{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# PyMOL image generation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys, os, time\n",
    "from Bio import SeqIO\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## ColabFold - run on GADI supercomputer"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<b>Settings:</b>  \n",
    "* amber relax\n",
    "* num relax = 1\n",
    "* use GPU relax"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Run pdb2pqr"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### First align structures to AtSAL1_updated"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<string>:1: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " PyMOL(TM) Molecular Graphics System, Version 2.3.0.\n",
      " Copyright (c) Schrodinger, LLC.\n",
      " All Rights Reserved.\n",
      " \n",
      "    Created by Warren L. DeLano, Ph.D. \n",
      " \n",
      "    PyMOL is user-supported open-source software.  Although some versions\n",
      "    are freely available, PyMOL is not in the public domain.\n",
      " \n",
      "    If PyMOL is helpful in your work or study, then please volunteer \n",
      "    support for our ongoing efforts to create open and affordable scientific\n",
      "    software by purchasing a PyMOL Maintenance and/or Support subscription.\n",
      "\n",
      "    More information can be found at \"http://www.pymol.org\".\n",
      " \n",
      "    Enter \"help\" for a list of commands.\n",
      "    Enter \"help <command-name>\" for information on a specific command.\n",
      "\n",
      " Hit ESC anytime to toggle between text and graphics.\n",
      "\n",
      " Detected OpenGL version 3.1. Shaders available.\n",
      " Geometry shaders not available\n",
      " Detected GLSL version 1.40.\n",
      " OpenGL graphics engine:\n",
      "  GL_VENDOR:   Mesa/X.org\n",
      "  GL_RENDERER: llvmpipe (LLVM 12.0.0, 256 bits)\n",
      "  GL_VERSION:  3.1 Mesa 21.2.6\n",
      " Detected 48 CPU cores.  Enabled multithreaded rendering.\n",
      "\n",
      "PyMOL>load /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/74987.pdb\n",
      " CmdLoad: PDB-string loaded into object \"74987\", state 1.\n",
      "PyMOL>load /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/AtSAL1_updated/AtSAL1_updated.pdb\n",
      " CmdLoad: PDB-string loaded into object \"AtSAL1_updated\", state 1.\n",
      "PyMOL>set_name AtSAL1_updated, AtSAL1_updated_pdb\n",
      "PyMOL>set_view (        0.980755210,   -0.023148270,   -0.193854913,        0.162131056,   -0.456586570,    0.874780297,        -0.108762056,   -0.889374375,   -0.444051415,        0.000122931,   -0.000242302,  -56.248649597,        -32.444423676,  -39.579414368,   -9.259656906,        44.643768311,   67.851081848,  -20.000000000 )\n",
      " Scene: view updated.\n",
      "PyMOL>align 74987, AtSAL1_updated_pdb\n",
      " Match: read scoring matrix.\n",
      " Match: assigning 352 x 352 pairwise scores.\n",
      " MatchAlign: aligning residues (352 vs 352)...\n",
      " MatchAlign: score 415.500\n",
      " ExecutiveAlign: 1982 atoms aligned.\n",
      " ExecutiveRMS: 159 atoms rejected during cycle 1 (RMSD=6.24).\n",
      " ExecutiveRMS: 149 atoms rejected during cycle 2 (RMSD=4.01).\n",
      " ExecutiveRMS: 156 atoms rejected during cycle 3 (RMSD=2.86).\n",
      " ExecutiveRMS: 140 atoms rejected during cycle 4 (RMSD=2.07).\n",
      " ExecutiveRMS: 84 atoms rejected during cycle 5 (RMSD=1.45).\n",
      " Executive: RMSD =    1.199 (1294 to 1294 atoms)\n",
      "PyMOL>delete AtSAL1_updated_pdb\n",
      "PyMOL>save /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/74987.pdb, 74987\n",
      " Save: wrote \"/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/74987.pdb\".\n",
      "PyMOL>reinitialize\n",
      "PyMOL>load /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/At_v_Sc_all-v-all_EMD_dx.csv\n",
      " Error: unsupported file type: csv\n",
      "PyMOL>load /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/AtSAL1_updated/AtSAL1_updated.pdb\n",
      " CmdLoad: PDB-string loaded into object \"AtSAL1_updated\", state 1.\n",
      "PyMOL>set_name AtSAL1_updated, AtSAL1_updated_pdb\n",
      "PyMOL>set_view (        0.980755210,   -0.023148270,   -0.193854913,        0.162131056,   -0.456586570,    0.874780297,        -0.108762056,   -0.889374375,   -0.444051415,        0.000122931,   -0.000242302,  -56.248649597,        -32.444423676,  -39.579414368,   -9.259656906,        44.643768311,   67.851081848,  -20.000000000 )\n",
      " Scene: view updated.\n",
      "PyMOL>align At_v_Sc_all-v-all_EMD_dx, AtSAL1_updated_pdb\n",
      "Selector-Error: Invalid selection name \"At_v_Sc_all-v-all_EMD_dx\".\n",
      "At_v_Sc_all-v-all_EMD_dx<--\n",
      "PyMOL>delete AtSAL1_updated_pdb\n",
      "PyMOL>save /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/At_v_Sc_all-v-all_EMD_dx.csv, At_v_Sc_all-v-all_EMD_dx\n",
      " Error: File format not supported for export\n",
      "PyMOL>reinitialize\n",
      "PyMOL>load /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/1ka1_hal2.pdb\n",
      " ObjectMolecule: Read crystal symmetry information.\n",
      " CmdLoad: PDB-string loaded into object \"1ka1_hal2\", state 1.\n",
      "PyMOL>load /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/AtSAL1_updated/AtSAL1_updated.pdb\n",
      " CmdLoad: PDB-string loaded into object \"AtSAL1_updated\", state 1.\n",
      "PyMOL>set_name AtSAL1_updated, AtSAL1_updated_pdb\n",
      "PyMOL>set_view (        0.980755210,   -0.023148270,   -0.193854913,        0.162131056,   -0.456586570,    0.874780297,        -0.108762056,   -0.889374375,   -0.444051415,        0.000122931,   -0.000242302,  -56.248649597,        -32.444423676,  -39.579414368,   -9.259656906,        44.643768311,   67.851081848,  -20.000000000 )\n",
      " Scene: view updated.\n",
      "PyMOL>align 1ka1_hal2, AtSAL1_updated_pdb\n",
      " Match: read scoring matrix.\n",
      " Match: assigning 354 x 352 pairwise scores.\n",
      " MatchAlign: aligning residues (354 vs 352)...\n",
      " MatchAlign: score 515.500\n",
      " ExecutiveAlign: 2082 atoms aligned.\n",
      " ExecutiveRMS: 154 atoms rejected during cycle 1 (RMSD=4.40).\n",
      " ExecutiveRMS: 135 atoms rejected during cycle 2 (RMSD=2.61).\n",
      " ExecutiveRMS: 144 atoms rejected during cycle 3 (RMSD=1.97).\n",
      " ExecutiveRMS: 75 atoms rejected during cycle 4 (RMSD=1.53).\n",
      " ExecutiveRMS: 75 atoms rejected during cycle 5 (RMSD=1.35).\n",
      " Executive: RMSD =    1.222 (1499 to 1499 atoms)\n",
      "PyMOL>delete AtSAL1_updated_pdb\n",
      "PyMOL>save /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/1ka1_hal2.pdb, 1ka1_hal2\n",
      " Save: wrote \"/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/1ka1_hal2.pdb\".\n",
      "PyMOL>reinitialize\n",
      "PyMOL>load /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/AtSAL1_updated.pdb\n",
      " CmdLoad: PDB-string loaded into object \"AtSAL1_updated\", state 1.\n",
      "PyMOL>load /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/AtSAL1_updated/AtSAL1_updated.pdb\n",
      " CmdLoad: PDB-string appended into object \"AtSAL1_updated\", state 2.\n",
      "PyMOL>set_name AtSAL1_updated, AtSAL1_updated_pdb\n",
      "PyMOL>set_view (        0.980755210,   -0.023148270,   -0.193854913,        0.162131056,   -0.456586570,    0.874780297,        -0.108762056,   -0.889374375,   -0.444051415,        0.000122931,   -0.000242302,  -56.248649597,        -32.444423676,  -39.579414368,   -9.259656906,        44.643768311,   67.851081848,  -20.000000000 )\n",
      " Scene: view updated.\n",
      "PyMOL>align AtSAL1_updated, AtSAL1_updated_pdb\n",
      " ExecutiveAlign: invalid selections for alignment.\n",
      "PyMOL>delete AtSAL1_updated_pdb\n",
      "PyMOL>save /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/AtSAL1_updated.pdb, AtSAL1_updated\n",
      "Selector-Error: Invalid selection name \"AtSAL1_updated\".\n",
      "AtSAL1_updated<--\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3/dist-packages/pymol/exporting.py\", line 866, in save\n",
      "    contents = func(**kw)\n",
      "  File \"/usr/lib/python3/dist-packages/pymol/exporting.py\", line 654, in get_str\n",
      "    return b.decode('utf-8')\n",
      "AttributeError: 'NoneType' object has no attribute 'decode'\n",
      "PyMOL>reinitialize\n",
      "PyMOL>load /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/AtAHL.pdb\n",
      " CmdLoad: PDB-string loaded into object \"AtAHL\", state 1.\n",
      "PyMOL>load /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/AtSAL1_updated/AtSAL1_updated.pdb\n",
      " CmdLoad: PDB-string loaded into object \"AtSAL1_updated\", state 1.\n",
      "PyMOL>set_name AtSAL1_updated, AtSAL1_updated_pdb\n",
      "PyMOL>set_view (        0.980755210,   -0.023148270,   -0.193854913,        0.162131056,   -0.456586570,    0.874780297,        -0.108762056,   -0.889374375,   -0.444051415,        0.000122931,   -0.000242302,  -56.248649597,        -32.444423676,  -39.579414368,   -9.259656906,        44.643768311,   67.851081848,  -20.000000000 )\n",
      " Scene: view updated.\n",
      "PyMOL>align AtAHL, AtSAL1_updated_pdb\n",
      " Match: read scoring matrix.\n",
      " Match: assigning 368 x 352 pairwise scores.\n",
      " MatchAlign: aligning residues (368 vs 352)...\n",
      " MatchAlign: score 718.500\n",
      " ExecutiveAlign: 2124 atoms aligned.\n",
      " ExecutiveRMS: 147 atoms rejected during cycle 1 (RMSD=3.98).\n",
      " ExecutiveRMS: 137 atoms rejected during cycle 2 (RMSD=2.22).\n",
      " ExecutiveRMS: 105 atoms rejected during cycle 3 (RMSD=1.49).\n",
      " ExecutiveRMS: 77 atoms rejected during cycle 4 (RMSD=1.24).\n",
      " ExecutiveRMS: 48 atoms rejected during cycle 5 (RMSD=1.13).\n",
      " Executive: RMSD =    1.067 (1610 to 1610 atoms)\n",
      "PyMOL>delete AtSAL1_updated_pdb\n",
      "PyMOL>save /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/AtAHL.pdb, AtAHL\n",
      " Save: wrote \"/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/AtAHL.pdb\".\n",
      "PyMOL>reinitialize\n"
     ]
    }
   ],
   "source": [
    "import __main__\n",
    "__main__.pymol_argv = ['pymol', '-qc']\n",
    "\n",
    "import os\n",
    "import pymolPy3\n",
    "\n",
    "pm = pymolPy3.pymolPy3()\n",
    "\n",
    "# Import files inc AtSAL1\n",
    "# assign directory\n",
    "\n",
    "directory = f'/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/'\n",
    "output_dir = f'/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/'\n",
    "\n",
    "AtSAL1_pdb = '/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/AtSAL1_updated/AtSAL1_updated.pdb'\n",
    "\n",
    "\n",
    "\"\"\"\n",
    "Structures need to be aligned BEFORE electrostatics\n",
    "\"\"\"\n",
    "\n",
    "\n",
    "# iterate over files in directory\n",
    "for filename in os.listdir(directory):\n",
    "    basename = os.path.basename(filename)\n",
    "    f = os.path.join(directory, filename)\n",
    "\n",
    "    # checking if it is a file, create short name\n",
    "    if os.path.isfile(f):\n",
    "        shortname = basename.split('.')[0]\n",
    "\n",
    "        # load structure\n",
    "        pm(f\"load {f}\")\n",
    "\n",
    "        # Load AtSAL1 reference\n",
    "        pm(f\"load {AtSAL1_pdb}\")\n",
    "        pm('set_name AtSAL1_updated, AtSAL1_updated_pdb')\n",
    "\n",
    "        pm(\"set_view (\\\n",
    "        0.980755210,   -0.023148270,   -0.193854913,\\\n",
    "        0.162131056,   -0.456586570,    0.874780297,\\\n",
    "        -0.108762056,   -0.889374375,   -0.444051415,\\\n",
    "        0.000122931,   -0.000242302,  -56.248649597,\\\n",
    "        -32.444423676,  -39.579414368,   -9.259656906,\\\n",
    "        44.643768311,   67.851081848,  -20.000000000 )\")\n",
    "        \n",
    "        # set structure name\n",
    "        pdb = f\"{shortname}\"\n",
    "\n",
    "        # Align \n",
    "        pm(f\"align {pdb}, AtSAL1_updated_pdb\")    \n",
    "\n",
    "\n",
    "\n",
    "        # Remove AtSAL1 reference from the session\n",
    "        pm(\"delete AtSAL1_updated_pdb\")\n",
    "\n",
    "        # Save file\n",
    "        pm(f\"save {output_dir}{basename}, {pdb}\")\n",
    "\n",
    "        # Clear\n",
    "        pm(\"reinitialize\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<string>:1: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses\n"
     ]
    }
   ],
   "source": [
    "pm(f\"save {output_dir}{basename}, {pdb}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### pdb2pqr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:PDB2PQR v3.6.2: biomolecular structure conversion software.\n",
      "INFO:Please cite:  Jurrus E, et al.  Improvements to the APBS biomolecular solvation software suite.  Protein Sci 27 112-128 (2018).\n",
      "INFO:Please cite:  Dolinsky TJ, et al.  PDB2PQR: expanding and upgrading automated preparation of biomolecular structures for molecular simulations. Nucleic Acids Res 35 W522-W525 (2007).\n",
      "INFO:Checking and transforming input arguments.\n",
      "INFO:Loading topology files.\n",
      "INFO:Loading molecule: /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/74987.pdb\n",
      "INFO:Setting up molecule.\n",
      "INFO:Created biomolecule object with 352 residues and 5354 atoms.\n",
      "INFO:Setting termini states for biomolecule chains.\n",
      "INFO:Loading forcefield.\n",
      "INFO:Loading hydrogen topology definitions.\n",
      "INFO:This biomolecule is clean.  No repair needed.\n",
      "INFO:Updating disulfide bridges.\n",
      "INFO:Debumping biomolecule.\n",
      "INFO:Assigning titration states with PROPKA.\n",
      "INFO:\n",
      "propka3.5.1                                                                                  2024-07-26\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "--                                                                           --\n",
      "--  PROPKA: A PROTEIN PKA PREDICTOR                                          --\n",
      "--                                                                           --\n",
      "--  VERSION 1.0,  04/25/2004,  IOWA CITY                                     --\n",
      "--  BY HUI LI                                                                --\n",
      "--                                                                           --\n",
      "--  VERSION 2.0,  11/05/2007, IOWA CITY/COPENHAGEN                           --\n",
      "--  BY DELPHINE C. BAS AND DAVID M. ROGERS                                   --\n",
      "--                                                                           --\n",
      "--  VERSION 3.0,  01/06/2011, COPENHAGEN                                     --\n",
      "--  BY MATS H.M. OLSSON AND CHRESTEN R. SONDERGARD                           --\n",
      "--                                                                           --\n",
      "--  VERSION 3.1,  07/01/2011, COPENHAGEN                                     --\n",
      "--  BY CHRESTEN R. SONDERGARD AND MATS H.M. OLSSON                           --\n",
      "--                                                                           --\n",
      "--  VERSION 3.2,  06/17/2020, PLANET EARTH                                   --\n",
      "--  SEE ABOVE FOR AUTHORS                                                    --\n",
      "--                                                                           --\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "References:\n",
      "\n",
      "Very Fast Empirical Prediction and Rationalization of Protein pKa Values.\n",
      "Hui Li, Andrew D. Robertson and Jan H. Jensen. PROTEINS: Structure, Function,\n",
      "and Bioinformatics. 61:704-721 (2005)\n",
      "\n",
      "Very Fast Prediction and Rationalization of pKa Values for Protein-Ligand\n",
      "Complexes.  Delphine C. Bas, David M. Rogers and Jan H. Jensen.  PROTEINS:\n",
      "Structure, Function, and Bioinformatics 73:765-783 (2008)\n",
      "\n",
      "PROPKA3: Consistent Treatment of Internal and Surface Residues in Empirical\n",
      "pKa predictions.  Mats H.M. Olsson, Chresten R. Sondergard, Michal Rostkowski,\n",
      "and Jan H. Jensen.  Journal of Chemical Theory and Computation, 7(2):525-537\n",
      "(2011)\n",
      "\n",
      "Improved Treatment of Ligands and Coupling Effects in Empirical Calculation\n",
      "and Rationalization of pKa Values.  Chresten R. Sondergaard, Mats H.M. Olsson,\n",
      "Michal Rostkowski, and Jan H. Jensen.  Journal of Chemical Theory and\n",
      "Computation, (2011)\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "\n",
      "INFO:Coupling map:\n",
      "\n",
      "INFO:Calculating pKas for Conformation container 1A with 3171 atoms and 875 groups\n",
      "INFO:pKa determinants:\n",
      "\n",
      "---------  -----   ------   ---------------------    --------------    --------------    --------------\n",
      "                            DESOLVATION  EFFECTS       SIDECHAIN          BACKBONE        COULOMBIC\n",
      " RESIDUE    pKa    BURIED     REGULAR      RE        HYDROGEN BOND     HYDROGEN BOND      INTERACTION\n",
      "---------  -----   ------   ---------   ---------    --------------    --------------    --------------\n",
      "\n",
      "ASP  10 A   4.95    44 %    1.49  405   0.14    0   -0.28 ARG  14 A    0.00 XXX   0 X   -0.05 ARG  17 A\n",
      "ASP  10 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.10 LYS  59 A\n",
      "ASP  10 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU   6 A\n",
      "ASP  10 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP  13 A\n",
      "ASP  10 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.29 GLU  56 A\n",
      "ASP  10 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.55 ARG  14 A\n",
      "\n",
      "ASP  13 A   4.41    60 %    1.88  448   0.32    0   -0.72 ARG  17 A   -0.16 GLN 194 A   -0.08 ARG  14 A\n",
      "ASP  13 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.00 LYS  91 A\n",
      "ASP  13 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.05 LYS 191 A\n",
      "ASP  13 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.59 ARG  17 A\n",
      "\n",
      "ASP  21 A   2.92    25 %    0.74  352   0.19    0   -0.11 ARG  24 A    0.00 XXX   0 X   -0.03 ARG  17 A\n",
      "ASP  21 A                                           -0.85 LYS  91 A    0.00 XXX   0 X   -0.32 ARG  24 A\n",
      "ASP  21 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.49 LYS  91 A\n",
      "\n",
      "ASP  38 A   2.45    12 %    0.70  316   0.00    0   -0.85 THR  40 A   -0.20 ASP  38 A   -0.26 LYS  36 A\n",
      "ASP  38 A                                            0.00 XXX   0 X   -0.66 THR  40 A   -0.08 LYS 286 A\n",
      "\n",
      "ASP  46 A   5.88   100 %    3.67  601   1.00    0   -0.85 THR  43 A   -0.70 THR  43 A   -0.59 LYS  36 A\n",
      "ASP  46 A                                           -0.64 THR 134 A   -0.38 ASP  46 A    0.56 GLU  69 A\n",
      "\n",
      "ASP  71 A   3.56    35 %    0.91  380   0.00    0    0.00 XXX   0 X   -0.46 HIS  73 A   -0.12 LYS  36 A\n",
      "ASP  71 A                                            0.00 XXX   0 X   -0.28 PHE  74 A   -0.05 ARG  76 A\n",
      "ASP  71 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.21 LYS 113 A\n",
      "ASP  71 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 322 A\n",
      "ASP  71 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.19 HIS  73 A\n",
      "ASP  71 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.18 ASP 109 A\n",
      "\n",
      "ASP  96 A   3.95     0 %    0.24  216   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.09 LYS  89 A\n",
      "ASP  96 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.00 LYS  91 A\n",
      "\n",
      "ASP 101 A   3.94     0 %    0.32  265   0.00    0    0.00 XXX   0 X   -0.03 ASP 101 A   -0.16 LYS  89 A\n",
      "ASP 101 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 103 A\n",
      "\n",
      "ASP 103 A   2.46     0 %    0.37  271   0.00    0   -0.85 SER 100 A   -0.81 SER 100 A   -0.04 LYS  89 A\n",
      "ASP 103 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 HIS  98 A\n",
      "\n",
      "ASP 109 A   2.64    27 %    1.12  356   0.23    0   -0.85 SER  72 A   -0.79 SER  72 A   -0.02 HIS  73 A\n",
      "ASP 109 A                                           -0.48 ARG  76 A    0.00 XXX   0 X   -0.37 ARG  76 A\n",
      "\n",
      "ASP 114 A   3.91     0 %    0.33  245   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.29 LYS 113 A\n",
      "ASP 114 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 ARG 322 A\n",
      "ASP 114 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.11 ASP  71 A\n",
      "ASP 114 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 109 A\n",
      "\n",
      "ASP 129 A  11.71*  100 %    3.55  628   0.27    0    0.42 ASP 289 A   -0.23 ILE 131 A   -0.17 LYS  36 A\n",
      "ASP 129 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.32 ASP  46 A\n",
      "ASP 129 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 GLU  70 A\n",
      "ASP 129 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.98 GLU  69 A\n",
      "ASP 129 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.31 ASP 132 A\n",
      "ASP 129 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.28 ASP 289 A\n",
      "\n",
      "ASP 132 A   6.29*  100 %    3.80  623   0.65    0   -0.13 SER 260 A   -0.48 SER 260 A    0.11 GLU  69 A\n",
      "ASP 132 A                                            0.00 XXX   0 X   -0.56 LEU 261 A   -0.70 LYS 263 A\n",
      "ASP 132 A                                            0.00 XXX   0 X   -0.19 CYS 262 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 141 A   3.75     0 %    0.27  267   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.31 LYS  23 A\n",
      "ASP 141 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 ARG  24 A\n",
      "\n",
      "ASP 154 A   4.07     0 %    0.28  239   0.00    0    0.00 XXX   0 X   -0.00 ASP 154 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 170 A   3.21     0 %    0.24  274   0.00    0   -0.42 ARG 271 A    0.00 XXX   0 X   -0.03 LYS 192 A\n",
      "ASP 170 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.37 ARG 271 A\n",
      "\n",
      "ASP 173 A   1.41     0 %    0.49  200   0.00    0   -0.85 SER 175 A   -0.08 ASP 173 A   -0.38 LYS 192 A\n",
      "ASP 173 A                                           -0.81 LYS 192 A   -0.76 SER 175 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 203 A   3.82     0 %    0.12  132   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.10 LYS 191 A\n",
      "\n",
      "ASP 210 A   2.58    45 %    1.85  406   0.24    0   -0.85 SER 273 A   -0.41 CYS 212 A   -0.43 ARG 219 A\n",
      "ASP 210 A                                           -0.78 ASN 334 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "ASP 210 A                                           -0.85 ARG 219 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "ASP 238 A   4.04     0 %    0.23  233   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.09 LYS 247 A\n",
      "ASP 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 249 A\n",
      "ASP 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 229 A\n",
      "ASP 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 GLU 250 A\n",
      "\n",
      "ASP 245 A   4.00     0 %    0.16  109   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.04 GLU 250 A\n",
      "\n",
      "ASP 289 A   7.94*  100 %    2.62  580   0.38    0   -0.42 ASP 129 A    0.00 XXX   0 X   -0.01 LYS  36 A\n",
      "ASP 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.00 LYS 263 A\n",
      "ASP 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.19 LYS 286 A\n",
      "ASP 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.76 GLU  69 A\n",
      "ASP 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.00 ASP 132 A\n",
      "\n",
      "ASP 306 A   3.95    39 %    1.62  391   0.52    0    0.00 XXX   0 X   -0.01 ASP 306 A   -0.01 ARG 283 A\n",
      "ASP 306 A                                            0.00 XXX   0 X   -0.48 GLU 308 A   -0.01 HIS 338 A\n",
      "ASP 306 A                                            0.00 XXX   0 X   -0.34 GLY 309 A    0.00 XXX   0 X\n",
      "ASP 306 A                                            0.00 XXX   0 X   -0.61 ASP 310 A    0.00 XXX   0 X\n",
      "ASP 306 A                                            0.00 XXX   0 X   -0.53 ALA 327 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 310 A   4.33     0 %    0.22  242   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.00 LYS 303 A\n",
      "ASP 310 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.28 ASP 306 A\n",
      "ASP 310 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 311 A\n",
      "\n",
      "ASP 317 A   3.26    37 %    1.59  385   0.12    0    0.00 XXX   0 X   -0.51 GLU 320 A   -0.05 LYS 286 A\n",
      "ASP 317 A                                            0.00 XXX   0 X   -0.03 ARG 321 A   -0.13 ARG 321 A\n",
      "ASP 317 A                                            0.00 XXX   0 X   -0.72 ARG 322 A   -0.10 ARG 322 A\n",
      "ASP 317 A                                            0.00 XXX   0 X   -0.69 LEU 323 A    0.00 XXX   0 X\n",
      "\n",
      "GLU   4 A   4.10    15 %    0.77  322   0.00    0    0.00 XXX   0 X   -0.76 GLU   4 A   -0.26 N+    1 A\n",
      "GLU   4 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.07 LYS   3 A\n",
      "GLU   4 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 HIS   2 A\n",
      "GLU   4 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 HIS 183 A\n",
      "\n",
      "GLU   6 A   4.46     0 %    0.19  222   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.05 LYS   3 A\n",
      "GLU   6 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.18 HIS   2 A\n",
      "\n",
      "GLU  31 A   4.62     6 %    0.36  298   0.00    0   -0.24 SER  29 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "GLU  35 A   4.73     0 %    0.22  190   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP  38 A\n",
      "\n",
      "GLU  56 A   2.83    47 %    1.86  412   0.21    0   -0.78 SER  93 A   -0.83 SER  93 A   -0.52 ARG  14 A\n",
      "GLU  56 A                                           -0.18 ARG  14 A    0.00 XXX   0 X   -0.58 LYS  59 A\n",
      "GLU  56 A                                           -0.85 LYS  59 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "GLU  69 A   4.06*   92 %    1.97  539   0.00    0   -0.35 TRP 288 A    0.00 XXX   0 X   -0.18 LYS 286 A\n",
      "GLU  69 A                                           -0.85 LYS  36 A    0.00 XXX   0 X   -0.02 ARG 322 A\n",
      "GLU  69 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 ASP  38 A\n",
      "GLU  69 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.20 LYS  36 A\n",
      "\n",
      "GLU  70 A   8.36*   97 %    3.42  553   0.03    0   -0.85 LYS  36 A    0.00 XXX   0 X    0.08 ASP  38 A\n",
      "GLU  70 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 ASP  71 A\n",
      "GLU  70 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.29 LYS  36 A\n",
      "GLU  70 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.00 ASP  46 A\n",
      "GLU  70 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.27 GLU  69 A\n",
      "\n",
      "GLU  87 A   5.73    40 %    1.00  393   0.16    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.00 ARG  24 A\n",
      "GLU  87 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.14 LYS  91 A\n",
      "GLU  87 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.17 ASP  21 A\n",
      "GLU  87 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU  31 A\n",
      "\n",
      "GLU 106 A   4.79     0 %    0.16  269   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 110 A\n",
      "GLU 106 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.11 ASP 103 A\n",
      "GLU 106 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 109 A\n",
      "\n",
      "GLU 142 A   4.62     0 %    0.09  177   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.01 LYS  23 A\n",
      "GLU 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 141 A\n",
      "\n",
      "GLU 156 A   5.01    18 %    0.37  332   0.02    0    0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 154 A\n",
      "GLU 156 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 HIS 298 A\n",
      "GLU 156 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.14 GLU 311 A\n",
      "GLU 156 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 315 A\n",
      "\n",
      "GLU 224 A   2.07    12 %    0.73  314   0.02    0   -0.52 ARG 135 A    0.00 XXX   0 X   -0.16 LYS 263 A\n",
      "GLU 224 A                                           -0.85 LYS 139 A    0.00 XXX   0 X   -0.42 ARG 135 A\n",
      "GLU 224 A                                           -0.84 CYS 258 A    0.00 XXX   0 X   -0.39 LYS 139 A\n",
      "\n",
      "GLU 229 A   3.42     0 %    0.34  244   0.00    0   -0.66 THR 227 A   -0.76 GLU 229 A    0.00 XXX   0 X\n",
      "\n",
      "GLU 244 A   4.57     0 %    0.07   82   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "GLU 250 A   2.96     0 %    0.53  224   0.00    0   -0.85 LYS 247 A   -0.84 LYS 247 A   -0.00 LYS 217 A\n",
      "GLU 250 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.38 LYS 247 A\n",
      "\n",
      "GLU 299 A   4.11    28 %    1.15  361   0.24    0    0.00 XXX   0 X   -0.79 GLY 185 A   -0.13 ARG 206 A\n",
      "GLU 299 A                                            0.00 XXX   0 X   -0.08 CYS 186 A   -0.10 HIS 183 A\n",
      "GLU 299 A                                            0.00 XXX   0 X   -0.58 GLY 187 A   -0.10 HIS 298 A\n",
      "\n",
      "GLU 308 A   5.24     0 %    0.26  271   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.10 ARG 283 A\n",
      "GLU 308 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.24 ASP 306 A\n",
      "GLU 308 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP 310 A\n",
      "GLU 308 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 C-  352 A\n",
      "\n",
      "GLU 311 A   3.29    11 %    0.28  311   0.00    0   -0.85 LYS 303 A    0.00 XXX   0 X   -0.21 HIS 298 A\n",
      "GLU 311 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.42 LYS 303 A\n",
      "\n",
      "GLU 315 A   4.29     0 %    0.41  259   0.00    0   -0.39 ASN 155 A    0.00 XXX   0 X   -0.24 ARG 321 A\n",
      "GLU 315 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 154 A\n",
      "\n",
      "GLU 316 A   5.33    15 %    0.68  322   0.08    0    0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 306 A\n",
      "GLU 316 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 317 A\n",
      "\n",
      "GLU 320 A   4.89     8 %    0.35  305   0.01    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.05 LYS 113 A\n",
      "GLU 320 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 LYS 286 A\n",
      "GLU 320 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.19 ARG 322 A\n",
      "GLU 320 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 114 A\n",
      "GLU 320 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.30 ASP 317 A\n",
      "\n",
      "GLU 343 A   4.70     0 %    0.20  205   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "C-  352 A   2.61     0 %    0.24  209   0.00    0   -0.57 ARG 283 A    0.00 XXX   0 X   -0.26 ARG 283 A\n",
      "\n",
      "HIS   2 A   6.46     0 %   -0.18  208   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.08 N+    1 A\n",
      "HIS   2 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 GLU   4 A\n",
      "HIS   2 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.18 GLU   6 A\n",
      "\n",
      "HIS  73 A   6.41     0 %   -0.19  218   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.06 ARG  76 A\n",
      "HIS  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 113 A\n",
      "HIS  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 ASP  71 A\n",
      "HIS  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 109 A\n",
      "\n",
      "HIS  98 A   6.37     0 %   -0.15  122   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.00 LYS  89 A\n",
      "HIS  98 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 103 A\n",
      "\n",
      "HIS 183 A   6.16    21 %   -0.82  341   0.00    0    0.00 XXX   0 X    0.05 HIS 183 A   -0.39 N+    1 A\n",
      "HIS 183 A                                            0.00 XXX   0 X    0.71 ILE 184 A   -0.03 HIS   2 A\n",
      "HIS 183 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 GLU   4 A\n",
      "HIS 183 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 GLU 299 A\n",
      "\n",
      "HIS 248 A   6.22     0 %   -0.17  173   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 217 A\n",
      "HIS 248 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 249 A\n",
      "\n",
      "HIS 290 A   3.27   100 %   -3.17  619   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.07 LYS 263 A\n",
      "\n",
      "HIS 298 A   5.12    76 %   -1.70  493   0.00    0    0.00 XXX   0 X    0.30 GLY 302 A   -0.32 LYS 303 A\n",
      "HIS 298 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 156 A\n",
      "HIS 298 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 GLU 299 A\n",
      "HIS 298 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.21 GLU 311 A\n",
      "\n",
      "HIS 338 A   4.12    64 %   -2.13  461   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.17 LYS 303 A\n",
      "HIS 338 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.10 HIS 298 A\n",
      "HIS 338 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 306 A\n",
      "\n",
      "CYS  16 A  14.27   100 %    3.67  626   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.05 ARG  17 A\n",
      "CYS  16 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.44 ASP  13 A\n",
      "CYS  16 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.84 CYS  19 A\n",
      "CYS  16 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.36 CYS 164 A\n",
      "\n",
      "CYS  19 A  13.25   100 %    3.84  640   0.00    0    0.00 XXX   0 X   -0.06 CYS  19 A   -0.03 LYS  23 A\n",
      "CYS  19 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.49 ASP  46 A\n",
      "CYS  19 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU  87 A\n",
      "\n",
      "CYS 119 A   9.04     0 %    0.20  187   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.14 LYS 121 A\n",
      "CYS 119 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 321 A\n",
      "\n",
      "CYS 164 A  12.58   100 %    3.66  580   0.00    0    0.00 XXX   0 X   -0.13 TYR 145 A   -0.02 LYS  23 A\n",
      "CYS 164 A                                            0.00 XXX   0 X   -0.10 ASN 166 A   -0.04 ARG 271 A\n",
      "CYS 164 A                                            0.00 XXX   0 X   -0.05 TRP 167 A    0.11 ASP 132 A\n",
      "CYS 164 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.17 CYS 257 A\n",
      "\n",
      "CYS 186 A   9.38     0 %    0.39  262   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.09 N+    1 A\n",
      "CYS 186 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 ARG 206 A\n",
      "CYS 186 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 GLU 299 A\n",
      "\n",
      "CYS 207 A  11.12    63 %    2.22  457   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.07 ARG 206 A\n",
      "CYS 207 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 271 A\n",
      "\n",
      "CYS 212 A   9.75    29 %    1.27  363   0.00    0   -0.00 SER 273 A   -0.71 LEU 214 A   -0.04 LYS 217 A\n",
      "CYS 212 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.12 ARG 219 A\n",
      "CYS 212 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.35 ASP 210 A\n",
      "\n",
      "CYS 221 A  11.78    93 %    2.98  541   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.06 ARG 219 A\n",
      "CYS 221 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.40 LYS 263 A\n",
      "CYS 221 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.26 CYS 257 A\n",
      "\n",
      "CYS 257 A   6.54*   61 %    1.43  451   0.00    0   -3.60 CYS 262 A    0.00 XXX   0 X   -0.02 LYS 139 A\n",
      "CYS 257 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.33 LYS 263 A\n",
      "CYS 257 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 GLU 224 A\n",
      "\n",
      "CYS 258 A  10.03    39 %    1.48  391   0.00    0   -0.71 ARG 135 A   -0.05 CYS 258 A   -0.39 ARG 135 A\n",
      "CYS 258 A                                           -0.23 LYS 139 A    0.00 XXX   0 X   -0.45 LYS 139 A\n",
      "CYS 258 A                                            0.84 GLU 224 A    0.00 XXX   0 X   -0.20 LYS 263 A\n",
      "CYS 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP 132 A\n",
      "CYS 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.15 CYS 257 A\n",
      "CYS 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.48 GLU 224 A\n",
      "\n",
      "CYS 262 A  17.21*  100 %    3.46  569   0.00    0    3.60 CYS 257 A   -0.07 CYS 262 A   -1.78 LYS 263 A\n",
      "CYS 262 A                                            0.00 XXX   0 X   -0.06 LYS 263 A    0.53 CYS 164 A\n",
      "CYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.44 CYS 221 A\n",
      "CYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 224 A\n",
      "CYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 CYS 258 A\n",
      "CYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.03 ASP 132 A\n",
      "CYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.81 CYS 257 A\n",
      "\n",
      "CYS 296 A  12.13   100 %    3.70  656   0.00    0   -0.85 SER 182 A    0.00 XXX   0 X    0.23 CYS 207 A\n",
      "CYS 296 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 299 A\n",
      "\n",
      "TYR   1 A  10.58     0 %    0.41  279   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU   4 A\n",
      "TYR   1 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP 154 A\n",
      "TYR   1 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 GLU 156 A\n",
      "\n",
      "TYR 116 A  10.46     2 %    0.31  288   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.16 CYS 119 A\n",
      "TYR 116 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 ARG 321 A\n",
      "\n",
      "TYR 125 A  12.77    85 %    2.97  518   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 315 A\n",
      "TYR 125 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP 317 A\n",
      "TYR 125 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.34 ARG 321 A\n",
      "\n",
      "TYR 145 A  16.29    97 %    3.60  552   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.78 CYS  16 A\n",
      "TYR 145 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.50 CYS  19 A\n",
      "TYR 145 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP  21 A\n",
      "TYR 145 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 141 A\n",
      "TYR 145 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.37 CYS 164 A\n",
      "\n",
      "TYR 264 A  13.96   100 %    3.82  649   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 132 A\n",
      "TYR 264 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP 289 A\n",
      "\n",
      "LYS   3 A  10.39     0 %   -0.23  184   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.07 GLU   4 A\n",
      "LYS   3 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU   6 A\n",
      "\n",
      "LYS  23 A   9.00    44 %   -1.88  405   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.03 CYS  19 A\n",
      "LYS  23 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.31 ASP 141 A\n",
      "LYS  23 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 142 A\n",
      "LYS  23 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 CYS 164 A\n",
      "\n",
      "LYS  30 A  10.47     0 %   -0.03   61   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS  32 A  10.45     0 %   -0.05  104   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS  36 A  13.02    75 %   -2.82  492   0.00    0    0.85 GLU  69 A    0.00 XXX   0 X    0.26 ASP  38 A\n",
      "LYS  36 A                                            0.85 GLU  70 A    0.00 XXX   0 X    0.12 ASP  71 A\n",
      "LYS  36 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.17 ASP 129 A\n",
      "LYS  36 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 289 A\n",
      "LYS  36 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.59 ASP  46 A\n",
      "LYS  36 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.20 GLU  69 A\n",
      "LYS  36 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.29 GLU  70 A\n",
      "\n",
      "LYS  59 A  10.02    37 %   -1.79  385   0.00    0    0.85 GLU  56 A    0.00 XXX   0 X    0.10 ASP  10 A\n",
      "LYS  59 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.22 ARG  14 A\n",
      "LYS  59 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.58 GLU  56 A\n",
      "\n",
      "LYS  89 A  10.40     0 %   -0.38  237   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP  96 A\n",
      "LYS  89 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 ASP 101 A\n",
      "LYS  89 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 103 A\n",
      "\n",
      "LYS  91 A  10.94    30 %   -0.88  365   0.00    0    0.85 ASP  21 A    0.00 XXX   0 X    0.00 ASP  13 A\n",
      "LYS  91 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.11 ARG  17 A\n",
      "LYS  91 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 ARG  24 A\n",
      "LYS  91 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.14 GLU  87 A\n",
      "LYS  91 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP  96 A\n",
      "LYS  91 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.49 ASP  21 A\n",
      "\n",
      "LYS 113 A  10.86     0 %   -0.16  251   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.21 ASP  71 A\n",
      "LYS 113 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.29 ASP 114 A\n",
      "LYS 113 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 320 A\n",
      "LYS 113 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 ARG 322 A\n",
      "\n",
      "LYS 121 A  10.58     0 %   -0.06  106   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.14 CYS 119 A\n",
      "\n",
      "LYS 139 A  11.66     0 %   -0.43  269   0.00    0    0.23 CYS 258 A    0.00 XXX   0 X   -0.35 ARG 135 A\n",
      "LYS 139 A                                            0.85 GLU 224 A    0.00 XXX   0 X    0.02 CYS 257 A\n",
      "LYS 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.45 CYS 258 A\n",
      "LYS 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.39 GLU 224 A\n",
      "\n",
      "LYS 191 A  10.20     2 %   -0.45  288   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP  13 A\n",
      "LYS 191 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP 203 A\n",
      "\n",
      "LYS 192 A  11.20     0 %   -0.52  249   0.00    0    0.81 ASP 173 A    0.00 XXX   0 X    0.03 ASP 170 A\n",
      "LYS 192 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 ASP 173 A\n",
      "\n",
      "LYS 217 A  10.32     0 %   -0.21  224   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.04 CYS 212 A\n",
      "LYS 217 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 GLU 250 A\n",
      "\n",
      "LYS 247 A  11.60     0 %   -0.22  164   0.00    0    0.85 GLU 250 A    0.00 XXX   0 X    0.09 ASP 238 A\n",
      "LYS 247 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 GLU 250 A\n",
      "\n",
      "LYS 249 A  10.46     0 %   -0.10  152   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.06 ASP 238 A\n",
      "\n",
      "LYS 263 A  11.10    90 %   -2.93  534   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.04 ARG 135 A\n",
      "LYS 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.40 CYS 221 A\n",
      "LYS 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 GLU 224 A\n",
      "LYS 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.33 CYS 257 A\n",
      "LYS 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 CYS 258 A\n",
      "LYS 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.78 CYS 262 A\n",
      "LYS 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP 289 A\n",
      "LYS 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.70 ASP 132 A\n",
      "\n",
      "LYS 280 A  10.26    15 %   -0.23  322   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.01 ARG 135 A\n",
      "\n",
      "LYS 286 A  10.05    45 %   -0.88  407   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP  38 A\n",
      "LYS 286 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.18 GLU  69 A\n",
      "LYS 286 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 ASP 289 A\n",
      "LYS 286 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 317 A\n",
      "LYS 286 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU 320 A\n",
      "LYS 286 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.11 ARG 322 A\n",
      "LYS 286 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 LYS  36 A\n",
      "\n",
      "LYS 303 A  11.21    13 %   -0.56  318   0.00    0    0.85 GLU 311 A    0.00 XXX   0 X    0.00 ASP 310 A\n",
      "LYS 303 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.42 GLU 311 A\n",
      "\n",
      "ARG  14 A  12.98    32 %   -1.07  370   0.00    0    0.28 ASP  10 A    0.00 XXX   0 X    0.08 ASP  13 A\n",
      "ARG  14 A                                            0.18 GLU  56 A    0.00 XXX   0 X   -0.05 ARG  17 A\n",
      "ARG  14 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.55 ASP  10 A\n",
      "ARG  14 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.52 GLU  56 A\n",
      "\n",
      "ARG  17 A  13.13    26 %   -0.81  355   0.00    0    0.72 ASP  13 A    0.00 XXX   0 X    0.05 ASP  10 A\n",
      "ARG  17 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 CYS  16 A\n",
      "ARG  17 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP  21 A\n",
      "ARG  17 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.59 ASP  13 A\n",
      "\n",
      "ARG  24 A  12.69     0 %   -0.25  263   0.00    0    0.11 ASP  21 A    0.00 XXX   0 X    0.32 ASP  21 A\n",
      "ARG  24 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 GLU  87 A\n",
      "ARG  24 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 141 A\n",
      "\n",
      "ARG  76 A  12.96     1 %   -0.44  283   0.00    0    0.48 ASP 109 A    0.00 XXX   0 X    0.05 ASP  71 A\n",
      "ARG  76 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.37 ASP 109 A\n",
      "\n",
      "ARG 110 A  10.93    57 %   -1.59  442   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 106 A\n",
      "\n",
      "ARG 135 A  14.08    12 %   -0.46  314   0.00    0    0.71 CYS 258 A    0.00 XXX   0 X    0.39 CYS 258 A\n",
      "ARG 135 A                                            0.52 GLU 224 A    0.00 XXX   0 X    0.42 GLU 224 A\n",
      "\n",
      "ARG 206 A  12.17     3 %   -0.59  289   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.06 CYS 186 A\n",
      "ARG 206 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 CYS 207 A\n",
      "ARG 206 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 GLU 299 A\n",
      "\n",
      "ARG 219 A  12.97    38 %   -0.99  388   0.00    0    0.85 ASP 210 A    0.00 XXX   0 X    0.12 CYS 212 A\n",
      "ARG 219 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 CYS 221 A\n",
      "ARG 219 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.43 ASP 210 A\n",
      "\n",
      "ARG 271 A  12.14    41 %   -1.19  395   0.00    0    0.42 ASP 170 A    0.00 XXX   0 X    0.04 CYS 164 A\n",
      "ARG 271 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 CYS 207 A\n",
      "ARG 271 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 219 A\n",
      "ARG 271 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.37 ASP 170 A\n",
      "\n",
      "ARG 278 A  11.64    29 %   -0.86  362   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "ARG 283 A  13.24     0 %   -0.20  243   0.00    0    0.57 C-  352 A    0.00 XXX   0 X    0.01 ASP 306 A\n",
      "ARG 283 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 GLU 308 A\n",
      "ARG 283 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.26 C-  352 A\n",
      "\n",
      "ARG 321 A  11.96    26 %   -1.28  353   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.01 TYR 116 A\n",
      "ARG 321 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 CYS 119 A\n",
      "ARG 321 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.34 TYR 125 A\n",
      "ARG 321 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.24 GLU 315 A\n",
      "ARG 321 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 ASP 317 A\n",
      "\n",
      "ARG 322 A  10.73    72 %   -2.17  484   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU  69 A\n",
      "ARG 322 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP  71 A\n",
      "ARG 322 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 ASP 114 A\n",
      "ARG 322 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP 317 A\n",
      "ARG 322 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 GLU 320 A\n",
      "\n",
      "N+    1 A   7.87     2 %   -0.47  287   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.26 GLU   4 A\n",
      "N+    1 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 CYS 186 A\n",
      "\n",
      "Coupled residues (marked *) were detected.Please rerun PropKa with the --display-coupled-residues \n",
      "or -d option for detailed information.\n",
      "\n",
      "INFO:pKa summary:\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "SUMMARY OF THIS PREDICTION\n",
      "       Group      pKa  model-pKa   ligand atom-type\n",
      "   ASP  10 A     4.95       3.80                      \n",
      "   ASP  13 A     4.41       3.80                      \n",
      "   ASP  21 A     2.92       3.80                      \n",
      "   ASP  38 A     2.45       3.80                      \n",
      "   ASP  46 A     5.88       3.80                      \n",
      "   ASP  71 A     3.56       3.80                      \n",
      "   ASP  96 A     3.95       3.80                      \n",
      "   ASP 101 A     3.94       3.80                      \n",
      "   ASP 103 A     2.46       3.80                      \n",
      "   ASP 109 A     2.64       3.80                      \n",
      "   ASP 114 A     3.91       3.80                      \n",
      "   ASP 129 A    11.71       3.80                      \n",
      "   ASP 132 A     6.29       3.80                      \n",
      "   ASP 141 A     3.75       3.80                      \n",
      "   ASP 154 A     4.07       3.80                      \n",
      "   ASP 170 A     3.21       3.80                      \n",
      "   ASP 173 A     1.41       3.80                      \n",
      "   ASP 203 A     3.82       3.80                      \n",
      "   ASP 210 A     2.58       3.80                      \n",
      "   ASP 238 A     4.04       3.80                      \n",
      "   ASP 245 A     4.00       3.80                      \n",
      "   ASP 289 A     7.94       3.80                      \n",
      "   ASP 306 A     3.95       3.80                      \n",
      "   ASP 310 A     4.33       3.80                      \n",
      "   ASP 317 A     3.26       3.80                      \n",
      "   GLU   4 A     4.10       4.50                      \n",
      "   GLU   6 A     4.46       4.50                      \n",
      "   GLU  31 A     4.62       4.50                      \n",
      "   GLU  35 A     4.73       4.50                      \n",
      "   GLU  56 A     2.83       4.50                      \n",
      "   GLU  69 A     4.06       4.50                      \n",
      "   GLU  70 A     8.36       4.50                      \n",
      "   GLU  87 A     5.73       4.50                      \n",
      "   GLU 106 A     4.79       4.50                      \n",
      "   GLU 142 A     4.62       4.50                      \n",
      "   GLU 156 A     5.01       4.50                      \n",
      "   GLU 224 A     2.07       4.50                      \n",
      "   GLU 229 A     3.42       4.50                      \n",
      "   GLU 244 A     4.57       4.50                      \n",
      "   GLU 250 A     2.96       4.50                      \n",
      "   GLU 299 A     4.11       4.50                      \n",
      "   GLU 308 A     5.24       4.50                      \n",
      "   GLU 311 A     3.29       4.50                      \n",
      "   GLU 315 A     4.29       4.50                      \n",
      "   GLU 316 A     5.33       4.50                      \n",
      "   GLU 320 A     4.89       4.50                      \n",
      "   GLU 343 A     4.70       4.50                      \n",
      "   C-  352 A     2.61       3.20                      \n",
      "   HIS   2 A     6.46       6.50                      \n",
      "   HIS  73 A     6.41       6.50                      \n",
      "   HIS  98 A     6.37       6.50                      \n",
      "   HIS 183 A     6.16       6.50                      \n",
      "   HIS 248 A     6.22       6.50                      \n",
      "   HIS 290 A     3.27       6.50                      \n",
      "   HIS 298 A     5.12       6.50                      \n",
      "   HIS 338 A     4.12       6.50                      \n",
      "   CYS  16 A    14.27       9.00                      \n",
      "   CYS  19 A    13.25       9.00                      \n",
      "   CYS 119 A     9.04       9.00                      \n",
      "   CYS 164 A    12.58       9.00                      \n",
      "   CYS 186 A     9.38       9.00                      \n",
      "   CYS 207 A    11.12       9.00                      \n",
      "   CYS 212 A     9.75       9.00                      \n",
      "   CYS 221 A    11.78       9.00                      \n",
      "   CYS 257 A     6.54       9.00                      \n",
      "   CYS 258 A    10.03       9.00                      \n",
      "   CYS 262 A    17.21       9.00                      \n",
      "   CYS 296 A    12.13       9.00                      \n",
      "   TYR   1 A    10.58      10.00                      \n",
      "   TYR 116 A    10.46      10.00                      \n",
      "   TYR 125 A    12.77      10.00                      \n",
      "   TYR 145 A    16.29      10.00                      \n",
      "   TYR 264 A    13.96      10.00                      \n",
      "   LYS   3 A    10.39      10.50                      \n",
      "   LYS  23 A     9.00      10.50                      \n",
      "   LYS  30 A    10.47      10.50                      \n",
      "   LYS  32 A    10.45      10.50                      \n",
      "   LYS  36 A    13.02      10.50                      \n",
      "   LYS  59 A    10.02      10.50                      \n",
      "   LYS  89 A    10.40      10.50                      \n",
      "   LYS  91 A    10.94      10.50                      \n",
      "   LYS 113 A    10.86      10.50                      \n",
      "   LYS 121 A    10.58      10.50                      \n",
      "   LYS 139 A    11.66      10.50                      \n",
      "   LYS 191 A    10.20      10.50                      \n",
      "   LYS 192 A    11.20      10.50                      \n",
      "   LYS 217 A    10.32      10.50                      \n",
      "   LYS 247 A    11.60      10.50                      \n",
      "   LYS 249 A    10.46      10.50                      \n",
      "   LYS 263 A    11.10      10.50                      \n",
      "   LYS 280 A    10.26      10.50                      \n",
      "   LYS 286 A    10.05      10.50                      \n",
      "   LYS 303 A    11.21      10.50                      \n",
      "   ARG  14 A    12.98      12.50                      \n",
      "   ARG  17 A    13.13      12.50                      \n",
      "   ARG  24 A    12.69      12.50                      \n",
      "   ARG  76 A    12.96      12.50                      \n",
      "   ARG 110 A    10.93      12.50                      \n",
      "   ARG 135 A    14.08      12.50                      \n",
      "   ARG 206 A    12.17      12.50                      \n",
      "   ARG 219 A    12.97      12.50                      \n",
      "   ARG 271 A    12.14      12.50                      \n",
      "   ARG 278 A    11.64      12.50                      \n",
      "   ARG 283 A    13.24      12.50                      \n",
      "   ARG 321 A    11.96      12.50                      \n",
      "   ARG 322 A    10.73      12.50                      \n",
      "   N+    1 A     7.87       8.00                      \n",
      "\n",
      "INFO:PROPKA information:\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "Free energy of   folding (kcal/mol) as a function of pH (using neutral reference)\n",
      "  0.00     34.30\n",
      "  1.00     33.99\n",
      "  2.00     32.12\n",
      "  3.00     25.92\n",
      "  4.00     18.89\n",
      "  5.00     20.28\n",
      "  6.00     24.08\n",
      "  7.00     26.09\n",
      "  8.00     28.36\n",
      "  9.00     32.60\n",
      " 10.00     42.82\n",
      " 11.00     57.76\n",
      " 12.00     71.32\n",
      " 13.00     81.81\n",
      " 14.00     89.66\n",
      "\n",
      "The pH of optimum stability is  4.3 for which the free energy is   18.4 kcal/mol at 298K\n",
      "Could not determine pH values where the free energy is within 80 % of minimum\n",
      "Could not determine the pH-range where the free energy is negative\n",
      "\n",
      "\n",
      "Protein charge of folded and unfolded state as a function of pH\n",
      "    pH  unfolded  folded\n",
      "  0.00     41.99   41.93\n",
      "  0.10     41.99   41.91\n",
      "  0.20     41.99   41.89\n",
      "  0.30     41.99   41.87\n",
      "  0.40     41.99   41.83\n",
      "  0.50     41.98   41.79\n",
      "  0.60     41.98   41.74\n",
      "  0.70     41.97   41.68\n",
      "  0.80     41.97   41.61\n",
      "  0.90     41.96   41.52\n",
      "  1.00     41.95   41.42\n",
      "  1.10     41.93   41.29\n",
      "  1.20     41.92   41.15\n",
      "  1.30     41.89   40.98\n",
      "  1.40     41.87   40.78\n",
      "  1.50     41.83   40.56\n",
      "  1.60     41.79   40.30\n",
      "  1.70     41.74   40.00\n",
      "  1.80     41.67   39.67\n",
      "  1.90     41.59   39.28\n",
      "  2.00     41.48   38.85\n",
      "  2.10     41.35   38.36\n",
      "  2.20     41.19   37.82\n",
      "  2.30     40.98   37.21\n",
      "  2.40     40.73   36.55\n",
      "  2.50     40.42   35.82\n",
      "  2.60     40.04   35.03\n",
      "  2.70     39.58   34.17\n",
      "  2.80     39.01   33.26\n",
      "  2.90     38.33   32.30\n",
      "  3.00     37.52   31.28\n",
      "  3.10     36.55   30.20\n",
      "  3.20     35.43   29.07\n",
      "  3.30     34.13   27.88\n",
      "  3.40     32.64   26.64\n",
      "  3.50     30.98   25.34\n",
      "  3.60     29.14   23.99\n",
      "  3.70     27.15   22.59\n",
      "  3.80     25.03   21.14\n",
      "  3.90     22.80   19.65\n",
      "  4.00     20.50   18.12\n",
      "  4.10     18.16   16.58\n",
      "  4.20     15.82   15.03\n",
      "  4.30     13.52   13.49\n",
      "  4.40     11.28   11.96\n",
      "  4.50      9.13   10.46\n",
      "  4.60      7.10    9.01\n",
      "  4.70      5.21    7.61\n",
      "  4.80      3.48    6.26\n",
      "  4.90      1.93    4.98\n",
      "  5.00      0.54    3.76\n",
      "  5.10     -0.69    2.62\n",
      "  5.20     -1.76    1.54\n",
      "  5.30     -2.70    0.52\n",
      "  5.40     -3.52   -0.43\n",
      "  5.50     -4.24   -1.33\n",
      "  5.60     -4.89   -2.18\n",
      "  5.70     -5.49   -2.99\n",
      "  5.80     -6.05   -3.76\n",
      "  5.90     -6.58   -4.51\n",
      "  6.00     -7.11   -5.23\n",
      "  6.10     -7.64   -5.93\n",
      "  6.20     -8.18   -6.60\n",
      "  6.30     -8.72   -7.24\n",
      "  6.40     -9.26   -7.86\n",
      "  6.50     -9.80   -8.44\n",
      "  6.60    -10.34   -8.98\n",
      "  6.70    -10.85   -9.48\n",
      "  6.80    -11.34   -9.94\n",
      "  6.90    -11.79  -10.36\n",
      "  7.00    -12.21  -10.74\n",
      "  7.10    -12.60  -11.08\n",
      "  7.20    -12.96  -11.39\n",
      "  7.30    -13.29  -11.69\n",
      "  7.40    -13.59  -11.96\n",
      "  7.50    -13.89  -12.23\n",
      "  7.60    -14.18  -12.49\n",
      "  7.70    -14.47  -12.75\n",
      "  7.80    -14.78  -13.01\n",
      "  7.90    -15.10  -13.27\n",
      "  8.00    -15.45  -13.54\n",
      "  8.10    -15.84  -13.82\n",
      "  8.20    -16.27  -14.10\n",
      "  8.30    -16.76  -14.39\n",
      "  8.40    -17.30  -14.69\n",
      "  8.50    -17.91  -15.00\n",
      "  8.60    -18.59  -15.33\n",
      "  8.70    -19.34  -15.67\n",
      "  8.80    -20.15  -16.04\n",
      "  8.90    -21.03  -16.42\n",
      "  9.00    -21.95  -16.84\n",
      "  9.10    -22.92  -17.29\n",
      "  9.20    -23.93  -17.77\n",
      "  9.30    -24.96  -18.30\n",
      "  9.40    -26.02  -18.88\n",
      "  9.50    -27.11  -19.51\n",
      "  9.60    -28.24  -20.20\n",
      "  9.70    -29.41  -20.95\n",
      "  9.80    -30.63  -21.77\n",
      "  9.90    -31.90  -22.67\n",
      " 10.00    -33.24  -23.64\n",
      " 10.10    -34.64  -24.67\n",
      " 10.20    -36.09  -25.77\n",
      " 10.30    -37.57  -26.92\n",
      " 10.40    -39.07  -28.12\n",
      " 10.50    -40.56  -29.34\n",
      " 10.60    -42.01  -30.59\n",
      " 10.70    -43.40  -31.84\n",
      " 10.80    -44.70  -33.09\n",
      " 10.90    -45.91  -34.32\n",
      " 11.00    -47.02  -35.53\n",
      " 11.10    -48.02  -36.72\n",
      " 11.20    -48.92  -37.88\n",
      " 11.30    -49.74  -39.01\n",
      " 11.40    -50.48  -40.11\n",
      " 11.50    -51.17  -41.18\n",
      " 11.60    -51.83  -42.22\n",
      " 11.70    -52.47  -43.23\n",
      " 11.80    -53.11  -44.22\n",
      " 11.90    -53.77  -45.19\n",
      " 12.00    -54.45  -46.13\n",
      " 12.10    -55.16  -47.06\n",
      " 12.20    -55.91  -47.96\n",
      " 12.30    -56.69  -48.85\n",
      " 12.40    -57.48  -49.72\n",
      " 12.50    -58.28  -50.57\n",
      " 12.60    -59.07  -51.42\n",
      " 12.70    -59.83  -52.25\n",
      " 12.80    -60.55  -53.07\n",
      " 12.90    -61.21  -53.86\n",
      " 13.00    -61.81  -54.64\n",
      " 13.10    -62.34  -55.39\n",
      " 13.20    -62.79  -56.10\n",
      " 13.30    -63.19  -56.77\n",
      " 13.40    -63.52  -57.41\n",
      " 13.50    -63.80  -58.00\n",
      " 13.60    -64.03  -58.55\n",
      " 13.70    -64.21  -59.05\n",
      " 13.80    -64.37  -59.52\n",
      " 13.90    -64.49  -59.95\n",
      " 14.00    -64.59  -60.34\n",
      "The pI is  5.35 (folded) and  5.04 (unfolded)\n",
      "\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "INFO:Applying pKa values at a pH of 7.00:\n",
      "INFO:Adding hydrogens to biomolecule.\n",
      "INFO:Debumping biomolecule (again).\n",
      "INFO:Optimizing hydrogen bonds\n",
      "INFO:Applying force field to biomolecule states.\n",
      "INFO:Regenerating headers.\n",
      "INFO:Regenerating PDB lines.\n",
      "WARNING:Ignoring 467 header lines in output.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "pH: 7.0\tname: 74987\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:PDB2PQR v3.6.2: biomolecular structure conversion software.\n",
      "INFO:Please cite:  Jurrus E, et al.  Improvements to the APBS biomolecular solvation software suite.  Protein Sci 27 112-128 (2018).\n",
      "INFO:Please cite:  Dolinsky TJ, et al.  PDB2PQR: expanding and upgrading automated preparation of biomolecular structures for molecular simulations. Nucleic Acids Res 35 W522-W525 (2007).\n",
      "INFO:Checking and transforming input arguments.\n",
      "INFO:Loading topology files.\n",
      "INFO:Loading molecule: /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/1ka1_hal2.pdb\n",
      "INFO:Setting up molecule.\n",
      "INFO:Created biomolecule object with 354 residues and 2718 atoms.\n",
      "WARNING:Multiple occupancies found: OG in SER A 84.\n",
      "WARNING:Multiple occupancies found in SER A 84. At least one of the instances is being ignored.\n",
      "INFO:Setting termini states for biomolecule chains.\n",
      "INFO:Loading forcefield.\n",
      "INFO:Loading hydrogen topology definitions.\n",
      "WARNING:Missing atom CB in residue LYS A 32\n",
      "WARNING:Missing atom CG in residue LYS A 32\n",
      "WARNING:Missing atom CD in residue LYS A 32\n",
      "WARNING:Missing atom CE in residue LYS A 32\n",
      "WARNING:Missing atom NZ in residue LYS A 32\n",
      "WARNING:Missing atom CB in residue ASP A 33\n",
      "WARNING:Missing atom CG in residue ASP A 33\n",
      "WARNING:Missing atom OD1 in residue ASP A 33\n",
      "WARNING:Missing atom OD2 in residue ASP A 33\n",
      "WARNING:Missing atom CG in residue LYS A 189\n",
      "WARNING:Missing atom CD in residue LYS A 189\n",
      "WARNING:Missing atom CE in residue LYS A 189\n",
      "WARNING:Missing atom NZ in residue LYS A 189\n",
      "WARNING:Missing atom CD in residue LYS A 229\n",
      "WARNING:Missing atom CE in residue LYS A 229\n",
      "WARNING:Missing atom NZ in residue LYS A 229\n",
      "WARNING:Missing atom OXT in residue ARG A 355\n",
      "WARNING:Missing atom CB in residue LYS A 32\n",
      "WARNING:Missing atom CG in residue LYS A 32\n",
      "WARNING:Missing atom CD in residue LYS A 32\n",
      "WARNING:Missing atom CE in residue LYS A 32\n",
      "WARNING:Missing atom NZ in residue LYS A 32\n",
      "WARNING:Missing atom CB in residue ASP A 33\n",
      "WARNING:Missing atom CG in residue ASP A 33\n",
      "WARNING:Missing atom OD1 in residue ASP A 33\n",
      "WARNING:Missing atom OD2 in residue ASP A 33\n",
      "WARNING:Missing atom CG in residue LYS A 189\n",
      "WARNING:Missing atom CD in residue LYS A 189\n",
      "WARNING:Missing atom CE in residue LYS A 189\n",
      "WARNING:Missing atom NZ in residue LYS A 189\n",
      "WARNING:Missing atom CD in residue LYS A 229\n",
      "WARNING:Missing atom CE in residue LYS A 229\n",
      "WARNING:Missing atom NZ in residue LYS A 229\n",
      "WARNING:Missing atom OXT in residue ARG A 355\n",
      "INFO:Attempting to repair 17 missing atoms in biomolecule.\n",
      "WARNING:Missing atom CB in residue LYS A 32\n",
      "WARNING:Missing atom CG in residue LYS A 32\n",
      "WARNING:Missing atom CD in residue LYS A 32\n",
      "WARNING:Missing atom CE in residue LYS A 32\n",
      "WARNING:Missing atom NZ in residue LYS A 32\n",
      "WARNING:Missing atom CB in residue ASP A 33\n",
      "WARNING:Missing atom CG in residue ASP A 33\n",
      "WARNING:Missing atom OD1 in residue ASP A 33\n",
      "WARNING:Missing atom OD2 in residue ASP A 33\n",
      "WARNING:Missing atom CG in residue LYS A 189\n",
      "WARNING:Missing atom CD in residue LYS A 189\n",
      "WARNING:Missing atom CE in residue LYS A 189\n",
      "WARNING:Missing atom NZ in residue LYS A 189\n",
      "WARNING:Missing atom CD in residue LYS A 229\n",
      "WARNING:Missing atom CE in residue LYS A 229\n",
      "WARNING:Missing atom NZ in residue LYS A 229\n",
      "WARNING:Missing atom OXT in residue ARG A 355\n",
      "INFO:Added atom CB to residue LYS A 32 at coordinates -45.594, -33.973, -23.265\n",
      "INFO:Added atom CG to residue LYS A 32 at coordinates -45.153, -35.334, -22.744\n",
      "INFO:Added atom CD to residue LYS A 32 at coordinates -43.702, -35.621, -23.091\n",
      "INFO:Added atom CE to residue LYS A 32 at coordinates -43.239, -36.968, -22.584\n",
      "INFO:Added atom NZ to residue LYS A 32 at coordinates -41.819, -37.204, -22.945\n",
      "INFO:Added atom CB to residue ASP A 33 at coordinates -49.706, -30.631, -25.302\n",
      "INFO:Added atom CG to residue ASP A 33 at coordinates -48.707, -31.048, -26.377\n",
      "INFO:Added atom OD1 to residue ASP A 33 at coordinates -47.538, -31.342, -26.061\n",
      "INFO:Added atom OD2 to residue ASP A 33 at coordinates -49.132, -31.087, -27.541\n",
      "INFO:Added atom CG to residue LYS A 189 at coordinates -35.105, -63.964, -9.656\n",
      "INFO:Added atom CD to residue LYS A 189 at coordinates -35.396, -65.193, -8.824\n",
      "INFO:Added atom CE to residue LYS A 189 at coordinates -34.533, -66.375, -9.200\n",
      "INFO:Added atom NZ to residue LYS A 189 at coordinates -34.854, -67.554, -8.358\n",
      "INFO:Added atom CD to residue LYS A 229 at coordinates -3.896, -48.760, -18.063\n",
      "INFO:Added atom CE to residue LYS A 229 at coordinates -3.416, -50.110, -17.576\n",
      "INFO:Added atom NZ to residue LYS A 229 at coordinates -2.176, -50.508, -18.287\n",
      "INFO:Added atom OXT to residue ARG A 355 at coordinates -9.183, -18.959, -17.852\n",
      "INFO:Updating disulfide bridges.\n",
      "INFO:Debumping biomolecule.\n",
      "INFO:Assigning titration states with PROPKA.\n",
      "INFO:\n",
      "propka3.5.1                                                                                  2024-07-26\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "--                                                                           --\n",
      "--  PROPKA: A PROTEIN PKA PREDICTOR                                          --\n",
      "--                                                                           --\n",
      "--  VERSION 1.0,  04/25/2004,  IOWA CITY                                     --\n",
      "--  BY HUI LI                                                                --\n",
      "--                                                                           --\n",
      "--  VERSION 2.0,  11/05/2007, IOWA CITY/COPENHAGEN                           --\n",
      "--  BY DELPHINE C. BAS AND DAVID M. ROGERS                                   --\n",
      "--                                                                           --\n",
      "--  VERSION 3.0,  01/06/2011, COPENHAGEN                                     --\n",
      "--  BY MATS H.M. OLSSON AND CHRESTEN R. SONDERGARD                           --\n",
      "--                                                                           --\n",
      "--  VERSION 3.1,  07/01/2011, COPENHAGEN                                     --\n",
      "--  BY CHRESTEN R. SONDERGARD AND MATS H.M. OLSSON                           --\n",
      "--                                                                           --\n",
      "--  VERSION 3.2,  06/17/2020, PLANET EARTH                                   --\n",
      "--  SEE ABOVE FOR AUTHORS                                                    --\n",
      "--                                                                           --\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "References:\n",
      "\n",
      "Very Fast Empirical Prediction and Rationalization of Protein pKa Values.\n",
      "Hui Li, Andrew D. Robertson and Jan H. Jensen. PROTEINS: Structure, Function,\n",
      "and Bioinformatics. 61:704-721 (2005)\n",
      "\n",
      "Very Fast Prediction and Rationalization of pKa Values for Protein-Ligand\n",
      "Complexes.  Delphine C. Bas, David M. Rogers and Jan H. Jensen.  PROTEINS:\n",
      "Structure, Function, and Bioinformatics 73:765-783 (2008)\n",
      "\n",
      "PROPKA3: Consistent Treatment of Internal and Surface Residues in Empirical\n",
      "pKa predictions.  Mats H.M. Olsson, Chresten R. Sondergard, Michal Rostkowski,\n",
      "and Jan H. Jensen.  Journal of Chemical Theory and Computation, 7(2):525-537\n",
      "(2011)\n",
      "\n",
      "Improved Treatment of Ligands and Coupling Effects in Empirical Calculation\n",
      "and Rationalization of pKa Values.  Chresten R. Sondergaard, Mats H.M. Olsson,\n",
      "Michal Rostkowski, and Jan H. Jensen.  Journal of Chemical Theory and\n",
      "Computation, (2011)\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "\n",
      "WARNING:Missing atoms or failed protonation for HIS  31 A (HIS) -- please check the structure\n",
      "WARNING:Group (HIS) for   231-  CG    31-HIS (A) [ -51.746  -37.773  -23.250] C\n",
      "WARNING:Expected 4 interaction atoms for acids, found:\n",
      "WARNING:                 0- HE2    31-HIS (A) [ -54.068  -39.642  -24.334] H\n",
      "WARNING:               233- ND1    31-HIS (A) [ -51.129  -38.973  -23.536] N\n",
      "WARNING:               235- NE2    31-HIS (A) [ -53.201  -39.225  -24.025] N\n",
      "WARNING:Expected 2 interaction atoms for bases, found:\n",
      "WARNING:               233- ND1    31-HIS (A) [ -51.129  -38.973  -23.536] N\n",
      "WARNING:               235- NE2    31-HIS (A) [ -53.201  -39.225  -24.025] N\n",
      "INFO:Coupling map:\n",
      "\n",
      "INFO:Calculating pKas for Conformation container 1A with 3249 atoms and 890 groups\n",
      "INFO:pKa determinants:\n",
      "\n",
      "---------  -----   ------   ---------------------    --------------    --------------    --------------\n",
      "                            DESOLVATION  EFFECTS       SIDECHAIN          BACKBONE        COULOMBIC\n",
      " RESIDUE    pKa    BURIED     REGULAR      RE        HYDROGEN BOND     HYDROGEN BOND      INTERACTION\n",
      "---------  -----   ------   ---------   ---------    --------------    --------------    --------------\n",
      "\n",
      "ASP  33 A   3.91     0 %    0.20  152   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS  32 A\n",
      "ASP  33 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 HIS  31 A\n",
      "\n",
      "ASP  41 A   2.21    11 %    0.60  312   0.00    0   -0.79 SER  43 A   -0.23 ASP  41 A   -0.28 LYS  39 A\n",
      "ASP  41 A                                            0.00 XXX   0 X   -0.73 SER  43 A   -0.04 LYS 148 A\n",
      "ASP  41 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.12 LYS 291 A\n",
      "ASP  41 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 HIS 241 A\n",
      "\n",
      "ASP  49 A   6.17*  100 %    3.70  606   1.05    0   -0.84 THR  46 A   -0.70 THR  46 A   -0.30 LYS  39 A\n",
      "ASP  49 A                                           -0.85 THR 147 A   -0.24 ASP  49 A    0.55 GLU  72 A\n",
      "\n",
      "ASP  66 A   3.97     0 %    0.19  204   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.11 LYS  61 A\n",
      "ASP  66 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP  67 A\n",
      "\n",
      "ASP  67 A   3.50    45 %    1.80  406   0.24    0   -0.81 TRP 139 A    0.00 XXX   0 X   -0.03 ARG   5 A\n",
      "ASP  67 A                                           -0.70 ARG 137 A    0.00 XXX   0 X   -0.15 LYS  61 A\n",
      "ASP  67 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 LYS  68 A\n",
      "ASP  67 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.57 ARG 137 A\n",
      "\n",
      "ASP  80 A   3.08     0 %    0.24  175   0.00    0   -0.70 ARG 122 A    0.00 XXX   0 X   -0.26 ARG 122 A\n",
      "\n",
      "ASP  94 A   2.91    69 %    1.87  474   0.67    0   -0.85 LYS  16 A    0.00 XXX   0 X   -0.00 ARG  23 A\n",
      "ASP  94 A                                           -0.85 LYS  91 A    0.00 XXX   0 X   -0.00 LYS 116 A\n",
      "ASP  94 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.05 LYS  16 A\n",
      "ASP  94 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.67 LYS  91 A\n",
      "\n",
      "ASP 104 A   3.88     0 %    0.20  138   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.04 LYS 102 A\n",
      "ASP 104 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 LYS 103 A\n",
      "\n",
      "ASP 105 A   4.10     0 %    0.13  159   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG   5 A\n",
      "ASP 105 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS 103 A\n",
      "ASP 105 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.21 ASP 104 A\n",
      "\n",
      "ASP 111 A   3.81     0 %    0.12  140   0.00    0    0.00 XXX   0 X   -0.11 ASP 111 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 120 A   2.75    13 %    0.72  318   0.05    0   -0.32 SER 117 A   -0.79 LYS 116 A   -0.02 LYS 116 A\n",
      "ASP 120 A                                            0.00 XXX   0 X   -0.69 SER 117 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 126 A   3.39    34 %    1.22  376   0.28    0   -0.16 SER  74 A   -0.85 SER  75 A   -0.06 ARG 122 A\n",
      "ASP 126 A                                           -0.85 SER  75 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "ASP 142 A  16.43*  100 %    3.57  642   1.03    0    0.48 GLU  72 A   -0.03 ILE 144 A   -0.28 LYS  39 A\n",
      "ASP 142 A                                            0.67 ASP 145 A    0.00 XXX   0 X    0.50 ASP  49 A\n",
      "ASP 142 A                                            0.37 ASP 294 A    0.00 XXX   0 X    0.17 CYS 140 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.88 CYS 159 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.87 GLU  72 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.59 GLU  73 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.57 ASP 145 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.23 ASP 294 A\n",
      "\n",
      "ASP 145 A   3.62*  100 %    2.82  672   0.55    0   -0.67 ASP 142 A   -0.04 GLY 146 A   -0.19 HIS 241 A\n",
      "ASP 145 A                                           -0.73 ASP 294 A   -0.57 GLN 265 A    0.17 ASP 263 A\n",
      "ASP 145 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 GLU 290 A\n",
      "ASP 145 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.44 HIS 295 A\n",
      "ASP 145 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.65 LYS 267 A\n",
      "ASP 145 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.55 ARG 281 A\n",
      "\n",
      "ASP 165 A   3.15     0 %    0.28  211   0.00    0    0.00 XXX   0 X   -0.81 GLY 136 A   -0.03 LYS  68 A\n",
      "ASP 165 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.00 ARG 134 A\n",
      "ASP 165 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 LYS 135 A\n",
      "\n",
      "ASP 187 A   2.15    10 %    1.13  308   0.05    0   -0.17 SER 181 A   -0.76 LEU 180 A   -0.41 HIS 191 A\n",
      "ASP 187 A                                           -0.85 TYR 196 A   -0.63 SER 181 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 213 A   3.05    18 %    0.79  331   0.14    0    0.00 XXX   0 X   -0.76 GLU 215 A   -0.09 ARG  15 A\n",
      "ASP 213 A                                            0.00 XXX   0 X   -0.84 SER 216 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 227 A   3.07     0 %    0.35  211   0.00    0    0.00 XXX   0 X   -0.84 LYS 229 A   -0.24 LYS 229 A\n",
      "\n",
      "ASP 230 A   3.96     0 %    0.30  181   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.03 ARG 223 A\n",
      "ASP 230 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.20 LYS 229 A\n",
      "ASP 230 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.09 LYS 258 A\n",
      "ASP 230 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP 227 A\n",
      "ASP 230 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 277 A\n",
      "\n",
      "ASP 245 A   3.84     0 %    0.18  234   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.09 LYS 239 A\n",
      "ASP 245 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 HIS 244 A\n",
      "ASP 245 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 GLU 246 A\n",
      "\n",
      "ASP 263 A   3.38    62 %    2.13  455   0.38    0   -1.55 ARG 152 A   -0.09 ASP 263 A   -0.03 LYS 148 A\n",
      "ASP 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.50 LYS 267 A\n",
      "ASP 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.20 HIS 241 A\n",
      "ASP 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.57 ARG 152 A\n",
      "\n",
      "ASP 277 A   3.47    56 %    2.43  438   0.35    0   -1.64 ARG 223 A   -0.78 ILE 232 A   -0.04 LYS 258 A\n",
      "ASP 277 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 HIS 224 A\n",
      "ASP 277 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.61 ARG 223 A\n",
      "\n",
      "ASP 294 A   9.23*  100 %    2.63  612   0.38    0   -0.37 ASP 142 A    0.00 XXX   0 X   -0.04 LYS  39 A\n",
      "ASP 294 A                                            0.73 ASP 145 A    0.00 XXX   0 X   -0.08 LYS 267 A\n",
      "ASP 294 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.35 LYS 291 A\n",
      "ASP 294 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.93 GLU  72 A\n",
      "ASP 294 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.64 ASP 145 A\n",
      "ASP 294 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.87 ARG 281 A\n",
      "ASP 294 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.84 GLU 290 A\n",
      "\n",
      "ASP 311 A   3.81    51 %    2.24  424   0.49    0    0.00 XXX   0 X   -0.12 ASP 311 A   -0.05 HIS 341 A\n",
      "ASP 311 A                                            0.00 XXX   0 X   -0.61 MET 313 A    0.00 XXX   0 X\n",
      "ASP 311 A                                            0.00 XXX   0 X   -0.83 GLU 314 A    0.00 XXX   0 X\n",
      "ASP 311 A                                            0.00 XXX   0 X   -0.40 ASP 315 A    0.00 XXX   0 X\n",
      "ASP 311 A                                            0.00 XXX   0 X   -0.72 VAL 316 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 315 A   6.34*   20 %    0.96  336   0.20    0    0.87 ASP 342 A    0.00 XXX   0 X   -0.04 ARG 338 A\n",
      "ASP 315 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.14 ASP 311 A\n",
      "ASP 315 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.40 ASP 342 A\n",
      "\n",
      "ASP 319 A   1.83    15 %    0.73  324   0.03    0   -1.58 ARG 134 A   -0.76 GLY 321 A   -0.38 ARG 134 A\n",
      "\n",
      "ASP 342 A   3.08*    0 %    0.48  257   0.00    0   -0.87 ASP 315 A    0.00 XXX   0 X   -0.15 ARG 338 A\n",
      "ASP 342 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.18 HIS 341 A\n",
      "\n",
      "ASP 350 A   3.76     0 %    0.21  206   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.25 LYS 253 A\n",
      "\n",
      "GLU   4 A   2.92     0 %    0.28  187   0.00    0   -0.77 N+    2 A   -0.70 GLU   4 A   -0.38 N+    2 A\n",
      "\n",
      "GLU   6 A   3.84    24 %    1.04  349   0.00    0   -0.65 ARG 137 A   -0.66 GLU   6 A   -0.05 N+    2 A\n",
      "GLU   6 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.12 ARG   5 A\n",
      "GLU   6 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU   4 A\n",
      "GLU   6 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.14 ASP  67 A\n",
      "GLU   6 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.43 ARG 137 A\n",
      "\n",
      "GLU  27 A   4.11    13 %    0.87  318   0.20    0   -0.85 SER  26 A   -0.45 GLU  27 A   -0.00 LYS  22 A\n",
      "GLU  27 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.09 ARG  23 A\n",
      "GLU  27 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.07 HIS  31 A\n",
      "\n",
      "GLU  72 A   5.47*  100 %    2.36  562   0.17    0   -0.34 LYS  39 A    0.00 XXX   0 X   -0.16 LYS 291 A\n",
      "GLU  72 A                                           -0.48 ASP 142 A    0.00 XXX   0 X   -0.05 ARG 324 A\n",
      "GLU  72 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.14 ASP  41 A\n",
      "GLU  72 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.42 ASP 145 A\n",
      "GLU  72 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.10 LYS  39 A\n",
      "\n",
      "GLU  73 A   8.38*   90 %    2.95  532   0.08    0   -0.85 LYS  39 A    0.00 XXX   0 X   -0.01 ARG 324 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 ASP  41 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.00 LYS  39 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.09 ASP  49 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.55 GLU  72 A\n",
      "\n",
      "GLU  89 A   5.51    46 %    1.01  409   0.11    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.20 ARG  23 A\n",
      "GLU  89 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.12 GLU  27 A\n",
      "GLU  89 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 HIS  31 A\n",
      "\n",
      "GLU  95 A   3.70     0 %    0.31  175   0.00    0   -0.68 LYS  99 A    0.00 XXX   0 X   -0.05 ARG  23 A\n",
      "GLU  95 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.38 LYS  99 A\n",
      "\n",
      "GLU 119 A   4.30     0 %    0.29  205   0.00    0    0.00 XXX   0 X   -0.63 GLU 119 A   -0.10 ARG 122 A\n",
      "GLU 119 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 ASP  80 A\n",
      "GLU 119 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 ASP 120 A\n",
      "\n",
      "GLU 131 A   4.45     1 %    0.36  283   0.01    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.41 ARG 324 A\n",
      "\n",
      "GLU 154 A   4.04     2 %    0.35  287   0.01    0   -0.72 ARG 152 A    0.00 XXX   0 X    0.02 GLU 238 A\n",
      "GLU 154 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.18 ASP 263 A\n",
      "GLU 154 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.31 ARG 152 A\n",
      "\n",
      "GLU 192 A   3.88     0 %    0.35  255   0.00    0    0.00 XXX   0 X   -0.04 GLY 190 A    0.02 ASP 187 A\n",
      "GLU 192 A                                            0.00 XXX   0 X   -0.73 GLU 192 A   -0.23 HIS 191 A\n",
      "\n",
      "GLU 215 A   3.63    10 %    0.43  309   0.07    0   -0.85 LYS 103 A    0.00 XXX   0 X   -0.16 ARG  15 A\n",
      "GLU 215 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.05 ARG 199 A\n",
      "GLU 215 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP 213 A\n",
      "GLU 215 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.41 LYS 103 A\n",
      "\n",
      "GLU 235 A   3.26    68 %    2.28  472   0.12    0   -0.46 THR 233 A    0.00 XXX   0 X   -0.30 HIS 244 A\n",
      "GLU 235 A                                           -0.85 SER 259 A    0.00 XXX   0 X   -0.65 LYS 251 A\n",
      "GLU 235 A                                           -0.85 LYS 251 A    0.00 XXX   0 X   -0.53 HIS 261 A\n",
      "\n",
      "GLU 238 A   3.63    15 %    0.62  323   0.00    0   -0.74 ARG 152 A    0.00 XXX   0 X   -0.35 LYS 148 A\n",
      "GLU 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.30 HIS 241 A\n",
      "GLU 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 HIS 244 A\n",
      "GLU 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.24 ASP 263 A\n",
      "GLU 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.33 ARG 152 A\n",
      "\n",
      "GLU 246 A   2.95    20 %    1.10  337   0.12    0   -0.73 SER 243 A    0.00 XXX   0 X   -0.43 ARG 355 A\n",
      "GLU 246 A                                           -1.62 ARG 355 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "GLU 290 A   1.29    98 %    3.05  556   0.08    0   -0.59 ARG 281 A    0.00 XXX   0 X   -0.00 LYS 267 A\n",
      "GLU 290 A                                           -1.60 HIS 295 A    0.00 XXX   0 X   -0.23 LYS 291 A\n",
      "GLU 290 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 LYS 329 A\n",
      "GLU 290 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.10 HIS 241 A\n",
      "GLU 290 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -2.03 ARG 281 A\n",
      "GLU 290 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.70 HIS 295 A\n",
      "\n",
      "GLU 304 A   3.18    26 %    1.06  354   0.21    0   -0.46 HIS 303 A   -0.84 GLY 203 A   -0.19 ARG 202 A\n",
      "GLU 304 A                                            0.00 XXX   0 X   -0.00 LEU 204 A   -0.05 LYS 219 A\n",
      "GLU 304 A                                            0.00 XXX   0 X   -0.74 GLY 205 A   -0.31 HIS 303 A\n",
      "\n",
      "GLU 314 A   5.06     0 %    0.26  273   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.01 LYS 329 A\n",
      "GLU 314 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.31 ASP 311 A\n",
      "\n",
      "GLU 339 A   4.02     0 %    0.19  164   0.00    0    0.00 XXX   0 X   -0.67 GLU 339 A   -0.07 ARG 338 A\n",
      "GLU 339 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP 342 A\n",
      "\n",
      "C-  355 A   3.34     0 %    0.16  125   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.07 ARG 355 A\n",
      "C-  355 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 246 A\n",
      "\n",
      "HIS  31 A   6.33     0 %   -0.29  196   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.07 GLU  27 A\n",
      "HIS  31 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP  33 A\n",
      "HIS  31 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU  89 A\n",
      "\n",
      "HIS 191 A   6.82     6 %   -0.67  298   0.00    0    0.00 XXX   0 X    0.37 LEU 188 A   -0.02 LYS  22 A\n",
      "HIS 191 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.41 ASP 187 A\n",
      "HIS 191 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.23 GLU 192 A\n",
      "\n",
      "HIS 221 A   5.75    12 %   -0.53  315   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 219 A\n",
      "HIS 221 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.17 HIS 224 A\n",
      "\n",
      "HIS 224 A   6.57    11 %   -0.55  311   0.00    0    0.00 XXX   0 X    0.58 SER 335 A    0.04 ASP 277 A\n",
      "\n",
      "HIS 241 A   4.98    65 %   -1.23  464   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.26 LYS 148 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 ARG 152 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.17 LYS 267 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.56 ARG 281 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP  41 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 ASP 145 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.30 GLU 238 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 HIS 244 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 ASP 263 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 GLU 290 A\n",
      "\n",
      "HIS 244 A   5.80    28 %   -0.91  359   0.00    0    0.00 XXX   0 X    0.04 GLY 236 A   -0.09 LYS 239 A\n",
      "HIS 244 A                                            0.00 XXX   0 X    0.22 VAL 237 A   -0.10 LYS 251 A\n",
      "HIS 244 A                                            0.00 XXX   0 X    0.01 SER 242 A   -0.01 LYS 267 A\n",
      "HIS 244 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.30 GLU 235 A\n",
      "HIS 244 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 238 A\n",
      "HIS 244 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP 245 A\n",
      "HIS 244 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.27 HIS 261 A\n",
      "\n",
      "HIS 261 A   6.60    35 %   -0.93  379   0.00    0    0.00 XXX   0 X    0.85 GLY 236 A   -0.29 LYS 251 A\n",
      "HIS 261 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 267 A\n",
      "HIS 261 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.53 GLU 235 A\n",
      "\n",
      "HIS 295 A   5.60   100 %   -3.33  658   0.00    0    1.60 GLU 290 A    0.00 XXX   0 X   -0.15 LYS 267 A\n",
      "HIS 295 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.16 ARG 281 A\n",
      "HIS 295 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.00 LYS 291 A\n",
      "HIS 295 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.44 ASP 145 A\n",
      "HIS 295 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.70 GLU 290 A\n",
      "\n",
      "HIS 303 A   6.22    34 %   -0.66  377   0.00    0    0.46 GLU 304 A    0.00 XXX   0 X   -0.39 ARG 202 A\n",
      "HIS 303 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.31 GLU 304 A\n",
      "\n",
      "HIS 309 A   2.56   100 %   -3.24  632   0.00    0    0.00 XXX   0 X    0.06 HIS 295 A   -0.74 HIS 295 A\n",
      "HIS 309 A                                            0.00 XXX   0 X    0.23 ILE 332 A   -0.25 HIS 341 A\n",
      "\n",
      "HIS 341 A   4.22    77 %   -2.52  498   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 311 A\n",
      "HIS 341 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.18 ASP 342 A\n",
      "\n",
      "CYS 140 A  12.78   100 %    3.75  624   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.13 ARG 324 A\n",
      "CYS 140 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 ASP 294 A\n",
      "\n",
      "CYS 159 A  15.19   100 %    3.86  700   0.00    0   -0.06 GLN 265 A    0.00 XXX   0 X   -0.03 LYS 267 A\n",
      "CYS 159 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 CYS 140 A\n",
      "CYS 159 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 CYS 172 A\n",
      "CYS 159 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.30 ASP 294 A\n",
      "CYS 159 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.32 ASP 145 A\n",
      "CYS 159 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.53 CYS 175 A\n",
      "\n",
      "CYS 172 A  12.08   100 %    3.66  715   0.00    0   -0.14 THR  11 A    0.00 XXX   0 X   -0.50 ARG 199 A\n",
      "CYS 172 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 CYS 175 A\n",
      "\n",
      "CYS 175 A   9.09*  100 %    3.67  714   0.00    0   -3.60 CYS 269 A    0.00 XXX   0 X    0.02 ASP 145 A\n",
      "\n",
      "CYS 269 A  19.48*  100 %    4.24  712   0.00    0    3.60 CYS 175 A   -0.02 CYS 269 A    0.49 CYS 159 A\n",
      "CYS 269 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 CYS 172 A\n",
      "CYS 269 A                                            0.00 XXX   0 X    0.00 XXX   0 X    2.03 CYS 175 A\n",
      "\n",
      "CYS 349 A   9.57     1 %    0.40  284   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 311 A\n",
      "CYS 349 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 314 A\n",
      "CYS 349 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.11 ASP 350 A\n",
      "\n",
      "TYR  50 A  12.47    60 %    1.83  448   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.23 ASP  49 A\n",
      "TYR  50 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU  72 A\n",
      "TYR  50 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 126 A\n",
      "TYR  50 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.29 LYS  39 A\n",
      "TYR  50 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.60 GLU  73 A\n",
      "\n",
      "TYR  97 A  12.15    89 %    3.48  531   0.00    0   -0.35 ARG  15 A    0.00 XXX   0 X   -1.07 ARG  15 A\n",
      "TYR  97 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP  94 A\n",
      "TYR  97 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 215 A\n",
      "TYR  97 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 TYR 101 A\n",
      "\n",
      "TYR 101 A  10.24    26 %    0.81  354   0.00    0   -0.53 SER 193 A    0.00 XXX   0 X   -0.10 ARG  15 A\n",
      "TYR 101 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 GLU 192 A\n",
      "\n",
      "TYR 130 A  10.10     4 %    0.46  293   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.36 LYS  68 A\n",
      "\n",
      "TYR 183 A  11.75    61 %    1.58  453   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 154 A\n",
      "TYR 183 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 CYS 175 A\n",
      "TYR 183 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP 187 A\n",
      "TYR 183 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 263 A\n",
      "\n",
      "TYR 196 A  13.17    38 %    2.33  389   0.00    0    0.85 ASP 187 A   -0.79 LEU 188 A    0.05 GLU 192 A\n",
      "TYR 196 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.27 TYR 208 A\n",
      "TYR 196 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.48 ASP 187 A\n",
      "\n",
      "TYR 208 A  11.41    46 %    1.34  411   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.07 ASP 187 A\n",
      "\n",
      "TYR 268 A  17.93   100 %    4.86  684   0.00    0    0.77 TYR 279 A    0.00 XXX   0 X    0.17 ASP 145 A\n",
      "TYR 268 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 CYS 159 A\n",
      "TYR 268 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.54 ARG 281 A\n",
      "TYR 268 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.46 GLU 290 A\n",
      "TYR 268 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 ASP 294 A\n",
      "TYR 268 A                                            0.00 XXX   0 X    0.00 XXX   0 X    2.03 TYR 279 A\n",
      "\n",
      "TYR 279 A  12.08   100 %    3.49  678   0.00    0   -0.51 ARG 281 A    0.00 XXX   0 X    0.50 ASP 145 A\n",
      "TYR 279 A                                           -0.77 TYR 268 A    0.00 XXX   0 X    0.01 CYS 159 A\n",
      "TYR 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.75 ARG 281 A\n",
      "TYR 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.36 ASP 294 A\n",
      "TYR 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.76 GLU 290 A\n",
      "\n",
      "TYR 288 A  12.91    68 %    1.44  472   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP  41 A\n",
      "TYR 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.17 GLU  72 A\n",
      "TYR 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP 142 A\n",
      "TYR 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 ASP 145 A\n",
      "TYR 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.51 ARG 281 A\n",
      "TYR 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.80 GLU 290 A\n",
      "TYR 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.85 ASP 294 A\n",
      "\n",
      "LYS  16 A   9.20    87 %   -2.94  524   0.00    0    0.85 ASP  94 A    0.00 XXX   0 X   -0.26 LYS  91 A\n",
      "LYS  16 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.05 ASP  94 A\n",
      "\n",
      "LYS  22 A   9.53    44 %   -0.97  405   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 GLU  27 A\n",
      "\n",
      "LYS  32 A  10.25     0 %   -0.31  236   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.06 ASP  33 A\n",
      "\n",
      "LYS  39 A  13.17    62 %   -1.79  454   0.00    0    0.34 GLU  72 A    0.00 XXX   0 X    0.28 ASP  41 A\n",
      "LYS  39 A                                            0.85 GLU  73 A    0.00 XXX   0 X    0.30 ASP  49 A\n",
      "LYS  39 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.28 ASP 142 A\n",
      "LYS  39 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 294 A\n",
      "LYS  39 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 324 A\n",
      "LYS  39 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.29 TYR  50 A\n",
      "LYS  39 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.10 GLU  72 A\n",
      "LYS  39 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.00 GLU  73 A\n",
      "\n",
      "LYS  61 A   9.38    37 %   -1.38  385   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.11 ASP  66 A\n",
      "LYS  61 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP  67 A\n",
      "\n",
      "LYS  68 A  10.47     6 %   -0.46  298   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP  67 A\n",
      "LYS  68 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 ARG 137 A\n",
      "LYS  68 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 165 A\n",
      "LYS  68 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.36 TYR 130 A\n",
      "\n",
      "LYS  91 A  11.20    36 %   -0.82  382   0.00    0    0.85 ASP  94 A    0.00 XXX   0 X    0.67 ASP  94 A\n",
      "\n",
      "LYS  99 A  11.34     0 %   -0.22  141   0.00    0    0.68 GLU  95 A    0.00 XXX   0 X   -0.00 ARG  23 A\n",
      "LYS  99 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 GLU  95 A\n",
      "\n",
      "LYS 102 A  10.42     0 %   -0.12  190   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 104 A\n",
      "\n",
      "LYS 103 A  11.30     6 %   -0.45  298   0.00    0    0.85 GLU 215 A    0.00 XXX   0 X   -0.09 ARG  15 A\n",
      "LYS 103 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP 104 A\n",
      "LYS 103 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 105 A\n",
      "LYS 103 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 199 A\n",
      "LYS 103 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.41 GLU 215 A\n",
      "\n",
      "LYS 116 A  10.23     0 %   -0.12  213   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP  94 A\n",
      "LYS 116 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 120 A\n",
      "LYS 116 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.18 LYS  91 A\n",
      "\n",
      "LYS 135 A  10.54     0 %   -0.04   82   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP 165 A\n",
      "\n",
      "LYS 148 A  10.50    12 %   -0.36  314   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP  41 A\n",
      "LYS 148 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.05 ARG 152 A\n",
      "LYS 148 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.35 GLU 238 A\n",
      "LYS 148 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 263 A\n",
      "\n",
      "LYS 189 A  10.46     0 %   -0.04  110   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 219 A  10.37     0 %   -0.19  220   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 304 A\n",
      "\n",
      "LYS 226 A  10.46     0 %   -0.04   96   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 229 A  10.85     0 %   -0.10  111   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.24 ASP 227 A\n",
      "LYS 229 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 ASP 230 A\n",
      "\n",
      "LYS 239 A  10.51     0 %   -0.08  150   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP 245 A\n",
      "\n",
      "LYS 251 A  10.84    33 %   -1.16  374   0.00    0    0.85 GLU 235 A    0.00 XXX   0 X    0.65 GLU 235 A\n",
      "\n",
      "LYS 253 A  10.54     0 %   -0.21  219   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.25 ASP 350 A\n",
      "\n",
      "LYS 258 A  10.36     0 %   -0.17  226   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.10 ARG 223 A\n",
      "LYS 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP 230 A\n",
      "LYS 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 277 A\n",
      "\n",
      "LYS 267 A   7.64   100 %   -3.32  608   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.00 ARG 152 A\n",
      "LYS 267 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 CYS 159 A\n",
      "LYS 267 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.50 ASP 263 A\n",
      "LYS 267 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.79 ARG 281 A\n",
      "LYS 267 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 GLU 290 A\n",
      "LYS 267 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP 294 A\n",
      "LYS 267 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.65 ASP 145 A\n",
      "\n",
      "LYS 285 A  10.40     0 %   -0.10  190   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 291 A  10.01    47 %   -1.26  413   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.12 ASP  41 A\n",
      "LYS 291 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 GLU  72 A\n",
      "LYS 291 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 ARG 281 A\n",
      "LYS 291 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.23 GLU 290 A\n",
      "LYS 291 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.35 ASP 294 A\n",
      "LYS 291 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.05 LYS  39 A\n",
      "\n",
      "LYS 329 A   9.59    18 %   -0.97  331   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU 290 A\n",
      "LYS 329 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 314 A\n",
      "LYS 329 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS 291 A\n",
      "\n",
      "ARG   5 A  12.07    10 %   -0.57  309   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.12 GLU   6 A\n",
      "ARG   5 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP  67 A\n",
      "ARG   5 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 105 A\n",
      "ARG   5 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 ARG 137 A\n",
      "\n",
      "ARG  15 A  12.14    68 %   -2.12  473   0.00    0    0.35 TYR  97 A    0.00 XXX   0 X    1.07 TYR  97 A\n",
      "ARG  15 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 TYR 101 A\n",
      "ARG  15 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP 213 A\n",
      "ARG  15 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 GLU 215 A\n",
      "\n",
      "ARG  23 A  11.87    37 %   -0.97  384   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.09 GLU  27 A\n",
      "ARG  23 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 GLU  89 A\n",
      "ARG  23 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP  94 A\n",
      "ARG  23 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU  95 A\n",
      "\n",
      "ARG 122 A  13.16     0 %   -0.45  261   0.00    0    0.70 ASP  80 A    0.00 XXX   0 X    0.10 GLU 119 A\n",
      "ARG 122 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 ASP 126 A\n",
      "ARG 122 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.26 ASP  80 A\n",
      "\n",
      "ARG 134 A  14.14     0 %   -0.33  237   0.00    0    1.58 ASP 319 A    0.00 XXX   0 X    0.00 ASP 165 A\n",
      "ARG 134 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 ASP 319 A\n",
      "\n",
      "ARG 137 A  13.63    35 %   -1.22  380   0.00    0    0.65 GLU   6 A    0.00 XXX   0 X    0.43 GLU   6 A\n",
      "ARG 137 A                                            0.70 ASP  67 A    0.00 XXX   0 X    0.57 ASP  67 A\n",
      "\n",
      "ARG 152 A  15.85    18 %   -0.87  332   0.00    0    0.72 GLU 154 A    0.00 XXX   0 X    0.31 GLU 154 A\n",
      "ARG 152 A                                            0.74 GLU 238 A    0.00 XXX   0 X    0.33 GLU 238 A\n",
      "ARG 152 A                                            1.55 ASP 263 A    0.00 XXX   0 X    0.57 ASP 263 A\n",
      "\n",
      "ARG 199 A  10.80    65 %   -2.25  462   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.50 CYS 172 A\n",
      "ARG 199 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 215 A\n",
      "\n",
      "ARG 202 A  11.55    41 %   -1.14  396   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.19 GLU 304 A\n",
      "\n",
      "ARG 223 A  13.63    34 %   -1.14  377   0.00    0    1.64 ASP 277 A    0.00 XXX   0 X    0.03 ASP 230 A\n",
      "ARG 223 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.61 ASP 277 A\n",
      "\n",
      "ARG 281 A  17.71   100 %   -2.15  598   0.00    0    0.51 TYR 279 A    0.00 XXX   0 X    0.54 TYR 268 A\n",
      "ARG 281 A                                            0.59 GLU 290 A    0.00 XXX   0 X    1.75 TYR 279 A\n",
      "ARG 281 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.51 TYR 288 A\n",
      "ARG 281 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.55 ASP 145 A\n",
      "ARG 281 A                                            0.00 XXX   0 X    0.00 XXX   0 X    2.03 GLU 290 A\n",
      "ARG 281 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.87 ASP 294 A\n",
      "\n",
      "ARG 324 A  11.59    54 %   -1.52  433   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU  72 A\n",
      "ARG 324 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU  73 A\n",
      "ARG 324 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.41 GLU 131 A\n",
      "ARG 324 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 CYS 140 A\n",
      "\n",
      "ARG 338 A  12.69     0 %   -0.07  146   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 315 A\n",
      "ARG 338 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 GLU 339 A\n",
      "ARG 338 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP 342 A\n",
      "\n",
      "ARG 355 A  14.03     6 %   -0.58  299   0.00    0    1.62 GLU 246 A    0.00 XXX   0 X    0.07 C-  355 A\n",
      "ARG 355 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.43 GLU 246 A\n",
      "\n",
      "N+    2 A   8.88     0 %   -0.32  186   0.00    0    0.77 GLU   4 A    0.00 XXX   0 X    0.05 GLU   6 A\n",
      "N+    2 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 GLU   4 A\n",
      "\n",
      "Coupled residues (marked *) were detected.Please rerun PropKa with the --display-coupled-residues \n",
      "or -d option for detailed information.\n",
      "\n",
      "INFO:pKa summary:\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "SUMMARY OF THIS PREDICTION\n",
      "       Group      pKa  model-pKa   ligand atom-type\n",
      "   ASP  33 A     3.91       3.80                      \n",
      "   ASP  41 A     2.21       3.80                      \n",
      "   ASP  49 A     6.17       3.80                      \n",
      "   ASP  66 A     3.97       3.80                      \n",
      "   ASP  67 A     3.50       3.80                      \n",
      "   ASP  80 A     3.08       3.80                      \n",
      "   ASP  94 A     2.91       3.80                      \n",
      "   ASP 104 A     3.88       3.80                      \n",
      "   ASP 105 A     4.10       3.80                      \n",
      "   ASP 111 A     3.81       3.80                      \n",
      "   ASP 120 A     2.75       3.80                      \n",
      "   ASP 126 A     3.39       3.80                      \n",
      "   ASP 142 A    16.43       3.80                      \n",
      "   ASP 145 A     3.62       3.80                      \n",
      "   ASP 165 A     3.15       3.80                      \n",
      "   ASP 187 A     2.15       3.80                      \n",
      "   ASP 213 A     3.05       3.80                      \n",
      "   ASP 227 A     3.07       3.80                      \n",
      "   ASP 230 A     3.96       3.80                      \n",
      "   ASP 245 A     3.84       3.80                      \n",
      "   ASP 263 A     3.38       3.80                      \n",
      "   ASP 277 A     3.47       3.80                      \n",
      "   ASP 294 A     9.23       3.80                      \n",
      "   ASP 311 A     3.81       3.80                      \n",
      "   ASP 315 A     6.34       3.80                      \n",
      "   ASP 319 A     1.83       3.80                      \n",
      "   ASP 342 A     3.08       3.80                      \n",
      "   ASP 350 A     3.76       3.80                      \n",
      "   GLU   4 A     2.92       4.50                      \n",
      "   GLU   6 A     3.84       4.50                      \n",
      "   GLU  27 A     4.11       4.50                      \n",
      "   GLU  72 A     5.47       4.50                      \n",
      "   GLU  73 A     8.38       4.50                      \n",
      "   GLU  89 A     5.51       4.50                      \n",
      "   GLU  95 A     3.70       4.50                      \n",
      "   GLU 119 A     4.30       4.50                      \n",
      "   GLU 131 A     4.45       4.50                      \n",
      "   GLU 154 A     4.04       4.50                      \n",
      "   GLU 192 A     3.88       4.50                      \n",
      "   GLU 215 A     3.63       4.50                      \n",
      "   GLU 235 A     3.26       4.50                      \n",
      "   GLU 238 A     3.63       4.50                      \n",
      "   GLU 246 A     2.95       4.50                      \n",
      "   GLU 290 A     1.29       4.50                      \n",
      "   GLU 304 A     3.18       4.50                      \n",
      "   GLU 314 A     5.06       4.50                      \n",
      "   GLU 339 A     4.02       4.50                      \n",
      "   C-  355 A     3.34       3.20                      \n",
      "   HIS  31 A     6.33       6.50                      \n",
      "   HIS 191 A     6.82       6.50                      \n",
      "   HIS 221 A     5.75       6.50                      \n",
      "   HIS 224 A     6.57       6.50                      \n",
      "   HIS 241 A     4.98       6.50                      \n",
      "   HIS 244 A     5.80       6.50                      \n",
      "   HIS 261 A     6.60       6.50                      \n",
      "   HIS 295 A     5.60       6.50                      \n",
      "   HIS 303 A     6.22       6.50                      \n",
      "   HIS 309 A     2.56       6.50                      \n",
      "   HIS 341 A     4.22       6.50                      \n",
      "   CYS 140 A    12.78       9.00                      \n",
      "   CYS 159 A    15.19       9.00                      \n",
      "   CYS 172 A    12.08       9.00                      \n",
      "   CYS 175 A     9.09       9.00                      \n",
      "   CYS 269 A    19.48       9.00                      \n",
      "   CYS 349 A     9.57       9.00                      \n",
      "   TYR  50 A    12.47      10.00                      \n",
      "   TYR  97 A    12.15      10.00                      \n",
      "   TYR 101 A    10.24      10.00                      \n",
      "   TYR 130 A    10.10      10.00                      \n",
      "   TYR 183 A    11.75      10.00                      \n",
      "   TYR 196 A    13.17      10.00                      \n",
      "   TYR 208 A    11.41      10.00                      \n",
      "   TYR 268 A    17.93      10.00                      \n",
      "   TYR 279 A    12.08      10.00                      \n",
      "   TYR 288 A    12.91      10.00                      \n",
      "   LYS  16 A     9.20      10.50                      \n",
      "   LYS  22 A     9.53      10.50                      \n",
      "   LYS  32 A    10.25      10.50                      \n",
      "   LYS  39 A    13.17      10.50                      \n",
      "   LYS  61 A     9.38      10.50                      \n",
      "   LYS  68 A    10.47      10.50                      \n",
      "   LYS  91 A    11.20      10.50                      \n",
      "   LYS  99 A    11.34      10.50                      \n",
      "   LYS 102 A    10.42      10.50                      \n",
      "   LYS 103 A    11.30      10.50                      \n",
      "   LYS 116 A    10.23      10.50                      \n",
      "   LYS 135 A    10.54      10.50                      \n",
      "   LYS 148 A    10.50      10.50                      \n",
      "   LYS 189 A    10.46      10.50                      \n",
      "   LYS 219 A    10.37      10.50                      \n",
      "   LYS 226 A    10.46      10.50                      \n",
      "   LYS 229 A    10.85      10.50                      \n",
      "   LYS 239 A    10.51      10.50                      \n",
      "   LYS 251 A    10.84      10.50                      \n",
      "   LYS 253 A    10.54      10.50                      \n",
      "   LYS 258 A    10.36      10.50                      \n",
      "   LYS 267 A     7.64      10.50                      \n",
      "   LYS 285 A    10.40      10.50                      \n",
      "   LYS 291 A    10.01      10.50                      \n",
      "   LYS 329 A     9.59      10.50                      \n",
      "   ARG   5 A    12.07      12.50                      \n",
      "   ARG  15 A    12.14      12.50                      \n",
      "   ARG  23 A    11.87      12.50                      \n",
      "   ARG 122 A    13.16      12.50                      \n",
      "   ARG 134 A    14.14      12.50                      \n",
      "   ARG 137 A    13.63      12.50                      \n",
      "   ARG 152 A    15.85      12.50                      \n",
      "   ARG 199 A    10.80      12.50                      \n",
      "   ARG 202 A    11.55      12.50                      \n",
      "   ARG 223 A    13.63      12.50                      \n",
      "   ARG 281 A    17.71      12.50                      \n",
      "   ARG 324 A    11.59      12.50                      \n",
      "   ARG 338 A    12.69      12.50                      \n",
      "   ARG 355 A    14.03      12.50                      \n",
      "   N+    2 A     8.88       8.00                      \n",
      "\n",
      "INFO:PROPKA information:\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "Free energy of   folding (kcal/mol) as a function of pH (using neutral reference)\n",
      "  0.00     44.66\n",
      "  1.00     44.24\n",
      "  2.00     41.82\n",
      "  3.00     33.82\n",
      "  4.00     21.09\n",
      "  5.00     16.07\n",
      "  6.00     17.12\n",
      "  7.00     18.86\n",
      "  8.00     22.13\n",
      "  9.00     26.07\n",
      " 10.00     32.15\n",
      " 11.00     44.82\n",
      " 12.00     59.71\n",
      " 13.00     73.20\n",
      " 14.00     85.66\n",
      "\n",
      "The pH of optimum stability is  5.0 for which the free energy is   16.1 kcal/mol at 298K\n",
      "Could not determine pH values where the free energy is within 80 % of minimum\n",
      "Could not determine the pH-range where the free energy is negative\n",
      "\n",
      "\n",
      "Protein charge of folded and unfolded state as a function of pH\n",
      "    pH  unfolded  folded\n",
      "  0.00     50.99   50.91\n",
      "  0.10     50.99   50.88\n",
      "  0.20     50.99   50.85\n",
      "  0.30     50.99   50.82\n",
      "  0.40     50.99   50.77\n",
      "  0.50     50.98   50.72\n",
      "  0.60     50.98   50.65\n",
      "  0.70     50.97   50.57\n",
      "  0.80     50.96   50.48\n",
      "  0.90     50.96   50.36\n",
      "  1.00     50.94   50.23\n",
      "  1.10     50.93   50.07\n",
      "  1.20     50.91   49.89\n",
      "  1.30     50.89   49.68\n",
      "  1.40     50.86   49.44\n",
      "  1.50     50.82   49.16\n",
      "  1.60     50.78   48.85\n",
      "  1.70     50.72   48.49\n",
      "  1.80     50.65   48.09\n",
      "  1.90     50.56   47.64\n",
      "  2.00     50.44   47.13\n",
      "  2.10     50.30   46.56\n",
      "  2.20     50.13   45.91\n",
      "  2.30     49.91   45.20\n",
      "  2.40     49.64   44.39\n",
      "  2.50     49.31   43.50\n",
      "  2.60     48.90   42.51\n",
      "  2.70     48.40   41.41\n",
      "  2.80     47.80   40.22\n",
      "  2.90     47.07   38.92\n",
      "  3.00     46.20   37.51\n",
      "  3.10     45.17   36.02\n",
      "  3.20     43.97   34.43\n",
      "  3.30     42.58   32.78\n",
      "  3.40     41.01   31.06\n",
      "  3.50     39.25   29.30\n",
      "  3.60     37.31   27.52\n",
      "  3.70     35.23   25.73\n",
      "  3.80     33.02   23.96\n",
      "  3.90     30.72   22.23\n",
      "  4.00     28.37   20.55\n",
      "  4.10     26.01   18.95\n",
      "  4.20     23.67   17.45\n",
      "  4.30     21.38   16.04\n",
      "  4.40     19.18   14.74\n",
      "  4.50     17.10   13.54\n",
      "  4.60     15.14   12.44\n",
      "  4.70     13.34   11.43\n",
      "  4.80     11.70   10.50\n",
      "  4.90     10.22    9.64\n",
      "  5.00      8.90    8.83\n",
      "  5.10      7.74    8.07\n",
      "  5.20      6.71    7.33\n",
      "  5.30      5.81    6.62\n",
      "  5.40      5.00    5.93\n",
      "  5.50      4.27    5.25\n",
      "  5.60      3.60    4.57\n",
      "  5.70      2.97    3.90\n",
      "  5.80      2.35    3.24\n",
      "  5.90      1.73    2.58\n",
      "  6.00      1.10    1.93\n",
      "  6.10      0.45    1.29\n",
      "  6.20     -0.22    0.66\n",
      "  6.30     -0.91    0.05\n",
      "  6.40     -1.61   -0.54\n",
      "  6.50     -2.31   -1.11\n",
      "  6.60     -3.01   -1.64\n",
      "  6.70     -3.68   -2.15\n",
      "  6.80     -4.31   -2.62\n",
      "  6.90     -4.91   -3.05\n",
      "  7.00     -5.45   -3.45\n",
      "  7.10     -5.94   -3.81\n",
      "  7.20     -6.38   -4.14\n",
      "  7.30     -6.78   -4.45\n",
      "  7.40     -7.13   -4.74\n",
      "  7.50     -7.46   -5.01\n",
      "  7.60     -7.76   -5.26\n",
      "  7.70     -8.04   -5.51\n",
      "  7.80     -8.32   -5.76\n",
      "  7.90     -8.60   -6.00\n",
      "  8.00     -8.88   -6.26\n",
      "  8.10     -9.18   -6.52\n",
      "  8.20     -9.50   -6.80\n",
      "  8.30     -9.84   -7.11\n",
      "  8.40    -10.22   -7.43\n",
      "  8.50    -10.65   -7.79\n",
      "  8.60    -11.11   -8.19\n",
      "  8.70    -11.64   -8.63\n",
      "  8.80    -12.21   -9.11\n",
      "  8.90    -12.85   -9.64\n",
      "  9.00    -13.55  -10.22\n",
      "  9.10    -14.32  -10.85\n",
      "  9.20    -15.17  -11.54\n",
      "  9.30    -16.09  -12.29\n",
      "  9.40    -17.10  -13.08\n",
      "  9.50    -18.21  -13.94\n",
      "  9.60    -19.42  -14.85\n",
      "  9.70    -20.76  -15.82\n",
      "  9.80    -22.21  -16.86\n",
      "  9.90    -23.79  -17.95\n",
      " 10.00    -25.49  -19.10\n",
      " 10.10    -27.30  -20.31\n",
      " 10.20    -29.18  -21.57\n",
      " 10.30    -31.13  -22.86\n",
      " 10.40    -33.09  -24.19\n",
      " 10.50    -35.05  -25.53\n",
      " 10.60    -36.95  -26.87\n",
      " 10.70    -38.76  -28.19\n",
      " 10.80    -40.46  -29.49\n",
      " 10.90    -42.03  -30.76\n",
      " 11.00    -43.45  -31.99\n",
      " 11.10    -44.73  -33.18\n",
      " 11.20    -45.88  -34.33\n",
      " 11.30    -46.90  -35.46\n",
      " 11.40    -47.83  -36.54\n",
      " 11.50    -48.67  -37.61\n",
      " 11.60    -49.47  -38.64\n",
      " 11.70    -50.22  -39.65\n",
      " 11.80    -50.97  -40.64\n",
      " 11.90    -51.72  -41.60\n",
      " 12.00    -52.49  -42.54\n",
      " 12.10    -53.29  -43.45\n",
      " 12.20    -54.12  -44.34\n",
      " 12.30    -54.97  -45.20\n",
      " 12.40    -55.84  -46.03\n",
      " 12.50    -56.72  -46.84\n",
      " 12.60    -57.58  -47.62\n",
      " 12.70    -58.41  -48.37\n",
      " 12.80    -59.18  -49.10\n",
      " 12.90    -59.90  -49.80\n",
      " 13.00    -60.55  -50.48\n",
      " 13.10    -61.12  -51.13\n",
      " 13.20    -61.61  -51.75\n",
      " 13.30    -62.04  -52.35\n",
      " 13.40    -62.40  -52.91\n",
      " 13.50    -62.70  -53.45\n",
      " 13.60    -62.95  -53.96\n",
      " 13.70    -63.15  -54.44\n",
      " 13.80    -63.32  -54.89\n",
      " 13.90    -63.45  -55.30\n",
      " 14.00    -63.56  -55.69\n",
      "The pI is  6.31 (folded) and  6.17 (unfolded)\n",
      "\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "INFO:Applying pKa values at a pH of 7.00:\n",
      "INFO:Adding hydrogens to biomolecule.\n",
      "INFO:Debumping biomolecule (again).\n",
      "INFO:Optimizing hydrogen bonds\n",
      "INFO:Applying force field to biomolecule states.\n",
      "INFO:Regenerating headers.\n",
      "INFO:Regenerating PDB lines.\n",
      "WARNING:Ignoring 466 header lines in output.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "pH: 7.0\tname: 1ka1_hal2\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:PDB2PQR v3.6.2: biomolecular structure conversion software.\n",
      "INFO:Please cite:  Jurrus E, et al.  Improvements to the APBS biomolecular solvation software suite.  Protein Sci 27 112-128 (2018).\n",
      "INFO:Please cite:  Dolinsky TJ, et al.  PDB2PQR: expanding and upgrading automated preparation of biomolecular structures for molecular simulations. Nucleic Acids Res 35 W522-W525 (2007).\n",
      "INFO:Checking and transforming input arguments.\n",
      "INFO:Loading topology files.\n",
      "INFO:Loading molecule: /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/AtSAL1_updated.pdb\n",
      "INFO:Setting up molecule.\n",
      "INFO:Created biomolecule object with 352 residues and 2628 atoms.\n",
      "INFO:Setting termini states for biomolecule chains.\n",
      "INFO:Loading forcefield.\n",
      "INFO:Loading hydrogen topology definitions.\n",
      "WARNING:Missing atom OXT in residue ALA A 352\n",
      "WARNING:Missing atom OXT in residue ALA A 352\n",
      "INFO:Attempting to repair 1 missing atoms in biomolecule.\n",
      "WARNING:Missing atom OXT in residue ALA A 352\n",
      "INFO:Added atom OXT to residue ALA A 352 at coordinates -4.698, -13.324, -18.987\n",
      "INFO:Updating disulfide bridges.\n",
      "INFO:Debumping biomolecule.\n",
      "INFO:Assigning titration states with PROPKA.\n",
      "INFO:\n",
      "propka3.5.1                                                                                  2024-07-26\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "--                                                                           --\n",
      "--  PROPKA: A PROTEIN PKA PREDICTOR                                          --\n",
      "--                                                                           --\n",
      "--  VERSION 1.0,  04/25/2004,  IOWA CITY                                     --\n",
      "--  BY HUI LI                                                                --\n",
      "--                                                                           --\n",
      "--  VERSION 2.0,  11/05/2007, IOWA CITY/COPENHAGEN                           --\n",
      "--  BY DELPHINE C. BAS AND DAVID M. ROGERS                                   --\n",
      "--                                                                           --\n",
      "--  VERSION 3.0,  01/06/2011, COPENHAGEN                                     --\n",
      "--  BY MATS H.M. OLSSON AND CHRESTEN R. SONDERGARD                           --\n",
      "--                                                                           --\n",
      "--  VERSION 3.1,  07/01/2011, COPENHAGEN                                     --\n",
      "--  BY CHRESTEN R. SONDERGARD AND MATS H.M. OLSSON                           --\n",
      "--                                                                           --\n",
      "--  VERSION 3.2,  06/17/2020, PLANET EARTH                                   --\n",
      "--  SEE ABOVE FOR AUTHORS                                                    --\n",
      "--                                                                           --\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "References:\n",
      "\n",
      "Very Fast Empirical Prediction and Rationalization of Protein pKa Values.\n",
      "Hui Li, Andrew D. Robertson and Jan H. Jensen. PROTEINS: Structure, Function,\n",
      "and Bioinformatics. 61:704-721 (2005)\n",
      "\n",
      "Very Fast Prediction and Rationalization of pKa Values for Protein-Ligand\n",
      "Complexes.  Delphine C. Bas, David M. Rogers and Jan H. Jensen.  PROTEINS:\n",
      "Structure, Function, and Bioinformatics 73:765-783 (2008)\n",
      "\n",
      "PROPKA3: Consistent Treatment of Internal and Surface Residues in Empirical\n",
      "pKa predictions.  Mats H.M. Olsson, Chresten R. Sondergard, Michal Rostkowski,\n",
      "and Jan H. Jensen.  Journal of Chemical Theory and Computation, 7(2):525-537\n",
      "(2011)\n",
      "\n",
      "Improved Treatment of Ligands and Coupling Effects in Empirical Calculation\n",
      "and Rationalization of pKa Values.  Chresten R. Sondergaard, Mats H.M. Olsson,\n",
      "Michal Rostkowski, and Jan H. Jensen.  Journal of Chemical Theory and\n",
      "Computation, (2011)\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "\n",
      "INFO:Coupling map:\n",
      "\n",
      "INFO:Calculating pKas for Conformation container 1A with 3086 atoms and 869 groups\n",
      "INFO:did not converge in 10 iterations\n",
      "INFO:pKa determinants:\n",
      "\n",
      "---------  -----   ------   ---------------------    --------------    --------------    --------------\n",
      "                            DESOLVATION  EFFECTS       SIDECHAIN          BACKBONE        COULOMBIC\n",
      " RESIDUE    pKa    BURIED     REGULAR      RE        HYDROGEN BOND     HYDROGEN BOND      INTERACTION\n",
      "---------  -----   ------   ---------   ---------    --------------    --------------    --------------\n",
      "\n",
      "ASP   8 A   3.99    18 %    0.58  331   0.18    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.03 N+    1 A\n",
      "ASP   8 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.15 LYS   5 A\n",
      "ASP   8 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.30 LYS  11 A\n",
      "ASP   8 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.12 LYS  12 A\n",
      "ASP   8 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU   6 A\n",
      "\n",
      "ASP  32 A   3.90     0 %    0.12  137   0.00    0    0.00 XXX   0 X   -0.02 VAL  33 A    0.00 XXX   0 X\n",
      "\n",
      "ASP  38 A   3.69     0 %    0.12  180   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.24 LYS  39 A\n",
      "\n",
      "ASP  46 A   7.13    91 %    2.32  536   0.55    0    0.00 XXX   0 X    0.00 XXX   0 X    0.25 GLU  71 A\n",
      "ASP  46 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 GLU  72 A\n",
      "ASP  46 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 137 A\n",
      "\n",
      "ASP  73 A   3.49     0 %    0.30  267   0.00    0    0.00 XXX   0 X   -0.69 GLY  75 A   -0.06 ARG  78 A\n",
      "ASP  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS 317 A\n",
      "ASP  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 ASP 118 A\n",
      "\n",
      "ASP  76 A   4.20     0 %    0.23  268   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.00 LYS  79 A\n",
      "ASP  76 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.14 ASP  73 A\n",
      "ASP  76 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP  80 A\n",
      "\n",
      "ASP  80 A   3.82     0 %    0.31  184   0.00    0   -0.20 SER  82 A   -0.22 ASP  80 A    0.03 ASP  38 A\n",
      "ASP  80 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP  84 A\n",
      "\n",
      "ASP  84 A   3.48     0 %    0.35  226   0.00    0    0.00 XXX   0 X   -0.06 ASP  84 A   -0.25 ARG  88 A\n",
      "ASP  84 A                                            0.00 XXX   0 X   -0.35 THR  85 A    0.00 XXX   0 X\n",
      "\n",
      "ASP  95 A   3.71     0 %    0.32  279   0.00    0    0.00 XXX   0 X   -0.01 ASP  95 A   -0.28 LYS  23 A\n",
      "ASP  95 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.12 LYS  91 A\n",
      "\n",
      "ASP 111 A   3.94     0 %    0.21  211   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.03 ARG  78 A\n",
      "ASP 111 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.14 ARG 115 A\n",
      "ASP 111 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP 112 A\n",
      "\n",
      "ASP 112 A   2.85     0 %    0.34  258   0.00    0   -0.49 SER 109 A   -0.62 SER 109 A   -0.18 ARG 115 A\n",
      "\n",
      "ASP 118 A   3.20    10 %    0.65  308   0.06    0   -0.77 SER  74 A   -0.25 SER  74 A   -0.26 ARG  78 A\n",
      "ASP 118 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 115 A\n",
      "\n",
      "ASP 134 A  13.40*  100 %    3.03  569   0.41    0    1.52 ASP 137 A   -0.26 ILE 136 A   -0.01 LYS 285 A\n",
      "ASP 134 A                                            0.96 ASP 288 A    0.00 XXX   0 X    0.29 GLU  71 A\n",
      "ASP 134 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.68 ASP  46 A\n",
      "ASP 134 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.89 ASP 137 A\n",
      "ASP 134 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.08 ASP 288 A\n",
      "\n",
      "ASP 137 A   2.84*  100 %    2.61  581   0.29    0   -1.52 ASP 134 A   -0.00 GLN 260 A   -0.47 LYS 262 A\n",
      "ASP 137 A                                           -1.46 ASP 288 A    0.00 XXX   0 X    0.05 GLU  71 A\n",
      "ASP 137 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 GLU 284 A\n",
      "ASP 137 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.65 ARG 276 A\n",
      "\n",
      "ASP 146 A   3.60    12 %    0.79  316   0.11    0   -0.79 ARG 144 A    0.00 XXX   0 X   -0.31 ARG 144 A\n",
      "\n",
      "ASP 186 A   3.58     0 %    0.15  116   0.00    0   -0.37 SER 184 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "ASP 206 A   3.75    53 %    1.44  431   0.23    0   -1.15 ARG  19 A    0.00 XXX   0 X   -0.59 ARG  19 A\n",
      "\n",
      "ASP 239 A   3.65     0 %    0.43  269   0.00    0    0.00 XXX   0 X   -0.34 ASP 239 A   -0.04 HIS 235 A\n",
      "ASP 239 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.17 HIS 238 A\n",
      "ASP 239 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 HIS 279 A\n",
      "\n",
      "ASP 258 A   6.14    44 %    1.75  404   0.62    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.00 LYS 140 A\n",
      "ASP 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.34 ARG 144 A\n",
      "ASP 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.05 LYS 262 A\n",
      "ASP 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.34 ASP 146 A\n",
      "ASP 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 232 A\n",
      "\n",
      "ASP 270 A   5.41    76 %    2.89  493   0.45    0   -0.11 SER 174 A   -0.21 ILE 175 A   -0.89 ARG 268 A\n",
      "ASP 270 A                                           -0.51 ARG 268 A   -0.01 ASP 270 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 288 A   7.25*   81 %    2.27  509   0.16    0   -0.96 ASP 134 A    0.00 XXX   0 X   -0.13 LYS 262 A\n",
      "ASP 288 A                                            1.46 ASP 137 A    0.00 XXX   0 X   -0.22 LYS 285 A\n",
      "ASP 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.62 ASP 137 A\n",
      "ASP 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.54 ARG 276 A\n",
      "ASP 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.54 GLU 284 A\n",
      "ASP 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.75 HIS 289 A\n",
      "\n",
      "ASP 305 A   4.25    25 %    1.48  352   0.32    0    0.00 XXX   0 X   -0.17 ASP 305 A   -0.12 LYS 309 A\n",
      "ASP 305 A                                            0.00 XXX   0 X   -0.07 ALA 307 A   -0.04 ARG 341 A\n",
      "ASP 305 A                                            0.00 XXX   0 X   -0.00 GLY 308 A    0.02 ASP 320 A\n",
      "ASP 305 A                                            0.00 XXX   0 X   -0.63 LYS 309 A    0.23 ASP 322 A\n",
      "ASP 305 A                                            0.00 XXX   0 X   -0.58 ASP 322 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 312 A   2.16    11 %    0.68  312   0.01    0   -0.85 SER 314 A   -0.15 SER 314 A   -0.10 LYS 159 A\n",
      "ASP 312 A                                           -0.85 LYS 315 A    0.00 XXX   0 X   -0.38 LYS 315 A\n",
      "\n",
      "ASP 320 A   3.88     0 %    0.22  254   0.00    0    0.00 XXX   0 X   -0.12 ASP 320 A   -0.13 LYS 315 A\n",
      "ASP 320 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.11 ASP 312 A\n",
      "\n",
      "ASP 322 A   3.54    11 %    0.50  311   0.00    0   -0.66 THR 323 A    0.00 XXX   0 X   -0.01 LYS 309 A\n",
      "ASP 322 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.10 ARG 341 A\n",
      "\n",
      "ASP 342 A   4.00     0 %    0.37  242   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.04 LYS 247 A\n",
      "ASP 342 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.05 LYS 338 A\n",
      "ASP 342 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 ARG 341 A\n",
      "\n",
      "GLU   4 A   4.37     0 %    0.15  174   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.19 N+    1 A\n",
      "GLU   4 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.24 LYS   5 A\n",
      "GLU   4 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU   6 A\n",
      "GLU   4 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 ASP   8 A\n",
      "\n",
      "GLU   6 A   3.70    28 %    1.00  361   0.00    0   -0.72 ARG 129 A   -0.65 GLU   6 A   -0.06 LYS   5 A\n",
      "GLU   6 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.37 ARG 129 A\n",
      "\n",
      "GLU  58 A   4.99    32 %    0.70  370   0.10    0   -0.31 CYS 119 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "GLU  60 A   4.70    11 %    0.55  313   0.02    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.40 LYS  12 A\n",
      "GLU  60 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS  59 A\n",
      "GLU  60 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP   8 A\n",
      "GLU  60 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 101 A\n",
      "\n",
      "GLU  64 A   4.73     0 %    0.31  278   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.32 ARG 129 A\n",
      "GLU  64 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 GLU   6 A\n",
      "GLU  64 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 HIS 130 A\n",
      "GLU  64 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 GLU 156 A\n",
      "GLU  64 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 157 A\n",
      "\n",
      "GLU  71 A   5.17    41 %    0.73  396   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.20 LYS 285 A\n",
      "GLU  71 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 317 A\n",
      "GLU  71 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 ASP 288 A\n",
      "\n",
      "GLU  72 A   5.72    30 %    0.59  365   0.07    0   -0.15 TYR  47 A    0.00 XXX   0 X   -0.01 LYS 317 A\n",
      "GLU  72 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.26 GLU  71 A\n",
      "GLU  72 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 ASP  73 A\n",
      "GLU  72 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.28 ASP  76 A\n",
      "\n",
      "GLU  87 A   4.66     0 %    0.21  201   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.08 LYS  91 A\n",
      "GLU  87 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP  84 A\n",
      "\n",
      "GLU 100 A   4.92     0 %    0.40  210   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 101 A\n",
      "\n",
      "GLU 101 A   4.59     0 %    0.15  174   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS  12 A\n",
      "\n",
      "GLU 123 A   4.29     1 %    0.40  284   0.01    0   -0.17 LYS 317 A    0.00 XXX   0 X   -0.45 LYS 317 A\n",
      "\n",
      "GLU 156 A   4.19    14 %    0.82  321   0.09    0   -0.82 ARG 129 A    0.00 XXX   0 X    0.09 GLU   6 A\n",
      "GLU 156 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 HIS 130 A\n",
      "GLU 156 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.45 ARG 129 A\n",
      "\n",
      "GLU 157 A   4.41     0 %    0.30  215   0.00    0    0.00 XXX   0 X   -0.50 GLY 128 A   -0.04 ARG 129 A\n",
      "GLU 157 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.05 HIS 130 A\n",
      "GLU 157 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 GLU 156 A\n",
      "\n",
      "GLU 187 A   4.76     0 %    0.24  215   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 186 A\n",
      "\n",
      "GLU 220 A   4.46     0 %    0.22  175   0.00    0    0.00 XXX   0 X   -0.30 GLU 220 A   -0.08 LYS 331 A\n",
      "GLU 220 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.12 GLU 224 A\n",
      "\n",
      "GLU 223 A   4.77     0 %    0.51  236   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.27 LYS 251 A\n",
      "GLU 223 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 224 A\n",
      "\n",
      "GLU 224 A   3.96     0 %    0.21  180   0.00    0   -0.71 ASN 221 A    0.00 XXX   0 X   -0.04 LYS 251 A\n",
      "\n",
      "GLU 229 A   5.80    55 %    1.19  436   0.40    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.13 ARG 256 A\n",
      "GLU 229 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.09 LYS 262 A\n",
      "GLU 229 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.07 HIS 238 A\n",
      "\n",
      "GLU 232 A   4.44     0 %    0.15  188   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS 140 A\n",
      "GLU 232 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.20 ARG 144 A\n",
      "GLU 232 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.00 ARG 256 A\n",
      "GLU 232 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 146 A\n",
      "\n",
      "GLU 284 A   2.33    71 %    2.47  480   0.16    0   -0.21 ARG 276 A   -0.15 LYS 285 A   -0.19 LYS 262 A\n",
      "GLU 284 A                                           -0.11 TYR 282 A    0.00 XXX   0 X   -0.19 LYS 285 A\n",
      "GLU 284 A                                           -1.60 HIS 289 A    0.00 XXX   0 X   -1.06 ARG 276 A\n",
      "GLU 284 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.29 HIS 289 A\n",
      "\n",
      "GLU 298 A   3.85    28 %    1.05  360   0.21    0    0.00 XXX   0 X   -0.73 GLY 197 A   -0.09 LYS 213 A\n",
      "GLU 298 A                                            0.00 XXX   0 X   -0.33 SER 198 A    0.00 XXX   0 X\n",
      "GLU 298 A                                            0.00 XXX   0 X   -0.75 GLY 199 A    0.00 XXX   0 X\n",
      "\n",
      "GLU 330 A   4.59     0 %    0.16  232   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 331 A\n",
      "\n",
      "GLU 346 A   4.58     0 %    0.39  216   0.00    0   -0.10 GLN 347 A    0.00 XXX   0 X   -0.33 LYS 247 A\n",
      "GLU 346 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 LYS 349 A\n",
      "GLU 346 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.14 ASP 342 A\n",
      "\n",
      "GLU 348 A   4.77     0 %    0.30  273   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.09 LYS 349 A\n",
      "GLU 348 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 239 A\n",
      "GLU 348 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 322 A\n",
      "\n",
      "C-  352 A   3.34     0 %    0.14   58   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "HIS 130 A   6.21    28 %   -0.99  361   0.00    0    0.00 XXX   0 X    0.01 GLY 124 A    0.01 GLU  64 A\n",
      "HIS 130 A                                            0.00 XXX   0 X    0.43 ASN 127 A    0.04 GLU 156 A\n",
      "HIS 130 A                                            0.00 XXX   0 X    0.15 ARG 129 A    0.05 GLU 157 A\n",
      "\n",
      "HIS 235 A   6.45     0 %   -0.18  141   0.00    0    0.00 XXX   0 X    0.10 ALA 234 A    0.04 ASP 239 A\n",
      "\n",
      "HIS 238 A   6.25     0 %   -0.30  241   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.00 ARG 256 A\n",
      "HIS 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 GLU 229 A\n",
      "HIS 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.19 HIS 235 A\n",
      "HIS 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.17 ASP 239 A\n",
      "\n",
      "HIS 279 A   6.15     0 %   -0.30  228   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.07 HIS 235 A\n",
      "HIS 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 239 A\n",
      "\n",
      "HIS 289 A   5.35   100 %   -3.14  578   0.00    0    1.60 GLU 284 A    0.00 XXX   0 X   -0.40 LYS 262 A\n",
      "HIS 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.23 ARG 276 A\n",
      "HIS 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 LYS 285 A\n",
      "HIS 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.29 GLU 284 A\n",
      "HIS 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.75 ASP 288 A\n",
      "\n",
      "CYS  21 A  12.37   100 %    3.39  590   0.00    0   -0.13 SER  49 A   -0.12 CYS  21 A    0.23 ASP  46 A\n",
      "\n",
      "CYS 119 A  10.39    24 %    0.60  349   0.00    0    0.31 GLU  58 A    0.00 XXX   0 X   -0.02 ARG 115 A\n",
      "CYS 119 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 118 A\n",
      "CYS 119 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.46 GLU  58 A\n",
      "\n",
      "CYS 167 A  13.04   100 %    3.60  645   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.29 ARG 268 A\n",
      "CYS 167 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.73 CYS 190 A\n",
      "\n",
      "CYS 190 A  10.90    74 %    2.85  488   0.00    0   -0.22 ARG 268 A   -0.02 GLY 189 A   -0.84 ARG 268 A\n",
      "CYS 190 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 187 A\n",
      "CYS 190 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.11 ASP 270 A\n",
      "\n",
      "TYR   3 A  10.76     8 %    0.53  303   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.10 GLU   6 A\n",
      "TYR   3 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.19 ARG 129 A\n",
      "TYR   3 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.29 GLU 156 A\n",
      "TYR   3 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 157 A\n",
      "\n",
      "TYR  47 A  12.70    45 %    1.22  407   0.00    0    0.15 GLU  72 A    0.00 XXX   0 X    0.26 ASP  46 A\n",
      "TYR  47 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU  71 A\n",
      "TYR  47 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.14 ASP  73 A\n",
      "TYR  47 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.39 ASP  76 A\n",
      "TYR  47 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP 118 A\n",
      "TYR  47 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.51 GLU  72 A\n",
      "\n",
      "TYR 148 A  15.46    84 %    4.04  516   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    1.36 CYS  21 A\n",
      "TYR 148 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 CYS 167 A\n",
      "\n",
      "TYR 201 A  10.14     0 %    0.32  267   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 298 A\n",
      "TYR 201 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.20 LYS 213 A\n",
      "\n",
      "TYR 263 A  15.09*  100 %    3.51  633   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP 137 A\n",
      "TYR 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.83 ARG 276 A\n",
      "TYR 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.46 GLU 284 A\n",
      "TYR 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 ASP 288 A\n",
      "TYR 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.61 TYR 274 A\n",
      "\n",
      "TYR 274 A   9.17*  100 %    3.44  594   0.00    0   -0.85 LYS 262 A    0.00 XXX   0 X    0.22 ASP 137 A\n",
      "TYR 274 A                                           -0.20 ARG 276 A    0.00 XXX   0 X    0.08 GLU 229 A\n",
      "TYR 274 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.42 GLU 284 A\n",
      "TYR 274 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.12 ASP 288 A\n",
      "TYR 274 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -2.03 LYS 262 A\n",
      "TYR 274 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -2.03 ARG 276 A\n",
      "\n",
      "TYR 282 A  12.77    42 %    1.48  398   0.00    0    0.11 GLU 284 A    0.00 XXX   0 X    0.09 GLU  71 A\n",
      "TYR 282 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 134 A\n",
      "TYR 282 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP 137 A\n",
      "TYR 282 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.34 ARG 276 A\n",
      "TYR 282 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.07 ARG 283 A\n",
      "TYR 282 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 TYR 274 A\n",
      "TYR 282 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 TYR 318 A\n",
      "TYR 282 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.71 GLU 284 A\n",
      "TYR 282 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.53 ASP 288 A\n",
      "\n",
      "TYR 318 A   9.59     0 %    0.35  256   0.00    0   -0.13 ARG 283 A    0.00 XXX   0 X    0.06 GLU  71 A\n",
      "TYR 318 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 123 A\n",
      "TYR 318 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.26 LYS 285 A\n",
      "TYR 318 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.09 LYS 317 A\n",
      "TYR 318 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.38 ARG 283 A\n",
      "\n",
      "LYS   5 A  10.80     0 %   -0.15  167   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.24 GLU   4 A\n",
      "LYS   5 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 GLU   6 A\n",
      "LYS   5 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP   8 A\n",
      "\n",
      "LYS  11 A   9.47    50 %   -1.27  420   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.30 ASP   8 A\n",
      "LYS  11 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS  12 A\n",
      "\n",
      "LYS  12 A  10.57    27 %   -0.51  358   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.12 ASP   8 A\n",
      "LYS  12 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.40 GLU  60 A\n",
      "LYS  12 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 GLU 101 A\n",
      "\n",
      "LYS  23 A  10.16    17 %   -0.59  328   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.28 ASP  95 A\n",
      "LYS  23 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 LYS  91 A\n",
      "\n",
      "LYS  26 A  10.23     3 %   -0.27  291   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS  39 A  10.48     0 %   -0.26  206   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.24 ASP  38 A\n",
      "\n",
      "LYS  59 A  10.21     0 %   -0.32  249   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU  60 A\n",
      "\n",
      "LYS  79 A  10.44     0 %   -0.06   88   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP  76 A\n",
      "\n",
      "LYS  91 A  10.52     0 %   -0.17  208   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU  87 A\n",
      "LYS  91 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG  88 A\n",
      "LYS  91 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.12 ASP  95 A\n",
      "\n",
      "LYS 140 A   9.86    17 %   -0.56  329   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.08 ARG 144 A\n",
      "LYS 140 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 232 A\n",
      "LYS 140 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP 258 A\n",
      "LYS 140 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS  39 A\n",
      "\n",
      "LYS 159 A  10.42     0 %   -0.14  238   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP 312 A\n",
      "LYS 159 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 LYS 315 A\n",
      "\n",
      "LYS 180 A  10.43     0 %   -0.12   93   0.00    0    0.06 LYS 182 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 182 A  10.14     0 %   -0.31  138   0.00    0   -0.06 LYS 180 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 208 A  10.45     0 %   -0.05   83   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 213 A  10.19     9 %   -0.60  306   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.09 GLU 298 A\n",
      "LYS 213 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 TYR 201 A\n",
      "\n",
      "LYS 247 A  10.38     0 %   -0.49  262   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 342 A\n",
      "LYS 247 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.33 GLU 346 A\n",
      "\n",
      "LYS 251 A  10.60     0 %   -0.22  135   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.27 GLU 223 A\n",
      "LYS 251 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 GLU 224 A\n",
      "\n",
      "LYS 262 A   9.41   100 %   -2.90  586   0.00    0    0.85 TYR 274 A    0.00 XXX   0 X    0.47 ASP 137 A\n",
      "LYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 GLU 229 A\n",
      "LYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 258 A\n",
      "LYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -2.00 ARG 276 A\n",
      "LYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 GLU 284 A\n",
      "LYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 ASP 288 A\n",
      "LYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    2.03 TYR 274 A\n",
      "\n",
      "LYS 280 A  10.46     0 %   -0.04   81   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 285 A  10.26    17 %   -0.88  329   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.20 GLU  71 A\n",
      "LYS 285 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 134 A\n",
      "LYS 285 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.24 ARG 283 A\n",
      "LYS 285 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 GLU 284 A\n",
      "LYS 285 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.22 ASP 288 A\n",
      "LYS 285 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.26 TYR 318 A\n",
      "\n",
      "LYS 309 A  10.51     0 %   -0.11  165   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.12 ASP 305 A\n",
      "LYS 309 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 322 A\n",
      "LYS 309 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 ARG 341 A\n",
      "\n",
      "LYS 315 A  11.51     0 %   -0.35  220   0.00    0    0.85 ASP 312 A    0.00 XXX   0 X    0.13 ASP 320 A\n",
      "LYS 315 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 ASP 312 A\n",
      "\n",
      "LYS 317 A  10.13    35 %   -1.18  378   0.00    0    0.17 GLU 123 A    0.00 XXX   0 X    0.06 GLU  71 A\n",
      "LYS 317 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU  72 A\n",
      "LYS 317 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP  73 A\n",
      "LYS 317 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 TYR 318 A\n",
      "LYS 317 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.45 GLU 123 A\n",
      "\n",
      "LYS 331 A  10.52     0 %   -0.12  161   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU 220 A\n",
      "LYS 331 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 GLU 330 A\n",
      "\n",
      "LYS 338 A  10.23     0 %   -0.27  219   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.05 ARG 341 A\n",
      "LYS 338 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 342 A\n",
      "\n",
      "LYS 349 A  10.31     0 %   -0.30  147   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 346 A\n",
      "LYS 349 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 GLU 348 A\n",
      "\n",
      "ARG  19 A  13.45    31 %   -0.78  369   0.00    0    1.15 ASP 206 A    0.00 XXX   0 X    0.59 ASP 206 A\n",
      "\n",
      "ARG  78 A  12.45     0 %   -0.37  250   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.06 ASP  73 A\n",
      "ARG  78 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 111 A\n",
      "ARG  78 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.26 ASP 118 A\n",
      "ARG  78 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 ARG 115 A\n",
      "\n",
      "ARG  88 A  12.29    11 %   -0.46  311   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.25 ASP  84 A\n",
      "\n",
      "ARG 115 A  12.63     0 %   -0.24  199   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.14 ASP 111 A\n",
      "ARG 115 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.18 ASP 112 A\n",
      "ARG 115 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 118 A\n",
      "ARG 115 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 CYS 119 A\n",
      "\n",
      "ARG 129 A  14.57    24 %   -0.85  348   0.00    0    0.72 GLU   6 A    0.00 XXX   0 X    0.19 TYR   3 A\n",
      "ARG 129 A                                            0.82 GLU 156 A    0.00 XXX   0 X    0.32 GLU  64 A\n",
      "ARG 129 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 GLU 157 A\n",
      "ARG 129 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.37 GLU   6 A\n",
      "ARG 129 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.45 GLU 156 A\n",
      "\n",
      "ARG 144 A  13.46     3 %   -0.68  291   0.00    0    0.79 ASP 146 A    0.00 XXX   0 X    0.20 GLU 232 A\n",
      "ARG 144 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.34 ASP 258 A\n",
      "ARG 144 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.31 ASP 146 A\n",
      "\n",
      "ARG 256 A  12.33     0 %   -0.30  275   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.13 GLU 229 A\n",
      "ARG 256 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 GLU 232 A\n",
      "\n",
      "ARG 268 A  12.83    75 %   -2.41  490   0.00    0    0.22 CYS 190 A    0.00 XXX   0 X    0.29 CYS 167 A\n",
      "ARG 268 A                                            0.51 ASP 270 A    0.00 XXX   0 X    0.84 CYS 190 A\n",
      "ARG 268 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.89 ASP 270 A\n",
      "\n",
      "ARG 276 A  16.42    98 %   -1.95  557   0.00    0    0.20 TYR 274 A    0.00 XXX   0 X    0.83 TYR 263 A\n",
      "ARG 276 A                                            0.21 GLU 284 A    0.00 XXX   0 X    0.34 TYR 282 A\n",
      "ARG 276 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.65 ASP 137 A\n",
      "ARG 276 A                                            0.00 XXX   0 X    0.00 XXX   0 X    2.03 TYR 274 A\n",
      "ARG 276 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.06 GLU 284 A\n",
      "ARG 276 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.54 ASP 288 A\n",
      "\n",
      "ARG 283 A  12.77     0 %   -0.31  205   0.00    0    0.13 TYR 318 A    0.00 XXX   0 X    0.07 TYR 282 A\n",
      "ARG 283 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 TYR 318 A\n",
      "\n",
      "ARG 341 A  12.40     1 %   -0.32  283   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 305 A\n",
      "ARG 341 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP 322 A\n",
      "ARG 341 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP 342 A\n",
      "\n",
      "N+    1 A   7.94     0 %   -0.28  245   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.19 GLU   4 A\n",
      "N+    1 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP   8 A\n",
      "N+    1 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 LYS  11 A\n",
      "\n",
      "Coupled residues (marked *) were detected.Please rerun PropKa with the --display-coupled-residues \n",
      "or -d option for detailed information.\n",
      "\n",
      "INFO:pKa summary:\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "SUMMARY OF THIS PREDICTION\n",
      "       Group      pKa  model-pKa   ligand atom-type\n",
      "   ASP   8 A     3.99       3.80                      \n",
      "   ASP  32 A     3.90       3.80                      \n",
      "   ASP  38 A     3.69       3.80                      \n",
      "   ASP  46 A     7.13       3.80                      \n",
      "   ASP  73 A     3.49       3.80                      \n",
      "   ASP  76 A     4.20       3.80                      \n",
      "   ASP  80 A     3.82       3.80                      \n",
      "   ASP  84 A     3.48       3.80                      \n",
      "   ASP  95 A     3.71       3.80                      \n",
      "   ASP 111 A     3.94       3.80                      \n",
      "   ASP 112 A     2.85       3.80                      \n",
      "   ASP 118 A     3.20       3.80                      \n",
      "   ASP 134 A    13.40       3.80                      \n",
      "   ASP 137 A     2.84       3.80                      \n",
      "   ASP 146 A     3.60       3.80                      \n",
      "   ASP 186 A     3.58       3.80                      \n",
      "   ASP 206 A     3.75       3.80                      \n",
      "   ASP 239 A     3.65       3.80                      \n",
      "   ASP 258 A     6.14       3.80                      \n",
      "   ASP 270 A     5.41       3.80                      \n",
      "   ASP 288 A     7.25       3.80                      \n",
      "   ASP 305 A     4.25       3.80                      \n",
      "   ASP 312 A     2.16       3.80                      \n",
      "   ASP 320 A     3.88       3.80                      \n",
      "   ASP 322 A     3.54       3.80                      \n",
      "   ASP 342 A     4.00       3.80                      \n",
      "   GLU   4 A     4.37       4.50                      \n",
      "   GLU   6 A     3.70       4.50                      \n",
      "   GLU  58 A     4.99       4.50                      \n",
      "   GLU  60 A     4.70       4.50                      \n",
      "   GLU  64 A     4.73       4.50                      \n",
      "   GLU  71 A     5.17       4.50                      \n",
      "   GLU  72 A     5.72       4.50                      \n",
      "   GLU  87 A     4.66       4.50                      \n",
      "   GLU 100 A     4.92       4.50                      \n",
      "   GLU 101 A     4.59       4.50                      \n",
      "   GLU 123 A     4.29       4.50                      \n",
      "   GLU 156 A     4.19       4.50                      \n",
      "   GLU 157 A     4.41       4.50                      \n",
      "   GLU 187 A     4.76       4.50                      \n",
      "   GLU 220 A     4.46       4.50                      \n",
      "   GLU 223 A     4.77       4.50                      \n",
      "   GLU 224 A     3.96       4.50                      \n",
      "   GLU 229 A     5.80       4.50                      \n",
      "   GLU 232 A     4.44       4.50                      \n",
      "   GLU 284 A     2.33       4.50                      \n",
      "   GLU 298 A     3.85       4.50                      \n",
      "   GLU 330 A     4.59       4.50                      \n",
      "   GLU 346 A     4.58       4.50                      \n",
      "   GLU 348 A     4.77       4.50                      \n",
      "   C-  352 A     3.34       3.20                      \n",
      "   HIS 130 A     6.21       6.50                      \n",
      "   HIS 235 A     6.45       6.50                      \n",
      "   HIS 238 A     6.25       6.50                      \n",
      "   HIS 279 A     6.15       6.50                      \n",
      "   HIS 289 A     5.35       6.50                      \n",
      "   CYS  21 A    12.37       9.00                      \n",
      "   CYS 119 A    10.39       9.00                      \n",
      "   CYS 167 A    13.04       9.00                      \n",
      "   CYS 190 A    10.90       9.00                      \n",
      "   TYR   3 A    10.76      10.00                      \n",
      "   TYR  47 A    12.70      10.00                      \n",
      "   TYR 148 A    15.46      10.00                      \n",
      "   TYR 201 A    10.14      10.00                      \n",
      "   TYR 263 A    15.09      10.00                      \n",
      "   TYR 274 A     9.17      10.00                      \n",
      "   TYR 282 A    12.77      10.00                      \n",
      "   TYR 318 A     9.59      10.00                      \n",
      "   LYS   5 A    10.80      10.50                      \n",
      "   LYS  11 A     9.47      10.50                      \n",
      "   LYS  12 A    10.57      10.50                      \n",
      "   LYS  23 A    10.16      10.50                      \n",
      "   LYS  26 A    10.23      10.50                      \n",
      "   LYS  39 A    10.48      10.50                      \n",
      "   LYS  59 A    10.21      10.50                      \n",
      "   LYS  79 A    10.44      10.50                      \n",
      "   LYS  91 A    10.52      10.50                      \n",
      "   LYS 140 A     9.86      10.50                      \n",
      "   LYS 159 A    10.42      10.50                      \n",
      "   LYS 180 A    10.43      10.50                      \n",
      "   LYS 182 A    10.14      10.50                      \n",
      "   LYS 208 A    10.45      10.50                      \n",
      "   LYS 213 A    10.19      10.50                      \n",
      "   LYS 247 A    10.38      10.50                      \n",
      "   LYS 251 A    10.60      10.50                      \n",
      "   LYS 262 A     9.41      10.50                      \n",
      "   LYS 280 A    10.46      10.50                      \n",
      "   LYS 285 A    10.26      10.50                      \n",
      "   LYS 309 A    10.51      10.50                      \n",
      "   LYS 315 A    11.51      10.50                      \n",
      "   LYS 317 A    10.13      10.50                      \n",
      "   LYS 331 A    10.52      10.50                      \n",
      "   LYS 338 A    10.23      10.50                      \n",
      "   LYS 349 A    10.31      10.50                      \n",
      "   ARG  19 A    13.45      12.50                      \n",
      "   ARG  78 A    12.45      12.50                      \n",
      "   ARG  88 A    12.29      12.50                      \n",
      "   ARG 115 A    12.63      12.50                      \n",
      "   ARG 129 A    14.57      12.50                      \n",
      "   ARG 144 A    13.46      12.50                      \n",
      "   ARG 256 A    12.33      12.50                      \n",
      "   ARG 268 A    12.83      12.50                      \n",
      "   ARG 276 A    16.42      12.50                      \n",
      "   ARG 283 A    12.77      12.50                      \n",
      "   ARG 341 A    12.40      12.50                      \n",
      "   N+    1 A     7.94       8.00                      \n",
      "\n",
      "INFO:PROPKA information:\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "Free energy of   folding (kcal/mol) as a function of pH (using neutral reference)\n",
      "  0.00     29.52\n",
      "  1.00     29.44\n",
      "  2.00     28.85\n",
      "  3.00     26.51\n",
      "  4.00     24.84\n",
      "  5.00     30.16\n",
      "  6.00     36.69\n",
      "  7.00     40.13\n",
      "  8.00     42.33\n",
      "  9.00     44.66\n",
      " 10.00     48.46\n",
      " 11.00     55.32\n",
      " 12.00     64.91\n",
      " 13.00     74.71\n",
      " 14.00     82.36\n",
      "\n",
      "The pH of optimum stability is  3.8 for which the free energy is   24.7 kcal/mol at 298K\n",
      "Could not determine pH values where the free energy is within 80 % of minimum\n",
      "Could not determine the pH-range where the free energy is negative\n",
      "\n",
      "\n",
      "Protein charge of folded and unfolded state as a function of pH\n",
      "    pH  unfolded  folded\n",
      "  0.00     42.99   42.98\n",
      "  0.10     42.99   42.98\n",
      "  0.20     42.99   42.97\n",
      "  0.30     42.99   42.96\n",
      "  0.40     42.99   42.95\n",
      "  0.50     42.98   42.94\n",
      "  0.60     42.98   42.92\n",
      "  0.70     42.97   42.90\n",
      "  0.80     42.97   42.88\n",
      "  0.90     42.96   42.85\n",
      "  1.00     42.94   42.81\n",
      "  1.10     42.93   42.76\n",
      "  1.20     42.91   42.71\n",
      "  1.30     42.89   42.64\n",
      "  1.40     42.86   42.55\n",
      "  1.50     42.83   42.44\n",
      "  1.60     42.78   42.32\n",
      "  1.70     42.73   42.16\n",
      "  1.80     42.66   41.98\n",
      "  1.90     42.57   41.77\n",
      "  2.00     42.46   41.52\n",
      "  2.10     42.32   41.23\n",
      "  2.20     42.15   40.90\n",
      "  2.30     41.94   40.52\n",
      "  2.40     41.68   40.08\n",
      "  2.50     41.35   39.59\n",
      "  2.60     40.96   39.03\n",
      "  2.70     40.47   38.40\n",
      "  2.80     39.88   37.69\n",
      "  2.90     39.17   36.89\n",
      "  3.00     38.32   36.00\n",
      "  3.10     37.31   35.00\n",
      "  3.20     36.13   33.89\n",
      "  3.30     34.77   32.68\n",
      "  3.40     33.21   31.35\n",
      "  3.50     31.47   29.91\n",
      "  3.60     29.54   28.38\n",
      "  3.70     27.44   26.75\n",
      "  3.80     25.20   25.06\n",
      "  3.90     22.85   23.30\n",
      "  4.00     20.41   21.51\n",
      "  4.10     17.94   19.69\n",
      "  4.20     15.46   17.87\n",
      "  4.30     13.00   16.06\n",
      "  4.40     10.61   14.27\n",
      "  4.50      8.32   12.53\n",
      "  4.60      6.16   10.84\n",
      "  4.70      4.14    9.21\n",
      "  4.80      2.30    7.66\n",
      "  4.90      0.64    6.18\n",
      "  5.00     -0.83    4.80\n",
      "  5.10     -2.12    3.51\n",
      "  5.20     -3.24    2.30\n",
      "  5.30     -4.21    1.18\n",
      "  5.40     -5.04    0.13\n",
      "  5.50     -5.76   -0.84\n",
      "  5.60     -6.39   -1.75\n",
      "  5.70     -6.94   -2.60\n",
      "  5.80     -7.44   -3.40\n",
      "  5.90     -7.89   -4.15\n",
      "  6.00     -8.32   -4.86\n",
      "  6.10     -8.72   -5.53\n",
      "  6.20     -9.12   -6.16\n",
      "  6.30     -9.51   -6.75\n",
      "  6.40     -9.89   -7.31\n",
      "  6.50    -10.26   -7.82\n",
      "  6.60    -10.62   -8.31\n",
      "  6.70    -10.96   -8.75\n",
      "  6.80    -11.28   -9.16\n",
      "  6.90    -11.58   -9.54\n",
      "  7.00    -11.85   -9.90\n",
      "  7.10    -12.10  -10.23\n",
      "  7.20    -12.34  -10.54\n",
      "  7.30    -12.55  -10.82\n",
      "  7.40    -12.74  -11.09\n",
      "  7.50    -12.93  -11.35\n",
      "  7.60    -13.11  -11.58\n",
      "  7.70    -13.29  -11.81\n",
      "  7.80    -13.47  -12.02\n",
      "  7.90    -13.66  -12.23\n",
      "  8.00    -13.86  -12.44\n",
      "  8.10    -14.08  -12.64\n",
      "  8.20    -14.31  -12.84\n",
      "  8.30    -14.57  -13.05\n",
      "  8.40    -14.85  -13.27\n",
      "  8.50    -15.17  -13.51\n",
      "  8.60    -15.53  -13.77\n",
      "  8.70    -15.93  -14.06\n",
      "  8.80    -16.37  -14.40\n",
      "  8.90    -16.87  -14.78\n",
      "  9.00    -17.42  -15.22\n",
      "  9.10    -18.04  -15.72\n",
      "  9.20    -18.72  -16.30\n",
      "  9.30    -19.49  -16.96\n",
      "  9.40    -20.34  -17.71\n",
      "  9.50    -21.30  -18.55\n",
      "  9.60    -22.37  -19.50\n",
      "  9.70    -23.56  -20.55\n",
      "  9.80    -24.88  -21.72\n",
      "  9.90    -26.33  -22.98\n",
      " 10.00    -27.91  -24.34\n",
      " 10.10    -29.60  -25.79\n",
      " 10.20    -31.40  -27.31\n",
      " 10.30    -33.26  -28.87\n",
      " 10.40    -35.16  -30.45\n",
      " 10.50    -37.06  -32.01\n",
      " 10.60    -38.92  -33.54\n",
      " 10.70    -40.70  -34.99\n",
      " 10.80    -42.38  -36.37\n",
      " 10.90    -43.92  -37.64\n",
      " 11.00    -45.32  -38.81\n",
      " 11.10    -46.58  -39.88\n",
      " 11.20    -47.70  -40.84\n",
      " 11.30    -48.69  -41.72\n",
      " 11.40    -49.58  -42.51\n",
      " 11.50    -50.38  -43.25\n",
      " 11.60    -51.11  -43.93\n",
      " 11.70    -51.80  -44.59\n",
      " 11.80    -52.46  -45.22\n",
      " 11.90    -53.11  -45.85\n",
      " 12.00    -53.76  -46.49\n",
      " 12.10    -54.43  -47.14\n",
      " 12.20    -55.11  -47.80\n",
      " 12.30    -55.81  -48.49\n",
      " 12.40    -56.51  -49.20\n",
      " 12.50    -57.22  -49.92\n",
      " 12.60    -57.90  -50.65\n",
      " 12.70    -58.56  -51.38\n",
      " 12.80    -59.18  -52.10\n",
      " 12.90    -59.75  -52.81\n",
      " 13.00    -60.27  -53.49\n",
      " 13.10    -60.72  -54.14\n",
      " 13.20    -61.11  -54.75\n",
      " 13.30    -61.45  -55.32\n",
      " 13.40    -61.73  -55.85\n",
      " 13.50    -61.97  -56.34\n",
      " 13.60    -62.17  -56.79\n",
      " 13.70    -62.33  -57.20\n",
      " 13.80    -62.46  -57.56\n",
      " 13.90    -62.57  -57.89\n",
      " 14.00    -62.65  -58.19\n",
      "The pI is  5.41 (folded) and  4.94 (unfolded)\n",
      "\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "INFO:Applying pKa values at a pH of 7.00:\n",
      "INFO:Adding hydrogens to biomolecule.\n",
      "INFO:Debumping biomolecule (again).\n",
      "WARNING:WARNING: Unable to debump LEU A 29\n",
      "INFO:Optimizing hydrogen bonds\n",
      "INFO:Applying force field to biomolecule states.\n",
      "INFO:Regenerating headers.\n",
      "INFO:Regenerating PDB lines.\n",
      "WARNING:Ignoring 467 header lines in output.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "pH: 7.0\tname: AtSAL1_updated\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:PDB2PQR v3.6.2: biomolecular structure conversion software.\n",
      "INFO:Please cite:  Jurrus E, et al.  Improvements to the APBS biomolecular solvation software suite.  Protein Sci 27 112-128 (2018).\n",
      "INFO:Please cite:  Dolinsky TJ, et al.  PDB2PQR: expanding and upgrading automated preparation of biomolecular structures for molecular simulations. Nucleic Acids Res 35 W522-W525 (2007).\n",
      "INFO:Checking and transforming input arguments.\n",
      "INFO:Loading topology files.\n",
      "INFO:Loading molecule: /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/AtAHL.pdb\n",
      "INFO:Setting up molecule.\n",
      "INFO:Created biomolecule object with 368 residues and 5474 atoms.\n",
      "INFO:Setting termini states for biomolecule chains.\n",
      "INFO:Loading forcefield.\n",
      "INFO:Loading hydrogen topology definitions.\n",
      "INFO:This biomolecule is clean.  No repair needed.\n",
      "INFO:Updating disulfide bridges.\n",
      "INFO:Debumping biomolecule.\n",
      "INFO:Assigning titration states with PROPKA.\n",
      "INFO:\n",
      "propka3.5.1                                                                                  2024-07-26\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "--                                                                           --\n",
      "--  PROPKA: A PROTEIN PKA PREDICTOR                                          --\n",
      "--                                                                           --\n",
      "--  VERSION 1.0,  04/25/2004,  IOWA CITY                                     --\n",
      "--  BY HUI LI                                                                --\n",
      "--                                                                           --\n",
      "--  VERSION 2.0,  11/05/2007, IOWA CITY/COPENHAGEN                           --\n",
      "--  BY DELPHINE C. BAS AND DAVID M. ROGERS                                   --\n",
      "--                                                                           --\n",
      "--  VERSION 3.0,  01/06/2011, COPENHAGEN                                     --\n",
      "--  BY MATS H.M. OLSSON AND CHRESTEN R. SONDERGARD                           --\n",
      "--                                                                           --\n",
      "--  VERSION 3.1,  07/01/2011, COPENHAGEN                                     --\n",
      "--  BY CHRESTEN R. SONDERGARD AND MATS H.M. OLSSON                           --\n",
      "--                                                                           --\n",
      "--  VERSION 3.2,  06/17/2020, PLANET EARTH                                   --\n",
      "--  SEE ABOVE FOR AUTHORS                                                    --\n",
      "--                                                                           --\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "References:\n",
      "\n",
      "Very Fast Empirical Prediction and Rationalization of Protein pKa Values.\n",
      "Hui Li, Andrew D. Robertson and Jan H. Jensen. PROTEINS: Structure, Function,\n",
      "and Bioinformatics. 61:704-721 (2005)\n",
      "\n",
      "Very Fast Prediction and Rationalization of pKa Values for Protein-Ligand\n",
      "Complexes.  Delphine C. Bas, David M. Rogers and Jan H. Jensen.  PROTEINS:\n",
      "Structure, Function, and Bioinformatics 73:765-783 (2008)\n",
      "\n",
      "PROPKA3: Consistent Treatment of Internal and Surface Residues in Empirical\n",
      "pKa predictions.  Mats H.M. Olsson, Chresten R. Sondergard, Michal Rostkowski,\n",
      "and Jan H. Jensen.  Journal of Chemical Theory and Computation, 7(2):525-537\n",
      "(2011)\n",
      "\n",
      "Improved Treatment of Ligands and Coupling Effects in Empirical Calculation\n",
      "and Rationalization of pKa Values.  Chresten R. Sondergaard, Mats H.M. Olsson,\n",
      "Michal Rostkowski, and Jan H. Jensen.  Journal of Chemical Theory and\n",
      "Computation, (2011)\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "\n",
      "INFO:Coupling map:\n",
      "\n",
      "INFO:Calculating pKas for Conformation container 1A with 3218 atoms and 894 groups\n",
      "INFO:did not converge in 10 iterations\n",
      "INFO:pKa determinants:\n",
      "\n",
      "---------  -----   ------   ---------------------    --------------    --------------    --------------\n",
      "                            DESOLVATION  EFFECTS       SIDECHAIN          BACKBONE        COULOMBIC\n",
      " RESIDUE    pKa    BURIED     REGULAR      RE        HYDROGEN BOND     HYDROGEN BOND      INTERACTION\n",
      "---------  -----   ------   ---------   ---------    --------------    --------------    --------------\n",
      "\n",
      "ASP   6 A   3.00    28 %    0.95  359   0.08    0   -1.34 ARG  10 A    0.00 XXX   0 X   -0.50 ARG  10 A\n",
      "\n",
      "ASP  38 A   4.10     0 %    0.13  168   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.04 LYS  35 A\n",
      "ASP  38 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 ASP  39 A\n",
      "ASP  38 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP  40 A\n",
      "ASP  38 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP  74 A\n",
      "ASP  38 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 GLU  80 A\n",
      "\n",
      "ASP  39 A   2.55    20 %    0.70  337   0.00    0   -0.85 SER  41 A   -0.18 ASP  39 A   -0.26 LYS  37 A\n",
      "ASP  39 A                                            0.00 XXX   0 X   -0.61 SER  41 A   -0.27 LYS 302 A\n",
      "ASP  39 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.23 GLU  72 A\n",
      "\n",
      "ASP  40 A   4.01     0 %    0.14  138   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP  39 A\n",
      "\n",
      "ASP  47 A   5.77*  100 %    3.69  595   1.08    0   -0.85 THR  44 A   -0.64 THR  44 A    0.41 GLU  72 A\n",
      "ASP  47 A                                           -0.59 THR 144 A   -0.44 ASP  47 A   -0.69 LYS  37 A\n",
      "\n",
      "ASP  64 A   2.40     0 %    0.38  280   0.00    0   -0.30 LYS 120 A    0.00 XXX   0 X   -0.34 LYS 120 A\n",
      "ASP  64 A                                           -0.68 ARG 124 A    0.00 XXX   0 X   -0.46 ARG 124 A\n",
      "\n",
      "ASP  74 A   3.59    27 %    0.81  358   0.01    0    0.00 XXX   0 X   -0.53 GLU  76 A   -0.02 LYS  35 A\n",
      "ASP  74 A                                            0.00 XXX   0 X   -0.39 THR  77 A   -0.11 LYS  37 A\n",
      "ASP  74 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU  80 A\n",
      "\n",
      "ASP  82 A   3.93     0 %    0.12  119   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.07 LYS  35 A\n",
      "ASP  82 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 GLU  80 A\n",
      "\n",
      "ASP 139 A  11.53*  100 %    3.50  634   0.39    0    0.49 ASP 305 A   -0.33 VAL 141 A   -0.41 LYS  37 A\n",
      "ASP 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.21 LYS 279 A\n",
      "ASP 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.50 LYS 302 A\n",
      "ASP 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.40 ASP  47 A\n",
      "ASP 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.37 GLU  73 A\n",
      "ASP 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.29 GLU  72 A\n",
      "ASP 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.51 ASP 142 A\n",
      "ASP 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.21 ASP 305 A\n",
      "\n",
      "ASP 142 A   5.13*  100 %    3.07  665   0.33    0   -0.07 SER 276 A   -0.04 GLY 143 A   -0.34 LYS 293 A\n",
      "ASP 142 A                                            0.00 XXX   0 X   -0.28 MET 277 A   -0.13 LYS 302 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.33 GLU  72 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 249 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.12 GLU 301 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.69 LYS 279 A\n",
      "\n",
      "ASP 151 A   4.73    32 %    0.77  370   0.33    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.15 ARG 149 A\n",
      "ASP 151 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 HIS  27 A\n",
      "\n",
      "ASP 237 A   3.65     0 %    0.27  219   0.00    0    0.00 XXX   0 X   -0.21 ASP 237 A   -0.31 LYS 354 A\n",
      "ASP 237 A                                            0.00 XXX   0 X   -0.00 ASP 238 A    0.10 ASP 238 A\n",
      "\n",
      "ASP 238 A   3.28     0 %    0.44  227   0.00    0    0.00 XXX   0 X   -0.76 VAL 240 A   -0.20 LYS 354 A\n",
      "\n",
      "ASP 287 A   3.73    64 %    2.21  461   0.60    0   -0.54 TYR 175 A    0.00 XXX   0 X   -0.47 LYS 179 A\n",
      "ASP 287 A                                           -0.37 LYS 179 A    0.00 XXX   0 X   -0.85 ARG 285 A\n",
      "ASP 287 A                                           -0.66 ARG 285 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "ASP 305 A   6.18*  100 %    2.94  592   0.45    0   -0.49 ASP 139 A   -0.22 LYS 302 A   -0.07 LYS  37 A\n",
      "ASP 305 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.99 GLU  72 A\n",
      "ASP 305 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.26 ASP 142 A\n",
      "ASP 305 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.66 LYS 279 A\n",
      "ASP 305 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.74 LYS 293 A\n",
      "ASP 305 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.72 GLU 301 A\n",
      "ASP 305 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.89 LYS 302 A\n",
      "ASP 305 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.91 HIS 306 A\n",
      "\n",
      "ASP 322 A   3.09    35 %    1.70  379   0.53    0   -0.75 ARG 326 A   -0.40 GLY 324 A   -0.10 ARG 341 A\n",
      "ASP 322 A                                            0.00 XXX   0 X   -0.03 GLY 325 A    0.28 ASP 340 A\n",
      "ASP 322 A                                            0.00 XXX   0 X   -0.78 ARG 326 A   -0.03 HIS 352 A\n",
      "ASP 322 A                                            0.00 XXX   0 X   -0.72 ASP 340 A   -0.42 ARG 326 A\n",
      "\n",
      "ASP 329 A   2.58    14 %    0.73  320   0.01    0   -0.85 SER 331 A   -0.78 SER 331 A   -0.01 ARG 326 A\n",
      "ASP 329 A                                           -0.01 LYS 332 A    0.00 XXX   0 X   -0.32 LYS 332 A\n",
      "\n",
      "ASP 340 A   2.65    15 %    0.87  323   0.00    0   -1.39 ARG 341 A    0.00 XXX   0 X   -0.19 ARG 326 A\n",
      "ASP 340 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.44 ARG 341 A\n",
      "\n",
      "GLU   2 A   3.46     0 %    0.26  241   0.00    0   -0.85 LYS 208 A    0.00 XXX   0 X   -0.14 N+    1 A\n",
      "GLU   2 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 GLU   4 A\n",
      "GLU   2 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP   6 A\n",
      "GLU   2 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.38 LYS 208 A\n",
      "\n",
      "GLU   4 A   2.86    16 %    0.81  327   0.00    0   -0.85 N+    1 A   -0.68 GLU   4 A   -0.01 LYS 208 A\n",
      "GLU   4 A                                           -0.25 ARG 134 A    0.00 XXX   0 X   -0.41 N+    1 A\n",
      "GLU   4 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.26 ARG 134 A\n",
      "\n",
      "GLU  24 A   4.74     4 %    0.29  293   0.02    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS  21 A\n",
      "GLU  24 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS  25 A\n",
      "GLU  24 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 HIS  27 A\n",
      "GLU  24 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU  96 A\n",
      "\n",
      "GLU  60 A   5.00    13 %    0.50  319   0.03    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.02 LYS 120 A\n",
      "GLU  60 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.07 ARG 124 A\n",
      "GLU  60 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP  64 A\n",
      "GLU  60 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 117 A\n",
      "\n",
      "GLU  72 A   1.96*   90 %    2.13  534   0.00    0   -0.71 TRP 304 A    0.00 XXX   0 X   -1.16 LYS  37 A\n",
      "GLU  72 A                                           -0.85 LYS  37 A    0.00 XXX   0 X   -1.10 LYS 302 A\n",
      "GLU  72 A                                           -0.84 LYS 302 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "GLU  73 A   7.93*   92 %    3.22  539   0.02    0   -0.85 LYS  37 A    0.00 XXX   0 X   -0.23 LYS 302 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP  39 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 ASP  74 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.18 LYS  37 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.24 ASP  47 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.93 GLU  72 A\n",
      "\n",
      "GLU  76 A   4.99     0 %    0.13  203   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP  38 A\n",
      "GLU  76 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.27 ASP  74 A\n",
      "GLU  76 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU  80 A\n",
      "\n",
      "GLU  80 A   3.53     0 %    0.26  214   0.00    0   -0.85 LYS  35 A    0.00 XXX   0 X   -0.38 LYS  35 A\n",
      "\n",
      "GLU  96 A   3.28    17 %    0.51  330   0.04    0   -0.20 LYS  21 A    0.00 XXX   0 X   -0.37 LYS  21 A\n",
      "GLU  96 A                                           -0.76 LYS  25 A    0.00 XXX   0 X   -0.44 LYS  25 A\n",
      "\n",
      "GLU 100 A   4.18    21 %    0.73  339   0.12    0   -0.85 LYS  21 A    0.00 XXX   0 X   -0.01 LYS  25 A\n",
      "GLU 100 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 LYS 108 A\n",
      "GLU 100 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.17 GLU  96 A\n",
      "GLU 100 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.47 LYS  21 A\n",
      "\n",
      "GLU 117 A   3.73     0 %    0.34  255   0.00    0    0.00 XXX   0 X   -0.81 GLY 114 A   -0.26 LYS 111 A\n",
      "GLU 117 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS 120 A\n",
      "\n",
      "GLU 161 A   3.76     0 %    0.38  263   0.00    0   -0.74 ARG 134 A    0.00 XXX   0 X   -0.04 N+    1 A\n",
      "GLU 161 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 LYS 164 A\n",
      "GLU 161 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU   4 A\n",
      "GLU 161 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.38 ARG 134 A\n",
      "\n",
      "GLU 180 A   4.49     0 %    0.11  156   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.12 LYS 178 A\n",
      "\n",
      "GLU 226 A   4.56     0 %    0.06  112   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "GLU 246 A   5.27    37 %    1.07  386   0.02    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.20 ARG 273 A\n",
      "GLU 246 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.12 HIS 255 A\n",
      "\n",
      "GLU 249 A   5.27    46 %    0.93  410   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.10 ARG 149 A\n",
      "GLU 249 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 293 A\n",
      "GLU 249 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP  39 A\n",
      "GLU 249 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 HIS 255 A\n",
      "\n",
      "GLU 289 A   3.91    55 %    1.68  435   0.54    0   -0.85 SER 234 A   -0.79 THR 243 A   -0.18 LYS 179 A\n",
      "GLU 289 A                                           -0.62 THR 243 A   -0.00 GLU 289 A   -0.45 LYS 269 A\n",
      "GLU 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP 287 A\n",
      "\n",
      "GLU 301 A   1.69    92 %    3.27  539   0.00    0   -0.85 LYS 293 A    0.00 XXX   0 X   -0.03 LYS 302 A\n",
      "GLU 301 A                                           -1.60 HIS 306 A    0.00 XXX   0 X   -0.52 LYS 279 A\n",
      "GLU 301 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.62 LYS 293 A\n",
      "GLU 301 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.45 HIS 306 A\n",
      "\n",
      "GLU 314 A   5.47    70 %    1.78  477   0.58    0   -0.74 ARG 209 A    0.00 XXX   0 X    0.16 GLU 315 A\n",
      "GLU 314 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.81 ARG 209 A\n",
      "\n",
      "GLU 315 A   4.02    32 %    1.27  372   0.32    0   -0.15 SER 211 A   -0.81 GLY 210 A   -0.02 LYS 208 A\n",
      "GLU 315 A                                            0.00 XXX   0 X   -0.11 SER 211 A   -0.28 ARG 209 A\n",
      "GLU 315 A                                            0.00 XXX   0 X   -0.70 GLY 212 A    0.00 XXX   0 X\n",
      "\n",
      "GLU 337 A   3.62     0 %    0.36  250   0.00    0   -0.85 LYS 300 A    0.00 XXX   0 X   -0.01 LYS 332 A\n",
      "GLU 337 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.38 LYS 300 A\n",
      "\n",
      "GLU 353 A   3.83     0 %    0.26  212   0.00    0   -0.84 GLN 349 A    0.00 XXX   0 X   -0.01 LYS 354 A\n",
      "GLU 353 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 HIS 352 A\n",
      "\n",
      "GLU 364 A   4.60     0 %    0.12  133   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.01 ARG 341 A\n",
      "\n",
      "C-  368 A   3.32     0 %    0.12   79   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "HIS  13 A   4.57    81 %   -2.69  507   0.00    0    0.82 GLN 217 A    0.00 XXX   0 X   -0.06 ARG  10 A\n",
      "\n",
      "HIS  27 A   6.82     6 %   -0.41  298   0.00    0    0.00 XXX   0 X    0.68 GLN  23 A    0.01 GLU  24 A\n",
      "HIS  27 A                                            0.00 XXX   0 X    0.01 ARG 149 A    0.02 ASP 151 A\n",
      "\n",
      "HIS  33 A   6.06    11 %   -0.44  312   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "HIS 135 A   5.85    35 %   -1.10  379   0.00    0    0.00 XXX   0 X    0.45 LYS 132 A    0.00 XXX   0 X\n",
      "\n",
      "HIS 255 A   6.41    20 %   -0.86  336   0.00    0    0.00 XXX   0 X    0.78 GLU 249 A   -0.09 ARG 250 A\n",
      "HIS 255 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 273 A\n",
      "HIS 255 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 LYS 293 A\n",
      "HIS 255 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.12 GLU 246 A\n",
      "HIS 255 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 249 A\n",
      "\n",
      "HIS 306 A   5.07   100 %   -3.28  641   0.00    0    1.60 GLU 301 A    0.00 XXX   0 X   -1.25 LYS 279 A\n",
      "HIS 306 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.85 LYS 293 A\n",
      "HIS 306 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.45 GLU 301 A\n",
      "HIS 306 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.91 ASP 305 A\n",
      "\n",
      "HIS 352 A   4.72    65 %   -1.90  462   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 322 A\n",
      "HIS 352 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU 353 A\n",
      "\n",
      "CYS  19 A  13.08   100 %    3.78  655   0.00    0    0.00 XXX   0 X   -0.08 CYS  19 A    0.39 ASP  47 A\n",
      "\n",
      "CYS 125 A  11.63    96 %    3.54  549   0.00    0   -0.05 GLN  51 A    0.00 XXX   0 X   -0.23 ARG 124 A\n",
      "CYS 125 A                                           -0.63 SER  55 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "CYS 172 A  13.72   100 %    3.47  673   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.18 ARG 285 A\n",
      "CYS 172 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 ASP 142 A\n",
      "CYS 172 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.35 CYS 203 A\n",
      "\n",
      "CYS 181 A   9.45     5 %    0.55  295   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 178 A\n",
      "CYS 181 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.10 LYS 201 A\n",
      "CYS 181 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 ARG 285 A\n",
      "CYS 181 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.11 GLU 180 A\n",
      "CYS 181 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 287 A\n",
      "\n",
      "CYS 186 A   9.82    31 %    0.88  368   0.00    0    0.00 XXX   0 X   -0.08 CYS 186 A   -0.15 ARG 273 A\n",
      "CYS 186 A                                            0.00 XXX   0 X   -0.02 ASN 187 A    0.19 ASP 151 A\n",
      "\n",
      "CYS 203 A  11.78   100 %    3.38  575   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.65 ARG 285 A\n",
      "CYS 203 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 ASP 287 A\n",
      "\n",
      "CYS 245 A  12.50   100 %    3.40  607   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.11 LYS 279 A\n",
      "CYS 245 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 246 A\n",
      "CYS 245 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 287 A\n",
      "CYS 245 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.12 GLU 289 A\n",
      "\n",
      "CYS 346 A  12.74   100 %    3.74  577   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "TYR 104 A  10.78    56 %    2.22  438   0.00    0   -0.85 SER  17 A   -0.68 GLY 222 A    0.31 GLU 100 A\n",
      "TYR 104 A                                            0.00 XXX   0 X   -0.01 GLY 223 A   -0.21 LYS  21 A\n",
      "\n",
      "TYR 153 A  15.70   100 %    4.01  560   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    1.59 CYS  19 A\n",
      "TYR 153 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU  24 A\n",
      "TYR 153 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 CYS 172 A\n",
      "\n",
      "TYR 175 A  15.46   100 %    3.86  560   0.00    0   -0.21 ARG 285 A    0.00 XXX   0 X    0.40 CYS 172 A\n",
      "TYR 175 A                                            0.54 ASP 287 A    0.00 XXX   0 X    0.54 CYS 203 A\n",
      "TYR 175 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.43 CYS 245 A\n",
      "TYR 175 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.24 ARG 285 A\n",
      "TYR 175 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.15 ASP 287 A\n",
      "\n",
      "TYR 206 A  11.50    48 %    1.46  417   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU   2 A\n",
      "TYR 206 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP   6 A\n",
      "TYR 206 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.00 ARG  10 A\n",
      "\n",
      "TYR 275 A  12.11    30 %    1.75  364   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.17 ARG 149 A\n",
      "TYR 275 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 ASP 151 A\n",
      "TYR 275 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 CYS 186 A\n",
      "TYR 275 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.49 GLU 249 A\n",
      "TYR 275 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 ARG 250 A\n",
      "TYR 275 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 ARG 273 A\n",
      "\n",
      "TYR 280 A  14.55   100 %    3.81  663   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.14 ASP 142 A\n",
      "TYR 280 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.40 GLU 301 A\n",
      "TYR 280 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.12 ASP 305 A\n",
      "TYR 280 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU 314 A\n",
      "\n",
      "TYR 299 A  14.29    74 %    1.49  488   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP  39 A\n",
      "TYR 299 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.23 GLU  72 A\n",
      "TYR 299 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP 139 A\n",
      "TYR 299 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.32 ASP 142 A\n",
      "TYR 299 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU 249 A\n",
      "TYR 299 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 TYR 335 A\n",
      "TYR 299 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.93 GLU 301 A\n",
      "TYR 299 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.05 ASP 305 A\n",
      "\n",
      "TYR 335 A  10.09     6 %    0.28  299   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP  39 A\n",
      "TYR 335 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU  72 A\n",
      "TYR 335 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP 305 A\n",
      "TYR 335 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 337 A\n",
      "TYR 335 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.10 LYS 300 A\n",
      "TYR 335 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.18 LYS 302 A\n",
      "\n",
      "TYR 360 A  10.04     0 %    0.35  261   0.00    0   -0.33 ARG 341 A    0.00 XXX   0 X    0.11 ASP 322 A\n",
      "TYR 360 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.07 ARG 326 A\n",
      "TYR 360 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.28 ASP 340 A\n",
      "TYR 360 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.30 ARG 341 A\n",
      "\n",
      "LYS  21 A  11.63    24 %   -1.00  348   0.00    0    0.20 GLU  96 A    0.00 XXX   0 X    0.03 GLU  24 A\n",
      "LYS  21 A                                            0.85 GLU 100 A    0.00 XXX   0 X    0.21 TYR 104 A\n",
      "LYS  21 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.37 GLU  96 A\n",
      "LYS  21 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.47 GLU 100 A\n",
      "\n",
      "LYS  25 A  11.09    17 %   -0.54  328   0.00    0    0.76 GLU  96 A    0.00 XXX   0 X    0.03 GLU  24 A\n",
      "LYS  25 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 100 A\n",
      "LYS  25 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.11 LYS  21 A\n",
      "LYS  25 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.44 GLU  96 A\n",
      "\n",
      "LYS  35 A  11.28     0 %   -0.58  275   0.00    0    0.85 GLU  80 A    0.00 XXX   0 X    0.04 ASP  38 A\n",
      "LYS  35 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP  74 A\n",
      "LYS  35 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 ASP  82 A\n",
      "LYS  35 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 GLU  80 A\n",
      "\n",
      "LYS  37 A  13.37    74 %   -2.71  488   0.00    0    0.85 GLU  72 A    0.00 XXX   0 X    0.26 ASP  39 A\n",
      "LYS  37 A                                            0.85 GLU  73 A    0.00 XXX   0 X    0.11 ASP  74 A\n",
      "LYS  37 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.41 ASP 139 A\n",
      "LYS  37 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 ASP 305 A\n",
      "LYS  37 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.69 ASP  47 A\n",
      "LYS  37 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.16 GLU  72 A\n",
      "LYS  37 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.18 GLU  73 A\n",
      "\n",
      "LYS 108 A  10.20     0 %   -0.32  220   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 100 A\n",
      "\n",
      "LYS 111 A  10.66     0 %   -0.10  128   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.26 GLU 117 A\n",
      "\n",
      "LYS 120 A  10.80     0 %   -0.19  244   0.00    0    0.30 ASP  64 A    0.00 XXX   0 X    0.02 GLU  60 A\n",
      "LYS 120 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 117 A\n",
      "LYS 120 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.20 ARG 124 A\n",
      "LYS 120 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.34 ASP  64 A\n",
      "\n",
      "LYS 132 A  10.46     0 %   -0.04   93   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 164 A  10.39     0 %   -0.10  198   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 161 A\n",
      "LYS 164 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 209 A\n",
      "\n",
      "LYS 178 A  10.56     0 %   -0.13  165   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.12 GLU 180 A\n",
      "LYS 178 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 CYS 181 A\n",
      "\n",
      "LYS 179 A  10.81    19 %   -0.61  334   0.00    0    0.37 ASP 287 A    0.00 XXX   0 X   -0.11 ARG 285 A\n",
      "LYS 179 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.18 GLU 289 A\n",
      "LYS 179 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.47 ASP 287 A\n",
      "\n",
      "LYS 191 A  10.47     0 %   -0.03   79   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 193 A  10.48     0 %   -0.02   43   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 201 A  10.15    29 %   -0.45  362   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.10 CYS 181 A\n",
      "\n",
      "LYS 208 A  11.45     0 %   -0.31  254   0.00    0    0.85 GLU   2 A    0.00 XXX   0 X    0.01 GLU   4 A\n",
      "LYS 208 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 315 A\n",
      "LYS 208 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 GLU   2 A\n",
      "\n",
      "LYS 232 A  10.41     0 %   -0.09  201   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 269 A  10.30    12 %   -0.57  314   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.45 GLU 289 A\n",
      "LYS 269 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 LYS 179 A\n",
      "\n",
      "LYS 279 A   9.58   100 %   -3.37  661   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.21 ASP 139 A\n",
      "LYS 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.11 CYS 245 A\n",
      "LYS 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.74 LYS 293 A\n",
      "LYS 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.69 ASP 142 A\n",
      "LYS 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.52 GLU 301 A\n",
      "LYS 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.66 ASP 305 A\n",
      "\n",
      "LYS 293 A  10.97    95 %   -3.08  548   0.00    0    0.85 GLU 301 A    0.00 XXX   0 X    0.34 ASP 142 A\n",
      "LYS 293 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 GLU 249 A\n",
      "LYS 293 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 302 A\n",
      "LYS 293 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.62 GLU 301 A\n",
      "LYS 293 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.74 ASP 305 A\n",
      "\n",
      "LYS 300 A  11.52     0 %   -0.32  217   0.00    0    0.85 GLU 337 A    0.00 XXX   0 X    0.10 TYR 335 A\n",
      "LYS 300 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 GLU 337 A\n",
      "\n",
      "LYS 302 A  12.40    70 %   -1.84  477   0.00    0    0.84 GLU  72 A    0.00 XXX   0 X    0.27 ASP  39 A\n",
      "LYS 302 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.23 GLU  73 A\n",
      "LYS 302 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.50 ASP 139 A\n",
      "LYS 302 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 ASP 142 A\n",
      "LYS 302 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 301 A\n",
      "LYS 302 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.44 LYS  37 A\n",
      "LYS 302 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.18 TYR 335 A\n",
      "LYS 302 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.10 GLU  72 A\n",
      "LYS 302 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.89 ASP 305 A\n",
      "\n",
      "LYS 332 A  10.67     0 %   -0.17  202   0.00    0    0.01 ASP 329 A    0.00 XXX   0 X    0.32 ASP 329 A\n",
      "LYS 332 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 337 A\n",
      "\n",
      "LYS 354 A  10.57     0 %   -0.46  252   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.31 ASP 237 A\n",
      "LYS 354 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 ASP 238 A\n",
      "LYS 354 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 353 A\n",
      "\n",
      "ARG  10 A  13.55    29 %   -0.79  363   0.00    0    1.34 ASP   6 A    0.00 XXX   0 X    0.00 TYR 206 A\n",
      "ARG  10 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.50 ASP   6 A\n",
      "\n",
      "ARG 124 A  12.59    43 %   -1.35  401   0.00    0    0.68 ASP  64 A    0.00 XXX   0 X    0.07 GLU  60 A\n",
      "ARG 124 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.23 CYS 125 A\n",
      "ARG 124 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.46 ASP  64 A\n",
      "\n",
      "ARG 134 A  13.76     4 %   -0.37  293   0.00    0    0.25 GLU   4 A    0.00 XXX   0 X    0.26 GLU   4 A\n",
      "ARG 134 A                                            0.74 GLU 161 A    0.00 XXX   0 X    0.38 GLU 161 A\n",
      "\n",
      "ARG 149 A  12.62     0 %   -0.29  273   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP 151 A\n",
      "ARG 149 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 GLU 249 A\n",
      "ARG 149 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.17 TYR 275 A\n",
      "\n",
      "ARG 209 A  12.83    59 %   -1.51  447   0.00    0    0.74 GLU 314 A    0.00 XXX   0 X    0.28 GLU 315 A\n",
      "ARG 209 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.81 GLU 314 A\n",
      "\n",
      "ARG 250 A  12.37     0 %   -0.09  157   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.01 TYR 275 A\n",
      "ARG 250 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.05 ARG 273 A\n",
      "\n",
      "ARG 268 A  12.32     0 %   -0.18  156   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "ARG 273 A  12.69     0 %   -0.19  279   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.15 CYS 186 A\n",
      "ARG 273 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 GLU 246 A\n",
      "ARG 273 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 TYR 275 A\n",
      "\n",
      "ARG 285 A  13.75    70 %   -2.59  478   0.00    0    0.21 TYR 175 A    0.00 XXX   0 X    0.18 CYS 172 A\n",
      "ARG 285 A                                            0.66 ASP 287 A    0.00 XXX   0 X    1.24 TYR 175 A\n",
      "ARG 285 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 CYS 181 A\n",
      "ARG 285 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.65 CYS 203 A\n",
      "ARG 285 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.85 ASP 287 A\n",
      "\n",
      "ARG 326 A  13.60     0 %   -0.31  246   0.00    0    0.75 ASP 322 A    0.00 XXX   0 X    0.01 ASP 329 A\n",
      "ARG 326 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 ASP 340 A\n",
      "ARG 326 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 TYR 360 A\n",
      "ARG 326 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 ARG 341 A\n",
      "ARG 326 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.42 ASP 322 A\n",
      "\n",
      "ARG 341 A  14.03    17 %   -1.04  330   0.00    0    0.33 TYR 360 A    0.00 XXX   0 X    0.10 ASP 322 A\n",
      "ARG 341 A                                            1.39 ASP 340 A    0.00 XXX   0 X    0.30 TYR 360 A\n",
      "ARG 341 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 364 A\n",
      "ARG 341 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.44 ASP 340 A\n",
      "\n",
      "N+    1 A   8.68     0 %   -0.50  274   0.00    0    0.85 GLU   4 A    0.00 XXX   0 X    0.14 GLU   2 A\n",
      "N+    1 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.16 ARG 134 A\n",
      "N+    1 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 GLU 161 A\n",
      "N+    1 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.10 LYS 208 A\n",
      "N+    1 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.41 GLU   4 A\n",
      "\n",
      "Coupled residues (marked *) were detected.Please rerun PropKa with the --display-coupled-residues \n",
      "or -d option for detailed information.\n",
      "\n",
      "INFO:pKa summary:\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "SUMMARY OF THIS PREDICTION\n",
      "       Group      pKa  model-pKa   ligand atom-type\n",
      "   ASP   6 A     3.00       3.80                      \n",
      "   ASP  38 A     4.10       3.80                      \n",
      "   ASP  39 A     2.55       3.80                      \n",
      "   ASP  40 A     4.01       3.80                      \n",
      "   ASP  47 A     5.77       3.80                      \n",
      "   ASP  64 A     2.40       3.80                      \n",
      "   ASP  74 A     3.59       3.80                      \n",
      "   ASP  82 A     3.93       3.80                      \n",
      "   ASP 139 A    11.53       3.80                      \n",
      "   ASP 142 A     5.13       3.80                      \n",
      "   ASP 151 A     4.73       3.80                      \n",
      "   ASP 237 A     3.65       3.80                      \n",
      "   ASP 238 A     3.28       3.80                      \n",
      "   ASP 287 A     3.73       3.80                      \n",
      "   ASP 305 A     6.18       3.80                      \n",
      "   ASP 322 A     3.09       3.80                      \n",
      "   ASP 329 A     2.58       3.80                      \n",
      "   ASP 340 A     2.65       3.80                      \n",
      "   GLU   2 A     3.46       4.50                      \n",
      "   GLU   4 A     2.86       4.50                      \n",
      "   GLU  24 A     4.74       4.50                      \n",
      "   GLU  60 A     5.00       4.50                      \n",
      "   GLU  72 A     1.96       4.50                      \n",
      "   GLU  73 A     7.93       4.50                      \n",
      "   GLU  76 A     4.99       4.50                      \n",
      "   GLU  80 A     3.53       4.50                      \n",
      "   GLU  96 A     3.28       4.50                      \n",
      "   GLU 100 A     4.18       4.50                      \n",
      "   GLU 117 A     3.73       4.50                      \n",
      "   GLU 161 A     3.76       4.50                      \n",
      "   GLU 180 A     4.49       4.50                      \n",
      "   GLU 226 A     4.56       4.50                      \n",
      "   GLU 246 A     5.27       4.50                      \n",
      "   GLU 249 A     5.27       4.50                      \n",
      "   GLU 289 A     3.91       4.50                      \n",
      "   GLU 301 A     1.69       4.50                      \n",
      "   GLU 314 A     5.47       4.50                      \n",
      "   GLU 315 A     4.02       4.50                      \n",
      "   GLU 337 A     3.62       4.50                      \n",
      "   GLU 353 A     3.83       4.50                      \n",
      "   GLU 364 A     4.60       4.50                      \n",
      "   C-  368 A     3.32       3.20                      \n",
      "   HIS  13 A     4.57       6.50                      \n",
      "   HIS  27 A     6.82       6.50                      \n",
      "   HIS  33 A     6.06       6.50                      \n",
      "   HIS 135 A     5.85       6.50                      \n",
      "   HIS 255 A     6.41       6.50                      \n",
      "   HIS 306 A     5.07       6.50                      \n",
      "   HIS 352 A     4.72       6.50                      \n",
      "   CYS  19 A    13.08       9.00                      \n",
      "   CYS 125 A    11.63       9.00                      \n",
      "   CYS 172 A    13.72       9.00                      \n",
      "   CYS 181 A     9.45       9.00                      \n",
      "   CYS 186 A     9.82       9.00                      \n",
      "   CYS 203 A    11.78       9.00                      \n",
      "   CYS 245 A    12.50       9.00                      \n",
      "   CYS 346 A    12.74       9.00                      \n",
      "   TYR 104 A    10.78      10.00                      \n",
      "   TYR 153 A    15.70      10.00                      \n",
      "   TYR 175 A    15.46      10.00                      \n",
      "   TYR 206 A    11.50      10.00                      \n",
      "   TYR 275 A    12.11      10.00                      \n",
      "   TYR 280 A    14.55      10.00                      \n",
      "   TYR 299 A    14.29      10.00                      \n",
      "   TYR 335 A    10.09      10.00                      \n",
      "   TYR 360 A    10.04      10.00                      \n",
      "   LYS  21 A    11.63      10.50                      \n",
      "   LYS  25 A    11.09      10.50                      \n",
      "   LYS  35 A    11.28      10.50                      \n",
      "   LYS  37 A    13.37      10.50                      \n",
      "   LYS 108 A    10.20      10.50                      \n",
      "   LYS 111 A    10.66      10.50                      \n",
      "   LYS 120 A    10.80      10.50                      \n",
      "   LYS 132 A    10.46      10.50                      \n",
      "   LYS 164 A    10.39      10.50                      \n",
      "   LYS 178 A    10.56      10.50                      \n",
      "   LYS 179 A    10.81      10.50                      \n",
      "   LYS 191 A    10.47      10.50                      \n",
      "   LYS 193 A    10.48      10.50                      \n",
      "   LYS 201 A    10.15      10.50                      \n",
      "   LYS 208 A    11.45      10.50                      \n",
      "   LYS 232 A    10.41      10.50                      \n",
      "   LYS 269 A    10.30      10.50                      \n",
      "   LYS 279 A     9.58      10.50                      \n",
      "   LYS 293 A    10.97      10.50                      \n",
      "   LYS 300 A    11.52      10.50                      \n",
      "   LYS 302 A    12.40      10.50                      \n",
      "   LYS 332 A    10.67      10.50                      \n",
      "   LYS 354 A    10.57      10.50                      \n",
      "   ARG  10 A    13.55      12.50                      \n",
      "   ARG 124 A    12.59      12.50                      \n",
      "   ARG 134 A    13.76      12.50                      \n",
      "   ARG 149 A    12.62      12.50                      \n",
      "   ARG 209 A    12.83      12.50                      \n",
      "   ARG 250 A    12.37      12.50                      \n",
      "   ARG 268 A    12.32      12.50                      \n",
      "   ARG 273 A    12.69      12.50                      \n",
      "   ARG 285 A    13.75      12.50                      \n",
      "   ARG 326 A    13.60      12.50                      \n",
      "   ARG 341 A    14.03      12.50                      \n",
      "   N+    1 A     8.68       8.00                      \n",
      "\n",
      "INFO:PROPKA information:\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "Free energy of   folding (kcal/mol) as a function of pH (using neutral reference)\n",
      "  0.00     28.05\n",
      "  1.00     27.81\n",
      "  2.00     26.19\n",
      "  3.00     20.32\n",
      "  4.00     11.83\n",
      "  5.00     10.29\n",
      "  6.00     12.13\n",
      "  7.00     13.31\n",
      "  8.00     15.91\n",
      "  9.00     20.92\n",
      " 10.00     32.09\n",
      " 11.00     51.34\n",
      " 12.00     71.39\n",
      " 13.00     87.31\n",
      " 14.00     98.55\n",
      "\n",
      "The pH of optimum stability is  4.6 for which the free energy is    9.9 kcal/mol at 298K\n",
      "Could not determine pH values where the free energy is within 80 % of minimum\n",
      "Could not determine the pH-range where the free energy is negative\n",
      "\n",
      "\n",
      "Protein charge of folded and unfolded state as a function of pH\n",
      "    pH  unfolded  folded\n",
      "  0.00     42.00   41.95\n",
      "  0.10     41.99   41.94\n",
      "  0.20     41.99   41.92\n",
      "  0.30     41.99   41.90\n",
      "  0.40     41.99   41.88\n",
      "  0.50     41.99   41.85\n",
      "  0.60     41.98   41.81\n",
      "  0.70     41.98   41.76\n",
      "  0.80     41.97   41.70\n",
      "  0.90     41.97   41.63\n",
      "  1.00     41.96   41.54\n",
      "  1.10     41.95   41.44\n",
      "  1.20     41.93   41.31\n",
      "  1.30     41.92   41.16\n",
      "  1.40     41.89   40.98\n",
      "  1.50     41.87   40.78\n",
      "  1.60     41.83   40.54\n",
      "  1.70     41.79   40.26\n",
      "  1.80     41.74   39.95\n",
      "  1.90     41.67   39.59\n",
      "  2.00     41.59   39.19\n",
      "  2.10     41.48   38.75\n",
      "  2.20     41.35   38.25\n",
      "  2.30     41.19   37.70\n",
      "  2.40     40.99   37.10\n",
      "  2.50     40.75   36.44\n",
      "  2.60     40.44   35.72\n",
      "  2.70     40.08   34.94\n",
      "  2.80     39.63   34.10\n",
      "  2.90     39.09   33.19\n",
      "  3.00     38.44   32.23\n",
      "  3.10     37.68   31.20\n",
      "  3.20     36.79   30.11\n",
      "  3.30     35.75   28.97\n",
      "  3.40     34.56   27.77\n",
      "  3.50     33.23   26.53\n",
      "  3.60     31.74   25.25\n",
      "  3.70     30.11   23.95\n",
      "  3.80     28.36   22.63\n",
      "  3.90     26.50   21.29\n",
      "  4.00     24.55   19.97\n",
      "  4.10     22.54   18.65\n",
      "  4.20     20.50   17.34\n",
      "  4.30     18.46   16.06\n",
      "  4.40     16.44   14.80\n",
      "  4.50     14.47   13.57\n",
      "  4.60     12.59   12.37\n",
      "  4.70     10.83   11.19\n",
      "  4.80      9.20   10.06\n",
      "  4.90      7.72    8.95\n",
      "  5.00      6.39    7.89\n",
      "  5.10      5.22    6.87\n",
      "  5.20      4.19    5.90\n",
      "  5.30      3.29    4.97\n",
      "  5.40      2.50    4.10\n",
      "  5.50      1.81    3.27\n",
      "  5.60      1.19    2.50\n",
      "  5.70      0.62    1.77\n",
      "  5.80      0.10    1.08\n",
      "  5.90     -0.40    0.44\n",
      "  6.00     -0.88   -0.15\n",
      "  6.10     -1.36   -0.71\n",
      "  6.20     -1.85   -1.22\n",
      "  6.30     -2.33   -1.69\n",
      "  6.40     -2.82   -2.13\n",
      "  6.50     -3.30   -2.53\n",
      "  6.60     -3.77   -2.89\n",
      "  6.70     -4.22   -3.22\n",
      "  6.80     -4.65   -3.52\n",
      "  6.90     -5.05   -3.79\n",
      "  7.00     -5.42   -4.03\n",
      "  7.10     -5.76   -4.25\n",
      "  7.20     -6.07   -4.45\n",
      "  7.30     -6.36   -4.64\n",
      "  7.40     -6.62   -4.81\n",
      "  7.50     -6.87   -4.97\n",
      "  7.60     -7.12   -5.12\n",
      "  7.70     -7.36   -5.27\n",
      "  7.80     -7.62   -5.42\n",
      "  7.90     -7.88   -5.57\n",
      "  8.00     -8.17   -5.72\n",
      "  8.10     -8.48   -5.88\n",
      "  8.20     -8.82   -6.04\n",
      "  8.30     -9.20   -6.22\n",
      "  8.40     -9.63   -6.40\n",
      "  8.50    -10.11   -6.59\n",
      "  8.60    -10.65   -6.81\n",
      "  8.70    -11.25   -7.04\n",
      "  8.80    -11.91   -7.29\n",
      "  8.90    -12.63   -7.57\n",
      "  9.00    -13.41   -7.89\n",
      "  9.10    -14.26   -8.24\n",
      "  9.20    -15.17   -8.64\n",
      "  9.30    -16.14   -9.09\n",
      "  9.40    -17.18   -9.59\n",
      "  9.50    -18.30  -10.16\n",
      "  9.60    -19.51  -10.80\n",
      "  9.70    -20.81  -11.51\n",
      "  9.80    -22.22  -12.29\n",
      "  9.90    -23.72  -13.16\n",
      " 10.00    -25.32  -14.10\n",
      " 10.10    -27.01  -15.11\n",
      " 10.20    -28.77  -16.20\n",
      " 10.30    -30.57  -17.34\n",
      " 10.40    -32.39  -18.53\n",
      " 10.50    -34.20  -19.75\n",
      " 10.60    -35.95  -21.00\n",
      " 10.70    -37.62  -22.26\n",
      " 10.80    -39.18  -23.51\n",
      " 10.90    -40.61  -24.76\n",
      " 11.00    -41.91  -25.98\n",
      " 11.10    -43.08  -27.19\n",
      " 11.20    -44.11  -28.36\n",
      " 11.30    -45.04  -29.51\n",
      " 11.40    -45.86  -30.63\n",
      " 11.50    -46.61  -31.73\n",
      " 11.60    -47.30  -32.79\n",
      " 11.70    -47.95  -33.84\n",
      " 11.80    -48.58  -34.85\n",
      " 11.90    -49.21  -35.85\n",
      " 12.00    -49.84  -36.83\n",
      " 12.10    -50.49  -37.80\n",
      " 12.20    -51.16  -38.75\n",
      " 12.30    -51.85  -39.69\n",
      " 12.40    -52.54  -40.62\n",
      " 12.50    -53.24  -41.54\n",
      " 12.60    -53.92  -42.44\n",
      " 12.70    -54.58  -43.33\n",
      " 12.80    -55.20  -44.20\n",
      " 12.90    -55.76  -45.04\n",
      " 13.00    -56.27  -45.87\n",
      " 13.10    -56.73  -46.67\n",
      " 13.20    -57.12  -47.44\n",
      " 13.30    -57.45  -48.19\n",
      " 13.40    -57.74  -48.93\n",
      " 13.50    -57.97  -49.64\n",
      " 13.60    -58.17  -50.32\n",
      " 13.70    -58.33  -50.98\n",
      " 13.80    -58.46  -51.62\n",
      " 13.90    -58.57  -52.22\n",
      " 14.00    -58.65  -52.79\n",
      "The pI is  5.97 (folded) and  5.82 (unfolded)\n",
      "\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "INFO:Applying pKa values at a pH of 7.00:\n",
      "INFO:Adding hydrogens to biomolecule.\n",
      "INFO:Debumping biomolecule (again).\n",
      "INFO:Optimizing hydrogen bonds\n",
      "INFO:Applying force field to biomolecule states.\n",
      "INFO:Regenerating headers.\n",
      "INFO:Regenerating PDB lines.\n",
      "WARNING:Ignoring 466 header lines in output.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "pH: 7.0\tname: AtAHL\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:PDB2PQR v3.6.2: biomolecular structure conversion software.\n",
      "INFO:Please cite:  Jurrus E, et al.  Improvements to the APBS biomolecular solvation software suite.  Protein Sci 27 112-128 (2018).\n",
      "INFO:Please cite:  Dolinsky TJ, et al.  PDB2PQR: expanding and upgrading automated preparation of biomolecular structures for molecular simulations. Nucleic Acids Res 35 W522-W525 (2007).\n",
      "INFO:Checking and transforming input arguments.\n",
      "INFO:Loading topology files.\n",
      "INFO:Loading molecule: /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/74987.pdb\n",
      "INFO:Setting up molecule.\n",
      "INFO:Created biomolecule object with 352 residues and 5354 atoms.\n",
      "INFO:Setting termini states for biomolecule chains.\n",
      "INFO:Loading forcefield.\n",
      "INFO:Loading hydrogen topology definitions.\n",
      "INFO:This biomolecule is clean.  No repair needed.\n",
      "INFO:Updating disulfide bridges.\n",
      "INFO:Debumping biomolecule.\n",
      "INFO:Assigning titration states with PROPKA.\n",
      "INFO:\n",
      "propka3.5.1                                                                                  2024-07-26\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "--                                                                           --\n",
      "--  PROPKA: A PROTEIN PKA PREDICTOR                                          --\n",
      "--                                                                           --\n",
      "--  VERSION 1.0,  04/25/2004,  IOWA CITY                                     --\n",
      "--  BY HUI LI                                                                --\n",
      "--                                                                           --\n",
      "--  VERSION 2.0,  11/05/2007, IOWA CITY/COPENHAGEN                           --\n",
      "--  BY DELPHINE C. BAS AND DAVID M. ROGERS                                   --\n",
      "--                                                                           --\n",
      "--  VERSION 3.0,  01/06/2011, COPENHAGEN                                     --\n",
      "--  BY MATS H.M. OLSSON AND CHRESTEN R. SONDERGARD                           --\n",
      "--                                                                           --\n",
      "--  VERSION 3.1,  07/01/2011, COPENHAGEN                                     --\n",
      "--  BY CHRESTEN R. SONDERGARD AND MATS H.M. OLSSON                           --\n",
      "--                                                                           --\n",
      "--  VERSION 3.2,  06/17/2020, PLANET EARTH                                   --\n",
      "--  SEE ABOVE FOR AUTHORS                                                    --\n",
      "--                                                                           --\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "References:\n",
      "\n",
      "Very Fast Empirical Prediction and Rationalization of Protein pKa Values.\n",
      "Hui Li, Andrew D. Robertson and Jan H. Jensen. PROTEINS: Structure, Function,\n",
      "and Bioinformatics. 61:704-721 (2005)\n",
      "\n",
      "Very Fast Prediction and Rationalization of pKa Values for Protein-Ligand\n",
      "Complexes.  Delphine C. Bas, David M. Rogers and Jan H. Jensen.  PROTEINS:\n",
      "Structure, Function, and Bioinformatics 73:765-783 (2008)\n",
      "\n",
      "PROPKA3: Consistent Treatment of Internal and Surface Residues in Empirical\n",
      "pKa predictions.  Mats H.M. Olsson, Chresten R. Sondergard, Michal Rostkowski,\n",
      "and Jan H. Jensen.  Journal of Chemical Theory and Computation, 7(2):525-537\n",
      "(2011)\n",
      "\n",
      "Improved Treatment of Ligands and Coupling Effects in Empirical Calculation\n",
      "and Rationalization of pKa Values.  Chresten R. Sondergaard, Mats H.M. Olsson,\n",
      "Michal Rostkowski, and Jan H. Jensen.  Journal of Chemical Theory and\n",
      "Computation, (2011)\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "\n",
      "INFO:Coupling map:\n",
      "\n",
      "INFO:Calculating pKas for Conformation container 1A with 3171 atoms and 875 groups\n",
      "INFO:pKa determinants:\n",
      "\n",
      "---------  -----   ------   ---------------------    --------------    --------------    --------------\n",
      "                            DESOLVATION  EFFECTS       SIDECHAIN          BACKBONE        COULOMBIC\n",
      " RESIDUE    pKa    BURIED     REGULAR      RE        HYDROGEN BOND     HYDROGEN BOND      INTERACTION\n",
      "---------  -----   ------   ---------   ---------    --------------    --------------    --------------\n",
      "\n",
      "ASP  10 A   4.95    44 %    1.49  405   0.14    0   -0.28 ARG  14 A    0.00 XXX   0 X   -0.05 ARG  17 A\n",
      "ASP  10 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.10 LYS  59 A\n",
      "ASP  10 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU   6 A\n",
      "ASP  10 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP  13 A\n",
      "ASP  10 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.29 GLU  56 A\n",
      "ASP  10 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.55 ARG  14 A\n",
      "\n",
      "ASP  13 A   4.41    60 %    1.88  448   0.32    0   -0.72 ARG  17 A   -0.16 GLN 194 A   -0.08 ARG  14 A\n",
      "ASP  13 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.00 LYS  91 A\n",
      "ASP  13 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.05 LYS 191 A\n",
      "ASP  13 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.59 ARG  17 A\n",
      "\n",
      "ASP  21 A   2.92    25 %    0.74  352   0.19    0   -0.11 ARG  24 A    0.00 XXX   0 X   -0.03 ARG  17 A\n",
      "ASP  21 A                                           -0.85 LYS  91 A    0.00 XXX   0 X   -0.32 ARG  24 A\n",
      "ASP  21 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.49 LYS  91 A\n",
      "\n",
      "ASP  38 A   2.45    12 %    0.70  316   0.00    0   -0.85 THR  40 A   -0.20 ASP  38 A   -0.26 LYS  36 A\n",
      "ASP  38 A                                            0.00 XXX   0 X   -0.66 THR  40 A   -0.08 LYS 286 A\n",
      "\n",
      "ASP  46 A   5.88   100 %    3.67  601   1.00    0   -0.85 THR  43 A   -0.70 THR  43 A   -0.59 LYS  36 A\n",
      "ASP  46 A                                           -0.64 THR 134 A   -0.38 ASP  46 A    0.56 GLU  69 A\n",
      "\n",
      "ASP  71 A   3.56    35 %    0.91  380   0.00    0    0.00 XXX   0 X   -0.46 HIS  73 A   -0.12 LYS  36 A\n",
      "ASP  71 A                                            0.00 XXX   0 X   -0.28 PHE  74 A   -0.05 ARG  76 A\n",
      "ASP  71 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.21 LYS 113 A\n",
      "ASP  71 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 322 A\n",
      "ASP  71 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.19 HIS  73 A\n",
      "ASP  71 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.18 ASP 109 A\n",
      "\n",
      "ASP  96 A   3.95     0 %    0.24  216   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.09 LYS  89 A\n",
      "ASP  96 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.00 LYS  91 A\n",
      "\n",
      "ASP 101 A   3.94     0 %    0.32  265   0.00    0    0.00 XXX   0 X   -0.03 ASP 101 A   -0.16 LYS  89 A\n",
      "ASP 101 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 103 A\n",
      "\n",
      "ASP 103 A   2.46     0 %    0.37  271   0.00    0   -0.85 SER 100 A   -0.81 SER 100 A   -0.04 LYS  89 A\n",
      "ASP 103 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 HIS  98 A\n",
      "\n",
      "ASP 109 A   2.64    27 %    1.12  356   0.23    0   -0.85 SER  72 A   -0.79 SER  72 A   -0.02 HIS  73 A\n",
      "ASP 109 A                                           -0.48 ARG  76 A    0.00 XXX   0 X   -0.37 ARG  76 A\n",
      "\n",
      "ASP 114 A   3.91     0 %    0.33  245   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.29 LYS 113 A\n",
      "ASP 114 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 ARG 322 A\n",
      "ASP 114 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.11 ASP  71 A\n",
      "ASP 114 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 109 A\n",
      "\n",
      "ASP 129 A  11.71*  100 %    3.55  628   0.27    0    0.42 ASP 289 A   -0.23 ILE 131 A   -0.17 LYS  36 A\n",
      "ASP 129 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.32 ASP  46 A\n",
      "ASP 129 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 GLU  70 A\n",
      "ASP 129 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.98 GLU  69 A\n",
      "ASP 129 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.31 ASP 132 A\n",
      "ASP 129 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.28 ASP 289 A\n",
      "\n",
      "ASP 132 A   6.29*  100 %    3.80  623   0.65    0   -0.13 SER 260 A   -0.48 SER 260 A    0.11 GLU  69 A\n",
      "ASP 132 A                                            0.00 XXX   0 X   -0.56 LEU 261 A   -0.70 LYS 263 A\n",
      "ASP 132 A                                            0.00 XXX   0 X   -0.19 CYS 262 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 141 A   3.75     0 %    0.27  267   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.31 LYS  23 A\n",
      "ASP 141 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 ARG  24 A\n",
      "\n",
      "ASP 154 A   4.07     0 %    0.28  239   0.00    0    0.00 XXX   0 X   -0.00 ASP 154 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 170 A   3.21     0 %    0.24  274   0.00    0   -0.42 ARG 271 A    0.00 XXX   0 X   -0.03 LYS 192 A\n",
      "ASP 170 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.37 ARG 271 A\n",
      "\n",
      "ASP 173 A   1.41     0 %    0.49  200   0.00    0   -0.85 SER 175 A   -0.08 ASP 173 A   -0.38 LYS 192 A\n",
      "ASP 173 A                                           -0.81 LYS 192 A   -0.76 SER 175 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 203 A   3.82     0 %    0.12  132   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.10 LYS 191 A\n",
      "\n",
      "ASP 210 A   2.58    45 %    1.85  406   0.24    0   -0.85 SER 273 A   -0.41 CYS 212 A   -0.43 ARG 219 A\n",
      "ASP 210 A                                           -0.78 ASN 334 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "ASP 210 A                                           -0.85 ARG 219 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "ASP 238 A   4.04     0 %    0.23  233   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.09 LYS 247 A\n",
      "ASP 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 249 A\n",
      "ASP 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 229 A\n",
      "ASP 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 GLU 250 A\n",
      "\n",
      "ASP 245 A   4.00     0 %    0.16  109   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.04 GLU 250 A\n",
      "\n",
      "ASP 289 A   7.94*  100 %    2.62  580   0.38    0   -0.42 ASP 129 A    0.00 XXX   0 X   -0.01 LYS  36 A\n",
      "ASP 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.00 LYS 263 A\n",
      "ASP 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.19 LYS 286 A\n",
      "ASP 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.76 GLU  69 A\n",
      "ASP 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.00 ASP 132 A\n",
      "\n",
      "ASP 306 A   3.95    39 %    1.62  391   0.52    0    0.00 XXX   0 X   -0.01 ASP 306 A   -0.01 ARG 283 A\n",
      "ASP 306 A                                            0.00 XXX   0 X   -0.48 GLU 308 A   -0.01 HIS 338 A\n",
      "ASP 306 A                                            0.00 XXX   0 X   -0.34 GLY 309 A    0.00 XXX   0 X\n",
      "ASP 306 A                                            0.00 XXX   0 X   -0.61 ASP 310 A    0.00 XXX   0 X\n",
      "ASP 306 A                                            0.00 XXX   0 X   -0.53 ALA 327 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 310 A   4.33     0 %    0.22  242   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.00 LYS 303 A\n",
      "ASP 310 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.28 ASP 306 A\n",
      "ASP 310 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 311 A\n",
      "\n",
      "ASP 317 A   3.26    37 %    1.59  385   0.12    0    0.00 XXX   0 X   -0.51 GLU 320 A   -0.05 LYS 286 A\n",
      "ASP 317 A                                            0.00 XXX   0 X   -0.03 ARG 321 A   -0.13 ARG 321 A\n",
      "ASP 317 A                                            0.00 XXX   0 X   -0.72 ARG 322 A   -0.10 ARG 322 A\n",
      "ASP 317 A                                            0.00 XXX   0 X   -0.69 LEU 323 A    0.00 XXX   0 X\n",
      "\n",
      "GLU   4 A   4.10    15 %    0.77  322   0.00    0    0.00 XXX   0 X   -0.76 GLU   4 A   -0.26 N+    1 A\n",
      "GLU   4 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.07 LYS   3 A\n",
      "GLU   4 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 HIS   2 A\n",
      "GLU   4 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 HIS 183 A\n",
      "\n",
      "GLU   6 A   4.46     0 %    0.19  222   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.05 LYS   3 A\n",
      "GLU   6 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.18 HIS   2 A\n",
      "\n",
      "GLU  31 A   4.62     6 %    0.36  298   0.00    0   -0.24 SER  29 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "GLU  35 A   4.73     0 %    0.22  190   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP  38 A\n",
      "\n",
      "GLU  56 A   2.83    47 %    1.86  412   0.21    0   -0.78 SER  93 A   -0.83 SER  93 A   -0.52 ARG  14 A\n",
      "GLU  56 A                                           -0.18 ARG  14 A    0.00 XXX   0 X   -0.58 LYS  59 A\n",
      "GLU  56 A                                           -0.85 LYS  59 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "GLU  69 A   4.06*   92 %    1.97  539   0.00    0   -0.35 TRP 288 A    0.00 XXX   0 X   -0.18 LYS 286 A\n",
      "GLU  69 A                                           -0.85 LYS  36 A    0.00 XXX   0 X   -0.02 ARG 322 A\n",
      "GLU  69 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 ASP  38 A\n",
      "GLU  69 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.20 LYS  36 A\n",
      "\n",
      "GLU  70 A   8.36*   97 %    3.42  553   0.03    0   -0.85 LYS  36 A    0.00 XXX   0 X    0.08 ASP  38 A\n",
      "GLU  70 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 ASP  71 A\n",
      "GLU  70 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.29 LYS  36 A\n",
      "GLU  70 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.00 ASP  46 A\n",
      "GLU  70 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.27 GLU  69 A\n",
      "\n",
      "GLU  87 A   5.73    40 %    1.00  393   0.16    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.00 ARG  24 A\n",
      "GLU  87 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.14 LYS  91 A\n",
      "GLU  87 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.17 ASP  21 A\n",
      "GLU  87 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU  31 A\n",
      "\n",
      "GLU 106 A   4.79     0 %    0.16  269   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 110 A\n",
      "GLU 106 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.11 ASP 103 A\n",
      "GLU 106 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 109 A\n",
      "\n",
      "GLU 142 A   4.62     0 %    0.09  177   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.01 LYS  23 A\n",
      "GLU 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 141 A\n",
      "\n",
      "GLU 156 A   5.01    18 %    0.37  332   0.02    0    0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 154 A\n",
      "GLU 156 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 HIS 298 A\n",
      "GLU 156 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.14 GLU 311 A\n",
      "GLU 156 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 315 A\n",
      "\n",
      "GLU 224 A   2.07    12 %    0.73  314   0.02    0   -0.52 ARG 135 A    0.00 XXX   0 X   -0.16 LYS 263 A\n",
      "GLU 224 A                                           -0.85 LYS 139 A    0.00 XXX   0 X   -0.42 ARG 135 A\n",
      "GLU 224 A                                           -0.84 CYS 258 A    0.00 XXX   0 X   -0.39 LYS 139 A\n",
      "\n",
      "GLU 229 A   3.42     0 %    0.34  244   0.00    0   -0.66 THR 227 A   -0.76 GLU 229 A    0.00 XXX   0 X\n",
      "\n",
      "GLU 244 A   4.57     0 %    0.07   82   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "GLU 250 A   2.96     0 %    0.53  224   0.00    0   -0.85 LYS 247 A   -0.84 LYS 247 A   -0.00 LYS 217 A\n",
      "GLU 250 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.38 LYS 247 A\n",
      "\n",
      "GLU 299 A   4.11    28 %    1.15  361   0.24    0    0.00 XXX   0 X   -0.79 GLY 185 A   -0.13 ARG 206 A\n",
      "GLU 299 A                                            0.00 XXX   0 X   -0.08 CYS 186 A   -0.10 HIS 183 A\n",
      "GLU 299 A                                            0.00 XXX   0 X   -0.58 GLY 187 A   -0.10 HIS 298 A\n",
      "\n",
      "GLU 308 A   5.24     0 %    0.26  271   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.10 ARG 283 A\n",
      "GLU 308 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.24 ASP 306 A\n",
      "GLU 308 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP 310 A\n",
      "GLU 308 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 C-  352 A\n",
      "\n",
      "GLU 311 A   3.29    11 %    0.28  311   0.00    0   -0.85 LYS 303 A    0.00 XXX   0 X   -0.21 HIS 298 A\n",
      "GLU 311 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.42 LYS 303 A\n",
      "\n",
      "GLU 315 A   4.29     0 %    0.41  259   0.00    0   -0.39 ASN 155 A    0.00 XXX   0 X   -0.24 ARG 321 A\n",
      "GLU 315 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 154 A\n",
      "\n",
      "GLU 316 A   5.33    15 %    0.68  322   0.08    0    0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 306 A\n",
      "GLU 316 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 317 A\n",
      "\n",
      "GLU 320 A   4.89     8 %    0.35  305   0.01    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.05 LYS 113 A\n",
      "GLU 320 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 LYS 286 A\n",
      "GLU 320 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.19 ARG 322 A\n",
      "GLU 320 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 114 A\n",
      "GLU 320 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.30 ASP 317 A\n",
      "\n",
      "GLU 343 A   4.70     0 %    0.20  205   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "C-  352 A   2.61     0 %    0.24  209   0.00    0   -0.57 ARG 283 A    0.00 XXX   0 X   -0.26 ARG 283 A\n",
      "\n",
      "HIS   2 A   6.46     0 %   -0.18  208   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.08 N+    1 A\n",
      "HIS   2 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 GLU   4 A\n",
      "HIS   2 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.18 GLU   6 A\n",
      "\n",
      "HIS  73 A   6.41     0 %   -0.19  218   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.06 ARG  76 A\n",
      "HIS  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 113 A\n",
      "HIS  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 ASP  71 A\n",
      "HIS  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 109 A\n",
      "\n",
      "HIS  98 A   6.37     0 %   -0.15  122   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.00 LYS  89 A\n",
      "HIS  98 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 103 A\n",
      "\n",
      "HIS 183 A   6.16    21 %   -0.82  341   0.00    0    0.00 XXX   0 X    0.05 HIS 183 A   -0.39 N+    1 A\n",
      "HIS 183 A                                            0.00 XXX   0 X    0.71 ILE 184 A   -0.03 HIS   2 A\n",
      "HIS 183 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 GLU   4 A\n",
      "HIS 183 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 GLU 299 A\n",
      "\n",
      "HIS 248 A   6.22     0 %   -0.17  173   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 217 A\n",
      "HIS 248 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 249 A\n",
      "\n",
      "HIS 290 A   3.27   100 %   -3.17  619   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.07 LYS 263 A\n",
      "\n",
      "HIS 298 A   5.12    76 %   -1.70  493   0.00    0    0.00 XXX   0 X    0.30 GLY 302 A   -0.32 LYS 303 A\n",
      "HIS 298 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 156 A\n",
      "HIS 298 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 GLU 299 A\n",
      "HIS 298 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.21 GLU 311 A\n",
      "\n",
      "HIS 338 A   4.12    64 %   -2.13  461   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.17 LYS 303 A\n",
      "HIS 338 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.10 HIS 298 A\n",
      "HIS 338 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 306 A\n",
      "\n",
      "CYS  16 A  14.27   100 %    3.67  626   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.05 ARG  17 A\n",
      "CYS  16 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.44 ASP  13 A\n",
      "CYS  16 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.84 CYS  19 A\n",
      "CYS  16 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.36 CYS 164 A\n",
      "\n",
      "CYS  19 A  13.25   100 %    3.84  640   0.00    0    0.00 XXX   0 X   -0.06 CYS  19 A   -0.03 LYS  23 A\n",
      "CYS  19 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.49 ASP  46 A\n",
      "CYS  19 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU  87 A\n",
      "\n",
      "CYS 119 A   9.04     0 %    0.20  187   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.14 LYS 121 A\n",
      "CYS 119 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 321 A\n",
      "\n",
      "CYS 164 A  12.58   100 %    3.66  580   0.00    0    0.00 XXX   0 X   -0.13 TYR 145 A   -0.02 LYS  23 A\n",
      "CYS 164 A                                            0.00 XXX   0 X   -0.10 ASN 166 A   -0.04 ARG 271 A\n",
      "CYS 164 A                                            0.00 XXX   0 X   -0.05 TRP 167 A    0.11 ASP 132 A\n",
      "CYS 164 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.17 CYS 257 A\n",
      "\n",
      "CYS 186 A   9.38     0 %    0.39  262   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.09 N+    1 A\n",
      "CYS 186 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 ARG 206 A\n",
      "CYS 186 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 GLU 299 A\n",
      "\n",
      "CYS 207 A  11.12    63 %    2.22  457   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.07 ARG 206 A\n",
      "CYS 207 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 271 A\n",
      "\n",
      "CYS 212 A   9.75    29 %    1.27  363   0.00    0   -0.00 SER 273 A   -0.71 LEU 214 A   -0.04 LYS 217 A\n",
      "CYS 212 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.12 ARG 219 A\n",
      "CYS 212 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.35 ASP 210 A\n",
      "\n",
      "CYS 221 A  11.78    93 %    2.98  541   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.06 ARG 219 A\n",
      "CYS 221 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.40 LYS 263 A\n",
      "CYS 221 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.26 CYS 257 A\n",
      "\n",
      "CYS 257 A   6.54*   61 %    1.43  451   0.00    0   -3.60 CYS 262 A    0.00 XXX   0 X   -0.02 LYS 139 A\n",
      "CYS 257 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.33 LYS 263 A\n",
      "CYS 257 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 GLU 224 A\n",
      "\n",
      "CYS 258 A  10.03    39 %    1.48  391   0.00    0   -0.71 ARG 135 A   -0.05 CYS 258 A   -0.39 ARG 135 A\n",
      "CYS 258 A                                           -0.23 LYS 139 A    0.00 XXX   0 X   -0.45 LYS 139 A\n",
      "CYS 258 A                                            0.84 GLU 224 A    0.00 XXX   0 X   -0.20 LYS 263 A\n",
      "CYS 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP 132 A\n",
      "CYS 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.15 CYS 257 A\n",
      "CYS 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.48 GLU 224 A\n",
      "\n",
      "CYS 262 A  17.21*  100 %    3.46  569   0.00    0    3.60 CYS 257 A   -0.07 CYS 262 A   -1.78 LYS 263 A\n",
      "CYS 262 A                                            0.00 XXX   0 X   -0.06 LYS 263 A    0.53 CYS 164 A\n",
      "CYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.44 CYS 221 A\n",
      "CYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 224 A\n",
      "CYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 CYS 258 A\n",
      "CYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.03 ASP 132 A\n",
      "CYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.81 CYS 257 A\n",
      "\n",
      "CYS 296 A  12.13   100 %    3.70  656   0.00    0   -0.85 SER 182 A    0.00 XXX   0 X    0.23 CYS 207 A\n",
      "CYS 296 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 299 A\n",
      "\n",
      "TYR   1 A  10.58     0 %    0.41  279   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU   4 A\n",
      "TYR   1 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP 154 A\n",
      "TYR   1 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 GLU 156 A\n",
      "\n",
      "TYR 116 A  10.46     2 %    0.31  288   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.16 CYS 119 A\n",
      "TYR 116 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 ARG 321 A\n",
      "\n",
      "TYR 125 A  12.77    85 %    2.97  518   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 315 A\n",
      "TYR 125 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP 317 A\n",
      "TYR 125 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.34 ARG 321 A\n",
      "\n",
      "TYR 145 A  16.29    97 %    3.60  552   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.78 CYS  16 A\n",
      "TYR 145 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.50 CYS  19 A\n",
      "TYR 145 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP  21 A\n",
      "TYR 145 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 141 A\n",
      "TYR 145 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.37 CYS 164 A\n",
      "\n",
      "TYR 264 A  13.96   100 %    3.82  649   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 132 A\n",
      "TYR 264 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP 289 A\n",
      "\n",
      "LYS   3 A  10.39     0 %   -0.23  184   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.07 GLU   4 A\n",
      "LYS   3 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU   6 A\n",
      "\n",
      "LYS  23 A   9.00    44 %   -1.88  405   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.03 CYS  19 A\n",
      "LYS  23 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.31 ASP 141 A\n",
      "LYS  23 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 142 A\n",
      "LYS  23 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 CYS 164 A\n",
      "\n",
      "LYS  30 A  10.47     0 %   -0.03   61   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS  32 A  10.45     0 %   -0.05  104   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS  36 A  13.02    75 %   -2.82  492   0.00    0    0.85 GLU  69 A    0.00 XXX   0 X    0.26 ASP  38 A\n",
      "LYS  36 A                                            0.85 GLU  70 A    0.00 XXX   0 X    0.12 ASP  71 A\n",
      "LYS  36 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.17 ASP 129 A\n",
      "LYS  36 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 289 A\n",
      "LYS  36 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.59 ASP  46 A\n",
      "LYS  36 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.20 GLU  69 A\n",
      "LYS  36 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.29 GLU  70 A\n",
      "\n",
      "LYS  59 A  10.02    37 %   -1.79  385   0.00    0    0.85 GLU  56 A    0.00 XXX   0 X    0.10 ASP  10 A\n",
      "LYS  59 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.22 ARG  14 A\n",
      "LYS  59 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.58 GLU  56 A\n",
      "\n",
      "LYS  89 A  10.40     0 %   -0.38  237   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP  96 A\n",
      "LYS  89 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 ASP 101 A\n",
      "LYS  89 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 103 A\n",
      "\n",
      "LYS  91 A  10.94    30 %   -0.88  365   0.00    0    0.85 ASP  21 A    0.00 XXX   0 X    0.00 ASP  13 A\n",
      "LYS  91 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.11 ARG  17 A\n",
      "LYS  91 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 ARG  24 A\n",
      "LYS  91 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.14 GLU  87 A\n",
      "LYS  91 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP  96 A\n",
      "LYS  91 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.49 ASP  21 A\n",
      "\n",
      "LYS 113 A  10.86     0 %   -0.16  251   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.21 ASP  71 A\n",
      "LYS 113 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.29 ASP 114 A\n",
      "LYS 113 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 320 A\n",
      "LYS 113 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 ARG 322 A\n",
      "\n",
      "LYS 121 A  10.58     0 %   -0.06  106   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.14 CYS 119 A\n",
      "\n",
      "LYS 139 A  11.66     0 %   -0.43  269   0.00    0    0.23 CYS 258 A    0.00 XXX   0 X   -0.35 ARG 135 A\n",
      "LYS 139 A                                            0.85 GLU 224 A    0.00 XXX   0 X    0.02 CYS 257 A\n",
      "LYS 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.45 CYS 258 A\n",
      "LYS 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.39 GLU 224 A\n",
      "\n",
      "LYS 191 A  10.20     2 %   -0.45  288   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP  13 A\n",
      "LYS 191 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP 203 A\n",
      "\n",
      "LYS 192 A  11.20     0 %   -0.52  249   0.00    0    0.81 ASP 173 A    0.00 XXX   0 X    0.03 ASP 170 A\n",
      "LYS 192 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 ASP 173 A\n",
      "\n",
      "LYS 217 A  10.32     0 %   -0.21  224   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.04 CYS 212 A\n",
      "LYS 217 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 GLU 250 A\n",
      "\n",
      "LYS 247 A  11.60     0 %   -0.22  164   0.00    0    0.85 GLU 250 A    0.00 XXX   0 X    0.09 ASP 238 A\n",
      "LYS 247 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 GLU 250 A\n",
      "\n",
      "LYS 249 A  10.46     0 %   -0.10  152   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.06 ASP 238 A\n",
      "\n",
      "LYS 263 A  11.10    90 %   -2.93  534   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.04 ARG 135 A\n",
      "LYS 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.40 CYS 221 A\n",
      "LYS 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 GLU 224 A\n",
      "LYS 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.33 CYS 257 A\n",
      "LYS 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 CYS 258 A\n",
      "LYS 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.78 CYS 262 A\n",
      "LYS 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP 289 A\n",
      "LYS 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.70 ASP 132 A\n",
      "\n",
      "LYS 280 A  10.26    15 %   -0.23  322   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.01 ARG 135 A\n",
      "\n",
      "LYS 286 A  10.05    45 %   -0.88  407   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP  38 A\n",
      "LYS 286 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.18 GLU  69 A\n",
      "LYS 286 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 ASP 289 A\n",
      "LYS 286 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 317 A\n",
      "LYS 286 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU 320 A\n",
      "LYS 286 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.11 ARG 322 A\n",
      "LYS 286 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 LYS  36 A\n",
      "\n",
      "LYS 303 A  11.21    13 %   -0.56  318   0.00    0    0.85 GLU 311 A    0.00 XXX   0 X    0.00 ASP 310 A\n",
      "LYS 303 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.42 GLU 311 A\n",
      "\n",
      "ARG  14 A  12.98    32 %   -1.07  370   0.00    0    0.28 ASP  10 A    0.00 XXX   0 X    0.08 ASP  13 A\n",
      "ARG  14 A                                            0.18 GLU  56 A    0.00 XXX   0 X   -0.05 ARG  17 A\n",
      "ARG  14 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.55 ASP  10 A\n",
      "ARG  14 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.52 GLU  56 A\n",
      "\n",
      "ARG  17 A  13.13    26 %   -0.81  355   0.00    0    0.72 ASP  13 A    0.00 XXX   0 X    0.05 ASP  10 A\n",
      "ARG  17 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 CYS  16 A\n",
      "ARG  17 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP  21 A\n",
      "ARG  17 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.59 ASP  13 A\n",
      "\n",
      "ARG  24 A  12.69     0 %   -0.25  263   0.00    0    0.11 ASP  21 A    0.00 XXX   0 X    0.32 ASP  21 A\n",
      "ARG  24 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 GLU  87 A\n",
      "ARG  24 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 141 A\n",
      "\n",
      "ARG  76 A  12.96     1 %   -0.44  283   0.00    0    0.48 ASP 109 A    0.00 XXX   0 X    0.05 ASP  71 A\n",
      "ARG  76 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.37 ASP 109 A\n",
      "\n",
      "ARG 110 A  10.93    57 %   -1.59  442   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 106 A\n",
      "\n",
      "ARG 135 A  14.08    12 %   -0.46  314   0.00    0    0.71 CYS 258 A    0.00 XXX   0 X    0.39 CYS 258 A\n",
      "ARG 135 A                                            0.52 GLU 224 A    0.00 XXX   0 X    0.42 GLU 224 A\n",
      "\n",
      "ARG 206 A  12.17     3 %   -0.59  289   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.06 CYS 186 A\n",
      "ARG 206 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 CYS 207 A\n",
      "ARG 206 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 GLU 299 A\n",
      "\n",
      "ARG 219 A  12.97    38 %   -0.99  388   0.00    0    0.85 ASP 210 A    0.00 XXX   0 X    0.12 CYS 212 A\n",
      "ARG 219 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 CYS 221 A\n",
      "ARG 219 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.43 ASP 210 A\n",
      "\n",
      "ARG 271 A  12.14    41 %   -1.19  395   0.00    0    0.42 ASP 170 A    0.00 XXX   0 X    0.04 CYS 164 A\n",
      "ARG 271 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 CYS 207 A\n",
      "ARG 271 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 219 A\n",
      "ARG 271 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.37 ASP 170 A\n",
      "\n",
      "ARG 278 A  11.64    29 %   -0.86  362   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "ARG 283 A  13.24     0 %   -0.20  243   0.00    0    0.57 C-  352 A    0.00 XXX   0 X    0.01 ASP 306 A\n",
      "ARG 283 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 GLU 308 A\n",
      "ARG 283 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.26 C-  352 A\n",
      "\n",
      "ARG 321 A  11.96    26 %   -1.28  353   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.01 TYR 116 A\n",
      "ARG 321 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 CYS 119 A\n",
      "ARG 321 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.34 TYR 125 A\n",
      "ARG 321 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.24 GLU 315 A\n",
      "ARG 321 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 ASP 317 A\n",
      "\n",
      "ARG 322 A  10.73    72 %   -2.17  484   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU  69 A\n",
      "ARG 322 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP  71 A\n",
      "ARG 322 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 ASP 114 A\n",
      "ARG 322 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP 317 A\n",
      "ARG 322 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 GLU 320 A\n",
      "\n",
      "N+    1 A   7.87     2 %   -0.47  287   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.26 GLU   4 A\n",
      "N+    1 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 CYS 186 A\n",
      "\n",
      "Coupled residues (marked *) were detected.Please rerun PropKa with the --display-coupled-residues \n",
      "or -d option for detailed information.\n",
      "\n",
      "INFO:pKa summary:\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "SUMMARY OF THIS PREDICTION\n",
      "       Group      pKa  model-pKa   ligand atom-type\n",
      "   ASP  10 A     4.95       3.80                      \n",
      "   ASP  13 A     4.41       3.80                      \n",
      "   ASP  21 A     2.92       3.80                      \n",
      "   ASP  38 A     2.45       3.80                      \n",
      "   ASP  46 A     5.88       3.80                      \n",
      "   ASP  71 A     3.56       3.80                      \n",
      "   ASP  96 A     3.95       3.80                      \n",
      "   ASP 101 A     3.94       3.80                      \n",
      "   ASP 103 A     2.46       3.80                      \n",
      "   ASP 109 A     2.64       3.80                      \n",
      "   ASP 114 A     3.91       3.80                      \n",
      "   ASP 129 A    11.71       3.80                      \n",
      "   ASP 132 A     6.29       3.80                      \n",
      "   ASP 141 A     3.75       3.80                      \n",
      "   ASP 154 A     4.07       3.80                      \n",
      "   ASP 170 A     3.21       3.80                      \n",
      "   ASP 173 A     1.41       3.80                      \n",
      "   ASP 203 A     3.82       3.80                      \n",
      "   ASP 210 A     2.58       3.80                      \n",
      "   ASP 238 A     4.04       3.80                      \n",
      "   ASP 245 A     4.00       3.80                      \n",
      "   ASP 289 A     7.94       3.80                      \n",
      "   ASP 306 A     3.95       3.80                      \n",
      "   ASP 310 A     4.33       3.80                      \n",
      "   ASP 317 A     3.26       3.80                      \n",
      "   GLU   4 A     4.10       4.50                      \n",
      "   GLU   6 A     4.46       4.50                      \n",
      "   GLU  31 A     4.62       4.50                      \n",
      "   GLU  35 A     4.73       4.50                      \n",
      "   GLU  56 A     2.83       4.50                      \n",
      "   GLU  69 A     4.06       4.50                      \n",
      "   GLU  70 A     8.36       4.50                      \n",
      "   GLU  87 A     5.73       4.50                      \n",
      "   GLU 106 A     4.79       4.50                      \n",
      "   GLU 142 A     4.62       4.50                      \n",
      "   GLU 156 A     5.01       4.50                      \n",
      "   GLU 224 A     2.07       4.50                      \n",
      "   GLU 229 A     3.42       4.50                      \n",
      "   GLU 244 A     4.57       4.50                      \n",
      "   GLU 250 A     2.96       4.50                      \n",
      "   GLU 299 A     4.11       4.50                      \n",
      "   GLU 308 A     5.24       4.50                      \n",
      "   GLU 311 A     3.29       4.50                      \n",
      "   GLU 315 A     4.29       4.50                      \n",
      "   GLU 316 A     5.33       4.50                      \n",
      "   GLU 320 A     4.89       4.50                      \n",
      "   GLU 343 A     4.70       4.50                      \n",
      "   C-  352 A     2.61       3.20                      \n",
      "   HIS   2 A     6.46       6.50                      \n",
      "   HIS  73 A     6.41       6.50                      \n",
      "   HIS  98 A     6.37       6.50                      \n",
      "   HIS 183 A     6.16       6.50                      \n",
      "   HIS 248 A     6.22       6.50                      \n",
      "   HIS 290 A     3.27       6.50                      \n",
      "   HIS 298 A     5.12       6.50                      \n",
      "   HIS 338 A     4.12       6.50                      \n",
      "   CYS  16 A    14.27       9.00                      \n",
      "   CYS  19 A    13.25       9.00                      \n",
      "   CYS 119 A     9.04       9.00                      \n",
      "   CYS 164 A    12.58       9.00                      \n",
      "   CYS 186 A     9.38       9.00                      \n",
      "   CYS 207 A    11.12       9.00                      \n",
      "   CYS 212 A     9.75       9.00                      \n",
      "   CYS 221 A    11.78       9.00                      \n",
      "   CYS 257 A     6.54       9.00                      \n",
      "   CYS 258 A    10.03       9.00                      \n",
      "   CYS 262 A    17.21       9.00                      \n",
      "   CYS 296 A    12.13       9.00                      \n",
      "   TYR   1 A    10.58      10.00                      \n",
      "   TYR 116 A    10.46      10.00                      \n",
      "   TYR 125 A    12.77      10.00                      \n",
      "   TYR 145 A    16.29      10.00                      \n",
      "   TYR 264 A    13.96      10.00                      \n",
      "   LYS   3 A    10.39      10.50                      \n",
      "   LYS  23 A     9.00      10.50                      \n",
      "   LYS  30 A    10.47      10.50                      \n",
      "   LYS  32 A    10.45      10.50                      \n",
      "   LYS  36 A    13.02      10.50                      \n",
      "   LYS  59 A    10.02      10.50                      \n",
      "   LYS  89 A    10.40      10.50                      \n",
      "   LYS  91 A    10.94      10.50                      \n",
      "   LYS 113 A    10.86      10.50                      \n",
      "   LYS 121 A    10.58      10.50                      \n",
      "   LYS 139 A    11.66      10.50                      \n",
      "   LYS 191 A    10.20      10.50                      \n",
      "   LYS 192 A    11.20      10.50                      \n",
      "   LYS 217 A    10.32      10.50                      \n",
      "   LYS 247 A    11.60      10.50                      \n",
      "   LYS 249 A    10.46      10.50                      \n",
      "   LYS 263 A    11.10      10.50                      \n",
      "   LYS 280 A    10.26      10.50                      \n",
      "   LYS 286 A    10.05      10.50                      \n",
      "   LYS 303 A    11.21      10.50                      \n",
      "   ARG  14 A    12.98      12.50                      \n",
      "   ARG  17 A    13.13      12.50                      \n",
      "   ARG  24 A    12.69      12.50                      \n",
      "   ARG  76 A    12.96      12.50                      \n",
      "   ARG 110 A    10.93      12.50                      \n",
      "   ARG 135 A    14.08      12.50                      \n",
      "   ARG 206 A    12.17      12.50                      \n",
      "   ARG 219 A    12.97      12.50                      \n",
      "   ARG 271 A    12.14      12.50                      \n",
      "   ARG 278 A    11.64      12.50                      \n",
      "   ARG 283 A    13.24      12.50                      \n",
      "   ARG 321 A    11.96      12.50                      \n",
      "   ARG 322 A    10.73      12.50                      \n",
      "   N+    1 A     7.87       8.00                      \n",
      "\n",
      "INFO:PROPKA information:\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "Free energy of   folding (kcal/mol) as a function of pH (using neutral reference)\n",
      "  0.00     34.30\n",
      "  1.00     33.99\n",
      "  2.00     32.12\n",
      "  3.00     25.92\n",
      "  4.00     18.89\n",
      "  5.00     20.28\n",
      "  6.00     24.08\n",
      "  7.00     26.09\n",
      "  8.00     28.36\n",
      "  9.00     32.60\n",
      " 10.00     42.82\n",
      " 11.00     57.76\n",
      " 12.00     71.32\n",
      " 13.00     81.81\n",
      " 14.00     89.66\n",
      "\n",
      "The pH of optimum stability is  4.3 for which the free energy is   18.4 kcal/mol at 298K\n",
      "Could not determine pH values where the free energy is within 80 % of minimum\n",
      "Could not determine the pH-range where the free energy is negative\n",
      "\n",
      "\n",
      "Protein charge of folded and unfolded state as a function of pH\n",
      "    pH  unfolded  folded\n",
      "  0.00     41.99   41.93\n",
      "  0.10     41.99   41.91\n",
      "  0.20     41.99   41.89\n",
      "  0.30     41.99   41.87\n",
      "  0.40     41.99   41.83\n",
      "  0.50     41.98   41.79\n",
      "  0.60     41.98   41.74\n",
      "  0.70     41.97   41.68\n",
      "  0.80     41.97   41.61\n",
      "  0.90     41.96   41.52\n",
      "  1.00     41.95   41.42\n",
      "  1.10     41.93   41.29\n",
      "  1.20     41.92   41.15\n",
      "  1.30     41.89   40.98\n",
      "  1.40     41.87   40.78\n",
      "  1.50     41.83   40.56\n",
      "  1.60     41.79   40.30\n",
      "  1.70     41.74   40.00\n",
      "  1.80     41.67   39.67\n",
      "  1.90     41.59   39.28\n",
      "  2.00     41.48   38.85\n",
      "  2.10     41.35   38.36\n",
      "  2.20     41.19   37.82\n",
      "  2.30     40.98   37.21\n",
      "  2.40     40.73   36.55\n",
      "  2.50     40.42   35.82\n",
      "  2.60     40.04   35.03\n",
      "  2.70     39.58   34.17\n",
      "  2.80     39.01   33.26\n",
      "  2.90     38.33   32.30\n",
      "  3.00     37.52   31.28\n",
      "  3.10     36.55   30.20\n",
      "  3.20     35.43   29.07\n",
      "  3.30     34.13   27.88\n",
      "  3.40     32.64   26.64\n",
      "  3.50     30.98   25.34\n",
      "  3.60     29.14   23.99\n",
      "  3.70     27.15   22.59\n",
      "  3.80     25.03   21.14\n",
      "  3.90     22.80   19.65\n",
      "  4.00     20.50   18.12\n",
      "  4.10     18.16   16.58\n",
      "  4.20     15.82   15.03\n",
      "  4.30     13.52   13.49\n",
      "  4.40     11.28   11.96\n",
      "  4.50      9.13   10.46\n",
      "  4.60      7.10    9.01\n",
      "  4.70      5.21    7.61\n",
      "  4.80      3.48    6.26\n",
      "  4.90      1.93    4.98\n",
      "  5.00      0.54    3.76\n",
      "  5.10     -0.69    2.62\n",
      "  5.20     -1.76    1.54\n",
      "  5.30     -2.70    0.52\n",
      "  5.40     -3.52   -0.43\n",
      "  5.50     -4.24   -1.33\n",
      "  5.60     -4.89   -2.18\n",
      "  5.70     -5.49   -2.99\n",
      "  5.80     -6.05   -3.76\n",
      "  5.90     -6.58   -4.51\n",
      "  6.00     -7.11   -5.23\n",
      "  6.10     -7.64   -5.93\n",
      "  6.20     -8.18   -6.60\n",
      "  6.30     -8.72   -7.24\n",
      "  6.40     -9.26   -7.86\n",
      "  6.50     -9.80   -8.44\n",
      "  6.60    -10.34   -8.98\n",
      "  6.70    -10.85   -9.48\n",
      "  6.80    -11.34   -9.94\n",
      "  6.90    -11.79  -10.36\n",
      "  7.00    -12.21  -10.74\n",
      "  7.10    -12.60  -11.08\n",
      "  7.20    -12.96  -11.39\n",
      "  7.30    -13.29  -11.69\n",
      "  7.40    -13.59  -11.96\n",
      "  7.50    -13.89  -12.23\n",
      "  7.60    -14.18  -12.49\n",
      "  7.70    -14.47  -12.75\n",
      "  7.80    -14.78  -13.01\n",
      "  7.90    -15.10  -13.27\n",
      "  8.00    -15.45  -13.54\n",
      "  8.10    -15.84  -13.82\n",
      "  8.20    -16.27  -14.10\n",
      "  8.30    -16.76  -14.39\n",
      "  8.40    -17.30  -14.69\n",
      "  8.50    -17.91  -15.00\n",
      "  8.60    -18.59  -15.33\n",
      "  8.70    -19.34  -15.67\n",
      "  8.80    -20.15  -16.04\n",
      "  8.90    -21.03  -16.42\n",
      "  9.00    -21.95  -16.84\n",
      "  9.10    -22.92  -17.29\n",
      "  9.20    -23.93  -17.77\n",
      "  9.30    -24.96  -18.30\n",
      "  9.40    -26.02  -18.88\n",
      "  9.50    -27.11  -19.51\n",
      "  9.60    -28.24  -20.20\n",
      "  9.70    -29.41  -20.95\n",
      "  9.80    -30.63  -21.77\n",
      "  9.90    -31.90  -22.67\n",
      " 10.00    -33.24  -23.64\n",
      " 10.10    -34.64  -24.67\n",
      " 10.20    -36.09  -25.77\n",
      " 10.30    -37.57  -26.92\n",
      " 10.40    -39.07  -28.12\n",
      " 10.50    -40.56  -29.34\n",
      " 10.60    -42.01  -30.59\n",
      " 10.70    -43.40  -31.84\n",
      " 10.80    -44.70  -33.09\n",
      " 10.90    -45.91  -34.32\n",
      " 11.00    -47.02  -35.53\n",
      " 11.10    -48.02  -36.72\n",
      " 11.20    -48.92  -37.88\n",
      " 11.30    -49.74  -39.01\n",
      " 11.40    -50.48  -40.11\n",
      " 11.50    -51.17  -41.18\n",
      " 11.60    -51.83  -42.22\n",
      " 11.70    -52.47  -43.23\n",
      " 11.80    -53.11  -44.22\n",
      " 11.90    -53.77  -45.19\n",
      " 12.00    -54.45  -46.13\n",
      " 12.10    -55.16  -47.06\n",
      " 12.20    -55.91  -47.96\n",
      " 12.30    -56.69  -48.85\n",
      " 12.40    -57.48  -49.72\n",
      " 12.50    -58.28  -50.57\n",
      " 12.60    -59.07  -51.42\n",
      " 12.70    -59.83  -52.25\n",
      " 12.80    -60.55  -53.07\n",
      " 12.90    -61.21  -53.86\n",
      " 13.00    -61.81  -54.64\n",
      " 13.10    -62.34  -55.39\n",
      " 13.20    -62.79  -56.10\n",
      " 13.30    -63.19  -56.77\n",
      " 13.40    -63.52  -57.41\n",
      " 13.50    -63.80  -58.00\n",
      " 13.60    -64.03  -58.55\n",
      " 13.70    -64.21  -59.05\n",
      " 13.80    -64.37  -59.52\n",
      " 13.90    -64.49  -59.95\n",
      " 14.00    -64.59  -60.34\n",
      "The pI is  5.35 (folded) and  5.04 (unfolded)\n",
      "\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "INFO:Applying pKa values at a pH of 7.30:\n",
      "INFO:Adding hydrogens to biomolecule.\n",
      "INFO:Debumping biomolecule (again).\n",
      "INFO:Optimizing hydrogen bonds\n",
      "INFO:Applying force field to biomolecule states.\n",
      "INFO:Regenerating headers.\n",
      "INFO:Regenerating PDB lines.\n",
      "WARNING:Ignoring 467 header lines in output.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "pH: 7.3\tname: 74987\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:PDB2PQR v3.6.2: biomolecular structure conversion software.\n",
      "INFO:Please cite:  Jurrus E, et al.  Improvements to the APBS biomolecular solvation software suite.  Protein Sci 27 112-128 (2018).\n",
      "INFO:Please cite:  Dolinsky TJ, et al.  PDB2PQR: expanding and upgrading automated preparation of biomolecular structures for molecular simulations. Nucleic Acids Res 35 W522-W525 (2007).\n",
      "INFO:Checking and transforming input arguments.\n",
      "INFO:Loading topology files.\n",
      "INFO:Loading molecule: /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/1ka1_hal2.pdb\n",
      "INFO:Setting up molecule.\n",
      "INFO:Created biomolecule object with 354 residues and 2718 atoms.\n",
      "WARNING:Multiple occupancies found: OG in SER A 84.\n",
      "WARNING:Multiple occupancies found in SER A 84. At least one of the instances is being ignored.\n",
      "INFO:Setting termini states for biomolecule chains.\n",
      "INFO:Loading forcefield.\n",
      "INFO:Loading hydrogen topology definitions.\n",
      "WARNING:Missing atom CB in residue LYS A 32\n",
      "WARNING:Missing atom CG in residue LYS A 32\n",
      "WARNING:Missing atom CD in residue LYS A 32\n",
      "WARNING:Missing atom CE in residue LYS A 32\n",
      "WARNING:Missing atom NZ in residue LYS A 32\n",
      "WARNING:Missing atom CB in residue ASP A 33\n",
      "WARNING:Missing atom CG in residue ASP A 33\n",
      "WARNING:Missing atom OD1 in residue ASP A 33\n",
      "WARNING:Missing atom OD2 in residue ASP A 33\n",
      "WARNING:Missing atom CG in residue LYS A 189\n",
      "WARNING:Missing atom CD in residue LYS A 189\n",
      "WARNING:Missing atom CE in residue LYS A 189\n",
      "WARNING:Missing atom NZ in residue LYS A 189\n",
      "WARNING:Missing atom CD in residue LYS A 229\n",
      "WARNING:Missing atom CE in residue LYS A 229\n",
      "WARNING:Missing atom NZ in residue LYS A 229\n",
      "WARNING:Missing atom OXT in residue ARG A 355\n",
      "WARNING:Missing atom CB in residue LYS A 32\n",
      "WARNING:Missing atom CG in residue LYS A 32\n",
      "WARNING:Missing atom CD in residue LYS A 32\n",
      "WARNING:Missing atom CE in residue LYS A 32\n",
      "WARNING:Missing atom NZ in residue LYS A 32\n",
      "WARNING:Missing atom CB in residue ASP A 33\n",
      "WARNING:Missing atom CG in residue ASP A 33\n",
      "WARNING:Missing atom OD1 in residue ASP A 33\n",
      "WARNING:Missing atom OD2 in residue ASP A 33\n",
      "WARNING:Missing atom CG in residue LYS A 189\n",
      "WARNING:Missing atom CD in residue LYS A 189\n",
      "WARNING:Missing atom CE in residue LYS A 189\n",
      "WARNING:Missing atom NZ in residue LYS A 189\n",
      "WARNING:Missing atom CD in residue LYS A 229\n",
      "WARNING:Missing atom CE in residue LYS A 229\n",
      "WARNING:Missing atom NZ in residue LYS A 229\n",
      "WARNING:Missing atom OXT in residue ARG A 355\n",
      "INFO:Attempting to repair 17 missing atoms in biomolecule.\n",
      "WARNING:Missing atom CB in residue LYS A 32\n",
      "WARNING:Missing atom CG in residue LYS A 32\n",
      "WARNING:Missing atom CD in residue LYS A 32\n",
      "WARNING:Missing atom CE in residue LYS A 32\n",
      "WARNING:Missing atom NZ in residue LYS A 32\n",
      "WARNING:Missing atom CB in residue ASP A 33\n",
      "WARNING:Missing atom CG in residue ASP A 33\n",
      "WARNING:Missing atom OD1 in residue ASP A 33\n",
      "WARNING:Missing atom OD2 in residue ASP A 33\n",
      "WARNING:Missing atom CG in residue LYS A 189\n",
      "WARNING:Missing atom CD in residue LYS A 189\n",
      "WARNING:Missing atom CE in residue LYS A 189\n",
      "WARNING:Missing atom NZ in residue LYS A 189\n",
      "WARNING:Missing atom CD in residue LYS A 229\n",
      "WARNING:Missing atom CE in residue LYS A 229\n",
      "WARNING:Missing atom NZ in residue LYS A 229\n",
      "WARNING:Missing atom OXT in residue ARG A 355\n",
      "INFO:Added atom CB to residue LYS A 32 at coordinates -45.594, -33.973, -23.265\n",
      "INFO:Added atom CG to residue LYS A 32 at coordinates -45.153, -35.334, -22.744\n",
      "INFO:Added atom CD to residue LYS A 32 at coordinates -43.702, -35.621, -23.091\n",
      "INFO:Added atom CE to residue LYS A 32 at coordinates -43.239, -36.968, -22.584\n",
      "INFO:Added atom NZ to residue LYS A 32 at coordinates -41.819, -37.204, -22.945\n",
      "INFO:Added atom CB to residue ASP A 33 at coordinates -49.706, -30.631, -25.302\n",
      "INFO:Added atom CG to residue ASP A 33 at coordinates -48.707, -31.048, -26.377\n",
      "INFO:Added atom OD1 to residue ASP A 33 at coordinates -47.538, -31.342, -26.061\n",
      "INFO:Added atom OD2 to residue ASP A 33 at coordinates -49.132, -31.087, -27.541\n",
      "INFO:Added atom CG to residue LYS A 189 at coordinates -35.105, -63.964, -9.656\n",
      "INFO:Added atom CD to residue LYS A 189 at coordinates -35.396, -65.193, -8.824\n",
      "INFO:Added atom CE to residue LYS A 189 at coordinates -34.533, -66.375, -9.200\n",
      "INFO:Added atom NZ to residue LYS A 189 at coordinates -34.854, -67.554, -8.358\n",
      "INFO:Added atom CD to residue LYS A 229 at coordinates -3.896, -48.760, -18.063\n",
      "INFO:Added atom CE to residue LYS A 229 at coordinates -3.416, -50.110, -17.576\n",
      "INFO:Added atom NZ to residue LYS A 229 at coordinates -2.176, -50.508, -18.287\n",
      "INFO:Added atom OXT to residue ARG A 355 at coordinates -9.183, -18.959, -17.852\n",
      "INFO:Updating disulfide bridges.\n",
      "INFO:Debumping biomolecule.\n",
      "INFO:Assigning titration states with PROPKA.\n",
      "INFO:\n",
      "propka3.5.1                                                                                  2024-07-26\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "--                                                                           --\n",
      "--  PROPKA: A PROTEIN PKA PREDICTOR                                          --\n",
      "--                                                                           --\n",
      "--  VERSION 1.0,  04/25/2004,  IOWA CITY                                     --\n",
      "--  BY HUI LI                                                                --\n",
      "--                                                                           --\n",
      "--  VERSION 2.0,  11/05/2007, IOWA CITY/COPENHAGEN                           --\n",
      "--  BY DELPHINE C. BAS AND DAVID M. ROGERS                                   --\n",
      "--                                                                           --\n",
      "--  VERSION 3.0,  01/06/2011, COPENHAGEN                                     --\n",
      "--  BY MATS H.M. OLSSON AND CHRESTEN R. SONDERGARD                           --\n",
      "--                                                                           --\n",
      "--  VERSION 3.1,  07/01/2011, COPENHAGEN                                     --\n",
      "--  BY CHRESTEN R. SONDERGARD AND MATS H.M. OLSSON                           --\n",
      "--                                                                           --\n",
      "--  VERSION 3.2,  06/17/2020, PLANET EARTH                                   --\n",
      "--  SEE ABOVE FOR AUTHORS                                                    --\n",
      "--                                                                           --\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "References:\n",
      "\n",
      "Very Fast Empirical Prediction and Rationalization of Protein pKa Values.\n",
      "Hui Li, Andrew D. Robertson and Jan H. Jensen. PROTEINS: Structure, Function,\n",
      "and Bioinformatics. 61:704-721 (2005)\n",
      "\n",
      "Very Fast Prediction and Rationalization of pKa Values for Protein-Ligand\n",
      "Complexes.  Delphine C. Bas, David M. Rogers and Jan H. Jensen.  PROTEINS:\n",
      "Structure, Function, and Bioinformatics 73:765-783 (2008)\n",
      "\n",
      "PROPKA3: Consistent Treatment of Internal and Surface Residues in Empirical\n",
      "pKa predictions.  Mats H.M. Olsson, Chresten R. Sondergard, Michal Rostkowski,\n",
      "and Jan H. Jensen.  Journal of Chemical Theory and Computation, 7(2):525-537\n",
      "(2011)\n",
      "\n",
      "Improved Treatment of Ligands and Coupling Effects in Empirical Calculation\n",
      "and Rationalization of pKa Values.  Chresten R. Sondergaard, Mats H.M. Olsson,\n",
      "Michal Rostkowski, and Jan H. Jensen.  Journal of Chemical Theory and\n",
      "Computation, (2011)\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "\n",
      "WARNING:Missing atoms or failed protonation for HIS  31 A (HIS) -- please check the structure\n",
      "WARNING:Group (HIS) for   231-  CG    31-HIS (A) [ -51.746  -37.773  -23.250] C\n",
      "WARNING:Expected 4 interaction atoms for acids, found:\n",
      "WARNING:                 0- HE2    31-HIS (A) [ -54.068  -39.642  -24.334] H\n",
      "WARNING:               233- ND1    31-HIS (A) [ -51.129  -38.973  -23.536] N\n",
      "WARNING:               235- NE2    31-HIS (A) [ -53.201  -39.225  -24.025] N\n",
      "WARNING:Expected 2 interaction atoms for bases, found:\n",
      "WARNING:               233- ND1    31-HIS (A) [ -51.129  -38.973  -23.536] N\n",
      "WARNING:               235- NE2    31-HIS (A) [ -53.201  -39.225  -24.025] N\n",
      "INFO:Coupling map:\n",
      "\n",
      "INFO:Calculating pKas for Conformation container 1A with 3249 atoms and 890 groups\n",
      "INFO:pKa determinants:\n",
      "\n",
      "---------  -----   ------   ---------------------    --------------    --------------    --------------\n",
      "                            DESOLVATION  EFFECTS       SIDECHAIN          BACKBONE        COULOMBIC\n",
      " RESIDUE    pKa    BURIED     REGULAR      RE        HYDROGEN BOND     HYDROGEN BOND      INTERACTION\n",
      "---------  -----   ------   ---------   ---------    --------------    --------------    --------------\n",
      "\n",
      "ASP  33 A   3.91     0 %    0.20  152   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS  32 A\n",
      "ASP  33 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 HIS  31 A\n",
      "\n",
      "ASP  41 A   2.21    11 %    0.60  312   0.00    0   -0.79 SER  43 A   -0.23 ASP  41 A   -0.28 LYS  39 A\n",
      "ASP  41 A                                            0.00 XXX   0 X   -0.73 SER  43 A   -0.04 LYS 148 A\n",
      "ASP  41 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.12 LYS 291 A\n",
      "ASP  41 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 HIS 241 A\n",
      "\n",
      "ASP  49 A   6.17*  100 %    3.70  606   1.05    0   -0.84 THR  46 A   -0.70 THR  46 A   -0.30 LYS  39 A\n",
      "ASP  49 A                                           -0.85 THR 147 A   -0.24 ASP  49 A    0.55 GLU  72 A\n",
      "\n",
      "ASP  66 A   3.97     0 %    0.19  204   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.11 LYS  61 A\n",
      "ASP  66 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP  67 A\n",
      "\n",
      "ASP  67 A   3.50    45 %    1.80  406   0.24    0   -0.81 TRP 139 A    0.00 XXX   0 X   -0.03 ARG   5 A\n",
      "ASP  67 A                                           -0.70 ARG 137 A    0.00 XXX   0 X   -0.15 LYS  61 A\n",
      "ASP  67 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 LYS  68 A\n",
      "ASP  67 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.57 ARG 137 A\n",
      "\n",
      "ASP  80 A   3.08     0 %    0.24  175   0.00    0   -0.70 ARG 122 A    0.00 XXX   0 X   -0.26 ARG 122 A\n",
      "\n",
      "ASP  94 A   2.91    69 %    1.87  474   0.67    0   -0.85 LYS  16 A    0.00 XXX   0 X   -0.00 ARG  23 A\n",
      "ASP  94 A                                           -0.85 LYS  91 A    0.00 XXX   0 X   -0.00 LYS 116 A\n",
      "ASP  94 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.05 LYS  16 A\n",
      "ASP  94 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.67 LYS  91 A\n",
      "\n",
      "ASP 104 A   3.88     0 %    0.20  138   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.04 LYS 102 A\n",
      "ASP 104 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 LYS 103 A\n",
      "\n",
      "ASP 105 A   4.10     0 %    0.13  159   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG   5 A\n",
      "ASP 105 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS 103 A\n",
      "ASP 105 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.21 ASP 104 A\n",
      "\n",
      "ASP 111 A   3.81     0 %    0.12  140   0.00    0    0.00 XXX   0 X   -0.11 ASP 111 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 120 A   2.75    13 %    0.72  318   0.05    0   -0.32 SER 117 A   -0.79 LYS 116 A   -0.02 LYS 116 A\n",
      "ASP 120 A                                            0.00 XXX   0 X   -0.69 SER 117 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 126 A   3.39    34 %    1.22  376   0.28    0   -0.16 SER  74 A   -0.85 SER  75 A   -0.06 ARG 122 A\n",
      "ASP 126 A                                           -0.85 SER  75 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "ASP 142 A  16.43*  100 %    3.57  642   1.03    0    0.48 GLU  72 A   -0.03 ILE 144 A   -0.28 LYS  39 A\n",
      "ASP 142 A                                            0.67 ASP 145 A    0.00 XXX   0 X    0.50 ASP  49 A\n",
      "ASP 142 A                                            0.37 ASP 294 A    0.00 XXX   0 X    0.17 CYS 140 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.88 CYS 159 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.87 GLU  72 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.59 GLU  73 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.57 ASP 145 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.23 ASP 294 A\n",
      "\n",
      "ASP 145 A   3.62*  100 %    2.82  672   0.55    0   -0.67 ASP 142 A   -0.04 GLY 146 A   -0.19 HIS 241 A\n",
      "ASP 145 A                                           -0.73 ASP 294 A   -0.57 GLN 265 A    0.17 ASP 263 A\n",
      "ASP 145 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 GLU 290 A\n",
      "ASP 145 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.44 HIS 295 A\n",
      "ASP 145 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.65 LYS 267 A\n",
      "ASP 145 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.55 ARG 281 A\n",
      "\n",
      "ASP 165 A   3.15     0 %    0.28  211   0.00    0    0.00 XXX   0 X   -0.81 GLY 136 A   -0.03 LYS  68 A\n",
      "ASP 165 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.00 ARG 134 A\n",
      "ASP 165 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 LYS 135 A\n",
      "\n",
      "ASP 187 A   2.15    10 %    1.13  308   0.05    0   -0.17 SER 181 A   -0.76 LEU 180 A   -0.41 HIS 191 A\n",
      "ASP 187 A                                           -0.85 TYR 196 A   -0.63 SER 181 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 213 A   3.05    18 %    0.79  331   0.14    0    0.00 XXX   0 X   -0.76 GLU 215 A   -0.09 ARG  15 A\n",
      "ASP 213 A                                            0.00 XXX   0 X   -0.84 SER 216 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 227 A   3.07     0 %    0.35  211   0.00    0    0.00 XXX   0 X   -0.84 LYS 229 A   -0.24 LYS 229 A\n",
      "\n",
      "ASP 230 A   3.96     0 %    0.30  181   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.03 ARG 223 A\n",
      "ASP 230 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.20 LYS 229 A\n",
      "ASP 230 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.09 LYS 258 A\n",
      "ASP 230 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP 227 A\n",
      "ASP 230 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 277 A\n",
      "\n",
      "ASP 245 A   3.84     0 %    0.18  234   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.09 LYS 239 A\n",
      "ASP 245 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 HIS 244 A\n",
      "ASP 245 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 GLU 246 A\n",
      "\n",
      "ASP 263 A   3.38    62 %    2.13  455   0.38    0   -1.55 ARG 152 A   -0.09 ASP 263 A   -0.03 LYS 148 A\n",
      "ASP 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.50 LYS 267 A\n",
      "ASP 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.20 HIS 241 A\n",
      "ASP 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.57 ARG 152 A\n",
      "\n",
      "ASP 277 A   3.47    56 %    2.43  438   0.35    0   -1.64 ARG 223 A   -0.78 ILE 232 A   -0.04 LYS 258 A\n",
      "ASP 277 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 HIS 224 A\n",
      "ASP 277 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.61 ARG 223 A\n",
      "\n",
      "ASP 294 A   9.23*  100 %    2.63  612   0.38    0   -0.37 ASP 142 A    0.00 XXX   0 X   -0.04 LYS  39 A\n",
      "ASP 294 A                                            0.73 ASP 145 A    0.00 XXX   0 X   -0.08 LYS 267 A\n",
      "ASP 294 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.35 LYS 291 A\n",
      "ASP 294 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.93 GLU  72 A\n",
      "ASP 294 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.64 ASP 145 A\n",
      "ASP 294 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.87 ARG 281 A\n",
      "ASP 294 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.84 GLU 290 A\n",
      "\n",
      "ASP 311 A   3.81    51 %    2.24  424   0.49    0    0.00 XXX   0 X   -0.12 ASP 311 A   -0.05 HIS 341 A\n",
      "ASP 311 A                                            0.00 XXX   0 X   -0.61 MET 313 A    0.00 XXX   0 X\n",
      "ASP 311 A                                            0.00 XXX   0 X   -0.83 GLU 314 A    0.00 XXX   0 X\n",
      "ASP 311 A                                            0.00 XXX   0 X   -0.40 ASP 315 A    0.00 XXX   0 X\n",
      "ASP 311 A                                            0.00 XXX   0 X   -0.72 VAL 316 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 315 A   6.34*   20 %    0.96  336   0.20    0    0.87 ASP 342 A    0.00 XXX   0 X   -0.04 ARG 338 A\n",
      "ASP 315 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.14 ASP 311 A\n",
      "ASP 315 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.40 ASP 342 A\n",
      "\n",
      "ASP 319 A   1.83    15 %    0.73  324   0.03    0   -1.58 ARG 134 A   -0.76 GLY 321 A   -0.38 ARG 134 A\n",
      "\n",
      "ASP 342 A   3.08*    0 %    0.48  257   0.00    0   -0.87 ASP 315 A    0.00 XXX   0 X   -0.15 ARG 338 A\n",
      "ASP 342 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.18 HIS 341 A\n",
      "\n",
      "ASP 350 A   3.76     0 %    0.21  206   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.25 LYS 253 A\n",
      "\n",
      "GLU   4 A   2.92     0 %    0.28  187   0.00    0   -0.77 N+    2 A   -0.70 GLU   4 A   -0.38 N+    2 A\n",
      "\n",
      "GLU   6 A   3.84    24 %    1.04  349   0.00    0   -0.65 ARG 137 A   -0.66 GLU   6 A   -0.05 N+    2 A\n",
      "GLU   6 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.12 ARG   5 A\n",
      "GLU   6 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU   4 A\n",
      "GLU   6 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.14 ASP  67 A\n",
      "GLU   6 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.43 ARG 137 A\n",
      "\n",
      "GLU  27 A   4.11    13 %    0.87  318   0.20    0   -0.85 SER  26 A   -0.45 GLU  27 A   -0.00 LYS  22 A\n",
      "GLU  27 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.09 ARG  23 A\n",
      "GLU  27 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.07 HIS  31 A\n",
      "\n",
      "GLU  72 A   5.47*  100 %    2.36  562   0.17    0   -0.34 LYS  39 A    0.00 XXX   0 X   -0.16 LYS 291 A\n",
      "GLU  72 A                                           -0.48 ASP 142 A    0.00 XXX   0 X   -0.05 ARG 324 A\n",
      "GLU  72 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.14 ASP  41 A\n",
      "GLU  72 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.42 ASP 145 A\n",
      "GLU  72 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.10 LYS  39 A\n",
      "\n",
      "GLU  73 A   8.38*   90 %    2.95  532   0.08    0   -0.85 LYS  39 A    0.00 XXX   0 X   -0.01 ARG 324 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 ASP  41 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.00 LYS  39 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.09 ASP  49 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.55 GLU  72 A\n",
      "\n",
      "GLU  89 A   5.51    46 %    1.01  409   0.11    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.20 ARG  23 A\n",
      "GLU  89 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.12 GLU  27 A\n",
      "GLU  89 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 HIS  31 A\n",
      "\n",
      "GLU  95 A   3.70     0 %    0.31  175   0.00    0   -0.68 LYS  99 A    0.00 XXX   0 X   -0.05 ARG  23 A\n",
      "GLU  95 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.38 LYS  99 A\n",
      "\n",
      "GLU 119 A   4.30     0 %    0.29  205   0.00    0    0.00 XXX   0 X   -0.63 GLU 119 A   -0.10 ARG 122 A\n",
      "GLU 119 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 ASP  80 A\n",
      "GLU 119 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 ASP 120 A\n",
      "\n",
      "GLU 131 A   4.45     1 %    0.36  283   0.01    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.41 ARG 324 A\n",
      "\n",
      "GLU 154 A   4.04     2 %    0.35  287   0.01    0   -0.72 ARG 152 A    0.00 XXX   0 X    0.02 GLU 238 A\n",
      "GLU 154 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.18 ASP 263 A\n",
      "GLU 154 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.31 ARG 152 A\n",
      "\n",
      "GLU 192 A   3.88     0 %    0.35  255   0.00    0    0.00 XXX   0 X   -0.04 GLY 190 A    0.02 ASP 187 A\n",
      "GLU 192 A                                            0.00 XXX   0 X   -0.73 GLU 192 A   -0.23 HIS 191 A\n",
      "\n",
      "GLU 215 A   3.63    10 %    0.43  309   0.07    0   -0.85 LYS 103 A    0.00 XXX   0 X   -0.16 ARG  15 A\n",
      "GLU 215 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.05 ARG 199 A\n",
      "GLU 215 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP 213 A\n",
      "GLU 215 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.41 LYS 103 A\n",
      "\n",
      "GLU 235 A   3.26    68 %    2.28  472   0.12    0   -0.46 THR 233 A    0.00 XXX   0 X   -0.30 HIS 244 A\n",
      "GLU 235 A                                           -0.85 SER 259 A    0.00 XXX   0 X   -0.65 LYS 251 A\n",
      "GLU 235 A                                           -0.85 LYS 251 A    0.00 XXX   0 X   -0.53 HIS 261 A\n",
      "\n",
      "GLU 238 A   3.63    15 %    0.62  323   0.00    0   -0.74 ARG 152 A    0.00 XXX   0 X   -0.35 LYS 148 A\n",
      "GLU 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.30 HIS 241 A\n",
      "GLU 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 HIS 244 A\n",
      "GLU 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.24 ASP 263 A\n",
      "GLU 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.33 ARG 152 A\n",
      "\n",
      "GLU 246 A   2.95    20 %    1.10  337   0.12    0   -0.73 SER 243 A    0.00 XXX   0 X   -0.43 ARG 355 A\n",
      "GLU 246 A                                           -1.62 ARG 355 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "GLU 290 A   1.29    98 %    3.05  556   0.08    0   -0.59 ARG 281 A    0.00 XXX   0 X   -0.00 LYS 267 A\n",
      "GLU 290 A                                           -1.60 HIS 295 A    0.00 XXX   0 X   -0.23 LYS 291 A\n",
      "GLU 290 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 LYS 329 A\n",
      "GLU 290 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.10 HIS 241 A\n",
      "GLU 290 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -2.03 ARG 281 A\n",
      "GLU 290 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.70 HIS 295 A\n",
      "\n",
      "GLU 304 A   3.18    26 %    1.06  354   0.21    0   -0.46 HIS 303 A   -0.84 GLY 203 A   -0.19 ARG 202 A\n",
      "GLU 304 A                                            0.00 XXX   0 X   -0.00 LEU 204 A   -0.05 LYS 219 A\n",
      "GLU 304 A                                            0.00 XXX   0 X   -0.74 GLY 205 A   -0.31 HIS 303 A\n",
      "\n",
      "GLU 314 A   5.06     0 %    0.26  273   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.01 LYS 329 A\n",
      "GLU 314 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.31 ASP 311 A\n",
      "\n",
      "GLU 339 A   4.02     0 %    0.19  164   0.00    0    0.00 XXX   0 X   -0.67 GLU 339 A   -0.07 ARG 338 A\n",
      "GLU 339 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP 342 A\n",
      "\n",
      "C-  355 A   3.34     0 %    0.16  125   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.07 ARG 355 A\n",
      "C-  355 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 246 A\n",
      "\n",
      "HIS  31 A   6.33     0 %   -0.29  196   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.07 GLU  27 A\n",
      "HIS  31 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP  33 A\n",
      "HIS  31 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU  89 A\n",
      "\n",
      "HIS 191 A   6.82     6 %   -0.67  298   0.00    0    0.00 XXX   0 X    0.37 LEU 188 A   -0.02 LYS  22 A\n",
      "HIS 191 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.41 ASP 187 A\n",
      "HIS 191 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.23 GLU 192 A\n",
      "\n",
      "HIS 221 A   5.75    12 %   -0.53  315   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 219 A\n",
      "HIS 221 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.17 HIS 224 A\n",
      "\n",
      "HIS 224 A   6.57    11 %   -0.55  311   0.00    0    0.00 XXX   0 X    0.58 SER 335 A    0.04 ASP 277 A\n",
      "\n",
      "HIS 241 A   4.98    65 %   -1.23  464   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.26 LYS 148 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 ARG 152 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.17 LYS 267 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.56 ARG 281 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP  41 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 ASP 145 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.30 GLU 238 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 HIS 244 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 ASP 263 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 GLU 290 A\n",
      "\n",
      "HIS 244 A   5.80    28 %   -0.91  359   0.00    0    0.00 XXX   0 X    0.04 GLY 236 A   -0.09 LYS 239 A\n",
      "HIS 244 A                                            0.00 XXX   0 X    0.22 VAL 237 A   -0.10 LYS 251 A\n",
      "HIS 244 A                                            0.00 XXX   0 X    0.01 SER 242 A   -0.01 LYS 267 A\n",
      "HIS 244 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.30 GLU 235 A\n",
      "HIS 244 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 238 A\n",
      "HIS 244 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP 245 A\n",
      "HIS 244 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.27 HIS 261 A\n",
      "\n",
      "HIS 261 A   6.60    35 %   -0.93  379   0.00    0    0.00 XXX   0 X    0.85 GLY 236 A   -0.29 LYS 251 A\n",
      "HIS 261 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 267 A\n",
      "HIS 261 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.53 GLU 235 A\n",
      "\n",
      "HIS 295 A   5.60   100 %   -3.33  658   0.00    0    1.60 GLU 290 A    0.00 XXX   0 X   -0.15 LYS 267 A\n",
      "HIS 295 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.16 ARG 281 A\n",
      "HIS 295 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.00 LYS 291 A\n",
      "HIS 295 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.44 ASP 145 A\n",
      "HIS 295 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.70 GLU 290 A\n",
      "\n",
      "HIS 303 A   6.22    34 %   -0.66  377   0.00    0    0.46 GLU 304 A    0.00 XXX   0 X   -0.39 ARG 202 A\n",
      "HIS 303 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.31 GLU 304 A\n",
      "\n",
      "HIS 309 A   2.56   100 %   -3.24  632   0.00    0    0.00 XXX   0 X    0.06 HIS 295 A   -0.74 HIS 295 A\n",
      "HIS 309 A                                            0.00 XXX   0 X    0.23 ILE 332 A   -0.25 HIS 341 A\n",
      "\n",
      "HIS 341 A   4.22    77 %   -2.52  498   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 311 A\n",
      "HIS 341 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.18 ASP 342 A\n",
      "\n",
      "CYS 140 A  12.78   100 %    3.75  624   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.13 ARG 324 A\n",
      "CYS 140 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 ASP 294 A\n",
      "\n",
      "CYS 159 A  15.19   100 %    3.86  700   0.00    0   -0.06 GLN 265 A    0.00 XXX   0 X   -0.03 LYS 267 A\n",
      "CYS 159 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 CYS 140 A\n",
      "CYS 159 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 CYS 172 A\n",
      "CYS 159 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.30 ASP 294 A\n",
      "CYS 159 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.32 ASP 145 A\n",
      "CYS 159 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.53 CYS 175 A\n",
      "\n",
      "CYS 172 A  12.08   100 %    3.66  715   0.00    0   -0.14 THR  11 A    0.00 XXX   0 X   -0.50 ARG 199 A\n",
      "CYS 172 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 CYS 175 A\n",
      "\n",
      "CYS 175 A   9.09*  100 %    3.67  714   0.00    0   -3.60 CYS 269 A    0.00 XXX   0 X    0.02 ASP 145 A\n",
      "\n",
      "CYS 269 A  19.48*  100 %    4.24  712   0.00    0    3.60 CYS 175 A   -0.02 CYS 269 A    0.49 CYS 159 A\n",
      "CYS 269 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 CYS 172 A\n",
      "CYS 269 A                                            0.00 XXX   0 X    0.00 XXX   0 X    2.03 CYS 175 A\n",
      "\n",
      "CYS 349 A   9.57     1 %    0.40  284   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 311 A\n",
      "CYS 349 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 314 A\n",
      "CYS 349 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.11 ASP 350 A\n",
      "\n",
      "TYR  50 A  12.47    60 %    1.83  448   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.23 ASP  49 A\n",
      "TYR  50 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU  72 A\n",
      "TYR  50 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 126 A\n",
      "TYR  50 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.29 LYS  39 A\n",
      "TYR  50 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.60 GLU  73 A\n",
      "\n",
      "TYR  97 A  12.15    89 %    3.48  531   0.00    0   -0.35 ARG  15 A    0.00 XXX   0 X   -1.07 ARG  15 A\n",
      "TYR  97 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP  94 A\n",
      "TYR  97 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 215 A\n",
      "TYR  97 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 TYR 101 A\n",
      "\n",
      "TYR 101 A  10.24    26 %    0.81  354   0.00    0   -0.53 SER 193 A    0.00 XXX   0 X   -0.10 ARG  15 A\n",
      "TYR 101 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 GLU 192 A\n",
      "\n",
      "TYR 130 A  10.10     4 %    0.46  293   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.36 LYS  68 A\n",
      "\n",
      "TYR 183 A  11.75    61 %    1.58  453   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 154 A\n",
      "TYR 183 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 CYS 175 A\n",
      "TYR 183 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP 187 A\n",
      "TYR 183 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 263 A\n",
      "\n",
      "TYR 196 A  13.17    38 %    2.33  389   0.00    0    0.85 ASP 187 A   -0.79 LEU 188 A    0.05 GLU 192 A\n",
      "TYR 196 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.27 TYR 208 A\n",
      "TYR 196 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.48 ASP 187 A\n",
      "\n",
      "TYR 208 A  11.41    46 %    1.34  411   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.07 ASP 187 A\n",
      "\n",
      "TYR 268 A  17.93   100 %    4.86  684   0.00    0    0.77 TYR 279 A    0.00 XXX   0 X    0.17 ASP 145 A\n",
      "TYR 268 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 CYS 159 A\n",
      "TYR 268 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.54 ARG 281 A\n",
      "TYR 268 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.46 GLU 290 A\n",
      "TYR 268 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 ASP 294 A\n",
      "TYR 268 A                                            0.00 XXX   0 X    0.00 XXX   0 X    2.03 TYR 279 A\n",
      "\n",
      "TYR 279 A  12.08   100 %    3.49  678   0.00    0   -0.51 ARG 281 A    0.00 XXX   0 X    0.50 ASP 145 A\n",
      "TYR 279 A                                           -0.77 TYR 268 A    0.00 XXX   0 X    0.01 CYS 159 A\n",
      "TYR 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.75 ARG 281 A\n",
      "TYR 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.36 ASP 294 A\n",
      "TYR 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.76 GLU 290 A\n",
      "\n",
      "TYR 288 A  12.91    68 %    1.44  472   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP  41 A\n",
      "TYR 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.17 GLU  72 A\n",
      "TYR 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP 142 A\n",
      "TYR 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 ASP 145 A\n",
      "TYR 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.51 ARG 281 A\n",
      "TYR 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.80 GLU 290 A\n",
      "TYR 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.85 ASP 294 A\n",
      "\n",
      "LYS  16 A   9.20    87 %   -2.94  524   0.00    0    0.85 ASP  94 A    0.00 XXX   0 X   -0.26 LYS  91 A\n",
      "LYS  16 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.05 ASP  94 A\n",
      "\n",
      "LYS  22 A   9.53    44 %   -0.97  405   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 GLU  27 A\n",
      "\n",
      "LYS  32 A  10.25     0 %   -0.31  236   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.06 ASP  33 A\n",
      "\n",
      "LYS  39 A  13.17    62 %   -1.79  454   0.00    0    0.34 GLU  72 A    0.00 XXX   0 X    0.28 ASP  41 A\n",
      "LYS  39 A                                            0.85 GLU  73 A    0.00 XXX   0 X    0.30 ASP  49 A\n",
      "LYS  39 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.28 ASP 142 A\n",
      "LYS  39 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 294 A\n",
      "LYS  39 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 324 A\n",
      "LYS  39 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.29 TYR  50 A\n",
      "LYS  39 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.10 GLU  72 A\n",
      "LYS  39 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.00 GLU  73 A\n",
      "\n",
      "LYS  61 A   9.38    37 %   -1.38  385   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.11 ASP  66 A\n",
      "LYS  61 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP  67 A\n",
      "\n",
      "LYS  68 A  10.47     6 %   -0.46  298   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP  67 A\n",
      "LYS  68 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 ARG 137 A\n",
      "LYS  68 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 165 A\n",
      "LYS  68 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.36 TYR 130 A\n",
      "\n",
      "LYS  91 A  11.20    36 %   -0.82  382   0.00    0    0.85 ASP  94 A    0.00 XXX   0 X    0.67 ASP  94 A\n",
      "\n",
      "LYS  99 A  11.34     0 %   -0.22  141   0.00    0    0.68 GLU  95 A    0.00 XXX   0 X   -0.00 ARG  23 A\n",
      "LYS  99 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 GLU  95 A\n",
      "\n",
      "LYS 102 A  10.42     0 %   -0.12  190   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 104 A\n",
      "\n",
      "LYS 103 A  11.30     6 %   -0.45  298   0.00    0    0.85 GLU 215 A    0.00 XXX   0 X   -0.09 ARG  15 A\n",
      "LYS 103 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP 104 A\n",
      "LYS 103 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 105 A\n",
      "LYS 103 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 199 A\n",
      "LYS 103 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.41 GLU 215 A\n",
      "\n",
      "LYS 116 A  10.23     0 %   -0.12  213   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP  94 A\n",
      "LYS 116 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 120 A\n",
      "LYS 116 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.18 LYS  91 A\n",
      "\n",
      "LYS 135 A  10.54     0 %   -0.04   82   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP 165 A\n",
      "\n",
      "LYS 148 A  10.50    12 %   -0.36  314   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP  41 A\n",
      "LYS 148 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.05 ARG 152 A\n",
      "LYS 148 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.35 GLU 238 A\n",
      "LYS 148 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 263 A\n",
      "\n",
      "LYS 189 A  10.46     0 %   -0.04  110   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 219 A  10.37     0 %   -0.19  220   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 304 A\n",
      "\n",
      "LYS 226 A  10.46     0 %   -0.04   96   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 229 A  10.85     0 %   -0.10  111   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.24 ASP 227 A\n",
      "LYS 229 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 ASP 230 A\n",
      "\n",
      "LYS 239 A  10.51     0 %   -0.08  150   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP 245 A\n",
      "\n",
      "LYS 251 A  10.84    33 %   -1.16  374   0.00    0    0.85 GLU 235 A    0.00 XXX   0 X    0.65 GLU 235 A\n",
      "\n",
      "LYS 253 A  10.54     0 %   -0.21  219   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.25 ASP 350 A\n",
      "\n",
      "LYS 258 A  10.36     0 %   -0.17  226   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.10 ARG 223 A\n",
      "LYS 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP 230 A\n",
      "LYS 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 277 A\n",
      "\n",
      "LYS 267 A   7.64   100 %   -3.32  608   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.00 ARG 152 A\n",
      "LYS 267 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 CYS 159 A\n",
      "LYS 267 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.50 ASP 263 A\n",
      "LYS 267 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.79 ARG 281 A\n",
      "LYS 267 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 GLU 290 A\n",
      "LYS 267 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP 294 A\n",
      "LYS 267 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.65 ASP 145 A\n",
      "\n",
      "LYS 285 A  10.40     0 %   -0.10  190   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 291 A  10.01    47 %   -1.26  413   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.12 ASP  41 A\n",
      "LYS 291 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 GLU  72 A\n",
      "LYS 291 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 ARG 281 A\n",
      "LYS 291 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.23 GLU 290 A\n",
      "LYS 291 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.35 ASP 294 A\n",
      "LYS 291 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.05 LYS  39 A\n",
      "\n",
      "LYS 329 A   9.59    18 %   -0.97  331   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU 290 A\n",
      "LYS 329 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 314 A\n",
      "LYS 329 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS 291 A\n",
      "\n",
      "ARG   5 A  12.07    10 %   -0.57  309   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.12 GLU   6 A\n",
      "ARG   5 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP  67 A\n",
      "ARG   5 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 105 A\n",
      "ARG   5 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 ARG 137 A\n",
      "\n",
      "ARG  15 A  12.14    68 %   -2.12  473   0.00    0    0.35 TYR  97 A    0.00 XXX   0 X    1.07 TYR  97 A\n",
      "ARG  15 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 TYR 101 A\n",
      "ARG  15 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP 213 A\n",
      "ARG  15 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 GLU 215 A\n",
      "\n",
      "ARG  23 A  11.87    37 %   -0.97  384   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.09 GLU  27 A\n",
      "ARG  23 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 GLU  89 A\n",
      "ARG  23 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP  94 A\n",
      "ARG  23 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU  95 A\n",
      "\n",
      "ARG 122 A  13.16     0 %   -0.45  261   0.00    0    0.70 ASP  80 A    0.00 XXX   0 X    0.10 GLU 119 A\n",
      "ARG 122 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 ASP 126 A\n",
      "ARG 122 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.26 ASP  80 A\n",
      "\n",
      "ARG 134 A  14.14     0 %   -0.33  237   0.00    0    1.58 ASP 319 A    0.00 XXX   0 X    0.00 ASP 165 A\n",
      "ARG 134 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 ASP 319 A\n",
      "\n",
      "ARG 137 A  13.63    35 %   -1.22  380   0.00    0    0.65 GLU   6 A    0.00 XXX   0 X    0.43 GLU   6 A\n",
      "ARG 137 A                                            0.70 ASP  67 A    0.00 XXX   0 X    0.57 ASP  67 A\n",
      "\n",
      "ARG 152 A  15.85    18 %   -0.87  332   0.00    0    0.72 GLU 154 A    0.00 XXX   0 X    0.31 GLU 154 A\n",
      "ARG 152 A                                            0.74 GLU 238 A    0.00 XXX   0 X    0.33 GLU 238 A\n",
      "ARG 152 A                                            1.55 ASP 263 A    0.00 XXX   0 X    0.57 ASP 263 A\n",
      "\n",
      "ARG 199 A  10.80    65 %   -2.25  462   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.50 CYS 172 A\n",
      "ARG 199 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 215 A\n",
      "\n",
      "ARG 202 A  11.55    41 %   -1.14  396   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.19 GLU 304 A\n",
      "\n",
      "ARG 223 A  13.63    34 %   -1.14  377   0.00    0    1.64 ASP 277 A    0.00 XXX   0 X    0.03 ASP 230 A\n",
      "ARG 223 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.61 ASP 277 A\n",
      "\n",
      "ARG 281 A  17.71   100 %   -2.15  598   0.00    0    0.51 TYR 279 A    0.00 XXX   0 X    0.54 TYR 268 A\n",
      "ARG 281 A                                            0.59 GLU 290 A    0.00 XXX   0 X    1.75 TYR 279 A\n",
      "ARG 281 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.51 TYR 288 A\n",
      "ARG 281 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.55 ASP 145 A\n",
      "ARG 281 A                                            0.00 XXX   0 X    0.00 XXX   0 X    2.03 GLU 290 A\n",
      "ARG 281 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.87 ASP 294 A\n",
      "\n",
      "ARG 324 A  11.59    54 %   -1.52  433   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU  72 A\n",
      "ARG 324 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU  73 A\n",
      "ARG 324 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.41 GLU 131 A\n",
      "ARG 324 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 CYS 140 A\n",
      "\n",
      "ARG 338 A  12.69     0 %   -0.07  146   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 315 A\n",
      "ARG 338 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 GLU 339 A\n",
      "ARG 338 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP 342 A\n",
      "\n",
      "ARG 355 A  14.03     6 %   -0.58  299   0.00    0    1.62 GLU 246 A    0.00 XXX   0 X    0.07 C-  355 A\n",
      "ARG 355 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.43 GLU 246 A\n",
      "\n",
      "N+    2 A   8.88     0 %   -0.32  186   0.00    0    0.77 GLU   4 A    0.00 XXX   0 X    0.05 GLU   6 A\n",
      "N+    2 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 GLU   4 A\n",
      "\n",
      "Coupled residues (marked *) were detected.Please rerun PropKa with the --display-coupled-residues \n",
      "or -d option for detailed information.\n",
      "\n",
      "INFO:pKa summary:\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "SUMMARY OF THIS PREDICTION\n",
      "       Group      pKa  model-pKa   ligand atom-type\n",
      "   ASP  33 A     3.91       3.80                      \n",
      "   ASP  41 A     2.21       3.80                      \n",
      "   ASP  49 A     6.17       3.80                      \n",
      "   ASP  66 A     3.97       3.80                      \n",
      "   ASP  67 A     3.50       3.80                      \n",
      "   ASP  80 A     3.08       3.80                      \n",
      "   ASP  94 A     2.91       3.80                      \n",
      "   ASP 104 A     3.88       3.80                      \n",
      "   ASP 105 A     4.10       3.80                      \n",
      "   ASP 111 A     3.81       3.80                      \n",
      "   ASP 120 A     2.75       3.80                      \n",
      "   ASP 126 A     3.39       3.80                      \n",
      "   ASP 142 A    16.43       3.80                      \n",
      "   ASP 145 A     3.62       3.80                      \n",
      "   ASP 165 A     3.15       3.80                      \n",
      "   ASP 187 A     2.15       3.80                      \n",
      "   ASP 213 A     3.05       3.80                      \n",
      "   ASP 227 A     3.07       3.80                      \n",
      "   ASP 230 A     3.96       3.80                      \n",
      "   ASP 245 A     3.84       3.80                      \n",
      "   ASP 263 A     3.38       3.80                      \n",
      "   ASP 277 A     3.47       3.80                      \n",
      "   ASP 294 A     9.23       3.80                      \n",
      "   ASP 311 A     3.81       3.80                      \n",
      "   ASP 315 A     6.34       3.80                      \n",
      "   ASP 319 A     1.83       3.80                      \n",
      "   ASP 342 A     3.08       3.80                      \n",
      "   ASP 350 A     3.76       3.80                      \n",
      "   GLU   4 A     2.92       4.50                      \n",
      "   GLU   6 A     3.84       4.50                      \n",
      "   GLU  27 A     4.11       4.50                      \n",
      "   GLU  72 A     5.47       4.50                      \n",
      "   GLU  73 A     8.38       4.50                      \n",
      "   GLU  89 A     5.51       4.50                      \n",
      "   GLU  95 A     3.70       4.50                      \n",
      "   GLU 119 A     4.30       4.50                      \n",
      "   GLU 131 A     4.45       4.50                      \n",
      "   GLU 154 A     4.04       4.50                      \n",
      "   GLU 192 A     3.88       4.50                      \n",
      "   GLU 215 A     3.63       4.50                      \n",
      "   GLU 235 A     3.26       4.50                      \n",
      "   GLU 238 A     3.63       4.50                      \n",
      "   GLU 246 A     2.95       4.50                      \n",
      "   GLU 290 A     1.29       4.50                      \n",
      "   GLU 304 A     3.18       4.50                      \n",
      "   GLU 314 A     5.06       4.50                      \n",
      "   GLU 339 A     4.02       4.50                      \n",
      "   C-  355 A     3.34       3.20                      \n",
      "   HIS  31 A     6.33       6.50                      \n",
      "   HIS 191 A     6.82       6.50                      \n",
      "   HIS 221 A     5.75       6.50                      \n",
      "   HIS 224 A     6.57       6.50                      \n",
      "   HIS 241 A     4.98       6.50                      \n",
      "   HIS 244 A     5.80       6.50                      \n",
      "   HIS 261 A     6.60       6.50                      \n",
      "   HIS 295 A     5.60       6.50                      \n",
      "   HIS 303 A     6.22       6.50                      \n",
      "   HIS 309 A     2.56       6.50                      \n",
      "   HIS 341 A     4.22       6.50                      \n",
      "   CYS 140 A    12.78       9.00                      \n",
      "   CYS 159 A    15.19       9.00                      \n",
      "   CYS 172 A    12.08       9.00                      \n",
      "   CYS 175 A     9.09       9.00                      \n",
      "   CYS 269 A    19.48       9.00                      \n",
      "   CYS 349 A     9.57       9.00                      \n",
      "   TYR  50 A    12.47      10.00                      \n",
      "   TYR  97 A    12.15      10.00                      \n",
      "   TYR 101 A    10.24      10.00                      \n",
      "   TYR 130 A    10.10      10.00                      \n",
      "   TYR 183 A    11.75      10.00                      \n",
      "   TYR 196 A    13.17      10.00                      \n",
      "   TYR 208 A    11.41      10.00                      \n",
      "   TYR 268 A    17.93      10.00                      \n",
      "   TYR 279 A    12.08      10.00                      \n",
      "   TYR 288 A    12.91      10.00                      \n",
      "   LYS  16 A     9.20      10.50                      \n",
      "   LYS  22 A     9.53      10.50                      \n",
      "   LYS  32 A    10.25      10.50                      \n",
      "   LYS  39 A    13.17      10.50                      \n",
      "   LYS  61 A     9.38      10.50                      \n",
      "   LYS  68 A    10.47      10.50                      \n",
      "   LYS  91 A    11.20      10.50                      \n",
      "   LYS  99 A    11.34      10.50                      \n",
      "   LYS 102 A    10.42      10.50                      \n",
      "   LYS 103 A    11.30      10.50                      \n",
      "   LYS 116 A    10.23      10.50                      \n",
      "   LYS 135 A    10.54      10.50                      \n",
      "   LYS 148 A    10.50      10.50                      \n",
      "   LYS 189 A    10.46      10.50                      \n",
      "   LYS 219 A    10.37      10.50                      \n",
      "   LYS 226 A    10.46      10.50                      \n",
      "   LYS 229 A    10.85      10.50                      \n",
      "   LYS 239 A    10.51      10.50                      \n",
      "   LYS 251 A    10.84      10.50                      \n",
      "   LYS 253 A    10.54      10.50                      \n",
      "   LYS 258 A    10.36      10.50                      \n",
      "   LYS 267 A     7.64      10.50                      \n",
      "   LYS 285 A    10.40      10.50                      \n",
      "   LYS 291 A    10.01      10.50                      \n",
      "   LYS 329 A     9.59      10.50                      \n",
      "   ARG   5 A    12.07      12.50                      \n",
      "   ARG  15 A    12.14      12.50                      \n",
      "   ARG  23 A    11.87      12.50                      \n",
      "   ARG 122 A    13.16      12.50                      \n",
      "   ARG 134 A    14.14      12.50                      \n",
      "   ARG 137 A    13.63      12.50                      \n",
      "   ARG 152 A    15.85      12.50                      \n",
      "   ARG 199 A    10.80      12.50                      \n",
      "   ARG 202 A    11.55      12.50                      \n",
      "   ARG 223 A    13.63      12.50                      \n",
      "   ARG 281 A    17.71      12.50                      \n",
      "   ARG 324 A    11.59      12.50                      \n",
      "   ARG 338 A    12.69      12.50                      \n",
      "   ARG 355 A    14.03      12.50                      \n",
      "   N+    2 A     8.88       8.00                      \n",
      "\n",
      "INFO:PROPKA information:\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "Free energy of   folding (kcal/mol) as a function of pH (using neutral reference)\n",
      "  0.00     44.66\n",
      "  1.00     44.24\n",
      "  2.00     41.82\n",
      "  3.00     33.82\n",
      "  4.00     21.09\n",
      "  5.00     16.07\n",
      "  6.00     17.12\n",
      "  7.00     18.86\n",
      "  8.00     22.13\n",
      "  9.00     26.07\n",
      " 10.00     32.15\n",
      " 11.00     44.82\n",
      " 12.00     59.71\n",
      " 13.00     73.20\n",
      " 14.00     85.66\n",
      "\n",
      "The pH of optimum stability is  5.0 for which the free energy is   16.1 kcal/mol at 298K\n",
      "Could not determine pH values where the free energy is within 80 % of minimum\n",
      "Could not determine the pH-range where the free energy is negative\n",
      "\n",
      "\n",
      "Protein charge of folded and unfolded state as a function of pH\n",
      "    pH  unfolded  folded\n",
      "  0.00     50.99   50.91\n",
      "  0.10     50.99   50.88\n",
      "  0.20     50.99   50.85\n",
      "  0.30     50.99   50.82\n",
      "  0.40     50.99   50.77\n",
      "  0.50     50.98   50.72\n",
      "  0.60     50.98   50.65\n",
      "  0.70     50.97   50.57\n",
      "  0.80     50.96   50.48\n",
      "  0.90     50.96   50.36\n",
      "  1.00     50.94   50.23\n",
      "  1.10     50.93   50.07\n",
      "  1.20     50.91   49.89\n",
      "  1.30     50.89   49.68\n",
      "  1.40     50.86   49.44\n",
      "  1.50     50.82   49.16\n",
      "  1.60     50.78   48.85\n",
      "  1.70     50.72   48.49\n",
      "  1.80     50.65   48.09\n",
      "  1.90     50.56   47.64\n",
      "  2.00     50.44   47.13\n",
      "  2.10     50.30   46.56\n",
      "  2.20     50.13   45.91\n",
      "  2.30     49.91   45.20\n",
      "  2.40     49.64   44.39\n",
      "  2.50     49.31   43.50\n",
      "  2.60     48.90   42.51\n",
      "  2.70     48.40   41.41\n",
      "  2.80     47.80   40.22\n",
      "  2.90     47.07   38.92\n",
      "  3.00     46.20   37.51\n",
      "  3.10     45.17   36.02\n",
      "  3.20     43.97   34.43\n",
      "  3.30     42.58   32.78\n",
      "  3.40     41.01   31.06\n",
      "  3.50     39.25   29.30\n",
      "  3.60     37.31   27.52\n",
      "  3.70     35.23   25.73\n",
      "  3.80     33.02   23.96\n",
      "  3.90     30.72   22.23\n",
      "  4.00     28.37   20.55\n",
      "  4.10     26.01   18.95\n",
      "  4.20     23.67   17.45\n",
      "  4.30     21.38   16.04\n",
      "  4.40     19.18   14.74\n",
      "  4.50     17.10   13.54\n",
      "  4.60     15.14   12.44\n",
      "  4.70     13.34   11.43\n",
      "  4.80     11.70   10.50\n",
      "  4.90     10.22    9.64\n",
      "  5.00      8.90    8.83\n",
      "  5.10      7.74    8.07\n",
      "  5.20      6.71    7.33\n",
      "  5.30      5.81    6.62\n",
      "  5.40      5.00    5.93\n",
      "  5.50      4.27    5.25\n",
      "  5.60      3.60    4.57\n",
      "  5.70      2.97    3.90\n",
      "  5.80      2.35    3.24\n",
      "  5.90      1.73    2.58\n",
      "  6.00      1.10    1.93\n",
      "  6.10      0.45    1.29\n",
      "  6.20     -0.22    0.66\n",
      "  6.30     -0.91    0.05\n",
      "  6.40     -1.61   -0.54\n",
      "  6.50     -2.31   -1.11\n",
      "  6.60     -3.01   -1.64\n",
      "  6.70     -3.68   -2.15\n",
      "  6.80     -4.31   -2.62\n",
      "  6.90     -4.91   -3.05\n",
      "  7.00     -5.45   -3.45\n",
      "  7.10     -5.94   -3.81\n",
      "  7.20     -6.38   -4.14\n",
      "  7.30     -6.78   -4.45\n",
      "  7.40     -7.13   -4.74\n",
      "  7.50     -7.46   -5.01\n",
      "  7.60     -7.76   -5.26\n",
      "  7.70     -8.04   -5.51\n",
      "  7.80     -8.32   -5.76\n",
      "  7.90     -8.60   -6.00\n",
      "  8.00     -8.88   -6.26\n",
      "  8.10     -9.18   -6.52\n",
      "  8.20     -9.50   -6.80\n",
      "  8.30     -9.84   -7.11\n",
      "  8.40    -10.22   -7.43\n",
      "  8.50    -10.65   -7.79\n",
      "  8.60    -11.11   -8.19\n",
      "  8.70    -11.64   -8.63\n",
      "  8.80    -12.21   -9.11\n",
      "  8.90    -12.85   -9.64\n",
      "  9.00    -13.55  -10.22\n",
      "  9.10    -14.32  -10.85\n",
      "  9.20    -15.17  -11.54\n",
      "  9.30    -16.09  -12.29\n",
      "  9.40    -17.10  -13.08\n",
      "  9.50    -18.21  -13.94\n",
      "  9.60    -19.42  -14.85\n",
      "  9.70    -20.76  -15.82\n",
      "  9.80    -22.21  -16.86\n",
      "  9.90    -23.79  -17.95\n",
      " 10.00    -25.49  -19.10\n",
      " 10.10    -27.30  -20.31\n",
      " 10.20    -29.18  -21.57\n",
      " 10.30    -31.13  -22.86\n",
      " 10.40    -33.09  -24.19\n",
      " 10.50    -35.05  -25.53\n",
      " 10.60    -36.95  -26.87\n",
      " 10.70    -38.76  -28.19\n",
      " 10.80    -40.46  -29.49\n",
      " 10.90    -42.03  -30.76\n",
      " 11.00    -43.45  -31.99\n",
      " 11.10    -44.73  -33.18\n",
      " 11.20    -45.88  -34.33\n",
      " 11.30    -46.90  -35.46\n",
      " 11.40    -47.83  -36.54\n",
      " 11.50    -48.67  -37.61\n",
      " 11.60    -49.47  -38.64\n",
      " 11.70    -50.22  -39.65\n",
      " 11.80    -50.97  -40.64\n",
      " 11.90    -51.72  -41.60\n",
      " 12.00    -52.49  -42.54\n",
      " 12.10    -53.29  -43.45\n",
      " 12.20    -54.12  -44.34\n",
      " 12.30    -54.97  -45.20\n",
      " 12.40    -55.84  -46.03\n",
      " 12.50    -56.72  -46.84\n",
      " 12.60    -57.58  -47.62\n",
      " 12.70    -58.41  -48.37\n",
      " 12.80    -59.18  -49.10\n",
      " 12.90    -59.90  -49.80\n",
      " 13.00    -60.55  -50.48\n",
      " 13.10    -61.12  -51.13\n",
      " 13.20    -61.61  -51.75\n",
      " 13.30    -62.04  -52.35\n",
      " 13.40    -62.40  -52.91\n",
      " 13.50    -62.70  -53.45\n",
      " 13.60    -62.95  -53.96\n",
      " 13.70    -63.15  -54.44\n",
      " 13.80    -63.32  -54.89\n",
      " 13.90    -63.45  -55.30\n",
      " 14.00    -63.56  -55.69\n",
      "The pI is  6.31 (folded) and  6.17 (unfolded)\n",
      "\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "INFO:Applying pKa values at a pH of 7.30:\n",
      "INFO:Adding hydrogens to biomolecule.\n",
      "INFO:Debumping biomolecule (again).\n",
      "INFO:Optimizing hydrogen bonds\n",
      "INFO:Applying force field to biomolecule states.\n",
      "INFO:Regenerating headers.\n",
      "INFO:Regenerating PDB lines.\n",
      "WARNING:Ignoring 466 header lines in output.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "pH: 7.3\tname: 1ka1_hal2\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:PDB2PQR v3.6.2: biomolecular structure conversion software.\n",
      "INFO:Please cite:  Jurrus E, et al.  Improvements to the APBS biomolecular solvation software suite.  Protein Sci 27 112-128 (2018).\n",
      "INFO:Please cite:  Dolinsky TJ, et al.  PDB2PQR: expanding and upgrading automated preparation of biomolecular structures for molecular simulations. Nucleic Acids Res 35 W522-W525 (2007).\n",
      "INFO:Checking and transforming input arguments.\n",
      "INFO:Loading topology files.\n",
      "INFO:Loading molecule: /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/AtSAL1_updated.pdb\n",
      "INFO:Setting up molecule.\n",
      "INFO:Created biomolecule object with 352 residues and 2628 atoms.\n",
      "INFO:Setting termini states for biomolecule chains.\n",
      "INFO:Loading forcefield.\n",
      "INFO:Loading hydrogen topology definitions.\n",
      "WARNING:Missing atom OXT in residue ALA A 352\n",
      "WARNING:Missing atom OXT in residue ALA A 352\n",
      "INFO:Attempting to repair 1 missing atoms in biomolecule.\n",
      "WARNING:Missing atom OXT in residue ALA A 352\n",
      "INFO:Added atom OXT to residue ALA A 352 at coordinates -4.698, -13.324, -18.987\n",
      "INFO:Updating disulfide bridges.\n",
      "INFO:Debumping biomolecule.\n",
      "INFO:Assigning titration states with PROPKA.\n",
      "INFO:\n",
      "propka3.5.1                                                                                  2024-07-26\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "--                                                                           --\n",
      "--  PROPKA: A PROTEIN PKA PREDICTOR                                          --\n",
      "--                                                                           --\n",
      "--  VERSION 1.0,  04/25/2004,  IOWA CITY                                     --\n",
      "--  BY HUI LI                                                                --\n",
      "--                                                                           --\n",
      "--  VERSION 2.0,  11/05/2007, IOWA CITY/COPENHAGEN                           --\n",
      "--  BY DELPHINE C. BAS AND DAVID M. ROGERS                                   --\n",
      "--                                                                           --\n",
      "--  VERSION 3.0,  01/06/2011, COPENHAGEN                                     --\n",
      "--  BY MATS H.M. OLSSON AND CHRESTEN R. SONDERGARD                           --\n",
      "--                                                                           --\n",
      "--  VERSION 3.1,  07/01/2011, COPENHAGEN                                     --\n",
      "--  BY CHRESTEN R. SONDERGARD AND MATS H.M. OLSSON                           --\n",
      "--                                                                           --\n",
      "--  VERSION 3.2,  06/17/2020, PLANET EARTH                                   --\n",
      "--  SEE ABOVE FOR AUTHORS                                                    --\n",
      "--                                                                           --\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "References:\n",
      "\n",
      "Very Fast Empirical Prediction and Rationalization of Protein pKa Values.\n",
      "Hui Li, Andrew D. Robertson and Jan H. Jensen. PROTEINS: Structure, Function,\n",
      "and Bioinformatics. 61:704-721 (2005)\n",
      "\n",
      "Very Fast Prediction and Rationalization of pKa Values for Protein-Ligand\n",
      "Complexes.  Delphine C. Bas, David M. Rogers and Jan H. Jensen.  PROTEINS:\n",
      "Structure, Function, and Bioinformatics 73:765-783 (2008)\n",
      "\n",
      "PROPKA3: Consistent Treatment of Internal and Surface Residues in Empirical\n",
      "pKa predictions.  Mats H.M. Olsson, Chresten R. Sondergard, Michal Rostkowski,\n",
      "and Jan H. Jensen.  Journal of Chemical Theory and Computation, 7(2):525-537\n",
      "(2011)\n",
      "\n",
      "Improved Treatment of Ligands and Coupling Effects in Empirical Calculation\n",
      "and Rationalization of pKa Values.  Chresten R. Sondergaard, Mats H.M. Olsson,\n",
      "Michal Rostkowski, and Jan H. Jensen.  Journal of Chemical Theory and\n",
      "Computation, (2011)\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "\n",
      "INFO:Coupling map:\n",
      "\n",
      "INFO:Calculating pKas for Conformation container 1A with 3086 atoms and 869 groups\n",
      "INFO:did not converge in 10 iterations\n",
      "INFO:pKa determinants:\n",
      "\n",
      "---------  -----   ------   ---------------------    --------------    --------------    --------------\n",
      "                            DESOLVATION  EFFECTS       SIDECHAIN          BACKBONE        COULOMBIC\n",
      " RESIDUE    pKa    BURIED     REGULAR      RE        HYDROGEN BOND     HYDROGEN BOND      INTERACTION\n",
      "---------  -----   ------   ---------   ---------    --------------    --------------    --------------\n",
      "\n",
      "ASP   8 A   3.99    18 %    0.58  331   0.18    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.03 N+    1 A\n",
      "ASP   8 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.15 LYS   5 A\n",
      "ASP   8 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.30 LYS  11 A\n",
      "ASP   8 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.12 LYS  12 A\n",
      "ASP   8 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU   6 A\n",
      "\n",
      "ASP  32 A   3.90     0 %    0.12  137   0.00    0    0.00 XXX   0 X   -0.02 VAL  33 A    0.00 XXX   0 X\n",
      "\n",
      "ASP  38 A   3.69     0 %    0.12  180   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.24 LYS  39 A\n",
      "\n",
      "ASP  46 A   7.13    91 %    2.32  536   0.55    0    0.00 XXX   0 X    0.00 XXX   0 X    0.25 GLU  71 A\n",
      "ASP  46 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 GLU  72 A\n",
      "ASP  46 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 137 A\n",
      "\n",
      "ASP  73 A   3.49     0 %    0.30  267   0.00    0    0.00 XXX   0 X   -0.69 GLY  75 A   -0.06 ARG  78 A\n",
      "ASP  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS 317 A\n",
      "ASP  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 ASP 118 A\n",
      "\n",
      "ASP  76 A   4.20     0 %    0.23  268   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.00 LYS  79 A\n",
      "ASP  76 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.14 ASP  73 A\n",
      "ASP  76 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP  80 A\n",
      "\n",
      "ASP  80 A   3.82     0 %    0.31  184   0.00    0   -0.20 SER  82 A   -0.22 ASP  80 A    0.03 ASP  38 A\n",
      "ASP  80 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP  84 A\n",
      "\n",
      "ASP  84 A   3.48     0 %    0.35  226   0.00    0    0.00 XXX   0 X   -0.06 ASP  84 A   -0.25 ARG  88 A\n",
      "ASP  84 A                                            0.00 XXX   0 X   -0.35 THR  85 A    0.00 XXX   0 X\n",
      "\n",
      "ASP  95 A   3.71     0 %    0.32  279   0.00    0    0.00 XXX   0 X   -0.01 ASP  95 A   -0.28 LYS  23 A\n",
      "ASP  95 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.12 LYS  91 A\n",
      "\n",
      "ASP 111 A   3.94     0 %    0.21  211   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.03 ARG  78 A\n",
      "ASP 111 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.14 ARG 115 A\n",
      "ASP 111 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP 112 A\n",
      "\n",
      "ASP 112 A   2.85     0 %    0.34  258   0.00    0   -0.49 SER 109 A   -0.62 SER 109 A   -0.18 ARG 115 A\n",
      "\n",
      "ASP 118 A   3.20    10 %    0.65  308   0.06    0   -0.77 SER  74 A   -0.25 SER  74 A   -0.26 ARG  78 A\n",
      "ASP 118 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 115 A\n",
      "\n",
      "ASP 134 A  13.40*  100 %    3.03  569   0.41    0    1.52 ASP 137 A   -0.26 ILE 136 A   -0.01 LYS 285 A\n",
      "ASP 134 A                                            0.96 ASP 288 A    0.00 XXX   0 X    0.29 GLU  71 A\n",
      "ASP 134 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.68 ASP  46 A\n",
      "ASP 134 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.89 ASP 137 A\n",
      "ASP 134 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.08 ASP 288 A\n",
      "\n",
      "ASP 137 A   2.84*  100 %    2.61  581   0.29    0   -1.52 ASP 134 A   -0.00 GLN 260 A   -0.47 LYS 262 A\n",
      "ASP 137 A                                           -1.46 ASP 288 A    0.00 XXX   0 X    0.05 GLU  71 A\n",
      "ASP 137 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 GLU 284 A\n",
      "ASP 137 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.65 ARG 276 A\n",
      "\n",
      "ASP 146 A   3.60    12 %    0.79  316   0.11    0   -0.79 ARG 144 A    0.00 XXX   0 X   -0.31 ARG 144 A\n",
      "\n",
      "ASP 186 A   3.58     0 %    0.15  116   0.00    0   -0.37 SER 184 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "ASP 206 A   3.75    53 %    1.44  431   0.23    0   -1.15 ARG  19 A    0.00 XXX   0 X   -0.59 ARG  19 A\n",
      "\n",
      "ASP 239 A   3.65     0 %    0.43  269   0.00    0    0.00 XXX   0 X   -0.34 ASP 239 A   -0.04 HIS 235 A\n",
      "ASP 239 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.17 HIS 238 A\n",
      "ASP 239 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 HIS 279 A\n",
      "\n",
      "ASP 258 A   6.14    44 %    1.75  404   0.62    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.00 LYS 140 A\n",
      "ASP 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.34 ARG 144 A\n",
      "ASP 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.05 LYS 262 A\n",
      "ASP 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.34 ASP 146 A\n",
      "ASP 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 232 A\n",
      "\n",
      "ASP 270 A   5.41    76 %    2.89  493   0.45    0   -0.11 SER 174 A   -0.21 ILE 175 A   -0.89 ARG 268 A\n",
      "ASP 270 A                                           -0.51 ARG 268 A   -0.01 ASP 270 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 288 A   7.25*   81 %    2.27  509   0.16    0   -0.96 ASP 134 A    0.00 XXX   0 X   -0.13 LYS 262 A\n",
      "ASP 288 A                                            1.46 ASP 137 A    0.00 XXX   0 X   -0.22 LYS 285 A\n",
      "ASP 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.62 ASP 137 A\n",
      "ASP 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.54 ARG 276 A\n",
      "ASP 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.54 GLU 284 A\n",
      "ASP 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.75 HIS 289 A\n",
      "\n",
      "ASP 305 A   4.25    25 %    1.48  352   0.32    0    0.00 XXX   0 X   -0.17 ASP 305 A   -0.12 LYS 309 A\n",
      "ASP 305 A                                            0.00 XXX   0 X   -0.07 ALA 307 A   -0.04 ARG 341 A\n",
      "ASP 305 A                                            0.00 XXX   0 X   -0.00 GLY 308 A    0.02 ASP 320 A\n",
      "ASP 305 A                                            0.00 XXX   0 X   -0.63 LYS 309 A    0.23 ASP 322 A\n",
      "ASP 305 A                                            0.00 XXX   0 X   -0.58 ASP 322 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 312 A   2.16    11 %    0.68  312   0.01    0   -0.85 SER 314 A   -0.15 SER 314 A   -0.10 LYS 159 A\n",
      "ASP 312 A                                           -0.85 LYS 315 A    0.00 XXX   0 X   -0.38 LYS 315 A\n",
      "\n",
      "ASP 320 A   3.88     0 %    0.22  254   0.00    0    0.00 XXX   0 X   -0.12 ASP 320 A   -0.13 LYS 315 A\n",
      "ASP 320 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.11 ASP 312 A\n",
      "\n",
      "ASP 322 A   3.54    11 %    0.50  311   0.00    0   -0.66 THR 323 A    0.00 XXX   0 X   -0.01 LYS 309 A\n",
      "ASP 322 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.10 ARG 341 A\n",
      "\n",
      "ASP 342 A   4.00     0 %    0.37  242   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.04 LYS 247 A\n",
      "ASP 342 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.05 LYS 338 A\n",
      "ASP 342 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 ARG 341 A\n",
      "\n",
      "GLU   4 A   4.37     0 %    0.15  174   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.19 N+    1 A\n",
      "GLU   4 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.24 LYS   5 A\n",
      "GLU   4 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU   6 A\n",
      "GLU   4 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 ASP   8 A\n",
      "\n",
      "GLU   6 A   3.70    28 %    1.00  361   0.00    0   -0.72 ARG 129 A   -0.65 GLU   6 A   -0.06 LYS   5 A\n",
      "GLU   6 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.37 ARG 129 A\n",
      "\n",
      "GLU  58 A   4.99    32 %    0.70  370   0.10    0   -0.31 CYS 119 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "GLU  60 A   4.70    11 %    0.55  313   0.02    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.40 LYS  12 A\n",
      "GLU  60 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS  59 A\n",
      "GLU  60 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP   8 A\n",
      "GLU  60 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 101 A\n",
      "\n",
      "GLU  64 A   4.73     0 %    0.31  278   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.32 ARG 129 A\n",
      "GLU  64 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 GLU   6 A\n",
      "GLU  64 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 HIS 130 A\n",
      "GLU  64 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 GLU 156 A\n",
      "GLU  64 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 157 A\n",
      "\n",
      "GLU  71 A   5.17    41 %    0.73  396   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.20 LYS 285 A\n",
      "GLU  71 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 317 A\n",
      "GLU  71 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 ASP 288 A\n",
      "\n",
      "GLU  72 A   5.72    30 %    0.59  365   0.07    0   -0.15 TYR  47 A    0.00 XXX   0 X   -0.01 LYS 317 A\n",
      "GLU  72 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.26 GLU  71 A\n",
      "GLU  72 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 ASP  73 A\n",
      "GLU  72 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.28 ASP  76 A\n",
      "\n",
      "GLU  87 A   4.66     0 %    0.21  201   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.08 LYS  91 A\n",
      "GLU  87 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP  84 A\n",
      "\n",
      "GLU 100 A   4.92     0 %    0.40  210   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 101 A\n",
      "\n",
      "GLU 101 A   4.59     0 %    0.15  174   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS  12 A\n",
      "\n",
      "GLU 123 A   4.29     1 %    0.40  284   0.01    0   -0.17 LYS 317 A    0.00 XXX   0 X   -0.45 LYS 317 A\n",
      "\n",
      "GLU 156 A   4.19    14 %    0.82  321   0.09    0   -0.82 ARG 129 A    0.00 XXX   0 X    0.09 GLU   6 A\n",
      "GLU 156 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 HIS 130 A\n",
      "GLU 156 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.45 ARG 129 A\n",
      "\n",
      "GLU 157 A   4.41     0 %    0.30  215   0.00    0    0.00 XXX   0 X   -0.50 GLY 128 A   -0.04 ARG 129 A\n",
      "GLU 157 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.05 HIS 130 A\n",
      "GLU 157 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 GLU 156 A\n",
      "\n",
      "GLU 187 A   4.76     0 %    0.24  215   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 186 A\n",
      "\n",
      "GLU 220 A   4.46     0 %    0.22  175   0.00    0    0.00 XXX   0 X   -0.30 GLU 220 A   -0.08 LYS 331 A\n",
      "GLU 220 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.12 GLU 224 A\n",
      "\n",
      "GLU 223 A   4.77     0 %    0.51  236   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.27 LYS 251 A\n",
      "GLU 223 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 224 A\n",
      "\n",
      "GLU 224 A   3.96     0 %    0.21  180   0.00    0   -0.71 ASN 221 A    0.00 XXX   0 X   -0.04 LYS 251 A\n",
      "\n",
      "GLU 229 A   5.80    55 %    1.19  436   0.40    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.13 ARG 256 A\n",
      "GLU 229 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.09 LYS 262 A\n",
      "GLU 229 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.07 HIS 238 A\n",
      "\n",
      "GLU 232 A   4.44     0 %    0.15  188   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS 140 A\n",
      "GLU 232 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.20 ARG 144 A\n",
      "GLU 232 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.00 ARG 256 A\n",
      "GLU 232 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 146 A\n",
      "\n",
      "GLU 284 A   2.33    71 %    2.47  480   0.16    0   -0.21 ARG 276 A   -0.15 LYS 285 A   -0.19 LYS 262 A\n",
      "GLU 284 A                                           -0.11 TYR 282 A    0.00 XXX   0 X   -0.19 LYS 285 A\n",
      "GLU 284 A                                           -1.60 HIS 289 A    0.00 XXX   0 X   -1.06 ARG 276 A\n",
      "GLU 284 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.29 HIS 289 A\n",
      "\n",
      "GLU 298 A   3.85    28 %    1.05  360   0.21    0    0.00 XXX   0 X   -0.73 GLY 197 A   -0.09 LYS 213 A\n",
      "GLU 298 A                                            0.00 XXX   0 X   -0.33 SER 198 A    0.00 XXX   0 X\n",
      "GLU 298 A                                            0.00 XXX   0 X   -0.75 GLY 199 A    0.00 XXX   0 X\n",
      "\n",
      "GLU 330 A   4.59     0 %    0.16  232   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 331 A\n",
      "\n",
      "GLU 346 A   4.58     0 %    0.39  216   0.00    0   -0.10 GLN 347 A    0.00 XXX   0 X   -0.33 LYS 247 A\n",
      "GLU 346 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 LYS 349 A\n",
      "GLU 346 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.14 ASP 342 A\n",
      "\n",
      "GLU 348 A   4.77     0 %    0.30  273   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.09 LYS 349 A\n",
      "GLU 348 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 239 A\n",
      "GLU 348 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 322 A\n",
      "\n",
      "C-  352 A   3.34     0 %    0.14   58   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "HIS 130 A   6.21    28 %   -0.99  361   0.00    0    0.00 XXX   0 X    0.01 GLY 124 A    0.01 GLU  64 A\n",
      "HIS 130 A                                            0.00 XXX   0 X    0.43 ASN 127 A    0.04 GLU 156 A\n",
      "HIS 130 A                                            0.00 XXX   0 X    0.15 ARG 129 A    0.05 GLU 157 A\n",
      "\n",
      "HIS 235 A   6.45     0 %   -0.18  141   0.00    0    0.00 XXX   0 X    0.10 ALA 234 A    0.04 ASP 239 A\n",
      "\n",
      "HIS 238 A   6.25     0 %   -0.30  241   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.00 ARG 256 A\n",
      "HIS 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 GLU 229 A\n",
      "HIS 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.19 HIS 235 A\n",
      "HIS 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.17 ASP 239 A\n",
      "\n",
      "HIS 279 A   6.15     0 %   -0.30  228   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.07 HIS 235 A\n",
      "HIS 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 239 A\n",
      "\n",
      "HIS 289 A   5.35   100 %   -3.14  578   0.00    0    1.60 GLU 284 A    0.00 XXX   0 X   -0.40 LYS 262 A\n",
      "HIS 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.23 ARG 276 A\n",
      "HIS 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 LYS 285 A\n",
      "HIS 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.29 GLU 284 A\n",
      "HIS 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.75 ASP 288 A\n",
      "\n",
      "CYS  21 A  12.37   100 %    3.39  590   0.00    0   -0.13 SER  49 A   -0.12 CYS  21 A    0.23 ASP  46 A\n",
      "\n",
      "CYS 119 A  10.39    24 %    0.60  349   0.00    0    0.31 GLU  58 A    0.00 XXX   0 X   -0.02 ARG 115 A\n",
      "CYS 119 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 118 A\n",
      "CYS 119 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.46 GLU  58 A\n",
      "\n",
      "CYS 167 A  13.04   100 %    3.60  645   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.29 ARG 268 A\n",
      "CYS 167 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.73 CYS 190 A\n",
      "\n",
      "CYS 190 A  10.90    74 %    2.85  488   0.00    0   -0.22 ARG 268 A   -0.02 GLY 189 A   -0.84 ARG 268 A\n",
      "CYS 190 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 187 A\n",
      "CYS 190 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.11 ASP 270 A\n",
      "\n",
      "TYR   3 A  10.76     8 %    0.53  303   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.10 GLU   6 A\n",
      "TYR   3 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.19 ARG 129 A\n",
      "TYR   3 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.29 GLU 156 A\n",
      "TYR   3 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 157 A\n",
      "\n",
      "TYR  47 A  12.70    45 %    1.22  407   0.00    0    0.15 GLU  72 A    0.00 XXX   0 X    0.26 ASP  46 A\n",
      "TYR  47 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU  71 A\n",
      "TYR  47 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.14 ASP  73 A\n",
      "TYR  47 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.39 ASP  76 A\n",
      "TYR  47 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP 118 A\n",
      "TYR  47 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.51 GLU  72 A\n",
      "\n",
      "TYR 148 A  15.46    84 %    4.04  516   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    1.36 CYS  21 A\n",
      "TYR 148 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 CYS 167 A\n",
      "\n",
      "TYR 201 A  10.14     0 %    0.32  267   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 298 A\n",
      "TYR 201 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.20 LYS 213 A\n",
      "\n",
      "TYR 263 A  15.09*  100 %    3.51  633   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP 137 A\n",
      "TYR 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.83 ARG 276 A\n",
      "TYR 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.46 GLU 284 A\n",
      "TYR 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 ASP 288 A\n",
      "TYR 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.61 TYR 274 A\n",
      "\n",
      "TYR 274 A   9.17*  100 %    3.44  594   0.00    0   -0.85 LYS 262 A    0.00 XXX   0 X    0.22 ASP 137 A\n",
      "TYR 274 A                                           -0.20 ARG 276 A    0.00 XXX   0 X    0.08 GLU 229 A\n",
      "TYR 274 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.42 GLU 284 A\n",
      "TYR 274 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.12 ASP 288 A\n",
      "TYR 274 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -2.03 LYS 262 A\n",
      "TYR 274 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -2.03 ARG 276 A\n",
      "\n",
      "TYR 282 A  12.77    42 %    1.48  398   0.00    0    0.11 GLU 284 A    0.00 XXX   0 X    0.09 GLU  71 A\n",
      "TYR 282 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 134 A\n",
      "TYR 282 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP 137 A\n",
      "TYR 282 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.34 ARG 276 A\n",
      "TYR 282 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.07 ARG 283 A\n",
      "TYR 282 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 TYR 274 A\n",
      "TYR 282 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 TYR 318 A\n",
      "TYR 282 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.71 GLU 284 A\n",
      "TYR 282 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.53 ASP 288 A\n",
      "\n",
      "TYR 318 A   9.59     0 %    0.35  256   0.00    0   -0.13 ARG 283 A    0.00 XXX   0 X    0.06 GLU  71 A\n",
      "TYR 318 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 123 A\n",
      "TYR 318 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.26 LYS 285 A\n",
      "TYR 318 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.09 LYS 317 A\n",
      "TYR 318 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.38 ARG 283 A\n",
      "\n",
      "LYS   5 A  10.80     0 %   -0.15  167   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.24 GLU   4 A\n",
      "LYS   5 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 GLU   6 A\n",
      "LYS   5 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP   8 A\n",
      "\n",
      "LYS  11 A   9.47    50 %   -1.27  420   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.30 ASP   8 A\n",
      "LYS  11 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS  12 A\n",
      "\n",
      "LYS  12 A  10.57    27 %   -0.51  358   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.12 ASP   8 A\n",
      "LYS  12 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.40 GLU  60 A\n",
      "LYS  12 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 GLU 101 A\n",
      "\n",
      "LYS  23 A  10.16    17 %   -0.59  328   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.28 ASP  95 A\n",
      "LYS  23 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 LYS  91 A\n",
      "\n",
      "LYS  26 A  10.23     3 %   -0.27  291   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS  39 A  10.48     0 %   -0.26  206   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.24 ASP  38 A\n",
      "\n",
      "LYS  59 A  10.21     0 %   -0.32  249   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU  60 A\n",
      "\n",
      "LYS  79 A  10.44     0 %   -0.06   88   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP  76 A\n",
      "\n",
      "LYS  91 A  10.52     0 %   -0.17  208   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU  87 A\n",
      "LYS  91 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG  88 A\n",
      "LYS  91 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.12 ASP  95 A\n",
      "\n",
      "LYS 140 A   9.86    17 %   -0.56  329   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.08 ARG 144 A\n",
      "LYS 140 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 232 A\n",
      "LYS 140 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP 258 A\n",
      "LYS 140 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS  39 A\n",
      "\n",
      "LYS 159 A  10.42     0 %   -0.14  238   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP 312 A\n",
      "LYS 159 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 LYS 315 A\n",
      "\n",
      "LYS 180 A  10.43     0 %   -0.12   93   0.00    0    0.06 LYS 182 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 182 A  10.14     0 %   -0.31  138   0.00    0   -0.06 LYS 180 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 208 A  10.45     0 %   -0.05   83   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 213 A  10.19     9 %   -0.60  306   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.09 GLU 298 A\n",
      "LYS 213 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 TYR 201 A\n",
      "\n",
      "LYS 247 A  10.38     0 %   -0.49  262   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 342 A\n",
      "LYS 247 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.33 GLU 346 A\n",
      "\n",
      "LYS 251 A  10.60     0 %   -0.22  135   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.27 GLU 223 A\n",
      "LYS 251 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 GLU 224 A\n",
      "\n",
      "LYS 262 A   9.41   100 %   -2.90  586   0.00    0    0.85 TYR 274 A    0.00 XXX   0 X    0.47 ASP 137 A\n",
      "LYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 GLU 229 A\n",
      "LYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 258 A\n",
      "LYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -2.00 ARG 276 A\n",
      "LYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 GLU 284 A\n",
      "LYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 ASP 288 A\n",
      "LYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    2.03 TYR 274 A\n",
      "\n",
      "LYS 280 A  10.46     0 %   -0.04   81   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 285 A  10.26    17 %   -0.88  329   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.20 GLU  71 A\n",
      "LYS 285 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 134 A\n",
      "LYS 285 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.24 ARG 283 A\n",
      "LYS 285 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 GLU 284 A\n",
      "LYS 285 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.22 ASP 288 A\n",
      "LYS 285 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.26 TYR 318 A\n",
      "\n",
      "LYS 309 A  10.51     0 %   -0.11  165   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.12 ASP 305 A\n",
      "LYS 309 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 322 A\n",
      "LYS 309 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 ARG 341 A\n",
      "\n",
      "LYS 315 A  11.51     0 %   -0.35  220   0.00    0    0.85 ASP 312 A    0.00 XXX   0 X    0.13 ASP 320 A\n",
      "LYS 315 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 ASP 312 A\n",
      "\n",
      "LYS 317 A  10.13    35 %   -1.18  378   0.00    0    0.17 GLU 123 A    0.00 XXX   0 X    0.06 GLU  71 A\n",
      "LYS 317 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU  72 A\n",
      "LYS 317 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP  73 A\n",
      "LYS 317 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 TYR 318 A\n",
      "LYS 317 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.45 GLU 123 A\n",
      "\n",
      "LYS 331 A  10.52     0 %   -0.12  161   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU 220 A\n",
      "LYS 331 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 GLU 330 A\n",
      "\n",
      "LYS 338 A  10.23     0 %   -0.27  219   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.05 ARG 341 A\n",
      "LYS 338 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 342 A\n",
      "\n",
      "LYS 349 A  10.31     0 %   -0.30  147   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 346 A\n",
      "LYS 349 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 GLU 348 A\n",
      "\n",
      "ARG  19 A  13.45    31 %   -0.78  369   0.00    0    1.15 ASP 206 A    0.00 XXX   0 X    0.59 ASP 206 A\n",
      "\n",
      "ARG  78 A  12.45     0 %   -0.37  250   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.06 ASP  73 A\n",
      "ARG  78 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 111 A\n",
      "ARG  78 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.26 ASP 118 A\n",
      "ARG  78 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 ARG 115 A\n",
      "\n",
      "ARG  88 A  12.29    11 %   -0.46  311   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.25 ASP  84 A\n",
      "\n",
      "ARG 115 A  12.63     0 %   -0.24  199   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.14 ASP 111 A\n",
      "ARG 115 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.18 ASP 112 A\n",
      "ARG 115 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 118 A\n",
      "ARG 115 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 CYS 119 A\n",
      "\n",
      "ARG 129 A  14.57    24 %   -0.85  348   0.00    0    0.72 GLU   6 A    0.00 XXX   0 X    0.19 TYR   3 A\n",
      "ARG 129 A                                            0.82 GLU 156 A    0.00 XXX   0 X    0.32 GLU  64 A\n",
      "ARG 129 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 GLU 157 A\n",
      "ARG 129 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.37 GLU   6 A\n",
      "ARG 129 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.45 GLU 156 A\n",
      "\n",
      "ARG 144 A  13.46     3 %   -0.68  291   0.00    0    0.79 ASP 146 A    0.00 XXX   0 X    0.20 GLU 232 A\n",
      "ARG 144 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.34 ASP 258 A\n",
      "ARG 144 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.31 ASP 146 A\n",
      "\n",
      "ARG 256 A  12.33     0 %   -0.30  275   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.13 GLU 229 A\n",
      "ARG 256 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 GLU 232 A\n",
      "\n",
      "ARG 268 A  12.83    75 %   -2.41  490   0.00    0    0.22 CYS 190 A    0.00 XXX   0 X    0.29 CYS 167 A\n",
      "ARG 268 A                                            0.51 ASP 270 A    0.00 XXX   0 X    0.84 CYS 190 A\n",
      "ARG 268 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.89 ASP 270 A\n",
      "\n",
      "ARG 276 A  16.42    98 %   -1.95  557   0.00    0    0.20 TYR 274 A    0.00 XXX   0 X    0.83 TYR 263 A\n",
      "ARG 276 A                                            0.21 GLU 284 A    0.00 XXX   0 X    0.34 TYR 282 A\n",
      "ARG 276 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.65 ASP 137 A\n",
      "ARG 276 A                                            0.00 XXX   0 X    0.00 XXX   0 X    2.03 TYR 274 A\n",
      "ARG 276 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.06 GLU 284 A\n",
      "ARG 276 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.54 ASP 288 A\n",
      "\n",
      "ARG 283 A  12.77     0 %   -0.31  205   0.00    0    0.13 TYR 318 A    0.00 XXX   0 X    0.07 TYR 282 A\n",
      "ARG 283 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 TYR 318 A\n",
      "\n",
      "ARG 341 A  12.40     1 %   -0.32  283   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 305 A\n",
      "ARG 341 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP 322 A\n",
      "ARG 341 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP 342 A\n",
      "\n",
      "N+    1 A   7.94     0 %   -0.28  245   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.19 GLU   4 A\n",
      "N+    1 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP   8 A\n",
      "N+    1 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 LYS  11 A\n",
      "\n",
      "Coupled residues (marked *) were detected.Please rerun PropKa with the --display-coupled-residues \n",
      "or -d option for detailed information.\n",
      "\n",
      "INFO:pKa summary:\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "SUMMARY OF THIS PREDICTION\n",
      "       Group      pKa  model-pKa   ligand atom-type\n",
      "   ASP   8 A     3.99       3.80                      \n",
      "   ASP  32 A     3.90       3.80                      \n",
      "   ASP  38 A     3.69       3.80                      \n",
      "   ASP  46 A     7.13       3.80                      \n",
      "   ASP  73 A     3.49       3.80                      \n",
      "   ASP  76 A     4.20       3.80                      \n",
      "   ASP  80 A     3.82       3.80                      \n",
      "   ASP  84 A     3.48       3.80                      \n",
      "   ASP  95 A     3.71       3.80                      \n",
      "   ASP 111 A     3.94       3.80                      \n",
      "   ASP 112 A     2.85       3.80                      \n",
      "   ASP 118 A     3.20       3.80                      \n",
      "   ASP 134 A    13.40       3.80                      \n",
      "   ASP 137 A     2.84       3.80                      \n",
      "   ASP 146 A     3.60       3.80                      \n",
      "   ASP 186 A     3.58       3.80                      \n",
      "   ASP 206 A     3.75       3.80                      \n",
      "   ASP 239 A     3.65       3.80                      \n",
      "   ASP 258 A     6.14       3.80                      \n",
      "   ASP 270 A     5.41       3.80                      \n",
      "   ASP 288 A     7.25       3.80                      \n",
      "   ASP 305 A     4.25       3.80                      \n",
      "   ASP 312 A     2.16       3.80                      \n",
      "   ASP 320 A     3.88       3.80                      \n",
      "   ASP 322 A     3.54       3.80                      \n",
      "   ASP 342 A     4.00       3.80                      \n",
      "   GLU   4 A     4.37       4.50                      \n",
      "   GLU   6 A     3.70       4.50                      \n",
      "   GLU  58 A     4.99       4.50                      \n",
      "   GLU  60 A     4.70       4.50                      \n",
      "   GLU  64 A     4.73       4.50                      \n",
      "   GLU  71 A     5.17       4.50                      \n",
      "   GLU  72 A     5.72       4.50                      \n",
      "   GLU  87 A     4.66       4.50                      \n",
      "   GLU 100 A     4.92       4.50                      \n",
      "   GLU 101 A     4.59       4.50                      \n",
      "   GLU 123 A     4.29       4.50                      \n",
      "   GLU 156 A     4.19       4.50                      \n",
      "   GLU 157 A     4.41       4.50                      \n",
      "   GLU 187 A     4.76       4.50                      \n",
      "   GLU 220 A     4.46       4.50                      \n",
      "   GLU 223 A     4.77       4.50                      \n",
      "   GLU 224 A     3.96       4.50                      \n",
      "   GLU 229 A     5.80       4.50                      \n",
      "   GLU 232 A     4.44       4.50                      \n",
      "   GLU 284 A     2.33       4.50                      \n",
      "   GLU 298 A     3.85       4.50                      \n",
      "   GLU 330 A     4.59       4.50                      \n",
      "   GLU 346 A     4.58       4.50                      \n",
      "   GLU 348 A     4.77       4.50                      \n",
      "   C-  352 A     3.34       3.20                      \n",
      "   HIS 130 A     6.21       6.50                      \n",
      "   HIS 235 A     6.45       6.50                      \n",
      "   HIS 238 A     6.25       6.50                      \n",
      "   HIS 279 A     6.15       6.50                      \n",
      "   HIS 289 A     5.35       6.50                      \n",
      "   CYS  21 A    12.37       9.00                      \n",
      "   CYS 119 A    10.39       9.00                      \n",
      "   CYS 167 A    13.04       9.00                      \n",
      "   CYS 190 A    10.90       9.00                      \n",
      "   TYR   3 A    10.76      10.00                      \n",
      "   TYR  47 A    12.70      10.00                      \n",
      "   TYR 148 A    15.46      10.00                      \n",
      "   TYR 201 A    10.14      10.00                      \n",
      "   TYR 263 A    15.09      10.00                      \n",
      "   TYR 274 A     9.17      10.00                      \n",
      "   TYR 282 A    12.77      10.00                      \n",
      "   TYR 318 A     9.59      10.00                      \n",
      "   LYS   5 A    10.80      10.50                      \n",
      "   LYS  11 A     9.47      10.50                      \n",
      "   LYS  12 A    10.57      10.50                      \n",
      "   LYS  23 A    10.16      10.50                      \n",
      "   LYS  26 A    10.23      10.50                      \n",
      "   LYS  39 A    10.48      10.50                      \n",
      "   LYS  59 A    10.21      10.50                      \n",
      "   LYS  79 A    10.44      10.50                      \n",
      "   LYS  91 A    10.52      10.50                      \n",
      "   LYS 140 A     9.86      10.50                      \n",
      "   LYS 159 A    10.42      10.50                      \n",
      "   LYS 180 A    10.43      10.50                      \n",
      "   LYS 182 A    10.14      10.50                      \n",
      "   LYS 208 A    10.45      10.50                      \n",
      "   LYS 213 A    10.19      10.50                      \n",
      "   LYS 247 A    10.38      10.50                      \n",
      "   LYS 251 A    10.60      10.50                      \n",
      "   LYS 262 A     9.41      10.50                      \n",
      "   LYS 280 A    10.46      10.50                      \n",
      "   LYS 285 A    10.26      10.50                      \n",
      "   LYS 309 A    10.51      10.50                      \n",
      "   LYS 315 A    11.51      10.50                      \n",
      "   LYS 317 A    10.13      10.50                      \n",
      "   LYS 331 A    10.52      10.50                      \n",
      "   LYS 338 A    10.23      10.50                      \n",
      "   LYS 349 A    10.31      10.50                      \n",
      "   ARG  19 A    13.45      12.50                      \n",
      "   ARG  78 A    12.45      12.50                      \n",
      "   ARG  88 A    12.29      12.50                      \n",
      "   ARG 115 A    12.63      12.50                      \n",
      "   ARG 129 A    14.57      12.50                      \n",
      "   ARG 144 A    13.46      12.50                      \n",
      "   ARG 256 A    12.33      12.50                      \n",
      "   ARG 268 A    12.83      12.50                      \n",
      "   ARG 276 A    16.42      12.50                      \n",
      "   ARG 283 A    12.77      12.50                      \n",
      "   ARG 341 A    12.40      12.50                      \n",
      "   N+    1 A     7.94       8.00                      \n",
      "\n",
      "INFO:PROPKA information:\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "Free energy of   folding (kcal/mol) as a function of pH (using neutral reference)\n",
      "  0.00     29.52\n",
      "  1.00     29.44\n",
      "  2.00     28.85\n",
      "  3.00     26.51\n",
      "  4.00     24.84\n",
      "  5.00     30.16\n",
      "  6.00     36.69\n",
      "  7.00     40.13\n",
      "  8.00     42.33\n",
      "  9.00     44.66\n",
      " 10.00     48.46\n",
      " 11.00     55.32\n",
      " 12.00     64.91\n",
      " 13.00     74.71\n",
      " 14.00     82.36\n",
      "\n",
      "The pH of optimum stability is  3.8 for which the free energy is   24.7 kcal/mol at 298K\n",
      "Could not determine pH values where the free energy is within 80 % of minimum\n",
      "Could not determine the pH-range where the free energy is negative\n",
      "\n",
      "\n",
      "Protein charge of folded and unfolded state as a function of pH\n",
      "    pH  unfolded  folded\n",
      "  0.00     42.99   42.98\n",
      "  0.10     42.99   42.98\n",
      "  0.20     42.99   42.97\n",
      "  0.30     42.99   42.96\n",
      "  0.40     42.99   42.95\n",
      "  0.50     42.98   42.94\n",
      "  0.60     42.98   42.92\n",
      "  0.70     42.97   42.90\n",
      "  0.80     42.97   42.88\n",
      "  0.90     42.96   42.85\n",
      "  1.00     42.94   42.81\n",
      "  1.10     42.93   42.76\n",
      "  1.20     42.91   42.71\n",
      "  1.30     42.89   42.64\n",
      "  1.40     42.86   42.55\n",
      "  1.50     42.83   42.44\n",
      "  1.60     42.78   42.32\n",
      "  1.70     42.73   42.16\n",
      "  1.80     42.66   41.98\n",
      "  1.90     42.57   41.77\n",
      "  2.00     42.46   41.52\n",
      "  2.10     42.32   41.23\n",
      "  2.20     42.15   40.90\n",
      "  2.30     41.94   40.52\n",
      "  2.40     41.68   40.08\n",
      "  2.50     41.35   39.59\n",
      "  2.60     40.96   39.03\n",
      "  2.70     40.47   38.40\n",
      "  2.80     39.88   37.69\n",
      "  2.90     39.17   36.89\n",
      "  3.00     38.32   36.00\n",
      "  3.10     37.31   35.00\n",
      "  3.20     36.13   33.89\n",
      "  3.30     34.77   32.68\n",
      "  3.40     33.21   31.35\n",
      "  3.50     31.47   29.91\n",
      "  3.60     29.54   28.38\n",
      "  3.70     27.44   26.75\n",
      "  3.80     25.20   25.06\n",
      "  3.90     22.85   23.30\n",
      "  4.00     20.41   21.51\n",
      "  4.10     17.94   19.69\n",
      "  4.20     15.46   17.87\n",
      "  4.30     13.00   16.06\n",
      "  4.40     10.61   14.27\n",
      "  4.50      8.32   12.53\n",
      "  4.60      6.16   10.84\n",
      "  4.70      4.14    9.21\n",
      "  4.80      2.30    7.66\n",
      "  4.90      0.64    6.18\n",
      "  5.00     -0.83    4.80\n",
      "  5.10     -2.12    3.51\n",
      "  5.20     -3.24    2.30\n",
      "  5.30     -4.21    1.18\n",
      "  5.40     -5.04    0.13\n",
      "  5.50     -5.76   -0.84\n",
      "  5.60     -6.39   -1.75\n",
      "  5.70     -6.94   -2.60\n",
      "  5.80     -7.44   -3.40\n",
      "  5.90     -7.89   -4.15\n",
      "  6.00     -8.32   -4.86\n",
      "  6.10     -8.72   -5.53\n",
      "  6.20     -9.12   -6.16\n",
      "  6.30     -9.51   -6.75\n",
      "  6.40     -9.89   -7.31\n",
      "  6.50    -10.26   -7.82\n",
      "  6.60    -10.62   -8.31\n",
      "  6.70    -10.96   -8.75\n",
      "  6.80    -11.28   -9.16\n",
      "  6.90    -11.58   -9.54\n",
      "  7.00    -11.85   -9.90\n",
      "  7.10    -12.10  -10.23\n",
      "  7.20    -12.34  -10.54\n",
      "  7.30    -12.55  -10.82\n",
      "  7.40    -12.74  -11.09\n",
      "  7.50    -12.93  -11.35\n",
      "  7.60    -13.11  -11.58\n",
      "  7.70    -13.29  -11.81\n",
      "  7.80    -13.47  -12.02\n",
      "  7.90    -13.66  -12.23\n",
      "  8.00    -13.86  -12.44\n",
      "  8.10    -14.08  -12.64\n",
      "  8.20    -14.31  -12.84\n",
      "  8.30    -14.57  -13.05\n",
      "  8.40    -14.85  -13.27\n",
      "  8.50    -15.17  -13.51\n",
      "  8.60    -15.53  -13.77\n",
      "  8.70    -15.93  -14.06\n",
      "  8.80    -16.37  -14.40\n",
      "  8.90    -16.87  -14.78\n",
      "  9.00    -17.42  -15.22\n",
      "  9.10    -18.04  -15.72\n",
      "  9.20    -18.72  -16.30\n",
      "  9.30    -19.49  -16.96\n",
      "  9.40    -20.34  -17.71\n",
      "  9.50    -21.30  -18.55\n",
      "  9.60    -22.37  -19.50\n",
      "  9.70    -23.56  -20.55\n",
      "  9.80    -24.88  -21.72\n",
      "  9.90    -26.33  -22.98\n",
      " 10.00    -27.91  -24.34\n",
      " 10.10    -29.60  -25.79\n",
      " 10.20    -31.40  -27.31\n",
      " 10.30    -33.26  -28.87\n",
      " 10.40    -35.16  -30.45\n",
      " 10.50    -37.06  -32.01\n",
      " 10.60    -38.92  -33.54\n",
      " 10.70    -40.70  -34.99\n",
      " 10.80    -42.38  -36.37\n",
      " 10.90    -43.92  -37.64\n",
      " 11.00    -45.32  -38.81\n",
      " 11.10    -46.58  -39.88\n",
      " 11.20    -47.70  -40.84\n",
      " 11.30    -48.69  -41.72\n",
      " 11.40    -49.58  -42.51\n",
      " 11.50    -50.38  -43.25\n",
      " 11.60    -51.11  -43.93\n",
      " 11.70    -51.80  -44.59\n",
      " 11.80    -52.46  -45.22\n",
      " 11.90    -53.11  -45.85\n",
      " 12.00    -53.76  -46.49\n",
      " 12.10    -54.43  -47.14\n",
      " 12.20    -55.11  -47.80\n",
      " 12.30    -55.81  -48.49\n",
      " 12.40    -56.51  -49.20\n",
      " 12.50    -57.22  -49.92\n",
      " 12.60    -57.90  -50.65\n",
      " 12.70    -58.56  -51.38\n",
      " 12.80    -59.18  -52.10\n",
      " 12.90    -59.75  -52.81\n",
      " 13.00    -60.27  -53.49\n",
      " 13.10    -60.72  -54.14\n",
      " 13.20    -61.11  -54.75\n",
      " 13.30    -61.45  -55.32\n",
      " 13.40    -61.73  -55.85\n",
      " 13.50    -61.97  -56.34\n",
      " 13.60    -62.17  -56.79\n",
      " 13.70    -62.33  -57.20\n",
      " 13.80    -62.46  -57.56\n",
      " 13.90    -62.57  -57.89\n",
      " 14.00    -62.65  -58.19\n",
      "The pI is  5.41 (folded) and  4.94 (unfolded)\n",
      "\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "INFO:Applying pKa values at a pH of 7.30:\n",
      "INFO:Adding hydrogens to biomolecule.\n",
      "INFO:Debumping biomolecule (again).\n",
      "WARNING:WARNING: Unable to debump LEU A 29\n",
      "INFO:Optimizing hydrogen bonds\n",
      "INFO:Applying force field to biomolecule states.\n",
      "INFO:Regenerating headers.\n",
      "INFO:Regenerating PDB lines.\n",
      "WARNING:Ignoring 467 header lines in output.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "pH: 7.3\tname: AtSAL1_updated\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:PDB2PQR v3.6.2: biomolecular structure conversion software.\n",
      "INFO:Please cite:  Jurrus E, et al.  Improvements to the APBS biomolecular solvation software suite.  Protein Sci 27 112-128 (2018).\n",
      "INFO:Please cite:  Dolinsky TJ, et al.  PDB2PQR: expanding and upgrading automated preparation of biomolecular structures for molecular simulations. Nucleic Acids Res 35 W522-W525 (2007).\n",
      "INFO:Checking and transforming input arguments.\n",
      "INFO:Loading topology files.\n",
      "INFO:Loading molecule: /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/AtAHL.pdb\n",
      "INFO:Setting up molecule.\n",
      "INFO:Created biomolecule object with 368 residues and 5474 atoms.\n",
      "INFO:Setting termini states for biomolecule chains.\n",
      "INFO:Loading forcefield.\n",
      "INFO:Loading hydrogen topology definitions.\n",
      "INFO:This biomolecule is clean.  No repair needed.\n",
      "INFO:Updating disulfide bridges.\n",
      "INFO:Debumping biomolecule.\n",
      "INFO:Assigning titration states with PROPKA.\n",
      "INFO:\n",
      "propka3.5.1                                                                                  2024-07-26\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "--                                                                           --\n",
      "--  PROPKA: A PROTEIN PKA PREDICTOR                                          --\n",
      "--                                                                           --\n",
      "--  VERSION 1.0,  04/25/2004,  IOWA CITY                                     --\n",
      "--  BY HUI LI                                                                --\n",
      "--                                                                           --\n",
      "--  VERSION 2.0,  11/05/2007, IOWA CITY/COPENHAGEN                           --\n",
      "--  BY DELPHINE C. BAS AND DAVID M. ROGERS                                   --\n",
      "--                                                                           --\n",
      "--  VERSION 3.0,  01/06/2011, COPENHAGEN                                     --\n",
      "--  BY MATS H.M. OLSSON AND CHRESTEN R. SONDERGARD                           --\n",
      "--                                                                           --\n",
      "--  VERSION 3.1,  07/01/2011, COPENHAGEN                                     --\n",
      "--  BY CHRESTEN R. SONDERGARD AND MATS H.M. OLSSON                           --\n",
      "--                                                                           --\n",
      "--  VERSION 3.2,  06/17/2020, PLANET EARTH                                   --\n",
      "--  SEE ABOVE FOR AUTHORS                                                    --\n",
      "--                                                                           --\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "References:\n",
      "\n",
      "Very Fast Empirical Prediction and Rationalization of Protein pKa Values.\n",
      "Hui Li, Andrew D. Robertson and Jan H. Jensen. PROTEINS: Structure, Function,\n",
      "and Bioinformatics. 61:704-721 (2005)\n",
      "\n",
      "Very Fast Prediction and Rationalization of pKa Values for Protein-Ligand\n",
      "Complexes.  Delphine C. Bas, David M. Rogers and Jan H. Jensen.  PROTEINS:\n",
      "Structure, Function, and Bioinformatics 73:765-783 (2008)\n",
      "\n",
      "PROPKA3: Consistent Treatment of Internal and Surface Residues in Empirical\n",
      "pKa predictions.  Mats H.M. Olsson, Chresten R. Sondergard, Michal Rostkowski,\n",
      "and Jan H. Jensen.  Journal of Chemical Theory and Computation, 7(2):525-537\n",
      "(2011)\n",
      "\n",
      "Improved Treatment of Ligands and Coupling Effects in Empirical Calculation\n",
      "and Rationalization of pKa Values.  Chresten R. Sondergaard, Mats H.M. Olsson,\n",
      "Michal Rostkowski, and Jan H. Jensen.  Journal of Chemical Theory and\n",
      "Computation, (2011)\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "\n",
      "INFO:Coupling map:\n",
      "\n",
      "INFO:Calculating pKas for Conformation container 1A with 3218 atoms and 894 groups\n",
      "INFO:did not converge in 10 iterations\n",
      "INFO:pKa determinants:\n",
      "\n",
      "---------  -----   ------   ---------------------    --------------    --------------    --------------\n",
      "                            DESOLVATION  EFFECTS       SIDECHAIN          BACKBONE        COULOMBIC\n",
      " RESIDUE    pKa    BURIED     REGULAR      RE        HYDROGEN BOND     HYDROGEN BOND      INTERACTION\n",
      "---------  -----   ------   ---------   ---------    --------------    --------------    --------------\n",
      "\n",
      "ASP   6 A   3.00    28 %    0.95  359   0.08    0   -1.34 ARG  10 A    0.00 XXX   0 X   -0.50 ARG  10 A\n",
      "\n",
      "ASP  38 A   4.10     0 %    0.13  168   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.04 LYS  35 A\n",
      "ASP  38 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 ASP  39 A\n",
      "ASP  38 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP  40 A\n",
      "ASP  38 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP  74 A\n",
      "ASP  38 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 GLU  80 A\n",
      "\n",
      "ASP  39 A   2.55    20 %    0.70  337   0.00    0   -0.85 SER  41 A   -0.18 ASP  39 A   -0.26 LYS  37 A\n",
      "ASP  39 A                                            0.00 XXX   0 X   -0.61 SER  41 A   -0.27 LYS 302 A\n",
      "ASP  39 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.23 GLU  72 A\n",
      "\n",
      "ASP  40 A   4.01     0 %    0.14  138   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP  39 A\n",
      "\n",
      "ASP  47 A   5.77*  100 %    3.69  595   1.08    0   -0.85 THR  44 A   -0.64 THR  44 A    0.41 GLU  72 A\n",
      "ASP  47 A                                           -0.59 THR 144 A   -0.44 ASP  47 A   -0.69 LYS  37 A\n",
      "\n",
      "ASP  64 A   2.40     0 %    0.38  280   0.00    0   -0.30 LYS 120 A    0.00 XXX   0 X   -0.34 LYS 120 A\n",
      "ASP  64 A                                           -0.68 ARG 124 A    0.00 XXX   0 X   -0.46 ARG 124 A\n",
      "\n",
      "ASP  74 A   3.59    27 %    0.81  358   0.01    0    0.00 XXX   0 X   -0.53 GLU  76 A   -0.02 LYS  35 A\n",
      "ASP  74 A                                            0.00 XXX   0 X   -0.39 THR  77 A   -0.11 LYS  37 A\n",
      "ASP  74 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU  80 A\n",
      "\n",
      "ASP  82 A   3.93     0 %    0.12  119   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.07 LYS  35 A\n",
      "ASP  82 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 GLU  80 A\n",
      "\n",
      "ASP 139 A  11.53*  100 %    3.50  634   0.39    0    0.49 ASP 305 A   -0.33 VAL 141 A   -0.41 LYS  37 A\n",
      "ASP 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.21 LYS 279 A\n",
      "ASP 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.50 LYS 302 A\n",
      "ASP 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.40 ASP  47 A\n",
      "ASP 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.37 GLU  73 A\n",
      "ASP 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.29 GLU  72 A\n",
      "ASP 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.51 ASP 142 A\n",
      "ASP 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.21 ASP 305 A\n",
      "\n",
      "ASP 142 A   5.13*  100 %    3.07  665   0.33    0   -0.07 SER 276 A   -0.04 GLY 143 A   -0.34 LYS 293 A\n",
      "ASP 142 A                                            0.00 XXX   0 X   -0.28 MET 277 A   -0.13 LYS 302 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.33 GLU  72 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 249 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.12 GLU 301 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.69 LYS 279 A\n",
      "\n",
      "ASP 151 A   4.73    32 %    0.77  370   0.33    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.15 ARG 149 A\n",
      "ASP 151 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 HIS  27 A\n",
      "\n",
      "ASP 237 A   3.65     0 %    0.27  219   0.00    0    0.00 XXX   0 X   -0.21 ASP 237 A   -0.31 LYS 354 A\n",
      "ASP 237 A                                            0.00 XXX   0 X   -0.00 ASP 238 A    0.10 ASP 238 A\n",
      "\n",
      "ASP 238 A   3.28     0 %    0.44  227   0.00    0    0.00 XXX   0 X   -0.76 VAL 240 A   -0.20 LYS 354 A\n",
      "\n",
      "ASP 287 A   3.73    64 %    2.21  461   0.60    0   -0.54 TYR 175 A    0.00 XXX   0 X   -0.47 LYS 179 A\n",
      "ASP 287 A                                           -0.37 LYS 179 A    0.00 XXX   0 X   -0.85 ARG 285 A\n",
      "ASP 287 A                                           -0.66 ARG 285 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "ASP 305 A   6.18*  100 %    2.94  592   0.45    0   -0.49 ASP 139 A   -0.22 LYS 302 A   -0.07 LYS  37 A\n",
      "ASP 305 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.99 GLU  72 A\n",
      "ASP 305 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.26 ASP 142 A\n",
      "ASP 305 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.66 LYS 279 A\n",
      "ASP 305 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.74 LYS 293 A\n",
      "ASP 305 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.72 GLU 301 A\n",
      "ASP 305 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.89 LYS 302 A\n",
      "ASP 305 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.91 HIS 306 A\n",
      "\n",
      "ASP 322 A   3.09    35 %    1.70  379   0.53    0   -0.75 ARG 326 A   -0.40 GLY 324 A   -0.10 ARG 341 A\n",
      "ASP 322 A                                            0.00 XXX   0 X   -0.03 GLY 325 A    0.28 ASP 340 A\n",
      "ASP 322 A                                            0.00 XXX   0 X   -0.78 ARG 326 A   -0.03 HIS 352 A\n",
      "ASP 322 A                                            0.00 XXX   0 X   -0.72 ASP 340 A   -0.42 ARG 326 A\n",
      "\n",
      "ASP 329 A   2.58    14 %    0.73  320   0.01    0   -0.85 SER 331 A   -0.78 SER 331 A   -0.01 ARG 326 A\n",
      "ASP 329 A                                           -0.01 LYS 332 A    0.00 XXX   0 X   -0.32 LYS 332 A\n",
      "\n",
      "ASP 340 A   2.65    15 %    0.87  323   0.00    0   -1.39 ARG 341 A    0.00 XXX   0 X   -0.19 ARG 326 A\n",
      "ASP 340 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.44 ARG 341 A\n",
      "\n",
      "GLU   2 A   3.46     0 %    0.26  241   0.00    0   -0.85 LYS 208 A    0.00 XXX   0 X   -0.14 N+    1 A\n",
      "GLU   2 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 GLU   4 A\n",
      "GLU   2 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP   6 A\n",
      "GLU   2 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.38 LYS 208 A\n",
      "\n",
      "GLU   4 A   2.86    16 %    0.81  327   0.00    0   -0.85 N+    1 A   -0.68 GLU   4 A   -0.01 LYS 208 A\n",
      "GLU   4 A                                           -0.25 ARG 134 A    0.00 XXX   0 X   -0.41 N+    1 A\n",
      "GLU   4 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.26 ARG 134 A\n",
      "\n",
      "GLU  24 A   4.74     4 %    0.29  293   0.02    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS  21 A\n",
      "GLU  24 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS  25 A\n",
      "GLU  24 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 HIS  27 A\n",
      "GLU  24 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU  96 A\n",
      "\n",
      "GLU  60 A   5.00    13 %    0.50  319   0.03    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.02 LYS 120 A\n",
      "GLU  60 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.07 ARG 124 A\n",
      "GLU  60 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP  64 A\n",
      "GLU  60 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 117 A\n",
      "\n",
      "GLU  72 A   1.96*   90 %    2.13  534   0.00    0   -0.71 TRP 304 A    0.00 XXX   0 X   -1.16 LYS  37 A\n",
      "GLU  72 A                                           -0.85 LYS  37 A    0.00 XXX   0 X   -1.10 LYS 302 A\n",
      "GLU  72 A                                           -0.84 LYS 302 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "GLU  73 A   7.93*   92 %    3.22  539   0.02    0   -0.85 LYS  37 A    0.00 XXX   0 X   -0.23 LYS 302 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP  39 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 ASP  74 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.18 LYS  37 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.24 ASP  47 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.93 GLU  72 A\n",
      "\n",
      "GLU  76 A   4.99     0 %    0.13  203   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP  38 A\n",
      "GLU  76 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.27 ASP  74 A\n",
      "GLU  76 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU  80 A\n",
      "\n",
      "GLU  80 A   3.53     0 %    0.26  214   0.00    0   -0.85 LYS  35 A    0.00 XXX   0 X   -0.38 LYS  35 A\n",
      "\n",
      "GLU  96 A   3.28    17 %    0.51  330   0.04    0   -0.20 LYS  21 A    0.00 XXX   0 X   -0.37 LYS  21 A\n",
      "GLU  96 A                                           -0.76 LYS  25 A    0.00 XXX   0 X   -0.44 LYS  25 A\n",
      "\n",
      "GLU 100 A   4.18    21 %    0.73  339   0.12    0   -0.85 LYS  21 A    0.00 XXX   0 X   -0.01 LYS  25 A\n",
      "GLU 100 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 LYS 108 A\n",
      "GLU 100 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.17 GLU  96 A\n",
      "GLU 100 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.47 LYS  21 A\n",
      "\n",
      "GLU 117 A   3.73     0 %    0.34  255   0.00    0    0.00 XXX   0 X   -0.81 GLY 114 A   -0.26 LYS 111 A\n",
      "GLU 117 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS 120 A\n",
      "\n",
      "GLU 161 A   3.76     0 %    0.38  263   0.00    0   -0.74 ARG 134 A    0.00 XXX   0 X   -0.04 N+    1 A\n",
      "GLU 161 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 LYS 164 A\n",
      "GLU 161 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU   4 A\n",
      "GLU 161 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.38 ARG 134 A\n",
      "\n",
      "GLU 180 A   4.49     0 %    0.11  156   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.12 LYS 178 A\n",
      "\n",
      "GLU 226 A   4.56     0 %    0.06  112   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "GLU 246 A   5.27    37 %    1.07  386   0.02    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.20 ARG 273 A\n",
      "GLU 246 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.12 HIS 255 A\n",
      "\n",
      "GLU 249 A   5.27    46 %    0.93  410   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.10 ARG 149 A\n",
      "GLU 249 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 293 A\n",
      "GLU 249 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP  39 A\n",
      "GLU 249 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 HIS 255 A\n",
      "\n",
      "GLU 289 A   3.91    55 %    1.68  435   0.54    0   -0.85 SER 234 A   -0.79 THR 243 A   -0.18 LYS 179 A\n",
      "GLU 289 A                                           -0.62 THR 243 A   -0.00 GLU 289 A   -0.45 LYS 269 A\n",
      "GLU 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP 287 A\n",
      "\n",
      "GLU 301 A   1.69    92 %    3.27  539   0.00    0   -0.85 LYS 293 A    0.00 XXX   0 X   -0.03 LYS 302 A\n",
      "GLU 301 A                                           -1.60 HIS 306 A    0.00 XXX   0 X   -0.52 LYS 279 A\n",
      "GLU 301 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.62 LYS 293 A\n",
      "GLU 301 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.45 HIS 306 A\n",
      "\n",
      "GLU 314 A   5.47    70 %    1.78  477   0.58    0   -0.74 ARG 209 A    0.00 XXX   0 X    0.16 GLU 315 A\n",
      "GLU 314 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.81 ARG 209 A\n",
      "\n",
      "GLU 315 A   4.02    32 %    1.27  372   0.32    0   -0.15 SER 211 A   -0.81 GLY 210 A   -0.02 LYS 208 A\n",
      "GLU 315 A                                            0.00 XXX   0 X   -0.11 SER 211 A   -0.28 ARG 209 A\n",
      "GLU 315 A                                            0.00 XXX   0 X   -0.70 GLY 212 A    0.00 XXX   0 X\n",
      "\n",
      "GLU 337 A   3.62     0 %    0.36  250   0.00    0   -0.85 LYS 300 A    0.00 XXX   0 X   -0.01 LYS 332 A\n",
      "GLU 337 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.38 LYS 300 A\n",
      "\n",
      "GLU 353 A   3.83     0 %    0.26  212   0.00    0   -0.84 GLN 349 A    0.00 XXX   0 X   -0.01 LYS 354 A\n",
      "GLU 353 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 HIS 352 A\n",
      "\n",
      "GLU 364 A   4.60     0 %    0.12  133   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.01 ARG 341 A\n",
      "\n",
      "C-  368 A   3.32     0 %    0.12   79   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "HIS  13 A   4.57    81 %   -2.69  507   0.00    0    0.82 GLN 217 A    0.00 XXX   0 X   -0.06 ARG  10 A\n",
      "\n",
      "HIS  27 A   6.82     6 %   -0.41  298   0.00    0    0.00 XXX   0 X    0.68 GLN  23 A    0.01 GLU  24 A\n",
      "HIS  27 A                                            0.00 XXX   0 X    0.01 ARG 149 A    0.02 ASP 151 A\n",
      "\n",
      "HIS  33 A   6.06    11 %   -0.44  312   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "HIS 135 A   5.85    35 %   -1.10  379   0.00    0    0.00 XXX   0 X    0.45 LYS 132 A    0.00 XXX   0 X\n",
      "\n",
      "HIS 255 A   6.41    20 %   -0.86  336   0.00    0    0.00 XXX   0 X    0.78 GLU 249 A   -0.09 ARG 250 A\n",
      "HIS 255 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 273 A\n",
      "HIS 255 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 LYS 293 A\n",
      "HIS 255 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.12 GLU 246 A\n",
      "HIS 255 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 249 A\n",
      "\n",
      "HIS 306 A   5.07   100 %   -3.28  641   0.00    0    1.60 GLU 301 A    0.00 XXX   0 X   -1.25 LYS 279 A\n",
      "HIS 306 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.85 LYS 293 A\n",
      "HIS 306 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.45 GLU 301 A\n",
      "HIS 306 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.91 ASP 305 A\n",
      "\n",
      "HIS 352 A   4.72    65 %   -1.90  462   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 322 A\n",
      "HIS 352 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU 353 A\n",
      "\n",
      "CYS  19 A  13.08   100 %    3.78  655   0.00    0    0.00 XXX   0 X   -0.08 CYS  19 A    0.39 ASP  47 A\n",
      "\n",
      "CYS 125 A  11.63    96 %    3.54  549   0.00    0   -0.05 GLN  51 A    0.00 XXX   0 X   -0.23 ARG 124 A\n",
      "CYS 125 A                                           -0.63 SER  55 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "CYS 172 A  13.72   100 %    3.47  673   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.18 ARG 285 A\n",
      "CYS 172 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 ASP 142 A\n",
      "CYS 172 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.35 CYS 203 A\n",
      "\n",
      "CYS 181 A   9.45     5 %    0.55  295   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 178 A\n",
      "CYS 181 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.10 LYS 201 A\n",
      "CYS 181 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 ARG 285 A\n",
      "CYS 181 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.11 GLU 180 A\n",
      "CYS 181 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 287 A\n",
      "\n",
      "CYS 186 A   9.82    31 %    0.88  368   0.00    0    0.00 XXX   0 X   -0.08 CYS 186 A   -0.15 ARG 273 A\n",
      "CYS 186 A                                            0.00 XXX   0 X   -0.02 ASN 187 A    0.19 ASP 151 A\n",
      "\n",
      "CYS 203 A  11.78   100 %    3.38  575   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.65 ARG 285 A\n",
      "CYS 203 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 ASP 287 A\n",
      "\n",
      "CYS 245 A  12.50   100 %    3.40  607   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.11 LYS 279 A\n",
      "CYS 245 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 246 A\n",
      "CYS 245 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 287 A\n",
      "CYS 245 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.12 GLU 289 A\n",
      "\n",
      "CYS 346 A  12.74   100 %    3.74  577   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "TYR 104 A  10.78    56 %    2.22  438   0.00    0   -0.85 SER  17 A   -0.68 GLY 222 A    0.31 GLU 100 A\n",
      "TYR 104 A                                            0.00 XXX   0 X   -0.01 GLY 223 A   -0.21 LYS  21 A\n",
      "\n",
      "TYR 153 A  15.70   100 %    4.01  560   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    1.59 CYS  19 A\n",
      "TYR 153 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU  24 A\n",
      "TYR 153 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 CYS 172 A\n",
      "\n",
      "TYR 175 A  15.46   100 %    3.86  560   0.00    0   -0.21 ARG 285 A    0.00 XXX   0 X    0.40 CYS 172 A\n",
      "TYR 175 A                                            0.54 ASP 287 A    0.00 XXX   0 X    0.54 CYS 203 A\n",
      "TYR 175 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.43 CYS 245 A\n",
      "TYR 175 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.24 ARG 285 A\n",
      "TYR 175 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.15 ASP 287 A\n",
      "\n",
      "TYR 206 A  11.50    48 %    1.46  417   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU   2 A\n",
      "TYR 206 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP   6 A\n",
      "TYR 206 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.00 ARG  10 A\n",
      "\n",
      "TYR 275 A  12.11    30 %    1.75  364   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.17 ARG 149 A\n",
      "TYR 275 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 ASP 151 A\n",
      "TYR 275 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 CYS 186 A\n",
      "TYR 275 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.49 GLU 249 A\n",
      "TYR 275 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 ARG 250 A\n",
      "TYR 275 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 ARG 273 A\n",
      "\n",
      "TYR 280 A  14.55   100 %    3.81  663   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.14 ASP 142 A\n",
      "TYR 280 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.40 GLU 301 A\n",
      "TYR 280 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.12 ASP 305 A\n",
      "TYR 280 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU 314 A\n",
      "\n",
      "TYR 299 A  14.29    74 %    1.49  488   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP  39 A\n",
      "TYR 299 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.23 GLU  72 A\n",
      "TYR 299 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP 139 A\n",
      "TYR 299 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.32 ASP 142 A\n",
      "TYR 299 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU 249 A\n",
      "TYR 299 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 TYR 335 A\n",
      "TYR 299 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.93 GLU 301 A\n",
      "TYR 299 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.05 ASP 305 A\n",
      "\n",
      "TYR 335 A  10.09     6 %    0.28  299   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP  39 A\n",
      "TYR 335 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU  72 A\n",
      "TYR 335 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP 305 A\n",
      "TYR 335 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 337 A\n",
      "TYR 335 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.10 LYS 300 A\n",
      "TYR 335 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.18 LYS 302 A\n",
      "\n",
      "TYR 360 A  10.04     0 %    0.35  261   0.00    0   -0.33 ARG 341 A    0.00 XXX   0 X    0.11 ASP 322 A\n",
      "TYR 360 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.07 ARG 326 A\n",
      "TYR 360 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.28 ASP 340 A\n",
      "TYR 360 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.30 ARG 341 A\n",
      "\n",
      "LYS  21 A  11.63    24 %   -1.00  348   0.00    0    0.20 GLU  96 A    0.00 XXX   0 X    0.03 GLU  24 A\n",
      "LYS  21 A                                            0.85 GLU 100 A    0.00 XXX   0 X    0.21 TYR 104 A\n",
      "LYS  21 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.37 GLU  96 A\n",
      "LYS  21 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.47 GLU 100 A\n",
      "\n",
      "LYS  25 A  11.09    17 %   -0.54  328   0.00    0    0.76 GLU  96 A    0.00 XXX   0 X    0.03 GLU  24 A\n",
      "LYS  25 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 100 A\n",
      "LYS  25 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.11 LYS  21 A\n",
      "LYS  25 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.44 GLU  96 A\n",
      "\n",
      "LYS  35 A  11.28     0 %   -0.58  275   0.00    0    0.85 GLU  80 A    0.00 XXX   0 X    0.04 ASP  38 A\n",
      "LYS  35 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP  74 A\n",
      "LYS  35 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 ASP  82 A\n",
      "LYS  35 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 GLU  80 A\n",
      "\n",
      "LYS  37 A  13.37    74 %   -2.71  488   0.00    0    0.85 GLU  72 A    0.00 XXX   0 X    0.26 ASP  39 A\n",
      "LYS  37 A                                            0.85 GLU  73 A    0.00 XXX   0 X    0.11 ASP  74 A\n",
      "LYS  37 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.41 ASP 139 A\n",
      "LYS  37 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 ASP 305 A\n",
      "LYS  37 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.69 ASP  47 A\n",
      "LYS  37 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.16 GLU  72 A\n",
      "LYS  37 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.18 GLU  73 A\n",
      "\n",
      "LYS 108 A  10.20     0 %   -0.32  220   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 100 A\n",
      "\n",
      "LYS 111 A  10.66     0 %   -0.10  128   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.26 GLU 117 A\n",
      "\n",
      "LYS 120 A  10.80     0 %   -0.19  244   0.00    0    0.30 ASP  64 A    0.00 XXX   0 X    0.02 GLU  60 A\n",
      "LYS 120 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 117 A\n",
      "LYS 120 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.20 ARG 124 A\n",
      "LYS 120 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.34 ASP  64 A\n",
      "\n",
      "LYS 132 A  10.46     0 %   -0.04   93   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 164 A  10.39     0 %   -0.10  198   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 161 A\n",
      "LYS 164 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 209 A\n",
      "\n",
      "LYS 178 A  10.56     0 %   -0.13  165   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.12 GLU 180 A\n",
      "LYS 178 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 CYS 181 A\n",
      "\n",
      "LYS 179 A  10.81    19 %   -0.61  334   0.00    0    0.37 ASP 287 A    0.00 XXX   0 X   -0.11 ARG 285 A\n",
      "LYS 179 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.18 GLU 289 A\n",
      "LYS 179 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.47 ASP 287 A\n",
      "\n",
      "LYS 191 A  10.47     0 %   -0.03   79   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 193 A  10.48     0 %   -0.02   43   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 201 A  10.15    29 %   -0.45  362   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.10 CYS 181 A\n",
      "\n",
      "LYS 208 A  11.45     0 %   -0.31  254   0.00    0    0.85 GLU   2 A    0.00 XXX   0 X    0.01 GLU   4 A\n",
      "LYS 208 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 315 A\n",
      "LYS 208 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 GLU   2 A\n",
      "\n",
      "LYS 232 A  10.41     0 %   -0.09  201   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 269 A  10.30    12 %   -0.57  314   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.45 GLU 289 A\n",
      "LYS 269 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 LYS 179 A\n",
      "\n",
      "LYS 279 A   9.58   100 %   -3.37  661   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.21 ASP 139 A\n",
      "LYS 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.11 CYS 245 A\n",
      "LYS 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.74 LYS 293 A\n",
      "LYS 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.69 ASP 142 A\n",
      "LYS 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.52 GLU 301 A\n",
      "LYS 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.66 ASP 305 A\n",
      "\n",
      "LYS 293 A  10.97    95 %   -3.08  548   0.00    0    0.85 GLU 301 A    0.00 XXX   0 X    0.34 ASP 142 A\n",
      "LYS 293 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 GLU 249 A\n",
      "LYS 293 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 302 A\n",
      "LYS 293 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.62 GLU 301 A\n",
      "LYS 293 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.74 ASP 305 A\n",
      "\n",
      "LYS 300 A  11.52     0 %   -0.32  217   0.00    0    0.85 GLU 337 A    0.00 XXX   0 X    0.10 TYR 335 A\n",
      "LYS 300 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 GLU 337 A\n",
      "\n",
      "LYS 302 A  12.40    70 %   -1.84  477   0.00    0    0.84 GLU  72 A    0.00 XXX   0 X    0.27 ASP  39 A\n",
      "LYS 302 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.23 GLU  73 A\n",
      "LYS 302 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.50 ASP 139 A\n",
      "LYS 302 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 ASP 142 A\n",
      "LYS 302 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 301 A\n",
      "LYS 302 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.44 LYS  37 A\n",
      "LYS 302 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.18 TYR 335 A\n",
      "LYS 302 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.10 GLU  72 A\n",
      "LYS 302 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.89 ASP 305 A\n",
      "\n",
      "LYS 332 A  10.67     0 %   -0.17  202   0.00    0    0.01 ASP 329 A    0.00 XXX   0 X    0.32 ASP 329 A\n",
      "LYS 332 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 337 A\n",
      "\n",
      "LYS 354 A  10.57     0 %   -0.46  252   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.31 ASP 237 A\n",
      "LYS 354 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 ASP 238 A\n",
      "LYS 354 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 353 A\n",
      "\n",
      "ARG  10 A  13.55    29 %   -0.79  363   0.00    0    1.34 ASP   6 A    0.00 XXX   0 X    0.00 TYR 206 A\n",
      "ARG  10 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.50 ASP   6 A\n",
      "\n",
      "ARG 124 A  12.59    43 %   -1.35  401   0.00    0    0.68 ASP  64 A    0.00 XXX   0 X    0.07 GLU  60 A\n",
      "ARG 124 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.23 CYS 125 A\n",
      "ARG 124 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.46 ASP  64 A\n",
      "\n",
      "ARG 134 A  13.76     4 %   -0.37  293   0.00    0    0.25 GLU   4 A    0.00 XXX   0 X    0.26 GLU   4 A\n",
      "ARG 134 A                                            0.74 GLU 161 A    0.00 XXX   0 X    0.38 GLU 161 A\n",
      "\n",
      "ARG 149 A  12.62     0 %   -0.29  273   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP 151 A\n",
      "ARG 149 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 GLU 249 A\n",
      "ARG 149 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.17 TYR 275 A\n",
      "\n",
      "ARG 209 A  12.83    59 %   -1.51  447   0.00    0    0.74 GLU 314 A    0.00 XXX   0 X    0.28 GLU 315 A\n",
      "ARG 209 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.81 GLU 314 A\n",
      "\n",
      "ARG 250 A  12.37     0 %   -0.09  157   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.01 TYR 275 A\n",
      "ARG 250 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.05 ARG 273 A\n",
      "\n",
      "ARG 268 A  12.32     0 %   -0.18  156   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "ARG 273 A  12.69     0 %   -0.19  279   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.15 CYS 186 A\n",
      "ARG 273 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 GLU 246 A\n",
      "ARG 273 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 TYR 275 A\n",
      "\n",
      "ARG 285 A  13.75    70 %   -2.59  478   0.00    0    0.21 TYR 175 A    0.00 XXX   0 X    0.18 CYS 172 A\n",
      "ARG 285 A                                            0.66 ASP 287 A    0.00 XXX   0 X    1.24 TYR 175 A\n",
      "ARG 285 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 CYS 181 A\n",
      "ARG 285 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.65 CYS 203 A\n",
      "ARG 285 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.85 ASP 287 A\n",
      "\n",
      "ARG 326 A  13.60     0 %   -0.31  246   0.00    0    0.75 ASP 322 A    0.00 XXX   0 X    0.01 ASP 329 A\n",
      "ARG 326 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 ASP 340 A\n",
      "ARG 326 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 TYR 360 A\n",
      "ARG 326 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 ARG 341 A\n",
      "ARG 326 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.42 ASP 322 A\n",
      "\n",
      "ARG 341 A  14.03    17 %   -1.04  330   0.00    0    0.33 TYR 360 A    0.00 XXX   0 X    0.10 ASP 322 A\n",
      "ARG 341 A                                            1.39 ASP 340 A    0.00 XXX   0 X    0.30 TYR 360 A\n",
      "ARG 341 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 364 A\n",
      "ARG 341 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.44 ASP 340 A\n",
      "\n",
      "N+    1 A   8.68     0 %   -0.50  274   0.00    0    0.85 GLU   4 A    0.00 XXX   0 X    0.14 GLU   2 A\n",
      "N+    1 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.16 ARG 134 A\n",
      "N+    1 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 GLU 161 A\n",
      "N+    1 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.10 LYS 208 A\n",
      "N+    1 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.41 GLU   4 A\n",
      "\n",
      "Coupled residues (marked *) were detected.Please rerun PropKa with the --display-coupled-residues \n",
      "or -d option for detailed information.\n",
      "\n",
      "INFO:pKa summary:\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "SUMMARY OF THIS PREDICTION\n",
      "       Group      pKa  model-pKa   ligand atom-type\n",
      "   ASP   6 A     3.00       3.80                      \n",
      "   ASP  38 A     4.10       3.80                      \n",
      "   ASP  39 A     2.55       3.80                      \n",
      "   ASP  40 A     4.01       3.80                      \n",
      "   ASP  47 A     5.77       3.80                      \n",
      "   ASP  64 A     2.40       3.80                      \n",
      "   ASP  74 A     3.59       3.80                      \n",
      "   ASP  82 A     3.93       3.80                      \n",
      "   ASP 139 A    11.53       3.80                      \n",
      "   ASP 142 A     5.13       3.80                      \n",
      "   ASP 151 A     4.73       3.80                      \n",
      "   ASP 237 A     3.65       3.80                      \n",
      "   ASP 238 A     3.28       3.80                      \n",
      "   ASP 287 A     3.73       3.80                      \n",
      "   ASP 305 A     6.18       3.80                      \n",
      "   ASP 322 A     3.09       3.80                      \n",
      "   ASP 329 A     2.58       3.80                      \n",
      "   ASP 340 A     2.65       3.80                      \n",
      "   GLU   2 A     3.46       4.50                      \n",
      "   GLU   4 A     2.86       4.50                      \n",
      "   GLU  24 A     4.74       4.50                      \n",
      "   GLU  60 A     5.00       4.50                      \n",
      "   GLU  72 A     1.96       4.50                      \n",
      "   GLU  73 A     7.93       4.50                      \n",
      "   GLU  76 A     4.99       4.50                      \n",
      "   GLU  80 A     3.53       4.50                      \n",
      "   GLU  96 A     3.28       4.50                      \n",
      "   GLU 100 A     4.18       4.50                      \n",
      "   GLU 117 A     3.73       4.50                      \n",
      "   GLU 161 A     3.76       4.50                      \n",
      "   GLU 180 A     4.49       4.50                      \n",
      "   GLU 226 A     4.56       4.50                      \n",
      "   GLU 246 A     5.27       4.50                      \n",
      "   GLU 249 A     5.27       4.50                      \n",
      "   GLU 289 A     3.91       4.50                      \n",
      "   GLU 301 A     1.69       4.50                      \n",
      "   GLU 314 A     5.47       4.50                      \n",
      "   GLU 315 A     4.02       4.50                      \n",
      "   GLU 337 A     3.62       4.50                      \n",
      "   GLU 353 A     3.83       4.50                      \n",
      "   GLU 364 A     4.60       4.50                      \n",
      "   C-  368 A     3.32       3.20                      \n",
      "   HIS  13 A     4.57       6.50                      \n",
      "   HIS  27 A     6.82       6.50                      \n",
      "   HIS  33 A     6.06       6.50                      \n",
      "   HIS 135 A     5.85       6.50                      \n",
      "   HIS 255 A     6.41       6.50                      \n",
      "   HIS 306 A     5.07       6.50                      \n",
      "   HIS 352 A     4.72       6.50                      \n",
      "   CYS  19 A    13.08       9.00                      \n",
      "   CYS 125 A    11.63       9.00                      \n",
      "   CYS 172 A    13.72       9.00                      \n",
      "   CYS 181 A     9.45       9.00                      \n",
      "   CYS 186 A     9.82       9.00                      \n",
      "   CYS 203 A    11.78       9.00                      \n",
      "   CYS 245 A    12.50       9.00                      \n",
      "   CYS 346 A    12.74       9.00                      \n",
      "   TYR 104 A    10.78      10.00                      \n",
      "   TYR 153 A    15.70      10.00                      \n",
      "   TYR 175 A    15.46      10.00                      \n",
      "   TYR 206 A    11.50      10.00                      \n",
      "   TYR 275 A    12.11      10.00                      \n",
      "   TYR 280 A    14.55      10.00                      \n",
      "   TYR 299 A    14.29      10.00                      \n",
      "   TYR 335 A    10.09      10.00                      \n",
      "   TYR 360 A    10.04      10.00                      \n",
      "   LYS  21 A    11.63      10.50                      \n",
      "   LYS  25 A    11.09      10.50                      \n",
      "   LYS  35 A    11.28      10.50                      \n",
      "   LYS  37 A    13.37      10.50                      \n",
      "   LYS 108 A    10.20      10.50                      \n",
      "   LYS 111 A    10.66      10.50                      \n",
      "   LYS 120 A    10.80      10.50                      \n",
      "   LYS 132 A    10.46      10.50                      \n",
      "   LYS 164 A    10.39      10.50                      \n",
      "   LYS 178 A    10.56      10.50                      \n",
      "   LYS 179 A    10.81      10.50                      \n",
      "   LYS 191 A    10.47      10.50                      \n",
      "   LYS 193 A    10.48      10.50                      \n",
      "   LYS 201 A    10.15      10.50                      \n",
      "   LYS 208 A    11.45      10.50                      \n",
      "   LYS 232 A    10.41      10.50                      \n",
      "   LYS 269 A    10.30      10.50                      \n",
      "   LYS 279 A     9.58      10.50                      \n",
      "   LYS 293 A    10.97      10.50                      \n",
      "   LYS 300 A    11.52      10.50                      \n",
      "   LYS 302 A    12.40      10.50                      \n",
      "   LYS 332 A    10.67      10.50                      \n",
      "   LYS 354 A    10.57      10.50                      \n",
      "   ARG  10 A    13.55      12.50                      \n",
      "   ARG 124 A    12.59      12.50                      \n",
      "   ARG 134 A    13.76      12.50                      \n",
      "   ARG 149 A    12.62      12.50                      \n",
      "   ARG 209 A    12.83      12.50                      \n",
      "   ARG 250 A    12.37      12.50                      \n",
      "   ARG 268 A    12.32      12.50                      \n",
      "   ARG 273 A    12.69      12.50                      \n",
      "   ARG 285 A    13.75      12.50                      \n",
      "   ARG 326 A    13.60      12.50                      \n",
      "   ARG 341 A    14.03      12.50                      \n",
      "   N+    1 A     8.68       8.00                      \n",
      "\n",
      "INFO:PROPKA information:\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "Free energy of   folding (kcal/mol) as a function of pH (using neutral reference)\n",
      "  0.00     28.05\n",
      "  1.00     27.81\n",
      "  2.00     26.19\n",
      "  3.00     20.32\n",
      "  4.00     11.83\n",
      "  5.00     10.29\n",
      "  6.00     12.13\n",
      "  7.00     13.31\n",
      "  8.00     15.91\n",
      "  9.00     20.92\n",
      " 10.00     32.09\n",
      " 11.00     51.34\n",
      " 12.00     71.39\n",
      " 13.00     87.31\n",
      " 14.00     98.55\n",
      "\n",
      "The pH of optimum stability is  4.6 for which the free energy is    9.9 kcal/mol at 298K\n",
      "Could not determine pH values where the free energy is within 80 % of minimum\n",
      "Could not determine the pH-range where the free energy is negative\n",
      "\n",
      "\n",
      "Protein charge of folded and unfolded state as a function of pH\n",
      "    pH  unfolded  folded\n",
      "  0.00     42.00   41.95\n",
      "  0.10     41.99   41.94\n",
      "  0.20     41.99   41.92\n",
      "  0.30     41.99   41.90\n",
      "  0.40     41.99   41.88\n",
      "  0.50     41.99   41.85\n",
      "  0.60     41.98   41.81\n",
      "  0.70     41.98   41.76\n",
      "  0.80     41.97   41.70\n",
      "  0.90     41.97   41.63\n",
      "  1.00     41.96   41.54\n",
      "  1.10     41.95   41.44\n",
      "  1.20     41.93   41.31\n",
      "  1.30     41.92   41.16\n",
      "  1.40     41.89   40.98\n",
      "  1.50     41.87   40.78\n",
      "  1.60     41.83   40.54\n",
      "  1.70     41.79   40.26\n",
      "  1.80     41.74   39.95\n",
      "  1.90     41.67   39.59\n",
      "  2.00     41.59   39.19\n",
      "  2.10     41.48   38.75\n",
      "  2.20     41.35   38.25\n",
      "  2.30     41.19   37.70\n",
      "  2.40     40.99   37.10\n",
      "  2.50     40.75   36.44\n",
      "  2.60     40.44   35.72\n",
      "  2.70     40.08   34.94\n",
      "  2.80     39.63   34.10\n",
      "  2.90     39.09   33.19\n",
      "  3.00     38.44   32.23\n",
      "  3.10     37.68   31.20\n",
      "  3.20     36.79   30.11\n",
      "  3.30     35.75   28.97\n",
      "  3.40     34.56   27.77\n",
      "  3.50     33.23   26.53\n",
      "  3.60     31.74   25.25\n",
      "  3.70     30.11   23.95\n",
      "  3.80     28.36   22.63\n",
      "  3.90     26.50   21.29\n",
      "  4.00     24.55   19.97\n",
      "  4.10     22.54   18.65\n",
      "  4.20     20.50   17.34\n",
      "  4.30     18.46   16.06\n",
      "  4.40     16.44   14.80\n",
      "  4.50     14.47   13.57\n",
      "  4.60     12.59   12.37\n",
      "  4.70     10.83   11.19\n",
      "  4.80      9.20   10.06\n",
      "  4.90      7.72    8.95\n",
      "  5.00      6.39    7.89\n",
      "  5.10      5.22    6.87\n",
      "  5.20      4.19    5.90\n",
      "  5.30      3.29    4.97\n",
      "  5.40      2.50    4.10\n",
      "  5.50      1.81    3.27\n",
      "  5.60      1.19    2.50\n",
      "  5.70      0.62    1.77\n",
      "  5.80      0.10    1.08\n",
      "  5.90     -0.40    0.44\n",
      "  6.00     -0.88   -0.15\n",
      "  6.10     -1.36   -0.71\n",
      "  6.20     -1.85   -1.22\n",
      "  6.30     -2.33   -1.69\n",
      "  6.40     -2.82   -2.13\n",
      "  6.50     -3.30   -2.53\n",
      "  6.60     -3.77   -2.89\n",
      "  6.70     -4.22   -3.22\n",
      "  6.80     -4.65   -3.52\n",
      "  6.90     -5.05   -3.79\n",
      "  7.00     -5.42   -4.03\n",
      "  7.10     -5.76   -4.25\n",
      "  7.20     -6.07   -4.45\n",
      "  7.30     -6.36   -4.64\n",
      "  7.40     -6.62   -4.81\n",
      "  7.50     -6.87   -4.97\n",
      "  7.60     -7.12   -5.12\n",
      "  7.70     -7.36   -5.27\n",
      "  7.80     -7.62   -5.42\n",
      "  7.90     -7.88   -5.57\n",
      "  8.00     -8.17   -5.72\n",
      "  8.10     -8.48   -5.88\n",
      "  8.20     -8.82   -6.04\n",
      "  8.30     -9.20   -6.22\n",
      "  8.40     -9.63   -6.40\n",
      "  8.50    -10.11   -6.59\n",
      "  8.60    -10.65   -6.81\n",
      "  8.70    -11.25   -7.04\n",
      "  8.80    -11.91   -7.29\n",
      "  8.90    -12.63   -7.57\n",
      "  9.00    -13.41   -7.89\n",
      "  9.10    -14.26   -8.24\n",
      "  9.20    -15.17   -8.64\n",
      "  9.30    -16.14   -9.09\n",
      "  9.40    -17.18   -9.59\n",
      "  9.50    -18.30  -10.16\n",
      "  9.60    -19.51  -10.80\n",
      "  9.70    -20.81  -11.51\n",
      "  9.80    -22.22  -12.29\n",
      "  9.90    -23.72  -13.16\n",
      " 10.00    -25.32  -14.10\n",
      " 10.10    -27.01  -15.11\n",
      " 10.20    -28.77  -16.20\n",
      " 10.30    -30.57  -17.34\n",
      " 10.40    -32.39  -18.53\n",
      " 10.50    -34.20  -19.75\n",
      " 10.60    -35.95  -21.00\n",
      " 10.70    -37.62  -22.26\n",
      " 10.80    -39.18  -23.51\n",
      " 10.90    -40.61  -24.76\n",
      " 11.00    -41.91  -25.98\n",
      " 11.10    -43.08  -27.19\n",
      " 11.20    -44.11  -28.36\n",
      " 11.30    -45.04  -29.51\n",
      " 11.40    -45.86  -30.63\n",
      " 11.50    -46.61  -31.73\n",
      " 11.60    -47.30  -32.79\n",
      " 11.70    -47.95  -33.84\n",
      " 11.80    -48.58  -34.85\n",
      " 11.90    -49.21  -35.85\n",
      " 12.00    -49.84  -36.83\n",
      " 12.10    -50.49  -37.80\n",
      " 12.20    -51.16  -38.75\n",
      " 12.30    -51.85  -39.69\n",
      " 12.40    -52.54  -40.62\n",
      " 12.50    -53.24  -41.54\n",
      " 12.60    -53.92  -42.44\n",
      " 12.70    -54.58  -43.33\n",
      " 12.80    -55.20  -44.20\n",
      " 12.90    -55.76  -45.04\n",
      " 13.00    -56.27  -45.87\n",
      " 13.10    -56.73  -46.67\n",
      " 13.20    -57.12  -47.44\n",
      " 13.30    -57.45  -48.19\n",
      " 13.40    -57.74  -48.93\n",
      " 13.50    -57.97  -49.64\n",
      " 13.60    -58.17  -50.32\n",
      " 13.70    -58.33  -50.98\n",
      " 13.80    -58.46  -51.62\n",
      " 13.90    -58.57  -52.22\n",
      " 14.00    -58.65  -52.79\n",
      "The pI is  5.97 (folded) and  5.82 (unfolded)\n",
      "\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "INFO:Applying pKa values at a pH of 7.30:\n",
      "INFO:Adding hydrogens to biomolecule.\n",
      "INFO:Debumping biomolecule (again).\n",
      "INFO:Optimizing hydrogen bonds\n",
      "INFO:Applying force field to biomolecule states.\n",
      "INFO:Regenerating headers.\n",
      "INFO:Regenerating PDB lines.\n",
      "WARNING:Ignoring 466 header lines in output.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "pH: 7.3\tname: AtAHL\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:PDB2PQR v3.6.2: biomolecular structure conversion software.\n",
      "INFO:Please cite:  Jurrus E, et al.  Improvements to the APBS biomolecular solvation software suite.  Protein Sci 27 112-128 (2018).\n",
      "INFO:Please cite:  Dolinsky TJ, et al.  PDB2PQR: expanding and upgrading automated preparation of biomolecular structures for molecular simulations. Nucleic Acids Res 35 W522-W525 (2007).\n",
      "INFO:Checking and transforming input arguments.\n",
      "INFO:Loading topology files.\n",
      "INFO:Loading molecule: /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/74987.pdb\n",
      "INFO:Setting up molecule.\n",
      "INFO:Created biomolecule object with 352 residues and 5354 atoms.\n",
      "INFO:Setting termini states for biomolecule chains.\n",
      "INFO:Loading forcefield.\n",
      "INFO:Loading hydrogen topology definitions.\n",
      "INFO:This biomolecule is clean.  No repair needed.\n",
      "INFO:Updating disulfide bridges.\n",
      "INFO:Debumping biomolecule.\n",
      "INFO:Assigning titration states with PROPKA.\n",
      "INFO:\n",
      "propka3.5.1                                                                                  2024-07-26\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "--                                                                           --\n",
      "--  PROPKA: A PROTEIN PKA PREDICTOR                                          --\n",
      "--                                                                           --\n",
      "--  VERSION 1.0,  04/25/2004,  IOWA CITY                                     --\n",
      "--  BY HUI LI                                                                --\n",
      "--                                                                           --\n",
      "--  VERSION 2.0,  11/05/2007, IOWA CITY/COPENHAGEN                           --\n",
      "--  BY DELPHINE C. BAS AND DAVID M. ROGERS                                   --\n",
      "--                                                                           --\n",
      "--  VERSION 3.0,  01/06/2011, COPENHAGEN                                     --\n",
      "--  BY MATS H.M. OLSSON AND CHRESTEN R. SONDERGARD                           --\n",
      "--                                                                           --\n",
      "--  VERSION 3.1,  07/01/2011, COPENHAGEN                                     --\n",
      "--  BY CHRESTEN R. SONDERGARD AND MATS H.M. OLSSON                           --\n",
      "--                                                                           --\n",
      "--  VERSION 3.2,  06/17/2020, PLANET EARTH                                   --\n",
      "--  SEE ABOVE FOR AUTHORS                                                    --\n",
      "--                                                                           --\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "References:\n",
      "\n",
      "Very Fast Empirical Prediction and Rationalization of Protein pKa Values.\n",
      "Hui Li, Andrew D. Robertson and Jan H. Jensen. PROTEINS: Structure, Function,\n",
      "and Bioinformatics. 61:704-721 (2005)\n",
      "\n",
      "Very Fast Prediction and Rationalization of pKa Values for Protein-Ligand\n",
      "Complexes.  Delphine C. Bas, David M. Rogers and Jan H. Jensen.  PROTEINS:\n",
      "Structure, Function, and Bioinformatics 73:765-783 (2008)\n",
      "\n",
      "PROPKA3: Consistent Treatment of Internal and Surface Residues in Empirical\n",
      "pKa predictions.  Mats H.M. Olsson, Chresten R. Sondergard, Michal Rostkowski,\n",
      "and Jan H. Jensen.  Journal of Chemical Theory and Computation, 7(2):525-537\n",
      "(2011)\n",
      "\n",
      "Improved Treatment of Ligands and Coupling Effects in Empirical Calculation\n",
      "and Rationalization of pKa Values.  Chresten R. Sondergaard, Mats H.M. Olsson,\n",
      "Michal Rostkowski, and Jan H. Jensen.  Journal of Chemical Theory and\n",
      "Computation, (2011)\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "\n",
      "INFO:Coupling map:\n",
      "\n",
      "INFO:Calculating pKas for Conformation container 1A with 3171 atoms and 875 groups\n",
      "INFO:pKa determinants:\n",
      "\n",
      "---------  -----   ------   ---------------------    --------------    --------------    --------------\n",
      "                            DESOLVATION  EFFECTS       SIDECHAIN          BACKBONE        COULOMBIC\n",
      " RESIDUE    pKa    BURIED     REGULAR      RE        HYDROGEN BOND     HYDROGEN BOND      INTERACTION\n",
      "---------  -----   ------   ---------   ---------    --------------    --------------    --------------\n",
      "\n",
      "ASP  10 A   4.95    44 %    1.49  405   0.14    0   -0.28 ARG  14 A    0.00 XXX   0 X   -0.05 ARG  17 A\n",
      "ASP  10 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.10 LYS  59 A\n",
      "ASP  10 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU   6 A\n",
      "ASP  10 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP  13 A\n",
      "ASP  10 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.29 GLU  56 A\n",
      "ASP  10 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.55 ARG  14 A\n",
      "\n",
      "ASP  13 A   4.41    60 %    1.88  448   0.32    0   -0.72 ARG  17 A   -0.16 GLN 194 A   -0.08 ARG  14 A\n",
      "ASP  13 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.00 LYS  91 A\n",
      "ASP  13 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.05 LYS 191 A\n",
      "ASP  13 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.59 ARG  17 A\n",
      "\n",
      "ASP  21 A   2.92    25 %    0.74  352   0.19    0   -0.11 ARG  24 A    0.00 XXX   0 X   -0.03 ARG  17 A\n",
      "ASP  21 A                                           -0.85 LYS  91 A    0.00 XXX   0 X   -0.32 ARG  24 A\n",
      "ASP  21 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.49 LYS  91 A\n",
      "\n",
      "ASP  38 A   2.45    12 %    0.70  316   0.00    0   -0.85 THR  40 A   -0.20 ASP  38 A   -0.26 LYS  36 A\n",
      "ASP  38 A                                            0.00 XXX   0 X   -0.66 THR  40 A   -0.08 LYS 286 A\n",
      "\n",
      "ASP  46 A   5.88   100 %    3.67  601   1.00    0   -0.85 THR  43 A   -0.70 THR  43 A   -0.59 LYS  36 A\n",
      "ASP  46 A                                           -0.64 THR 134 A   -0.38 ASP  46 A    0.56 GLU  69 A\n",
      "\n",
      "ASP  71 A   3.56    35 %    0.91  380   0.00    0    0.00 XXX   0 X   -0.46 HIS  73 A   -0.12 LYS  36 A\n",
      "ASP  71 A                                            0.00 XXX   0 X   -0.28 PHE  74 A   -0.05 ARG  76 A\n",
      "ASP  71 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.21 LYS 113 A\n",
      "ASP  71 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 322 A\n",
      "ASP  71 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.19 HIS  73 A\n",
      "ASP  71 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.18 ASP 109 A\n",
      "\n",
      "ASP  96 A   3.95     0 %    0.24  216   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.09 LYS  89 A\n",
      "ASP  96 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.00 LYS  91 A\n",
      "\n",
      "ASP 101 A   3.94     0 %    0.32  265   0.00    0    0.00 XXX   0 X   -0.03 ASP 101 A   -0.16 LYS  89 A\n",
      "ASP 101 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 103 A\n",
      "\n",
      "ASP 103 A   2.46     0 %    0.37  271   0.00    0   -0.85 SER 100 A   -0.81 SER 100 A   -0.04 LYS  89 A\n",
      "ASP 103 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 HIS  98 A\n",
      "\n",
      "ASP 109 A   2.64    27 %    1.12  356   0.23    0   -0.85 SER  72 A   -0.79 SER  72 A   -0.02 HIS  73 A\n",
      "ASP 109 A                                           -0.48 ARG  76 A    0.00 XXX   0 X   -0.37 ARG  76 A\n",
      "\n",
      "ASP 114 A   3.91     0 %    0.33  245   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.29 LYS 113 A\n",
      "ASP 114 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 ARG 322 A\n",
      "ASP 114 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.11 ASP  71 A\n",
      "ASP 114 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 109 A\n",
      "\n",
      "ASP 129 A  11.71*  100 %    3.55  628   0.27    0    0.42 ASP 289 A   -0.23 ILE 131 A   -0.17 LYS  36 A\n",
      "ASP 129 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.32 ASP  46 A\n",
      "ASP 129 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 GLU  70 A\n",
      "ASP 129 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.98 GLU  69 A\n",
      "ASP 129 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.31 ASP 132 A\n",
      "ASP 129 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.28 ASP 289 A\n",
      "\n",
      "ASP 132 A   6.29*  100 %    3.80  623   0.65    0   -0.13 SER 260 A   -0.48 SER 260 A    0.11 GLU  69 A\n",
      "ASP 132 A                                            0.00 XXX   0 X   -0.56 LEU 261 A   -0.70 LYS 263 A\n",
      "ASP 132 A                                            0.00 XXX   0 X   -0.19 CYS 262 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 141 A   3.75     0 %    0.27  267   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.31 LYS  23 A\n",
      "ASP 141 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 ARG  24 A\n",
      "\n",
      "ASP 154 A   4.07     0 %    0.28  239   0.00    0    0.00 XXX   0 X   -0.00 ASP 154 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 170 A   3.21     0 %    0.24  274   0.00    0   -0.42 ARG 271 A    0.00 XXX   0 X   -0.03 LYS 192 A\n",
      "ASP 170 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.37 ARG 271 A\n",
      "\n",
      "ASP 173 A   1.41     0 %    0.49  200   0.00    0   -0.85 SER 175 A   -0.08 ASP 173 A   -0.38 LYS 192 A\n",
      "ASP 173 A                                           -0.81 LYS 192 A   -0.76 SER 175 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 203 A   3.82     0 %    0.12  132   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.10 LYS 191 A\n",
      "\n",
      "ASP 210 A   2.58    45 %    1.85  406   0.24    0   -0.85 SER 273 A   -0.41 CYS 212 A   -0.43 ARG 219 A\n",
      "ASP 210 A                                           -0.78 ASN 334 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "ASP 210 A                                           -0.85 ARG 219 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "ASP 238 A   4.04     0 %    0.23  233   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.09 LYS 247 A\n",
      "ASP 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 249 A\n",
      "ASP 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 229 A\n",
      "ASP 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 GLU 250 A\n",
      "\n",
      "ASP 245 A   4.00     0 %    0.16  109   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.04 GLU 250 A\n",
      "\n",
      "ASP 289 A   7.94*  100 %    2.62  580   0.38    0   -0.42 ASP 129 A    0.00 XXX   0 X   -0.01 LYS  36 A\n",
      "ASP 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.00 LYS 263 A\n",
      "ASP 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.19 LYS 286 A\n",
      "ASP 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.76 GLU  69 A\n",
      "ASP 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.00 ASP 132 A\n",
      "\n",
      "ASP 306 A   3.95    39 %    1.62  391   0.52    0    0.00 XXX   0 X   -0.01 ASP 306 A   -0.01 ARG 283 A\n",
      "ASP 306 A                                            0.00 XXX   0 X   -0.48 GLU 308 A   -0.01 HIS 338 A\n",
      "ASP 306 A                                            0.00 XXX   0 X   -0.34 GLY 309 A    0.00 XXX   0 X\n",
      "ASP 306 A                                            0.00 XXX   0 X   -0.61 ASP 310 A    0.00 XXX   0 X\n",
      "ASP 306 A                                            0.00 XXX   0 X   -0.53 ALA 327 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 310 A   4.33     0 %    0.22  242   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.00 LYS 303 A\n",
      "ASP 310 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.28 ASP 306 A\n",
      "ASP 310 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 311 A\n",
      "\n",
      "ASP 317 A   3.26    37 %    1.59  385   0.12    0    0.00 XXX   0 X   -0.51 GLU 320 A   -0.05 LYS 286 A\n",
      "ASP 317 A                                            0.00 XXX   0 X   -0.03 ARG 321 A   -0.13 ARG 321 A\n",
      "ASP 317 A                                            0.00 XXX   0 X   -0.72 ARG 322 A   -0.10 ARG 322 A\n",
      "ASP 317 A                                            0.00 XXX   0 X   -0.69 LEU 323 A    0.00 XXX   0 X\n",
      "\n",
      "GLU   4 A   4.10    15 %    0.77  322   0.00    0    0.00 XXX   0 X   -0.76 GLU   4 A   -0.26 N+    1 A\n",
      "GLU   4 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.07 LYS   3 A\n",
      "GLU   4 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 HIS   2 A\n",
      "GLU   4 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 HIS 183 A\n",
      "\n",
      "GLU   6 A   4.46     0 %    0.19  222   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.05 LYS   3 A\n",
      "GLU   6 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.18 HIS   2 A\n",
      "\n",
      "GLU  31 A   4.62     6 %    0.36  298   0.00    0   -0.24 SER  29 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "GLU  35 A   4.73     0 %    0.22  190   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP  38 A\n",
      "\n",
      "GLU  56 A   2.83    47 %    1.86  412   0.21    0   -0.78 SER  93 A   -0.83 SER  93 A   -0.52 ARG  14 A\n",
      "GLU  56 A                                           -0.18 ARG  14 A    0.00 XXX   0 X   -0.58 LYS  59 A\n",
      "GLU  56 A                                           -0.85 LYS  59 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "GLU  69 A   4.06*   92 %    1.97  539   0.00    0   -0.35 TRP 288 A    0.00 XXX   0 X   -0.18 LYS 286 A\n",
      "GLU  69 A                                           -0.85 LYS  36 A    0.00 XXX   0 X   -0.02 ARG 322 A\n",
      "GLU  69 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 ASP  38 A\n",
      "GLU  69 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.20 LYS  36 A\n",
      "\n",
      "GLU  70 A   8.36*   97 %    3.42  553   0.03    0   -0.85 LYS  36 A    0.00 XXX   0 X    0.08 ASP  38 A\n",
      "GLU  70 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 ASP  71 A\n",
      "GLU  70 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.29 LYS  36 A\n",
      "GLU  70 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.00 ASP  46 A\n",
      "GLU  70 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.27 GLU  69 A\n",
      "\n",
      "GLU  87 A   5.73    40 %    1.00  393   0.16    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.00 ARG  24 A\n",
      "GLU  87 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.14 LYS  91 A\n",
      "GLU  87 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.17 ASP  21 A\n",
      "GLU  87 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU  31 A\n",
      "\n",
      "GLU 106 A   4.79     0 %    0.16  269   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 110 A\n",
      "GLU 106 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.11 ASP 103 A\n",
      "GLU 106 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 109 A\n",
      "\n",
      "GLU 142 A   4.62     0 %    0.09  177   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.01 LYS  23 A\n",
      "GLU 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 141 A\n",
      "\n",
      "GLU 156 A   5.01    18 %    0.37  332   0.02    0    0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 154 A\n",
      "GLU 156 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 HIS 298 A\n",
      "GLU 156 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.14 GLU 311 A\n",
      "GLU 156 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 315 A\n",
      "\n",
      "GLU 224 A   2.07    12 %    0.73  314   0.02    0   -0.52 ARG 135 A    0.00 XXX   0 X   -0.16 LYS 263 A\n",
      "GLU 224 A                                           -0.85 LYS 139 A    0.00 XXX   0 X   -0.42 ARG 135 A\n",
      "GLU 224 A                                           -0.84 CYS 258 A    0.00 XXX   0 X   -0.39 LYS 139 A\n",
      "\n",
      "GLU 229 A   3.42     0 %    0.34  244   0.00    0   -0.66 THR 227 A   -0.76 GLU 229 A    0.00 XXX   0 X\n",
      "\n",
      "GLU 244 A   4.57     0 %    0.07   82   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "GLU 250 A   2.96     0 %    0.53  224   0.00    0   -0.85 LYS 247 A   -0.84 LYS 247 A   -0.00 LYS 217 A\n",
      "GLU 250 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.38 LYS 247 A\n",
      "\n",
      "GLU 299 A   4.11    28 %    1.15  361   0.24    0    0.00 XXX   0 X   -0.79 GLY 185 A   -0.13 ARG 206 A\n",
      "GLU 299 A                                            0.00 XXX   0 X   -0.08 CYS 186 A   -0.10 HIS 183 A\n",
      "GLU 299 A                                            0.00 XXX   0 X   -0.58 GLY 187 A   -0.10 HIS 298 A\n",
      "\n",
      "GLU 308 A   5.24     0 %    0.26  271   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.10 ARG 283 A\n",
      "GLU 308 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.24 ASP 306 A\n",
      "GLU 308 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP 310 A\n",
      "GLU 308 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 C-  352 A\n",
      "\n",
      "GLU 311 A   3.29    11 %    0.28  311   0.00    0   -0.85 LYS 303 A    0.00 XXX   0 X   -0.21 HIS 298 A\n",
      "GLU 311 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.42 LYS 303 A\n",
      "\n",
      "GLU 315 A   4.29     0 %    0.41  259   0.00    0   -0.39 ASN 155 A    0.00 XXX   0 X   -0.24 ARG 321 A\n",
      "GLU 315 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 154 A\n",
      "\n",
      "GLU 316 A   5.33    15 %    0.68  322   0.08    0    0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 306 A\n",
      "GLU 316 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 317 A\n",
      "\n",
      "GLU 320 A   4.89     8 %    0.35  305   0.01    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.05 LYS 113 A\n",
      "GLU 320 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 LYS 286 A\n",
      "GLU 320 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.19 ARG 322 A\n",
      "GLU 320 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 114 A\n",
      "GLU 320 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.30 ASP 317 A\n",
      "\n",
      "GLU 343 A   4.70     0 %    0.20  205   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "C-  352 A   2.61     0 %    0.24  209   0.00    0   -0.57 ARG 283 A    0.00 XXX   0 X   -0.26 ARG 283 A\n",
      "\n",
      "HIS   2 A   6.46     0 %   -0.18  208   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.08 N+    1 A\n",
      "HIS   2 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 GLU   4 A\n",
      "HIS   2 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.18 GLU   6 A\n",
      "\n",
      "HIS  73 A   6.41     0 %   -0.19  218   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.06 ARG  76 A\n",
      "HIS  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 113 A\n",
      "HIS  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 ASP  71 A\n",
      "HIS  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 109 A\n",
      "\n",
      "HIS  98 A   6.37     0 %   -0.15  122   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.00 LYS  89 A\n",
      "HIS  98 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 103 A\n",
      "\n",
      "HIS 183 A   6.16    21 %   -0.82  341   0.00    0    0.00 XXX   0 X    0.05 HIS 183 A   -0.39 N+    1 A\n",
      "HIS 183 A                                            0.00 XXX   0 X    0.71 ILE 184 A   -0.03 HIS   2 A\n",
      "HIS 183 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 GLU   4 A\n",
      "HIS 183 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 GLU 299 A\n",
      "\n",
      "HIS 248 A   6.22     0 %   -0.17  173   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 217 A\n",
      "HIS 248 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 249 A\n",
      "\n",
      "HIS 290 A   3.27   100 %   -3.17  619   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.07 LYS 263 A\n",
      "\n",
      "HIS 298 A   5.12    76 %   -1.70  493   0.00    0    0.00 XXX   0 X    0.30 GLY 302 A   -0.32 LYS 303 A\n",
      "HIS 298 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 156 A\n",
      "HIS 298 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 GLU 299 A\n",
      "HIS 298 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.21 GLU 311 A\n",
      "\n",
      "HIS 338 A   4.12    64 %   -2.13  461   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.17 LYS 303 A\n",
      "HIS 338 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.10 HIS 298 A\n",
      "HIS 338 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 306 A\n",
      "\n",
      "CYS  16 A  14.27   100 %    3.67  626   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.05 ARG  17 A\n",
      "CYS  16 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.44 ASP  13 A\n",
      "CYS  16 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.84 CYS  19 A\n",
      "CYS  16 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.36 CYS 164 A\n",
      "\n",
      "CYS  19 A  13.25   100 %    3.84  640   0.00    0    0.00 XXX   0 X   -0.06 CYS  19 A   -0.03 LYS  23 A\n",
      "CYS  19 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.49 ASP  46 A\n",
      "CYS  19 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU  87 A\n",
      "\n",
      "CYS 119 A   9.04     0 %    0.20  187   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.14 LYS 121 A\n",
      "CYS 119 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 321 A\n",
      "\n",
      "CYS 164 A  12.58   100 %    3.66  580   0.00    0    0.00 XXX   0 X   -0.13 TYR 145 A   -0.02 LYS  23 A\n",
      "CYS 164 A                                            0.00 XXX   0 X   -0.10 ASN 166 A   -0.04 ARG 271 A\n",
      "CYS 164 A                                            0.00 XXX   0 X   -0.05 TRP 167 A    0.11 ASP 132 A\n",
      "CYS 164 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.17 CYS 257 A\n",
      "\n",
      "CYS 186 A   9.38     0 %    0.39  262   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.09 N+    1 A\n",
      "CYS 186 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 ARG 206 A\n",
      "CYS 186 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 GLU 299 A\n",
      "\n",
      "CYS 207 A  11.12    63 %    2.22  457   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.07 ARG 206 A\n",
      "CYS 207 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 271 A\n",
      "\n",
      "CYS 212 A   9.75    29 %    1.27  363   0.00    0   -0.00 SER 273 A   -0.71 LEU 214 A   -0.04 LYS 217 A\n",
      "CYS 212 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.12 ARG 219 A\n",
      "CYS 212 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.35 ASP 210 A\n",
      "\n",
      "CYS 221 A  11.78    93 %    2.98  541   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.06 ARG 219 A\n",
      "CYS 221 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.40 LYS 263 A\n",
      "CYS 221 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.26 CYS 257 A\n",
      "\n",
      "CYS 257 A   6.54*   61 %    1.43  451   0.00    0   -3.60 CYS 262 A    0.00 XXX   0 X   -0.02 LYS 139 A\n",
      "CYS 257 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.33 LYS 263 A\n",
      "CYS 257 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 GLU 224 A\n",
      "\n",
      "CYS 258 A  10.03    39 %    1.48  391   0.00    0   -0.71 ARG 135 A   -0.05 CYS 258 A   -0.39 ARG 135 A\n",
      "CYS 258 A                                           -0.23 LYS 139 A    0.00 XXX   0 X   -0.45 LYS 139 A\n",
      "CYS 258 A                                            0.84 GLU 224 A    0.00 XXX   0 X   -0.20 LYS 263 A\n",
      "CYS 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP 132 A\n",
      "CYS 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.15 CYS 257 A\n",
      "CYS 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.48 GLU 224 A\n",
      "\n",
      "CYS 262 A  17.21*  100 %    3.46  569   0.00    0    3.60 CYS 257 A   -0.07 CYS 262 A   -1.78 LYS 263 A\n",
      "CYS 262 A                                            0.00 XXX   0 X   -0.06 LYS 263 A    0.53 CYS 164 A\n",
      "CYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.44 CYS 221 A\n",
      "CYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 224 A\n",
      "CYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 CYS 258 A\n",
      "CYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.03 ASP 132 A\n",
      "CYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.81 CYS 257 A\n",
      "\n",
      "CYS 296 A  12.13   100 %    3.70  656   0.00    0   -0.85 SER 182 A    0.00 XXX   0 X    0.23 CYS 207 A\n",
      "CYS 296 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 299 A\n",
      "\n",
      "TYR   1 A  10.58     0 %    0.41  279   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU   4 A\n",
      "TYR   1 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP 154 A\n",
      "TYR   1 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 GLU 156 A\n",
      "\n",
      "TYR 116 A  10.46     2 %    0.31  288   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.16 CYS 119 A\n",
      "TYR 116 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 ARG 321 A\n",
      "\n",
      "TYR 125 A  12.77    85 %    2.97  518   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 315 A\n",
      "TYR 125 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP 317 A\n",
      "TYR 125 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.34 ARG 321 A\n",
      "\n",
      "TYR 145 A  16.29    97 %    3.60  552   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.78 CYS  16 A\n",
      "TYR 145 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.50 CYS  19 A\n",
      "TYR 145 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP  21 A\n",
      "TYR 145 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 141 A\n",
      "TYR 145 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.37 CYS 164 A\n",
      "\n",
      "TYR 264 A  13.96   100 %    3.82  649   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 132 A\n",
      "TYR 264 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP 289 A\n",
      "\n",
      "LYS   3 A  10.39     0 %   -0.23  184   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.07 GLU   4 A\n",
      "LYS   3 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU   6 A\n",
      "\n",
      "LYS  23 A   9.00    44 %   -1.88  405   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.03 CYS  19 A\n",
      "LYS  23 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.31 ASP 141 A\n",
      "LYS  23 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 142 A\n",
      "LYS  23 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 CYS 164 A\n",
      "\n",
      "LYS  30 A  10.47     0 %   -0.03   61   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS  32 A  10.45     0 %   -0.05  104   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS  36 A  13.02    75 %   -2.82  492   0.00    0    0.85 GLU  69 A    0.00 XXX   0 X    0.26 ASP  38 A\n",
      "LYS  36 A                                            0.85 GLU  70 A    0.00 XXX   0 X    0.12 ASP  71 A\n",
      "LYS  36 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.17 ASP 129 A\n",
      "LYS  36 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 289 A\n",
      "LYS  36 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.59 ASP  46 A\n",
      "LYS  36 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.20 GLU  69 A\n",
      "LYS  36 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.29 GLU  70 A\n",
      "\n",
      "LYS  59 A  10.02    37 %   -1.79  385   0.00    0    0.85 GLU  56 A    0.00 XXX   0 X    0.10 ASP  10 A\n",
      "LYS  59 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.22 ARG  14 A\n",
      "LYS  59 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.58 GLU  56 A\n",
      "\n",
      "LYS  89 A  10.40     0 %   -0.38  237   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP  96 A\n",
      "LYS  89 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 ASP 101 A\n",
      "LYS  89 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 103 A\n",
      "\n",
      "LYS  91 A  10.94    30 %   -0.88  365   0.00    0    0.85 ASP  21 A    0.00 XXX   0 X    0.00 ASP  13 A\n",
      "LYS  91 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.11 ARG  17 A\n",
      "LYS  91 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 ARG  24 A\n",
      "LYS  91 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.14 GLU  87 A\n",
      "LYS  91 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP  96 A\n",
      "LYS  91 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.49 ASP  21 A\n",
      "\n",
      "LYS 113 A  10.86     0 %   -0.16  251   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.21 ASP  71 A\n",
      "LYS 113 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.29 ASP 114 A\n",
      "LYS 113 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 320 A\n",
      "LYS 113 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 ARG 322 A\n",
      "\n",
      "LYS 121 A  10.58     0 %   -0.06  106   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.14 CYS 119 A\n",
      "\n",
      "LYS 139 A  11.66     0 %   -0.43  269   0.00    0    0.23 CYS 258 A    0.00 XXX   0 X   -0.35 ARG 135 A\n",
      "LYS 139 A                                            0.85 GLU 224 A    0.00 XXX   0 X    0.02 CYS 257 A\n",
      "LYS 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.45 CYS 258 A\n",
      "LYS 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.39 GLU 224 A\n",
      "\n",
      "LYS 191 A  10.20     2 %   -0.45  288   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP  13 A\n",
      "LYS 191 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP 203 A\n",
      "\n",
      "LYS 192 A  11.20     0 %   -0.52  249   0.00    0    0.81 ASP 173 A    0.00 XXX   0 X    0.03 ASP 170 A\n",
      "LYS 192 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 ASP 173 A\n",
      "\n",
      "LYS 217 A  10.32     0 %   -0.21  224   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.04 CYS 212 A\n",
      "LYS 217 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 GLU 250 A\n",
      "\n",
      "LYS 247 A  11.60     0 %   -0.22  164   0.00    0    0.85 GLU 250 A    0.00 XXX   0 X    0.09 ASP 238 A\n",
      "LYS 247 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 GLU 250 A\n",
      "\n",
      "LYS 249 A  10.46     0 %   -0.10  152   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.06 ASP 238 A\n",
      "\n",
      "LYS 263 A  11.10    90 %   -2.93  534   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.04 ARG 135 A\n",
      "LYS 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.40 CYS 221 A\n",
      "LYS 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 GLU 224 A\n",
      "LYS 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.33 CYS 257 A\n",
      "LYS 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 CYS 258 A\n",
      "LYS 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.78 CYS 262 A\n",
      "LYS 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP 289 A\n",
      "LYS 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.70 ASP 132 A\n",
      "\n",
      "LYS 280 A  10.26    15 %   -0.23  322   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.01 ARG 135 A\n",
      "\n",
      "LYS 286 A  10.05    45 %   -0.88  407   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP  38 A\n",
      "LYS 286 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.18 GLU  69 A\n",
      "LYS 286 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 ASP 289 A\n",
      "LYS 286 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 317 A\n",
      "LYS 286 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU 320 A\n",
      "LYS 286 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.11 ARG 322 A\n",
      "LYS 286 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 LYS  36 A\n",
      "\n",
      "LYS 303 A  11.21    13 %   -0.56  318   0.00    0    0.85 GLU 311 A    0.00 XXX   0 X    0.00 ASP 310 A\n",
      "LYS 303 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.42 GLU 311 A\n",
      "\n",
      "ARG  14 A  12.98    32 %   -1.07  370   0.00    0    0.28 ASP  10 A    0.00 XXX   0 X    0.08 ASP  13 A\n",
      "ARG  14 A                                            0.18 GLU  56 A    0.00 XXX   0 X   -0.05 ARG  17 A\n",
      "ARG  14 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.55 ASP  10 A\n",
      "ARG  14 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.52 GLU  56 A\n",
      "\n",
      "ARG  17 A  13.13    26 %   -0.81  355   0.00    0    0.72 ASP  13 A    0.00 XXX   0 X    0.05 ASP  10 A\n",
      "ARG  17 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 CYS  16 A\n",
      "ARG  17 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP  21 A\n",
      "ARG  17 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.59 ASP  13 A\n",
      "\n",
      "ARG  24 A  12.69     0 %   -0.25  263   0.00    0    0.11 ASP  21 A    0.00 XXX   0 X    0.32 ASP  21 A\n",
      "ARG  24 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 GLU  87 A\n",
      "ARG  24 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 141 A\n",
      "\n",
      "ARG  76 A  12.96     1 %   -0.44  283   0.00    0    0.48 ASP 109 A    0.00 XXX   0 X    0.05 ASP  71 A\n",
      "ARG  76 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.37 ASP 109 A\n",
      "\n",
      "ARG 110 A  10.93    57 %   -1.59  442   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 106 A\n",
      "\n",
      "ARG 135 A  14.08    12 %   -0.46  314   0.00    0    0.71 CYS 258 A    0.00 XXX   0 X    0.39 CYS 258 A\n",
      "ARG 135 A                                            0.52 GLU 224 A    0.00 XXX   0 X    0.42 GLU 224 A\n",
      "\n",
      "ARG 206 A  12.17     3 %   -0.59  289   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.06 CYS 186 A\n",
      "ARG 206 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 CYS 207 A\n",
      "ARG 206 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 GLU 299 A\n",
      "\n",
      "ARG 219 A  12.97    38 %   -0.99  388   0.00    0    0.85 ASP 210 A    0.00 XXX   0 X    0.12 CYS 212 A\n",
      "ARG 219 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 CYS 221 A\n",
      "ARG 219 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.43 ASP 210 A\n",
      "\n",
      "ARG 271 A  12.14    41 %   -1.19  395   0.00    0    0.42 ASP 170 A    0.00 XXX   0 X    0.04 CYS 164 A\n",
      "ARG 271 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 CYS 207 A\n",
      "ARG 271 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 219 A\n",
      "ARG 271 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.37 ASP 170 A\n",
      "\n",
      "ARG 278 A  11.64    29 %   -0.86  362   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "ARG 283 A  13.24     0 %   -0.20  243   0.00    0    0.57 C-  352 A    0.00 XXX   0 X    0.01 ASP 306 A\n",
      "ARG 283 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 GLU 308 A\n",
      "ARG 283 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.26 C-  352 A\n",
      "\n",
      "ARG 321 A  11.96    26 %   -1.28  353   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.01 TYR 116 A\n",
      "ARG 321 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 CYS 119 A\n",
      "ARG 321 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.34 TYR 125 A\n",
      "ARG 321 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.24 GLU 315 A\n",
      "ARG 321 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 ASP 317 A\n",
      "\n",
      "ARG 322 A  10.73    72 %   -2.17  484   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU  69 A\n",
      "ARG 322 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP  71 A\n",
      "ARG 322 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 ASP 114 A\n",
      "ARG 322 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP 317 A\n",
      "ARG 322 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 GLU 320 A\n",
      "\n",
      "N+    1 A   7.87     2 %   -0.47  287   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.26 GLU   4 A\n",
      "N+    1 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 CYS 186 A\n",
      "\n",
      "Coupled residues (marked *) were detected.Please rerun PropKa with the --display-coupled-residues \n",
      "or -d option for detailed information.\n",
      "\n",
      "INFO:pKa summary:\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "SUMMARY OF THIS PREDICTION\n",
      "       Group      pKa  model-pKa   ligand atom-type\n",
      "   ASP  10 A     4.95       3.80                      \n",
      "   ASP  13 A     4.41       3.80                      \n",
      "   ASP  21 A     2.92       3.80                      \n",
      "   ASP  38 A     2.45       3.80                      \n",
      "   ASP  46 A     5.88       3.80                      \n",
      "   ASP  71 A     3.56       3.80                      \n",
      "   ASP  96 A     3.95       3.80                      \n",
      "   ASP 101 A     3.94       3.80                      \n",
      "   ASP 103 A     2.46       3.80                      \n",
      "   ASP 109 A     2.64       3.80                      \n",
      "   ASP 114 A     3.91       3.80                      \n",
      "   ASP 129 A    11.71       3.80                      \n",
      "   ASP 132 A     6.29       3.80                      \n",
      "   ASP 141 A     3.75       3.80                      \n",
      "   ASP 154 A     4.07       3.80                      \n",
      "   ASP 170 A     3.21       3.80                      \n",
      "   ASP 173 A     1.41       3.80                      \n",
      "   ASP 203 A     3.82       3.80                      \n",
      "   ASP 210 A     2.58       3.80                      \n",
      "   ASP 238 A     4.04       3.80                      \n",
      "   ASP 245 A     4.00       3.80                      \n",
      "   ASP 289 A     7.94       3.80                      \n",
      "   ASP 306 A     3.95       3.80                      \n",
      "   ASP 310 A     4.33       3.80                      \n",
      "   ASP 317 A     3.26       3.80                      \n",
      "   GLU   4 A     4.10       4.50                      \n",
      "   GLU   6 A     4.46       4.50                      \n",
      "   GLU  31 A     4.62       4.50                      \n",
      "   GLU  35 A     4.73       4.50                      \n",
      "   GLU  56 A     2.83       4.50                      \n",
      "   GLU  69 A     4.06       4.50                      \n",
      "   GLU  70 A     8.36       4.50                      \n",
      "   GLU  87 A     5.73       4.50                      \n",
      "   GLU 106 A     4.79       4.50                      \n",
      "   GLU 142 A     4.62       4.50                      \n",
      "   GLU 156 A     5.01       4.50                      \n",
      "   GLU 224 A     2.07       4.50                      \n",
      "   GLU 229 A     3.42       4.50                      \n",
      "   GLU 244 A     4.57       4.50                      \n",
      "   GLU 250 A     2.96       4.50                      \n",
      "   GLU 299 A     4.11       4.50                      \n",
      "   GLU 308 A     5.24       4.50                      \n",
      "   GLU 311 A     3.29       4.50                      \n",
      "   GLU 315 A     4.29       4.50                      \n",
      "   GLU 316 A     5.33       4.50                      \n",
      "   GLU 320 A     4.89       4.50                      \n",
      "   GLU 343 A     4.70       4.50                      \n",
      "   C-  352 A     2.61       3.20                      \n",
      "   HIS   2 A     6.46       6.50                      \n",
      "   HIS  73 A     6.41       6.50                      \n",
      "   HIS  98 A     6.37       6.50                      \n",
      "   HIS 183 A     6.16       6.50                      \n",
      "   HIS 248 A     6.22       6.50                      \n",
      "   HIS 290 A     3.27       6.50                      \n",
      "   HIS 298 A     5.12       6.50                      \n",
      "   HIS 338 A     4.12       6.50                      \n",
      "   CYS  16 A    14.27       9.00                      \n",
      "   CYS  19 A    13.25       9.00                      \n",
      "   CYS 119 A     9.04       9.00                      \n",
      "   CYS 164 A    12.58       9.00                      \n",
      "   CYS 186 A     9.38       9.00                      \n",
      "   CYS 207 A    11.12       9.00                      \n",
      "   CYS 212 A     9.75       9.00                      \n",
      "   CYS 221 A    11.78       9.00                      \n",
      "   CYS 257 A     6.54       9.00                      \n",
      "   CYS 258 A    10.03       9.00                      \n",
      "   CYS 262 A    17.21       9.00                      \n",
      "   CYS 296 A    12.13       9.00                      \n",
      "   TYR   1 A    10.58      10.00                      \n",
      "   TYR 116 A    10.46      10.00                      \n",
      "   TYR 125 A    12.77      10.00                      \n",
      "   TYR 145 A    16.29      10.00                      \n",
      "   TYR 264 A    13.96      10.00                      \n",
      "   LYS   3 A    10.39      10.50                      \n",
      "   LYS  23 A     9.00      10.50                      \n",
      "   LYS  30 A    10.47      10.50                      \n",
      "   LYS  32 A    10.45      10.50                      \n",
      "   LYS  36 A    13.02      10.50                      \n",
      "   LYS  59 A    10.02      10.50                      \n",
      "   LYS  89 A    10.40      10.50                      \n",
      "   LYS  91 A    10.94      10.50                      \n",
      "   LYS 113 A    10.86      10.50                      \n",
      "   LYS 121 A    10.58      10.50                      \n",
      "   LYS 139 A    11.66      10.50                      \n",
      "   LYS 191 A    10.20      10.50                      \n",
      "   LYS 192 A    11.20      10.50                      \n",
      "   LYS 217 A    10.32      10.50                      \n",
      "   LYS 247 A    11.60      10.50                      \n",
      "   LYS 249 A    10.46      10.50                      \n",
      "   LYS 263 A    11.10      10.50                      \n",
      "   LYS 280 A    10.26      10.50                      \n",
      "   LYS 286 A    10.05      10.50                      \n",
      "   LYS 303 A    11.21      10.50                      \n",
      "   ARG  14 A    12.98      12.50                      \n",
      "   ARG  17 A    13.13      12.50                      \n",
      "   ARG  24 A    12.69      12.50                      \n",
      "   ARG  76 A    12.96      12.50                      \n",
      "   ARG 110 A    10.93      12.50                      \n",
      "   ARG 135 A    14.08      12.50                      \n",
      "   ARG 206 A    12.17      12.50                      \n",
      "   ARG 219 A    12.97      12.50                      \n",
      "   ARG 271 A    12.14      12.50                      \n",
      "   ARG 278 A    11.64      12.50                      \n",
      "   ARG 283 A    13.24      12.50                      \n",
      "   ARG 321 A    11.96      12.50                      \n",
      "   ARG 322 A    10.73      12.50                      \n",
      "   N+    1 A     7.87       8.00                      \n",
      "\n",
      "INFO:PROPKA information:\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "Free energy of   folding (kcal/mol) as a function of pH (using neutral reference)\n",
      "  0.00     34.30\n",
      "  1.00     33.99\n",
      "  2.00     32.12\n",
      "  3.00     25.92\n",
      "  4.00     18.89\n",
      "  5.00     20.28\n",
      "  6.00     24.08\n",
      "  7.00     26.09\n",
      "  8.00     28.36\n",
      "  9.00     32.60\n",
      " 10.00     42.82\n",
      " 11.00     57.76\n",
      " 12.00     71.32\n",
      " 13.00     81.81\n",
      " 14.00     89.66\n",
      "\n",
      "The pH of optimum stability is  4.3 for which the free energy is   18.4 kcal/mol at 298K\n",
      "Could not determine pH values where the free energy is within 80 % of minimum\n",
      "Could not determine the pH-range where the free energy is negative\n",
      "\n",
      "\n",
      "Protein charge of folded and unfolded state as a function of pH\n",
      "    pH  unfolded  folded\n",
      "  0.00     41.99   41.93\n",
      "  0.10     41.99   41.91\n",
      "  0.20     41.99   41.89\n",
      "  0.30     41.99   41.87\n",
      "  0.40     41.99   41.83\n",
      "  0.50     41.98   41.79\n",
      "  0.60     41.98   41.74\n",
      "  0.70     41.97   41.68\n",
      "  0.80     41.97   41.61\n",
      "  0.90     41.96   41.52\n",
      "  1.00     41.95   41.42\n",
      "  1.10     41.93   41.29\n",
      "  1.20     41.92   41.15\n",
      "  1.30     41.89   40.98\n",
      "  1.40     41.87   40.78\n",
      "  1.50     41.83   40.56\n",
      "  1.60     41.79   40.30\n",
      "  1.70     41.74   40.00\n",
      "  1.80     41.67   39.67\n",
      "  1.90     41.59   39.28\n",
      "  2.00     41.48   38.85\n",
      "  2.10     41.35   38.36\n",
      "  2.20     41.19   37.82\n",
      "  2.30     40.98   37.21\n",
      "  2.40     40.73   36.55\n",
      "  2.50     40.42   35.82\n",
      "  2.60     40.04   35.03\n",
      "  2.70     39.58   34.17\n",
      "  2.80     39.01   33.26\n",
      "  2.90     38.33   32.30\n",
      "  3.00     37.52   31.28\n",
      "  3.10     36.55   30.20\n",
      "  3.20     35.43   29.07\n",
      "  3.30     34.13   27.88\n",
      "  3.40     32.64   26.64\n",
      "  3.50     30.98   25.34\n",
      "  3.60     29.14   23.99\n",
      "  3.70     27.15   22.59\n",
      "  3.80     25.03   21.14\n",
      "  3.90     22.80   19.65\n",
      "  4.00     20.50   18.12\n",
      "  4.10     18.16   16.58\n",
      "  4.20     15.82   15.03\n",
      "  4.30     13.52   13.49\n",
      "  4.40     11.28   11.96\n",
      "  4.50      9.13   10.46\n",
      "  4.60      7.10    9.01\n",
      "  4.70      5.21    7.61\n",
      "  4.80      3.48    6.26\n",
      "  4.90      1.93    4.98\n",
      "  5.00      0.54    3.76\n",
      "  5.10     -0.69    2.62\n",
      "  5.20     -1.76    1.54\n",
      "  5.30     -2.70    0.52\n",
      "  5.40     -3.52   -0.43\n",
      "  5.50     -4.24   -1.33\n",
      "  5.60     -4.89   -2.18\n",
      "  5.70     -5.49   -2.99\n",
      "  5.80     -6.05   -3.76\n",
      "  5.90     -6.58   -4.51\n",
      "  6.00     -7.11   -5.23\n",
      "  6.10     -7.64   -5.93\n",
      "  6.20     -8.18   -6.60\n",
      "  6.30     -8.72   -7.24\n",
      "  6.40     -9.26   -7.86\n",
      "  6.50     -9.80   -8.44\n",
      "  6.60    -10.34   -8.98\n",
      "  6.70    -10.85   -9.48\n",
      "  6.80    -11.34   -9.94\n",
      "  6.90    -11.79  -10.36\n",
      "  7.00    -12.21  -10.74\n",
      "  7.10    -12.60  -11.08\n",
      "  7.20    -12.96  -11.39\n",
      "  7.30    -13.29  -11.69\n",
      "  7.40    -13.59  -11.96\n",
      "  7.50    -13.89  -12.23\n",
      "  7.60    -14.18  -12.49\n",
      "  7.70    -14.47  -12.75\n",
      "  7.80    -14.78  -13.01\n",
      "  7.90    -15.10  -13.27\n",
      "  8.00    -15.45  -13.54\n",
      "  8.10    -15.84  -13.82\n",
      "  8.20    -16.27  -14.10\n",
      "  8.30    -16.76  -14.39\n",
      "  8.40    -17.30  -14.69\n",
      "  8.50    -17.91  -15.00\n",
      "  8.60    -18.59  -15.33\n",
      "  8.70    -19.34  -15.67\n",
      "  8.80    -20.15  -16.04\n",
      "  8.90    -21.03  -16.42\n",
      "  9.00    -21.95  -16.84\n",
      "  9.10    -22.92  -17.29\n",
      "  9.20    -23.93  -17.77\n",
      "  9.30    -24.96  -18.30\n",
      "  9.40    -26.02  -18.88\n",
      "  9.50    -27.11  -19.51\n",
      "  9.60    -28.24  -20.20\n",
      "  9.70    -29.41  -20.95\n",
      "  9.80    -30.63  -21.77\n",
      "  9.90    -31.90  -22.67\n",
      " 10.00    -33.24  -23.64\n",
      " 10.10    -34.64  -24.67\n",
      " 10.20    -36.09  -25.77\n",
      " 10.30    -37.57  -26.92\n",
      " 10.40    -39.07  -28.12\n",
      " 10.50    -40.56  -29.34\n",
      " 10.60    -42.01  -30.59\n",
      " 10.70    -43.40  -31.84\n",
      " 10.80    -44.70  -33.09\n",
      " 10.90    -45.91  -34.32\n",
      " 11.00    -47.02  -35.53\n",
      " 11.10    -48.02  -36.72\n",
      " 11.20    -48.92  -37.88\n",
      " 11.30    -49.74  -39.01\n",
      " 11.40    -50.48  -40.11\n",
      " 11.50    -51.17  -41.18\n",
      " 11.60    -51.83  -42.22\n",
      " 11.70    -52.47  -43.23\n",
      " 11.80    -53.11  -44.22\n",
      " 11.90    -53.77  -45.19\n",
      " 12.00    -54.45  -46.13\n",
      " 12.10    -55.16  -47.06\n",
      " 12.20    -55.91  -47.96\n",
      " 12.30    -56.69  -48.85\n",
      " 12.40    -57.48  -49.72\n",
      " 12.50    -58.28  -50.57\n",
      " 12.60    -59.07  -51.42\n",
      " 12.70    -59.83  -52.25\n",
      " 12.80    -60.55  -53.07\n",
      " 12.90    -61.21  -53.86\n",
      " 13.00    -61.81  -54.64\n",
      " 13.10    -62.34  -55.39\n",
      " 13.20    -62.79  -56.10\n",
      " 13.30    -63.19  -56.77\n",
      " 13.40    -63.52  -57.41\n",
      " 13.50    -63.80  -58.00\n",
      " 13.60    -64.03  -58.55\n",
      " 13.70    -64.21  -59.05\n",
      " 13.80    -64.37  -59.52\n",
      " 13.90    -64.49  -59.95\n",
      " 14.00    -64.59  -60.34\n",
      "The pI is  5.35 (folded) and  5.04 (unfolded)\n",
      "\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "INFO:Applying pKa values at a pH of 7.60:\n",
      "INFO:Adding hydrogens to biomolecule.\n",
      "INFO:Debumping biomolecule (again).\n",
      "INFO:Optimizing hydrogen bonds\n",
      "INFO:Applying force field to biomolecule states.\n",
      "INFO:Regenerating headers.\n",
      "INFO:Regenerating PDB lines.\n",
      "WARNING:Ignoring 467 header lines in output.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "pH: 7.6\tname: 74987\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:PDB2PQR v3.6.2: biomolecular structure conversion software.\n",
      "INFO:Please cite:  Jurrus E, et al.  Improvements to the APBS biomolecular solvation software suite.  Protein Sci 27 112-128 (2018).\n",
      "INFO:Please cite:  Dolinsky TJ, et al.  PDB2PQR: expanding and upgrading automated preparation of biomolecular structures for molecular simulations. Nucleic Acids Res 35 W522-W525 (2007).\n",
      "INFO:Checking and transforming input arguments.\n",
      "INFO:Loading topology files.\n",
      "INFO:Loading molecule: /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/1ka1_hal2.pdb\n",
      "INFO:Setting up molecule.\n",
      "INFO:Created biomolecule object with 354 residues and 2718 atoms.\n",
      "WARNING:Multiple occupancies found: OG in SER A 84.\n",
      "WARNING:Multiple occupancies found in SER A 84. At least one of the instances is being ignored.\n",
      "INFO:Setting termini states for biomolecule chains.\n",
      "INFO:Loading forcefield.\n",
      "INFO:Loading hydrogen topology definitions.\n",
      "WARNING:Missing atom CB in residue LYS A 32\n",
      "WARNING:Missing atom CG in residue LYS A 32\n",
      "WARNING:Missing atom CD in residue LYS A 32\n",
      "WARNING:Missing atom CE in residue LYS A 32\n",
      "WARNING:Missing atom NZ in residue LYS A 32\n",
      "WARNING:Missing atom CB in residue ASP A 33\n",
      "WARNING:Missing atom CG in residue ASP A 33\n",
      "WARNING:Missing atom OD1 in residue ASP A 33\n",
      "WARNING:Missing atom OD2 in residue ASP A 33\n",
      "WARNING:Missing atom CG in residue LYS A 189\n",
      "WARNING:Missing atom CD in residue LYS A 189\n",
      "WARNING:Missing atom CE in residue LYS A 189\n",
      "WARNING:Missing atom NZ in residue LYS A 189\n",
      "WARNING:Missing atom CD in residue LYS A 229\n",
      "WARNING:Missing atom CE in residue LYS A 229\n",
      "WARNING:Missing atom NZ in residue LYS A 229\n",
      "WARNING:Missing atom OXT in residue ARG A 355\n",
      "WARNING:Missing atom CB in residue LYS A 32\n",
      "WARNING:Missing atom CG in residue LYS A 32\n",
      "WARNING:Missing atom CD in residue LYS A 32\n",
      "WARNING:Missing atom CE in residue LYS A 32\n",
      "WARNING:Missing atom NZ in residue LYS A 32\n",
      "WARNING:Missing atom CB in residue ASP A 33\n",
      "WARNING:Missing atom CG in residue ASP A 33\n",
      "WARNING:Missing atom OD1 in residue ASP A 33\n",
      "WARNING:Missing atom OD2 in residue ASP A 33\n",
      "WARNING:Missing atom CG in residue LYS A 189\n",
      "WARNING:Missing atom CD in residue LYS A 189\n",
      "WARNING:Missing atom CE in residue LYS A 189\n",
      "WARNING:Missing atom NZ in residue LYS A 189\n",
      "WARNING:Missing atom CD in residue LYS A 229\n",
      "WARNING:Missing atom CE in residue LYS A 229\n",
      "WARNING:Missing atom NZ in residue LYS A 229\n",
      "WARNING:Missing atom OXT in residue ARG A 355\n",
      "INFO:Attempting to repair 17 missing atoms in biomolecule.\n",
      "WARNING:Missing atom CB in residue LYS A 32\n",
      "WARNING:Missing atom CG in residue LYS A 32\n",
      "WARNING:Missing atom CD in residue LYS A 32\n",
      "WARNING:Missing atom CE in residue LYS A 32\n",
      "WARNING:Missing atom NZ in residue LYS A 32\n",
      "WARNING:Missing atom CB in residue ASP A 33\n",
      "WARNING:Missing atom CG in residue ASP A 33\n",
      "WARNING:Missing atom OD1 in residue ASP A 33\n",
      "WARNING:Missing atom OD2 in residue ASP A 33\n",
      "WARNING:Missing atom CG in residue LYS A 189\n",
      "WARNING:Missing atom CD in residue LYS A 189\n",
      "WARNING:Missing atom CE in residue LYS A 189\n",
      "WARNING:Missing atom NZ in residue LYS A 189\n",
      "WARNING:Missing atom CD in residue LYS A 229\n",
      "WARNING:Missing atom CE in residue LYS A 229\n",
      "WARNING:Missing atom NZ in residue LYS A 229\n",
      "WARNING:Missing atom OXT in residue ARG A 355\n",
      "INFO:Added atom CB to residue LYS A 32 at coordinates -45.594, -33.973, -23.265\n",
      "INFO:Added atom CG to residue LYS A 32 at coordinates -45.153, -35.334, -22.744\n",
      "INFO:Added atom CD to residue LYS A 32 at coordinates -43.702, -35.621, -23.091\n",
      "INFO:Added atom CE to residue LYS A 32 at coordinates -43.239, -36.968, -22.584\n",
      "INFO:Added atom NZ to residue LYS A 32 at coordinates -41.819, -37.204, -22.945\n",
      "INFO:Added atom CB to residue ASP A 33 at coordinates -49.706, -30.631, -25.302\n",
      "INFO:Added atom CG to residue ASP A 33 at coordinates -48.707, -31.048, -26.377\n",
      "INFO:Added atom OD1 to residue ASP A 33 at coordinates -47.538, -31.342, -26.061\n",
      "INFO:Added atom OD2 to residue ASP A 33 at coordinates -49.132, -31.087, -27.541\n",
      "INFO:Added atom CG to residue LYS A 189 at coordinates -35.105, -63.964, -9.656\n",
      "INFO:Added atom CD to residue LYS A 189 at coordinates -35.396, -65.193, -8.824\n",
      "INFO:Added atom CE to residue LYS A 189 at coordinates -34.533, -66.375, -9.200\n",
      "INFO:Added atom NZ to residue LYS A 189 at coordinates -34.854, -67.554, -8.358\n",
      "INFO:Added atom CD to residue LYS A 229 at coordinates -3.896, -48.760, -18.063\n",
      "INFO:Added atom CE to residue LYS A 229 at coordinates -3.416, -50.110, -17.576\n",
      "INFO:Added atom NZ to residue LYS A 229 at coordinates -2.176, -50.508, -18.287\n",
      "INFO:Added atom OXT to residue ARG A 355 at coordinates -9.183, -18.959, -17.852\n",
      "INFO:Updating disulfide bridges.\n",
      "INFO:Debumping biomolecule.\n",
      "INFO:Assigning titration states with PROPKA.\n",
      "INFO:\n",
      "propka3.5.1                                                                                  2024-07-26\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "--                                                                           --\n",
      "--  PROPKA: A PROTEIN PKA PREDICTOR                                          --\n",
      "--                                                                           --\n",
      "--  VERSION 1.0,  04/25/2004,  IOWA CITY                                     --\n",
      "--  BY HUI LI                                                                --\n",
      "--                                                                           --\n",
      "--  VERSION 2.0,  11/05/2007, IOWA CITY/COPENHAGEN                           --\n",
      "--  BY DELPHINE C. BAS AND DAVID M. ROGERS                                   --\n",
      "--                                                                           --\n",
      "--  VERSION 3.0,  01/06/2011, COPENHAGEN                                     --\n",
      "--  BY MATS H.M. OLSSON AND CHRESTEN R. SONDERGARD                           --\n",
      "--                                                                           --\n",
      "--  VERSION 3.1,  07/01/2011, COPENHAGEN                                     --\n",
      "--  BY CHRESTEN R. SONDERGARD AND MATS H.M. OLSSON                           --\n",
      "--                                                                           --\n",
      "--  VERSION 3.2,  06/17/2020, PLANET EARTH                                   --\n",
      "--  SEE ABOVE FOR AUTHORS                                                    --\n",
      "--                                                                           --\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "References:\n",
      "\n",
      "Very Fast Empirical Prediction and Rationalization of Protein pKa Values.\n",
      "Hui Li, Andrew D. Robertson and Jan H. Jensen. PROTEINS: Structure, Function,\n",
      "and Bioinformatics. 61:704-721 (2005)\n",
      "\n",
      "Very Fast Prediction and Rationalization of pKa Values for Protein-Ligand\n",
      "Complexes.  Delphine C. Bas, David M. Rogers and Jan H. Jensen.  PROTEINS:\n",
      "Structure, Function, and Bioinformatics 73:765-783 (2008)\n",
      "\n",
      "PROPKA3: Consistent Treatment of Internal and Surface Residues in Empirical\n",
      "pKa predictions.  Mats H.M. Olsson, Chresten R. Sondergard, Michal Rostkowski,\n",
      "and Jan H. Jensen.  Journal of Chemical Theory and Computation, 7(2):525-537\n",
      "(2011)\n",
      "\n",
      "Improved Treatment of Ligands and Coupling Effects in Empirical Calculation\n",
      "and Rationalization of pKa Values.  Chresten R. Sondergaard, Mats H.M. Olsson,\n",
      "Michal Rostkowski, and Jan H. Jensen.  Journal of Chemical Theory and\n",
      "Computation, (2011)\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "\n",
      "WARNING:Missing atoms or failed protonation for HIS  31 A (HIS) -- please check the structure\n",
      "WARNING:Group (HIS) for   231-  CG    31-HIS (A) [ -51.746  -37.773  -23.250] C\n",
      "WARNING:Expected 4 interaction atoms for acids, found:\n",
      "WARNING:                 0- HE2    31-HIS (A) [ -54.068  -39.642  -24.334] H\n",
      "WARNING:               233- ND1    31-HIS (A) [ -51.129  -38.973  -23.536] N\n",
      "WARNING:               235- NE2    31-HIS (A) [ -53.201  -39.225  -24.025] N\n",
      "WARNING:Expected 2 interaction atoms for bases, found:\n",
      "WARNING:               233- ND1    31-HIS (A) [ -51.129  -38.973  -23.536] N\n",
      "WARNING:               235- NE2    31-HIS (A) [ -53.201  -39.225  -24.025] N\n",
      "INFO:Coupling map:\n",
      "\n",
      "INFO:Calculating pKas for Conformation container 1A with 3249 atoms and 890 groups\n",
      "INFO:pKa determinants:\n",
      "\n",
      "---------  -----   ------   ---------------------    --------------    --------------    --------------\n",
      "                            DESOLVATION  EFFECTS       SIDECHAIN          BACKBONE        COULOMBIC\n",
      " RESIDUE    pKa    BURIED     REGULAR      RE        HYDROGEN BOND     HYDROGEN BOND      INTERACTION\n",
      "---------  -----   ------   ---------   ---------    --------------    --------------    --------------\n",
      "\n",
      "ASP  33 A   3.91     0 %    0.20  152   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS  32 A\n",
      "ASP  33 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 HIS  31 A\n",
      "\n",
      "ASP  41 A   2.21    11 %    0.60  312   0.00    0   -0.79 SER  43 A   -0.23 ASP  41 A   -0.28 LYS  39 A\n",
      "ASP  41 A                                            0.00 XXX   0 X   -0.73 SER  43 A   -0.04 LYS 148 A\n",
      "ASP  41 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.12 LYS 291 A\n",
      "ASP  41 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 HIS 241 A\n",
      "\n",
      "ASP  49 A   6.17*  100 %    3.70  606   1.05    0   -0.84 THR  46 A   -0.70 THR  46 A   -0.30 LYS  39 A\n",
      "ASP  49 A                                           -0.85 THR 147 A   -0.24 ASP  49 A    0.55 GLU  72 A\n",
      "\n",
      "ASP  66 A   3.97     0 %    0.19  204   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.11 LYS  61 A\n",
      "ASP  66 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP  67 A\n",
      "\n",
      "ASP  67 A   3.50    45 %    1.80  406   0.24    0   -0.81 TRP 139 A    0.00 XXX   0 X   -0.03 ARG   5 A\n",
      "ASP  67 A                                           -0.70 ARG 137 A    0.00 XXX   0 X   -0.15 LYS  61 A\n",
      "ASP  67 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 LYS  68 A\n",
      "ASP  67 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.57 ARG 137 A\n",
      "\n",
      "ASP  80 A   3.08     0 %    0.24  175   0.00    0   -0.70 ARG 122 A    0.00 XXX   0 X   -0.26 ARG 122 A\n",
      "\n",
      "ASP  94 A   2.91    69 %    1.87  474   0.67    0   -0.85 LYS  16 A    0.00 XXX   0 X   -0.00 ARG  23 A\n",
      "ASP  94 A                                           -0.85 LYS  91 A    0.00 XXX   0 X   -0.00 LYS 116 A\n",
      "ASP  94 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.05 LYS  16 A\n",
      "ASP  94 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.67 LYS  91 A\n",
      "\n",
      "ASP 104 A   3.88     0 %    0.20  138   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.04 LYS 102 A\n",
      "ASP 104 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 LYS 103 A\n",
      "\n",
      "ASP 105 A   4.10     0 %    0.13  159   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG   5 A\n",
      "ASP 105 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS 103 A\n",
      "ASP 105 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.21 ASP 104 A\n",
      "\n",
      "ASP 111 A   3.81     0 %    0.12  140   0.00    0    0.00 XXX   0 X   -0.11 ASP 111 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 120 A   2.75    13 %    0.72  318   0.05    0   -0.32 SER 117 A   -0.79 LYS 116 A   -0.02 LYS 116 A\n",
      "ASP 120 A                                            0.00 XXX   0 X   -0.69 SER 117 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 126 A   3.39    34 %    1.22  376   0.28    0   -0.16 SER  74 A   -0.85 SER  75 A   -0.06 ARG 122 A\n",
      "ASP 126 A                                           -0.85 SER  75 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "ASP 142 A  16.43*  100 %    3.57  642   1.03    0    0.48 GLU  72 A   -0.03 ILE 144 A   -0.28 LYS  39 A\n",
      "ASP 142 A                                            0.67 ASP 145 A    0.00 XXX   0 X    0.50 ASP  49 A\n",
      "ASP 142 A                                            0.37 ASP 294 A    0.00 XXX   0 X    0.17 CYS 140 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.88 CYS 159 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.87 GLU  72 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.59 GLU  73 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.57 ASP 145 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.23 ASP 294 A\n",
      "\n",
      "ASP 145 A   3.62*  100 %    2.82  672   0.55    0   -0.67 ASP 142 A   -0.04 GLY 146 A   -0.19 HIS 241 A\n",
      "ASP 145 A                                           -0.73 ASP 294 A   -0.57 GLN 265 A    0.17 ASP 263 A\n",
      "ASP 145 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 GLU 290 A\n",
      "ASP 145 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.44 HIS 295 A\n",
      "ASP 145 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.65 LYS 267 A\n",
      "ASP 145 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.55 ARG 281 A\n",
      "\n",
      "ASP 165 A   3.15     0 %    0.28  211   0.00    0    0.00 XXX   0 X   -0.81 GLY 136 A   -0.03 LYS  68 A\n",
      "ASP 165 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.00 ARG 134 A\n",
      "ASP 165 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 LYS 135 A\n",
      "\n",
      "ASP 187 A   2.15    10 %    1.13  308   0.05    0   -0.17 SER 181 A   -0.76 LEU 180 A   -0.41 HIS 191 A\n",
      "ASP 187 A                                           -0.85 TYR 196 A   -0.63 SER 181 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 213 A   3.05    18 %    0.79  331   0.14    0    0.00 XXX   0 X   -0.76 GLU 215 A   -0.09 ARG  15 A\n",
      "ASP 213 A                                            0.00 XXX   0 X   -0.84 SER 216 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 227 A   3.07     0 %    0.35  211   0.00    0    0.00 XXX   0 X   -0.84 LYS 229 A   -0.24 LYS 229 A\n",
      "\n",
      "ASP 230 A   3.96     0 %    0.30  181   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.03 ARG 223 A\n",
      "ASP 230 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.20 LYS 229 A\n",
      "ASP 230 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.09 LYS 258 A\n",
      "ASP 230 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP 227 A\n",
      "ASP 230 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 277 A\n",
      "\n",
      "ASP 245 A   3.84     0 %    0.18  234   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.09 LYS 239 A\n",
      "ASP 245 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 HIS 244 A\n",
      "ASP 245 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 GLU 246 A\n",
      "\n",
      "ASP 263 A   3.38    62 %    2.13  455   0.38    0   -1.55 ARG 152 A   -0.09 ASP 263 A   -0.03 LYS 148 A\n",
      "ASP 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.50 LYS 267 A\n",
      "ASP 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.20 HIS 241 A\n",
      "ASP 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.57 ARG 152 A\n",
      "\n",
      "ASP 277 A   3.47    56 %    2.43  438   0.35    0   -1.64 ARG 223 A   -0.78 ILE 232 A   -0.04 LYS 258 A\n",
      "ASP 277 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 HIS 224 A\n",
      "ASP 277 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.61 ARG 223 A\n",
      "\n",
      "ASP 294 A   9.23*  100 %    2.63  612   0.38    0   -0.37 ASP 142 A    0.00 XXX   0 X   -0.04 LYS  39 A\n",
      "ASP 294 A                                            0.73 ASP 145 A    0.00 XXX   0 X   -0.08 LYS 267 A\n",
      "ASP 294 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.35 LYS 291 A\n",
      "ASP 294 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.93 GLU  72 A\n",
      "ASP 294 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.64 ASP 145 A\n",
      "ASP 294 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.87 ARG 281 A\n",
      "ASP 294 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.84 GLU 290 A\n",
      "\n",
      "ASP 311 A   3.81    51 %    2.24  424   0.49    0    0.00 XXX   0 X   -0.12 ASP 311 A   -0.05 HIS 341 A\n",
      "ASP 311 A                                            0.00 XXX   0 X   -0.61 MET 313 A    0.00 XXX   0 X\n",
      "ASP 311 A                                            0.00 XXX   0 X   -0.83 GLU 314 A    0.00 XXX   0 X\n",
      "ASP 311 A                                            0.00 XXX   0 X   -0.40 ASP 315 A    0.00 XXX   0 X\n",
      "ASP 311 A                                            0.00 XXX   0 X   -0.72 VAL 316 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 315 A   6.34*   20 %    0.96  336   0.20    0    0.87 ASP 342 A    0.00 XXX   0 X   -0.04 ARG 338 A\n",
      "ASP 315 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.14 ASP 311 A\n",
      "ASP 315 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.40 ASP 342 A\n",
      "\n",
      "ASP 319 A   1.83    15 %    0.73  324   0.03    0   -1.58 ARG 134 A   -0.76 GLY 321 A   -0.38 ARG 134 A\n",
      "\n",
      "ASP 342 A   3.08*    0 %    0.48  257   0.00    0   -0.87 ASP 315 A    0.00 XXX   0 X   -0.15 ARG 338 A\n",
      "ASP 342 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.18 HIS 341 A\n",
      "\n",
      "ASP 350 A   3.76     0 %    0.21  206   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.25 LYS 253 A\n",
      "\n",
      "GLU   4 A   2.92     0 %    0.28  187   0.00    0   -0.77 N+    2 A   -0.70 GLU   4 A   -0.38 N+    2 A\n",
      "\n",
      "GLU   6 A   3.84    24 %    1.04  349   0.00    0   -0.65 ARG 137 A   -0.66 GLU   6 A   -0.05 N+    2 A\n",
      "GLU   6 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.12 ARG   5 A\n",
      "GLU   6 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU   4 A\n",
      "GLU   6 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.14 ASP  67 A\n",
      "GLU   6 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.43 ARG 137 A\n",
      "\n",
      "GLU  27 A   4.11    13 %    0.87  318   0.20    0   -0.85 SER  26 A   -0.45 GLU  27 A   -0.00 LYS  22 A\n",
      "GLU  27 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.09 ARG  23 A\n",
      "GLU  27 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.07 HIS  31 A\n",
      "\n",
      "GLU  72 A   5.47*  100 %    2.36  562   0.17    0   -0.34 LYS  39 A    0.00 XXX   0 X   -0.16 LYS 291 A\n",
      "GLU  72 A                                           -0.48 ASP 142 A    0.00 XXX   0 X   -0.05 ARG 324 A\n",
      "GLU  72 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.14 ASP  41 A\n",
      "GLU  72 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.42 ASP 145 A\n",
      "GLU  72 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.10 LYS  39 A\n",
      "\n",
      "GLU  73 A   8.38*   90 %    2.95  532   0.08    0   -0.85 LYS  39 A    0.00 XXX   0 X   -0.01 ARG 324 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 ASP  41 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.00 LYS  39 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.09 ASP  49 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.55 GLU  72 A\n",
      "\n",
      "GLU  89 A   5.51    46 %    1.01  409   0.11    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.20 ARG  23 A\n",
      "GLU  89 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.12 GLU  27 A\n",
      "GLU  89 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 HIS  31 A\n",
      "\n",
      "GLU  95 A   3.70     0 %    0.31  175   0.00    0   -0.68 LYS  99 A    0.00 XXX   0 X   -0.05 ARG  23 A\n",
      "GLU  95 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.38 LYS  99 A\n",
      "\n",
      "GLU 119 A   4.30     0 %    0.29  205   0.00    0    0.00 XXX   0 X   -0.63 GLU 119 A   -0.10 ARG 122 A\n",
      "GLU 119 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 ASP  80 A\n",
      "GLU 119 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 ASP 120 A\n",
      "\n",
      "GLU 131 A   4.45     1 %    0.36  283   0.01    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.41 ARG 324 A\n",
      "\n",
      "GLU 154 A   4.04     2 %    0.35  287   0.01    0   -0.72 ARG 152 A    0.00 XXX   0 X    0.02 GLU 238 A\n",
      "GLU 154 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.18 ASP 263 A\n",
      "GLU 154 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.31 ARG 152 A\n",
      "\n",
      "GLU 192 A   3.88     0 %    0.35  255   0.00    0    0.00 XXX   0 X   -0.04 GLY 190 A    0.02 ASP 187 A\n",
      "GLU 192 A                                            0.00 XXX   0 X   -0.73 GLU 192 A   -0.23 HIS 191 A\n",
      "\n",
      "GLU 215 A   3.63    10 %    0.43  309   0.07    0   -0.85 LYS 103 A    0.00 XXX   0 X   -0.16 ARG  15 A\n",
      "GLU 215 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.05 ARG 199 A\n",
      "GLU 215 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP 213 A\n",
      "GLU 215 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.41 LYS 103 A\n",
      "\n",
      "GLU 235 A   3.26    68 %    2.28  472   0.12    0   -0.46 THR 233 A    0.00 XXX   0 X   -0.30 HIS 244 A\n",
      "GLU 235 A                                           -0.85 SER 259 A    0.00 XXX   0 X   -0.65 LYS 251 A\n",
      "GLU 235 A                                           -0.85 LYS 251 A    0.00 XXX   0 X   -0.53 HIS 261 A\n",
      "\n",
      "GLU 238 A   3.63    15 %    0.62  323   0.00    0   -0.74 ARG 152 A    0.00 XXX   0 X   -0.35 LYS 148 A\n",
      "GLU 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.30 HIS 241 A\n",
      "GLU 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 HIS 244 A\n",
      "GLU 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.24 ASP 263 A\n",
      "GLU 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.33 ARG 152 A\n",
      "\n",
      "GLU 246 A   2.95    20 %    1.10  337   0.12    0   -0.73 SER 243 A    0.00 XXX   0 X   -0.43 ARG 355 A\n",
      "GLU 246 A                                           -1.62 ARG 355 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "GLU 290 A   1.29    98 %    3.05  556   0.08    0   -0.59 ARG 281 A    0.00 XXX   0 X   -0.00 LYS 267 A\n",
      "GLU 290 A                                           -1.60 HIS 295 A    0.00 XXX   0 X   -0.23 LYS 291 A\n",
      "GLU 290 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 LYS 329 A\n",
      "GLU 290 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.10 HIS 241 A\n",
      "GLU 290 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -2.03 ARG 281 A\n",
      "GLU 290 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.70 HIS 295 A\n",
      "\n",
      "GLU 304 A   3.18    26 %    1.06  354   0.21    0   -0.46 HIS 303 A   -0.84 GLY 203 A   -0.19 ARG 202 A\n",
      "GLU 304 A                                            0.00 XXX   0 X   -0.00 LEU 204 A   -0.05 LYS 219 A\n",
      "GLU 304 A                                            0.00 XXX   0 X   -0.74 GLY 205 A   -0.31 HIS 303 A\n",
      "\n",
      "GLU 314 A   5.06     0 %    0.26  273   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.01 LYS 329 A\n",
      "GLU 314 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.31 ASP 311 A\n",
      "\n",
      "GLU 339 A   4.02     0 %    0.19  164   0.00    0    0.00 XXX   0 X   -0.67 GLU 339 A   -0.07 ARG 338 A\n",
      "GLU 339 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP 342 A\n",
      "\n",
      "C-  355 A   3.34     0 %    0.16  125   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.07 ARG 355 A\n",
      "C-  355 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 246 A\n",
      "\n",
      "HIS  31 A   6.33     0 %   -0.29  196   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.07 GLU  27 A\n",
      "HIS  31 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP  33 A\n",
      "HIS  31 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU  89 A\n",
      "\n",
      "HIS 191 A   6.82     6 %   -0.67  298   0.00    0    0.00 XXX   0 X    0.37 LEU 188 A   -0.02 LYS  22 A\n",
      "HIS 191 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.41 ASP 187 A\n",
      "HIS 191 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.23 GLU 192 A\n",
      "\n",
      "HIS 221 A   5.75    12 %   -0.53  315   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 219 A\n",
      "HIS 221 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.17 HIS 224 A\n",
      "\n",
      "HIS 224 A   6.57    11 %   -0.55  311   0.00    0    0.00 XXX   0 X    0.58 SER 335 A    0.04 ASP 277 A\n",
      "\n",
      "HIS 241 A   4.98    65 %   -1.23  464   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.26 LYS 148 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 ARG 152 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.17 LYS 267 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.56 ARG 281 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP  41 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 ASP 145 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.30 GLU 238 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 HIS 244 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 ASP 263 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 GLU 290 A\n",
      "\n",
      "HIS 244 A   5.80    28 %   -0.91  359   0.00    0    0.00 XXX   0 X    0.04 GLY 236 A   -0.09 LYS 239 A\n",
      "HIS 244 A                                            0.00 XXX   0 X    0.22 VAL 237 A   -0.10 LYS 251 A\n",
      "HIS 244 A                                            0.00 XXX   0 X    0.01 SER 242 A   -0.01 LYS 267 A\n",
      "HIS 244 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.30 GLU 235 A\n",
      "HIS 244 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 238 A\n",
      "HIS 244 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP 245 A\n",
      "HIS 244 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.27 HIS 261 A\n",
      "\n",
      "HIS 261 A   6.60    35 %   -0.93  379   0.00    0    0.00 XXX   0 X    0.85 GLY 236 A   -0.29 LYS 251 A\n",
      "HIS 261 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 267 A\n",
      "HIS 261 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.53 GLU 235 A\n",
      "\n",
      "HIS 295 A   5.60   100 %   -3.33  658   0.00    0    1.60 GLU 290 A    0.00 XXX   0 X   -0.15 LYS 267 A\n",
      "HIS 295 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.16 ARG 281 A\n",
      "HIS 295 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.00 LYS 291 A\n",
      "HIS 295 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.44 ASP 145 A\n",
      "HIS 295 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.70 GLU 290 A\n",
      "\n",
      "HIS 303 A   6.22    34 %   -0.66  377   0.00    0    0.46 GLU 304 A    0.00 XXX   0 X   -0.39 ARG 202 A\n",
      "HIS 303 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.31 GLU 304 A\n",
      "\n",
      "HIS 309 A   2.56   100 %   -3.24  632   0.00    0    0.00 XXX   0 X    0.06 HIS 295 A   -0.74 HIS 295 A\n",
      "HIS 309 A                                            0.00 XXX   0 X    0.23 ILE 332 A   -0.25 HIS 341 A\n",
      "\n",
      "HIS 341 A   4.22    77 %   -2.52  498   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 311 A\n",
      "HIS 341 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.18 ASP 342 A\n",
      "\n",
      "CYS 140 A  12.78   100 %    3.75  624   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.13 ARG 324 A\n",
      "CYS 140 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 ASP 294 A\n",
      "\n",
      "CYS 159 A  15.19   100 %    3.86  700   0.00    0   -0.06 GLN 265 A    0.00 XXX   0 X   -0.03 LYS 267 A\n",
      "CYS 159 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 CYS 140 A\n",
      "CYS 159 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 CYS 172 A\n",
      "CYS 159 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.30 ASP 294 A\n",
      "CYS 159 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.32 ASP 145 A\n",
      "CYS 159 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.53 CYS 175 A\n",
      "\n",
      "CYS 172 A  12.08   100 %    3.66  715   0.00    0   -0.14 THR  11 A    0.00 XXX   0 X   -0.50 ARG 199 A\n",
      "CYS 172 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 CYS 175 A\n",
      "\n",
      "CYS 175 A   9.09*  100 %    3.67  714   0.00    0   -3.60 CYS 269 A    0.00 XXX   0 X    0.02 ASP 145 A\n",
      "\n",
      "CYS 269 A  19.48*  100 %    4.24  712   0.00    0    3.60 CYS 175 A   -0.02 CYS 269 A    0.49 CYS 159 A\n",
      "CYS 269 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 CYS 172 A\n",
      "CYS 269 A                                            0.00 XXX   0 X    0.00 XXX   0 X    2.03 CYS 175 A\n",
      "\n",
      "CYS 349 A   9.57     1 %    0.40  284   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 311 A\n",
      "CYS 349 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 314 A\n",
      "CYS 349 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.11 ASP 350 A\n",
      "\n",
      "TYR  50 A  12.47    60 %    1.83  448   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.23 ASP  49 A\n",
      "TYR  50 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU  72 A\n",
      "TYR  50 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 126 A\n",
      "TYR  50 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.29 LYS  39 A\n",
      "TYR  50 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.60 GLU  73 A\n",
      "\n",
      "TYR  97 A  12.15    89 %    3.48  531   0.00    0   -0.35 ARG  15 A    0.00 XXX   0 X   -1.07 ARG  15 A\n",
      "TYR  97 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP  94 A\n",
      "TYR  97 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 215 A\n",
      "TYR  97 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 TYR 101 A\n",
      "\n",
      "TYR 101 A  10.24    26 %    0.81  354   0.00    0   -0.53 SER 193 A    0.00 XXX   0 X   -0.10 ARG  15 A\n",
      "TYR 101 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 GLU 192 A\n",
      "\n",
      "TYR 130 A  10.10     4 %    0.46  293   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.36 LYS  68 A\n",
      "\n",
      "TYR 183 A  11.75    61 %    1.58  453   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 154 A\n",
      "TYR 183 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 CYS 175 A\n",
      "TYR 183 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP 187 A\n",
      "TYR 183 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 263 A\n",
      "\n",
      "TYR 196 A  13.17    38 %    2.33  389   0.00    0    0.85 ASP 187 A   -0.79 LEU 188 A    0.05 GLU 192 A\n",
      "TYR 196 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.27 TYR 208 A\n",
      "TYR 196 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.48 ASP 187 A\n",
      "\n",
      "TYR 208 A  11.41    46 %    1.34  411   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.07 ASP 187 A\n",
      "\n",
      "TYR 268 A  17.93   100 %    4.86  684   0.00    0    0.77 TYR 279 A    0.00 XXX   0 X    0.17 ASP 145 A\n",
      "TYR 268 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 CYS 159 A\n",
      "TYR 268 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.54 ARG 281 A\n",
      "TYR 268 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.46 GLU 290 A\n",
      "TYR 268 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 ASP 294 A\n",
      "TYR 268 A                                            0.00 XXX   0 X    0.00 XXX   0 X    2.03 TYR 279 A\n",
      "\n",
      "TYR 279 A  12.08   100 %    3.49  678   0.00    0   -0.51 ARG 281 A    0.00 XXX   0 X    0.50 ASP 145 A\n",
      "TYR 279 A                                           -0.77 TYR 268 A    0.00 XXX   0 X    0.01 CYS 159 A\n",
      "TYR 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.75 ARG 281 A\n",
      "TYR 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.36 ASP 294 A\n",
      "TYR 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.76 GLU 290 A\n",
      "\n",
      "TYR 288 A  12.91    68 %    1.44  472   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP  41 A\n",
      "TYR 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.17 GLU  72 A\n",
      "TYR 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP 142 A\n",
      "TYR 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 ASP 145 A\n",
      "TYR 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.51 ARG 281 A\n",
      "TYR 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.80 GLU 290 A\n",
      "TYR 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.85 ASP 294 A\n",
      "\n",
      "LYS  16 A   9.20    87 %   -2.94  524   0.00    0    0.85 ASP  94 A    0.00 XXX   0 X   -0.26 LYS  91 A\n",
      "LYS  16 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.05 ASP  94 A\n",
      "\n",
      "LYS  22 A   9.53    44 %   -0.97  405   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 GLU  27 A\n",
      "\n",
      "LYS  32 A  10.25     0 %   -0.31  236   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.06 ASP  33 A\n",
      "\n",
      "LYS  39 A  13.17    62 %   -1.79  454   0.00    0    0.34 GLU  72 A    0.00 XXX   0 X    0.28 ASP  41 A\n",
      "LYS  39 A                                            0.85 GLU  73 A    0.00 XXX   0 X    0.30 ASP  49 A\n",
      "LYS  39 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.28 ASP 142 A\n",
      "LYS  39 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 294 A\n",
      "LYS  39 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 324 A\n",
      "LYS  39 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.29 TYR  50 A\n",
      "LYS  39 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.10 GLU  72 A\n",
      "LYS  39 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.00 GLU  73 A\n",
      "\n",
      "LYS  61 A   9.38    37 %   -1.38  385   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.11 ASP  66 A\n",
      "LYS  61 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP  67 A\n",
      "\n",
      "LYS  68 A  10.47     6 %   -0.46  298   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP  67 A\n",
      "LYS  68 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 ARG 137 A\n",
      "LYS  68 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 165 A\n",
      "LYS  68 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.36 TYR 130 A\n",
      "\n",
      "LYS  91 A  11.20    36 %   -0.82  382   0.00    0    0.85 ASP  94 A    0.00 XXX   0 X    0.67 ASP  94 A\n",
      "\n",
      "LYS  99 A  11.34     0 %   -0.22  141   0.00    0    0.68 GLU  95 A    0.00 XXX   0 X   -0.00 ARG  23 A\n",
      "LYS  99 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 GLU  95 A\n",
      "\n",
      "LYS 102 A  10.42     0 %   -0.12  190   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 104 A\n",
      "\n",
      "LYS 103 A  11.30     6 %   -0.45  298   0.00    0    0.85 GLU 215 A    0.00 XXX   0 X   -0.09 ARG  15 A\n",
      "LYS 103 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP 104 A\n",
      "LYS 103 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 105 A\n",
      "LYS 103 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 199 A\n",
      "LYS 103 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.41 GLU 215 A\n",
      "\n",
      "LYS 116 A  10.23     0 %   -0.12  213   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP  94 A\n",
      "LYS 116 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 120 A\n",
      "LYS 116 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.18 LYS  91 A\n",
      "\n",
      "LYS 135 A  10.54     0 %   -0.04   82   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP 165 A\n",
      "\n",
      "LYS 148 A  10.50    12 %   -0.36  314   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP  41 A\n",
      "LYS 148 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.05 ARG 152 A\n",
      "LYS 148 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.35 GLU 238 A\n",
      "LYS 148 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 263 A\n",
      "\n",
      "LYS 189 A  10.46     0 %   -0.04  110   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 219 A  10.37     0 %   -0.19  220   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 304 A\n",
      "\n",
      "LYS 226 A  10.46     0 %   -0.04   96   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 229 A  10.85     0 %   -0.10  111   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.24 ASP 227 A\n",
      "LYS 229 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 ASP 230 A\n",
      "\n",
      "LYS 239 A  10.51     0 %   -0.08  150   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP 245 A\n",
      "\n",
      "LYS 251 A  10.84    33 %   -1.16  374   0.00    0    0.85 GLU 235 A    0.00 XXX   0 X    0.65 GLU 235 A\n",
      "\n",
      "LYS 253 A  10.54     0 %   -0.21  219   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.25 ASP 350 A\n",
      "\n",
      "LYS 258 A  10.36     0 %   -0.17  226   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.10 ARG 223 A\n",
      "LYS 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP 230 A\n",
      "LYS 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 277 A\n",
      "\n",
      "LYS 267 A   7.64   100 %   -3.32  608   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.00 ARG 152 A\n",
      "LYS 267 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 CYS 159 A\n",
      "LYS 267 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.50 ASP 263 A\n",
      "LYS 267 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.79 ARG 281 A\n",
      "LYS 267 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 GLU 290 A\n",
      "LYS 267 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP 294 A\n",
      "LYS 267 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.65 ASP 145 A\n",
      "\n",
      "LYS 285 A  10.40     0 %   -0.10  190   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 291 A  10.01    47 %   -1.26  413   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.12 ASP  41 A\n",
      "LYS 291 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 GLU  72 A\n",
      "LYS 291 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 ARG 281 A\n",
      "LYS 291 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.23 GLU 290 A\n",
      "LYS 291 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.35 ASP 294 A\n",
      "LYS 291 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.05 LYS  39 A\n",
      "\n",
      "LYS 329 A   9.59    18 %   -0.97  331   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU 290 A\n",
      "LYS 329 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 314 A\n",
      "LYS 329 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS 291 A\n",
      "\n",
      "ARG   5 A  12.07    10 %   -0.57  309   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.12 GLU   6 A\n",
      "ARG   5 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP  67 A\n",
      "ARG   5 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 105 A\n",
      "ARG   5 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 ARG 137 A\n",
      "\n",
      "ARG  15 A  12.14    68 %   -2.12  473   0.00    0    0.35 TYR  97 A    0.00 XXX   0 X    1.07 TYR  97 A\n",
      "ARG  15 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 TYR 101 A\n",
      "ARG  15 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP 213 A\n",
      "ARG  15 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 GLU 215 A\n",
      "\n",
      "ARG  23 A  11.87    37 %   -0.97  384   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.09 GLU  27 A\n",
      "ARG  23 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 GLU  89 A\n",
      "ARG  23 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP  94 A\n",
      "ARG  23 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU  95 A\n",
      "\n",
      "ARG 122 A  13.16     0 %   -0.45  261   0.00    0    0.70 ASP  80 A    0.00 XXX   0 X    0.10 GLU 119 A\n",
      "ARG 122 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 ASP 126 A\n",
      "ARG 122 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.26 ASP  80 A\n",
      "\n",
      "ARG 134 A  14.14     0 %   -0.33  237   0.00    0    1.58 ASP 319 A    0.00 XXX   0 X    0.00 ASP 165 A\n",
      "ARG 134 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 ASP 319 A\n",
      "\n",
      "ARG 137 A  13.63    35 %   -1.22  380   0.00    0    0.65 GLU   6 A    0.00 XXX   0 X    0.43 GLU   6 A\n",
      "ARG 137 A                                            0.70 ASP  67 A    0.00 XXX   0 X    0.57 ASP  67 A\n",
      "\n",
      "ARG 152 A  15.85    18 %   -0.87  332   0.00    0    0.72 GLU 154 A    0.00 XXX   0 X    0.31 GLU 154 A\n",
      "ARG 152 A                                            0.74 GLU 238 A    0.00 XXX   0 X    0.33 GLU 238 A\n",
      "ARG 152 A                                            1.55 ASP 263 A    0.00 XXX   0 X    0.57 ASP 263 A\n",
      "\n",
      "ARG 199 A  10.80    65 %   -2.25  462   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.50 CYS 172 A\n",
      "ARG 199 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 215 A\n",
      "\n",
      "ARG 202 A  11.55    41 %   -1.14  396   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.19 GLU 304 A\n",
      "\n",
      "ARG 223 A  13.63    34 %   -1.14  377   0.00    0    1.64 ASP 277 A    0.00 XXX   0 X    0.03 ASP 230 A\n",
      "ARG 223 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.61 ASP 277 A\n",
      "\n",
      "ARG 281 A  17.71   100 %   -2.15  598   0.00    0    0.51 TYR 279 A    0.00 XXX   0 X    0.54 TYR 268 A\n",
      "ARG 281 A                                            0.59 GLU 290 A    0.00 XXX   0 X    1.75 TYR 279 A\n",
      "ARG 281 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.51 TYR 288 A\n",
      "ARG 281 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.55 ASP 145 A\n",
      "ARG 281 A                                            0.00 XXX   0 X    0.00 XXX   0 X    2.03 GLU 290 A\n",
      "ARG 281 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.87 ASP 294 A\n",
      "\n",
      "ARG 324 A  11.59    54 %   -1.52  433   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU  72 A\n",
      "ARG 324 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU  73 A\n",
      "ARG 324 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.41 GLU 131 A\n",
      "ARG 324 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 CYS 140 A\n",
      "\n",
      "ARG 338 A  12.69     0 %   -0.07  146   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 315 A\n",
      "ARG 338 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 GLU 339 A\n",
      "ARG 338 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP 342 A\n",
      "\n",
      "ARG 355 A  14.03     6 %   -0.58  299   0.00    0    1.62 GLU 246 A    0.00 XXX   0 X    0.07 C-  355 A\n",
      "ARG 355 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.43 GLU 246 A\n",
      "\n",
      "N+    2 A   8.88     0 %   -0.32  186   0.00    0    0.77 GLU   4 A    0.00 XXX   0 X    0.05 GLU   6 A\n",
      "N+    2 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 GLU   4 A\n",
      "\n",
      "Coupled residues (marked *) were detected.Please rerun PropKa with the --display-coupled-residues \n",
      "or -d option for detailed information.\n",
      "\n",
      "INFO:pKa summary:\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "SUMMARY OF THIS PREDICTION\n",
      "       Group      pKa  model-pKa   ligand atom-type\n",
      "   ASP  33 A     3.91       3.80                      \n",
      "   ASP  41 A     2.21       3.80                      \n",
      "   ASP  49 A     6.17       3.80                      \n",
      "   ASP  66 A     3.97       3.80                      \n",
      "   ASP  67 A     3.50       3.80                      \n",
      "   ASP  80 A     3.08       3.80                      \n",
      "   ASP  94 A     2.91       3.80                      \n",
      "   ASP 104 A     3.88       3.80                      \n",
      "   ASP 105 A     4.10       3.80                      \n",
      "   ASP 111 A     3.81       3.80                      \n",
      "   ASP 120 A     2.75       3.80                      \n",
      "   ASP 126 A     3.39       3.80                      \n",
      "   ASP 142 A    16.43       3.80                      \n",
      "   ASP 145 A     3.62       3.80                      \n",
      "   ASP 165 A     3.15       3.80                      \n",
      "   ASP 187 A     2.15       3.80                      \n",
      "   ASP 213 A     3.05       3.80                      \n",
      "   ASP 227 A     3.07       3.80                      \n",
      "   ASP 230 A     3.96       3.80                      \n",
      "   ASP 245 A     3.84       3.80                      \n",
      "   ASP 263 A     3.38       3.80                      \n",
      "   ASP 277 A     3.47       3.80                      \n",
      "   ASP 294 A     9.23       3.80                      \n",
      "   ASP 311 A     3.81       3.80                      \n",
      "   ASP 315 A     6.34       3.80                      \n",
      "   ASP 319 A     1.83       3.80                      \n",
      "   ASP 342 A     3.08       3.80                      \n",
      "   ASP 350 A     3.76       3.80                      \n",
      "   GLU   4 A     2.92       4.50                      \n",
      "   GLU   6 A     3.84       4.50                      \n",
      "   GLU  27 A     4.11       4.50                      \n",
      "   GLU  72 A     5.47       4.50                      \n",
      "   GLU  73 A     8.38       4.50                      \n",
      "   GLU  89 A     5.51       4.50                      \n",
      "   GLU  95 A     3.70       4.50                      \n",
      "   GLU 119 A     4.30       4.50                      \n",
      "   GLU 131 A     4.45       4.50                      \n",
      "   GLU 154 A     4.04       4.50                      \n",
      "   GLU 192 A     3.88       4.50                      \n",
      "   GLU 215 A     3.63       4.50                      \n",
      "   GLU 235 A     3.26       4.50                      \n",
      "   GLU 238 A     3.63       4.50                      \n",
      "   GLU 246 A     2.95       4.50                      \n",
      "   GLU 290 A     1.29       4.50                      \n",
      "   GLU 304 A     3.18       4.50                      \n",
      "   GLU 314 A     5.06       4.50                      \n",
      "   GLU 339 A     4.02       4.50                      \n",
      "   C-  355 A     3.34       3.20                      \n",
      "   HIS  31 A     6.33       6.50                      \n",
      "   HIS 191 A     6.82       6.50                      \n",
      "   HIS 221 A     5.75       6.50                      \n",
      "   HIS 224 A     6.57       6.50                      \n",
      "   HIS 241 A     4.98       6.50                      \n",
      "   HIS 244 A     5.80       6.50                      \n",
      "   HIS 261 A     6.60       6.50                      \n",
      "   HIS 295 A     5.60       6.50                      \n",
      "   HIS 303 A     6.22       6.50                      \n",
      "   HIS 309 A     2.56       6.50                      \n",
      "   HIS 341 A     4.22       6.50                      \n",
      "   CYS 140 A    12.78       9.00                      \n",
      "   CYS 159 A    15.19       9.00                      \n",
      "   CYS 172 A    12.08       9.00                      \n",
      "   CYS 175 A     9.09       9.00                      \n",
      "   CYS 269 A    19.48       9.00                      \n",
      "   CYS 349 A     9.57       9.00                      \n",
      "   TYR  50 A    12.47      10.00                      \n",
      "   TYR  97 A    12.15      10.00                      \n",
      "   TYR 101 A    10.24      10.00                      \n",
      "   TYR 130 A    10.10      10.00                      \n",
      "   TYR 183 A    11.75      10.00                      \n",
      "   TYR 196 A    13.17      10.00                      \n",
      "   TYR 208 A    11.41      10.00                      \n",
      "   TYR 268 A    17.93      10.00                      \n",
      "   TYR 279 A    12.08      10.00                      \n",
      "   TYR 288 A    12.91      10.00                      \n",
      "   LYS  16 A     9.20      10.50                      \n",
      "   LYS  22 A     9.53      10.50                      \n",
      "   LYS  32 A    10.25      10.50                      \n",
      "   LYS  39 A    13.17      10.50                      \n",
      "   LYS  61 A     9.38      10.50                      \n",
      "   LYS  68 A    10.47      10.50                      \n",
      "   LYS  91 A    11.20      10.50                      \n",
      "   LYS  99 A    11.34      10.50                      \n",
      "   LYS 102 A    10.42      10.50                      \n",
      "   LYS 103 A    11.30      10.50                      \n",
      "   LYS 116 A    10.23      10.50                      \n",
      "   LYS 135 A    10.54      10.50                      \n",
      "   LYS 148 A    10.50      10.50                      \n",
      "   LYS 189 A    10.46      10.50                      \n",
      "   LYS 219 A    10.37      10.50                      \n",
      "   LYS 226 A    10.46      10.50                      \n",
      "   LYS 229 A    10.85      10.50                      \n",
      "   LYS 239 A    10.51      10.50                      \n",
      "   LYS 251 A    10.84      10.50                      \n",
      "   LYS 253 A    10.54      10.50                      \n",
      "   LYS 258 A    10.36      10.50                      \n",
      "   LYS 267 A     7.64      10.50                      \n",
      "   LYS 285 A    10.40      10.50                      \n",
      "   LYS 291 A    10.01      10.50                      \n",
      "   LYS 329 A     9.59      10.50                      \n",
      "   ARG   5 A    12.07      12.50                      \n",
      "   ARG  15 A    12.14      12.50                      \n",
      "   ARG  23 A    11.87      12.50                      \n",
      "   ARG 122 A    13.16      12.50                      \n",
      "   ARG 134 A    14.14      12.50                      \n",
      "   ARG 137 A    13.63      12.50                      \n",
      "   ARG 152 A    15.85      12.50                      \n",
      "   ARG 199 A    10.80      12.50                      \n",
      "   ARG 202 A    11.55      12.50                      \n",
      "   ARG 223 A    13.63      12.50                      \n",
      "   ARG 281 A    17.71      12.50                      \n",
      "   ARG 324 A    11.59      12.50                      \n",
      "   ARG 338 A    12.69      12.50                      \n",
      "   ARG 355 A    14.03      12.50                      \n",
      "   N+    2 A     8.88       8.00                      \n",
      "\n",
      "INFO:PROPKA information:\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "Free energy of   folding (kcal/mol) as a function of pH (using neutral reference)\n",
      "  0.00     44.66\n",
      "  1.00     44.24\n",
      "  2.00     41.82\n",
      "  3.00     33.82\n",
      "  4.00     21.09\n",
      "  5.00     16.07\n",
      "  6.00     17.12\n",
      "  7.00     18.86\n",
      "  8.00     22.13\n",
      "  9.00     26.07\n",
      " 10.00     32.15\n",
      " 11.00     44.82\n",
      " 12.00     59.71\n",
      " 13.00     73.20\n",
      " 14.00     85.66\n",
      "\n",
      "The pH of optimum stability is  5.0 for which the free energy is   16.1 kcal/mol at 298K\n",
      "Could not determine pH values where the free energy is within 80 % of minimum\n",
      "Could not determine the pH-range where the free energy is negative\n",
      "\n",
      "\n",
      "Protein charge of folded and unfolded state as a function of pH\n",
      "    pH  unfolded  folded\n",
      "  0.00     50.99   50.91\n",
      "  0.10     50.99   50.88\n",
      "  0.20     50.99   50.85\n",
      "  0.30     50.99   50.82\n",
      "  0.40     50.99   50.77\n",
      "  0.50     50.98   50.72\n",
      "  0.60     50.98   50.65\n",
      "  0.70     50.97   50.57\n",
      "  0.80     50.96   50.48\n",
      "  0.90     50.96   50.36\n",
      "  1.00     50.94   50.23\n",
      "  1.10     50.93   50.07\n",
      "  1.20     50.91   49.89\n",
      "  1.30     50.89   49.68\n",
      "  1.40     50.86   49.44\n",
      "  1.50     50.82   49.16\n",
      "  1.60     50.78   48.85\n",
      "  1.70     50.72   48.49\n",
      "  1.80     50.65   48.09\n",
      "  1.90     50.56   47.64\n",
      "  2.00     50.44   47.13\n",
      "  2.10     50.30   46.56\n",
      "  2.20     50.13   45.91\n",
      "  2.30     49.91   45.20\n",
      "  2.40     49.64   44.39\n",
      "  2.50     49.31   43.50\n",
      "  2.60     48.90   42.51\n",
      "  2.70     48.40   41.41\n",
      "  2.80     47.80   40.22\n",
      "  2.90     47.07   38.92\n",
      "  3.00     46.20   37.51\n",
      "  3.10     45.17   36.02\n",
      "  3.20     43.97   34.43\n",
      "  3.30     42.58   32.78\n",
      "  3.40     41.01   31.06\n",
      "  3.50     39.25   29.30\n",
      "  3.60     37.31   27.52\n",
      "  3.70     35.23   25.73\n",
      "  3.80     33.02   23.96\n",
      "  3.90     30.72   22.23\n",
      "  4.00     28.37   20.55\n",
      "  4.10     26.01   18.95\n",
      "  4.20     23.67   17.45\n",
      "  4.30     21.38   16.04\n",
      "  4.40     19.18   14.74\n",
      "  4.50     17.10   13.54\n",
      "  4.60     15.14   12.44\n",
      "  4.70     13.34   11.43\n",
      "  4.80     11.70   10.50\n",
      "  4.90     10.22    9.64\n",
      "  5.00      8.90    8.83\n",
      "  5.10      7.74    8.07\n",
      "  5.20      6.71    7.33\n",
      "  5.30      5.81    6.62\n",
      "  5.40      5.00    5.93\n",
      "  5.50      4.27    5.25\n",
      "  5.60      3.60    4.57\n",
      "  5.70      2.97    3.90\n",
      "  5.80      2.35    3.24\n",
      "  5.90      1.73    2.58\n",
      "  6.00      1.10    1.93\n",
      "  6.10      0.45    1.29\n",
      "  6.20     -0.22    0.66\n",
      "  6.30     -0.91    0.05\n",
      "  6.40     -1.61   -0.54\n",
      "  6.50     -2.31   -1.11\n",
      "  6.60     -3.01   -1.64\n",
      "  6.70     -3.68   -2.15\n",
      "  6.80     -4.31   -2.62\n",
      "  6.90     -4.91   -3.05\n",
      "  7.00     -5.45   -3.45\n",
      "  7.10     -5.94   -3.81\n",
      "  7.20     -6.38   -4.14\n",
      "  7.30     -6.78   -4.45\n",
      "  7.40     -7.13   -4.74\n",
      "  7.50     -7.46   -5.01\n",
      "  7.60     -7.76   -5.26\n",
      "  7.70     -8.04   -5.51\n",
      "  7.80     -8.32   -5.76\n",
      "  7.90     -8.60   -6.00\n",
      "  8.00     -8.88   -6.26\n",
      "  8.10     -9.18   -6.52\n",
      "  8.20     -9.50   -6.80\n",
      "  8.30     -9.84   -7.11\n",
      "  8.40    -10.22   -7.43\n",
      "  8.50    -10.65   -7.79\n",
      "  8.60    -11.11   -8.19\n",
      "  8.70    -11.64   -8.63\n",
      "  8.80    -12.21   -9.11\n",
      "  8.90    -12.85   -9.64\n",
      "  9.00    -13.55  -10.22\n",
      "  9.10    -14.32  -10.85\n",
      "  9.20    -15.17  -11.54\n",
      "  9.30    -16.09  -12.29\n",
      "  9.40    -17.10  -13.08\n",
      "  9.50    -18.21  -13.94\n",
      "  9.60    -19.42  -14.85\n",
      "  9.70    -20.76  -15.82\n",
      "  9.80    -22.21  -16.86\n",
      "  9.90    -23.79  -17.95\n",
      " 10.00    -25.49  -19.10\n",
      " 10.10    -27.30  -20.31\n",
      " 10.20    -29.18  -21.57\n",
      " 10.30    -31.13  -22.86\n",
      " 10.40    -33.09  -24.19\n",
      " 10.50    -35.05  -25.53\n",
      " 10.60    -36.95  -26.87\n",
      " 10.70    -38.76  -28.19\n",
      " 10.80    -40.46  -29.49\n",
      " 10.90    -42.03  -30.76\n",
      " 11.00    -43.45  -31.99\n",
      " 11.10    -44.73  -33.18\n",
      " 11.20    -45.88  -34.33\n",
      " 11.30    -46.90  -35.46\n",
      " 11.40    -47.83  -36.54\n",
      " 11.50    -48.67  -37.61\n",
      " 11.60    -49.47  -38.64\n",
      " 11.70    -50.22  -39.65\n",
      " 11.80    -50.97  -40.64\n",
      " 11.90    -51.72  -41.60\n",
      " 12.00    -52.49  -42.54\n",
      " 12.10    -53.29  -43.45\n",
      " 12.20    -54.12  -44.34\n",
      " 12.30    -54.97  -45.20\n",
      " 12.40    -55.84  -46.03\n",
      " 12.50    -56.72  -46.84\n",
      " 12.60    -57.58  -47.62\n",
      " 12.70    -58.41  -48.37\n",
      " 12.80    -59.18  -49.10\n",
      " 12.90    -59.90  -49.80\n",
      " 13.00    -60.55  -50.48\n",
      " 13.10    -61.12  -51.13\n",
      " 13.20    -61.61  -51.75\n",
      " 13.30    -62.04  -52.35\n",
      " 13.40    -62.40  -52.91\n",
      " 13.50    -62.70  -53.45\n",
      " 13.60    -62.95  -53.96\n",
      " 13.70    -63.15  -54.44\n",
      " 13.80    -63.32  -54.89\n",
      " 13.90    -63.45  -55.30\n",
      " 14.00    -63.56  -55.69\n",
      "The pI is  6.31 (folded) and  6.17 (unfolded)\n",
      "\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "INFO:Applying pKa values at a pH of 7.60:\n",
      "INFO:Adding hydrogens to biomolecule.\n",
      "INFO:Debumping biomolecule (again).\n",
      "INFO:Optimizing hydrogen bonds\n",
      "INFO:Applying force field to biomolecule states.\n",
      "INFO:Regenerating headers.\n",
      "INFO:Regenerating PDB lines.\n",
      "WARNING:Ignoring 466 header lines in output.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "pH: 7.6\tname: 1ka1_hal2\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:PDB2PQR v3.6.2: biomolecular structure conversion software.\n",
      "INFO:Please cite:  Jurrus E, et al.  Improvements to the APBS biomolecular solvation software suite.  Protein Sci 27 112-128 (2018).\n",
      "INFO:Please cite:  Dolinsky TJ, et al.  PDB2PQR: expanding and upgrading automated preparation of biomolecular structures for molecular simulations. Nucleic Acids Res 35 W522-W525 (2007).\n",
      "INFO:Checking and transforming input arguments.\n",
      "INFO:Loading topology files.\n",
      "INFO:Loading molecule: /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/AtSAL1_updated.pdb\n",
      "INFO:Setting up molecule.\n",
      "INFO:Created biomolecule object with 352 residues and 2628 atoms.\n",
      "INFO:Setting termini states for biomolecule chains.\n",
      "INFO:Loading forcefield.\n",
      "INFO:Loading hydrogen topology definitions.\n",
      "WARNING:Missing atom OXT in residue ALA A 352\n",
      "WARNING:Missing atom OXT in residue ALA A 352\n",
      "INFO:Attempting to repair 1 missing atoms in biomolecule.\n",
      "WARNING:Missing atom OXT in residue ALA A 352\n",
      "INFO:Added atom OXT to residue ALA A 352 at coordinates -4.698, -13.324, -18.987\n",
      "INFO:Updating disulfide bridges.\n",
      "INFO:Debumping biomolecule.\n",
      "INFO:Assigning titration states with PROPKA.\n",
      "INFO:\n",
      "propka3.5.1                                                                                  2024-07-26\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "--                                                                           --\n",
      "--  PROPKA: A PROTEIN PKA PREDICTOR                                          --\n",
      "--                                                                           --\n",
      "--  VERSION 1.0,  04/25/2004,  IOWA CITY                                     --\n",
      "--  BY HUI LI                                                                --\n",
      "--                                                                           --\n",
      "--  VERSION 2.0,  11/05/2007, IOWA CITY/COPENHAGEN                           --\n",
      "--  BY DELPHINE C. BAS AND DAVID M. ROGERS                                   --\n",
      "--                                                                           --\n",
      "--  VERSION 3.0,  01/06/2011, COPENHAGEN                                     --\n",
      "--  BY MATS H.M. OLSSON AND CHRESTEN R. SONDERGARD                           --\n",
      "--                                                                           --\n",
      "--  VERSION 3.1,  07/01/2011, COPENHAGEN                                     --\n",
      "--  BY CHRESTEN R. SONDERGARD AND MATS H.M. OLSSON                           --\n",
      "--                                                                           --\n",
      "--  VERSION 3.2,  06/17/2020, PLANET EARTH                                   --\n",
      "--  SEE ABOVE FOR AUTHORS                                                    --\n",
      "--                                                                           --\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "References:\n",
      "\n",
      "Very Fast Empirical Prediction and Rationalization of Protein pKa Values.\n",
      "Hui Li, Andrew D. Robertson and Jan H. Jensen. PROTEINS: Structure, Function,\n",
      "and Bioinformatics. 61:704-721 (2005)\n",
      "\n",
      "Very Fast Prediction and Rationalization of pKa Values for Protein-Ligand\n",
      "Complexes.  Delphine C. Bas, David M. Rogers and Jan H. Jensen.  PROTEINS:\n",
      "Structure, Function, and Bioinformatics 73:765-783 (2008)\n",
      "\n",
      "PROPKA3: Consistent Treatment of Internal and Surface Residues in Empirical\n",
      "pKa predictions.  Mats H.M. Olsson, Chresten R. Sondergard, Michal Rostkowski,\n",
      "and Jan H. Jensen.  Journal of Chemical Theory and Computation, 7(2):525-537\n",
      "(2011)\n",
      "\n",
      "Improved Treatment of Ligands and Coupling Effects in Empirical Calculation\n",
      "and Rationalization of pKa Values.  Chresten R. Sondergaard, Mats H.M. Olsson,\n",
      "Michal Rostkowski, and Jan H. Jensen.  Journal of Chemical Theory and\n",
      "Computation, (2011)\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "\n",
      "INFO:Coupling map:\n",
      "\n",
      "INFO:Calculating pKas for Conformation container 1A with 3086 atoms and 869 groups\n",
      "INFO:did not converge in 10 iterations\n",
      "INFO:pKa determinants:\n",
      "\n",
      "---------  -----   ------   ---------------------    --------------    --------------    --------------\n",
      "                            DESOLVATION  EFFECTS       SIDECHAIN          BACKBONE        COULOMBIC\n",
      " RESIDUE    pKa    BURIED     REGULAR      RE        HYDROGEN BOND     HYDROGEN BOND      INTERACTION\n",
      "---------  -----   ------   ---------   ---------    --------------    --------------    --------------\n",
      "\n",
      "ASP   8 A   3.99    18 %    0.58  331   0.18    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.03 N+    1 A\n",
      "ASP   8 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.15 LYS   5 A\n",
      "ASP   8 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.30 LYS  11 A\n",
      "ASP   8 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.12 LYS  12 A\n",
      "ASP   8 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU   6 A\n",
      "\n",
      "ASP  32 A   3.90     0 %    0.12  137   0.00    0    0.00 XXX   0 X   -0.02 VAL  33 A    0.00 XXX   0 X\n",
      "\n",
      "ASP  38 A   3.69     0 %    0.12  180   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.24 LYS  39 A\n",
      "\n",
      "ASP  46 A   7.13    91 %    2.32  536   0.55    0    0.00 XXX   0 X    0.00 XXX   0 X    0.25 GLU  71 A\n",
      "ASP  46 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 GLU  72 A\n",
      "ASP  46 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 137 A\n",
      "\n",
      "ASP  73 A   3.49     0 %    0.30  267   0.00    0    0.00 XXX   0 X   -0.69 GLY  75 A   -0.06 ARG  78 A\n",
      "ASP  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS 317 A\n",
      "ASP  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 ASP 118 A\n",
      "\n",
      "ASP  76 A   4.20     0 %    0.23  268   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.00 LYS  79 A\n",
      "ASP  76 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.14 ASP  73 A\n",
      "ASP  76 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP  80 A\n",
      "\n",
      "ASP  80 A   3.82     0 %    0.31  184   0.00    0   -0.20 SER  82 A   -0.22 ASP  80 A    0.03 ASP  38 A\n",
      "ASP  80 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP  84 A\n",
      "\n",
      "ASP  84 A   3.48     0 %    0.35  226   0.00    0    0.00 XXX   0 X   -0.06 ASP  84 A   -0.25 ARG  88 A\n",
      "ASP  84 A                                            0.00 XXX   0 X   -0.35 THR  85 A    0.00 XXX   0 X\n",
      "\n",
      "ASP  95 A   3.71     0 %    0.32  279   0.00    0    0.00 XXX   0 X   -0.01 ASP  95 A   -0.28 LYS  23 A\n",
      "ASP  95 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.12 LYS  91 A\n",
      "\n",
      "ASP 111 A   3.94     0 %    0.21  211   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.03 ARG  78 A\n",
      "ASP 111 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.14 ARG 115 A\n",
      "ASP 111 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP 112 A\n",
      "\n",
      "ASP 112 A   2.85     0 %    0.34  258   0.00    0   -0.49 SER 109 A   -0.62 SER 109 A   -0.18 ARG 115 A\n",
      "\n",
      "ASP 118 A   3.20    10 %    0.65  308   0.06    0   -0.77 SER  74 A   -0.25 SER  74 A   -0.26 ARG  78 A\n",
      "ASP 118 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 115 A\n",
      "\n",
      "ASP 134 A  13.40*  100 %    3.03  569   0.41    0    1.52 ASP 137 A   -0.26 ILE 136 A   -0.01 LYS 285 A\n",
      "ASP 134 A                                            0.96 ASP 288 A    0.00 XXX   0 X    0.29 GLU  71 A\n",
      "ASP 134 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.68 ASP  46 A\n",
      "ASP 134 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.89 ASP 137 A\n",
      "ASP 134 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.08 ASP 288 A\n",
      "\n",
      "ASP 137 A   2.84*  100 %    2.61  581   0.29    0   -1.52 ASP 134 A   -0.00 GLN 260 A   -0.47 LYS 262 A\n",
      "ASP 137 A                                           -1.46 ASP 288 A    0.00 XXX   0 X    0.05 GLU  71 A\n",
      "ASP 137 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 GLU 284 A\n",
      "ASP 137 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.65 ARG 276 A\n",
      "\n",
      "ASP 146 A   3.60    12 %    0.79  316   0.11    0   -0.79 ARG 144 A    0.00 XXX   0 X   -0.31 ARG 144 A\n",
      "\n",
      "ASP 186 A   3.58     0 %    0.15  116   0.00    0   -0.37 SER 184 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "ASP 206 A   3.75    53 %    1.44  431   0.23    0   -1.15 ARG  19 A    0.00 XXX   0 X   -0.59 ARG  19 A\n",
      "\n",
      "ASP 239 A   3.65     0 %    0.43  269   0.00    0    0.00 XXX   0 X   -0.34 ASP 239 A   -0.04 HIS 235 A\n",
      "ASP 239 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.17 HIS 238 A\n",
      "ASP 239 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 HIS 279 A\n",
      "\n",
      "ASP 258 A   6.14    44 %    1.75  404   0.62    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.00 LYS 140 A\n",
      "ASP 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.34 ARG 144 A\n",
      "ASP 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.05 LYS 262 A\n",
      "ASP 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.34 ASP 146 A\n",
      "ASP 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 232 A\n",
      "\n",
      "ASP 270 A   5.41    76 %    2.89  493   0.45    0   -0.11 SER 174 A   -0.21 ILE 175 A   -0.89 ARG 268 A\n",
      "ASP 270 A                                           -0.51 ARG 268 A   -0.01 ASP 270 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 288 A   7.25*   81 %    2.27  509   0.16    0   -0.96 ASP 134 A    0.00 XXX   0 X   -0.13 LYS 262 A\n",
      "ASP 288 A                                            1.46 ASP 137 A    0.00 XXX   0 X   -0.22 LYS 285 A\n",
      "ASP 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.62 ASP 137 A\n",
      "ASP 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.54 ARG 276 A\n",
      "ASP 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.54 GLU 284 A\n",
      "ASP 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.75 HIS 289 A\n",
      "\n",
      "ASP 305 A   4.25    25 %    1.48  352   0.32    0    0.00 XXX   0 X   -0.17 ASP 305 A   -0.12 LYS 309 A\n",
      "ASP 305 A                                            0.00 XXX   0 X   -0.07 ALA 307 A   -0.04 ARG 341 A\n",
      "ASP 305 A                                            0.00 XXX   0 X   -0.00 GLY 308 A    0.02 ASP 320 A\n",
      "ASP 305 A                                            0.00 XXX   0 X   -0.63 LYS 309 A    0.23 ASP 322 A\n",
      "ASP 305 A                                            0.00 XXX   0 X   -0.58 ASP 322 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 312 A   2.16    11 %    0.68  312   0.01    0   -0.85 SER 314 A   -0.15 SER 314 A   -0.10 LYS 159 A\n",
      "ASP 312 A                                           -0.85 LYS 315 A    0.00 XXX   0 X   -0.38 LYS 315 A\n",
      "\n",
      "ASP 320 A   3.88     0 %    0.22  254   0.00    0    0.00 XXX   0 X   -0.12 ASP 320 A   -0.13 LYS 315 A\n",
      "ASP 320 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.11 ASP 312 A\n",
      "\n",
      "ASP 322 A   3.54    11 %    0.50  311   0.00    0   -0.66 THR 323 A    0.00 XXX   0 X   -0.01 LYS 309 A\n",
      "ASP 322 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.10 ARG 341 A\n",
      "\n",
      "ASP 342 A   4.00     0 %    0.37  242   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.04 LYS 247 A\n",
      "ASP 342 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.05 LYS 338 A\n",
      "ASP 342 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 ARG 341 A\n",
      "\n",
      "GLU   4 A   4.37     0 %    0.15  174   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.19 N+    1 A\n",
      "GLU   4 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.24 LYS   5 A\n",
      "GLU   4 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU   6 A\n",
      "GLU   4 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 ASP   8 A\n",
      "\n",
      "GLU   6 A   3.70    28 %    1.00  361   0.00    0   -0.72 ARG 129 A   -0.65 GLU   6 A   -0.06 LYS   5 A\n",
      "GLU   6 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.37 ARG 129 A\n",
      "\n",
      "GLU  58 A   4.99    32 %    0.70  370   0.10    0   -0.31 CYS 119 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "GLU  60 A   4.70    11 %    0.55  313   0.02    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.40 LYS  12 A\n",
      "GLU  60 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS  59 A\n",
      "GLU  60 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP   8 A\n",
      "GLU  60 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 101 A\n",
      "\n",
      "GLU  64 A   4.73     0 %    0.31  278   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.32 ARG 129 A\n",
      "GLU  64 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 GLU   6 A\n",
      "GLU  64 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 HIS 130 A\n",
      "GLU  64 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 GLU 156 A\n",
      "GLU  64 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 157 A\n",
      "\n",
      "GLU  71 A   5.17    41 %    0.73  396   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.20 LYS 285 A\n",
      "GLU  71 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 317 A\n",
      "GLU  71 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 ASP 288 A\n",
      "\n",
      "GLU  72 A   5.72    30 %    0.59  365   0.07    0   -0.15 TYR  47 A    0.00 XXX   0 X   -0.01 LYS 317 A\n",
      "GLU  72 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.26 GLU  71 A\n",
      "GLU  72 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 ASP  73 A\n",
      "GLU  72 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.28 ASP  76 A\n",
      "\n",
      "GLU  87 A   4.66     0 %    0.21  201   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.08 LYS  91 A\n",
      "GLU  87 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP  84 A\n",
      "\n",
      "GLU 100 A   4.92     0 %    0.40  210   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 101 A\n",
      "\n",
      "GLU 101 A   4.59     0 %    0.15  174   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS  12 A\n",
      "\n",
      "GLU 123 A   4.29     1 %    0.40  284   0.01    0   -0.17 LYS 317 A    0.00 XXX   0 X   -0.45 LYS 317 A\n",
      "\n",
      "GLU 156 A   4.19    14 %    0.82  321   0.09    0   -0.82 ARG 129 A    0.00 XXX   0 X    0.09 GLU   6 A\n",
      "GLU 156 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 HIS 130 A\n",
      "GLU 156 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.45 ARG 129 A\n",
      "\n",
      "GLU 157 A   4.41     0 %    0.30  215   0.00    0    0.00 XXX   0 X   -0.50 GLY 128 A   -0.04 ARG 129 A\n",
      "GLU 157 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.05 HIS 130 A\n",
      "GLU 157 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 GLU 156 A\n",
      "\n",
      "GLU 187 A   4.76     0 %    0.24  215   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 186 A\n",
      "\n",
      "GLU 220 A   4.46     0 %    0.22  175   0.00    0    0.00 XXX   0 X   -0.30 GLU 220 A   -0.08 LYS 331 A\n",
      "GLU 220 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.12 GLU 224 A\n",
      "\n",
      "GLU 223 A   4.77     0 %    0.51  236   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.27 LYS 251 A\n",
      "GLU 223 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 224 A\n",
      "\n",
      "GLU 224 A   3.96     0 %    0.21  180   0.00    0   -0.71 ASN 221 A    0.00 XXX   0 X   -0.04 LYS 251 A\n",
      "\n",
      "GLU 229 A   5.80    55 %    1.19  436   0.40    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.13 ARG 256 A\n",
      "GLU 229 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.09 LYS 262 A\n",
      "GLU 229 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.07 HIS 238 A\n",
      "\n",
      "GLU 232 A   4.44     0 %    0.15  188   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS 140 A\n",
      "GLU 232 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.20 ARG 144 A\n",
      "GLU 232 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.00 ARG 256 A\n",
      "GLU 232 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 146 A\n",
      "\n",
      "GLU 284 A   2.33    71 %    2.47  480   0.16    0   -0.21 ARG 276 A   -0.15 LYS 285 A   -0.19 LYS 262 A\n",
      "GLU 284 A                                           -0.11 TYR 282 A    0.00 XXX   0 X   -0.19 LYS 285 A\n",
      "GLU 284 A                                           -1.60 HIS 289 A    0.00 XXX   0 X   -1.06 ARG 276 A\n",
      "GLU 284 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.29 HIS 289 A\n",
      "\n",
      "GLU 298 A   3.85    28 %    1.05  360   0.21    0    0.00 XXX   0 X   -0.73 GLY 197 A   -0.09 LYS 213 A\n",
      "GLU 298 A                                            0.00 XXX   0 X   -0.33 SER 198 A    0.00 XXX   0 X\n",
      "GLU 298 A                                            0.00 XXX   0 X   -0.75 GLY 199 A    0.00 XXX   0 X\n",
      "\n",
      "GLU 330 A   4.59     0 %    0.16  232   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 331 A\n",
      "\n",
      "GLU 346 A   4.58     0 %    0.39  216   0.00    0   -0.10 GLN 347 A    0.00 XXX   0 X   -0.33 LYS 247 A\n",
      "GLU 346 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 LYS 349 A\n",
      "GLU 346 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.14 ASP 342 A\n",
      "\n",
      "GLU 348 A   4.77     0 %    0.30  273   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.09 LYS 349 A\n",
      "GLU 348 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 239 A\n",
      "GLU 348 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 322 A\n",
      "\n",
      "C-  352 A   3.34     0 %    0.14   58   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "HIS 130 A   6.21    28 %   -0.99  361   0.00    0    0.00 XXX   0 X    0.01 GLY 124 A    0.01 GLU  64 A\n",
      "HIS 130 A                                            0.00 XXX   0 X    0.43 ASN 127 A    0.04 GLU 156 A\n",
      "HIS 130 A                                            0.00 XXX   0 X    0.15 ARG 129 A    0.05 GLU 157 A\n",
      "\n",
      "HIS 235 A   6.45     0 %   -0.18  141   0.00    0    0.00 XXX   0 X    0.10 ALA 234 A    0.04 ASP 239 A\n",
      "\n",
      "HIS 238 A   6.25     0 %   -0.30  241   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.00 ARG 256 A\n",
      "HIS 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 GLU 229 A\n",
      "HIS 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.19 HIS 235 A\n",
      "HIS 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.17 ASP 239 A\n",
      "\n",
      "HIS 279 A   6.15     0 %   -0.30  228   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.07 HIS 235 A\n",
      "HIS 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 239 A\n",
      "\n",
      "HIS 289 A   5.35   100 %   -3.14  578   0.00    0    1.60 GLU 284 A    0.00 XXX   0 X   -0.40 LYS 262 A\n",
      "HIS 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.23 ARG 276 A\n",
      "HIS 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 LYS 285 A\n",
      "HIS 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.29 GLU 284 A\n",
      "HIS 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.75 ASP 288 A\n",
      "\n",
      "CYS  21 A  12.37   100 %    3.39  590   0.00    0   -0.13 SER  49 A   -0.12 CYS  21 A    0.23 ASP  46 A\n",
      "\n",
      "CYS 119 A  10.39    24 %    0.60  349   0.00    0    0.31 GLU  58 A    0.00 XXX   0 X   -0.02 ARG 115 A\n",
      "CYS 119 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 118 A\n",
      "CYS 119 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.46 GLU  58 A\n",
      "\n",
      "CYS 167 A  13.04   100 %    3.60  645   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.29 ARG 268 A\n",
      "CYS 167 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.73 CYS 190 A\n",
      "\n",
      "CYS 190 A  10.90    74 %    2.85  488   0.00    0   -0.22 ARG 268 A   -0.02 GLY 189 A   -0.84 ARG 268 A\n",
      "CYS 190 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 187 A\n",
      "CYS 190 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.11 ASP 270 A\n",
      "\n",
      "TYR   3 A  10.76     8 %    0.53  303   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.10 GLU   6 A\n",
      "TYR   3 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.19 ARG 129 A\n",
      "TYR   3 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.29 GLU 156 A\n",
      "TYR   3 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 157 A\n",
      "\n",
      "TYR  47 A  12.70    45 %    1.22  407   0.00    0    0.15 GLU  72 A    0.00 XXX   0 X    0.26 ASP  46 A\n",
      "TYR  47 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU  71 A\n",
      "TYR  47 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.14 ASP  73 A\n",
      "TYR  47 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.39 ASP  76 A\n",
      "TYR  47 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP 118 A\n",
      "TYR  47 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.51 GLU  72 A\n",
      "\n",
      "TYR 148 A  15.46    84 %    4.04  516   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    1.36 CYS  21 A\n",
      "TYR 148 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 CYS 167 A\n",
      "\n",
      "TYR 201 A  10.14     0 %    0.32  267   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 298 A\n",
      "TYR 201 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.20 LYS 213 A\n",
      "\n",
      "TYR 263 A  15.09*  100 %    3.51  633   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP 137 A\n",
      "TYR 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.83 ARG 276 A\n",
      "TYR 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.46 GLU 284 A\n",
      "TYR 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 ASP 288 A\n",
      "TYR 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.61 TYR 274 A\n",
      "\n",
      "TYR 274 A   9.17*  100 %    3.44  594   0.00    0   -0.85 LYS 262 A    0.00 XXX   0 X    0.22 ASP 137 A\n",
      "TYR 274 A                                           -0.20 ARG 276 A    0.00 XXX   0 X    0.08 GLU 229 A\n",
      "TYR 274 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.42 GLU 284 A\n",
      "TYR 274 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.12 ASP 288 A\n",
      "TYR 274 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -2.03 LYS 262 A\n",
      "TYR 274 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -2.03 ARG 276 A\n",
      "\n",
      "TYR 282 A  12.77    42 %    1.48  398   0.00    0    0.11 GLU 284 A    0.00 XXX   0 X    0.09 GLU  71 A\n",
      "TYR 282 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 134 A\n",
      "TYR 282 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP 137 A\n",
      "TYR 282 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.34 ARG 276 A\n",
      "TYR 282 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.07 ARG 283 A\n",
      "TYR 282 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 TYR 274 A\n",
      "TYR 282 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 TYR 318 A\n",
      "TYR 282 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.71 GLU 284 A\n",
      "TYR 282 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.53 ASP 288 A\n",
      "\n",
      "TYR 318 A   9.59     0 %    0.35  256   0.00    0   -0.13 ARG 283 A    0.00 XXX   0 X    0.06 GLU  71 A\n",
      "TYR 318 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 123 A\n",
      "TYR 318 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.26 LYS 285 A\n",
      "TYR 318 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.09 LYS 317 A\n",
      "TYR 318 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.38 ARG 283 A\n",
      "\n",
      "LYS   5 A  10.80     0 %   -0.15  167   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.24 GLU   4 A\n",
      "LYS   5 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 GLU   6 A\n",
      "LYS   5 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP   8 A\n",
      "\n",
      "LYS  11 A   9.47    50 %   -1.27  420   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.30 ASP   8 A\n",
      "LYS  11 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS  12 A\n",
      "\n",
      "LYS  12 A  10.57    27 %   -0.51  358   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.12 ASP   8 A\n",
      "LYS  12 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.40 GLU  60 A\n",
      "LYS  12 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 GLU 101 A\n",
      "\n",
      "LYS  23 A  10.16    17 %   -0.59  328   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.28 ASP  95 A\n",
      "LYS  23 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 LYS  91 A\n",
      "\n",
      "LYS  26 A  10.23     3 %   -0.27  291   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS  39 A  10.48     0 %   -0.26  206   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.24 ASP  38 A\n",
      "\n",
      "LYS  59 A  10.21     0 %   -0.32  249   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU  60 A\n",
      "\n",
      "LYS  79 A  10.44     0 %   -0.06   88   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP  76 A\n",
      "\n",
      "LYS  91 A  10.52     0 %   -0.17  208   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU  87 A\n",
      "LYS  91 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG  88 A\n",
      "LYS  91 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.12 ASP  95 A\n",
      "\n",
      "LYS 140 A   9.86    17 %   -0.56  329   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.08 ARG 144 A\n",
      "LYS 140 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 232 A\n",
      "LYS 140 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP 258 A\n",
      "LYS 140 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS  39 A\n",
      "\n",
      "LYS 159 A  10.42     0 %   -0.14  238   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP 312 A\n",
      "LYS 159 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 LYS 315 A\n",
      "\n",
      "LYS 180 A  10.43     0 %   -0.12   93   0.00    0    0.06 LYS 182 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 182 A  10.14     0 %   -0.31  138   0.00    0   -0.06 LYS 180 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 208 A  10.45     0 %   -0.05   83   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 213 A  10.19     9 %   -0.60  306   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.09 GLU 298 A\n",
      "LYS 213 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 TYR 201 A\n",
      "\n",
      "LYS 247 A  10.38     0 %   -0.49  262   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 342 A\n",
      "LYS 247 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.33 GLU 346 A\n",
      "\n",
      "LYS 251 A  10.60     0 %   -0.22  135   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.27 GLU 223 A\n",
      "LYS 251 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 GLU 224 A\n",
      "\n",
      "LYS 262 A   9.41   100 %   -2.90  586   0.00    0    0.85 TYR 274 A    0.00 XXX   0 X    0.47 ASP 137 A\n",
      "LYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 GLU 229 A\n",
      "LYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 258 A\n",
      "LYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -2.00 ARG 276 A\n",
      "LYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 GLU 284 A\n",
      "LYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 ASP 288 A\n",
      "LYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    2.03 TYR 274 A\n",
      "\n",
      "LYS 280 A  10.46     0 %   -0.04   81   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 285 A  10.26    17 %   -0.88  329   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.20 GLU  71 A\n",
      "LYS 285 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 134 A\n",
      "LYS 285 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.24 ARG 283 A\n",
      "LYS 285 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 GLU 284 A\n",
      "LYS 285 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.22 ASP 288 A\n",
      "LYS 285 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.26 TYR 318 A\n",
      "\n",
      "LYS 309 A  10.51     0 %   -0.11  165   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.12 ASP 305 A\n",
      "LYS 309 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 322 A\n",
      "LYS 309 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 ARG 341 A\n",
      "\n",
      "LYS 315 A  11.51     0 %   -0.35  220   0.00    0    0.85 ASP 312 A    0.00 XXX   0 X    0.13 ASP 320 A\n",
      "LYS 315 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 ASP 312 A\n",
      "\n",
      "LYS 317 A  10.13    35 %   -1.18  378   0.00    0    0.17 GLU 123 A    0.00 XXX   0 X    0.06 GLU  71 A\n",
      "LYS 317 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU  72 A\n",
      "LYS 317 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP  73 A\n",
      "LYS 317 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 TYR 318 A\n",
      "LYS 317 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.45 GLU 123 A\n",
      "\n",
      "LYS 331 A  10.52     0 %   -0.12  161   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU 220 A\n",
      "LYS 331 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 GLU 330 A\n",
      "\n",
      "LYS 338 A  10.23     0 %   -0.27  219   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.05 ARG 341 A\n",
      "LYS 338 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 342 A\n",
      "\n",
      "LYS 349 A  10.31     0 %   -0.30  147   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 346 A\n",
      "LYS 349 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 GLU 348 A\n",
      "\n",
      "ARG  19 A  13.45    31 %   -0.78  369   0.00    0    1.15 ASP 206 A    0.00 XXX   0 X    0.59 ASP 206 A\n",
      "\n",
      "ARG  78 A  12.45     0 %   -0.37  250   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.06 ASP  73 A\n",
      "ARG  78 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 111 A\n",
      "ARG  78 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.26 ASP 118 A\n",
      "ARG  78 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 ARG 115 A\n",
      "\n",
      "ARG  88 A  12.29    11 %   -0.46  311   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.25 ASP  84 A\n",
      "\n",
      "ARG 115 A  12.63     0 %   -0.24  199   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.14 ASP 111 A\n",
      "ARG 115 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.18 ASP 112 A\n",
      "ARG 115 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 118 A\n",
      "ARG 115 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 CYS 119 A\n",
      "\n",
      "ARG 129 A  14.57    24 %   -0.85  348   0.00    0    0.72 GLU   6 A    0.00 XXX   0 X    0.19 TYR   3 A\n",
      "ARG 129 A                                            0.82 GLU 156 A    0.00 XXX   0 X    0.32 GLU  64 A\n",
      "ARG 129 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 GLU 157 A\n",
      "ARG 129 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.37 GLU   6 A\n",
      "ARG 129 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.45 GLU 156 A\n",
      "\n",
      "ARG 144 A  13.46     3 %   -0.68  291   0.00    0    0.79 ASP 146 A    0.00 XXX   0 X    0.20 GLU 232 A\n",
      "ARG 144 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.34 ASP 258 A\n",
      "ARG 144 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.31 ASP 146 A\n",
      "\n",
      "ARG 256 A  12.33     0 %   -0.30  275   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.13 GLU 229 A\n",
      "ARG 256 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 GLU 232 A\n",
      "\n",
      "ARG 268 A  12.83    75 %   -2.41  490   0.00    0    0.22 CYS 190 A    0.00 XXX   0 X    0.29 CYS 167 A\n",
      "ARG 268 A                                            0.51 ASP 270 A    0.00 XXX   0 X    0.84 CYS 190 A\n",
      "ARG 268 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.89 ASP 270 A\n",
      "\n",
      "ARG 276 A  16.42    98 %   -1.95  557   0.00    0    0.20 TYR 274 A    0.00 XXX   0 X    0.83 TYR 263 A\n",
      "ARG 276 A                                            0.21 GLU 284 A    0.00 XXX   0 X    0.34 TYR 282 A\n",
      "ARG 276 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.65 ASP 137 A\n",
      "ARG 276 A                                            0.00 XXX   0 X    0.00 XXX   0 X    2.03 TYR 274 A\n",
      "ARG 276 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.06 GLU 284 A\n",
      "ARG 276 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.54 ASP 288 A\n",
      "\n",
      "ARG 283 A  12.77     0 %   -0.31  205   0.00    0    0.13 TYR 318 A    0.00 XXX   0 X    0.07 TYR 282 A\n",
      "ARG 283 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 TYR 318 A\n",
      "\n",
      "ARG 341 A  12.40     1 %   -0.32  283   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 305 A\n",
      "ARG 341 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP 322 A\n",
      "ARG 341 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP 342 A\n",
      "\n",
      "N+    1 A   7.94     0 %   -0.28  245   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.19 GLU   4 A\n",
      "N+    1 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP   8 A\n",
      "N+    1 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 LYS  11 A\n",
      "\n",
      "Coupled residues (marked *) were detected.Please rerun PropKa with the --display-coupled-residues \n",
      "or -d option for detailed information.\n",
      "\n",
      "INFO:pKa summary:\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "SUMMARY OF THIS PREDICTION\n",
      "       Group      pKa  model-pKa   ligand atom-type\n",
      "   ASP   8 A     3.99       3.80                      \n",
      "   ASP  32 A     3.90       3.80                      \n",
      "   ASP  38 A     3.69       3.80                      \n",
      "   ASP  46 A     7.13       3.80                      \n",
      "   ASP  73 A     3.49       3.80                      \n",
      "   ASP  76 A     4.20       3.80                      \n",
      "   ASP  80 A     3.82       3.80                      \n",
      "   ASP  84 A     3.48       3.80                      \n",
      "   ASP  95 A     3.71       3.80                      \n",
      "   ASP 111 A     3.94       3.80                      \n",
      "   ASP 112 A     2.85       3.80                      \n",
      "   ASP 118 A     3.20       3.80                      \n",
      "   ASP 134 A    13.40       3.80                      \n",
      "   ASP 137 A     2.84       3.80                      \n",
      "   ASP 146 A     3.60       3.80                      \n",
      "   ASP 186 A     3.58       3.80                      \n",
      "   ASP 206 A     3.75       3.80                      \n",
      "   ASP 239 A     3.65       3.80                      \n",
      "   ASP 258 A     6.14       3.80                      \n",
      "   ASP 270 A     5.41       3.80                      \n",
      "   ASP 288 A     7.25       3.80                      \n",
      "   ASP 305 A     4.25       3.80                      \n",
      "   ASP 312 A     2.16       3.80                      \n",
      "   ASP 320 A     3.88       3.80                      \n",
      "   ASP 322 A     3.54       3.80                      \n",
      "   ASP 342 A     4.00       3.80                      \n",
      "   GLU   4 A     4.37       4.50                      \n",
      "   GLU   6 A     3.70       4.50                      \n",
      "   GLU  58 A     4.99       4.50                      \n",
      "   GLU  60 A     4.70       4.50                      \n",
      "   GLU  64 A     4.73       4.50                      \n",
      "   GLU  71 A     5.17       4.50                      \n",
      "   GLU  72 A     5.72       4.50                      \n",
      "   GLU  87 A     4.66       4.50                      \n",
      "   GLU 100 A     4.92       4.50                      \n",
      "   GLU 101 A     4.59       4.50                      \n",
      "   GLU 123 A     4.29       4.50                      \n",
      "   GLU 156 A     4.19       4.50                      \n",
      "   GLU 157 A     4.41       4.50                      \n",
      "   GLU 187 A     4.76       4.50                      \n",
      "   GLU 220 A     4.46       4.50                      \n",
      "   GLU 223 A     4.77       4.50                      \n",
      "   GLU 224 A     3.96       4.50                      \n",
      "   GLU 229 A     5.80       4.50                      \n",
      "   GLU 232 A     4.44       4.50                      \n",
      "   GLU 284 A     2.33       4.50                      \n",
      "   GLU 298 A     3.85       4.50                      \n",
      "   GLU 330 A     4.59       4.50                      \n",
      "   GLU 346 A     4.58       4.50                      \n",
      "   GLU 348 A     4.77       4.50                      \n",
      "   C-  352 A     3.34       3.20                      \n",
      "   HIS 130 A     6.21       6.50                      \n",
      "   HIS 235 A     6.45       6.50                      \n",
      "   HIS 238 A     6.25       6.50                      \n",
      "   HIS 279 A     6.15       6.50                      \n",
      "   HIS 289 A     5.35       6.50                      \n",
      "   CYS  21 A    12.37       9.00                      \n",
      "   CYS 119 A    10.39       9.00                      \n",
      "   CYS 167 A    13.04       9.00                      \n",
      "   CYS 190 A    10.90       9.00                      \n",
      "   TYR   3 A    10.76      10.00                      \n",
      "   TYR  47 A    12.70      10.00                      \n",
      "   TYR 148 A    15.46      10.00                      \n",
      "   TYR 201 A    10.14      10.00                      \n",
      "   TYR 263 A    15.09      10.00                      \n",
      "   TYR 274 A     9.17      10.00                      \n",
      "   TYR 282 A    12.77      10.00                      \n",
      "   TYR 318 A     9.59      10.00                      \n",
      "   LYS   5 A    10.80      10.50                      \n",
      "   LYS  11 A     9.47      10.50                      \n",
      "   LYS  12 A    10.57      10.50                      \n",
      "   LYS  23 A    10.16      10.50                      \n",
      "   LYS  26 A    10.23      10.50                      \n",
      "   LYS  39 A    10.48      10.50                      \n",
      "   LYS  59 A    10.21      10.50                      \n",
      "   LYS  79 A    10.44      10.50                      \n",
      "   LYS  91 A    10.52      10.50                      \n",
      "   LYS 140 A     9.86      10.50                      \n",
      "   LYS 159 A    10.42      10.50                      \n",
      "   LYS 180 A    10.43      10.50                      \n",
      "   LYS 182 A    10.14      10.50                      \n",
      "   LYS 208 A    10.45      10.50                      \n",
      "   LYS 213 A    10.19      10.50                      \n",
      "   LYS 247 A    10.38      10.50                      \n",
      "   LYS 251 A    10.60      10.50                      \n",
      "   LYS 262 A     9.41      10.50                      \n",
      "   LYS 280 A    10.46      10.50                      \n",
      "   LYS 285 A    10.26      10.50                      \n",
      "   LYS 309 A    10.51      10.50                      \n",
      "   LYS 315 A    11.51      10.50                      \n",
      "   LYS 317 A    10.13      10.50                      \n",
      "   LYS 331 A    10.52      10.50                      \n",
      "   LYS 338 A    10.23      10.50                      \n",
      "   LYS 349 A    10.31      10.50                      \n",
      "   ARG  19 A    13.45      12.50                      \n",
      "   ARG  78 A    12.45      12.50                      \n",
      "   ARG  88 A    12.29      12.50                      \n",
      "   ARG 115 A    12.63      12.50                      \n",
      "   ARG 129 A    14.57      12.50                      \n",
      "   ARG 144 A    13.46      12.50                      \n",
      "   ARG 256 A    12.33      12.50                      \n",
      "   ARG 268 A    12.83      12.50                      \n",
      "   ARG 276 A    16.42      12.50                      \n",
      "   ARG 283 A    12.77      12.50                      \n",
      "   ARG 341 A    12.40      12.50                      \n",
      "   N+    1 A     7.94       8.00                      \n",
      "\n",
      "INFO:PROPKA information:\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "Free energy of   folding (kcal/mol) as a function of pH (using neutral reference)\n",
      "  0.00     29.52\n",
      "  1.00     29.44\n",
      "  2.00     28.85\n",
      "  3.00     26.51\n",
      "  4.00     24.84\n",
      "  5.00     30.16\n",
      "  6.00     36.69\n",
      "  7.00     40.13\n",
      "  8.00     42.33\n",
      "  9.00     44.66\n",
      " 10.00     48.46\n",
      " 11.00     55.32\n",
      " 12.00     64.91\n",
      " 13.00     74.71\n",
      " 14.00     82.36\n",
      "\n",
      "The pH of optimum stability is  3.8 for which the free energy is   24.7 kcal/mol at 298K\n",
      "Could not determine pH values where the free energy is within 80 % of minimum\n",
      "Could not determine the pH-range where the free energy is negative\n",
      "\n",
      "\n",
      "Protein charge of folded and unfolded state as a function of pH\n",
      "    pH  unfolded  folded\n",
      "  0.00     42.99   42.98\n",
      "  0.10     42.99   42.98\n",
      "  0.20     42.99   42.97\n",
      "  0.30     42.99   42.96\n",
      "  0.40     42.99   42.95\n",
      "  0.50     42.98   42.94\n",
      "  0.60     42.98   42.92\n",
      "  0.70     42.97   42.90\n",
      "  0.80     42.97   42.88\n",
      "  0.90     42.96   42.85\n",
      "  1.00     42.94   42.81\n",
      "  1.10     42.93   42.76\n",
      "  1.20     42.91   42.71\n",
      "  1.30     42.89   42.64\n",
      "  1.40     42.86   42.55\n",
      "  1.50     42.83   42.44\n",
      "  1.60     42.78   42.32\n",
      "  1.70     42.73   42.16\n",
      "  1.80     42.66   41.98\n",
      "  1.90     42.57   41.77\n",
      "  2.00     42.46   41.52\n",
      "  2.10     42.32   41.23\n",
      "  2.20     42.15   40.90\n",
      "  2.30     41.94   40.52\n",
      "  2.40     41.68   40.08\n",
      "  2.50     41.35   39.59\n",
      "  2.60     40.96   39.03\n",
      "  2.70     40.47   38.40\n",
      "  2.80     39.88   37.69\n",
      "  2.90     39.17   36.89\n",
      "  3.00     38.32   36.00\n",
      "  3.10     37.31   35.00\n",
      "  3.20     36.13   33.89\n",
      "  3.30     34.77   32.68\n",
      "  3.40     33.21   31.35\n",
      "  3.50     31.47   29.91\n",
      "  3.60     29.54   28.38\n",
      "  3.70     27.44   26.75\n",
      "  3.80     25.20   25.06\n",
      "  3.90     22.85   23.30\n",
      "  4.00     20.41   21.51\n",
      "  4.10     17.94   19.69\n",
      "  4.20     15.46   17.87\n",
      "  4.30     13.00   16.06\n",
      "  4.40     10.61   14.27\n",
      "  4.50      8.32   12.53\n",
      "  4.60      6.16   10.84\n",
      "  4.70      4.14    9.21\n",
      "  4.80      2.30    7.66\n",
      "  4.90      0.64    6.18\n",
      "  5.00     -0.83    4.80\n",
      "  5.10     -2.12    3.51\n",
      "  5.20     -3.24    2.30\n",
      "  5.30     -4.21    1.18\n",
      "  5.40     -5.04    0.13\n",
      "  5.50     -5.76   -0.84\n",
      "  5.60     -6.39   -1.75\n",
      "  5.70     -6.94   -2.60\n",
      "  5.80     -7.44   -3.40\n",
      "  5.90     -7.89   -4.15\n",
      "  6.00     -8.32   -4.86\n",
      "  6.10     -8.72   -5.53\n",
      "  6.20     -9.12   -6.16\n",
      "  6.30     -9.51   -6.75\n",
      "  6.40     -9.89   -7.31\n",
      "  6.50    -10.26   -7.82\n",
      "  6.60    -10.62   -8.31\n",
      "  6.70    -10.96   -8.75\n",
      "  6.80    -11.28   -9.16\n",
      "  6.90    -11.58   -9.54\n",
      "  7.00    -11.85   -9.90\n",
      "  7.10    -12.10  -10.23\n",
      "  7.20    -12.34  -10.54\n",
      "  7.30    -12.55  -10.82\n",
      "  7.40    -12.74  -11.09\n",
      "  7.50    -12.93  -11.35\n",
      "  7.60    -13.11  -11.58\n",
      "  7.70    -13.29  -11.81\n",
      "  7.80    -13.47  -12.02\n",
      "  7.90    -13.66  -12.23\n",
      "  8.00    -13.86  -12.44\n",
      "  8.10    -14.08  -12.64\n",
      "  8.20    -14.31  -12.84\n",
      "  8.30    -14.57  -13.05\n",
      "  8.40    -14.85  -13.27\n",
      "  8.50    -15.17  -13.51\n",
      "  8.60    -15.53  -13.77\n",
      "  8.70    -15.93  -14.06\n",
      "  8.80    -16.37  -14.40\n",
      "  8.90    -16.87  -14.78\n",
      "  9.00    -17.42  -15.22\n",
      "  9.10    -18.04  -15.72\n",
      "  9.20    -18.72  -16.30\n",
      "  9.30    -19.49  -16.96\n",
      "  9.40    -20.34  -17.71\n",
      "  9.50    -21.30  -18.55\n",
      "  9.60    -22.37  -19.50\n",
      "  9.70    -23.56  -20.55\n",
      "  9.80    -24.88  -21.72\n",
      "  9.90    -26.33  -22.98\n",
      " 10.00    -27.91  -24.34\n",
      " 10.10    -29.60  -25.79\n",
      " 10.20    -31.40  -27.31\n",
      " 10.30    -33.26  -28.87\n",
      " 10.40    -35.16  -30.45\n",
      " 10.50    -37.06  -32.01\n",
      " 10.60    -38.92  -33.54\n",
      " 10.70    -40.70  -34.99\n",
      " 10.80    -42.38  -36.37\n",
      " 10.90    -43.92  -37.64\n",
      " 11.00    -45.32  -38.81\n",
      " 11.10    -46.58  -39.88\n",
      " 11.20    -47.70  -40.84\n",
      " 11.30    -48.69  -41.72\n",
      " 11.40    -49.58  -42.51\n",
      " 11.50    -50.38  -43.25\n",
      " 11.60    -51.11  -43.93\n",
      " 11.70    -51.80  -44.59\n",
      " 11.80    -52.46  -45.22\n",
      " 11.90    -53.11  -45.85\n",
      " 12.00    -53.76  -46.49\n",
      " 12.10    -54.43  -47.14\n",
      " 12.20    -55.11  -47.80\n",
      " 12.30    -55.81  -48.49\n",
      " 12.40    -56.51  -49.20\n",
      " 12.50    -57.22  -49.92\n",
      " 12.60    -57.90  -50.65\n",
      " 12.70    -58.56  -51.38\n",
      " 12.80    -59.18  -52.10\n",
      " 12.90    -59.75  -52.81\n",
      " 13.00    -60.27  -53.49\n",
      " 13.10    -60.72  -54.14\n",
      " 13.20    -61.11  -54.75\n",
      " 13.30    -61.45  -55.32\n",
      " 13.40    -61.73  -55.85\n",
      " 13.50    -61.97  -56.34\n",
      " 13.60    -62.17  -56.79\n",
      " 13.70    -62.33  -57.20\n",
      " 13.80    -62.46  -57.56\n",
      " 13.90    -62.57  -57.89\n",
      " 14.00    -62.65  -58.19\n",
      "The pI is  5.41 (folded) and  4.94 (unfolded)\n",
      "\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "INFO:Applying pKa values at a pH of 7.60:\n",
      "INFO:Adding hydrogens to biomolecule.\n",
      "INFO:Debumping biomolecule (again).\n",
      "WARNING:WARNING: Unable to debump LEU A 29\n",
      "INFO:Optimizing hydrogen bonds\n",
      "INFO:Applying force field to biomolecule states.\n",
      "INFO:Regenerating headers.\n",
      "INFO:Regenerating PDB lines.\n",
      "WARNING:Ignoring 467 header lines in output.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "pH: 7.6\tname: AtSAL1_updated\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:PDB2PQR v3.6.2: biomolecular structure conversion software.\n",
      "INFO:Please cite:  Jurrus E, et al.  Improvements to the APBS biomolecular solvation software suite.  Protein Sci 27 112-128 (2018).\n",
      "INFO:Please cite:  Dolinsky TJ, et al.  PDB2PQR: expanding and upgrading automated preparation of biomolecular structures for molecular simulations. Nucleic Acids Res 35 W522-W525 (2007).\n",
      "INFO:Checking and transforming input arguments.\n",
      "INFO:Loading topology files.\n",
      "INFO:Loading molecule: /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/AtAHL.pdb\n",
      "INFO:Setting up molecule.\n",
      "INFO:Created biomolecule object with 368 residues and 5474 atoms.\n",
      "INFO:Setting termini states for biomolecule chains.\n",
      "INFO:Loading forcefield.\n",
      "INFO:Loading hydrogen topology definitions.\n",
      "INFO:This biomolecule is clean.  No repair needed.\n",
      "INFO:Updating disulfide bridges.\n",
      "INFO:Debumping biomolecule.\n",
      "INFO:Assigning titration states with PROPKA.\n",
      "INFO:\n",
      "propka3.5.1                                                                                  2024-07-26\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "--                                                                           --\n",
      "--  PROPKA: A PROTEIN PKA PREDICTOR                                          --\n",
      "--                                                                           --\n",
      "--  VERSION 1.0,  04/25/2004,  IOWA CITY                                     --\n",
      "--  BY HUI LI                                                                --\n",
      "--                                                                           --\n",
      "--  VERSION 2.0,  11/05/2007, IOWA CITY/COPENHAGEN                           --\n",
      "--  BY DELPHINE C. BAS AND DAVID M. ROGERS                                   --\n",
      "--                                                                           --\n",
      "--  VERSION 3.0,  01/06/2011, COPENHAGEN                                     --\n",
      "--  BY MATS H.M. OLSSON AND CHRESTEN R. SONDERGARD                           --\n",
      "--                                                                           --\n",
      "--  VERSION 3.1,  07/01/2011, COPENHAGEN                                     --\n",
      "--  BY CHRESTEN R. SONDERGARD AND MATS H.M. OLSSON                           --\n",
      "--                                                                           --\n",
      "--  VERSION 3.2,  06/17/2020, PLANET EARTH                                   --\n",
      "--  SEE ABOVE FOR AUTHORS                                                    --\n",
      "--                                                                           --\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "References:\n",
      "\n",
      "Very Fast Empirical Prediction and Rationalization of Protein pKa Values.\n",
      "Hui Li, Andrew D. Robertson and Jan H. Jensen. PROTEINS: Structure, Function,\n",
      "and Bioinformatics. 61:704-721 (2005)\n",
      "\n",
      "Very Fast Prediction and Rationalization of pKa Values for Protein-Ligand\n",
      "Complexes.  Delphine C. Bas, David M. Rogers and Jan H. Jensen.  PROTEINS:\n",
      "Structure, Function, and Bioinformatics 73:765-783 (2008)\n",
      "\n",
      "PROPKA3: Consistent Treatment of Internal and Surface Residues in Empirical\n",
      "pKa predictions.  Mats H.M. Olsson, Chresten R. Sondergard, Michal Rostkowski,\n",
      "and Jan H. Jensen.  Journal of Chemical Theory and Computation, 7(2):525-537\n",
      "(2011)\n",
      "\n",
      "Improved Treatment of Ligands and Coupling Effects in Empirical Calculation\n",
      "and Rationalization of pKa Values.  Chresten R. Sondergaard, Mats H.M. Olsson,\n",
      "Michal Rostkowski, and Jan H. Jensen.  Journal of Chemical Theory and\n",
      "Computation, (2011)\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "\n",
      "INFO:Coupling map:\n",
      "\n",
      "INFO:Calculating pKas for Conformation container 1A with 3218 atoms and 894 groups\n",
      "INFO:did not converge in 10 iterations\n",
      "INFO:pKa determinants:\n",
      "\n",
      "---------  -----   ------   ---------------------    --------------    --------------    --------------\n",
      "                            DESOLVATION  EFFECTS       SIDECHAIN          BACKBONE        COULOMBIC\n",
      " RESIDUE    pKa    BURIED     REGULAR      RE        HYDROGEN BOND     HYDROGEN BOND      INTERACTION\n",
      "---------  -----   ------   ---------   ---------    --------------    --------------    --------------\n",
      "\n",
      "ASP   6 A   3.00    28 %    0.95  359   0.08    0   -1.34 ARG  10 A    0.00 XXX   0 X   -0.50 ARG  10 A\n",
      "\n",
      "ASP  38 A   4.10     0 %    0.13  168   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.04 LYS  35 A\n",
      "ASP  38 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 ASP  39 A\n",
      "ASP  38 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP  40 A\n",
      "ASP  38 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP  74 A\n",
      "ASP  38 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 GLU  80 A\n",
      "\n",
      "ASP  39 A   2.55    20 %    0.70  337   0.00    0   -0.85 SER  41 A   -0.18 ASP  39 A   -0.26 LYS  37 A\n",
      "ASP  39 A                                            0.00 XXX   0 X   -0.61 SER  41 A   -0.27 LYS 302 A\n",
      "ASP  39 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.23 GLU  72 A\n",
      "\n",
      "ASP  40 A   4.01     0 %    0.14  138   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP  39 A\n",
      "\n",
      "ASP  47 A   5.77*  100 %    3.69  595   1.08    0   -0.85 THR  44 A   -0.64 THR  44 A    0.41 GLU  72 A\n",
      "ASP  47 A                                           -0.59 THR 144 A   -0.44 ASP  47 A   -0.69 LYS  37 A\n",
      "\n",
      "ASP  64 A   2.40     0 %    0.38  280   0.00    0   -0.30 LYS 120 A    0.00 XXX   0 X   -0.34 LYS 120 A\n",
      "ASP  64 A                                           -0.68 ARG 124 A    0.00 XXX   0 X   -0.46 ARG 124 A\n",
      "\n",
      "ASP  74 A   3.59    27 %    0.81  358   0.01    0    0.00 XXX   0 X   -0.53 GLU  76 A   -0.02 LYS  35 A\n",
      "ASP  74 A                                            0.00 XXX   0 X   -0.39 THR  77 A   -0.11 LYS  37 A\n",
      "ASP  74 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU  80 A\n",
      "\n",
      "ASP  82 A   3.93     0 %    0.12  119   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.07 LYS  35 A\n",
      "ASP  82 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 GLU  80 A\n",
      "\n",
      "ASP 139 A  11.53*  100 %    3.50  634   0.39    0    0.49 ASP 305 A   -0.33 VAL 141 A   -0.41 LYS  37 A\n",
      "ASP 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.21 LYS 279 A\n",
      "ASP 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.50 LYS 302 A\n",
      "ASP 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.40 ASP  47 A\n",
      "ASP 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.37 GLU  73 A\n",
      "ASP 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.29 GLU  72 A\n",
      "ASP 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.51 ASP 142 A\n",
      "ASP 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.21 ASP 305 A\n",
      "\n",
      "ASP 142 A   5.13*  100 %    3.07  665   0.33    0   -0.07 SER 276 A   -0.04 GLY 143 A   -0.34 LYS 293 A\n",
      "ASP 142 A                                            0.00 XXX   0 X   -0.28 MET 277 A   -0.13 LYS 302 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.33 GLU  72 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 249 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.12 GLU 301 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.69 LYS 279 A\n",
      "\n",
      "ASP 151 A   4.73    32 %    0.77  370   0.33    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.15 ARG 149 A\n",
      "ASP 151 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 HIS  27 A\n",
      "\n",
      "ASP 237 A   3.65     0 %    0.27  219   0.00    0    0.00 XXX   0 X   -0.21 ASP 237 A   -0.31 LYS 354 A\n",
      "ASP 237 A                                            0.00 XXX   0 X   -0.00 ASP 238 A    0.10 ASP 238 A\n",
      "\n",
      "ASP 238 A   3.28     0 %    0.44  227   0.00    0    0.00 XXX   0 X   -0.76 VAL 240 A   -0.20 LYS 354 A\n",
      "\n",
      "ASP 287 A   3.73    64 %    2.21  461   0.60    0   -0.54 TYR 175 A    0.00 XXX   0 X   -0.47 LYS 179 A\n",
      "ASP 287 A                                           -0.37 LYS 179 A    0.00 XXX   0 X   -0.85 ARG 285 A\n",
      "ASP 287 A                                           -0.66 ARG 285 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "ASP 305 A   6.18*  100 %    2.94  592   0.45    0   -0.49 ASP 139 A   -0.22 LYS 302 A   -0.07 LYS  37 A\n",
      "ASP 305 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.99 GLU  72 A\n",
      "ASP 305 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.26 ASP 142 A\n",
      "ASP 305 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.66 LYS 279 A\n",
      "ASP 305 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.74 LYS 293 A\n",
      "ASP 305 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.72 GLU 301 A\n",
      "ASP 305 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.89 LYS 302 A\n",
      "ASP 305 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.91 HIS 306 A\n",
      "\n",
      "ASP 322 A   3.09    35 %    1.70  379   0.53    0   -0.75 ARG 326 A   -0.40 GLY 324 A   -0.10 ARG 341 A\n",
      "ASP 322 A                                            0.00 XXX   0 X   -0.03 GLY 325 A    0.28 ASP 340 A\n",
      "ASP 322 A                                            0.00 XXX   0 X   -0.78 ARG 326 A   -0.03 HIS 352 A\n",
      "ASP 322 A                                            0.00 XXX   0 X   -0.72 ASP 340 A   -0.42 ARG 326 A\n",
      "\n",
      "ASP 329 A   2.58    14 %    0.73  320   0.01    0   -0.85 SER 331 A   -0.78 SER 331 A   -0.01 ARG 326 A\n",
      "ASP 329 A                                           -0.01 LYS 332 A    0.00 XXX   0 X   -0.32 LYS 332 A\n",
      "\n",
      "ASP 340 A   2.65    15 %    0.87  323   0.00    0   -1.39 ARG 341 A    0.00 XXX   0 X   -0.19 ARG 326 A\n",
      "ASP 340 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.44 ARG 341 A\n",
      "\n",
      "GLU   2 A   3.46     0 %    0.26  241   0.00    0   -0.85 LYS 208 A    0.00 XXX   0 X   -0.14 N+    1 A\n",
      "GLU   2 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 GLU   4 A\n",
      "GLU   2 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP   6 A\n",
      "GLU   2 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.38 LYS 208 A\n",
      "\n",
      "GLU   4 A   2.86    16 %    0.81  327   0.00    0   -0.85 N+    1 A   -0.68 GLU   4 A   -0.01 LYS 208 A\n",
      "GLU   4 A                                           -0.25 ARG 134 A    0.00 XXX   0 X   -0.41 N+    1 A\n",
      "GLU   4 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.26 ARG 134 A\n",
      "\n",
      "GLU  24 A   4.74     4 %    0.29  293   0.02    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS  21 A\n",
      "GLU  24 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS  25 A\n",
      "GLU  24 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 HIS  27 A\n",
      "GLU  24 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU  96 A\n",
      "\n",
      "GLU  60 A   5.00    13 %    0.50  319   0.03    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.02 LYS 120 A\n",
      "GLU  60 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.07 ARG 124 A\n",
      "GLU  60 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP  64 A\n",
      "GLU  60 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 117 A\n",
      "\n",
      "GLU  72 A   1.96*   90 %    2.13  534   0.00    0   -0.71 TRP 304 A    0.00 XXX   0 X   -1.16 LYS  37 A\n",
      "GLU  72 A                                           -0.85 LYS  37 A    0.00 XXX   0 X   -1.10 LYS 302 A\n",
      "GLU  72 A                                           -0.84 LYS 302 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "GLU  73 A   7.93*   92 %    3.22  539   0.02    0   -0.85 LYS  37 A    0.00 XXX   0 X   -0.23 LYS 302 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP  39 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 ASP  74 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.18 LYS  37 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.24 ASP  47 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.93 GLU  72 A\n",
      "\n",
      "GLU  76 A   4.99     0 %    0.13  203   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP  38 A\n",
      "GLU  76 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.27 ASP  74 A\n",
      "GLU  76 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU  80 A\n",
      "\n",
      "GLU  80 A   3.53     0 %    0.26  214   0.00    0   -0.85 LYS  35 A    0.00 XXX   0 X   -0.38 LYS  35 A\n",
      "\n",
      "GLU  96 A   3.28    17 %    0.51  330   0.04    0   -0.20 LYS  21 A    0.00 XXX   0 X   -0.37 LYS  21 A\n",
      "GLU  96 A                                           -0.76 LYS  25 A    0.00 XXX   0 X   -0.44 LYS  25 A\n",
      "\n",
      "GLU 100 A   4.18    21 %    0.73  339   0.12    0   -0.85 LYS  21 A    0.00 XXX   0 X   -0.01 LYS  25 A\n",
      "GLU 100 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 LYS 108 A\n",
      "GLU 100 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.17 GLU  96 A\n",
      "GLU 100 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.47 LYS  21 A\n",
      "\n",
      "GLU 117 A   3.73     0 %    0.34  255   0.00    0    0.00 XXX   0 X   -0.81 GLY 114 A   -0.26 LYS 111 A\n",
      "GLU 117 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS 120 A\n",
      "\n",
      "GLU 161 A   3.76     0 %    0.38  263   0.00    0   -0.74 ARG 134 A    0.00 XXX   0 X   -0.04 N+    1 A\n",
      "GLU 161 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 LYS 164 A\n",
      "GLU 161 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU   4 A\n",
      "GLU 161 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.38 ARG 134 A\n",
      "\n",
      "GLU 180 A   4.49     0 %    0.11  156   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.12 LYS 178 A\n",
      "\n",
      "GLU 226 A   4.56     0 %    0.06  112   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "GLU 246 A   5.27    37 %    1.07  386   0.02    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.20 ARG 273 A\n",
      "GLU 246 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.12 HIS 255 A\n",
      "\n",
      "GLU 249 A   5.27    46 %    0.93  410   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.10 ARG 149 A\n",
      "GLU 249 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 293 A\n",
      "GLU 249 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP  39 A\n",
      "GLU 249 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 HIS 255 A\n",
      "\n",
      "GLU 289 A   3.91    55 %    1.68  435   0.54    0   -0.85 SER 234 A   -0.79 THR 243 A   -0.18 LYS 179 A\n",
      "GLU 289 A                                           -0.62 THR 243 A   -0.00 GLU 289 A   -0.45 LYS 269 A\n",
      "GLU 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP 287 A\n",
      "\n",
      "GLU 301 A   1.69    92 %    3.27  539   0.00    0   -0.85 LYS 293 A    0.00 XXX   0 X   -0.03 LYS 302 A\n",
      "GLU 301 A                                           -1.60 HIS 306 A    0.00 XXX   0 X   -0.52 LYS 279 A\n",
      "GLU 301 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.62 LYS 293 A\n",
      "GLU 301 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.45 HIS 306 A\n",
      "\n",
      "GLU 314 A   5.47    70 %    1.78  477   0.58    0   -0.74 ARG 209 A    0.00 XXX   0 X    0.16 GLU 315 A\n",
      "GLU 314 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.81 ARG 209 A\n",
      "\n",
      "GLU 315 A   4.02    32 %    1.27  372   0.32    0   -0.15 SER 211 A   -0.81 GLY 210 A   -0.02 LYS 208 A\n",
      "GLU 315 A                                            0.00 XXX   0 X   -0.11 SER 211 A   -0.28 ARG 209 A\n",
      "GLU 315 A                                            0.00 XXX   0 X   -0.70 GLY 212 A    0.00 XXX   0 X\n",
      "\n",
      "GLU 337 A   3.62     0 %    0.36  250   0.00    0   -0.85 LYS 300 A    0.00 XXX   0 X   -0.01 LYS 332 A\n",
      "GLU 337 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.38 LYS 300 A\n",
      "\n",
      "GLU 353 A   3.83     0 %    0.26  212   0.00    0   -0.84 GLN 349 A    0.00 XXX   0 X   -0.01 LYS 354 A\n",
      "GLU 353 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 HIS 352 A\n",
      "\n",
      "GLU 364 A   4.60     0 %    0.12  133   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.01 ARG 341 A\n",
      "\n",
      "C-  368 A   3.32     0 %    0.12   79   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "HIS  13 A   4.57    81 %   -2.69  507   0.00    0    0.82 GLN 217 A    0.00 XXX   0 X   -0.06 ARG  10 A\n",
      "\n",
      "HIS  27 A   6.82     6 %   -0.41  298   0.00    0    0.00 XXX   0 X    0.68 GLN  23 A    0.01 GLU  24 A\n",
      "HIS  27 A                                            0.00 XXX   0 X    0.01 ARG 149 A    0.02 ASP 151 A\n",
      "\n",
      "HIS  33 A   6.06    11 %   -0.44  312   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "HIS 135 A   5.85    35 %   -1.10  379   0.00    0    0.00 XXX   0 X    0.45 LYS 132 A    0.00 XXX   0 X\n",
      "\n",
      "HIS 255 A   6.41    20 %   -0.86  336   0.00    0    0.00 XXX   0 X    0.78 GLU 249 A   -0.09 ARG 250 A\n",
      "HIS 255 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 273 A\n",
      "HIS 255 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 LYS 293 A\n",
      "HIS 255 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.12 GLU 246 A\n",
      "HIS 255 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 249 A\n",
      "\n",
      "HIS 306 A   5.07   100 %   -3.28  641   0.00    0    1.60 GLU 301 A    0.00 XXX   0 X   -1.25 LYS 279 A\n",
      "HIS 306 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.85 LYS 293 A\n",
      "HIS 306 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.45 GLU 301 A\n",
      "HIS 306 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.91 ASP 305 A\n",
      "\n",
      "HIS 352 A   4.72    65 %   -1.90  462   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 322 A\n",
      "HIS 352 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU 353 A\n",
      "\n",
      "CYS  19 A  13.08   100 %    3.78  655   0.00    0    0.00 XXX   0 X   -0.08 CYS  19 A    0.39 ASP  47 A\n",
      "\n",
      "CYS 125 A  11.63    96 %    3.54  549   0.00    0   -0.05 GLN  51 A    0.00 XXX   0 X   -0.23 ARG 124 A\n",
      "CYS 125 A                                           -0.63 SER  55 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "CYS 172 A  13.72   100 %    3.47  673   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.18 ARG 285 A\n",
      "CYS 172 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 ASP 142 A\n",
      "CYS 172 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.35 CYS 203 A\n",
      "\n",
      "CYS 181 A   9.45     5 %    0.55  295   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 178 A\n",
      "CYS 181 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.10 LYS 201 A\n",
      "CYS 181 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 ARG 285 A\n",
      "CYS 181 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.11 GLU 180 A\n",
      "CYS 181 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 287 A\n",
      "\n",
      "CYS 186 A   9.82    31 %    0.88  368   0.00    0    0.00 XXX   0 X   -0.08 CYS 186 A   -0.15 ARG 273 A\n",
      "CYS 186 A                                            0.00 XXX   0 X   -0.02 ASN 187 A    0.19 ASP 151 A\n",
      "\n",
      "CYS 203 A  11.78   100 %    3.38  575   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.65 ARG 285 A\n",
      "CYS 203 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 ASP 287 A\n",
      "\n",
      "CYS 245 A  12.50   100 %    3.40  607   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.11 LYS 279 A\n",
      "CYS 245 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 246 A\n",
      "CYS 245 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 287 A\n",
      "CYS 245 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.12 GLU 289 A\n",
      "\n",
      "CYS 346 A  12.74   100 %    3.74  577   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "TYR 104 A  10.78    56 %    2.22  438   0.00    0   -0.85 SER  17 A   -0.68 GLY 222 A    0.31 GLU 100 A\n",
      "TYR 104 A                                            0.00 XXX   0 X   -0.01 GLY 223 A   -0.21 LYS  21 A\n",
      "\n",
      "TYR 153 A  15.70   100 %    4.01  560   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    1.59 CYS  19 A\n",
      "TYR 153 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU  24 A\n",
      "TYR 153 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 CYS 172 A\n",
      "\n",
      "TYR 175 A  15.46   100 %    3.86  560   0.00    0   -0.21 ARG 285 A    0.00 XXX   0 X    0.40 CYS 172 A\n",
      "TYR 175 A                                            0.54 ASP 287 A    0.00 XXX   0 X    0.54 CYS 203 A\n",
      "TYR 175 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.43 CYS 245 A\n",
      "TYR 175 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.24 ARG 285 A\n",
      "TYR 175 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.15 ASP 287 A\n",
      "\n",
      "TYR 206 A  11.50    48 %    1.46  417   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU   2 A\n",
      "TYR 206 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP   6 A\n",
      "TYR 206 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.00 ARG  10 A\n",
      "\n",
      "TYR 275 A  12.11    30 %    1.75  364   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.17 ARG 149 A\n",
      "TYR 275 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 ASP 151 A\n",
      "TYR 275 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 CYS 186 A\n",
      "TYR 275 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.49 GLU 249 A\n",
      "TYR 275 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 ARG 250 A\n",
      "TYR 275 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 ARG 273 A\n",
      "\n",
      "TYR 280 A  14.55   100 %    3.81  663   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.14 ASP 142 A\n",
      "TYR 280 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.40 GLU 301 A\n",
      "TYR 280 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.12 ASP 305 A\n",
      "TYR 280 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU 314 A\n",
      "\n",
      "TYR 299 A  14.29    74 %    1.49  488   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP  39 A\n",
      "TYR 299 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.23 GLU  72 A\n",
      "TYR 299 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP 139 A\n",
      "TYR 299 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.32 ASP 142 A\n",
      "TYR 299 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU 249 A\n",
      "TYR 299 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 TYR 335 A\n",
      "TYR 299 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.93 GLU 301 A\n",
      "TYR 299 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.05 ASP 305 A\n",
      "\n",
      "TYR 335 A  10.09     6 %    0.28  299   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP  39 A\n",
      "TYR 335 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU  72 A\n",
      "TYR 335 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP 305 A\n",
      "TYR 335 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 337 A\n",
      "TYR 335 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.10 LYS 300 A\n",
      "TYR 335 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.18 LYS 302 A\n",
      "\n",
      "TYR 360 A  10.04     0 %    0.35  261   0.00    0   -0.33 ARG 341 A    0.00 XXX   0 X    0.11 ASP 322 A\n",
      "TYR 360 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.07 ARG 326 A\n",
      "TYR 360 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.28 ASP 340 A\n",
      "TYR 360 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.30 ARG 341 A\n",
      "\n",
      "LYS  21 A  11.63    24 %   -1.00  348   0.00    0    0.20 GLU  96 A    0.00 XXX   0 X    0.03 GLU  24 A\n",
      "LYS  21 A                                            0.85 GLU 100 A    0.00 XXX   0 X    0.21 TYR 104 A\n",
      "LYS  21 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.37 GLU  96 A\n",
      "LYS  21 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.47 GLU 100 A\n",
      "\n",
      "LYS  25 A  11.09    17 %   -0.54  328   0.00    0    0.76 GLU  96 A    0.00 XXX   0 X    0.03 GLU  24 A\n",
      "LYS  25 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 100 A\n",
      "LYS  25 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.11 LYS  21 A\n",
      "LYS  25 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.44 GLU  96 A\n",
      "\n",
      "LYS  35 A  11.28     0 %   -0.58  275   0.00    0    0.85 GLU  80 A    0.00 XXX   0 X    0.04 ASP  38 A\n",
      "LYS  35 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP  74 A\n",
      "LYS  35 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 ASP  82 A\n",
      "LYS  35 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 GLU  80 A\n",
      "\n",
      "LYS  37 A  13.37    74 %   -2.71  488   0.00    0    0.85 GLU  72 A    0.00 XXX   0 X    0.26 ASP  39 A\n",
      "LYS  37 A                                            0.85 GLU  73 A    0.00 XXX   0 X    0.11 ASP  74 A\n",
      "LYS  37 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.41 ASP 139 A\n",
      "LYS  37 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 ASP 305 A\n",
      "LYS  37 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.69 ASP  47 A\n",
      "LYS  37 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.16 GLU  72 A\n",
      "LYS  37 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.18 GLU  73 A\n",
      "\n",
      "LYS 108 A  10.20     0 %   -0.32  220   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 100 A\n",
      "\n",
      "LYS 111 A  10.66     0 %   -0.10  128   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.26 GLU 117 A\n",
      "\n",
      "LYS 120 A  10.80     0 %   -0.19  244   0.00    0    0.30 ASP  64 A    0.00 XXX   0 X    0.02 GLU  60 A\n",
      "LYS 120 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 117 A\n",
      "LYS 120 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.20 ARG 124 A\n",
      "LYS 120 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.34 ASP  64 A\n",
      "\n",
      "LYS 132 A  10.46     0 %   -0.04   93   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 164 A  10.39     0 %   -0.10  198   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 161 A\n",
      "LYS 164 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 209 A\n",
      "\n",
      "LYS 178 A  10.56     0 %   -0.13  165   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.12 GLU 180 A\n",
      "LYS 178 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 CYS 181 A\n",
      "\n",
      "LYS 179 A  10.81    19 %   -0.61  334   0.00    0    0.37 ASP 287 A    0.00 XXX   0 X   -0.11 ARG 285 A\n",
      "LYS 179 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.18 GLU 289 A\n",
      "LYS 179 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.47 ASP 287 A\n",
      "\n",
      "LYS 191 A  10.47     0 %   -0.03   79   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 193 A  10.48     0 %   -0.02   43   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 201 A  10.15    29 %   -0.45  362   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.10 CYS 181 A\n",
      "\n",
      "LYS 208 A  11.45     0 %   -0.31  254   0.00    0    0.85 GLU   2 A    0.00 XXX   0 X    0.01 GLU   4 A\n",
      "LYS 208 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 315 A\n",
      "LYS 208 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 GLU   2 A\n",
      "\n",
      "LYS 232 A  10.41     0 %   -0.09  201   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 269 A  10.30    12 %   -0.57  314   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.45 GLU 289 A\n",
      "LYS 269 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 LYS 179 A\n",
      "\n",
      "LYS 279 A   9.58   100 %   -3.37  661   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.21 ASP 139 A\n",
      "LYS 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.11 CYS 245 A\n",
      "LYS 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.74 LYS 293 A\n",
      "LYS 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.69 ASP 142 A\n",
      "LYS 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.52 GLU 301 A\n",
      "LYS 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.66 ASP 305 A\n",
      "\n",
      "LYS 293 A  10.97    95 %   -3.08  548   0.00    0    0.85 GLU 301 A    0.00 XXX   0 X    0.34 ASP 142 A\n",
      "LYS 293 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 GLU 249 A\n",
      "LYS 293 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 302 A\n",
      "LYS 293 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.62 GLU 301 A\n",
      "LYS 293 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.74 ASP 305 A\n",
      "\n",
      "LYS 300 A  11.52     0 %   -0.32  217   0.00    0    0.85 GLU 337 A    0.00 XXX   0 X    0.10 TYR 335 A\n",
      "LYS 300 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 GLU 337 A\n",
      "\n",
      "LYS 302 A  12.40    70 %   -1.84  477   0.00    0    0.84 GLU  72 A    0.00 XXX   0 X    0.27 ASP  39 A\n",
      "LYS 302 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.23 GLU  73 A\n",
      "LYS 302 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.50 ASP 139 A\n",
      "LYS 302 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 ASP 142 A\n",
      "LYS 302 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 301 A\n",
      "LYS 302 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.44 LYS  37 A\n",
      "LYS 302 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.18 TYR 335 A\n",
      "LYS 302 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.10 GLU  72 A\n",
      "LYS 302 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.89 ASP 305 A\n",
      "\n",
      "LYS 332 A  10.67     0 %   -0.17  202   0.00    0    0.01 ASP 329 A    0.00 XXX   0 X    0.32 ASP 329 A\n",
      "LYS 332 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 337 A\n",
      "\n",
      "LYS 354 A  10.57     0 %   -0.46  252   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.31 ASP 237 A\n",
      "LYS 354 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 ASP 238 A\n",
      "LYS 354 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 353 A\n",
      "\n",
      "ARG  10 A  13.55    29 %   -0.79  363   0.00    0    1.34 ASP   6 A    0.00 XXX   0 X    0.00 TYR 206 A\n",
      "ARG  10 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.50 ASP   6 A\n",
      "\n",
      "ARG 124 A  12.59    43 %   -1.35  401   0.00    0    0.68 ASP  64 A    0.00 XXX   0 X    0.07 GLU  60 A\n",
      "ARG 124 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.23 CYS 125 A\n",
      "ARG 124 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.46 ASP  64 A\n",
      "\n",
      "ARG 134 A  13.76     4 %   -0.37  293   0.00    0    0.25 GLU   4 A    0.00 XXX   0 X    0.26 GLU   4 A\n",
      "ARG 134 A                                            0.74 GLU 161 A    0.00 XXX   0 X    0.38 GLU 161 A\n",
      "\n",
      "ARG 149 A  12.62     0 %   -0.29  273   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP 151 A\n",
      "ARG 149 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 GLU 249 A\n",
      "ARG 149 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.17 TYR 275 A\n",
      "\n",
      "ARG 209 A  12.83    59 %   -1.51  447   0.00    0    0.74 GLU 314 A    0.00 XXX   0 X    0.28 GLU 315 A\n",
      "ARG 209 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.81 GLU 314 A\n",
      "\n",
      "ARG 250 A  12.37     0 %   -0.09  157   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.01 TYR 275 A\n",
      "ARG 250 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.05 ARG 273 A\n",
      "\n",
      "ARG 268 A  12.32     0 %   -0.18  156   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "ARG 273 A  12.69     0 %   -0.19  279   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.15 CYS 186 A\n",
      "ARG 273 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 GLU 246 A\n",
      "ARG 273 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 TYR 275 A\n",
      "\n",
      "ARG 285 A  13.75    70 %   -2.59  478   0.00    0    0.21 TYR 175 A    0.00 XXX   0 X    0.18 CYS 172 A\n",
      "ARG 285 A                                            0.66 ASP 287 A    0.00 XXX   0 X    1.24 TYR 175 A\n",
      "ARG 285 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 CYS 181 A\n",
      "ARG 285 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.65 CYS 203 A\n",
      "ARG 285 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.85 ASP 287 A\n",
      "\n",
      "ARG 326 A  13.60     0 %   -0.31  246   0.00    0    0.75 ASP 322 A    0.00 XXX   0 X    0.01 ASP 329 A\n",
      "ARG 326 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 ASP 340 A\n",
      "ARG 326 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 TYR 360 A\n",
      "ARG 326 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 ARG 341 A\n",
      "ARG 326 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.42 ASP 322 A\n",
      "\n",
      "ARG 341 A  14.03    17 %   -1.04  330   0.00    0    0.33 TYR 360 A    0.00 XXX   0 X    0.10 ASP 322 A\n",
      "ARG 341 A                                            1.39 ASP 340 A    0.00 XXX   0 X    0.30 TYR 360 A\n",
      "ARG 341 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 364 A\n",
      "ARG 341 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.44 ASP 340 A\n",
      "\n",
      "N+    1 A   8.68     0 %   -0.50  274   0.00    0    0.85 GLU   4 A    0.00 XXX   0 X    0.14 GLU   2 A\n",
      "N+    1 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.16 ARG 134 A\n",
      "N+    1 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 GLU 161 A\n",
      "N+    1 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.10 LYS 208 A\n",
      "N+    1 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.41 GLU   4 A\n",
      "\n",
      "Coupled residues (marked *) were detected.Please rerun PropKa with the --display-coupled-residues \n",
      "or -d option for detailed information.\n",
      "\n",
      "INFO:pKa summary:\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "SUMMARY OF THIS PREDICTION\n",
      "       Group      pKa  model-pKa   ligand atom-type\n",
      "   ASP   6 A     3.00       3.80                      \n",
      "   ASP  38 A     4.10       3.80                      \n",
      "   ASP  39 A     2.55       3.80                      \n",
      "   ASP  40 A     4.01       3.80                      \n",
      "   ASP  47 A     5.77       3.80                      \n",
      "   ASP  64 A     2.40       3.80                      \n",
      "   ASP  74 A     3.59       3.80                      \n",
      "   ASP  82 A     3.93       3.80                      \n",
      "   ASP 139 A    11.53       3.80                      \n",
      "   ASP 142 A     5.13       3.80                      \n",
      "   ASP 151 A     4.73       3.80                      \n",
      "   ASP 237 A     3.65       3.80                      \n",
      "   ASP 238 A     3.28       3.80                      \n",
      "   ASP 287 A     3.73       3.80                      \n",
      "   ASP 305 A     6.18       3.80                      \n",
      "   ASP 322 A     3.09       3.80                      \n",
      "   ASP 329 A     2.58       3.80                      \n",
      "   ASP 340 A     2.65       3.80                      \n",
      "   GLU   2 A     3.46       4.50                      \n",
      "   GLU   4 A     2.86       4.50                      \n",
      "   GLU  24 A     4.74       4.50                      \n",
      "   GLU  60 A     5.00       4.50                      \n",
      "   GLU  72 A     1.96       4.50                      \n",
      "   GLU  73 A     7.93       4.50                      \n",
      "   GLU  76 A     4.99       4.50                      \n",
      "   GLU  80 A     3.53       4.50                      \n",
      "   GLU  96 A     3.28       4.50                      \n",
      "   GLU 100 A     4.18       4.50                      \n",
      "   GLU 117 A     3.73       4.50                      \n",
      "   GLU 161 A     3.76       4.50                      \n",
      "   GLU 180 A     4.49       4.50                      \n",
      "   GLU 226 A     4.56       4.50                      \n",
      "   GLU 246 A     5.27       4.50                      \n",
      "   GLU 249 A     5.27       4.50                      \n",
      "   GLU 289 A     3.91       4.50                      \n",
      "   GLU 301 A     1.69       4.50                      \n",
      "   GLU 314 A     5.47       4.50                      \n",
      "   GLU 315 A     4.02       4.50                      \n",
      "   GLU 337 A     3.62       4.50                      \n",
      "   GLU 353 A     3.83       4.50                      \n",
      "   GLU 364 A     4.60       4.50                      \n",
      "   C-  368 A     3.32       3.20                      \n",
      "   HIS  13 A     4.57       6.50                      \n",
      "   HIS  27 A     6.82       6.50                      \n",
      "   HIS  33 A     6.06       6.50                      \n",
      "   HIS 135 A     5.85       6.50                      \n",
      "   HIS 255 A     6.41       6.50                      \n",
      "   HIS 306 A     5.07       6.50                      \n",
      "   HIS 352 A     4.72       6.50                      \n",
      "   CYS  19 A    13.08       9.00                      \n",
      "   CYS 125 A    11.63       9.00                      \n",
      "   CYS 172 A    13.72       9.00                      \n",
      "   CYS 181 A     9.45       9.00                      \n",
      "   CYS 186 A     9.82       9.00                      \n",
      "   CYS 203 A    11.78       9.00                      \n",
      "   CYS 245 A    12.50       9.00                      \n",
      "   CYS 346 A    12.74       9.00                      \n",
      "   TYR 104 A    10.78      10.00                      \n",
      "   TYR 153 A    15.70      10.00                      \n",
      "   TYR 175 A    15.46      10.00                      \n",
      "   TYR 206 A    11.50      10.00                      \n",
      "   TYR 275 A    12.11      10.00                      \n",
      "   TYR 280 A    14.55      10.00                      \n",
      "   TYR 299 A    14.29      10.00                      \n",
      "   TYR 335 A    10.09      10.00                      \n",
      "   TYR 360 A    10.04      10.00                      \n",
      "   LYS  21 A    11.63      10.50                      \n",
      "   LYS  25 A    11.09      10.50                      \n",
      "   LYS  35 A    11.28      10.50                      \n",
      "   LYS  37 A    13.37      10.50                      \n",
      "   LYS 108 A    10.20      10.50                      \n",
      "   LYS 111 A    10.66      10.50                      \n",
      "   LYS 120 A    10.80      10.50                      \n",
      "   LYS 132 A    10.46      10.50                      \n",
      "   LYS 164 A    10.39      10.50                      \n",
      "   LYS 178 A    10.56      10.50                      \n",
      "   LYS 179 A    10.81      10.50                      \n",
      "   LYS 191 A    10.47      10.50                      \n",
      "   LYS 193 A    10.48      10.50                      \n",
      "   LYS 201 A    10.15      10.50                      \n",
      "   LYS 208 A    11.45      10.50                      \n",
      "   LYS 232 A    10.41      10.50                      \n",
      "   LYS 269 A    10.30      10.50                      \n",
      "   LYS 279 A     9.58      10.50                      \n",
      "   LYS 293 A    10.97      10.50                      \n",
      "   LYS 300 A    11.52      10.50                      \n",
      "   LYS 302 A    12.40      10.50                      \n",
      "   LYS 332 A    10.67      10.50                      \n",
      "   LYS 354 A    10.57      10.50                      \n",
      "   ARG  10 A    13.55      12.50                      \n",
      "   ARG 124 A    12.59      12.50                      \n",
      "   ARG 134 A    13.76      12.50                      \n",
      "   ARG 149 A    12.62      12.50                      \n",
      "   ARG 209 A    12.83      12.50                      \n",
      "   ARG 250 A    12.37      12.50                      \n",
      "   ARG 268 A    12.32      12.50                      \n",
      "   ARG 273 A    12.69      12.50                      \n",
      "   ARG 285 A    13.75      12.50                      \n",
      "   ARG 326 A    13.60      12.50                      \n",
      "   ARG 341 A    14.03      12.50                      \n",
      "   N+    1 A     8.68       8.00                      \n",
      "\n",
      "INFO:PROPKA information:\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "Free energy of   folding (kcal/mol) as a function of pH (using neutral reference)\n",
      "  0.00     28.05\n",
      "  1.00     27.81\n",
      "  2.00     26.19\n",
      "  3.00     20.32\n",
      "  4.00     11.83\n",
      "  5.00     10.29\n",
      "  6.00     12.13\n",
      "  7.00     13.31\n",
      "  8.00     15.91\n",
      "  9.00     20.92\n",
      " 10.00     32.09\n",
      " 11.00     51.34\n",
      " 12.00     71.39\n",
      " 13.00     87.31\n",
      " 14.00     98.55\n",
      "\n",
      "The pH of optimum stability is  4.6 for which the free energy is    9.9 kcal/mol at 298K\n",
      "Could not determine pH values where the free energy is within 80 % of minimum\n",
      "Could not determine the pH-range where the free energy is negative\n",
      "\n",
      "\n",
      "Protein charge of folded and unfolded state as a function of pH\n",
      "    pH  unfolded  folded\n",
      "  0.00     42.00   41.95\n",
      "  0.10     41.99   41.94\n",
      "  0.20     41.99   41.92\n",
      "  0.30     41.99   41.90\n",
      "  0.40     41.99   41.88\n",
      "  0.50     41.99   41.85\n",
      "  0.60     41.98   41.81\n",
      "  0.70     41.98   41.76\n",
      "  0.80     41.97   41.70\n",
      "  0.90     41.97   41.63\n",
      "  1.00     41.96   41.54\n",
      "  1.10     41.95   41.44\n",
      "  1.20     41.93   41.31\n",
      "  1.30     41.92   41.16\n",
      "  1.40     41.89   40.98\n",
      "  1.50     41.87   40.78\n",
      "  1.60     41.83   40.54\n",
      "  1.70     41.79   40.26\n",
      "  1.80     41.74   39.95\n",
      "  1.90     41.67   39.59\n",
      "  2.00     41.59   39.19\n",
      "  2.10     41.48   38.75\n",
      "  2.20     41.35   38.25\n",
      "  2.30     41.19   37.70\n",
      "  2.40     40.99   37.10\n",
      "  2.50     40.75   36.44\n",
      "  2.60     40.44   35.72\n",
      "  2.70     40.08   34.94\n",
      "  2.80     39.63   34.10\n",
      "  2.90     39.09   33.19\n",
      "  3.00     38.44   32.23\n",
      "  3.10     37.68   31.20\n",
      "  3.20     36.79   30.11\n",
      "  3.30     35.75   28.97\n",
      "  3.40     34.56   27.77\n",
      "  3.50     33.23   26.53\n",
      "  3.60     31.74   25.25\n",
      "  3.70     30.11   23.95\n",
      "  3.80     28.36   22.63\n",
      "  3.90     26.50   21.29\n",
      "  4.00     24.55   19.97\n",
      "  4.10     22.54   18.65\n",
      "  4.20     20.50   17.34\n",
      "  4.30     18.46   16.06\n",
      "  4.40     16.44   14.80\n",
      "  4.50     14.47   13.57\n",
      "  4.60     12.59   12.37\n",
      "  4.70     10.83   11.19\n",
      "  4.80      9.20   10.06\n",
      "  4.90      7.72    8.95\n",
      "  5.00      6.39    7.89\n",
      "  5.10      5.22    6.87\n",
      "  5.20      4.19    5.90\n",
      "  5.30      3.29    4.97\n",
      "  5.40      2.50    4.10\n",
      "  5.50      1.81    3.27\n",
      "  5.60      1.19    2.50\n",
      "  5.70      0.62    1.77\n",
      "  5.80      0.10    1.08\n",
      "  5.90     -0.40    0.44\n",
      "  6.00     -0.88   -0.15\n",
      "  6.10     -1.36   -0.71\n",
      "  6.20     -1.85   -1.22\n",
      "  6.30     -2.33   -1.69\n",
      "  6.40     -2.82   -2.13\n",
      "  6.50     -3.30   -2.53\n",
      "  6.60     -3.77   -2.89\n",
      "  6.70     -4.22   -3.22\n",
      "  6.80     -4.65   -3.52\n",
      "  6.90     -5.05   -3.79\n",
      "  7.00     -5.42   -4.03\n",
      "  7.10     -5.76   -4.25\n",
      "  7.20     -6.07   -4.45\n",
      "  7.30     -6.36   -4.64\n",
      "  7.40     -6.62   -4.81\n",
      "  7.50     -6.87   -4.97\n",
      "  7.60     -7.12   -5.12\n",
      "  7.70     -7.36   -5.27\n",
      "  7.80     -7.62   -5.42\n",
      "  7.90     -7.88   -5.57\n",
      "  8.00     -8.17   -5.72\n",
      "  8.10     -8.48   -5.88\n",
      "  8.20     -8.82   -6.04\n",
      "  8.30     -9.20   -6.22\n",
      "  8.40     -9.63   -6.40\n",
      "  8.50    -10.11   -6.59\n",
      "  8.60    -10.65   -6.81\n",
      "  8.70    -11.25   -7.04\n",
      "  8.80    -11.91   -7.29\n",
      "  8.90    -12.63   -7.57\n",
      "  9.00    -13.41   -7.89\n",
      "  9.10    -14.26   -8.24\n",
      "  9.20    -15.17   -8.64\n",
      "  9.30    -16.14   -9.09\n",
      "  9.40    -17.18   -9.59\n",
      "  9.50    -18.30  -10.16\n",
      "  9.60    -19.51  -10.80\n",
      "  9.70    -20.81  -11.51\n",
      "  9.80    -22.22  -12.29\n",
      "  9.90    -23.72  -13.16\n",
      " 10.00    -25.32  -14.10\n",
      " 10.10    -27.01  -15.11\n",
      " 10.20    -28.77  -16.20\n",
      " 10.30    -30.57  -17.34\n",
      " 10.40    -32.39  -18.53\n",
      " 10.50    -34.20  -19.75\n",
      " 10.60    -35.95  -21.00\n",
      " 10.70    -37.62  -22.26\n",
      " 10.80    -39.18  -23.51\n",
      " 10.90    -40.61  -24.76\n",
      " 11.00    -41.91  -25.98\n",
      " 11.10    -43.08  -27.19\n",
      " 11.20    -44.11  -28.36\n",
      " 11.30    -45.04  -29.51\n",
      " 11.40    -45.86  -30.63\n",
      " 11.50    -46.61  -31.73\n",
      " 11.60    -47.30  -32.79\n",
      " 11.70    -47.95  -33.84\n",
      " 11.80    -48.58  -34.85\n",
      " 11.90    -49.21  -35.85\n",
      " 12.00    -49.84  -36.83\n",
      " 12.10    -50.49  -37.80\n",
      " 12.20    -51.16  -38.75\n",
      " 12.30    -51.85  -39.69\n",
      " 12.40    -52.54  -40.62\n",
      " 12.50    -53.24  -41.54\n",
      " 12.60    -53.92  -42.44\n",
      " 12.70    -54.58  -43.33\n",
      " 12.80    -55.20  -44.20\n",
      " 12.90    -55.76  -45.04\n",
      " 13.00    -56.27  -45.87\n",
      " 13.10    -56.73  -46.67\n",
      " 13.20    -57.12  -47.44\n",
      " 13.30    -57.45  -48.19\n",
      " 13.40    -57.74  -48.93\n",
      " 13.50    -57.97  -49.64\n",
      " 13.60    -58.17  -50.32\n",
      " 13.70    -58.33  -50.98\n",
      " 13.80    -58.46  -51.62\n",
      " 13.90    -58.57  -52.22\n",
      " 14.00    -58.65  -52.79\n",
      "The pI is  5.97 (folded) and  5.82 (unfolded)\n",
      "\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "INFO:Applying pKa values at a pH of 7.60:\n",
      "INFO:Adding hydrogens to biomolecule.\n",
      "INFO:Debumping biomolecule (again).\n",
      "INFO:Optimizing hydrogen bonds\n",
      "INFO:Applying force field to biomolecule states.\n",
      "INFO:Regenerating headers.\n",
      "INFO:Regenerating PDB lines.\n",
      "WARNING:Ignoring 466 header lines in output.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "pH: 7.6\tname: AtAHL\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:PDB2PQR v3.6.2: biomolecular structure conversion software.\n",
      "INFO:Please cite:  Jurrus E, et al.  Improvements to the APBS biomolecular solvation software suite.  Protein Sci 27 112-128 (2018).\n",
      "INFO:Please cite:  Dolinsky TJ, et al.  PDB2PQR: expanding and upgrading automated preparation of biomolecular structures for molecular simulations. Nucleic Acids Res 35 W522-W525 (2007).\n",
      "INFO:Checking and transforming input arguments.\n",
      "INFO:Loading topology files.\n",
      "INFO:Loading molecule: /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/74987.pdb\n",
      "INFO:Setting up molecule.\n",
      "INFO:Created biomolecule object with 352 residues and 5354 atoms.\n",
      "INFO:Setting termini states for biomolecule chains.\n",
      "INFO:Loading forcefield.\n",
      "INFO:Loading hydrogen topology definitions.\n",
      "INFO:This biomolecule is clean.  No repair needed.\n",
      "INFO:Updating disulfide bridges.\n",
      "INFO:Debumping biomolecule.\n",
      "INFO:Assigning titration states with PROPKA.\n",
      "INFO:\n",
      "propka3.5.1                                                                                  2024-07-26\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "--                                                                           --\n",
      "--  PROPKA: A PROTEIN PKA PREDICTOR                                          --\n",
      "--                                                                           --\n",
      "--  VERSION 1.0,  04/25/2004,  IOWA CITY                                     --\n",
      "--  BY HUI LI                                                                --\n",
      "--                                                                           --\n",
      "--  VERSION 2.0,  11/05/2007, IOWA CITY/COPENHAGEN                           --\n",
      "--  BY DELPHINE C. BAS AND DAVID M. ROGERS                                   --\n",
      "--                                                                           --\n",
      "--  VERSION 3.0,  01/06/2011, COPENHAGEN                                     --\n",
      "--  BY MATS H.M. OLSSON AND CHRESTEN R. SONDERGARD                           --\n",
      "--                                                                           --\n",
      "--  VERSION 3.1,  07/01/2011, COPENHAGEN                                     --\n",
      "--  BY CHRESTEN R. SONDERGARD AND MATS H.M. OLSSON                           --\n",
      "--                                                                           --\n",
      "--  VERSION 3.2,  06/17/2020, PLANET EARTH                                   --\n",
      "--  SEE ABOVE FOR AUTHORS                                                    --\n",
      "--                                                                           --\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "References:\n",
      "\n",
      "Very Fast Empirical Prediction and Rationalization of Protein pKa Values.\n",
      "Hui Li, Andrew D. Robertson and Jan H. Jensen. PROTEINS: Structure, Function,\n",
      "and Bioinformatics. 61:704-721 (2005)\n",
      "\n",
      "Very Fast Prediction and Rationalization of pKa Values for Protein-Ligand\n",
      "Complexes.  Delphine C. Bas, David M. Rogers and Jan H. Jensen.  PROTEINS:\n",
      "Structure, Function, and Bioinformatics 73:765-783 (2008)\n",
      "\n",
      "PROPKA3: Consistent Treatment of Internal and Surface Residues in Empirical\n",
      "pKa predictions.  Mats H.M. Olsson, Chresten R. Sondergard, Michal Rostkowski,\n",
      "and Jan H. Jensen.  Journal of Chemical Theory and Computation, 7(2):525-537\n",
      "(2011)\n",
      "\n",
      "Improved Treatment of Ligands and Coupling Effects in Empirical Calculation\n",
      "and Rationalization of pKa Values.  Chresten R. Sondergaard, Mats H.M. Olsson,\n",
      "Michal Rostkowski, and Jan H. Jensen.  Journal of Chemical Theory and\n",
      "Computation, (2011)\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "\n",
      "INFO:Coupling map:\n",
      "\n",
      "INFO:Calculating pKas for Conformation container 1A with 3171 atoms and 875 groups\n",
      "INFO:pKa determinants:\n",
      "\n",
      "---------  -----   ------   ---------------------    --------------    --------------    --------------\n",
      "                            DESOLVATION  EFFECTS       SIDECHAIN          BACKBONE        COULOMBIC\n",
      " RESIDUE    pKa    BURIED     REGULAR      RE        HYDROGEN BOND     HYDROGEN BOND      INTERACTION\n",
      "---------  -----   ------   ---------   ---------    --------------    --------------    --------------\n",
      "\n",
      "ASP  10 A   4.95    44 %    1.49  405   0.14    0   -0.28 ARG  14 A    0.00 XXX   0 X   -0.05 ARG  17 A\n",
      "ASP  10 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.10 LYS  59 A\n",
      "ASP  10 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU   6 A\n",
      "ASP  10 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP  13 A\n",
      "ASP  10 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.29 GLU  56 A\n",
      "ASP  10 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.55 ARG  14 A\n",
      "\n",
      "ASP  13 A   4.41    60 %    1.88  448   0.32    0   -0.72 ARG  17 A   -0.16 GLN 194 A   -0.08 ARG  14 A\n",
      "ASP  13 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.00 LYS  91 A\n",
      "ASP  13 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.05 LYS 191 A\n",
      "ASP  13 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.59 ARG  17 A\n",
      "\n",
      "ASP  21 A   2.92    25 %    0.74  352   0.19    0   -0.11 ARG  24 A    0.00 XXX   0 X   -0.03 ARG  17 A\n",
      "ASP  21 A                                           -0.85 LYS  91 A    0.00 XXX   0 X   -0.32 ARG  24 A\n",
      "ASP  21 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.49 LYS  91 A\n",
      "\n",
      "ASP  38 A   2.45    12 %    0.70  316   0.00    0   -0.85 THR  40 A   -0.20 ASP  38 A   -0.26 LYS  36 A\n",
      "ASP  38 A                                            0.00 XXX   0 X   -0.66 THR  40 A   -0.08 LYS 286 A\n",
      "\n",
      "ASP  46 A   5.88   100 %    3.67  601   1.00    0   -0.85 THR  43 A   -0.70 THR  43 A   -0.59 LYS  36 A\n",
      "ASP  46 A                                           -0.64 THR 134 A   -0.38 ASP  46 A    0.56 GLU  69 A\n",
      "\n",
      "ASP  71 A   3.56    35 %    0.91  380   0.00    0    0.00 XXX   0 X   -0.46 HIS  73 A   -0.12 LYS  36 A\n",
      "ASP  71 A                                            0.00 XXX   0 X   -0.28 PHE  74 A   -0.05 ARG  76 A\n",
      "ASP  71 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.21 LYS 113 A\n",
      "ASP  71 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 322 A\n",
      "ASP  71 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.19 HIS  73 A\n",
      "ASP  71 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.18 ASP 109 A\n",
      "\n",
      "ASP  96 A   3.95     0 %    0.24  216   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.09 LYS  89 A\n",
      "ASP  96 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.00 LYS  91 A\n",
      "\n",
      "ASP 101 A   3.94     0 %    0.32  265   0.00    0    0.00 XXX   0 X   -0.03 ASP 101 A   -0.16 LYS  89 A\n",
      "ASP 101 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 103 A\n",
      "\n",
      "ASP 103 A   2.46     0 %    0.37  271   0.00    0   -0.85 SER 100 A   -0.81 SER 100 A   -0.04 LYS  89 A\n",
      "ASP 103 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 HIS  98 A\n",
      "\n",
      "ASP 109 A   2.64    27 %    1.12  356   0.23    0   -0.85 SER  72 A   -0.79 SER  72 A   -0.02 HIS  73 A\n",
      "ASP 109 A                                           -0.48 ARG  76 A    0.00 XXX   0 X   -0.37 ARG  76 A\n",
      "\n",
      "ASP 114 A   3.91     0 %    0.33  245   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.29 LYS 113 A\n",
      "ASP 114 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 ARG 322 A\n",
      "ASP 114 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.11 ASP  71 A\n",
      "ASP 114 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 109 A\n",
      "\n",
      "ASP 129 A  11.71*  100 %    3.55  628   0.27    0    0.42 ASP 289 A   -0.23 ILE 131 A   -0.17 LYS  36 A\n",
      "ASP 129 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.32 ASP  46 A\n",
      "ASP 129 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 GLU  70 A\n",
      "ASP 129 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.98 GLU  69 A\n",
      "ASP 129 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.31 ASP 132 A\n",
      "ASP 129 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.28 ASP 289 A\n",
      "\n",
      "ASP 132 A   6.29*  100 %    3.80  623   0.65    0   -0.13 SER 260 A   -0.48 SER 260 A    0.11 GLU  69 A\n",
      "ASP 132 A                                            0.00 XXX   0 X   -0.56 LEU 261 A   -0.70 LYS 263 A\n",
      "ASP 132 A                                            0.00 XXX   0 X   -0.19 CYS 262 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 141 A   3.75     0 %    0.27  267   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.31 LYS  23 A\n",
      "ASP 141 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 ARG  24 A\n",
      "\n",
      "ASP 154 A   4.07     0 %    0.28  239   0.00    0    0.00 XXX   0 X   -0.00 ASP 154 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 170 A   3.21     0 %    0.24  274   0.00    0   -0.42 ARG 271 A    0.00 XXX   0 X   -0.03 LYS 192 A\n",
      "ASP 170 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.37 ARG 271 A\n",
      "\n",
      "ASP 173 A   1.41     0 %    0.49  200   0.00    0   -0.85 SER 175 A   -0.08 ASP 173 A   -0.38 LYS 192 A\n",
      "ASP 173 A                                           -0.81 LYS 192 A   -0.76 SER 175 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 203 A   3.82     0 %    0.12  132   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.10 LYS 191 A\n",
      "\n",
      "ASP 210 A   2.58    45 %    1.85  406   0.24    0   -0.85 SER 273 A   -0.41 CYS 212 A   -0.43 ARG 219 A\n",
      "ASP 210 A                                           -0.78 ASN 334 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "ASP 210 A                                           -0.85 ARG 219 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "ASP 238 A   4.04     0 %    0.23  233   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.09 LYS 247 A\n",
      "ASP 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 249 A\n",
      "ASP 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 229 A\n",
      "ASP 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 GLU 250 A\n",
      "\n",
      "ASP 245 A   4.00     0 %    0.16  109   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.04 GLU 250 A\n",
      "\n",
      "ASP 289 A   7.94*  100 %    2.62  580   0.38    0   -0.42 ASP 129 A    0.00 XXX   0 X   -0.01 LYS  36 A\n",
      "ASP 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.00 LYS 263 A\n",
      "ASP 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.19 LYS 286 A\n",
      "ASP 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.76 GLU  69 A\n",
      "ASP 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.00 ASP 132 A\n",
      "\n",
      "ASP 306 A   3.95    39 %    1.62  391   0.52    0    0.00 XXX   0 X   -0.01 ASP 306 A   -0.01 ARG 283 A\n",
      "ASP 306 A                                            0.00 XXX   0 X   -0.48 GLU 308 A   -0.01 HIS 338 A\n",
      "ASP 306 A                                            0.00 XXX   0 X   -0.34 GLY 309 A    0.00 XXX   0 X\n",
      "ASP 306 A                                            0.00 XXX   0 X   -0.61 ASP 310 A    0.00 XXX   0 X\n",
      "ASP 306 A                                            0.00 XXX   0 X   -0.53 ALA 327 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 310 A   4.33     0 %    0.22  242   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.00 LYS 303 A\n",
      "ASP 310 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.28 ASP 306 A\n",
      "ASP 310 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 311 A\n",
      "\n",
      "ASP 317 A   3.26    37 %    1.59  385   0.12    0    0.00 XXX   0 X   -0.51 GLU 320 A   -0.05 LYS 286 A\n",
      "ASP 317 A                                            0.00 XXX   0 X   -0.03 ARG 321 A   -0.13 ARG 321 A\n",
      "ASP 317 A                                            0.00 XXX   0 X   -0.72 ARG 322 A   -0.10 ARG 322 A\n",
      "ASP 317 A                                            0.00 XXX   0 X   -0.69 LEU 323 A    0.00 XXX   0 X\n",
      "\n",
      "GLU   4 A   4.10    15 %    0.77  322   0.00    0    0.00 XXX   0 X   -0.76 GLU   4 A   -0.26 N+    1 A\n",
      "GLU   4 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.07 LYS   3 A\n",
      "GLU   4 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 HIS   2 A\n",
      "GLU   4 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 HIS 183 A\n",
      "\n",
      "GLU   6 A   4.46     0 %    0.19  222   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.05 LYS   3 A\n",
      "GLU   6 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.18 HIS   2 A\n",
      "\n",
      "GLU  31 A   4.62     6 %    0.36  298   0.00    0   -0.24 SER  29 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "GLU  35 A   4.73     0 %    0.22  190   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP  38 A\n",
      "\n",
      "GLU  56 A   2.83    47 %    1.86  412   0.21    0   -0.78 SER  93 A   -0.83 SER  93 A   -0.52 ARG  14 A\n",
      "GLU  56 A                                           -0.18 ARG  14 A    0.00 XXX   0 X   -0.58 LYS  59 A\n",
      "GLU  56 A                                           -0.85 LYS  59 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "GLU  69 A   4.06*   92 %    1.97  539   0.00    0   -0.35 TRP 288 A    0.00 XXX   0 X   -0.18 LYS 286 A\n",
      "GLU  69 A                                           -0.85 LYS  36 A    0.00 XXX   0 X   -0.02 ARG 322 A\n",
      "GLU  69 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 ASP  38 A\n",
      "GLU  69 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.20 LYS  36 A\n",
      "\n",
      "GLU  70 A   8.36*   97 %    3.42  553   0.03    0   -0.85 LYS  36 A    0.00 XXX   0 X    0.08 ASP  38 A\n",
      "GLU  70 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 ASP  71 A\n",
      "GLU  70 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.29 LYS  36 A\n",
      "GLU  70 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.00 ASP  46 A\n",
      "GLU  70 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.27 GLU  69 A\n",
      "\n",
      "GLU  87 A   5.73    40 %    1.00  393   0.16    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.00 ARG  24 A\n",
      "GLU  87 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.14 LYS  91 A\n",
      "GLU  87 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.17 ASP  21 A\n",
      "GLU  87 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU  31 A\n",
      "\n",
      "GLU 106 A   4.79     0 %    0.16  269   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 110 A\n",
      "GLU 106 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.11 ASP 103 A\n",
      "GLU 106 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 109 A\n",
      "\n",
      "GLU 142 A   4.62     0 %    0.09  177   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.01 LYS  23 A\n",
      "GLU 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 141 A\n",
      "\n",
      "GLU 156 A   5.01    18 %    0.37  332   0.02    0    0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 154 A\n",
      "GLU 156 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 HIS 298 A\n",
      "GLU 156 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.14 GLU 311 A\n",
      "GLU 156 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 315 A\n",
      "\n",
      "GLU 224 A   2.07    12 %    0.73  314   0.02    0   -0.52 ARG 135 A    0.00 XXX   0 X   -0.16 LYS 263 A\n",
      "GLU 224 A                                           -0.85 LYS 139 A    0.00 XXX   0 X   -0.42 ARG 135 A\n",
      "GLU 224 A                                           -0.84 CYS 258 A    0.00 XXX   0 X   -0.39 LYS 139 A\n",
      "\n",
      "GLU 229 A   3.42     0 %    0.34  244   0.00    0   -0.66 THR 227 A   -0.76 GLU 229 A    0.00 XXX   0 X\n",
      "\n",
      "GLU 244 A   4.57     0 %    0.07   82   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "GLU 250 A   2.96     0 %    0.53  224   0.00    0   -0.85 LYS 247 A   -0.84 LYS 247 A   -0.00 LYS 217 A\n",
      "GLU 250 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.38 LYS 247 A\n",
      "\n",
      "GLU 299 A   4.11    28 %    1.15  361   0.24    0    0.00 XXX   0 X   -0.79 GLY 185 A   -0.13 ARG 206 A\n",
      "GLU 299 A                                            0.00 XXX   0 X   -0.08 CYS 186 A   -0.10 HIS 183 A\n",
      "GLU 299 A                                            0.00 XXX   0 X   -0.58 GLY 187 A   -0.10 HIS 298 A\n",
      "\n",
      "GLU 308 A   5.24     0 %    0.26  271   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.10 ARG 283 A\n",
      "GLU 308 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.24 ASP 306 A\n",
      "GLU 308 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP 310 A\n",
      "GLU 308 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 C-  352 A\n",
      "\n",
      "GLU 311 A   3.29    11 %    0.28  311   0.00    0   -0.85 LYS 303 A    0.00 XXX   0 X   -0.21 HIS 298 A\n",
      "GLU 311 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.42 LYS 303 A\n",
      "\n",
      "GLU 315 A   4.29     0 %    0.41  259   0.00    0   -0.39 ASN 155 A    0.00 XXX   0 X   -0.24 ARG 321 A\n",
      "GLU 315 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 154 A\n",
      "\n",
      "GLU 316 A   5.33    15 %    0.68  322   0.08    0    0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 306 A\n",
      "GLU 316 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 317 A\n",
      "\n",
      "GLU 320 A   4.89     8 %    0.35  305   0.01    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.05 LYS 113 A\n",
      "GLU 320 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 LYS 286 A\n",
      "GLU 320 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.19 ARG 322 A\n",
      "GLU 320 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 114 A\n",
      "GLU 320 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.30 ASP 317 A\n",
      "\n",
      "GLU 343 A   4.70     0 %    0.20  205   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "C-  352 A   2.61     0 %    0.24  209   0.00    0   -0.57 ARG 283 A    0.00 XXX   0 X   -0.26 ARG 283 A\n",
      "\n",
      "HIS   2 A   6.46     0 %   -0.18  208   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.08 N+    1 A\n",
      "HIS   2 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 GLU   4 A\n",
      "HIS   2 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.18 GLU   6 A\n",
      "\n",
      "HIS  73 A   6.41     0 %   -0.19  218   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.06 ARG  76 A\n",
      "HIS  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 113 A\n",
      "HIS  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 ASP  71 A\n",
      "HIS  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 109 A\n",
      "\n",
      "HIS  98 A   6.37     0 %   -0.15  122   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.00 LYS  89 A\n",
      "HIS  98 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 103 A\n",
      "\n",
      "HIS 183 A   6.16    21 %   -0.82  341   0.00    0    0.00 XXX   0 X    0.05 HIS 183 A   -0.39 N+    1 A\n",
      "HIS 183 A                                            0.00 XXX   0 X    0.71 ILE 184 A   -0.03 HIS   2 A\n",
      "HIS 183 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 GLU   4 A\n",
      "HIS 183 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 GLU 299 A\n",
      "\n",
      "HIS 248 A   6.22     0 %   -0.17  173   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 217 A\n",
      "HIS 248 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 249 A\n",
      "\n",
      "HIS 290 A   3.27   100 %   -3.17  619   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.07 LYS 263 A\n",
      "\n",
      "HIS 298 A   5.12    76 %   -1.70  493   0.00    0    0.00 XXX   0 X    0.30 GLY 302 A   -0.32 LYS 303 A\n",
      "HIS 298 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 156 A\n",
      "HIS 298 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 GLU 299 A\n",
      "HIS 298 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.21 GLU 311 A\n",
      "\n",
      "HIS 338 A   4.12    64 %   -2.13  461   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.17 LYS 303 A\n",
      "HIS 338 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.10 HIS 298 A\n",
      "HIS 338 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 306 A\n",
      "\n",
      "CYS  16 A  14.27   100 %    3.67  626   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.05 ARG  17 A\n",
      "CYS  16 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.44 ASP  13 A\n",
      "CYS  16 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.84 CYS  19 A\n",
      "CYS  16 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.36 CYS 164 A\n",
      "\n",
      "CYS  19 A  13.25   100 %    3.84  640   0.00    0    0.00 XXX   0 X   -0.06 CYS  19 A   -0.03 LYS  23 A\n",
      "CYS  19 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.49 ASP  46 A\n",
      "CYS  19 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU  87 A\n",
      "\n",
      "CYS 119 A   9.04     0 %    0.20  187   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.14 LYS 121 A\n",
      "CYS 119 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 321 A\n",
      "\n",
      "CYS 164 A  12.58   100 %    3.66  580   0.00    0    0.00 XXX   0 X   -0.13 TYR 145 A   -0.02 LYS  23 A\n",
      "CYS 164 A                                            0.00 XXX   0 X   -0.10 ASN 166 A   -0.04 ARG 271 A\n",
      "CYS 164 A                                            0.00 XXX   0 X   -0.05 TRP 167 A    0.11 ASP 132 A\n",
      "CYS 164 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.17 CYS 257 A\n",
      "\n",
      "CYS 186 A   9.38     0 %    0.39  262   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.09 N+    1 A\n",
      "CYS 186 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 ARG 206 A\n",
      "CYS 186 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 GLU 299 A\n",
      "\n",
      "CYS 207 A  11.12    63 %    2.22  457   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.07 ARG 206 A\n",
      "CYS 207 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 271 A\n",
      "\n",
      "CYS 212 A   9.75    29 %    1.27  363   0.00    0   -0.00 SER 273 A   -0.71 LEU 214 A   -0.04 LYS 217 A\n",
      "CYS 212 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.12 ARG 219 A\n",
      "CYS 212 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.35 ASP 210 A\n",
      "\n",
      "CYS 221 A  11.78    93 %    2.98  541   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.06 ARG 219 A\n",
      "CYS 221 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.40 LYS 263 A\n",
      "CYS 221 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.26 CYS 257 A\n",
      "\n",
      "CYS 257 A   6.54*   61 %    1.43  451   0.00    0   -3.60 CYS 262 A    0.00 XXX   0 X   -0.02 LYS 139 A\n",
      "CYS 257 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.33 LYS 263 A\n",
      "CYS 257 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 GLU 224 A\n",
      "\n",
      "CYS 258 A  10.03    39 %    1.48  391   0.00    0   -0.71 ARG 135 A   -0.05 CYS 258 A   -0.39 ARG 135 A\n",
      "CYS 258 A                                           -0.23 LYS 139 A    0.00 XXX   0 X   -0.45 LYS 139 A\n",
      "CYS 258 A                                            0.84 GLU 224 A    0.00 XXX   0 X   -0.20 LYS 263 A\n",
      "CYS 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP 132 A\n",
      "CYS 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.15 CYS 257 A\n",
      "CYS 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.48 GLU 224 A\n",
      "\n",
      "CYS 262 A  17.21*  100 %    3.46  569   0.00    0    3.60 CYS 257 A   -0.07 CYS 262 A   -1.78 LYS 263 A\n",
      "CYS 262 A                                            0.00 XXX   0 X   -0.06 LYS 263 A    0.53 CYS 164 A\n",
      "CYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.44 CYS 221 A\n",
      "CYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 224 A\n",
      "CYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 CYS 258 A\n",
      "CYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.03 ASP 132 A\n",
      "CYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.81 CYS 257 A\n",
      "\n",
      "CYS 296 A  12.13   100 %    3.70  656   0.00    0   -0.85 SER 182 A    0.00 XXX   0 X    0.23 CYS 207 A\n",
      "CYS 296 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 299 A\n",
      "\n",
      "TYR   1 A  10.58     0 %    0.41  279   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU   4 A\n",
      "TYR   1 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP 154 A\n",
      "TYR   1 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 GLU 156 A\n",
      "\n",
      "TYR 116 A  10.46     2 %    0.31  288   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.16 CYS 119 A\n",
      "TYR 116 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 ARG 321 A\n",
      "\n",
      "TYR 125 A  12.77    85 %    2.97  518   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 315 A\n",
      "TYR 125 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP 317 A\n",
      "TYR 125 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.34 ARG 321 A\n",
      "\n",
      "TYR 145 A  16.29    97 %    3.60  552   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.78 CYS  16 A\n",
      "TYR 145 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.50 CYS  19 A\n",
      "TYR 145 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP  21 A\n",
      "TYR 145 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 141 A\n",
      "TYR 145 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.37 CYS 164 A\n",
      "\n",
      "TYR 264 A  13.96   100 %    3.82  649   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 132 A\n",
      "TYR 264 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP 289 A\n",
      "\n",
      "LYS   3 A  10.39     0 %   -0.23  184   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.07 GLU   4 A\n",
      "LYS   3 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU   6 A\n",
      "\n",
      "LYS  23 A   9.00    44 %   -1.88  405   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.03 CYS  19 A\n",
      "LYS  23 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.31 ASP 141 A\n",
      "LYS  23 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 142 A\n",
      "LYS  23 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 CYS 164 A\n",
      "\n",
      "LYS  30 A  10.47     0 %   -0.03   61   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS  32 A  10.45     0 %   -0.05  104   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS  36 A  13.02    75 %   -2.82  492   0.00    0    0.85 GLU  69 A    0.00 XXX   0 X    0.26 ASP  38 A\n",
      "LYS  36 A                                            0.85 GLU  70 A    0.00 XXX   0 X    0.12 ASP  71 A\n",
      "LYS  36 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.17 ASP 129 A\n",
      "LYS  36 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 289 A\n",
      "LYS  36 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.59 ASP  46 A\n",
      "LYS  36 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.20 GLU  69 A\n",
      "LYS  36 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.29 GLU  70 A\n",
      "\n",
      "LYS  59 A  10.02    37 %   -1.79  385   0.00    0    0.85 GLU  56 A    0.00 XXX   0 X    0.10 ASP  10 A\n",
      "LYS  59 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.22 ARG  14 A\n",
      "LYS  59 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.58 GLU  56 A\n",
      "\n",
      "LYS  89 A  10.40     0 %   -0.38  237   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP  96 A\n",
      "LYS  89 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 ASP 101 A\n",
      "LYS  89 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 103 A\n",
      "\n",
      "LYS  91 A  10.94    30 %   -0.88  365   0.00    0    0.85 ASP  21 A    0.00 XXX   0 X    0.00 ASP  13 A\n",
      "LYS  91 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.11 ARG  17 A\n",
      "LYS  91 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 ARG  24 A\n",
      "LYS  91 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.14 GLU  87 A\n",
      "LYS  91 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP  96 A\n",
      "LYS  91 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.49 ASP  21 A\n",
      "\n",
      "LYS 113 A  10.86     0 %   -0.16  251   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.21 ASP  71 A\n",
      "LYS 113 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.29 ASP 114 A\n",
      "LYS 113 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 320 A\n",
      "LYS 113 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 ARG 322 A\n",
      "\n",
      "LYS 121 A  10.58     0 %   -0.06  106   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.14 CYS 119 A\n",
      "\n",
      "LYS 139 A  11.66     0 %   -0.43  269   0.00    0    0.23 CYS 258 A    0.00 XXX   0 X   -0.35 ARG 135 A\n",
      "LYS 139 A                                            0.85 GLU 224 A    0.00 XXX   0 X    0.02 CYS 257 A\n",
      "LYS 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.45 CYS 258 A\n",
      "LYS 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.39 GLU 224 A\n",
      "\n",
      "LYS 191 A  10.20     2 %   -0.45  288   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP  13 A\n",
      "LYS 191 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP 203 A\n",
      "\n",
      "LYS 192 A  11.20     0 %   -0.52  249   0.00    0    0.81 ASP 173 A    0.00 XXX   0 X    0.03 ASP 170 A\n",
      "LYS 192 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 ASP 173 A\n",
      "\n",
      "LYS 217 A  10.32     0 %   -0.21  224   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.04 CYS 212 A\n",
      "LYS 217 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 GLU 250 A\n",
      "\n",
      "LYS 247 A  11.60     0 %   -0.22  164   0.00    0    0.85 GLU 250 A    0.00 XXX   0 X    0.09 ASP 238 A\n",
      "LYS 247 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 GLU 250 A\n",
      "\n",
      "LYS 249 A  10.46     0 %   -0.10  152   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.06 ASP 238 A\n",
      "\n",
      "LYS 263 A  11.10    90 %   -2.93  534   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.04 ARG 135 A\n",
      "LYS 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.40 CYS 221 A\n",
      "LYS 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 GLU 224 A\n",
      "LYS 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.33 CYS 257 A\n",
      "LYS 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 CYS 258 A\n",
      "LYS 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.78 CYS 262 A\n",
      "LYS 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP 289 A\n",
      "LYS 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.70 ASP 132 A\n",
      "\n",
      "LYS 280 A  10.26    15 %   -0.23  322   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.01 ARG 135 A\n",
      "\n",
      "LYS 286 A  10.05    45 %   -0.88  407   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP  38 A\n",
      "LYS 286 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.18 GLU  69 A\n",
      "LYS 286 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 ASP 289 A\n",
      "LYS 286 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 317 A\n",
      "LYS 286 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU 320 A\n",
      "LYS 286 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.11 ARG 322 A\n",
      "LYS 286 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 LYS  36 A\n",
      "\n",
      "LYS 303 A  11.21    13 %   -0.56  318   0.00    0    0.85 GLU 311 A    0.00 XXX   0 X    0.00 ASP 310 A\n",
      "LYS 303 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.42 GLU 311 A\n",
      "\n",
      "ARG  14 A  12.98    32 %   -1.07  370   0.00    0    0.28 ASP  10 A    0.00 XXX   0 X    0.08 ASP  13 A\n",
      "ARG  14 A                                            0.18 GLU  56 A    0.00 XXX   0 X   -0.05 ARG  17 A\n",
      "ARG  14 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.55 ASP  10 A\n",
      "ARG  14 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.52 GLU  56 A\n",
      "\n",
      "ARG  17 A  13.13    26 %   -0.81  355   0.00    0    0.72 ASP  13 A    0.00 XXX   0 X    0.05 ASP  10 A\n",
      "ARG  17 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 CYS  16 A\n",
      "ARG  17 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP  21 A\n",
      "ARG  17 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.59 ASP  13 A\n",
      "\n",
      "ARG  24 A  12.69     0 %   -0.25  263   0.00    0    0.11 ASP  21 A    0.00 XXX   0 X    0.32 ASP  21 A\n",
      "ARG  24 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 GLU  87 A\n",
      "ARG  24 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 141 A\n",
      "\n",
      "ARG  76 A  12.96     1 %   -0.44  283   0.00    0    0.48 ASP 109 A    0.00 XXX   0 X    0.05 ASP  71 A\n",
      "ARG  76 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.37 ASP 109 A\n",
      "\n",
      "ARG 110 A  10.93    57 %   -1.59  442   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 106 A\n",
      "\n",
      "ARG 135 A  14.08    12 %   -0.46  314   0.00    0    0.71 CYS 258 A    0.00 XXX   0 X    0.39 CYS 258 A\n",
      "ARG 135 A                                            0.52 GLU 224 A    0.00 XXX   0 X    0.42 GLU 224 A\n",
      "\n",
      "ARG 206 A  12.17     3 %   -0.59  289   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.06 CYS 186 A\n",
      "ARG 206 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 CYS 207 A\n",
      "ARG 206 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 GLU 299 A\n",
      "\n",
      "ARG 219 A  12.97    38 %   -0.99  388   0.00    0    0.85 ASP 210 A    0.00 XXX   0 X    0.12 CYS 212 A\n",
      "ARG 219 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 CYS 221 A\n",
      "ARG 219 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.43 ASP 210 A\n",
      "\n",
      "ARG 271 A  12.14    41 %   -1.19  395   0.00    0    0.42 ASP 170 A    0.00 XXX   0 X    0.04 CYS 164 A\n",
      "ARG 271 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 CYS 207 A\n",
      "ARG 271 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 219 A\n",
      "ARG 271 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.37 ASP 170 A\n",
      "\n",
      "ARG 278 A  11.64    29 %   -0.86  362   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "ARG 283 A  13.24     0 %   -0.20  243   0.00    0    0.57 C-  352 A    0.00 XXX   0 X    0.01 ASP 306 A\n",
      "ARG 283 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 GLU 308 A\n",
      "ARG 283 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.26 C-  352 A\n",
      "\n",
      "ARG 321 A  11.96    26 %   -1.28  353   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.01 TYR 116 A\n",
      "ARG 321 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 CYS 119 A\n",
      "ARG 321 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.34 TYR 125 A\n",
      "ARG 321 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.24 GLU 315 A\n",
      "ARG 321 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 ASP 317 A\n",
      "\n",
      "ARG 322 A  10.73    72 %   -2.17  484   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU  69 A\n",
      "ARG 322 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP  71 A\n",
      "ARG 322 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 ASP 114 A\n",
      "ARG 322 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP 317 A\n",
      "ARG 322 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 GLU 320 A\n",
      "\n",
      "N+    1 A   7.87     2 %   -0.47  287   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.26 GLU   4 A\n",
      "N+    1 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 CYS 186 A\n",
      "\n",
      "Coupled residues (marked *) were detected.Please rerun PropKa with the --display-coupled-residues \n",
      "or -d option for detailed information.\n",
      "\n",
      "INFO:pKa summary:\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "SUMMARY OF THIS PREDICTION\n",
      "       Group      pKa  model-pKa   ligand atom-type\n",
      "   ASP  10 A     4.95       3.80                      \n",
      "   ASP  13 A     4.41       3.80                      \n",
      "   ASP  21 A     2.92       3.80                      \n",
      "   ASP  38 A     2.45       3.80                      \n",
      "   ASP  46 A     5.88       3.80                      \n",
      "   ASP  71 A     3.56       3.80                      \n",
      "   ASP  96 A     3.95       3.80                      \n",
      "   ASP 101 A     3.94       3.80                      \n",
      "   ASP 103 A     2.46       3.80                      \n",
      "   ASP 109 A     2.64       3.80                      \n",
      "   ASP 114 A     3.91       3.80                      \n",
      "   ASP 129 A    11.71       3.80                      \n",
      "   ASP 132 A     6.29       3.80                      \n",
      "   ASP 141 A     3.75       3.80                      \n",
      "   ASP 154 A     4.07       3.80                      \n",
      "   ASP 170 A     3.21       3.80                      \n",
      "   ASP 173 A     1.41       3.80                      \n",
      "   ASP 203 A     3.82       3.80                      \n",
      "   ASP 210 A     2.58       3.80                      \n",
      "   ASP 238 A     4.04       3.80                      \n",
      "   ASP 245 A     4.00       3.80                      \n",
      "   ASP 289 A     7.94       3.80                      \n",
      "   ASP 306 A     3.95       3.80                      \n",
      "   ASP 310 A     4.33       3.80                      \n",
      "   ASP 317 A     3.26       3.80                      \n",
      "   GLU   4 A     4.10       4.50                      \n",
      "   GLU   6 A     4.46       4.50                      \n",
      "   GLU  31 A     4.62       4.50                      \n",
      "   GLU  35 A     4.73       4.50                      \n",
      "   GLU  56 A     2.83       4.50                      \n",
      "   GLU  69 A     4.06       4.50                      \n",
      "   GLU  70 A     8.36       4.50                      \n",
      "   GLU  87 A     5.73       4.50                      \n",
      "   GLU 106 A     4.79       4.50                      \n",
      "   GLU 142 A     4.62       4.50                      \n",
      "   GLU 156 A     5.01       4.50                      \n",
      "   GLU 224 A     2.07       4.50                      \n",
      "   GLU 229 A     3.42       4.50                      \n",
      "   GLU 244 A     4.57       4.50                      \n",
      "   GLU 250 A     2.96       4.50                      \n",
      "   GLU 299 A     4.11       4.50                      \n",
      "   GLU 308 A     5.24       4.50                      \n",
      "   GLU 311 A     3.29       4.50                      \n",
      "   GLU 315 A     4.29       4.50                      \n",
      "   GLU 316 A     5.33       4.50                      \n",
      "   GLU 320 A     4.89       4.50                      \n",
      "   GLU 343 A     4.70       4.50                      \n",
      "   C-  352 A     2.61       3.20                      \n",
      "   HIS   2 A     6.46       6.50                      \n",
      "   HIS  73 A     6.41       6.50                      \n",
      "   HIS  98 A     6.37       6.50                      \n",
      "   HIS 183 A     6.16       6.50                      \n",
      "   HIS 248 A     6.22       6.50                      \n",
      "   HIS 290 A     3.27       6.50                      \n",
      "   HIS 298 A     5.12       6.50                      \n",
      "   HIS 338 A     4.12       6.50                      \n",
      "   CYS  16 A    14.27       9.00                      \n",
      "   CYS  19 A    13.25       9.00                      \n",
      "   CYS 119 A     9.04       9.00                      \n",
      "   CYS 164 A    12.58       9.00                      \n",
      "   CYS 186 A     9.38       9.00                      \n",
      "   CYS 207 A    11.12       9.00                      \n",
      "   CYS 212 A     9.75       9.00                      \n",
      "   CYS 221 A    11.78       9.00                      \n",
      "   CYS 257 A     6.54       9.00                      \n",
      "   CYS 258 A    10.03       9.00                      \n",
      "   CYS 262 A    17.21       9.00                      \n",
      "   CYS 296 A    12.13       9.00                      \n",
      "   TYR   1 A    10.58      10.00                      \n",
      "   TYR 116 A    10.46      10.00                      \n",
      "   TYR 125 A    12.77      10.00                      \n",
      "   TYR 145 A    16.29      10.00                      \n",
      "   TYR 264 A    13.96      10.00                      \n",
      "   LYS   3 A    10.39      10.50                      \n",
      "   LYS  23 A     9.00      10.50                      \n",
      "   LYS  30 A    10.47      10.50                      \n",
      "   LYS  32 A    10.45      10.50                      \n",
      "   LYS  36 A    13.02      10.50                      \n",
      "   LYS  59 A    10.02      10.50                      \n",
      "   LYS  89 A    10.40      10.50                      \n",
      "   LYS  91 A    10.94      10.50                      \n",
      "   LYS 113 A    10.86      10.50                      \n",
      "   LYS 121 A    10.58      10.50                      \n",
      "   LYS 139 A    11.66      10.50                      \n",
      "   LYS 191 A    10.20      10.50                      \n",
      "   LYS 192 A    11.20      10.50                      \n",
      "   LYS 217 A    10.32      10.50                      \n",
      "   LYS 247 A    11.60      10.50                      \n",
      "   LYS 249 A    10.46      10.50                      \n",
      "   LYS 263 A    11.10      10.50                      \n",
      "   LYS 280 A    10.26      10.50                      \n",
      "   LYS 286 A    10.05      10.50                      \n",
      "   LYS 303 A    11.21      10.50                      \n",
      "   ARG  14 A    12.98      12.50                      \n",
      "   ARG  17 A    13.13      12.50                      \n",
      "   ARG  24 A    12.69      12.50                      \n",
      "   ARG  76 A    12.96      12.50                      \n",
      "   ARG 110 A    10.93      12.50                      \n",
      "   ARG 135 A    14.08      12.50                      \n",
      "   ARG 206 A    12.17      12.50                      \n",
      "   ARG 219 A    12.97      12.50                      \n",
      "   ARG 271 A    12.14      12.50                      \n",
      "   ARG 278 A    11.64      12.50                      \n",
      "   ARG 283 A    13.24      12.50                      \n",
      "   ARG 321 A    11.96      12.50                      \n",
      "   ARG 322 A    10.73      12.50                      \n",
      "   N+    1 A     7.87       8.00                      \n",
      "\n",
      "INFO:PROPKA information:\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "Free energy of   folding (kcal/mol) as a function of pH (using neutral reference)\n",
      "  0.00     34.30\n",
      "  1.00     33.99\n",
      "  2.00     32.12\n",
      "  3.00     25.92\n",
      "  4.00     18.89\n",
      "  5.00     20.28\n",
      "  6.00     24.08\n",
      "  7.00     26.09\n",
      "  8.00     28.36\n",
      "  9.00     32.60\n",
      " 10.00     42.82\n",
      " 11.00     57.76\n",
      " 12.00     71.32\n",
      " 13.00     81.81\n",
      " 14.00     89.66\n",
      "\n",
      "The pH of optimum stability is  4.3 for which the free energy is   18.4 kcal/mol at 298K\n",
      "Could not determine pH values where the free energy is within 80 % of minimum\n",
      "Could not determine the pH-range where the free energy is negative\n",
      "\n",
      "\n",
      "Protein charge of folded and unfolded state as a function of pH\n",
      "    pH  unfolded  folded\n",
      "  0.00     41.99   41.93\n",
      "  0.10     41.99   41.91\n",
      "  0.20     41.99   41.89\n",
      "  0.30     41.99   41.87\n",
      "  0.40     41.99   41.83\n",
      "  0.50     41.98   41.79\n",
      "  0.60     41.98   41.74\n",
      "  0.70     41.97   41.68\n",
      "  0.80     41.97   41.61\n",
      "  0.90     41.96   41.52\n",
      "  1.00     41.95   41.42\n",
      "  1.10     41.93   41.29\n",
      "  1.20     41.92   41.15\n",
      "  1.30     41.89   40.98\n",
      "  1.40     41.87   40.78\n",
      "  1.50     41.83   40.56\n",
      "  1.60     41.79   40.30\n",
      "  1.70     41.74   40.00\n",
      "  1.80     41.67   39.67\n",
      "  1.90     41.59   39.28\n",
      "  2.00     41.48   38.85\n",
      "  2.10     41.35   38.36\n",
      "  2.20     41.19   37.82\n",
      "  2.30     40.98   37.21\n",
      "  2.40     40.73   36.55\n",
      "  2.50     40.42   35.82\n",
      "  2.60     40.04   35.03\n",
      "  2.70     39.58   34.17\n",
      "  2.80     39.01   33.26\n",
      "  2.90     38.33   32.30\n",
      "  3.00     37.52   31.28\n",
      "  3.10     36.55   30.20\n",
      "  3.20     35.43   29.07\n",
      "  3.30     34.13   27.88\n",
      "  3.40     32.64   26.64\n",
      "  3.50     30.98   25.34\n",
      "  3.60     29.14   23.99\n",
      "  3.70     27.15   22.59\n",
      "  3.80     25.03   21.14\n",
      "  3.90     22.80   19.65\n",
      "  4.00     20.50   18.12\n",
      "  4.10     18.16   16.58\n",
      "  4.20     15.82   15.03\n",
      "  4.30     13.52   13.49\n",
      "  4.40     11.28   11.96\n",
      "  4.50      9.13   10.46\n",
      "  4.60      7.10    9.01\n",
      "  4.70      5.21    7.61\n",
      "  4.80      3.48    6.26\n",
      "  4.90      1.93    4.98\n",
      "  5.00      0.54    3.76\n",
      "  5.10     -0.69    2.62\n",
      "  5.20     -1.76    1.54\n",
      "  5.30     -2.70    0.52\n",
      "  5.40     -3.52   -0.43\n",
      "  5.50     -4.24   -1.33\n",
      "  5.60     -4.89   -2.18\n",
      "  5.70     -5.49   -2.99\n",
      "  5.80     -6.05   -3.76\n",
      "  5.90     -6.58   -4.51\n",
      "  6.00     -7.11   -5.23\n",
      "  6.10     -7.64   -5.93\n",
      "  6.20     -8.18   -6.60\n",
      "  6.30     -8.72   -7.24\n",
      "  6.40     -9.26   -7.86\n",
      "  6.50     -9.80   -8.44\n",
      "  6.60    -10.34   -8.98\n",
      "  6.70    -10.85   -9.48\n",
      "  6.80    -11.34   -9.94\n",
      "  6.90    -11.79  -10.36\n",
      "  7.00    -12.21  -10.74\n",
      "  7.10    -12.60  -11.08\n",
      "  7.20    -12.96  -11.39\n",
      "  7.30    -13.29  -11.69\n",
      "  7.40    -13.59  -11.96\n",
      "  7.50    -13.89  -12.23\n",
      "  7.60    -14.18  -12.49\n",
      "  7.70    -14.47  -12.75\n",
      "  7.80    -14.78  -13.01\n",
      "  7.90    -15.10  -13.27\n",
      "  8.00    -15.45  -13.54\n",
      "  8.10    -15.84  -13.82\n",
      "  8.20    -16.27  -14.10\n",
      "  8.30    -16.76  -14.39\n",
      "  8.40    -17.30  -14.69\n",
      "  8.50    -17.91  -15.00\n",
      "  8.60    -18.59  -15.33\n",
      "  8.70    -19.34  -15.67\n",
      "  8.80    -20.15  -16.04\n",
      "  8.90    -21.03  -16.42\n",
      "  9.00    -21.95  -16.84\n",
      "  9.10    -22.92  -17.29\n",
      "  9.20    -23.93  -17.77\n",
      "  9.30    -24.96  -18.30\n",
      "  9.40    -26.02  -18.88\n",
      "  9.50    -27.11  -19.51\n",
      "  9.60    -28.24  -20.20\n",
      "  9.70    -29.41  -20.95\n",
      "  9.80    -30.63  -21.77\n",
      "  9.90    -31.90  -22.67\n",
      " 10.00    -33.24  -23.64\n",
      " 10.10    -34.64  -24.67\n",
      " 10.20    -36.09  -25.77\n",
      " 10.30    -37.57  -26.92\n",
      " 10.40    -39.07  -28.12\n",
      " 10.50    -40.56  -29.34\n",
      " 10.60    -42.01  -30.59\n",
      " 10.70    -43.40  -31.84\n",
      " 10.80    -44.70  -33.09\n",
      " 10.90    -45.91  -34.32\n",
      " 11.00    -47.02  -35.53\n",
      " 11.10    -48.02  -36.72\n",
      " 11.20    -48.92  -37.88\n",
      " 11.30    -49.74  -39.01\n",
      " 11.40    -50.48  -40.11\n",
      " 11.50    -51.17  -41.18\n",
      " 11.60    -51.83  -42.22\n",
      " 11.70    -52.47  -43.23\n",
      " 11.80    -53.11  -44.22\n",
      " 11.90    -53.77  -45.19\n",
      " 12.00    -54.45  -46.13\n",
      " 12.10    -55.16  -47.06\n",
      " 12.20    -55.91  -47.96\n",
      " 12.30    -56.69  -48.85\n",
      " 12.40    -57.48  -49.72\n",
      " 12.50    -58.28  -50.57\n",
      " 12.60    -59.07  -51.42\n",
      " 12.70    -59.83  -52.25\n",
      " 12.80    -60.55  -53.07\n",
      " 12.90    -61.21  -53.86\n",
      " 13.00    -61.81  -54.64\n",
      " 13.10    -62.34  -55.39\n",
      " 13.20    -62.79  -56.10\n",
      " 13.30    -63.19  -56.77\n",
      " 13.40    -63.52  -57.41\n",
      " 13.50    -63.80  -58.00\n",
      " 13.60    -64.03  -58.55\n",
      " 13.70    -64.21  -59.05\n",
      " 13.80    -64.37  -59.52\n",
      " 13.90    -64.49  -59.95\n",
      " 14.00    -64.59  -60.34\n",
      "The pI is  5.35 (folded) and  5.04 (unfolded)\n",
      "\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "INFO:Applying pKa values at a pH of 7.90:\n",
      "INFO:Adding hydrogens to biomolecule.\n",
      "INFO:Debumping biomolecule (again).\n",
      "INFO:Optimizing hydrogen bonds\n",
      "INFO:Applying force field to biomolecule states.\n",
      "INFO:Regenerating headers.\n",
      "INFO:Regenerating PDB lines.\n",
      "WARNING:Ignoring 467 header lines in output.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "pH: 7.9\tname: 74987\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:PDB2PQR v3.6.2: biomolecular structure conversion software.\n",
      "INFO:Please cite:  Jurrus E, et al.  Improvements to the APBS biomolecular solvation software suite.  Protein Sci 27 112-128 (2018).\n",
      "INFO:Please cite:  Dolinsky TJ, et al.  PDB2PQR: expanding and upgrading automated preparation of biomolecular structures for molecular simulations. Nucleic Acids Res 35 W522-W525 (2007).\n",
      "INFO:Checking and transforming input arguments.\n",
      "INFO:Loading topology files.\n",
      "INFO:Loading molecule: /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/1ka1_hal2.pdb\n",
      "INFO:Setting up molecule.\n",
      "INFO:Created biomolecule object with 354 residues and 2718 atoms.\n",
      "WARNING:Multiple occupancies found: OG in SER A 84.\n",
      "WARNING:Multiple occupancies found in SER A 84. At least one of the instances is being ignored.\n",
      "INFO:Setting termini states for biomolecule chains.\n",
      "INFO:Loading forcefield.\n",
      "INFO:Loading hydrogen topology definitions.\n",
      "WARNING:Missing atom CB in residue LYS A 32\n",
      "WARNING:Missing atom CG in residue LYS A 32\n",
      "WARNING:Missing atom CD in residue LYS A 32\n",
      "WARNING:Missing atom CE in residue LYS A 32\n",
      "WARNING:Missing atom NZ in residue LYS A 32\n",
      "WARNING:Missing atom CB in residue ASP A 33\n",
      "WARNING:Missing atom CG in residue ASP A 33\n",
      "WARNING:Missing atom OD1 in residue ASP A 33\n",
      "WARNING:Missing atom OD2 in residue ASP A 33\n",
      "WARNING:Missing atom CG in residue LYS A 189\n",
      "WARNING:Missing atom CD in residue LYS A 189\n",
      "WARNING:Missing atom CE in residue LYS A 189\n",
      "WARNING:Missing atom NZ in residue LYS A 189\n",
      "WARNING:Missing atom CD in residue LYS A 229\n",
      "WARNING:Missing atom CE in residue LYS A 229\n",
      "WARNING:Missing atom NZ in residue LYS A 229\n",
      "WARNING:Missing atom OXT in residue ARG A 355\n",
      "WARNING:Missing atom CB in residue LYS A 32\n",
      "WARNING:Missing atom CG in residue LYS A 32\n",
      "WARNING:Missing atom CD in residue LYS A 32\n",
      "WARNING:Missing atom CE in residue LYS A 32\n",
      "WARNING:Missing atom NZ in residue LYS A 32\n",
      "WARNING:Missing atom CB in residue ASP A 33\n",
      "WARNING:Missing atom CG in residue ASP A 33\n",
      "WARNING:Missing atom OD1 in residue ASP A 33\n",
      "WARNING:Missing atom OD2 in residue ASP A 33\n",
      "WARNING:Missing atom CG in residue LYS A 189\n",
      "WARNING:Missing atom CD in residue LYS A 189\n",
      "WARNING:Missing atom CE in residue LYS A 189\n",
      "WARNING:Missing atom NZ in residue LYS A 189\n",
      "WARNING:Missing atom CD in residue LYS A 229\n",
      "WARNING:Missing atom CE in residue LYS A 229\n",
      "WARNING:Missing atom NZ in residue LYS A 229\n",
      "WARNING:Missing atom OXT in residue ARG A 355\n",
      "INFO:Attempting to repair 17 missing atoms in biomolecule.\n",
      "WARNING:Missing atom CB in residue LYS A 32\n",
      "WARNING:Missing atom CG in residue LYS A 32\n",
      "WARNING:Missing atom CD in residue LYS A 32\n",
      "WARNING:Missing atom CE in residue LYS A 32\n",
      "WARNING:Missing atom NZ in residue LYS A 32\n",
      "WARNING:Missing atom CB in residue ASP A 33\n",
      "WARNING:Missing atom CG in residue ASP A 33\n",
      "WARNING:Missing atom OD1 in residue ASP A 33\n",
      "WARNING:Missing atom OD2 in residue ASP A 33\n",
      "WARNING:Missing atom CG in residue LYS A 189\n",
      "WARNING:Missing atom CD in residue LYS A 189\n",
      "WARNING:Missing atom CE in residue LYS A 189\n",
      "WARNING:Missing atom NZ in residue LYS A 189\n",
      "WARNING:Missing atom CD in residue LYS A 229\n",
      "WARNING:Missing atom CE in residue LYS A 229\n",
      "WARNING:Missing atom NZ in residue LYS A 229\n",
      "WARNING:Missing atom OXT in residue ARG A 355\n",
      "INFO:Added atom CB to residue LYS A 32 at coordinates -45.594, -33.973, -23.265\n",
      "INFO:Added atom CG to residue LYS A 32 at coordinates -45.153, -35.334, -22.744\n",
      "INFO:Added atom CD to residue LYS A 32 at coordinates -43.702, -35.621, -23.091\n",
      "INFO:Added atom CE to residue LYS A 32 at coordinates -43.239, -36.968, -22.584\n",
      "INFO:Added atom NZ to residue LYS A 32 at coordinates -41.819, -37.204, -22.945\n",
      "INFO:Added atom CB to residue ASP A 33 at coordinates -49.706, -30.631, -25.302\n",
      "INFO:Added atom CG to residue ASP A 33 at coordinates -48.707, -31.048, -26.377\n",
      "INFO:Added atom OD1 to residue ASP A 33 at coordinates -47.538, -31.342, -26.061\n",
      "INFO:Added atom OD2 to residue ASP A 33 at coordinates -49.132, -31.087, -27.541\n",
      "INFO:Added atom CG to residue LYS A 189 at coordinates -35.105, -63.964, -9.656\n",
      "INFO:Added atom CD to residue LYS A 189 at coordinates -35.396, -65.193, -8.824\n",
      "INFO:Added atom CE to residue LYS A 189 at coordinates -34.533, -66.375, -9.200\n",
      "INFO:Added atom NZ to residue LYS A 189 at coordinates -34.854, -67.554, -8.358\n",
      "INFO:Added atom CD to residue LYS A 229 at coordinates -3.896, -48.760, -18.063\n",
      "INFO:Added atom CE to residue LYS A 229 at coordinates -3.416, -50.110, -17.576\n",
      "INFO:Added atom NZ to residue LYS A 229 at coordinates -2.176, -50.508, -18.287\n",
      "INFO:Added atom OXT to residue ARG A 355 at coordinates -9.183, -18.959, -17.852\n",
      "INFO:Updating disulfide bridges.\n",
      "INFO:Debumping biomolecule.\n",
      "INFO:Assigning titration states with PROPKA.\n",
      "INFO:\n",
      "propka3.5.1                                                                                  2024-07-26\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "--                                                                           --\n",
      "--  PROPKA: A PROTEIN PKA PREDICTOR                                          --\n",
      "--                                                                           --\n",
      "--  VERSION 1.0,  04/25/2004,  IOWA CITY                                     --\n",
      "--  BY HUI LI                                                                --\n",
      "--                                                                           --\n",
      "--  VERSION 2.0,  11/05/2007, IOWA CITY/COPENHAGEN                           --\n",
      "--  BY DELPHINE C. BAS AND DAVID M. ROGERS                                   --\n",
      "--                                                                           --\n",
      "--  VERSION 3.0,  01/06/2011, COPENHAGEN                                     --\n",
      "--  BY MATS H.M. OLSSON AND CHRESTEN R. SONDERGARD                           --\n",
      "--                                                                           --\n",
      "--  VERSION 3.1,  07/01/2011, COPENHAGEN                                     --\n",
      "--  BY CHRESTEN R. SONDERGARD AND MATS H.M. OLSSON                           --\n",
      "--                                                                           --\n",
      "--  VERSION 3.2,  06/17/2020, PLANET EARTH                                   --\n",
      "--  SEE ABOVE FOR AUTHORS                                                    --\n",
      "--                                                                           --\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "References:\n",
      "\n",
      "Very Fast Empirical Prediction and Rationalization of Protein pKa Values.\n",
      "Hui Li, Andrew D. Robertson and Jan H. Jensen. PROTEINS: Structure, Function,\n",
      "and Bioinformatics. 61:704-721 (2005)\n",
      "\n",
      "Very Fast Prediction and Rationalization of pKa Values for Protein-Ligand\n",
      "Complexes.  Delphine C. Bas, David M. Rogers and Jan H. Jensen.  PROTEINS:\n",
      "Structure, Function, and Bioinformatics 73:765-783 (2008)\n",
      "\n",
      "PROPKA3: Consistent Treatment of Internal and Surface Residues in Empirical\n",
      "pKa predictions.  Mats H.M. Olsson, Chresten R. Sondergard, Michal Rostkowski,\n",
      "and Jan H. Jensen.  Journal of Chemical Theory and Computation, 7(2):525-537\n",
      "(2011)\n",
      "\n",
      "Improved Treatment of Ligands and Coupling Effects in Empirical Calculation\n",
      "and Rationalization of pKa Values.  Chresten R. Sondergaard, Mats H.M. Olsson,\n",
      "Michal Rostkowski, and Jan H. Jensen.  Journal of Chemical Theory and\n",
      "Computation, (2011)\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "\n",
      "WARNING:Missing atoms or failed protonation for HIS  31 A (HIS) -- please check the structure\n",
      "WARNING:Group (HIS) for   231-  CG    31-HIS (A) [ -51.746  -37.773  -23.250] C\n",
      "WARNING:Expected 4 interaction atoms for acids, found:\n",
      "WARNING:                 0- HE2    31-HIS (A) [ -54.068  -39.642  -24.334] H\n",
      "WARNING:               233- ND1    31-HIS (A) [ -51.129  -38.973  -23.536] N\n",
      "WARNING:               235- NE2    31-HIS (A) [ -53.201  -39.225  -24.025] N\n",
      "WARNING:Expected 2 interaction atoms for bases, found:\n",
      "WARNING:               233- ND1    31-HIS (A) [ -51.129  -38.973  -23.536] N\n",
      "WARNING:               235- NE2    31-HIS (A) [ -53.201  -39.225  -24.025] N\n",
      "INFO:Coupling map:\n",
      "\n",
      "INFO:Calculating pKas for Conformation container 1A with 3249 atoms and 890 groups\n",
      "INFO:pKa determinants:\n",
      "\n",
      "---------  -----   ------   ---------------------    --------------    --------------    --------------\n",
      "                            DESOLVATION  EFFECTS       SIDECHAIN          BACKBONE        COULOMBIC\n",
      " RESIDUE    pKa    BURIED     REGULAR      RE        HYDROGEN BOND     HYDROGEN BOND      INTERACTION\n",
      "---------  -----   ------   ---------   ---------    --------------    --------------    --------------\n",
      "\n",
      "ASP  33 A   3.91     0 %    0.20  152   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS  32 A\n",
      "ASP  33 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 HIS  31 A\n",
      "\n",
      "ASP  41 A   2.21    11 %    0.60  312   0.00    0   -0.79 SER  43 A   -0.23 ASP  41 A   -0.28 LYS  39 A\n",
      "ASP  41 A                                            0.00 XXX   0 X   -0.73 SER  43 A   -0.04 LYS 148 A\n",
      "ASP  41 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.12 LYS 291 A\n",
      "ASP  41 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 HIS 241 A\n",
      "\n",
      "ASP  49 A   6.17*  100 %    3.70  606   1.05    0   -0.84 THR  46 A   -0.70 THR  46 A   -0.30 LYS  39 A\n",
      "ASP  49 A                                           -0.85 THR 147 A   -0.24 ASP  49 A    0.55 GLU  72 A\n",
      "\n",
      "ASP  66 A   3.97     0 %    0.19  204   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.11 LYS  61 A\n",
      "ASP  66 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP  67 A\n",
      "\n",
      "ASP  67 A   3.50    45 %    1.80  406   0.24    0   -0.81 TRP 139 A    0.00 XXX   0 X   -0.03 ARG   5 A\n",
      "ASP  67 A                                           -0.70 ARG 137 A    0.00 XXX   0 X   -0.15 LYS  61 A\n",
      "ASP  67 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 LYS  68 A\n",
      "ASP  67 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.57 ARG 137 A\n",
      "\n",
      "ASP  80 A   3.08     0 %    0.24  175   0.00    0   -0.70 ARG 122 A    0.00 XXX   0 X   -0.26 ARG 122 A\n",
      "\n",
      "ASP  94 A   2.91    69 %    1.87  474   0.67    0   -0.85 LYS  16 A    0.00 XXX   0 X   -0.00 ARG  23 A\n",
      "ASP  94 A                                           -0.85 LYS  91 A    0.00 XXX   0 X   -0.00 LYS 116 A\n",
      "ASP  94 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.05 LYS  16 A\n",
      "ASP  94 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.67 LYS  91 A\n",
      "\n",
      "ASP 104 A   3.88     0 %    0.20  138   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.04 LYS 102 A\n",
      "ASP 104 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 LYS 103 A\n",
      "\n",
      "ASP 105 A   4.10     0 %    0.13  159   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG   5 A\n",
      "ASP 105 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS 103 A\n",
      "ASP 105 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.21 ASP 104 A\n",
      "\n",
      "ASP 111 A   3.81     0 %    0.12  140   0.00    0    0.00 XXX   0 X   -0.11 ASP 111 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 120 A   2.75    13 %    0.72  318   0.05    0   -0.32 SER 117 A   -0.79 LYS 116 A   -0.02 LYS 116 A\n",
      "ASP 120 A                                            0.00 XXX   0 X   -0.69 SER 117 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 126 A   3.39    34 %    1.22  376   0.28    0   -0.16 SER  74 A   -0.85 SER  75 A   -0.06 ARG 122 A\n",
      "ASP 126 A                                           -0.85 SER  75 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "ASP 142 A  16.43*  100 %    3.57  642   1.03    0    0.48 GLU  72 A   -0.03 ILE 144 A   -0.28 LYS  39 A\n",
      "ASP 142 A                                            0.67 ASP 145 A    0.00 XXX   0 X    0.50 ASP  49 A\n",
      "ASP 142 A                                            0.37 ASP 294 A    0.00 XXX   0 X    0.17 CYS 140 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.88 CYS 159 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.87 GLU  72 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.59 GLU  73 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.57 ASP 145 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.23 ASP 294 A\n",
      "\n",
      "ASP 145 A   3.62*  100 %    2.82  672   0.55    0   -0.67 ASP 142 A   -0.04 GLY 146 A   -0.19 HIS 241 A\n",
      "ASP 145 A                                           -0.73 ASP 294 A   -0.57 GLN 265 A    0.17 ASP 263 A\n",
      "ASP 145 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 GLU 290 A\n",
      "ASP 145 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.44 HIS 295 A\n",
      "ASP 145 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.65 LYS 267 A\n",
      "ASP 145 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.55 ARG 281 A\n",
      "\n",
      "ASP 165 A   3.15     0 %    0.28  211   0.00    0    0.00 XXX   0 X   -0.81 GLY 136 A   -0.03 LYS  68 A\n",
      "ASP 165 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.00 ARG 134 A\n",
      "ASP 165 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 LYS 135 A\n",
      "\n",
      "ASP 187 A   2.15    10 %    1.13  308   0.05    0   -0.17 SER 181 A   -0.76 LEU 180 A   -0.41 HIS 191 A\n",
      "ASP 187 A                                           -0.85 TYR 196 A   -0.63 SER 181 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 213 A   3.05    18 %    0.79  331   0.14    0    0.00 XXX   0 X   -0.76 GLU 215 A   -0.09 ARG  15 A\n",
      "ASP 213 A                                            0.00 XXX   0 X   -0.84 SER 216 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 227 A   3.07     0 %    0.35  211   0.00    0    0.00 XXX   0 X   -0.84 LYS 229 A   -0.24 LYS 229 A\n",
      "\n",
      "ASP 230 A   3.96     0 %    0.30  181   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.03 ARG 223 A\n",
      "ASP 230 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.20 LYS 229 A\n",
      "ASP 230 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.09 LYS 258 A\n",
      "ASP 230 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP 227 A\n",
      "ASP 230 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 277 A\n",
      "\n",
      "ASP 245 A   3.84     0 %    0.18  234   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.09 LYS 239 A\n",
      "ASP 245 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 HIS 244 A\n",
      "ASP 245 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 GLU 246 A\n",
      "\n",
      "ASP 263 A   3.38    62 %    2.13  455   0.38    0   -1.55 ARG 152 A   -0.09 ASP 263 A   -0.03 LYS 148 A\n",
      "ASP 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.50 LYS 267 A\n",
      "ASP 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.20 HIS 241 A\n",
      "ASP 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.57 ARG 152 A\n",
      "\n",
      "ASP 277 A   3.47    56 %    2.43  438   0.35    0   -1.64 ARG 223 A   -0.78 ILE 232 A   -0.04 LYS 258 A\n",
      "ASP 277 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 HIS 224 A\n",
      "ASP 277 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.61 ARG 223 A\n",
      "\n",
      "ASP 294 A   9.23*  100 %    2.63  612   0.38    0   -0.37 ASP 142 A    0.00 XXX   0 X   -0.04 LYS  39 A\n",
      "ASP 294 A                                            0.73 ASP 145 A    0.00 XXX   0 X   -0.08 LYS 267 A\n",
      "ASP 294 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.35 LYS 291 A\n",
      "ASP 294 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.93 GLU  72 A\n",
      "ASP 294 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.64 ASP 145 A\n",
      "ASP 294 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.87 ARG 281 A\n",
      "ASP 294 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.84 GLU 290 A\n",
      "\n",
      "ASP 311 A   3.81    51 %    2.24  424   0.49    0    0.00 XXX   0 X   -0.12 ASP 311 A   -0.05 HIS 341 A\n",
      "ASP 311 A                                            0.00 XXX   0 X   -0.61 MET 313 A    0.00 XXX   0 X\n",
      "ASP 311 A                                            0.00 XXX   0 X   -0.83 GLU 314 A    0.00 XXX   0 X\n",
      "ASP 311 A                                            0.00 XXX   0 X   -0.40 ASP 315 A    0.00 XXX   0 X\n",
      "ASP 311 A                                            0.00 XXX   0 X   -0.72 VAL 316 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 315 A   6.34*   20 %    0.96  336   0.20    0    0.87 ASP 342 A    0.00 XXX   0 X   -0.04 ARG 338 A\n",
      "ASP 315 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.14 ASP 311 A\n",
      "ASP 315 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.40 ASP 342 A\n",
      "\n",
      "ASP 319 A   1.83    15 %    0.73  324   0.03    0   -1.58 ARG 134 A   -0.76 GLY 321 A   -0.38 ARG 134 A\n",
      "\n",
      "ASP 342 A   3.08*    0 %    0.48  257   0.00    0   -0.87 ASP 315 A    0.00 XXX   0 X   -0.15 ARG 338 A\n",
      "ASP 342 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.18 HIS 341 A\n",
      "\n",
      "ASP 350 A   3.76     0 %    0.21  206   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.25 LYS 253 A\n",
      "\n",
      "GLU   4 A   2.92     0 %    0.28  187   0.00    0   -0.77 N+    2 A   -0.70 GLU   4 A   -0.38 N+    2 A\n",
      "\n",
      "GLU   6 A   3.84    24 %    1.04  349   0.00    0   -0.65 ARG 137 A   -0.66 GLU   6 A   -0.05 N+    2 A\n",
      "GLU   6 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.12 ARG   5 A\n",
      "GLU   6 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU   4 A\n",
      "GLU   6 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.14 ASP  67 A\n",
      "GLU   6 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.43 ARG 137 A\n",
      "\n",
      "GLU  27 A   4.11    13 %    0.87  318   0.20    0   -0.85 SER  26 A   -0.45 GLU  27 A   -0.00 LYS  22 A\n",
      "GLU  27 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.09 ARG  23 A\n",
      "GLU  27 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.07 HIS  31 A\n",
      "\n",
      "GLU  72 A   5.47*  100 %    2.36  562   0.17    0   -0.34 LYS  39 A    0.00 XXX   0 X   -0.16 LYS 291 A\n",
      "GLU  72 A                                           -0.48 ASP 142 A    0.00 XXX   0 X   -0.05 ARG 324 A\n",
      "GLU  72 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.14 ASP  41 A\n",
      "GLU  72 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.42 ASP 145 A\n",
      "GLU  72 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.10 LYS  39 A\n",
      "\n",
      "GLU  73 A   8.38*   90 %    2.95  532   0.08    0   -0.85 LYS  39 A    0.00 XXX   0 X   -0.01 ARG 324 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 ASP  41 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.00 LYS  39 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.09 ASP  49 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.55 GLU  72 A\n",
      "\n",
      "GLU  89 A   5.51    46 %    1.01  409   0.11    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.20 ARG  23 A\n",
      "GLU  89 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.12 GLU  27 A\n",
      "GLU  89 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 HIS  31 A\n",
      "\n",
      "GLU  95 A   3.70     0 %    0.31  175   0.00    0   -0.68 LYS  99 A    0.00 XXX   0 X   -0.05 ARG  23 A\n",
      "GLU  95 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.38 LYS  99 A\n",
      "\n",
      "GLU 119 A   4.30     0 %    0.29  205   0.00    0    0.00 XXX   0 X   -0.63 GLU 119 A   -0.10 ARG 122 A\n",
      "GLU 119 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 ASP  80 A\n",
      "GLU 119 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 ASP 120 A\n",
      "\n",
      "GLU 131 A   4.45     1 %    0.36  283   0.01    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.41 ARG 324 A\n",
      "\n",
      "GLU 154 A   4.04     2 %    0.35  287   0.01    0   -0.72 ARG 152 A    0.00 XXX   0 X    0.02 GLU 238 A\n",
      "GLU 154 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.18 ASP 263 A\n",
      "GLU 154 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.31 ARG 152 A\n",
      "\n",
      "GLU 192 A   3.88     0 %    0.35  255   0.00    0    0.00 XXX   0 X   -0.04 GLY 190 A    0.02 ASP 187 A\n",
      "GLU 192 A                                            0.00 XXX   0 X   -0.73 GLU 192 A   -0.23 HIS 191 A\n",
      "\n",
      "GLU 215 A   3.63    10 %    0.43  309   0.07    0   -0.85 LYS 103 A    0.00 XXX   0 X   -0.16 ARG  15 A\n",
      "GLU 215 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.05 ARG 199 A\n",
      "GLU 215 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP 213 A\n",
      "GLU 215 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.41 LYS 103 A\n",
      "\n",
      "GLU 235 A   3.26    68 %    2.28  472   0.12    0   -0.46 THR 233 A    0.00 XXX   0 X   -0.30 HIS 244 A\n",
      "GLU 235 A                                           -0.85 SER 259 A    0.00 XXX   0 X   -0.65 LYS 251 A\n",
      "GLU 235 A                                           -0.85 LYS 251 A    0.00 XXX   0 X   -0.53 HIS 261 A\n",
      "\n",
      "GLU 238 A   3.63    15 %    0.62  323   0.00    0   -0.74 ARG 152 A    0.00 XXX   0 X   -0.35 LYS 148 A\n",
      "GLU 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.30 HIS 241 A\n",
      "GLU 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 HIS 244 A\n",
      "GLU 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.24 ASP 263 A\n",
      "GLU 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.33 ARG 152 A\n",
      "\n",
      "GLU 246 A   2.95    20 %    1.10  337   0.12    0   -0.73 SER 243 A    0.00 XXX   0 X   -0.43 ARG 355 A\n",
      "GLU 246 A                                           -1.62 ARG 355 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "GLU 290 A   1.29    98 %    3.05  556   0.08    0   -0.59 ARG 281 A    0.00 XXX   0 X   -0.00 LYS 267 A\n",
      "GLU 290 A                                           -1.60 HIS 295 A    0.00 XXX   0 X   -0.23 LYS 291 A\n",
      "GLU 290 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 LYS 329 A\n",
      "GLU 290 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.10 HIS 241 A\n",
      "GLU 290 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -2.03 ARG 281 A\n",
      "GLU 290 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.70 HIS 295 A\n",
      "\n",
      "GLU 304 A   3.18    26 %    1.06  354   0.21    0   -0.46 HIS 303 A   -0.84 GLY 203 A   -0.19 ARG 202 A\n",
      "GLU 304 A                                            0.00 XXX   0 X   -0.00 LEU 204 A   -0.05 LYS 219 A\n",
      "GLU 304 A                                            0.00 XXX   0 X   -0.74 GLY 205 A   -0.31 HIS 303 A\n",
      "\n",
      "GLU 314 A   5.06     0 %    0.26  273   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.01 LYS 329 A\n",
      "GLU 314 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.31 ASP 311 A\n",
      "\n",
      "GLU 339 A   4.02     0 %    0.19  164   0.00    0    0.00 XXX   0 X   -0.67 GLU 339 A   -0.07 ARG 338 A\n",
      "GLU 339 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP 342 A\n",
      "\n",
      "C-  355 A   3.34     0 %    0.16  125   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.07 ARG 355 A\n",
      "C-  355 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 246 A\n",
      "\n",
      "HIS  31 A   6.33     0 %   -0.29  196   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.07 GLU  27 A\n",
      "HIS  31 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP  33 A\n",
      "HIS  31 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU  89 A\n",
      "\n",
      "HIS 191 A   6.82     6 %   -0.67  298   0.00    0    0.00 XXX   0 X    0.37 LEU 188 A   -0.02 LYS  22 A\n",
      "HIS 191 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.41 ASP 187 A\n",
      "HIS 191 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.23 GLU 192 A\n",
      "\n",
      "HIS 221 A   5.75    12 %   -0.53  315   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 219 A\n",
      "HIS 221 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.17 HIS 224 A\n",
      "\n",
      "HIS 224 A   6.57    11 %   -0.55  311   0.00    0    0.00 XXX   0 X    0.58 SER 335 A    0.04 ASP 277 A\n",
      "\n",
      "HIS 241 A   4.98    65 %   -1.23  464   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.26 LYS 148 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 ARG 152 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.17 LYS 267 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.56 ARG 281 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP  41 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 ASP 145 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.30 GLU 238 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 HIS 244 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 ASP 263 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 GLU 290 A\n",
      "\n",
      "HIS 244 A   5.80    28 %   -0.91  359   0.00    0    0.00 XXX   0 X    0.04 GLY 236 A   -0.09 LYS 239 A\n",
      "HIS 244 A                                            0.00 XXX   0 X    0.22 VAL 237 A   -0.10 LYS 251 A\n",
      "HIS 244 A                                            0.00 XXX   0 X    0.01 SER 242 A   -0.01 LYS 267 A\n",
      "HIS 244 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.30 GLU 235 A\n",
      "HIS 244 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 238 A\n",
      "HIS 244 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP 245 A\n",
      "HIS 244 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.27 HIS 261 A\n",
      "\n",
      "HIS 261 A   6.60    35 %   -0.93  379   0.00    0    0.00 XXX   0 X    0.85 GLY 236 A   -0.29 LYS 251 A\n",
      "HIS 261 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 267 A\n",
      "HIS 261 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.53 GLU 235 A\n",
      "\n",
      "HIS 295 A   5.60   100 %   -3.33  658   0.00    0    1.60 GLU 290 A    0.00 XXX   0 X   -0.15 LYS 267 A\n",
      "HIS 295 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.16 ARG 281 A\n",
      "HIS 295 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.00 LYS 291 A\n",
      "HIS 295 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.44 ASP 145 A\n",
      "HIS 295 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.70 GLU 290 A\n",
      "\n",
      "HIS 303 A   6.22    34 %   -0.66  377   0.00    0    0.46 GLU 304 A    0.00 XXX   0 X   -0.39 ARG 202 A\n",
      "HIS 303 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.31 GLU 304 A\n",
      "\n",
      "HIS 309 A   2.56   100 %   -3.24  632   0.00    0    0.00 XXX   0 X    0.06 HIS 295 A   -0.74 HIS 295 A\n",
      "HIS 309 A                                            0.00 XXX   0 X    0.23 ILE 332 A   -0.25 HIS 341 A\n",
      "\n",
      "HIS 341 A   4.22    77 %   -2.52  498   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 311 A\n",
      "HIS 341 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.18 ASP 342 A\n",
      "\n",
      "CYS 140 A  12.78   100 %    3.75  624   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.13 ARG 324 A\n",
      "CYS 140 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 ASP 294 A\n",
      "\n",
      "CYS 159 A  15.19   100 %    3.86  700   0.00    0   -0.06 GLN 265 A    0.00 XXX   0 X   -0.03 LYS 267 A\n",
      "CYS 159 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 CYS 140 A\n",
      "CYS 159 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 CYS 172 A\n",
      "CYS 159 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.30 ASP 294 A\n",
      "CYS 159 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.32 ASP 145 A\n",
      "CYS 159 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.53 CYS 175 A\n",
      "\n",
      "CYS 172 A  12.08   100 %    3.66  715   0.00    0   -0.14 THR  11 A    0.00 XXX   0 X   -0.50 ARG 199 A\n",
      "CYS 172 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 CYS 175 A\n",
      "\n",
      "CYS 175 A   9.09*  100 %    3.67  714   0.00    0   -3.60 CYS 269 A    0.00 XXX   0 X    0.02 ASP 145 A\n",
      "\n",
      "CYS 269 A  19.48*  100 %    4.24  712   0.00    0    3.60 CYS 175 A   -0.02 CYS 269 A    0.49 CYS 159 A\n",
      "CYS 269 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 CYS 172 A\n",
      "CYS 269 A                                            0.00 XXX   0 X    0.00 XXX   0 X    2.03 CYS 175 A\n",
      "\n",
      "CYS 349 A   9.57     1 %    0.40  284   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 311 A\n",
      "CYS 349 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 314 A\n",
      "CYS 349 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.11 ASP 350 A\n",
      "\n",
      "TYR  50 A  12.47    60 %    1.83  448   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.23 ASP  49 A\n",
      "TYR  50 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU  72 A\n",
      "TYR  50 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 126 A\n",
      "TYR  50 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.29 LYS  39 A\n",
      "TYR  50 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.60 GLU  73 A\n",
      "\n",
      "TYR  97 A  12.15    89 %    3.48  531   0.00    0   -0.35 ARG  15 A    0.00 XXX   0 X   -1.07 ARG  15 A\n",
      "TYR  97 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP  94 A\n",
      "TYR  97 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 215 A\n",
      "TYR  97 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 TYR 101 A\n",
      "\n",
      "TYR 101 A  10.24    26 %    0.81  354   0.00    0   -0.53 SER 193 A    0.00 XXX   0 X   -0.10 ARG  15 A\n",
      "TYR 101 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 GLU 192 A\n",
      "\n",
      "TYR 130 A  10.10     4 %    0.46  293   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.36 LYS  68 A\n",
      "\n",
      "TYR 183 A  11.75    61 %    1.58  453   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 154 A\n",
      "TYR 183 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 CYS 175 A\n",
      "TYR 183 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP 187 A\n",
      "TYR 183 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 263 A\n",
      "\n",
      "TYR 196 A  13.17    38 %    2.33  389   0.00    0    0.85 ASP 187 A   -0.79 LEU 188 A    0.05 GLU 192 A\n",
      "TYR 196 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.27 TYR 208 A\n",
      "TYR 196 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.48 ASP 187 A\n",
      "\n",
      "TYR 208 A  11.41    46 %    1.34  411   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.07 ASP 187 A\n",
      "\n",
      "TYR 268 A  17.93   100 %    4.86  684   0.00    0    0.77 TYR 279 A    0.00 XXX   0 X    0.17 ASP 145 A\n",
      "TYR 268 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 CYS 159 A\n",
      "TYR 268 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.54 ARG 281 A\n",
      "TYR 268 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.46 GLU 290 A\n",
      "TYR 268 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 ASP 294 A\n",
      "TYR 268 A                                            0.00 XXX   0 X    0.00 XXX   0 X    2.03 TYR 279 A\n",
      "\n",
      "TYR 279 A  12.08   100 %    3.49  678   0.00    0   -0.51 ARG 281 A    0.00 XXX   0 X    0.50 ASP 145 A\n",
      "TYR 279 A                                           -0.77 TYR 268 A    0.00 XXX   0 X    0.01 CYS 159 A\n",
      "TYR 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.75 ARG 281 A\n",
      "TYR 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.36 ASP 294 A\n",
      "TYR 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.76 GLU 290 A\n",
      "\n",
      "TYR 288 A  12.91    68 %    1.44  472   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP  41 A\n",
      "TYR 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.17 GLU  72 A\n",
      "TYR 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP 142 A\n",
      "TYR 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 ASP 145 A\n",
      "TYR 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.51 ARG 281 A\n",
      "TYR 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.80 GLU 290 A\n",
      "TYR 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.85 ASP 294 A\n",
      "\n",
      "LYS  16 A   9.20    87 %   -2.94  524   0.00    0    0.85 ASP  94 A    0.00 XXX   0 X   -0.26 LYS  91 A\n",
      "LYS  16 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.05 ASP  94 A\n",
      "\n",
      "LYS  22 A   9.53    44 %   -0.97  405   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 GLU  27 A\n",
      "\n",
      "LYS  32 A  10.25     0 %   -0.31  236   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.06 ASP  33 A\n",
      "\n",
      "LYS  39 A  13.17    62 %   -1.79  454   0.00    0    0.34 GLU  72 A    0.00 XXX   0 X    0.28 ASP  41 A\n",
      "LYS  39 A                                            0.85 GLU  73 A    0.00 XXX   0 X    0.30 ASP  49 A\n",
      "LYS  39 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.28 ASP 142 A\n",
      "LYS  39 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 294 A\n",
      "LYS  39 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 324 A\n",
      "LYS  39 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.29 TYR  50 A\n",
      "LYS  39 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.10 GLU  72 A\n",
      "LYS  39 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.00 GLU  73 A\n",
      "\n",
      "LYS  61 A   9.38    37 %   -1.38  385   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.11 ASP  66 A\n",
      "LYS  61 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP  67 A\n",
      "\n",
      "LYS  68 A  10.47     6 %   -0.46  298   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP  67 A\n",
      "LYS  68 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 ARG 137 A\n",
      "LYS  68 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 165 A\n",
      "LYS  68 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.36 TYR 130 A\n",
      "\n",
      "LYS  91 A  11.20    36 %   -0.82  382   0.00    0    0.85 ASP  94 A    0.00 XXX   0 X    0.67 ASP  94 A\n",
      "\n",
      "LYS  99 A  11.34     0 %   -0.22  141   0.00    0    0.68 GLU  95 A    0.00 XXX   0 X   -0.00 ARG  23 A\n",
      "LYS  99 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 GLU  95 A\n",
      "\n",
      "LYS 102 A  10.42     0 %   -0.12  190   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 104 A\n",
      "\n",
      "LYS 103 A  11.30     6 %   -0.45  298   0.00    0    0.85 GLU 215 A    0.00 XXX   0 X   -0.09 ARG  15 A\n",
      "LYS 103 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP 104 A\n",
      "LYS 103 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 105 A\n",
      "LYS 103 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 199 A\n",
      "LYS 103 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.41 GLU 215 A\n",
      "\n",
      "LYS 116 A  10.23     0 %   -0.12  213   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP  94 A\n",
      "LYS 116 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 120 A\n",
      "LYS 116 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.18 LYS  91 A\n",
      "\n",
      "LYS 135 A  10.54     0 %   -0.04   82   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP 165 A\n",
      "\n",
      "LYS 148 A  10.50    12 %   -0.36  314   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP  41 A\n",
      "LYS 148 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.05 ARG 152 A\n",
      "LYS 148 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.35 GLU 238 A\n",
      "LYS 148 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 263 A\n",
      "\n",
      "LYS 189 A  10.46     0 %   -0.04  110   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 219 A  10.37     0 %   -0.19  220   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 304 A\n",
      "\n",
      "LYS 226 A  10.46     0 %   -0.04   96   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 229 A  10.85     0 %   -0.10  111   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.24 ASP 227 A\n",
      "LYS 229 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 ASP 230 A\n",
      "\n",
      "LYS 239 A  10.51     0 %   -0.08  150   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP 245 A\n",
      "\n",
      "LYS 251 A  10.84    33 %   -1.16  374   0.00    0    0.85 GLU 235 A    0.00 XXX   0 X    0.65 GLU 235 A\n",
      "\n",
      "LYS 253 A  10.54     0 %   -0.21  219   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.25 ASP 350 A\n",
      "\n",
      "LYS 258 A  10.36     0 %   -0.17  226   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.10 ARG 223 A\n",
      "LYS 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP 230 A\n",
      "LYS 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 277 A\n",
      "\n",
      "LYS 267 A   7.64   100 %   -3.32  608   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.00 ARG 152 A\n",
      "LYS 267 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 CYS 159 A\n",
      "LYS 267 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.50 ASP 263 A\n",
      "LYS 267 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.79 ARG 281 A\n",
      "LYS 267 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 GLU 290 A\n",
      "LYS 267 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP 294 A\n",
      "LYS 267 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.65 ASP 145 A\n",
      "\n",
      "LYS 285 A  10.40     0 %   -0.10  190   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 291 A  10.01    47 %   -1.26  413   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.12 ASP  41 A\n",
      "LYS 291 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 GLU  72 A\n",
      "LYS 291 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 ARG 281 A\n",
      "LYS 291 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.23 GLU 290 A\n",
      "LYS 291 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.35 ASP 294 A\n",
      "LYS 291 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.05 LYS  39 A\n",
      "\n",
      "LYS 329 A   9.59    18 %   -0.97  331   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU 290 A\n",
      "LYS 329 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 314 A\n",
      "LYS 329 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS 291 A\n",
      "\n",
      "ARG   5 A  12.07    10 %   -0.57  309   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.12 GLU   6 A\n",
      "ARG   5 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP  67 A\n",
      "ARG   5 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 105 A\n",
      "ARG   5 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 ARG 137 A\n",
      "\n",
      "ARG  15 A  12.14    68 %   -2.12  473   0.00    0    0.35 TYR  97 A    0.00 XXX   0 X    1.07 TYR  97 A\n",
      "ARG  15 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 TYR 101 A\n",
      "ARG  15 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP 213 A\n",
      "ARG  15 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 GLU 215 A\n",
      "\n",
      "ARG  23 A  11.87    37 %   -0.97  384   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.09 GLU  27 A\n",
      "ARG  23 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 GLU  89 A\n",
      "ARG  23 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP  94 A\n",
      "ARG  23 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU  95 A\n",
      "\n",
      "ARG 122 A  13.16     0 %   -0.45  261   0.00    0    0.70 ASP  80 A    0.00 XXX   0 X    0.10 GLU 119 A\n",
      "ARG 122 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 ASP 126 A\n",
      "ARG 122 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.26 ASP  80 A\n",
      "\n",
      "ARG 134 A  14.14     0 %   -0.33  237   0.00    0    1.58 ASP 319 A    0.00 XXX   0 X    0.00 ASP 165 A\n",
      "ARG 134 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 ASP 319 A\n",
      "\n",
      "ARG 137 A  13.63    35 %   -1.22  380   0.00    0    0.65 GLU   6 A    0.00 XXX   0 X    0.43 GLU   6 A\n",
      "ARG 137 A                                            0.70 ASP  67 A    0.00 XXX   0 X    0.57 ASP  67 A\n",
      "\n",
      "ARG 152 A  15.85    18 %   -0.87  332   0.00    0    0.72 GLU 154 A    0.00 XXX   0 X    0.31 GLU 154 A\n",
      "ARG 152 A                                            0.74 GLU 238 A    0.00 XXX   0 X    0.33 GLU 238 A\n",
      "ARG 152 A                                            1.55 ASP 263 A    0.00 XXX   0 X    0.57 ASP 263 A\n",
      "\n",
      "ARG 199 A  10.80    65 %   -2.25  462   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.50 CYS 172 A\n",
      "ARG 199 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 215 A\n",
      "\n",
      "ARG 202 A  11.55    41 %   -1.14  396   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.19 GLU 304 A\n",
      "\n",
      "ARG 223 A  13.63    34 %   -1.14  377   0.00    0    1.64 ASP 277 A    0.00 XXX   0 X    0.03 ASP 230 A\n",
      "ARG 223 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.61 ASP 277 A\n",
      "\n",
      "ARG 281 A  17.71   100 %   -2.15  598   0.00    0    0.51 TYR 279 A    0.00 XXX   0 X    0.54 TYR 268 A\n",
      "ARG 281 A                                            0.59 GLU 290 A    0.00 XXX   0 X    1.75 TYR 279 A\n",
      "ARG 281 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.51 TYR 288 A\n",
      "ARG 281 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.55 ASP 145 A\n",
      "ARG 281 A                                            0.00 XXX   0 X    0.00 XXX   0 X    2.03 GLU 290 A\n",
      "ARG 281 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.87 ASP 294 A\n",
      "\n",
      "ARG 324 A  11.59    54 %   -1.52  433   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU  72 A\n",
      "ARG 324 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU  73 A\n",
      "ARG 324 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.41 GLU 131 A\n",
      "ARG 324 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 CYS 140 A\n",
      "\n",
      "ARG 338 A  12.69     0 %   -0.07  146   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 315 A\n",
      "ARG 338 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 GLU 339 A\n",
      "ARG 338 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP 342 A\n",
      "\n",
      "ARG 355 A  14.03     6 %   -0.58  299   0.00    0    1.62 GLU 246 A    0.00 XXX   0 X    0.07 C-  355 A\n",
      "ARG 355 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.43 GLU 246 A\n",
      "\n",
      "N+    2 A   8.88     0 %   -0.32  186   0.00    0    0.77 GLU   4 A    0.00 XXX   0 X    0.05 GLU   6 A\n",
      "N+    2 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 GLU   4 A\n",
      "\n",
      "Coupled residues (marked *) were detected.Please rerun PropKa with the --display-coupled-residues \n",
      "or -d option for detailed information.\n",
      "\n",
      "INFO:pKa summary:\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "SUMMARY OF THIS PREDICTION\n",
      "       Group      pKa  model-pKa   ligand atom-type\n",
      "   ASP  33 A     3.91       3.80                      \n",
      "   ASP  41 A     2.21       3.80                      \n",
      "   ASP  49 A     6.17       3.80                      \n",
      "   ASP  66 A     3.97       3.80                      \n",
      "   ASP  67 A     3.50       3.80                      \n",
      "   ASP  80 A     3.08       3.80                      \n",
      "   ASP  94 A     2.91       3.80                      \n",
      "   ASP 104 A     3.88       3.80                      \n",
      "   ASP 105 A     4.10       3.80                      \n",
      "   ASP 111 A     3.81       3.80                      \n",
      "   ASP 120 A     2.75       3.80                      \n",
      "   ASP 126 A     3.39       3.80                      \n",
      "   ASP 142 A    16.43       3.80                      \n",
      "   ASP 145 A     3.62       3.80                      \n",
      "   ASP 165 A     3.15       3.80                      \n",
      "   ASP 187 A     2.15       3.80                      \n",
      "   ASP 213 A     3.05       3.80                      \n",
      "   ASP 227 A     3.07       3.80                      \n",
      "   ASP 230 A     3.96       3.80                      \n",
      "   ASP 245 A     3.84       3.80                      \n",
      "   ASP 263 A     3.38       3.80                      \n",
      "   ASP 277 A     3.47       3.80                      \n",
      "   ASP 294 A     9.23       3.80                      \n",
      "   ASP 311 A     3.81       3.80                      \n",
      "   ASP 315 A     6.34       3.80                      \n",
      "   ASP 319 A     1.83       3.80                      \n",
      "   ASP 342 A     3.08       3.80                      \n",
      "   ASP 350 A     3.76       3.80                      \n",
      "   GLU   4 A     2.92       4.50                      \n",
      "   GLU   6 A     3.84       4.50                      \n",
      "   GLU  27 A     4.11       4.50                      \n",
      "   GLU  72 A     5.47       4.50                      \n",
      "   GLU  73 A     8.38       4.50                      \n",
      "   GLU  89 A     5.51       4.50                      \n",
      "   GLU  95 A     3.70       4.50                      \n",
      "   GLU 119 A     4.30       4.50                      \n",
      "   GLU 131 A     4.45       4.50                      \n",
      "   GLU 154 A     4.04       4.50                      \n",
      "   GLU 192 A     3.88       4.50                      \n",
      "   GLU 215 A     3.63       4.50                      \n",
      "   GLU 235 A     3.26       4.50                      \n",
      "   GLU 238 A     3.63       4.50                      \n",
      "   GLU 246 A     2.95       4.50                      \n",
      "   GLU 290 A     1.29       4.50                      \n",
      "   GLU 304 A     3.18       4.50                      \n",
      "   GLU 314 A     5.06       4.50                      \n",
      "   GLU 339 A     4.02       4.50                      \n",
      "   C-  355 A     3.34       3.20                      \n",
      "   HIS  31 A     6.33       6.50                      \n",
      "   HIS 191 A     6.82       6.50                      \n",
      "   HIS 221 A     5.75       6.50                      \n",
      "   HIS 224 A     6.57       6.50                      \n",
      "   HIS 241 A     4.98       6.50                      \n",
      "   HIS 244 A     5.80       6.50                      \n",
      "   HIS 261 A     6.60       6.50                      \n",
      "   HIS 295 A     5.60       6.50                      \n",
      "   HIS 303 A     6.22       6.50                      \n",
      "   HIS 309 A     2.56       6.50                      \n",
      "   HIS 341 A     4.22       6.50                      \n",
      "   CYS 140 A    12.78       9.00                      \n",
      "   CYS 159 A    15.19       9.00                      \n",
      "   CYS 172 A    12.08       9.00                      \n",
      "   CYS 175 A     9.09       9.00                      \n",
      "   CYS 269 A    19.48       9.00                      \n",
      "   CYS 349 A     9.57       9.00                      \n",
      "   TYR  50 A    12.47      10.00                      \n",
      "   TYR  97 A    12.15      10.00                      \n",
      "   TYR 101 A    10.24      10.00                      \n",
      "   TYR 130 A    10.10      10.00                      \n",
      "   TYR 183 A    11.75      10.00                      \n",
      "   TYR 196 A    13.17      10.00                      \n",
      "   TYR 208 A    11.41      10.00                      \n",
      "   TYR 268 A    17.93      10.00                      \n",
      "   TYR 279 A    12.08      10.00                      \n",
      "   TYR 288 A    12.91      10.00                      \n",
      "   LYS  16 A     9.20      10.50                      \n",
      "   LYS  22 A     9.53      10.50                      \n",
      "   LYS  32 A    10.25      10.50                      \n",
      "   LYS  39 A    13.17      10.50                      \n",
      "   LYS  61 A     9.38      10.50                      \n",
      "   LYS  68 A    10.47      10.50                      \n",
      "   LYS  91 A    11.20      10.50                      \n",
      "   LYS  99 A    11.34      10.50                      \n",
      "   LYS 102 A    10.42      10.50                      \n",
      "   LYS 103 A    11.30      10.50                      \n",
      "   LYS 116 A    10.23      10.50                      \n",
      "   LYS 135 A    10.54      10.50                      \n",
      "   LYS 148 A    10.50      10.50                      \n",
      "   LYS 189 A    10.46      10.50                      \n",
      "   LYS 219 A    10.37      10.50                      \n",
      "   LYS 226 A    10.46      10.50                      \n",
      "   LYS 229 A    10.85      10.50                      \n",
      "   LYS 239 A    10.51      10.50                      \n",
      "   LYS 251 A    10.84      10.50                      \n",
      "   LYS 253 A    10.54      10.50                      \n",
      "   LYS 258 A    10.36      10.50                      \n",
      "   LYS 267 A     7.64      10.50                      \n",
      "   LYS 285 A    10.40      10.50                      \n",
      "   LYS 291 A    10.01      10.50                      \n",
      "   LYS 329 A     9.59      10.50                      \n",
      "   ARG   5 A    12.07      12.50                      \n",
      "   ARG  15 A    12.14      12.50                      \n",
      "   ARG  23 A    11.87      12.50                      \n",
      "   ARG 122 A    13.16      12.50                      \n",
      "   ARG 134 A    14.14      12.50                      \n",
      "   ARG 137 A    13.63      12.50                      \n",
      "   ARG 152 A    15.85      12.50                      \n",
      "   ARG 199 A    10.80      12.50                      \n",
      "   ARG 202 A    11.55      12.50                      \n",
      "   ARG 223 A    13.63      12.50                      \n",
      "   ARG 281 A    17.71      12.50                      \n",
      "   ARG 324 A    11.59      12.50                      \n",
      "   ARG 338 A    12.69      12.50                      \n",
      "   ARG 355 A    14.03      12.50                      \n",
      "   N+    2 A     8.88       8.00                      \n",
      "\n",
      "INFO:PROPKA information:\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "Free energy of   folding (kcal/mol) as a function of pH (using neutral reference)\n",
      "  0.00     44.66\n",
      "  1.00     44.24\n",
      "  2.00     41.82\n",
      "  3.00     33.82\n",
      "  4.00     21.09\n",
      "  5.00     16.07\n",
      "  6.00     17.12\n",
      "  7.00     18.86\n",
      "  8.00     22.13\n",
      "  9.00     26.07\n",
      " 10.00     32.15\n",
      " 11.00     44.82\n",
      " 12.00     59.71\n",
      " 13.00     73.20\n",
      " 14.00     85.66\n",
      "\n",
      "The pH of optimum stability is  5.0 for which the free energy is   16.1 kcal/mol at 298K\n",
      "Could not determine pH values where the free energy is within 80 % of minimum\n",
      "Could not determine the pH-range where the free energy is negative\n",
      "\n",
      "\n",
      "Protein charge of folded and unfolded state as a function of pH\n",
      "    pH  unfolded  folded\n",
      "  0.00     50.99   50.91\n",
      "  0.10     50.99   50.88\n",
      "  0.20     50.99   50.85\n",
      "  0.30     50.99   50.82\n",
      "  0.40     50.99   50.77\n",
      "  0.50     50.98   50.72\n",
      "  0.60     50.98   50.65\n",
      "  0.70     50.97   50.57\n",
      "  0.80     50.96   50.48\n",
      "  0.90     50.96   50.36\n",
      "  1.00     50.94   50.23\n",
      "  1.10     50.93   50.07\n",
      "  1.20     50.91   49.89\n",
      "  1.30     50.89   49.68\n",
      "  1.40     50.86   49.44\n",
      "  1.50     50.82   49.16\n",
      "  1.60     50.78   48.85\n",
      "  1.70     50.72   48.49\n",
      "  1.80     50.65   48.09\n",
      "  1.90     50.56   47.64\n",
      "  2.00     50.44   47.13\n",
      "  2.10     50.30   46.56\n",
      "  2.20     50.13   45.91\n",
      "  2.30     49.91   45.20\n",
      "  2.40     49.64   44.39\n",
      "  2.50     49.31   43.50\n",
      "  2.60     48.90   42.51\n",
      "  2.70     48.40   41.41\n",
      "  2.80     47.80   40.22\n",
      "  2.90     47.07   38.92\n",
      "  3.00     46.20   37.51\n",
      "  3.10     45.17   36.02\n",
      "  3.20     43.97   34.43\n",
      "  3.30     42.58   32.78\n",
      "  3.40     41.01   31.06\n",
      "  3.50     39.25   29.30\n",
      "  3.60     37.31   27.52\n",
      "  3.70     35.23   25.73\n",
      "  3.80     33.02   23.96\n",
      "  3.90     30.72   22.23\n",
      "  4.00     28.37   20.55\n",
      "  4.10     26.01   18.95\n",
      "  4.20     23.67   17.45\n",
      "  4.30     21.38   16.04\n",
      "  4.40     19.18   14.74\n",
      "  4.50     17.10   13.54\n",
      "  4.60     15.14   12.44\n",
      "  4.70     13.34   11.43\n",
      "  4.80     11.70   10.50\n",
      "  4.90     10.22    9.64\n",
      "  5.00      8.90    8.83\n",
      "  5.10      7.74    8.07\n",
      "  5.20      6.71    7.33\n",
      "  5.30      5.81    6.62\n",
      "  5.40      5.00    5.93\n",
      "  5.50      4.27    5.25\n",
      "  5.60      3.60    4.57\n",
      "  5.70      2.97    3.90\n",
      "  5.80      2.35    3.24\n",
      "  5.90      1.73    2.58\n",
      "  6.00      1.10    1.93\n",
      "  6.10      0.45    1.29\n",
      "  6.20     -0.22    0.66\n",
      "  6.30     -0.91    0.05\n",
      "  6.40     -1.61   -0.54\n",
      "  6.50     -2.31   -1.11\n",
      "  6.60     -3.01   -1.64\n",
      "  6.70     -3.68   -2.15\n",
      "  6.80     -4.31   -2.62\n",
      "  6.90     -4.91   -3.05\n",
      "  7.00     -5.45   -3.45\n",
      "  7.10     -5.94   -3.81\n",
      "  7.20     -6.38   -4.14\n",
      "  7.30     -6.78   -4.45\n",
      "  7.40     -7.13   -4.74\n",
      "  7.50     -7.46   -5.01\n",
      "  7.60     -7.76   -5.26\n",
      "  7.70     -8.04   -5.51\n",
      "  7.80     -8.32   -5.76\n",
      "  7.90     -8.60   -6.00\n",
      "  8.00     -8.88   -6.26\n",
      "  8.10     -9.18   -6.52\n",
      "  8.20     -9.50   -6.80\n",
      "  8.30     -9.84   -7.11\n",
      "  8.40    -10.22   -7.43\n",
      "  8.50    -10.65   -7.79\n",
      "  8.60    -11.11   -8.19\n",
      "  8.70    -11.64   -8.63\n",
      "  8.80    -12.21   -9.11\n",
      "  8.90    -12.85   -9.64\n",
      "  9.00    -13.55  -10.22\n",
      "  9.10    -14.32  -10.85\n",
      "  9.20    -15.17  -11.54\n",
      "  9.30    -16.09  -12.29\n",
      "  9.40    -17.10  -13.08\n",
      "  9.50    -18.21  -13.94\n",
      "  9.60    -19.42  -14.85\n",
      "  9.70    -20.76  -15.82\n",
      "  9.80    -22.21  -16.86\n",
      "  9.90    -23.79  -17.95\n",
      " 10.00    -25.49  -19.10\n",
      " 10.10    -27.30  -20.31\n",
      " 10.20    -29.18  -21.57\n",
      " 10.30    -31.13  -22.86\n",
      " 10.40    -33.09  -24.19\n",
      " 10.50    -35.05  -25.53\n",
      " 10.60    -36.95  -26.87\n",
      " 10.70    -38.76  -28.19\n",
      " 10.80    -40.46  -29.49\n",
      " 10.90    -42.03  -30.76\n",
      " 11.00    -43.45  -31.99\n",
      " 11.10    -44.73  -33.18\n",
      " 11.20    -45.88  -34.33\n",
      " 11.30    -46.90  -35.46\n",
      " 11.40    -47.83  -36.54\n",
      " 11.50    -48.67  -37.61\n",
      " 11.60    -49.47  -38.64\n",
      " 11.70    -50.22  -39.65\n",
      " 11.80    -50.97  -40.64\n",
      " 11.90    -51.72  -41.60\n",
      " 12.00    -52.49  -42.54\n",
      " 12.10    -53.29  -43.45\n",
      " 12.20    -54.12  -44.34\n",
      " 12.30    -54.97  -45.20\n",
      " 12.40    -55.84  -46.03\n",
      " 12.50    -56.72  -46.84\n",
      " 12.60    -57.58  -47.62\n",
      " 12.70    -58.41  -48.37\n",
      " 12.80    -59.18  -49.10\n",
      " 12.90    -59.90  -49.80\n",
      " 13.00    -60.55  -50.48\n",
      " 13.10    -61.12  -51.13\n",
      " 13.20    -61.61  -51.75\n",
      " 13.30    -62.04  -52.35\n",
      " 13.40    -62.40  -52.91\n",
      " 13.50    -62.70  -53.45\n",
      " 13.60    -62.95  -53.96\n",
      " 13.70    -63.15  -54.44\n",
      " 13.80    -63.32  -54.89\n",
      " 13.90    -63.45  -55.30\n",
      " 14.00    -63.56  -55.69\n",
      "The pI is  6.31 (folded) and  6.17 (unfolded)\n",
      "\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "INFO:Applying pKa values at a pH of 7.90:\n",
      "INFO:Adding hydrogens to biomolecule.\n",
      "INFO:Debumping biomolecule (again).\n",
      "INFO:Optimizing hydrogen bonds\n",
      "INFO:Applying force field to biomolecule states.\n",
      "INFO:Regenerating headers.\n",
      "INFO:Regenerating PDB lines.\n",
      "WARNING:Ignoring 466 header lines in output.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "pH: 7.9\tname: 1ka1_hal2\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:PDB2PQR v3.6.2: biomolecular structure conversion software.\n",
      "INFO:Please cite:  Jurrus E, et al.  Improvements to the APBS biomolecular solvation software suite.  Protein Sci 27 112-128 (2018).\n",
      "INFO:Please cite:  Dolinsky TJ, et al.  PDB2PQR: expanding and upgrading automated preparation of biomolecular structures for molecular simulations. Nucleic Acids Res 35 W522-W525 (2007).\n",
      "INFO:Checking and transforming input arguments.\n",
      "INFO:Loading topology files.\n",
      "INFO:Loading molecule: /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/AtSAL1_updated.pdb\n",
      "INFO:Setting up molecule.\n",
      "INFO:Created biomolecule object with 352 residues and 2628 atoms.\n",
      "INFO:Setting termini states for biomolecule chains.\n",
      "INFO:Loading forcefield.\n",
      "INFO:Loading hydrogen topology definitions.\n",
      "WARNING:Missing atom OXT in residue ALA A 352\n",
      "WARNING:Missing atom OXT in residue ALA A 352\n",
      "INFO:Attempting to repair 1 missing atoms in biomolecule.\n",
      "WARNING:Missing atom OXT in residue ALA A 352\n",
      "INFO:Added atom OXT to residue ALA A 352 at coordinates -4.698, -13.324, -18.987\n",
      "INFO:Updating disulfide bridges.\n",
      "INFO:Debumping biomolecule.\n",
      "INFO:Assigning titration states with PROPKA.\n",
      "INFO:\n",
      "propka3.5.1                                                                                  2024-07-26\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "--                                                                           --\n",
      "--  PROPKA: A PROTEIN PKA PREDICTOR                                          --\n",
      "--                                                                           --\n",
      "--  VERSION 1.0,  04/25/2004,  IOWA CITY                                     --\n",
      "--  BY HUI LI                                                                --\n",
      "--                                                                           --\n",
      "--  VERSION 2.0,  11/05/2007, IOWA CITY/COPENHAGEN                           --\n",
      "--  BY DELPHINE C. BAS AND DAVID M. ROGERS                                   --\n",
      "--                                                                           --\n",
      "--  VERSION 3.0,  01/06/2011, COPENHAGEN                                     --\n",
      "--  BY MATS H.M. OLSSON AND CHRESTEN R. SONDERGARD                           --\n",
      "--                                                                           --\n",
      "--  VERSION 3.1,  07/01/2011, COPENHAGEN                                     --\n",
      "--  BY CHRESTEN R. SONDERGARD AND MATS H.M. OLSSON                           --\n",
      "--                                                                           --\n",
      "--  VERSION 3.2,  06/17/2020, PLANET EARTH                                   --\n",
      "--  SEE ABOVE FOR AUTHORS                                                    --\n",
      "--                                                                           --\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "References:\n",
      "\n",
      "Very Fast Empirical Prediction and Rationalization of Protein pKa Values.\n",
      "Hui Li, Andrew D. Robertson and Jan H. Jensen. PROTEINS: Structure, Function,\n",
      "and Bioinformatics. 61:704-721 (2005)\n",
      "\n",
      "Very Fast Prediction and Rationalization of pKa Values for Protein-Ligand\n",
      "Complexes.  Delphine C. Bas, David M. Rogers and Jan H. Jensen.  PROTEINS:\n",
      "Structure, Function, and Bioinformatics 73:765-783 (2008)\n",
      "\n",
      "PROPKA3: Consistent Treatment of Internal and Surface Residues in Empirical\n",
      "pKa predictions.  Mats H.M. Olsson, Chresten R. Sondergard, Michal Rostkowski,\n",
      "and Jan H. Jensen.  Journal of Chemical Theory and Computation, 7(2):525-537\n",
      "(2011)\n",
      "\n",
      "Improved Treatment of Ligands and Coupling Effects in Empirical Calculation\n",
      "and Rationalization of pKa Values.  Chresten R. Sondergaard, Mats H.M. Olsson,\n",
      "Michal Rostkowski, and Jan H. Jensen.  Journal of Chemical Theory and\n",
      "Computation, (2011)\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "\n",
      "INFO:Coupling map:\n",
      "\n",
      "INFO:Calculating pKas for Conformation container 1A with 3086 atoms and 869 groups\n",
      "INFO:did not converge in 10 iterations\n",
      "INFO:pKa determinants:\n",
      "\n",
      "---------  -----   ------   ---------------------    --------------    --------------    --------------\n",
      "                            DESOLVATION  EFFECTS       SIDECHAIN          BACKBONE        COULOMBIC\n",
      " RESIDUE    pKa    BURIED     REGULAR      RE        HYDROGEN BOND     HYDROGEN BOND      INTERACTION\n",
      "---------  -----   ------   ---------   ---------    --------------    --------------    --------------\n",
      "\n",
      "ASP   8 A   3.99    18 %    0.58  331   0.18    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.03 N+    1 A\n",
      "ASP   8 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.15 LYS   5 A\n",
      "ASP   8 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.30 LYS  11 A\n",
      "ASP   8 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.12 LYS  12 A\n",
      "ASP   8 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU   6 A\n",
      "\n",
      "ASP  32 A   3.90     0 %    0.12  137   0.00    0    0.00 XXX   0 X   -0.02 VAL  33 A    0.00 XXX   0 X\n",
      "\n",
      "ASP  38 A   3.69     0 %    0.12  180   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.24 LYS  39 A\n",
      "\n",
      "ASP  46 A   7.13    91 %    2.32  536   0.55    0    0.00 XXX   0 X    0.00 XXX   0 X    0.25 GLU  71 A\n",
      "ASP  46 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 GLU  72 A\n",
      "ASP  46 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 137 A\n",
      "\n",
      "ASP  73 A   3.49     0 %    0.30  267   0.00    0    0.00 XXX   0 X   -0.69 GLY  75 A   -0.06 ARG  78 A\n",
      "ASP  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS 317 A\n",
      "ASP  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 ASP 118 A\n",
      "\n",
      "ASP  76 A   4.20     0 %    0.23  268   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.00 LYS  79 A\n",
      "ASP  76 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.14 ASP  73 A\n",
      "ASP  76 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP  80 A\n",
      "\n",
      "ASP  80 A   3.82     0 %    0.31  184   0.00    0   -0.20 SER  82 A   -0.22 ASP  80 A    0.03 ASP  38 A\n",
      "ASP  80 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP  84 A\n",
      "\n",
      "ASP  84 A   3.48     0 %    0.35  226   0.00    0    0.00 XXX   0 X   -0.06 ASP  84 A   -0.25 ARG  88 A\n",
      "ASP  84 A                                            0.00 XXX   0 X   -0.35 THR  85 A    0.00 XXX   0 X\n",
      "\n",
      "ASP  95 A   3.71     0 %    0.32  279   0.00    0    0.00 XXX   0 X   -0.01 ASP  95 A   -0.28 LYS  23 A\n",
      "ASP  95 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.12 LYS  91 A\n",
      "\n",
      "ASP 111 A   3.94     0 %    0.21  211   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.03 ARG  78 A\n",
      "ASP 111 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.14 ARG 115 A\n",
      "ASP 111 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP 112 A\n",
      "\n",
      "ASP 112 A   2.85     0 %    0.34  258   0.00    0   -0.49 SER 109 A   -0.62 SER 109 A   -0.18 ARG 115 A\n",
      "\n",
      "ASP 118 A   3.20    10 %    0.65  308   0.06    0   -0.77 SER  74 A   -0.25 SER  74 A   -0.26 ARG  78 A\n",
      "ASP 118 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 115 A\n",
      "\n",
      "ASP 134 A  13.40*  100 %    3.03  569   0.41    0    1.52 ASP 137 A   -0.26 ILE 136 A   -0.01 LYS 285 A\n",
      "ASP 134 A                                            0.96 ASP 288 A    0.00 XXX   0 X    0.29 GLU  71 A\n",
      "ASP 134 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.68 ASP  46 A\n",
      "ASP 134 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.89 ASP 137 A\n",
      "ASP 134 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.08 ASP 288 A\n",
      "\n",
      "ASP 137 A   2.84*  100 %    2.61  581   0.29    0   -1.52 ASP 134 A   -0.00 GLN 260 A   -0.47 LYS 262 A\n",
      "ASP 137 A                                           -1.46 ASP 288 A    0.00 XXX   0 X    0.05 GLU  71 A\n",
      "ASP 137 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 GLU 284 A\n",
      "ASP 137 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.65 ARG 276 A\n",
      "\n",
      "ASP 146 A   3.60    12 %    0.79  316   0.11    0   -0.79 ARG 144 A    0.00 XXX   0 X   -0.31 ARG 144 A\n",
      "\n",
      "ASP 186 A   3.58     0 %    0.15  116   0.00    0   -0.37 SER 184 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "ASP 206 A   3.75    53 %    1.44  431   0.23    0   -1.15 ARG  19 A    0.00 XXX   0 X   -0.59 ARG  19 A\n",
      "\n",
      "ASP 239 A   3.65     0 %    0.43  269   0.00    0    0.00 XXX   0 X   -0.34 ASP 239 A   -0.04 HIS 235 A\n",
      "ASP 239 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.17 HIS 238 A\n",
      "ASP 239 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 HIS 279 A\n",
      "\n",
      "ASP 258 A   6.14    44 %    1.75  404   0.62    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.00 LYS 140 A\n",
      "ASP 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.34 ARG 144 A\n",
      "ASP 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.05 LYS 262 A\n",
      "ASP 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.34 ASP 146 A\n",
      "ASP 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 232 A\n",
      "\n",
      "ASP 270 A   5.41    76 %    2.89  493   0.45    0   -0.11 SER 174 A   -0.21 ILE 175 A   -0.89 ARG 268 A\n",
      "ASP 270 A                                           -0.51 ARG 268 A   -0.01 ASP 270 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 288 A   7.25*   81 %    2.27  509   0.16    0   -0.96 ASP 134 A    0.00 XXX   0 X   -0.13 LYS 262 A\n",
      "ASP 288 A                                            1.46 ASP 137 A    0.00 XXX   0 X   -0.22 LYS 285 A\n",
      "ASP 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.62 ASP 137 A\n",
      "ASP 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.54 ARG 276 A\n",
      "ASP 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.54 GLU 284 A\n",
      "ASP 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.75 HIS 289 A\n",
      "\n",
      "ASP 305 A   4.25    25 %    1.48  352   0.32    0    0.00 XXX   0 X   -0.17 ASP 305 A   -0.12 LYS 309 A\n",
      "ASP 305 A                                            0.00 XXX   0 X   -0.07 ALA 307 A   -0.04 ARG 341 A\n",
      "ASP 305 A                                            0.00 XXX   0 X   -0.00 GLY 308 A    0.02 ASP 320 A\n",
      "ASP 305 A                                            0.00 XXX   0 X   -0.63 LYS 309 A    0.23 ASP 322 A\n",
      "ASP 305 A                                            0.00 XXX   0 X   -0.58 ASP 322 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 312 A   2.16    11 %    0.68  312   0.01    0   -0.85 SER 314 A   -0.15 SER 314 A   -0.10 LYS 159 A\n",
      "ASP 312 A                                           -0.85 LYS 315 A    0.00 XXX   0 X   -0.38 LYS 315 A\n",
      "\n",
      "ASP 320 A   3.88     0 %    0.22  254   0.00    0    0.00 XXX   0 X   -0.12 ASP 320 A   -0.13 LYS 315 A\n",
      "ASP 320 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.11 ASP 312 A\n",
      "\n",
      "ASP 322 A   3.54    11 %    0.50  311   0.00    0   -0.66 THR 323 A    0.00 XXX   0 X   -0.01 LYS 309 A\n",
      "ASP 322 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.10 ARG 341 A\n",
      "\n",
      "ASP 342 A   4.00     0 %    0.37  242   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.04 LYS 247 A\n",
      "ASP 342 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.05 LYS 338 A\n",
      "ASP 342 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 ARG 341 A\n",
      "\n",
      "GLU   4 A   4.37     0 %    0.15  174   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.19 N+    1 A\n",
      "GLU   4 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.24 LYS   5 A\n",
      "GLU   4 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU   6 A\n",
      "GLU   4 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 ASP   8 A\n",
      "\n",
      "GLU   6 A   3.70    28 %    1.00  361   0.00    0   -0.72 ARG 129 A   -0.65 GLU   6 A   -0.06 LYS   5 A\n",
      "GLU   6 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.37 ARG 129 A\n",
      "\n",
      "GLU  58 A   4.99    32 %    0.70  370   0.10    0   -0.31 CYS 119 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "GLU  60 A   4.70    11 %    0.55  313   0.02    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.40 LYS  12 A\n",
      "GLU  60 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS  59 A\n",
      "GLU  60 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP   8 A\n",
      "GLU  60 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 101 A\n",
      "\n",
      "GLU  64 A   4.73     0 %    0.31  278   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.32 ARG 129 A\n",
      "GLU  64 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 GLU   6 A\n",
      "GLU  64 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 HIS 130 A\n",
      "GLU  64 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 GLU 156 A\n",
      "GLU  64 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 157 A\n",
      "\n",
      "GLU  71 A   5.17    41 %    0.73  396   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.20 LYS 285 A\n",
      "GLU  71 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 317 A\n",
      "GLU  71 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 ASP 288 A\n",
      "\n",
      "GLU  72 A   5.72    30 %    0.59  365   0.07    0   -0.15 TYR  47 A    0.00 XXX   0 X   -0.01 LYS 317 A\n",
      "GLU  72 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.26 GLU  71 A\n",
      "GLU  72 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 ASP  73 A\n",
      "GLU  72 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.28 ASP  76 A\n",
      "\n",
      "GLU  87 A   4.66     0 %    0.21  201   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.08 LYS  91 A\n",
      "GLU  87 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP  84 A\n",
      "\n",
      "GLU 100 A   4.92     0 %    0.40  210   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 101 A\n",
      "\n",
      "GLU 101 A   4.59     0 %    0.15  174   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS  12 A\n",
      "\n",
      "GLU 123 A   4.29     1 %    0.40  284   0.01    0   -0.17 LYS 317 A    0.00 XXX   0 X   -0.45 LYS 317 A\n",
      "\n",
      "GLU 156 A   4.19    14 %    0.82  321   0.09    0   -0.82 ARG 129 A    0.00 XXX   0 X    0.09 GLU   6 A\n",
      "GLU 156 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 HIS 130 A\n",
      "GLU 156 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.45 ARG 129 A\n",
      "\n",
      "GLU 157 A   4.41     0 %    0.30  215   0.00    0    0.00 XXX   0 X   -0.50 GLY 128 A   -0.04 ARG 129 A\n",
      "GLU 157 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.05 HIS 130 A\n",
      "GLU 157 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 GLU 156 A\n",
      "\n",
      "GLU 187 A   4.76     0 %    0.24  215   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 186 A\n",
      "\n",
      "GLU 220 A   4.46     0 %    0.22  175   0.00    0    0.00 XXX   0 X   -0.30 GLU 220 A   -0.08 LYS 331 A\n",
      "GLU 220 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.12 GLU 224 A\n",
      "\n",
      "GLU 223 A   4.77     0 %    0.51  236   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.27 LYS 251 A\n",
      "GLU 223 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 224 A\n",
      "\n",
      "GLU 224 A   3.96     0 %    0.21  180   0.00    0   -0.71 ASN 221 A    0.00 XXX   0 X   -0.04 LYS 251 A\n",
      "\n",
      "GLU 229 A   5.80    55 %    1.19  436   0.40    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.13 ARG 256 A\n",
      "GLU 229 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.09 LYS 262 A\n",
      "GLU 229 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.07 HIS 238 A\n",
      "\n",
      "GLU 232 A   4.44     0 %    0.15  188   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS 140 A\n",
      "GLU 232 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.20 ARG 144 A\n",
      "GLU 232 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.00 ARG 256 A\n",
      "GLU 232 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 146 A\n",
      "\n",
      "GLU 284 A   2.33    71 %    2.47  480   0.16    0   -0.21 ARG 276 A   -0.15 LYS 285 A   -0.19 LYS 262 A\n",
      "GLU 284 A                                           -0.11 TYR 282 A    0.00 XXX   0 X   -0.19 LYS 285 A\n",
      "GLU 284 A                                           -1.60 HIS 289 A    0.00 XXX   0 X   -1.06 ARG 276 A\n",
      "GLU 284 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.29 HIS 289 A\n",
      "\n",
      "GLU 298 A   3.85    28 %    1.05  360   0.21    0    0.00 XXX   0 X   -0.73 GLY 197 A   -0.09 LYS 213 A\n",
      "GLU 298 A                                            0.00 XXX   0 X   -0.33 SER 198 A    0.00 XXX   0 X\n",
      "GLU 298 A                                            0.00 XXX   0 X   -0.75 GLY 199 A    0.00 XXX   0 X\n",
      "\n",
      "GLU 330 A   4.59     0 %    0.16  232   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 331 A\n",
      "\n",
      "GLU 346 A   4.58     0 %    0.39  216   0.00    0   -0.10 GLN 347 A    0.00 XXX   0 X   -0.33 LYS 247 A\n",
      "GLU 346 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 LYS 349 A\n",
      "GLU 346 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.14 ASP 342 A\n",
      "\n",
      "GLU 348 A   4.77     0 %    0.30  273   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.09 LYS 349 A\n",
      "GLU 348 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 239 A\n",
      "GLU 348 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 322 A\n",
      "\n",
      "C-  352 A   3.34     0 %    0.14   58   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "HIS 130 A   6.21    28 %   -0.99  361   0.00    0    0.00 XXX   0 X    0.01 GLY 124 A    0.01 GLU  64 A\n",
      "HIS 130 A                                            0.00 XXX   0 X    0.43 ASN 127 A    0.04 GLU 156 A\n",
      "HIS 130 A                                            0.00 XXX   0 X    0.15 ARG 129 A    0.05 GLU 157 A\n",
      "\n",
      "HIS 235 A   6.45     0 %   -0.18  141   0.00    0    0.00 XXX   0 X    0.10 ALA 234 A    0.04 ASP 239 A\n",
      "\n",
      "HIS 238 A   6.25     0 %   -0.30  241   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.00 ARG 256 A\n",
      "HIS 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 GLU 229 A\n",
      "HIS 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.19 HIS 235 A\n",
      "HIS 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.17 ASP 239 A\n",
      "\n",
      "HIS 279 A   6.15     0 %   -0.30  228   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.07 HIS 235 A\n",
      "HIS 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 239 A\n",
      "\n",
      "HIS 289 A   5.35   100 %   -3.14  578   0.00    0    1.60 GLU 284 A    0.00 XXX   0 X   -0.40 LYS 262 A\n",
      "HIS 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.23 ARG 276 A\n",
      "HIS 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 LYS 285 A\n",
      "HIS 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.29 GLU 284 A\n",
      "HIS 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.75 ASP 288 A\n",
      "\n",
      "CYS  21 A  12.37   100 %    3.39  590   0.00    0   -0.13 SER  49 A   -0.12 CYS  21 A    0.23 ASP  46 A\n",
      "\n",
      "CYS 119 A  10.39    24 %    0.60  349   0.00    0    0.31 GLU  58 A    0.00 XXX   0 X   -0.02 ARG 115 A\n",
      "CYS 119 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 118 A\n",
      "CYS 119 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.46 GLU  58 A\n",
      "\n",
      "CYS 167 A  13.04   100 %    3.60  645   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.29 ARG 268 A\n",
      "CYS 167 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.73 CYS 190 A\n",
      "\n",
      "CYS 190 A  10.90    74 %    2.85  488   0.00    0   -0.22 ARG 268 A   -0.02 GLY 189 A   -0.84 ARG 268 A\n",
      "CYS 190 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 187 A\n",
      "CYS 190 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.11 ASP 270 A\n",
      "\n",
      "TYR   3 A  10.76     8 %    0.53  303   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.10 GLU   6 A\n",
      "TYR   3 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.19 ARG 129 A\n",
      "TYR   3 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.29 GLU 156 A\n",
      "TYR   3 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 157 A\n",
      "\n",
      "TYR  47 A  12.70    45 %    1.22  407   0.00    0    0.15 GLU  72 A    0.00 XXX   0 X    0.26 ASP  46 A\n",
      "TYR  47 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU  71 A\n",
      "TYR  47 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.14 ASP  73 A\n",
      "TYR  47 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.39 ASP  76 A\n",
      "TYR  47 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP 118 A\n",
      "TYR  47 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.51 GLU  72 A\n",
      "\n",
      "TYR 148 A  15.46    84 %    4.04  516   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    1.36 CYS  21 A\n",
      "TYR 148 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 CYS 167 A\n",
      "\n",
      "TYR 201 A  10.14     0 %    0.32  267   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 298 A\n",
      "TYR 201 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.20 LYS 213 A\n",
      "\n",
      "TYR 263 A  15.09*  100 %    3.51  633   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP 137 A\n",
      "TYR 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.83 ARG 276 A\n",
      "TYR 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.46 GLU 284 A\n",
      "TYR 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 ASP 288 A\n",
      "TYR 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.61 TYR 274 A\n",
      "\n",
      "TYR 274 A   9.17*  100 %    3.44  594   0.00    0   -0.85 LYS 262 A    0.00 XXX   0 X    0.22 ASP 137 A\n",
      "TYR 274 A                                           -0.20 ARG 276 A    0.00 XXX   0 X    0.08 GLU 229 A\n",
      "TYR 274 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.42 GLU 284 A\n",
      "TYR 274 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.12 ASP 288 A\n",
      "TYR 274 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -2.03 LYS 262 A\n",
      "TYR 274 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -2.03 ARG 276 A\n",
      "\n",
      "TYR 282 A  12.77    42 %    1.48  398   0.00    0    0.11 GLU 284 A    0.00 XXX   0 X    0.09 GLU  71 A\n",
      "TYR 282 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 134 A\n",
      "TYR 282 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP 137 A\n",
      "TYR 282 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.34 ARG 276 A\n",
      "TYR 282 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.07 ARG 283 A\n",
      "TYR 282 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 TYR 274 A\n",
      "TYR 282 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 TYR 318 A\n",
      "TYR 282 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.71 GLU 284 A\n",
      "TYR 282 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.53 ASP 288 A\n",
      "\n",
      "TYR 318 A   9.59     0 %    0.35  256   0.00    0   -0.13 ARG 283 A    0.00 XXX   0 X    0.06 GLU  71 A\n",
      "TYR 318 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 123 A\n",
      "TYR 318 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.26 LYS 285 A\n",
      "TYR 318 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.09 LYS 317 A\n",
      "TYR 318 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.38 ARG 283 A\n",
      "\n",
      "LYS   5 A  10.80     0 %   -0.15  167   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.24 GLU   4 A\n",
      "LYS   5 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 GLU   6 A\n",
      "LYS   5 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP   8 A\n",
      "\n",
      "LYS  11 A   9.47    50 %   -1.27  420   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.30 ASP   8 A\n",
      "LYS  11 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS  12 A\n",
      "\n",
      "LYS  12 A  10.57    27 %   -0.51  358   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.12 ASP   8 A\n",
      "LYS  12 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.40 GLU  60 A\n",
      "LYS  12 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 GLU 101 A\n",
      "\n",
      "LYS  23 A  10.16    17 %   -0.59  328   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.28 ASP  95 A\n",
      "LYS  23 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 LYS  91 A\n",
      "\n",
      "LYS  26 A  10.23     3 %   -0.27  291   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS  39 A  10.48     0 %   -0.26  206   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.24 ASP  38 A\n",
      "\n",
      "LYS  59 A  10.21     0 %   -0.32  249   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU  60 A\n",
      "\n",
      "LYS  79 A  10.44     0 %   -0.06   88   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP  76 A\n",
      "\n",
      "LYS  91 A  10.52     0 %   -0.17  208   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU  87 A\n",
      "LYS  91 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG  88 A\n",
      "LYS  91 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.12 ASP  95 A\n",
      "\n",
      "LYS 140 A   9.86    17 %   -0.56  329   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.08 ARG 144 A\n",
      "LYS 140 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 232 A\n",
      "LYS 140 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP 258 A\n",
      "LYS 140 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS  39 A\n",
      "\n",
      "LYS 159 A  10.42     0 %   -0.14  238   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP 312 A\n",
      "LYS 159 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 LYS 315 A\n",
      "\n",
      "LYS 180 A  10.43     0 %   -0.12   93   0.00    0    0.06 LYS 182 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 182 A  10.14     0 %   -0.31  138   0.00    0   -0.06 LYS 180 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 208 A  10.45     0 %   -0.05   83   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 213 A  10.19     9 %   -0.60  306   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.09 GLU 298 A\n",
      "LYS 213 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 TYR 201 A\n",
      "\n",
      "LYS 247 A  10.38     0 %   -0.49  262   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 342 A\n",
      "LYS 247 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.33 GLU 346 A\n",
      "\n",
      "LYS 251 A  10.60     0 %   -0.22  135   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.27 GLU 223 A\n",
      "LYS 251 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 GLU 224 A\n",
      "\n",
      "LYS 262 A   9.41   100 %   -2.90  586   0.00    0    0.85 TYR 274 A    0.00 XXX   0 X    0.47 ASP 137 A\n",
      "LYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 GLU 229 A\n",
      "LYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 258 A\n",
      "LYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -2.00 ARG 276 A\n",
      "LYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 GLU 284 A\n",
      "LYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 ASP 288 A\n",
      "LYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    2.03 TYR 274 A\n",
      "\n",
      "LYS 280 A  10.46     0 %   -0.04   81   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 285 A  10.26    17 %   -0.88  329   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.20 GLU  71 A\n",
      "LYS 285 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 134 A\n",
      "LYS 285 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.24 ARG 283 A\n",
      "LYS 285 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 GLU 284 A\n",
      "LYS 285 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.22 ASP 288 A\n",
      "LYS 285 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.26 TYR 318 A\n",
      "\n",
      "LYS 309 A  10.51     0 %   -0.11  165   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.12 ASP 305 A\n",
      "LYS 309 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 322 A\n",
      "LYS 309 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 ARG 341 A\n",
      "\n",
      "LYS 315 A  11.51     0 %   -0.35  220   0.00    0    0.85 ASP 312 A    0.00 XXX   0 X    0.13 ASP 320 A\n",
      "LYS 315 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 ASP 312 A\n",
      "\n",
      "LYS 317 A  10.13    35 %   -1.18  378   0.00    0    0.17 GLU 123 A    0.00 XXX   0 X    0.06 GLU  71 A\n",
      "LYS 317 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU  72 A\n",
      "LYS 317 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP  73 A\n",
      "LYS 317 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 TYR 318 A\n",
      "LYS 317 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.45 GLU 123 A\n",
      "\n",
      "LYS 331 A  10.52     0 %   -0.12  161   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU 220 A\n",
      "LYS 331 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 GLU 330 A\n",
      "\n",
      "LYS 338 A  10.23     0 %   -0.27  219   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.05 ARG 341 A\n",
      "LYS 338 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 342 A\n",
      "\n",
      "LYS 349 A  10.31     0 %   -0.30  147   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 346 A\n",
      "LYS 349 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 GLU 348 A\n",
      "\n",
      "ARG  19 A  13.45    31 %   -0.78  369   0.00    0    1.15 ASP 206 A    0.00 XXX   0 X    0.59 ASP 206 A\n",
      "\n",
      "ARG  78 A  12.45     0 %   -0.37  250   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.06 ASP  73 A\n",
      "ARG  78 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 111 A\n",
      "ARG  78 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.26 ASP 118 A\n",
      "ARG  78 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 ARG 115 A\n",
      "\n",
      "ARG  88 A  12.29    11 %   -0.46  311   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.25 ASP  84 A\n",
      "\n",
      "ARG 115 A  12.63     0 %   -0.24  199   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.14 ASP 111 A\n",
      "ARG 115 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.18 ASP 112 A\n",
      "ARG 115 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 118 A\n",
      "ARG 115 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 CYS 119 A\n",
      "\n",
      "ARG 129 A  14.57    24 %   -0.85  348   0.00    0    0.72 GLU   6 A    0.00 XXX   0 X    0.19 TYR   3 A\n",
      "ARG 129 A                                            0.82 GLU 156 A    0.00 XXX   0 X    0.32 GLU  64 A\n",
      "ARG 129 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 GLU 157 A\n",
      "ARG 129 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.37 GLU   6 A\n",
      "ARG 129 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.45 GLU 156 A\n",
      "\n",
      "ARG 144 A  13.46     3 %   -0.68  291   0.00    0    0.79 ASP 146 A    0.00 XXX   0 X    0.20 GLU 232 A\n",
      "ARG 144 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.34 ASP 258 A\n",
      "ARG 144 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.31 ASP 146 A\n",
      "\n",
      "ARG 256 A  12.33     0 %   -0.30  275   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.13 GLU 229 A\n",
      "ARG 256 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 GLU 232 A\n",
      "\n",
      "ARG 268 A  12.83    75 %   -2.41  490   0.00    0    0.22 CYS 190 A    0.00 XXX   0 X    0.29 CYS 167 A\n",
      "ARG 268 A                                            0.51 ASP 270 A    0.00 XXX   0 X    0.84 CYS 190 A\n",
      "ARG 268 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.89 ASP 270 A\n",
      "\n",
      "ARG 276 A  16.42    98 %   -1.95  557   0.00    0    0.20 TYR 274 A    0.00 XXX   0 X    0.83 TYR 263 A\n",
      "ARG 276 A                                            0.21 GLU 284 A    0.00 XXX   0 X    0.34 TYR 282 A\n",
      "ARG 276 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.65 ASP 137 A\n",
      "ARG 276 A                                            0.00 XXX   0 X    0.00 XXX   0 X    2.03 TYR 274 A\n",
      "ARG 276 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.06 GLU 284 A\n",
      "ARG 276 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.54 ASP 288 A\n",
      "\n",
      "ARG 283 A  12.77     0 %   -0.31  205   0.00    0    0.13 TYR 318 A    0.00 XXX   0 X    0.07 TYR 282 A\n",
      "ARG 283 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 TYR 318 A\n",
      "\n",
      "ARG 341 A  12.40     1 %   -0.32  283   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 305 A\n",
      "ARG 341 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP 322 A\n",
      "ARG 341 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP 342 A\n",
      "\n",
      "N+    1 A   7.94     0 %   -0.28  245   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.19 GLU   4 A\n",
      "N+    1 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP   8 A\n",
      "N+    1 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 LYS  11 A\n",
      "\n",
      "Coupled residues (marked *) were detected.Please rerun PropKa with the --display-coupled-residues \n",
      "or -d option for detailed information.\n",
      "\n",
      "INFO:pKa summary:\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "SUMMARY OF THIS PREDICTION\n",
      "       Group      pKa  model-pKa   ligand atom-type\n",
      "   ASP   8 A     3.99       3.80                      \n",
      "   ASP  32 A     3.90       3.80                      \n",
      "   ASP  38 A     3.69       3.80                      \n",
      "   ASP  46 A     7.13       3.80                      \n",
      "   ASP  73 A     3.49       3.80                      \n",
      "   ASP  76 A     4.20       3.80                      \n",
      "   ASP  80 A     3.82       3.80                      \n",
      "   ASP  84 A     3.48       3.80                      \n",
      "   ASP  95 A     3.71       3.80                      \n",
      "   ASP 111 A     3.94       3.80                      \n",
      "   ASP 112 A     2.85       3.80                      \n",
      "   ASP 118 A     3.20       3.80                      \n",
      "   ASP 134 A    13.40       3.80                      \n",
      "   ASP 137 A     2.84       3.80                      \n",
      "   ASP 146 A     3.60       3.80                      \n",
      "   ASP 186 A     3.58       3.80                      \n",
      "   ASP 206 A     3.75       3.80                      \n",
      "   ASP 239 A     3.65       3.80                      \n",
      "   ASP 258 A     6.14       3.80                      \n",
      "   ASP 270 A     5.41       3.80                      \n",
      "   ASP 288 A     7.25       3.80                      \n",
      "   ASP 305 A     4.25       3.80                      \n",
      "   ASP 312 A     2.16       3.80                      \n",
      "   ASP 320 A     3.88       3.80                      \n",
      "   ASP 322 A     3.54       3.80                      \n",
      "   ASP 342 A     4.00       3.80                      \n",
      "   GLU   4 A     4.37       4.50                      \n",
      "   GLU   6 A     3.70       4.50                      \n",
      "   GLU  58 A     4.99       4.50                      \n",
      "   GLU  60 A     4.70       4.50                      \n",
      "   GLU  64 A     4.73       4.50                      \n",
      "   GLU  71 A     5.17       4.50                      \n",
      "   GLU  72 A     5.72       4.50                      \n",
      "   GLU  87 A     4.66       4.50                      \n",
      "   GLU 100 A     4.92       4.50                      \n",
      "   GLU 101 A     4.59       4.50                      \n",
      "   GLU 123 A     4.29       4.50                      \n",
      "   GLU 156 A     4.19       4.50                      \n",
      "   GLU 157 A     4.41       4.50                      \n",
      "   GLU 187 A     4.76       4.50                      \n",
      "   GLU 220 A     4.46       4.50                      \n",
      "   GLU 223 A     4.77       4.50                      \n",
      "   GLU 224 A     3.96       4.50                      \n",
      "   GLU 229 A     5.80       4.50                      \n",
      "   GLU 232 A     4.44       4.50                      \n",
      "   GLU 284 A     2.33       4.50                      \n",
      "   GLU 298 A     3.85       4.50                      \n",
      "   GLU 330 A     4.59       4.50                      \n",
      "   GLU 346 A     4.58       4.50                      \n",
      "   GLU 348 A     4.77       4.50                      \n",
      "   C-  352 A     3.34       3.20                      \n",
      "   HIS 130 A     6.21       6.50                      \n",
      "   HIS 235 A     6.45       6.50                      \n",
      "   HIS 238 A     6.25       6.50                      \n",
      "   HIS 279 A     6.15       6.50                      \n",
      "   HIS 289 A     5.35       6.50                      \n",
      "   CYS  21 A    12.37       9.00                      \n",
      "   CYS 119 A    10.39       9.00                      \n",
      "   CYS 167 A    13.04       9.00                      \n",
      "   CYS 190 A    10.90       9.00                      \n",
      "   TYR   3 A    10.76      10.00                      \n",
      "   TYR  47 A    12.70      10.00                      \n",
      "   TYR 148 A    15.46      10.00                      \n",
      "   TYR 201 A    10.14      10.00                      \n",
      "   TYR 263 A    15.09      10.00                      \n",
      "   TYR 274 A     9.17      10.00                      \n",
      "   TYR 282 A    12.77      10.00                      \n",
      "   TYR 318 A     9.59      10.00                      \n",
      "   LYS   5 A    10.80      10.50                      \n",
      "   LYS  11 A     9.47      10.50                      \n",
      "   LYS  12 A    10.57      10.50                      \n",
      "   LYS  23 A    10.16      10.50                      \n",
      "   LYS  26 A    10.23      10.50                      \n",
      "   LYS  39 A    10.48      10.50                      \n",
      "   LYS  59 A    10.21      10.50                      \n",
      "   LYS  79 A    10.44      10.50                      \n",
      "   LYS  91 A    10.52      10.50                      \n",
      "   LYS 140 A     9.86      10.50                      \n",
      "   LYS 159 A    10.42      10.50                      \n",
      "   LYS 180 A    10.43      10.50                      \n",
      "   LYS 182 A    10.14      10.50                      \n",
      "   LYS 208 A    10.45      10.50                      \n",
      "   LYS 213 A    10.19      10.50                      \n",
      "   LYS 247 A    10.38      10.50                      \n",
      "   LYS 251 A    10.60      10.50                      \n",
      "   LYS 262 A     9.41      10.50                      \n",
      "   LYS 280 A    10.46      10.50                      \n",
      "   LYS 285 A    10.26      10.50                      \n",
      "   LYS 309 A    10.51      10.50                      \n",
      "   LYS 315 A    11.51      10.50                      \n",
      "   LYS 317 A    10.13      10.50                      \n",
      "   LYS 331 A    10.52      10.50                      \n",
      "   LYS 338 A    10.23      10.50                      \n",
      "   LYS 349 A    10.31      10.50                      \n",
      "   ARG  19 A    13.45      12.50                      \n",
      "   ARG  78 A    12.45      12.50                      \n",
      "   ARG  88 A    12.29      12.50                      \n",
      "   ARG 115 A    12.63      12.50                      \n",
      "   ARG 129 A    14.57      12.50                      \n",
      "   ARG 144 A    13.46      12.50                      \n",
      "   ARG 256 A    12.33      12.50                      \n",
      "   ARG 268 A    12.83      12.50                      \n",
      "   ARG 276 A    16.42      12.50                      \n",
      "   ARG 283 A    12.77      12.50                      \n",
      "   ARG 341 A    12.40      12.50                      \n",
      "   N+    1 A     7.94       8.00                      \n",
      "\n",
      "INFO:PROPKA information:\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "Free energy of   folding (kcal/mol) as a function of pH (using neutral reference)\n",
      "  0.00     29.52\n",
      "  1.00     29.44\n",
      "  2.00     28.85\n",
      "  3.00     26.51\n",
      "  4.00     24.84\n",
      "  5.00     30.16\n",
      "  6.00     36.69\n",
      "  7.00     40.13\n",
      "  8.00     42.33\n",
      "  9.00     44.66\n",
      " 10.00     48.46\n",
      " 11.00     55.32\n",
      " 12.00     64.91\n",
      " 13.00     74.71\n",
      " 14.00     82.36\n",
      "\n",
      "The pH of optimum stability is  3.8 for which the free energy is   24.7 kcal/mol at 298K\n",
      "Could not determine pH values where the free energy is within 80 % of minimum\n",
      "Could not determine the pH-range where the free energy is negative\n",
      "\n",
      "\n",
      "Protein charge of folded and unfolded state as a function of pH\n",
      "    pH  unfolded  folded\n",
      "  0.00     42.99   42.98\n",
      "  0.10     42.99   42.98\n",
      "  0.20     42.99   42.97\n",
      "  0.30     42.99   42.96\n",
      "  0.40     42.99   42.95\n",
      "  0.50     42.98   42.94\n",
      "  0.60     42.98   42.92\n",
      "  0.70     42.97   42.90\n",
      "  0.80     42.97   42.88\n",
      "  0.90     42.96   42.85\n",
      "  1.00     42.94   42.81\n",
      "  1.10     42.93   42.76\n",
      "  1.20     42.91   42.71\n",
      "  1.30     42.89   42.64\n",
      "  1.40     42.86   42.55\n",
      "  1.50     42.83   42.44\n",
      "  1.60     42.78   42.32\n",
      "  1.70     42.73   42.16\n",
      "  1.80     42.66   41.98\n",
      "  1.90     42.57   41.77\n",
      "  2.00     42.46   41.52\n",
      "  2.10     42.32   41.23\n",
      "  2.20     42.15   40.90\n",
      "  2.30     41.94   40.52\n",
      "  2.40     41.68   40.08\n",
      "  2.50     41.35   39.59\n",
      "  2.60     40.96   39.03\n",
      "  2.70     40.47   38.40\n",
      "  2.80     39.88   37.69\n",
      "  2.90     39.17   36.89\n",
      "  3.00     38.32   36.00\n",
      "  3.10     37.31   35.00\n",
      "  3.20     36.13   33.89\n",
      "  3.30     34.77   32.68\n",
      "  3.40     33.21   31.35\n",
      "  3.50     31.47   29.91\n",
      "  3.60     29.54   28.38\n",
      "  3.70     27.44   26.75\n",
      "  3.80     25.20   25.06\n",
      "  3.90     22.85   23.30\n",
      "  4.00     20.41   21.51\n",
      "  4.10     17.94   19.69\n",
      "  4.20     15.46   17.87\n",
      "  4.30     13.00   16.06\n",
      "  4.40     10.61   14.27\n",
      "  4.50      8.32   12.53\n",
      "  4.60      6.16   10.84\n",
      "  4.70      4.14    9.21\n",
      "  4.80      2.30    7.66\n",
      "  4.90      0.64    6.18\n",
      "  5.00     -0.83    4.80\n",
      "  5.10     -2.12    3.51\n",
      "  5.20     -3.24    2.30\n",
      "  5.30     -4.21    1.18\n",
      "  5.40     -5.04    0.13\n",
      "  5.50     -5.76   -0.84\n",
      "  5.60     -6.39   -1.75\n",
      "  5.70     -6.94   -2.60\n",
      "  5.80     -7.44   -3.40\n",
      "  5.90     -7.89   -4.15\n",
      "  6.00     -8.32   -4.86\n",
      "  6.10     -8.72   -5.53\n",
      "  6.20     -9.12   -6.16\n",
      "  6.30     -9.51   -6.75\n",
      "  6.40     -9.89   -7.31\n",
      "  6.50    -10.26   -7.82\n",
      "  6.60    -10.62   -8.31\n",
      "  6.70    -10.96   -8.75\n",
      "  6.80    -11.28   -9.16\n",
      "  6.90    -11.58   -9.54\n",
      "  7.00    -11.85   -9.90\n",
      "  7.10    -12.10  -10.23\n",
      "  7.20    -12.34  -10.54\n",
      "  7.30    -12.55  -10.82\n",
      "  7.40    -12.74  -11.09\n",
      "  7.50    -12.93  -11.35\n",
      "  7.60    -13.11  -11.58\n",
      "  7.70    -13.29  -11.81\n",
      "  7.80    -13.47  -12.02\n",
      "  7.90    -13.66  -12.23\n",
      "  8.00    -13.86  -12.44\n",
      "  8.10    -14.08  -12.64\n",
      "  8.20    -14.31  -12.84\n",
      "  8.30    -14.57  -13.05\n",
      "  8.40    -14.85  -13.27\n",
      "  8.50    -15.17  -13.51\n",
      "  8.60    -15.53  -13.77\n",
      "  8.70    -15.93  -14.06\n",
      "  8.80    -16.37  -14.40\n",
      "  8.90    -16.87  -14.78\n",
      "  9.00    -17.42  -15.22\n",
      "  9.10    -18.04  -15.72\n",
      "  9.20    -18.72  -16.30\n",
      "  9.30    -19.49  -16.96\n",
      "  9.40    -20.34  -17.71\n",
      "  9.50    -21.30  -18.55\n",
      "  9.60    -22.37  -19.50\n",
      "  9.70    -23.56  -20.55\n",
      "  9.80    -24.88  -21.72\n",
      "  9.90    -26.33  -22.98\n",
      " 10.00    -27.91  -24.34\n",
      " 10.10    -29.60  -25.79\n",
      " 10.20    -31.40  -27.31\n",
      " 10.30    -33.26  -28.87\n",
      " 10.40    -35.16  -30.45\n",
      " 10.50    -37.06  -32.01\n",
      " 10.60    -38.92  -33.54\n",
      " 10.70    -40.70  -34.99\n",
      " 10.80    -42.38  -36.37\n",
      " 10.90    -43.92  -37.64\n",
      " 11.00    -45.32  -38.81\n",
      " 11.10    -46.58  -39.88\n",
      " 11.20    -47.70  -40.84\n",
      " 11.30    -48.69  -41.72\n",
      " 11.40    -49.58  -42.51\n",
      " 11.50    -50.38  -43.25\n",
      " 11.60    -51.11  -43.93\n",
      " 11.70    -51.80  -44.59\n",
      " 11.80    -52.46  -45.22\n",
      " 11.90    -53.11  -45.85\n",
      " 12.00    -53.76  -46.49\n",
      " 12.10    -54.43  -47.14\n",
      " 12.20    -55.11  -47.80\n",
      " 12.30    -55.81  -48.49\n",
      " 12.40    -56.51  -49.20\n",
      " 12.50    -57.22  -49.92\n",
      " 12.60    -57.90  -50.65\n",
      " 12.70    -58.56  -51.38\n",
      " 12.80    -59.18  -52.10\n",
      " 12.90    -59.75  -52.81\n",
      " 13.00    -60.27  -53.49\n",
      " 13.10    -60.72  -54.14\n",
      " 13.20    -61.11  -54.75\n",
      " 13.30    -61.45  -55.32\n",
      " 13.40    -61.73  -55.85\n",
      " 13.50    -61.97  -56.34\n",
      " 13.60    -62.17  -56.79\n",
      " 13.70    -62.33  -57.20\n",
      " 13.80    -62.46  -57.56\n",
      " 13.90    -62.57  -57.89\n",
      " 14.00    -62.65  -58.19\n",
      "The pI is  5.41 (folded) and  4.94 (unfolded)\n",
      "\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "INFO:Applying pKa values at a pH of 7.90:\n",
      "INFO:Adding hydrogens to biomolecule.\n",
      "INFO:Debumping biomolecule (again).\n",
      "WARNING:WARNING: Unable to debump LEU A 29\n",
      "INFO:Optimizing hydrogen bonds\n",
      "INFO:Applying force field to biomolecule states.\n",
      "INFO:Regenerating headers.\n",
      "INFO:Regenerating PDB lines.\n",
      "WARNING:Ignoring 467 header lines in output.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "pH: 7.9\tname: AtSAL1_updated\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:PDB2PQR v3.6.2: biomolecular structure conversion software.\n",
      "INFO:Please cite:  Jurrus E, et al.  Improvements to the APBS biomolecular solvation software suite.  Protein Sci 27 112-128 (2018).\n",
      "INFO:Please cite:  Dolinsky TJ, et al.  PDB2PQR: expanding and upgrading automated preparation of biomolecular structures for molecular simulations. Nucleic Acids Res 35 W522-W525 (2007).\n",
      "INFO:Checking and transforming input arguments.\n",
      "INFO:Loading topology files.\n",
      "INFO:Loading molecule: /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/AtAHL.pdb\n",
      "INFO:Setting up molecule.\n",
      "INFO:Created biomolecule object with 368 residues and 5474 atoms.\n",
      "INFO:Setting termini states for biomolecule chains.\n",
      "INFO:Loading forcefield.\n",
      "INFO:Loading hydrogen topology definitions.\n",
      "INFO:This biomolecule is clean.  No repair needed.\n",
      "INFO:Updating disulfide bridges.\n",
      "INFO:Debumping biomolecule.\n",
      "INFO:Assigning titration states with PROPKA.\n",
      "INFO:\n",
      "propka3.5.1                                                                                  2024-07-26\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "--                                                                           --\n",
      "--  PROPKA: A PROTEIN PKA PREDICTOR                                          --\n",
      "--                                                                           --\n",
      "--  VERSION 1.0,  04/25/2004,  IOWA CITY                                     --\n",
      "--  BY HUI LI                                                                --\n",
      "--                                                                           --\n",
      "--  VERSION 2.0,  11/05/2007, IOWA CITY/COPENHAGEN                           --\n",
      "--  BY DELPHINE C. BAS AND DAVID M. ROGERS                                   --\n",
      "--                                                                           --\n",
      "--  VERSION 3.0,  01/06/2011, COPENHAGEN                                     --\n",
      "--  BY MATS H.M. OLSSON AND CHRESTEN R. SONDERGARD                           --\n",
      "--                                                                           --\n",
      "--  VERSION 3.1,  07/01/2011, COPENHAGEN                                     --\n",
      "--  BY CHRESTEN R. SONDERGARD AND MATS H.M. OLSSON                           --\n",
      "--                                                                           --\n",
      "--  VERSION 3.2,  06/17/2020, PLANET EARTH                                   --\n",
      "--  SEE ABOVE FOR AUTHORS                                                    --\n",
      "--                                                                           --\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "References:\n",
      "\n",
      "Very Fast Empirical Prediction and Rationalization of Protein pKa Values.\n",
      "Hui Li, Andrew D. Robertson and Jan H. Jensen. PROTEINS: Structure, Function,\n",
      "and Bioinformatics. 61:704-721 (2005)\n",
      "\n",
      "Very Fast Prediction and Rationalization of pKa Values for Protein-Ligand\n",
      "Complexes.  Delphine C. Bas, David M. Rogers and Jan H. Jensen.  PROTEINS:\n",
      "Structure, Function, and Bioinformatics 73:765-783 (2008)\n",
      "\n",
      "PROPKA3: Consistent Treatment of Internal and Surface Residues in Empirical\n",
      "pKa predictions.  Mats H.M. Olsson, Chresten R. Sondergard, Michal Rostkowski,\n",
      "and Jan H. Jensen.  Journal of Chemical Theory and Computation, 7(2):525-537\n",
      "(2011)\n",
      "\n",
      "Improved Treatment of Ligands and Coupling Effects in Empirical Calculation\n",
      "and Rationalization of pKa Values.  Chresten R. Sondergaard, Mats H.M. Olsson,\n",
      "Michal Rostkowski, and Jan H. Jensen.  Journal of Chemical Theory and\n",
      "Computation, (2011)\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "\n",
      "INFO:Coupling map:\n",
      "\n",
      "INFO:Calculating pKas for Conformation container 1A with 3218 atoms and 894 groups\n",
      "INFO:did not converge in 10 iterations\n",
      "INFO:pKa determinants:\n",
      "\n",
      "---------  -----   ------   ---------------------    --------------    --------------    --------------\n",
      "                            DESOLVATION  EFFECTS       SIDECHAIN          BACKBONE        COULOMBIC\n",
      " RESIDUE    pKa    BURIED     REGULAR      RE        HYDROGEN BOND     HYDROGEN BOND      INTERACTION\n",
      "---------  -----   ------   ---------   ---------    --------------    --------------    --------------\n",
      "\n",
      "ASP   6 A   3.00    28 %    0.95  359   0.08    0   -1.34 ARG  10 A    0.00 XXX   0 X   -0.50 ARG  10 A\n",
      "\n",
      "ASP  38 A   4.10     0 %    0.13  168   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.04 LYS  35 A\n",
      "ASP  38 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 ASP  39 A\n",
      "ASP  38 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP  40 A\n",
      "ASP  38 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP  74 A\n",
      "ASP  38 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 GLU  80 A\n",
      "\n",
      "ASP  39 A   2.55    20 %    0.70  337   0.00    0   -0.85 SER  41 A   -0.18 ASP  39 A   -0.26 LYS  37 A\n",
      "ASP  39 A                                            0.00 XXX   0 X   -0.61 SER  41 A   -0.27 LYS 302 A\n",
      "ASP  39 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.23 GLU  72 A\n",
      "\n",
      "ASP  40 A   4.01     0 %    0.14  138   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP  39 A\n",
      "\n",
      "ASP  47 A   5.77*  100 %    3.69  595   1.08    0   -0.85 THR  44 A   -0.64 THR  44 A    0.41 GLU  72 A\n",
      "ASP  47 A                                           -0.59 THR 144 A   -0.44 ASP  47 A   -0.69 LYS  37 A\n",
      "\n",
      "ASP  64 A   2.40     0 %    0.38  280   0.00    0   -0.30 LYS 120 A    0.00 XXX   0 X   -0.34 LYS 120 A\n",
      "ASP  64 A                                           -0.68 ARG 124 A    0.00 XXX   0 X   -0.46 ARG 124 A\n",
      "\n",
      "ASP  74 A   3.59    27 %    0.81  358   0.01    0    0.00 XXX   0 X   -0.53 GLU  76 A   -0.02 LYS  35 A\n",
      "ASP  74 A                                            0.00 XXX   0 X   -0.39 THR  77 A   -0.11 LYS  37 A\n",
      "ASP  74 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU  80 A\n",
      "\n",
      "ASP  82 A   3.93     0 %    0.12  119   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.07 LYS  35 A\n",
      "ASP  82 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 GLU  80 A\n",
      "\n",
      "ASP 139 A  11.53*  100 %    3.50  634   0.39    0    0.49 ASP 305 A   -0.33 VAL 141 A   -0.41 LYS  37 A\n",
      "ASP 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.21 LYS 279 A\n",
      "ASP 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.50 LYS 302 A\n",
      "ASP 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.40 ASP  47 A\n",
      "ASP 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.37 GLU  73 A\n",
      "ASP 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.29 GLU  72 A\n",
      "ASP 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.51 ASP 142 A\n",
      "ASP 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.21 ASP 305 A\n",
      "\n",
      "ASP 142 A   5.13*  100 %    3.07  665   0.33    0   -0.07 SER 276 A   -0.04 GLY 143 A   -0.34 LYS 293 A\n",
      "ASP 142 A                                            0.00 XXX   0 X   -0.28 MET 277 A   -0.13 LYS 302 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.33 GLU  72 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 249 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.12 GLU 301 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.69 LYS 279 A\n",
      "\n",
      "ASP 151 A   4.73    32 %    0.77  370   0.33    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.15 ARG 149 A\n",
      "ASP 151 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 HIS  27 A\n",
      "\n",
      "ASP 237 A   3.65     0 %    0.27  219   0.00    0    0.00 XXX   0 X   -0.21 ASP 237 A   -0.31 LYS 354 A\n",
      "ASP 237 A                                            0.00 XXX   0 X   -0.00 ASP 238 A    0.10 ASP 238 A\n",
      "\n",
      "ASP 238 A   3.28     0 %    0.44  227   0.00    0    0.00 XXX   0 X   -0.76 VAL 240 A   -0.20 LYS 354 A\n",
      "\n",
      "ASP 287 A   3.73    64 %    2.21  461   0.60    0   -0.54 TYR 175 A    0.00 XXX   0 X   -0.47 LYS 179 A\n",
      "ASP 287 A                                           -0.37 LYS 179 A    0.00 XXX   0 X   -0.85 ARG 285 A\n",
      "ASP 287 A                                           -0.66 ARG 285 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "ASP 305 A   6.18*  100 %    2.94  592   0.45    0   -0.49 ASP 139 A   -0.22 LYS 302 A   -0.07 LYS  37 A\n",
      "ASP 305 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.99 GLU  72 A\n",
      "ASP 305 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.26 ASP 142 A\n",
      "ASP 305 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.66 LYS 279 A\n",
      "ASP 305 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.74 LYS 293 A\n",
      "ASP 305 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.72 GLU 301 A\n",
      "ASP 305 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.89 LYS 302 A\n",
      "ASP 305 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.91 HIS 306 A\n",
      "\n",
      "ASP 322 A   3.09    35 %    1.70  379   0.53    0   -0.75 ARG 326 A   -0.40 GLY 324 A   -0.10 ARG 341 A\n",
      "ASP 322 A                                            0.00 XXX   0 X   -0.03 GLY 325 A    0.28 ASP 340 A\n",
      "ASP 322 A                                            0.00 XXX   0 X   -0.78 ARG 326 A   -0.03 HIS 352 A\n",
      "ASP 322 A                                            0.00 XXX   0 X   -0.72 ASP 340 A   -0.42 ARG 326 A\n",
      "\n",
      "ASP 329 A   2.58    14 %    0.73  320   0.01    0   -0.85 SER 331 A   -0.78 SER 331 A   -0.01 ARG 326 A\n",
      "ASP 329 A                                           -0.01 LYS 332 A    0.00 XXX   0 X   -0.32 LYS 332 A\n",
      "\n",
      "ASP 340 A   2.65    15 %    0.87  323   0.00    0   -1.39 ARG 341 A    0.00 XXX   0 X   -0.19 ARG 326 A\n",
      "ASP 340 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.44 ARG 341 A\n",
      "\n",
      "GLU   2 A   3.46     0 %    0.26  241   0.00    0   -0.85 LYS 208 A    0.00 XXX   0 X   -0.14 N+    1 A\n",
      "GLU   2 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 GLU   4 A\n",
      "GLU   2 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP   6 A\n",
      "GLU   2 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.38 LYS 208 A\n",
      "\n",
      "GLU   4 A   2.86    16 %    0.81  327   0.00    0   -0.85 N+    1 A   -0.68 GLU   4 A   -0.01 LYS 208 A\n",
      "GLU   4 A                                           -0.25 ARG 134 A    0.00 XXX   0 X   -0.41 N+    1 A\n",
      "GLU   4 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.26 ARG 134 A\n",
      "\n",
      "GLU  24 A   4.74     4 %    0.29  293   0.02    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS  21 A\n",
      "GLU  24 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS  25 A\n",
      "GLU  24 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 HIS  27 A\n",
      "GLU  24 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU  96 A\n",
      "\n",
      "GLU  60 A   5.00    13 %    0.50  319   0.03    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.02 LYS 120 A\n",
      "GLU  60 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.07 ARG 124 A\n",
      "GLU  60 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP  64 A\n",
      "GLU  60 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 117 A\n",
      "\n",
      "GLU  72 A   1.96*   90 %    2.13  534   0.00    0   -0.71 TRP 304 A    0.00 XXX   0 X   -1.16 LYS  37 A\n",
      "GLU  72 A                                           -0.85 LYS  37 A    0.00 XXX   0 X   -1.10 LYS 302 A\n",
      "GLU  72 A                                           -0.84 LYS 302 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "GLU  73 A   7.93*   92 %    3.22  539   0.02    0   -0.85 LYS  37 A    0.00 XXX   0 X   -0.23 LYS 302 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP  39 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 ASP  74 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.18 LYS  37 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.24 ASP  47 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.93 GLU  72 A\n",
      "\n",
      "GLU  76 A   4.99     0 %    0.13  203   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP  38 A\n",
      "GLU  76 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.27 ASP  74 A\n",
      "GLU  76 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU  80 A\n",
      "\n",
      "GLU  80 A   3.53     0 %    0.26  214   0.00    0   -0.85 LYS  35 A    0.00 XXX   0 X   -0.38 LYS  35 A\n",
      "\n",
      "GLU  96 A   3.28    17 %    0.51  330   0.04    0   -0.20 LYS  21 A    0.00 XXX   0 X   -0.37 LYS  21 A\n",
      "GLU  96 A                                           -0.76 LYS  25 A    0.00 XXX   0 X   -0.44 LYS  25 A\n",
      "\n",
      "GLU 100 A   4.18    21 %    0.73  339   0.12    0   -0.85 LYS  21 A    0.00 XXX   0 X   -0.01 LYS  25 A\n",
      "GLU 100 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 LYS 108 A\n",
      "GLU 100 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.17 GLU  96 A\n",
      "GLU 100 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.47 LYS  21 A\n",
      "\n",
      "GLU 117 A   3.73     0 %    0.34  255   0.00    0    0.00 XXX   0 X   -0.81 GLY 114 A   -0.26 LYS 111 A\n",
      "GLU 117 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS 120 A\n",
      "\n",
      "GLU 161 A   3.76     0 %    0.38  263   0.00    0   -0.74 ARG 134 A    0.00 XXX   0 X   -0.04 N+    1 A\n",
      "GLU 161 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 LYS 164 A\n",
      "GLU 161 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU   4 A\n",
      "GLU 161 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.38 ARG 134 A\n",
      "\n",
      "GLU 180 A   4.49     0 %    0.11  156   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.12 LYS 178 A\n",
      "\n",
      "GLU 226 A   4.56     0 %    0.06  112   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "GLU 246 A   5.27    37 %    1.07  386   0.02    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.20 ARG 273 A\n",
      "GLU 246 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.12 HIS 255 A\n",
      "\n",
      "GLU 249 A   5.27    46 %    0.93  410   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.10 ARG 149 A\n",
      "GLU 249 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 293 A\n",
      "GLU 249 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP  39 A\n",
      "GLU 249 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 HIS 255 A\n",
      "\n",
      "GLU 289 A   3.91    55 %    1.68  435   0.54    0   -0.85 SER 234 A   -0.79 THR 243 A   -0.18 LYS 179 A\n",
      "GLU 289 A                                           -0.62 THR 243 A   -0.00 GLU 289 A   -0.45 LYS 269 A\n",
      "GLU 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP 287 A\n",
      "\n",
      "GLU 301 A   1.69    92 %    3.27  539   0.00    0   -0.85 LYS 293 A    0.00 XXX   0 X   -0.03 LYS 302 A\n",
      "GLU 301 A                                           -1.60 HIS 306 A    0.00 XXX   0 X   -0.52 LYS 279 A\n",
      "GLU 301 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.62 LYS 293 A\n",
      "GLU 301 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.45 HIS 306 A\n",
      "\n",
      "GLU 314 A   5.47    70 %    1.78  477   0.58    0   -0.74 ARG 209 A    0.00 XXX   0 X    0.16 GLU 315 A\n",
      "GLU 314 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.81 ARG 209 A\n",
      "\n",
      "GLU 315 A   4.02    32 %    1.27  372   0.32    0   -0.15 SER 211 A   -0.81 GLY 210 A   -0.02 LYS 208 A\n",
      "GLU 315 A                                            0.00 XXX   0 X   -0.11 SER 211 A   -0.28 ARG 209 A\n",
      "GLU 315 A                                            0.00 XXX   0 X   -0.70 GLY 212 A    0.00 XXX   0 X\n",
      "\n",
      "GLU 337 A   3.62     0 %    0.36  250   0.00    0   -0.85 LYS 300 A    0.00 XXX   0 X   -0.01 LYS 332 A\n",
      "GLU 337 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.38 LYS 300 A\n",
      "\n",
      "GLU 353 A   3.83     0 %    0.26  212   0.00    0   -0.84 GLN 349 A    0.00 XXX   0 X   -0.01 LYS 354 A\n",
      "GLU 353 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 HIS 352 A\n",
      "\n",
      "GLU 364 A   4.60     0 %    0.12  133   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.01 ARG 341 A\n",
      "\n",
      "C-  368 A   3.32     0 %    0.12   79   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "HIS  13 A   4.57    81 %   -2.69  507   0.00    0    0.82 GLN 217 A    0.00 XXX   0 X   -0.06 ARG  10 A\n",
      "\n",
      "HIS  27 A   6.82     6 %   -0.41  298   0.00    0    0.00 XXX   0 X    0.68 GLN  23 A    0.01 GLU  24 A\n",
      "HIS  27 A                                            0.00 XXX   0 X    0.01 ARG 149 A    0.02 ASP 151 A\n",
      "\n",
      "HIS  33 A   6.06    11 %   -0.44  312   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "HIS 135 A   5.85    35 %   -1.10  379   0.00    0    0.00 XXX   0 X    0.45 LYS 132 A    0.00 XXX   0 X\n",
      "\n",
      "HIS 255 A   6.41    20 %   -0.86  336   0.00    0    0.00 XXX   0 X    0.78 GLU 249 A   -0.09 ARG 250 A\n",
      "HIS 255 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 273 A\n",
      "HIS 255 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 LYS 293 A\n",
      "HIS 255 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.12 GLU 246 A\n",
      "HIS 255 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 249 A\n",
      "\n",
      "HIS 306 A   5.07   100 %   -3.28  641   0.00    0    1.60 GLU 301 A    0.00 XXX   0 X   -1.25 LYS 279 A\n",
      "HIS 306 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.85 LYS 293 A\n",
      "HIS 306 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.45 GLU 301 A\n",
      "HIS 306 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.91 ASP 305 A\n",
      "\n",
      "HIS 352 A   4.72    65 %   -1.90  462   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 322 A\n",
      "HIS 352 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU 353 A\n",
      "\n",
      "CYS  19 A  13.08   100 %    3.78  655   0.00    0    0.00 XXX   0 X   -0.08 CYS  19 A    0.39 ASP  47 A\n",
      "\n",
      "CYS 125 A  11.63    96 %    3.54  549   0.00    0   -0.05 GLN  51 A    0.00 XXX   0 X   -0.23 ARG 124 A\n",
      "CYS 125 A                                           -0.63 SER  55 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "CYS 172 A  13.72   100 %    3.47  673   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.18 ARG 285 A\n",
      "CYS 172 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 ASP 142 A\n",
      "CYS 172 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.35 CYS 203 A\n",
      "\n",
      "CYS 181 A   9.45     5 %    0.55  295   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 178 A\n",
      "CYS 181 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.10 LYS 201 A\n",
      "CYS 181 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 ARG 285 A\n",
      "CYS 181 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.11 GLU 180 A\n",
      "CYS 181 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 287 A\n",
      "\n",
      "CYS 186 A   9.82    31 %    0.88  368   0.00    0    0.00 XXX   0 X   -0.08 CYS 186 A   -0.15 ARG 273 A\n",
      "CYS 186 A                                            0.00 XXX   0 X   -0.02 ASN 187 A    0.19 ASP 151 A\n",
      "\n",
      "CYS 203 A  11.78   100 %    3.38  575   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.65 ARG 285 A\n",
      "CYS 203 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 ASP 287 A\n",
      "\n",
      "CYS 245 A  12.50   100 %    3.40  607   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.11 LYS 279 A\n",
      "CYS 245 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 246 A\n",
      "CYS 245 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 287 A\n",
      "CYS 245 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.12 GLU 289 A\n",
      "\n",
      "CYS 346 A  12.74   100 %    3.74  577   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "TYR 104 A  10.78    56 %    2.22  438   0.00    0   -0.85 SER  17 A   -0.68 GLY 222 A    0.31 GLU 100 A\n",
      "TYR 104 A                                            0.00 XXX   0 X   -0.01 GLY 223 A   -0.21 LYS  21 A\n",
      "\n",
      "TYR 153 A  15.70   100 %    4.01  560   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    1.59 CYS  19 A\n",
      "TYR 153 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU  24 A\n",
      "TYR 153 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 CYS 172 A\n",
      "\n",
      "TYR 175 A  15.46   100 %    3.86  560   0.00    0   -0.21 ARG 285 A    0.00 XXX   0 X    0.40 CYS 172 A\n",
      "TYR 175 A                                            0.54 ASP 287 A    0.00 XXX   0 X    0.54 CYS 203 A\n",
      "TYR 175 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.43 CYS 245 A\n",
      "TYR 175 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.24 ARG 285 A\n",
      "TYR 175 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.15 ASP 287 A\n",
      "\n",
      "TYR 206 A  11.50    48 %    1.46  417   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU   2 A\n",
      "TYR 206 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP   6 A\n",
      "TYR 206 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.00 ARG  10 A\n",
      "\n",
      "TYR 275 A  12.11    30 %    1.75  364   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.17 ARG 149 A\n",
      "TYR 275 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 ASP 151 A\n",
      "TYR 275 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 CYS 186 A\n",
      "TYR 275 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.49 GLU 249 A\n",
      "TYR 275 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 ARG 250 A\n",
      "TYR 275 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 ARG 273 A\n",
      "\n",
      "TYR 280 A  14.55   100 %    3.81  663   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.14 ASP 142 A\n",
      "TYR 280 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.40 GLU 301 A\n",
      "TYR 280 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.12 ASP 305 A\n",
      "TYR 280 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU 314 A\n",
      "\n",
      "TYR 299 A  14.29    74 %    1.49  488   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP  39 A\n",
      "TYR 299 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.23 GLU  72 A\n",
      "TYR 299 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP 139 A\n",
      "TYR 299 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.32 ASP 142 A\n",
      "TYR 299 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU 249 A\n",
      "TYR 299 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 TYR 335 A\n",
      "TYR 299 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.93 GLU 301 A\n",
      "TYR 299 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.05 ASP 305 A\n",
      "\n",
      "TYR 335 A  10.09     6 %    0.28  299   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP  39 A\n",
      "TYR 335 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU  72 A\n",
      "TYR 335 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP 305 A\n",
      "TYR 335 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 337 A\n",
      "TYR 335 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.10 LYS 300 A\n",
      "TYR 335 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.18 LYS 302 A\n",
      "\n",
      "TYR 360 A  10.04     0 %    0.35  261   0.00    0   -0.33 ARG 341 A    0.00 XXX   0 X    0.11 ASP 322 A\n",
      "TYR 360 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.07 ARG 326 A\n",
      "TYR 360 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.28 ASP 340 A\n",
      "TYR 360 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.30 ARG 341 A\n",
      "\n",
      "LYS  21 A  11.63    24 %   -1.00  348   0.00    0    0.20 GLU  96 A    0.00 XXX   0 X    0.03 GLU  24 A\n",
      "LYS  21 A                                            0.85 GLU 100 A    0.00 XXX   0 X    0.21 TYR 104 A\n",
      "LYS  21 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.37 GLU  96 A\n",
      "LYS  21 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.47 GLU 100 A\n",
      "\n",
      "LYS  25 A  11.09    17 %   -0.54  328   0.00    0    0.76 GLU  96 A    0.00 XXX   0 X    0.03 GLU  24 A\n",
      "LYS  25 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 100 A\n",
      "LYS  25 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.11 LYS  21 A\n",
      "LYS  25 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.44 GLU  96 A\n",
      "\n",
      "LYS  35 A  11.28     0 %   -0.58  275   0.00    0    0.85 GLU  80 A    0.00 XXX   0 X    0.04 ASP  38 A\n",
      "LYS  35 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP  74 A\n",
      "LYS  35 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 ASP  82 A\n",
      "LYS  35 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 GLU  80 A\n",
      "\n",
      "LYS  37 A  13.37    74 %   -2.71  488   0.00    0    0.85 GLU  72 A    0.00 XXX   0 X    0.26 ASP  39 A\n",
      "LYS  37 A                                            0.85 GLU  73 A    0.00 XXX   0 X    0.11 ASP  74 A\n",
      "LYS  37 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.41 ASP 139 A\n",
      "LYS  37 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 ASP 305 A\n",
      "LYS  37 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.69 ASP  47 A\n",
      "LYS  37 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.16 GLU  72 A\n",
      "LYS  37 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.18 GLU  73 A\n",
      "\n",
      "LYS 108 A  10.20     0 %   -0.32  220   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 100 A\n",
      "\n",
      "LYS 111 A  10.66     0 %   -0.10  128   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.26 GLU 117 A\n",
      "\n",
      "LYS 120 A  10.80     0 %   -0.19  244   0.00    0    0.30 ASP  64 A    0.00 XXX   0 X    0.02 GLU  60 A\n",
      "LYS 120 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 117 A\n",
      "LYS 120 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.20 ARG 124 A\n",
      "LYS 120 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.34 ASP  64 A\n",
      "\n",
      "LYS 132 A  10.46     0 %   -0.04   93   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 164 A  10.39     0 %   -0.10  198   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 161 A\n",
      "LYS 164 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 209 A\n",
      "\n",
      "LYS 178 A  10.56     0 %   -0.13  165   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.12 GLU 180 A\n",
      "LYS 178 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 CYS 181 A\n",
      "\n",
      "LYS 179 A  10.81    19 %   -0.61  334   0.00    0    0.37 ASP 287 A    0.00 XXX   0 X   -0.11 ARG 285 A\n",
      "LYS 179 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.18 GLU 289 A\n",
      "LYS 179 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.47 ASP 287 A\n",
      "\n",
      "LYS 191 A  10.47     0 %   -0.03   79   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 193 A  10.48     0 %   -0.02   43   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 201 A  10.15    29 %   -0.45  362   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.10 CYS 181 A\n",
      "\n",
      "LYS 208 A  11.45     0 %   -0.31  254   0.00    0    0.85 GLU   2 A    0.00 XXX   0 X    0.01 GLU   4 A\n",
      "LYS 208 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 315 A\n",
      "LYS 208 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 GLU   2 A\n",
      "\n",
      "LYS 232 A  10.41     0 %   -0.09  201   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 269 A  10.30    12 %   -0.57  314   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.45 GLU 289 A\n",
      "LYS 269 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 LYS 179 A\n",
      "\n",
      "LYS 279 A   9.58   100 %   -3.37  661   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.21 ASP 139 A\n",
      "LYS 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.11 CYS 245 A\n",
      "LYS 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.74 LYS 293 A\n",
      "LYS 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.69 ASP 142 A\n",
      "LYS 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.52 GLU 301 A\n",
      "LYS 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.66 ASP 305 A\n",
      "\n",
      "LYS 293 A  10.97    95 %   -3.08  548   0.00    0    0.85 GLU 301 A    0.00 XXX   0 X    0.34 ASP 142 A\n",
      "LYS 293 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 GLU 249 A\n",
      "LYS 293 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 302 A\n",
      "LYS 293 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.62 GLU 301 A\n",
      "LYS 293 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.74 ASP 305 A\n",
      "\n",
      "LYS 300 A  11.52     0 %   -0.32  217   0.00    0    0.85 GLU 337 A    0.00 XXX   0 X    0.10 TYR 335 A\n",
      "LYS 300 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 GLU 337 A\n",
      "\n",
      "LYS 302 A  12.40    70 %   -1.84  477   0.00    0    0.84 GLU  72 A    0.00 XXX   0 X    0.27 ASP  39 A\n",
      "LYS 302 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.23 GLU  73 A\n",
      "LYS 302 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.50 ASP 139 A\n",
      "LYS 302 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 ASP 142 A\n",
      "LYS 302 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 301 A\n",
      "LYS 302 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.44 LYS  37 A\n",
      "LYS 302 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.18 TYR 335 A\n",
      "LYS 302 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.10 GLU  72 A\n",
      "LYS 302 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.89 ASP 305 A\n",
      "\n",
      "LYS 332 A  10.67     0 %   -0.17  202   0.00    0    0.01 ASP 329 A    0.00 XXX   0 X    0.32 ASP 329 A\n",
      "LYS 332 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 337 A\n",
      "\n",
      "LYS 354 A  10.57     0 %   -0.46  252   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.31 ASP 237 A\n",
      "LYS 354 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 ASP 238 A\n",
      "LYS 354 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 353 A\n",
      "\n",
      "ARG  10 A  13.55    29 %   -0.79  363   0.00    0    1.34 ASP   6 A    0.00 XXX   0 X    0.00 TYR 206 A\n",
      "ARG  10 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.50 ASP   6 A\n",
      "\n",
      "ARG 124 A  12.59    43 %   -1.35  401   0.00    0    0.68 ASP  64 A    0.00 XXX   0 X    0.07 GLU  60 A\n",
      "ARG 124 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.23 CYS 125 A\n",
      "ARG 124 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.46 ASP  64 A\n",
      "\n",
      "ARG 134 A  13.76     4 %   -0.37  293   0.00    0    0.25 GLU   4 A    0.00 XXX   0 X    0.26 GLU   4 A\n",
      "ARG 134 A                                            0.74 GLU 161 A    0.00 XXX   0 X    0.38 GLU 161 A\n",
      "\n",
      "ARG 149 A  12.62     0 %   -0.29  273   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP 151 A\n",
      "ARG 149 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 GLU 249 A\n",
      "ARG 149 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.17 TYR 275 A\n",
      "\n",
      "ARG 209 A  12.83    59 %   -1.51  447   0.00    0    0.74 GLU 314 A    0.00 XXX   0 X    0.28 GLU 315 A\n",
      "ARG 209 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.81 GLU 314 A\n",
      "\n",
      "ARG 250 A  12.37     0 %   -0.09  157   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.01 TYR 275 A\n",
      "ARG 250 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.05 ARG 273 A\n",
      "\n",
      "ARG 268 A  12.32     0 %   -0.18  156   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "ARG 273 A  12.69     0 %   -0.19  279   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.15 CYS 186 A\n",
      "ARG 273 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 GLU 246 A\n",
      "ARG 273 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 TYR 275 A\n",
      "\n",
      "ARG 285 A  13.75    70 %   -2.59  478   0.00    0    0.21 TYR 175 A    0.00 XXX   0 X    0.18 CYS 172 A\n",
      "ARG 285 A                                            0.66 ASP 287 A    0.00 XXX   0 X    1.24 TYR 175 A\n",
      "ARG 285 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 CYS 181 A\n",
      "ARG 285 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.65 CYS 203 A\n",
      "ARG 285 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.85 ASP 287 A\n",
      "\n",
      "ARG 326 A  13.60     0 %   -0.31  246   0.00    0    0.75 ASP 322 A    0.00 XXX   0 X    0.01 ASP 329 A\n",
      "ARG 326 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 ASP 340 A\n",
      "ARG 326 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 TYR 360 A\n",
      "ARG 326 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 ARG 341 A\n",
      "ARG 326 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.42 ASP 322 A\n",
      "\n",
      "ARG 341 A  14.03    17 %   -1.04  330   0.00    0    0.33 TYR 360 A    0.00 XXX   0 X    0.10 ASP 322 A\n",
      "ARG 341 A                                            1.39 ASP 340 A    0.00 XXX   0 X    0.30 TYR 360 A\n",
      "ARG 341 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 364 A\n",
      "ARG 341 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.44 ASP 340 A\n",
      "\n",
      "N+    1 A   8.68     0 %   -0.50  274   0.00    0    0.85 GLU   4 A    0.00 XXX   0 X    0.14 GLU   2 A\n",
      "N+    1 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.16 ARG 134 A\n",
      "N+    1 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 GLU 161 A\n",
      "N+    1 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.10 LYS 208 A\n",
      "N+    1 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.41 GLU   4 A\n",
      "\n",
      "Coupled residues (marked *) were detected.Please rerun PropKa with the --display-coupled-residues \n",
      "or -d option for detailed information.\n",
      "\n",
      "INFO:pKa summary:\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "SUMMARY OF THIS PREDICTION\n",
      "       Group      pKa  model-pKa   ligand atom-type\n",
      "   ASP   6 A     3.00       3.80                      \n",
      "   ASP  38 A     4.10       3.80                      \n",
      "   ASP  39 A     2.55       3.80                      \n",
      "   ASP  40 A     4.01       3.80                      \n",
      "   ASP  47 A     5.77       3.80                      \n",
      "   ASP  64 A     2.40       3.80                      \n",
      "   ASP  74 A     3.59       3.80                      \n",
      "   ASP  82 A     3.93       3.80                      \n",
      "   ASP 139 A    11.53       3.80                      \n",
      "   ASP 142 A     5.13       3.80                      \n",
      "   ASP 151 A     4.73       3.80                      \n",
      "   ASP 237 A     3.65       3.80                      \n",
      "   ASP 238 A     3.28       3.80                      \n",
      "   ASP 287 A     3.73       3.80                      \n",
      "   ASP 305 A     6.18       3.80                      \n",
      "   ASP 322 A     3.09       3.80                      \n",
      "   ASP 329 A     2.58       3.80                      \n",
      "   ASP 340 A     2.65       3.80                      \n",
      "   GLU   2 A     3.46       4.50                      \n",
      "   GLU   4 A     2.86       4.50                      \n",
      "   GLU  24 A     4.74       4.50                      \n",
      "   GLU  60 A     5.00       4.50                      \n",
      "   GLU  72 A     1.96       4.50                      \n",
      "   GLU  73 A     7.93       4.50                      \n",
      "   GLU  76 A     4.99       4.50                      \n",
      "   GLU  80 A     3.53       4.50                      \n",
      "   GLU  96 A     3.28       4.50                      \n",
      "   GLU 100 A     4.18       4.50                      \n",
      "   GLU 117 A     3.73       4.50                      \n",
      "   GLU 161 A     3.76       4.50                      \n",
      "   GLU 180 A     4.49       4.50                      \n",
      "   GLU 226 A     4.56       4.50                      \n",
      "   GLU 246 A     5.27       4.50                      \n",
      "   GLU 249 A     5.27       4.50                      \n",
      "   GLU 289 A     3.91       4.50                      \n",
      "   GLU 301 A     1.69       4.50                      \n",
      "   GLU 314 A     5.47       4.50                      \n",
      "   GLU 315 A     4.02       4.50                      \n",
      "   GLU 337 A     3.62       4.50                      \n",
      "   GLU 353 A     3.83       4.50                      \n",
      "   GLU 364 A     4.60       4.50                      \n",
      "   C-  368 A     3.32       3.20                      \n",
      "   HIS  13 A     4.57       6.50                      \n",
      "   HIS  27 A     6.82       6.50                      \n",
      "   HIS  33 A     6.06       6.50                      \n",
      "   HIS 135 A     5.85       6.50                      \n",
      "   HIS 255 A     6.41       6.50                      \n",
      "   HIS 306 A     5.07       6.50                      \n",
      "   HIS 352 A     4.72       6.50                      \n",
      "   CYS  19 A    13.08       9.00                      \n",
      "   CYS 125 A    11.63       9.00                      \n",
      "   CYS 172 A    13.72       9.00                      \n",
      "   CYS 181 A     9.45       9.00                      \n",
      "   CYS 186 A     9.82       9.00                      \n",
      "   CYS 203 A    11.78       9.00                      \n",
      "   CYS 245 A    12.50       9.00                      \n",
      "   CYS 346 A    12.74       9.00                      \n",
      "   TYR 104 A    10.78      10.00                      \n",
      "   TYR 153 A    15.70      10.00                      \n",
      "   TYR 175 A    15.46      10.00                      \n",
      "   TYR 206 A    11.50      10.00                      \n",
      "   TYR 275 A    12.11      10.00                      \n",
      "   TYR 280 A    14.55      10.00                      \n",
      "   TYR 299 A    14.29      10.00                      \n",
      "   TYR 335 A    10.09      10.00                      \n",
      "   TYR 360 A    10.04      10.00                      \n",
      "   LYS  21 A    11.63      10.50                      \n",
      "   LYS  25 A    11.09      10.50                      \n",
      "   LYS  35 A    11.28      10.50                      \n",
      "   LYS  37 A    13.37      10.50                      \n",
      "   LYS 108 A    10.20      10.50                      \n",
      "   LYS 111 A    10.66      10.50                      \n",
      "   LYS 120 A    10.80      10.50                      \n",
      "   LYS 132 A    10.46      10.50                      \n",
      "   LYS 164 A    10.39      10.50                      \n",
      "   LYS 178 A    10.56      10.50                      \n",
      "   LYS 179 A    10.81      10.50                      \n",
      "   LYS 191 A    10.47      10.50                      \n",
      "   LYS 193 A    10.48      10.50                      \n",
      "   LYS 201 A    10.15      10.50                      \n",
      "   LYS 208 A    11.45      10.50                      \n",
      "   LYS 232 A    10.41      10.50                      \n",
      "   LYS 269 A    10.30      10.50                      \n",
      "   LYS 279 A     9.58      10.50                      \n",
      "   LYS 293 A    10.97      10.50                      \n",
      "   LYS 300 A    11.52      10.50                      \n",
      "   LYS 302 A    12.40      10.50                      \n",
      "   LYS 332 A    10.67      10.50                      \n",
      "   LYS 354 A    10.57      10.50                      \n",
      "   ARG  10 A    13.55      12.50                      \n",
      "   ARG 124 A    12.59      12.50                      \n",
      "   ARG 134 A    13.76      12.50                      \n",
      "   ARG 149 A    12.62      12.50                      \n",
      "   ARG 209 A    12.83      12.50                      \n",
      "   ARG 250 A    12.37      12.50                      \n",
      "   ARG 268 A    12.32      12.50                      \n",
      "   ARG 273 A    12.69      12.50                      \n",
      "   ARG 285 A    13.75      12.50                      \n",
      "   ARG 326 A    13.60      12.50                      \n",
      "   ARG 341 A    14.03      12.50                      \n",
      "   N+    1 A     8.68       8.00                      \n",
      "\n",
      "INFO:PROPKA information:\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "Free energy of   folding (kcal/mol) as a function of pH (using neutral reference)\n",
      "  0.00     28.05\n",
      "  1.00     27.81\n",
      "  2.00     26.19\n",
      "  3.00     20.32\n",
      "  4.00     11.83\n",
      "  5.00     10.29\n",
      "  6.00     12.13\n",
      "  7.00     13.31\n",
      "  8.00     15.91\n",
      "  9.00     20.92\n",
      " 10.00     32.09\n",
      " 11.00     51.34\n",
      " 12.00     71.39\n",
      " 13.00     87.31\n",
      " 14.00     98.55\n",
      "\n",
      "The pH of optimum stability is  4.6 for which the free energy is    9.9 kcal/mol at 298K\n",
      "Could not determine pH values where the free energy is within 80 % of minimum\n",
      "Could not determine the pH-range where the free energy is negative\n",
      "\n",
      "\n",
      "Protein charge of folded and unfolded state as a function of pH\n",
      "    pH  unfolded  folded\n",
      "  0.00     42.00   41.95\n",
      "  0.10     41.99   41.94\n",
      "  0.20     41.99   41.92\n",
      "  0.30     41.99   41.90\n",
      "  0.40     41.99   41.88\n",
      "  0.50     41.99   41.85\n",
      "  0.60     41.98   41.81\n",
      "  0.70     41.98   41.76\n",
      "  0.80     41.97   41.70\n",
      "  0.90     41.97   41.63\n",
      "  1.00     41.96   41.54\n",
      "  1.10     41.95   41.44\n",
      "  1.20     41.93   41.31\n",
      "  1.30     41.92   41.16\n",
      "  1.40     41.89   40.98\n",
      "  1.50     41.87   40.78\n",
      "  1.60     41.83   40.54\n",
      "  1.70     41.79   40.26\n",
      "  1.80     41.74   39.95\n",
      "  1.90     41.67   39.59\n",
      "  2.00     41.59   39.19\n",
      "  2.10     41.48   38.75\n",
      "  2.20     41.35   38.25\n",
      "  2.30     41.19   37.70\n",
      "  2.40     40.99   37.10\n",
      "  2.50     40.75   36.44\n",
      "  2.60     40.44   35.72\n",
      "  2.70     40.08   34.94\n",
      "  2.80     39.63   34.10\n",
      "  2.90     39.09   33.19\n",
      "  3.00     38.44   32.23\n",
      "  3.10     37.68   31.20\n",
      "  3.20     36.79   30.11\n",
      "  3.30     35.75   28.97\n",
      "  3.40     34.56   27.77\n",
      "  3.50     33.23   26.53\n",
      "  3.60     31.74   25.25\n",
      "  3.70     30.11   23.95\n",
      "  3.80     28.36   22.63\n",
      "  3.90     26.50   21.29\n",
      "  4.00     24.55   19.97\n",
      "  4.10     22.54   18.65\n",
      "  4.20     20.50   17.34\n",
      "  4.30     18.46   16.06\n",
      "  4.40     16.44   14.80\n",
      "  4.50     14.47   13.57\n",
      "  4.60     12.59   12.37\n",
      "  4.70     10.83   11.19\n",
      "  4.80      9.20   10.06\n",
      "  4.90      7.72    8.95\n",
      "  5.00      6.39    7.89\n",
      "  5.10      5.22    6.87\n",
      "  5.20      4.19    5.90\n",
      "  5.30      3.29    4.97\n",
      "  5.40      2.50    4.10\n",
      "  5.50      1.81    3.27\n",
      "  5.60      1.19    2.50\n",
      "  5.70      0.62    1.77\n",
      "  5.80      0.10    1.08\n",
      "  5.90     -0.40    0.44\n",
      "  6.00     -0.88   -0.15\n",
      "  6.10     -1.36   -0.71\n",
      "  6.20     -1.85   -1.22\n",
      "  6.30     -2.33   -1.69\n",
      "  6.40     -2.82   -2.13\n",
      "  6.50     -3.30   -2.53\n",
      "  6.60     -3.77   -2.89\n",
      "  6.70     -4.22   -3.22\n",
      "  6.80     -4.65   -3.52\n",
      "  6.90     -5.05   -3.79\n",
      "  7.00     -5.42   -4.03\n",
      "  7.10     -5.76   -4.25\n",
      "  7.20     -6.07   -4.45\n",
      "  7.30     -6.36   -4.64\n",
      "  7.40     -6.62   -4.81\n",
      "  7.50     -6.87   -4.97\n",
      "  7.60     -7.12   -5.12\n",
      "  7.70     -7.36   -5.27\n",
      "  7.80     -7.62   -5.42\n",
      "  7.90     -7.88   -5.57\n",
      "  8.00     -8.17   -5.72\n",
      "  8.10     -8.48   -5.88\n",
      "  8.20     -8.82   -6.04\n",
      "  8.30     -9.20   -6.22\n",
      "  8.40     -9.63   -6.40\n",
      "  8.50    -10.11   -6.59\n",
      "  8.60    -10.65   -6.81\n",
      "  8.70    -11.25   -7.04\n",
      "  8.80    -11.91   -7.29\n",
      "  8.90    -12.63   -7.57\n",
      "  9.00    -13.41   -7.89\n",
      "  9.10    -14.26   -8.24\n",
      "  9.20    -15.17   -8.64\n",
      "  9.30    -16.14   -9.09\n",
      "  9.40    -17.18   -9.59\n",
      "  9.50    -18.30  -10.16\n",
      "  9.60    -19.51  -10.80\n",
      "  9.70    -20.81  -11.51\n",
      "  9.80    -22.22  -12.29\n",
      "  9.90    -23.72  -13.16\n",
      " 10.00    -25.32  -14.10\n",
      " 10.10    -27.01  -15.11\n",
      " 10.20    -28.77  -16.20\n",
      " 10.30    -30.57  -17.34\n",
      " 10.40    -32.39  -18.53\n",
      " 10.50    -34.20  -19.75\n",
      " 10.60    -35.95  -21.00\n",
      " 10.70    -37.62  -22.26\n",
      " 10.80    -39.18  -23.51\n",
      " 10.90    -40.61  -24.76\n",
      " 11.00    -41.91  -25.98\n",
      " 11.10    -43.08  -27.19\n",
      " 11.20    -44.11  -28.36\n",
      " 11.30    -45.04  -29.51\n",
      " 11.40    -45.86  -30.63\n",
      " 11.50    -46.61  -31.73\n",
      " 11.60    -47.30  -32.79\n",
      " 11.70    -47.95  -33.84\n",
      " 11.80    -48.58  -34.85\n",
      " 11.90    -49.21  -35.85\n",
      " 12.00    -49.84  -36.83\n",
      " 12.10    -50.49  -37.80\n",
      " 12.20    -51.16  -38.75\n",
      " 12.30    -51.85  -39.69\n",
      " 12.40    -52.54  -40.62\n",
      " 12.50    -53.24  -41.54\n",
      " 12.60    -53.92  -42.44\n",
      " 12.70    -54.58  -43.33\n",
      " 12.80    -55.20  -44.20\n",
      " 12.90    -55.76  -45.04\n",
      " 13.00    -56.27  -45.87\n",
      " 13.10    -56.73  -46.67\n",
      " 13.20    -57.12  -47.44\n",
      " 13.30    -57.45  -48.19\n",
      " 13.40    -57.74  -48.93\n",
      " 13.50    -57.97  -49.64\n",
      " 13.60    -58.17  -50.32\n",
      " 13.70    -58.33  -50.98\n",
      " 13.80    -58.46  -51.62\n",
      " 13.90    -58.57  -52.22\n",
      " 14.00    -58.65  -52.79\n",
      "The pI is  5.97 (folded) and  5.82 (unfolded)\n",
      "\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "INFO:Applying pKa values at a pH of 7.90:\n",
      "INFO:Adding hydrogens to biomolecule.\n",
      "INFO:Debumping biomolecule (again).\n",
      "INFO:Optimizing hydrogen bonds\n",
      "INFO:Applying force field to biomolecule states.\n",
      "INFO:Regenerating headers.\n",
      "INFO:Regenerating PDB lines.\n",
      "WARNING:Ignoring 466 header lines in output.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "pH: 7.9\tname: AtAHL\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:PDB2PQR v3.6.2: biomolecular structure conversion software.\n",
      "INFO:Please cite:  Jurrus E, et al.  Improvements to the APBS biomolecular solvation software suite.  Protein Sci 27 112-128 (2018).\n",
      "INFO:Please cite:  Dolinsky TJ, et al.  PDB2PQR: expanding and upgrading automated preparation of biomolecular structures for molecular simulations. Nucleic Acids Res 35 W522-W525 (2007).\n",
      "INFO:Checking and transforming input arguments.\n",
      "INFO:Loading topology files.\n",
      "INFO:Loading molecule: /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/74987.pdb\n",
      "INFO:Setting up molecule.\n",
      "INFO:Created biomolecule object with 352 residues and 5354 atoms.\n",
      "INFO:Setting termini states for biomolecule chains.\n",
      "INFO:Loading forcefield.\n",
      "INFO:Loading hydrogen topology definitions.\n",
      "INFO:This biomolecule is clean.  No repair needed.\n",
      "INFO:Updating disulfide bridges.\n",
      "INFO:Debumping biomolecule.\n",
      "INFO:Assigning titration states with PROPKA.\n",
      "INFO:\n",
      "propka3.5.1                                                                                  2024-07-26\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "--                                                                           --\n",
      "--  PROPKA: A PROTEIN PKA PREDICTOR                                          --\n",
      "--                                                                           --\n",
      "--  VERSION 1.0,  04/25/2004,  IOWA CITY                                     --\n",
      "--  BY HUI LI                                                                --\n",
      "--                                                                           --\n",
      "--  VERSION 2.0,  11/05/2007, IOWA CITY/COPENHAGEN                           --\n",
      "--  BY DELPHINE C. BAS AND DAVID M. ROGERS                                   --\n",
      "--                                                                           --\n",
      "--  VERSION 3.0,  01/06/2011, COPENHAGEN                                     --\n",
      "--  BY MATS H.M. OLSSON AND CHRESTEN R. SONDERGARD                           --\n",
      "--                                                                           --\n",
      "--  VERSION 3.1,  07/01/2011, COPENHAGEN                                     --\n",
      "--  BY CHRESTEN R. SONDERGARD AND MATS H.M. OLSSON                           --\n",
      "--                                                                           --\n",
      "--  VERSION 3.2,  06/17/2020, PLANET EARTH                                   --\n",
      "--  SEE ABOVE FOR AUTHORS                                                    --\n",
      "--                                                                           --\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "References:\n",
      "\n",
      "Very Fast Empirical Prediction and Rationalization of Protein pKa Values.\n",
      "Hui Li, Andrew D. Robertson and Jan H. Jensen. PROTEINS: Structure, Function,\n",
      "and Bioinformatics. 61:704-721 (2005)\n",
      "\n",
      "Very Fast Prediction and Rationalization of pKa Values for Protein-Ligand\n",
      "Complexes.  Delphine C. Bas, David M. Rogers and Jan H. Jensen.  PROTEINS:\n",
      "Structure, Function, and Bioinformatics 73:765-783 (2008)\n",
      "\n",
      "PROPKA3: Consistent Treatment of Internal and Surface Residues in Empirical\n",
      "pKa predictions.  Mats H.M. Olsson, Chresten R. Sondergard, Michal Rostkowski,\n",
      "and Jan H. Jensen.  Journal of Chemical Theory and Computation, 7(2):525-537\n",
      "(2011)\n",
      "\n",
      "Improved Treatment of Ligands and Coupling Effects in Empirical Calculation\n",
      "and Rationalization of pKa Values.  Chresten R. Sondergaard, Mats H.M. Olsson,\n",
      "Michal Rostkowski, and Jan H. Jensen.  Journal of Chemical Theory and\n",
      "Computation, (2011)\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "\n",
      "INFO:Coupling map:\n",
      "\n",
      "INFO:Calculating pKas for Conformation container 1A with 3171 atoms and 875 groups\n",
      "INFO:pKa determinants:\n",
      "\n",
      "---------  -----   ------   ---------------------    --------------    --------------    --------------\n",
      "                            DESOLVATION  EFFECTS       SIDECHAIN          BACKBONE        COULOMBIC\n",
      " RESIDUE    pKa    BURIED     REGULAR      RE        HYDROGEN BOND     HYDROGEN BOND      INTERACTION\n",
      "---------  -----   ------   ---------   ---------    --------------    --------------    --------------\n",
      "\n",
      "ASP  10 A   4.95    44 %    1.49  405   0.14    0   -0.28 ARG  14 A    0.00 XXX   0 X   -0.05 ARG  17 A\n",
      "ASP  10 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.10 LYS  59 A\n",
      "ASP  10 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU   6 A\n",
      "ASP  10 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP  13 A\n",
      "ASP  10 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.29 GLU  56 A\n",
      "ASP  10 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.55 ARG  14 A\n",
      "\n",
      "ASP  13 A   4.41    60 %    1.88  448   0.32    0   -0.72 ARG  17 A   -0.16 GLN 194 A   -0.08 ARG  14 A\n",
      "ASP  13 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.00 LYS  91 A\n",
      "ASP  13 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.05 LYS 191 A\n",
      "ASP  13 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.59 ARG  17 A\n",
      "\n",
      "ASP  21 A   2.92    25 %    0.74  352   0.19    0   -0.11 ARG  24 A    0.00 XXX   0 X   -0.03 ARG  17 A\n",
      "ASP  21 A                                           -0.85 LYS  91 A    0.00 XXX   0 X   -0.32 ARG  24 A\n",
      "ASP  21 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.49 LYS  91 A\n",
      "\n",
      "ASP  38 A   2.45    12 %    0.70  316   0.00    0   -0.85 THR  40 A   -0.20 ASP  38 A   -0.26 LYS  36 A\n",
      "ASP  38 A                                            0.00 XXX   0 X   -0.66 THR  40 A   -0.08 LYS 286 A\n",
      "\n",
      "ASP  46 A   5.88   100 %    3.67  601   1.00    0   -0.85 THR  43 A   -0.70 THR  43 A   -0.59 LYS  36 A\n",
      "ASP  46 A                                           -0.64 THR 134 A   -0.38 ASP  46 A    0.56 GLU  69 A\n",
      "\n",
      "ASP  71 A   3.56    35 %    0.91  380   0.00    0    0.00 XXX   0 X   -0.46 HIS  73 A   -0.12 LYS  36 A\n",
      "ASP  71 A                                            0.00 XXX   0 X   -0.28 PHE  74 A   -0.05 ARG  76 A\n",
      "ASP  71 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.21 LYS 113 A\n",
      "ASP  71 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 322 A\n",
      "ASP  71 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.19 HIS  73 A\n",
      "ASP  71 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.18 ASP 109 A\n",
      "\n",
      "ASP  96 A   3.95     0 %    0.24  216   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.09 LYS  89 A\n",
      "ASP  96 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.00 LYS  91 A\n",
      "\n",
      "ASP 101 A   3.94     0 %    0.32  265   0.00    0    0.00 XXX   0 X   -0.03 ASP 101 A   -0.16 LYS  89 A\n",
      "ASP 101 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 103 A\n",
      "\n",
      "ASP 103 A   2.46     0 %    0.37  271   0.00    0   -0.85 SER 100 A   -0.81 SER 100 A   -0.04 LYS  89 A\n",
      "ASP 103 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 HIS  98 A\n",
      "\n",
      "ASP 109 A   2.64    27 %    1.12  356   0.23    0   -0.85 SER  72 A   -0.79 SER  72 A   -0.02 HIS  73 A\n",
      "ASP 109 A                                           -0.48 ARG  76 A    0.00 XXX   0 X   -0.37 ARG  76 A\n",
      "\n",
      "ASP 114 A   3.91     0 %    0.33  245   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.29 LYS 113 A\n",
      "ASP 114 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 ARG 322 A\n",
      "ASP 114 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.11 ASP  71 A\n",
      "ASP 114 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 109 A\n",
      "\n",
      "ASP 129 A  11.71*  100 %    3.55  628   0.27    0    0.42 ASP 289 A   -0.23 ILE 131 A   -0.17 LYS  36 A\n",
      "ASP 129 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.32 ASP  46 A\n",
      "ASP 129 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 GLU  70 A\n",
      "ASP 129 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.98 GLU  69 A\n",
      "ASP 129 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.31 ASP 132 A\n",
      "ASP 129 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.28 ASP 289 A\n",
      "\n",
      "ASP 132 A   6.29*  100 %    3.80  623   0.65    0   -0.13 SER 260 A   -0.48 SER 260 A    0.11 GLU  69 A\n",
      "ASP 132 A                                            0.00 XXX   0 X   -0.56 LEU 261 A   -0.70 LYS 263 A\n",
      "ASP 132 A                                            0.00 XXX   0 X   -0.19 CYS 262 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 141 A   3.75     0 %    0.27  267   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.31 LYS  23 A\n",
      "ASP 141 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 ARG  24 A\n",
      "\n",
      "ASP 154 A   4.07     0 %    0.28  239   0.00    0    0.00 XXX   0 X   -0.00 ASP 154 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 170 A   3.21     0 %    0.24  274   0.00    0   -0.42 ARG 271 A    0.00 XXX   0 X   -0.03 LYS 192 A\n",
      "ASP 170 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.37 ARG 271 A\n",
      "\n",
      "ASP 173 A   1.41     0 %    0.49  200   0.00    0   -0.85 SER 175 A   -0.08 ASP 173 A   -0.38 LYS 192 A\n",
      "ASP 173 A                                           -0.81 LYS 192 A   -0.76 SER 175 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 203 A   3.82     0 %    0.12  132   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.10 LYS 191 A\n",
      "\n",
      "ASP 210 A   2.58    45 %    1.85  406   0.24    0   -0.85 SER 273 A   -0.41 CYS 212 A   -0.43 ARG 219 A\n",
      "ASP 210 A                                           -0.78 ASN 334 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "ASP 210 A                                           -0.85 ARG 219 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "ASP 238 A   4.04     0 %    0.23  233   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.09 LYS 247 A\n",
      "ASP 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 249 A\n",
      "ASP 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 229 A\n",
      "ASP 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 GLU 250 A\n",
      "\n",
      "ASP 245 A   4.00     0 %    0.16  109   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.04 GLU 250 A\n",
      "\n",
      "ASP 289 A   7.94*  100 %    2.62  580   0.38    0   -0.42 ASP 129 A    0.00 XXX   0 X   -0.01 LYS  36 A\n",
      "ASP 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.00 LYS 263 A\n",
      "ASP 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.19 LYS 286 A\n",
      "ASP 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.76 GLU  69 A\n",
      "ASP 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.00 ASP 132 A\n",
      "\n",
      "ASP 306 A   3.95    39 %    1.62  391   0.52    0    0.00 XXX   0 X   -0.01 ASP 306 A   -0.01 ARG 283 A\n",
      "ASP 306 A                                            0.00 XXX   0 X   -0.48 GLU 308 A   -0.01 HIS 338 A\n",
      "ASP 306 A                                            0.00 XXX   0 X   -0.34 GLY 309 A    0.00 XXX   0 X\n",
      "ASP 306 A                                            0.00 XXX   0 X   -0.61 ASP 310 A    0.00 XXX   0 X\n",
      "ASP 306 A                                            0.00 XXX   0 X   -0.53 ALA 327 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 310 A   4.33     0 %    0.22  242   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.00 LYS 303 A\n",
      "ASP 310 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.28 ASP 306 A\n",
      "ASP 310 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 311 A\n",
      "\n",
      "ASP 317 A   3.26    37 %    1.59  385   0.12    0    0.00 XXX   0 X   -0.51 GLU 320 A   -0.05 LYS 286 A\n",
      "ASP 317 A                                            0.00 XXX   0 X   -0.03 ARG 321 A   -0.13 ARG 321 A\n",
      "ASP 317 A                                            0.00 XXX   0 X   -0.72 ARG 322 A   -0.10 ARG 322 A\n",
      "ASP 317 A                                            0.00 XXX   0 X   -0.69 LEU 323 A    0.00 XXX   0 X\n",
      "\n",
      "GLU   4 A   4.10    15 %    0.77  322   0.00    0    0.00 XXX   0 X   -0.76 GLU   4 A   -0.26 N+    1 A\n",
      "GLU   4 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.07 LYS   3 A\n",
      "GLU   4 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 HIS   2 A\n",
      "GLU   4 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 HIS 183 A\n",
      "\n",
      "GLU   6 A   4.46     0 %    0.19  222   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.05 LYS   3 A\n",
      "GLU   6 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.18 HIS   2 A\n",
      "\n",
      "GLU  31 A   4.62     6 %    0.36  298   0.00    0   -0.24 SER  29 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "GLU  35 A   4.73     0 %    0.22  190   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP  38 A\n",
      "\n",
      "GLU  56 A   2.83    47 %    1.86  412   0.21    0   -0.78 SER  93 A   -0.83 SER  93 A   -0.52 ARG  14 A\n",
      "GLU  56 A                                           -0.18 ARG  14 A    0.00 XXX   0 X   -0.58 LYS  59 A\n",
      "GLU  56 A                                           -0.85 LYS  59 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "GLU  69 A   4.06*   92 %    1.97  539   0.00    0   -0.35 TRP 288 A    0.00 XXX   0 X   -0.18 LYS 286 A\n",
      "GLU  69 A                                           -0.85 LYS  36 A    0.00 XXX   0 X   -0.02 ARG 322 A\n",
      "GLU  69 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 ASP  38 A\n",
      "GLU  69 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.20 LYS  36 A\n",
      "\n",
      "GLU  70 A   8.36*   97 %    3.42  553   0.03    0   -0.85 LYS  36 A    0.00 XXX   0 X    0.08 ASP  38 A\n",
      "GLU  70 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 ASP  71 A\n",
      "GLU  70 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.29 LYS  36 A\n",
      "GLU  70 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.00 ASP  46 A\n",
      "GLU  70 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.27 GLU  69 A\n",
      "\n",
      "GLU  87 A   5.73    40 %    1.00  393   0.16    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.00 ARG  24 A\n",
      "GLU  87 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.14 LYS  91 A\n",
      "GLU  87 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.17 ASP  21 A\n",
      "GLU  87 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU  31 A\n",
      "\n",
      "GLU 106 A   4.79     0 %    0.16  269   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 110 A\n",
      "GLU 106 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.11 ASP 103 A\n",
      "GLU 106 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 109 A\n",
      "\n",
      "GLU 142 A   4.62     0 %    0.09  177   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.01 LYS  23 A\n",
      "GLU 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 141 A\n",
      "\n",
      "GLU 156 A   5.01    18 %    0.37  332   0.02    0    0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 154 A\n",
      "GLU 156 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 HIS 298 A\n",
      "GLU 156 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.14 GLU 311 A\n",
      "GLU 156 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 315 A\n",
      "\n",
      "GLU 224 A   2.07    12 %    0.73  314   0.02    0   -0.52 ARG 135 A    0.00 XXX   0 X   -0.16 LYS 263 A\n",
      "GLU 224 A                                           -0.85 LYS 139 A    0.00 XXX   0 X   -0.42 ARG 135 A\n",
      "GLU 224 A                                           -0.84 CYS 258 A    0.00 XXX   0 X   -0.39 LYS 139 A\n",
      "\n",
      "GLU 229 A   3.42     0 %    0.34  244   0.00    0   -0.66 THR 227 A   -0.76 GLU 229 A    0.00 XXX   0 X\n",
      "\n",
      "GLU 244 A   4.57     0 %    0.07   82   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "GLU 250 A   2.96     0 %    0.53  224   0.00    0   -0.85 LYS 247 A   -0.84 LYS 247 A   -0.00 LYS 217 A\n",
      "GLU 250 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.38 LYS 247 A\n",
      "\n",
      "GLU 299 A   4.11    28 %    1.15  361   0.24    0    0.00 XXX   0 X   -0.79 GLY 185 A   -0.13 ARG 206 A\n",
      "GLU 299 A                                            0.00 XXX   0 X   -0.08 CYS 186 A   -0.10 HIS 183 A\n",
      "GLU 299 A                                            0.00 XXX   0 X   -0.58 GLY 187 A   -0.10 HIS 298 A\n",
      "\n",
      "GLU 308 A   5.24     0 %    0.26  271   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.10 ARG 283 A\n",
      "GLU 308 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.24 ASP 306 A\n",
      "GLU 308 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP 310 A\n",
      "GLU 308 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 C-  352 A\n",
      "\n",
      "GLU 311 A   3.29    11 %    0.28  311   0.00    0   -0.85 LYS 303 A    0.00 XXX   0 X   -0.21 HIS 298 A\n",
      "GLU 311 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.42 LYS 303 A\n",
      "\n",
      "GLU 315 A   4.29     0 %    0.41  259   0.00    0   -0.39 ASN 155 A    0.00 XXX   0 X   -0.24 ARG 321 A\n",
      "GLU 315 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 154 A\n",
      "\n",
      "GLU 316 A   5.33    15 %    0.68  322   0.08    0    0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 306 A\n",
      "GLU 316 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 317 A\n",
      "\n",
      "GLU 320 A   4.89     8 %    0.35  305   0.01    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.05 LYS 113 A\n",
      "GLU 320 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 LYS 286 A\n",
      "GLU 320 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.19 ARG 322 A\n",
      "GLU 320 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 114 A\n",
      "GLU 320 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.30 ASP 317 A\n",
      "\n",
      "GLU 343 A   4.70     0 %    0.20  205   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "C-  352 A   2.61     0 %    0.24  209   0.00    0   -0.57 ARG 283 A    0.00 XXX   0 X   -0.26 ARG 283 A\n",
      "\n",
      "HIS   2 A   6.46     0 %   -0.18  208   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.08 N+    1 A\n",
      "HIS   2 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 GLU   4 A\n",
      "HIS   2 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.18 GLU   6 A\n",
      "\n",
      "HIS  73 A   6.41     0 %   -0.19  218   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.06 ARG  76 A\n",
      "HIS  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 113 A\n",
      "HIS  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 ASP  71 A\n",
      "HIS  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 109 A\n",
      "\n",
      "HIS  98 A   6.37     0 %   -0.15  122   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.00 LYS  89 A\n",
      "HIS  98 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 103 A\n",
      "\n",
      "HIS 183 A   6.16    21 %   -0.82  341   0.00    0    0.00 XXX   0 X    0.05 HIS 183 A   -0.39 N+    1 A\n",
      "HIS 183 A                                            0.00 XXX   0 X    0.71 ILE 184 A   -0.03 HIS   2 A\n",
      "HIS 183 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 GLU   4 A\n",
      "HIS 183 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 GLU 299 A\n",
      "\n",
      "HIS 248 A   6.22     0 %   -0.17  173   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 217 A\n",
      "HIS 248 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 249 A\n",
      "\n",
      "HIS 290 A   3.27   100 %   -3.17  619   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.07 LYS 263 A\n",
      "\n",
      "HIS 298 A   5.12    76 %   -1.70  493   0.00    0    0.00 XXX   0 X    0.30 GLY 302 A   -0.32 LYS 303 A\n",
      "HIS 298 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 156 A\n",
      "HIS 298 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 GLU 299 A\n",
      "HIS 298 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.21 GLU 311 A\n",
      "\n",
      "HIS 338 A   4.12    64 %   -2.13  461   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.17 LYS 303 A\n",
      "HIS 338 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.10 HIS 298 A\n",
      "HIS 338 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 306 A\n",
      "\n",
      "CYS  16 A  14.27   100 %    3.67  626   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.05 ARG  17 A\n",
      "CYS  16 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.44 ASP  13 A\n",
      "CYS  16 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.84 CYS  19 A\n",
      "CYS  16 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.36 CYS 164 A\n",
      "\n",
      "CYS  19 A  13.25   100 %    3.84  640   0.00    0    0.00 XXX   0 X   -0.06 CYS  19 A   -0.03 LYS  23 A\n",
      "CYS  19 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.49 ASP  46 A\n",
      "CYS  19 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU  87 A\n",
      "\n",
      "CYS 119 A   9.04     0 %    0.20  187   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.14 LYS 121 A\n",
      "CYS 119 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 321 A\n",
      "\n",
      "CYS 164 A  12.58   100 %    3.66  580   0.00    0    0.00 XXX   0 X   -0.13 TYR 145 A   -0.02 LYS  23 A\n",
      "CYS 164 A                                            0.00 XXX   0 X   -0.10 ASN 166 A   -0.04 ARG 271 A\n",
      "CYS 164 A                                            0.00 XXX   0 X   -0.05 TRP 167 A    0.11 ASP 132 A\n",
      "CYS 164 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.17 CYS 257 A\n",
      "\n",
      "CYS 186 A   9.38     0 %    0.39  262   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.09 N+    1 A\n",
      "CYS 186 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 ARG 206 A\n",
      "CYS 186 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 GLU 299 A\n",
      "\n",
      "CYS 207 A  11.12    63 %    2.22  457   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.07 ARG 206 A\n",
      "CYS 207 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 271 A\n",
      "\n",
      "CYS 212 A   9.75    29 %    1.27  363   0.00    0   -0.00 SER 273 A   -0.71 LEU 214 A   -0.04 LYS 217 A\n",
      "CYS 212 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.12 ARG 219 A\n",
      "CYS 212 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.35 ASP 210 A\n",
      "\n",
      "CYS 221 A  11.78    93 %    2.98  541   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.06 ARG 219 A\n",
      "CYS 221 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.40 LYS 263 A\n",
      "CYS 221 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.26 CYS 257 A\n",
      "\n",
      "CYS 257 A   6.54*   61 %    1.43  451   0.00    0   -3.60 CYS 262 A    0.00 XXX   0 X   -0.02 LYS 139 A\n",
      "CYS 257 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.33 LYS 263 A\n",
      "CYS 257 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 GLU 224 A\n",
      "\n",
      "CYS 258 A  10.03    39 %    1.48  391   0.00    0   -0.71 ARG 135 A   -0.05 CYS 258 A   -0.39 ARG 135 A\n",
      "CYS 258 A                                           -0.23 LYS 139 A    0.00 XXX   0 X   -0.45 LYS 139 A\n",
      "CYS 258 A                                            0.84 GLU 224 A    0.00 XXX   0 X   -0.20 LYS 263 A\n",
      "CYS 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP 132 A\n",
      "CYS 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.15 CYS 257 A\n",
      "CYS 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.48 GLU 224 A\n",
      "\n",
      "CYS 262 A  17.21*  100 %    3.46  569   0.00    0    3.60 CYS 257 A   -0.07 CYS 262 A   -1.78 LYS 263 A\n",
      "CYS 262 A                                            0.00 XXX   0 X   -0.06 LYS 263 A    0.53 CYS 164 A\n",
      "CYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.44 CYS 221 A\n",
      "CYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 224 A\n",
      "CYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 CYS 258 A\n",
      "CYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.03 ASP 132 A\n",
      "CYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.81 CYS 257 A\n",
      "\n",
      "CYS 296 A  12.13   100 %    3.70  656   0.00    0   -0.85 SER 182 A    0.00 XXX   0 X    0.23 CYS 207 A\n",
      "CYS 296 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 299 A\n",
      "\n",
      "TYR   1 A  10.58     0 %    0.41  279   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU   4 A\n",
      "TYR   1 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP 154 A\n",
      "TYR   1 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 GLU 156 A\n",
      "\n",
      "TYR 116 A  10.46     2 %    0.31  288   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.16 CYS 119 A\n",
      "TYR 116 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 ARG 321 A\n",
      "\n",
      "TYR 125 A  12.77    85 %    2.97  518   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 315 A\n",
      "TYR 125 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP 317 A\n",
      "TYR 125 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.34 ARG 321 A\n",
      "\n",
      "TYR 145 A  16.29    97 %    3.60  552   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.78 CYS  16 A\n",
      "TYR 145 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.50 CYS  19 A\n",
      "TYR 145 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP  21 A\n",
      "TYR 145 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 141 A\n",
      "TYR 145 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.37 CYS 164 A\n",
      "\n",
      "TYR 264 A  13.96   100 %    3.82  649   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 132 A\n",
      "TYR 264 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP 289 A\n",
      "\n",
      "LYS   3 A  10.39     0 %   -0.23  184   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.07 GLU   4 A\n",
      "LYS   3 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU   6 A\n",
      "\n",
      "LYS  23 A   9.00    44 %   -1.88  405   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.03 CYS  19 A\n",
      "LYS  23 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.31 ASP 141 A\n",
      "LYS  23 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 142 A\n",
      "LYS  23 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 CYS 164 A\n",
      "\n",
      "LYS  30 A  10.47     0 %   -0.03   61   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS  32 A  10.45     0 %   -0.05  104   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS  36 A  13.02    75 %   -2.82  492   0.00    0    0.85 GLU  69 A    0.00 XXX   0 X    0.26 ASP  38 A\n",
      "LYS  36 A                                            0.85 GLU  70 A    0.00 XXX   0 X    0.12 ASP  71 A\n",
      "LYS  36 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.17 ASP 129 A\n",
      "LYS  36 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 289 A\n",
      "LYS  36 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.59 ASP  46 A\n",
      "LYS  36 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.20 GLU  69 A\n",
      "LYS  36 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.29 GLU  70 A\n",
      "\n",
      "LYS  59 A  10.02    37 %   -1.79  385   0.00    0    0.85 GLU  56 A    0.00 XXX   0 X    0.10 ASP  10 A\n",
      "LYS  59 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.22 ARG  14 A\n",
      "LYS  59 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.58 GLU  56 A\n",
      "\n",
      "LYS  89 A  10.40     0 %   -0.38  237   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP  96 A\n",
      "LYS  89 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 ASP 101 A\n",
      "LYS  89 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 103 A\n",
      "\n",
      "LYS  91 A  10.94    30 %   -0.88  365   0.00    0    0.85 ASP  21 A    0.00 XXX   0 X    0.00 ASP  13 A\n",
      "LYS  91 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.11 ARG  17 A\n",
      "LYS  91 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 ARG  24 A\n",
      "LYS  91 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.14 GLU  87 A\n",
      "LYS  91 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP  96 A\n",
      "LYS  91 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.49 ASP  21 A\n",
      "\n",
      "LYS 113 A  10.86     0 %   -0.16  251   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.21 ASP  71 A\n",
      "LYS 113 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.29 ASP 114 A\n",
      "LYS 113 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 320 A\n",
      "LYS 113 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 ARG 322 A\n",
      "\n",
      "LYS 121 A  10.58     0 %   -0.06  106   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.14 CYS 119 A\n",
      "\n",
      "LYS 139 A  11.66     0 %   -0.43  269   0.00    0    0.23 CYS 258 A    0.00 XXX   0 X   -0.35 ARG 135 A\n",
      "LYS 139 A                                            0.85 GLU 224 A    0.00 XXX   0 X    0.02 CYS 257 A\n",
      "LYS 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.45 CYS 258 A\n",
      "LYS 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.39 GLU 224 A\n",
      "\n",
      "LYS 191 A  10.20     2 %   -0.45  288   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP  13 A\n",
      "LYS 191 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP 203 A\n",
      "\n",
      "LYS 192 A  11.20     0 %   -0.52  249   0.00    0    0.81 ASP 173 A    0.00 XXX   0 X    0.03 ASP 170 A\n",
      "LYS 192 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 ASP 173 A\n",
      "\n",
      "LYS 217 A  10.32     0 %   -0.21  224   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.04 CYS 212 A\n",
      "LYS 217 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 GLU 250 A\n",
      "\n",
      "LYS 247 A  11.60     0 %   -0.22  164   0.00    0    0.85 GLU 250 A    0.00 XXX   0 X    0.09 ASP 238 A\n",
      "LYS 247 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 GLU 250 A\n",
      "\n",
      "LYS 249 A  10.46     0 %   -0.10  152   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.06 ASP 238 A\n",
      "\n",
      "LYS 263 A  11.10    90 %   -2.93  534   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.04 ARG 135 A\n",
      "LYS 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.40 CYS 221 A\n",
      "LYS 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 GLU 224 A\n",
      "LYS 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.33 CYS 257 A\n",
      "LYS 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 CYS 258 A\n",
      "LYS 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.78 CYS 262 A\n",
      "LYS 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP 289 A\n",
      "LYS 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.70 ASP 132 A\n",
      "\n",
      "LYS 280 A  10.26    15 %   -0.23  322   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.01 ARG 135 A\n",
      "\n",
      "LYS 286 A  10.05    45 %   -0.88  407   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP  38 A\n",
      "LYS 286 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.18 GLU  69 A\n",
      "LYS 286 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 ASP 289 A\n",
      "LYS 286 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 317 A\n",
      "LYS 286 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU 320 A\n",
      "LYS 286 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.11 ARG 322 A\n",
      "LYS 286 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 LYS  36 A\n",
      "\n",
      "LYS 303 A  11.21    13 %   -0.56  318   0.00    0    0.85 GLU 311 A    0.00 XXX   0 X    0.00 ASP 310 A\n",
      "LYS 303 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.42 GLU 311 A\n",
      "\n",
      "ARG  14 A  12.98    32 %   -1.07  370   0.00    0    0.28 ASP  10 A    0.00 XXX   0 X    0.08 ASP  13 A\n",
      "ARG  14 A                                            0.18 GLU  56 A    0.00 XXX   0 X   -0.05 ARG  17 A\n",
      "ARG  14 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.55 ASP  10 A\n",
      "ARG  14 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.52 GLU  56 A\n",
      "\n",
      "ARG  17 A  13.13    26 %   -0.81  355   0.00    0    0.72 ASP  13 A    0.00 XXX   0 X    0.05 ASP  10 A\n",
      "ARG  17 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 CYS  16 A\n",
      "ARG  17 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP  21 A\n",
      "ARG  17 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.59 ASP  13 A\n",
      "\n",
      "ARG  24 A  12.69     0 %   -0.25  263   0.00    0    0.11 ASP  21 A    0.00 XXX   0 X    0.32 ASP  21 A\n",
      "ARG  24 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 GLU  87 A\n",
      "ARG  24 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 141 A\n",
      "\n",
      "ARG  76 A  12.96     1 %   -0.44  283   0.00    0    0.48 ASP 109 A    0.00 XXX   0 X    0.05 ASP  71 A\n",
      "ARG  76 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.37 ASP 109 A\n",
      "\n",
      "ARG 110 A  10.93    57 %   -1.59  442   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 106 A\n",
      "\n",
      "ARG 135 A  14.08    12 %   -0.46  314   0.00    0    0.71 CYS 258 A    0.00 XXX   0 X    0.39 CYS 258 A\n",
      "ARG 135 A                                            0.52 GLU 224 A    0.00 XXX   0 X    0.42 GLU 224 A\n",
      "\n",
      "ARG 206 A  12.17     3 %   -0.59  289   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.06 CYS 186 A\n",
      "ARG 206 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 CYS 207 A\n",
      "ARG 206 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 GLU 299 A\n",
      "\n",
      "ARG 219 A  12.97    38 %   -0.99  388   0.00    0    0.85 ASP 210 A    0.00 XXX   0 X    0.12 CYS 212 A\n",
      "ARG 219 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 CYS 221 A\n",
      "ARG 219 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.43 ASP 210 A\n",
      "\n",
      "ARG 271 A  12.14    41 %   -1.19  395   0.00    0    0.42 ASP 170 A    0.00 XXX   0 X    0.04 CYS 164 A\n",
      "ARG 271 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 CYS 207 A\n",
      "ARG 271 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 219 A\n",
      "ARG 271 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.37 ASP 170 A\n",
      "\n",
      "ARG 278 A  11.64    29 %   -0.86  362   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "ARG 283 A  13.24     0 %   -0.20  243   0.00    0    0.57 C-  352 A    0.00 XXX   0 X    0.01 ASP 306 A\n",
      "ARG 283 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 GLU 308 A\n",
      "ARG 283 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.26 C-  352 A\n",
      "\n",
      "ARG 321 A  11.96    26 %   -1.28  353   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.01 TYR 116 A\n",
      "ARG 321 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 CYS 119 A\n",
      "ARG 321 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.34 TYR 125 A\n",
      "ARG 321 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.24 GLU 315 A\n",
      "ARG 321 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 ASP 317 A\n",
      "\n",
      "ARG 322 A  10.73    72 %   -2.17  484   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU  69 A\n",
      "ARG 322 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP  71 A\n",
      "ARG 322 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 ASP 114 A\n",
      "ARG 322 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP 317 A\n",
      "ARG 322 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 GLU 320 A\n",
      "\n",
      "N+    1 A   7.87     2 %   -0.47  287   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.26 GLU   4 A\n",
      "N+    1 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 CYS 186 A\n",
      "\n",
      "Coupled residues (marked *) were detected.Please rerun PropKa with the --display-coupled-residues \n",
      "or -d option for detailed information.\n",
      "\n",
      "INFO:pKa summary:\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "SUMMARY OF THIS PREDICTION\n",
      "       Group      pKa  model-pKa   ligand atom-type\n",
      "   ASP  10 A     4.95       3.80                      \n",
      "   ASP  13 A     4.41       3.80                      \n",
      "   ASP  21 A     2.92       3.80                      \n",
      "   ASP  38 A     2.45       3.80                      \n",
      "   ASP  46 A     5.88       3.80                      \n",
      "   ASP  71 A     3.56       3.80                      \n",
      "   ASP  96 A     3.95       3.80                      \n",
      "   ASP 101 A     3.94       3.80                      \n",
      "   ASP 103 A     2.46       3.80                      \n",
      "   ASP 109 A     2.64       3.80                      \n",
      "   ASP 114 A     3.91       3.80                      \n",
      "   ASP 129 A    11.71       3.80                      \n",
      "   ASP 132 A     6.29       3.80                      \n",
      "   ASP 141 A     3.75       3.80                      \n",
      "   ASP 154 A     4.07       3.80                      \n",
      "   ASP 170 A     3.21       3.80                      \n",
      "   ASP 173 A     1.41       3.80                      \n",
      "   ASP 203 A     3.82       3.80                      \n",
      "   ASP 210 A     2.58       3.80                      \n",
      "   ASP 238 A     4.04       3.80                      \n",
      "   ASP 245 A     4.00       3.80                      \n",
      "   ASP 289 A     7.94       3.80                      \n",
      "   ASP 306 A     3.95       3.80                      \n",
      "   ASP 310 A     4.33       3.80                      \n",
      "   ASP 317 A     3.26       3.80                      \n",
      "   GLU   4 A     4.10       4.50                      \n",
      "   GLU   6 A     4.46       4.50                      \n",
      "   GLU  31 A     4.62       4.50                      \n",
      "   GLU  35 A     4.73       4.50                      \n",
      "   GLU  56 A     2.83       4.50                      \n",
      "   GLU  69 A     4.06       4.50                      \n",
      "   GLU  70 A     8.36       4.50                      \n",
      "   GLU  87 A     5.73       4.50                      \n",
      "   GLU 106 A     4.79       4.50                      \n",
      "   GLU 142 A     4.62       4.50                      \n",
      "   GLU 156 A     5.01       4.50                      \n",
      "   GLU 224 A     2.07       4.50                      \n",
      "   GLU 229 A     3.42       4.50                      \n",
      "   GLU 244 A     4.57       4.50                      \n",
      "   GLU 250 A     2.96       4.50                      \n",
      "   GLU 299 A     4.11       4.50                      \n",
      "   GLU 308 A     5.24       4.50                      \n",
      "   GLU 311 A     3.29       4.50                      \n",
      "   GLU 315 A     4.29       4.50                      \n",
      "   GLU 316 A     5.33       4.50                      \n",
      "   GLU 320 A     4.89       4.50                      \n",
      "   GLU 343 A     4.70       4.50                      \n",
      "   C-  352 A     2.61       3.20                      \n",
      "   HIS   2 A     6.46       6.50                      \n",
      "   HIS  73 A     6.41       6.50                      \n",
      "   HIS  98 A     6.37       6.50                      \n",
      "   HIS 183 A     6.16       6.50                      \n",
      "   HIS 248 A     6.22       6.50                      \n",
      "   HIS 290 A     3.27       6.50                      \n",
      "   HIS 298 A     5.12       6.50                      \n",
      "   HIS 338 A     4.12       6.50                      \n",
      "   CYS  16 A    14.27       9.00                      \n",
      "   CYS  19 A    13.25       9.00                      \n",
      "   CYS 119 A     9.04       9.00                      \n",
      "   CYS 164 A    12.58       9.00                      \n",
      "   CYS 186 A     9.38       9.00                      \n",
      "   CYS 207 A    11.12       9.00                      \n",
      "   CYS 212 A     9.75       9.00                      \n",
      "   CYS 221 A    11.78       9.00                      \n",
      "   CYS 257 A     6.54       9.00                      \n",
      "   CYS 258 A    10.03       9.00                      \n",
      "   CYS 262 A    17.21       9.00                      \n",
      "   CYS 296 A    12.13       9.00                      \n",
      "   TYR   1 A    10.58      10.00                      \n",
      "   TYR 116 A    10.46      10.00                      \n",
      "   TYR 125 A    12.77      10.00                      \n",
      "   TYR 145 A    16.29      10.00                      \n",
      "   TYR 264 A    13.96      10.00                      \n",
      "   LYS   3 A    10.39      10.50                      \n",
      "   LYS  23 A     9.00      10.50                      \n",
      "   LYS  30 A    10.47      10.50                      \n",
      "   LYS  32 A    10.45      10.50                      \n",
      "   LYS  36 A    13.02      10.50                      \n",
      "   LYS  59 A    10.02      10.50                      \n",
      "   LYS  89 A    10.40      10.50                      \n",
      "   LYS  91 A    10.94      10.50                      \n",
      "   LYS 113 A    10.86      10.50                      \n",
      "   LYS 121 A    10.58      10.50                      \n",
      "   LYS 139 A    11.66      10.50                      \n",
      "   LYS 191 A    10.20      10.50                      \n",
      "   LYS 192 A    11.20      10.50                      \n",
      "   LYS 217 A    10.32      10.50                      \n",
      "   LYS 247 A    11.60      10.50                      \n",
      "   LYS 249 A    10.46      10.50                      \n",
      "   LYS 263 A    11.10      10.50                      \n",
      "   LYS 280 A    10.26      10.50                      \n",
      "   LYS 286 A    10.05      10.50                      \n",
      "   LYS 303 A    11.21      10.50                      \n",
      "   ARG  14 A    12.98      12.50                      \n",
      "   ARG  17 A    13.13      12.50                      \n",
      "   ARG  24 A    12.69      12.50                      \n",
      "   ARG  76 A    12.96      12.50                      \n",
      "   ARG 110 A    10.93      12.50                      \n",
      "   ARG 135 A    14.08      12.50                      \n",
      "   ARG 206 A    12.17      12.50                      \n",
      "   ARG 219 A    12.97      12.50                      \n",
      "   ARG 271 A    12.14      12.50                      \n",
      "   ARG 278 A    11.64      12.50                      \n",
      "   ARG 283 A    13.24      12.50                      \n",
      "   ARG 321 A    11.96      12.50                      \n",
      "   ARG 322 A    10.73      12.50                      \n",
      "   N+    1 A     7.87       8.00                      \n",
      "\n",
      "INFO:PROPKA information:\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "Free energy of   folding (kcal/mol) as a function of pH (using neutral reference)\n",
      "  0.00     34.30\n",
      "  1.00     33.99\n",
      "  2.00     32.12\n",
      "  3.00     25.92\n",
      "  4.00     18.89\n",
      "  5.00     20.28\n",
      "  6.00     24.08\n",
      "  7.00     26.09\n",
      "  8.00     28.36\n",
      "  9.00     32.60\n",
      " 10.00     42.82\n",
      " 11.00     57.76\n",
      " 12.00     71.32\n",
      " 13.00     81.81\n",
      " 14.00     89.66\n",
      "\n",
      "The pH of optimum stability is  4.3 for which the free energy is   18.4 kcal/mol at 298K\n",
      "Could not determine pH values where the free energy is within 80 % of minimum\n",
      "Could not determine the pH-range where the free energy is negative\n",
      "\n",
      "\n",
      "Protein charge of folded and unfolded state as a function of pH\n",
      "    pH  unfolded  folded\n",
      "  0.00     41.99   41.93\n",
      "  0.10     41.99   41.91\n",
      "  0.20     41.99   41.89\n",
      "  0.30     41.99   41.87\n",
      "  0.40     41.99   41.83\n",
      "  0.50     41.98   41.79\n",
      "  0.60     41.98   41.74\n",
      "  0.70     41.97   41.68\n",
      "  0.80     41.97   41.61\n",
      "  0.90     41.96   41.52\n",
      "  1.00     41.95   41.42\n",
      "  1.10     41.93   41.29\n",
      "  1.20     41.92   41.15\n",
      "  1.30     41.89   40.98\n",
      "  1.40     41.87   40.78\n",
      "  1.50     41.83   40.56\n",
      "  1.60     41.79   40.30\n",
      "  1.70     41.74   40.00\n",
      "  1.80     41.67   39.67\n",
      "  1.90     41.59   39.28\n",
      "  2.00     41.48   38.85\n",
      "  2.10     41.35   38.36\n",
      "  2.20     41.19   37.82\n",
      "  2.30     40.98   37.21\n",
      "  2.40     40.73   36.55\n",
      "  2.50     40.42   35.82\n",
      "  2.60     40.04   35.03\n",
      "  2.70     39.58   34.17\n",
      "  2.80     39.01   33.26\n",
      "  2.90     38.33   32.30\n",
      "  3.00     37.52   31.28\n",
      "  3.10     36.55   30.20\n",
      "  3.20     35.43   29.07\n",
      "  3.30     34.13   27.88\n",
      "  3.40     32.64   26.64\n",
      "  3.50     30.98   25.34\n",
      "  3.60     29.14   23.99\n",
      "  3.70     27.15   22.59\n",
      "  3.80     25.03   21.14\n",
      "  3.90     22.80   19.65\n",
      "  4.00     20.50   18.12\n",
      "  4.10     18.16   16.58\n",
      "  4.20     15.82   15.03\n",
      "  4.30     13.52   13.49\n",
      "  4.40     11.28   11.96\n",
      "  4.50      9.13   10.46\n",
      "  4.60      7.10    9.01\n",
      "  4.70      5.21    7.61\n",
      "  4.80      3.48    6.26\n",
      "  4.90      1.93    4.98\n",
      "  5.00      0.54    3.76\n",
      "  5.10     -0.69    2.62\n",
      "  5.20     -1.76    1.54\n",
      "  5.30     -2.70    0.52\n",
      "  5.40     -3.52   -0.43\n",
      "  5.50     -4.24   -1.33\n",
      "  5.60     -4.89   -2.18\n",
      "  5.70     -5.49   -2.99\n",
      "  5.80     -6.05   -3.76\n",
      "  5.90     -6.58   -4.51\n",
      "  6.00     -7.11   -5.23\n",
      "  6.10     -7.64   -5.93\n",
      "  6.20     -8.18   -6.60\n",
      "  6.30     -8.72   -7.24\n",
      "  6.40     -9.26   -7.86\n",
      "  6.50     -9.80   -8.44\n",
      "  6.60    -10.34   -8.98\n",
      "  6.70    -10.85   -9.48\n",
      "  6.80    -11.34   -9.94\n",
      "  6.90    -11.79  -10.36\n",
      "  7.00    -12.21  -10.74\n",
      "  7.10    -12.60  -11.08\n",
      "  7.20    -12.96  -11.39\n",
      "  7.30    -13.29  -11.69\n",
      "  7.40    -13.59  -11.96\n",
      "  7.50    -13.89  -12.23\n",
      "  7.60    -14.18  -12.49\n",
      "  7.70    -14.47  -12.75\n",
      "  7.80    -14.78  -13.01\n",
      "  7.90    -15.10  -13.27\n",
      "  8.00    -15.45  -13.54\n",
      "  8.10    -15.84  -13.82\n",
      "  8.20    -16.27  -14.10\n",
      "  8.30    -16.76  -14.39\n",
      "  8.40    -17.30  -14.69\n",
      "  8.50    -17.91  -15.00\n",
      "  8.60    -18.59  -15.33\n",
      "  8.70    -19.34  -15.67\n",
      "  8.80    -20.15  -16.04\n",
      "  8.90    -21.03  -16.42\n",
      "  9.00    -21.95  -16.84\n",
      "  9.10    -22.92  -17.29\n",
      "  9.20    -23.93  -17.77\n",
      "  9.30    -24.96  -18.30\n",
      "  9.40    -26.02  -18.88\n",
      "  9.50    -27.11  -19.51\n",
      "  9.60    -28.24  -20.20\n",
      "  9.70    -29.41  -20.95\n",
      "  9.80    -30.63  -21.77\n",
      "  9.90    -31.90  -22.67\n",
      " 10.00    -33.24  -23.64\n",
      " 10.10    -34.64  -24.67\n",
      " 10.20    -36.09  -25.77\n",
      " 10.30    -37.57  -26.92\n",
      " 10.40    -39.07  -28.12\n",
      " 10.50    -40.56  -29.34\n",
      " 10.60    -42.01  -30.59\n",
      " 10.70    -43.40  -31.84\n",
      " 10.80    -44.70  -33.09\n",
      " 10.90    -45.91  -34.32\n",
      " 11.00    -47.02  -35.53\n",
      " 11.10    -48.02  -36.72\n",
      " 11.20    -48.92  -37.88\n",
      " 11.30    -49.74  -39.01\n",
      " 11.40    -50.48  -40.11\n",
      " 11.50    -51.17  -41.18\n",
      " 11.60    -51.83  -42.22\n",
      " 11.70    -52.47  -43.23\n",
      " 11.80    -53.11  -44.22\n",
      " 11.90    -53.77  -45.19\n",
      " 12.00    -54.45  -46.13\n",
      " 12.10    -55.16  -47.06\n",
      " 12.20    -55.91  -47.96\n",
      " 12.30    -56.69  -48.85\n",
      " 12.40    -57.48  -49.72\n",
      " 12.50    -58.28  -50.57\n",
      " 12.60    -59.07  -51.42\n",
      " 12.70    -59.83  -52.25\n",
      " 12.80    -60.55  -53.07\n",
      " 12.90    -61.21  -53.86\n",
      " 13.00    -61.81  -54.64\n",
      " 13.10    -62.34  -55.39\n",
      " 13.20    -62.79  -56.10\n",
      " 13.30    -63.19  -56.77\n",
      " 13.40    -63.52  -57.41\n",
      " 13.50    -63.80  -58.00\n",
      " 13.60    -64.03  -58.55\n",
      " 13.70    -64.21  -59.05\n",
      " 13.80    -64.37  -59.52\n",
      " 13.90    -64.49  -59.95\n",
      " 14.00    -64.59  -60.34\n",
      "The pI is  5.35 (folded) and  5.04 (unfolded)\n",
      "\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "INFO:Applying pKa values at a pH of 8.20:\n",
      "INFO:Adding hydrogens to biomolecule.\n",
      "INFO:Debumping biomolecule (again).\n",
      "INFO:Optimizing hydrogen bonds\n",
      "INFO:Applying force field to biomolecule states.\n",
      "INFO:Regenerating headers.\n",
      "INFO:Regenerating PDB lines.\n",
      "WARNING:Ignoring 467 header lines in output.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "pH: 8.2\tname: 74987\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:PDB2PQR v3.6.2: biomolecular structure conversion software.\n",
      "INFO:Please cite:  Jurrus E, et al.  Improvements to the APBS biomolecular solvation software suite.  Protein Sci 27 112-128 (2018).\n",
      "INFO:Please cite:  Dolinsky TJ, et al.  PDB2PQR: expanding and upgrading automated preparation of biomolecular structures for molecular simulations. Nucleic Acids Res 35 W522-W525 (2007).\n",
      "INFO:Checking and transforming input arguments.\n",
      "INFO:Loading topology files.\n",
      "INFO:Loading molecule: /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/1ka1_hal2.pdb\n",
      "INFO:Setting up molecule.\n",
      "INFO:Created biomolecule object with 354 residues and 2718 atoms.\n",
      "WARNING:Multiple occupancies found: OG in SER A 84.\n",
      "WARNING:Multiple occupancies found in SER A 84. At least one of the instances is being ignored.\n",
      "INFO:Setting termini states for biomolecule chains.\n",
      "INFO:Loading forcefield.\n",
      "INFO:Loading hydrogen topology definitions.\n",
      "WARNING:Missing atom CB in residue LYS A 32\n",
      "WARNING:Missing atom CG in residue LYS A 32\n",
      "WARNING:Missing atom CD in residue LYS A 32\n",
      "WARNING:Missing atom CE in residue LYS A 32\n",
      "WARNING:Missing atom NZ in residue LYS A 32\n",
      "WARNING:Missing atom CB in residue ASP A 33\n",
      "WARNING:Missing atom CG in residue ASP A 33\n",
      "WARNING:Missing atom OD1 in residue ASP A 33\n",
      "WARNING:Missing atom OD2 in residue ASP A 33\n",
      "WARNING:Missing atom CG in residue LYS A 189\n",
      "WARNING:Missing atom CD in residue LYS A 189\n",
      "WARNING:Missing atom CE in residue LYS A 189\n",
      "WARNING:Missing atom NZ in residue LYS A 189\n",
      "WARNING:Missing atom CD in residue LYS A 229\n",
      "WARNING:Missing atom CE in residue LYS A 229\n",
      "WARNING:Missing atom NZ in residue LYS A 229\n",
      "WARNING:Missing atom OXT in residue ARG A 355\n",
      "WARNING:Missing atom CB in residue LYS A 32\n",
      "WARNING:Missing atom CG in residue LYS A 32\n",
      "WARNING:Missing atom CD in residue LYS A 32\n",
      "WARNING:Missing atom CE in residue LYS A 32\n",
      "WARNING:Missing atom NZ in residue LYS A 32\n",
      "WARNING:Missing atom CB in residue ASP A 33\n",
      "WARNING:Missing atom CG in residue ASP A 33\n",
      "WARNING:Missing atom OD1 in residue ASP A 33\n",
      "WARNING:Missing atom OD2 in residue ASP A 33\n",
      "WARNING:Missing atom CG in residue LYS A 189\n",
      "WARNING:Missing atom CD in residue LYS A 189\n",
      "WARNING:Missing atom CE in residue LYS A 189\n",
      "WARNING:Missing atom NZ in residue LYS A 189\n",
      "WARNING:Missing atom CD in residue LYS A 229\n",
      "WARNING:Missing atom CE in residue LYS A 229\n",
      "WARNING:Missing atom NZ in residue LYS A 229\n",
      "WARNING:Missing atom OXT in residue ARG A 355\n",
      "INFO:Attempting to repair 17 missing atoms in biomolecule.\n",
      "WARNING:Missing atom CB in residue LYS A 32\n",
      "WARNING:Missing atom CG in residue LYS A 32\n",
      "WARNING:Missing atom CD in residue LYS A 32\n",
      "WARNING:Missing atom CE in residue LYS A 32\n",
      "WARNING:Missing atom NZ in residue LYS A 32\n",
      "WARNING:Missing atom CB in residue ASP A 33\n",
      "WARNING:Missing atom CG in residue ASP A 33\n",
      "WARNING:Missing atom OD1 in residue ASP A 33\n",
      "WARNING:Missing atom OD2 in residue ASP A 33\n",
      "WARNING:Missing atom CG in residue LYS A 189\n",
      "WARNING:Missing atom CD in residue LYS A 189\n",
      "WARNING:Missing atom CE in residue LYS A 189\n",
      "WARNING:Missing atom NZ in residue LYS A 189\n",
      "WARNING:Missing atom CD in residue LYS A 229\n",
      "WARNING:Missing atom CE in residue LYS A 229\n",
      "WARNING:Missing atom NZ in residue LYS A 229\n",
      "WARNING:Missing atom OXT in residue ARG A 355\n",
      "INFO:Added atom CB to residue LYS A 32 at coordinates -45.594, -33.973, -23.265\n",
      "INFO:Added atom CG to residue LYS A 32 at coordinates -45.153, -35.334, -22.744\n",
      "INFO:Added atom CD to residue LYS A 32 at coordinates -43.702, -35.621, -23.091\n",
      "INFO:Added atom CE to residue LYS A 32 at coordinates -43.239, -36.968, -22.584\n",
      "INFO:Added atom NZ to residue LYS A 32 at coordinates -41.819, -37.204, -22.945\n",
      "INFO:Added atom CB to residue ASP A 33 at coordinates -49.706, -30.631, -25.302\n",
      "INFO:Added atom CG to residue ASP A 33 at coordinates -48.707, -31.048, -26.377\n",
      "INFO:Added atom OD1 to residue ASP A 33 at coordinates -47.538, -31.342, -26.061\n",
      "INFO:Added atom OD2 to residue ASP A 33 at coordinates -49.132, -31.087, -27.541\n",
      "INFO:Added atom CG to residue LYS A 189 at coordinates -35.105, -63.964, -9.656\n",
      "INFO:Added atom CD to residue LYS A 189 at coordinates -35.396, -65.193, -8.824\n",
      "INFO:Added atom CE to residue LYS A 189 at coordinates -34.533, -66.375, -9.200\n",
      "INFO:Added atom NZ to residue LYS A 189 at coordinates -34.854, -67.554, -8.358\n",
      "INFO:Added atom CD to residue LYS A 229 at coordinates -3.896, -48.760, -18.063\n",
      "INFO:Added atom CE to residue LYS A 229 at coordinates -3.416, -50.110, -17.576\n",
      "INFO:Added atom NZ to residue LYS A 229 at coordinates -2.176, -50.508, -18.287\n",
      "INFO:Added atom OXT to residue ARG A 355 at coordinates -9.183, -18.959, -17.852\n",
      "INFO:Updating disulfide bridges.\n",
      "INFO:Debumping biomolecule.\n",
      "INFO:Assigning titration states with PROPKA.\n",
      "INFO:\n",
      "propka3.5.1                                                                                  2024-07-26\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "--                                                                           --\n",
      "--  PROPKA: A PROTEIN PKA PREDICTOR                                          --\n",
      "--                                                                           --\n",
      "--  VERSION 1.0,  04/25/2004,  IOWA CITY                                     --\n",
      "--  BY HUI LI                                                                --\n",
      "--                                                                           --\n",
      "--  VERSION 2.0,  11/05/2007, IOWA CITY/COPENHAGEN                           --\n",
      "--  BY DELPHINE C. BAS AND DAVID M. ROGERS                                   --\n",
      "--                                                                           --\n",
      "--  VERSION 3.0,  01/06/2011, COPENHAGEN                                     --\n",
      "--  BY MATS H.M. OLSSON AND CHRESTEN R. SONDERGARD                           --\n",
      "--                                                                           --\n",
      "--  VERSION 3.1,  07/01/2011, COPENHAGEN                                     --\n",
      "--  BY CHRESTEN R. SONDERGARD AND MATS H.M. OLSSON                           --\n",
      "--                                                                           --\n",
      "--  VERSION 3.2,  06/17/2020, PLANET EARTH                                   --\n",
      "--  SEE ABOVE FOR AUTHORS                                                    --\n",
      "--                                                                           --\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "References:\n",
      "\n",
      "Very Fast Empirical Prediction and Rationalization of Protein pKa Values.\n",
      "Hui Li, Andrew D. Robertson and Jan H. Jensen. PROTEINS: Structure, Function,\n",
      "and Bioinformatics. 61:704-721 (2005)\n",
      "\n",
      "Very Fast Prediction and Rationalization of pKa Values for Protein-Ligand\n",
      "Complexes.  Delphine C. Bas, David M. Rogers and Jan H. Jensen.  PROTEINS:\n",
      "Structure, Function, and Bioinformatics 73:765-783 (2008)\n",
      "\n",
      "PROPKA3: Consistent Treatment of Internal and Surface Residues in Empirical\n",
      "pKa predictions.  Mats H.M. Olsson, Chresten R. Sondergard, Michal Rostkowski,\n",
      "and Jan H. Jensen.  Journal of Chemical Theory and Computation, 7(2):525-537\n",
      "(2011)\n",
      "\n",
      "Improved Treatment of Ligands and Coupling Effects in Empirical Calculation\n",
      "and Rationalization of pKa Values.  Chresten R. Sondergaard, Mats H.M. Olsson,\n",
      "Michal Rostkowski, and Jan H. Jensen.  Journal of Chemical Theory and\n",
      "Computation, (2011)\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "\n",
      "WARNING:Missing atoms or failed protonation for HIS  31 A (HIS) -- please check the structure\n",
      "WARNING:Group (HIS) for   231-  CG    31-HIS (A) [ -51.746  -37.773  -23.250] C\n",
      "WARNING:Expected 4 interaction atoms for acids, found:\n",
      "WARNING:                 0- HE2    31-HIS (A) [ -54.068  -39.642  -24.334] H\n",
      "WARNING:               233- ND1    31-HIS (A) [ -51.129  -38.973  -23.536] N\n",
      "WARNING:               235- NE2    31-HIS (A) [ -53.201  -39.225  -24.025] N\n",
      "WARNING:Expected 2 interaction atoms for bases, found:\n",
      "WARNING:               233- ND1    31-HIS (A) [ -51.129  -38.973  -23.536] N\n",
      "WARNING:               235- NE2    31-HIS (A) [ -53.201  -39.225  -24.025] N\n",
      "INFO:Coupling map:\n",
      "\n",
      "INFO:Calculating pKas for Conformation container 1A with 3249 atoms and 890 groups\n",
      "INFO:pKa determinants:\n",
      "\n",
      "---------  -----   ------   ---------------------    --------------    --------------    --------------\n",
      "                            DESOLVATION  EFFECTS       SIDECHAIN          BACKBONE        COULOMBIC\n",
      " RESIDUE    pKa    BURIED     REGULAR      RE        HYDROGEN BOND     HYDROGEN BOND      INTERACTION\n",
      "---------  -----   ------   ---------   ---------    --------------    --------------    --------------\n",
      "\n",
      "ASP  33 A   3.91     0 %    0.20  152   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS  32 A\n",
      "ASP  33 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 HIS  31 A\n",
      "\n",
      "ASP  41 A   2.21    11 %    0.60  312   0.00    0   -0.79 SER  43 A   -0.23 ASP  41 A   -0.28 LYS  39 A\n",
      "ASP  41 A                                            0.00 XXX   0 X   -0.73 SER  43 A   -0.04 LYS 148 A\n",
      "ASP  41 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.12 LYS 291 A\n",
      "ASP  41 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 HIS 241 A\n",
      "\n",
      "ASP  49 A   6.17*  100 %    3.70  606   1.05    0   -0.84 THR  46 A   -0.70 THR  46 A   -0.30 LYS  39 A\n",
      "ASP  49 A                                           -0.85 THR 147 A   -0.24 ASP  49 A    0.55 GLU  72 A\n",
      "\n",
      "ASP  66 A   3.97     0 %    0.19  204   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.11 LYS  61 A\n",
      "ASP  66 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP  67 A\n",
      "\n",
      "ASP  67 A   3.50    45 %    1.80  406   0.24    0   -0.81 TRP 139 A    0.00 XXX   0 X   -0.03 ARG   5 A\n",
      "ASP  67 A                                           -0.70 ARG 137 A    0.00 XXX   0 X   -0.15 LYS  61 A\n",
      "ASP  67 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 LYS  68 A\n",
      "ASP  67 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.57 ARG 137 A\n",
      "\n",
      "ASP  80 A   3.08     0 %    0.24  175   0.00    0   -0.70 ARG 122 A    0.00 XXX   0 X   -0.26 ARG 122 A\n",
      "\n",
      "ASP  94 A   2.91    69 %    1.87  474   0.67    0   -0.85 LYS  16 A    0.00 XXX   0 X   -0.00 ARG  23 A\n",
      "ASP  94 A                                           -0.85 LYS  91 A    0.00 XXX   0 X   -0.00 LYS 116 A\n",
      "ASP  94 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.05 LYS  16 A\n",
      "ASP  94 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.67 LYS  91 A\n",
      "\n",
      "ASP 104 A   3.88     0 %    0.20  138   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.04 LYS 102 A\n",
      "ASP 104 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 LYS 103 A\n",
      "\n",
      "ASP 105 A   4.10     0 %    0.13  159   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG   5 A\n",
      "ASP 105 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS 103 A\n",
      "ASP 105 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.21 ASP 104 A\n",
      "\n",
      "ASP 111 A   3.81     0 %    0.12  140   0.00    0    0.00 XXX   0 X   -0.11 ASP 111 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 120 A   2.75    13 %    0.72  318   0.05    0   -0.32 SER 117 A   -0.79 LYS 116 A   -0.02 LYS 116 A\n",
      "ASP 120 A                                            0.00 XXX   0 X   -0.69 SER 117 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 126 A   3.39    34 %    1.22  376   0.28    0   -0.16 SER  74 A   -0.85 SER  75 A   -0.06 ARG 122 A\n",
      "ASP 126 A                                           -0.85 SER  75 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "ASP 142 A  16.43*  100 %    3.57  642   1.03    0    0.48 GLU  72 A   -0.03 ILE 144 A   -0.28 LYS  39 A\n",
      "ASP 142 A                                            0.67 ASP 145 A    0.00 XXX   0 X    0.50 ASP  49 A\n",
      "ASP 142 A                                            0.37 ASP 294 A    0.00 XXX   0 X    0.17 CYS 140 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.88 CYS 159 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.87 GLU  72 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.59 GLU  73 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.57 ASP 145 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.23 ASP 294 A\n",
      "\n",
      "ASP 145 A   3.62*  100 %    2.82  672   0.55    0   -0.67 ASP 142 A   -0.04 GLY 146 A   -0.19 HIS 241 A\n",
      "ASP 145 A                                           -0.73 ASP 294 A   -0.57 GLN 265 A    0.17 ASP 263 A\n",
      "ASP 145 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 GLU 290 A\n",
      "ASP 145 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.44 HIS 295 A\n",
      "ASP 145 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.65 LYS 267 A\n",
      "ASP 145 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.55 ARG 281 A\n",
      "\n",
      "ASP 165 A   3.15     0 %    0.28  211   0.00    0    0.00 XXX   0 X   -0.81 GLY 136 A   -0.03 LYS  68 A\n",
      "ASP 165 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.00 ARG 134 A\n",
      "ASP 165 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 LYS 135 A\n",
      "\n",
      "ASP 187 A   2.15    10 %    1.13  308   0.05    0   -0.17 SER 181 A   -0.76 LEU 180 A   -0.41 HIS 191 A\n",
      "ASP 187 A                                           -0.85 TYR 196 A   -0.63 SER 181 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 213 A   3.05    18 %    0.79  331   0.14    0    0.00 XXX   0 X   -0.76 GLU 215 A   -0.09 ARG  15 A\n",
      "ASP 213 A                                            0.00 XXX   0 X   -0.84 SER 216 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 227 A   3.07     0 %    0.35  211   0.00    0    0.00 XXX   0 X   -0.84 LYS 229 A   -0.24 LYS 229 A\n",
      "\n",
      "ASP 230 A   3.96     0 %    0.30  181   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.03 ARG 223 A\n",
      "ASP 230 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.20 LYS 229 A\n",
      "ASP 230 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.09 LYS 258 A\n",
      "ASP 230 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP 227 A\n",
      "ASP 230 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 277 A\n",
      "\n",
      "ASP 245 A   3.84     0 %    0.18  234   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.09 LYS 239 A\n",
      "ASP 245 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 HIS 244 A\n",
      "ASP 245 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 GLU 246 A\n",
      "\n",
      "ASP 263 A   3.38    62 %    2.13  455   0.38    0   -1.55 ARG 152 A   -0.09 ASP 263 A   -0.03 LYS 148 A\n",
      "ASP 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.50 LYS 267 A\n",
      "ASP 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.20 HIS 241 A\n",
      "ASP 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.57 ARG 152 A\n",
      "\n",
      "ASP 277 A   3.47    56 %    2.43  438   0.35    0   -1.64 ARG 223 A   -0.78 ILE 232 A   -0.04 LYS 258 A\n",
      "ASP 277 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 HIS 224 A\n",
      "ASP 277 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.61 ARG 223 A\n",
      "\n",
      "ASP 294 A   9.23*  100 %    2.63  612   0.38    0   -0.37 ASP 142 A    0.00 XXX   0 X   -0.04 LYS  39 A\n",
      "ASP 294 A                                            0.73 ASP 145 A    0.00 XXX   0 X   -0.08 LYS 267 A\n",
      "ASP 294 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.35 LYS 291 A\n",
      "ASP 294 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.93 GLU  72 A\n",
      "ASP 294 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.64 ASP 145 A\n",
      "ASP 294 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.87 ARG 281 A\n",
      "ASP 294 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.84 GLU 290 A\n",
      "\n",
      "ASP 311 A   3.81    51 %    2.24  424   0.49    0    0.00 XXX   0 X   -0.12 ASP 311 A   -0.05 HIS 341 A\n",
      "ASP 311 A                                            0.00 XXX   0 X   -0.61 MET 313 A    0.00 XXX   0 X\n",
      "ASP 311 A                                            0.00 XXX   0 X   -0.83 GLU 314 A    0.00 XXX   0 X\n",
      "ASP 311 A                                            0.00 XXX   0 X   -0.40 ASP 315 A    0.00 XXX   0 X\n",
      "ASP 311 A                                            0.00 XXX   0 X   -0.72 VAL 316 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 315 A   6.34*   20 %    0.96  336   0.20    0    0.87 ASP 342 A    0.00 XXX   0 X   -0.04 ARG 338 A\n",
      "ASP 315 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.14 ASP 311 A\n",
      "ASP 315 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.40 ASP 342 A\n",
      "\n",
      "ASP 319 A   1.83    15 %    0.73  324   0.03    0   -1.58 ARG 134 A   -0.76 GLY 321 A   -0.38 ARG 134 A\n",
      "\n",
      "ASP 342 A   3.08*    0 %    0.48  257   0.00    0   -0.87 ASP 315 A    0.00 XXX   0 X   -0.15 ARG 338 A\n",
      "ASP 342 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.18 HIS 341 A\n",
      "\n",
      "ASP 350 A   3.76     0 %    0.21  206   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.25 LYS 253 A\n",
      "\n",
      "GLU   4 A   2.92     0 %    0.28  187   0.00    0   -0.77 N+    2 A   -0.70 GLU   4 A   -0.38 N+    2 A\n",
      "\n",
      "GLU   6 A   3.84    24 %    1.04  349   0.00    0   -0.65 ARG 137 A   -0.66 GLU   6 A   -0.05 N+    2 A\n",
      "GLU   6 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.12 ARG   5 A\n",
      "GLU   6 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU   4 A\n",
      "GLU   6 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.14 ASP  67 A\n",
      "GLU   6 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.43 ARG 137 A\n",
      "\n",
      "GLU  27 A   4.11    13 %    0.87  318   0.20    0   -0.85 SER  26 A   -0.45 GLU  27 A   -0.00 LYS  22 A\n",
      "GLU  27 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.09 ARG  23 A\n",
      "GLU  27 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.07 HIS  31 A\n",
      "\n",
      "GLU  72 A   5.47*  100 %    2.36  562   0.17    0   -0.34 LYS  39 A    0.00 XXX   0 X   -0.16 LYS 291 A\n",
      "GLU  72 A                                           -0.48 ASP 142 A    0.00 XXX   0 X   -0.05 ARG 324 A\n",
      "GLU  72 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.14 ASP  41 A\n",
      "GLU  72 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.42 ASP 145 A\n",
      "GLU  72 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.10 LYS  39 A\n",
      "\n",
      "GLU  73 A   8.38*   90 %    2.95  532   0.08    0   -0.85 LYS  39 A    0.00 XXX   0 X   -0.01 ARG 324 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 ASP  41 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.00 LYS  39 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.09 ASP  49 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.55 GLU  72 A\n",
      "\n",
      "GLU  89 A   5.51    46 %    1.01  409   0.11    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.20 ARG  23 A\n",
      "GLU  89 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.12 GLU  27 A\n",
      "GLU  89 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 HIS  31 A\n",
      "\n",
      "GLU  95 A   3.70     0 %    0.31  175   0.00    0   -0.68 LYS  99 A    0.00 XXX   0 X   -0.05 ARG  23 A\n",
      "GLU  95 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.38 LYS  99 A\n",
      "\n",
      "GLU 119 A   4.30     0 %    0.29  205   0.00    0    0.00 XXX   0 X   -0.63 GLU 119 A   -0.10 ARG 122 A\n",
      "GLU 119 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 ASP  80 A\n",
      "GLU 119 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 ASP 120 A\n",
      "\n",
      "GLU 131 A   4.45     1 %    0.36  283   0.01    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.41 ARG 324 A\n",
      "\n",
      "GLU 154 A   4.04     2 %    0.35  287   0.01    0   -0.72 ARG 152 A    0.00 XXX   0 X    0.02 GLU 238 A\n",
      "GLU 154 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.18 ASP 263 A\n",
      "GLU 154 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.31 ARG 152 A\n",
      "\n",
      "GLU 192 A   3.88     0 %    0.35  255   0.00    0    0.00 XXX   0 X   -0.04 GLY 190 A    0.02 ASP 187 A\n",
      "GLU 192 A                                            0.00 XXX   0 X   -0.73 GLU 192 A   -0.23 HIS 191 A\n",
      "\n",
      "GLU 215 A   3.63    10 %    0.43  309   0.07    0   -0.85 LYS 103 A    0.00 XXX   0 X   -0.16 ARG  15 A\n",
      "GLU 215 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.05 ARG 199 A\n",
      "GLU 215 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP 213 A\n",
      "GLU 215 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.41 LYS 103 A\n",
      "\n",
      "GLU 235 A   3.26    68 %    2.28  472   0.12    0   -0.46 THR 233 A    0.00 XXX   0 X   -0.30 HIS 244 A\n",
      "GLU 235 A                                           -0.85 SER 259 A    0.00 XXX   0 X   -0.65 LYS 251 A\n",
      "GLU 235 A                                           -0.85 LYS 251 A    0.00 XXX   0 X   -0.53 HIS 261 A\n",
      "\n",
      "GLU 238 A   3.63    15 %    0.62  323   0.00    0   -0.74 ARG 152 A    0.00 XXX   0 X   -0.35 LYS 148 A\n",
      "GLU 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.30 HIS 241 A\n",
      "GLU 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 HIS 244 A\n",
      "GLU 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.24 ASP 263 A\n",
      "GLU 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.33 ARG 152 A\n",
      "\n",
      "GLU 246 A   2.95    20 %    1.10  337   0.12    0   -0.73 SER 243 A    0.00 XXX   0 X   -0.43 ARG 355 A\n",
      "GLU 246 A                                           -1.62 ARG 355 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "GLU 290 A   1.29    98 %    3.05  556   0.08    0   -0.59 ARG 281 A    0.00 XXX   0 X   -0.00 LYS 267 A\n",
      "GLU 290 A                                           -1.60 HIS 295 A    0.00 XXX   0 X   -0.23 LYS 291 A\n",
      "GLU 290 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 LYS 329 A\n",
      "GLU 290 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.10 HIS 241 A\n",
      "GLU 290 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -2.03 ARG 281 A\n",
      "GLU 290 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.70 HIS 295 A\n",
      "\n",
      "GLU 304 A   3.18    26 %    1.06  354   0.21    0   -0.46 HIS 303 A   -0.84 GLY 203 A   -0.19 ARG 202 A\n",
      "GLU 304 A                                            0.00 XXX   0 X   -0.00 LEU 204 A   -0.05 LYS 219 A\n",
      "GLU 304 A                                            0.00 XXX   0 X   -0.74 GLY 205 A   -0.31 HIS 303 A\n",
      "\n",
      "GLU 314 A   5.06     0 %    0.26  273   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.01 LYS 329 A\n",
      "GLU 314 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.31 ASP 311 A\n",
      "\n",
      "GLU 339 A   4.02     0 %    0.19  164   0.00    0    0.00 XXX   0 X   -0.67 GLU 339 A   -0.07 ARG 338 A\n",
      "GLU 339 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP 342 A\n",
      "\n",
      "C-  355 A   3.34     0 %    0.16  125   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.07 ARG 355 A\n",
      "C-  355 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 246 A\n",
      "\n",
      "HIS  31 A   6.33     0 %   -0.29  196   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.07 GLU  27 A\n",
      "HIS  31 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP  33 A\n",
      "HIS  31 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU  89 A\n",
      "\n",
      "HIS 191 A   6.82     6 %   -0.67  298   0.00    0    0.00 XXX   0 X    0.37 LEU 188 A   -0.02 LYS  22 A\n",
      "HIS 191 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.41 ASP 187 A\n",
      "HIS 191 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.23 GLU 192 A\n",
      "\n",
      "HIS 221 A   5.75    12 %   -0.53  315   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 219 A\n",
      "HIS 221 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.17 HIS 224 A\n",
      "\n",
      "HIS 224 A   6.57    11 %   -0.55  311   0.00    0    0.00 XXX   0 X    0.58 SER 335 A    0.04 ASP 277 A\n",
      "\n",
      "HIS 241 A   4.98    65 %   -1.23  464   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.26 LYS 148 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 ARG 152 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.17 LYS 267 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.56 ARG 281 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP  41 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 ASP 145 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.30 GLU 238 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 HIS 244 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 ASP 263 A\n",
      "HIS 241 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 GLU 290 A\n",
      "\n",
      "HIS 244 A   5.80    28 %   -0.91  359   0.00    0    0.00 XXX   0 X    0.04 GLY 236 A   -0.09 LYS 239 A\n",
      "HIS 244 A                                            0.00 XXX   0 X    0.22 VAL 237 A   -0.10 LYS 251 A\n",
      "HIS 244 A                                            0.00 XXX   0 X    0.01 SER 242 A   -0.01 LYS 267 A\n",
      "HIS 244 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.30 GLU 235 A\n",
      "HIS 244 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 238 A\n",
      "HIS 244 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP 245 A\n",
      "HIS 244 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.27 HIS 261 A\n",
      "\n",
      "HIS 261 A   6.60    35 %   -0.93  379   0.00    0    0.00 XXX   0 X    0.85 GLY 236 A   -0.29 LYS 251 A\n",
      "HIS 261 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 267 A\n",
      "HIS 261 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.53 GLU 235 A\n",
      "\n",
      "HIS 295 A   5.60   100 %   -3.33  658   0.00    0    1.60 GLU 290 A    0.00 XXX   0 X   -0.15 LYS 267 A\n",
      "HIS 295 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.16 ARG 281 A\n",
      "HIS 295 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.00 LYS 291 A\n",
      "HIS 295 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.44 ASP 145 A\n",
      "HIS 295 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.70 GLU 290 A\n",
      "\n",
      "HIS 303 A   6.22    34 %   -0.66  377   0.00    0    0.46 GLU 304 A    0.00 XXX   0 X   -0.39 ARG 202 A\n",
      "HIS 303 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.31 GLU 304 A\n",
      "\n",
      "HIS 309 A   2.56   100 %   -3.24  632   0.00    0    0.00 XXX   0 X    0.06 HIS 295 A   -0.74 HIS 295 A\n",
      "HIS 309 A                                            0.00 XXX   0 X    0.23 ILE 332 A   -0.25 HIS 341 A\n",
      "\n",
      "HIS 341 A   4.22    77 %   -2.52  498   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 311 A\n",
      "HIS 341 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.18 ASP 342 A\n",
      "\n",
      "CYS 140 A  12.78   100 %    3.75  624   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.13 ARG 324 A\n",
      "CYS 140 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 ASP 294 A\n",
      "\n",
      "CYS 159 A  15.19   100 %    3.86  700   0.00    0   -0.06 GLN 265 A    0.00 XXX   0 X   -0.03 LYS 267 A\n",
      "CYS 159 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 CYS 140 A\n",
      "CYS 159 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 CYS 172 A\n",
      "CYS 159 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.30 ASP 294 A\n",
      "CYS 159 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.32 ASP 145 A\n",
      "CYS 159 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.53 CYS 175 A\n",
      "\n",
      "CYS 172 A  12.08   100 %    3.66  715   0.00    0   -0.14 THR  11 A    0.00 XXX   0 X   -0.50 ARG 199 A\n",
      "CYS 172 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 CYS 175 A\n",
      "\n",
      "CYS 175 A   9.09*  100 %    3.67  714   0.00    0   -3.60 CYS 269 A    0.00 XXX   0 X    0.02 ASP 145 A\n",
      "\n",
      "CYS 269 A  19.48*  100 %    4.24  712   0.00    0    3.60 CYS 175 A   -0.02 CYS 269 A    0.49 CYS 159 A\n",
      "CYS 269 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 CYS 172 A\n",
      "CYS 269 A                                            0.00 XXX   0 X    0.00 XXX   0 X    2.03 CYS 175 A\n",
      "\n",
      "CYS 349 A   9.57     1 %    0.40  284   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 311 A\n",
      "CYS 349 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 314 A\n",
      "CYS 349 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.11 ASP 350 A\n",
      "\n",
      "TYR  50 A  12.47    60 %    1.83  448   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.23 ASP  49 A\n",
      "TYR  50 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU  72 A\n",
      "TYR  50 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 126 A\n",
      "TYR  50 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.29 LYS  39 A\n",
      "TYR  50 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.60 GLU  73 A\n",
      "\n",
      "TYR  97 A  12.15    89 %    3.48  531   0.00    0   -0.35 ARG  15 A    0.00 XXX   0 X   -1.07 ARG  15 A\n",
      "TYR  97 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP  94 A\n",
      "TYR  97 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 215 A\n",
      "TYR  97 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 TYR 101 A\n",
      "\n",
      "TYR 101 A  10.24    26 %    0.81  354   0.00    0   -0.53 SER 193 A    0.00 XXX   0 X   -0.10 ARG  15 A\n",
      "TYR 101 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 GLU 192 A\n",
      "\n",
      "TYR 130 A  10.10     4 %    0.46  293   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.36 LYS  68 A\n",
      "\n",
      "TYR 183 A  11.75    61 %    1.58  453   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 154 A\n",
      "TYR 183 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 CYS 175 A\n",
      "TYR 183 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP 187 A\n",
      "TYR 183 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 263 A\n",
      "\n",
      "TYR 196 A  13.17    38 %    2.33  389   0.00    0    0.85 ASP 187 A   -0.79 LEU 188 A    0.05 GLU 192 A\n",
      "TYR 196 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.27 TYR 208 A\n",
      "TYR 196 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.48 ASP 187 A\n",
      "\n",
      "TYR 208 A  11.41    46 %    1.34  411   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.07 ASP 187 A\n",
      "\n",
      "TYR 268 A  17.93   100 %    4.86  684   0.00    0    0.77 TYR 279 A    0.00 XXX   0 X    0.17 ASP 145 A\n",
      "TYR 268 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 CYS 159 A\n",
      "TYR 268 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.54 ARG 281 A\n",
      "TYR 268 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.46 GLU 290 A\n",
      "TYR 268 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 ASP 294 A\n",
      "TYR 268 A                                            0.00 XXX   0 X    0.00 XXX   0 X    2.03 TYR 279 A\n",
      "\n",
      "TYR 279 A  12.08   100 %    3.49  678   0.00    0   -0.51 ARG 281 A    0.00 XXX   0 X    0.50 ASP 145 A\n",
      "TYR 279 A                                           -0.77 TYR 268 A    0.00 XXX   0 X    0.01 CYS 159 A\n",
      "TYR 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.75 ARG 281 A\n",
      "TYR 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.36 ASP 294 A\n",
      "TYR 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.76 GLU 290 A\n",
      "\n",
      "TYR 288 A  12.91    68 %    1.44  472   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP  41 A\n",
      "TYR 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.17 GLU  72 A\n",
      "TYR 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP 142 A\n",
      "TYR 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 ASP 145 A\n",
      "TYR 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.51 ARG 281 A\n",
      "TYR 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.80 GLU 290 A\n",
      "TYR 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.85 ASP 294 A\n",
      "\n",
      "LYS  16 A   9.20    87 %   -2.94  524   0.00    0    0.85 ASP  94 A    0.00 XXX   0 X   -0.26 LYS  91 A\n",
      "LYS  16 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.05 ASP  94 A\n",
      "\n",
      "LYS  22 A   9.53    44 %   -0.97  405   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 GLU  27 A\n",
      "\n",
      "LYS  32 A  10.25     0 %   -0.31  236   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.06 ASP  33 A\n",
      "\n",
      "LYS  39 A  13.17    62 %   -1.79  454   0.00    0    0.34 GLU  72 A    0.00 XXX   0 X    0.28 ASP  41 A\n",
      "LYS  39 A                                            0.85 GLU  73 A    0.00 XXX   0 X    0.30 ASP  49 A\n",
      "LYS  39 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.28 ASP 142 A\n",
      "LYS  39 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 294 A\n",
      "LYS  39 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 324 A\n",
      "LYS  39 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.29 TYR  50 A\n",
      "LYS  39 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.10 GLU  72 A\n",
      "LYS  39 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.00 GLU  73 A\n",
      "\n",
      "LYS  61 A   9.38    37 %   -1.38  385   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.11 ASP  66 A\n",
      "LYS  61 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP  67 A\n",
      "\n",
      "LYS  68 A  10.47     6 %   -0.46  298   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP  67 A\n",
      "LYS  68 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 ARG 137 A\n",
      "LYS  68 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 165 A\n",
      "LYS  68 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.36 TYR 130 A\n",
      "\n",
      "LYS  91 A  11.20    36 %   -0.82  382   0.00    0    0.85 ASP  94 A    0.00 XXX   0 X    0.67 ASP  94 A\n",
      "\n",
      "LYS  99 A  11.34     0 %   -0.22  141   0.00    0    0.68 GLU  95 A    0.00 XXX   0 X   -0.00 ARG  23 A\n",
      "LYS  99 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 GLU  95 A\n",
      "\n",
      "LYS 102 A  10.42     0 %   -0.12  190   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 104 A\n",
      "\n",
      "LYS 103 A  11.30     6 %   -0.45  298   0.00    0    0.85 GLU 215 A    0.00 XXX   0 X   -0.09 ARG  15 A\n",
      "LYS 103 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP 104 A\n",
      "LYS 103 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 105 A\n",
      "LYS 103 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 199 A\n",
      "LYS 103 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.41 GLU 215 A\n",
      "\n",
      "LYS 116 A  10.23     0 %   -0.12  213   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP  94 A\n",
      "LYS 116 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 120 A\n",
      "LYS 116 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.18 LYS  91 A\n",
      "\n",
      "LYS 135 A  10.54     0 %   -0.04   82   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP 165 A\n",
      "\n",
      "LYS 148 A  10.50    12 %   -0.36  314   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP  41 A\n",
      "LYS 148 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.05 ARG 152 A\n",
      "LYS 148 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.35 GLU 238 A\n",
      "LYS 148 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 263 A\n",
      "\n",
      "LYS 189 A  10.46     0 %   -0.04  110   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 219 A  10.37     0 %   -0.19  220   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 304 A\n",
      "\n",
      "LYS 226 A  10.46     0 %   -0.04   96   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 229 A  10.85     0 %   -0.10  111   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.24 ASP 227 A\n",
      "LYS 229 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 ASP 230 A\n",
      "\n",
      "LYS 239 A  10.51     0 %   -0.08  150   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP 245 A\n",
      "\n",
      "LYS 251 A  10.84    33 %   -1.16  374   0.00    0    0.85 GLU 235 A    0.00 XXX   0 X    0.65 GLU 235 A\n",
      "\n",
      "LYS 253 A  10.54     0 %   -0.21  219   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.25 ASP 350 A\n",
      "\n",
      "LYS 258 A  10.36     0 %   -0.17  226   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.10 ARG 223 A\n",
      "LYS 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP 230 A\n",
      "LYS 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 277 A\n",
      "\n",
      "LYS 267 A   7.64   100 %   -3.32  608   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.00 ARG 152 A\n",
      "LYS 267 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 CYS 159 A\n",
      "LYS 267 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.50 ASP 263 A\n",
      "LYS 267 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.79 ARG 281 A\n",
      "LYS 267 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 GLU 290 A\n",
      "LYS 267 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP 294 A\n",
      "LYS 267 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.65 ASP 145 A\n",
      "\n",
      "LYS 285 A  10.40     0 %   -0.10  190   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 291 A  10.01    47 %   -1.26  413   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.12 ASP  41 A\n",
      "LYS 291 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 GLU  72 A\n",
      "LYS 291 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 ARG 281 A\n",
      "LYS 291 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.23 GLU 290 A\n",
      "LYS 291 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.35 ASP 294 A\n",
      "LYS 291 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.05 LYS  39 A\n",
      "\n",
      "LYS 329 A   9.59    18 %   -0.97  331   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU 290 A\n",
      "LYS 329 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 314 A\n",
      "LYS 329 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS 291 A\n",
      "\n",
      "ARG   5 A  12.07    10 %   -0.57  309   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.12 GLU   6 A\n",
      "ARG   5 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP  67 A\n",
      "ARG   5 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 105 A\n",
      "ARG   5 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 ARG 137 A\n",
      "\n",
      "ARG  15 A  12.14    68 %   -2.12  473   0.00    0    0.35 TYR  97 A    0.00 XXX   0 X    1.07 TYR  97 A\n",
      "ARG  15 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 TYR 101 A\n",
      "ARG  15 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP 213 A\n",
      "ARG  15 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 GLU 215 A\n",
      "\n",
      "ARG  23 A  11.87    37 %   -0.97  384   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.09 GLU  27 A\n",
      "ARG  23 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 GLU  89 A\n",
      "ARG  23 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP  94 A\n",
      "ARG  23 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU  95 A\n",
      "\n",
      "ARG 122 A  13.16     0 %   -0.45  261   0.00    0    0.70 ASP  80 A    0.00 XXX   0 X    0.10 GLU 119 A\n",
      "ARG 122 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 ASP 126 A\n",
      "ARG 122 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.26 ASP  80 A\n",
      "\n",
      "ARG 134 A  14.14     0 %   -0.33  237   0.00    0    1.58 ASP 319 A    0.00 XXX   0 X    0.00 ASP 165 A\n",
      "ARG 134 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 ASP 319 A\n",
      "\n",
      "ARG 137 A  13.63    35 %   -1.22  380   0.00    0    0.65 GLU   6 A    0.00 XXX   0 X    0.43 GLU   6 A\n",
      "ARG 137 A                                            0.70 ASP  67 A    0.00 XXX   0 X    0.57 ASP  67 A\n",
      "\n",
      "ARG 152 A  15.85    18 %   -0.87  332   0.00    0    0.72 GLU 154 A    0.00 XXX   0 X    0.31 GLU 154 A\n",
      "ARG 152 A                                            0.74 GLU 238 A    0.00 XXX   0 X    0.33 GLU 238 A\n",
      "ARG 152 A                                            1.55 ASP 263 A    0.00 XXX   0 X    0.57 ASP 263 A\n",
      "\n",
      "ARG 199 A  10.80    65 %   -2.25  462   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.50 CYS 172 A\n",
      "ARG 199 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 215 A\n",
      "\n",
      "ARG 202 A  11.55    41 %   -1.14  396   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.19 GLU 304 A\n",
      "\n",
      "ARG 223 A  13.63    34 %   -1.14  377   0.00    0    1.64 ASP 277 A    0.00 XXX   0 X    0.03 ASP 230 A\n",
      "ARG 223 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.61 ASP 277 A\n",
      "\n",
      "ARG 281 A  17.71   100 %   -2.15  598   0.00    0    0.51 TYR 279 A    0.00 XXX   0 X    0.54 TYR 268 A\n",
      "ARG 281 A                                            0.59 GLU 290 A    0.00 XXX   0 X    1.75 TYR 279 A\n",
      "ARG 281 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.51 TYR 288 A\n",
      "ARG 281 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.55 ASP 145 A\n",
      "ARG 281 A                                            0.00 XXX   0 X    0.00 XXX   0 X    2.03 GLU 290 A\n",
      "ARG 281 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.87 ASP 294 A\n",
      "\n",
      "ARG 324 A  11.59    54 %   -1.52  433   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU  72 A\n",
      "ARG 324 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU  73 A\n",
      "ARG 324 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.41 GLU 131 A\n",
      "ARG 324 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 CYS 140 A\n",
      "\n",
      "ARG 338 A  12.69     0 %   -0.07  146   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 315 A\n",
      "ARG 338 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 GLU 339 A\n",
      "ARG 338 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP 342 A\n",
      "\n",
      "ARG 355 A  14.03     6 %   -0.58  299   0.00    0    1.62 GLU 246 A    0.00 XXX   0 X    0.07 C-  355 A\n",
      "ARG 355 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.43 GLU 246 A\n",
      "\n",
      "N+    2 A   8.88     0 %   -0.32  186   0.00    0    0.77 GLU   4 A    0.00 XXX   0 X    0.05 GLU   6 A\n",
      "N+    2 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 GLU   4 A\n",
      "\n",
      "Coupled residues (marked *) were detected.Please rerun PropKa with the --display-coupled-residues \n",
      "or -d option for detailed information.\n",
      "\n",
      "INFO:pKa summary:\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "SUMMARY OF THIS PREDICTION\n",
      "       Group      pKa  model-pKa   ligand atom-type\n",
      "   ASP  33 A     3.91       3.80                      \n",
      "   ASP  41 A     2.21       3.80                      \n",
      "   ASP  49 A     6.17       3.80                      \n",
      "   ASP  66 A     3.97       3.80                      \n",
      "   ASP  67 A     3.50       3.80                      \n",
      "   ASP  80 A     3.08       3.80                      \n",
      "   ASP  94 A     2.91       3.80                      \n",
      "   ASP 104 A     3.88       3.80                      \n",
      "   ASP 105 A     4.10       3.80                      \n",
      "   ASP 111 A     3.81       3.80                      \n",
      "   ASP 120 A     2.75       3.80                      \n",
      "   ASP 126 A     3.39       3.80                      \n",
      "   ASP 142 A    16.43       3.80                      \n",
      "   ASP 145 A     3.62       3.80                      \n",
      "   ASP 165 A     3.15       3.80                      \n",
      "   ASP 187 A     2.15       3.80                      \n",
      "   ASP 213 A     3.05       3.80                      \n",
      "   ASP 227 A     3.07       3.80                      \n",
      "   ASP 230 A     3.96       3.80                      \n",
      "   ASP 245 A     3.84       3.80                      \n",
      "   ASP 263 A     3.38       3.80                      \n",
      "   ASP 277 A     3.47       3.80                      \n",
      "   ASP 294 A     9.23       3.80                      \n",
      "   ASP 311 A     3.81       3.80                      \n",
      "   ASP 315 A     6.34       3.80                      \n",
      "   ASP 319 A     1.83       3.80                      \n",
      "   ASP 342 A     3.08       3.80                      \n",
      "   ASP 350 A     3.76       3.80                      \n",
      "   GLU   4 A     2.92       4.50                      \n",
      "   GLU   6 A     3.84       4.50                      \n",
      "   GLU  27 A     4.11       4.50                      \n",
      "   GLU  72 A     5.47       4.50                      \n",
      "   GLU  73 A     8.38       4.50                      \n",
      "   GLU  89 A     5.51       4.50                      \n",
      "   GLU  95 A     3.70       4.50                      \n",
      "   GLU 119 A     4.30       4.50                      \n",
      "   GLU 131 A     4.45       4.50                      \n",
      "   GLU 154 A     4.04       4.50                      \n",
      "   GLU 192 A     3.88       4.50                      \n",
      "   GLU 215 A     3.63       4.50                      \n",
      "   GLU 235 A     3.26       4.50                      \n",
      "   GLU 238 A     3.63       4.50                      \n",
      "   GLU 246 A     2.95       4.50                      \n",
      "   GLU 290 A     1.29       4.50                      \n",
      "   GLU 304 A     3.18       4.50                      \n",
      "   GLU 314 A     5.06       4.50                      \n",
      "   GLU 339 A     4.02       4.50                      \n",
      "   C-  355 A     3.34       3.20                      \n",
      "   HIS  31 A     6.33       6.50                      \n",
      "   HIS 191 A     6.82       6.50                      \n",
      "   HIS 221 A     5.75       6.50                      \n",
      "   HIS 224 A     6.57       6.50                      \n",
      "   HIS 241 A     4.98       6.50                      \n",
      "   HIS 244 A     5.80       6.50                      \n",
      "   HIS 261 A     6.60       6.50                      \n",
      "   HIS 295 A     5.60       6.50                      \n",
      "   HIS 303 A     6.22       6.50                      \n",
      "   HIS 309 A     2.56       6.50                      \n",
      "   HIS 341 A     4.22       6.50                      \n",
      "   CYS 140 A    12.78       9.00                      \n",
      "   CYS 159 A    15.19       9.00                      \n",
      "   CYS 172 A    12.08       9.00                      \n",
      "   CYS 175 A     9.09       9.00                      \n",
      "   CYS 269 A    19.48       9.00                      \n",
      "   CYS 349 A     9.57       9.00                      \n",
      "   TYR  50 A    12.47      10.00                      \n",
      "   TYR  97 A    12.15      10.00                      \n",
      "   TYR 101 A    10.24      10.00                      \n",
      "   TYR 130 A    10.10      10.00                      \n",
      "   TYR 183 A    11.75      10.00                      \n",
      "   TYR 196 A    13.17      10.00                      \n",
      "   TYR 208 A    11.41      10.00                      \n",
      "   TYR 268 A    17.93      10.00                      \n",
      "   TYR 279 A    12.08      10.00                      \n",
      "   TYR 288 A    12.91      10.00                      \n",
      "   LYS  16 A     9.20      10.50                      \n",
      "   LYS  22 A     9.53      10.50                      \n",
      "   LYS  32 A    10.25      10.50                      \n",
      "   LYS  39 A    13.17      10.50                      \n",
      "   LYS  61 A     9.38      10.50                      \n",
      "   LYS  68 A    10.47      10.50                      \n",
      "   LYS  91 A    11.20      10.50                      \n",
      "   LYS  99 A    11.34      10.50                      \n",
      "   LYS 102 A    10.42      10.50                      \n",
      "   LYS 103 A    11.30      10.50                      \n",
      "   LYS 116 A    10.23      10.50                      \n",
      "   LYS 135 A    10.54      10.50                      \n",
      "   LYS 148 A    10.50      10.50                      \n",
      "   LYS 189 A    10.46      10.50                      \n",
      "   LYS 219 A    10.37      10.50                      \n",
      "   LYS 226 A    10.46      10.50                      \n",
      "   LYS 229 A    10.85      10.50                      \n",
      "   LYS 239 A    10.51      10.50                      \n",
      "   LYS 251 A    10.84      10.50                      \n",
      "   LYS 253 A    10.54      10.50                      \n",
      "   LYS 258 A    10.36      10.50                      \n",
      "   LYS 267 A     7.64      10.50                      \n",
      "   LYS 285 A    10.40      10.50                      \n",
      "   LYS 291 A    10.01      10.50                      \n",
      "   LYS 329 A     9.59      10.50                      \n",
      "   ARG   5 A    12.07      12.50                      \n",
      "   ARG  15 A    12.14      12.50                      \n",
      "   ARG  23 A    11.87      12.50                      \n",
      "   ARG 122 A    13.16      12.50                      \n",
      "   ARG 134 A    14.14      12.50                      \n",
      "   ARG 137 A    13.63      12.50                      \n",
      "   ARG 152 A    15.85      12.50                      \n",
      "   ARG 199 A    10.80      12.50                      \n",
      "   ARG 202 A    11.55      12.50                      \n",
      "   ARG 223 A    13.63      12.50                      \n",
      "   ARG 281 A    17.71      12.50                      \n",
      "   ARG 324 A    11.59      12.50                      \n",
      "   ARG 338 A    12.69      12.50                      \n",
      "   ARG 355 A    14.03      12.50                      \n",
      "   N+    2 A     8.88       8.00                      \n",
      "\n",
      "INFO:PROPKA information:\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "Free energy of   folding (kcal/mol) as a function of pH (using neutral reference)\n",
      "  0.00     44.66\n",
      "  1.00     44.24\n",
      "  2.00     41.82\n",
      "  3.00     33.82\n",
      "  4.00     21.09\n",
      "  5.00     16.07\n",
      "  6.00     17.12\n",
      "  7.00     18.86\n",
      "  8.00     22.13\n",
      "  9.00     26.07\n",
      " 10.00     32.15\n",
      " 11.00     44.82\n",
      " 12.00     59.71\n",
      " 13.00     73.20\n",
      " 14.00     85.66\n",
      "\n",
      "The pH of optimum stability is  5.0 for which the free energy is   16.1 kcal/mol at 298K\n",
      "Could not determine pH values where the free energy is within 80 % of minimum\n",
      "Could not determine the pH-range where the free energy is negative\n",
      "\n",
      "\n",
      "Protein charge of folded and unfolded state as a function of pH\n",
      "    pH  unfolded  folded\n",
      "  0.00     50.99   50.91\n",
      "  0.10     50.99   50.88\n",
      "  0.20     50.99   50.85\n",
      "  0.30     50.99   50.82\n",
      "  0.40     50.99   50.77\n",
      "  0.50     50.98   50.72\n",
      "  0.60     50.98   50.65\n",
      "  0.70     50.97   50.57\n",
      "  0.80     50.96   50.48\n",
      "  0.90     50.96   50.36\n",
      "  1.00     50.94   50.23\n",
      "  1.10     50.93   50.07\n",
      "  1.20     50.91   49.89\n",
      "  1.30     50.89   49.68\n",
      "  1.40     50.86   49.44\n",
      "  1.50     50.82   49.16\n",
      "  1.60     50.78   48.85\n",
      "  1.70     50.72   48.49\n",
      "  1.80     50.65   48.09\n",
      "  1.90     50.56   47.64\n",
      "  2.00     50.44   47.13\n",
      "  2.10     50.30   46.56\n",
      "  2.20     50.13   45.91\n",
      "  2.30     49.91   45.20\n",
      "  2.40     49.64   44.39\n",
      "  2.50     49.31   43.50\n",
      "  2.60     48.90   42.51\n",
      "  2.70     48.40   41.41\n",
      "  2.80     47.80   40.22\n",
      "  2.90     47.07   38.92\n",
      "  3.00     46.20   37.51\n",
      "  3.10     45.17   36.02\n",
      "  3.20     43.97   34.43\n",
      "  3.30     42.58   32.78\n",
      "  3.40     41.01   31.06\n",
      "  3.50     39.25   29.30\n",
      "  3.60     37.31   27.52\n",
      "  3.70     35.23   25.73\n",
      "  3.80     33.02   23.96\n",
      "  3.90     30.72   22.23\n",
      "  4.00     28.37   20.55\n",
      "  4.10     26.01   18.95\n",
      "  4.20     23.67   17.45\n",
      "  4.30     21.38   16.04\n",
      "  4.40     19.18   14.74\n",
      "  4.50     17.10   13.54\n",
      "  4.60     15.14   12.44\n",
      "  4.70     13.34   11.43\n",
      "  4.80     11.70   10.50\n",
      "  4.90     10.22    9.64\n",
      "  5.00      8.90    8.83\n",
      "  5.10      7.74    8.07\n",
      "  5.20      6.71    7.33\n",
      "  5.30      5.81    6.62\n",
      "  5.40      5.00    5.93\n",
      "  5.50      4.27    5.25\n",
      "  5.60      3.60    4.57\n",
      "  5.70      2.97    3.90\n",
      "  5.80      2.35    3.24\n",
      "  5.90      1.73    2.58\n",
      "  6.00      1.10    1.93\n",
      "  6.10      0.45    1.29\n",
      "  6.20     -0.22    0.66\n",
      "  6.30     -0.91    0.05\n",
      "  6.40     -1.61   -0.54\n",
      "  6.50     -2.31   -1.11\n",
      "  6.60     -3.01   -1.64\n",
      "  6.70     -3.68   -2.15\n",
      "  6.80     -4.31   -2.62\n",
      "  6.90     -4.91   -3.05\n",
      "  7.00     -5.45   -3.45\n",
      "  7.10     -5.94   -3.81\n",
      "  7.20     -6.38   -4.14\n",
      "  7.30     -6.78   -4.45\n",
      "  7.40     -7.13   -4.74\n",
      "  7.50     -7.46   -5.01\n",
      "  7.60     -7.76   -5.26\n",
      "  7.70     -8.04   -5.51\n",
      "  7.80     -8.32   -5.76\n",
      "  7.90     -8.60   -6.00\n",
      "  8.00     -8.88   -6.26\n",
      "  8.10     -9.18   -6.52\n",
      "  8.20     -9.50   -6.80\n",
      "  8.30     -9.84   -7.11\n",
      "  8.40    -10.22   -7.43\n",
      "  8.50    -10.65   -7.79\n",
      "  8.60    -11.11   -8.19\n",
      "  8.70    -11.64   -8.63\n",
      "  8.80    -12.21   -9.11\n",
      "  8.90    -12.85   -9.64\n",
      "  9.00    -13.55  -10.22\n",
      "  9.10    -14.32  -10.85\n",
      "  9.20    -15.17  -11.54\n",
      "  9.30    -16.09  -12.29\n",
      "  9.40    -17.10  -13.08\n",
      "  9.50    -18.21  -13.94\n",
      "  9.60    -19.42  -14.85\n",
      "  9.70    -20.76  -15.82\n",
      "  9.80    -22.21  -16.86\n",
      "  9.90    -23.79  -17.95\n",
      " 10.00    -25.49  -19.10\n",
      " 10.10    -27.30  -20.31\n",
      " 10.20    -29.18  -21.57\n",
      " 10.30    -31.13  -22.86\n",
      " 10.40    -33.09  -24.19\n",
      " 10.50    -35.05  -25.53\n",
      " 10.60    -36.95  -26.87\n",
      " 10.70    -38.76  -28.19\n",
      " 10.80    -40.46  -29.49\n",
      " 10.90    -42.03  -30.76\n",
      " 11.00    -43.45  -31.99\n",
      " 11.10    -44.73  -33.18\n",
      " 11.20    -45.88  -34.33\n",
      " 11.30    -46.90  -35.46\n",
      " 11.40    -47.83  -36.54\n",
      " 11.50    -48.67  -37.61\n",
      " 11.60    -49.47  -38.64\n",
      " 11.70    -50.22  -39.65\n",
      " 11.80    -50.97  -40.64\n",
      " 11.90    -51.72  -41.60\n",
      " 12.00    -52.49  -42.54\n",
      " 12.10    -53.29  -43.45\n",
      " 12.20    -54.12  -44.34\n",
      " 12.30    -54.97  -45.20\n",
      " 12.40    -55.84  -46.03\n",
      " 12.50    -56.72  -46.84\n",
      " 12.60    -57.58  -47.62\n",
      " 12.70    -58.41  -48.37\n",
      " 12.80    -59.18  -49.10\n",
      " 12.90    -59.90  -49.80\n",
      " 13.00    -60.55  -50.48\n",
      " 13.10    -61.12  -51.13\n",
      " 13.20    -61.61  -51.75\n",
      " 13.30    -62.04  -52.35\n",
      " 13.40    -62.40  -52.91\n",
      " 13.50    -62.70  -53.45\n",
      " 13.60    -62.95  -53.96\n",
      " 13.70    -63.15  -54.44\n",
      " 13.80    -63.32  -54.89\n",
      " 13.90    -63.45  -55.30\n",
      " 14.00    -63.56  -55.69\n",
      "The pI is  6.31 (folded) and  6.17 (unfolded)\n",
      "\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "INFO:Applying pKa values at a pH of 8.20:\n",
      "INFO:Adding hydrogens to biomolecule.\n",
      "INFO:Debumping biomolecule (again).\n",
      "INFO:Optimizing hydrogen bonds\n",
      "INFO:Applying force field to biomolecule states.\n",
      "INFO:Regenerating headers.\n",
      "INFO:Regenerating PDB lines.\n",
      "WARNING:Ignoring 466 header lines in output.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "pH: 8.2\tname: 1ka1_hal2\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:PDB2PQR v3.6.2: biomolecular structure conversion software.\n",
      "INFO:Please cite:  Jurrus E, et al.  Improvements to the APBS biomolecular solvation software suite.  Protein Sci 27 112-128 (2018).\n",
      "INFO:Please cite:  Dolinsky TJ, et al.  PDB2PQR: expanding and upgrading automated preparation of biomolecular structures for molecular simulations. Nucleic Acids Res 35 W522-W525 (2007).\n",
      "INFO:Checking and transforming input arguments.\n",
      "INFO:Loading topology files.\n",
      "INFO:Loading molecule: /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/AtSAL1_updated.pdb\n",
      "INFO:Setting up molecule.\n",
      "INFO:Created biomolecule object with 352 residues and 2628 atoms.\n",
      "INFO:Setting termini states for biomolecule chains.\n",
      "INFO:Loading forcefield.\n",
      "INFO:Loading hydrogen topology definitions.\n",
      "WARNING:Missing atom OXT in residue ALA A 352\n",
      "WARNING:Missing atom OXT in residue ALA A 352\n",
      "INFO:Attempting to repair 1 missing atoms in biomolecule.\n",
      "WARNING:Missing atom OXT in residue ALA A 352\n",
      "INFO:Added atom OXT to residue ALA A 352 at coordinates -4.698, -13.324, -18.987\n",
      "INFO:Updating disulfide bridges.\n",
      "INFO:Debumping biomolecule.\n",
      "INFO:Assigning titration states with PROPKA.\n",
      "INFO:\n",
      "propka3.5.1                                                                                  2024-07-26\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "--                                                                           --\n",
      "--  PROPKA: A PROTEIN PKA PREDICTOR                                          --\n",
      "--                                                                           --\n",
      "--  VERSION 1.0,  04/25/2004,  IOWA CITY                                     --\n",
      "--  BY HUI LI                                                                --\n",
      "--                                                                           --\n",
      "--  VERSION 2.0,  11/05/2007, IOWA CITY/COPENHAGEN                           --\n",
      "--  BY DELPHINE C. BAS AND DAVID M. ROGERS                                   --\n",
      "--                                                                           --\n",
      "--  VERSION 3.0,  01/06/2011, COPENHAGEN                                     --\n",
      "--  BY MATS H.M. OLSSON AND CHRESTEN R. SONDERGARD                           --\n",
      "--                                                                           --\n",
      "--  VERSION 3.1,  07/01/2011, COPENHAGEN                                     --\n",
      "--  BY CHRESTEN R. SONDERGARD AND MATS H.M. OLSSON                           --\n",
      "--                                                                           --\n",
      "--  VERSION 3.2,  06/17/2020, PLANET EARTH                                   --\n",
      "--  SEE ABOVE FOR AUTHORS                                                    --\n",
      "--                                                                           --\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "References:\n",
      "\n",
      "Very Fast Empirical Prediction and Rationalization of Protein pKa Values.\n",
      "Hui Li, Andrew D. Robertson and Jan H. Jensen. PROTEINS: Structure, Function,\n",
      "and Bioinformatics. 61:704-721 (2005)\n",
      "\n",
      "Very Fast Prediction and Rationalization of pKa Values for Protein-Ligand\n",
      "Complexes.  Delphine C. Bas, David M. Rogers and Jan H. Jensen.  PROTEINS:\n",
      "Structure, Function, and Bioinformatics 73:765-783 (2008)\n",
      "\n",
      "PROPKA3: Consistent Treatment of Internal and Surface Residues in Empirical\n",
      "pKa predictions.  Mats H.M. Olsson, Chresten R. Sondergard, Michal Rostkowski,\n",
      "and Jan H. Jensen.  Journal of Chemical Theory and Computation, 7(2):525-537\n",
      "(2011)\n",
      "\n",
      "Improved Treatment of Ligands and Coupling Effects in Empirical Calculation\n",
      "and Rationalization of pKa Values.  Chresten R. Sondergaard, Mats H.M. Olsson,\n",
      "Michal Rostkowski, and Jan H. Jensen.  Journal of Chemical Theory and\n",
      "Computation, (2011)\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "\n",
      "INFO:Coupling map:\n",
      "\n",
      "INFO:Calculating pKas for Conformation container 1A with 3086 atoms and 869 groups\n",
      "INFO:did not converge in 10 iterations\n",
      "INFO:pKa determinants:\n",
      "\n",
      "---------  -----   ------   ---------------------    --------------    --------------    --------------\n",
      "                            DESOLVATION  EFFECTS       SIDECHAIN          BACKBONE        COULOMBIC\n",
      " RESIDUE    pKa    BURIED     REGULAR      RE        HYDROGEN BOND     HYDROGEN BOND      INTERACTION\n",
      "---------  -----   ------   ---------   ---------    --------------    --------------    --------------\n",
      "\n",
      "ASP   8 A   3.99    18 %    0.58  331   0.18    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.03 N+    1 A\n",
      "ASP   8 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.15 LYS   5 A\n",
      "ASP   8 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.30 LYS  11 A\n",
      "ASP   8 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.12 LYS  12 A\n",
      "ASP   8 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU   6 A\n",
      "\n",
      "ASP  32 A   3.90     0 %    0.12  137   0.00    0    0.00 XXX   0 X   -0.02 VAL  33 A    0.00 XXX   0 X\n",
      "\n",
      "ASP  38 A   3.69     0 %    0.12  180   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.24 LYS  39 A\n",
      "\n",
      "ASP  46 A   7.13    91 %    2.32  536   0.55    0    0.00 XXX   0 X    0.00 XXX   0 X    0.25 GLU  71 A\n",
      "ASP  46 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 GLU  72 A\n",
      "ASP  46 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 137 A\n",
      "\n",
      "ASP  73 A   3.49     0 %    0.30  267   0.00    0    0.00 XXX   0 X   -0.69 GLY  75 A   -0.06 ARG  78 A\n",
      "ASP  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS 317 A\n",
      "ASP  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 ASP 118 A\n",
      "\n",
      "ASP  76 A   4.20     0 %    0.23  268   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.00 LYS  79 A\n",
      "ASP  76 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.14 ASP  73 A\n",
      "ASP  76 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP  80 A\n",
      "\n",
      "ASP  80 A   3.82     0 %    0.31  184   0.00    0   -0.20 SER  82 A   -0.22 ASP  80 A    0.03 ASP  38 A\n",
      "ASP  80 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP  84 A\n",
      "\n",
      "ASP  84 A   3.48     0 %    0.35  226   0.00    0    0.00 XXX   0 X   -0.06 ASP  84 A   -0.25 ARG  88 A\n",
      "ASP  84 A                                            0.00 XXX   0 X   -0.35 THR  85 A    0.00 XXX   0 X\n",
      "\n",
      "ASP  95 A   3.71     0 %    0.32  279   0.00    0    0.00 XXX   0 X   -0.01 ASP  95 A   -0.28 LYS  23 A\n",
      "ASP  95 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.12 LYS  91 A\n",
      "\n",
      "ASP 111 A   3.94     0 %    0.21  211   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.03 ARG  78 A\n",
      "ASP 111 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.14 ARG 115 A\n",
      "ASP 111 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP 112 A\n",
      "\n",
      "ASP 112 A   2.85     0 %    0.34  258   0.00    0   -0.49 SER 109 A   -0.62 SER 109 A   -0.18 ARG 115 A\n",
      "\n",
      "ASP 118 A   3.20    10 %    0.65  308   0.06    0   -0.77 SER  74 A   -0.25 SER  74 A   -0.26 ARG  78 A\n",
      "ASP 118 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 115 A\n",
      "\n",
      "ASP 134 A  13.40*  100 %    3.03  569   0.41    0    1.52 ASP 137 A   -0.26 ILE 136 A   -0.01 LYS 285 A\n",
      "ASP 134 A                                            0.96 ASP 288 A    0.00 XXX   0 X    0.29 GLU  71 A\n",
      "ASP 134 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.68 ASP  46 A\n",
      "ASP 134 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.89 ASP 137 A\n",
      "ASP 134 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.08 ASP 288 A\n",
      "\n",
      "ASP 137 A   2.84*  100 %    2.61  581   0.29    0   -1.52 ASP 134 A   -0.00 GLN 260 A   -0.47 LYS 262 A\n",
      "ASP 137 A                                           -1.46 ASP 288 A    0.00 XXX   0 X    0.05 GLU  71 A\n",
      "ASP 137 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 GLU 284 A\n",
      "ASP 137 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.65 ARG 276 A\n",
      "\n",
      "ASP 146 A   3.60    12 %    0.79  316   0.11    0   -0.79 ARG 144 A    0.00 XXX   0 X   -0.31 ARG 144 A\n",
      "\n",
      "ASP 186 A   3.58     0 %    0.15  116   0.00    0   -0.37 SER 184 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "ASP 206 A   3.75    53 %    1.44  431   0.23    0   -1.15 ARG  19 A    0.00 XXX   0 X   -0.59 ARG  19 A\n",
      "\n",
      "ASP 239 A   3.65     0 %    0.43  269   0.00    0    0.00 XXX   0 X   -0.34 ASP 239 A   -0.04 HIS 235 A\n",
      "ASP 239 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.17 HIS 238 A\n",
      "ASP 239 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 HIS 279 A\n",
      "\n",
      "ASP 258 A   6.14    44 %    1.75  404   0.62    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.00 LYS 140 A\n",
      "ASP 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.34 ARG 144 A\n",
      "ASP 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.05 LYS 262 A\n",
      "ASP 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.34 ASP 146 A\n",
      "ASP 258 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 232 A\n",
      "\n",
      "ASP 270 A   5.41    76 %    2.89  493   0.45    0   -0.11 SER 174 A   -0.21 ILE 175 A   -0.89 ARG 268 A\n",
      "ASP 270 A                                           -0.51 ARG 268 A   -0.01 ASP 270 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 288 A   7.25*   81 %    2.27  509   0.16    0   -0.96 ASP 134 A    0.00 XXX   0 X   -0.13 LYS 262 A\n",
      "ASP 288 A                                            1.46 ASP 137 A    0.00 XXX   0 X   -0.22 LYS 285 A\n",
      "ASP 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.62 ASP 137 A\n",
      "ASP 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.54 ARG 276 A\n",
      "ASP 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.54 GLU 284 A\n",
      "ASP 288 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.75 HIS 289 A\n",
      "\n",
      "ASP 305 A   4.25    25 %    1.48  352   0.32    0    0.00 XXX   0 X   -0.17 ASP 305 A   -0.12 LYS 309 A\n",
      "ASP 305 A                                            0.00 XXX   0 X   -0.07 ALA 307 A   -0.04 ARG 341 A\n",
      "ASP 305 A                                            0.00 XXX   0 X   -0.00 GLY 308 A    0.02 ASP 320 A\n",
      "ASP 305 A                                            0.00 XXX   0 X   -0.63 LYS 309 A    0.23 ASP 322 A\n",
      "ASP 305 A                                            0.00 XXX   0 X   -0.58 ASP 322 A    0.00 XXX   0 X\n",
      "\n",
      "ASP 312 A   2.16    11 %    0.68  312   0.01    0   -0.85 SER 314 A   -0.15 SER 314 A   -0.10 LYS 159 A\n",
      "ASP 312 A                                           -0.85 LYS 315 A    0.00 XXX   0 X   -0.38 LYS 315 A\n",
      "\n",
      "ASP 320 A   3.88     0 %    0.22  254   0.00    0    0.00 XXX   0 X   -0.12 ASP 320 A   -0.13 LYS 315 A\n",
      "ASP 320 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.11 ASP 312 A\n",
      "\n",
      "ASP 322 A   3.54    11 %    0.50  311   0.00    0   -0.66 THR 323 A    0.00 XXX   0 X   -0.01 LYS 309 A\n",
      "ASP 322 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.10 ARG 341 A\n",
      "\n",
      "ASP 342 A   4.00     0 %    0.37  242   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.04 LYS 247 A\n",
      "ASP 342 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.05 LYS 338 A\n",
      "ASP 342 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 ARG 341 A\n",
      "\n",
      "GLU   4 A   4.37     0 %    0.15  174   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.19 N+    1 A\n",
      "GLU   4 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.24 LYS   5 A\n",
      "GLU   4 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU   6 A\n",
      "GLU   4 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 ASP   8 A\n",
      "\n",
      "GLU   6 A   3.70    28 %    1.00  361   0.00    0   -0.72 ARG 129 A   -0.65 GLU   6 A   -0.06 LYS   5 A\n",
      "GLU   6 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.37 ARG 129 A\n",
      "\n",
      "GLU  58 A   4.99    32 %    0.70  370   0.10    0   -0.31 CYS 119 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "GLU  60 A   4.70    11 %    0.55  313   0.02    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.40 LYS  12 A\n",
      "GLU  60 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS  59 A\n",
      "GLU  60 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP   8 A\n",
      "GLU  60 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 101 A\n",
      "\n",
      "GLU  64 A   4.73     0 %    0.31  278   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.32 ARG 129 A\n",
      "GLU  64 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 GLU   6 A\n",
      "GLU  64 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 HIS 130 A\n",
      "GLU  64 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.16 GLU 156 A\n",
      "GLU  64 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 157 A\n",
      "\n",
      "GLU  71 A   5.17    41 %    0.73  396   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.20 LYS 285 A\n",
      "GLU  71 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 317 A\n",
      "GLU  71 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 ASP 288 A\n",
      "\n",
      "GLU  72 A   5.72    30 %    0.59  365   0.07    0   -0.15 TYR  47 A    0.00 XXX   0 X   -0.01 LYS 317 A\n",
      "GLU  72 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.26 GLU  71 A\n",
      "GLU  72 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 ASP  73 A\n",
      "GLU  72 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.28 ASP  76 A\n",
      "\n",
      "GLU  87 A   4.66     0 %    0.21  201   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.08 LYS  91 A\n",
      "GLU  87 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP  84 A\n",
      "\n",
      "GLU 100 A   4.92     0 %    0.40  210   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 101 A\n",
      "\n",
      "GLU 101 A   4.59     0 %    0.15  174   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS  12 A\n",
      "\n",
      "GLU 123 A   4.29     1 %    0.40  284   0.01    0   -0.17 LYS 317 A    0.00 XXX   0 X   -0.45 LYS 317 A\n",
      "\n",
      "GLU 156 A   4.19    14 %    0.82  321   0.09    0   -0.82 ARG 129 A    0.00 XXX   0 X    0.09 GLU   6 A\n",
      "GLU 156 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 HIS 130 A\n",
      "GLU 156 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.45 ARG 129 A\n",
      "\n",
      "GLU 157 A   4.41     0 %    0.30  215   0.00    0    0.00 XXX   0 X   -0.50 GLY 128 A   -0.04 ARG 129 A\n",
      "GLU 157 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.05 HIS 130 A\n",
      "GLU 157 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 GLU 156 A\n",
      "\n",
      "GLU 187 A   4.76     0 %    0.24  215   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 186 A\n",
      "\n",
      "GLU 220 A   4.46     0 %    0.22  175   0.00    0    0.00 XXX   0 X   -0.30 GLU 220 A   -0.08 LYS 331 A\n",
      "GLU 220 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.12 GLU 224 A\n",
      "\n",
      "GLU 223 A   4.77     0 %    0.51  236   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.27 LYS 251 A\n",
      "GLU 223 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 224 A\n",
      "\n",
      "GLU 224 A   3.96     0 %    0.21  180   0.00    0   -0.71 ASN 221 A    0.00 XXX   0 X   -0.04 LYS 251 A\n",
      "\n",
      "GLU 229 A   5.80    55 %    1.19  436   0.40    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.13 ARG 256 A\n",
      "GLU 229 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.09 LYS 262 A\n",
      "GLU 229 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.07 HIS 238 A\n",
      "\n",
      "GLU 232 A   4.44     0 %    0.15  188   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS 140 A\n",
      "GLU 232 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.20 ARG 144 A\n",
      "GLU 232 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.00 ARG 256 A\n",
      "GLU 232 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 146 A\n",
      "\n",
      "GLU 284 A   2.33    71 %    2.47  480   0.16    0   -0.21 ARG 276 A   -0.15 LYS 285 A   -0.19 LYS 262 A\n",
      "GLU 284 A                                           -0.11 TYR 282 A    0.00 XXX   0 X   -0.19 LYS 285 A\n",
      "GLU 284 A                                           -1.60 HIS 289 A    0.00 XXX   0 X   -1.06 ARG 276 A\n",
      "GLU 284 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.29 HIS 289 A\n",
      "\n",
      "GLU 298 A   3.85    28 %    1.05  360   0.21    0    0.00 XXX   0 X   -0.73 GLY 197 A   -0.09 LYS 213 A\n",
      "GLU 298 A                                            0.00 XXX   0 X   -0.33 SER 198 A    0.00 XXX   0 X\n",
      "GLU 298 A                                            0.00 XXX   0 X   -0.75 GLY 199 A    0.00 XXX   0 X\n",
      "\n",
      "GLU 330 A   4.59     0 %    0.16  232   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 331 A\n",
      "\n",
      "GLU 346 A   4.58     0 %    0.39  216   0.00    0   -0.10 GLN 347 A    0.00 XXX   0 X   -0.33 LYS 247 A\n",
      "GLU 346 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 LYS 349 A\n",
      "GLU 346 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.14 ASP 342 A\n",
      "\n",
      "GLU 348 A   4.77     0 %    0.30  273   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.09 LYS 349 A\n",
      "GLU 348 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 239 A\n",
      "GLU 348 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 322 A\n",
      "\n",
      "C-  352 A   3.34     0 %    0.14   58   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "HIS 130 A   6.21    28 %   -0.99  361   0.00    0    0.00 XXX   0 X    0.01 GLY 124 A    0.01 GLU  64 A\n",
      "HIS 130 A                                            0.00 XXX   0 X    0.43 ASN 127 A    0.04 GLU 156 A\n",
      "HIS 130 A                                            0.00 XXX   0 X    0.15 ARG 129 A    0.05 GLU 157 A\n",
      "\n",
      "HIS 235 A   6.45     0 %   -0.18  141   0.00    0    0.00 XXX   0 X    0.10 ALA 234 A    0.04 ASP 239 A\n",
      "\n",
      "HIS 238 A   6.25     0 %   -0.30  241   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.00 ARG 256 A\n",
      "HIS 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 GLU 229 A\n",
      "HIS 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.19 HIS 235 A\n",
      "HIS 238 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.17 ASP 239 A\n",
      "\n",
      "HIS 279 A   6.15     0 %   -0.30  228   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.07 HIS 235 A\n",
      "HIS 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 239 A\n",
      "\n",
      "HIS 289 A   5.35   100 %   -3.14  578   0.00    0    1.60 GLU 284 A    0.00 XXX   0 X   -0.40 LYS 262 A\n",
      "HIS 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.23 ARG 276 A\n",
      "HIS 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 LYS 285 A\n",
      "HIS 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.29 GLU 284 A\n",
      "HIS 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.75 ASP 288 A\n",
      "\n",
      "CYS  21 A  12.37   100 %    3.39  590   0.00    0   -0.13 SER  49 A   -0.12 CYS  21 A    0.23 ASP  46 A\n",
      "\n",
      "CYS 119 A  10.39    24 %    0.60  349   0.00    0    0.31 GLU  58 A    0.00 XXX   0 X   -0.02 ARG 115 A\n",
      "CYS 119 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 118 A\n",
      "CYS 119 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.46 GLU  58 A\n",
      "\n",
      "CYS 167 A  13.04   100 %    3.60  645   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.29 ARG 268 A\n",
      "CYS 167 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.73 CYS 190 A\n",
      "\n",
      "CYS 190 A  10.90    74 %    2.85  488   0.00    0   -0.22 ARG 268 A   -0.02 GLY 189 A   -0.84 ARG 268 A\n",
      "CYS 190 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 187 A\n",
      "CYS 190 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.11 ASP 270 A\n",
      "\n",
      "TYR   3 A  10.76     8 %    0.53  303   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.10 GLU   6 A\n",
      "TYR   3 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.19 ARG 129 A\n",
      "TYR   3 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.29 GLU 156 A\n",
      "TYR   3 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 157 A\n",
      "\n",
      "TYR  47 A  12.70    45 %    1.22  407   0.00    0    0.15 GLU  72 A    0.00 XXX   0 X    0.26 ASP  46 A\n",
      "TYR  47 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU  71 A\n",
      "TYR  47 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.14 ASP  73 A\n",
      "TYR  47 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.39 ASP  76 A\n",
      "TYR  47 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP 118 A\n",
      "TYR  47 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.51 GLU  72 A\n",
      "\n",
      "TYR 148 A  15.46    84 %    4.04  516   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    1.36 CYS  21 A\n",
      "TYR 148 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 CYS 167 A\n",
      "\n",
      "TYR 201 A  10.14     0 %    0.32  267   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 298 A\n",
      "TYR 201 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.20 LYS 213 A\n",
      "\n",
      "TYR 263 A  15.09*  100 %    3.51  633   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP 137 A\n",
      "TYR 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.83 ARG 276 A\n",
      "TYR 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.46 GLU 284 A\n",
      "TYR 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 ASP 288 A\n",
      "TYR 263 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.61 TYR 274 A\n",
      "\n",
      "TYR 274 A   9.17*  100 %    3.44  594   0.00    0   -0.85 LYS 262 A    0.00 XXX   0 X    0.22 ASP 137 A\n",
      "TYR 274 A                                           -0.20 ARG 276 A    0.00 XXX   0 X    0.08 GLU 229 A\n",
      "TYR 274 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.42 GLU 284 A\n",
      "TYR 274 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.12 ASP 288 A\n",
      "TYR 274 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -2.03 LYS 262 A\n",
      "TYR 274 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -2.03 ARG 276 A\n",
      "\n",
      "TYR 282 A  12.77    42 %    1.48  398   0.00    0    0.11 GLU 284 A    0.00 XXX   0 X    0.09 GLU  71 A\n",
      "TYR 282 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 134 A\n",
      "TYR 282 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP 137 A\n",
      "TYR 282 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.34 ARG 276 A\n",
      "TYR 282 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.07 ARG 283 A\n",
      "TYR 282 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 TYR 274 A\n",
      "TYR 282 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 TYR 318 A\n",
      "TYR 282 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.71 GLU 284 A\n",
      "TYR 282 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.53 ASP 288 A\n",
      "\n",
      "TYR 318 A   9.59     0 %    0.35  256   0.00    0   -0.13 ARG 283 A    0.00 XXX   0 X    0.06 GLU  71 A\n",
      "TYR 318 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 123 A\n",
      "TYR 318 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.26 LYS 285 A\n",
      "TYR 318 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.09 LYS 317 A\n",
      "TYR 318 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.38 ARG 283 A\n",
      "\n",
      "LYS   5 A  10.80     0 %   -0.15  167   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.24 GLU   4 A\n",
      "LYS   5 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 GLU   6 A\n",
      "LYS   5 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP   8 A\n",
      "\n",
      "LYS  11 A   9.47    50 %   -1.27  420   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.30 ASP   8 A\n",
      "LYS  11 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS  12 A\n",
      "\n",
      "LYS  12 A  10.57    27 %   -0.51  358   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.12 ASP   8 A\n",
      "LYS  12 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.40 GLU  60 A\n",
      "LYS  12 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 GLU 101 A\n",
      "\n",
      "LYS  23 A  10.16    17 %   -0.59  328   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.28 ASP  95 A\n",
      "LYS  23 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 LYS  91 A\n",
      "\n",
      "LYS  26 A  10.23     3 %   -0.27  291   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS  39 A  10.48     0 %   -0.26  206   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.24 ASP  38 A\n",
      "\n",
      "LYS  59 A  10.21     0 %   -0.32  249   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU  60 A\n",
      "\n",
      "LYS  79 A  10.44     0 %   -0.06   88   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP  76 A\n",
      "\n",
      "LYS  91 A  10.52     0 %   -0.17  208   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU  87 A\n",
      "LYS  91 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG  88 A\n",
      "LYS  91 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.12 ASP  95 A\n",
      "\n",
      "LYS 140 A   9.86    17 %   -0.56  329   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.08 ARG 144 A\n",
      "LYS 140 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 232 A\n",
      "LYS 140 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP 258 A\n",
      "LYS 140 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS  39 A\n",
      "\n",
      "LYS 159 A  10.42     0 %   -0.14  238   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP 312 A\n",
      "LYS 159 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 LYS 315 A\n",
      "\n",
      "LYS 180 A  10.43     0 %   -0.12   93   0.00    0    0.06 LYS 182 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 182 A  10.14     0 %   -0.31  138   0.00    0   -0.06 LYS 180 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 208 A  10.45     0 %   -0.05   83   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 213 A  10.19     9 %   -0.60  306   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.09 GLU 298 A\n",
      "LYS 213 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 TYR 201 A\n",
      "\n",
      "LYS 247 A  10.38     0 %   -0.49  262   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 342 A\n",
      "LYS 247 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.33 GLU 346 A\n",
      "\n",
      "LYS 251 A  10.60     0 %   -0.22  135   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.27 GLU 223 A\n",
      "LYS 251 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 GLU 224 A\n",
      "\n",
      "LYS 262 A   9.41   100 %   -2.90  586   0.00    0    0.85 TYR 274 A    0.00 XXX   0 X    0.47 ASP 137 A\n",
      "LYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 GLU 229 A\n",
      "LYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 258 A\n",
      "LYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -2.00 ARG 276 A\n",
      "LYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 GLU 284 A\n",
      "LYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 ASP 288 A\n",
      "LYS 262 A                                            0.00 XXX   0 X    0.00 XXX   0 X    2.03 TYR 274 A\n",
      "\n",
      "LYS 280 A  10.46     0 %   -0.04   81   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 285 A  10.26    17 %   -0.88  329   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.20 GLU  71 A\n",
      "LYS 285 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 134 A\n",
      "LYS 285 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.24 ARG 283 A\n",
      "LYS 285 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 GLU 284 A\n",
      "LYS 285 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.22 ASP 288 A\n",
      "LYS 285 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.26 TYR 318 A\n",
      "\n",
      "LYS 309 A  10.51     0 %   -0.11  165   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.12 ASP 305 A\n",
      "LYS 309 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 322 A\n",
      "LYS 309 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 ARG 341 A\n",
      "\n",
      "LYS 315 A  11.51     0 %   -0.35  220   0.00    0    0.85 ASP 312 A    0.00 XXX   0 X    0.13 ASP 320 A\n",
      "LYS 315 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 ASP 312 A\n",
      "\n",
      "LYS 317 A  10.13    35 %   -1.18  378   0.00    0    0.17 GLU 123 A    0.00 XXX   0 X    0.06 GLU  71 A\n",
      "LYS 317 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU  72 A\n",
      "LYS 317 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP  73 A\n",
      "LYS 317 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 TYR 318 A\n",
      "LYS 317 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.45 GLU 123 A\n",
      "\n",
      "LYS 331 A  10.52     0 %   -0.12  161   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU 220 A\n",
      "LYS 331 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 GLU 330 A\n",
      "\n",
      "LYS 338 A  10.23     0 %   -0.27  219   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.05 ARG 341 A\n",
      "LYS 338 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 342 A\n",
      "\n",
      "LYS 349 A  10.31     0 %   -0.30  147   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 346 A\n",
      "LYS 349 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 GLU 348 A\n",
      "\n",
      "ARG  19 A  13.45    31 %   -0.78  369   0.00    0    1.15 ASP 206 A    0.00 XXX   0 X    0.59 ASP 206 A\n",
      "\n",
      "ARG  78 A  12.45     0 %   -0.37  250   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.06 ASP  73 A\n",
      "ARG  78 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 111 A\n",
      "ARG  78 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.26 ASP 118 A\n",
      "ARG  78 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.04 ARG 115 A\n",
      "\n",
      "ARG  88 A  12.29    11 %   -0.46  311   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.25 ASP  84 A\n",
      "\n",
      "ARG 115 A  12.63     0 %   -0.24  199   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.14 ASP 111 A\n",
      "ARG 115 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.18 ASP 112 A\n",
      "ARG 115 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP 118 A\n",
      "ARG 115 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 CYS 119 A\n",
      "\n",
      "ARG 129 A  14.57    24 %   -0.85  348   0.00    0    0.72 GLU   6 A    0.00 XXX   0 X    0.19 TYR   3 A\n",
      "ARG 129 A                                            0.82 GLU 156 A    0.00 XXX   0 X    0.32 GLU  64 A\n",
      "ARG 129 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 GLU 157 A\n",
      "ARG 129 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.37 GLU   6 A\n",
      "ARG 129 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.45 GLU 156 A\n",
      "\n",
      "ARG 144 A  13.46     3 %   -0.68  291   0.00    0    0.79 ASP 146 A    0.00 XXX   0 X    0.20 GLU 232 A\n",
      "ARG 144 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.34 ASP 258 A\n",
      "ARG 144 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.31 ASP 146 A\n",
      "\n",
      "ARG 256 A  12.33     0 %   -0.30  275   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.13 GLU 229 A\n",
      "ARG 256 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 GLU 232 A\n",
      "\n",
      "ARG 268 A  12.83    75 %   -2.41  490   0.00    0    0.22 CYS 190 A    0.00 XXX   0 X    0.29 CYS 167 A\n",
      "ARG 268 A                                            0.51 ASP 270 A    0.00 XXX   0 X    0.84 CYS 190 A\n",
      "ARG 268 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.89 ASP 270 A\n",
      "\n",
      "ARG 276 A  16.42    98 %   -1.95  557   0.00    0    0.20 TYR 274 A    0.00 XXX   0 X    0.83 TYR 263 A\n",
      "ARG 276 A                                            0.21 GLU 284 A    0.00 XXX   0 X    0.34 TYR 282 A\n",
      "ARG 276 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.65 ASP 137 A\n",
      "ARG 276 A                                            0.00 XXX   0 X    0.00 XXX   0 X    2.03 TYR 274 A\n",
      "ARG 276 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.06 GLU 284 A\n",
      "ARG 276 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.54 ASP 288 A\n",
      "\n",
      "ARG 283 A  12.77     0 %   -0.31  205   0.00    0    0.13 TYR 318 A    0.00 XXX   0 X    0.07 TYR 282 A\n",
      "ARG 283 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 TYR 318 A\n",
      "\n",
      "ARG 341 A  12.40     1 %   -0.32  283   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP 305 A\n",
      "ARG 341 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 ASP 322 A\n",
      "ARG 341 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP 342 A\n",
      "\n",
      "N+    1 A   7.94     0 %   -0.28  245   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.19 GLU   4 A\n",
      "N+    1 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP   8 A\n",
      "N+    1 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 LYS  11 A\n",
      "\n",
      "Coupled residues (marked *) were detected.Please rerun PropKa with the --display-coupled-residues \n",
      "or -d option for detailed information.\n",
      "\n",
      "INFO:pKa summary:\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "SUMMARY OF THIS PREDICTION\n",
      "       Group      pKa  model-pKa   ligand atom-type\n",
      "   ASP   8 A     3.99       3.80                      \n",
      "   ASP  32 A     3.90       3.80                      \n",
      "   ASP  38 A     3.69       3.80                      \n",
      "   ASP  46 A     7.13       3.80                      \n",
      "   ASP  73 A     3.49       3.80                      \n",
      "   ASP  76 A     4.20       3.80                      \n",
      "   ASP  80 A     3.82       3.80                      \n",
      "   ASP  84 A     3.48       3.80                      \n",
      "   ASP  95 A     3.71       3.80                      \n",
      "   ASP 111 A     3.94       3.80                      \n",
      "   ASP 112 A     2.85       3.80                      \n",
      "   ASP 118 A     3.20       3.80                      \n",
      "   ASP 134 A    13.40       3.80                      \n",
      "   ASP 137 A     2.84       3.80                      \n",
      "   ASP 146 A     3.60       3.80                      \n",
      "   ASP 186 A     3.58       3.80                      \n",
      "   ASP 206 A     3.75       3.80                      \n",
      "   ASP 239 A     3.65       3.80                      \n",
      "   ASP 258 A     6.14       3.80                      \n",
      "   ASP 270 A     5.41       3.80                      \n",
      "   ASP 288 A     7.25       3.80                      \n",
      "   ASP 305 A     4.25       3.80                      \n",
      "   ASP 312 A     2.16       3.80                      \n",
      "   ASP 320 A     3.88       3.80                      \n",
      "   ASP 322 A     3.54       3.80                      \n",
      "   ASP 342 A     4.00       3.80                      \n",
      "   GLU   4 A     4.37       4.50                      \n",
      "   GLU   6 A     3.70       4.50                      \n",
      "   GLU  58 A     4.99       4.50                      \n",
      "   GLU  60 A     4.70       4.50                      \n",
      "   GLU  64 A     4.73       4.50                      \n",
      "   GLU  71 A     5.17       4.50                      \n",
      "   GLU  72 A     5.72       4.50                      \n",
      "   GLU  87 A     4.66       4.50                      \n",
      "   GLU 100 A     4.92       4.50                      \n",
      "   GLU 101 A     4.59       4.50                      \n",
      "   GLU 123 A     4.29       4.50                      \n",
      "   GLU 156 A     4.19       4.50                      \n",
      "   GLU 157 A     4.41       4.50                      \n",
      "   GLU 187 A     4.76       4.50                      \n",
      "   GLU 220 A     4.46       4.50                      \n",
      "   GLU 223 A     4.77       4.50                      \n",
      "   GLU 224 A     3.96       4.50                      \n",
      "   GLU 229 A     5.80       4.50                      \n",
      "   GLU 232 A     4.44       4.50                      \n",
      "   GLU 284 A     2.33       4.50                      \n",
      "   GLU 298 A     3.85       4.50                      \n",
      "   GLU 330 A     4.59       4.50                      \n",
      "   GLU 346 A     4.58       4.50                      \n",
      "   GLU 348 A     4.77       4.50                      \n",
      "   C-  352 A     3.34       3.20                      \n",
      "   HIS 130 A     6.21       6.50                      \n",
      "   HIS 235 A     6.45       6.50                      \n",
      "   HIS 238 A     6.25       6.50                      \n",
      "   HIS 279 A     6.15       6.50                      \n",
      "   HIS 289 A     5.35       6.50                      \n",
      "   CYS  21 A    12.37       9.00                      \n",
      "   CYS 119 A    10.39       9.00                      \n",
      "   CYS 167 A    13.04       9.00                      \n",
      "   CYS 190 A    10.90       9.00                      \n",
      "   TYR   3 A    10.76      10.00                      \n",
      "   TYR  47 A    12.70      10.00                      \n",
      "   TYR 148 A    15.46      10.00                      \n",
      "   TYR 201 A    10.14      10.00                      \n",
      "   TYR 263 A    15.09      10.00                      \n",
      "   TYR 274 A     9.17      10.00                      \n",
      "   TYR 282 A    12.77      10.00                      \n",
      "   TYR 318 A     9.59      10.00                      \n",
      "   LYS   5 A    10.80      10.50                      \n",
      "   LYS  11 A     9.47      10.50                      \n",
      "   LYS  12 A    10.57      10.50                      \n",
      "   LYS  23 A    10.16      10.50                      \n",
      "   LYS  26 A    10.23      10.50                      \n",
      "   LYS  39 A    10.48      10.50                      \n",
      "   LYS  59 A    10.21      10.50                      \n",
      "   LYS  79 A    10.44      10.50                      \n",
      "   LYS  91 A    10.52      10.50                      \n",
      "   LYS 140 A     9.86      10.50                      \n",
      "   LYS 159 A    10.42      10.50                      \n",
      "   LYS 180 A    10.43      10.50                      \n",
      "   LYS 182 A    10.14      10.50                      \n",
      "   LYS 208 A    10.45      10.50                      \n",
      "   LYS 213 A    10.19      10.50                      \n",
      "   LYS 247 A    10.38      10.50                      \n",
      "   LYS 251 A    10.60      10.50                      \n",
      "   LYS 262 A     9.41      10.50                      \n",
      "   LYS 280 A    10.46      10.50                      \n",
      "   LYS 285 A    10.26      10.50                      \n",
      "   LYS 309 A    10.51      10.50                      \n",
      "   LYS 315 A    11.51      10.50                      \n",
      "   LYS 317 A    10.13      10.50                      \n",
      "   LYS 331 A    10.52      10.50                      \n",
      "   LYS 338 A    10.23      10.50                      \n",
      "   LYS 349 A    10.31      10.50                      \n",
      "   ARG  19 A    13.45      12.50                      \n",
      "   ARG  78 A    12.45      12.50                      \n",
      "   ARG  88 A    12.29      12.50                      \n",
      "   ARG 115 A    12.63      12.50                      \n",
      "   ARG 129 A    14.57      12.50                      \n",
      "   ARG 144 A    13.46      12.50                      \n",
      "   ARG 256 A    12.33      12.50                      \n",
      "   ARG 268 A    12.83      12.50                      \n",
      "   ARG 276 A    16.42      12.50                      \n",
      "   ARG 283 A    12.77      12.50                      \n",
      "   ARG 341 A    12.40      12.50                      \n",
      "   N+    1 A     7.94       8.00                      \n",
      "\n",
      "INFO:PROPKA information:\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "Free energy of   folding (kcal/mol) as a function of pH (using neutral reference)\n",
      "  0.00     29.52\n",
      "  1.00     29.44\n",
      "  2.00     28.85\n",
      "  3.00     26.51\n",
      "  4.00     24.84\n",
      "  5.00     30.16\n",
      "  6.00     36.69\n",
      "  7.00     40.13\n",
      "  8.00     42.33\n",
      "  9.00     44.66\n",
      " 10.00     48.46\n",
      " 11.00     55.32\n",
      " 12.00     64.91\n",
      " 13.00     74.71\n",
      " 14.00     82.36\n",
      "\n",
      "The pH of optimum stability is  3.8 for which the free energy is   24.7 kcal/mol at 298K\n",
      "Could not determine pH values where the free energy is within 80 % of minimum\n",
      "Could not determine the pH-range where the free energy is negative\n",
      "\n",
      "\n",
      "Protein charge of folded and unfolded state as a function of pH\n",
      "    pH  unfolded  folded\n",
      "  0.00     42.99   42.98\n",
      "  0.10     42.99   42.98\n",
      "  0.20     42.99   42.97\n",
      "  0.30     42.99   42.96\n",
      "  0.40     42.99   42.95\n",
      "  0.50     42.98   42.94\n",
      "  0.60     42.98   42.92\n",
      "  0.70     42.97   42.90\n",
      "  0.80     42.97   42.88\n",
      "  0.90     42.96   42.85\n",
      "  1.00     42.94   42.81\n",
      "  1.10     42.93   42.76\n",
      "  1.20     42.91   42.71\n",
      "  1.30     42.89   42.64\n",
      "  1.40     42.86   42.55\n",
      "  1.50     42.83   42.44\n",
      "  1.60     42.78   42.32\n",
      "  1.70     42.73   42.16\n",
      "  1.80     42.66   41.98\n",
      "  1.90     42.57   41.77\n",
      "  2.00     42.46   41.52\n",
      "  2.10     42.32   41.23\n",
      "  2.20     42.15   40.90\n",
      "  2.30     41.94   40.52\n",
      "  2.40     41.68   40.08\n",
      "  2.50     41.35   39.59\n",
      "  2.60     40.96   39.03\n",
      "  2.70     40.47   38.40\n",
      "  2.80     39.88   37.69\n",
      "  2.90     39.17   36.89\n",
      "  3.00     38.32   36.00\n",
      "  3.10     37.31   35.00\n",
      "  3.20     36.13   33.89\n",
      "  3.30     34.77   32.68\n",
      "  3.40     33.21   31.35\n",
      "  3.50     31.47   29.91\n",
      "  3.60     29.54   28.38\n",
      "  3.70     27.44   26.75\n",
      "  3.80     25.20   25.06\n",
      "  3.90     22.85   23.30\n",
      "  4.00     20.41   21.51\n",
      "  4.10     17.94   19.69\n",
      "  4.20     15.46   17.87\n",
      "  4.30     13.00   16.06\n",
      "  4.40     10.61   14.27\n",
      "  4.50      8.32   12.53\n",
      "  4.60      6.16   10.84\n",
      "  4.70      4.14    9.21\n",
      "  4.80      2.30    7.66\n",
      "  4.90      0.64    6.18\n",
      "  5.00     -0.83    4.80\n",
      "  5.10     -2.12    3.51\n",
      "  5.20     -3.24    2.30\n",
      "  5.30     -4.21    1.18\n",
      "  5.40     -5.04    0.13\n",
      "  5.50     -5.76   -0.84\n",
      "  5.60     -6.39   -1.75\n",
      "  5.70     -6.94   -2.60\n",
      "  5.80     -7.44   -3.40\n",
      "  5.90     -7.89   -4.15\n",
      "  6.00     -8.32   -4.86\n",
      "  6.10     -8.72   -5.53\n",
      "  6.20     -9.12   -6.16\n",
      "  6.30     -9.51   -6.75\n",
      "  6.40     -9.89   -7.31\n",
      "  6.50    -10.26   -7.82\n",
      "  6.60    -10.62   -8.31\n",
      "  6.70    -10.96   -8.75\n",
      "  6.80    -11.28   -9.16\n",
      "  6.90    -11.58   -9.54\n",
      "  7.00    -11.85   -9.90\n",
      "  7.10    -12.10  -10.23\n",
      "  7.20    -12.34  -10.54\n",
      "  7.30    -12.55  -10.82\n",
      "  7.40    -12.74  -11.09\n",
      "  7.50    -12.93  -11.35\n",
      "  7.60    -13.11  -11.58\n",
      "  7.70    -13.29  -11.81\n",
      "  7.80    -13.47  -12.02\n",
      "  7.90    -13.66  -12.23\n",
      "  8.00    -13.86  -12.44\n",
      "  8.10    -14.08  -12.64\n",
      "  8.20    -14.31  -12.84\n",
      "  8.30    -14.57  -13.05\n",
      "  8.40    -14.85  -13.27\n",
      "  8.50    -15.17  -13.51\n",
      "  8.60    -15.53  -13.77\n",
      "  8.70    -15.93  -14.06\n",
      "  8.80    -16.37  -14.40\n",
      "  8.90    -16.87  -14.78\n",
      "  9.00    -17.42  -15.22\n",
      "  9.10    -18.04  -15.72\n",
      "  9.20    -18.72  -16.30\n",
      "  9.30    -19.49  -16.96\n",
      "  9.40    -20.34  -17.71\n",
      "  9.50    -21.30  -18.55\n",
      "  9.60    -22.37  -19.50\n",
      "  9.70    -23.56  -20.55\n",
      "  9.80    -24.88  -21.72\n",
      "  9.90    -26.33  -22.98\n",
      " 10.00    -27.91  -24.34\n",
      " 10.10    -29.60  -25.79\n",
      " 10.20    -31.40  -27.31\n",
      " 10.30    -33.26  -28.87\n",
      " 10.40    -35.16  -30.45\n",
      " 10.50    -37.06  -32.01\n",
      " 10.60    -38.92  -33.54\n",
      " 10.70    -40.70  -34.99\n",
      " 10.80    -42.38  -36.37\n",
      " 10.90    -43.92  -37.64\n",
      " 11.00    -45.32  -38.81\n",
      " 11.10    -46.58  -39.88\n",
      " 11.20    -47.70  -40.84\n",
      " 11.30    -48.69  -41.72\n",
      " 11.40    -49.58  -42.51\n",
      " 11.50    -50.38  -43.25\n",
      " 11.60    -51.11  -43.93\n",
      " 11.70    -51.80  -44.59\n",
      " 11.80    -52.46  -45.22\n",
      " 11.90    -53.11  -45.85\n",
      " 12.00    -53.76  -46.49\n",
      " 12.10    -54.43  -47.14\n",
      " 12.20    -55.11  -47.80\n",
      " 12.30    -55.81  -48.49\n",
      " 12.40    -56.51  -49.20\n",
      " 12.50    -57.22  -49.92\n",
      " 12.60    -57.90  -50.65\n",
      " 12.70    -58.56  -51.38\n",
      " 12.80    -59.18  -52.10\n",
      " 12.90    -59.75  -52.81\n",
      " 13.00    -60.27  -53.49\n",
      " 13.10    -60.72  -54.14\n",
      " 13.20    -61.11  -54.75\n",
      " 13.30    -61.45  -55.32\n",
      " 13.40    -61.73  -55.85\n",
      " 13.50    -61.97  -56.34\n",
      " 13.60    -62.17  -56.79\n",
      " 13.70    -62.33  -57.20\n",
      " 13.80    -62.46  -57.56\n",
      " 13.90    -62.57  -57.89\n",
      " 14.00    -62.65  -58.19\n",
      "The pI is  5.41 (folded) and  4.94 (unfolded)\n",
      "\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "INFO:Applying pKa values at a pH of 8.20:\n",
      "INFO:Adding hydrogens to biomolecule.\n",
      "INFO:Debumping biomolecule (again).\n",
      "WARNING:WARNING: Unable to debump LEU A 29\n",
      "INFO:Optimizing hydrogen bonds\n",
      "INFO:Applying force field to biomolecule states.\n",
      "INFO:Regenerating headers.\n",
      "INFO:Regenerating PDB lines.\n",
      "WARNING:Ignoring 467 header lines in output.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "pH: 8.2\tname: AtSAL1_updated\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:PDB2PQR v3.6.2: biomolecular structure conversion software.\n",
      "INFO:Please cite:  Jurrus E, et al.  Improvements to the APBS biomolecular solvation software suite.  Protein Sci 27 112-128 (2018).\n",
      "INFO:Please cite:  Dolinsky TJ, et al.  PDB2PQR: expanding and upgrading automated preparation of biomolecular structures for molecular simulations. Nucleic Acids Res 35 W522-W525 (2007).\n",
      "INFO:Checking and transforming input arguments.\n",
      "INFO:Loading topology files.\n",
      "INFO:Loading molecule: /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/AtAHL.pdb\n",
      "INFO:Setting up molecule.\n",
      "INFO:Created biomolecule object with 368 residues and 5474 atoms.\n",
      "INFO:Setting termini states for biomolecule chains.\n",
      "INFO:Loading forcefield.\n",
      "INFO:Loading hydrogen topology definitions.\n",
      "INFO:This biomolecule is clean.  No repair needed.\n",
      "INFO:Updating disulfide bridges.\n",
      "INFO:Debumping biomolecule.\n",
      "INFO:Assigning titration states with PROPKA.\n",
      "INFO:\n",
      "propka3.5.1                                                                                  2024-07-26\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "--                                                                           --\n",
      "--  PROPKA: A PROTEIN PKA PREDICTOR                                          --\n",
      "--                                                                           --\n",
      "--  VERSION 1.0,  04/25/2004,  IOWA CITY                                     --\n",
      "--  BY HUI LI                                                                --\n",
      "--                                                                           --\n",
      "--  VERSION 2.0,  11/05/2007, IOWA CITY/COPENHAGEN                           --\n",
      "--  BY DELPHINE C. BAS AND DAVID M. ROGERS                                   --\n",
      "--                                                                           --\n",
      "--  VERSION 3.0,  01/06/2011, COPENHAGEN                                     --\n",
      "--  BY MATS H.M. OLSSON AND CHRESTEN R. SONDERGARD                           --\n",
      "--                                                                           --\n",
      "--  VERSION 3.1,  07/01/2011, COPENHAGEN                                     --\n",
      "--  BY CHRESTEN R. SONDERGARD AND MATS H.M. OLSSON                           --\n",
      "--                                                                           --\n",
      "--  VERSION 3.2,  06/17/2020, PLANET EARTH                                   --\n",
      "--  SEE ABOVE FOR AUTHORS                                                    --\n",
      "--                                                                           --\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "References:\n",
      "\n",
      "Very Fast Empirical Prediction and Rationalization of Protein pKa Values.\n",
      "Hui Li, Andrew D. Robertson and Jan H. Jensen. PROTEINS: Structure, Function,\n",
      "and Bioinformatics. 61:704-721 (2005)\n",
      "\n",
      "Very Fast Prediction and Rationalization of pKa Values for Protein-Ligand\n",
      "Complexes.  Delphine C. Bas, David M. Rogers and Jan H. Jensen.  PROTEINS:\n",
      "Structure, Function, and Bioinformatics 73:765-783 (2008)\n",
      "\n",
      "PROPKA3: Consistent Treatment of Internal and Surface Residues in Empirical\n",
      "pKa predictions.  Mats H.M. Olsson, Chresten R. Sondergard, Michal Rostkowski,\n",
      "and Jan H. Jensen.  Journal of Chemical Theory and Computation, 7(2):525-537\n",
      "(2011)\n",
      "\n",
      "Improved Treatment of Ligands and Coupling Effects in Empirical Calculation\n",
      "and Rationalization of pKa Values.  Chresten R. Sondergaard, Mats H.M. Olsson,\n",
      "Michal Rostkowski, and Jan H. Jensen.  Journal of Chemical Theory and\n",
      "Computation, (2011)\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "\n",
      "INFO:Coupling map:\n",
      "\n",
      "INFO:Calculating pKas for Conformation container 1A with 3218 atoms and 894 groups\n",
      "INFO:did not converge in 10 iterations\n",
      "INFO:pKa determinants:\n",
      "\n",
      "---------  -----   ------   ---------------------    --------------    --------------    --------------\n",
      "                            DESOLVATION  EFFECTS       SIDECHAIN          BACKBONE        COULOMBIC\n",
      " RESIDUE    pKa    BURIED     REGULAR      RE        HYDROGEN BOND     HYDROGEN BOND      INTERACTION\n",
      "---------  -----   ------   ---------   ---------    --------------    --------------    --------------\n",
      "\n",
      "ASP   6 A   3.00    28 %    0.95  359   0.08    0   -1.34 ARG  10 A    0.00 XXX   0 X   -0.50 ARG  10 A\n",
      "\n",
      "ASP  38 A   4.10     0 %    0.13  168   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.04 LYS  35 A\n",
      "ASP  38 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 ASP  39 A\n",
      "ASP  38 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP  40 A\n",
      "ASP  38 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP  74 A\n",
      "ASP  38 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 GLU  80 A\n",
      "\n",
      "ASP  39 A   2.55    20 %    0.70  337   0.00    0   -0.85 SER  41 A   -0.18 ASP  39 A   -0.26 LYS  37 A\n",
      "ASP  39 A                                            0.00 XXX   0 X   -0.61 SER  41 A   -0.27 LYS 302 A\n",
      "ASP  39 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.23 GLU  72 A\n",
      "\n",
      "ASP  40 A   4.01     0 %    0.14  138   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP  39 A\n",
      "\n",
      "ASP  47 A   5.77*  100 %    3.69  595   1.08    0   -0.85 THR  44 A   -0.64 THR  44 A    0.41 GLU  72 A\n",
      "ASP  47 A                                           -0.59 THR 144 A   -0.44 ASP  47 A   -0.69 LYS  37 A\n",
      "\n",
      "ASP  64 A   2.40     0 %    0.38  280   0.00    0   -0.30 LYS 120 A    0.00 XXX   0 X   -0.34 LYS 120 A\n",
      "ASP  64 A                                           -0.68 ARG 124 A    0.00 XXX   0 X   -0.46 ARG 124 A\n",
      "\n",
      "ASP  74 A   3.59    27 %    0.81  358   0.01    0    0.00 XXX   0 X   -0.53 GLU  76 A   -0.02 LYS  35 A\n",
      "ASP  74 A                                            0.00 XXX   0 X   -0.39 THR  77 A   -0.11 LYS  37 A\n",
      "ASP  74 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU  80 A\n",
      "\n",
      "ASP  82 A   3.93     0 %    0.12  119   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.07 LYS  35 A\n",
      "ASP  82 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 GLU  80 A\n",
      "\n",
      "ASP 139 A  11.53*  100 %    3.50  634   0.39    0    0.49 ASP 305 A   -0.33 VAL 141 A   -0.41 LYS  37 A\n",
      "ASP 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.21 LYS 279 A\n",
      "ASP 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.50 LYS 302 A\n",
      "ASP 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.40 ASP  47 A\n",
      "ASP 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.37 GLU  73 A\n",
      "ASP 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.29 GLU  72 A\n",
      "ASP 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.51 ASP 142 A\n",
      "ASP 139 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.21 ASP 305 A\n",
      "\n",
      "ASP 142 A   5.13*  100 %    3.07  665   0.33    0   -0.07 SER 276 A   -0.04 GLY 143 A   -0.34 LYS 293 A\n",
      "ASP 142 A                                            0.00 XXX   0 X   -0.28 MET 277 A   -0.13 LYS 302 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.33 GLU  72 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 249 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.12 GLU 301 A\n",
      "ASP 142 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.69 LYS 279 A\n",
      "\n",
      "ASP 151 A   4.73    32 %    0.77  370   0.33    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.15 ARG 149 A\n",
      "ASP 151 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 HIS  27 A\n",
      "\n",
      "ASP 237 A   3.65     0 %    0.27  219   0.00    0    0.00 XXX   0 X   -0.21 ASP 237 A   -0.31 LYS 354 A\n",
      "ASP 237 A                                            0.00 XXX   0 X   -0.00 ASP 238 A    0.10 ASP 238 A\n",
      "\n",
      "ASP 238 A   3.28     0 %    0.44  227   0.00    0    0.00 XXX   0 X   -0.76 VAL 240 A   -0.20 LYS 354 A\n",
      "\n",
      "ASP 287 A   3.73    64 %    2.21  461   0.60    0   -0.54 TYR 175 A    0.00 XXX   0 X   -0.47 LYS 179 A\n",
      "ASP 287 A                                           -0.37 LYS 179 A    0.00 XXX   0 X   -0.85 ARG 285 A\n",
      "ASP 287 A                                           -0.66 ARG 285 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "ASP 305 A   6.18*  100 %    2.94  592   0.45    0   -0.49 ASP 139 A   -0.22 LYS 302 A   -0.07 LYS  37 A\n",
      "ASP 305 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.99 GLU  72 A\n",
      "ASP 305 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.26 ASP 142 A\n",
      "ASP 305 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.66 LYS 279 A\n",
      "ASP 305 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.74 LYS 293 A\n",
      "ASP 305 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.72 GLU 301 A\n",
      "ASP 305 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.89 LYS 302 A\n",
      "ASP 305 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.91 HIS 306 A\n",
      "\n",
      "ASP 322 A   3.09    35 %    1.70  379   0.53    0   -0.75 ARG 326 A   -0.40 GLY 324 A   -0.10 ARG 341 A\n",
      "ASP 322 A                                            0.00 XXX   0 X   -0.03 GLY 325 A    0.28 ASP 340 A\n",
      "ASP 322 A                                            0.00 XXX   0 X   -0.78 ARG 326 A   -0.03 HIS 352 A\n",
      "ASP 322 A                                            0.00 XXX   0 X   -0.72 ASP 340 A   -0.42 ARG 326 A\n",
      "\n",
      "ASP 329 A   2.58    14 %    0.73  320   0.01    0   -0.85 SER 331 A   -0.78 SER 331 A   -0.01 ARG 326 A\n",
      "ASP 329 A                                           -0.01 LYS 332 A    0.00 XXX   0 X   -0.32 LYS 332 A\n",
      "\n",
      "ASP 340 A   2.65    15 %    0.87  323   0.00    0   -1.39 ARG 341 A    0.00 XXX   0 X   -0.19 ARG 326 A\n",
      "ASP 340 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.44 ARG 341 A\n",
      "\n",
      "GLU   2 A   3.46     0 %    0.26  241   0.00    0   -0.85 LYS 208 A    0.00 XXX   0 X   -0.14 N+    1 A\n",
      "GLU   2 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 GLU   4 A\n",
      "GLU   2 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP   6 A\n",
      "GLU   2 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.38 LYS 208 A\n",
      "\n",
      "GLU   4 A   2.86    16 %    0.81  327   0.00    0   -0.85 N+    1 A   -0.68 GLU   4 A   -0.01 LYS 208 A\n",
      "GLU   4 A                                           -0.25 ARG 134 A    0.00 XXX   0 X   -0.41 N+    1 A\n",
      "GLU   4 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.26 ARG 134 A\n",
      "\n",
      "GLU  24 A   4.74     4 %    0.29  293   0.02    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS  21 A\n",
      "GLU  24 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS  25 A\n",
      "GLU  24 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 HIS  27 A\n",
      "GLU  24 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU  96 A\n",
      "\n",
      "GLU  60 A   5.00    13 %    0.50  319   0.03    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.02 LYS 120 A\n",
      "GLU  60 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.07 ARG 124 A\n",
      "GLU  60 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP  64 A\n",
      "GLU  60 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 117 A\n",
      "\n",
      "GLU  72 A   1.96*   90 %    2.13  534   0.00    0   -0.71 TRP 304 A    0.00 XXX   0 X   -1.16 LYS  37 A\n",
      "GLU  72 A                                           -0.85 LYS  37 A    0.00 XXX   0 X   -1.10 LYS 302 A\n",
      "GLU  72 A                                           -0.84 LYS 302 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "GLU  73 A   7.93*   92 %    3.22  539   0.02    0   -0.85 LYS  37 A    0.00 XXX   0 X   -0.23 LYS 302 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP  39 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 ASP  74 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.18 LYS  37 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.24 ASP  47 A\n",
      "GLU  73 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.93 GLU  72 A\n",
      "\n",
      "GLU  76 A   4.99     0 %    0.13  203   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP  38 A\n",
      "GLU  76 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.27 ASP  74 A\n",
      "GLU  76 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU  80 A\n",
      "\n",
      "GLU  80 A   3.53     0 %    0.26  214   0.00    0   -0.85 LYS  35 A    0.00 XXX   0 X   -0.38 LYS  35 A\n",
      "\n",
      "GLU  96 A   3.28    17 %    0.51  330   0.04    0   -0.20 LYS  21 A    0.00 XXX   0 X   -0.37 LYS  21 A\n",
      "GLU  96 A                                           -0.76 LYS  25 A    0.00 XXX   0 X   -0.44 LYS  25 A\n",
      "\n",
      "GLU 100 A   4.18    21 %    0.73  339   0.12    0   -0.85 LYS  21 A    0.00 XXX   0 X   -0.01 LYS  25 A\n",
      "GLU 100 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 LYS 108 A\n",
      "GLU 100 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.17 GLU  96 A\n",
      "GLU 100 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.47 LYS  21 A\n",
      "\n",
      "GLU 117 A   3.73     0 %    0.34  255   0.00    0    0.00 XXX   0 X   -0.81 GLY 114 A   -0.26 LYS 111 A\n",
      "GLU 117 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 LYS 120 A\n",
      "\n",
      "GLU 161 A   3.76     0 %    0.38  263   0.00    0   -0.74 ARG 134 A    0.00 XXX   0 X   -0.04 N+    1 A\n",
      "GLU 161 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 LYS 164 A\n",
      "GLU 161 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU   4 A\n",
      "GLU 161 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.38 ARG 134 A\n",
      "\n",
      "GLU 180 A   4.49     0 %    0.11  156   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.12 LYS 178 A\n",
      "\n",
      "GLU 226 A   4.56     0 %    0.06  112   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "GLU 246 A   5.27    37 %    1.07  386   0.02    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.20 ARG 273 A\n",
      "GLU 246 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.12 HIS 255 A\n",
      "\n",
      "GLU 249 A   5.27    46 %    0.93  410   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.10 ARG 149 A\n",
      "GLU 249 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 293 A\n",
      "GLU 249 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP  39 A\n",
      "GLU 249 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 HIS 255 A\n",
      "\n",
      "GLU 289 A   3.91    55 %    1.68  435   0.54    0   -0.85 SER 234 A   -0.79 THR 243 A   -0.18 LYS 179 A\n",
      "GLU 289 A                                           -0.62 THR 243 A   -0.00 GLU 289 A   -0.45 LYS 269 A\n",
      "GLU 289 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP 287 A\n",
      "\n",
      "GLU 301 A   1.69    92 %    3.27  539   0.00    0   -0.85 LYS 293 A    0.00 XXX   0 X   -0.03 LYS 302 A\n",
      "GLU 301 A                                           -1.60 HIS 306 A    0.00 XXX   0 X   -0.52 LYS 279 A\n",
      "GLU 301 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.62 LYS 293 A\n",
      "GLU 301 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.45 HIS 306 A\n",
      "\n",
      "GLU 314 A   5.47    70 %    1.78  477   0.58    0   -0.74 ARG 209 A    0.00 XXX   0 X    0.16 GLU 315 A\n",
      "GLU 314 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.81 ARG 209 A\n",
      "\n",
      "GLU 315 A   4.02    32 %    1.27  372   0.32    0   -0.15 SER 211 A   -0.81 GLY 210 A   -0.02 LYS 208 A\n",
      "GLU 315 A                                            0.00 XXX   0 X   -0.11 SER 211 A   -0.28 ARG 209 A\n",
      "GLU 315 A                                            0.00 XXX   0 X   -0.70 GLY 212 A    0.00 XXX   0 X\n",
      "\n",
      "GLU 337 A   3.62     0 %    0.36  250   0.00    0   -0.85 LYS 300 A    0.00 XXX   0 X   -0.01 LYS 332 A\n",
      "GLU 337 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.38 LYS 300 A\n",
      "\n",
      "GLU 353 A   3.83     0 %    0.26  212   0.00    0   -0.84 GLN 349 A    0.00 XXX   0 X   -0.01 LYS 354 A\n",
      "GLU 353 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 HIS 352 A\n",
      "\n",
      "GLU 364 A   4.60     0 %    0.12  133   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.01 ARG 341 A\n",
      "\n",
      "C-  368 A   3.32     0 %    0.12   79   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "HIS  13 A   4.57    81 %   -2.69  507   0.00    0    0.82 GLN 217 A    0.00 XXX   0 X   -0.06 ARG  10 A\n",
      "\n",
      "HIS  27 A   6.82     6 %   -0.41  298   0.00    0    0.00 XXX   0 X    0.68 GLN  23 A    0.01 GLU  24 A\n",
      "HIS  27 A                                            0.00 XXX   0 X    0.01 ARG 149 A    0.02 ASP 151 A\n",
      "\n",
      "HIS  33 A   6.06    11 %   -0.44  312   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "HIS 135 A   5.85    35 %   -1.10  379   0.00    0    0.00 XXX   0 X    0.45 LYS 132 A    0.00 XXX   0 X\n",
      "\n",
      "HIS 255 A   6.41    20 %   -0.86  336   0.00    0    0.00 XXX   0 X    0.78 GLU 249 A   -0.09 ARG 250 A\n",
      "HIS 255 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 273 A\n",
      "HIS 255 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 LYS 293 A\n",
      "HIS 255 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.12 GLU 246 A\n",
      "HIS 255 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 249 A\n",
      "\n",
      "HIS 306 A   5.07   100 %   -3.28  641   0.00    0    1.60 GLU 301 A    0.00 XXX   0 X   -1.25 LYS 279 A\n",
      "HIS 306 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.85 LYS 293 A\n",
      "HIS 306 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.45 GLU 301 A\n",
      "HIS 306 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.91 ASP 305 A\n",
      "\n",
      "HIS 352 A   4.72    65 %   -1.90  462   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.03 ASP 322 A\n",
      "HIS 352 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU 353 A\n",
      "\n",
      "CYS  19 A  13.08   100 %    3.78  655   0.00    0    0.00 XXX   0 X   -0.08 CYS  19 A    0.39 ASP  47 A\n",
      "\n",
      "CYS 125 A  11.63    96 %    3.54  549   0.00    0   -0.05 GLN  51 A    0.00 XXX   0 X   -0.23 ARG 124 A\n",
      "CYS 125 A                                           -0.63 SER  55 A    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "CYS 172 A  13.72   100 %    3.47  673   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.18 ARG 285 A\n",
      "CYS 172 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 ASP 142 A\n",
      "CYS 172 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.35 CYS 203 A\n",
      "\n",
      "CYS 181 A   9.45     5 %    0.55  295   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 178 A\n",
      "CYS 181 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.10 LYS 201 A\n",
      "CYS 181 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 ARG 285 A\n",
      "CYS 181 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.11 GLU 180 A\n",
      "CYS 181 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP 287 A\n",
      "\n",
      "CYS 186 A   9.82    31 %    0.88  368   0.00    0    0.00 XXX   0 X   -0.08 CYS 186 A   -0.15 ARG 273 A\n",
      "CYS 186 A                                            0.00 XXX   0 X   -0.02 ASN 187 A    0.19 ASP 151 A\n",
      "\n",
      "CYS 203 A  11.78   100 %    3.38  575   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.65 ARG 285 A\n",
      "CYS 203 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 ASP 287 A\n",
      "\n",
      "CYS 245 A  12.50   100 %    3.40  607   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.11 LYS 279 A\n",
      "CYS 245 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 246 A\n",
      "CYS 245 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 ASP 287 A\n",
      "CYS 245 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.12 GLU 289 A\n",
      "\n",
      "CYS 346 A  12.74   100 %    3.74  577   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "TYR 104 A  10.78    56 %    2.22  438   0.00    0   -0.85 SER  17 A   -0.68 GLY 222 A    0.31 GLU 100 A\n",
      "TYR 104 A                                            0.00 XXX   0 X   -0.01 GLY 223 A   -0.21 LYS  21 A\n",
      "\n",
      "TYR 153 A  15.70   100 %    4.01  560   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    1.59 CYS  19 A\n",
      "TYR 153 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU  24 A\n",
      "TYR 153 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 CYS 172 A\n",
      "\n",
      "TYR 175 A  15.46   100 %    3.86  560   0.00    0   -0.21 ARG 285 A    0.00 XXX   0 X    0.40 CYS 172 A\n",
      "TYR 175 A                                            0.54 ASP 287 A    0.00 XXX   0 X    0.54 CYS 203 A\n",
      "TYR 175 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.43 CYS 245 A\n",
      "TYR 175 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -1.24 ARG 285 A\n",
      "TYR 175 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.15 ASP 287 A\n",
      "\n",
      "TYR 206 A  11.50    48 %    1.46  417   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU   2 A\n",
      "TYR 206 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 ASP   6 A\n",
      "TYR 206 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.00 ARG  10 A\n",
      "\n",
      "TYR 275 A  12.11    30 %    1.75  364   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X   -0.17 ARG 149 A\n",
      "TYR 275 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 ASP 151 A\n",
      "TYR 275 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 CYS 186 A\n",
      "TYR 275 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.49 GLU 249 A\n",
      "TYR 275 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.01 ARG 250 A\n",
      "TYR 275 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 ARG 273 A\n",
      "\n",
      "TYR 280 A  14.55   100 %    3.81  663   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.14 ASP 142 A\n",
      "TYR 280 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.40 GLU 301 A\n",
      "TYR 280 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.12 ASP 305 A\n",
      "TYR 280 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU 314 A\n",
      "\n",
      "TYR 299 A  14.29    74 %    1.49  488   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.08 ASP  39 A\n",
      "TYR 299 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.23 GLU  72 A\n",
      "TYR 299 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.09 ASP 139 A\n",
      "TYR 299 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.32 ASP 142 A\n",
      "TYR 299 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.08 GLU 249 A\n",
      "TYR 299 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 TYR 335 A\n",
      "TYR 299 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.93 GLU 301 A\n",
      "TYR 299 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.05 ASP 305 A\n",
      "\n",
      "TYR 335 A  10.09     6 %    0.28  299   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.01 ASP  39 A\n",
      "TYR 335 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU  72 A\n",
      "TYR 335 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.00 ASP 305 A\n",
      "TYR 335 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.05 GLU 337 A\n",
      "TYR 335 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.10 LYS 300 A\n",
      "TYR 335 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.18 LYS 302 A\n",
      "\n",
      "TYR 360 A  10.04     0 %    0.35  261   0.00    0   -0.33 ARG 341 A    0.00 XXX   0 X    0.11 ASP 322 A\n",
      "TYR 360 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.07 ARG 326 A\n",
      "TYR 360 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.28 ASP 340 A\n",
      "TYR 360 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.30 ARG 341 A\n",
      "\n",
      "LYS  21 A  11.63    24 %   -1.00  348   0.00    0    0.20 GLU  96 A    0.00 XXX   0 X    0.03 GLU  24 A\n",
      "LYS  21 A                                            0.85 GLU 100 A    0.00 XXX   0 X    0.21 TYR 104 A\n",
      "LYS  21 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.37 GLU  96 A\n",
      "LYS  21 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.47 GLU 100 A\n",
      "\n",
      "LYS  25 A  11.09    17 %   -0.54  328   0.00    0    0.76 GLU  96 A    0.00 XXX   0 X    0.03 GLU  24 A\n",
      "LYS  25 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 100 A\n",
      "LYS  25 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.11 LYS  21 A\n",
      "LYS  25 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.44 GLU  96 A\n",
      "\n",
      "LYS  35 A  11.28     0 %   -0.58  275   0.00    0    0.85 GLU  80 A    0.00 XXX   0 X    0.04 ASP  38 A\n",
      "LYS  35 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 ASP  74 A\n",
      "LYS  35 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 ASP  82 A\n",
      "LYS  35 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 GLU  80 A\n",
      "\n",
      "LYS  37 A  13.37    74 %   -2.71  488   0.00    0    0.85 GLU  72 A    0.00 XXX   0 X    0.26 ASP  39 A\n",
      "LYS  37 A                                            0.85 GLU  73 A    0.00 XXX   0 X    0.11 ASP  74 A\n",
      "LYS  37 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.41 ASP 139 A\n",
      "LYS  37 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 ASP 305 A\n",
      "LYS  37 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.69 ASP  47 A\n",
      "LYS  37 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.16 GLU  72 A\n",
      "LYS  37 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.18 GLU  73 A\n",
      "\n",
      "LYS 108 A  10.20     0 %   -0.32  220   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 100 A\n",
      "\n",
      "LYS 111 A  10.66     0 %   -0.10  128   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.26 GLU 117 A\n",
      "\n",
      "LYS 120 A  10.80     0 %   -0.19  244   0.00    0    0.30 ASP  64 A    0.00 XXX   0 X    0.02 GLU  60 A\n",
      "LYS 120 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 117 A\n",
      "LYS 120 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.20 ARG 124 A\n",
      "LYS 120 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.34 ASP  64 A\n",
      "\n",
      "LYS 132 A  10.46     0 %   -0.04   93   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 164 A  10.39     0 %   -0.10  198   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 161 A\n",
      "LYS 164 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.02 ARG 209 A\n",
      "\n",
      "LYS 178 A  10.56     0 %   -0.13  165   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.12 GLU 180 A\n",
      "LYS 178 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 CYS 181 A\n",
      "\n",
      "LYS 179 A  10.81    19 %   -0.61  334   0.00    0    0.37 ASP 287 A    0.00 XXX   0 X   -0.11 ARG 285 A\n",
      "LYS 179 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.18 GLU 289 A\n",
      "LYS 179 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.47 ASP 287 A\n",
      "\n",
      "LYS 191 A  10.47     0 %   -0.03   79   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 193 A  10.48     0 %   -0.02   43   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 201 A  10.15    29 %   -0.45  362   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.10 CYS 181 A\n",
      "\n",
      "LYS 208 A  11.45     0 %   -0.31  254   0.00    0    0.85 GLU   2 A    0.00 XXX   0 X    0.01 GLU   4 A\n",
      "LYS 208 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.02 GLU 315 A\n",
      "LYS 208 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 GLU   2 A\n",
      "\n",
      "LYS 232 A  10.41     0 %   -0.09  201   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "LYS 269 A  10.30    12 %   -0.57  314   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.45 GLU 289 A\n",
      "LYS 269 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.08 LYS 179 A\n",
      "\n",
      "LYS 279 A   9.58   100 %   -3.37  661   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.21 ASP 139 A\n",
      "LYS 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.11 CYS 245 A\n",
      "LYS 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.74 LYS 293 A\n",
      "LYS 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.69 ASP 142 A\n",
      "LYS 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.52 GLU 301 A\n",
      "LYS 279 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.66 ASP 305 A\n",
      "\n",
      "LYS 293 A  10.97    95 %   -3.08  548   0.00    0    0.85 GLU 301 A    0.00 XXX   0 X    0.34 ASP 142 A\n",
      "LYS 293 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 GLU 249 A\n",
      "LYS 293 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.06 LYS 302 A\n",
      "LYS 293 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.62 GLU 301 A\n",
      "LYS 293 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.74 ASP 305 A\n",
      "\n",
      "LYS 300 A  11.52     0 %   -0.32  217   0.00    0    0.85 GLU 337 A    0.00 XXX   0 X    0.10 TYR 335 A\n",
      "LYS 300 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.38 GLU 337 A\n",
      "\n",
      "LYS 302 A  12.40    70 %   -1.84  477   0.00    0    0.84 GLU  72 A    0.00 XXX   0 X    0.27 ASP  39 A\n",
      "LYS 302 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.23 GLU  73 A\n",
      "LYS 302 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.50 ASP 139 A\n",
      "LYS 302 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.13 ASP 142 A\n",
      "LYS 302 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 GLU 301 A\n",
      "LYS 302 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.44 LYS  37 A\n",
      "LYS 302 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.18 TYR 335 A\n",
      "LYS 302 A                                            0.00 XXX   0 X    0.00 XXX   0 X    1.10 GLU  72 A\n",
      "LYS 302 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.89 ASP 305 A\n",
      "\n",
      "LYS 332 A  10.67     0 %   -0.17  202   0.00    0    0.01 ASP 329 A    0.00 XXX   0 X    0.32 ASP 329 A\n",
      "LYS 332 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 337 A\n",
      "\n",
      "LYS 354 A  10.57     0 %   -0.46  252   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.31 ASP 237 A\n",
      "LYS 354 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 ASP 238 A\n",
      "LYS 354 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 353 A\n",
      "\n",
      "ARG  10 A  13.55    29 %   -0.79  363   0.00    0    1.34 ASP   6 A    0.00 XXX   0 X    0.00 TYR 206 A\n",
      "ARG  10 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.50 ASP   6 A\n",
      "\n",
      "ARG 124 A  12.59    43 %   -1.35  401   0.00    0    0.68 ASP  64 A    0.00 XXX   0 X    0.07 GLU  60 A\n",
      "ARG 124 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.23 CYS 125 A\n",
      "ARG 124 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.46 ASP  64 A\n",
      "\n",
      "ARG 134 A  13.76     4 %   -0.37  293   0.00    0    0.25 GLU   4 A    0.00 XXX   0 X    0.26 GLU   4 A\n",
      "ARG 134 A                                            0.74 GLU 161 A    0.00 XXX   0 X    0.38 GLU 161 A\n",
      "\n",
      "ARG 149 A  12.62     0 %   -0.29  273   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.15 ASP 151 A\n",
      "ARG 149 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.10 GLU 249 A\n",
      "ARG 149 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.17 TYR 275 A\n",
      "\n",
      "ARG 209 A  12.83    59 %   -1.51  447   0.00    0    0.74 GLU 314 A    0.00 XXX   0 X    0.28 GLU 315 A\n",
      "ARG 209 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.81 GLU 314 A\n",
      "\n",
      "ARG 250 A  12.37     0 %   -0.09  157   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.01 TYR 275 A\n",
      "ARG 250 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.05 ARG 273 A\n",
      "\n",
      "ARG 268 A  12.32     0 %   -0.18  156   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.00 XXX   0 X\n",
      "\n",
      "ARG 273 A  12.69     0 %   -0.19  279   0.00    0    0.00 XXX   0 X    0.00 XXX   0 X    0.15 CYS 186 A\n",
      "ARG 273 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.20 GLU 246 A\n",
      "ARG 273 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.03 TYR 275 A\n",
      "\n",
      "ARG 285 A  13.75    70 %   -2.59  478   0.00    0    0.21 TYR 175 A    0.00 XXX   0 X    0.18 CYS 172 A\n",
      "ARG 285 A                                            0.66 ASP 287 A    0.00 XXX   0 X    1.24 TYR 175 A\n",
      "ARG 285 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.06 CYS 181 A\n",
      "ARG 285 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.65 CYS 203 A\n",
      "ARG 285 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.85 ASP 287 A\n",
      "\n",
      "ARG 326 A  13.60     0 %   -0.31  246   0.00    0    0.75 ASP 322 A    0.00 XXX   0 X    0.01 ASP 329 A\n",
      "ARG 326 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.19 ASP 340 A\n",
      "ARG 326 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.07 TYR 360 A\n",
      "ARG 326 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.03 ARG 341 A\n",
      "ARG 326 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.42 ASP 322 A\n",
      "\n",
      "ARG 341 A  14.03    17 %   -1.04  330   0.00    0    0.33 TYR 360 A    0.00 XXX   0 X    0.10 ASP 322 A\n",
      "ARG 341 A                                            1.39 ASP 340 A    0.00 XXX   0 X    0.30 TYR 360 A\n",
      "ARG 341 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.01 GLU 364 A\n",
      "ARG 341 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.44 ASP 340 A\n",
      "\n",
      "N+    1 A   8.68     0 %   -0.50  274   0.00    0    0.85 GLU   4 A    0.00 XXX   0 X    0.14 GLU   2 A\n",
      "N+    1 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.16 ARG 134 A\n",
      "N+    1 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.04 GLU 161 A\n",
      "N+    1 A                                            0.00 XXX   0 X    0.00 XXX   0 X   -0.10 LYS 208 A\n",
      "N+    1 A                                            0.00 XXX   0 X    0.00 XXX   0 X    0.41 GLU   4 A\n",
      "\n",
      "Coupled residues (marked *) were detected.Please rerun PropKa with the --display-coupled-residues \n",
      "or -d option for detailed information.\n",
      "\n",
      "INFO:pKa summary:\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "SUMMARY OF THIS PREDICTION\n",
      "       Group      pKa  model-pKa   ligand atom-type\n",
      "   ASP   6 A     3.00       3.80                      \n",
      "   ASP  38 A     4.10       3.80                      \n",
      "   ASP  39 A     2.55       3.80                      \n",
      "   ASP  40 A     4.01       3.80                      \n",
      "   ASP  47 A     5.77       3.80                      \n",
      "   ASP  64 A     2.40       3.80                      \n",
      "   ASP  74 A     3.59       3.80                      \n",
      "   ASP  82 A     3.93       3.80                      \n",
      "   ASP 139 A    11.53       3.80                      \n",
      "   ASP 142 A     5.13       3.80                      \n",
      "   ASP 151 A     4.73       3.80                      \n",
      "   ASP 237 A     3.65       3.80                      \n",
      "   ASP 238 A     3.28       3.80                      \n",
      "   ASP 287 A     3.73       3.80                      \n",
      "   ASP 305 A     6.18       3.80                      \n",
      "   ASP 322 A     3.09       3.80                      \n",
      "   ASP 329 A     2.58       3.80                      \n",
      "   ASP 340 A     2.65       3.80                      \n",
      "   GLU   2 A     3.46       4.50                      \n",
      "   GLU   4 A     2.86       4.50                      \n",
      "   GLU  24 A     4.74       4.50                      \n",
      "   GLU  60 A     5.00       4.50                      \n",
      "   GLU  72 A     1.96       4.50                      \n",
      "   GLU  73 A     7.93       4.50                      \n",
      "   GLU  76 A     4.99       4.50                      \n",
      "   GLU  80 A     3.53       4.50                      \n",
      "   GLU  96 A     3.28       4.50                      \n",
      "   GLU 100 A     4.18       4.50                      \n",
      "   GLU 117 A     3.73       4.50                      \n",
      "   GLU 161 A     3.76       4.50                      \n",
      "   GLU 180 A     4.49       4.50                      \n",
      "   GLU 226 A     4.56       4.50                      \n",
      "   GLU 246 A     5.27       4.50                      \n",
      "   GLU 249 A     5.27       4.50                      \n",
      "   GLU 289 A     3.91       4.50                      \n",
      "   GLU 301 A     1.69       4.50                      \n",
      "   GLU 314 A     5.47       4.50                      \n",
      "   GLU 315 A     4.02       4.50                      \n",
      "   GLU 337 A     3.62       4.50                      \n",
      "   GLU 353 A     3.83       4.50                      \n",
      "   GLU 364 A     4.60       4.50                      \n",
      "   C-  368 A     3.32       3.20                      \n",
      "   HIS  13 A     4.57       6.50                      \n",
      "   HIS  27 A     6.82       6.50                      \n",
      "   HIS  33 A     6.06       6.50                      \n",
      "   HIS 135 A     5.85       6.50                      \n",
      "   HIS 255 A     6.41       6.50                      \n",
      "   HIS 306 A     5.07       6.50                      \n",
      "   HIS 352 A     4.72       6.50                      \n",
      "   CYS  19 A    13.08       9.00                      \n",
      "   CYS 125 A    11.63       9.00                      \n",
      "   CYS 172 A    13.72       9.00                      \n",
      "   CYS 181 A     9.45       9.00                      \n",
      "   CYS 186 A     9.82       9.00                      \n",
      "   CYS 203 A    11.78       9.00                      \n",
      "   CYS 245 A    12.50       9.00                      \n",
      "   CYS 346 A    12.74       9.00                      \n",
      "   TYR 104 A    10.78      10.00                      \n",
      "   TYR 153 A    15.70      10.00                      \n",
      "   TYR 175 A    15.46      10.00                      \n",
      "   TYR 206 A    11.50      10.00                      \n",
      "   TYR 275 A    12.11      10.00                      \n",
      "   TYR 280 A    14.55      10.00                      \n",
      "   TYR 299 A    14.29      10.00                      \n",
      "   TYR 335 A    10.09      10.00                      \n",
      "   TYR 360 A    10.04      10.00                      \n",
      "   LYS  21 A    11.63      10.50                      \n",
      "   LYS  25 A    11.09      10.50                      \n",
      "   LYS  35 A    11.28      10.50                      \n",
      "   LYS  37 A    13.37      10.50                      \n",
      "   LYS 108 A    10.20      10.50                      \n",
      "   LYS 111 A    10.66      10.50                      \n",
      "   LYS 120 A    10.80      10.50                      \n",
      "   LYS 132 A    10.46      10.50                      \n",
      "   LYS 164 A    10.39      10.50                      \n",
      "   LYS 178 A    10.56      10.50                      \n",
      "   LYS 179 A    10.81      10.50                      \n",
      "   LYS 191 A    10.47      10.50                      \n",
      "   LYS 193 A    10.48      10.50                      \n",
      "   LYS 201 A    10.15      10.50                      \n",
      "   LYS 208 A    11.45      10.50                      \n",
      "   LYS 232 A    10.41      10.50                      \n",
      "   LYS 269 A    10.30      10.50                      \n",
      "   LYS 279 A     9.58      10.50                      \n",
      "   LYS 293 A    10.97      10.50                      \n",
      "   LYS 300 A    11.52      10.50                      \n",
      "   LYS 302 A    12.40      10.50                      \n",
      "   LYS 332 A    10.67      10.50                      \n",
      "   LYS 354 A    10.57      10.50                      \n",
      "   ARG  10 A    13.55      12.50                      \n",
      "   ARG 124 A    12.59      12.50                      \n",
      "   ARG 134 A    13.76      12.50                      \n",
      "   ARG 149 A    12.62      12.50                      \n",
      "   ARG 209 A    12.83      12.50                      \n",
      "   ARG 250 A    12.37      12.50                      \n",
      "   ARG 268 A    12.32      12.50                      \n",
      "   ARG 273 A    12.69      12.50                      \n",
      "   ARG 285 A    13.75      12.50                      \n",
      "   ARG 326 A    13.60      12.50                      \n",
      "   ARG 341 A    14.03      12.50                      \n",
      "   N+    1 A     8.68       8.00                      \n",
      "\n",
      "INFO:PROPKA information:\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "Free energy of   folding (kcal/mol) as a function of pH (using neutral reference)\n",
      "  0.00     28.05\n",
      "  1.00     27.81\n",
      "  2.00     26.19\n",
      "  3.00     20.32\n",
      "  4.00     11.83\n",
      "  5.00     10.29\n",
      "  6.00     12.13\n",
      "  7.00     13.31\n",
      "  8.00     15.91\n",
      "  9.00     20.92\n",
      " 10.00     32.09\n",
      " 11.00     51.34\n",
      " 12.00     71.39\n",
      " 13.00     87.31\n",
      " 14.00     98.55\n",
      "\n",
      "The pH of optimum stability is  4.6 for which the free energy is    9.9 kcal/mol at 298K\n",
      "Could not determine pH values where the free energy is within 80 % of minimum\n",
      "Could not determine the pH-range where the free energy is negative\n",
      "\n",
      "\n",
      "Protein charge of folded and unfolded state as a function of pH\n",
      "    pH  unfolded  folded\n",
      "  0.00     42.00   41.95\n",
      "  0.10     41.99   41.94\n",
      "  0.20     41.99   41.92\n",
      "  0.30     41.99   41.90\n",
      "  0.40     41.99   41.88\n",
      "  0.50     41.99   41.85\n",
      "  0.60     41.98   41.81\n",
      "  0.70     41.98   41.76\n",
      "  0.80     41.97   41.70\n",
      "  0.90     41.97   41.63\n",
      "  1.00     41.96   41.54\n",
      "  1.10     41.95   41.44\n",
      "  1.20     41.93   41.31\n",
      "  1.30     41.92   41.16\n",
      "  1.40     41.89   40.98\n",
      "  1.50     41.87   40.78\n",
      "  1.60     41.83   40.54\n",
      "  1.70     41.79   40.26\n",
      "  1.80     41.74   39.95\n",
      "  1.90     41.67   39.59\n",
      "  2.00     41.59   39.19\n",
      "  2.10     41.48   38.75\n",
      "  2.20     41.35   38.25\n",
      "  2.30     41.19   37.70\n",
      "  2.40     40.99   37.10\n",
      "  2.50     40.75   36.44\n",
      "  2.60     40.44   35.72\n",
      "  2.70     40.08   34.94\n",
      "  2.80     39.63   34.10\n",
      "  2.90     39.09   33.19\n",
      "  3.00     38.44   32.23\n",
      "  3.10     37.68   31.20\n",
      "  3.20     36.79   30.11\n",
      "  3.30     35.75   28.97\n",
      "  3.40     34.56   27.77\n",
      "  3.50     33.23   26.53\n",
      "  3.60     31.74   25.25\n",
      "  3.70     30.11   23.95\n",
      "  3.80     28.36   22.63\n",
      "  3.90     26.50   21.29\n",
      "  4.00     24.55   19.97\n",
      "  4.10     22.54   18.65\n",
      "  4.20     20.50   17.34\n",
      "  4.30     18.46   16.06\n",
      "  4.40     16.44   14.80\n",
      "  4.50     14.47   13.57\n",
      "  4.60     12.59   12.37\n",
      "  4.70     10.83   11.19\n",
      "  4.80      9.20   10.06\n",
      "  4.90      7.72    8.95\n",
      "  5.00      6.39    7.89\n",
      "  5.10      5.22    6.87\n",
      "  5.20      4.19    5.90\n",
      "  5.30      3.29    4.97\n",
      "  5.40      2.50    4.10\n",
      "  5.50      1.81    3.27\n",
      "  5.60      1.19    2.50\n",
      "  5.70      0.62    1.77\n",
      "  5.80      0.10    1.08\n",
      "  5.90     -0.40    0.44\n",
      "  6.00     -0.88   -0.15\n",
      "  6.10     -1.36   -0.71\n",
      "  6.20     -1.85   -1.22\n",
      "  6.30     -2.33   -1.69\n",
      "  6.40     -2.82   -2.13\n",
      "  6.50     -3.30   -2.53\n",
      "  6.60     -3.77   -2.89\n",
      "  6.70     -4.22   -3.22\n",
      "  6.80     -4.65   -3.52\n",
      "  6.90     -5.05   -3.79\n",
      "  7.00     -5.42   -4.03\n",
      "  7.10     -5.76   -4.25\n",
      "  7.20     -6.07   -4.45\n",
      "  7.30     -6.36   -4.64\n",
      "  7.40     -6.62   -4.81\n",
      "  7.50     -6.87   -4.97\n",
      "  7.60     -7.12   -5.12\n",
      "  7.70     -7.36   -5.27\n",
      "  7.80     -7.62   -5.42\n",
      "  7.90     -7.88   -5.57\n",
      "  8.00     -8.17   -5.72\n",
      "  8.10     -8.48   -5.88\n",
      "  8.20     -8.82   -6.04\n",
      "  8.30     -9.20   -6.22\n",
      "  8.40     -9.63   -6.40\n",
      "  8.50    -10.11   -6.59\n",
      "  8.60    -10.65   -6.81\n",
      "  8.70    -11.25   -7.04\n",
      "  8.80    -11.91   -7.29\n",
      "  8.90    -12.63   -7.57\n",
      "  9.00    -13.41   -7.89\n",
      "  9.10    -14.26   -8.24\n",
      "  9.20    -15.17   -8.64\n",
      "  9.30    -16.14   -9.09\n",
      "  9.40    -17.18   -9.59\n",
      "  9.50    -18.30  -10.16\n",
      "  9.60    -19.51  -10.80\n",
      "  9.70    -20.81  -11.51\n",
      "  9.80    -22.22  -12.29\n",
      "  9.90    -23.72  -13.16\n",
      " 10.00    -25.32  -14.10\n",
      " 10.10    -27.01  -15.11\n",
      " 10.20    -28.77  -16.20\n",
      " 10.30    -30.57  -17.34\n",
      " 10.40    -32.39  -18.53\n",
      " 10.50    -34.20  -19.75\n",
      " 10.60    -35.95  -21.00\n",
      " 10.70    -37.62  -22.26\n",
      " 10.80    -39.18  -23.51\n",
      " 10.90    -40.61  -24.76\n",
      " 11.00    -41.91  -25.98\n",
      " 11.10    -43.08  -27.19\n",
      " 11.20    -44.11  -28.36\n",
      " 11.30    -45.04  -29.51\n",
      " 11.40    -45.86  -30.63\n",
      " 11.50    -46.61  -31.73\n",
      " 11.60    -47.30  -32.79\n",
      " 11.70    -47.95  -33.84\n",
      " 11.80    -48.58  -34.85\n",
      " 11.90    -49.21  -35.85\n",
      " 12.00    -49.84  -36.83\n",
      " 12.10    -50.49  -37.80\n",
      " 12.20    -51.16  -38.75\n",
      " 12.30    -51.85  -39.69\n",
      " 12.40    -52.54  -40.62\n",
      " 12.50    -53.24  -41.54\n",
      " 12.60    -53.92  -42.44\n",
      " 12.70    -54.58  -43.33\n",
      " 12.80    -55.20  -44.20\n",
      " 12.90    -55.76  -45.04\n",
      " 13.00    -56.27  -45.87\n",
      " 13.10    -56.73  -46.67\n",
      " 13.20    -57.12  -47.44\n",
      " 13.30    -57.45  -48.19\n",
      " 13.40    -57.74  -48.93\n",
      " 13.50    -57.97  -49.64\n",
      " 13.60    -58.17  -50.32\n",
      " 13.70    -58.33  -50.98\n",
      " 13.80    -58.46  -51.62\n",
      " 13.90    -58.57  -52.22\n",
      " 14.00    -58.65  -52.79\n",
      "The pI is  5.97 (folded) and  5.82 (unfolded)\n",
      "\n",
      "--------------------------------------------------------------------------------------------------------\n",
      "INFO:Applying pKa values at a pH of 8.20:\n",
      "INFO:Adding hydrogens to biomolecule.\n",
      "INFO:Debumping biomolecule (again).\n",
      "INFO:Optimizing hydrogen bonds\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "pH: 8.2\tname: AtAHL\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:Applying force field to biomolecule states.\n",
      "INFO:Regenerating headers.\n",
      "INFO:Regenerating PDB lines.\n",
      "WARNING:Ignoring 466 header lines in output.\n"
     ]
    }
   ],
   "source": [
    "# assign directory\n",
    "ph = [(7.0,'7_0'), (7.3,'7_3'), (7.6,'7_6'), (7.9,'7_9'), (8.2,'8_2')]\n",
    "directory = '/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/'\n",
    "\n",
    "for i in ph:\n",
    "    # Make dir\n",
    "    os.system(f\"mkdir {directory}pdb2pqr_pH{i[1]}\") \n",
    "    output_dir = f\"{directory}pdb2pqr_pH{i[1]}\"\n",
    "\n",
    "    # iterate over files in directory\n",
    "    for filename in os.listdir(directory):\n",
    "        basename = os.path.basename(filename)\n",
    "        f = os.path.join(directory, filename)\n",
    "        # checking if it is a file\n",
    "        if os.path.isfile(f):\n",
    "            if f.endswith(\".pdb\"):\n",
    "                shortname = basename.split('.')[0]\n",
    "\n",
    "                # Run pdb2pqr locally \n",
    "                os.system(f\"pdb2pqr --titration-state-method propka --with-ph {i[0]} --ff=AMBER {f} {output_dir}/{shortname}.pqr\")\n",
    "                print(f\"\\npH: {i[0]}\\tname: {shortname}\\n\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Run  (inputgen)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing: pH7.0\tAtAHL.pqr\n",
      "Processing: pH7.0\tAtSAL1_updated.pqr\n",
      "Processing: pH7.0\t1ka1_hal2.pqr\n",
      "Processing: pH7.0\t74987.pqr\n",
      "Processing: pH7.3\tAtAHL.pqr\n",
      "Processing: pH7.3\tAtSAL1_updated.pqr\n",
      "Processing: pH7.3\t1ka1_hal2.pqr\n",
      "Processing: pH7.3\t74987.pqr\n",
      "Processing: pH7.6\tAtAHL.pqr\n",
      "Processing: pH7.6\tAtSAL1_updated.pqr\n",
      "Processing: pH7.6\t1ka1_hal2.pqr\n",
      "Processing: pH7.6\t74987.pqr\n",
      "Processing: pH7.9\tAtAHL.pqr\n",
      "Processing: pH7.9\tAtSAL1_updated.pqr\n",
      "Processing: pH7.9\t1ka1_hal2.pqr\n",
      "Processing: pH7.9\t74987.pqr\n",
      "Processing: pH8.2\tAtAHL.pqr\n",
      "Processing: pH8.2\tAtSAL1_updated.pqr\n",
      "Processing: pH8.2\t1ka1_hal2.pqr\n",
      "Processing: pH8.2\t74987.pqr\n"
     ]
    }
   ],
   "source": [
    "# assign directory\n",
    "ph = [(7.0,'7_0'), (7.3,'7_3'), (7.6,'7_6'), (7.9,'7_9'), (8.2,'8_2')]\n",
    "\n",
    "for i in ph:\n",
    "    directory = f'/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/pdb2pqr_pH{i[1]}'\n",
    "    # iterate over files in directory\n",
    "    for filename in os.listdir(directory):\n",
    "        basename = os.path.basename(filename)\n",
    "        f = os.path.join(directory, filename)\n",
    "        # checking if it is a file\n",
    "        if os.path.isfile(f):\n",
    "            if f.endswith(\".pqr\"):\n",
    "                shortname = basename.split('-')[0]\n",
    "                print(f'Processing: pH{i[0]}\\t{basename}')\n",
    "                os.system(f\"inputgen {f}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Run apbs locally"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing: pH 7.0\tAtSAL1_updated.in\n",
      "\n",
      "\n",
      "----------------------------------------------------------------------\n",
      "    APBS -- Adaptive Poisson-Boltzmann Solver\n",
      "    Version 3.0\n",
      "    \n",
      "    Nathan A. Baker (nathan.baker@pnnl.gov)\n",
      "    Pacific Northwest National Laboratory\n",
      "    \n",
      "    Additional contributing authors listed in the code documentation.\n",
      "    \n",
      "    Copyright (c) 2010-2020 Battelle Memorial Institute. Developed at the Pacific\n",
      "    Northwest National Laboratory, operated by Battelle Memorial Institute, Pacific\n",
      "    Northwest Division for the U.S. Department of Energy.\n",
      "    \n",
      "    Portions Copyright (c) 2002-2010, Washington University in St. Louis.\n",
      "    Portions Copyright (c) 2002-2020, Nathan A. Baker.\n",
      "    Portions Copyright (c) 1999-2002, The Regents of the University of California.\n",
      "    Portions Copyright (c) 1995, Michael Holst.\n",
      "    All rights reserved.\n",
      "    \n",
      "    Redistribution and use in source and binary forms, with or without\n",
      "    modification, are permitted provided that the following conditions are met:\n",
      "    \n",
      "    * Redistributions of source code must retain the above copyright notice, this\n",
      "      list of conditions and the following disclaimer.\n",
      "    \n",
      "    * Redistributions in binary form must reproduce the above copyright notice,\n",
      "      this list of conditions and the following disclaimer in the documentation\n",
      "      and/or other materials provided with the distribution.\n",
      "    \n",
      "    * Neither the name of the developer nor the names of its contributors may be\n",
      "      used to endorse or promote products derived from this software without\n",
      "      specific prior written permission.\n",
      "    \n",
      "    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS \"AS IS\" AND\n",
      "    ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED\n",
      "    WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE\n",
      "    DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR\n",
      "    ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES\n",
      "    (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;\n",
      "    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND\n",
      "    ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT\n",
      "    (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS\n",
      "    SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.\n",
      "----------------------------------------------------------------------\n",
      "    APBS uses FETK (the Finite Element ToolKit) to solve the\n",
      "    Poisson-Boltzmann equation numerically.  FETK is a portable collection\n",
      "    of finite element modeling class libraries developed by the Michael Holst\n",
      "    research group and written in an object-oriented form of C.  FEtk is\n",
      "    designed to solve general coupled systems of nonlinear partial differential\n",
      "    equations using adaptive finite element methods, inexact Newton methods,\n",
      "    and algebraic multilevel methods.  More information about FEtk may be found\n",
      "    at <http://www.FEtk.ORG>.\n",
      "----------------------------------------------------------------------\n",
      "    APBS also uses Aqua to solve the Poisson-Boltzmann equation numerically.  \n",
      "    Aqua is a modified form of the Holst group PMG library <http://www.FEtk.ORG>\n",
      "    which has been modified by Patrice Koehl\n",
      "    <http://koehllab.genomecenter.ucdavis.edu/> for improved efficiency and\n",
      "    memory usage when solving the Poisson-Boltzmann equation.\n",
      "----------------------------------------------------------------------\n",
      "    Please cite your use of APBS as:\n",
      "\n",
      "    Baker NA, Sept D, Joseph S, Holst MJ, McCammon JA. Electrostatics of\n",
      "    nanosystems: application to microtubules and the ribosome. Proc.\n",
      "    Natl. Acad. Sci. USA 98, 10037-10041 2001.\n",
      "    \n",
      "\n",
      "This executable compiled on May 29 2020 at 16:07:17\n",
      "\n",
      "Parsing input file AtSAL1_updated.in...\n",
      "rank 0 size 1...\n",
      "Parsed input file.\n",
      "Got paths for 1 molecules\n",
      "Reading PQR-format atom data from AtSAL1_updated.pqr.\n",
      "  5263 atoms\n",
      "  Centered at (-3.168e+01, -3.844e+01, -7.218e+00)\n",
      "  Net charge -1.00e+01 e\n",
      "Preparing to run 4 PBE calculations.\n",
      "----------------------------------------\n",
      "CALCULATION #1: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  976.597 MB total, 976.597 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 161 x 161\n",
      "  Grid spacings: 0.692 x 0.565 x 0.571\n",
      "  Grid lengths: 110.799 x 90.410 x 91.309\n",
      "  Grid center: (-31.682, -38.444, -7.218)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Single Debye-Huckel sphere boundary conditions\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 78.540\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 2.166013354830E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "CALCULATION #2: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  976.597 MB total, 1945.352 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 161 x 161\n",
      "  Grid spacings: 0.532 x 0.457 x 0.461\n",
      "  Grid lengths: 85.176 x 73.182 x 73.711\n",
      "  Grid center: (-31.682, -38.444, -7.218)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Boundary conditions from focusing\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 78.540\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Potential to be written to pot.dx\n",
      "  Total electrostatic energy = 3.283713368468E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "  Writing potential to pot.dx\n",
      "----------------------------------------\n",
      "CALCULATION #3: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  970.057 MB total, 1945.352 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 161 x 161\n",
      "  Grid spacings: 0.692 x 0.565 x 0.571\n",
      "  Grid lengths: 110.799 x 90.410 x 91.309\n",
      "  Grid center: (-31.682, -38.444, -7.218)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Single Debye-Huckel sphere boundary conditions\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 2.000\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 2.267983170689E+05 kJ/mol\n",
      "  Calculating forces...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[focusFillBound()]: WARNING:\n",
      "    Unusually large potential values\n",
      "    detected on the focusing boundary!\n",
      "    Convergence not guaranteed for NPBE/NRPBE calculations!\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "----------------------------------------\n",
      "CALCULATION #4: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  970.057 MB total, 1945.352 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 161 x 161\n",
      "  Grid spacings: 0.532 x 0.457 x 0.461\n",
      "  Grid lengths: 85.176 x 73.182 x 73.711\n",
      "  Grid center: (-31.682, -38.444, -7.218)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Boundary conditions from focusing\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 2.000\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 3.384338925125E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "PRINT STATEMENTS\n",
      "\n",
      "print energy 2 - 1 end\n",
      "  Local net energy (PE 0) = 1.006255566567E+04 kJ/mol\n",
      "  Global net ELEC energy = 1.006255566567E+04 kJ/mol\n",
      "----------------------------------------\n",
      "CLEANING UP AND SHUTTING DOWN...\n",
      "Destroying force arrays.\n",
      "No energy arrays to destroy.\n",
      "Destroying multigrid structures.\n",
      "Destroying finite element structures.\n",
      "Destroying 1 molecules\n",
      "Final memory usage:  0.001 MB total, 1945.352 MB high water\n",
      "\n",
      "\n",
      "Thanks for using APBS!\n",
      "\n",
      "Processing: pH 7.0\tAtAHL.in\n",
      "\n",
      "\n",
      "----------------------------------------------------------------------\n",
      "    APBS -- Adaptive Poisson-Boltzmann Solver\n",
      "    Version 3.0\n",
      "    \n",
      "    Nathan A. Baker (nathan.baker@pnnl.gov)\n",
      "    Pacific Northwest National Laboratory\n",
      "    \n",
      "    Additional contributing authors listed in the code documentation.\n",
      "    \n",
      "    Copyright (c) 2010-2020 Battelle Memorial Institute. Developed at the Pacific\n",
      "    Northwest National Laboratory, operated by Battelle Memorial Institute, Pacific\n",
      "    Northwest Division for the U.S. Department of Energy.\n",
      "    \n",
      "    Portions Copyright (c) 2002-2010, Washington University in St. Louis.\n",
      "    Portions Copyright (c) 2002-2020, Nathan A. Baker.\n",
      "    Portions Copyright (c) 1999-2002, The Regents of the University of California.\n",
      "    Portions Copyright (c) 1995, Michael Holst.\n",
      "    All rights reserved.\n",
      "    \n",
      "    Redistribution and use in source and binary forms, with or without\n",
      "    modification, are permitted provided that the following conditions are met:\n",
      "    \n",
      "    * Redistributions of source code must retain the above copyright notice, this\n",
      "      list of conditions and the following disclaimer.\n",
      "    \n",
      "    * Redistributions in binary form must reproduce the above copyright notice,\n",
      "      this list of conditions and the following disclaimer in the documentation\n",
      "      and/or other materials provided with the distribution.\n",
      "    \n",
      "    * Neither the name of the developer nor the names of its contributors may be\n",
      "      used to endorse or promote products derived from this software without\n",
      "      specific prior written permission.\n",
      "    \n",
      "    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS \"AS IS\" AND\n",
      "    ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED\n",
      "    WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE\n",
      "    DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR\n",
      "    ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES\n",
      "    (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;\n",
      "    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND\n",
      "    ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT\n",
      "    (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS\n",
      "    SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.\n",
      "----------------------------------------------------------------------\n",
      "    APBS uses FETK (the Finite Element ToolKit) to solve the\n",
      "    Poisson-Boltzmann equation numerically.  FETK is a portable collection\n",
      "    of finite element modeling class libraries developed by the Michael Holst\n",
      "    research group and written in an object-oriented form of C.  FEtk is\n",
      "    designed to solve general coupled systems of nonlinear partial differential\n",
      "    equations using adaptive finite element methods, inexact Newton methods,\n",
      "    and algebraic multilevel methods.  More information about FEtk may be found\n",
      "    at <http://www.FEtk.ORG>.\n",
      "----------------------------------------------------------------------\n",
      "    APBS also uses Aqua to solve the Poisson-Boltzmann equation numerically.  \n",
      "    Aqua is a modified form of the Holst group PMG library <http://www.FEtk.ORG>\n",
      "    which has been modified by Patrice Koehl\n",
      "    <http://koehllab.genomecenter.ucdavis.edu/> for improved efficiency and\n",
      "    memory usage when solving the Poisson-Boltzmann equation.\n",
      "----------------------------------------------------------------------\n",
      "    Please cite your use of APBS as:\n",
      "\n",
      "    Baker NA, Sept D, Joseph S, Holst MJ, McCammon JA. Electrostatics of\n",
      "    nanosystems: application to microtubules and the ribosome. Proc.\n",
      "    Natl. Acad. Sci. USA 98, 10037-10041 2001.\n",
      "    \n",
      "\n",
      "This executable compiled on May 29 2020 at 16:07:17\n",
      "\n",
      "Parsing input file AtAHL.in...\n",
      "rank 0 size 1...\n",
      "Parsed input file.\n",
      "Got paths for 1 molecules\n",
      "Reading PQR-format atom data from AtAHL.pqr.\n",
      "  5476 atoms\n",
      "  Centered at (-3.306e+01, -4.212e+01, -8.250e+00)\n",
      "  Net charge -5.00e+00 e\n",
      "Preparing to run 4 PBE calculations.\n",
      "----------------------------------------\n",
      "CALCULATION #1: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  791.341 MB total, 791.341 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 129 x 161\n",
      "  Grid spacings: 0.704 x 0.689 x 0.585\n",
      "  Grid lengths: 112.579 x 88.177 x 93.611\n",
      "  Grid center: (-33.059, -42.116, -8.250)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Single Debye-Huckel sphere boundary conditions\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 78.540\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 1.835427995658E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "CALCULATION #2: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  791.341 MB total, 1574.583 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 129 x 161\n",
      "  Grid spacings: 0.539 x 0.561 x 0.469\n",
      "  Grid lengths: 86.223 x 71.869 x 75.065\n",
      "  Grid center: (-33.059, -42.116, -8.250)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Boundary conditions from focusing\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 78.540\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Potential to be written to pot.dx\n",
      "  Total electrostatic energy = 2.877710175835E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "  Writing potential to pot.dx\n",
      "----------------------------------------\n",
      "CALCULATION #3: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  784.584 MB total, 1574.583 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 129 x 161\n",
      "  Grid spacings: 0.704 x 0.689 x 0.585\n",
      "  Grid lengths: 112.579 x 88.177 x 93.611\n",
      "  Grid center: (-33.059, -42.116, -8.250)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Single Debye-Huckel sphere boundary conditions\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 2.000\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 1.910317696734E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "CALCULATION #4: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  784.584 MB total, 1574.583 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 129 x 161\n",
      "  Grid spacings: 0.539 x 0.561 x 0.469\n",
      "  Grid lengths: 86.223 x 71.869 x 75.065\n",
      "  Grid center: (-33.059, -42.116, -8.250)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Boundary conditions from focusing\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 2.000\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 2.951342259002E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "PRINT STATEMENTS\n",
      "\n",
      "print energy 2 - 1 end\n",
      "  Local net energy (PE 0) = 7.363208316716E+03 kJ/mol\n",
      "  Global net ELEC energy = 7.363208316716E+03 kJ/mol\n",
      "----------------------------------------\n",
      "CLEANING UP AND SHUTTING DOWN...\n",
      "Destroying force arrays.\n",
      "No energy arrays to destroy.\n",
      "Destroying multigrid structures.\n",
      "Destroying finite element structures.\n",
      "Destroying 1 molecules\n",
      "Final memory usage:  0.001 MB total, 1574.583 MB high water\n",
      "\n",
      "\n",
      "Thanks for using APBS!\n",
      "\n",
      "Processing: pH 7.0\t74987.in\n",
      "\n",
      "\n",
      "----------------------------------------------------------------------\n",
      "    APBS -- Adaptive Poisson-Boltzmann Solver\n",
      "    Version 3.0\n",
      "    \n",
      "    Nathan A. Baker (nathan.baker@pnnl.gov)\n",
      "    Pacific Northwest National Laboratory\n",
      "    \n",
      "    Additional contributing authors listed in the code documentation.\n",
      "    \n",
      "    Copyright (c) 2010-2020 Battelle Memorial Institute. Developed at the Pacific\n",
      "    Northwest National Laboratory, operated by Battelle Memorial Institute, Pacific\n",
      "    Northwest Division for the U.S. Department of Energy.\n",
      "    \n",
      "    Portions Copyright (c) 2002-2010, Washington University in St. Louis.\n",
      "    Portions Copyright (c) 2002-2020, Nathan A. Baker.\n",
      "    Portions Copyright (c) 1999-2002, The Regents of the University of California.\n",
      "    Portions Copyright (c) 1995, Michael Holst.\n",
      "    All rights reserved.\n",
      "    \n",
      "    Redistribution and use in source and binary forms, with or without\n",
      "    modification, are permitted provided that the following conditions are met:\n",
      "    \n",
      "    * Redistributions of source code must retain the above copyright notice, this\n",
      "      list of conditions and the following disclaimer.\n",
      "    \n",
      "    * Redistributions in binary form must reproduce the above copyright notice,\n",
      "      this list of conditions and the following disclaimer in the documentation\n",
      "      and/or other materials provided with the distribution.\n",
      "    \n",
      "    * Neither the name of the developer nor the names of its contributors may be\n",
      "      used to endorse or promote products derived from this software without\n",
      "      specific prior written permission.\n",
      "    \n",
      "    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS \"AS IS\" AND\n",
      "    ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED\n",
      "    WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE\n",
      "    DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR\n",
      "    ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES\n",
      "    (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;\n",
      "    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND\n",
      "    ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT\n",
      "    (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS\n",
      "    SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.\n",
      "----------------------------------------------------------------------\n",
      "    APBS uses FETK (the Finite Element ToolKit) to solve the\n",
      "    Poisson-Boltzmann equation numerically.  FETK is a portable collection\n",
      "    of finite element modeling class libraries developed by the Michael Holst\n",
      "    research group and written in an object-oriented form of C.  FEtk is\n",
      "    designed to solve general coupled systems of nonlinear partial differential\n",
      "    equations using adaptive finite element methods, inexact Newton methods,\n",
      "    and algebraic multilevel methods.  More information about FEtk may be found\n",
      "    at <http://www.FEtk.ORG>.\n",
      "----------------------------------------------------------------------\n",
      "    APBS also uses Aqua to solve the Poisson-Boltzmann equation numerically.  \n",
      "    Aqua is a modified form of the Holst group PMG library <http://www.FEtk.ORG>\n",
      "    which has been modified by Patrice Koehl\n",
      "    <http://koehllab.genomecenter.ucdavis.edu/> for improved efficiency and\n",
      "    memory usage when solving the Poisson-Boltzmann equation.\n",
      "----------------------------------------------------------------------\n",
      "    Please cite your use of APBS as:\n",
      "\n",
      "    Baker NA, Sept D, Joseph S, Holst MJ, McCammon JA. Electrostatics of\n",
      "    nanosystems: application to microtubules and the ribosome. Proc.\n",
      "    Natl. Acad. Sci. USA 98, 10037-10041 2001.\n",
      "    \n",
      "\n",
      "This executable compiled on May 29 2020 at 16:07:17\n",
      "\n",
      "Parsing input file 74987.in...\n",
      "rank 0 size 1...\n",
      "Parsed input file.\n",
      "Got paths for 1 molecules\n",
      "Reading PQR-format atom data from 74987.pqr.\n",
      "  5356 atoms\n",
      "  Centered at (-3.215e+01, -4.055e+01, -6.682e+00)\n",
      "  Net charge -1.20e+01 e\n",
      "Preparing to run 4 PBE calculations.\n",
      "----------------------------------------\n",
      "CALCULATION #1: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  938.276 MB total, 938.276 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 193 x 129 x 161\n",
      "  Grid spacings: 0.613 x 0.676 x 0.590\n",
      "  Grid lengths: 117.764 x 86.573 x 94.369\n",
      "  Grid center: (-32.145, -40.554, -6.682)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Single Debye-Huckel sphere boundary conditions\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 78.540\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 2.044159580805E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "CALCULATION #2: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  938.276 MB total, 1868.851 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 193 x 129 x 161\n",
      "  Grid spacings: 0.465 x 0.554 x 0.472\n",
      "  Grid lengths: 89.273 x 70.925 x 75.511\n",
      "  Grid center: (-32.145, -40.554, -6.682)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Boundary conditions from focusing\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 78.540\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Potential to be written to pot.dx\n",
      "  Total electrostatic energy = 3.137100422795E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "  Writing potential to pot.dx\n",
      "----------------------------------------\n",
      "CALCULATION #3: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  931.895 MB total, 1868.851 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 193 x 129 x 161\n",
      "  Grid spacings: 0.613 x 0.676 x 0.590\n",
      "  Grid lengths: 117.764 x 86.573 x 94.369\n",
      "  Grid center: (-32.145, -40.554, -6.682)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Single Debye-Huckel sphere boundary conditions\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 2.000\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 2.136271554391E+05 kJ/mol\n",
      "  Calculating forces...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[focusFillBound()]: WARNING:\n",
      "    Unusually large potential values\n",
      "    detected on the focusing boundary!\n",
      "    Convergence not guaranteed for NPBE/NRPBE calculations!\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "----------------------------------------\n",
      "CALCULATION #4: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  931.895 MB total, 1868.851 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 193 x 129 x 161\n",
      "  Grid spacings: 0.465 x 0.554 x 0.472\n",
      "  Grid lengths: 89.273 x 70.925 x 75.511\n",
      "  Grid center: (-32.145, -40.554, -6.682)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Boundary conditions from focusing\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 2.000\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 3.227705337148E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "PRINT STATEMENTS\n",
      "\n",
      "print energy 2 - 1 end\n",
      "  Local net energy (PE 0) = 9.060491435336E+03 kJ/mol\n",
      "  Global net ELEC energy = 9.060491435336E+03 kJ/mol\n",
      "----------------------------------------\n",
      "CLEANING UP AND SHUTTING DOWN...\n",
      "Destroying force arrays.\n",
      "No energy arrays to destroy.\n",
      "Destroying multigrid structures.\n",
      "Destroying finite element structures.\n",
      "Destroying 1 molecules\n",
      "Final memory usage:  0.001 MB total, 1868.851 MB high water\n",
      "\n",
      "\n",
      "Thanks for using APBS!\n",
      "\n",
      "Processing: pH 7.0\t1ka1_hal2.in\n",
      "\n",
      "\n",
      "----------------------------------------------------------------------\n",
      "    APBS -- Adaptive Poisson-Boltzmann Solver\n",
      "    Version 3.0\n",
      "    \n",
      "    Nathan A. Baker (nathan.baker@pnnl.gov)\n",
      "    Pacific Northwest National Laboratory\n",
      "    \n",
      "    Additional contributing authors listed in the code documentation.\n",
      "    \n",
      "    Copyright (c) 2010-2020 Battelle Memorial Institute. Developed at the Pacific\n",
      "    Northwest National Laboratory, operated by Battelle Memorial Institute, Pacific\n",
      "    Northwest Division for the U.S. Department of Energy.\n",
      "    \n",
      "    Portions Copyright (c) 2002-2010, Washington University in St. Louis.\n",
      "    Portions Copyright (c) 2002-2020, Nathan A. Baker.\n",
      "    Portions Copyright (c) 1999-2002, The Regents of the University of California.\n",
      "    Portions Copyright (c) 1995, Michael Holst.\n",
      "    All rights reserved.\n",
      "    \n",
      "    Redistribution and use in source and binary forms, with or without\n",
      "    modification, are permitted provided that the following conditions are met:\n",
      "    \n",
      "    * Redistributions of source code must retain the above copyright notice, this\n",
      "      list of conditions and the following disclaimer.\n",
      "    \n",
      "    * Redistributions in binary form must reproduce the above copyright notice,\n",
      "      this list of conditions and the following disclaimer in the documentation\n",
      "      and/or other materials provided with the distribution.\n",
      "    \n",
      "    * Neither the name of the developer nor the names of its contributors may be\n",
      "      used to endorse or promote products derived from this software without\n",
      "      specific prior written permission.\n",
      "    \n",
      "    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS \"AS IS\" AND\n",
      "    ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED\n",
      "    WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE\n",
      "    DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR\n",
      "    ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES\n",
      "    (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;\n",
      "    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND\n",
      "    ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT\n",
      "    (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS\n",
      "    SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.\n",
      "----------------------------------------------------------------------\n",
      "    APBS uses FETK (the Finite Element ToolKit) to solve the\n",
      "    Poisson-Boltzmann equation numerically.  FETK is a portable collection\n",
      "    of finite element modeling class libraries developed by the Michael Holst\n",
      "    research group and written in an object-oriented form of C.  FEtk is\n",
      "    designed to solve general coupled systems of nonlinear partial differential\n",
      "    equations using adaptive finite element methods, inexact Newton methods,\n",
      "    and algebraic multilevel methods.  More information about FEtk may be found\n",
      "    at <http://www.FEtk.ORG>.\n",
      "----------------------------------------------------------------------\n",
      "    APBS also uses Aqua to solve the Poisson-Boltzmann equation numerically.  \n",
      "    Aqua is a modified form of the Holst group PMG library <http://www.FEtk.ORG>\n",
      "    which has been modified by Patrice Koehl\n",
      "    <http://koehllab.genomecenter.ucdavis.edu/> for improved efficiency and\n",
      "    memory usage when solving the Poisson-Boltzmann equation.\n",
      "----------------------------------------------------------------------\n",
      "    Please cite your use of APBS as:\n",
      "\n",
      "    Baker NA, Sept D, Joseph S, Holst MJ, McCammon JA. Electrostatics of\n",
      "    nanosystems: application to microtubules and the ribosome. Proc.\n",
      "    Natl. Acad. Sci. USA 98, 10037-10041 2001.\n",
      "    \n",
      "\n",
      "This executable compiled on May 29 2020 at 16:07:17\n",
      "\n",
      "Parsing input file 1ka1_hal2.in...\n",
      "rank 0 size 1...\n",
      "Parsed input file.\n",
      "Got paths for 1 molecules\n",
      "Reading PQR-format atom data from 1ka1_hal2.pqr.\n",
      "  5457 atoms\n",
      "  Centered at (-3.270e+01, -4.262e+01, -4.458e+00)\n",
      "  Net charge -5.00e+00 e\n",
      "Preparing to run 4 PBE calculations.\n",
      "----------------------------------------\n",
      "CALCULATION #1: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  795.568 MB total, 795.568 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 161 x 129\n",
      "  Grid spacings: 0.687 x 0.572 x 0.663\n",
      "  Grid lengths: 109.929 x 91.584 x 84.857\n",
      "  Grid center: (-32.703, -42.625, -4.458)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Single Debye-Huckel sphere boundary conditions\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 78.540\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 2.006514037351E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "CALCULATION #2: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  795.568 MB total, 1583.946 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 161 x 129\n",
      "  Grid spacings: 0.529 x 0.462 x 0.546\n",
      "  Grid lengths: 84.664 x 73.873 x 69.916\n",
      "  Grid center: (-32.703, -42.625, -4.458)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Boundary conditions from focusing\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 78.540\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Potential to be written to pot.dx\n",
      "  Total electrostatic energy = 3.061855552354E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "  Writing potential to pot.dx\n",
      "----------------------------------------\n",
      "CALCULATION #3: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  789.716 MB total, 1583.946 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 161 x 129\n",
      "  Grid spacings: 0.687 x 0.572 x 0.663\n",
      "  Grid lengths: 109.929 x 91.584 x 84.857\n",
      "  Grid center: (-32.703, -42.625, -4.458)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Single Debye-Huckel sphere boundary conditions\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 2.000\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 2.080654185153E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "CALCULATION #4: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  789.716 MB total, 1583.946 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 161 x 129\n",
      "  Grid spacings: 0.529 x 0.462 x 0.546\n",
      "  Grid lengths: 84.664 x 73.873 x 69.916\n",
      "  Grid center: (-32.703, -42.625, -4.458)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Boundary conditions from focusing\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 2.000\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 3.134712849475E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "PRINT STATEMENTS\n",
      "\n",
      "print energy 2 - 1 end\n",
      "  Local net energy (PE 0) = 7.285729712055E+03 kJ/mol\n",
      "  Global net ELEC energy = 7.285729712055E+03 kJ/mol\n",
      "----------------------------------------\n",
      "CLEANING UP AND SHUTTING DOWN...\n",
      "Destroying force arrays.\n",
      "No energy arrays to destroy.\n",
      "Destroying multigrid structures.\n",
      "Destroying finite element structures.\n",
      "Destroying 1 molecules\n",
      "Final memory usage:  0.001 MB total, 1583.946 MB high water\n",
      "\n",
      "\n",
      "Thanks for using APBS!\n",
      "\n",
      "Processing: pH 7.3\tAtSAL1_updated.in\n",
      "\n",
      "\n",
      "----------------------------------------------------------------------\n",
      "    APBS -- Adaptive Poisson-Boltzmann Solver\n",
      "    Version 3.0\n",
      "    \n",
      "    Nathan A. Baker (nathan.baker@pnnl.gov)\n",
      "    Pacific Northwest National Laboratory\n",
      "    \n",
      "    Additional contributing authors listed in the code documentation.\n",
      "    \n",
      "    Copyright (c) 2010-2020 Battelle Memorial Institute. Developed at the Pacific\n",
      "    Northwest National Laboratory, operated by Battelle Memorial Institute, Pacific\n",
      "    Northwest Division for the U.S. Department of Energy.\n",
      "    \n",
      "    Portions Copyright (c) 2002-2010, Washington University in St. Louis.\n",
      "    Portions Copyright (c) 2002-2020, Nathan A. Baker.\n",
      "    Portions Copyright (c) 1999-2002, The Regents of the University of California.\n",
      "    Portions Copyright (c) 1995, Michael Holst.\n",
      "    All rights reserved.\n",
      "    \n",
      "    Redistribution and use in source and binary forms, with or without\n",
      "    modification, are permitted provided that the following conditions are met:\n",
      "    \n",
      "    * Redistributions of source code must retain the above copyright notice, this\n",
      "      list of conditions and the following disclaimer.\n",
      "    \n",
      "    * Redistributions in binary form must reproduce the above copyright notice,\n",
      "      this list of conditions and the following disclaimer in the documentation\n",
      "      and/or other materials provided with the distribution.\n",
      "    \n",
      "    * Neither the name of the developer nor the names of its contributors may be\n",
      "      used to endorse or promote products derived from this software without\n",
      "      specific prior written permission.\n",
      "    \n",
      "    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS \"AS IS\" AND\n",
      "    ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED\n",
      "    WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE\n",
      "    DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR\n",
      "    ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES\n",
      "    (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;\n",
      "    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND\n",
      "    ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT\n",
      "    (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS\n",
      "    SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.\n",
      "----------------------------------------------------------------------\n",
      "    APBS uses FETK (the Finite Element ToolKit) to solve the\n",
      "    Poisson-Boltzmann equation numerically.  FETK is a portable collection\n",
      "    of finite element modeling class libraries developed by the Michael Holst\n",
      "    research group and written in an object-oriented form of C.  FEtk is\n",
      "    designed to solve general coupled systems of nonlinear partial differential\n",
      "    equations using adaptive finite element methods, inexact Newton methods,\n",
      "    and algebraic multilevel methods.  More information about FEtk may be found\n",
      "    at <http://www.FEtk.ORG>.\n",
      "----------------------------------------------------------------------\n",
      "    APBS also uses Aqua to solve the Poisson-Boltzmann equation numerically.  \n",
      "    Aqua is a modified form of the Holst group PMG library <http://www.FEtk.ORG>\n",
      "    which has been modified by Patrice Koehl\n",
      "    <http://koehllab.genomecenter.ucdavis.edu/> for improved efficiency and\n",
      "    memory usage when solving the Poisson-Boltzmann equation.\n",
      "----------------------------------------------------------------------\n",
      "    Please cite your use of APBS as:\n",
      "\n",
      "    Baker NA, Sept D, Joseph S, Holst MJ, McCammon JA. Electrostatics of\n",
      "    nanosystems: application to microtubules and the ribosome. Proc.\n",
      "    Natl. Acad. Sci. USA 98, 10037-10041 2001.\n",
      "    \n",
      "\n",
      "This executable compiled on May 29 2020 at 16:07:17\n",
      "\n",
      "Parsing input file AtSAL1_updated.in...\n",
      "rank 0 size 1...\n",
      "Parsed input file.\n",
      "Got paths for 1 molecules\n",
      "Reading PQR-format atom data from AtSAL1_updated.pqr.\n",
      "  5261 atoms\n",
      "  Centered at (-3.168e+01, -3.844e+01, -7.218e+00)\n",
      "  Net charge -1.20e+01 e\n",
      "Preparing to run 4 PBE calculations.\n",
      "----------------------------------------\n",
      "CALCULATION #1: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  976.593 MB total, 976.593 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 161 x 161\n",
      "  Grid spacings: 0.692 x 0.565 x 0.571\n",
      "  Grid lengths: 110.799 x 90.410 x 91.309\n",
      "  Grid center: (-31.682, -38.444, -7.218)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Single Debye-Huckel sphere boundary conditions\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 78.540\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 2.172967233269E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "CALCULATION #2: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  976.593 MB total, 1945.342 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 161 x 161\n",
      "  Grid spacings: 0.532 x 0.457 x 0.461\n",
      "  Grid lengths: 85.176 x 73.182 x 73.711\n",
      "  Grid center: (-31.682, -38.444, -7.218)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Boundary conditions from focusing\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 78.540\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Potential to be written to pot.dx\n",
      "  Total electrostatic energy = 3.292470291842E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "  Writing potential to pot.dx\n",
      "----------------------------------------\n",
      "CALCULATION #3: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  970.052 MB total, 1945.342 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 161 x 161\n",
      "  Grid spacings: 0.692 x 0.565 x 0.571\n",
      "  Grid lengths: 110.799 x 90.410 x 91.309\n",
      "  Grid center: (-31.682, -38.444, -7.218)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Single Debye-Huckel sphere boundary conditions\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 2.000\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 2.283434323883E+05 kJ/mol\n",
      "  Calculating forces...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[focusFillBound()]: WARNING:\n",
      "    Unusually large potential values\n",
      "    detected on the focusing boundary!\n",
      "    Convergence not guaranteed for NPBE/NRPBE calculations!\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "----------------------------------------\n",
      "CALCULATION #4: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  970.052 MB total, 1945.342 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 161 x 161\n",
      "  Grid spacings: 0.532 x 0.457 x 0.461\n",
      "  Grid lengths: 85.176 x 73.182 x 73.711\n",
      "  Grid center: (-31.682, -38.444, -7.218)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Boundary conditions from focusing\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 2.000\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 3.401581366045E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "PRINT STATEMENTS\n",
      "\n",
      "print energy 2 - 1 end\n",
      "  Local net energy (PE 0) = 1.091110742026E+04 kJ/mol\n",
      "  Global net ELEC energy = 1.091110742026E+04 kJ/mol\n",
      "----------------------------------------\n",
      "CLEANING UP AND SHUTTING DOWN...\n",
      "Destroying force arrays.\n",
      "No energy arrays to destroy.\n",
      "Destroying multigrid structures.\n",
      "Destroying finite element structures.\n",
      "Destroying 1 molecules\n",
      "Final memory usage:  0.001 MB total, 1945.342 MB high water\n",
      "\n",
      "\n",
      "Thanks for using APBS!\n",
      "\n",
      "Processing: pH 7.3\tAtAHL.in\n",
      "\n",
      "\n",
      "----------------------------------------------------------------------\n",
      "    APBS -- Adaptive Poisson-Boltzmann Solver\n",
      "    Version 3.0\n",
      "    \n",
      "    Nathan A. Baker (nathan.baker@pnnl.gov)\n",
      "    Pacific Northwest National Laboratory\n",
      "    \n",
      "    Additional contributing authors listed in the code documentation.\n",
      "    \n",
      "    Copyright (c) 2010-2020 Battelle Memorial Institute. Developed at the Pacific\n",
      "    Northwest National Laboratory, operated by Battelle Memorial Institute, Pacific\n",
      "    Northwest Division for the U.S. Department of Energy.\n",
      "    \n",
      "    Portions Copyright (c) 2002-2010, Washington University in St. Louis.\n",
      "    Portions Copyright (c) 2002-2020, Nathan A. Baker.\n",
      "    Portions Copyright (c) 1999-2002, The Regents of the University of California.\n",
      "    Portions Copyright (c) 1995, Michael Holst.\n",
      "    All rights reserved.\n",
      "    \n",
      "    Redistribution and use in source and binary forms, with or without\n",
      "    modification, are permitted provided that the following conditions are met:\n",
      "    \n",
      "    * Redistributions of source code must retain the above copyright notice, this\n",
      "      list of conditions and the following disclaimer.\n",
      "    \n",
      "    * Redistributions in binary form must reproduce the above copyright notice,\n",
      "      this list of conditions and the following disclaimer in the documentation\n",
      "      and/or other materials provided with the distribution.\n",
      "    \n",
      "    * Neither the name of the developer nor the names of its contributors may be\n",
      "      used to endorse or promote products derived from this software without\n",
      "      specific prior written permission.\n",
      "    \n",
      "    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS \"AS IS\" AND\n",
      "    ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED\n",
      "    WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE\n",
      "    DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR\n",
      "    ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES\n",
      "    (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;\n",
      "    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND\n",
      "    ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT\n",
      "    (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS\n",
      "    SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.\n",
      "----------------------------------------------------------------------\n",
      "    APBS uses FETK (the Finite Element ToolKit) to solve the\n",
      "    Poisson-Boltzmann equation numerically.  FETK is a portable collection\n",
      "    of finite element modeling class libraries developed by the Michael Holst\n",
      "    research group and written in an object-oriented form of C.  FEtk is\n",
      "    designed to solve general coupled systems of nonlinear partial differential\n",
      "    equations using adaptive finite element methods, inexact Newton methods,\n",
      "    and algebraic multilevel methods.  More information about FEtk may be found\n",
      "    at <http://www.FEtk.ORG>.\n",
      "----------------------------------------------------------------------\n",
      "    APBS also uses Aqua to solve the Poisson-Boltzmann equation numerically.  \n",
      "    Aqua is a modified form of the Holst group PMG library <http://www.FEtk.ORG>\n",
      "    which has been modified by Patrice Koehl\n",
      "    <http://koehllab.genomecenter.ucdavis.edu/> for improved efficiency and\n",
      "    memory usage when solving the Poisson-Boltzmann equation.\n",
      "----------------------------------------------------------------------\n",
      "    Please cite your use of APBS as:\n",
      "\n",
      "    Baker NA, Sept D, Joseph S, Holst MJ, McCammon JA. Electrostatics of\n",
      "    nanosystems: application to microtubules and the ribosome. Proc.\n",
      "    Natl. Acad. Sci. USA 98, 10037-10041 2001.\n",
      "    \n",
      "\n",
      "This executable compiled on May 29 2020 at 16:07:17\n",
      "\n",
      "Parsing input file AtAHL.in...\n",
      "rank 0 size 1...\n",
      "Parsed input file.\n",
      "Got paths for 1 molecules\n",
      "Reading PQR-format atom data from AtAHL.pqr.\n",
      "  5476 atoms\n",
      "  Centered at (-3.306e+01, -4.212e+01, -8.250e+00)\n",
      "  Net charge -5.00e+00 e\n",
      "Preparing to run 4 PBE calculations.\n",
      "----------------------------------------\n",
      "CALCULATION #1: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  791.341 MB total, 791.341 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 129 x 161\n",
      "  Grid spacings: 0.704 x 0.689 x 0.585\n",
      "  Grid lengths: 112.579 x 88.177 x 93.611\n",
      "  Grid center: (-33.059, -42.116, -8.250)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Single Debye-Huckel sphere boundary conditions\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 78.540\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 1.835427995658E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "CALCULATION #2: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  791.341 MB total, 1574.583 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 129 x 161\n",
      "  Grid spacings: 0.539 x 0.561 x 0.469\n",
      "  Grid lengths: 86.223 x 71.869 x 75.065\n",
      "  Grid center: (-33.059, -42.116, -8.250)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Boundary conditions from focusing\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 78.540\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Potential to be written to pot.dx\n",
      "  Total electrostatic energy = 2.877710175835E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "  Writing potential to pot.dx\n",
      "----------------------------------------\n",
      "CALCULATION #3: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  784.584 MB total, 1574.583 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 129 x 161\n",
      "  Grid spacings: 0.704 x 0.689 x 0.585\n",
      "  Grid lengths: 112.579 x 88.177 x 93.611\n",
      "  Grid center: (-33.059, -42.116, -8.250)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Single Debye-Huckel sphere boundary conditions\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 2.000\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 1.910317696734E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "CALCULATION #4: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  784.584 MB total, 1574.583 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 129 x 161\n",
      "  Grid spacings: 0.539 x 0.561 x 0.469\n",
      "  Grid lengths: 86.223 x 71.869 x 75.065\n",
      "  Grid center: (-33.059, -42.116, -8.250)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Boundary conditions from focusing\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 2.000\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 2.951342259002E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "PRINT STATEMENTS\n",
      "\n",
      "print energy 2 - 1 end\n",
      "  Local net energy (PE 0) = 7.363208316716E+03 kJ/mol\n",
      "  Global net ELEC energy = 7.363208316716E+03 kJ/mol\n",
      "----------------------------------------\n",
      "CLEANING UP AND SHUTTING DOWN...\n",
      "Destroying force arrays.\n",
      "No energy arrays to destroy.\n",
      "Destroying multigrid structures.\n",
      "Destroying finite element structures.\n",
      "Destroying 1 molecules\n",
      "Final memory usage:  0.001 MB total, 1574.583 MB high water\n",
      "\n",
      "\n",
      "Thanks for using APBS!\n",
      "\n",
      "Processing: pH 7.3\t74987.in\n",
      "\n",
      "\n",
      "----------------------------------------------------------------------\n",
      "    APBS -- Adaptive Poisson-Boltzmann Solver\n",
      "    Version 3.0\n",
      "    \n",
      "    Nathan A. Baker (nathan.baker@pnnl.gov)\n",
      "    Pacific Northwest National Laboratory\n",
      "    \n",
      "    Additional contributing authors listed in the code documentation.\n",
      "    \n",
      "    Copyright (c) 2010-2020 Battelle Memorial Institute. Developed at the Pacific\n",
      "    Northwest National Laboratory, operated by Battelle Memorial Institute, Pacific\n",
      "    Northwest Division for the U.S. Department of Energy.\n",
      "    \n",
      "    Portions Copyright (c) 2002-2010, Washington University in St. Louis.\n",
      "    Portions Copyright (c) 2002-2020, Nathan A. Baker.\n",
      "    Portions Copyright (c) 1999-2002, The Regents of the University of California.\n",
      "    Portions Copyright (c) 1995, Michael Holst.\n",
      "    All rights reserved.\n",
      "    \n",
      "    Redistribution and use in source and binary forms, with or without\n",
      "    modification, are permitted provided that the following conditions are met:\n",
      "    \n",
      "    * Redistributions of source code must retain the above copyright notice, this\n",
      "      list of conditions and the following disclaimer.\n",
      "    \n",
      "    * Redistributions in binary form must reproduce the above copyright notice,\n",
      "      this list of conditions and the following disclaimer in the documentation\n",
      "      and/or other materials provided with the distribution.\n",
      "    \n",
      "    * Neither the name of the developer nor the names of its contributors may be\n",
      "      used to endorse or promote products derived from this software without\n",
      "      specific prior written permission.\n",
      "    \n",
      "    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS \"AS IS\" AND\n",
      "    ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED\n",
      "    WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE\n",
      "    DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR\n",
      "    ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES\n",
      "    (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;\n",
      "    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND\n",
      "    ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT\n",
      "    (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS\n",
      "    SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.\n",
      "----------------------------------------------------------------------\n",
      "    APBS uses FETK (the Finite Element ToolKit) to solve the\n",
      "    Poisson-Boltzmann equation numerically.  FETK is a portable collection\n",
      "    of finite element modeling class libraries developed by the Michael Holst\n",
      "    research group and written in an object-oriented form of C.  FEtk is\n",
      "    designed to solve general coupled systems of nonlinear partial differential\n",
      "    equations using adaptive finite element methods, inexact Newton methods,\n",
      "    and algebraic multilevel methods.  More information about FEtk may be found\n",
      "    at <http://www.FEtk.ORG>.\n",
      "----------------------------------------------------------------------\n",
      "    APBS also uses Aqua to solve the Poisson-Boltzmann equation numerically.  \n",
      "    Aqua is a modified form of the Holst group PMG library <http://www.FEtk.ORG>\n",
      "    which has been modified by Patrice Koehl\n",
      "    <http://koehllab.genomecenter.ucdavis.edu/> for improved efficiency and\n",
      "    memory usage when solving the Poisson-Boltzmann equation.\n",
      "----------------------------------------------------------------------\n",
      "    Please cite your use of APBS as:\n",
      "\n",
      "    Baker NA, Sept D, Joseph S, Holst MJ, McCammon JA. Electrostatics of\n",
      "    nanosystems: application to microtubules and the ribosome. Proc.\n",
      "    Natl. Acad. Sci. USA 98, 10037-10041 2001.\n",
      "    \n",
      "\n",
      "This executable compiled on May 29 2020 at 16:07:17\n",
      "\n",
      "Parsing input file 74987.in...\n",
      "rank 0 size 1...\n",
      "Parsed input file.\n",
      "Got paths for 1 molecules\n",
      "Reading PQR-format atom data from 74987.pqr.\n",
      "  5356 atoms\n",
      "  Centered at (-3.215e+01, -4.055e+01, -6.682e+00)\n",
      "  Net charge -1.20e+01 e\n",
      "Preparing to run 4 PBE calculations.\n",
      "----------------------------------------\n",
      "CALCULATION #1: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  938.276 MB total, 938.276 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 193 x 129 x 161\n",
      "  Grid spacings: 0.613 x 0.676 x 0.590\n",
      "  Grid lengths: 117.764 x 86.573 x 94.369\n",
      "  Grid center: (-32.145, -40.554, -6.682)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Single Debye-Huckel sphere boundary conditions\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 78.540\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 2.044159580805E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "CALCULATION #2: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  938.276 MB total, 1868.851 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 193 x 129 x 161\n",
      "  Grid spacings: 0.465 x 0.554 x 0.472\n",
      "  Grid lengths: 89.273 x 70.925 x 75.511\n",
      "  Grid center: (-32.145, -40.554, -6.682)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Boundary conditions from focusing\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 78.540\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Potential to be written to pot.dx\n",
      "  Total electrostatic energy = 3.137100422795E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "  Writing potential to pot.dx\n",
      "----------------------------------------\n",
      "CALCULATION #3: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  931.895 MB total, 1868.851 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 193 x 129 x 161\n",
      "  Grid spacings: 0.613 x 0.676 x 0.590\n",
      "  Grid lengths: 117.764 x 86.573 x 94.369\n",
      "  Grid center: (-32.145, -40.554, -6.682)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Single Debye-Huckel sphere boundary conditions\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 2.000\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 2.136271554391E+05 kJ/mol\n",
      "  Calculating forces...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[focusFillBound()]: WARNING:\n",
      "    Unusually large potential values\n",
      "    detected on the focusing boundary!\n",
      "    Convergence not guaranteed for NPBE/NRPBE calculations!\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "----------------------------------------\n",
      "CALCULATION #4: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  931.895 MB total, 1868.851 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 193 x 129 x 161\n",
      "  Grid spacings: 0.465 x 0.554 x 0.472\n",
      "  Grid lengths: 89.273 x 70.925 x 75.511\n",
      "  Grid center: (-32.145, -40.554, -6.682)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Boundary conditions from focusing\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 2.000\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 3.227705337148E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "PRINT STATEMENTS\n",
      "\n",
      "print energy 2 - 1 end\n",
      "  Local net energy (PE 0) = 9.060491435336E+03 kJ/mol\n",
      "  Global net ELEC energy = 9.060491435336E+03 kJ/mol\n",
      "----------------------------------------\n",
      "CLEANING UP AND SHUTTING DOWN...\n",
      "Destroying force arrays.\n",
      "No energy arrays to destroy.\n",
      "Destroying multigrid structures.\n",
      "Destroying finite element structures.\n",
      "Destroying 1 molecules\n",
      "Final memory usage:  0.001 MB total, 1868.851 MB high water\n",
      "\n",
      "\n",
      "Thanks for using APBS!\n",
      "\n",
      "Processing: pH 7.3\t1ka1_hal2.in\n",
      "\n",
      "\n",
      "----------------------------------------------------------------------\n",
      "    APBS -- Adaptive Poisson-Boltzmann Solver\n",
      "    Version 3.0\n",
      "    \n",
      "    Nathan A. Baker (nathan.baker@pnnl.gov)\n",
      "    Pacific Northwest National Laboratory\n",
      "    \n",
      "    Additional contributing authors listed in the code documentation.\n",
      "    \n",
      "    Copyright (c) 2010-2020 Battelle Memorial Institute. Developed at the Pacific\n",
      "    Northwest National Laboratory, operated by Battelle Memorial Institute, Pacific\n",
      "    Northwest Division for the U.S. Department of Energy.\n",
      "    \n",
      "    Portions Copyright (c) 2002-2010, Washington University in St. Louis.\n",
      "    Portions Copyright (c) 2002-2020, Nathan A. Baker.\n",
      "    Portions Copyright (c) 1999-2002, The Regents of the University of California.\n",
      "    Portions Copyright (c) 1995, Michael Holst.\n",
      "    All rights reserved.\n",
      "    \n",
      "    Redistribution and use in source and binary forms, with or without\n",
      "    modification, are permitted provided that the following conditions are met:\n",
      "    \n",
      "    * Redistributions of source code must retain the above copyright notice, this\n",
      "      list of conditions and the following disclaimer.\n",
      "    \n",
      "    * Redistributions in binary form must reproduce the above copyright notice,\n",
      "      this list of conditions and the following disclaimer in the documentation\n",
      "      and/or other materials provided with the distribution.\n",
      "    \n",
      "    * Neither the name of the developer nor the names of its contributors may be\n",
      "      used to endorse or promote products derived from this software without\n",
      "      specific prior written permission.\n",
      "    \n",
      "    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS \"AS IS\" AND\n",
      "    ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED\n",
      "    WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE\n",
      "    DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR\n",
      "    ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES\n",
      "    (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;\n",
      "    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND\n",
      "    ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT\n",
      "    (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS\n",
      "    SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.\n",
      "----------------------------------------------------------------------\n",
      "    APBS uses FETK (the Finite Element ToolKit) to solve the\n",
      "    Poisson-Boltzmann equation numerically.  FETK is a portable collection\n",
      "    of finite element modeling class libraries developed by the Michael Holst\n",
      "    research group and written in an object-oriented form of C.  FEtk is\n",
      "    designed to solve general coupled systems of nonlinear partial differential\n",
      "    equations using adaptive finite element methods, inexact Newton methods,\n",
      "    and algebraic multilevel methods.  More information about FEtk may be found\n",
      "    at <http://www.FEtk.ORG>.\n",
      "----------------------------------------------------------------------\n",
      "    APBS also uses Aqua to solve the Poisson-Boltzmann equation numerically.  \n",
      "    Aqua is a modified form of the Holst group PMG library <http://www.FEtk.ORG>\n",
      "    which has been modified by Patrice Koehl\n",
      "    <http://koehllab.genomecenter.ucdavis.edu/> for improved efficiency and\n",
      "    memory usage when solving the Poisson-Boltzmann equation.\n",
      "----------------------------------------------------------------------\n",
      "    Please cite your use of APBS as:\n",
      "\n",
      "    Baker NA, Sept D, Joseph S, Holst MJ, McCammon JA. Electrostatics of\n",
      "    nanosystems: application to microtubules and the ribosome. Proc.\n",
      "    Natl. Acad. Sci. USA 98, 10037-10041 2001.\n",
      "    \n",
      "\n",
      "This executable compiled on May 29 2020 at 16:07:17\n",
      "\n",
      "Parsing input file 1ka1_hal2.in...\n",
      "rank 0 size 1...\n",
      "Parsed input file.\n",
      "Got paths for 1 molecules\n",
      "Reading PQR-format atom data from 1ka1_hal2.pqr.\n",
      "  5457 atoms\n",
      "  Centered at (-3.270e+01, -4.262e+01, -4.458e+00)\n",
      "  Net charge -5.00e+00 e\n",
      "Preparing to run 4 PBE calculations.\n",
      "----------------------------------------\n",
      "CALCULATION #1: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  795.568 MB total, 795.568 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 161 x 129\n",
      "  Grid spacings: 0.687 x 0.572 x 0.663\n",
      "  Grid lengths: 109.929 x 91.584 x 84.857\n",
      "  Grid center: (-32.703, -42.625, -4.458)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Single Debye-Huckel sphere boundary conditions\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 78.540\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 2.006514037351E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "CALCULATION #2: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  795.568 MB total, 1583.946 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 161 x 129\n",
      "  Grid spacings: 0.529 x 0.462 x 0.546\n",
      "  Grid lengths: 84.664 x 73.873 x 69.916\n",
      "  Grid center: (-32.703, -42.625, -4.458)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Boundary conditions from focusing\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 78.540\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Potential to be written to pot.dx\n",
      "  Total electrostatic energy = 3.061855552354E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "  Writing potential to pot.dx\n",
      "----------------------------------------\n",
      "CALCULATION #3: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  789.716 MB total, 1583.946 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 161 x 129\n",
      "  Grid spacings: 0.687 x 0.572 x 0.663\n",
      "  Grid lengths: 109.929 x 91.584 x 84.857\n",
      "  Grid center: (-32.703, -42.625, -4.458)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Single Debye-Huckel sphere boundary conditions\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 2.000\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 2.080654185153E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "CALCULATION #4: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  789.716 MB total, 1583.946 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 161 x 129\n",
      "  Grid spacings: 0.529 x 0.462 x 0.546\n",
      "  Grid lengths: 84.664 x 73.873 x 69.916\n",
      "  Grid center: (-32.703, -42.625, -4.458)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Boundary conditions from focusing\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 2.000\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 3.134712849475E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "PRINT STATEMENTS\n",
      "\n",
      "print energy 2 - 1 end\n",
      "  Local net energy (PE 0) = 7.285729712055E+03 kJ/mol\n",
      "  Global net ELEC energy = 7.285729712055E+03 kJ/mol\n",
      "----------------------------------------\n",
      "CLEANING UP AND SHUTTING DOWN...\n",
      "Destroying force arrays.\n",
      "No energy arrays to destroy.\n",
      "Destroying multigrid structures.\n",
      "Destroying finite element structures.\n",
      "Destroying 1 molecules\n",
      "Final memory usage:  0.001 MB total, 1583.946 MB high water\n",
      "\n",
      "\n",
      "Thanks for using APBS!\n",
      "\n",
      "Processing: pH 7.6\tAtSAL1_updated.in\n",
      "\n",
      "\n",
      "----------------------------------------------------------------------\n",
      "    APBS -- Adaptive Poisson-Boltzmann Solver\n",
      "    Version 3.0\n",
      "    \n",
      "    Nathan A. Baker (nathan.baker@pnnl.gov)\n",
      "    Pacific Northwest National Laboratory\n",
      "    \n",
      "    Additional contributing authors listed in the code documentation.\n",
      "    \n",
      "    Copyright (c) 2010-2020 Battelle Memorial Institute. Developed at the Pacific\n",
      "    Northwest National Laboratory, operated by Battelle Memorial Institute, Pacific\n",
      "    Northwest Division for the U.S. Department of Energy.\n",
      "    \n",
      "    Portions Copyright (c) 2002-2010, Washington University in St. Louis.\n",
      "    Portions Copyright (c) 2002-2020, Nathan A. Baker.\n",
      "    Portions Copyright (c) 1999-2002, The Regents of the University of California.\n",
      "    Portions Copyright (c) 1995, Michael Holst.\n",
      "    All rights reserved.\n",
      "    \n",
      "    Redistribution and use in source and binary forms, with or without\n",
      "    modification, are permitted provided that the following conditions are met:\n",
      "    \n",
      "    * Redistributions of source code must retain the above copyright notice, this\n",
      "      list of conditions and the following disclaimer.\n",
      "    \n",
      "    * Redistributions in binary form must reproduce the above copyright notice,\n",
      "      this list of conditions and the following disclaimer in the documentation\n",
      "      and/or other materials provided with the distribution.\n",
      "    \n",
      "    * Neither the name of the developer nor the names of its contributors may be\n",
      "      used to endorse or promote products derived from this software without\n",
      "      specific prior written permission.\n",
      "    \n",
      "    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS \"AS IS\" AND\n",
      "    ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED\n",
      "    WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE\n",
      "    DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR\n",
      "    ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES\n",
      "    (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;\n",
      "    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND\n",
      "    ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT\n",
      "    (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS\n",
      "    SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.\n",
      "----------------------------------------------------------------------\n",
      "    APBS uses FETK (the Finite Element ToolKit) to solve the\n",
      "    Poisson-Boltzmann equation numerically.  FETK is a portable collection\n",
      "    of finite element modeling class libraries developed by the Michael Holst\n",
      "    research group and written in an object-oriented form of C.  FEtk is\n",
      "    designed to solve general coupled systems of nonlinear partial differential\n",
      "    equations using adaptive finite element methods, inexact Newton methods,\n",
      "    and algebraic multilevel methods.  More information about FEtk may be found\n",
      "    at <http://www.FEtk.ORG>.\n",
      "----------------------------------------------------------------------\n",
      "    APBS also uses Aqua to solve the Poisson-Boltzmann equation numerically.  \n",
      "    Aqua is a modified form of the Holst group PMG library <http://www.FEtk.ORG>\n",
      "    which has been modified by Patrice Koehl\n",
      "    <http://koehllab.genomecenter.ucdavis.edu/> for improved efficiency and\n",
      "    memory usage when solving the Poisson-Boltzmann equation.\n",
      "----------------------------------------------------------------------\n",
      "    Please cite your use of APBS as:\n",
      "\n",
      "    Baker NA, Sept D, Joseph S, Holst MJ, McCammon JA. Electrostatics of\n",
      "    nanosystems: application to microtubules and the ribosome. Proc.\n",
      "    Natl. Acad. Sci. USA 98, 10037-10041 2001.\n",
      "    \n",
      "\n",
      "This executable compiled on May 29 2020 at 16:07:17\n",
      "\n",
      "Parsing input file AtSAL1_updated.in...\n",
      "rank 0 size 1...\n",
      "Parsed input file.\n",
      "Got paths for 1 molecules\n",
      "Reading PQR-format atom data from AtSAL1_updated.pqr.\n",
      "  5261 atoms\n",
      "  Centered at (-3.168e+01, -3.844e+01, -7.218e+00)\n",
      "  Net charge -1.20e+01 e\n",
      "Preparing to run 4 PBE calculations.\n",
      "----------------------------------------\n",
      "CALCULATION #1: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  976.593 MB total, 976.593 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 161 x 161\n",
      "  Grid spacings: 0.692 x 0.565 x 0.571\n",
      "  Grid lengths: 110.799 x 90.410 x 91.309\n",
      "  Grid center: (-31.682, -38.444, -7.218)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Single Debye-Huckel sphere boundary conditions\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 78.540\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 2.172967233269E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "CALCULATION #2: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  976.593 MB total, 1945.342 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 161 x 161\n",
      "  Grid spacings: 0.532 x 0.457 x 0.461\n",
      "  Grid lengths: 85.176 x 73.182 x 73.711\n",
      "  Grid center: (-31.682, -38.444, -7.218)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Boundary conditions from focusing\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 78.540\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Potential to be written to pot.dx\n",
      "  Total electrostatic energy = 3.292470291842E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "  Writing potential to pot.dx\n",
      "----------------------------------------\n",
      "CALCULATION #3: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  970.052 MB total, 1945.342 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 161 x 161\n",
      "  Grid spacings: 0.692 x 0.565 x 0.571\n",
      "  Grid lengths: 110.799 x 90.410 x 91.309\n",
      "  Grid center: (-31.682, -38.444, -7.218)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Single Debye-Huckel sphere boundary conditions\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 2.000\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 2.283434323883E+05 kJ/mol\n",
      "  Calculating forces...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[focusFillBound()]: WARNING:\n",
      "    Unusually large potential values\n",
      "    detected on the focusing boundary!\n",
      "    Convergence not guaranteed for NPBE/NRPBE calculations!\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "----------------------------------------\n",
      "CALCULATION #4: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  970.052 MB total, 1945.342 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 161 x 161\n",
      "  Grid spacings: 0.532 x 0.457 x 0.461\n",
      "  Grid lengths: 85.176 x 73.182 x 73.711\n",
      "  Grid center: (-31.682, -38.444, -7.218)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Boundary conditions from focusing\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 2.000\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 3.401581366045E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "PRINT STATEMENTS\n",
      "\n",
      "print energy 2 - 1 end\n",
      "  Local net energy (PE 0) = 1.091110742026E+04 kJ/mol\n",
      "  Global net ELEC energy = 1.091110742026E+04 kJ/mol\n",
      "----------------------------------------\n",
      "CLEANING UP AND SHUTTING DOWN...\n",
      "Destroying force arrays.\n",
      "No energy arrays to destroy.\n",
      "Destroying multigrid structures.\n",
      "Destroying finite element structures.\n",
      "Destroying 1 molecules\n",
      "Final memory usage:  0.001 MB total, 1945.342 MB high water\n",
      "\n",
      "\n",
      "Thanks for using APBS!\n",
      "\n",
      "Processing: pH 7.6\tAtAHL.in\n",
      "\n",
      "\n",
      "----------------------------------------------------------------------\n",
      "    APBS -- Adaptive Poisson-Boltzmann Solver\n",
      "    Version 3.0\n",
      "    \n",
      "    Nathan A. Baker (nathan.baker@pnnl.gov)\n",
      "    Pacific Northwest National Laboratory\n",
      "    \n",
      "    Additional contributing authors listed in the code documentation.\n",
      "    \n",
      "    Copyright (c) 2010-2020 Battelle Memorial Institute. Developed at the Pacific\n",
      "    Northwest National Laboratory, operated by Battelle Memorial Institute, Pacific\n",
      "    Northwest Division for the U.S. Department of Energy.\n",
      "    \n",
      "    Portions Copyright (c) 2002-2010, Washington University in St. Louis.\n",
      "    Portions Copyright (c) 2002-2020, Nathan A. Baker.\n",
      "    Portions Copyright (c) 1999-2002, The Regents of the University of California.\n",
      "    Portions Copyright (c) 1995, Michael Holst.\n",
      "    All rights reserved.\n",
      "    \n",
      "    Redistribution and use in source and binary forms, with or without\n",
      "    modification, are permitted provided that the following conditions are met:\n",
      "    \n",
      "    * Redistributions of source code must retain the above copyright notice, this\n",
      "      list of conditions and the following disclaimer.\n",
      "    \n",
      "    * Redistributions in binary form must reproduce the above copyright notice,\n",
      "      this list of conditions and the following disclaimer in the documentation\n",
      "      and/or other materials provided with the distribution.\n",
      "    \n",
      "    * Neither the name of the developer nor the names of its contributors may be\n",
      "      used to endorse or promote products derived from this software without\n",
      "      specific prior written permission.\n",
      "    \n",
      "    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS \"AS IS\" AND\n",
      "    ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED\n",
      "    WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE\n",
      "    DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR\n",
      "    ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES\n",
      "    (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;\n",
      "    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND\n",
      "    ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT\n",
      "    (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS\n",
      "    SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.\n",
      "----------------------------------------------------------------------\n",
      "    APBS uses FETK (the Finite Element ToolKit) to solve the\n",
      "    Poisson-Boltzmann equation numerically.  FETK is a portable collection\n",
      "    of finite element modeling class libraries developed by the Michael Holst\n",
      "    research group and written in an object-oriented form of C.  FEtk is\n",
      "    designed to solve general coupled systems of nonlinear partial differential\n",
      "    equations using adaptive finite element methods, inexact Newton methods,\n",
      "    and algebraic multilevel methods.  More information about FEtk may be found\n",
      "    at <http://www.FEtk.ORG>.\n",
      "----------------------------------------------------------------------\n",
      "    APBS also uses Aqua to solve the Poisson-Boltzmann equation numerically.  \n",
      "    Aqua is a modified form of the Holst group PMG library <http://www.FEtk.ORG>\n",
      "    which has been modified by Patrice Koehl\n",
      "    <http://koehllab.genomecenter.ucdavis.edu/> for improved efficiency and\n",
      "    memory usage when solving the Poisson-Boltzmann equation.\n",
      "----------------------------------------------------------------------\n",
      "    Please cite your use of APBS as:\n",
      "\n",
      "    Baker NA, Sept D, Joseph S, Holst MJ, McCammon JA. Electrostatics of\n",
      "    nanosystems: application to microtubules and the ribosome. Proc.\n",
      "    Natl. Acad. Sci. USA 98, 10037-10041 2001.\n",
      "    \n",
      "\n",
      "This executable compiled on May 29 2020 at 16:07:17\n",
      "\n",
      "Parsing input file AtAHL.in...\n",
      "rank 0 size 1...\n",
      "Parsed input file.\n",
      "Got paths for 1 molecules\n",
      "Reading PQR-format atom data from AtAHL.pqr.\n",
      "  5476 atoms\n",
      "  Centered at (-3.306e+01, -4.212e+01, -8.250e+00)\n",
      "  Net charge -5.00e+00 e\n",
      "Preparing to run 4 PBE calculations.\n",
      "----------------------------------------\n",
      "CALCULATION #1: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  791.341 MB total, 791.341 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 129 x 161\n",
      "  Grid spacings: 0.704 x 0.689 x 0.585\n",
      "  Grid lengths: 112.579 x 88.177 x 93.611\n",
      "  Grid center: (-33.059, -42.116, -8.250)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Single Debye-Huckel sphere boundary conditions\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 78.540\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 1.835427995658E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "CALCULATION #2: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  791.341 MB total, 1574.583 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 129 x 161\n",
      "  Grid spacings: 0.539 x 0.561 x 0.469\n",
      "  Grid lengths: 86.223 x 71.869 x 75.065\n",
      "  Grid center: (-33.059, -42.116, -8.250)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Boundary conditions from focusing\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 78.540\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Potential to be written to pot.dx\n",
      "  Total electrostatic energy = 2.877710175835E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "  Writing potential to pot.dx\n",
      "----------------------------------------\n",
      "CALCULATION #3: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  784.584 MB total, 1574.583 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 129 x 161\n",
      "  Grid spacings: 0.704 x 0.689 x 0.585\n",
      "  Grid lengths: 112.579 x 88.177 x 93.611\n",
      "  Grid center: (-33.059, -42.116, -8.250)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Single Debye-Huckel sphere boundary conditions\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 2.000\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 1.910317696734E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "CALCULATION #4: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  784.584 MB total, 1574.583 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 129 x 161\n",
      "  Grid spacings: 0.539 x 0.561 x 0.469\n",
      "  Grid lengths: 86.223 x 71.869 x 75.065\n",
      "  Grid center: (-33.059, -42.116, -8.250)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Boundary conditions from focusing\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 2.000\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 2.951342259002E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "PRINT STATEMENTS\n",
      "\n",
      "print energy 2 - 1 end\n",
      "  Local net energy (PE 0) = 7.363208316716E+03 kJ/mol\n",
      "  Global net ELEC energy = 7.363208316716E+03 kJ/mol\n",
      "----------------------------------------\n",
      "CLEANING UP AND SHUTTING DOWN...\n",
      "Destroying force arrays.\n",
      "No energy arrays to destroy.\n",
      "Destroying multigrid structures.\n",
      "Destroying finite element structures.\n",
      "Destroying 1 molecules\n",
      "Final memory usage:  0.001 MB total, 1574.583 MB high water\n",
      "\n",
      "\n",
      "Thanks for using APBS!\n",
      "\n",
      "Processing: pH 7.6\t74987.in\n",
      "\n",
      "\n",
      "----------------------------------------------------------------------\n",
      "    APBS -- Adaptive Poisson-Boltzmann Solver\n",
      "    Version 3.0\n",
      "    \n",
      "    Nathan A. Baker (nathan.baker@pnnl.gov)\n",
      "    Pacific Northwest National Laboratory\n",
      "    \n",
      "    Additional contributing authors listed in the code documentation.\n",
      "    \n",
      "    Copyright (c) 2010-2020 Battelle Memorial Institute. Developed at the Pacific\n",
      "    Northwest National Laboratory, operated by Battelle Memorial Institute, Pacific\n",
      "    Northwest Division for the U.S. Department of Energy.\n",
      "    \n",
      "    Portions Copyright (c) 2002-2010, Washington University in St. Louis.\n",
      "    Portions Copyright (c) 2002-2020, Nathan A. Baker.\n",
      "    Portions Copyright (c) 1999-2002, The Regents of the University of California.\n",
      "    Portions Copyright (c) 1995, Michael Holst.\n",
      "    All rights reserved.\n",
      "    \n",
      "    Redistribution and use in source and binary forms, with or without\n",
      "    modification, are permitted provided that the following conditions are met:\n",
      "    \n",
      "    * Redistributions of source code must retain the above copyright notice, this\n",
      "      list of conditions and the following disclaimer.\n",
      "    \n",
      "    * Redistributions in binary form must reproduce the above copyright notice,\n",
      "      this list of conditions and the following disclaimer in the documentation\n",
      "      and/or other materials provided with the distribution.\n",
      "    \n",
      "    * Neither the name of the developer nor the names of its contributors may be\n",
      "      used to endorse or promote products derived from this software without\n",
      "      specific prior written permission.\n",
      "    \n",
      "    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS \"AS IS\" AND\n",
      "    ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED\n",
      "    WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE\n",
      "    DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR\n",
      "    ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES\n",
      "    (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;\n",
      "    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND\n",
      "    ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT\n",
      "    (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS\n",
      "    SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.\n",
      "----------------------------------------------------------------------\n",
      "    APBS uses FETK (the Finite Element ToolKit) to solve the\n",
      "    Poisson-Boltzmann equation numerically.  FETK is a portable collection\n",
      "    of finite element modeling class libraries developed by the Michael Holst\n",
      "    research group and written in an object-oriented form of C.  FEtk is\n",
      "    designed to solve general coupled systems of nonlinear partial differential\n",
      "    equations using adaptive finite element methods, inexact Newton methods,\n",
      "    and algebraic multilevel methods.  More information about FEtk may be found\n",
      "    at <http://www.FEtk.ORG>.\n",
      "----------------------------------------------------------------------\n",
      "    APBS also uses Aqua to solve the Poisson-Boltzmann equation numerically.  \n",
      "    Aqua is a modified form of the Holst group PMG library <http://www.FEtk.ORG>\n",
      "    which has been modified by Patrice Koehl\n",
      "    <http://koehllab.genomecenter.ucdavis.edu/> for improved efficiency and\n",
      "    memory usage when solving the Poisson-Boltzmann equation.\n",
      "----------------------------------------------------------------------\n",
      "    Please cite your use of APBS as:\n",
      "\n",
      "    Baker NA, Sept D, Joseph S, Holst MJ, McCammon JA. Electrostatics of\n",
      "    nanosystems: application to microtubules and the ribosome. Proc.\n",
      "    Natl. Acad. Sci. USA 98, 10037-10041 2001.\n",
      "    \n",
      "\n",
      "This executable compiled on May 29 2020 at 16:07:17\n",
      "\n",
      "Parsing input file 74987.in...\n",
      "rank 0 size 1...\n",
      "Parsed input file.\n",
      "Got paths for 1 molecules\n",
      "Reading PQR-format atom data from 74987.pqr.\n",
      "  5356 atoms\n",
      "  Centered at (-3.215e+01, -4.055e+01, -6.682e+00)\n",
      "  Net charge -1.20e+01 e\n",
      "Preparing to run 4 PBE calculations.\n",
      "----------------------------------------\n",
      "CALCULATION #1: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  938.276 MB total, 938.276 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 193 x 129 x 161\n",
      "  Grid spacings: 0.613 x 0.676 x 0.590\n",
      "  Grid lengths: 117.764 x 86.573 x 94.369\n",
      "  Grid center: (-32.145, -40.554, -6.682)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Single Debye-Huckel sphere boundary conditions\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 78.540\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 2.044159580805E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "CALCULATION #2: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  938.276 MB total, 1868.851 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 193 x 129 x 161\n",
      "  Grid spacings: 0.465 x 0.554 x 0.472\n",
      "  Grid lengths: 89.273 x 70.925 x 75.511\n",
      "  Grid center: (-32.145, -40.554, -6.682)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Boundary conditions from focusing\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 78.540\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Potential to be written to pot.dx\n",
      "  Total electrostatic energy = 3.137100422795E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "  Writing potential to pot.dx\n",
      "----------------------------------------\n",
      "CALCULATION #3: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  931.895 MB total, 1868.851 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 193 x 129 x 161\n",
      "  Grid spacings: 0.613 x 0.676 x 0.590\n",
      "  Grid lengths: 117.764 x 86.573 x 94.369\n",
      "  Grid center: (-32.145, -40.554, -6.682)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Single Debye-Huckel sphere boundary conditions\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 2.000\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 2.136271554391E+05 kJ/mol\n",
      "  Calculating forces...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[focusFillBound()]: WARNING:\n",
      "    Unusually large potential values\n",
      "    detected on the focusing boundary!\n",
      "    Convergence not guaranteed for NPBE/NRPBE calculations!\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "----------------------------------------\n",
      "CALCULATION #4: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  931.895 MB total, 1868.851 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 193 x 129 x 161\n",
      "  Grid spacings: 0.465 x 0.554 x 0.472\n",
      "  Grid lengths: 89.273 x 70.925 x 75.511\n",
      "  Grid center: (-32.145, -40.554, -6.682)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Boundary conditions from focusing\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 2.000\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 3.227705337148E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "PRINT STATEMENTS\n",
      "\n",
      "print energy 2 - 1 end\n",
      "  Local net energy (PE 0) = 9.060491435336E+03 kJ/mol\n",
      "  Global net ELEC energy = 9.060491435336E+03 kJ/mol\n",
      "----------------------------------------\n",
      "CLEANING UP AND SHUTTING DOWN...\n",
      "Destroying force arrays.\n",
      "No energy arrays to destroy.\n",
      "Destroying multigrid structures.\n",
      "Destroying finite element structures.\n",
      "Destroying 1 molecules\n",
      "Final memory usage:  0.001 MB total, 1868.851 MB high water\n",
      "\n",
      "\n",
      "Thanks for using APBS!\n",
      "\n",
      "Processing: pH 7.6\t1ka1_hal2.in\n",
      "\n",
      "\n",
      "----------------------------------------------------------------------\n",
      "    APBS -- Adaptive Poisson-Boltzmann Solver\n",
      "    Version 3.0\n",
      "    \n",
      "    Nathan A. Baker (nathan.baker@pnnl.gov)\n",
      "    Pacific Northwest National Laboratory\n",
      "    \n",
      "    Additional contributing authors listed in the code documentation.\n",
      "    \n",
      "    Copyright (c) 2010-2020 Battelle Memorial Institute. Developed at the Pacific\n",
      "    Northwest National Laboratory, operated by Battelle Memorial Institute, Pacific\n",
      "    Northwest Division for the U.S. Department of Energy.\n",
      "    \n",
      "    Portions Copyright (c) 2002-2010, Washington University in St. Louis.\n",
      "    Portions Copyright (c) 2002-2020, Nathan A. Baker.\n",
      "    Portions Copyright (c) 1999-2002, The Regents of the University of California.\n",
      "    Portions Copyright (c) 1995, Michael Holst.\n",
      "    All rights reserved.\n",
      "    \n",
      "    Redistribution and use in source and binary forms, with or without\n",
      "    modification, are permitted provided that the following conditions are met:\n",
      "    \n",
      "    * Redistributions of source code must retain the above copyright notice, this\n",
      "      list of conditions and the following disclaimer.\n",
      "    \n",
      "    * Redistributions in binary form must reproduce the above copyright notice,\n",
      "      this list of conditions and the following disclaimer in the documentation\n",
      "      and/or other materials provided with the distribution.\n",
      "    \n",
      "    * Neither the name of the developer nor the names of its contributors may be\n",
      "      used to endorse or promote products derived from this software without\n",
      "      specific prior written permission.\n",
      "    \n",
      "    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS \"AS IS\" AND\n",
      "    ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED\n",
      "    WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE\n",
      "    DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR\n",
      "    ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES\n",
      "    (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;\n",
      "    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND\n",
      "    ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT\n",
      "    (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS\n",
      "    SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.\n",
      "----------------------------------------------------------------------\n",
      "    APBS uses FETK (the Finite Element ToolKit) to solve the\n",
      "    Poisson-Boltzmann equation numerically.  FETK is a portable collection\n",
      "    of finite element modeling class libraries developed by the Michael Holst\n",
      "    research group and written in an object-oriented form of C.  FEtk is\n",
      "    designed to solve general coupled systems of nonlinear partial differential\n",
      "    equations using adaptive finite element methods, inexact Newton methods,\n",
      "    and algebraic multilevel methods.  More information about FEtk may be found\n",
      "    at <http://www.FEtk.ORG>.\n",
      "----------------------------------------------------------------------\n",
      "    APBS also uses Aqua to solve the Poisson-Boltzmann equation numerically.  \n",
      "    Aqua is a modified form of the Holst group PMG library <http://www.FEtk.ORG>\n",
      "    which has been modified by Patrice Koehl\n",
      "    <http://koehllab.genomecenter.ucdavis.edu/> for improved efficiency and\n",
      "    memory usage when solving the Poisson-Boltzmann equation.\n",
      "----------------------------------------------------------------------\n",
      "    Please cite your use of APBS as:\n",
      "\n",
      "    Baker NA, Sept D, Joseph S, Holst MJ, McCammon JA. Electrostatics of\n",
      "    nanosystems: application to microtubules and the ribosome. Proc.\n",
      "    Natl. Acad. Sci. USA 98, 10037-10041 2001.\n",
      "    \n",
      "\n",
      "This executable compiled on May 29 2020 at 16:07:17\n",
      "\n",
      "Parsing input file 1ka1_hal2.in...\n",
      "rank 0 size 1...\n",
      "Parsed input file.\n",
      "Got paths for 1 molecules\n",
      "Reading PQR-format atom data from 1ka1_hal2.pqr.\n",
      "  5457 atoms\n",
      "  Centered at (-3.270e+01, -4.262e+01, -4.458e+00)\n",
      "  Net charge -5.00e+00 e\n",
      "Preparing to run 4 PBE calculations.\n",
      "----------------------------------------\n",
      "CALCULATION #1: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  795.568 MB total, 795.568 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 161 x 129\n",
      "  Grid spacings: 0.687 x 0.572 x 0.663\n",
      "  Grid lengths: 109.929 x 91.584 x 84.857\n",
      "  Grid center: (-32.703, -42.625, -4.458)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Single Debye-Huckel sphere boundary conditions\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 78.540\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 2.006514037351E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "CALCULATION #2: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  795.568 MB total, 1583.946 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 161 x 129\n",
      "  Grid spacings: 0.529 x 0.462 x 0.546\n",
      "  Grid lengths: 84.664 x 73.873 x 69.916\n",
      "  Grid center: (-32.703, -42.625, -4.458)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Boundary conditions from focusing\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 78.540\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Potential to be written to pot.dx\n",
      "  Total electrostatic energy = 3.061855552354E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "  Writing potential to pot.dx\n",
      "----------------------------------------\n",
      "CALCULATION #3: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  789.716 MB total, 1583.946 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 161 x 129\n",
      "  Grid spacings: 0.687 x 0.572 x 0.663\n",
      "  Grid lengths: 109.929 x 91.584 x 84.857\n",
      "  Grid center: (-32.703, -42.625, -4.458)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Single Debye-Huckel sphere boundary conditions\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 2.000\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 2.080654185153E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "CALCULATION #4: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  789.716 MB total, 1583.946 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 161 x 129\n",
      "  Grid spacings: 0.529 x 0.462 x 0.546\n",
      "  Grid lengths: 84.664 x 73.873 x 69.916\n",
      "  Grid center: (-32.703, -42.625, -4.458)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Boundary conditions from focusing\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 2.000\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 3.134712849475E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "PRINT STATEMENTS\n",
      "\n",
      "print energy 2 - 1 end\n",
      "  Local net energy (PE 0) = 7.285729712055E+03 kJ/mol\n",
      "  Global net ELEC energy = 7.285729712055E+03 kJ/mol\n",
      "----------------------------------------\n",
      "CLEANING UP AND SHUTTING DOWN...\n",
      "Destroying force arrays.\n",
      "No energy arrays to destroy.\n",
      "Destroying multigrid structures.\n",
      "Destroying finite element structures.\n",
      "Destroying 1 molecules\n",
      "Final memory usage:  0.001 MB total, 1583.946 MB high water\n",
      "\n",
      "\n",
      "Thanks for using APBS!\n",
      "\n",
      "Processing: pH 7.9\tAtSAL1_updated.in\n",
      "\n",
      "\n",
      "----------------------------------------------------------------------\n",
      "    APBS -- Adaptive Poisson-Boltzmann Solver\n",
      "    Version 3.0\n",
      "    \n",
      "    Nathan A. Baker (nathan.baker@pnnl.gov)\n",
      "    Pacific Northwest National Laboratory\n",
      "    \n",
      "    Additional contributing authors listed in the code documentation.\n",
      "    \n",
      "    Copyright (c) 2010-2020 Battelle Memorial Institute. Developed at the Pacific\n",
      "    Northwest National Laboratory, operated by Battelle Memorial Institute, Pacific\n",
      "    Northwest Division for the U.S. Department of Energy.\n",
      "    \n",
      "    Portions Copyright (c) 2002-2010, Washington University in St. Louis.\n",
      "    Portions Copyright (c) 2002-2020, Nathan A. Baker.\n",
      "    Portions Copyright (c) 1999-2002, The Regents of the University of California.\n",
      "    Portions Copyright (c) 1995, Michael Holst.\n",
      "    All rights reserved.\n",
      "    \n",
      "    Redistribution and use in source and binary forms, with or without\n",
      "    modification, are permitted provided that the following conditions are met:\n",
      "    \n",
      "    * Redistributions of source code must retain the above copyright notice, this\n",
      "      list of conditions and the following disclaimer.\n",
      "    \n",
      "    * Redistributions in binary form must reproduce the above copyright notice,\n",
      "      this list of conditions and the following disclaimer in the documentation\n",
      "      and/or other materials provided with the distribution.\n",
      "    \n",
      "    * Neither the name of the developer nor the names of its contributors may be\n",
      "      used to endorse or promote products derived from this software without\n",
      "      specific prior written permission.\n",
      "    \n",
      "    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS \"AS IS\" AND\n",
      "    ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED\n",
      "    WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE\n",
      "    DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR\n",
      "    ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES\n",
      "    (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;\n",
      "    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND\n",
      "    ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT\n",
      "    (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS\n",
      "    SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.\n",
      "----------------------------------------------------------------------\n",
      "    APBS uses FETK (the Finite Element ToolKit) to solve the\n",
      "    Poisson-Boltzmann equation numerically.  FETK is a portable collection\n",
      "    of finite element modeling class libraries developed by the Michael Holst\n",
      "    research group and written in an object-oriented form of C.  FEtk is\n",
      "    designed to solve general coupled systems of nonlinear partial differential\n",
      "    equations using adaptive finite element methods, inexact Newton methods,\n",
      "    and algebraic multilevel methods.  More information about FEtk may be found\n",
      "    at <http://www.FEtk.ORG>.\n",
      "----------------------------------------------------------------------\n",
      "    APBS also uses Aqua to solve the Poisson-Boltzmann equation numerically.  \n",
      "    Aqua is a modified form of the Holst group PMG library <http://www.FEtk.ORG>\n",
      "    which has been modified by Patrice Koehl\n",
      "    <http://koehllab.genomecenter.ucdavis.edu/> for improved efficiency and\n",
      "    memory usage when solving the Poisson-Boltzmann equation.\n",
      "----------------------------------------------------------------------\n",
      "    Please cite your use of APBS as:\n",
      "\n",
      "    Baker NA, Sept D, Joseph S, Holst MJ, McCammon JA. Electrostatics of\n",
      "    nanosystems: application to microtubules and the ribosome. Proc.\n",
      "    Natl. Acad. Sci. USA 98, 10037-10041 2001.\n",
      "    \n",
      "\n",
      "This executable compiled on May 29 2020 at 16:07:17\n",
      "\n",
      "Parsing input file AtSAL1_updated.in...\n",
      "rank 0 size 1...\n",
      "Parsed input file.\n",
      "Got paths for 1 molecules\n",
      "Reading PQR-format atom data from AtSAL1_updated.pqr.\n",
      "  5261 atoms\n",
      "  Centered at (-3.168e+01, -3.844e+01, -7.218e+00)\n",
      "  Net charge -1.20e+01 e\n",
      "Preparing to run 4 PBE calculations.\n",
      "----------------------------------------\n",
      "CALCULATION #1: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  976.593 MB total, 976.593 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 161 x 161\n",
      "  Grid spacings: 0.692 x 0.565 x 0.571\n",
      "  Grid lengths: 110.799 x 90.410 x 91.309\n",
      "  Grid center: (-31.682, -38.444, -7.218)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Single Debye-Huckel sphere boundary conditions\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 78.540\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 2.172967233269E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "CALCULATION #2: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  976.593 MB total, 1945.342 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 161 x 161\n",
      "  Grid spacings: 0.532 x 0.457 x 0.461\n",
      "  Grid lengths: 85.176 x 73.182 x 73.711\n",
      "  Grid center: (-31.682, -38.444, -7.218)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Boundary conditions from focusing\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 78.540\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Potential to be written to pot.dx\n",
      "  Total electrostatic energy = 3.292470291842E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "  Writing potential to pot.dx\n",
      "----------------------------------------\n",
      "CALCULATION #3: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  970.052 MB total, 1945.342 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 161 x 161\n",
      "  Grid spacings: 0.692 x 0.565 x 0.571\n",
      "  Grid lengths: 110.799 x 90.410 x 91.309\n",
      "  Grid center: (-31.682, -38.444, -7.218)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Single Debye-Huckel sphere boundary conditions\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 2.000\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 2.283434323883E+05 kJ/mol\n",
      "  Calculating forces...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[focusFillBound()]: WARNING:\n",
      "    Unusually large potential values\n",
      "    detected on the focusing boundary!\n",
      "    Convergence not guaranteed for NPBE/NRPBE calculations!\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "----------------------------------------\n",
      "CALCULATION #4: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  970.052 MB total, 1945.342 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 161 x 161\n",
      "  Grid spacings: 0.532 x 0.457 x 0.461\n",
      "  Grid lengths: 85.176 x 73.182 x 73.711\n",
      "  Grid center: (-31.682, -38.444, -7.218)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Boundary conditions from focusing\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 2.000\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 3.401581366045E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "PRINT STATEMENTS\n",
      "\n",
      "print energy 2 - 1 end\n",
      "  Local net energy (PE 0) = 1.091110742026E+04 kJ/mol\n",
      "  Global net ELEC energy = 1.091110742026E+04 kJ/mol\n",
      "----------------------------------------\n",
      "CLEANING UP AND SHUTTING DOWN...\n",
      "Destroying force arrays.\n",
      "No energy arrays to destroy.\n",
      "Destroying multigrid structures.\n",
      "Destroying finite element structures.\n",
      "Destroying 1 molecules\n",
      "Final memory usage:  0.001 MB total, 1945.342 MB high water\n",
      "\n",
      "\n",
      "Thanks for using APBS!\n",
      "\n",
      "Processing: pH 7.9\tAtAHL.in\n",
      "\n",
      "\n",
      "----------------------------------------------------------------------\n",
      "    APBS -- Adaptive Poisson-Boltzmann Solver\n",
      "    Version 3.0\n",
      "    \n",
      "    Nathan A. Baker (nathan.baker@pnnl.gov)\n",
      "    Pacific Northwest National Laboratory\n",
      "    \n",
      "    Additional contributing authors listed in the code documentation.\n",
      "    \n",
      "    Copyright (c) 2010-2020 Battelle Memorial Institute. Developed at the Pacific\n",
      "    Northwest National Laboratory, operated by Battelle Memorial Institute, Pacific\n",
      "    Northwest Division for the U.S. Department of Energy.\n",
      "    \n",
      "    Portions Copyright (c) 2002-2010, Washington University in St. Louis.\n",
      "    Portions Copyright (c) 2002-2020, Nathan A. Baker.\n",
      "    Portions Copyright (c) 1999-2002, The Regents of the University of California.\n",
      "    Portions Copyright (c) 1995, Michael Holst.\n",
      "    All rights reserved.\n",
      "    \n",
      "    Redistribution and use in source and binary forms, with or without\n",
      "    modification, are permitted provided that the following conditions are met:\n",
      "    \n",
      "    * Redistributions of source code must retain the above copyright notice, this\n",
      "      list of conditions and the following disclaimer.\n",
      "    \n",
      "    * Redistributions in binary form must reproduce the above copyright notice,\n",
      "      this list of conditions and the following disclaimer in the documentation\n",
      "      and/or other materials provided with the distribution.\n",
      "    \n",
      "    * Neither the name of the developer nor the names of its contributors may be\n",
      "      used to endorse or promote products derived from this software without\n",
      "      specific prior written permission.\n",
      "    \n",
      "    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS \"AS IS\" AND\n",
      "    ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED\n",
      "    WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE\n",
      "    DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR\n",
      "    ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES\n",
      "    (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;\n",
      "    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND\n",
      "    ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT\n",
      "    (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS\n",
      "    SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.\n",
      "----------------------------------------------------------------------\n",
      "    APBS uses FETK (the Finite Element ToolKit) to solve the\n",
      "    Poisson-Boltzmann equation numerically.  FETK is a portable collection\n",
      "    of finite element modeling class libraries developed by the Michael Holst\n",
      "    research group and written in an object-oriented form of C.  FEtk is\n",
      "    designed to solve general coupled systems of nonlinear partial differential\n",
      "    equations using adaptive finite element methods, inexact Newton methods,\n",
      "    and algebraic multilevel methods.  More information about FEtk may be found\n",
      "    at <http://www.FEtk.ORG>.\n",
      "----------------------------------------------------------------------\n",
      "    APBS also uses Aqua to solve the Poisson-Boltzmann equation numerically.  \n",
      "    Aqua is a modified form of the Holst group PMG library <http://www.FEtk.ORG>\n",
      "    which has been modified by Patrice Koehl\n",
      "    <http://koehllab.genomecenter.ucdavis.edu/> for improved efficiency and\n",
      "    memory usage when solving the Poisson-Boltzmann equation.\n",
      "----------------------------------------------------------------------\n",
      "    Please cite your use of APBS as:\n",
      "\n",
      "    Baker NA, Sept D, Joseph S, Holst MJ, McCammon JA. Electrostatics of\n",
      "    nanosystems: application to microtubules and the ribosome. Proc.\n",
      "    Natl. Acad. Sci. USA 98, 10037-10041 2001.\n",
      "    \n",
      "\n",
      "This executable compiled on May 29 2020 at 16:07:17\n",
      "\n",
      "Parsing input file AtAHL.in...\n",
      "rank 0 size 1...\n",
      "Parsed input file.\n",
      "Got paths for 1 molecules\n",
      "Reading PQR-format atom data from AtAHL.pqr.\n",
      "  5476 atoms\n",
      "  Centered at (-3.306e+01, -4.212e+01, -8.250e+00)\n",
      "  Net charge -5.00e+00 e\n",
      "Preparing to run 4 PBE calculations.\n",
      "----------------------------------------\n",
      "CALCULATION #1: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  791.341 MB total, 791.341 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 129 x 161\n",
      "  Grid spacings: 0.704 x 0.689 x 0.585\n",
      "  Grid lengths: 112.579 x 88.177 x 93.611\n",
      "  Grid center: (-33.059, -42.116, -8.250)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Single Debye-Huckel sphere boundary conditions\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 78.540\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 1.835427995658E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "CALCULATION #2: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  791.341 MB total, 1574.583 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 129 x 161\n",
      "  Grid spacings: 0.539 x 0.561 x 0.469\n",
      "  Grid lengths: 86.223 x 71.869 x 75.065\n",
      "  Grid center: (-33.059, -42.116, -8.250)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Boundary conditions from focusing\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 78.540\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Potential to be written to pot.dx\n",
      "  Total electrostatic energy = 2.877710175835E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "  Writing potential to pot.dx\n",
      "----------------------------------------\n",
      "CALCULATION #3: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  784.584 MB total, 1574.583 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 129 x 161\n",
      "  Grid spacings: 0.704 x 0.689 x 0.585\n",
      "  Grid lengths: 112.579 x 88.177 x 93.611\n",
      "  Grid center: (-33.059, -42.116, -8.250)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Single Debye-Huckel sphere boundary conditions\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 2.000\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 1.910317696734E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "CALCULATION #4: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  784.584 MB total, 1574.583 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 129 x 161\n",
      "  Grid spacings: 0.539 x 0.561 x 0.469\n",
      "  Grid lengths: 86.223 x 71.869 x 75.065\n",
      "  Grid center: (-33.059, -42.116, -8.250)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Boundary conditions from focusing\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 2.000\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 2.951342259002E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "PRINT STATEMENTS\n",
      "\n",
      "print energy 2 - 1 end\n",
      "  Local net energy (PE 0) = 7.363208316716E+03 kJ/mol\n",
      "  Global net ELEC energy = 7.363208316716E+03 kJ/mol\n",
      "----------------------------------------\n",
      "CLEANING UP AND SHUTTING DOWN...\n",
      "Destroying force arrays.\n",
      "No energy arrays to destroy.\n",
      "Destroying multigrid structures.\n",
      "Destroying finite element structures.\n",
      "Destroying 1 molecules\n",
      "Final memory usage:  0.001 MB total, 1574.583 MB high water\n",
      "\n",
      "\n",
      "Thanks for using APBS!\n",
      "\n",
      "Processing: pH 7.9\t74987.in\n",
      "\n",
      "\n",
      "----------------------------------------------------------------------\n",
      "    APBS -- Adaptive Poisson-Boltzmann Solver\n",
      "    Version 3.0\n",
      "    \n",
      "    Nathan A. Baker (nathan.baker@pnnl.gov)\n",
      "    Pacific Northwest National Laboratory\n",
      "    \n",
      "    Additional contributing authors listed in the code documentation.\n",
      "    \n",
      "    Copyright (c) 2010-2020 Battelle Memorial Institute. Developed at the Pacific\n",
      "    Northwest National Laboratory, operated by Battelle Memorial Institute, Pacific\n",
      "    Northwest Division for the U.S. Department of Energy.\n",
      "    \n",
      "    Portions Copyright (c) 2002-2010, Washington University in St. Louis.\n",
      "    Portions Copyright (c) 2002-2020, Nathan A. Baker.\n",
      "    Portions Copyright (c) 1999-2002, The Regents of the University of California.\n",
      "    Portions Copyright (c) 1995, Michael Holst.\n",
      "    All rights reserved.\n",
      "    \n",
      "    Redistribution and use in source and binary forms, with or without\n",
      "    modification, are permitted provided that the following conditions are met:\n",
      "    \n",
      "    * Redistributions of source code must retain the above copyright notice, this\n",
      "      list of conditions and the following disclaimer.\n",
      "    \n",
      "    * Redistributions in binary form must reproduce the above copyright notice,\n",
      "      this list of conditions and the following disclaimer in the documentation\n",
      "      and/or other materials provided with the distribution.\n",
      "    \n",
      "    * Neither the name of the developer nor the names of its contributors may be\n",
      "      used to endorse or promote products derived from this software without\n",
      "      specific prior written permission.\n",
      "    \n",
      "    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS \"AS IS\" AND\n",
      "    ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED\n",
      "    WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE\n",
      "    DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR\n",
      "    ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES\n",
      "    (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;\n",
      "    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND\n",
      "    ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT\n",
      "    (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS\n",
      "    SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.\n",
      "----------------------------------------------------------------------\n",
      "    APBS uses FETK (the Finite Element ToolKit) to solve the\n",
      "    Poisson-Boltzmann equation numerically.  FETK is a portable collection\n",
      "    of finite element modeling class libraries developed by the Michael Holst\n",
      "    research group and written in an object-oriented form of C.  FEtk is\n",
      "    designed to solve general coupled systems of nonlinear partial differential\n",
      "    equations using adaptive finite element methods, inexact Newton methods,\n",
      "    and algebraic multilevel methods.  More information about FEtk may be found\n",
      "    at <http://www.FEtk.ORG>.\n",
      "----------------------------------------------------------------------\n",
      "    APBS also uses Aqua to solve the Poisson-Boltzmann equation numerically.  \n",
      "    Aqua is a modified form of the Holst group PMG library <http://www.FEtk.ORG>\n",
      "    which has been modified by Patrice Koehl\n",
      "    <http://koehllab.genomecenter.ucdavis.edu/> for improved efficiency and\n",
      "    memory usage when solving the Poisson-Boltzmann equation.\n",
      "----------------------------------------------------------------------\n",
      "    Please cite your use of APBS as:\n",
      "\n",
      "    Baker NA, Sept D, Joseph S, Holst MJ, McCammon JA. Electrostatics of\n",
      "    nanosystems: application to microtubules and the ribosome. Proc.\n",
      "    Natl. Acad. Sci. USA 98, 10037-10041 2001.\n",
      "    \n",
      "\n",
      "This executable compiled on May 29 2020 at 16:07:17\n",
      "\n",
      "Parsing input file 74987.in...\n",
      "rank 0 size 1...\n",
      "Parsed input file.\n",
      "Got paths for 1 molecules\n",
      "Reading PQR-format atom data from 74987.pqr.\n",
      "  5356 atoms\n",
      "  Centered at (-3.215e+01, -4.055e+01, -6.682e+00)\n",
      "  Net charge -1.20e+01 e\n",
      "Preparing to run 4 PBE calculations.\n",
      "----------------------------------------\n",
      "CALCULATION #1: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  938.276 MB total, 938.276 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 193 x 129 x 161\n",
      "  Grid spacings: 0.613 x 0.676 x 0.590\n",
      "  Grid lengths: 117.764 x 86.573 x 94.369\n",
      "  Grid center: (-32.145, -40.554, -6.682)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Single Debye-Huckel sphere boundary conditions\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 78.540\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 2.044159580805E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "CALCULATION #2: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  938.276 MB total, 1868.851 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 193 x 129 x 161\n",
      "  Grid spacings: 0.465 x 0.554 x 0.472\n",
      "  Grid lengths: 89.273 x 70.925 x 75.511\n",
      "  Grid center: (-32.145, -40.554, -6.682)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Boundary conditions from focusing\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 78.540\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Potential to be written to pot.dx\n",
      "  Total electrostatic energy = 3.137100422795E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "  Writing potential to pot.dx\n",
      "----------------------------------------\n",
      "CALCULATION #3: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  931.895 MB total, 1868.851 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 193 x 129 x 161\n",
      "  Grid spacings: 0.613 x 0.676 x 0.590\n",
      "  Grid lengths: 117.764 x 86.573 x 94.369\n",
      "  Grid center: (-32.145, -40.554, -6.682)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Single Debye-Huckel sphere boundary conditions\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 2.000\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 2.136271554391E+05 kJ/mol\n",
      "  Calculating forces...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[focusFillBound()]: WARNING:\n",
      "    Unusually large potential values\n",
      "    detected on the focusing boundary!\n",
      "    Convergence not guaranteed for NPBE/NRPBE calculations!\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "----------------------------------------\n",
      "CALCULATION #4: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  931.895 MB total, 1868.851 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 193 x 129 x 161\n",
      "  Grid spacings: 0.465 x 0.554 x 0.472\n",
      "  Grid lengths: 89.273 x 70.925 x 75.511\n",
      "  Grid center: (-32.145, -40.554, -6.682)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Boundary conditions from focusing\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 2.000\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 3.227705337148E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "PRINT STATEMENTS\n",
      "\n",
      "print energy 2 - 1 end\n",
      "  Local net energy (PE 0) = 9.060491435336E+03 kJ/mol\n",
      "  Global net ELEC energy = 9.060491435336E+03 kJ/mol\n",
      "----------------------------------------\n",
      "CLEANING UP AND SHUTTING DOWN...\n",
      "Destroying force arrays.\n",
      "No energy arrays to destroy.\n",
      "Destroying multigrid structures.\n",
      "Destroying finite element structures.\n",
      "Destroying 1 molecules\n",
      "Final memory usage:  0.001 MB total, 1868.851 MB high water\n",
      "\n",
      "\n",
      "Thanks for using APBS!\n",
      "\n",
      "Processing: pH 7.9\t1ka1_hal2.in\n",
      "\n",
      "\n",
      "----------------------------------------------------------------------\n",
      "    APBS -- Adaptive Poisson-Boltzmann Solver\n",
      "    Version 3.0\n",
      "    \n",
      "    Nathan A. Baker (nathan.baker@pnnl.gov)\n",
      "    Pacific Northwest National Laboratory\n",
      "    \n",
      "    Additional contributing authors listed in the code documentation.\n",
      "    \n",
      "    Copyright (c) 2010-2020 Battelle Memorial Institute. Developed at the Pacific\n",
      "    Northwest National Laboratory, operated by Battelle Memorial Institute, Pacific\n",
      "    Northwest Division for the U.S. Department of Energy.\n",
      "    \n",
      "    Portions Copyright (c) 2002-2010, Washington University in St. Louis.\n",
      "    Portions Copyright (c) 2002-2020, Nathan A. Baker.\n",
      "    Portions Copyright (c) 1999-2002, The Regents of the University of California.\n",
      "    Portions Copyright (c) 1995, Michael Holst.\n",
      "    All rights reserved.\n",
      "    \n",
      "    Redistribution and use in source and binary forms, with or without\n",
      "    modification, are permitted provided that the following conditions are met:\n",
      "    \n",
      "    * Redistributions of source code must retain the above copyright notice, this\n",
      "      list of conditions and the following disclaimer.\n",
      "    \n",
      "    * Redistributions in binary form must reproduce the above copyright notice,\n",
      "      this list of conditions and the following disclaimer in the documentation\n",
      "      and/or other materials provided with the distribution.\n",
      "    \n",
      "    * Neither the name of the developer nor the names of its contributors may be\n",
      "      used to endorse or promote products derived from this software without\n",
      "      specific prior written permission.\n",
      "    \n",
      "    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS \"AS IS\" AND\n",
      "    ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED\n",
      "    WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE\n",
      "    DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR\n",
      "    ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES\n",
      "    (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;\n",
      "    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND\n",
      "    ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT\n",
      "    (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS\n",
      "    SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.\n",
      "----------------------------------------------------------------------\n",
      "    APBS uses FETK (the Finite Element ToolKit) to solve the\n",
      "    Poisson-Boltzmann equation numerically.  FETK is a portable collection\n",
      "    of finite element modeling class libraries developed by the Michael Holst\n",
      "    research group and written in an object-oriented form of C.  FEtk is\n",
      "    designed to solve general coupled systems of nonlinear partial differential\n",
      "    equations using adaptive finite element methods, inexact Newton methods,\n",
      "    and algebraic multilevel methods.  More information about FEtk may be found\n",
      "    at <http://www.FEtk.ORG>.\n",
      "----------------------------------------------------------------------\n",
      "    APBS also uses Aqua to solve the Poisson-Boltzmann equation numerically.  \n",
      "    Aqua is a modified form of the Holst group PMG library <http://www.FEtk.ORG>\n",
      "    which has been modified by Patrice Koehl\n",
      "    <http://koehllab.genomecenter.ucdavis.edu/> for improved efficiency and\n",
      "    memory usage when solving the Poisson-Boltzmann equation.\n",
      "----------------------------------------------------------------------\n",
      "    Please cite your use of APBS as:\n",
      "\n",
      "    Baker NA, Sept D, Joseph S, Holst MJ, McCammon JA. Electrostatics of\n",
      "    nanosystems: application to microtubules and the ribosome. Proc.\n",
      "    Natl. Acad. Sci. USA 98, 10037-10041 2001.\n",
      "    \n",
      "\n",
      "This executable compiled on May 29 2020 at 16:07:17\n",
      "\n",
      "Parsing input file 1ka1_hal2.in...\n",
      "rank 0 size 1...\n",
      "Parsed input file.\n",
      "Got paths for 1 molecules\n",
      "Reading PQR-format atom data from 1ka1_hal2.pqr.\n",
      "  5456 atoms\n",
      "  Centered at (-3.270e+01, -4.262e+01, -4.458e+00)\n",
      "  Net charge -6.00e+00 e\n",
      "Preparing to run 4 PBE calculations.\n",
      "----------------------------------------\n",
      "CALCULATION #1: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  795.565 MB total, 795.565 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 161 x 129\n",
      "  Grid spacings: 0.687 x 0.572 x 0.663\n",
      "  Grid lengths: 109.929 x 91.584 x 84.857\n",
      "  Grid center: (-32.703, -42.625, -4.458)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Single Debye-Huckel sphere boundary conditions\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 78.540\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 2.007841772386E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "CALCULATION #2: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  795.565 MB total, 1583.939 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 161 x 129\n",
      "  Grid spacings: 0.529 x 0.462 x 0.546\n",
      "  Grid lengths: 84.664 x 73.873 x 69.916\n",
      "  Grid center: (-32.703, -42.625, -4.458)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Boundary conditions from focusing\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 78.540\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Potential to be written to pot.dx\n",
      "  Total electrostatic energy = 3.064614386616E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "  Writing potential to pot.dx\n",
      "----------------------------------------\n",
      "CALCULATION #3: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  789.712 MB total, 1583.939 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 161 x 129\n",
      "  Grid spacings: 0.687 x 0.572 x 0.663\n",
      "  Grid lengths: 109.929 x 91.584 x 84.857\n",
      "  Grid center: (-32.703, -42.625, -4.458)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Single Debye-Huckel sphere boundary conditions\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 2.000\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 2.082913994093E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "CALCULATION #4: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  789.712 MB total, 1583.939 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 161 x 129\n",
      "  Grid spacings: 0.529 x 0.462 x 0.546\n",
      "  Grid lengths: 84.664 x 73.873 x 69.916\n",
      "  Grid center: (-32.703, -42.625, -4.458)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Boundary conditions from focusing\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 2.000\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 3.138376812262E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "PRINT STATEMENTS\n",
      "\n",
      "print energy 2 - 1 end\n",
      "  Local net energy (PE 0) = 7.376242564653E+03 kJ/mol\n",
      "  Global net ELEC energy = 7.376242564653E+03 kJ/mol\n",
      "----------------------------------------\n",
      "CLEANING UP AND SHUTTING DOWN...\n",
      "Destroying force arrays.\n",
      "No energy arrays to destroy.\n",
      "Destroying multigrid structures.\n",
      "Destroying finite element structures.\n",
      "Destroying 1 molecules\n",
      "Final memory usage:  0.001 MB total, 1583.939 MB high water\n",
      "\n",
      "\n",
      "Thanks for using APBS!\n",
      "\n",
      "Processing: pH 8.2\tAtSAL1_updated.in\n",
      "\n",
      "\n",
      "----------------------------------------------------------------------\n",
      "    APBS -- Adaptive Poisson-Boltzmann Solver\n",
      "    Version 3.0\n",
      "    \n",
      "    Nathan A. Baker (nathan.baker@pnnl.gov)\n",
      "    Pacific Northwest National Laboratory\n",
      "    \n",
      "    Additional contributing authors listed in the code documentation.\n",
      "    \n",
      "    Copyright (c) 2010-2020 Battelle Memorial Institute. Developed at the Pacific\n",
      "    Northwest National Laboratory, operated by Battelle Memorial Institute, Pacific\n",
      "    Northwest Division for the U.S. Department of Energy.\n",
      "    \n",
      "    Portions Copyright (c) 2002-2010, Washington University in St. Louis.\n",
      "    Portions Copyright (c) 2002-2020, Nathan A. Baker.\n",
      "    Portions Copyright (c) 1999-2002, The Regents of the University of California.\n",
      "    Portions Copyright (c) 1995, Michael Holst.\n",
      "    All rights reserved.\n",
      "    \n",
      "    Redistribution and use in source and binary forms, with or without\n",
      "    modification, are permitted provided that the following conditions are met:\n",
      "    \n",
      "    * Redistributions of source code must retain the above copyright notice, this\n",
      "      list of conditions and the following disclaimer.\n",
      "    \n",
      "    * Redistributions in binary form must reproduce the above copyright notice,\n",
      "      this list of conditions and the following disclaimer in the documentation\n",
      "      and/or other materials provided with the distribution.\n",
      "    \n",
      "    * Neither the name of the developer nor the names of its contributors may be\n",
      "      used to endorse or promote products derived from this software without\n",
      "      specific prior written permission.\n",
      "    \n",
      "    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS \"AS IS\" AND\n",
      "    ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED\n",
      "    WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE\n",
      "    DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR\n",
      "    ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES\n",
      "    (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;\n",
      "    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND\n",
      "    ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT\n",
      "    (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS\n",
      "    SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.\n",
      "----------------------------------------------------------------------\n",
      "    APBS uses FETK (the Finite Element ToolKit) to solve the\n",
      "    Poisson-Boltzmann equation numerically.  FETK is a portable collection\n",
      "    of finite element modeling class libraries developed by the Michael Holst\n",
      "    research group and written in an object-oriented form of C.  FEtk is\n",
      "    designed to solve general coupled systems of nonlinear partial differential\n",
      "    equations using adaptive finite element methods, inexact Newton methods,\n",
      "    and algebraic multilevel methods.  More information about FEtk may be found\n",
      "    at <http://www.FEtk.ORG>.\n",
      "----------------------------------------------------------------------\n",
      "    APBS also uses Aqua to solve the Poisson-Boltzmann equation numerically.  \n",
      "    Aqua is a modified form of the Holst group PMG library <http://www.FEtk.ORG>\n",
      "    which has been modified by Patrice Koehl\n",
      "    <http://koehllab.genomecenter.ucdavis.edu/> for improved efficiency and\n",
      "    memory usage when solving the Poisson-Boltzmann equation.\n",
      "----------------------------------------------------------------------\n",
      "    Please cite your use of APBS as:\n",
      "\n",
      "    Baker NA, Sept D, Joseph S, Holst MJ, McCammon JA. Electrostatics of\n",
      "    nanosystems: application to microtubules and the ribosome. Proc.\n",
      "    Natl. Acad. Sci. USA 98, 10037-10041 2001.\n",
      "    \n",
      "\n",
      "This executable compiled on May 29 2020 at 16:07:17\n",
      "\n",
      "Parsing input file AtSAL1_updated.in...\n",
      "rank 0 size 1...\n",
      "Parsed input file.\n",
      "Got paths for 1 molecules\n",
      "Reading PQR-format atom data from AtSAL1_updated.pqr.\n",
      "  5261 atoms\n",
      "  Centered at (-3.168e+01, -3.844e+01, -7.218e+00)\n",
      "  Net charge -1.20e+01 e\n",
      "Preparing to run 4 PBE calculations.\n",
      "----------------------------------------\n",
      "CALCULATION #1: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  976.593 MB total, 976.593 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 161 x 161\n",
      "  Grid spacings: 0.692 x 0.565 x 0.571\n",
      "  Grid lengths: 110.799 x 90.410 x 91.309\n",
      "  Grid center: (-31.682, -38.444, -7.218)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Single Debye-Huckel sphere boundary conditions\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 78.540\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 2.172967233269E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "CALCULATION #2: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  976.593 MB total, 1945.342 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 161 x 161\n",
      "  Grid spacings: 0.532 x 0.457 x 0.461\n",
      "  Grid lengths: 85.176 x 73.182 x 73.711\n",
      "  Grid center: (-31.682, -38.444, -7.218)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Boundary conditions from focusing\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 78.540\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Potential to be written to pot.dx\n",
      "  Total electrostatic energy = 3.292470291842E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "  Writing potential to pot.dx\n",
      "----------------------------------------\n",
      "CALCULATION #3: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  970.052 MB total, 1945.342 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 161 x 161\n",
      "  Grid spacings: 0.692 x 0.565 x 0.571\n",
      "  Grid lengths: 110.799 x 90.410 x 91.309\n",
      "  Grid center: (-31.682, -38.444, -7.218)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Single Debye-Huckel sphere boundary conditions\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 2.000\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 2.283434323883E+05 kJ/mol\n",
      "  Calculating forces...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[focusFillBound()]: WARNING:\n",
      "    Unusually large potential values\n",
      "    detected on the focusing boundary!\n",
      "    Convergence not guaranteed for NPBE/NRPBE calculations!\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "----------------------------------------\n",
      "CALCULATION #4: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  970.052 MB total, 1945.342 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 161 x 161\n",
      "  Grid spacings: 0.532 x 0.457 x 0.461\n",
      "  Grid lengths: 85.176 x 73.182 x 73.711\n",
      "  Grid center: (-31.682, -38.444, -7.218)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Boundary conditions from focusing\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 2.000\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 3.401581366045E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "PRINT STATEMENTS\n",
      "\n",
      "print energy 2 - 1 end\n",
      "  Local net energy (PE 0) = 1.091110742026E+04 kJ/mol\n",
      "  Global net ELEC energy = 1.091110742026E+04 kJ/mol\n",
      "----------------------------------------\n",
      "CLEANING UP AND SHUTTING DOWN...\n",
      "Destroying force arrays.\n",
      "No energy arrays to destroy.\n",
      "Destroying multigrid structures.\n",
      "Destroying finite element structures.\n",
      "Destroying 1 molecules\n",
      "Final memory usage:  0.001 MB total, 1945.342 MB high water\n",
      "\n",
      "\n",
      "Thanks for using APBS!\n",
      "\n",
      "Processing: pH 8.2\tAtAHL.in\n",
      "\n",
      "\n",
      "----------------------------------------------------------------------\n",
      "    APBS -- Adaptive Poisson-Boltzmann Solver\n",
      "    Version 3.0\n",
      "    \n",
      "    Nathan A. Baker (nathan.baker@pnnl.gov)\n",
      "    Pacific Northwest National Laboratory\n",
      "    \n",
      "    Additional contributing authors listed in the code documentation.\n",
      "    \n",
      "    Copyright (c) 2010-2020 Battelle Memorial Institute. Developed at the Pacific\n",
      "    Northwest National Laboratory, operated by Battelle Memorial Institute, Pacific\n",
      "    Northwest Division for the U.S. Department of Energy.\n",
      "    \n",
      "    Portions Copyright (c) 2002-2010, Washington University in St. Louis.\n",
      "    Portions Copyright (c) 2002-2020, Nathan A. Baker.\n",
      "    Portions Copyright (c) 1999-2002, The Regents of the University of California.\n",
      "    Portions Copyright (c) 1995, Michael Holst.\n",
      "    All rights reserved.\n",
      "    \n",
      "    Redistribution and use in source and binary forms, with or without\n",
      "    modification, are permitted provided that the following conditions are met:\n",
      "    \n",
      "    * Redistributions of source code must retain the above copyright notice, this\n",
      "      list of conditions and the following disclaimer.\n",
      "    \n",
      "    * Redistributions in binary form must reproduce the above copyright notice,\n",
      "      this list of conditions and the following disclaimer in the documentation\n",
      "      and/or other materials provided with the distribution.\n",
      "    \n",
      "    * Neither the name of the developer nor the names of its contributors may be\n",
      "      used to endorse or promote products derived from this software without\n",
      "      specific prior written permission.\n",
      "    \n",
      "    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS \"AS IS\" AND\n",
      "    ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED\n",
      "    WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE\n",
      "    DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR\n",
      "    ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES\n",
      "    (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;\n",
      "    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND\n",
      "    ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT\n",
      "    (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS\n",
      "    SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.\n",
      "----------------------------------------------------------------------\n",
      "    APBS uses FETK (the Finite Element ToolKit) to solve the\n",
      "    Poisson-Boltzmann equation numerically.  FETK is a portable collection\n",
      "    of finite element modeling class libraries developed by the Michael Holst\n",
      "    research group and written in an object-oriented form of C.  FEtk is\n",
      "    designed to solve general coupled systems of nonlinear partial differential\n",
      "    equations using adaptive finite element methods, inexact Newton methods,\n",
      "    and algebraic multilevel methods.  More information about FEtk may be found\n",
      "    at <http://www.FEtk.ORG>.\n",
      "----------------------------------------------------------------------\n",
      "    APBS also uses Aqua to solve the Poisson-Boltzmann equation numerically.  \n",
      "    Aqua is a modified form of the Holst group PMG library <http://www.FEtk.ORG>\n",
      "    which has been modified by Patrice Koehl\n",
      "    <http://koehllab.genomecenter.ucdavis.edu/> for improved efficiency and\n",
      "    memory usage when solving the Poisson-Boltzmann equation.\n",
      "----------------------------------------------------------------------\n",
      "    Please cite your use of APBS as:\n",
      "\n",
      "    Baker NA, Sept D, Joseph S, Holst MJ, McCammon JA. Electrostatics of\n",
      "    nanosystems: application to microtubules and the ribosome. Proc.\n",
      "    Natl. Acad. Sci. USA 98, 10037-10041 2001.\n",
      "    \n",
      "\n",
      "This executable compiled on May 29 2020 at 16:07:17\n",
      "\n",
      "Parsing input file AtAHL.in...\n",
      "rank 0 size 1...\n",
      "Parsed input file.\n",
      "Got paths for 1 molecules\n",
      "Reading PQR-format atom data from AtAHL.pqr.\n",
      "  5475 atoms\n",
      "  Centered at (-3.306e+01, -4.212e+01, -8.250e+00)\n",
      "  Net charge -6.00e+00 e\n",
      "Preparing to run 4 PBE calculations.\n",
      "----------------------------------------\n",
      "CALCULATION #1: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  791.341 MB total, 791.341 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 129 x 161\n",
      "  Grid spacings: 0.704 x 0.689 x 0.585\n",
      "  Grid lengths: 112.579 x 88.177 x 93.611\n",
      "  Grid center: (-33.059, -42.116, -8.250)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Single Debye-Huckel sphere boundary conditions\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 78.540\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 1.838119647748E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "CALCULATION #2: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  791.341 MB total, 1574.584 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 129 x 161\n",
      "  Grid spacings: 0.539 x 0.561 x 0.469\n",
      "  Grid lengths: 86.223 x 71.869 x 75.065\n",
      "  Grid center: (-33.059, -42.116, -8.250)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Boundary conditions from focusing\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 78.540\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Potential to be written to pot.dx\n",
      "  Total electrostatic energy = 2.881459609442E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "  Writing potential to pot.dx\n",
      "----------------------------------------\n",
      "CALCULATION #3: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  784.584 MB total, 1574.584 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 129 x 161\n",
      "  Grid spacings: 0.704 x 0.689 x 0.585\n",
      "  Grid lengths: 112.579 x 88.177 x 93.611\n",
      "  Grid center: (-33.059, -42.116, -8.250)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Single Debye-Huckel sphere boundary conditions\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 2.000\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 1.916366817853E+05 kJ/mol\n",
      "  Calculating forces...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[focusFillBound()]: WARNING:\n",
      "    Unusually large potential values\n",
      "    detected on the focusing boundary!\n",
      "    Convergence not guaranteed for NPBE/NRPBE calculations!\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "----------------------------------------\n",
      "CALCULATION #4: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  784.584 MB total, 1574.584 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 129 x 161\n",
      "  Grid spacings: 0.539 x 0.561 x 0.469\n",
      "  Grid lengths: 86.223 x 71.869 x 75.065\n",
      "  Grid center: (-33.059, -42.116, -8.250)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Boundary conditions from focusing\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 2.000\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 2.958456063612E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "PRINT STATEMENTS\n",
      "\n",
      "print energy 2 - 1 end\n",
      "  Local net energy (PE 0) = 7.699645416924E+03 kJ/mol\n",
      "  Global net ELEC energy = 7.699645416924E+03 kJ/mol\n",
      "----------------------------------------\n",
      "CLEANING UP AND SHUTTING DOWN...\n",
      "Destroying force arrays.\n",
      "No energy arrays to destroy.\n",
      "Destroying multigrid structures.\n",
      "Destroying finite element structures.\n",
      "Destroying 1 molecules\n",
      "Final memory usage:  0.001 MB total, 1574.584 MB high water\n",
      "\n",
      "\n",
      "Thanks for using APBS!\n",
      "\n",
      "Processing: pH 8.2\t74987.in\n",
      "\n",
      "\n",
      "----------------------------------------------------------------------\n",
      "    APBS -- Adaptive Poisson-Boltzmann Solver\n",
      "    Version 3.0\n",
      "    \n",
      "    Nathan A. Baker (nathan.baker@pnnl.gov)\n",
      "    Pacific Northwest National Laboratory\n",
      "    \n",
      "    Additional contributing authors listed in the code documentation.\n",
      "    \n",
      "    Copyright (c) 2010-2020 Battelle Memorial Institute. Developed at the Pacific\n",
      "    Northwest National Laboratory, operated by Battelle Memorial Institute, Pacific\n",
      "    Northwest Division for the U.S. Department of Energy.\n",
      "    \n",
      "    Portions Copyright (c) 2002-2010, Washington University in St. Louis.\n",
      "    Portions Copyright (c) 2002-2020, Nathan A. Baker.\n",
      "    Portions Copyright (c) 1999-2002, The Regents of the University of California.\n",
      "    Portions Copyright (c) 1995, Michael Holst.\n",
      "    All rights reserved.\n",
      "    \n",
      "    Redistribution and use in source and binary forms, with or without\n",
      "    modification, are permitted provided that the following conditions are met:\n",
      "    \n",
      "    * Redistributions of source code must retain the above copyright notice, this\n",
      "      list of conditions and the following disclaimer.\n",
      "    \n",
      "    * Redistributions in binary form must reproduce the above copyright notice,\n",
      "      this list of conditions and the following disclaimer in the documentation\n",
      "      and/or other materials provided with the distribution.\n",
      "    \n",
      "    * Neither the name of the developer nor the names of its contributors may be\n",
      "      used to endorse or promote products derived from this software without\n",
      "      specific prior written permission.\n",
      "    \n",
      "    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS \"AS IS\" AND\n",
      "    ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED\n",
      "    WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE\n",
      "    DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR\n",
      "    ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES\n",
      "    (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;\n",
      "    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND\n",
      "    ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT\n",
      "    (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS\n",
      "    SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.\n",
      "----------------------------------------------------------------------\n",
      "    APBS uses FETK (the Finite Element ToolKit) to solve the\n",
      "    Poisson-Boltzmann equation numerically.  FETK is a portable collection\n",
      "    of finite element modeling class libraries developed by the Michael Holst\n",
      "    research group and written in an object-oriented form of C.  FEtk is\n",
      "    designed to solve general coupled systems of nonlinear partial differential\n",
      "    equations using adaptive finite element methods, inexact Newton methods,\n",
      "    and algebraic multilevel methods.  More information about FEtk may be found\n",
      "    at <http://www.FEtk.ORG>.\n",
      "----------------------------------------------------------------------\n",
      "    APBS also uses Aqua to solve the Poisson-Boltzmann equation numerically.  \n",
      "    Aqua is a modified form of the Holst group PMG library <http://www.FEtk.ORG>\n",
      "    which has been modified by Patrice Koehl\n",
      "    <http://koehllab.genomecenter.ucdavis.edu/> for improved efficiency and\n",
      "    memory usage when solving the Poisson-Boltzmann equation.\n",
      "----------------------------------------------------------------------\n",
      "    Please cite your use of APBS as:\n",
      "\n",
      "    Baker NA, Sept D, Joseph S, Holst MJ, McCammon JA. Electrostatics of\n",
      "    nanosystems: application to microtubules and the ribosome. Proc.\n",
      "    Natl. Acad. Sci. USA 98, 10037-10041 2001.\n",
      "    \n",
      "\n",
      "This executable compiled on May 29 2020 at 16:07:17\n",
      "\n",
      "Parsing input file 74987.in...\n",
      "rank 0 size 1...\n",
      "Parsed input file.\n",
      "Got paths for 1 molecules\n",
      "Reading PQR-format atom data from 74987.pqr.\n",
      "  5355 atoms\n",
      "  Centered at (-3.215e+01, -4.055e+01, -6.682e+00)\n",
      "  Net charge -1.30e+01 e\n",
      "Preparing to run 4 PBE calculations.\n",
      "----------------------------------------\n",
      "CALCULATION #1: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  938.273 MB total, 938.273 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 193 x 129 x 161\n",
      "  Grid spacings: 0.613 x 0.676 x 0.590\n",
      "  Grid lengths: 117.764 x 86.573 x 94.369\n",
      "  Grid center: (-32.145, -40.554, -6.682)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Single Debye-Huckel sphere boundary conditions\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 78.540\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 2.046968706799E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "CALCULATION #2: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  938.273 MB total, 1868.846 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 193 x 129 x 161\n",
      "  Grid spacings: 0.465 x 0.554 x 0.472\n",
      "  Grid lengths: 89.273 x 70.925 x 75.511\n",
      "  Grid center: (-32.145, -40.554, -6.682)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Boundary conditions from focusing\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 78.540\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Potential to be written to pot.dx\n",
      "  Total electrostatic energy = 3.140989698568E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "  Writing potential to pot.dx\n",
      "----------------------------------------\n",
      "CALCULATION #3: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  931.892 MB total, 1868.846 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 193 x 129 x 161\n",
      "  Grid spacings: 0.613 x 0.676 x 0.590\n",
      "  Grid lengths: 117.764 x 86.573 x 94.369\n",
      "  Grid center: (-32.145, -40.554, -6.682)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Single Debye-Huckel sphere boundary conditions\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 2.000\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 2.142802065494E+05 kJ/mol\n",
      "  Calculating forces...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[focusFillBound()]: WARNING:\n",
      "    Unusually large potential values\n",
      "    detected on the focusing boundary!\n",
      "    Convergence not guaranteed for NPBE/NRPBE calculations!\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "----------------------------------------\n",
      "CALCULATION #4: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  931.892 MB total, 1868.846 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 193 x 129 x 161\n",
      "  Grid spacings: 0.465 x 0.554 x 0.472\n",
      "  Grid lengths: 89.273 x 70.925 x 75.511\n",
      "  Grid center: (-32.145, -40.554, -6.682)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Boundary conditions from focusing\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 2.000\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 3.235314296487E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "PRINT STATEMENTS\n",
      "\n",
      "print energy 2 - 1 end\n",
      "  Local net energy (PE 0) = 9.432459791894E+03 kJ/mol\n",
      "  Global net ELEC energy = 9.432459791894E+03 kJ/mol\n",
      "----------------------------------------\n",
      "CLEANING UP AND SHUTTING DOWN...\n",
      "Destroying force arrays.\n",
      "No energy arrays to destroy.\n",
      "Destroying multigrid structures.\n",
      "Destroying finite element structures.\n",
      "Destroying 1 molecules\n",
      "Final memory usage:  0.001 MB total, 1868.846 MB high water\n",
      "\n",
      "\n",
      "Thanks for using APBS!\n",
      "\n",
      "Processing: pH 8.2\t1ka1_hal2.in\n",
      "\n",
      "\n",
      "----------------------------------------------------------------------\n",
      "    APBS -- Adaptive Poisson-Boltzmann Solver\n",
      "    Version 3.0\n",
      "    \n",
      "    Nathan A. Baker (nathan.baker@pnnl.gov)\n",
      "    Pacific Northwest National Laboratory\n",
      "    \n",
      "    Additional contributing authors listed in the code documentation.\n",
      "    \n",
      "    Copyright (c) 2010-2020 Battelle Memorial Institute. Developed at the Pacific\n",
      "    Northwest National Laboratory, operated by Battelle Memorial Institute, Pacific\n",
      "    Northwest Division for the U.S. Department of Energy.\n",
      "    \n",
      "    Portions Copyright (c) 2002-2010, Washington University in St. Louis.\n",
      "    Portions Copyright (c) 2002-2020, Nathan A. Baker.\n",
      "    Portions Copyright (c) 1999-2002, The Regents of the University of California.\n",
      "    Portions Copyright (c) 1995, Michael Holst.\n",
      "    All rights reserved.\n",
      "    \n",
      "    Redistribution and use in source and binary forms, with or without\n",
      "    modification, are permitted provided that the following conditions are met:\n",
      "    \n",
      "    * Redistributions of source code must retain the above copyright notice, this\n",
      "      list of conditions and the following disclaimer.\n",
      "    \n",
      "    * Redistributions in binary form must reproduce the above copyright notice,\n",
      "      this list of conditions and the following disclaimer in the documentation\n",
      "      and/or other materials provided with the distribution.\n",
      "    \n",
      "    * Neither the name of the developer nor the names of its contributors may be\n",
      "      used to endorse or promote products derived from this software without\n",
      "      specific prior written permission.\n",
      "    \n",
      "    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS \"AS IS\" AND\n",
      "    ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED\n",
      "    WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE\n",
      "    DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR\n",
      "    ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES\n",
      "    (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;\n",
      "    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND\n",
      "    ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT\n",
      "    (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS\n",
      "    SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.\n",
      "----------------------------------------------------------------------\n",
      "    APBS uses FETK (the Finite Element ToolKit) to solve the\n",
      "    Poisson-Boltzmann equation numerically.  FETK is a portable collection\n",
      "    of finite element modeling class libraries developed by the Michael Holst\n",
      "    research group and written in an object-oriented form of C.  FEtk is\n",
      "    designed to solve general coupled systems of nonlinear partial differential\n",
      "    equations using adaptive finite element methods, inexact Newton methods,\n",
      "    and algebraic multilevel methods.  More information about FEtk may be found\n",
      "    at <http://www.FEtk.ORG>.\n",
      "----------------------------------------------------------------------\n",
      "    APBS also uses Aqua to solve the Poisson-Boltzmann equation numerically.  \n",
      "    Aqua is a modified form of the Holst group PMG library <http://www.FEtk.ORG>\n",
      "    which has been modified by Patrice Koehl\n",
      "    <http://koehllab.genomecenter.ucdavis.edu/> for improved efficiency and\n",
      "    memory usage when solving the Poisson-Boltzmann equation.\n",
      "----------------------------------------------------------------------\n",
      "    Please cite your use of APBS as:\n",
      "\n",
      "    Baker NA, Sept D, Joseph S, Holst MJ, McCammon JA. Electrostatics of\n",
      "    nanosystems: application to microtubules and the ribosome. Proc.\n",
      "    Natl. Acad. Sci. USA 98, 10037-10041 2001.\n",
      "    \n",
      "\n",
      "This executable compiled on May 29 2020 at 16:07:17\n",
      "\n",
      "Parsing input file 1ka1_hal2.in...\n",
      "rank 0 size 1...\n",
      "Parsed input file.\n",
      "Got paths for 1 molecules\n",
      "Reading PQR-format atom data from 1ka1_hal2.pqr.\n",
      "  5456 atoms\n",
      "  Centered at (-3.270e+01, -4.262e+01, -4.458e+00)\n",
      "  Net charge -6.00e+00 e\n",
      "Preparing to run 4 PBE calculations.\n",
      "----------------------------------------\n",
      "CALCULATION #1: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  795.565 MB total, 795.565 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 161 x 129\n",
      "  Grid spacings: 0.687 x 0.572 x 0.663\n",
      "  Grid lengths: 109.929 x 91.584 x 84.857\n",
      "  Grid center: (-32.703, -42.625, -4.458)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Single Debye-Huckel sphere boundary conditions\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 78.540\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 2.007841772386E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "CALCULATION #2: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  795.565 MB total, 1583.939 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 161 x 129\n",
      "  Grid spacings: 0.529 x 0.462 x 0.546\n",
      "  Grid lengths: 84.664 x 73.873 x 69.916\n",
      "  Grid center: (-32.703, -42.625, -4.458)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Boundary conditions from focusing\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 78.540\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Potential to be written to pot.dx\n",
      "  Total electrostatic energy = 3.064614386616E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "  Writing potential to pot.dx\n",
      "----------------------------------------\n",
      "CALCULATION #3: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  789.712 MB total, 1583.939 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 161 x 129\n",
      "  Grid spacings: 0.687 x 0.572 x 0.663\n",
      "  Grid lengths: 109.929 x 91.584 x 84.857\n",
      "  Grid center: (-32.703, -42.625, -4.458)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Single Debye-Huckel sphere boundary conditions\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 2.000\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 2.082913994093E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "CALCULATION #4: MULTIGRID\n",
      "  Setting up problem...\n",
      "  Vpbe_ctor:  Using max ion radius (0 A) for exclusion function\n",
      "  Debye length:  0 A\n",
      "  Current memory usage:  789.712 MB total, 1583.939 MB high water\n",
      "  Using cubic spline charge discretization.\n",
      "  Grid dimensions: 161 x 161 x 129\n",
      "  Grid spacings: 0.529 x 0.462 x 0.546\n",
      "  Grid lengths: 84.664 x 73.873 x 69.916\n",
      "  Grid center: (-32.703, -42.625, -4.458)\n",
      "  Multigrid levels: 4\n",
      "  Molecule ID: 1\n",
      "  Linearized traditional PBE\n",
      "  Boundary conditions from focusing\n",
      "  0 ion species (0.000 M ionic strength):\n",
      "  Solute dielectric: 2.000\n",
      "  Solvent dielectric: 2.000\n",
      "  Using \"molecular\" surface definition;harmonic average smoothing\n",
      "  Solvent probe radius: 1.400 A\n",
      "  Temperature:  298.150 K\n",
      "  Electrostatic energies will be calculated\n",
      "  Total electrostatic energy = 3.138376812262E+05 kJ/mol\n",
      "  Calculating forces...\n",
      "----------------------------------------\n",
      "PRINT STATEMENTS\n",
      "\n",
      "print energy 2 - 1 end\n",
      "  Local net energy (PE 0) = 7.376242564653E+03 kJ/mol\n",
      "  Global net ELEC energy = 7.376242564653E+03 kJ/mol\n",
      "----------------------------------------\n",
      "CLEANING UP AND SHUTTING DOWN...\n",
      "Destroying force arrays.\n",
      "No energy arrays to destroy.\n",
      "Destroying multigrid structures.\n",
      "Destroying finite element structures.\n",
      "Destroying 1 molecules\n",
      "Final memory usage:  0.001 MB total, 1583.939 MB high water\n",
      "\n",
      "\n",
      "Thanks for using APBS!\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# assign directory\n",
    "ph = [(7.0,'7_0'), (7.3,'7_3'), (7.6,'7_6'), (7.9,'7_9'), (8.2,'8_2')]\n",
    "\n",
    "for i in ph:\n",
    "    # iterate over files in directory\n",
    "    directory = f'/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/pdb2pqr_pH{i[1]}'\n",
    "    os.chdir(f\"{directory}\")\n",
    "\n",
    "    for filename in os.listdir(directory):\n",
    "        basename = os.path.basename(filename)\n",
    "        f = os.path.join(directory, filename)\n",
    "        # checking if it is a file\n",
    "        if os.path.isfile(f):\n",
    "            if f.endswith(\".in\"):\n",
    "                shortname = basename.split('.')[0]\n",
    "                print(f'Processing: pH {i[0]}\\t{basename}')\n",
    "\n",
    "                # Runs apbs\n",
    "                os.system(f\"apbs {basename}\")\n",
    "                # Moves and renames file\n",
    "                os.system(f\"mv pot.dx {directory}/{shortname}.dx\")\n",
    "            else:\n",
    "                pass"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### PyMol - Generate and save images"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\"\"\"\n",
    "* Opens command line version of pymol\n",
    "* Loads structure and electrostatics files\n",
    "* Shows surface charge using a two-colour gradient. Two colours are used to ensure accurate \n",
    "machine-readable comparison of continuous data surface charge\n",
    "* Sets display preferences and saves ray-image\n",
    "\"\"\"\n",
    "\n",
    "\n",
    "import __main__\n",
    "__main__.pymol_argv = ['pymol', '-qc']\n",
    "\n",
    "import os\n",
    "import pymolPy3\n",
    "\n",
    "pm = pymolPy3.pymolPy3()\n",
    "pm(\"bg white\")\n",
    "\n",
    "# Import files inc AtSAL1\n",
    "# assign directory\n",
    "ph = [(7.0,'7_0'), (7.3,'7_3'), (7.6,'7_6'), (7.9,'7_9'), (8.2,'8_2')]\n",
    "\n",
    "for i in ph:\n",
    "    # Make dir\n",
    "    directory = f'/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/'\n",
    "    output_dir = f'{directory}pdb2pqr_pH{i[1]}/images_v2'\n",
    "    os.system(f\"mkdir {output_dir}\") \n",
    "\n",
    "    # iterate over files in directory\n",
    "    for filename in os.listdir(directory):\n",
    "        # checking if it is a file, create short name\n",
    "        f = os.path.join(directory, filename)\n",
    "        if os.path.isfile(f):\n",
    "            basename = os.path.basename(filename)\n",
    "            if basename.endswith('.pdb'):\n",
    "                shortname = basename.split('.')[0]\n",
    "\n",
    "                # set structure name\n",
    "                pdb = f\"{shortname}_pdb\"\n",
    "                dx = f\"{shortname}_dx\"\n",
    "\n",
    "                # load structure\n",
    "                pm(f\"load {f}\")\n",
    "                pm(f\"set_name {basename.split('.')[0]}, {pdb}\")\n",
    "\n",
    "                # load dx file\n",
    "                pm(f\"load {directory}pdb2pqr_pH{i[1]}/{shortname.split('.')[0]}.dx\")\n",
    "                pm(f\"set_name {shortname.split('.')[0]}, {dx}\")\n",
    "\n",
    "                pm(\"hide all\")\n",
    "\n",
    "                # Set view\n",
    "                pm(\"set_view (\\\n",
    "                0.783801198,   -0.439554513,    0.438685596,\\\n",
    "                -0.517596722,   -0.072069705,    0.852583110,\\\n",
    "                -0.343140632,   -0.895317495,   -0.284001738,\\\n",
    "                0.000009127,   -0.000110667, -110.469139099,\\\n",
    "                -41.782485962,  -34.828662872,   -5.124740601,\\\n",
    "                95.614555359,  118.323310852,  -20.000000000 )\")\n",
    "\n",
    "                pm(\"bg white\")\n",
    "\n",
    "                # Centre image\n",
    "                pm(\"center\")         \n",
    "\n",
    "                # Visualise surface charge\n",
    "                ramp_name = f'{shortname}_ramp'\n",
    "                map_name = f'{shortname}_dx'\n",
    "\n",
    "                pm(f\"ramp_new {ramp_name}, {map_name}, [-5, 0, 5], [blue, green]\")\n",
    "                pm(f\"set surface_ramp_above_mode, 1, {pdb}\")\n",
    "                pm(f\"set surface_color, {ramp_name}, {pdb}\")\n",
    "                pm(f\"show surface, {pdb}\")\n",
    "\n",
    "                pm(\"set max_ups, 0\")\n",
    "                pm(\"set use_shaders, 0\")\n",
    "                pm(\"set sphere_mode, 9\")\n",
    "                pm(\"set specular, 0.0\")\n",
    "                pm(\"set ray_shadow, 0\")\n",
    "\n",
    "\n",
    "                pm(\"ray\")\n",
    "                pm(f\"png {output_dir}/{shortname}\")\n",
    "                pm(\"reinitialize\")\n",
    "\n",
    "            else:\n",
    "                pass"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Cropping images"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import cv2\n",
    "from IPython.display import Image"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Image cropping function\n",
    "def crop_images(img, roi):\n",
    "    x_min, y_min, x_max, y_max = roi\n",
    "    return img[y_min:y_max, x_min:x_max]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Cropped image coordinates \n",
    "roi = (160, 72, 480, 260)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# iterates through file structure loading images, checking the file and then crops them using crop_images(). Writes to file\n",
    "\n",
    "ph = [(7.0,'7_0'), (7.3,'7_3'), (7.6,'7_6'), (7.9,'7_9'), (8.2,'8_2')]\n",
    "files = ['images_v2']\n",
    "\n",
    "for i in ph:\n",
    "    for f in files:\n",
    "        # Make dir\n",
    "        directory = f'/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/pdb2pqr_pH{i[1]}/{f}'\n",
    "        output_dir = f\"/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/pdb2pqr_pH{i[1]}/{f}/cropped\"\n",
    "        os.system(f\"mkdir {output_dir}\") \n",
    "\n",
    "        # iterate over files in directory\n",
    "        for filename in os.listdir(directory):\n",
    "            # checking if it is a file, create short name\n",
    "            full_file = os.path.join(directory, filename)\n",
    "            if os.path.isfile(full_file):\n",
    "                basename = os.path.basename(filename)\n",
    "                if basename.endswith('.png'):\n",
    "                    shortname = basename.split('.')[0]\n",
    "\n",
    "\n",
    "                    #load image\n",
    "                    img = cv2.imread(full_file, cv2.IMREAD_UNCHANGED)\n",
    "\n",
    "                    img_cropped = crop_images(img, roi)\n",
    "\n",
    "                    cv2.imwrite(f\"{output_dir}/{shortname}.png\", img_cropped)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Image comparison"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Move and rename images"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ph = [(7.0,'7_0'), (7.3,'7_3'), (7.6,'7_6'), (7.9,'7_9'), (8.2,'8_2')]\n",
    "\n",
    "count = 0\n",
    "\n",
    "for i in ph:\n",
    "    # Make dir\n",
    "    base_dir = f'/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/'\n",
    "    file_dir = f'{base_dir}pdb2pqr_pH{i[1]}/images_v2/cropped'\n",
    "    output_dir = f'{base_dir}/all_cropped/'\n",
    "\n",
    "    # iterate over files in directory\n",
    "    for filename in os.listdir(file_dir):\n",
    "        # checking if it is a file, create short name\n",
    "        full_file = os.path.join(file_dir, filename)\n",
    "        if os.path.isfile(full_file):\n",
    "            count += 1\n",
    "            basename = os.path.basename(filename)\n",
    "            if basename.endswith('.png'):\n",
    "                shortname = basename.split('.')[0]\n",
    "                # copy and rename\n",
    "                os.system(f\"cp {file_dir}/{filename} {output_dir}/{shortname}-pH_{i[1]}.png\")\n",
    "print(count)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Including spatial data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Image comparison using Earth Mover's Distance (EMD) aka Wasserstein Distance"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import cv2\n",
    "import os\n",
    "import pandas as pd\n",
    "from scipy.stats import wasserstein_distance\n",
    "from concurrent.futures import ThreadPoolExecutor, as_completed\n",
    "import itertools\n",
    "from tqdm import tqdm\n",
    "import sys\n",
    "import multiprocessing\n",
    "\n",
    "def load_image(path):\n",
    "    \"\"\"\n",
    "    Load an image from the given path.\n",
    "    \n",
    "    Args:\n",
    "    path (str): The path to the image file.\n",
    "    \n",
    "    Returns:\n",
    "    numpy.ndarray: The loaded image, or None if the image could not be loaded.\n",
    "    \"\"\"\n",
    "    img = cv2.imread(path, cv2.IMREAD_UNCHANGED)\n",
    "    if img is None:\n",
    "        print(f\"Error loading image: {path}\")\n",
    "        return None\n",
    "    return img\n",
    "\n",
    "def spatial_binning(image, bin_size=(8, 8)):\n",
    "    \"\"\"\n",
    "    Perform spatial binning on an image.\n",
    "    \n",
    "    Args:\n",
    "    image (numpy.ndarray): The image to be binned.\n",
    "    bin_size (tuple): The size of the bins (rows, columns).\n",
    "    \n",
    "    Returns:\n",
    "    numpy.ndarray: The flattened array of mean color values for each bin.\n",
    "    \"\"\"\n",
    "    height, width, _ = image.shape\n",
    "    bins_h = np.linspace(0, height, bin_size[0] + 1, dtype=int)\n",
    "    bins_w = np.linspace(0, width, bin_size[1] + 1, dtype=int)\n",
    "    color_distribution = []\n",
    "    for i in range(bin_size[0]):\n",
    "        for j in range(bin_size[1]):\n",
    "            bin_image = image[bins_h[i]:bins_h[i + 1], bins_w[j]:bins_w[j + 1], :]\n",
    "            color_distribution.append(np.mean(bin_image, axis=(0, 1)))\n",
    "    return np.array(color_distribution).flatten()\n",
    "\n",
    "def emd_distance(bins1, bins2):\n",
    "    \"\"\"\n",
    "    Calculate the Earth Mover's Distance (EMD) between two binned images.\n",
    "    \n",
    "    Args:\n",
    "    bins1 (numpy.ndarray): The first binned image.\n",
    "    bins2 (numpy.ndarray): The second binned image.\n",
    "    \n",
    "    Returns:\n",
    "    float: The normalized EMD score, or NaN if the range of bins1 is zero.\n",
    "    \"\"\"\n",
    "    range_bins1 = np.max(bins1) - np.min(bins1)\n",
    "    if range_bins1 == 0:\n",
    "        return np.nan  # Handle division by zero\n",
    "    distance = wasserstein_distance(bins1, bins2)\n",
    "    return distance / range_bins1\n",
    "\n",
    "def compare_two_images(data):\n",
    "    \"\"\"\n",
    "    Compare two images and calculate the EMD score between them.\n",
    "    \n",
    "    Args:\n",
    "    data (tuple): A tuple containing paths and keys for the two images.\n",
    "    \n",
    "    Returns:\n",
    "    tuple: A tuple containing details and the EMD score, or None if an image failed to load.\n",
    "    \"\"\"\n",
    "    path1, path2, key1, key2 = data\n",
    "    img1 = load_image(path1)\n",
    "    img2 = load_image(path2)\n",
    "    if img1 is None or img2 is None:\n",
    "        return None\n",
    "    bins1 = spatial_binning(img1)\n",
    "    bins2 = spatial_binning(img2)\n",
    "    emd_score = emd_distance(bins1, bins2)\n",
    "    id1, pH1 = parse_details(key1)\n",
    "    id2, pH2 = parse_details(key2)\n",
    "    return key1, id1, pH1, key2, id2, pH2, emd_score\n",
    "\n",
    "def process_images(file_dir):\n",
    "    \"\"\"\n",
    "    Process the images in a directory and create a dictionary of image paths.\n",
    "    \n",
    "    Args:\n",
    "    file_dir (str): The directory containing the images.\n",
    "    \n",
    "    Returns:\n",
    "    dict: A dictionary with image basenames as keys and full file paths as values.\n",
    "    \"\"\"\n",
    "    images = {}\n",
    "    for filename in os.listdir(file_dir):\n",
    "        full_file_path = os.path.join(file_dir, filename)\n",
    "        basename = os.path.splitext(filename)[0]\n",
    "        images[basename] = full_file_path\n",
    "    return images\n",
    "\n",
    "def parse_details(basename):\n",
    "    \"\"\"\n",
    "    Parse details from the image basename.\n",
    "    \n",
    "    Args:\n",
    "    basename (str): The basename of the image file.\n",
    "    \n",
    "    Returns:\n",
    "    tuple: A tuple containing the ID and pH value extracted from the basename.\n",
    "    \"\"\"\n",
    "    id_name, pH = basename.split('-')\n",
    "    pH = pH.lstrip('pH_')\n",
    "    return id_name, pH\n",
    "\n",
    "def main():\n",
    "    \"\"\"\n",
    "    Main function to process all image pairs and calculate their EMD scores.\n",
    "    \"\"\"\n",
    "    base_dir = '/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/'\n",
    "    file_dir = os.path.join(base_dir, 'all_cropped/')\n",
    "\n",
    "    # Process images and generate image pairs\n",
    "    images = process_images(file_dir)\n",
    "    image_pairs = list(itertools.combinations(images.items(), 2))\n",
    "\n",
    "    num_images = len(image_pairs)\n",
    "    if num_images == 0:\n",
    "        print(\"No image pairs found. Exiting.\")\n",
    "        sys.exit(1)\n",
    "\n",
    "    # Limiting the number of threads based on available CPU cores\n",
    "    num_cores = multiprocessing.cpu_count()\n",
    "    max_workers = min(num_cores * 2, len(image_pairs))\n",
    "    \n",
    "    # Use ThreadPoolExecutor for concurrent image processing\n",
    "    with ThreadPoolExecutor(max_workers=max_workers) as executor, tqdm(total=num_images, desc=\"Processing images\") as pbar:\n",
    "        results = []\n",
    "        futures = []\n",
    "        for pair in image_pairs:\n",
    "            futures.append(executor.submit(compare_two_images, (pair[0][1], pair[1][1], pair[0][0], pair[1][0])))\n",
    "        for future in as_completed(futures):\n",
    "            result = future.result()\n",
    "            if result is not None:\n",
    "                results.append(result)\n",
    "            pbar.update(1)\n",
    "\n",
    "    # Save results to a CSV file\n",
    "    df = pd.DataFrame(results, columns=['basename1', 'id1', 'pH1', 'basename2', 'id2', 'pH2', 'emd_score'])\n",
    "    output_file = os.path.join(base_dir, 'all-v-all_EMD_w_spatial_binning.csv')\n",
    "    df.to_csv(output_file, index=False)\n",
    "    print(f\"Results saved to {output_file}\")\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    main()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Merge results with taxonomic information"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "index_df = pd.read_csv(\"/home/mattm/projects/NP/wrk_index.csv\")\n",
    "\n",
    "index_df = index_df[['Unid', 'Classification', 'Brief_Classification', 'Species']]\n",
    "\n",
    "index_df.loc[len(index_df.index)] = ['AtSAL1updated', 'CoreEudicots/Rosids_Brassicales', 'CoreEudicots/Rosids', 'Arabidopsis_thaliana'] \n",
    "\n",
    "results_df = pd.read_csv('/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/all-v-all_EMD_w_spatial_binning.csv', low_memory=False)\n",
    "\n",
    "detailed_results_df = results_df.merge(index_df, left_on='id1', right_on='Unid', how='outer' )\n",
    "\n",
    "base_dir = '/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/'\n",
    "output_file = os.path.join(base_dir, 'all-v-all_EMD_w_spatial_binning_w_tax.csv')\n",
    "\n",
    "detailed_results_df.to_csv(output_file, index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 3D Active site isolation and comparison"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Not for publication - only provide active site prism corners"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Code used to determine 3D coordinates of active site\n",
    "\n",
    "import __main__\n",
    "__main__.pymol_argv = ['pymol', '-qc']\n",
    "\n",
    "import os\n",
    "import pymolPy3\n",
    "\n",
    "pm = pymolPy3.pymolPy3()\n",
    "\n",
    "AtSAL1_pdb = '/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/AtSAL1_updated/AtSAL1_updated.pdb'\n",
    "\n",
    "\n",
    "\"\"\"\n",
    "Structures need to be aligned BEFORE electrostatics\n",
    "\"\"\"\n",
    "\n",
    "# Load AtSAL1 reference\n",
    "pm(f\"load {AtSAL1_pdb}\")\n",
    "pm('set_name AtSAL1_updated, AtSAL1_updated_pdb')\n",
    "\n",
    "pm(\"set_view (\\\n",
    "0.980755210,   -0.023148270,   -0.193854913,\\\n",
    "0.162131056,   -0.456586570,    0.874780297,\\\n",
    "-0.108762056,   -0.889374375,   -0.444051415,\\\n",
    "0.000122931,   -0.000242302,  -56.248649597,\\\n",
    "-32.444423676,  -39.579414368,   -9.259656906,\\\n",
    "44.643768311,   67.851081848,  -20.000000000 )\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Create a cube to isolate the 3D active site space\n",
    "## Not for publication"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Bottom plane\n",
    "pm(f'pseudoatom corner1, pos=[-43.55064101, -41.4898718,  -12.31982906]')\n",
    "pm(f'pseudoatom corner2, pos=[-26, -45, -17]')\n",
    "pm(f'pseudoatom corner3, pos=[-21.12, -35.24, -5.29]')\n",
    "pm(f'pseudoatom corner4, pos=[-38.67295493, -31.73449962,  -0.61338245]')\n",
    "# Top plane\n",
    "pm(f'pseudoatom corner5, pos=[-43.8596559 , -26.06520805, -25.0471327]')\n",
    "pm(f'pseudoatom corner6, pos=[-26.30901489, -29.57533625, -29.72730364]')\n",
    "pm(f'pseudoatom corner7, pos=[-21.42901489, -19.81533625, -18.01730364]')\n",
    "pm(f'pseudoatom corner8, pos=[-38.98196982, -16.30983587, -13.34068609]')\n",
    "\n",
    "pm(f'distance box_edge, corner1, corner2')\n",
    "pm(f'distance box_edge, corner2, corner3')\n",
    "pm(f'distance box_edge, corner3, corner4')\n",
    "pm(f'distance box_edge, corner4, corner1')\n",
    "pm(f'distance box_edge, corner5, corner6')\n",
    "pm(f'distance box_edge, corner6, corner7')\n",
    "pm(f'distance box_edge, corner7, corner8')\n",
    "pm(f'distance box_edge, corner8, corner5')\n",
    "pm(f'distance box_edge, corner1, corner5')\n",
    "pm(f'distance box_edge, corner2, corner6')\n",
    "pm(f'distance box_edge, corner3, corner7')\n",
    "pm(f'distance box_edge, corner4, corner8')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "corner1 = [-43.55064101, -41.4898718,  -12.31982906]\n",
    "corner2 = [-26, -45, -17]\n",
    "corner3 = [-21.12, -35.24, -5.29]\n",
    "corner4 = [-38.67295493, -31.73449962,  -0.61338245]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Not for publication"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Find the coordinates of the top plane\n",
    "import numpy as np\n",
    "\n",
    "distance_above = -40\n",
    "bottom_corners = np.array([corner1, corner2, corner3, corner4])\n",
    "\n",
    "normal_vector = np.cross(bottom_corners[1]-bottom_corners[0], bottom_corners[2]-bottom_corners[0])\n",
    "normal_vector = normal_vector.astype(float)  # Cast to float to ensure consistent data types\n",
    "normal_vector /= np.linalg.norm(normal_vector)\n",
    "\n",
    "top_corners = bottom_corners + distance_above * normal_vector\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Not for publication"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import __main__\n",
    "__main__.pymol_argv = ['pymol', '-qc']\n",
    "\n",
    "import os\n",
    "import pymolPy3\n",
    "\n",
    "pm = pymolPy3.pymolPy3()\n",
    "pm(\"bg white\")\n",
    "\n",
    "# Import files inc AtSAL1\n",
    "# assign directory\n",
    "\n",
    "# Make dir\n",
    "directory = '/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/AtSAL1_updated/'\n",
    "\n",
    "# set structure name\n",
    "pdb = f\"AtSAL1_updated\"\n",
    "dx = f\"AtSAL1_test\"\n",
    "\n",
    "f = 'AtSAL1_updated'\n",
    "dx = 'AtSAL1_test'\n",
    "\n",
    "# load structure\n",
    "pm(f\"load {directory}{f}.pdb\")\n",
    "\n",
    "# load dx file\n",
    "pm(f\"load {directory}{dx}.dx\")\n",
    "\n",
    "pm(\"hide all\")\n",
    "\n",
    "# Set view\n",
    "pm(\"set_view (\\\n",
    "0.783801198,   -0.439554513,    0.438685596,\\\n",
    "-0.517596722,   -0.072069705,    0.852583110,\\\n",
    "-0.343140632,   -0.895317495,   -0.284001738,\\\n",
    "0.000009127,   -0.000110667, -110.469139099,\\\n",
    "-41.782485962,  -34.828662872,   -5.124740601,\\\n",
    "95.614555359,  118.323310852,  -20.000000000 )\")\n",
    "\n",
    "# Centre image\n",
    "pm(\"center\")         \n",
    "\n",
    "# Visualise surface charge\n",
    "ramp_name = f'{dx}_ramp'\n",
    "map_name = f'{dx}'\n",
    "\n",
    "pm(f\"ramp_new {ramp_name}, {map_name}, [-5, 5], [blue, green]\")\n",
    "pm(f\"set surface_ramp_above_mode, 1, {pdb}\")\n",
    "pm(f\"set surface_color, {ramp_name}, {pdb}\")\n",
    "pm(f\"show surface, {pdb}\")\n",
    "\n",
    "pm(\"set max_ups, 0\")\n",
    "pm(\"set use_shaders, 0\")\n",
    "pm(\"set sphere_mode, 9\")\n",
    "pm(\"set specular, 0.0\")\n",
    "pm(\"set ray_shadow, 0\")\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Fills the rest of the grid with 0s to verify the above works"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Not for publication - only show resulting image"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "from scipy.spatial import Delaunay\n",
    "\n",
    "def parse_dx_file(filename, prism_corners):\n",
    "    \"\"\"\n",
    "    Parses a DX file to isolate a prism volume in the grid based on corner coordinates.\n",
    "\n",
    "    Args:\n",
    "        filename (str): Path to the DX file.\n",
    "        prism_corners (list): List of corner coordinates as NumPy arrays (shape: (3,)).\n",
    "\n",
    "    Returns:\n",
    "        tuple: (prism_data, origin, delta)\n",
    "          - prism_data (numpy.ndarray): Prism data subset from the original grid.\n",
    "          - origin (list): Origin of the grid (shape: (3,)).\n",
    "          - delta (list): Grid spacing along each axis (shape: (3,)).\n",
    "    \"\"\"\n",
    "    def parse_grid_data(lines):\n",
    "        \"\"\"\n",
    "        Parses the grid data section within the object containing \"positions\".\n",
    "\n",
    "        Args:\n",
    "            lines (list): List of lines from the DX file containing grid data.\n",
    "\n",
    "        Returns:\n",
    "            numpy.ndarray: Grid data as a NumPy array.\n",
    "        \"\"\"\n",
    "        data = []\n",
    "        for line in lines:\n",
    "            if line.strip().startswith('attribute'):\n",
    "                break\n",
    "            values = list(map(float, line.split()))\n",
    "            data.extend(values)\n",
    "        return np.array(data)\n",
    "\n",
    "    def point_in_prism(point, delaunay):\n",
    "        \"\"\"\n",
    "        Check if a point is inside the prism using Delaunay triangulation.\n",
    "\n",
    "        Args:\n",
    "            point (np.ndarray): The point to check.\n",
    "            delaunay (scipy.spatial.Delaunay): Delaunay triangulation object.\n",
    "\n",
    "        Returns:\n",
    "            bool: True if the point is inside the prism, False otherwise.\n",
    "        \"\"\"\n",
    "        return delaunay.find_simplex(point) >= 0\n",
    "\n",
    "    def isolate_prism_data(grid_data, origin, delta, prism_corners):\n",
    "        \"\"\"\n",
    "        Extracts the data for a prism defined by corner coordinates within the grid.\n",
    "\n",
    "        Args:\n",
    "            grid_data (numpy.ndarray): Full grid data.\n",
    "            origin (np.ndarray): Origin of the grid.\n",
    "            delta (np.ndarray): Grid spacing along each axis.\n",
    "            prism_corners (list): Corner coordinates of the prism.\n",
    "\n",
    "        Returns:\n",
    "            numpy.ndarray: Data from the grid corresponding to the prism volume.\n",
    "        \"\"\"\n",
    "        dimensions = grid_data.shape\n",
    "        prism_data = np.zeros_like(grid_data)\n",
    "\n",
    "        # Calculate the bounding box in real-world coordinates\n",
    "        bbox_min = np.min(prism_corners, axis=0)\n",
    "        bbox_max = np.max(prism_corners, axis=0)\n",
    "\n",
    "        # Calculate the grid indices for the bounding box\n",
    "        grid_min = np.floor((bbox_min - origin) / np.diag(delta)).astype(int)\n",
    "        grid_max = np.ceil((bbox_max - origin) / np.diag(delta)).astype(int)\n",
    "\n",
    "        # Clip the indices to the grid dimensions\n",
    "        grid_min = np.clip(grid_min, 0, np.array(grid_data.shape) - 1)\n",
    "        grid_max = np.clip(grid_max, 0, np.array(grid_data.shape) - 1)\n",
    "\n",
    "        # Generate grid points within the bounding box\n",
    "        grid_points = np.mgrid[\n",
    "            grid_min[0]:grid_max[0]+1,\n",
    "            grid_min[1]:grid_max[1]+1,\n",
    "            grid_min[2]:grid_max[2]+1\n",
    "        ].reshape(3, -1).T\n",
    "\n",
    "        # Convert grid points to real-world coordinates\n",
    "        world_points = origin + grid_points * np.diag(delta)\n",
    "\n",
    "        # Create Delaunay triangulation for the prism corners\n",
    "        delaunay = Delaunay(prism_corners)\n",
    "\n",
    "        # Check points inside the prism using Delaunay\n",
    "        mask = np.array([point_in_prism(point, delaunay) for point in world_points])\n",
    "\n",
    "        mask = mask.reshape(\n",
    "            grid_max[0] - grid_min[0] + 1,\n",
    "            grid_max[1] - grid_min[1] + 1,\n",
    "            grid_max[2] - grid_min[2] + 1\n",
    "        )\n",
    "\n",
    "        prism_data[\n",
    "            grid_min[0]:grid_max[0]+1,\n",
    "            grid_min[1]:grid_max[1]+1,\n",
    "            grid_min[2]:grid_max[2]+1\n",
    "        ][mask] = grid_data[\n",
    "            grid_min[0]:grid_max[0]+1,\n",
    "            grid_min[1]:grid_max[1]+1,\n",
    "            grid_min[2]:grid_max[2]+1\n",
    "        ][mask]\n",
    "\n",
    "        return prism_data\n",
    "\n",
    "    def write_dx_file(filename, prism_data, origin, delta):\n",
    "        \"\"\"\n",
    "        Writes prism data into a DX file in the same format as the provided example.\n",
    "\n",
    "        Args:\n",
    "            filename (str): Path to the DX file to write\n",
    "        \n",
    "        Writes prism data into a DX file in the same format as the provided example.\n",
    "\n",
    "        Args:\n",
    "            filename (str): Path to the DX file to write.\n",
    "            prism_data (numpy.ndarray): Prism data subset from the original grid.\n",
    "            origin (list): Origin of the grid (shape: (3,)).\n",
    "            delta (list): Grid spacing along each axis (shape: (3,)).\n",
    "        \"\"\"\n",
    "        with open(filename, 'w') as f:\n",
    "            # Write header information\n",
    "            f.write(\"# Data from 3.0\\n\")\n",
    "            f.write(\"# \\n\")\n",
    "            f.write(\"# POTENTIAL (kT/e)\\n\")\n",
    "            f.write(\"# \\n\")\n",
    "            f.write(\"object 1 class gridpositions counts {} {} {}\\n\".format(*prism_data.shape))\n",
    "            f.write(f\"origin {origin[0]:.6e} {origin[1]:.6e} {origin[2]:.6e}\\n\")\n",
    "            f.write(f\"delta {delta[0][0]:.6e} {delta[0][1]:.6e} {delta[0][2]:.6e}\\n\")\n",
    "            f.write(f\"delta {delta[1][0]:.6e} {delta[1][1]:.6e} {delta[1][2]:.6e}\\n\")\n",
    "            f.write(f\"delta {delta[2][0]:.6e} {delta[2][1]:.6e} {delta[2][2]:.6e}\\n\")\n",
    "            f.write(\"object 2 class gridconnections counts {} {} {}\\n\".format(*prism_data.shape))\n",
    "            f.write(\"object 3 class array type double rank 0 items {} data follows\\n\".format(np.prod(prism_data.shape)))\n",
    "\n",
    "            # Write prism data\n",
    "            num_values = prism_data.size\n",
    "            num_written = 0\n",
    "            for i in range(prism_data.shape[0]):\n",
    "                for j in range(prism_data.shape[1]):\n",
    "                    for k in range(prism_data.shape[2]):\n",
    "                        f.write(\"{:.6e} \".format(prism_data[i, j, k]))\n",
    "                        num_written += 1\n",
    "                        # Insert newline after every three values except for the last line\n",
    "                        if num_written % 3 == 0 and num_written < num_values:\n",
    "                            f.write(\"\\n\")\n",
    "            f.write(\"\\n\")\n",
    "\n",
    "            # Write footer\n",
    "            f.write(\"attribute \\\"dep\\\" string \\\"positions\\\"\\n\")\n",
    "            f.write(\"object \\\"regular positions regular connections\\\" class field\\n\")\n",
    "            f.write(\"component \\\"positions\\\" value 1\\n\")\n",
    "            f.write(\"component \\\"connections\\\" value 2\\n\")\n",
    "            f.write(\"component \\\"data\\\" value 3\\n\")\n",
    "\n",
    "    with open(filename, 'r') as file:\n",
    "        lines = file.readlines()\n",
    "\n",
    "    grid_data = None\n",
    "    origin = None\n",
    "    delta = np.zeros((3, 3))\n",
    "    dimensions = None\n",
    "    delta_index = 0\n",
    "\n",
    "    for line in lines:\n",
    "        if line.startswith('object 1 class gridpositions counts'):\n",
    "            dimensions = np.array(list(map(int, line.split()[5:8])))\n",
    "        elif line.startswith('origin'):\n",
    "            origin = np.array(list(map(float, line.split()[1:4])))\n",
    "        elif line.startswith('delta'):\n",
    "            values = list(map(float, line.split()[1:]))\n",
    "            delta[delta_index, :len(values)] = values\n",
    "            delta_index += 1\n",
    "            if delta_index == 3:\n",
    "                break\n",
    "\n",
    "    data_start_line = next(i for i, line in enumerate(lines) if line.startswith('object 3 class array type'))\n",
    "    data_lines = lines[data_start_line+1:]\n",
    "    grid_data = parse_grid_data(data_lines)\n",
    "    grid_data = grid_data.reshape(dimensions)\n",
    "\n",
    "    prism_data = isolate_prism_data(grid_data, origin, delta, prism_corners)\n",
    "\n",
    "    write_dx_file(\"/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/AtSAL1_test.dx\", prism_data, origin, delta)\n",
    "\n",
    "    return prism_data, origin, delta\n",
    "\n",
    "# Prism corners\n",
    "prism_corners = np.array([\n",
    "    [-43.55064101, -41.4898718, -12.31982906],\n",
    "    [-26, -45, -17],\n",
    "    [-21.12, -35.24, -5.29],\n",
    "    [-38.67295493, -31.73449962, -0.61338245],\n",
    "    [-43.8596559, -26.06520805, -25.0471327],\n",
    "    [-26.30901489, -29.57533625, -29.72730364],\n",
    "    [-21.42901489, -19.81533625, -18.01730364],\n",
    "    [-38.98196982, -16.30983587, -13.34068609]\n",
    "])\n",
    "\n",
    "filename = '/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/AtSAL1_updated/pdb2pqr_pH8_2/AtSAL1_updated.dx'\n",
    "prism_data, origin, delta = parse_dx_file(filename, prism_corners)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Generate image of above to show it works"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import __main__\n",
    "__main__.pymol_argv = ['pymol', '-qc']\n",
    "\n",
    "import os\n",
    "import pymolPy3\n",
    "\n",
    "pm = pymolPy3.pymolPy3()\n",
    "pm(\"bg white\")\n",
    "\n",
    "# Import files inc AtSAL1\n",
    "# assign directory\n",
    "\n",
    "# Make dir\n",
    "directory = '/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/AtSAL1_updated/'\n",
    "\n",
    "# set structure name\n",
    "pdb = f\"AtSAL1_updated\"\n",
    "\n",
    "f = 'AtSAL1_updated'\n",
    "dx = 'AtSAL1_test2'\n",
    "\n",
    "# load structure\n",
    "pm(f\"load {directory}{pdb}.pdb\")\n",
    "\n",
    "# load dx file\n",
    "pm(f\"load {directory}{dx}.dx\")\n",
    "\n",
    "pm(\"hide all\")\n",
    "\n",
    "# Set view\n",
    "pm(\"set_view (\\\n",
    "0.783801198,   -0.439554513,    0.438685596,\\\n",
    "-0.517596722,   -0.072069705,    0.852583110,\\\n",
    "-0.343140632,   -0.895317495,   -0.284001738,\\\n",
    "0.000009127,   -0.000110667, -110.469139099,\\\n",
    "-41.782485962,  -34.828662872,   -5.124740601,\\\n",
    "95.614555359,  118.323310852,  -20.000000000 )\")\n",
    "\n",
    "# Centre image\n",
    "pm(\"center\")         \n",
    "\n",
    "# Visualise surface charge\n",
    "ramp_name = f'{dx}_ramp'\n",
    "map_name = f'{dx}'\n",
    "\n",
    "pm(f\"ramp_new {ramp_name}, {map_name}, [-5, 5], [blue, green]\")\n",
    "pm(f\"set surface_ramp_above_mode, 1, {pdb}\")\n",
    "pm(f\"set surface_color, {ramp_name}, {pdb}\")\n",
    "pm(f\"show surface, {pdb}\")\n",
    "\n",
    "pm(\"set max_ups, 0\")\n",
    "pm(\"set use_shaders, 0\")\n",
    "pm(\"set sphere_mode, 9\")\n",
    "pm(\"set specular, 0.0\")\n",
    "pm(\"set ray_shadow, 0\")\n",
    "# Bottom plane\n",
    "pm(f'pseudoatom corner1, pos=[-43.55064101, -41.4898718,  -12.31982906]')\n",
    "pm(f'pseudoatom corner2, pos=[-26, -45, -17]')\n",
    "pm(f'pseudoatom corner3, pos=[-21.12, -35.24, -5.29]')\n",
    "pm(f'pseudoatom corner4, pos=[-38.67295493, -31.73449962,  -0.61338245]')\n",
    "# Top plane\n",
    "pm(f'pseudoatom corner5, pos=[-43.8596559 , -26.06520805, -25.0471327]')\n",
    "pm(f'pseudoatom corner6, pos=[-26.30901489, -29.57533625, -29.72730364]')\n",
    "pm(f'pseudoatom corner7, pos=[-21.42901489, -19.81533625, -18.01730364]')\n",
    "pm(f'pseudoatom corner8, pos=[-38.98196982, -16.30983587, -13.34068609]')\n",
    "\n",
    "pm(f'distance box_edge, corner1, corner2')\n",
    "pm(f'distance box_edge, corner2, corner3')\n",
    "pm(f'distance box_edge, corner3, corner4')\n",
    "pm(f'distance box_edge, corner4, corner1')\n",
    "pm(f'distance box_edge, corner5, corner6')\n",
    "pm(f'distance box_edge, corner6, corner7')\n",
    "pm(f'distance box_edge, corner7, corner8')\n",
    "pm(f'distance box_edge, corner8, corner5')\n",
    "pm(f'distance box_edge, corner1, corner5')\n",
    "pm(f'distance box_edge, corner2, corner6')\n",
    "pm(f'distance box_edge, corner3, corner7')\n",
    "pm(f'distance box_edge, corner4, corner8')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Same as above, but just removes points outside of the prism rather than make them zero"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Crops the .dx files in 3D"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "from scipy.spatial import Delaunay\n",
    "import os\n",
    "import concurrent.futures\n",
    "\n",
    "def parse_dx_file(filename, prism_corners, basename, output_dir):\n",
    "    \"\"\"\n",
    "    Parses a DX file to isolate a prism volume in the grid based on corner coordinates\n",
    "    and writes the output to a new DX file.\n",
    "\n",
    "    Args:\n",
    "        filename (str): Path to the DX file.\n",
    "        prism_corners (list): List of corner coordinates as NumPy arrays (shape: (3,)).\n",
    "        basename (str): Base name for the output file.\n",
    "        output_dir (str): Directory where the output file will be saved.\n",
    "\n",
    "    Returns:\n",
    "        None\n",
    "    \"\"\"\n",
    "    \n",
    "    def parse_grid_data(lines):\n",
    "        \"\"\"\n",
    "        Parses the grid data section within the object containing \"positions\".\n",
    "\n",
    "        Args:\n",
    "            lines (list): List of lines from the DX file containing grid data.\n",
    "\n",
    "        Returns:\n",
    "            numpy.ndarray: Grid data as a NumPy array.\n",
    "        \"\"\"\n",
    "        data = []\n",
    "        for line in lines:\n",
    "            if line.strip().startswith('attribute'):\n",
    "                break\n",
    "            values = list(map(float, line.split()))\n",
    "            data.extend(values)\n",
    "        return np.array(data)\n",
    "\n",
    "    def point_in_prism(point, delaunay):\n",
    "        \"\"\"\n",
    "        Checks if a point is inside the prism using Delaunay triangulation.\n",
    "\n",
    "        Args:\n",
    "            point (np.ndarray): The point to check.\n",
    "            delaunay (scipy.spatial.Delaunay): Delaunay triangulation object.\n",
    "\n",
    "        Returns:\n",
    "            bool: True if the point is inside the prism, False otherwise.\n",
    "        \"\"\"\n",
    "        return delaunay.find_simplex(point) >= 0\n",
    "\n",
    "    def isolate_prism_data(grid_data, origin, delta, prism_corners):\n",
    "        \"\"\"\n",
    "        Extracts the data for a prism defined by corner coordinates within the grid.\n",
    "\n",
    "        Args:\n",
    "            grid_data (numpy.ndarray): Full grid data.\n",
    "            origin (np.ndarray): Origin of the grid.\n",
    "            delta (np.ndarray): Grid spacing along each axis.\n",
    "            prism_corners (list): Corner coordinates of the prism.\n",
    "\n",
    "        Returns:\n",
    "            tuple: (inside_points, inside_values, grid_min, grid_max)\n",
    "        \"\"\"\n",
    "        bbox_min = np.min(prism_corners, axis=0)\n",
    "        bbox_max = np.max(prism_corners, axis=0)\n",
    "\n",
    "        grid_min = np.floor((bbox_min - origin) / np.diag(delta)).astype(int)\n",
    "        grid_max = np.ceil((bbox_max - origin) / np.diag(delta)).astype(int)\n",
    "\n",
    "        grid_min = np.clip(grid_min, 0, np.array(grid_data.shape) - 1)\n",
    "        grid_max = np.clip(grid_max, 0, np.array(grid_data.shape) - 1)\n",
    "\n",
    "        grid_points = np.mgrid[\n",
    "            grid_min[0]:grid_max[0]+1,\n",
    "            grid_min[1]:grid_max[1]+1,\n",
    "            grid_min[2]:grid_max[2]+1\n",
    "        ].reshape(3, -1).T\n",
    "\n",
    "        world_points = origin + grid_points * np.diag(delta)\n",
    "\n",
    "        delaunay = Delaunay(prism_corners)\n",
    "\n",
    "        mask = np.array([point_in_prism(point, delaunay) for point in world_points])\n",
    "\n",
    "        inside_points = grid_points[mask]\n",
    "        inside_values = grid_data[\n",
    "            grid_min[0]:grid_max[0]+1,\n",
    "            grid_min[1]:grid_max[1]+1,\n",
    "            grid_min[2]:grid_max[2]+1\n",
    "        ].flatten()[mask]\n",
    "\n",
    "        return inside_points, inside_values\n",
    "\n",
    "    def write_dx_file(filename, inside_points, inside_values, origin, delta):\n",
    "        \"\"\"\n",
    "        Writes prism data into a DX file in the same format as the provided example.\n",
    "\n",
    "        Args:\n",
    "            filename (str): Path to the DX file to write.\n",
    "            inside_points (np.ndarray): Points inside the prism.\n",
    "            inside_values (np.ndarray): Values at the points inside the prism.\n",
    "            origin (list): Origin of the grid (shape: (3,)).\n",
    "            delta (list): Grid spacing along each axis (shape: (3,)).\n",
    "        \"\"\"\n",
    "        min_indices = np.min(inside_points, axis=0)\n",
    "        max_indices = np.max(inside_points, axis=0)\n",
    "        counts = max_indices - min_indices + 1\n",
    "\n",
    "        with open(filename, 'w') as f:\n",
    "            f.write(\"# Data from 3.0\\n\")\n",
    "            f.write(\"# \\n\")\n",
    "            f.write(\"# POTENTIAL (kT/e)\\n\")\n",
    "            f.write(\"# \\n\")\n",
    "            f.write(\"object 1 class gridpositions counts {} {} {}\\n\".format(*counts))\n",
    "            f.write(f\"origin {origin[0]:.6e} {origin[1]:.6e} {origin[2]:.6e}\\n\")\n",
    "            for i in range(3):\n",
    "                f.write(f\"delta {delta[i][0]:.6e} {delta[i][1]:.6e} {delta[i][2]:.6e}\\n\")\n",
    "            f.write(\"object 2 class gridconnections counts {} {} {}\\n\".format(*counts))\n",
    "            f.write(\"object 3 class array type double rank 0 items {} data follows\\n\".format(inside_values.size))\n",
    "\n",
    "            for i, value in enumerate(inside_values):\n",
    "                f.write(\"{:.6e} \".format(value))\n",
    "                if (i + 1) % 3 == 0 and (i + 1) < inside_values.size:\n",
    "                    f.write(\"\\n\")\n",
    "            f.write(\"\\n\")\n",
    "\n",
    "            f.write(\"attribute \\\"dep\\\" string \\\"positions\\\"\\n\")\n",
    "            f.write(\"object \\\"regular positions regular connections\\\" class field\\n\")\n",
    "            f.write(\"component \\\"positions\\\" value 1\\n\")\n",
    "            f.write(\"component \\\"connections\\\" value 2\\n\")\n",
    "            f.write(\"component \\\"data\\\" value 3\\n\")\n",
    "\n",
    "    # Read DX file lines\n",
    "    with open(filename, 'r') as file:\n",
    "        lines = file.readlines()\n",
    "\n",
    "    # Initialize variables\n",
    "    origin = None\n",
    "    delta = np.zeros((3, 3))\n",
    "    dimensions = None\n",
    "    delta_index = 0\n",
    "\n",
    "    # Parse file for grid metadata\n",
    "    for line in lines:\n",
    "        if line.startswith('object 1 class gridpositions counts'):\n",
    "            dimensions = np.array(list(map(int, line.split()[5:8])))\n",
    "        elif line.startswith('origin'):\n",
    "            origin = np.array(list(map(float, line.split()[1:4])))\n",
    "        elif line.startswith('delta'):\n",
    "            values = list(map(float, line.split()[1:]))\n",
    "            delta[delta_index, :len(values)] = values\n",
    "            delta_index += 1\n",
    "            if delta_index == 3:\n",
    "                break\n",
    "\n",
    "    # Extract grid data\n",
    "    data_start_line = next(i for i, line in enumerate(lines) if line.startswith('object 3 class array type'))\n",
    "    data_lines = lines[data_start_line + 1:]\n",
    "    grid_data = parse_grid_data(data_lines).reshape(dimensions)\n",
    "\n",
    "    # Isolate prism data\n",
    "    inside_points, inside_values = isolate_prism_data(grid_data, origin, delta, prism_corners)\n",
    "\n",
    "    # Write the isolated data to a new DX file\n",
    "    output_filepath = os.path.join(output_dir, basename)\n",
    "    write_dx_file(output_filepath, inside_points, inside_values, origin, delta)\n",
    "\n",
    "def process_directory(directory, prism_corners):\n",
    "    \"\"\"\n",
    "    Processes all .dx files in the specified directory in parallel.\n",
    "    \n",
    "    Args:\n",
    "        directory (str): Directory containing .dx files.\n",
    "        prism_corners (np.ndarray): Array of prism corners for data extraction.\n",
    "    \"\"\"\n",
    "    output_dir = os.path.join(directory, \"cropped\")\n",
    "    os.makedirs(output_dir, exist_ok=True)\n",
    "    \n",
    "    # Collect all .dx files\n",
    "    dx_files = [f for f in os.listdir(directory) if f.endswith('.dx')]\n",
    "    \n",
    "    # Use concurrent processing\n",
    "    with concurrent.futures.ProcessPoolExecutor() as executor:\n",
    "        # Submit tasks\n",
    "        futures = [executor.submit(parse_dx_file, os.path.join(directory, f), prism_corners, f, output_dir) for f in dx_files]\n",
    "        # Optionally handle exceptions and retrieve results\n",
    "        for future in concurrent.futures.as_completed(futures):\n",
    "            try:\n",
    "                future.result()  # If needed, do something with the results or just ensure all tasks complete\n",
    "            except Exception as exc:\n",
    "                print(f'File processing generated an exception: {exc}')\n",
    "\n",
    "# Define prism corners\n",
    "prism_corners = np.array([\n",
    "    [-43.55064101, -41.4898718, -12.31982906],\n",
    "    [-26, -45, -17],\n",
    "    [-21.12, -35.24, -5.29],\n",
    "    [-38.67295493, -31.73449962, -0.61338245],\n",
    "    [-43.8596559, -26.06520805, -25.0471327],\n",
    "    [-26.30901489, -29.57533625, -29.72730364],\n",
    "    [-21.42901489, -19.81533625, -18.01730364],\n",
    "    [-38.98196982, -16.30983587, -13.34068609]\n",
    "])\n",
    "\n",
    "# Define pH levels and corresponding directory suffixes\n",
    "ph_levels = [(7.0, '7_0'), (7.3, '7_3'), (7.6, '7_6'), (7.9, '7_9'), (8.2, '8_2')]\n",
    "\n",
    "# Process each pH level\n",
    "for pH, suffix in ph_levels:\n",
    "    directory = f'/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/hal2p/pdb2pqr_pH{suffix}/'\n",
    "    process_directory(directory, prism_corners)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "\n",
    "# Define pH conditions with their corresponding folder names\n",
    "ph_conditions = [(7.0, '7_0'), (7.3, '7_3'), (7.6, '7_6'), (7.9, '7_9'), (8.2, '8_2')]\n",
    "\n",
    "# Initialize a counter to keep track of the number of files processed\n",
    "file_count = 0\n",
    "\n",
    "# Base directory for project files\n",
    "base_dir = '/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/hal2p/'\n",
    "\n",
    "# Loop over each pH condition to process files\n",
    "for ph_value, folder_suffix in ph_conditions:\n",
    "    input_dir = f'{base_dir}pdb2pqr_pH{folder_suffix}/cropped'\n",
    "    output_dir = f'{base_dir}all_cropped/'\n",
    "\n",
    "    # Ensure the output directory exists\n",
    "    os.makedirs(output_dir, exist_ok=True)\n",
    "\n",
    "    # Iterate over all files in the input directory\n",
    "    for filename in os.listdir(input_dir):\n",
    "        full_file_path = os.path.join(input_dir, filename)\n",
    "\n",
    "        # Check if the path is a file and ends with '.dx'\n",
    "        if os.path.isfile(full_file_path) and filename.endswith('.dx'):\n",
    "            file_count += 1\n",
    "            base_filename = os.path.splitext(filename)[0]\n",
    "            new_filename = f\"{base_filename}-pH_{folder_suffix}.dx\"\n",
    "            os.system(f\"cp '{full_file_path}' '{output_dir}{new_filename}'\")\n",
    "\n",
    "# Output the total count of processed files\n",
    "print(file_count)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Create 3D representation of a .dx file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import plotly.graph_objects as go\n",
    "\n",
    "def read_potential_file(filename):\n",
    "    counts = None\n",
    "    origin = None\n",
    "    deltas = []\n",
    "    potentials = []\n",
    "    \n",
    "    with open(filename, 'r') as f:\n",
    "        for line in f:\n",
    "            if line.startswith('object 1'):\n",
    "                counts = [int(x) for x in line.split('counts')[1].split()]\n",
    "            elif line.startswith('origin'):\n",
    "                origin = [float(x) for x in line.split()[1:]]\n",
    "            elif line.startswith('delta'):\n",
    "                delta = [float(x) for x in line.split()[1:]]\n",
    "                deltas.append(delta)\n",
    "            elif line[0].isdigit() or line[0] == '-':\n",
    "                values = [float(x) for x in line.split()]\n",
    "                potentials.extend(values)\n",
    "    \n",
    "    return np.array(counts), np.array(origin), np.array(deltas), np.array(potentials)\n",
    "\n",
    "def create_grid_coordinates(counts, origin, deltas):\n",
    "    x = np.linspace(origin[0], origin[0] + counts[0] * deltas[0][0], counts[0])\n",
    "    y = np.linspace(origin[1], origin[1] + counts[1] * deltas[1][1], counts[1])\n",
    "    z = np.linspace(origin[2], origin[2] + counts[2] * deltas[2][2], counts[2])\n",
    "    \n",
    "    X, Y, Z = np.meshgrid(x, y, z, indexing='ij')\n",
    "    return X, Y, Z\n",
    "\n",
    "def visualize_potentials(filename, sample_rate=5, filter_range=(-0.1, 0.1), \n",
    "                        point_style='dynamic', normal_point_size=4, \n",
    "                        range_point_size=1, min_dynamic_size=2, max_dynamic_size=8,\n",
    "                        background_opacity=0.8, protein_opacity=0.8, use_gray=True):\n",
    "    \"\"\"\n",
    "    Visualization with multiple point style options and customizable point sizes and colors.\n",
    "    \n",
    "    Parameters:\n",
    "    -----------\n",
    "    filename : str\n",
    "        Path to the potential data file\n",
    "    sample_rate : int\n",
    "        Sample every nth point to reduce plot size\n",
    "    filter_range : tuple\n",
    "        (min, max) range of values to filter or show as small points\n",
    "    point_style : str\n",
    "        'dynamic' - Point size scales with absolute potential value\n",
    "        'hide' - Don't show values in filter_range\n",
    "        'small' - Show values in filter_range with specified size\n",
    "    normal_point_size : int\n",
    "        Size of points outside filter_range (for 'hide'/'small' styles)\n",
    "    range_point_size : int\n",
    "        Size of points within filter_range (if point_style='small')\n",
    "    min_dynamic_size : int\n",
    "        Minimum point size for dynamic sizing\n",
    "    max_dynamic_size : int\n",
    "        Maximum point size for dynamic sizing\n",
    "    opacity : float\n",
    "        Opacity of points (0-1)\n",
    "    use_gray : bool\n",
    "        If True, use light gray for points in range; if False, keep viridis color scale\n",
    "    \"\"\"\n",
    "    # Read data\n",
    "    counts, origin, deltas, potentials = read_potential_file(filename)\n",
    "    potentials = potentials.reshape(counts)\n",
    "    X, Y, Z = create_grid_coordinates(counts, origin, deltas)\n",
    "    \n",
    "    # Sample the data\n",
    "    X = X[::sample_rate, ::sample_rate, ::sample_rate]\n",
    "    Y = Y[::sample_rate, ::sample_rate, ::sample_rate]\n",
    "    Z = Z[::sample_rate, ::sample_rate, ::sample_rate]\n",
    "    potentials = potentials[::sample_rate, ::sample_rate, ::sample_rate]\n",
    "    \n",
    "    # Create mask for values outside filter range\n",
    "    outside_range = (potentials < filter_range[0]) | (potentials > filter_range[1])\n",
    "    \n",
    "    # Create figure\n",
    "    fig = go.Figure()\n",
    "    \n",
    "    if point_style == 'dynamic':\n",
    "        # Keep all points but scale their sizes\n",
    "        sign = np.sign(potentials)\n",
    "        log_potentials = sign * np.log1p(np.abs(potentials))\n",
    "        \n",
    "        # Calculate point sizes based on absolute values\n",
    "        abs_potentials = np.abs(potentials)\n",
    "        max_abs_potential = np.max(abs_potentials)\n",
    "        normalized_sizes = (abs_potentials / max_abs_potential)\n",
    "        point_sizes = min_dynamic_size + (max_dynamic_size - min_dynamic_size) * normalized_sizes\n",
    "        \n",
    "        scatter = go.Scatter3d(\n",
    "            x=X.flatten(),\n",
    "            y=Y.flatten(),\n",
    "            z=Z.flatten(),\n",
    "            mode='markers',\n",
    "            marker=dict(\n",
    "                size=point_sizes.flatten(),\n",
    "                color=log_potentials.flatten(),\n",
    "                colorscale='viridis',\n",
    "                opacity=background_opacity,\n",
    "                colorbar=dict(\n",
    "                    title='Log Potential (kT/e)',\n",
    "                    ticktext=[f'{v:.1e}' for v in np.linspace(potentials.min(), potentials.max(), 6)],\n",
    "                    tickvals=np.linspace(log_potentials.min(), log_potentials.max(), 6)\n",
    "                )\n",
    "            ),\n",
    "            hovertemplate='x: %{x:.2f}<br>y: %{y:.2f}<br>z: %{z:.2f}<br>potential: %{text:.3e}<br>point size: %{marker.size:.1f}<extra></extra>',\n",
    "            text=potentials.flatten()\n",
    "        )\n",
    "        fig.add_trace(scatter)\n",
    "        \n",
    "    else:\n",
    "        # Process and plot values outside filter range\n",
    "        X_outside = X[outside_range]\n",
    "        Y_outside = Y[outside_range]\n",
    "        Z_outside = Z[outside_range]\n",
    "        potentials_outside = potentials[outside_range]\n",
    "        \n",
    "        # Log transform the values\n",
    "        sign_outside = np.sign(potentials_outside)\n",
    "        log_potentials_outside = sign_outside * np.log1p(np.abs(potentials_outside))\n",
    "        \n",
    "        scatter_outside = go.Scatter3d(\n",
    "            x=X_outside.flatten(),\n",
    "            y=Y_outside.flatten(),\n",
    "            z=Z_outside.flatten(),\n",
    "            mode='markers',\n",
    "            marker=dict(\n",
    "                size=normal_point_size,\n",
    "                color=log_potentials_outside.flatten(),\n",
    "                colorscale='viridis',\n",
    "                opacity=background_opacity,\n",
    "                colorbar=dict(\n",
    "                    title='Log Potential (kT/e)',\n",
    "                    ticktext=[f'{v:.1e}' for v in np.linspace(potentials_outside.min(), potentials_outside.max(), 6)],\n",
    "                    tickvals=np.linspace(log_potentials_outside.min(), log_potentials_outside.max(), 6)\n",
    "                )\n",
    "            ),\n",
    "            name='Outside range',\n",
    "            hovertemplate='x: %{x:.2f}<br>y: %{y:.2f}<br>z: %{z:.2f}<br>potential: %{text:.3e}<extra></extra>',\n",
    "            text=potentials_outside.flatten()\n",
    "        )\n",
    "        fig.add_trace(scatter_outside)\n",
    "        \n",
    "        # Process and plot values within filter range if point_style is 'small'\n",
    "        if point_style == 'small':\n",
    "            within_range = ~outside_range\n",
    "            X_within = X[within_range]\n",
    "            Y_within = Y[within_range]\n",
    "            Z_within = Z[within_range]\n",
    "            potentials_within = potentials[within_range]\n",
    "            \n",
    "            # Calculate colors for within-range points\n",
    "            if use_gray:\n",
    "                color = 'lightgray'\n",
    "                showscale = False\n",
    "            else:\n",
    "                sign_within = np.sign(potentials_within)\n",
    "                color = sign_within * np.log1p(np.abs(potentials_within))\n",
    "                showscale = False  # Don't show second colorbar\n",
    "            \n",
    "            scatter_within = go.Scatter3d(\n",
    "                x=X_within.flatten(),\n",
    "                y=Y_within.flatten(),\n",
    "                z=Z_within.flatten(),\n",
    "                mode='markers',\n",
    "                marker=dict(\n",
    "                    size=range_point_size,\n",
    "                    color=color if use_gray else color.flatten(),\n",
    "                    colorscale='viridis' if not use_gray else None,\n",
    "                    opacity=protein_opacity,\n",
    "                    showscale=showscale\n",
    "                ),\n",
    "                name='Within range',\n",
    "                hovertemplate='x: %{x:.2f}<br>y: %{y:.2f}<br>z: %{z:.2f}<br>potential: %{text:.3e}<extra></extra>',\n",
    "                text=potentials_within.flatten()\n",
    "            )\n",
    "            fig.add_trace(scatter_within)\n",
    "    \n",
    "    # Update layout\n",
    "    style_text = {\n",
    "        'dynamic': 'point size scaled with potential magnitude',\n",
    "        'hide': f'values {filter_range[0]:.1e} to {filter_range[1]:.1e} hidden',\n",
    "        'small': f'values {filter_range[0]:.1e} to {filter_range[1]:.1e} shown with size {range_point_size}'\n",
    "    }\n",
    "    color_text = ' (gray)' if use_gray and point_style == 'small' else ''\n",
    "    \n",
    "    fig.update_layout(\n",
    "        title=f'3D Electrostatic Potential Map<br>{style_text[point_style]}{color_text}',\n",
    "        legend=dict(\n",
    "            x=0,          # Move legend to the left side\n",
    "            y=1,          # Keep at top\n",
    "            xanchor='left',\n",
    "            yanchor='top',\n",
    "            bgcolor='rgba(255,255,255,0.7)'  # Semi-transparent white background\n",
    "        ),\n",
    "        scene=dict(\n",
    "            xaxis_title='X',\n",
    "            yaxis_title='Y',\n",
    "            zaxis_title='Z',\n",
    "            camera=dict(\n",
    "                eye=dict(x=1.5, y=1.5, z=1.5)\n",
    "            )\n",
    "        ),\n",
    "        width=900,\n",
    "        height=900\n",
    "    )\n",
    "    \n",
    "    return fig\n",
    "\n",
    "# Example usage:\n",
    "# Dynamic point sizing\n",
    "# fig = visualize_potentials('potential.dx', point_style='dynamic', \n",
    "#                          min_dynamic_size=2, max_dynamic_size=8)\n",
    "\n",
    "# Hide values between -0.1 and 0.1\n",
    "# fig = visualize_potentials('potential.dx', point_style='hide',\n",
    "#                          filter_range=(-0.1, 0.1))\n",
    "\n",
    "# Show values between -0.1 and 0.1 with specific point sizes (gray)\n",
    "# fig = visualize_potentials('potential.dx', point_style='small',\n",
    "#                          filter_range=(-0.1, 0.1),\n",
    "#                          normal_point_size=4,\n",
    "#                          range_point_size=1,\n",
    "#                          use_gray=True)\n",
    "\n",
    "# Show values between -0.1 and 0.1 with specific point sizes (viridis)\n",
    "# fig = visualize_potentials('potential.dx', point_style='small',\n",
    "#                          filter_range=(-0.1, 0.1),\n",
    "#                          normal_point_size=4,\n",
    "#                          range_point_size=1,\n",
    "#                          use_gray=False)\n",
    "\n",
    "# fig.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "fig = visualize_potentials('/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/pdb2pqr_pH7_9/AtSAL1.dx', \n",
    "                          point_style='small',\n",
    "                          sample_rate=2,\n",
    "                          filter_range=(-3, 3),\n",
    "                          normal_point_size=2,\n",
    "                          range_point_size=2,\n",
    "                          background_opacity=0.05,\n",
    "                          protein_opacity=0.05,\n",
    "                          use_gray=False)\n",
    "fig.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# AtCNP\n",
    "\n",
    "fig = visualize_potentials('/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/pdb2pqr_pH7_9/74987.dx', \n",
    "                          point_style='small',\n",
    "                          sample_rate=2,\n",
    "                          filter_range=(-3, 3),\n",
    "                          normal_point_size=2,\n",
    "                          range_point_size=2,\n",
    "                          background_opacity=0.05,\n",
    "                          protein_opacity=0.05,\n",
    "                          use_gray=False)\n",
    "fig.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "fig = visualize_potentials('/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/pdb2pqr_pH7_9/AtAHL.dx', \n",
    "                          point_style='small',\n",
    "                          sample_rate=2,\n",
    "                          filter_range=(-3, 3),\n",
    "                          normal_point_size=2,\n",
    "                          range_point_size=2,\n",
    "                          background_opacity=0.05,\n",
    "                          protein_opacity=0.05,\n",
    "                          use_gray=False)\n",
    "fig.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Pairwise comparisions of .dx files using Earth Mover Distance - no redundancy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing DX files: 100%|| 190/190 [00:07<00:00, 24.08it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Results saved to /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/all-v-all_EMD_dx.csv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "import numpy as np\n",
    "import os\n",
    "import pandas as pd\n",
    "from scipy.stats import wasserstein_distance\n",
    "from concurrent.futures import ProcessPoolExecutor, as_completed\n",
    "import itertools\n",
    "from tqdm import tqdm\n",
    "import sys\n",
    "import multiprocessing\n",
    "\n",
    "def parse_grid_data(lines):\n",
    "    \"\"\"\n",
    "    Parses the grid data section within the object containing \"positions\".\n",
    "\n",
    "    Args:\n",
    "        lines (list): List of lines from the DX file containing grid data.\n",
    "\n",
    "    Returns:\n",
    "        numpy.ndarray: Grid data as a NumPy array.\n",
    "    \"\"\"\n",
    "    data = []\n",
    "    start_parsing = False\n",
    "    for line in lines:\n",
    "        line = line.strip()\n",
    "        if line.endswith('data follows'):\n",
    "            start_parsing = True\n",
    "            continue\n",
    "        if start_parsing:\n",
    "            if line.startswith('attribute'):\n",
    "                break\n",
    "            values = list(map(float, line.split()))\n",
    "            data.extend(values)\n",
    "    return np.array(data)\n",
    "\n",
    "def load_dx_file(path):\n",
    "    \"\"\"\n",
    "    Loads a DX file and parses its grid data.\n",
    "\n",
    "    Args:\n",
    "        path (str): Path to the DX file.\n",
    "\n",
    "    Returns:\n",
    "        numpy.ndarray: Parsed grid data.\n",
    "    \"\"\"\n",
    "    with open(path, 'r') as file:\n",
    "        lines = file.readlines()\n",
    "    return parse_grid_data(lines)\n",
    "\n",
    "def emd_distance(bins1, bins2):\n",
    "    \"\"\"\n",
    "    Calculates the Earth Mover's Distance (EMD) between two sets of bins.\n",
    "\n",
    "    Args:\n",
    "        bins1 (numpy.ndarray): First set of bins.\n",
    "        bins2 (numpy.ndarray): Second set of bins.\n",
    "\n",
    "    Returns:\n",
    "        float: Normalized EMD distance.\n",
    "    \"\"\"\n",
    "    range_bins1 = np.max(bins1) - np.min(bins1)\n",
    "    if range_bins1 == 0:\n",
    "        return np.nan  # or any default value indicating division by zero\n",
    "    distance = wasserstein_distance(bins1, bins2)\n",
    "    return distance / range_bins1\n",
    "\n",
    "def compare_dx_batch(batch):\n",
    "    \"\"\"\n",
    "    Compares DX files in a batch and calculates EMD scores.\n",
    "\n",
    "    Args:\n",
    "        batch (list): List of pairs of DX file paths to compare.\n",
    "\n",
    "    Returns:\n",
    "        list: Results of comparisons including EMD scores.\n",
    "    \"\"\"\n",
    "    results = []\n",
    "    for pair in batch:\n",
    "        (key1, path1), (key2, path2) = pair\n",
    "        data1 = load_dx_file(path1)\n",
    "        data2 = load_dx_file(path2)\n",
    "        if data1 is not None and data2 is not None:\n",
    "            emd_score = emd_distance(data1, data2)\n",
    "            id1, pH1 = parse_details(key1)\n",
    "            id2, pH2 = parse_details(key2)\n",
    "            results.append((key1, id1, pH1, key2, id2, pH2, emd_score))\n",
    "    return results\n",
    "\n",
    "def process_dx_files(file_dir):\n",
    "    \"\"\"\n",
    "    Processes the directory to collect DX files.\n",
    "\n",
    "    Args:\n",
    "        file_dir (str): Directory containing DX files.\n",
    "\n",
    "    Returns:\n",
    "        dict: Dictionary of DX files with basenames as keys and paths as values.\n",
    "    \"\"\"\n",
    "    dx_files = {}\n",
    "    for filename in os.listdir(file_dir):\n",
    "        if filename.endswith('.dx'):\n",
    "            full_file_path = os.path.join(file_dir, filename)\n",
    "            basename = os.path.splitext(filename)[0]\n",
    "            dx_files[basename] = full_file_path\n",
    "    return dx_files\n",
    "\n",
    "def parse_details(basename):\n",
    "    \"\"\"\n",
    "    Parses details from the basename of a DX file.\n",
    "\n",
    "    Args:\n",
    "        basename (str): Basename of the DX file.\n",
    "\n",
    "    Returns:\n",
    "        tuple: Parsed ID and pH value.\n",
    "    \"\"\"\n",
    "    id_name, pH = basename.split('-')\n",
    "    pH = pH.lstrip('pH_')\n",
    "    return id_name, pH\n",
    "\n",
    "def main():\n",
    "    \"\"\"\n",
    "    Main function to process all DX files and compute EMD scores.\n",
    "    \"\"\"\n",
    "    base_dir = '/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/hal2p/'\n",
    "    file_dir = os.path.join(base_dir, 'all_cropped/')\n",
    "\n",
    "    dx_files = process_dx_files(file_dir)\n",
    "    dx_pairs = list(itertools.combinations(dx_files.items(), 2))\n",
    "\n",
    "    num_files = len(dx_pairs)\n",
    "    if num_files == 0:\n",
    "        print(\"No DX file pairs found. Exiting.\")\n",
    "        sys.exit(1)\n",
    "\n",
    "    batch_size = 500  # Define your batch size\n",
    "    num_batches = (num_files + batch_size - 1) // batch_size\n",
    "\n",
    "    with ProcessPoolExecutor(max_workers=multiprocessing.cpu_count()) as executor, tqdm(total=num_files, desc=\"Processing DX files\") as pbar:\n",
    "        futures = []\n",
    "        for i in range(num_batches):\n",
    "            start_idx = i * batch_size\n",
    "            end_idx = min((i + 1) * batch_size, num_files)\n",
    "            batch = dx_pairs[start_idx:end_idx]\n",
    "            futures.append(executor.submit(compare_dx_batch, batch))\n",
    "        \n",
    "        results = []\n",
    "        for future in as_completed(futures):\n",
    "            results.extend(future.result())\n",
    "            pbar.update(len(future.result()))\n",
    "\n",
    "    df = pd.DataFrame(results, columns=['basename1', 'id1', 'pH1', 'basename2', 'id2', 'pH2', 'emd_score'])\n",
    "    output_file = os.path.join(base_dir, 'all-v-all_EMD_dx.csv')\n",
    "    df.to_csv(output_file, index=False)\n",
    "    print(f\"Results saved to {output_file}\")\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    main()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Pairwise comparisions of .dx files using Earth Mover Distance - with redundancy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing DX files: 100%|| 400/400 [00:16<00:00, 24.90it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Results saved to /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/all-v-all_EMD_dx.csv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "import numpy as np\n",
    "import os\n",
    "import pandas as pd\n",
    "from scipy.stats import wasserstein_distance\n",
    "from concurrent.futures import ProcessPoolExecutor, as_completed\n",
    "import itertools\n",
    "from tqdm import tqdm\n",
    "import sys\n",
    "import multiprocessing\n",
    "\n",
    "def parse_grid_data(lines):\n",
    "    \"\"\"\n",
    "    Parses the grid data section within the object containing \"positions\".\n",
    "\n",
    "    Args:\n",
    "        lines (list): List of lines from the DX file containing grid data.\n",
    "\n",
    "    Returns:\n",
    "        numpy.ndarray: Grid data as a NumPy array.\n",
    "    \"\"\"\n",
    "    data = []\n",
    "    start_parsing = False\n",
    "    for line in lines:\n",
    "        line = line.strip()\n",
    "        if line.endswith('data follows'):\n",
    "            start_parsing = True\n",
    "            continue\n",
    "        if start_parsing:\n",
    "            if line.startswith('attribute'):\n",
    "                break\n",
    "            values = list(map(float, line.split()))\n",
    "            data.extend(values)\n",
    "    return np.array(data)\n",
    "\n",
    "def load_dx_file(path):\n",
    "    \"\"\"\n",
    "    Loads a DX file and parses its grid data.\n",
    "\n",
    "    Args:\n",
    "        path (str): Path to the DX file.\n",
    "\n",
    "    Returns:\n",
    "        numpy.ndarray: Parsed grid data.\n",
    "    \"\"\"\n",
    "    with open(path, 'r') as file:\n",
    "        lines = file.readlines()\n",
    "    return parse_grid_data(lines)\n",
    "\n",
    "def emd_distance(bins1, bins2):\n",
    "    \"\"\"\n",
    "    Calculates the Earth Mover's Distance (EMD) between two sets of bins.\n",
    "\n",
    "    Args:\n",
    "        bins1 (numpy.ndarray): First set of bins.\n",
    "        bins2 (numpy.ndarray): Second set of bins.\n",
    "\n",
    "    Returns:\n",
    "        float: Normalized EMD distance.\n",
    "    \"\"\"\n",
    "    range_bins1 = np.max(bins1) - np.min(bins1)\n",
    "    if range_bins1 == 0:\n",
    "        return np.nan  # or any default value indicating division by zero\n",
    "    distance = wasserstein_distance(bins1, bins2)\n",
    "    return distance / range_bins1\n",
    "\n",
    "def compare_dx_batch(batch):\n",
    "    \"\"\"\n",
    "    Compares DX files in a batch and calculates EMD scores.\n",
    "\n",
    "    Args:\n",
    "        batch (list): List of pairs of DX file paths to compare.\n",
    "\n",
    "    Returns:\n",
    "        list: Results of comparisons including EMD scores.\n",
    "    \"\"\"\n",
    "    results = []\n",
    "    for pair in batch:\n",
    "        (key1, path1), (key2, path2) = pair\n",
    "        data1 = load_dx_file(path1)\n",
    "        data2 = load_dx_file(path2)\n",
    "        if data1 is not None and data2 is not None:\n",
    "            emd_score = emd_distance(data1, data2)\n",
    "            id1, pH1 = parse_details(key1)\n",
    "            id2, pH2 = parse_details(key2)\n",
    "            results.append((key1, id1, pH1, key2, id2, pH2, emd_score))\n",
    "    return results\n",
    "\n",
    "def process_dx_files(file_dir):\n",
    "    \"\"\"\n",
    "    Processes the directory to collect DX files.\n",
    "\n",
    "    Args:\n",
    "        file_dir (str): Directory containing DX files.\n",
    "\n",
    "    Returns:\n",
    "        dict: Dictionary of DX files with basenames as keys and paths as values.\n",
    "    \"\"\"\n",
    "    dx_files = {}\n",
    "    for filename in os.listdir(file_dir):\n",
    "        if filename.endswith('.dx'):\n",
    "            full_file_path = os.path.join(file_dir, filename)\n",
    "            basename = os.path.splitext(filename)[0]\n",
    "            dx_files[basename] = full_file_path\n",
    "    return dx_files\n",
    "\n",
    "def parse_details(basename):\n",
    "    \"\"\"\n",
    "    Parses details from the basename of a DX file.\n",
    "\n",
    "    Args:\n",
    "        basename (str): Basename of the DX file.\n",
    "\n",
    "    Returns:\n",
    "        tuple: Parsed ID and pH value.\n",
    "    \"\"\"\n",
    "    id_name, pH = basename.split('-')\n",
    "    pH = pH.lstrip('pH_')\n",
    "    return id_name, pH\n",
    "\n",
    "def main():\n",
    "    \"\"\"\n",
    "    Main function to process all DX files and compute EMD scores.\n",
    "    \"\"\"\n",
    "    base_dir = '/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/'\n",
    "    file_dir = os.path.join(base_dir, 'all_cropped/')\n",
    "\n",
    "    dx_files = process_dx_files(file_dir)\n",
    "    dx_pairs = list(itertools.product(dx_files.items(), repeat=2))\n",
    "\n",
    "    num_files = len(dx_pairs)\n",
    "    if num_files == 0:\n",
    "        print(\"No DX file pairs found. Exiting.\")\n",
    "        sys.exit(1)\n",
    "\n",
    "    batch_size = 500  # Define your batch size\n",
    "    num_batches = (num_files + batch_size - 1) // batch_size\n",
    "\n",
    "    with ProcessPoolExecutor(max_workers=multiprocessing.cpu_count()) as executor, tqdm(total=num_files, desc=\"Processing DX files\") as pbar:\n",
    "        futures = []\n",
    "        for i in range(num_batches):\n",
    "            start_idx = i * batch_size\n",
    "            end_idx = min((i + 1) * batch_size, num_files)\n",
    "            batch = dx_pairs[start_idx:end_idx]\n",
    "            futures.append(executor.submit(compare_dx_batch, batch))\n",
    "        \n",
    "        results = []\n",
    "        for future in as_completed(futures):\n",
    "            results.extend(future.result())\n",
    "            pbar.update(len(future.result()))\n",
    "\n",
    "    df = pd.DataFrame(results, columns=['basename1', 'id1', 'pH1', 'basename2', 'id2', 'pH2', 'emd_score'])\n",
    "    output_file = os.path.join(base_dir, 'all-v-all_EMD_dx.csv')\n",
    "    df.to_csv(output_file, index=False)\n",
    "    print(f\"Results saved to {output_file}\")\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    main()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Merge with taxonomic information"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "index_df = pd.read_csv(\"/home/mattm/projects/NP/wrk_index.csv\")\n",
    "\n",
    "index_df = index_df[['Unid', 'Classification', 'Brief_Classification', 'Species']]\n",
    "\n",
    "index_df.loc[len(index_df.index)] = ['AtSAL1updated', 'CoreEudicots/Rosids_Brassicales', 'CoreEudicots/Rosids', 'Arabidopsis_thaliana'] \n",
    "\n",
    "results_df = pd.read_csv('/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/all-v-all_EMD_dx.csv', low_memory=False)\n",
    "\n",
    "detailed_results_df = results_df.merge(index_df, left_on='id1', right_on='Unid', how='outer' )\n",
    "\n",
    "base_dir = '/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/'\n",
    "output_file = os.path.join(base_dir, 'all-v-all_EMD_dx.csv')\n",
    "\n",
    "detailed_results_df.to_csv(output_file, index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## One-v-all\n",
    "### Compares the AtSAL1-updated structure to all others"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing DX files:   0%|          | 0/2525 [00:00<?, ?it/s]"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing DX files: 100%|| 2525/2525 [00:14<00:00, 170.65it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Results saved to /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/AtSAL1-pH_8_2_one-v-all_EMD_dx.csv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "import numpy as np\n",
    "import os\n",
    "import pandas as pd\n",
    "from scipy.stats import wasserstein_distance\n",
    "from concurrent.futures import ProcessPoolExecutor, as_completed\n",
    "from tqdm import tqdm\n",
    "import sys\n",
    "import multiprocessing\n",
    "\n",
    "def parse_grid_data(lines):\n",
    "    \"\"\"\n",
    "    Parses the grid data section within the object containing \"positions\".\n",
    "\n",
    "    Args:\n",
    "        lines (list): List of lines from the DX file containing grid data.\n",
    "\n",
    "    Returns:\n",
    "        numpy.ndarray: Grid data as a NumPy array.\n",
    "    \"\"\"\n",
    "    data = []\n",
    "    start_parsing = False\n",
    "    for line in lines:\n",
    "        line = line.strip()\n",
    "        if line.endswith('data follows'):\n",
    "            start_parsing = True\n",
    "            continue\n",
    "        if start_parsing:\n",
    "            if line.startswith('attribute'):\n",
    "                break\n",
    "            values = list(map(float, line.split()))\n",
    "            data.extend(values)\n",
    "    return np.array(data)\n",
    "\n",
    "def load_dx_file(path):\n",
    "    \"\"\"\n",
    "    Loads a DX file and parses its grid data.\n",
    "\n",
    "    Args:\n",
    "        path (str): Path to the DX file.\n",
    "\n",
    "    Returns:\n",
    "        numpy.ndarray: Parsed grid data.\n",
    "    \"\"\"\n",
    "    with open(path, 'r') as file:\n",
    "        lines = file.readlines()\n",
    "    return parse_grid_data(lines)\n",
    "\n",
    "def emd_distance(bins1, bins2):\n",
    "    \"\"\"\n",
    "    Calculates the Earth Mover's Distance (EMD) between two sets of bins.\n",
    "\n",
    "    Args:\n",
    "        bins1 (numpy.ndarray): First set of bins.\n",
    "        bins2 (numpy.ndarray): Second set of bins.\n",
    "\n",
    "    Returns:\n",
    "        float: Normalized EMD distance.\n",
    "    \"\"\"\n",
    "    range_bins1 = np.max(bins1) - np.min(bins1)\n",
    "    if range_bins1 == 0:\n",
    "        return np.nan  # or any default value indicating division by zero\n",
    "    distance = wasserstein_distance(bins1, bins2)\n",
    "    return distance / range_bins1\n",
    "\n",
    "def compare_dx_batch(one_file_data, one_file_key, batch):\n",
    "    \"\"\"\n",
    "    Compares the 'one' DX file with a batch of other DX files and calculates EMD scores.\n",
    "\n",
    "    Args:\n",
    "        one_file_data (numpy.ndarray): The data of the 'one' DX file.\n",
    "        one_file_key (str): The key of the 'one' DX file.\n",
    "        batch (list): List of other DX file paths to compare with 'one' file.\n",
    "\n",
    "    Returns:\n",
    "        list: Results of comparisons including EMD scores.\n",
    "    \"\"\"\n",
    "    results = []\n",
    "    for key2, path2 in batch:\n",
    "        data2 = load_dx_file(path2)\n",
    "        if one_file_data is not None and data2 is not None:\n",
    "            emd_score = emd_distance(one_file_data, data2)\n",
    "            id1, pH1 = parse_details(one_file_key)\n",
    "            id2, pH2 = parse_details(key2)\n",
    "            results.append((one_file_key, id1, pH1, key2, id2, pH2, emd_score))\n",
    "    return results\n",
    "\n",
    "def process_dx_files(file_dir):\n",
    "    \"\"\"\n",
    "    Processes the directory to collect DX files.\n",
    "\n",
    "    Args:\n",
    "        file_dir (str): Directory containing DX files.\n",
    "\n",
    "    Returns:\n",
    "        dict: Dictionary of DX files with basenames as keys and paths as values.\n",
    "    \"\"\"\n",
    "    dx_files = {}\n",
    "    for filename in os.listdir(file_dir):\n",
    "        if filename.endswith('.dx'):\n",
    "            full_file_path = os.path.join(file_dir, filename)\n",
    "            basename = os.path.splitext(filename)[0]\n",
    "            dx_files[basename] = full_file_path\n",
    "    return dx_files\n",
    "\n",
    "def parse_details(basename):\n",
    "    \"\"\"\n",
    "    Parses details from the basename of a DX file.\n",
    "\n",
    "    Args:\n",
    "        basename (str): Basename of the DX file.\n",
    "\n",
    "    Returns:\n",
    "        tuple: Parsed ID and pH value.\n",
    "    \"\"\"\n",
    "    id_name, pH = basename.split('-')\n",
    "    pH = pH.lstrip('pH_')\n",
    "    return id_name, pH\n",
    "\n",
    "def main():\n",
    "    \"\"\"\n",
    "    Main function to process all DX files and compute EMD scores.\n",
    "    \"\"\"\n",
    "    base_dir = '/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/'\n",
    "    file_dir = os.path.join(base_dir, 'all_cropped/')\n",
    "    # one_file_path = os.path.join(base_dir, 'AtSAL1_updated/all_cropped/AtSAL1_updated-pH_7_9.dx') # Change this as reference\n",
    "    # one_file_path = os.path.join(base_dir, 'AtSAL1_updated/all_cropped/AtSAL1_updated-pH_7_6.dx') # Change this as reference\n",
    "    one_file_path = os.path.join(base_dir, 'AtSAL1_updated/all_cropped/AtSAL1_updated-pH_8_2.dx') # Change this as reference\n",
    "    # one_file_path = os.path.join(base_dir, 'hal2p/all_cropped/1ka1_hal2-pH_7_9.dx') # Change this as reference\n",
    "    # one_file_path = os.path.join(base_dir, 'hal2p/all_cropped/1ka1_hal2-pH_7_6.dx') # Change this as reference\n",
    "    # one_file_path = os.path.join(base_dir, 'hal2p/all_cropped/1ka1_hal2-pH_8_2.dx') # Change this as reference\n",
    "\n",
    "    dx_files = process_dx_files(file_dir)\n",
    "    dx_items = list(dx_files.items())\n",
    "\n",
    "    if len(dx_items) == 0:\n",
    "        print(\"No DX files found. Exiting.\")\n",
    "        sys.exit(1)\n",
    "\n",
    "    one_file_key = os.path.splitext(os.path.basename(one_file_path))[0]\n",
    "    one_file_data = load_dx_file(one_file_path)\n",
    "\n",
    "    # Create pairs of (one, all others)\n",
    "    dx_pairs = [(key, path) for key, path in dx_items if key != one_file_key]\n",
    "\n",
    "    num_files = len(dx_pairs)\n",
    "    if num_files == 0:\n",
    "        print(\"No DX file pairs found for one-vs-all comparison. Exiting.\")\n",
    "        sys.exit(1)\n",
    "\n",
    "    batch_size = 500  # Define your batch size\n",
    "    num_batches = (num_files + batch_size - 1) // batch_size\n",
    "\n",
    "    with ProcessPoolExecutor(max_workers=multiprocessing.cpu_count()) as executor, tqdm(total=num_files, desc=\"Processing DX files\") as pbar:\n",
    "        futures = []\n",
    "        for i in range(num_batches):\n",
    "            start_idx = i * batch_size\n",
    "            end_idx = min((i + 1) * batch_size, num_files)\n",
    "            batch = dx_pairs[start_idx:end_idx]\n",
    "            futures.append(executor.submit(compare_dx_batch, one_file_data, one_file_key, batch))\n",
    "        \n",
    "        results = []\n",
    "        for future in as_completed(futures):\n",
    "            results.extend(future.result())\n",
    "            pbar.update(len(future.result()))\n",
    "\n",
    "    df = pd.DataFrame(results, columns=['basename1', 'id1', 'pH1', 'basename2', 'id2', 'pH2', 'emd_score'])\n",
    "    output_file = os.path.join(base_dir, 'AtSAL1-pH_8_2_one-v-all_EMD_dx.csv') # Change this depending on reference\n",
    "    df.to_csv(output_file, index=False)\n",
    "    print(f\"Results saved to {output_file}\")\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    main()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# one-vs-five EMD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2024-06-14 13:35:53,682 - INFO - Loading DX file: /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/AtSAL1_updated/all_cropped/AtSAL1_updated-pH_7_9.dx\n",
      "Processing DX files:   0%|          | 0/505 [00:00<?, ?it/s]2024-06-14 13:35:54,718 - INFO - Loading DX file: /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/all_cropped/36524-pH_7_3.dx\n",
      "2024-06-14 13:35:54,739 - INFO - Loading DX file: /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/all_cropped/36524-pH_7_9.dx\n",
      "2024-06-14 13:35:54,753 - INFO - Loading DX file: /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/all_cropped/36524-pH_8_2.dx\n",
      "2024-06-14 13:35:54,766 - INFO - Loading DX file: /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/all_cropped/36524-pH_7_6.dx\n",
      "2024-06-14 13:35:54,780 - INFO - Loading DX file: /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/all_cropped/36524-pH_7_0.dx\n",
      "2024-06-14 13:35:54,793 - INFO - Comparing AtSAL1_updated-pH_7_9 with batch of size 5\n",
      "2024-06-14 13:36:44,641 - ERROR - Error processing batch: Unable to allocate 622. GiB for an array with shape (316638, 263865) and data type float64\n",
      "2024-06-14 13:36:47,829 - INFO - Loading DX file: /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/all_cropped/71776-pH_7_0.dx\n",
      "2024-06-14 13:36:47,850 - INFO - Loading DX file: /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/all_cropped/71776-pH_8_2.dx\n",
      "2024-06-14 13:36:47,864 - INFO - Loading DX file: /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/all_cropped/71776-pH_7_6.dx\n",
      "2024-06-14 13:36:47,876 - INFO - Loading DX file: /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/all_cropped/71776-pH_7_3.dx\n",
      "2024-06-14 13:36:47,889 - INFO - Loading DX file: /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/all_cropped/71776-pH_7_9.dx\n",
      "2024-06-14 13:36:47,902 - INFO - Comparing AtSAL1_updated-pH_7_9 with batch of size 5\n",
      "2024-06-14 13:37:37,791 - ERROR - Error processing batch: Unable to allocate 622. GiB for an array with shape (316638, 263865) and data type float64\n",
      "2024-06-14 13:37:40,975 - INFO - Loading DX file: /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/all_cropped/60097-pH_7_0.dx\n",
      "2024-06-14 13:37:41,001 - INFO - Loading DX file: /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/all_cropped/60097-pH_7_6.dx\n",
      "2024-06-14 13:37:41,020 - INFO - Loading DX file: /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/all_cropped/60097-pH_8_2.dx\n",
      "2024-06-14 13:37:41,038 - INFO - Loading DX file: /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/all_cropped/60097-pH_7_9.dx\n",
      "2024-06-14 13:37:41,057 - INFO - Loading DX file: /home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/all_cropped/60097-pH_7_3.dx\n",
      "2024-06-14 13:37:41,075 - INFO - Comparing AtSAL1_updated-pH_7_9 with batch of size 5\n",
      "Processing DX files:   0%|          | 0/505 [01:56<?, ?it/s]\n"
     ]
    }
   ],
   "source": [
    "import numpy as np\n",
    "import os\n",
    "import pandas as pd\n",
    "from scipy.stats import wasserstein_distance\n",
    "from scipy.optimize import linprog\n",
    "from concurrent.futures import ProcessPoolExecutor, as_completed\n",
    "from tqdm import tqdm\n",
    "import sys\n",
    "import multiprocessing\n",
    "from collections import defaultdict\n",
    "import logging\n",
    "import gc  # For garbage collection\n",
    "\n",
    "# Setup logging\n",
    "logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')\n",
    "\n",
    "def parse_grid_data(lines):\n",
    "    \"\"\"\n",
    "    Parses the grid data section within the object containing \"positions\".\n",
    "\n",
    "    Args:\n",
    "        lines (list): List of lines from the DX file containing grid data.\n",
    "\n",
    "    Returns:\n",
    "        numpy.ndarray: Grid data as a NumPy array.\n",
    "    \"\"\"\n",
    "    data = []\n",
    "    start_parsing = False\n",
    "    for line in lines:\n",
    "        line = line.strip()\n",
    "        if line.endswith('data follows'):\n",
    "            start_parsing = True\n",
    "            continue\n",
    "        if start_parsing:\n",
    "            if line.startswith('attribute'):\n",
    "                break\n",
    "            values = list(map(float, line.split()))\n",
    "            data.extend(values)\n",
    "    return np.array(data)\n",
    "\n",
    "def load_dx_file(path):\n",
    "    \"\"\"\n",
    "    Loads a DX file and parses its grid data.\n",
    "\n",
    "    Args:\n",
    "        path (str): Path to the DX file.\n",
    "\n",
    "    Returns:\n",
    "        numpy.ndarray: Parsed grid data.\n",
    "    \"\"\"\n",
    "    logging.info(f\"Loading DX file: {path}\")\n",
    "    with open(path, 'r') as file:\n",
    "        lines = file.readlines()\n",
    "    return parse_grid_data(lines)\n",
    "\n",
    "def generalized_emd_distance(single_hist, set_hists, timeout=300):\n",
    "    \"\"\"\n",
    "    Calculates the generalized Earth Mover's Distance (EMD) between a single histogram and a set of histograms.\n",
    "\n",
    "    Args:\n",
    "        single_hist (numpy.ndarray): Histogram of the single image.\n",
    "        set_hists (list of numpy.ndarray): List of histograms of the set of images.\n",
    "        timeout (int): Time limit for the linear programming solver.\n",
    "\n",
    "    Returns:\n",
    "        float: Generalized EMD distance.\n",
    "    \"\"\"\n",
    "    num_bins = len(single_hist)\n",
    "    num_distributions = len(set_hists)\n",
    "    \n",
    "    # Create the cost matrix\n",
    "    cost_matrix = np.zeros((num_bins, num_distributions * num_bins))\n",
    "    for i in range(num_bins):\n",
    "        for j in range(num_distributions):\n",
    "            cost_matrix[i, j * num_bins:(j + 1) * num_bins] = np.abs(np.arange(num_bins) - i)\n",
    "    \n",
    "    # Flatten histograms\n",
    "    c = cost_matrix.flatten()\n",
    "    \n",
    "    # Create equality constraints\n",
    "    A_eq = np.zeros((num_bins * (num_distributions + 1), num_bins * num_distributions))\n",
    "    b_eq = np.zeros(num_bins * (num_distributions + 1))\n",
    "    \n",
    "    # Single histogram constraint\n",
    "    for i in range(num_bins):\n",
    "        A_eq[i, i::num_bins] = 1\n",
    "        b_eq[i] = single_hist[i]\n",
    "    \n",
    "    # Set histograms constraints\n",
    "    for j in range(num_distributions):\n",
    "        for i in range(num_bins):\n",
    "            A_eq[num_bins * (j + 1) + i, j * num_bins + i] = 1\n",
    "            b_eq[num_bins * (j + 1) + i] = set_hists[j][i]\n",
    "    \n",
    "    # Bounds for the flow variables\n",
    "    bounds = [(0, None) for _ in range(num_bins * num_distributions)]\n",
    "    \n",
    "    # Solve the linear programming problem with a timeout\n",
    "    try:\n",
    "        result = linprog(c, A_eq=A_eq, b_eq=b_eq, bounds=bounds, method='highs', options={'maxiter': timeout})\n",
    "    except TimeoutError:\n",
    "        logging.warning(\"Linear programming solver exceeded the time limit\")\n",
    "        return None\n",
    "    \n",
    "    return result.fun if result.success else None\n",
    "\n",
    "def compare_dx_batch(one_file_data, one_file_key, batch):\n",
    "    \"\"\"\n",
    "    Compares the 'one' DX file with a batch of other DX files and calculates generalized EMD scores.\n",
    "\n",
    "    Args:\n",
    "        one_file_data (numpy.ndarray): The data of the 'one' DX file.\n",
    "        one_file_key (str): The key of the 'one' DX file.\n",
    "        batch (list): List of other DX file paths to compare with 'one' file.\n",
    "\n",
    "    Returns:\n",
    "        list: Results of comparisons including generalized EMD scores.\n",
    "    \"\"\"\n",
    "    set_hists = []\n",
    "    for key2, path2 in batch:\n",
    "        try:\n",
    "            data2 = load_dx_file(path2)\n",
    "            if data2 is not None:\n",
    "                set_hists.append(data2)\n",
    "        except Exception as e:\n",
    "            logging.error(f\"Error loading file {path2}: {e}\")\n",
    "            return []\n",
    "\n",
    "    if not set_hists:\n",
    "        return []\n",
    "\n",
    "    logging.info(f\"Comparing {one_file_key} with batch of size {len(batch)}\")\n",
    "    emd_score = generalized_emd_distance(one_file_data, set_hists)\n",
    "    \n",
    "    results = []\n",
    "    for key2, path2 in batch:\n",
    "        id1, pH1 = parse_details(one_file_key)\n",
    "        id2, pH2 = parse_details(key2)\n",
    "        results.append((one_file_key, id1, pH1, key2, id2, pH2, emd_score))\n",
    "    return results\n",
    "\n",
    "def process_dx_files(file_dir):\n",
    "    \"\"\"\n",
    "    Processes the directory to collect DX files and group them by their base names.\n",
    "\n",
    "    Args:\n",
    "        file_dir (str): Directory containing DX files.\n",
    "\n",
    "    Returns:\n",
    "        dict: Dictionary of grouped DX files with base names as keys and lists of paths as values.\n",
    "    \"\"\"\n",
    "    dx_files = defaultdict(list)\n",
    "    for filename in os.listdir(file_dir):\n",
    "        if filename.endswith('.dx'):\n",
    "            full_file_path = os.path.join(file_dir, filename)\n",
    "            basename = filename.split('-')[0]\n",
    "            dx_files[basename].append(full_file_path)\n",
    "    return dx_files\n",
    "\n",
    "def parse_details(basename):\n",
    "    \"\"\"\n",
    "    Parses details from the basename of a DX file.\n",
    "\n",
    "    Args:\n",
    "        basename (str): Basename of the DX file.\n",
    "\n",
    "    Returns:\n",
    "        tuple: Parsed ID and pH value.\n",
    "    \"\"\"\n",
    "    parts = basename.split('-')\n",
    "    id_name = parts[0]\n",
    "    pH = parts[1].lstrip('pH_')\n",
    "    return id_name, pH\n",
    "\n",
    "def main():\n",
    "    \"\"\"\n",
    "    Main function to process all DX files and compute generalized EMD scores.\n",
    "    \"\"\"\n",
    "    base_dir = '/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/'\n",
    "    file_dir = os.path.join(base_dir, 'all_cropped/')\n",
    "    one_file_path = os.path.join(base_dir, 'AtSAL1_updated/all_cropped/AtSAL1_updated-pH_7_9.dx') # Change this as reference\n",
    "\n",
    "    dx_files = process_dx_files(file_dir)\n",
    "\n",
    "    if len(dx_files) == 0:\n",
    "        print(\"No DX files found. Exiting.\")\n",
    "        sys.exit(1)\n",
    "\n",
    "    one_file_key = os.path.splitext(os.path.basename(one_file_path))[0]\n",
    "    one_file_data = load_dx_file(one_file_path)\n",
    "\n",
    "    results = []\n",
    "    max_workers = 1  # Limit to 1 worker to minimize memory usage\n",
    "    with ProcessPoolExecutor(max_workers=max_workers) as executor, tqdm(total=len(dx_files), desc=\"Processing DX files\") as pbar:\n",
    "        futures = []\n",
    "        for key, paths in dx_files.items():\n",
    "            if len(paths) == 5:\n",
    "                batch = [(os.path.splitext(os.path.basename(path))[0], path) for path in paths]\n",
    "                futures.append(executor.submit(compare_dx_batch, one_file_data, one_file_key, batch))\n",
    "        \n",
    "        for future in as_completed(futures):\n",
    "            try:\n",
    "                batch_results = future.result()\n",
    "                results.extend(batch_results)\n",
    "                pbar.update(1)\n",
    "            except Exception as e:\n",
    "                logging.error(f\"Error processing batch: {e}\")\n",
    "            finally:\n",
    "                gc.collect()  # Force garbage collection after each batch\n",
    "\n",
    "    df = pd.DataFrame(results, columns=['basename1', 'id1', 'pH1', 'basename2', 'id2', 'pH2', 'emd_score'])\n",
    "    output_file = os.path.join(base_dir, 'one-v-set_EMD_dx.csv')\n",
    "    df.to_csv(output_file, index=False)\n",
    "    print(f\"Results saved to {output_file}\")\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    main()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### For the above, need to diagnose why this is using so much memory and find a way to reduce it. Start with removing paraellel processing?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Checking edge score distro, log transform - don't include code in publication"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# This is for joining the SAL1 and HAL2 results\n",
    "sal_df = pd.read_csv('/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/one-v-all_EMD_dx.csv')\n",
    "hal2_df = pd.read_csv('/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/hal2p/one-v-all_EMD_dx.csv')\n",
    "\n",
    "sal_df = sal_df[['id2', 'pH2', 'emd_score']]\n",
    "sal_df['ref_protein'] = 'AtSAL1_updated'\n",
    "hal2_df = hal2_df[['id2', 'pH2', 'emd_score']]\n",
    "hal2_df['ref_protein'] = '1qgx_hal2p'\n",
    "\n",
    "dfs = [sal_df, hal2_df]\n",
    "concat_df = pd.concat(dfs)\n",
    "concat_df.to_csv('/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/hal2p/one-v-all_EMD_dx_SAL1-HAL2.csv')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/hal2p/one-v-all_EMD_dx_SAL1-HAL2.csv')\n",
    "column_data = df['emd_score']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAjsAAAGxCAYAAACEFXd4AAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjkuMCwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy80BEi2AAAACXBIWXMAAA9hAAAPYQGoP6dpAABpO0lEQVR4nO3deXxU1f3/8ddMkpnsO9kghCWsEhZRMIoryFpFRa2KgpWvVkVrxVq/tNaNWqxatLWg3/anoHWrWFdcEBFwYUfCTiBsAbKSfRKyzv39ETKSsiVhkjuZvJ+Pxzwg955753MZIO+ce+45FsMwDERERES8lNXsAkRERERak8KOiIiIeDWFHREREfFqCjsiIiLi1RR2RERExKsp7IiIiIhXU9gRERERr6awIyIiIl7N1+wCPIHT6SQrK4uQkBAsFovZ5YiIiEgTGIZBWVkZCQkJWK2n7r9R2AGysrJITEw0uwwRERFpgYMHD9KlS5dT7lfYAUJCQoD6P6zQ0FCTqxEREZGmKC0tJTEx0fV9/FQUdsB16yo0NFRhR0REpJ050xAUDVAWERERr6awIyIiIl5NYUdERES8msKOiIiIeDWFHREREfFqCjsiIiLi1RR2RERExKsp7IiIiIhXU9gRERERr6awIyIiIl5NYUdERES8msKOiIiIeDWFHREREfFqCjsiIiLi1XzNLkA6juzsbIqKipp1TEREBPHx8a1UkYiIdAQKO9ImsrOz6d27Dw5HWbOOCw4OYdeudAUeERFpMYUdaRNFRUU4HGXc8eQ8ohOSmnTMkawDvPb4vRQVFSnsiIhIiynsSJuKTkgiLinZ7DJERKQD0QBlERER8WoKOyIiIuLVFHZERETEqynsiIiIiFdT2BERERGvprAjIiIiXk1hR0RERLyawo6IiIh4NYUdERER8WoKOyIiIuLVFHZERETEqynsiIiIiFdT2BERERGvprAjIiIiXk1hR0RERLyawo6IiIh4NVPDzssvv8zAgQMJDQ0lNDSU1NRUvvjiC9f+yy67DIvF0uh19913NzpHZmYmEyZMIDAwkJiYGB5++GFqa2vb+lKkFWVkZLB9+/Ymv7Kzs80uWUREPIivmW/epUsXnnnmGXr16oVhGLz++utMnDiRjRs3cs455wBw55138tRTT7mOCQwMdP2+rq6OCRMmEBcXx8qVK8nOzmbKlCn4+fnxpz/9qc2vR9zLUVwIWJg4cWKzjgsODmHXrnTi4+NbpzAREWlXTA07V111VaOvn376aV5++WVWr17tCjuBgYHExcWd9PivvvqK7du38/XXXxMbG8vgwYOZNWsWjzzyCE888QQ2m63Vr0HcyzAMACwWC5UVDsDg+hnP0KNfSpOOP5J1gNcev5eioiKFHRERAUwOO8erq6tj4cKFlJeXk5qa6tr+1ltv8eabbxIXF8dVV13FH/7wB1fvzqpVq0hJSSE2NtbVfsyYMdxzzz1s27aNIUOGnPS9qqqqqKqqcn1dWlraSlclTWEYBnvyy8nIc3C4+CgV1bV0CrHjVx2Ob0QCEbGdiUtKNrtMERFpp0wPO1u2bCE1NZXKykqCg4P58MMP6d+/PwC33HILSUlJJCQksHnzZh555BHS09P54IMPAMjJyWkUdADX1zk5Oad8z9mzZ/Pkk0+20hVJc9TUOflmZx47c8oabc8trQJCSLhjLhkVtfSqc+Lro/H0IiLSfKaHnT59+pCWlkZJSQnvv/8+U6dOZcWKFfTv35+77rrL1S4lJYX4+HhGjhzJnj176NmzZ4vfc+bMmcyYMcP1dWlpKYmJiWd1HdJ8jqpaPtx4mMLyaiwWGNwlnG7RQYT4+5JXWsXaHfso9A1gb6UfCzcc4prBnQmw+ZhdtoiItDOm/6hss9lITk5m6NChzJ49m0GDBvHXv/71pG2HDx8O1D+dAxAXF0dubm6jNg1fn2qcD4Ddbnc9AdbwkrZV5zT4Yks2heXVBNl8mDSkC5f07kTXyEAiAm30iQthsP0I+R/Nxs/iJK+sivd/PER5lZ60ExGR5jE97Pw3p9PZaDzN8dLS0gBcA09TU1PZsmULeXl5rjZLliwhNDTUdStMPNPKPUfIKqnE5mPl+qFd6BwRcEIbiwUq0n9gWGgZwXZfCsurWbjhEBXVCjwiItJ0poadmTNn8u2337J//362bNnCzJkzWb58OZMnT2bPnj3MmjWLDRs2sH//fj755BOmTJnCJZdcwsCBAwEYPXo0/fv357bbbmPTpk0sXryYRx99lOnTp2O32828NDmNvfkOfswsBuDK/rGEB57+qbkgHyc3DO1CqL8vJUdr+HxLDnVOow0qFRERb2Bq2MnLy2PKlCn06dOHkSNHsm7dOhYvXsyVV16JzWbj66+/ZvTo0fTt25eHHnqISZMm8emnn7qO9/HxYdGiRfj4+JCamsqtt97KlClTGs3LI56lzjD4dvcRAIYkhpMcE9yk40ID/Lh6UAI2HyuHi4+yLD3P9Zi6iIjI6Zg6QPnVV1895b7ExERWrFhxxnMkJSXx+eefu7MsaUV7S6DkaA2BNh9Se0Y169ioYDvjBsTxyaYstmWVEhfqz4DOYa1UqYiIeAuPG7Mj3svi58+2ovrfD+seiV8LHiXvFh3EhcdC0opd+RSWV7uzRBER8UKmP3ouHUfo+ddQVQdhAX4MSGh5j8zQpAgyiyo4WHiUL7fmcOP5XfC1mpPbs7OzKSoqatYxERERmt1ZRKQNKexIm6iqdRJy3tUApPaIwsdqafG5LBYLY/rH8daaTPIdVazaU8DFvTq5q9Qmy87OpnfvPjgcZWdufByt3SUi0rYUdqRNfH+gHJ+AUAJ9oVds0wYln06Q3ZdR/WL4dHM2P2YW06NTMJ3DT3x8vTUVFRXhcJRxx5PziE5IatIxWrtLRKTtKexIm/gsvb73o2coWC0t79U5Xo9OwfSPD2V7dilLtucyeXhXt5y3uaITkrR2l4iIB9MAZWl127JK2HmkCqOuhh5unqz6kt7RBNvr59/5PuOIe08uIiJeQWFHWt2bqzMBqNi1Cn9f9/TqNLD7+nBl//rFXzcfKiG3QnPviIhIYwo70qocVbV8nHYYgLKNX7TKe3SNDCTl2Hw7a/PAYmvbsTsiIuLZFHakVS3dkUtFdR0JIb5UHdzSau8zIjmasAA/Kmoh4or/abX3ERGR9kcDlKXZmjO3zLs/1K9CPyC0mlWtWJPN18qV/WJ5/8dDhAwaw9pDFWgtWBERAYUdaabmzC1jsQWQeP9bWHxtLPjjAwBUVJS3Wm2dIwLoHQa7SuBvq45wzYjqMy4yKiIi3k9hR5qlOXPLZJYZrMqFYD+YcMOtfDR3FlWVVa1aX0oUbNt7kMKoRB7/ZBt/vWlIq76fiIh4PoUdaZGmzC2zfnMWUE7fhAjC/BLapC5fq4Ujn71A56lz+Dgti7HnxDEuRZP3iYh0ZBqgLK2iutbJ/oIKAHrHhrTte2fv4oYB9U9n/f6jrRxxtG5vkoiIeDaFHWkV+wvKqXMahAX4ER3c9uNmbhkYQd+4EArLq3n0w60YhubfERHpqBR2pFXsL6gfiNyzUxAWNy0P0Rx+Phb+cuMgfK0WvtyWw4cbD7d5DSIi4hkUdsTtDMPgwLFbWN2igkyr45yEMH41shcAv/twC1sPl5hWi4iImEdhR9wu31FFRXUdfj4W4sP9Ta1l+uXJXNq7E5U1Tu56Y73G74iIdEAKO+J2Db06iRGB+FrN/SvmY7Xwt5uH0CM6iKySSu56Yz3lVbWm1iQiIm1Lj56L2zWEnaSoQJMrqRcW4Mc/p57HtXN/4MfMYn6xYB0LfnE+gTbz/vpnZGQ0q31ERATx8XqEXkSkJRR2xK2qauvIKjkKQJKJ43X+W89Owfxr2nBu/X9rWLuvkDsWrOP/bj2PsEC/sz63YRiUVdZQW2fgNAyC7b7Y/XxO2tZRXAhYmDhxYrPeIzg4hF270hV4RERaQGFH3Opg4VEMAyIC/QgLOPsg4U6DEsN5Y9owbnt1Lav3FvKzv3/Hy5OHMuDYiulNZRgGu3IdvLO5iNibnubDfVCzZ3+jNpFBNrpGBjKkazih/j/9OVRWOACD62c8Q49+KU16vyNZB3jt8XspKipS2BERaQGFHXGrA8ceOfekXp3jDekawbt3XcA9b23gYOFRrnt5JXdc1J27LulBZNCp5wOqcxr8mFnEV9ty+Gp7rutWnX/SIGqcYLWAn48VC1BZ66SwvJrC8mo2HypmQEIYFyZHYff9qbcnIrbzGWegFhER91DYEbc6WFR/C6trpGeM1zmZAZ3DWHTfxTy0cBNf78jllRV7+Neq/Yw5J44LekbRs1Mwdl8r5VW1ZOQ72HywhKU7czniqHadw+ZrZVCsnS9fe56b/2c6vZOT8bHWzydUUV1LVnElmw4Vc6joKJsPl5BZVMFVA9tmyQwREWlMYUfcpqyyhpKjNViABJMfOT+TsEA//jllKMvS85izZBdbD5fywcbDfHCayQdD/X0Z2S+WMefEcnGvThzYs4v3f/UFEfb7XEEHINDmS3JMMMkxwRwsrOCr7bkUV9Tw73UH6e9rb4vLExGR4yjsiNscLq7v1ekUYm90y8ZTWSwWrugby+V9Yli1t4AfMo6wem8heWWV1NQa+PlaSO4UTO/YEC7u1YnhPSLx82neo/SJkYHcPCyRz7Zkk1Vcyea6aGzxvVvpikRE5GQUdsRtDh+7hdUlIsDkSprHYrFwYc9oLuwZ3SrnD7T5ct2QLny6KYsDhRXEXP845XWt8lYiInISCjviNoeOhZ3OHhB2PG0eGx+rhfEp8bz53Q7KAsP4sayOgbVObL6a11NEpLUp7IhbOKpqKT42XqdzmHlhx5PnsbH5Whlsz2dZjh9Hw2JYviuP0f3jWu39RESknsKOuEXDLaxOIfZTTqjXFjx9HhubxcmRT58n7tY/syO7jKTIIPrEhbTqe4qIdHQKO+IWh4rq553xhFtY4Nnz2FQd3k4P/0r2Vgbwzc48EsL9CfH3rAkYRUS8iQYMiFscOvYkVpdwzwg7nq5HQCVxof5U1zn5bvcRs8sREfFqCjty1iqqaymuqAEgQWGnSawWuKJvDBZgd57DNfO0iIi4n8KOnLWckkoAIgNt+Js4Xqe96RRiZ1BiOADL0vOprXOaW5CIiJdS2JGzln0s7MR7+KzJnuiCHpEE2X0oOVrDjweLzS5HRMQrKezIWXOFnTCFneay+/owIrl+MsMN+4uoqK41uSIREe9jath5+eWXGThwIKGhoYSGhpKamsoXX3zh2l9ZWcn06dOJiooiODiYSZMmkZub2+gcmZmZTJgwgcDAQGJiYnj44YeprdU3jLZS5zTILW0IOxqv0xJ9YkOICbFTXedk7b5Cs8sREfE6poadLl268Mwzz7BhwwbWr1/PFVdcwcSJE9m2bRsADz74IJ9++ikLFy5kxYoVZGVlcd1117mOr6urY8KECVRXV7Ny5Upef/11FixYwGOPPWbWJXU4RxxV1DoN7L5WIgL1+HRLWCwWV+/OlsMlFFdUn+EIERFpDlPDzlVXXcX48ePp1asXvXv35umnnyY4OJjVq1dTUlLCq6++ypw5c7jiiisYOnQo8+fPZ+XKlaxevRqAr776iu3bt/Pmm28yePBgxo0bx6xZs5g7dy7V1fqG0RYabmHFhfljsVjO0FpOJTEykKSoQJwGrNpTYHY5IiJexWMmFayrq2PhwoWUl5eTmprKhg0bqKmpYdSoUa42ffv2pWvXrqxatYoLLriAVatWkZKSQmxsrKvNmDFjuOeee9i2bRtDhgw56XtVVVVRVVXl+rq0tLT1LszLZZfUz6+T4AW3sJq7nlZz25/JRT2jOVCQya48B+c7qogOtrv1/CIiHZXpYWfLli2kpqZSWVlJcHAwH374If379yctLQ2bzUZ4eHij9rGxseTk5ACQk5PTKOg07G/YdyqzZ8/mySefdO+FdFDeMDi5petpNaiocM8cOZ1C7PSKCWZ3noPVewv42cAEt5xXRKSjMz3s9OnTh7S0NEpKSnj//feZOnUqK1asaNX3nDlzJjNmzHB9XVpaSmJiYqu+pzdyVNVSVlmLBYgNbb9hpyXraQHsTlvNh3NnUVVZdebGTTS8eyS78xzsyS8nr6ySmJD2++cqIuIpTA87NpuN5OT6NYyGDh3KunXr+Otf/8rPf/5zqqurKS4ubtS7k5ubS1xc/UrRcXFxrF27ttH5Gp7WamhzMna7HbtdtwjOVsNkglHBNmy+7X8Wg+aup3UkK9PtNUQF2+kTG0J6bhlr9hZy1SD17oiInC2P+w7ldDqpqqpi6NCh+Pn5sXTpUte+9PR0MjMzSU1NBSA1NZUtW7aQl5fnarNkyRJCQ0Pp379/m9fe0TQ8ct6ee3U80fDukViAvUfKXX/GIiLScqb27MycOZNx48bRtWtXysrKePvtt1m+fDmLFy8mLCyMadOmMWPGDCIjIwkNDeX+++8nNTWVCy64AIDRo0fTv39/brvtNp599llycnJ49NFHmT59unpu2oDCTuuICLLRNy6EHTllrNpbwAURZlckItK+mRp28vLymDJlCtnZ2YSFhTFw4EAWL17MlVdeCcALL7yA1Wpl0qRJVFVVMWbMGObNm+c63sfHh0WLFnHPPfeQmppKUFAQU6dO5amnnjLrkjoMwzDILasfqxKnsON2w7pHsjO3jAMFFSTrj1dE5KyYGnZeffXV0+739/dn7ty5zJ0795RtkpKS+Pzzz91dmpyBowaqa534WC1EBtnMLsfrhAfa6B8fyrasUrZqUmURkbPicWN2pH0oOPYAUkyIHR+rJhNsDcO6RWK1QO5RsHc5x+xyRETaLYUdaZHCY+NmNV6n9YQG+NE/IRSA8BGTTa5GRKT9UtiRFik81rMTG6qB4K3p/G6RWAH/pIFsyjlqdjkiIu2Swo40n9WHYlfYUc9Oawr196NHWP3v30wrwjAMcwsSEWmHFHak2fyik6gzwO5rJTxAK523tn4RYNRWsy2viu8zjphdjohIu6OwI81mj+8FQEyoXSudt4FAXwtlG+ufOJyzZJd6d0REmklhR5rNFle/pILWbWo7JWvex+5jYWNmMcvT880uR0SkXVHYkWazxfQE6h87l7bhLC/mZ33qn8xS746ISPMo7Eiz1DoNbDHdAOiksNOmrh8QRqDNhy2HS/h6R96ZDxAREUBhR5rpYEkNFl8bvhY0OLmNhfn7MPXCbgA8vzidOqd6d0REmsLU5SKk/dl7bIKdCDsanGyCX17Sg7fXZJKeW8a/1x3kluFd3f4e2dnZFBUVNeuYiIgI4uPj3V6LiIg7KOxIs+wprAYgXHewTBEeaOOBkb14atF2/vJVOj8bFE+ov/t62LKzs+nduw8OR1mzjgsODmHXrnQFHhHxSAo70iwZx8JOhMKOaW5LTeLN1QfYe6ScucsymDmun9vOXVRUhMNRxh1PziM6IalJxxzJOsBrj99LUVGRwo6IeCSFHWkyp9NodBtLzOHnY+X3E/ox7fX1zP9+PzcMTSQ5Jtit7xGdkERcUrJbzykiYhYNUJYmO1hUQUWNgVFbTajN7Go6tiv6xnBZn05U1zn53QdbcGqwsojIKSnsSJNtPVwKQHX+fqwanGwqi8XCrIkDCPDzYe3+Qt5dd9DskkREPJbCjjTZtqwSAKpz95pciQAkRgby0OjeAMz+Ygc5JZUmVyQi4pkUdqTJtmYd69nJ3WNyJdLgFxd1Z1CXMMoqa3nw32mae0dE5CQUdqRJDMNgu6tnR2HHU/hYLcz5+WACbT6s2lvA3GUZZpckIuJxFHakSfLKqjjiqMZqgZr8/WaXI8fp2SmYWRMHAPDi17tYtafA5IpERDyLwo40ydbD9b06iWF+GLXVJlcj/23S0C5cd25nnAbc/eYGMvIcZpckIuIxFHakSbYdG6/TI1LPnHuqp69JYXBiOCVHa7h9/lryyjRgWUQEFHakiRqexEqO1GyCnirA5sOrU8+jW1Qgh4qOMuXVteSVKvCIiCjsSJM0zLHTUz07Hi0q2M6CXwyjU4idnTllXP/KKjILKswuS0TEVAo7ckbFFdUcLj4K6DZWe9AtOoj3706la2QgmYUVXPfySlbuOWJ2WSIiplHYkTPafmy8TtfIQIJtPiZXI02RFFUfePrGhXDEUcWt/28Nf/16N7V1TrNLExFpcwo7ckZbj43XOSch1ORKpDliQv354N4LuWFoF5wGvPD1LibO/YFNB4vNLk1EpE0p7MgZNTyJNaBzmMmVSHMF2nx57oZBzLlxEKH+vmzLKuWaeT8w84Mt5JdVmV2eiEibUNiRM2qYY6e/enbarevO7cLShy7j2iGdMQx4Z20mlz23jLnLMqisqTO7PBGRVqWwI6d1tLqOfUfKAd3Gau86hdh54eeDee+XqQzqEkZ5dR3PLU7niueX83HaYZxaV0tEvJTCjpzWrtwynAZEB9uICfE3uxxxg2HdI/nw3ot48eeDSQjzJ6ukkgfeTePaeT+wLVfz8oiI9/E1uwDxbDuy68fr9ItXr443sVotXDOkM2MHxPHq9/uYtyyDTYdK2HSohMjR91KrXh4R8SLq2ZHT2plTBkDfuBCTK5HW4O/nw/TLk1n+8OXcPCwRgJAh4/nqIFpuQkS8hnp25LS2q2en3crOzqaoqKjJ7W/r60uko46/rimiLCSahesPMW5AHD06BbdilSIirU9hR07JMAx2Hgs7feMUdtqT7Oxsevfug8NR1uxjrf7BDPntWxyp8mHR5mwu7dOJQV3C3V+kiEgbMTXszJ49mw8++ICdO3cSEBDAhRdeyJ///Gf69OnjanPZZZexYsWKRsf98pe/5JVXXnF9nZmZyT333MOyZcsIDg5m6tSpzJ49G19fZbmzkVVSSWllLb5WCz1jgswuR5qhqKgIh6OMO56cR3RCUpOP2522mg/nzmJwYCnZUUlsyypleXo+FmCgAo+ItFOmpoEVK1Ywffp0zj//fGpra/nd737H6NGj2b59O0FBP31zvfPOO3nqqadcXwcGBrp+X1dXx4QJE4iLi2PlypVkZ2czZcoU/Pz8+NOf/tSm1+NtGnp1kmOCsftqmYj2KDohibik5Ca3P5KVCYDVAiP7xhDg58P6A0UsS8/Hz8eq25ki0i6ZGna+/PLLRl8vWLCAmJgYNmzYwCWXXOLaHhgYSFxc3EnP8dVXX7F9+3a+/vprYmNjGTx4MLNmzeKRRx7hiSeewGbTwpUtpcHJHZvFYuHCnlHU1hmkHSpmyfZcAm0+JEWpl09E2hePehqrpKR+pt7IyMhG29966y2io6MZMGAAM2fOpKKiwrVv1apVpKSkEBsb69o2ZswYSktL2bZtW9sU7qU0OFksFguX9I6mX1wIBvDl1hxKjtaYXZaISLN4zKAWp9PJr3/9ay666CIGDBjg2n7LLbeQlJREQkICmzdv5pFHHiE9PZ0PPvgAgJycnEZBB3B9nZOTc9L3qqqqoqrqp3WBSktL3X05XsE1OFlhp0OzWCxc0TeGwopqckurWLQ5ixvPS8TPx6N+VhIROSWPCTvTp09n69atfP/9942233XXXa7fp6SkEB8fz8iRI9mzZw89e/Zs0XvNnj2bJ5988qzq9XaVNT8tE9EvXrexOjpfHysTUuJ5Z+1Bjjiq+XZXPiP7xZ75QBERD+ARP5rdd999LFq0iGXLltGlS5fTth0+fDgAGRkZAMTFxZGbm9uoTcPXpxrnM3PmTEpKSlyvgwcPnu0leJ2GZSKigmx0CrabXY54gBB/P8YNqP83tTWrlP3HwrCIiKcztWfHMAzuv/9+PvzwQ5YvX0737t3PeExaWhoA8fHxAKSmpvL000+Tl5dHTEwMAEuWLCE0NJT+/fuf9Bx2ux27Xd/AT2dn9rHByfEhWCwWk6sR+Cngu7ttcyRGBjI4MZy0g8V8vSOXWy9o+mPtIiJmMTXsTJ8+nbfffpuPP/6YkJAQ1xibsLAwAgIC2LNnD2+//Tbjx48nKiqKzZs38+CDD3LJJZcwcOBAAEaPHk3//v257bbbePbZZ8nJyeHRRx9l+vTpCjRnwTU4WZMJms5RXAhYmDhxYrOPrahwf+/LRT2jOFBQTlFFDSt25TNIEyyLiIczNey8/PLLQP3EgcebP38+t99+Ozabja+//poXX3yR8vJyEhMTmTRpEo8++qirrY+PD4sWLeKee+4hNTWVoKAgpk6d2mheHmm+nTkanOwpKiscgMH1M56hR7+UJh3TMDlgVWXVmRs3k6+PldHnxPHeuoPszCkjLsHtbyEi4lam38Y6ncTExBNmTz6ZpKQkPv/8c3eV1eEZhsGOY7exNDjZc0TEdm7yBIENkwO2lrhQf1K6hLH5UAk/5gNWTTopIp7LY57GEnOcbLHII+W1lBytwccCtQWH2F582LWvtcaCSPuT2iOK3bkOSmvqCD2v+bfYRETaisJOB3aqxSIDepxHzA1PcDRvP4MH/uykx7bGWBBpX/z9fBjRK5ol23MJu+hmCitqzS5JROSkFHY6sFMtFrm90GBLISR378Zt//ys0TGtORZE2p9+cSH8uCeXAgJ4a1MxI84zuyIRkRMp7MgJi0VuLMsGHCTGRhGX1HjpjtYeCyLti8ViYVA0fHMYFmeUkZFXRnKMxnmJiGfxiEkFxbMcKasGIDpEj+7LmXUKsFCxaxVOA575YqfZ5YiInEBhRxqprXNSVHEs7GjmZGmiohULsFrg6x15rNtfaHY5IiKNKOxIIwXl1RhAgJ8PQTY9TixNU1t4mDHJ9bevXvx6l8nViIg0prAjjRxx1A88jgq2aZkIaZYbU8Lw87HwQ0YBa/epd0dEPIfCjjRyxKFbWNIyscF+3HBeIgB/XareHRHxHAo70khDz45WOpeWuPeynurdERGPo7AjLoZhuMJOdLDN5GqkPeoSEajeHRHxOC0KO3v37nV3HeIByqvqqKxxYrFAZJDCjrSMendExNO0KOwkJydz+eWX8+abb1JZWenumsQkDb06EYE2fH3U6Scto94dEfE0LfqO9uOPPzJw4EBmzJhBXFwcv/zlL1m7dq27a5M2lq9bWOIm6t0REU/SorAzePBg/vrXv5KVlcVrr71GdnY2I0aMYMCAAcyZM4f8/Hx31ylt4KfxOhqcLGdHvTsi4knO6l6Fr68v1113HQsXLuTPf/4zGRkZ/OY3vyExMZEpU6aQnZ3trjqlDRTosXNxo3sv64mvtb53Z8OBIrPLEZEO7KzCzvr167n33nuJj49nzpw5/OY3v2HPnj0sWbKErKwsJk6c6K46pZXV1jkpPLZMhB47F3foEhHIpHO7APDSN7tNrkZEOrIWrXo+Z84c5s+fT3p6OuPHj+eNN95g/PjxWK312al79+4sWLCAbt26ubNWaUWFFdUYBvj7Wgmya5kIab6MjIwTto1ONFi4AZan5/PRtxvpHf1TkI6IiCA+Pr4tSxSRDqpFYefll1/mjjvu4Pbbbz/lf1YxMTG8+uqrZ1WctJ3jVzrXMhHSHI7iQsByyp7cqAkzCB5wBXe9+B/yP3zatT04OIRdu9IVeESk1bUo7OzefeYuaZvNxtSpU1tyejGBBidLS1VWOACD62c8Q49+KSfsL602+CITAnunctffFxFut3Ak6wCvPX4vRUVFCjsi0upaFHbmz59PcHAwN9xwQ6PtCxcupKKiQiGnHdJj53K2ImI7E5eUfML2OCCjMpvdeQ72VgUzvrfCjYi0rRYNUJ49ezbR0dEnbI+JieFPf/rTWRclbavxMhHq2RH3G9Y9EoDdeQ4Ky6tNrkZEOpoWhZ3MzEy6d+9+wvakpCQyMzPPuihpW5V11C8TAURpmQhpBdHBdnp2CgJg3X5NMigibatFYScmJobNmzefsH3Tpk1ERUWddVHStorrO3W0TIS0qvO71ffupOeU4agxTK5GRDqSFn1nu/nmm/nVr37FsmXLqKuro66ujm+++YYHHniAm266yd01SisrPnZXQeN1pDXFhvqTFBWIAWzXHIMi0oZaNEB51qxZ7N+/n5EjR+LrW38Kp9PJlClTNGanHWro2YkO0XgdaV3Du0dyoKCC/aXgE9rJ7HJEpINoUdix2Wz8+9//ZtasWWzatImAgABSUlJISkpyd33SBkpcPTsKO9K64sMCSIwI4GDRUcKGX292OSLSQbQo7DTo3bs3vXv3dlctYgYfX0p1G0va0LDukRwsOkzwwNEUVNSaXY6IdAAtCjt1dXUsWLCApUuXkpeXh9PpbLT/m2++cUtx0vr8ohIxALuvlWD7WWVfkSbpHB5AtD8cqfRj4dYSLj7P7IpExNu16LvbAw88wIIFC5gwYQIDBgzQ8gLtmK1T/RQC0cFaJkLahsViYUAkLM+Cz3eV8nBhBYmRgWaXJSJerEVh59133+W9995j/Pjx7q5H2pgtpj7saKVzaUuxgRaO7t9IQLfBvPD1LubcONjskkTEi7Xo0XObzUZy8onTwkv74xfTDYCoEI3XkbZVvOJ1AD7ceJj0nDKTqxERb9aisPPQQw/x17/+FcPQxGDtmWEYrttY6tmRtlads5sRSYEYBjz75U6zyxERL9ai21jff/89y5Yt44svvuCcc87Bz8+v0f4PPvjALcVJ6yqqrMMnKBwLEKllIsQEUwZHsOrgUZbuzGPFrnwu7a25d0TE/VrUsxMeHs61117LpZdeSnR0NGFhYY1e0j7sK6x/5jzYD/y0TISYoEuYjamp3QB48tNtVNc6T3+AiEgLtKhnZ/78+e6uQ0ywt6g+7ITrDpaY6IFRvfg47TB788t5feV+7rykh9kliYiXafGP87W1tXz99df83//9H2Vl9YMLs7KycDgcTT7H7NmzOf/88wkJCSEmJoZrrrmG9PT0Rm0qKyuZPn06UVFRBAcHM2nSJHJzcxu1yczMZMKECQQGBhITE8PDDz9Mba0mKzuTfQ1hR3ewxERhAX48MrYvAH9dupuckkqTKxIRb9OisHPgwAFSUlKYOHEi06dPJz8/H4A///nP/OY3v2nyeVasWMH06dNZvXo1S5YsoaamhtGjR1NeXu5q8+CDD/Lpp5+ycOFCVqxYQVZWFtddd51rf11dHRMmTKC6upqVK1fy+uuvs2DBAh577LGWXFqHsk89O+Ihrh/ahUGJ4Tiqapn5wWY9/CAibtWisPPAAw9w3nnnUVRUREBAgGv7tddey9KlS5t8ni+//JLbb7+dc845h0GDBrFgwQIyMzPZsGEDACUlJbz66qvMmTOHK664gqFDhzJ//nxWrlzJ6tWrAfjqq6/Yvn07b775JoMHD2bcuHHMmjWLuXPnUl1d3ZLL6xCqaus4VFIDKOyI+axWC89fPxCbj5Vl6fm8v+GQ2SWJiBdpUdj57rvvePTRR7HZGt//6NatG4cPH25xMSUlJQBERkYCsGHDBmpqahg1apSrTd++fenatSurVq0CYNWqVaSkpBAbG+tqM2bMGEpLS9m2bVuLa/F2u3Md1BlQd7SUAB+zqxGBXrEhPHhl/Vp7Ty3aTlbxUZMrEhFv0aIByk6nk7q6uhO2Hzp0iJCQkBYV4nQ6+fWvf81FF13EgAEDAMjJycFmsxEeHt6obWxsLDk5Oa42xwedhv0N+06mqqqKqqoq19elpaUtqrk925Fdf801efuxWAaZXI1IvTsv7s6X23LYdLCY6W//yLt3XYDd96c0np2dTVFRUbPPW11dfcIPZ2cSERFBfHx8s99LRDxPi8LO6NGjefHFF/nHP/4B1K9143A4ePzxx1u8hMT06dPZunUr33//fYuOb47Zs2fz5JNPtvr7eLId2fWDyqvz9wEKO+IZfH2svHTTEH720ndszCzmiU+2M/u6FKA+6PTu3QeHowWzLVss0MxxQMHBIezala7AI+IFWhR2/vKXvzBmzBj69+9PZWUlt9xyC7t37yY6Opp33nmn2ee77777WLRoEd9++y1dunRxbY+Li6O6upri4uJGvTu5ubnExcW52qxdu7bR+Rqe1mpo899mzpzJjBkzXF+XlpaSmJjY7Lrbs5059T071Xn7TK5EpLGuUYH87eYh/GLBOt5Zm0n/hFBuuyCJoqIiHI4y7nhyHtEJSU0+3+601Xw4dxbXz3iGHv1SmnTMkawDvPb4vRQVFSnsiHiBFoWdLl26sGnTJt599102b96Mw+Fg2rRpTJ48udGA5TMxDIP777+fDz/8kOXLl9O9e/dG+4cOHYqfnx9Lly5l0qRJAKSnp5OZmUlqaioAqampPP300+Tl5RETEwPAkiVLCA0NpX///id9X7vdjt3ecUflGoZx3G0shR3xPJf1ieE3o/vw3OJ0Hvt4KwF+PvQ/9l9LdEIScUlNX5vvSFYmABGxnZt1nIh4jxaFHQBfX19uvfXWs3rz6dOn8/bbb/Pxxx8TEhLiGmMTFhZGQEAAYWFhTJs2jRkzZhAZGUloaCj3338/qampXHDBBUD9LbX+/ftz22238eyzz5KTk8Ojjz7K9OnTO3SgOZ3c0iqKKmqwWqD6SKbZ5Yic1L2X9SS3tJI3Vh3g4fc3MePCaLNLEpF2qkVh54033jjt/ilTpjTpPC+//DIAl112WaPt8+fP5/bbbwfghRdewGq1MmnSJKqqqhgzZgzz5s1ztfXx8WHRokXcc889pKamEhQUxNSpU3nqqaeafkEdTEOvTpdQP/bV1ZhcjcjJWSwWnrz6HOqcBm+tyeQvPxwh/JKpODUHj4g0U4vCzgMPPNDo65qaGioqKrDZbAQGBjY57DRl4jB/f3/mzp3L3LlzT9kmKSmJzz//vEnvKbD9WNjpHmHjO5NrETkdi8XCrIkDCLL78o9v9xKWegPfZsG42BrCAvzOfAIREVo4z05RUVGjl8PhID09nREjRrRogLK0rZ059U+z9IjUOhHi+axWC78b349HLu6Es7qS3KPwr9UHWLWngKraE6fAEBH5b25b6rpXr14888wzJ/T6iOfZcVzPjkh7cWn3YLJf/zWxAVDnNFi7v5DXvt/Pt7vyKa7QbOkicmotHqB80pP5+pKVleXOU4qbVdbUsTe/frHWHgo70s7UFh7i0gRwBMSxem8hhRXVbDxYzMaDxSSE+9M/PpReMSHYfN32c5yIeIEWhZ1PPvmk0deGYZCdnc3f//53LrroIrcUJq1jV24ZTgMig2xEaJ0IaYcsFgu9YkNIjgnmQGEFaQeLySyoIKu4kqziSlbsyic5Jpj+8aF0Dm/6VBgi4r1aFHauueaaRl9bLBY6derEFVdcwV/+8hd31CWtpOEWVr/4ECwWi8nViLScxWKhW1QQ3aKCcFTWsiOnlO3ZpRRX1LAju4wd2WWE+vuS4AwCi3p6RDqyFq+NJe1TwzIR/eJCTa5ExH2C/X05v1sk5yVFkF1SyY7sUnblOiitrKWUSOJv/yuFNW69ay8i7Yj+9XcwP/XshAIdbwFU8W4Wi4WE8AASwgO4pHcnth4uYeXuXIjpzvoyA799hZzfLUK9miIdTIvCzvHrSp3JnDlzWvIW0gqOXyaib3wIFCvsiPfy87EypGsEHNrE52u3E5xyJav2FpBdcpTxKfH4+ejWlkhH0aKws3HjRjZu3EhNTQ19+vQBYNeuXfj4+HDuuee62umnJ8+SVVJJaWUtvlYLyTHB7Ck2uyKR1udncVLw+V8ZfsGF7DwazP6CCj7dnMXVAxPwVeAR6RBaFHauuuoqQkJCeP3114mIiADqJxr8xS9+wcUXX8xDDz3k1iLFPXZk1ffkJMcEY/fVk1jSsXS2V9O/b2c+SjvMwcKjfLo5m6sGxivwiHQALfpX/pe//IXZs2e7gg5AREQEf/zjH/U0lgfbmXP8eB2RjichPICJgzrja7WQWVjB8l35ZpckIm2gRWGntLSU/PwT/5PIz8+nrKzsrIuS1uF6Eis+xORKRMzTOSKAnw2MB2BbVinbskpMrkhEWluLws61117LL37xCz744AMOHTrEoUOH+M9//sO0adO47rrr3F2juIlrcLIeO5cOLikqiNQeUQAsS88nr6zS5IpEpDW1KOy88sorjBs3jltuuYWkpCSSkpK45ZZbGDt2LPPmzXN3jeIGFdW17CsoB3QbSwTg/G4RdI8Oos5p8NW2XOqchtkliUgraVHYCQwMZN68eRQUFLiezCosLGTevHkEBQW5u0Zxg125DgwDooPtdAqxm12OiOksFgtX9o8lwM+HgvJqNhwoMrskEWklZ/UYQnZ2NtnZ2fTq1YugoCAMQz8Zearjl4kQkXoBfj5c0jsagLX7CynS6ukiXqlFYaegoICRI0fSu3dvxo8fT3Z2NgDTpk3TY+ceqiHs9NctLJFG+sSGkBQZSJ3T4JudefqhTcQLtWienQcffBA/Pz8yMzPp16+fa/vPf/5zZsyYocfPPVCjmZNFPERGRkartG0Oi8XC5X1j+NfqAxwqOsr+ggq0VrqId2lR2Pnqq69YvHgxXbp0abS9V69eHDhwwC2FifsYhsFO12Pn6tkR8zmKCwELEydObPaxFRXlbq8nLMCPwYnhbDhQxA8ZR7giTr07It6kRWGnvLycwMDAE7YXFhZit2vwq6c5VHSUsqpabD5WenYKNrscESorHIDB9TOeoUe/lCYdszttNR/OnUVVZVWr1HReUgRbD5dQUF7Nfk0XJuJVWhR2Lr74Yt544w1mzZoF1HcDO51Onn32WS6//HK3Fihnr+EWVnJMsBY/FI8SEduZuKTkJrU9kpXZqrX4+/kwrFsk32UcYWsBWHxtrfp+ItJ2WhR2nn32WUaOHMn69euprq7mt7/9Ldu2baOwsJAffvjB3TXKWdqhW1giTTKwSxgbDxbjqKolKGWU2eWIiJu06Mf8AQMGsGvXLkaMGMHEiRMpLy/nuuuuY+PGjfTs2dPdNcpZ0mPnIk3j62PlvKT6Nf/Chk+iVhMNiniFZvfs1NTUMHbsWF555RV+//vft0ZN4mY7tACoSJOdkxDK6j35VIbFsnyfg4EDzK5IRM5Ws3t2/Pz82Lx5c2vUIq2grLKGAwUVgObYEWkKXx8rvcPrf//elhItIyHiBVp0G+vWW2/l1VdfdXct0goaxuskhPkTEaQBlyJNkRwGdZUODpXWsHhbjtnliMhZatEA5draWl577TW+/vprhg4desJ6WHPmzHFLcXL2tmeVANA/Qb06Ik3lZ7VQ9uMiwi+8ide+38f4lHizSxKRs9CssLN37166devG1q1bOffccwHYtWtXozYWi8V91clZ296wTERCmMmViLQvjh8/I3rETaw/UMSmg8UMSgw3uyQRaaFmhZ1evXqRnZ3NsmXLgPrlIf72t78RGxvbKsVJ82RnZ1NU1Hjl5g178wAIrSth+/btjfa11vT7It6grryIS7oF881eB/N/2MeLNw0xuyQRaaFmhZ3/XiDviy++oLzc/VO3S/NlZ2fTu3cfHI7jpn61+tL1wYVYfP24++c/o64076THtsb0+yLeYGK/UL7Z62DR5mxmju9HbKi/2SWJSAu0aMxOA60O7DmKiopwOMq448l5RCck1W+rMvjqIPhZYcbzr51wi7G1p98Xae96RdkZ1i2StfsL+deqA/xmTB+zSxKRFmhW2LFYLCd8w9QYHc8SnZDkmn6/MLsUyCU2NID4bl1OaNva0++LeIPbL+rG2v2FvLvuIA+M6qUlV0TaoWbfxrr99ttdi31WVlZy9913n/A01gcffOC+CqXF8svqe2yiQ7Q4q0hLXdk/lk4hdvLLqliyPVdPZom0Q80KO1OnTm309a233urWYsS9GsJOJ4UdkRbz87Fy43ldmLtsD++szVTYEWmHmhV25s+f31p1iJsZhkG+41jYCVbYETkbN53flXnL9/Dd7iMcKCgnKSrozAeJiMfQzWcvVVpZS3WtEx+LhUjNnCxyVhIjA7m4VycA3ll70ORqRKS5TA073377LVdddRUJCQlYLBY++uijRvtvv/1216DohtfYsWMbtSksLGTy5MmEhoYSHh7OtGnTcDgcbXgVnqnhFlZUsA0fqwaRi5ytW4Z1BeD9DQeprnWaXI2INIepYae8vJxBgwYxd+7cU7YZO3Ys2dnZrtc777zTaP/kyZPZtm0bS5YsYdGiRXz77bfcddddrV26x3MNTtYtLBG3GNkvhpgQO0cc1SzZnmt2OSLSDGc1z87ZGjduHOPGjTttG7vdTlxc3En37dixgy+//JJ169Zx3nnnAfDSSy8xfvx4nn/+eRISEtxec3vRMF4nRoOTRdyifqByIn9flsHbaw8wYaAGKou0Fx4/Zmf58uXExMTQp08f7rnnHgoKClz7Vq1aRXh4uCvoAIwaNQqr1cqaNWvMKNdj6LFzEfe7aVgiFgv8kFHA/iOaeVykvfDosDN27FjeeOMNli5dyp///GdWrFjBuHHjqKurAyAnJ4eYmJhGx/j6+hIZGUlOTs4pz1tVVUVpaWmjlzc5Wl2Ho6oW0JNYIu7UJSKQS3sfG6i8TpNyirQXHh12brrpJq6++mpSUlK45pprWLRoEevWrWP58uVndd7Zs2cTFhbmeiUmJrqnYA/RcAsrLMAPm69Hf8Qi7c7NDQOV1x/SQGWRdqJdfSfs0aMH0dHRrtW64+LiyMtrvLhlbW0thYWFpxznAzBz5kxKSkpcr4MHvetR0oZbWBqvI+J+I/vGEBtqp6BcA5VF2ot2FXYOHTpEQUEB8fH1AwNTU1MpLi5mw4YNrjbffPMNTqeT4cOHn/I8drud0NDQRi9vovE6Iq3H18fK9UPr15r793rv+kFJxFuZGnYcDgdpaWmkpaUBsG/fPtLS0sjMzMThcPDwww+zevVq9u/fz9KlS5k4cSLJycmMGTMGgH79+jF27FjuvPNO1q5dyw8//MB9993HTTfdpCexgBiN1xFpFTeeV3/r+7vd+RwuPmpyNSJyJqaGnfXr1zNkyBCGDBkCwIwZMxgyZAiPPfYYPj4+bN68mauvvprevXszbdo0hg4dynfffedaiBTgrbfeom/fvowcOZLx48czYsQI/vGPf5h1SaardRoUlVcDWhNLpLUkRQWR2iMKw6gfuyMins3UeXYuu+wyDMM45f7Fixef8RyRkZG8/fbb7iyrXSupBgMI8PMh0OZjdjkiXuvn5yeyam8B760/yP1XJGPVTOUiHqtdjdmRMyuqv4NFTIgdi0X/+Yq0lrED4gjx9+Vw8VF+2HPE7HJE5DQUdrxM8bGwo8HJIq3L38+HawZ3BuDf6zRQWcSTmXobS9yv+LieHRE5Ow3TXJzK+VE1/Av4cms2q3/cQlJ8tOtpURHxHAo73sRipbh+bLJmThY5C47iQsDCxIkTz9g2buqLEJfM6LsexUhfxq5d6Qo8Ih5GYceL+EYkUGeAn4+FsEA/s8sRabcqKxyAwfUznqFHv5TTtt1dbPDjEeh65e1s2/AJRUVFCjsiHkZhx4vYYnsAEB1sx6rBySJnLSK2M3FJyadtE55Qx6bv9+Fw2rDF9WqjykSkOTRA2YvYYurDjm5hibQdfz8fkjsFAxA8cLTJ1YjIySjseJGGnh1NJijSts5JqF9yJqj/pVTWaHFQEU+jsOMlDMP4qWdHYUekTXWJCCDIF6z2QL7PLDe7HBH5Lwo7XqLgaB0+QeFYgKggm9nliHQoFouFHsfWE168u8zcYkTkBAo7XmJvYf0z56G2+lWZRaRtdQsFw1nHtrwq9uY7zC5HRI6j74peYk9h/WyC4bqDJWKKQF8LR/f9CMB7WhxUxKMo7HiJPcd6dsJ1B0vENI7NXwHw/oZD1NRpoLKIp1DY8RINYSdSPTsipjmasZYIfx+OOKr4aluu2eWIyDEKO16gqLyaXEctoNtYIqZy1jG6V/2cO2+uPmByMSLSQGHHC2zLKgWgpjALm49mThYx07heoVgtsGpvARl5ejJLxBMo7HiBLYdLAKjOPf0KzSLS+mKCfbmibywAb67ONLkaEQGFHa+wtSHs5OwxuRIRAbgtNQmA//x4iIrqWpOrERGFHS+wNUs9OyKe5OLkaJKiAimrrOWTtCyzyxHp8LTqeTtXcrSGAwUVAFTnqmdHxGwZGfU/dIzq5s+rBRX8c3k6KUFlWCwnH08XERFBfHx8W5Yo0uEo7LRz247dwooL9uVApWZtFTGLo7gQsDBx4kQArP4hdJn+OnsK4dwrr6M6e9dJjwsODmHXrnQFHpFWpLDTzjXcwuoZaWONybWIdGSVFQ7A4PoZz9CjXwoAa3IN9pdB6vQ5DI89sWfnSNYBXnv8XoqKihR2RFqRwk47t+Vw/WPnyVGaYEfEE0TEdiYuKRmAYeFH2b/+EAfLLYxO6E6An4/J1Yl0TBqg3M41PInVK0rrRIh4mrhQfzqF2KlzGmw/Nh+WiLQ9hZ12rKyyhn1HygHoqXUiRDyOxWJhYJcwANIOFuN0GiZXJNIxKey0Yw0zJ3cODyDMX93jIp6ob2wIAX4+OKpqycjXQwQiZlDYaccabmGdkxBqciUiciq+PlZX786PmUUYhnp3RNqawk471hB2UjqHmVyJiJzOwC5h+Fgt5JZWkV1SaXY5Ih2Owk471rAm1gCFHRGPFmjzpW9cCFDfuyMibUthp51yVNWy99jgZIUdEc83JDEcgD355RSWV5tbjEgHo7DTTu3ILsUwfnq0VUQ8W1SwnZ6dggBYv7/Q5GpEOhaFnXZqy6GGW1ganCzSXpzXLRKAnblllB6tMbkakY5DYaed2qrxOiLtTlyoP10jAzEMWH9AY3dE2orCTjvVsCbWgASFHZH25PxuEQBszy6lolaPoYu0BYWddqiiupaMvPrJyVK6KOyItCedwwNICPOvX0JCQ3dE2oTCTju0I7sUpwHRwXZiQ/3NLkdEmsFisXBhz2gA9paCb1isyRWJeD9Tw863337LVVddRUJCAhaLhY8++qjRfsMweOyxx4iPjycgIIBRo0axe/fuRm0KCwuZPHkyoaGhhIeHM23aNBwO756SPe1g/S2swYnq1RFpjzpHBNSP3QHCLrrF7HJEvJ6pYae8vJxBgwYxd+7ck+5/9tln+dvf/sYrr7zCmjVrCAoKYsyYMVRW/jQD6eTJk9m2bRtLlixh0aJFfPvtt9x1111tdQmm2HSwGIBBXcJNrUNEWi61ZxQAQedcRmax5t0RaU2+Zr75uHHjGDdu3En3GYbBiy++yKOPPsrEiRMBeOONN4iNjeWjjz7ipptuYseOHXz55ZesW7eO8847D4CXXnqJ8ePH8/zzz5OQkNBm19KWNh8qBmDQsUnKRKT9iQv1p3MQHC734Z/rCxl7odkViXgvjx2zs2/fPnJychg1apRrW1hYGMOHD2fVqlUArFq1ivDwcFfQARg1ahRWq5U1a9a0ec1tobiimv0FFQCuxQVFpH0aFAVGXQ0bso6yLD3P7HJEvJbHhp2cnBwAYmMbD96LjY117cvJySEmJqbRfl9fXyIjI11tTqaqqorS0tJGr/Zi07HJBLtHBxEeaDO5GhE5GyE2C6XrPwHgj4u2U1PnNLkiEe/ksWGnNc2ePZuwsDDXKzEx0eySmqxhvI56dUS8Q8nKfxPmb2VPfjmvr9xvdjkiXsljw05cXBwAubm5jbbn5ua69sXFxZGX17jrt7a2lsLCQlebk5k5cyYlJSWu18GDB91cfevR4GQR72JUVzB1SP0yEnOW7OJQUYXJFYl4H48NO927dycuLo6lS5e6tpWWlrJmzRpSU1MBSE1Npbi4mA0bNrjafPPNNzidToYPH37Kc9vtdkJDQxu92gPDMFy3sTQ4WcR7jE4OZli3SCqq6/jdh1sxDM2sLOJOpoYdh8NBWloaaWlpQP2g5LS0NDIzM7FYLPz617/mj3/8I5988glbtmxhypQpJCQkcM011wDQr18/xo4dy5133snatWv54YcfuO+++7jpppu88kmsrJJKjjiq8LVaOCehfQQ0ETkzq8XC7Ekp2HytfLsrn4/SDptdkohXMTXsrF+/niFDhjBkyBAAZsyYwZAhQ3jssccA+O1vf8v999/PXXfdxfnnn4/D4eDLL7/E3/+nWYPfeust+vbty8iRIxk/fjwjRozgH//4hynX09oabmH1jQ/B38/H3GJExK16dgrmgZG9AHj84226nSXiRqbOs3PZZZedtrvWYrHw1FNP8dRTT52yTWRkJG+//XZrlOdxfhqcHG5qHSLSOu66pAdfbc9l08FiHng3jX/fdQG+Ph472kCk3dC/onYk7VjYGaywI+KV/HysvHTTEELsvmw4UMSLX+8+80EickYKO+1EndNg62ENThbxdl2jAvnTdSkAzF2eweJtp54zTESaRmGnndiT76C8uo5Amw/JMcFmlyMireiqQQlMSU3CMODX76a5ftARkZZR2GknGm5hpXQOw8dqMbcYEWl1j/2sPxf3iuZoTR3TXl/H4eKjZpck0m4p7LQTrskEdQtLpEPw9bEyd/K59IoJJre0ipv/sZrsEgUekZZQ2GknNjdMJqjBySIdRqi/H6/fMYyukYFkFlZwkwKPSIuY+ui5NE1lTR07susXKx2UqDWxRDqShPAA3rnrAm76xyoOFFQwad5KXr39fPrFN39i0ezsbIqKipp1TEREBPHx8c1+LxFPorDjgf77P6Sd+ZXUOg3C/a2UZO2jNPvEMTsZGRltWaKItKHO4QG8e1cqU15dw578cm54ZRUv3TKEy/vENPkc2dnZ9O7dB4ejrFnvHRwcwq5d6Qo80q4p7HiYk/2HFDL0KiJH/ZKsrasZ8OT40x5fUVHe2iWKiAk6hwfwwT0X8cs317N6byF3LFjH9MuS+fWoXk2aeLCoqAiHo4w7npxHdEJSk97zSNYBXnv8XoqKihR2pF1T2PEwJ/sPaXWOwQEHDBs2jHPGfnbS43anrebDubOoqqxqy3JFpA2FBfrxxh3DeeLTbby9JpO/L8tg7b5CnrthIElRQU06R3RCEnFJya1cqYhnUdjxUMf/h1R8eD9QQ3LXBOJO8R/akazMtitORExj87Xyp2tTuKBHFDP/s5m1+wsZ8+K3/GZ0H35xUXdNTSFyEgo7Hq68qpaSozUAxIX6n6G1iHQU58dY+NuEeP666gibcyr542c7eG/NXn6dGk23CNsJ7TWuTzoyhR0Pl1NaCUBUkA27VjoXEU4c2xc8cDQRV0xj1xG456P9lKxaSMmq98BZe8KxGtcnHZHCjofLLqkPO/Fh6tURkXonG9tXUWuwIR+yyv0IH3ELSVfcwvkxEOVff1tL4/qkI1PY8XDZx6aIjw8LMLkSEfE0/z3YuHsPg915Dpan51NSXcfSQzCkaxipPaI0rk86NIUdD1bnNMgtq/8pLD5cPTsi3qq542lO1d5isdA7NoTEiEBW7MonPbeMHzOL2ZtfTi+nnztKFWmXFHY8WH5ZFXVOA38/K+EB+o9KxNs4igsBCxMnTmzR8acafxNg82HsgDh6xwWzbGc+xUdrWE8sIUOvxjDOomCRdkphx4M1rIETHxaAxaLHSUW8TWWFAzC4fsYz9OiX0uTjmjr+pkd0MAnDA/h6Ry578suJHHUXW8qr6V3nbNJEhCLeQmHHgzUMTo7T4GQRrxYR27lZE/01Z/yNv58PE1Li+fK7daRXhpBTbWPhhkNcNSiBYLu+BUjHoGjvoQzDcIWdBIUdETkLFouFRD8Huf9+FD+Lk7yyKt5bf5CiimqzSxNpEwo7HqqiFhxVtVgtEKvJBEXEDaoObmV4aBnhgX6UVdaycP0h8sv0KLp4P4UdD5VfP1yHmBB//HRvXUTcJNDHyQ1Du9Ap2M7Rmjr+86MCj3g/fRf1UPn1d7BI0CPnIuJmgTZfJp3bmfgwf6pqnXy48TAFDgUe8V4KOx6qoWenc7gmExQR97P7+TBxcAIxIfU9PB9uPOxah0/E2yjseCBrYBhlx/7PiVfYEZFWYvf14ZohnYkKtlFeXcfHaYc5WlNndlkibqew44HsXfoD9Yt/BmjxTxFpRQF+PlwzuDPBdl+KKmpYtCmL2jqn2WWJuJXCjgfy73IOAAnq1RGRNhBs9+WawQnYfK1klVTy9c48DE21LF5EYccD2RMHABqvIyJtJyrYzs9S4rFYID2njLSDxWaXJOI2CjsepqLaiS2mO6AnsUSkbSVGBnJxcjQA32UcIe+oenfEOyjseJjt+ZVYrD4E+UKIvxb/FJG2NTgxnL5xIRgGrMwBn5Bos0sSOWsKOx5mU079BDsxuoMlIiawWCxc0TeGTsF2quqg0zW/o1oDlqWdU9jxMJty6ifYiQk0uRAR6bD8fKz8bGA8NivYE3ozb02BBixLu6aw40FKKmrYU1C/MF+senZExEShAX6kxoHhrOOrDAcL1x8yuySRFlPY8SCr9xVgADUFBwnwtZhdjoh0cHGBFoq/exOAP3y8lZ05pSZXJNIyCjseZNWeAgAqD2wyuRIRkXqlq9/nvM4BVNU6ufetH3FU1ZpdkkizKex4kJV7jgBQeWCzyZWIiDQweOiiTsSH+bM3v5zffbBF43ek3fHosPPEE09gsVgavfr27evaX1lZyfTp04mKiiI4OJhJkyaRm5trYsUtl19Wxa5cBwCVmVtMrkZE5Cdh/j78/ZYh+FotfLIpi7fXZppdkkizeHTYATjnnHPIzs52vb7//nvXvgcffJBPP/2UhQsXsmLFCrKysrjuuutMrLblVu2tv4XVI8KGs7LM5GpERBobmhTJb8f2AeDJT7ez9XCJyRWJNJ3Hhx1fX1/i4uJcr+jo+gmuSkpKePXVV5kzZw5XXHEFQ4cOZf78+axcuZLVq1ebXHXzfb87H4BB8Zo1WUQ8050X92BUvxiqa53c9/aPlFbWmF2SSJN4fNjZvXs3CQkJ9OjRg8mTJ5OZWd99umHDBmpqahg1apSrbd++fenatSurVq0yq9wWMQyDFbvqw87QBE2wIyKeyWKx8PwNg+gcHsD+ggr+9z+bNX5H2gWPDjvDhw9nwYIFfPnll7z88svs27ePiy++mLKyMnJycrDZbISHhzc6JjY2lpycnNOet6qqitLS0kYvM+3MKSO3tAp/PysDYu2m1iIicjrhgTbmTj4XPx8Ln2/J4Y1VB8wuSeSMPDrsjBs3jhtuuIGBAwcyZswYPv/8c4qLi3nvvffO6ryzZ88mLCzM9UpMTHRTxS3T0KuT2iMKm49HfyQiIgxODGfmuH4A/PGz7WzSCuni4XzNLqA5wsPD6d27NxkZGVx55ZVUV1dTXFzcqHcnNzeXuLi4055n5syZzJgxw/V1aWmpqYFnRXp92Lm0dyfgqGl1iIicTEZGxgnbhkcYXNg1kJWZFdz1+hr+NiGBELsPABEREcTHx7d1mSKn1K7CjsPhYM+ePdx2220MHToUPz8/li5dyqRJkwBIT08nMzOT1NTU057Hbrdjt3vG7SJHVS3rDxQCcFmfGCry1CUsIp7BUVwIWJg4ceJJ91vsQcRPfZFc4rlq1r/J//BpAIKDQ9i1K12BRzyGR4ed3/zmN1x11VUkJSWRlZXF448/jo+PDzfffDNhYWFMmzaNGTNmEBkZSWhoKPfffz+pqalccMEFZpfeZCszjlBTZ5AUFUi36CC255ldkYhIvcoKB2Bw/Yxn6NEv5aRtCisNlh6CwN6pTHxuEVEVmbz2+L0UFRUp7IjH8Oiwc+jQIW6++WYKCgro1KkTI0aMYPXq1XTq1AmAF154AavVyqRJk6iqqmLMmDHMmzfP5KqbZ/mu429hiYh4nojYzsQlJZ90XxxQE1TM8vR8NhfA5Z27tm1xIk3g0WHn3XffPe1+f39/5s6dy9y5c9uoIvcyDINlO+u7ci7ro7AjIu3TwM5hZBUdZVeeg1U5YPUPMbskkUb06I+JthwuIbukkkCbDxf2jDa7HBGRFrFYLFzRL4bwAD8qaiFqwoM4Nf+OeBCFHRN9ta1+Ha9Le3fC38/H5GpERFrO7uvD+JR4fCwQmDyM/2zTchLiORR2TLR4W/3kh2POOf2j8iIi7UGnEDtDjnVSv76xiHX7C80tSOQYhR2T7M13sDvPga/VwuV9YswuR0TELXqEgmPbMpwG3PPmBg4VVZhdkojCjlmWbK+/hZXaM4qwQD+TqxERcQ+LxULh4rn0iLBxxFHN/7y+HkdVrdllSQensGOShltYo/vHmlyJiIh7GTWVPH5FLJ1C7OzMKeOBdzZS59SAZTGPwo4JskuOsvHYWjKjFHZExAt1CvLln1POw+5rZenOPJ75YofZJUkHprBjgk83ZWEYcH63COLDAswuR0SkVQxODOf5GwYB8M/v9vHu2kyTK5KOSmHHBB9tzAJg4uDOJlciItK6rhqUwK9H9QLg0Y+2sjxda+JI21PYaWO7c8vYnl2Kr9XChBStGyMi3u+Bkb2YODiBWqfB3W9uYL0eSZc25tHLRXijj9IOA/XLQ0QE2UyuRkSkdWRkZDT6+o5zbBzOD2D94aNMfXUNz4yJo2ekvVGbiIgILR4qrUJhpw0ZhsHHabqFJSLey1FcCFiYOHHiCfssvnZibnwSEgdw77+3kfPWI9QWZbn2BweHsGtXugKPuJ3CThtaf6CIQ0VHCbb7MqqfnsISEe9TWeEADK6f8Qw9+qWcsL+6zmBZFhQTQa97/8HIzhDoZ+FI1gFee/xeioqKFHbE7RR22tDba+qfRBifEkeATWthiYj3iojtTFxS8kn33dClloUbDlFcUcO3eX5MOlc93dK6NEC5jRSWV/PZ5mwAJg9PMrkaERHzBNp8uW5IZ0L9fSk5WsP7Gw7hqNGkg9J6FHbayML1B6muc5LSOYxBieFmlyMiYqoQfz+uH9qF8AA/Sitr+eYQ+EXrB0FpHQo7bcDpNHj72GRat17Q1eRqREQ8Q0PgiQyycbQO4ib/mS05R80uS7yQwk4b+C7jCAcKKgjx9+WqQQlmlyMi4jGC7L7cMLQL0f5g9Q/m91/n8J8Nh8wuS7yMwk4b+H/f7QVg0rldCLRpTLiIyPH8/Xy4NAHK03+g1gkPLdzEHxdtp7bOaXZp4iUUdlrZpoPFfLf7CD5WC3dc1N3sckREPJKv1cKRj57h5oHhAPy/7/dxyz/XkFWs21py9hR2WtncZfWziE4clEDXqECTqxER8WQGtw2OYN7kcwmy+bB2fyHj//YdX2zJNrswaecUdlpRek4ZX23PxWKBey/vaXY5IiLtwviUeD771cUM7BJGcUUN97z1I3f/awO5pZVmlybtlMJOK5q3vL5XZ+w5cSTHhJhcjYhI+9EtOoj3776Q+y5Pxtdq4cttOYz8ywrmLc+gsqbO7PKknVHYaSWOqlq+230EgOmXn3wWUREROTWbr5XfjOnDp/ePYFBiOI6qWp79Mp0rnl/Ov1btV+iRJlPYaSXBdl++/e3lvHTzEAZ0DjO7HBGRdqtffCgf3nMhL/x8EAlh/mSVVPKHj7cx4s/fMOerdA4VVZhdong4PQfdioLtmldHRMQdrFYL1w7pwrgB8by3/iD/t2Ivh4uP8rdvMnhpWQYX9oxifEo8Y86JIzrYbna54mEUdkREpN3w9/NhSmo3bh7WlS+35rDguww2HCrjh4wCfsgo4PcfbqVHhI2Bcf4MigtgQKw/QbbGNzGqq6ux2WzNfu+IiAityN5OKeyIiEi74+dj5bwYC7fM/BmVvkEE9hlBYN8R2OOS2VtUzd6iaj7aUYrhrKMm/wBV2elUZaVTnZVOTeFhMJo/YWFwcAi7dqUr8LRDCjutLDs7m6Kioia3z8jIaMVqRES8R1FREQ5HGXc8+WeiE+oXEa2sNcg7CrlHIe8oOGp8sMX2wBbbg5DB4wBwVpUT7mfQOSqYKH+IsoO/r+W073Uk6wCvPX4vRUVFCjvtkMJOK8rOzqZ37z44HGXNPraiorwVKhIR8T7RCUnEJf301Gu34/aVVdaQU1JJTumxV3EF2IMoBUqP+zk0xN+H+FB/YsP8iQv1JybEjq+PnuHxFgo7reinnzrmuX7qOJPdaav5cO4sqiqrWrk6ERHvF+LvR4i/H71i6+c627zyG96a9ywjpz+DJSyenJJKCiuqKauspazSwa48BwBWC0QH24kL9ScuzB+fasPMy5CzpLDTBv77p47TOZKV2crViIh4rubcym/JbX+rBWry9pHoX02//rEAVNXWkVta9VMPUEklR2vqyCurIq+sis2HSwDoct+b/GlFLlcWB3JBjyh6xQRjtZ7+9pd4BoUdERExnaO4ELAwceLEZh97trf97b4+dI0MpGtk/fqFhmFQVlnrCj45pZXklVZCUDjfH6jg+wPbAIgMsjG8eyTDu0dyQc8oeseEKPx4KIUdERExXWWFAzC4fsYz9OiX0qRjWuu2v8ViITTAj9AAP3ofu/11eP9u/v7kw/z2+f/HXocP6w8UUlhezRdbc/hiaw4AEYF+DO8exfAekVzQI4o+sQo/nkJhR0REPEZEbGePvO3vY7FQdXgH5wUe4aaBydQMD2Z3QRVbcivZnHOU7flVFFXU8OW2HL7cVh9+QmxWzk0M4dJ+nbmgRxR94xR+zKKwIyIicgZnvM1m9cUWl4x/1xT8u6Zg79yfMvxZsaeEFXvqx/yEBfgxvHsk53WLIKVzOAM6hxLi79d2F9GBeU3YmTt3Ls899xw5OTkMGjSIl156iWHDhpldloiIeIHm3mZzGgb7DuWwZPGXpF4zlf0OKyVHa/hqey5fbc91tesc6kdypI1e0XZ6RdroFmGja1y05vJxM68IO//+97+ZMWMGr7zyCsOHD+fFF19kzJgxpKenExMTY3Z5IiLiJZpzm62ipIjSNf9h8Zr3weqDLbYn/l1TsMX3xh7XC9+wGA6X1nC4tIYV+38aZO2s2Mx5vbvQv0skPaKDSYwMpEtEAF0iAtQT1EJeEXbmzJnDnXfeyS9+8QsAXnnlFT777DNee+01/vd//9fk6kREpCM6U29QZZ1BUSUUVUFhVf2vFbVgDQznx0MOfjzkOOGYsAA/YkPtRAXZiQ6xEx1sIyrIRpDdt/5l8yXI7uP6fYDNB1+rBZ/jXr6uX634WC1YLfWDsi2A5djvvU27DzvV1dVs2LCBmTNnurZZrVZGjRrFqlWrTKxMRETk9L1B3f7r64P7djP38Qd5eNZfqPKP4HBpDXnlteQ5aimtclJytIaSozXAiUHI3RrCz3//vv5rS30wOrbTwn//3tL4WKuFT+67mG7RQa1e98m0+7Bz5MgR6urqiI2NbbQ9NjaWnTt3nvSYqqoqqqp+elSxpKR+8Fhpaalba3M46v8yZu3dSdXRiiYdk3doX/2vmXsIDPBv8nu15Li2OsbT69M1tY/6dE3toz5vvKa2fK9Du7dSnbObp++8+sSdfv74hnTCJygMn8BwrIGh+ASGYQ0IxS8gmKHDU6m1+FJVa3C0xkllrUFVnYHTAKcT6oz635tl7949RNq6ufWcDd+3DeMMF2a0c4cPHzYAY+XKlY22P/zww8awYcNOeszjjz9uAHrppZdeeumllxe8Dh48eNqs0O57dqKjo/Hx8SE3N7fR9tzcXOLi4k56zMyZM5kxY4bra6fTSWFhIVFRUSe9V1laWkpiYiIHDx4kNDTUvRfg4XTtunZde8eha9e1t7drNwyDsrIyEhISTtuu3Ycdm83G0KFDWbp0Kddccw1QH16WLl3Kfffdd9Jj7HY7dru90bbw8PAzvldoaGi7+4vgLrp2XXtHo2vXtXc07fXaw8LCztim3YcdgBkzZjB16lTOO+88hg0bxosvvkh5ebnr6SwRERHpuLwi7Pz85z8nPz+fxx57jJycHAYPHsyXX355wqBlERER6Xi8IuwA3Hfffae8bXW27HY7jz/++Am3vjoCXbuuvaPRtevaO5qOcO0WwzjT81oiIiIi7ZfV7AJEREREWpPCjoiIiHg1hR0RERHxal4ZdubOnUu3bt3w9/dn+PDhrF279rTtFy5cSN++ffH39yclJYXPP/+80X7DMHjssceIj48nICCAUaNGsXv37kZtCgsLmTx5MqGhoYSHhzNt2jTXchENNm/ezMUXX4y/vz+JiYk8++yz7rng47T1te/fv59p06bRvXt3AgIC6NmzJ48//jjV1dWN2lgslhNeq1evbtfXDtCtW7cTruuZZ55p1MYbP/fly5ef9DO1WCysW7cOaL+f+wcffMDo0aNdk4ympaWdcI7KykqmT59OVFQUwcHBTJo06YSJTTMzM5kwYQKBgYHExMTw8MMPU1tbe9bXe7y2vvbCwkLuv/9++vTpQ0BAAF27duVXv/qVa8mdBif73N999123XHMDMz73yy677ITruvvuuxu18cbP/VT/li0WCwsXLnS1a4vPvcXOfsEGz/Luu+8aNpvNeO2114xt27YZd955pxEeHm7k5uaetP0PP/xg+Pj4GM8++6yxfft249FHHzX8/PyMLVu2uNo888wzRlhYmPHRRx8ZmzZtMq6++mqje/fuxtGjR11txo4dawwaNMhYvXq18d133xnJycnGzTff7NpfUlJixMbGGpMnTza2bt1qvPPOO0ZAQIDxf//3f+362r/44gvj9ttvNxYvXmzs2bPH+Pjjj42YmBjjoYcecp1j3759BmB8/fXXRnZ2tutVXV3drq/dMAwjKSnJeOqppxpdl8PhcO331s+9qqqq0TVnZ2cb//M//2N0797dcDqdhmG038/9jTfeMJ588knjn//8pwEYGzduPOE8d999t5GYmGgsXbrUWL9+vXHBBRcYF154oWt/bW2tMWDAAGPUqFHGxo0bjc8//9yIjo42Zs6c2a6vfcuWLcZ1111nfPLJJ0ZGRoaxdOlSo1evXsakSZMatQOM+fPnN/rcj/930x6v3TAM49JLLzXuvPPORtdVUlLi2u+tn3ttbe0J/96ffPJJIzg42CgrK3O1a+3P/Wx4XdgZNmyYMX36dNfXdXV1RkJCgjF79uyTtr/xxhuNCRMmNNo2fPhw45e//KVhGIbhdDqNuLg447nnnnPtLy4uNux2u/HOO+8YhmEY27dvNwBj3bp1rjZffPGFYbFYjMOHDxuGYRjz5s0zIiIijKqqKlebRx55xOjTp89ZXvFPzLj2k3n22WeN7t27u75u+KZ3sv883MWsa09KSjJeeOGFU9bVUT736upqo1OnTsZTTz3l2tYeP/fjnar+4uJiw8/Pz1i4cKFr244dOwzAWLVqlWEYhvH5558bVqvVyMnJcbV5+eWXjdDQ0EZ/F86GGdd+Mu+9955hs9mMmpoa1zbA+PDDD5t2IS1g1rVfeumlxgMPPHDKujrS5z548GDjjjvuaLSttT/3s+FVt7Gqq6vZsGEDo0aNcm2zWq2MGjWKVatWnfSYVatWNWoPMGbMGFf7ffv2kZOT06hNWFgYw4cPd7VZtWoV4eHhnHfeea42o0aNwmq1smbNGlebSy65BJvN1uh90tPTKSoqOssrN+/aT6akpITIyMgTtl999dXExMQwYsQIPvnkk2Zd3+mYfe3PPPMMUVFRDBkyhOeee65Rl3VH+dw/+eQTCgoKTjpreXv63Jtiw4YN1NTUNDpP37596dq1a6P/E1JSUhpNbDpmzBhKS0vZtm1bk9/rVMy69pMpKSkhNDQUX9/G07ZNnz6d6Ohohg0bxmuvvXbmVambyOxrf+utt4iOjmbAgAHMnDmTioqKRu/TET73DRs2kJaWxrRp007Y11qf+9nymkkFAY4cOUJdXd0JMyfHxsayc+fOkx6Tk5Nz0vY5OTmu/Q3bTtcmJiam0X5fX18iIyMbtenevfsJ52jYFxER0eTrPBmzrv2/ZWRk8NJLL/H888+7tgUHB/OXv/yFiy66CKvVyn/+8x+uueYaPvroI66++urmXehJmHntv/rVrzj33HOJjIxk5cqVzJw5k+zsbObMmeM6T0f43F999VXGjBlDly5dXNva4+feFDk5OdhsthPW0/vvP8OTvU/DvrNl1rWfrI5Zs2Zx1113Ndr+1FNPccUVVxAYGMhXX33Fvffei8Ph4Fe/+lWL3+v49zTr2m+55RaSkpJISEhg8+bNPPLII6Snp/PBBx+c9n0a9p0tT/ncX331Vfr168eFF17YaHtrfu5ny6vCjpjr8OHDjB07lhtuuIE777zTtT06OrrRKvPnn38+WVlZPPfcc275pmem469r4MCB2Gw2fvnLXzJ79myvno30eIcOHWLx4sW89957jbZ78+cu9StlT5gwgf79+/PEE0802veHP/zB9fshQ4ZQXl7Oc8895xHf9M7G8aEuJSWF+Ph4Ro4cyZ49e+jZs6eJlbWdo0eP8vbbbzf6jBt48ufuVbexoqOj8fHxOeGpiNzcXOLi4k56TFxc3GnbN/x6pjZ5eXmN9tfW1lJYWNiozcnOcfx7nA2zrr1BVlYWl19+ORdeeCH/+Mc/zljv8OHDycjIOGO7pjD72o83fPhwamtr2b9//2nf5/j3OBuecO3z588nKiqqSQHG0z/3poiLi6O6upri4uJTnqc9fu7NUVZWxtixYwkJCeHDDz/Ez8/vtO2HDx/OoUOHqKqqavZ7/Tezr/14w4cPB3D9nfb2zx3g/fffp6KigilTppyxrTs/97PlVWHHZrMxdOhQli5d6trmdDpZunQpqampJz0mNTW1UXuAJUuWuNp3796duLi4Rm1KS0tZs2aNq01qairFxcVs2LDB1eabb77B6XS6/jGkpqby7bffUlNT0+h9+vTpc9a3MsC8a4f6Hp3LLruMoUOHMn/+fKzWM/+1SktLIz4+vlnXeCpmXvt/S0tLw2q1um5revPnDvWPqM+fP58pU6ac8RseeP7n3hRDhw7Fz8+v0XnS09PJzMxs9H/Cli1bGv0QtGTJEkJDQ+nfv3+T3+tUzLp2qP+7MHr0aGw2G5988gn+/v5nPCYtLY2IiAi39Haaee3/reER7Ya/0978uTd49dVXufrqq+nUqdMZ27rzcz9rJg+Qdrt3333XsNvtxoIFC4zt27cbd911lxEeHu4aHX/bbbcZ//u//+tq/8MPPxi+vr7G888/b+zYscN4/PHHT/oYbnh4uPHxxx8bmzdvNiZOnHjSR8+HDBlirFmzxvj++++NXr16NXr0vLi42IiNjTVuu+02Y+vWrca7775rBAYGuv0R5La+9kOHDhnJycnGyJEjjUOHDjV65LDBggULjLffftvYsWOHsWPHDuPpp582rFar8dprr7Xra1+5cqXxwgsvGGlpacaePXuMN9980+jUqZMxZcoU1zm89XNv8PXXXxuAsWPHjhPqaq+fe0FBgbFx40bjs88+MwDj3XffNTZu3Njo7/Tdd99tdO3a1fjmm2+M9evXG6mpqUZqaqprf8MjyKNHjzbS0tKML7/80ujUqZPbH0Fu62svKSkxhg8fbqSkpBgZGRmN/r3X1tYahmEYn3zyifHPf/7T2LJli7F7925j3rx5RmBgoPHYY4+162vPyMgwnnrqKWP9+vXGvn37jI8//tjo0aOHcckll7jO4a2fe4Pdu3cbFovF+OKLL06oqy0+97PhdWHHMAzjpZdeMrp27WrYbDZj2LBhxurVq137Lr30UmPq1KmN2r/33ntG7969DZvNZpxzzjnGZ5991mi/0+k0/vCHPxixsbGG3W43Ro4caaSnpzdqU1BQYNx8881GcHCwERoaavziF79oNP+AYRjGpk2bjBEjRhh2u93o3Lmz8cwzz7j3wo22v/b58+cbwElfDRYsWGD069fPCAwMNEJDQ41hw4Y1emy3vV77hg0bjOHDhxthYWGGv7+/0a9fP+NPf/qTUVlZ2eg83vi5N7j55psbzS9zvPb6uZ/q7/Tjjz/uanP06FHj3nvvNSIiIozAwEDj2muvPeEbw/79+41x48YZAQEBRnR0tPHQQw81ejy7PV77smXLTvnvfd++fYZh1E+7MXjwYCM4ONgICgoyBg0aZLzyyitGXV1du772zMxM45JLLjEiIyMNu91uJCcnGw8//HCjeXYMwzs/9wYzZ840EhMTT/pZttXn3lJa9VxERES8mleN2RERERH5bwo7IiIi4tUUdkRERMSrKeyIiIiIV1PYEREREa+msCMiIiJeTWFHREREvJrCjoiIiHg1hR0Rabcuu+wyfv3rX5tdhoh4OIUdERER8WoKOyIiblZdXW12CSJyHIUdEXE7p9PJ7Nmz6d69OwEBAQwaNIj3338fgOXLl2OxWFi8eDFDhgwhICCAK664gry8PL744gv69etHaGgot9xyCxUVFa5zlpeXM2XKFIKDg4mPj+cvf/lLs2qaN28evXr1wt/fn9jYWK6//vpG9T777LMkJydjt9vp2rUrTz/9tGv/li1buOKKKwgICCAqKoq77roLh8Ph2n/77bdzzTXX8PTTT5OQkECfPn0AOHjwIDfeeCPh4eFERkYyceJE9u/f35I/UhE5C75mFyAi3mf27Nm8+eabvPLKK/Tq1Ytvv/2WW2+9lU6dOrnaPPHEE/z9738nMDCQG2+8kRtvvBG73c7bb7+Nw+Hg2muv5aWXXuKRRx4B4OGHH2bFihV8/PHHxMTE8Lvf/Y4ff/yRwYMHn7Ge9evX86tf/Yp//etfXHjhhRQWFvLdd9+59s+cOZN//vOfvPDCC4wYMYLs7Gx27twJ1IesMWPGkJqayrp168jLy+N//ud/uO+++1iwYIHrHEuXLiU0NJQlS5YAUFNT4zruu+++w9fXlz/+8Y+MHTuWzZs3Y7PZ3PAnLSJNYvay6yLiXSorK43AwEBj5cqVjbZPmzbNuPnmm41ly5YZgPH111+79s2ePdsAjD179ri2/fKXvzTGjBljGIZhlJWVGTabzXjvvfdc+wsKCoyAgADjgQceOGNN//nPf4zQ0FCjtLT0hH2lpaWG3W43/vnPf5702H/84x9GRESE4XA4XNs+++wzw2q1Gjk5OYZhGMbUqVON2NhYo6qqytXmX//6l9GnTx/D6XS6tlVVVRkBAQHG4sWLz1iziLiPenZExK0yMjKoqKjgyiuvbLS9urqaIUOGuL4eOHCg6/exsbEEBgbSo0ePRtvWrl0LwJ49e6iurmb48OGu/ZGRka7bRWdy5ZVXkpSURI8ePRg7dixjx47l2muvJTAwkB07dlBVVcXIkSNPeuyOHTsYNGgQQUFBrm0XXXQRTqeT9PR0YmNjAUhJSWnUW7Np0yYyMjIICQlpdL7Kykr27NnTpLpFxD0UdkTErRrGsnz22Wd07ty50T673e76Ru/n5+fabrFYGn3dsM3pdLqlppCQEH788UeWL1/OV199xWOPPcYTTzzBunXrCAgIcMt7HB+GoP7PYejQobz11lsntD3+dp6ItD4NUBYRt+rfvz92u53MzEySk5MbvRITE1t0zp49e+Ln58eaNWtc24qKiti1a1eTz+Hr68uoUaN49tln2bx5M/v37+ebb76hV69eBAQEsHTp0pMe169fPzZt2kR5eblr2w8//IDVaj1tz9K5557L7t27iYmJOeHPISwsrMl1i8jZU8+OiLhVSEgIv/nNb3jwwQdxOp2MGDGCkpISfvjhB0JDQ0lKSmr2OYODg5k2bRoPP/wwUVFRxMTE8Pvf/x6rtWk/ry1atIi9e/dyySWXEBERweeff47T6aRPnz74+/vzyCOP8Nvf/habzcZFF11Efn4+27ZtY9q0aUyePJnHH3+cqVOn8sQTT5Cfn8/999/Pbbfd5rqFdTKTJ0/mueeeY+LEiTz11FN06dKFAwcO8MEHH/Db3/6WLl26NPvPQURaRmFHRNxu1qxZdOrUidmzZ7N3717Cw8M599xz+d3vftfiW1PPPfccDoeDq666ipCQEB566CFKSkqadGx4eDgffPABTzzxBJWVlfTq1Yt33nmHc845B4A//OEP+Pr68thjj5GVlUV8fDx33303AIGBgSxevJgHHniA888/n8DAQCZNmsScOXNO+56BgYF8++23PPLII1x33XWUlZXRuXNnRo4cSWhoaIv+DESkZSyGYRhmFyEiIiLSWjRmR0RERLyawo6ItHvfffcdwcHBp3yJSMem21gi0u4dPXqUw4cPn3J/cnJyG1YjIp5GYUdERES8mm5jiYiIiFdT2BERERGvprAjIiIiXk1hR0RERLyawo6IiIh4NYUdERER8WoKOyIiIuLVFHZERETEq/1/O2c/SzboqhcAAAAASUVORK5CYII=",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sns.histplot(column_data, kde=True)\n",
    "plt.xlabel('emd_score')  # Set the x-axis label to your column's name\n",
    "plt.ylabel('Frequency')    # Set the y-axis label\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Original range: [-7.474862721646301, -3.9923918933971856]\n",
      "Scaled range: [0.0, 1.0]\n"
     ]
    }
   ],
   "source": [
    "import numpy as np\n",
    "\n",
    "\n",
    "# Log transform\n",
    "log_transformed_data = np.log(column_data)\n",
    "\n",
    "min_val = np.min(log_transformed_data)\n",
    "max_val = np.max(log_transformed_data)\n",
    "\n",
    "\n",
    "scaled_values = (log_transformed_data - min_val) / (max_val - min_val)\n",
    "\n",
    "# Verify the results\n",
    "print(f\"Original range: [{min_val}, {max_val}]\")\n",
    "print(f\"Scaled range: [{scaled_values.min()}, {scaled_values.max()}]\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAjsAAAGxCAYAAACEFXd4AAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjkuMCwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy80BEi2AAAACXBIWXMAAA9hAAAPYQGoP6dpAABmwElEQVR4nO3dd3xUZb4G8Gd66iSZJJNCQiBA6EVAMAqKglRdUPZaQESXxYuC68paltW1gIq6iqiLoHsV1FWxYUMEkWahKEgPhCQEkpBMQuokmUx/7x8hI5FAkmEyZ8rz/XzmIzPnPWd+c0x5cs5bZEIIASIiIqIAJZe6ACIiIqKOxLBDREREAY1hh4iIiAIaww4REREFNIYdIiIiCmgMO0RERBTQGHaIiIgooDHsEBERUUBTSl2AL3A6nSguLkZkZCRkMpnU5RAREVEbCCFQW1uL5ORkyOXnv37DsAOguLgYqampUpdBREREbigsLERKSsp5tzPsAIiMjATQeLK0Wq3E1RAREVFbGI1GpKamun6Pnw/DDuC6daXVahl2iIiI/ExrXVDYQZmIiIgCGsMOERERBTSGHSIiIgpoDDtEREQU0Bh2iIiIKKAx7BAREVFAY9ghIiKigCZp2Fm+fDkGDBjgmt8mMzMT33zzjWv7qFGjIJPJmj3mzJnT7BgFBQWYNGkSwsLCoNfr8eCDD8Jut3v7oxAREZGPknRSwZSUFDz77LPo0aMHhBB4++23MXnyZOzduxd9+/YFAMyePRsLFy507RMWFub6t8PhwKRJk5CYmIjt27ejpKQEt99+O1QqFZ555hmvfx4iIiLyPTIhhJC6iLPpdDr861//wqxZszBq1CgMGjQIS5cubbHtN998g+uuuw7FxcVISEgAAKxYsQIPP/wwTp8+DbVa3ab3NBqNiIqKQk1NDWdQJiIi8hNt/f3tM312HA4HVq9ejfr6emRmZrpef++99xAXF4d+/fphwYIFMJlMrm07duxA//79XUEHAMaNGwej0YjDhw97tX4iIiLyTZKvjXXw4EFkZmbCbDYjIiICn332Gfr06QMAmDZtGtLS0pCcnIwDBw7g4YcfRnZ2NtasWQMAMBgMzYIOANdzg8Fw3ve0WCywWCyu50aj0dMfi4iIiHyE5GGnZ8+e2LdvH2pqavDJJ59g5syZ2LZtG/r06YO77rrL1a5///5ISkrC6NGjkZeXh27durn9nosXL8aTTz7pifKJiIjIx0l+G0utVqN79+4YMmQIFi9ejIEDB+Lll19use3w4cMBALm5uQCAxMRElJaWNmvT9DwxMfG877lgwQLU1NS4HoWFhZ74KEREROSDJL+y83tOp7PZLaaz7du3DwCQlJQEAMjMzMTTTz+NsrIy6PV6AMDGjRuh1Wpdt8JaotFooNFoPFs4EZEPMBqNzfo2tiYsLIwDMyjgSRp2FixYgAkTJqBz586ora3F+++/j61bt2LDhg3Iy8vD+++/j4kTJyI2NhYHDhzA/fffjyuvvBIDBgwAAIwdOxZ9+vTBjBkz8Pzzz8NgMODRRx/F3LlzGWaIKOgYjUZ0Te+GyoryNu+ji41D/vE8nwg8DGrUUSQNO2VlZbj99ttRUlKCqKgoDBgwABs2bMC1116LwsJCfPfdd1i6dCnq6+uRmpqKqVOn4tFHH3Xtr1AosHbtWtx9993IzMxEeHg4Zs6c2WxeHiKiYGEymVBZUY4HVnyBiOjYVtvXVVfghTmTYTKZJA8N/h7UyLdJGnbefPPN825LTU3Ftm3bWj1GWloa1q1b58myiIj8WkR0LLS6eKnLaBd/Dmrk+3yuzw4REQUvfwxq5PskH41FRERE1JEYdoiIiCigMewQERFRQGPYISIiooDGsENEREQBjWGHiIiIAhrDDhEREQU0hh0iIiIKaAw7REREFNAYdoiIiCigMewQERFRQOPaWERE5LfKysra3DYsLIyLhgYphh0iIvI7ZlM9IJNh4MCBbd5HFxuH/ON5DDxBiGGHiIj8js1qBoTAPUtWQ5+c0mr7uuoKvDBnMkwmE8NOEGLYISIiHyFDdYMDxQYjqupt0CjlCNMokBwVCm2oqsU9wqNioNXFe7lO8jcMO0REJCmj2YbVv5Yiefbr+OxoLYDac9qkxYbhktRopMWGe79A8nsMO0REJAmnU+Cj3YV4dv1RVJtsUOmSoZQD8ZEhiA1Xw+pwwthgh8FoxskKE05WmNAnSYurMnglh9qHYYeIiLwuq9iIRz8/iF8LqgEAaTEa/PrBi7jvoUcQG69v1rbaZMW+wmrsL6pBVokRp6obMFwnQdHktxh2iIiCnDeHb9dZ7Hhp4zGs2n4CDqdAuFqB+6/NwPhuoUj9+3qoFI+es090mBqjeurRQx+JDVkG1DTYsK1EDlV8F7froODCsENEFKS8OXxbCIGvD5Zg0doslBotAIBJ/ZPw6HW9kRQVCoPB0OoxOsWE4tZhnfH53lMoq7Ug4dZnUGUWSGpXJRSMGHaIiIKUt4Zv7zpegSUbj2FXfiUAoEtsGJ6c3M+tvjehKgVuvKQTPtqVh0po8cMpO1JSbOcdrUUEMOwQEQW9Cw3fdjgFjGYb6sx2VMGKkC6X4LChHs6QBugjNVAqWl51qLzOgvWHDPh87ynsPlkFAFAr5bhnVDfMuaobQlQKt+vVqBQYmejEx78UAPqu+HJ/Mf5naAo0SvePSYGNYYeIiM5xutaCw8U1yC6thdnmdL2ecPMizFp9FMBRKOQyJERqkBQdipgwNQCBBpsDeWX1MBjNrn3UCjluvjQVd4/qhuToUI/Up5IDZZ88iR73rkJFvRXrDhoweVAy5DKZR45PgYVhh4iIXOxOJ3bkVbhGSQGAUi5DZIgSKpkTRfm5SO2WgYp6O+xOgeIaM4przC0eq18nLf4wMBl/GNgJiVEhHq/VUVuOEZ2U2FLkQEGlCT/nV+Ky9FiPvw/5P4YdIiICABgbbFh7sASnaxs7EHeLD0e/TlHoHBMGuVwGY+VpPPHYvfi1pATx+gScrrWguKYBJdVm1DTYIJMBKoUcXePC0V0fgSgv9KOJCZFhdC89NmSVYld+JZKiQjjxIJ2DYYeIiGCy2vHZvlOoNtkQopJjTO8EdIuPOG97hVyGxKiQxis2nb1YaAt6JWlxqroBh4qN2HC4FNOGdUZECH+90W9a7llGRERBw+4U+HJ/MapNNkSGKHHrsM4XDDq+6KqMeMRHaNBgc+CbQyVwOoXUJZEPYdghIgpyPxscKDVaEKKUY8qgTtCG+N8wbqVCjon9E6FWyFFcY8b24xVSl0Q+hGGHiCiIhfW8AkV1AnIZcP3AZOjC1VKX5LboMDXG9G5camLPySrkl9dLXBH5Ct7UJCIKUhYHoLt2DgBgaJrOY8PCpdQjIRIDqxuwv6gGG7NKMX14Z4Rr3PtVZzQaYTKZ2tz+YpfSoI7DsENEFKT2V8igCI+BVg1c2jVG6nI8ZkSPOJyqbkB5nRUbsgy4YVCndh/DaDSia3o3VFaUt3kfd5fSoI7HsENEFIQMNWYU1MshnA5cmqCGUh44vRqUcjkm9EvCBz8XoLCyAb8WVKNHZPuOYTKZUFlRjgdWfIGI6Nbn7nF3KQ3yDoYdIqIgtONMB976w1sQ22tCu/ZtzyrpdrsdSmXrv2rac8y20IWrcVVGPDYdLcP2vHLEZLg3uiwiOva8S2mQ/2DYISIKMkVVJhRUmiCDQM1PHwBT2xZ23FklXSZXQDgdbW5vtVrb3LY1fZO1OFFRj7zT9dh2wgSZyvOzOJN/YNghIgoiQghsz2u8qtM1UuBETWmb923vKumlJ3Ox/OE72tS+qa3NZmtzPa2RyWQY0zsBpcYCGC126Mb8r8eOTf6FYYeIKIgUVjWgpMYMhVyG3tECW9w4xoVWST9bbXVFm9s3tfW0EJUC4/sm4tNfixAx4Fpsza3CLYmJHfJe5LsCp0caERG1am9BFQCgX7IWoUHy526nmFD002sAAM9tKkB5nUXiisjbGHaIiIJElcmKExWN88YMTI2WthgvuyQpBNayfFQ12LFgzUEIweUkggnDDhFRkNhXWA0A6BoXjpgw/50p2R0KuQzla5dAKZdhY1YpvtxfLHVJ5EWShp3ly5djwIAB0Gq10Gq1yMzMxDfffOPabjabMXfuXMTGxiIiIgJTp05FaWnzznQFBQWYNGkSwsLCoNfr8eCDD8Jut3v7oxAR+TSzzYGsYiMA4JIgu6rTxHY6H3cOa+yvs/CrLFSbPDfyi3ybpGEnJSUFzz77LPbs2YPdu3fjmmuuweTJk3H48GEAwP3334+vvvoKH3/8MbZt24bi4mLceOONrv0dDgcmTZoEq9WK7du34+2338aqVavw2GOPSfWRiIh8UlaxEXanQGyEGikx/r8shLtuvzQR3fURqKi3YvG6o1KXQ14iadi5/vrrMXHiRPTo0QMZGRl4+umnERERgZ07d6KmpgZvvvkmlixZgmuuuQZDhgzBypUrsX37duzcuRMA8O233yIrKwv//e9/MWjQIEyYMAGLFi3CsmXLPDpXAxGRPxNC4FBxDQBgUEo0ZDKZxBVJR6WQY/GN/QEAH+4uxC6ujh4UfKbPjsPhwOrVq1FfX4/MzEzs2bMHNpsNY8aMcbXp1asXOnfujB07dgAAduzYgf79+yMhIcHVZty4cTAaja6rQy2xWCwwGo3NHkREgaq4xowqkw0qhQwZCe1cNyEAXdpFh1uHpQIAFq7NgtPJzsqBTvKwc/DgQURERECj0WDOnDn47LPP0KdPHxgMBqjVakRHRzdrn5CQAIPBAAAwGAzNgk7T9qZt57N48WJERUW5HqmpqZ79UEREPuTwmas6GQmRUCsl/7HvEx4Y2xMRGiUOFxvxxf5TUpdDHUzyr/qePXti37592LVrF+6++27MnDkTWVlZHfqeCxYsQE1NjetRWFjYoe9HRCQVi92BnNI6AI3LJ1Cj2AgN7h7VDQDwr/XZMNvavqQF+R/Jw45arUb37t0xZMgQLF68GAMHDsTLL7+MxMREWK1WVFdXN2tfWlqKxDOzXyYmJp4zOqvpeeIFZsjUaDSuEWBNDyKiQJRtqIXdKaALVyNRy7WhzjZrRFckRYWguMaMlT+dkLoc6kCSh53fczqdsFgsGDJkCFQqFTZt2uTalp2djYKCAmRmZgIAMjMzcfDgwWar5W7cuBFarRZ9+vTxeu1ERL7m8Jnh5n2TtUHdMbklISoFHhjbEwDwxvd5MFk5bUmgkjTsLFiwAN9//z1OnDiBgwcPYsGCBdi6dSumT5+OqKgozJo1C/Pnz8eWLVuwZ88e3HnnncjMzMRll10GABg7diz69OmDGTNmYP/+/diwYQMeffRRzJ07FxqNRsqPRkQkucp6K8pqLZDJgF6J7JjcksmDkpEWG4Yqkw0f/sIuDYFK0rBTVlaG22+/HT179sTo0aPxyy+/YMOGDbj22msBAC+99BKuu+46TJ06FVdeeSUSExOxZs0a1/4KhQJr166FQqFAZmYmbrvtNtx+++1YuHChVB+JiMhnZBtqAQBpujCEqYNkIax2UirkuOvKdADA//2QD5vDKXFF1BEk/ep/8803L7g9JCQEy5Ytw7Jly87bJi0tDevWrfN0aUREfk0IgaOGxltYvRLZL/FCpg5OwUsbc3CqugFf7S/GjYNTpC6JPMzn+uwQEdHFK64xw2i2Q62QIz0+XOpyfFqISoE/jegCAFixLY+LhAYghh0iogDUdFWnmz4cKgV/1LfmtsvSEK5W4FhpHX7Or5S6HPIwfgcQEQUYh1O45tbhLay20Yao8IdByQDAjsoBiGGHiCjAFBltsNidCNcognrRz/a6+dLOAICvD5ag1sxh6IGEYYeIKMDkVdoAAD0TIiHn3DptNjAlCr0SI2GxO7Ehm7eyAgnDDhFRAJFrwlFobAw7vIXVPjKZDDcNbVwr8ctD5RJXQ57EsENEFEDCeo2AUwCxEWrER3Jy1fa64ZJOUCvkOHa6ASp9utTlkIcw7BARBZDwPqMAcMZkd8WEqzGmjx4AEN57pMTVkKcw7BARBYgSowUhnfsDaOyvQ+6Z2D8JABCWcQXn3AkQDDtERAFiw9HGTrWJEUpEhqgkrsZ/Xd1TD41CBpUuGVUNXD4iEDDsEBEFACEEvjnSGHa66Rh0Lka4RonLukQBAE5UWyWuhjyBYYeIKAAcLjbiZJUZTpsFXaLVUpfj967pEQMAOFFjk7gS8gQug0tEFAA+23sKANCQuwvqYX+QuBrfVVZW1qZ2PSOtEHYbaswqVNRZEBvBkW3+jGGHiMjP2R1OfLm/GABQf3grAIad3zOb6gGZDAMHDmzzPvFTH0NY92HILatj2PFzDDtERH5ue14FTtdaEBWiwMn8X6UuxyfZrGZACNyzZDX0ySmtti89mYt333sPYd2H4USFCcPTY71QJXUUhh0iIj+35tciAMCYDB0OOLmm04WER8VAq4tvtV1tdQUazgTHUqMZZpsDISpFR5dHHYQdlImI/FhNgw3fHDIAACb24dUHT3LUVkCrBgSAgkqT1OXQRWDYISLyY2sPFMNidyIjIQJ9EsKkLifgJIY3/po8WcGw488YdoiI/NjHuxtvYf3PkFTIuMK5xyWGNZ7Tk5X1nE3ZjzHsEBH5qZzSWuwrrIZSLsOUSzpJXU5Aig+VQSmXod7iQEU9Jxj0Vww7RER+6uM9jVd1ru6l5wrnHUQhl6FTTCgA3sryZww7RER+yGJ3uEZh/c+Q1odSk/vSdI19oU5W1ktcCbmLYYeIyA+tP2RAeZ0VCVoNrumll7qcgNb5TNgprjbD4WS/HX/EsENE5Ife3XESADBtWBqUCv4o70i6cDVCVQo4nAKlRrPU5ZAb+B1CRORnsoqN2H2yCkq5DLcOS5W6nIAnk8nQKbqx386p6gaJqyF3MOwQEfmZ/+5qvKozrl8i9NoQiasJDk2dlE9VMez4I4YdIiI/UtNgw+dnVjifcVmaxNUEj6YrO8U1DXCy347fYdghIvIj/915EiarAz0TIjG8q07qcoJGXIQaGqUcNodAWZ1F6nKonRh2iIj8hNnmwMqf8gEAc0alc8ZkL2rWb4e3svwOww4RkZ/4eE8Ryuus6BQdiusGJEtdTtBhJ2X/xbBDROQH7A4n3vg+DwBw15XpUHG4udc1dVIurm6Ak+tk+RV+txAR+YEv9hWjsLIBseFq3DSUw82lEB+hgUohg8XuRCXXyfIrDDtERD7ObHNgycZjAIBZI7siVK2QuKLgJJfLkHBmqL+hhpML+hOGHSIiH/fOjhM4Vd2ApKgQ/OmKrlKXE9SSohrDTgnDjl9h2CEi8mHVJiv+vTkXAHD/tRkIUfGqjpQSo3hlxx8x7BAR+bCXN+XAaLajZ0Ikpg7m6uZSSzxzG6vSZIXF5pC4Gmorhh0iIh+1t6AKb28/AQD4x6TeUMg5r47UwtRKRIWqAAAGLgrqNxh2iIh8kMXuwEOfHIBTAFMGJeOqjHipS6IzeCvL/zDsEBH5oGVb8pBTVofYcDUeu76v1OXQWZLO3Moq4ZUdvyFp2Fm8eDEuvfRSREZGQq/XY8qUKcjOzm7WZtSoUZDJZM0ec+bMadamoKAAkyZNQlhYGPR6PR588EHY7XZvfhQiIo/ZnluOf2/OAQA8ObkvdOFqiSuis519ZUdwckG/oJTyzbdt24a5c+fi0ksvhd1uxz/+8Q+MHTsWWVlZCA8Pd7WbPXs2Fi5c6HoeFhbm+rfD4cCkSZOQmJiI7du3o6SkBLfffjtUKhWeeeYZr34eIqKLdaq6AfM+2AunAKYOTsGk/klSl0S/ExehgULeOLlgtcmGGIZRnydp2Fm/fn2z56tWrYJer8eePXtw5ZVXul4PCwtDYmJii8f49ttvkZWVhe+++w4JCQkYNGgQFi1ahIcffhhPPPEE1Gp+ERKRfzBZ7Zjz7h5U1lvRr5MWT9/Qj4t9+iCFXIaESA2Ka8woMZoZdvyAT/XZqampAQDodLpmr7/33nuIi4tDv379sGDBAphMJte2HTt2oH///khISHC9Nm7cOBiNRhw+fNg7hRMRXSSzzYE/v70bB0/VICZMhRW3DeGcOj4s4cytrDL22/ELkl7ZOZvT6cRf//pXXHHFFejXr5/r9WnTpiEtLQ3Jyck4cOAAHn74YWRnZ2PNmjUAAIPB0CzoAHA9NxgMLb6XxWKBxWJxPTcajZ7+OEREbWa2OXDXu3uwPa8C4WoF3rzjUqTEhLW+I0kmIbIx7JQaLa20JF/gM2Fn7ty5OHToEH788cdmr991112uf/fv3x9JSUkYPXo08vLy0K1bN7fea/HixXjyyScvql4iIk+oqLPgf9/dg90nqxCqUmDlncMwuHOM1GVRK/RaDQDgdJ0FDic7Kfs6n7iNNW/ePKxduxZbtmxBSsqFZwgdPnw4ACA3t3H69MTERJSWljZr0/T8fP18FixYgJqaGtejsLDwYj8CEVG7HSutxQ2vbcfuk1WIDFHirTsuxbCuutZ3JMlFh6qgVsrhcAqugO4HJA07QgjMmzcPn332GTZv3oyuXVtf4G7fvn0AgKSkxhEKmZmZOHjwIMrKylxtNm7cCK1Wiz59+rR4DI1GA61W2+xBROQtQgi8u/Mk/vDvH1FQaUKqLhSf3XM5MrvFSl0atZFM1thJGQBK2W/H50l6G2vu3Ll4//338cUXXyAyMtLVxyYqKgqhoaHIy8vD+++/j4kTJyI2NhYHDhzA/fffjyuvvBIDBgwAAIwdOxZ9+vTBjBkz8Pzzz8NgMODRRx/F3LlzodFopPx4RETnKKoy4Z+fH8KW7NMAgJE94rD05kGIjeDPK3+j14agsKoBpbVmdA71iRsldB6Shp3ly5cDaJw48GwrV67EHXfcAbVaje+++w5Lly5FfX09UlNTMXXqVDz66KOutgqFAmvXrsXdd9+NzMxMhIeHY+bMmc3m5SEikprF7sBbP57Ay5uOwWxzQqWQYe4VnXDTJXrY6qpgqGt5v7CwMF599lFNV3bKjBZAHypxNXQhkoad1maeTE1NxbZt21o9TlpaGtatW+epsoiIPEYIga8PluC59UdRWNkAADAXHETxt6/h/mcKcX8r++ti45B/PI+BxwclnFk2orzOAoczROJq6EJ8ZjQWEVGg2X2iEk+vO4K9BdUAgLhwFY5+9BzmzL0PkZM/aHX/uuoKvDBnMkwmE8OOD4oMUSJEJYfZ5kRlg0PqcugCGHaIiC6C0WhsNtEpABRWm/Haj6ewJbcaABCqkuO2IQm4trMcwx/bjMiYhdDquIq5v5PJZEjQhuBkhQkVJoYdX8awQ0T0Oy0FmJbU1tZi+GWZqKqsAADI1KGIHjEdkYMnQaZQQTgdqDuwEUU/vodH66vQ1NvQauVQ5UCRENkYdsoZdnwaww4R0VmMRiO6pndDZUV5m/e59+VPUKPW4eeiBjTYG/sidtIqcWlyJGKG3ALceQsAoPRkLpY/fAdsNluH1E7eF3+mk3IFb2P5NIYdIqKzmEwmVFaU44EVXyAi+sLz3pSezMV/nv0HdlWHovTMlaCoUBWu7hmPtNjwc9rXVld0SM0kHf2ZsFNtdgBy/kr1Vfw/Q0TUgojo2Av2q3EKgV9OVCL5T8tQahJQyGW4NC0GQ9JioFRwzpVgERmihEYph8XuhCqus9Tl0Hkw7BARtVOt2YYNh0txqloOmVKOxDAZxg3sjOgwtdSlkZfJZDLER2pQVNUAdYJ76zVSx+OfH0RE7VBUZcIHPxfiVHUDFDKB8nVLMbKTgkEniDX121EnpEtcCZ0Pww4RURvtL6zGmr2n0GBzID5SgzGdnKg/+B1kMpnUpZGE9BEMO76OYYeIqBVCCPyUW46tx05DCKBnYiRuGpKCSJXUlZEvcF3Zie8KZysrA5A02GeHiOgCnEJgy9EyHCo2AgAu7xaLoWkxvJpDLjFhaihkADRhKKq2IDlJ6oro93hlh4joPIQQ2Hwm6MgAjO6lx6VddAw61IxcLkNMqAIAkF3W+mSU5H0MO0RELRBC4Iecchw+E3TG90tEv05RUpdFPir2TNg5dpphxxcx7BARteBgqQV7C6sBAGN6JyAjIVLagsin6cLOhJ2yBokroZYw7BAR/U5oj0zsKTEDAK7sEYc+yVxxnC7s7Cs7gp2UfQ7DDhHRWXJOmxB33XwAwMCUKFzSOUbiisgfxIQqIJwOVDXYUWq0SF0O/Q7DDhHRGUazDQ9/lQe5OhRJEUpc2eP8y0UQnU0pl8FWUQQAOFxcI3E19HsMO0REaOyQ/Mhnh1BstMJebcDVXcMgl3PUFbWdtTQPAHD4zDQF5DsYdoiIAHyypwhf7S+GQgac/uoFaJT88UjtYy07DoBXdnwRv5uJKOgVVJjw+JeHAQCzM5NhLT4qcUXkj3hlx3cx7BBRUBNC4B+fHYTJ6sCwrjrMGJoodUnkp2yljVd2iqoaUGOySVwNnY1hh4iC2id7ivBjbjk0SjmenzoACvbTITc5LfVI0qoBAIdLeCvLlzDsEFHQOl1rwVNfHwEA3H9tBrrEhUtcEfm7jPgwAEAWb2X5FIYdIgpaz35zFDUNNvRN1uLPI7pKXQ4FgJ76xrDDfju+haueE1FQ2l9YjU9/bZwX5akp/aBU+O7ffmVlZR5tRx0nIz4UAEdk+RqGHSIKOkIILFybBQC48ZJOPjtLstlUD8hkGDhwYLv2s1qtHVQRtSbjzJWdvNP1MNscCFEpJK6IAIYdIgpCX+4vxp6TVQhVKfDQ+F5Sl3NeNqsZEAL3LFkNfXJKq+1LT+Zi+cN3wGbjSCCpxIerEBuuRkW9FUcNtRiUGi11SQSGHSIKMla7E//akA0AuGdUNyRGhUhcUevCo2Kg1bW+dEVtdYUXqqELkclk6JOsxQ855cgqNjLs+AjfvUlNRNQBPvylAEVVDYiP1ODPI9OlLocCUJ9kLQD22/ElDDtEFDQarA68ujkXAHDvNd0RqmZ/CvK8vslRAICsEo7I8hVuhZ3jx497ug4iog73zo4TKKu1oFN0KG65tLPU5VCA6pPUeGXnaEktHE4hcTUEuNlnp3v37rjqqqswa9Ys/PGPf0RIiO/f8yai4GU0GlFeXYvXtuQAAO68VI/K8paHaXP4Nl2srnHhCFUp0GBzIL+8Ht31EVKXFPTcCju//vorVq5cifnz52PevHm4+eabMWvWLAwbNszT9RERncNoNMJkMrWpbW1tLYZflgl7+gjoRs+GraoYs6/9AyCcF9yPw7fJXQq5DL2SIrG3oBqHi2sYdnyAW2Fn0KBBePnll/Hiiy/iyy+/xKpVqzBixAhkZGTgT3/6E2bMmIH4+NZHDhARtZfRaETX9G6orChv+04KJXpe+2eYncBVA7qh5wffn7cph2+TJ/RN1mJvQTWySoyYPKiT1OUEvYsaeq5UKnHjjTdi0qRJeO2117BgwQI88MAD+Mc//oGbbroJzz33HJKSkjxVKxERTCYTKivK8cCKLxARHdtq+9KTuXj3vfdhdsoQrlHgku7JUMrP312Rw7fJE/oknemkzGUjfMJFhZ3du3fjrbfewurVqxEeHo4HHngAs2bNQlFREZ588klMnjwZP//8s6dqJSJyiYiObdPcM8aqCmiH/xEAMLhzzAWDDpGnNA0/zyo2QggBmUwmcUXBza2ws2TJEqxcuRLZ2dmYOHEi3nnnHUycOBHyMz9EunbtilWrVqFLly6erJWIqN1OmQCVLhlqOdDvzJBgoo7WKzESchlQUW9FWa0FCVoO5JGSW2Fn+fLl+NOf/oQ77rjjvLep9Ho93nzzzYsqjojoYuXUNP4R1i1aDrWSV3XIO0JUCnSLj0BOWR0OF9cw7EjMrbCTk5PTahu1Wo2ZM2e6c3giIo8wGM2osMggHDZ0jw6VuhwKMn2Ttcgpq0NWsRHX9EqQupyg5tafOStXrsTHH398zusff/wx3n777YsuiojIE/YVVgMA6o/8gFAl+0yQd/22bAQ7KUvNrbCzePFixMXFnfO6Xq/HM888067jXHrppYiMjIRer8eUKVOQnZ3drI3ZbMbcuXMRGxuLiIgITJ06FaWlpc3aFBQUYNKkSQgLC4Ner8eDDz4Iu93uzkcjogBRZ7Yjp7QWAFC7+wuJq6FgxGUjfIdbYaegoABdu3Y95/W0tDQUFBS0+Tjbtm3D3LlzsXPnTmzcuBE2mw1jx45FfX29q83999+Pr776Ch9//DG2bduG4uJi3Hjjja7tDocDkyZNgtVqxfbt2/H2229j1apVeOyxx9z5aEQUIA6eqoFTAHEaAWtpntTlUBBqWjbiZIUJRjPnbZKSW3129Ho9Dhw4cM5oq/379yM2tvV5L5qsX7++2fNVq1ZBr9djz549uPLKK1FTU4M333wT77//Pq655hoAjbfQevfujZ07d+Kyyy7Dt99+i6ysLHz33XdISEjAoEGDsGjRIjz88MN44oknoFar3fmIROTHHE6BQ2dWnO6mFdgjcT0UnGLC1UiOCkFxjRlHS2oxrKtO6pKClltXdm699Vb85S9/wZYtW+BwOOBwOLB582bcd999uOWWW9wupqam8YeTTtf4BbFnzx7YbDaMGTPG1aZXr17o3LkzduzYAQDYsWMH+vfvj4SE3zp/jRs3DkajEYcPH3a7FiLyX8dP18FkdSBMrUCncC7ESNL5rd9OjcSVBDe3ruwsWrQIJ06cwOjRo6FUNh7C6XTi9ttvb1efnbM5nU789a9/xRVXXIF+/foBAAwGA9RqNaKjo5u1TUhIgMFgcLU5O+g0bW/a1hKLxQKLxeJ6bjTyfipRIDlwqvEXS99kLeSydiwrQeRhfZKj8N2RMs6kLDG3wo5arcaHH36IRYsWYf/+/QgNDUX//v2RlpbmdiFz587FoUOH8OOPP7p9jLZavHgxnnzyyQ5/HyLyvqp6K4qqGiAD0K9TFGqLGXZIOk39djgiS1oXtVxERkYGMjIyLrqIefPmYe3atfj++++RkpLiej0xMRFWqxXV1dXNru6UlpYiMTHR1eb3S1I0jdZqavN7CxYswPz5813PjUYjUlNTL/pzEJH0Dp65XdAlLhzaEBVqJa6HglvfM7excspqYbU7ObGlRNwKOw6HA6tWrcKmTZtQVlYGp9PZbPvmzZvbdBwhBO6991589tln2Lp16zkjvIYMGQKVSoVNmzZh6tSpAIDs7GwUFBQgMzMTAJCZmYmnn34aZWVl0Ov1AICNGzdCq9WiT58+Lb6vRqOBRqNp12cmIt/ncAocLWmMN/07cWkIkl5KTCi0IUoYzXbklNW6hqOTd7kVdu677z6sWrUKkyZNQr9+/dxe4Gzu3Ll4//338cUXXyAyMtLVxyYqKgqhoaGIiorCrFmzMH/+fOh0Omi1Wtx7773IzMzEZZddBgAYO3Ys+vTpgxkzZuD555+HwWDAo48+irlz5zLQEAWZExX1aLA5EK5RIC02TOpyiCCTydAnWYudxyuRVWxk2JGIW2Fn9erV+OijjzBx4sSLevPly5cDAEaNGtXs9ZUrV+KOO+4AALz00kuQy+WYOnUqLBYLxo0bh9dee83VVqFQYO3atbj77ruRmZmJ8PBwzJw5EwsXLryo2ojI/zR1Au2VqIWcq0yTj+iTFIWdxytxuNiI/5G6mCDldgfl7t27X/SbC9H6kNCQkBAsW7YMy5YtO2+btLQ0rFu37qLrISL/ZbLacaKicULS3omREldD9JumfjucSVk6bvWU+tvf/oaXX365TWGFiMgbsg21cAogQatBbARvYZPvaJpr50ixEU4nf29Kwa0rOz/++CO2bNmCb775Bn379oVKpWq2fc2aNR4pjoiorZr+am4a6kvkK7rrI6BWyFFrsaOoqgGd2Z/M69wKO9HR0bjhhhs8XQsRkVtO11pQXmeFQiZDRgJvYZFvUSnkyEiMwKFTRhwurmHYkYBbYWflypWeroOIyG1NV3XS48MRolJIXA3RufomReHQKSOySoyY0D9J6nKCjtuzG9ntdnz33Xd4/fXXUVvbOK9FcXEx6urqPFYcEVFrHE6BbEPjzyDewiJf9dsaWeykLAW3ruycPHkS48ePR0FBASwWC6699lpERkbiueeeg8ViwYoVKzxdJxFRi1xz66gV6Kzj7QHyTa4RWQw7knDrys59992HoUOHoqqqCqGhoa7Xb7jhBmzatMljxRERtcY1t06SFnI559Yh39QrSQuZDDAYzaios7S+A3mUW1d2fvjhB2zfvh1qtbrZ6126dMGpU6c8UhgRUWs4tw75iwiNEl1iw5FfXo+sEiNG9oiXuqSg4taVHafTCYfDcc7rRUVFiIzkDxwi8o6c0jo4BaCP5Nw65Pu4Arp03Ao7Y8eOxdKlS13PZTIZ6urq8Pjjj1/0EhJERG11rKyxY3JPXtUhP9CH/XYk49ZtrBdffBHjxo1Dnz59YDabMW3aNOTk5CAuLg4ffPCBp2skIjpHvdWJ4mozAKCHPkLiaoha99uIrBqJKwk+boWdlJQU7N+/H6tXr8aBAwdQV1eHWbNmYfr06c06LBMRdZQT1TYAQFJUCCJDVK20JpJe04is4+X1MFntCFO79SuY3OD2mVYqlbjttts8WQsRUZudqLYCAGdMJr+hjwyBPlKDsloLsoqNGNpFJ3VJQcOtsPPOO+9ccPvtt9/uVjFERG2hiIxDWX3jIInuvIVFfqR/pyhsOlqGg6dqGHa8yK2wc9999zV7brPZYDKZoFarERYWxrBDRB0qrNcIAECn6FBEaHgrgPxHv7PCDnmPW6Oxqqqqmj3q6uqQnZ2NESNGsIMyEXW48DNhhx2Tyd/07xQFADjEsONVbq+N9Xs9evTAs88+e85VHyIiTyqusUCT3Asy8BYW+Z9+Z8JOblkdGqznzldHHcNjYQdo7LRcXFzsyUMSETWzOacKAJAQoUQ4b2GRn0nQahAXoYFTAFklnG/HW9z6SfHll182ey6EQElJCf7973/jiiuu8EhhREQt+e5YY9jpGsPh5uR/ZDIZ+nfSYkv2aRw6VYMhaTFSlxQU3Ao7U6ZMafZcJpMhPj4e11xzDV588UVP1EVEdI6TFfU4WmaCcDrQJYphh3xPWVlZq226RiuxBcCvJ8ox8/IuHV4TuRl2nE6np+sgImrV2gMlAABzwQGEDLla4mqIfmM21QMyGQYOHNhq29Duw6Gf+k+s2bobi67LgFar9UKFwY03vInIb3x9JuyYjvwA3MCwQ77DZjUDQuCeJauhT065YNt6qxMfHTZCHp2MSmMdw44XuBV25s+f3+a2S5YscectiIiaOX66DlklRijkgOnYDqnLIWpReFQMtLr4C7aJFAIh2bUw2xXIPd2ALhfORuQBboWdvXv3Yu/evbDZbOjZsycA4NixY1AoFBg8eLCrnUwm80yVRBT0mq7qXJqqxXFzrcTVELlPJpMhNlSBU7V2HC2rxxipCwoCboWd66+/HpGRkXj77bcRE9PYk7yqqgp33nknRo4cib/97W8eLZKI6OuDjWFnTEYMPpS4FqKLFRvWGHayy0xSlxIU3Jpn58UXX8TixYtdQQcAYmJi8NRTT3E0FhF5XG5ZLY4aaqFSyHBlt2ipyyG6aHFhCgDA0VKGHW9wK+wYjUacPn36nNdPnz6N2lpeXiYiz2oahTWyRzy0IRxXQf4vNqzx6/h4ZQPMNs6k3NHcCjs33HAD7rzzTqxZswZFRUUoKirCp59+ilmzZuHGG2/0dI1EFMSEEK6wM6l/ksTVEHlGuEoGh6kGDieQbeBFgo7mVthZsWIFJkyYgGnTpiEtLQ1paWmYNm0axo8fj9dee83TNRJREDtWWofcsjqoFXJc2zdB6nKIPEImk8FqyAUAroDuBW5dDw4LC8Nrr72Gf/3rX8jLywMAdOvWDeHh4R4tjoho7YHG9fauzIiHNkQF9nCgQGEtzUNo+hCugO4FF3Xzu6SkBCUlJbjyyisRGhoKIQSHmxORxwghXEPOrx/IW1gUWJqu7Px6ohwGg6HV9mFhYZyA0E1uhZ2KigrcdNNN2LJlC2QyGXJycpCeno5Zs2YhJiaGI7KIyCOySow4Xl4PjVKO0b15C4sCh9lUD0tp452RbIMRSSmpgMN+wX10sXHIP57HwOMGt8LO/fffD5VKhYKCAvTu3dv1+s0334z58+cz7BCRRzRd1bm6px4RGo7CosBhs5rhqCmFSiZgU6gw7z/fIS7s/F/jddUVeGHOZJhMJoYdN7j10+Pbb7/Fhg0bkJLSfI7rHj164OTJkx4pjIiCW7NRWAN4C4sCky5UjlKTQL0sDOm6KKnLCVhujcaqr69HWFjYOa9XVlZCo9FcdFFERIdOGVFQaUKISo5reumlLoeoQ8RoGvu5lhnNElcS2NwKOyNHjsQ777zjei6TyeB0OvH888/j6qu5EjERXby1BxtHYY3ulYBw3sKiABUTcibs1FokriSwufUT5Pnnn8fo0aOxe/duWK1WPPTQQzh8+DAqKyvx008/ebpGIgoyZ4/C4i0sCmRNV3Yq6qxwOAUUco5o7ghuXdnp168fjh07hhEjRmDy5Mmor6/HjTfeiL1796Jbt26erpGIgsz+ohoUVTUgTK3A1T15C4sCV7gK0CjlcAiBinpe3eko7b6yY7PZMH78eKxYsQKPPPJIR9REREFu7f4zt7B6JyBUrZC4GqKOI5PJoI/UoLCqAWVGC/SRIVKXFJDafWVHpVLhwIEDHVELERGcToF1BxtvYV3HW1gUBPTaxoDDfjsdx63bWLfddhvefPPNi37z77//Htdffz2Sk5Mhk8nw+eefN9t+xx13QCaTNXuMHz++WZvKykpMnz4dWq0W0dHRmDVrFurq6i66NiKSxt7CKhTXmBGuVuCqjHipyyHqcPrIxlHMZbUckdVR3OqgbLfb8dZbb+G7777DkCFDzlkTa8mSJW06Tn19PQYOHIg//elP510tffz48Vi5cqXr+e+Htk+fPh0lJSXYuHEjbDYb7rzzTtx11114//332/mpiEgqRqMRJlPjqlcf7SgEAIxIj0J1xelz2paVlXm1NqKO1hR2ytlJucO0K+wcP34cXbp0waFDhzB48GAAwLFjx5q1ac/aWBMmTMCECRMu2Eaj0SAxMbHFbUeOHMH69evxyy+/YOjQoQCAV199FRMnTsQLL7yA5OTkNtdCRNIwGo3omt4NlRXlAGTodM8qKCNj8e5Tf8Ubd/583v2sVqv3iiTqQFGhKqiVcljtTlTWWxEfyfnqPK1dYadHjx4oKSnBli1bADQuD/HKK68gIaHj1qzZunUr9Ho9YmJicM011+Cpp55CbGwsAGDHjh2Ijo52BR0AGDNmDORyOXbt2oUbbrihxWNaLBZYLL/dGzUajR1WPxFdmMlkQmVFOR5Y8QXqlFH4JqcOagXwwKIXW/wLt/RkLpY/fAdsNpsE1RJ5XlMn5aKqBpTVmhl2OkC7+uwIIZo9/+abb1BfX+/Rgs42fvx4vPPOO9i0aROee+45bNu2DRMmTIDD4QAAGAwG6PXNh6UqlUrodLoLriC7ePFiREVFuR6pqakd9hmIqG0iomNR3NA48qqbPhIxcXpodfHnPMKiYiSulMjzXP12jOyk3BEualrS34cfT7vllltc/+7fvz8GDBiAbt26YevWrRg9erTbx12wYAHmz5/vem40Ghl4iCTmFAI5ZY2DC3roIyWuhsi7moacl7KTcodo15WdphFRv3/NW9LT0xEXF4fc3FwAQGJi4jmdFe12OyorK8/bzwdo7Aek1WqbPYhIWqV1dpisDmiUcnTWnbv2HlEgS9A276RMntWuKztCCNxxxx2uEVFmsxlz5sw5ZzTWmjVrPFfhWYqKilBRUYGkpMa5NzIzM1FdXY09e/ZgyJAhAIDNmzfD6XRi+PDhHVIDEXWM/OrGPjjd4iM4GoWCTlSoChqlHBa7ExX1nFzQ09oVdmbOnNns+W233XZRb15XV+e6SgMA+fn52LdvH3Q6HXQ6HZ588klMnToViYmJyMvLw0MPPYTu3btj3LhxAIDevXtj/PjxmD17NlasWAGbzYZ58+bhlltu4UgsIn8ik+PkmbCTkRAhcTFE3ieTyaDXalBY2YBSzqTsce0KO2fPd+MJu3fvbrZKelM/mpkzZ2L58uU4cOAA3n77bVRXVyM5ORljx47FokWLms21895772HevHkYPXo05HI5pk6dildeecWjdRJRxwpJGwizXSBUpUBKDG9hUXBKiAxBYWUDyoxmoFOU1OUElIvqoHyxRo0adcFOzhs2bGj1GDqdjhMIEvm58D5XAgC663kLi4JXwpllI0o5Isvj3FougojIU6x2J8IyLgcA9EzgKCwKXk2dlCvqLbA7nBJXE1gYdohIUjtPGiHXhCNMJUNyNPspUPCK0CgRplbAKYDTdby640kMO0QkqW+zKwEAXaPVXp3KgsjXNM2kDPBWlqcx7BCRZExWO348XgMA6BqjkrgaIuk19dspM3JyQU9i2CEiyWzMKoXZ7oStqhhxYQqpyyGSHDspdwyGHSKSzFf7SwAApiPf8xYWEX5bI6vSZIXVzk7KnsKwQ0SSqDHZsO1Y43Iv9Ud+kLgaIt8QrlEiQtM4K0wZ18nyGIYdIpLEhsMG2BwC3WJDYCs/KXU5RD6jaQg6V0D3HIYdIpLEVweKAQBjeuokroTIt/zWb4dXdjyFYYeIvO50rQU/5ZYDAK7NYNghOpsr7NTyyo6nMOwQkdetO1gCpwAGpkQhJVrT+g5EQaSpk3JNgw1mm0PiagIDww4Red2aX4sAAH8Y1EniSoh8T4hKgajQxnmneCvLMxh2iMircstqsb+oBgq5DJMHJUtdDpFPauqkzFtZnsGwQ0RetebXUwCAURnxiIvgLSyilnAmZc9i2CEir3E6BT7b2xh2bhycInE1RL4rIZIzKXsSww4Rec3O4xUoqTEjMkSJ0b31UpdD5LPiIzWQAaiz2FFvsUtdjt9j2CEir/n0zC2s6wYkI0TFtbCIzketlEMXrgbATsqewLBDRF5hstrxzaHGtbCmDuYoLKLW6NlJ2WMYdojIKzYcNsBkdaCzLgxD0mKkLofI5/3Wb4dXdi4Www4ReUXTKKwbB3fiCudEbfDbiCwLhBASV+PfGHaIqMMZasz48czyEDdewlFYRG0RF6GGXAY02ByotzHsXAyGHSLqcJ/vOwUhgGFddOgcGyZ1OUR+QamQu+aiKq/niKyLwbBDRB1KCOFaHuJGdkwmapemdbLKTVwj62Iw7BBRh9pbWI1jpXXQKOWY0D9J6nKI/EpTvx2GnYvDsENEHeqDXQUAGufWaVrckIjapinsVDTYAbBjv7sYdoiowxjNNnx1oBgAcOuwVImrIfI/unA1FHIZrA5AGcMro+5i2CGiDvPFvmKYbU700Edwbh0iNyjkMsSf6aSsTuwhcTX+i2GHiDqEEALvn7mFdeuwzpxbh8hNCWdmUtYkMey4i2GHiDrEgaIaHCkxQq2UcxQW0UVo6rejZthxG8MOEXWID35uvKozqX8SosPUEldD5L9cYUffDXYnJxd0B8MOEXlcncWOL/c3dUzuLHE1RP4tJkwFpRyQq0NwopLrZLmDYYeIPO7LfcUwWR3oFh+OS7uwYzLRxZDJZIgLUwAAjpbWS1yNf2LYISKPa7qFxY7JRJ4RF6YEAGSVmiSuxD8ppS6AiALLwaIaHDxVA5VChhEpahgMhgu2Lysr81JlRP4rlld2LgrDDhF51Fs/5AAAqg9uRe/0SW3ez2q1dlBFRP4v/kzYySlvgNXuhFrJGzPtwbBDRB5TWW/FukONV2quGz0CXaZPbnWf0pO5WP7wHbDZbB1dHpHfilDL4WgwAqFaHDUYMSAlWuqS/ArDDhF5zAc/F8DiELCU5CBt0FBodfGt7lNbXeGFyoj8m0wmg9WQi9Cug3GgqIZhp514HYyIPMLmcOKdHScAALW7v2DHZCIPs5YcAwAcKKqWthA/xCs7RNQqo9EIk+nCo0C+PVqJUqMFMSFynDz6o5cqIwoeFkMugMbZyal9GHaI6IKMRiO6pndDZUX5BdslzngRmuSeyN/4DuC0s8MxkYc1Xdk5VlqLeosd4Rr+Cm8rSW9jff/997j++uuRnJwMmUyGzz//vNl2IQQee+wxJCUlITQ0FGPGjEFOTk6zNpWVlZg+fTq0Wi2io6Mxa9Ys1NXVefFTEAU2k8mEyopyPLDiCzyx+scWH/e8uRWa5J6Qy4Cp108AAHY4JvIwR10l9BEqOAWv7rSXpGGnvr4eAwcOxLJly1rc/vzzz+OVV17BihUrsGvXLoSHh2PcuHEwm3+bLnv69Ok4fPgwNm7ciLVr1+L777/HXXfd5a2PQBQ0IqJjodXFt/jIqW5s0zMxEjEx0VKWSRTQ+iWFAwD2FVZLW4ifkfQa2IQJEzBhwoQWtwkhsHTpUjz66KOYPLlx+Oo777yDhIQEfP7557jllltw5MgRrF+/Hr/88guGDh0KAHj11VcxceJEvPDCC0hOTvbaZyEKVrVmG3JON15NvSQ1BtbT1dIWRBTA+iaGY3NONfYVVkldil/x2dFY+fn5MBgMGDNmjOu1qKgoDB8+HDt27AAA7NixA9HR0a6gAwBjxoyBXC7Hrl27zntsi8UCo9HY7EFE7jlQVAMhgE7RoYiP1EhdDlFA65vYeGVnb0E1hOAK6G3ls2GnaYr5hISEZq8nJCS4thkMBuj1+mbblUoldDrdBaeoX7x4MaKiolyP1NRUD1dPFBxsDicOnWrsOzAoNVraYoiCQC99OBRyGcpqLSip4QrobeWzYacjLViwADU1Na5HYWGh1CUR+aWDp2pgtjsRFapCeny41OUQBbwQlRy9EiMBsN9Oe/hs2ElMTAQAlJaWNnu9tLTUtS0xMfGcRQTtdjsqKytdbVqi0Wig1WqbPYiofexOJ34taOw3MDQtBnJOIkjkFU1XURl22s5nw07Xrl2RmJiITZs2uV4zGo3YtWsXMjMzAQCZmZmorq7Gnj17XG02b94Mp9OJ4cOHe71momBypLgW9RYHIjRK9EqKlLocoqDRFHb2FrCTcltJOhqrrq4Oubm5ruf5+fnYt28fdDodOnfujL/+9a946qmn0KNHD3Tt2hX//Oc/kZycjClTpgAAevfujfHjx2P27NlYsWIFbDYb5s2bh1tuuYUjsYg6kNMpsPtkJQBgcOdoKOU++3cTUcC5pHM0gMbbyDaHEyoFv/9aI2nY2b17N66++mrX8/nz5wMAZs6ciVWrVuGhhx5CfX097rrrLlRXV2PEiBFYv349QkJCXPu89957mDdvHkaPHg25XI6pU6filVde8fpnIQom2aW1MJrtCFUp0K9TlNTlEAWV9LgIRIYoUWu2I9tQy+/BNpA07IwaNeqCQ+dkMhkWLlyIhQsXnreNTqfD+++/3xHlEVELhBDYfaLx8vklnaP5VyWRl8nlMgxKjcYPOeXYV1jNsNMG/ClFRO2Se7oOlSYr1Eo5BqTwhyyRFH7rt1MtaR3+gmGHiNpMCIFfzlzVGZQSDY1SIXFFRMHptxFZ7KTcFgw7RNRmJytMOF1rgfLMZXQikkbT91/e6XrUNHDR3dYw7BBRmwghsDO/AgDQPyUKoWpe1SGSSmyEBp11YQCAA0XV0hbjBxh2iKhNCmrsKDU2XtUZ0jlG6nKIgh777bQdww4RtU4mx68lDQAaf8CGayQdyElE4EzK7cGwQ0StCus1EtVmJ9RKOYak8aoOkS8YdGZywX2FXAG9NQw7RHRBNocT0SOnAwCGdI5BiIp9dYh8QZ8kLVQKGSrrrSisbJC6HJ/GsENEF/Tp/tNQxSQjVMkRWES+JESlQJ/kxrmu9nII+gUx7BDRedWYbHhrVwkA4JKkEKiV/JFB5EsGn7mVteckw86F8CcXEZ3Xv7fkwGhxwHr6BHrEqqUuh4h+Z2iaDgBcS7hQyxh2iKhFJ8rr8fb2kwCAqi1vQS6TSVwREf1e04CBowYj6ix2iavxXQw7RNSihWuzYHU4cVmaFub8X6Uuh4hakBgVgk7RoXAKYB/n2zkvTpZBROfYdKQUm4+WQaWQ4f6rUvGh1AUREQCgrKzsnNf6JoTiVHUDth4uQPfI367uhIWFQavVerM8n8WwQ0TNmG0OLFybBQD404iuSNOFSFwREZlN9YBMhoEDB56zLeKSiYgdew9e/eBr/POGx1yv62LjkH88j4EHDDtE9DvLt+bhZIUJ+kgN7r2mB+qqyqUuiSjo2axmQAjcs2Q19MkpzbZVmOz4MrsO2m6DMeeDHyCXyVBXXYEX5kyGyWRi2AHDDhGdJae0Fq9tzQUAPHZ9H0RolKiTuCYi+k14VAy0uvhmr0XECKhzj8PqcMKmjkJ8pEai6nwXOygTEQDA6RT4+5qDsDkERvfSY1L/JKlLIqI2kMtkSIxqvN1cXMOZlFvCsENEAID3dp3EnpNVCFcrsGhKP8g41JzIbyQ1hZ1qhp2WMOwQEU6U1+OZdUcBAA+O64nk6FCJKyKi9uh05nu2uNrMRUFbwLBDFOQcToG/fbwfDTYHLkvX4fbMLlKXRETtlBgVArkMqLPYYTRzcsHfY9ghCnJvfH8ce05WIVKjxAv/MxByOW9fEfkblUKOBG3jraxTVbyV9XsMO0RBbG9BFV78NhsA8Pgf+iIlJkziiojIXU23soqqTRJX4nsYdoiCVI3Jhnnv74XdKTCpfxKmDu4kdUlEdBE6xfzWb4eaY9ghCkJCCDz86QGcqm5AZ10YFk/tz9FXRH4uKSoEMgA1DTbUW51Sl+NTGHaIgtDr3x/H+sMGqBQyLJs2GNoQldQlEdFF0igVrgkFDXXspHw2hh2iILM1uwzPrW8cZv7Y9X3RPyVK4oqIyFOabmWVMuw0w7BDFESOn67DXz7YCyGAW4el4rbhnaUuiYg8qKmTMq/sNMewQxQkKuosuHPVLzCa7RjcORpP/KEv++kQBZimsFNjcUIRoZO4Gt/BsEMUBMw2B2a/sxsnK0xIiQnF6zOGQqNUSF0WEXlYiEoB/Zl+OyFpAyWuxndw1XOiAGdzOPGXD/bi14JqaEOUWHXnpdAICwyGqjbtX1ZW1sEVEpEnperCUFZrYdg5C8MOUQBzOAX+9tF+fJtVCrVSjjduHwp9iEDX9G6orChv17GsVmsHVUlEnpQaE4o9J6sQkjaQ62SdwbBDFKCcToF/rDmIL/cXQymXYcVtg3FZeiwMBgMqK8rxwIovEBEd2+pxSk/mYvnDd8Bms3mhaiK6WMnRoZDLAKU2HoXVFiQlSV2R9Bh2iAKQEAIL12bhw92FkMuAl2+5BNf0SmjWJiI6FlpdfKvHqq2u6KgyiagDqBRy6MOVMNTZ8XOBEcN6S12R9NhBmSjACCHw/IZsrNp+AgDwrz8OxKQB/NOOKJgkRTZey9hdWCtxJb6BV3aIAoDRaITJZIIQAv/+8RTe21MKAHj4ms64opMSBoPB1ZYdjokCX3KkEntLgF8La+FwCijkwT3NBMMOkZ8zGo1nOhxXQDf2bkReMhEAUPnd67jnua9wz3n2Y4djosAVF6aA01IPI8Jx8FQNBqVGS12SpBh2iPycyWRCZWUlRi78AgX1jXemr+gciowHHwbw8Dnt2eGYKPDJZTI0nNiH8J5XYGt2GcOO1AUQ0cWx2p2In/x3FNTLIZcB4/omIiMh8rzt2eGYKDiYj+85E3ZO469jMqQuR1I+3UH5iSeegEwma/bo1auXa7vZbMbcuXMRGxuLiIgITJ06FaWlpRJWTORdtWYb/vZlLsJ6Xg6FDJjUP+mCQYeIgkfD8T0AgP1F1aios0hcjbR8OuwAQN++fVFSUuJ6/Pjjj65t999/P7766it8/PHH2LZtG4qLi3HjjTdKWC2R95Qazbjp9Z34paAWTmsDxnQLR3p8hNRlEZGPcNRVoHtcKIQAfshp3ySigcbnw45SqURiYqLrERcXBwCoqanBm2++iSVLluCaa67BkCFDsHLlSmzfvh07d+6UuGqijpVbVosbX9uOIyVGxIQpUfr+35EcqZK6LCLyMZd3iQIAbMkO7lGYPh92cnJykJycjPT0dEyfPh0FBQUAgD179sBms2HMmDGutr169ULnzp2xY8cOqcol6nC7T1Ri6vIdOFXdgK5x4fi/m3vBWpondVlE5IMyu2gBAN8fOw2HM3iXjvDpsDN8+HCsWrUK69evx/Lly5Gfn4+RI0eitrYWBoMBarUa0dHRzfZJSEhoNqdISywWC4xGY7MHkT9Yf6gE0/5vF2oabBiUGo1P774cnaI0UpdFRD6qf1IEIkOUqDLZsL+oWupyJOPTo7EmTJjg+veAAQMwfPhwpKWl4aOPPkJoaKjbx128eDGefPJJT5RI5DVvbz+BJ746DCGAMb0T8OqtlyBUrYCBE6QS0XkoFTJc2SMeXx8swaYjpRjcOUbqkiTh01d2fi86OhoZGRnIzc1FYmIirFYrqqurm7UpLS1FYmLiBY+zYMEC1NTUuB6FhYUdWDXRxXE4BZ7+OguPf9kYdKYN74wVtw1GqFohdWlE5Aeu7dO4Lt6Gw8E7Wtmvwk5dXR3y8vKQlJSEIUOGQKVSYdOmTa7t2dnZKCgoQGZm5gWPo9FooNVqmz2IfFG9xY7/fXcP/vNDPgDgb9dm4Okp/aBU+NW3LhFJ6OpeeqgUMuSW1SHvdJ3U5UjCp29jPfDAA7j++uuRlpaG4uJiPP7441AoFLj11lsRFRWFWbNmYf78+dDpdNBqtbj33nuRmZmJyy67TOrSiS7asaLTmPfhQRw73QC1QoZ/ju2Ca3tGnjOXFNe6IqILiQpVIbNbHL4/dhobDhtwz6juUpfkdT4ddoqKinDrrbeioqIC8fHxGDFiBHbu3In4+HgAwEsvvQS5XI6pU6fCYrFg3LhxeO211ySumuji7cw+hZv+vQXy8Bg46qtxcs1TuP2Zoxfch2tdEdH5jO+b2Bh2DjHs+JzVq1dfcHtISAiWLVuGZcuWeakioo63/lAJ7lu9H/LwGGhVAmMvTUXkiP87b3uudUVErbm2TwIe+fwg9hfVoKSmAUlR7g/y8Uc+HXaIgokQAiu2Hcdz6xuv4DQc343pU0YjLl5/wf241hURtSY+UoOhaTH45UQVvj1cipmXd5G6JK9iL0ciH2C1O/HwpwdcQeePA+NR9slCqBUyiSsjokAxrm/jSOWvD5RIXIn3MewQSazaZMXtb+3CR7uLIJcBT1zfBw9c3RkQTqlLI6IAMmlAEmQy4OcTlThV3SB1OV7F21hEXmI0GmEymZq9llfegIe+ysWpGivC1HI8NSEdl3cN5QgrIvK4pKhQDO+qw87jlfhyXzHuHtVN6pK8hmGHyAuMRiO6pndDZcVvKw+HZmQibtJ8yNWhsFcbkPvpIkxddLLZfhxhRUSeNGVQJ+w8XonP955i2CEizzKZTKisKMcDK75AeJQO+wxm7DNYAABJEUqM6p+BkKvfc7XnCCsi6ggT+ifhsS8OI7u0FkdKjOidFByT6jLsEHmRJlKH74vsOF7eGHQGpUZjZPc4yOXNOyJzhBURdYSoUBWu7hWPDYdL8fm+U0ETdthBmchLlNFJWHusFsfL66GQy3Bt7wRclRF/TtAhIupIUwZ1AgB8ua8YDqeQuBrvYNgh8oKdJ2qQOPMlVJudCNco8MfBKeiTHBx/URGRb7m6lx7RYSqU1Jix7VhwDIZg2CHqQEIIvL4tD/O/yIUiJALxYQrcemlnJEaFSF0aEQWpEJUCUwenAADe31UgcTXewbBD1EFMVjvuW70Pi785CqcAavdvwIQeEQjXsKscEUnr1mGdAQCbj5ahOAjm3GHYIeoA+eX1uGHZdny5vxhKuQx/G5WKyvWvQsH+OUTkA7rrIzC8qw5OAXz4S6HU5XQ4/olJ5KaWJgkEgB+PV+OJ9SdQZ3VAF6bEM5PSkawK/L+ciMi/TBveGbvyK/HR7kLce013KBWBe/2DYYfIDS1NEgiZHFFX3IroK24FAJiLsnDgi2cx4fFKVxNOEkhEvmJ8v0TowtUoqTFjY1YpJvRPkrqkDsOwQ+SGsycJjIiOhcXuxLYTJpyqtQMAesepcenATCiu/xIAJwkkIt+jUSowfXhnvLo5F69/fxzj+yVCJgvMW+0MO0QXISI6FhaVFmuPFMNotkMpl2F0Lz16/W6iLk4SSERSaG2dvfHdwvD6Nhn2FVbjlxNVGNZV56XKvIthh+gi5FZYsb2oEA6nQFSoCpP6JyE+UiN1WUQU5MymekAmw8CBA1ttqxs3F5GDJmDZ5mwMm5Xpheq8j2GHyA02hxO6a+fgh4LGDspdYsMwrm8iQlQKiSsjIgJsVjMgBO5Zshr65JQLti05XYFvCp3YllOJ3LJadNdHeqlK72HYIWonQ40Z93xyDJGDrwMADO+qw/CuuoC9101E/is8KgZaXXyr7Ro2f4OwjEz8e3Mult5yiRcq8y6GHaIzzjeU/Gw/Fxjx+Pp8VJnscJjrMK5PAvp2jfVShUREHaNm+2qEZWTii/3FuHtUd/RMDKyrOww7RDjPUPKznRlWHnX5zZDJ5LCW5eP0Z08j4ZX3vVsoEVEHsJbmYXSPGGzKqcKL32bjjduHSl2SRzHsEOHcoeTNttmc2HaiHoY6BwAgI1aNrjGReKPawKHkRBQwZmcmY0tuFb7NKsW+wmoMSo2WuiSPCdzpEoncEBEdC60u3vWocIbhy+zGoKNSyDC+byImDEpDZHSM1KUSEXlUF10IbjyzQOgz645ACCFxRZ7DsEPUAovdgY1ZpfhyfzEabA7ERahx67DOAXcfm4jobH8d0wMhKjl+zq/El/uLpS7HYxh2iH6noNKE/+4sQFaJEQAwuHM0bh6aipgwtcSVERF1rJSYMMy7ujsA4Kmvj8BoDoxb9Qw7RGfINeHYUWjCZ3tPoc5iR1SoCn8ckoKRPeIDeoE8IqKzzb4yHV3jwnG61oKXNh6TuhyP4E9wCnoOp8BXh8uRPPt1HC1vXKhzQEoUpg/vjE7RoRJXR0TkXRqlAk/8oS8A4O3tJ7DnZGUre/g+hh0KWkIIbMwqxcSXf8DTG09CER6N6BA5brykE67uqYeKV3OIKEhdlRGPGy/pBKcA7v9wP+osdqlLuigcek5+oy2T/p0tLCwMWq32nNftDie+PliCN74/jsPFjf1yIjUKnPzmDcycPx/RujCP1UxE5K+emNwXu/IrUVBpwqKvsvDcHwdIXZLbGHbIL7Q66V8LdLFxyD+e5wo8hZUmfLS7EB/vLoLBaAYAhKoUuOOKLrihVwR6LvwMctnfOqR+IiJ/ow1R4cWbBuLW/+zEh7sLcXn3WEwe1EnqstzCsEN+4UKT/rWkrroCL8yZgv0ny3G48jS+OViC/UU1ru2x4WrMvLwLZlyWhphwNQwGQ0eWT0Tkly5Lj8U9o7ph2ZY8PPTJAXSNC8eAlGipy2o3hh3yK02T/rVECIGaBhsKKxtwvCoEKff+FzPeO+LaLpcBl3eLwy3DUnFtnwRolFyhnIioNfOv7YkjJbXYfLQMd72zB1/eewX0kSFSl9UuDDvk1+osdhRVmlBQZUJRVQNqzb91olOERSFUJcfw9FiM7p2A8X0TER+pkbBaIiLfVVZWdt5t/7g6GXmlRpysMmPa69uxcsZApCb4zyLIDDskmfZ0OG76JnQKgRMV9cgvr0dhpQlVpuYTXsllQFJUKPQhAt+88Bfk7NqI1E7JHq+diChQmE31gEyGgQMHXrCdMjoJCdOfQy50uPwfH2Dn4mnopNd5qcqLw7BDkmhXh2O5EiFdBiF24n344GANrI6aZpv1kRqk6sKQGhOK5OhQqBRyGCtP4/NTWRw+TkTUCpvVDAiBe5ashj455YJtqxsc+PqYEdb4rvjze/vxzp8v94sr5gw7JInWOhw7nALFtXbkV1tRWGODtXHBcVgdQJhagfT4cKTpwpESE4oQFfveEBFdrPComPP2iWyiBTBeCHy2rxhHSoEbXvsJq+68FN31vr1uIP/sJUmdvcp4WFQsyh2h2FHiwOpDtfjueD3yKhuDTohCwLhnLUalKDBrRFeM7pWA7voIBh0iIi+LDVPC8O4DSInSoKiqATcs246vfHzRUIYdkpTdKZBbVof1hwz4zw/5+OpACY4aamF1OBGuUWBQSjT+ODgFk1KdqPpuBfRhcshlMqnLJiIKavaqYvzn5p64tEsMai123PvBXvzto/2oMfnmwqG8jUVeZ7LasTmnCnHXP4gPDtbA7vytD06ERonu+gj00EcgKSoEsjPB5lSVVNUSEVFLYsJU+GD2ZXhlUw7+vSUXn/5ahM1HS/HguF64+dJUKOS+84cpww55haHGjE1HS/FdVil+yquA1e5EeJ+rYHcCkSG/BZxE7W8Bh4iIfJtSIcf8sT0xMiMe/1hzEDlldfjHZwfxnx+O43+vTMcNgzv5xJxmARN2li1bhn/9618wGAwYOHAgXn31VQwbNkzqsoJWZb0VP+dX4uf8SuzKr3CtQdUkWavGkY0fYPqMmUhPSWTAISLyY5d20WHdfSPx350nsfS7HOSX1+Pvaw7iufVHMXlQJ/xxSAr6dYqSrL6ACDsffvgh5s+fjxUrVmD48OFYunQpxo0bh+zsbOj1eqnLC3jldRYcKTHiaEktjhiMOFhUg5yyumZtZDLgktRojO6dgGv7JCDSWYfkR1Yifs6sDg06F5oky512RETUMpVCjjuv6Iqbhqbig58L8H8/5MNgNGPV9hONjzsvxaie0vxODoiws2TJEsyePRt33nknAGDFihX4+uuv8dZbb+Hvf/+7xNX5L4vdgWqTDVUmK6rqbag2WVFeZ0FRVcOZhwmFVQ2orLe2uH9GQgSGddVheNdYXJYe22wuBoOhvkNrb+skWb9ntbb8WYiIqLkL/ZF4XY8wjO/WBz+fNOLrrAocLKnH5d3ivFhdc34fdqxWK/bs2YMFCxa4XpPL5RgzZgx27NghYWXAzLd+xsnyOsggIJfLoJABcpkM8qb/ygGFTNbs3yqlAhq1CnKZDEpF4zaFXNbY7qz/KuWNrzduh2ubUzTOMux0CjiEgDjz3OEUcIrG9aMcZ7ZZ7U6YbQ6YbY3/bbA5Gv9rdaC6wQZT0+Q2rZABSI3RoHtcKHonRWFQl3gMSYuBLlzdsSf4AtozSRYAlJ7MxfKH74DN5psjCYiIfIU7f0zq4hNg/ssxqLXaDqzs/Pw+7JSXl8PhcCAhIaHZ6wkJCTh69GiL+1gsFlgsFtfzmprG0UBGo7HF9u7KLSpFYZXZo8f0NuF0wGmuhdNcB4e5Fk5TLRzGMtiNp2E3lsFeUw57TQlO2K34AUB0TAw2b9qEimIjKi5w3NOnTwMAKkoKG79xWlFd2jiHQ1XpKSjacNerqb3VbGrT8a3mBreO7wvtfamWYGvvS7X4e3tfqiXY2rf32KcLjwNC4Na/v4hYfVKr7U3Garz1xD0oLS1t/eDt1PR7Wwhx4YbCz506dUoAENu3b2/2+oMPPiiGDRvW4j6PP/64AMAHH3zwwQcffATAo7Cw8IJZwe+v7MTFxUGhUJyTGEtLS5GYmNjiPgsWLMD8+fNdz51OJyorKxEbG+vRzrJGoxGpqakoLCyEVqJLd8GA59l7eK69g+fZO3ievaMjz7MQArW1tUhOvvCCz34fdtRqNYYMGYJNmzZhypQpABrDy6ZNmzBv3rwW99FoNNBomi9cFh0d3WE1arVafiN5Ac+z9/BcewfPs3fwPHtHR53nqKioVtv4fdgBgPnz52PmzJkYOnQohg0bhqVLl6K+vt41OouIiIiCV0CEnZtvvhmnT5/GY489BoPBgEGDBmH9+vXndFomIiKi4BMQYQcA5s2bd97bVlLRaDR4/PHHz7llRp7F8+w9PNfewfPsHTzP3uEL51kmRGvjtYiIiIj8l1zqAoiIiIg6EsMOERERBTSGHSIiIgpoDDsXadmyZejSpQtCQkIwfPhw/Pzzzxds//HHH6NXr14ICQlB//79sW7dOi9V6t/ac57/85//YOTIkYiJiUFMTAzGjBnT6v8XatTer+cmq1evhkwmc811Ra1r77murq7G3LlzkZSUBI1Gg4yMDP78aIP2nuelS5eiZ8+eCA0NRWpqKu6//36Yzf697E9H+/7773H99dcjOTkZMpkMn3/+eav7bN26FYMHD4ZGo0H37t2xatWqji3SM4s2BKfVq1cLtVot3nrrLXH48GExe/ZsER0dLUpLS1ts/9NPPwmFQiGef/55kZWVJR599FGhUqnEwYMHvVy5f2nveZ42bZpYtmyZ2Lt3rzhy5Ii44447RFRUlCgqKvJy5f6lvee5SX5+vujUqZMYOXKkmDx5sneK9XPtPdcWi0UMHTpUTJw4Ufz4448iPz9fbN26Vezbt8/LlfuX9p7n9957T2g0GvHee++J/Px8sWHDBpGUlCTuv/9+L1fuX9atWyceeeQRsWbNGgFAfPbZZxdsf/z4cREWFibmz58vsrKyxKuvvioUCoVYv359h9XIsHMRhg0bJubOnet67nA4RHJysli8eHGL7W+66SYxadKkZq8NHz5c/O///m+H1unv2nuef89ut4vIyEjx9ttvd1SJAcGd82y328Xll18u/u///k/MnDmTYaeN2nuuly9fLtLT04XVavVWiQGhved57ty54pprrmn22vz588UVV1zRoXUGkraEnYceekj07du32Ws333yzGDduXIfVxdtYbrJardizZw/GjBnjek0ul2PMmDHYsWNHi/vs2LGjWXsAGDdu3Hnbk3vn+fdMJhNsNht0Ol1Hlen33D3PCxcuhF6vx6xZs7xRZkBw51x/+eWXyMzMxNy5c5GQkIB+/frhmWeegcPh8FbZfsed83z55Zdjz549rltdx48fx7p16zBx4kSv1BwspPhdGDCTCnpbeXk5HA7HObM0JyQk4OjRoy3uYzAYWmxvMBg6rE5/5855/r2HH34YycnJ53xz0W/cOc8//vgj3nzzTezbt88LFQYOd8718ePHsXnzZkyfPh3r1q1Dbm4u7rnnHthsNjz++OPeKNvvuHOep02bhvLycowYMQJCCNjtdsyZMwf/+Mc/vFFy0Djf70Kj0YiGhgaEhoZ6/D15ZYcC2rPPPovVq1fjs88+Q0hIiNTlBIza2lrMmDED//nPfxAXFyd1OQHP6XRCr9fjjTfewJAhQ3DzzTfjkUcewYoVK6QuLaBs3boVzzzzDF577TX8+uuvWLNmDb7++mssWrRI6tLoIvHKjpvi4uKgUChQWlra7PXS0lIkJia2uE9iYmK72pN757nJCy+8gGeffRbfffcdBgwY0JFl+r32nue8vDycOHEC119/ves1p9MJAFAqlcjOzka3bt06tmg/5c7XdFJSElQqFRQKheu13r17w2AwwGq1Qq1Wd2jN/sid8/zPf/4TM2bMwJ///GcAQP/+/VFfX4+77roLjzzyCORyXh/whPP9LtRqtR1yVQfglR23qdVqDBkyBJs2bXK95nQ6sWnTJmRmZra4T2ZmZrP2ALBx48bztif3zjMAPP/881i0aBHWr1+PoUOHeqNUv9be89yrVy8cPHgQ+/btcz3+8Ic/4Oqrr8a+ffuQmprqzfL9ijtf01dccQVyc3NdgRIAjh07hqSkJAad83DnPJtMpnMCTVPAFFxZyWMk+V3YYV2fg8Dq1auFRqMRq1atEllZWeKuu+4S0dHRwmAwCCGEmDFjhvj73//uav/TTz8JpVIpXnjhBXHkyBHx+OOPc+h5G7T3PD/77LNCrVaLTz75RJSUlLgetbW1Un0Ev9De8/x7HI3Vdu091wUFBSIyMlLMmzdPZGdni7Vr1wq9Xi+eeuopqT6CX2jveX788cdFZGSk+OCDD8Tx48fFt99+K7p16yZuuukmqT6CX6itrRV79+4Ve/fuFQDEkiVLxN69e8XJkyeFEEL8/e9/FzNmzHC1bxp6/uCDD4ojR46IZcuWcei5r3v11VdF586dhVqtFsOGDRM7d+50bbvqqqvEzJkzm7X/6KOPREZGhlCr1aJv377i66+/9nLF/qk95zktLU0AOOfx+OOPe79wP9Per+ezMey0T3vP9fbt28Xw4cOFRqMR6enp4umnnxZ2u93LVfuf9pxnm80mnnjiCdGtWzcREhIiUlNTxT333COqqqq8X7gf2bJlS4s/c5vO7cyZM8VVV111zj6DBg0SarVapKeni5UrV3ZojVz1nIiIiAIa++wQERFRQGPYISIiooDGsENEREQBjWGHiIiIAhrDDhEREQU0hh0iIiIKaAw7REREFNAYdoiIiCigMewQ0UVbtWoVoqOjL/o4MpkMn3/++UUfx5NOnDgBmUyGffv2SV0KEbmJYYeIiIgCGsMOEZGfstlsUpdA5BcYdoiCyCeffIL+/fsjNDQUsbGxGDNmDOrr6wEAb731Fvr27QuNRoOkpCTMmzfPtd+SJUvQv39/hIeHIzU1Fffccw/q6uou+F5ffPEFBg8ejJCQEKSnp+PJJ5+E3W53bc/JycGVV16JkJAQ9OnTBxs3bmzXZyksLMRNN92E6Oho6HQ6TJ48GSdOnHBtv+OOOzBlyhQ888wzSEhIQHR0NBYuXAi73Y4HH3wQOp0OKSkpWLlyZbPj/vzzz7jkkksQEhKCoUOHYu/evW2uqaqqCtOnT0d8fDxCQ0PRo0ePZscvKirCrbfeCp1Oh/DwcAwdOhS7du1ybV++fDm6desGtVqNnj174t133212fJlMhuXLl+MPf/gDwsPD8fTTTwNo/VwTBb0OXWaUiHxGcXGxUCqVYsmSJSI/P18cOHBALFu2TNTW1orXXntNhISEiKVLl4rs7Gzx888/i5deesm170svvSQ2b94s8vPzxaZNm0TPnj3F3Xff7dq+cuVKERUV5Xr+/fffC61WK1atWiXy8vLEt99+K7p06SKeeOIJIYQQDodD9OvXT4wePVrs27dPbNu2TVxyySUCgPjss89a/SxWq1X07t1b/OlPfxIHDhwQWVlZYtq0aaJnz57CYrEIIRpXWo6MjBRz584VR48eFW+++aYAIMaNGyeefvppcezYMbFo0SKhUqlEYWGhEEKI2tpaER8fL6ZNmyYOHTokvvrqK5Geni4AiL1797Za19y5c8WgQYPEL7/8IvLz88XGjRvFl19+6Tp2enq6GDlypPjhhx9ETk6O+PDDD8X27duFEEKsWbNGqFQqsWzZMpGdnS1efPFFoVAoxObNm13HByD0er146623RF5enjh58mSr55qIhGDYIQoSe/bsEQDEiRMnztmWnJwsHnnkkTYf6+OPPxaxsbGu578PO6NHjxbPPPNMs33effddkZSUJIQQYsOGDUKpVIpTp065tn/zzTdtDjvvvvuu6Nmzp3A6na7XLBaLCA0NFRs2bBBCNIadtLQ04XA4XG169uwpRo4c6Xput9tFeHi4+OCDD4QQQrz++usiNjZWNDQ0uNosX768zWHn+uuvF3feeWeL215//XURGRkpKioqWtx++eWXi9mzZzd77X/+53/ExIkTXc8BiL/+9a/N2rR2rolICKV015SIyJsGDhyI0aNHo3///hg3bhzGjh2LP/7xj7DZbCguLsbo0aPPu+93332HxYsX4+jRozAajbDb7TCbzTCZTAgLCzun/f79+/HTTz+5brMAgMPhcO1z5MgRpKamIjk52bU9MzOzzZ9l//79yM3NRWRkZLPXzWYz8vLyXM/79u0Lufy3u/UJCQno16+f67lCoUBsbCzKysoAAEeOHMGAAQMQEhLiVl133303pk6dil9//RVjx47FlClTcPnllwMA9u3bh0suuQQ6na7FfY8cOYK77rqr2WtXXHEFXn755WavDR06tNnz1s51S/9/iIINww5RkFAoFNi4cSO2b9+Ob7/9Fq+++ioeeeQRbNq06YL7nThxAtdddx3uvvtuPP3009DpdPjxxx8xa9YsWK3WFn+Z1tXV4cknn8SNN954zrazg4S76urqMGTIELz33nvnbIuPj3f9W6VSNdsmk8lafM3pdF50TQAwYcIEnDx5EuvWrcPGjRsxevRozJ07Fy+88AJCQ0M98h7h4eHNnnf0uSYKBOygTBREZDIZrrjiCjz55JPYu3cv1Go1Nm7ciC5dupw39OzZswdOpxMvvvgiLrvsMmRkZKC4uPiC7zN48GBkZ2eje/fu5zzkcjl69+6NwsJClJSUuPbZuXNnmz/H4MGDkZOTA71ef87xo6Ki2nyc3+vduzcOHDgAs9nsVl1AY9iaOXMm/vvf/2Lp0qV44403AAADBgzAvn37UFlZed73/umnn5q99tNPP6FPnz4XfL/WzjURMewQBY1du3bhmWeewe7du1FQUIA1a9bg9OnT6N27N5544gm8+OKLeOWVV5CTk4Nff/0Vr776KgCge/fusNlsePXVV3H8+HG8++67WLFixQXf67HHHsM777yDJ598EocPH8aRI0ewevVqPProowCAMWPGICMjAzNnzsT+/fvxww8/4JFHHmnzZ5k+fTri4uIwefJk/PDDD8jPz8fWrVvxl7/8BUVFRW6fo2nTpkEmk2H27NnIysrCunXr8MILL7R5/8ceewxffPEFcnNzcfjwYaxduxa9e/cGANx6661ITEzElClT8NNPP+H48eP49NNPsWPHDgDAgw8+iFWrVmH58uXIycnBkiVLsGbNGjzwwAOtvueFzjURgaOxiIJFVlaWGDdunIiPjxcajUZkZGSIV1991bV9xYoVomfPnkKlUomkpCRx7733urYtWbJEJCUlidDQUDFu3DjxzjvvCACiqqpKCHFuB2UhhFi/fr24/PLLRWhoqNBqtWLYsGHijTfecG3Pzs4WI0aMEGq1WmRkZIj169e3uYOyEEKUlJSI22+/XcTFxQmNRiPS09PF7NmzRU1NjRCisYPy5MmTm+1z1VVXifvuu6/Za2lpac1Gnu3YsUMMHDhQqNVqMWjQIPHpp5+2uYPyokWLRO/evUVoaKjQ6XRi8uTJ4vjx467tJ06cEFOnThVarVaEhYWJoUOHil27drm2v/baayI9PV2oVCqRkZEh3nnnnWbHP9/5ae1cEwU7mRBCSBu3iIiIiDoOb2MRERFRQGPYISKf88wzzyAiIqLFx4QJEySra86cOeeta86cOZLVRUQXxttYRORzKisrzztqKTQ0FJ06dfJyRY3KyspgNBpb3KbVaqHX671cERG1BcMOERERBTTexiIiIqKAxrBDREREAY1hh4iIiAIaww4REREFNIYdIiIiCmgMO0RERBTQGHaIiIgooDHsEBERUUD7fx0Sb9MHAfMgAAAAAElFTkSuQmCC",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sns.histplot(scaled_values, kde=True)\n",
    "plt.xlabel('scaled_emd_score')  # Set the x-axis label to your column's name\n",
    "plt.ylabel('Frequency')    # Set the y-axis label\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### create new column in the dataframe of normalised values - include in publication"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "\n",
    "# Read the CSV file into a DataFrame\n",
    "df = pd.read_csv('/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/hal2p/one-v-all_EMD_dx_SAL1-HAL2.csv')\n",
    "\n",
    "# Calculate the minimum and maximum values of the 'emd_score' column\n",
    "# Log transform\n",
    "log_transformed_data = np.log(column_data)\n",
    "\n",
    "min_val = np.min(log_transformed_data)\n",
    "max_val = np.max(log_transformed_data)\n",
    "\n",
    "\n",
    "scaled_values = (log_transformed_data - min_val) / (max_val - min_val)\n",
    "\n",
    "df['scaled_emd_score'] = scaled_values\n",
    "df2 = df[['id2', 'pH2', 'scaled_emd_score', 'ref_protein']]\n",
    "\n",
    "df.to_csv('/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/hal2p/one-v-all_EMD_dx_SAL1-HAL2_norm.csv', index=False)\n",
    "df2.to_csv('/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/hal2p/one-v-all_EMD_dx_SAL1-HAL2_norm_2.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Change the results csv for visulisation on phylogeny "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "        id2  pH-('7_0', 'AtSAL1_updated-pH_7_6')  \\\n",
      "0     10029                             0.005142   \n",
      "1     10086                             0.003430   \n",
      "2     10276                             0.006173   \n",
      "3     10748                             0.010092   \n",
      "4     10972                             0.010282   \n",
      "..      ...                                  ...   \n",
      "500   AtAHL                             0.002736   \n",
      "501  AtSAL1                             0.004041   \n",
      "502  AtSAL2                             0.005270   \n",
      "503  AtSAL3                             0.012906   \n",
      "504  AtSAL4                             0.003481   \n",
      "\n",
      "     pH-('7_3', 'AtSAL1_updated-pH_7_6')  pH-('7_6', 'AtSAL1_updated-pH_7_6')  \\\n",
      "0                               0.003467                             0.001930   \n",
      "1                               0.003430                             0.003430   \n",
      "2                               0.006173                             0.006173   \n",
      "3                               0.010092                             0.003989   \n",
      "4                               0.005432                             0.005432   \n",
      "..                                   ...                                  ...   \n",
      "500                             0.002736                             0.002736   \n",
      "501                             0.004041                             0.004041   \n",
      "502                             0.005270                             0.005270   \n",
      "503                             0.008984                             0.008984   \n",
      "504                             0.002151                             0.002151   \n",
      "\n",
      "     pH-('7_9', 'AtSAL1_updated-pH_7_6')  pH-('8_2', 'AtSAL1_updated-pH_7_6')  \\\n",
      "0                               0.001930                             0.001930   \n",
      "1                               0.003430                             0.003430   \n",
      "2                               0.006173                             0.002398   \n",
      "3                               0.003989                             0.003989   \n",
      "4                               0.005432                             0.005432   \n",
      "..                                   ...                                  ...   \n",
      "500                             0.002736                             0.003619   \n",
      "501                             0.003105                             0.003105   \n",
      "502                             0.002358                             0.003847   \n",
      "503                             0.006559                             0.003110   \n",
      "504                             0.002151                             0.002151   \n",
      "\n",
      "     pH-('7_0', 'AtSAL1_updated-pH_7_9')  pH-('7_3', 'AtSAL1_updated-pH_7_9')  \\\n",
      "0                               0.005142                             0.003467   \n",
      "1                               0.003430                             0.003430   \n",
      "2                               0.006173                             0.006173   \n",
      "3                               0.010092                             0.010092   \n",
      "4                               0.010282                             0.005432   \n",
      "..                                   ...                                  ...   \n",
      "500                             0.002736                             0.002736   \n",
      "501                             0.004041                             0.004041   \n",
      "502                             0.005270                             0.005270   \n",
      "503                             0.012906                             0.008984   \n",
      "504                             0.003481                             0.002151   \n",
      "\n",
      "     pH-('7_6', 'AtSAL1_updated-pH_7_9')  pH-('7_9', 'AtSAL1_updated-pH_7_9')  \\\n",
      "0                               0.001930                             0.001930   \n",
      "1                               0.003430                             0.003430   \n",
      "2                               0.006173                             0.006173   \n",
      "3                               0.003989                             0.003989   \n",
      "4                               0.005432                             0.005432   \n",
      "..                                   ...                                  ...   \n",
      "500                             0.002736                             0.002736   \n",
      "501                             0.004041                             0.003105   \n",
      "502                             0.005270                             0.002358   \n",
      "503                             0.008984                             0.006559   \n",
      "504                             0.002151                             0.002151   \n",
      "\n",
      "     ...  pH-('7_0', '1ka1_hal2-pH_7_9')  pH-('7_3', '1ka1_hal2-pH_7_9')  \\\n",
      "0    ...                        0.005162                        0.005217   \n",
      "1    ...                        0.001473                        0.001473   \n",
      "2    ...                        0.005606                        0.005606   \n",
      "3    ...                        0.009609                        0.009609   \n",
      "4    ...                        0.009656                        0.004716   \n",
      "..   ...                             ...                             ...   \n",
      "500  ...                        0.001501                        0.001501   \n",
      "501  ...                        0.002922                        0.002922   \n",
      "502  ...                        0.004847                        0.004847   \n",
      "503  ...                        0.012335                        0.008483   \n",
      "504  ...                        0.003243                        0.002083   \n",
      "\n",
      "     pH-('7_6', '1ka1_hal2-pH_7_9')  pH-('7_9', '1ka1_hal2-pH_7_9')  \\\n",
      "0                          0.004378                        0.004378   \n",
      "1                          0.001473                        0.001473   \n",
      "2                          0.005606                        0.005606   \n",
      "3                          0.003796                        0.003796   \n",
      "4                          0.004716                        0.004716   \n",
      "..                              ...                             ...   \n",
      "500                        0.001501                        0.001501   \n",
      "501                        0.002922                        0.002016   \n",
      "502                        0.004847                        0.001056   \n",
      "503                        0.008483                        0.006070   \n",
      "504                        0.002083                        0.002083   \n",
      "\n",
      "     pH-('8_2', '1ka1_hal2-pH_7_9')  pH-('7_0', '1ka1_hal2-pH_8_2')  \\\n",
      "0                          0.004378                        0.005162   \n",
      "1                          0.001473                        0.001473   \n",
      "2                          0.001743                        0.005606   \n",
      "3                          0.003796                        0.009609   \n",
      "4                          0.004716                        0.009656   \n",
      "..                              ...                             ...   \n",
      "500                        0.002856                        0.001501   \n",
      "501                        0.002016                        0.002922   \n",
      "502                        0.003528                        0.004847   \n",
      "503                        0.001825                        0.012335   \n",
      "504                        0.002083                        0.003243   \n",
      "\n",
      "     pH-('7_3', '1ka1_hal2-pH_8_2')  pH-('7_6', '1ka1_hal2-pH_8_2')  \\\n",
      "0                          0.005217                        0.004378   \n",
      "1                          0.001473                        0.001473   \n",
      "2                          0.005606                        0.005606   \n",
      "3                          0.009609                        0.003796   \n",
      "4                          0.004716                        0.004716   \n",
      "..                              ...                             ...   \n",
      "500                        0.001501                        0.001501   \n",
      "501                        0.002922                        0.002922   \n",
      "502                        0.004847                        0.004847   \n",
      "503                        0.008483                        0.008483   \n",
      "504                        0.002083                        0.002083   \n",
      "\n",
      "     pH-('7_9', '1ka1_hal2-pH_8_2')  pH-('8_2', '1ka1_hal2-pH_8_2')  \n",
      "0                          0.004378                        0.004378  \n",
      "1                          0.001473                        0.001473  \n",
      "2                          0.005606                        0.001743  \n",
      "3                          0.003796                        0.003796  \n",
      "4                          0.004716                        0.004716  \n",
      "..                              ...                             ...  \n",
      "500                        0.001501                        0.002856  \n",
      "501                        0.002016                        0.002016  \n",
      "502                        0.001056                        0.003528  \n",
      "503                        0.006070                        0.001825  \n",
      "504                        0.002083                        0.002083  \n",
      "\n",
      "[505 rows x 31 columns]\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Step 1: Read the CSV file into a DataFrame\n",
    "df = pd.read_csv('/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/AtSAL1_ScHAL2-one-v-all_EMD.csv')\n",
    "\n",
    "# Assuming the columns in your CSV are named 'ID', 'Condition', and 'Score'\n",
    "# Step 2: Pivot the DataFrame\n",
    "df_pivot = df.pivot(index='id2', columns=['pH2', 'ref_protein'], values='emd_score')\n",
    "\n",
    "# Rename the columns to include 'score_' prefix\n",
    "df_pivot.columns = [f'pH-{col}' for col in df_pivot.columns]\n",
    "\n",
    "# Step 3: Reset the index to turn the 'ID' index into a column\n",
    "df_pivot.reset_index(inplace=True)\n",
    "\n",
    "print(df_pivot)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['id2', 'pH-('7_0', 'AtSAL1_updated-pH_7_6')',\n",
       "       'pH-('7_3', 'AtSAL1_updated-pH_7_6')',\n",
       "       'pH-('7_6', 'AtSAL1_updated-pH_7_6')',\n",
       "       'pH-('7_9', 'AtSAL1_updated-pH_7_6')',\n",
       "       'pH-('8_2', 'AtSAL1_updated-pH_7_6')',\n",
       "       'pH-('7_0', 'AtSAL1_updated-pH_7_9')',\n",
       "       'pH-('7_3', 'AtSAL1_updated-pH_7_9')',\n",
       "       'pH-('7_6', 'AtSAL1_updated-pH_7_9')',\n",
       "       'pH-('7_9', 'AtSAL1_updated-pH_7_9')',\n",
       "       'pH-('8_2', 'AtSAL1_updated-pH_7_9')',\n",
       "       'pH-('7_0', 'AtSAL1_updated-pH_8_2')',\n",
       "       'pH-('7_3', 'AtSAL1_updated-pH_8_2')',\n",
       "       'pH-('7_6', 'AtSAL1_updated-pH_8_2')',\n",
       "       'pH-('7_9', 'AtSAL1_updated-pH_8_2')',\n",
       "       'pH-('8_2', 'AtSAL1_updated-pH_8_2')', 'pH-('7_0', '1ka1_hal2-pH_7_6')',\n",
       "       'pH-('7_3', '1ka1_hal2-pH_7_6')', 'pH-('7_6', '1ka1_hal2-pH_7_6')',\n",
       "       'pH-('7_9', '1ka1_hal2-pH_7_6')', 'pH-('8_2', '1ka1_hal2-pH_7_6')',\n",
       "       'pH-('7_0', '1ka1_hal2-pH_7_9')', 'pH-('7_3', '1ka1_hal2-pH_7_9')',\n",
       "       'pH-('7_6', '1ka1_hal2-pH_7_9')', 'pH-('7_9', '1ka1_hal2-pH_7_9')',\n",
       "       'pH-('8_2', '1ka1_hal2-pH_7_9')', 'pH-('7_0', '1ka1_hal2-pH_8_2')',\n",
       "       'pH-('7_3', '1ka1_hal2-pH_8_2')', 'pH-('7_6', '1ka1_hal2-pH_8_2')',\n",
       "       'pH-('7_9', '1ka1_hal2-pH_8_2')', 'pH-('8_2', '1ka1_hal2-pH_8_2')'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_pivot.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_pivot.columns = ['unid', \n",
    "       'pH-7_0-AtSAL1-pH_7_6',\n",
    "       'pH-7_3-AtSAL1-pH_7_6',\n",
    "       'pH-7_6-AtSAL1-pH_7_6',\n",
    "       'pH-7_9-AtSAL1-pH_7_6',\n",
    "       'pH-8_2-AtSAL1-pH_7_6',\n",
    "       'pH-7_0-AtSAL1-pH_7_9',\n",
    "       'pH-7_3-AtSAL1-pH_7_9',\n",
    "       'pH-7_6-AtSAL1-pH_7_9',\n",
    "       'pH-7_9-AtSAL1-pH_7_9',\n",
    "       'pH-8_2-AtSAL1-pH_7_9',\n",
    "       'pH-7_0-AtSAL1-pH_8_2',\n",
    "       'pH-7_3-AtSAL1-pH_8_2',\n",
    "       'pH-7_6-AtSAL1-pH_8_2',\n",
    "       'pH-7_9-AtSAL1-pH_8_2',\n",
    "       'pH-8_2-AtSAL1-pH_8_2', \n",
    "       'pH-7_0-ScHAL2-pH_7_6',\n",
    "       'pH-7_3-ScHAL2-pH_7_6', \n",
    "       'pH-7_6-ScHAL2-pH_7_6',\n",
    "       'pH-7_9-ScHAL2-pH_7_6', \n",
    "       'pH-8_2-ScHAL2-pH_7_6',\n",
    "       'pH-7_0-ScHAL2-pH_7_9', \n",
    "       'pH-7_3-ScHAL2-pH_7_9',\n",
    "       'pH-7_6-ScHAL2-pH_7_9', \n",
    "       'pH-7_9-ScHAL2-pH_7_9',\n",
    "       'pH-8_2-ScHAL2-pH_7_9', \n",
    "       'pH-7_0-ScHAL2-pH_8_2',\n",
    "       'pH-7_3-ScHAL2-pH_8_2', \n",
    "       'pH-7_6-ScHAL2-pH_8_2',\n",
    "       'pH-7_9-ScHAL2-pH_8_2', \n",
    "       'pH-8_2-ScHAL2-pH_8_2']\n",
    "df_pivot.to_csv('/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/AtSAL1_ScHAL2-one-v-all_EMD_2.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find the pLDDT scores of the active sites"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "from concurrent.futures import ProcessPoolExecutor\n",
    "from Bio.PDB import PDBParser, PDBIO, Select\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "\n",
    "def find_b_factors(pdb_file, output_dir):\n",
    "    \"\"\"\n",
    "    Find atoms within a specified rectangular prism in a PDB file and calculate their mean B-factor (pLDDT).\n",
    "\n",
    "    Parameters:\n",
    "    pdb_file (str): Path to the input PDB file.\n",
    "    output_dir (str): Directory to save the output PDB file.\n",
    "\n",
    "    Returns:\n",
    "    tuple: (short_name, mean_b_factor, std_error) for the processed PDB file.\n",
    "    \"\"\"\n",
    "\n",
    "    def compute_bounding_box(corner_coords):\n",
    "        \"\"\"\n",
    "        Compute the bounding box from corner coordinates.\n",
    "\n",
    "        Parameters:\n",
    "        corner_coords (numpy.ndarray): Array of corner coordinates.\n",
    "\n",
    "        Returns:\n",
    "        tuple: Minimum and maximum coordinates of the bounding box.\n",
    "        \"\"\"\n",
    "        min_coords = np.min(corner_coords, axis=0)\n",
    "        max_coords = np.max(corner_coords, axis=0)\n",
    "        return min_coords, max_coords\n",
    "\n",
    "    def find_atoms_in_prism(pdb_file, min_coords, max_coords):\n",
    "        \"\"\"\n",
    "        Read the PDB file and extract atoms within the rectangular prism.\n",
    "\n",
    "        Parameters:\n",
    "        pdb_file (str): Path to the input PDB file.\n",
    "        min_coords (numpy.ndarray): Minimum coordinates of the bounding box.\n",
    "        max_coords (numpy.ndarray): Maximum coordinates of the bounding box.\n",
    "\n",
    "        Returns:\n",
    "        tuple: List of atoms within the prism and the structure.\n",
    "        \"\"\"\n",
    "        parser = PDBParser(QUIET=True)\n",
    "        structure = parser.get_structure('protein', pdb_file)\n",
    "        \n",
    "        atoms_in_prism = []\n",
    "\n",
    "        for model in structure:\n",
    "            for chain in model:\n",
    "                for residue in chain:\n",
    "                    for atom in residue:\n",
    "                        coord = atom.get_coord()\n",
    "                        if np.all(coord >= min_coords) and np.all(coord <= max_coords):\n",
    "                            atoms_in_prism.append(atom)\n",
    "\n",
    "        return atoms_in_prism, structure\n",
    "\n",
    "    class AtomsSelect(Select):\n",
    "        \"\"\"\n",
    "        Custom selection class for PDBIO.\n",
    "        \"\"\"\n",
    "        def __init__(self, atoms):\n",
    "            self.atoms_set = set(atoms)\n",
    "\n",
    "        def accept_atom(self, atom):\n",
    "            return atom in self.atoms_set\n",
    "\n",
    "    def write_atoms_to_pdb(atoms, structure, output_file):\n",
    "        \"\"\"\n",
    "        Write atoms to a new PDB file.\n",
    "\n",
    "        Parameters:\n",
    "        atoms (list): List of atoms to write.\n",
    "        structure (Structure): The structure containing the atoms.\n",
    "        output_file (str): Path to the output PDB file.\n",
    "        \"\"\"\n",
    "        io = PDBIO()\n",
    "        io.set_structure(structure)\n",
    "        io.save(output_file, select=AtomsSelect(atoms))\n",
    "\n",
    "    def calculate_mean_b_factor(atoms):\n",
    "        \"\"\"\n",
    "        Calculate the mean B-factor of the given atoms.\n",
    "\n",
    "        Parameters:\n",
    "        atoms (list): List of atoms.\n",
    "\n",
    "        Returns:\n",
    "        tuple: Mean B-factor and standard error, or None if no atoms are found.\n",
    "        \"\"\"\n",
    "        b_factors = [atom.get_bfactor() for atom in atoms]\n",
    "        if b_factors:\n",
    "            mean_b_factor = np.mean(b_factors)\n",
    "            std_deviation = np.std(b_factors, ddof=1)  # Sample standard deviation\n",
    "            sample_size = len(b_factors)\n",
    "            std_error = std_deviation / np.sqrt(sample_size)\n",
    "            return mean_b_factor, std_error\n",
    "        else:\n",
    "            return None, None\n",
    "\n",
    "    # Define the coordinates of the corners of the rectangular prism\n",
    "    corner_coords = np.array([\n",
    "        [-43.55064101, -41.4898718, -12.31982906],\n",
    "        [-26, -45, -17],\n",
    "        [-21.12, -35.24, -5.29],\n",
    "        [-38.67295493, -31.73449962, -0.61338245],\n",
    "        [-43.8596559, -26.06520805, -25.0471327],\n",
    "        [-26.30901489, -29.57533625, -29.72730364],\n",
    "        [-21.42901489, -19.81533625, -18.01730364],\n",
    "        [-38.98196982, -16.30983587, -13.34068609]\n",
    "    ])\n",
    "\n",
    "    # Compute the bounding box from the corner coordinates\n",
    "    min_coords, max_coords = compute_bounding_box(corner_coords)\n",
    "\n",
    "    # Find atoms within the specified rectangular prism\n",
    "    atoms, structure = find_atoms_in_prism(pdb_file, min_coords, max_coords)\n",
    "\n",
    "    # Create the output file path\n",
    "    short_name = os.path.basename(pdb_file)\n",
    "    output_pdb_file = os.path.join(output_dir, short_name)\n",
    "\n",
    "    # Write atoms to a new PDB file\n",
    "    write_atoms_to_pdb(atoms, structure, output_pdb_file)\n",
    "\n",
    "    # Calculate mean B-factor\n",
    "    mean_b_factor, std_error = calculate_mean_b_factor(atoms)\n",
    "    if mean_b_factor is not None:\n",
    "        print(f\"Mean active site pLDDT for {short_name}: {mean_b_factor}, standard_error {std_error}\")\n",
    "    else:\n",
    "        print(f\"No atoms found within the specified prism in {pdb_file}.\")\n",
    "\n",
    "    short_name = short_name.split('.')[0]\n",
    "\n",
    "    return short_name, mean_b_factor, std_error\n",
    "\n",
    "def process_pdb_files(input_dir, output_dir, max_workers=4):\n",
    "    \"\"\"\n",
    "    Process all PDB files in the input directory in parallel and save results to a DataFrame.\n",
    "\n",
    "    Parameters:\n",
    "    input_dir (str): Directory containing input PDB files.\n",
    "    output_dir (str): Directory to save the output PDB files.\n",
    "    max_workers (int): Maximum number of worker processes.\n",
    "\n",
    "    Returns:\n",
    "    pd.DataFrame: DataFrame containing short_name, mean_b_factor, and std_error for each PDB file.\n",
    "    \"\"\"\n",
    "    # Ensure the output directory exists\n",
    "    os.makedirs(output_dir, exist_ok=True)\n",
    "\n",
    "    # List all PDB files in the input directory\n",
    "    pdb_files = [os.path.join(input_dir, f) for f in os.listdir(input_dir) if f.endswith('.pdb')]\n",
    "\n",
    "    results = []\n",
    "    # Process PDB files in parallel\n",
    "    with ProcessPoolExecutor(max_workers=max_workers) as executor:\n",
    "        futures = [executor.submit(find_b_factors, pdb_file, output_dir) for pdb_file in pdb_files]\n",
    "        for future in futures:\n",
    "            result = future.result()\n",
    "            results.append(result)\n",
    "\n",
    "    # Convert results to a pandas DataFrame\n",
    "    df = pd.DataFrame(results, columns=['short_name', 'mean_b_factor', 'std_error'])\n",
    "    return df\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    input_dir = '/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/'\n",
    "    output_dir = '/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/cropped_pdb'\n",
    "\n",
    "    # Process PDB files using multiple cores\n",
    "    df = process_pdb_files(input_dir, output_dir)\n",
    "\n",
    "    # Print the resulting DataFrame\n",
    "    df.to_csv(f'{input_dir}active_site_pLDDT.csv', index=False)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Check pLDDT distribution - don't include code in publication"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/active_site_pLDDT.csv')\n",
    "column_data = df['mean_b_factor']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAjIAAAGwCAYAAACzXI8XAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjkuMCwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy80BEi2AAAACXBIWXMAAA9hAAAPYQGoP6dpAABSy0lEQVR4nO3deXiTVd4+8PvJnu503xeg7Ptq1XGjIyoqqOMywoDKTx0HZ1TGZXgd9XVFdERwGdEZRZhx5RX3AUcQVATZdwpdKAS6p1uatEma5Pz+SFsplNKmSZ8kvT/XlQuaPDn5PqRpb845zzmSEEKAiIiIKAAp5C6AiIiIyFMMMkRERBSwGGSIiIgoYDHIEBERUcBikCEiIqKAxSBDREREAYtBhoiIiAKWSu4CfM3lcqG0tBTh4eGQJEnucoiIiKgLhBBoaGhAcnIyFIqz97sEfZApLS1FWlqa3GUQERGRB06cOIHU1NSzPh70QSY8PByA+x8iIiJC5mqIiIioK0wmE9LS0tp+j59N0AeZ1uGkiIgIBhkiIqIAc65pIZzsS0RERAGLQYaIiIgCFoMMERERBSwGGSIiIgpYDDJEREQUsBhkiIiIKGAxyBAREVHAYpAhIiKigMUgQ0RERAGLQYaIiIgCFoMMERERBSwGGSIiIgpYDDJEREQUsBhkiIiIKGCp5C6AiIhIDgaDAUaj0attxsbGIj093attUucYZIiIqM8xGAwYMnQomhobvdquPiQEh/PyGGZ6EYMMERH1OUajEU2NjZj5yItISB/glTYrDEV4b9FDMBqNDDK9iEGGiIj6rIT0AUjNHi53GdQDnOxLREREAYtBhoiIiAIWgwwREREFLAYZIiIiClgMMkRERBSwGGSIiIgoYDHIEBERUcBikCEiIqKAxSBDREREAYtBhoiIiAIWgwwREREFLAYZIiIiClgMMkRERBSwGGSIiIgoYDHIEBERUcBikCEiIqKAxSBDREREAYtBhoiIiAIWgwwREREFLAYZIiIiClgquQsgIqLgYzAYYDQavdpmbGws0tPTvdomBT4GGSIi8iqDwYAhQ4eiqbHRq+3qQ0JwOC+PYYbaYZAhIiKvMhqNaGpsxMxHXkRC+gCvtFlhKMJ7ix6C0WhkkKF2GGSIiMgnEtIHIDV7uNxlUJDjZF8iIiIKWLIGGafTicceewxZWVnQ6/UYMGAAnn76aQgh2o4RQuDxxx9HUlIS9Ho9cnNzUVBQIGPVRERE5C9kDTKLFi3CG2+8gddeew15eXlYtGgRXnjhBbz66qttx7zwwgt45ZVXsGzZMmzduhWhoaGYOnUqrFarjJUTERGRP5B1jszmzZsxffp0TJs2DQCQmZmJDz74ANu2bQPg7o1ZsmQJ/vrXv2L69OkAgJUrVyIhIQGfffYZbrnlFtlqJyIiIvnJ2iNz/vnnY/369cjPzwcA7N27F5s2bcKVV14JACguLkZ5eTlyc3PbnhMZGYnJkydjy5YtHbZps9lgMpna3YiIiCg4ydoj85e//AUmkwlDhgyBUqmE0+nEs88+i5kzZwIAysvLAQAJCQntnpeQkND22OkWLlyIJ5980reFExERkV+QtUfm448/xnvvvYf3338fu3btwooVK/C3v/0NK1as8LjNBQsWoL6+vu124sQJL1ZMRERE/kTWHpmHHnoIf/nLX9rmuowcORLHjx/HwoULMWfOHCQmJgIAKioqkJSU1Pa8iooKjBkzpsM2tVottFqtz2snIiIi+cnaI9PY2AiFon0JSqUSLpcLAJCVlYXExESsX7++7XGTyYStW7ciJyenV2slIiIi/yNrj8w111yDZ599Funp6Rg+fDh2796NxYsX44477gAASJKE+++/H8888wyys7ORlZWFxx57DMnJyZgxY4acpRMRkZ8SQsDpElApueZrXyBrkHn11Vfx2GOP4Q9/+AMqKyuRnJyMu+++G48//njbMQ8//DAsFgvuuusu1NXV4cILL8TatWuh0+lkrJyIiPzJoVITlv9UjHV5FTBZHXC6BIYmReDG8am4bmwK+oVq5C6RfETWIBMeHo4lS5ZgyZIlZz1GkiQ89dRTeOqpp3qvMCIiCggmazMe+b99WHPgzCtZ88pMeOqrQ3jluwIsmzUe5/WPkaFC8jVuGklERAEpv6IBd/9rJ4qNFigVEq4ckYjfnZeBjJhQAMB/D5XjX1uOo6DSjN+9vRXPzhiJmyamyVw1eRuDDBERBZwDJfW45a2fYbY5kBKlxxuzxmFUalS7Y2bnZOKmCWn486q9+HpfGR7+ZB+0agWmj0mRp2jyCc6EIiKigGKobsRty7fDbHNgYmY/fHHvBWeEmFY6tRKv3jIWd1yQBQD4n9X7UWy09GK15GsMMkREFDAabC7MWb4NRrMNQ5Mi8M5tExET1vnaYQqFhP+5aggmZUXDYnfi3vd3we4UvVQx+RqDDBERBYy/76hDsdGClCg93r19IsJ16i49T6VU4JVbxiI6VIODpSasOtTg40qptzDIEBFRQAgdmYutJTaolRLe/N14JER0bxmOxEgdnrtuBADgy3wLlGHRviiTehmDDBER+T2zA4iechcAYP6vB2NESqRH7UwdnojxGf1gdwKR59/izRJJJgwyRETk14QQ2F2jgkIbgmGxGtx1UX+P25IkCY9cMQQAEDZ6KhqavVUlyYVBhoiI/NpRowWVVgWEw457J0VCqZB61N6krGiMS9JCUiiRV6/0UpUkFwYZIiLyWw6nCz/kVwEATNtWIzHMO8uf3ToiHABwslEBs83hlTZJHgwyRETkt3adqIPJ6oBOKVD/8/95rd3+/dSwnjwIAQkHS+q91i71PgYZIiLyS03NTuw4VgMAGBnlhGi2erV98+41AIADpSa4XFxXJlAxyBARkV/adbwWzU6BuDAt0kJcXm/fcmQTNAoBs82B4mqu9huoGGSIiMjvNNmd2HuyDgAwuX80pJ7N7+2Y04HMUHdA2n+Sw0uBikGGiIj8zi5DS29MuBb9Y0N99jpZ4U4AwPGaRjRYeS12IGKQISIiv9LU/EtvzHlZ0ZB80h3jFqYCkqPcKwQXVJp99jrkOwwyRETkV/aX1LfNjcnyYW9Mq+x496XYhQwyAYlBhoiI/IbTJbDvRB0AYFx6lE97Y1oNjA8DAJTVWzm8FIAYZIiIyG/kVzTAYnciVKtEdkJ4r7xmmFaF5Ej38BJ7ZQIPgwwREfkFIQR2G+oAAKNTo3q8FUF3tIYmzpMJPN5Z65mIiKiHSuqaUGW2QaWQMPIsu1vn5eV55bVOb2dAXCi+z69CWb0VZqsDYTr+egwUfKeIiMgv7GtZy2VoUgR06vabOZpq3PstzZo1y6uvaTa7e2DCdWokRepQVm/FUaMZo1KjvPo65DsMMkREJDuLzYGiKneo6Kg3pslsAgBMu/tRDB41vsevl7fte6xZsRRW6y/bHmTGhqKs3orj1Y0MMgGEQYaIiGSXV2aCSwCJETrEhWvPelxMcgZSs4f3+PUqDEVn3JcRHYItRdU4WdsEp0v06hwd8hwn+xIRkayEEDhQ6u5xGZESIVsd8eFa6NVK2J0ulNU3yVYHdQ+DDBERyepEbRPqm5qhUSowqJcuue6IJElIjw4BAByvbpStDuoeBhkiIpLVgRL3JN8hSeFQK+X9tZQR4w4yhhoGmUDBIENERLKxNjtxtMoCABieLN+wUqvWHpnKBhsa7Q6Zq6GuYJAhIiLZFFSY4RQCMWEaxIWdfZJvbwnVqtrqMHB4KSAwyBARkWzyyt2TfIclRvTKvkpdkc7hpYDCIENERLKoa7SjrN4KCcDgRPkm+Z4urZ8egHulYfJ/DDJERCSLvLIGAO4ekFCt/yxrlhSphwTAZHVwN+wAwCBDRES9TgiBwy3DSkMT5Z/keyqNStG2KB97ZfwfgwwREfW6CpMNJqsDaqWE/nGhcpdzhpQo9/BSaZ31HEeS3BhkiIio1+VXuoeVsmJDZV87piMpnCcTMPzvu4eIiIKaEAIFFe4NIuVcybczyZHuIFNjsaPJ7pS5GuoMgwwREfWqcpMVZpt7WCmjZQE6f6PXKBEdqgEAlHLfJb/GIENERL2qtTemf1wYVH44rNSqdZ4Mh5f8m/9+BxERUdARQqCgsmVYKT5M5mo6lxylAwCU1DLI+DMGGSIi6jVl9e5hJY1S0bavkb9KbumRMZptcDhdMldDZ8MgQ0REvaa1N6Z/XKhfDysBQLhWhRCNEi7h3kSS/JN/fxcREVHQcA8ruS+7zvbzYSUAkCQJiRHu4aVyE9eT8VcMMkRE1CtK662w2JzuYaUY/x5WapUQ6Q4yFfUMMv6KQYaIiHpFQYW7N2ZAXChUisD49cMeGf8XGN9JREQU0FxCoLBlfky2ny6C15GECPeeSyarA412h8zVUEcYZIiIyOfK6qyw2J3QqPz/aqVTaVVKRIe4F8Zjr4x/YpAhIiKfa53kOyAuFEqFJHM13ZMQ6e6VqajnlUv+iEGGiIh8SgiBo0YLAGBgnP9frXQ6zpPxbwwyRETkU1VmGxqsDqgUUkANK7U6NcgIIWSuhk7HIENERD5VVOXujcmICfH7RfA6EhOmhVIhwe5woa6xWe5y6DSB9x1FREQB5WjVL5tEBiKlQkJcmHueDFf49T8MMkRE5DP1Tc0wmu2QAGTFhspdjsfiwt1BpopBxu8wyBARkc+09sakROmhVytlrsZz8eGtPTKc8OtvGGSIiMhnWufH9I8L3N4Y4NQgY+OEXz/DIENERD5hcwKldU0AgAEBOj+mVXSYBgoJsDlcaLByhV9/wiBDREQ+Ud6kgAAQG6ZBhF4tdzk9olIoEBPKCb/+iEGGiIh8orTJ/Ssm0HtjWnHCr39ikCEiIq+TVFpUWN1bEQRLkOGEX//EIENERF6nyxwNp5AQrlMhNkwjdzlewR4Z/8QgQ0REXheSnQMAGBAbBkkKrE0iz6Y1yFjsTlhsnPDrLxhkiIjIq5wuAf3ASQAC/7LrU6mVCvQLcU9a5oRf/8EgQ0REXnW42g5lSCTUCoGUKL3c5XgVh5f8D4MMERF51bYS9y/5JL0LCkVwDCu1at1zqdrMIOMvGGSIiMhrhBDYVuK+qidZ75K5Gu+LbQkyRrNd5kqoFYMMERF5zZGKBlRYnHA125CgC76l/GNbhpZqG+1wOIMvqAUiBhkiIvKa/x6sAABYj+2BKgh/w4RqlNCp3SsW11jYK+MPgvDbjIiI5PLtIXeQaSz4WeZKfEOSJA4v+RkGGSIi8orSuibsL6mHQgKaCrfKXY7P/BJkOOHXHzDIEBGRV/z3YDkAYHCMGq4mk8zV+E7rSsUMMv5B9iBTUlKCWbNmISYmBnq9HiNHjsSOHTvaHhdC4PHHH0dSUhL0ej1yc3NRUFAgY8VERNSRb1rmx5yXopO5Et86dWhJiOCb0BxoZA0ytbW1uOCCC6BWq7FmzRocOnQIL730Evr169d2zAsvvIBXXnkFy5Ytw9atWxEaGoqpU6fCauWmXURE/qLGYsfW4moAwOTU4A4yMaEaSACamp1otDvlLqfPU8n54osWLUJaWhqWL1/edl9WVlbb34UQWLJkCf76179i+vTpAICVK1ciISEBn332GW655ZZer5mIiM607lAFXAIYnhyB+FBZf7X4nEqpQFSIGrWNzTCabQjVBvf5+jtZe2S++OILTJgwATfeeCPi4+MxduxY/OMf/2h7vLi4GOXl5cjNzW27LzIyEpMnT8aWLVs6bNNms8FkMrW7ERGRb61tmR9zxfBEmSvpHbxyyX/IGmSOHj2KN954A9nZ2fjmm29wzz334E9/+hNWrFgBACgvd38wEhIS2j0vISGh7bHTLVy4EJGRkW23tLQ0354EEVEfZ7Y5sKnACACYOqKvBRlO+JWbrEHG5XJh3LhxeO655zB27FjcdddduPPOO7Fs2TKP21ywYAHq6+vbbidOnPBixUREdLoNhythd7rQPzYU2fFhcpfTK2Jarlyq5qJ4spM1yCQlJWHYsGHt7hs6dCgMBgMAIDHRnewrKiraHVNRUdH22Om0Wi0iIiLa3YiIyHdah5WmjkiEJAXXJpFnExPqDjI1FjtcvHJJVrIGmQsuuABHjhxpd19+fj4yMjIAuCf+JiYmYv369W2Pm0wmbN26FTk5Ob1aKxERncna7MTGw5UAgKl9ZH4MAETo1VAqJDhdAqamZrnL6dNkDTIPPPAAfv75Zzz33HMoLCzE+++/j7feegvz5s0D4F4K+v7778czzzyDL774Avv378fs2bORnJyMGTNmyFk6EREB+KnQCIvdiaRIHUalRMpdTq9RSBKiQ37plSH5yHrN2MSJE/Hpp59iwYIFeOqpp5CVlYUlS5Zg5syZbcc8/PDDsFgsuOuuu1BXV4cLL7wQa9euhU4X3OsUEBEFgrUHWoaVhidCoegbw0qtosM0qDLbUG2xo3+c3NX0XbJf/H711Vfj6quvPuvjkiThqaeewlNPPdWLVRER0bk4nC6sy3PPYexLw0qtWufJcMKvvGTfooCIiALTtmM1qG1sRr8QNSZm9jv3E4JM24RfriUjKwYZIiLyyDctw0q/HpYAlbLv/TqJbg0yjbxySU597zuPiIh6zOUSv1x23QeHlQBeueQvGGSIiKjbth+rQYXJhnCdChdmx8pdjiwUktTWK8N5MvJhkCEiom77cl8pAPfeSlqVUuZq5MMJv/JjkCEiom5xOF1Ys989rHT16GSZq5FXNCf8yo5BhoiIumXL0WpUW+yIDtXg/AExcpcjq1O3KiB5MMgQEVG3fLnXPax05YhEqPvg1UqnOvXKJV64JI++/R1IRETdYne42lbzvXpU3x5WAoBIvRqqliuXzA65q+mbGGSIiKjLvjtcCZPVgfhwLSZlRctdjuykU65cMjX3rS0a/AWDDBERddnqXScBANeNTYGyj+2tdDatQaaBQUYWDDJERNQlNRY7NhypBABcPy5V5mr8B3tk5MUgQ0REXfLVvlI0OwWGJ0dgcGK43OX4jRgGGVkxyBARUZd8sqsEAHtjThcTpgXQMrQk8ddqb+O/OBERnVNhpRl7T9RBqZBwbR9fBO90EToVVAoJLkhQRfXNfafkxCBDRETn9PGOEwCAiwfFIS5cK3M1/uXUK5fUsekyV9P3eBRkjh496u06iIjIT9kcTqxqCTK/ncRf1B1hkJGPR0Fm4MCBuPTSS/Hvf/8bVqvV2zUREZEfWXugHLWNzUiK1OHSwXFyl+OXWif8ahhkep1HQWbXrl0YNWoU5s+fj8TERNx9993Ytm2bt2sjIiI/8N5WAwDg5olpUPXxLQnOJjqstUcmQ+ZK+h6PviPHjBmDpUuXorS0FO+88w7Kyspw4YUXYsSIEVi8eDGqqqq8XScREcmgoKIB24proFRIuGUiexvOJibUPW9IHZ0Kp4ubLvWmHkVrlUqF66+/HqtWrcKiRYtQWFiIBx98EGlpaZg9ezbKysq8VScREcmgtTfmsiHxSIzUyVyN/4rQqaCUBCSVGuVmp9zl9Ck9CjI7duzAH/7wByQlJWHx4sV48MEHUVRUhG+//RalpaWYPn26t+okIqJeVt/Y3Ha10uwcDpl0RpIkhKvdPTEnG7h7ZG9SefKkxYsXY/ny5Thy5AiuuuoqrFy5EldddRUUCncuysrKwrvvvovMzExv1kpERL3o31uPo9HuxNCkCFw4MFbucvxehEqgzg6cNDHI9CaPgswbb7yBO+64A7fddhuSkpI6PCY+Ph5vv/12j4ojIiJ52BxOvLv5GADgrouyIElcfv9c2npkGGR6lUdBpqCg4JzHaDQazJkzx5PmiYhIZp/vLkVVgw1JkTpcPYor+XYFg4w8PJojs3z5cqxateqM+1etWoUVK1b0uCgiIpKP0yXw5g9FAIA7LsiCmpdcd0lES5ApaXDAxSuXeo1H350LFy5EbOyZ46Xx8fF47rnnelwUERHJ59PdJSiqsiAqRI1bJqXJXU7ACFUBwtkMq0OgtL5J7nL6DI+CjMFgQFZW1hn3Z2RkwGAw9LgoIiKSh83hxMvf5gMA7rl4AMJ1apkrChwKCWiuLQXg3mSTeodHQSY+Ph779u074/69e/ciJiamx0UREZE8Ptp+AiV1TYgP12J2Tqbc5QScZqP7cnUGmd7jUZD57W9/iz/96U/YsGEDnE4nnE4nvvvuO9x333245ZZbvF0jERH1AovNgVfWFwIA/jglG3qNUuaKAk9zNYNMb/PoqqWnn34ax44dw5QpU6BSuZtwuVyYPXs258gQEQWol7/Nh9FsQ3p0CG6ewLkxnmCQ6X0eBRmNRoOPPvoITz/9NPbu3Qu9Xo+RI0ciI4MrPxIRBaIDJfVY3rJuzJPTh0Oj4pVKnmgLMlVmCCG4/k4v8CjItBo0aBAGDRrkrVqIiIKSwWCA0Wj0WnuxsbFIT/feBo5Ol8Cjn+6H0yUwbVQSLh0c77W2+xpHTQkkAHWNzai22BEbppW7pKDnUZBxOp149913sX79elRWVsLlcrV7/LvvvvNKcUREgc5gMGDI0KFoamz0Wpv6kBAczsvzWpj5x49HsfdkPcK1Kjxx9TCvtNlXCYcd8aFKVFicKKgwM8j0Ao+CzH333Yd3330X06ZNw4gRI9h1RkR0FkajEU2NjZj5yItISB/Q4/YqDEV4b9FDMBqNXgky24pr8OI3RwAAj04bivgI7nDdU6kRKlRYnCisMiNnAK/k9TWPgsyHH36Ijz/+GFdddZW36yEiCkoJ6QOQmj1c7jLaqWqw4d73d8HpErhubApunsgJvt6QFqHCzjIbijjht1d4NJtLo9Fg4MCB3q6FiIh6idnmwF3/2oHKBhuy48Pw7HXsXfeWlAh3H0FBZYPMlfQNHgWZP//5z1i6dCmE4F4SRESBpsnuxB3vbsduQx0i9Wq8MWscQjQ9uvaDTpHaEmR4CXbv8Og7d9OmTdiwYQPWrFmD4cOHQ61uv4T16tWrvVIcERF5V63Fjnve24ltxTUI16qw8o5JGBgfLndZQSU13P2rtcJkg8najAhu8+BTHgWZqKgoXHfddd6uhYiIfOhAST3u/tdOlNQ1IUSjxLt3TMTotCi5ywo6oRoFEiK0qDDZUFhpxrj0fnKXFNQ8CjLLly/3dh1EROQjFpsDr28oxD9/LIbd6UJGTAje/N14DEmMkLu0oDUwPoxBppd4PCjqcDiwceNGFBUV4dZbb0V4eDhKS0sRERGBsLAwb9ZIREQeqLHY8fGOE3hnUzEqG2wAgNyhCXjpptGI1HO4w5cGxoXhp8JqXrnUCzwKMsePH8cVV1wBg8EAm82GX//61wgPD8eiRYtgs9mwbNkyb9dJRERdUGmyYuORKqzLq8DGI1WwO90LlmbEhOCv04Yhd2g8r07qBQMT3POOOOHX9zxeEG/ChAnYu3cvYmJ+Weznuuuuw5133um14oiIqHMnahqxrbjGfTtWg2Kjpd3jI1Mi8bvzMjB9bDK0Ku5m3VsGxrlHJgoYZHzOoyDz448/YvPmzdBoNO3uz8zMRElJiVcKIyKiMwkBaJKH4KX1xTj6n0qUmZ3tHpcA9O+nxoRkLSYl65DVTw2gCgf3VXXarrf3b+rrBsa7g8yJ2kZYm53QqRkifcWjIONyueB0Os+4/+TJkwgP52V8RETe5nC5cLDUhO0lEpJ+9zf8VA0ATginA/byQlhPHID1xAHYSvJwzGZBd3e88/b+TX1dbJgGUSFq1DU242iVBcOSObHaVzwKMpdffjmWLFmCt956CwAgSRLMZjOeeOIJbltARORFQgjkV5ixucgIk9UBQA1XsxX9RAOGJEchQSegzuoP5PQHcK1Hr+Ht/ZvI/XtxYFwYdhyvRUFlA4OMD3kUZF566SVMnToVw4YNg9Vqxa233oqCggLExsbigw8+8HaNRER9krXZiXV5FSiqcs97CdEokWgvwcbX78Ol//MyxozmTtX+bGC8O8jwyiXf8ijIpKamYu/evfjwww+xb98+mM1mzJ07FzNnzoRer/d2jUREfU55vRVf7y+D2eaAQgImZkZjfEY/7Nt4AMJmOXcDHsjLy/OrdgJd6zyZwioGGV/yeB0ZlUqFWbNmebMWIiKC+0qkL/eVotkpEKVX48oRiYiP0Pns9Uw17onA3v6Zbjb37V/grUGmoKJv/zv4mkdBZuXKlZ0+Pnv2bI+KISLq64qNFny9vwxOl0BatB5Xj0yGRuXR/r5d1mQ2AQCm3f0oBo8a3+P28rZ9jzUrlsJqtfa4rUDWGmSOVVvgcLqgUvr2feyrPF5H5lTNzc1obGyERqNBSEgIgwwRkQfKTda2ENM/NhRXjkjs1V9+MckZSM0e3uN2KgxFXqgm8CVH6qFXK9HU7MTxmkYMiOOq977g0Sektra23c1sNuPIkSO48MILOdmXiMgDDdZmfLm3FE6XQGZMCK4amcT/wQc4hUL6ZZ4MJ/z6jNc+JdnZ2Xj++efP6K0hIqLOOZwufLm3DI12J2JCNbhiRCKUCm4jEAwYZHzPq3FfpVKhtLTUm00SEQW9n4qqUWW2Qa9W4trR3EogmDDI+J5Hc2S++OKLdl8LIVBWVobXXnsNF1xwgVcKIyLqCww1jdhzog4A8OthCYjgrtRBhUHG9zwKMjNmzGj3tSRJiIuLw2WXXYaXXnrJG3UREQU9W7MT3x6qAODe3DErNlTmisjbTg0yLpeAgkOGXufxXktERNQzm4uqYbY5EKlX41fZsXKXQz6QER0CtVJCU7MTpfVNSO0XIndJQYdT4omIZFDVYMP+knoAQO7QeKh5hVJQUikVyIxx97RxeMk3POqRmT9/fpePXbx4sScvQUQUtIQQ2JhfCQEgOz6M/0sPctkJYSioNKOw0oxLBsfLXU7Q8SjI7N69G7t370ZzczMGDx4MAMjPz4dSqcS4cePajpMkjgUSEZ0uv8KM0jorVAqJQ0p9wMA4Tvj1JY+CzDXXXIPw8HCsWLEC/fr1A+BeJO/222/Hr371K/z5z3/2apFERMHC6RLYXGQE4N4IMlzHq5SC3QBeueRTHg3KvvTSS1i4cGFbiAGAfv364ZlnnuFVS0REnThUZoLJ6kCIRomx6VFyl0O9IDs+HABQUGmGEELmaoKPR0HGZDKhqqrqjPurqqrQ0NDQ46KIiIKRw+nCtuIaAO7eGE7w7Rv6x4VCkoD6pmYYzXa5ywk6Hn2KrrvuOtx+++1YvXo1Tp48iZMnT+KTTz7B3Llzcf3113u7RiKioHCg1ASzzYEwrQojkiPkLod6iU6tRFrLhG4OL3mfR3Nkli1bhgcffBC33normpub3Q2pVJg7dy5efPFFrxZIRBQMnALYcay1N6YfN4TsYwbGh8FQ04jCKjNyBsTIXU5Q8SjIhISE4O9//ztefPFFFBW5t2sfMGAAQkO5KiURUUdOWBSw2J0I06owPDlS7nKol2XHh+G7w5UorOD0C2/r0X8JysrKUFZWhuzsbISGhnISExFRhyTkm9wbQY5Ni+LO1n1Q25VLVRxa8jaPgkx1dTWmTJmCQYMG4aqrrkJZWRkAYO7cubz0mojoNPoBE9DgkKBRKjA8hXNj+iJuHuk7HgWZBx54AGq1GgaDASEhv6xIefPNN2Pt2rUeFfL8889DkiTcf//9bfdZrVbMmzcPMTExCAsLww033ICKigqP2icikkvEJPdFECNTIqFVKWWuhuTQGmQqTDaYrM0yVxNcPAoy//3vf7Fo0SKkpqa2uz87OxvHjx/vdnvbt2/Hm2++iVGjRrW7/4EHHsCXX36JVatW4fvvv0dpaSmviiKigFJU0wxd+khIEBiTFiV3OSSTCJ0aCRFaAOyV8TaPgozFYmnXE9OqpqYGWq22W22ZzWbMnDkT//jHP9otsFdfX4+3334bixcvxmWXXYbx48dj+fLl2Lx5M37++WdPyiYi6nVrCi0AgNQQF8J0Hl1fQUGCw0u+4VGQ+dWvfoWVK1e2fS1JElwuF1544QVceuml3Wpr3rx5mDZtGnJzc9vdv3PnTjQ3N7e7f8iQIUhPT8eWLVvO2p7NZoPJZGp3IyKSQ12jHZtONAEABoS7ZK6G5MY9l3zDo/8evPDCC5gyZQp27NgBu92Ohx9+GAcPHkRNTQ1++umnLrfz4YcfYteuXdi+ffsZj5WXl0Oj0SAqKqrd/QkJCSgvLz9rmwsXLsSTTz7Z5RqIiHxl1Y6TsDsBe0URotPS5C6HZDYwwb1VAYOMd3nUIzNixAjk5+fjwgsvxPTp02GxWHD99ddj9+7dGDBgQJfaOHHiBO677z6899570Ol0npTRoQULFqC+vr7tduLECa+1TUTUVS6XwL+3uucMNuz+DyRecd3nsUfGN7rdI9Pc3IwrrrgCy5Ytw6OPPurxC+/cuROVlZUYN25c231OpxM//PADXnvtNXzzzTew2+2oq6tr1ytTUVGBxMTEs7ar1Wq7PU+HiMjbfiw04nh1I0LUEgyHNgK4W+6SSGatc2RO1DbC2uyETs0r2Lyh2z0yarUa+/bt6/ELT5kyBfv378eePXvabhMmTMDMmTPb/q5Wq7F+/fq25xw5cgQGgwE5OTk9fn0iIl/6cJsBAHBpph6i2SZzNeQPYsM0iApRQwigiAvjeY1Hc2RmzZqFt99+G88//7zHLxweHo4RI0a0uy80NBQxMTFt98+dOxfz589HdHQ0IiIi8Mc//hE5OTk477zzPH5dIiJfqzbbsC7PveZVblYI/i5zPeQfJElCdnwYth+rRUGFmVtVeIlHQcbhcOCdd97BunXrMH78+DP2WFq8eLFXinv55ZehUChwww03wGazYerUqfj73/kjgYj826e7S9DsFBidGomMKLXc5ZAfGZQQju3HanGEey55TbeCzNGjR5GZmYkDBw60zW3Jz89vd4zUgxltGzdubPe1TqfD66+/jtdff93jNomIepMQAh9td19kcOOENADV8hZEfmVIovvKpSPlDDLe0q0gk52djbKyMmzYsAGAe0uCV155BQkJCT4pjogo0Ow5UYeCSjO0KgWuHZOMwkMMMvSLwYnuvbYYZLynW5N9T9/des2aNbBYLF4tiIgokH284yQA4KqRSYjQcViJ2hvcspZMSV0TGrjnkld4tI5Mq9ODDRFRX2ZtduKrfaUAgBvHp57jaOqLIkPUSIxwr52Wz3kyXtGtICNJ0hlzYHoyJ4aIKJisz6tEg9WB5EgdzusfI3c55KcGt8yTOczhJa/o1hwZIQRuu+22tgXnrFYrfv/7359x1dLq1au9VyERUYD4dLd7WGn62BQoFPxPHnVscGI4vs+vQj6DjFd0K8jMmTOn3dezZs3yajFERIGq2mzDxiNVAIDrx6bIXA35s9Z5MuyR8Y5uBZnly5f7qg4iooD21b4yOFwCI1IikN3yi4qoI61DS0cqGiCE4BSNHurRZF8iInJbvbsEAHDdWE7ypc4NjA+DQgLqGptR2cDtK3qKQYaIqIeOV1uw90QdFBJw7ehkucshP6dTK5EZ655byvVkeo5Bhoioh77eXwYAOH9ALOLCtTJXQ4GAK/x6D4MMEVEPfbXXHWSmjUqSuRIKFIMT3Cv8csJvzzHIEBH1wNEqMw6VmaBUSJg6PFHucihADE4MA8BF8byBQYaIqAf+0zKsdMHAWESHamSuhgJF655L+RUNcLq4Sn5PMMgQEfXAV/vcQebqkRxWoq5Ljw6BTq2AzeHC8WruWdgTDDJERB4qrDTjcHkDVAoJlw9PkLscCiBKhYRBLesNcXipZxhkiIg89HVLb8yF2bGICuGwEnXPIK7w6xUMMkREHvp6v3un66tHce0Y6j5egu0dDDJERB7Ir2hAfoUZaqWEXw/jsBJ136lbFZDnGGSIiDzQOsn3ouw4ROrVMldDgah188hjRguszU6ZqwlcDDJERN0khMDX+1qGlUbzaiXyTFy4Fv1C1HAJ98Rx8gyDDBFRNx2paEBRlQUalQK5QzmsRJ6RJOmX4SXOk/EYgwwRUTe1bklw8aA4hOs4rESeG9KyMB7nyXiOQYaIqBuEEG2bRF7NvZWoh3gJds8xyBARdcOhMhOKjRZoVQpM4bAS9dCQJHeQySszyVxJ4GKQISLqhtarlS4dHI8wrUrmaijQDUkMhyQBVQ02VDZY5S4nIDHIEBF1kftqJXeQmcZhJfKCEI0K/WNDAQCHStkr4wkGGSKiLjpQYoKhphE6tQKXDYmXuxwKEsOTIwG4hy2p+xhkiIi66KuWLQmmDElAKIeVyEuGJbuvXDrIHhmP8JNIRHQKg8EAo9F4xv1CCHy6owoAMDSsCbt27epSe3l5eV6tj4LP8JYgw6ElzzDIEBG1MBgMGDJ0KJoaG894TJM0CEmzF8Nlb8KffnMDhMPWrbbNZq7cSh0bluQOMseqLTDbHJxE3k381yIiamE0GtHU2IiZj7yIhPQB7R7bV6tEQQOQHqXBjUs/6HKbedu+x5oVS2G18ooU6lhMmBaJETqUm6w4XGbChMxouUsKKAwyRESnSUgfgNTs4W1fCyHwzU/HADgwekAqUuPDutxWhaHI+wVS0BmWHIFykxWHGGS6jZN9iYjOodxkhdnmgFopITMmRO5yKAi1zpM5WMJ5Mt3FIENEdA75Fe75Lf1jw6BS8scmeV9bkCmrl7mSwMNPJBFRJ4QQKKx0B5lBCV0fUiLqjmFJ7rVk8svNaHa6ZK4msDDIEBF1orTePaykUSqQHs1hJfKNtGg9InQq2J0u5HMn7G5hkCEi6kRByy+V/nGhHFYin5EkCSNT3b0y+09yeKk7+KkkIjoLV7thpXCZq6FgNzIlCgCwr4RBpjsYZIiIzqKszgqL3QmNisNK5Huj2CPjEQYZIqKzaJ2rMCAuFEqFJHM1FOxGpriDzOFyE2wOp8zVBA4uiEdE1AGXEChoHVaK57ASdZ2n+2sJIRCmkWC2C3y+cTsGRmsQGxuL9PR0L1cYXBhkiIg6UFLbhKZmJ7QqBdI4rERdYKpxbyo6a9Ysj9uIv+kp6LPGYe5DT8G8Zw30ISE4nJfHMNMJBhkiog60DisNjA/jsBJ1SZPZvSrvtLsfxeBR4z1q40CdEkdMwPjrf4/UqZfhvUUPwWg0Msh0gkGGiOg0LoFfhpV4tRJ1U0xyRru9urrDWmnGkf1lMEv6MzYupY5xsi8R0WkqrBJsDhdCNEqk9tPLXQ71IQkRWgBAtcUOLvDbNQwyRESnOWFx/2gcFB8OhcRhJeo9YVoV9GolhADqmvm91xUMMkREp5BUWpQ2tQSZRO6tRL1LkqS2XpkaG4NMVzDIEBGdQj9wIpxCQoROhcQIndzlUB+UGOn+vqu1M8h0BYMMEdEpQodcBMA9yVfisBLJICnSPS+r2sZf0V3BfyUiohYWuwv6ARMA8Golkk/r0FKjU4IiJEreYgIAgwwRUYutJVZIKg3C1S7EhmnkLof6KK1KiZhQ9/efNnmwzNX4PwYZIqIWmwxNAIC0EBeHlUhWrfNktMlDZK7E/zHIEBEBMJpt2FdpB+AOMkRy+iXIsEfmXBhkiIgArNlfBpcAbGX5CFPLXQ31dUktV8xpkrLhdAmZq/FvDDJERAC+2FsKALDk/SBzJURAdKgGKklAodHDUO+Quxy/xiBDRH3e8WoLth+rhUICGvN+lLscIkiShGituycmv9ouczX+jUGGiPq8T3aeBACMTtDCaa6WuRoit2iNO8gcrm6WuRL/xiBDRH2ayyXwya4SAMClmdwgkvxHrNY96TzPyB6ZzjDIEFGf9vPRapTUNSFcp8KkFG5JQP4jWisgXE5UWpwoq2+Suxy/xSBDRH3a/7UMK107OhkaJdeOIf+hVgD2iqMAgG3FNTJX478YZIioz2qwNuM/B8oAAL8ZnypzNURnsp08CIBBpjMMMkTUZ63ZXw5rswsD4kIxJi1K7nKIzmA94Q4y248xyJwNgwwR9Vmtw0q/GZ/GLQnIL7X2yORXmFFr4aTfjjDIEFGfdMxowbZjNVBIwHVjU+Quh6hDriYTUiNUANgrczYMMkTUJ32yy90b86vsuLZ9bYj80dBY907YDDIdY5Ahoj7H5RJti+Bxki/5u2Fx7s2/OOG3YwwyRNTnbDlajdJ6K8J1Kvx6WILc5RB1anicFgCwv6QeJitX+T0dgwwR9TnvbzMAcK8do1MrZa6GqHOxIUpkxYbCJYCtR9krczoGGSLqU6oabPjmQDkAYObkDJmrIeqaCwbGAAB+KjTKXIn/kTXILFy4EBMnTkR4eDji4+MxY8YMHDlypN0xVqsV8+bNQ0xMDMLCwnDDDTegoqJCpoqJKNB9vOMEHC6BcelRGJYcIXc5RF1ywYBYAAwyHZE1yHz//feYN28efv75Z3z77bdobm7G5ZdfDovF0nbMAw88gC+//BKrVq3C999/j9LSUlx//fUyVk1EgcrpEnh/q3tYib0xFEhyBsRAkoCCSjMqTVa5y/ErKjlffO3ate2+fvfddxEfH4+dO3fioosuQn19Pd5++228//77uOyyywAAy5cvx9ChQ/Hzzz/jvPPOk6NsIgpQP+RXoaSuCZF6NaaNSpK7HKIuiwrRYHhyBA6UmLC5qBozuPZRG7+aI1NfXw8AiI6OBgDs3LkTzc3NyM3NbTtmyJAhSE9Px5YtWzpsw2azwWQytbsREQHAv34+DsB9yTUn+VKguWCge3hpE4eX2vGbIONyuXD//ffjggsuwIgRIwAA5eXl0Gg0iIqKandsQkICysvLO2xn4cKFiIyMbLulpaX5unQiCgDFRgu+O1wJSQJmncdhJQo8rfNkNhcaIYSQuRr/4TdBZt68eThw4AA+/PDDHrWzYMEC1NfXt91OnDjhpQqJKJCt2HwMAHDp4HhkxYbKWwyRByZmRkOjVKC03oqjRsu5n9BH+EWQuffee/HVV19hw4YNSE39ZZXNxMRE2O121NXVtTu+oqICiYmJHbal1WoRERHR7kZEfZvJ2oxVO9z/qbn9gkx5iyHykF6jxKQs99SLDYcrZa7Gf8gaZIQQuPfee/Hpp5/iu+++Q1ZWVrvHx48fD7VajfXr17fdd+TIERgMBuTk5PR2uUQUoFbtOAmL3Yns+DBc2DLPgCgQXTI4DgCw8UiVzJX4D1mvWpo3bx7ef/99fP755wgPD2+b9xIZGQm9Xo/IyEjMnTsX8+fPR3R0NCIiIvDHP/4ROTk5vGKJiLrE4XS1DSvddkEmJEmStyCiHrhsSDye+ToPW4urYbY5EKaV9de4X5C1R+aNN95AfX09LrnkEiQlJbXdPvroo7ZjXn75ZVx99dW44YYbcNFFFyExMRGrV6+WsWoiCiRrDpTDUNOIfiFqXMdLVinAZcWGIiMmBM1OwcXxWsga5boy61qn0+H111/H66+/3gsVEVEwEUJg2fdFAIA552ciRMP/vVJgkyQJlw6Ox7ubj2HD4UpMHd7xfNG+xC8m+xIR+cKmQiMOlpqgVysxJydT7nKIvOLSIfEAgA1HKnkZNhhkiCiItfbG3DwxDf1CNTJXQ+Qdk7OioVcrUWGyIa+sQe5yZMcgQ0RBac+JOvxUWA2lQsL/+1XWuZ9AFCB0amXbbtjfHuImygwyRBSUlq7LBwDMGJOC1H4hMldD5F2Xt8yNWXuw41Xu+xIGGSIKOrsNtdhwpApKhYQ/XjZQ7nKIvO7XQxOgVEjIKzPheHXfXuWXQYaIgs6SdQUAgOvGpiCT2xFQEOoXqsF5/d2r/K490Ld7ZRhkiCio7DLU4vt89sZQ8LuCw0sAGGSIKIgIIbBozWEAwPVjU5ARw94YCl5ThydCkoDdhjqU1TfJXY5suDoUEQU0g8EAo9G9wumOUiu2FtdCrQByE+3YtWtXt9rKy8vzRYlEPhEfocO49H7YebwW/z1YgTnnZ8pdkiwYZIgoYBkMBgwZOhRNjY2ApEDS7a9CE5cB45ZPcMXC5R63azabvVglke9cOSIRO4/X4qt9pQwyRESBxmg0oqmxETMfeRFN0dnYWaOCWiEw64ZroLnxmm63l7fte6xZsRRWq9UH1RJ537RRSXj2P3nYfqwWJ2oakRbd95YaYJAhooAXnToA6yq1AJw4r38c+mf086idCkORdwsj8rGkSD1y+sdgc1E1vthbinmX9r0J7pzsS0QBL69eiUa7E1F6NUalRcpdDlGvmtGyq/vqXSf75N5LDDJEFNBUMakobHD/KLt4UBxUCv5Yo77lihGJ0KoUKKqy4GCpSe5yeh0/8UQUsIQQiM79PQQk9I8N5eJ31CdF6NTIHZYAAPh0d4nM1fQ+BhkiClgbjzdBnzkGCghcNChO7nKIZHPdGPfw0ud7StHsdMlcTe9ikCGigFTVYMPyPe5u9KGRTkTq1TJXRCSfiwfHITZMC6PZhnV9bEdsBhkiCkj/++VBmO0C9ooiDIroW/8DJTqdWqnATRNSAQDvbzPIXE3vYpAhooCzZn8Zvt5XBoUEVK95BQpJ7oqI5PfbSemQJODHAiMM1Y1yl9NrGGSIKKCU11ux4NP9AIAZg0Nhr+DaL0QAkBYdgl9lu+eKfbC97/TKMMgQUcBwuQQeXLUXdY3NGJESgZuHh8tdEpFfuXVSOgBg1Y4TsDv6xpArgwwRBYy3NxVjU6EROrUCS24eC7WSY0pEp5oyNB7x4VoYzXb8Z3+Z3OX0CgYZIgoI24/V4Pm1hwEAf502DAPjw2SuiMj/qJUKzM7JAAC89cPRPrHSL4MMEfm9qgYb5r23C06XwLWjkzFzcrrcJRH5rZmTM6BTK3CozIQtRdVyl+NzDDJE5NeanS788YNdqGywYWB8GBZePxKSxCElorPpF6rBjePTAAD/+PGozNX4HoMMEfktIQQe//wgfj5ag1CNEm/MHIdQrUrusoj83twLsyBJwIYjVSioaJC7HJ9ikCEiv/Xu5mP4YJsBkgS88tuxyE7gVUpEXZEZG4rLW/Zf+vvG4F6igEGGiPzSNwfL8fRXhwAAC64cgilDE2SuiCiwzLt0IADg8z0lKKoyy1yN77CPloh6jcFggNFoPOdxeVV2PPlDNVwCyM3SY3xoHXbt2nXmcXl5viiTKCiMSo1C7tB4rMurxKvrC7DklrFyl+QTDDJE1CsMBgOGDB2KpsbOl05Xx2YgYeYiKHVhaCzchrdfeAZvi84X9jKbg/d/m0Q9cX/uIKzLq8QXe0tx72XZQblsAYMMEfUKo9GIpsZGzHzkRSSkD+jwmIZm4PsKNWwuCdEaF6ZfMgaqy/7vrG3mbfsea1YshdVq9VXZRAFtREokcocmYF1eBZasy8drt46TuySvY5Ahol6VkD4AqdnDz7i/vqkZ3+w8CZvLgbgwLa4flwKdWtlpWxWG4J7ESOQN8389COsPV+CrfWW448JajEvvJ3dJXsUgQ0Syq7HY8enuEphtDkSHajBjbPI5QwxRX+GNuWCXZujx3bEmLPhoO966aTAyMjK8UJl/YJAhIllVNdjw6e4SNDU7ER2qwfVjUxCi4Y8mIlNNFQBg1qxZPW5LGRaN5DvfwpFqYMy1c7H3y3eQnh4cK2TzpwURyabcZMVnu0tgc7gQF67FjDHJDDFELZrMJgDAtLsfxeBR43vcXl69AofqgdDzb0VJeRWDDBFRT5TUNuGLvaWwO11IjNBhxphkaDmcRHSGmOSMDueVdVeC04XiHwvRFJmAjw81IGeSF4rzA1wQj4h63dEqMz7bUwK704XUKD2uG5vCEEPkY2qlAmP6OQAAnx+xIK/MJHNF3sEgQ0S9qqhBga/2lcHhEsiMCcH0McnQqPijiKg3JIcIWI78BJcAFqzeD6dLyF1Sj/GnBxH1CpcQiLpoDvbUqiAADE+OwNWjkqFS8scQUW+qXfcW9CoJe07U4Z1NxXKX02P8CUJEPmd3uPDK1jpE5twIADgvKxpThsRDqZBkroyo73Gaq3Hb6AgAwIvfHMHh8sAeYmKQISKfqrHYMevtrfjBYIVwOTE+2oHJ/WMgSQwxRHLJ7a/HlCHxsDtduP/DPbA5nHKX5DEGGSLymcPlJlz72iZsK66BXiWh8v+eRGZY5/smEZHvSZKE528YhZhQDQ6XN2Dhfw7LXZLHGGSIyCf+e7AcN/x9M07WNiE9OgQLp8TAWnzmDtZEJI+4cC1e+M0oAMC7m4/hq32lMlfkGQYZIvIqIQRe31CIu/+9Exa7Ezn9Y/D5vAuQHqmWuzQiOs2UoQn4/cXuTVwf+b99KKwMvJ3kGWSIyGuszU7c/9EevPjNEQgB/O68DKycOwn9QjVyl0ZEZ/Hg5YNwXv9oWOxO3P2vHahvapa7pG5hkCEiryivt+LmN7fg8z2lUCkkPD1jBJ6eMQJqXl5N5NdUSgVe+e1YJEXqUFRlwbz3dqHZGThz2fgThoh6bHOhEdNe+RF7T9YjKkSNlXMn4XfnBc/uukTBLj5ch3/OmYAQjRKbCo144ouDECIwFstjkCEij7lcAn/fWIhZb29FtcWOoUkR+HzeBTh/QKzcpRFRNw1PjsTSW8ZCkoD3txrwyvpCuUvqEm4aSUQeqW9qxp8/3ot1eRUAgN+MT8UzM0ZAxz2TiPxeXl5eh/fHAJg7JgL/3G3Cy+vyYa4ux5XZoZ22FRsbK+tO2gwyRNRtuw21uO/DPTDUNEKjVODJ6cNxy8Q0LnJH5OdMNVUAgFmzZnV6XOQFv0XUhTPx1q46LHzmSVgOrD/rsfqQEBzOy5MtzDDIEFGXOV0Cy74vwsvf5sPhEkiJ0uONWeMwKjVK7tKIqAuazO7tCKbd/SgGjxp/1uOEAPbWOlFkViJ22v2Y+rs/driYZYWhCO8teghGo5FBhoj8W3m9FQ98tAdbjlYDAK4elYRnrxuJSD3XhyEKNDHJGUjNHt7pMalCYGN+FfadrMfOGhUi4+IxMjWylyrsOgYZIjorg8GAqqoqbD5pxZs762G2C+hUEuaOjcBlmQJFefu73NbZxuSJyD9JkoRLBsVBArD3ZD2+O1IJs82B8/pH+9UwMoMMEXXIYDBg2PgchPzqdoQMygEA2MoLUfLFC3i41vOlzM3mwFs5lKivkiQJFw+Kg1atxLbiGmw7VgOzzYHL/Gj3egYZIjqDEAIfbzeg38zFUOrCIEFgSIQLQ9LSoZj0mkdt5m37HmtWLIXVavVytUTkS5IkIad/DMK0Kmw4XIlDZSZY7A5cNSJJ7tIAMMgQ0WkKKhrw1FeH8GNBPZS6MERpXLhqTCbiwrU9arfCUOSlColIDiNTIhGqUWLNgXIcr27EJ7tOYny43FVxQTwialFrseOJzw/giqU/4scCI9QKoHbDclya4OhxiCGi4NA/LgzXj0uBTq1AZYMN35WroU0ZJmtNDDJEfZzd4cI7m4px8YsbsGLLcThdAlOHJ2DJ1DiYtn0CPxkGJyI/kRSpxy0T0xEbpoHNJSHht8/h26ONstXDoSWiPsra7MSH2wx464ejKK13z1sZmhSBx64eivMHxGLXrl0yV0hE/ipSr8ZNE9LwxbZ8nGxUIUonX78IgwxRH9Ngbca/fzbg7U1HYTTbAQDx4VrM//Ug3DghzW+uRCAi/6ZWKjApxokdbz2CiTe9J1sdDDJEfcSBknp8uN2Az3eXosHmAACk9tPj9xcPwG/Gp3KPJCLqNkkCbCWHZa2BQYYoiNU3NuOr/aX4cNsJ7C+pb7u/f1wo/nDJQEwfkwy1klPliChwMcgQBZnKBiu+PVSBtQfKsaWoGg6XAAColRKmDk/EbyelI6d/DBQcQiKiIMAgQxTgbA4ndh2vw5YiIzYVGrH7RB2E+OXxwQnh+M34VFw/LgUxYbyMmoiCC4MMBQXR8pu7df8Pg8EAo9HotfZjY2O9vrOrpzXWWZ0orGlGYU0zDlfbcdhoh93Z/pjsaDUmp+hwXqoOyeEqAHU4nl+H4914He6NRESBgEGG/I612YmqBhuqzDb3n623U742mm2wNjthc7hgd7hgd7ogBKBSSFBIgLXJAuFohrA3wWVrhMtmgctqhstqhtNSC6e5puXPWjgtNXCaayEctrPWpA8JweG8PK+FGYPBgCFDh6KpsZO1FxRKqKNToI5Nb7llQJs4EKrIhDMOdZhrYDu+D1bDXjQV78bxBiPWeaVS7o1ERP6NQYZ6hdMlUGOxo6rBhsoGa6dBpcHq8Ph1WueDKDQhgAZASNe3nFdJAnoloFMK6JS//N1WV47v33sV+4vLEZ2QjFCNssc7v1ZVVcHqAK57ZCnC4tNhdUpodAIWhwSLQ0KjQ4LFAQh09DoC4Sqgn9aFaI1AnM6Fk7t3Yu1XSzHt7kcx+O57elRbK+6NRESBgEGGuszhdMFic8Jsd8Bic8Bsc//p/rsTZmszaix2VFvsqG20o9psR43FfatttMMlzv0arTQqBeLCtIgL1yI+3P1n2y1Mi9hwLUI0SmiUCmhU7ptCkuBwCuzZtw/XXDsDt/3v64hOGQCbw91zY3O4YG12umu2O9HY8qfF5oDDJeAQEhocQIPj9PCQisSZizBvTRWw5huoFBIi9GpE6tWI0KmgUyuhUSmgVSmhVSmgVEhwCgGXS8DpEnAJgUa7E2abA2ar+9+trtGG9Ac+xi4AqOzk30GpQHSoBtGhGsSEaRAXpkV8hBZaVftLpZvK3PsYxSRnIDV7eNf/oTvBvZGIKBAwyPRAV+c4CCHgcAFOIeB0AU7h7jlwttzncLl7LJwCaLLZoVCp4XSd/hz3n82tz3MJOATOOM7R0k5r+3aHE0JStH3tEF1//ql1OVzijHkY3SUBiNAqEKVToJ9OgX56JaJ0CoSpBOLCNIjSKRClU6KfToEQtXRar0dzy80M2ADYgEa4b6erNhTAUVeGCDWQGKnr0vtjd7rQaHOHDYvdAYutNfA4UFNnQllZOUJiktAsJDhaepdqLPae/YPA3QsUptcgRKNEuE6NSJ0aEXoVIluCUphW1ePeHyKiYBYQQeb111/Hiy++iPLycowePRqvvvoqJk2aJGtNf1+7B0/9YxVcUEJSayGpNJBUGija/v7LfZIyIP6Zu0w4muGyN7rnn7Tc3H9vhKvRBGdTfcufJrga6+FsNMHVVA+npQ4Qrg5alAB0o7umi7o6t0OSpJbeFCX6hWrOePzQ1iPY+Y+73ceqtFDown65aUMhqTWQlBpIKjUkpRpQKAHhgnC53OfrcsLVbIWwNbb8ezXCZbXA1ViHO5/9J4aMnezV8yYi6kv8/jfsRx99hPnz52PZsmWYPHkylixZgqlTp+LIkSOIj4+Xra49hlross/vQQsCCrhXRVQAUEiAs9kGq7keIWER0Gp1UEjuGRIK6ZfjTj1eksQvf2/589S/V5ceQ+GuzRg88SLEJ6W4Hz9LG+3vw2ntCigkQCUBhTt/wDcrlrjnYowa34Pzd2udh+Gt9k5t01tzO5rMJgDw6xqJiPoqvw8yixcvxp133onbb78dALBs2TJ8/fXXeOedd/CXv/xFtrouSNPhwzcXI/fm/4e4pBSoFAqoFBJUSgkqhXuehFopQalw3xRSy02Btr+fbuf6L/Desocw+8m3MCbn4h7XuHN9PnZu+QhDrpiCMeOH9rg9AKgr8e5cjNZ5GIEwtyMQaiQi6mv8OsjY7Xbs3LkTCxYsaLtPoVAgNzcXW7Zs6fA5NpsNNtsvl9HW17uXZTeZTF6tLSu0GQ27voL6oslQK91DGM6W29kv4u1c6y+38mP5KAoN6XGN3m7PF22yRtboT22yRtboT20GQo1VJ4sBuIfyvf17trU9Ic4x9UD4sZKSEgFAbN68ud39Dz30kJg0aVKHz3niiScE3BMueOONN9544423AL+dOHGi06zg1z0ynliwYAHmz5/f9rXL5UJNTQ1iYmKC6uoPk8mEtLQ0nDhxAhEREXKX43M83+DVl84V4PkGs750roDvz1cIgYaGBiQnJ3d6nF8HmdjYWCiVSlRUVLS7v6KiAomJiR0+R6vVQqttv59MVFSUr0qUXURERJ/4wLTi+QavvnSuAM83mPWlcwV8e76RkZHnPEbhk1f2Eo1Gg/Hjx2P9+vVt97lcLqxfvx45OTkyVkZERET+wK97ZABg/vz5mDNnDiZMmIBJkyZhyZIlsFgsbVcxERERUd/l90Hm5ptvRlVVFR5//HGUl5djzJgxWLt2LRISztw4ry/RarV44oknzhhGC1Y83+DVl84V4PkGs750roD/nK8kxLmuayIiIiLyT349R4aIiIioMwwyREREFLAYZIiIiChgMcgQERFRwGKQ8QOZmZmQJOmM27x583Ds2LEOH5MkCatWrTprm7fddtsZx19xxRW9eFYdczqdeOyxx5CVlQW9Xo8BAwbg6aefbreXhhACjz/+OJKSkqDX65Gbm4uCgoJztv36668jMzMTOp0OkydPxrZt23x5Kl1yrvNtbm7GI488gpEjRyI0NBTJycmYPXs2SktLO233f//3f894f4cMGdIbp9Sprry/nn5vBuL7C+Csn98XX3zxrO366/vb0NCA+++/HxkZGdDr9Tj//POxffv2tseD6bMLdH6+wfbZBc79/vrtZ7en+yFRz1VWVoqysrK227fffisAiA0bNgiHw9HusbKyMvHkk0+KsLAw0dDQcNY258yZI6644op2z6upqenFs+rYs88+K2JiYsRXX30liouLxapVq0RYWJhYunRp2zHPP/+8iIyMFJ999pnYu3evuPbaa0VWVpZoamo6a7sffvih0Gg04p133hEHDx4Ud955p4iKihIVFRW9cVpnda7zraurE7m5ueKjjz4Shw8fFlu2bBGTJk0S48eP77TdJ554QgwfPrzd+1tVVdUbp9Sprry/nnxvBur7K4Q44/P7zjvvCEmSRFFR0Vnb9df396abbhLDhg0T33//vSgoKBBPPPGEiIiIECdPnhRCBNdnV4jOzzfYPrtCnPv99dfPLoOMH7rvvvvEgAEDhMvl6vDxMWPGiDvuuKPTNubMmSOmT5/ug+p6Ztq0aWfUfv3114uZM2cKIYRwuVwiMTFRvPjii22P19XVCa1WKz744IOztjtp0iQxb968tq+dTqdITk4WCxcu9PIZdM+5zrcj27ZtEwDE8ePHz3rME088IUaPHu2tMr2mK+fryfdmML2/06dPF5dddlmn7frj+9vY2CiUSqX46quv2t0/btw48eijjwbdZ/dc59uRQP7sduV8/fWzy6ElP2O32/Hvf/8bd9xxR4ebXO7cuRN79uzB3Llzz9nWxo0bER8fj8GDB+Oee+5BdXW1L0rulvPPPx/r169Hfn4+AGDv3r3YtGkTrrzySgBAcXExysvLkZub2/acyMhITJ48GVu2bOmwTbvdjp07d7Z7jkKhQG5u7lmf01vOdb4dqa+vhyRJ59wjrKCgAMnJyejfvz9mzpwJg8HgzdI90tXz7c73ZjC9vxUVFfj666+79Pn1t/fX4XDA6XRCp9O1u1+v12PTpk1B99k91/l2JJA/u109X7/87HotEpFXfPTRR0KpVIqSkpIOH7/nnnvE0KFDz9nOBx98ID7//HOxb98+8emnn4qhQ4eKiRMnCofD4e2Su8XpdIpHHnlESJIkVCqVkCRJPPfcc22P//TTTwKAKC0tbfe8G2+8Udx0000dtllSUiIAiM2bN7e7/6GHHhKTJk3y/kl0w7nO93RNTU1i3Lhx4tZbb+203f/85z/i448/Fnv37hVr164VOTk5Ij09XZhMJm+fQrd05Xy7+70ZTO/vokWLRL9+/TodahHCf9/fnJwccfHFF4uSkhLhcDjEv/71L6FQKMSgQYOC7rMrROfne7pA/+wKce7z9dfPLoOMn7n88svF1Vdf3eFjjY2NIjIyUvztb3/rdrtFRUUCgFi3bl1PS+yRDz74QKSmpooPPvhA7Nu3T6xcuVJER0eLd999VwgRfEHmXOd7KrvdLq655hoxduxYUV9f363Xqa2tFREREeKf//ynt0r3SHfOt9W5vjeD5f0VQojBgweLe++9t9uv4y/vb2FhobjooosEAKFUKsXEiRPFzJkzxZAhQ4LusytE5+d7qmD47ArR9fNt5S+fXQYZP3Ls2DGhUCjEZ5991uHjK1euFGq1WlRWVnrUfmxsrFi2bFlPSuyx1NRU8dprr7W77+mnnxaDBw8WQvzywdi9e3e7Yy666CLxpz/9qcM2bTabUCqV4tNPP213/+zZs8W1117rtdo9ca7zbWW328WMGTPEqFGjhNFo9Oi1JkyYIP7yl794XKs3dPV8T9fZ92YwvL9CCPHDDz8IAGLPnj0evZY/vL+tzGZzW2C56aabxFVXXRV0n91TdXS+rYLls3uqzs73dP7w2eUcGT+yfPlyxMfHY9q0aR0+/vbbb+Paa69FXFxct9s+efIkqqurkZSU1NMye6SxsREKRftvO6VSCZfLBQDIyspCYmIi1q9f3/a4yWTC1q1bkZOT02GbGo0G48ePb/ccl8uF9evXn/U5veVc5wu4L+O86aabUFBQgHXr1iEmJqbbr2M2m1FUVOT3729HzvW9Gejvb6u3334b48ePx+jRo7v9Ov7y/rYKDQ1FUlISamtr8c0332D69OlB99k9VUfnCwTXZ/dUZzvf0/nNZ9drkYh6xOl0ivT0dPHII490+HhBQYGQJEmsWbOmw8cHDx4sVq9eLYQQoqGhQTz44INiy5Ytori4WKxbt06MGzdOZGdnC6vV6rNz6Io5c+aIlJSUtstVV69eLWJjY8XDDz/cdszzzz8voqKi2sZip0+ffsYlnJdddpl49dVX277+8MMPhVarFe+++644dOiQuOuuu0RUVJQoLy/v1fM73bnO1263i2uvvVakpqaKPXv2tLus0WaztbVz+vn++c9/Fhs3bhTFxcXip59+Erm5uSI2Ntbj3jpvOdf5dvV7M1je31b19fUiJCREvPHGGx22Eyjv79q1a8WaNWvE0aNHxX//+18xevRoMXnyZGG324UQwfXZFaLz8w22z64QnZ+vP392GWT8xDfffCMAiCNHjnT4+IIFC0RaWppwOp0dPg5ALF++XAjhnktz+eWXi7i4OKFWq0VGRoa48847/eIHg8lkEvfdd59IT08XOp1O9O/fXzz66KPtPvgul0s89thjIiEhQWi1WjFlypQz/l0yMjLEE0880e6+V199VaSnpwuNRiMmTZokfv755944pU6d63yLi4sFgA5vGzZsaGvn9PO9+eabRVJSktBoNCIlJUXcfPPNorCwsJfP7kznOt+ufm8Gy/vb6s033xR6vV7U1dV12E6gvL8fffSR6N+/v9BoNCIxMVHMmzev3TkF02dXiM7PN9g+u0J0fr7+/NmVhDhlCUoiIiKiAMI5MkRERBSwGGSIiIgoYDHIEBERUcBikCEiIqKAxSBDREREAYtBhoiIiAIWgwwREREFLAYZIiIiClgMMkRERBSwGGSIyC9ccskluP/++8/6uCRJbbfQ0FBkZ2fjtttuw86dO9sdt3HjxrbjFAoFIiMjMXbsWDz88MMoKytrOy4zM7Ndm6ffbrvtNh+dKRF5E4MMEQWM5cuXo6ysDAcPHsTrr78Os9mMyZMnY+XKlWcce+TIEZSWlmL79u145JFHsG7dOowYMQL79+8HAGzfvh1lZWUoKyvDJ5980vac1vuWLl3aq+dGRJ5RyV0AEfUNl1xyCUaMGAEA+Ne//gW1Wo177rkHTz31FCRJ6lIbUVFRSExMBODuUbn88ssxZ84c3HvvvbjmmmvQr1+/tmPj4+Pbjh80aBCmT5+OsWPH4p577sGmTZsQFxfXdmx0dHS75xBR4GCPDBH1mhUrVkClUmHbtm1YunQpFi9ejH/+8589avOBBx5AQ0MDvv32206P0+v1+P3vf4+ffvoJlZWVPXpNIvIf7JEhol6TlpaGl19+GZIkYfDgwdi/fz9efvll3HnnnR63OWTIEADAsWPHunVsfHy8x69JRP6DPTJE1GvOO++8dsNIOTk5KCgogNPp9LhNIQQAdGl4qjvHElFgYJAhooCWl5cHAMjKyurysZmZmb4siYh6EYMMEfWarVu3tvv6559/RnZ2NpRKpcdtLlmyBBEREcjNze30uKamJrz11lu46KKL2k30JaLAxjkyRNRrDAYD5s+fj7vvvhu7du3Cq6++ipdeeqnt8aqqKuzZs6fdc5KSkpCQkAAAqKurQ3l5OWw2G/Lz8/Hmm2/is88+w8qVK8+42qiyshJWqxUNDQ3YuXMnXnjhBRiNRqxevdrXp0lEvYhBhoh6zezZs9HU1IRJkyZBqVTivvvuw1133dX2+Pvvv4/333+/3XOefvpp/PWvfwUA3H777QAAnU6HlJQUXHjhhdi2bRvGjRt3xmsNHjwYkiQhLCwM/fv3x+WXX4758+e3Xb5NRMFBEq2z34iIfOiSSy7BmDFjsGTJErlLIaIgwjkyREREFLAYZIiIiChgcWiJiIiIAhZ7ZIiIiChgMcgQERFRwGKQISIiooDFIENEREQBi0GGiIiIAhaDDBEREQUsBhkiIiIKWAwyREREFLD+P8xNfhHquyE1AAAAAElFTkSuQmCC",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sns.histplot(column_data, kde=True)\n",
    "plt.xlabel('pLDDT')  # Set the x-axis label to your column's name\n",
    "plt.ylabel('Frequency')    # Set the y-axis label\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>short_name</th>\n",
       "      <th>mean_b_factor</th>\n",
       "      <th>std_error</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>30863</td>\n",
       "      <td>76.176428</td>\n",
       "      <td>0.576875</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>227</th>\n",
       "      <td>50678</td>\n",
       "      <td>77.396509</td>\n",
       "      <td>0.631230</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>266</th>\n",
       "      <td>96404</td>\n",
       "      <td>78.451445</td>\n",
       "      <td>0.521628</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>429</th>\n",
       "      <td>95143</td>\n",
       "      <td>79.429210</td>\n",
       "      <td>0.514830</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    short_name  mean_b_factor  std_error\n",
       "80       30863      76.176428   0.576875\n",
       "227      50678      77.396509   0.631230\n",
       "266      96404      78.451445   0.521628\n",
       "429      95143      79.429210   0.514830"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Active sites modelled with accuracy below pLDDT 80\n",
    "df.loc[df['mean_b_factor'] < 80]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Full structure pLDDT scores, broken down by paralogue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def read_list_items(file):\n",
    "    tip_lst = []\n",
    "    for i in file:\n",
    "        i = str(i.strip())\n",
    "        tip_lst.append(i)\n",
    "\n",
    "    return tip_lst"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('/home/mattm/projects/np_paper_structural/20240328_colabfold/tips_by_paralogue/Node_518_SAL_tips.txt', 'r') as SAL:\n",
    "    sal = read_list_items(SAL)\n",
    "\n",
    "with open('/home/mattm/projects/np_paper_structural/20240328_colabfold/tips_by_paralogue/Node_677_AHL_tips.txt', 'r') as AHL:\n",
    "    ahl = read_list_items(AHL)\n",
    "\n",
    "with open('/home/mattm/projects/np_paper_structural/20240328_colabfold/tips_by_paralogue/Node_863-CNP_tips.txt', 'r') as CNP:\n",
    "    cnp = read_list_items(CNP)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SAL: 160\n",
      "AHL: 187\n",
      "CNP: 150\n"
     ]
    }
   ],
   "source": [
    "print(f'SAL: {len(sal)}')\n",
    "print(f'AHL: {len(ahl)}')\n",
    "print(f'CNP: {len(cnp)}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "pLDDT_score = pd.read_csv('/home/mattm/projects/np_paper_structural/20240328_colabfold/pLDDT_scores.txt', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    structure  pLDDT    pTM paralogue\n",
      "0       50679   87.8  0.899       ahl\n",
      "1       77157   87.7  0.902       ahl\n",
      "2       23215   90.0  0.913       cnp\n",
      "3       91327   92.4  0.925       ahl\n",
      "4       29668   88.9  0.917       sal\n",
      "..        ...    ...    ...       ...\n",
      "500     50213   89.2  0.909       cnp\n",
      "501     31273   86.9  0.892       cnp\n",
      "502     49149   92.8  0.925       cnp\n",
      "503     31442   90.6  0.919       sal\n",
      "504     47649   85.1  0.893       cnp\n",
      "\n",
      "[505 rows x 4 columns]\n"
     ]
    }
   ],
   "source": [
    "def categorize_structure(value):\n",
    "    if value in sal:\n",
    "        return 'sal'\n",
    "    elif value in ahl:\n",
    "        return 'ahl'\n",
    "    elif value in cnp:\n",
    "        return 'cnp'\n",
    "    else:\n",
    "        return 'unknown'\n",
    "\n",
    "# Create the new column\n",
    "pLDDT_score['paralogue'] = pLDDT_score['structure'].apply(categorize_structure)\n",
    "\n",
    "# Display the result\n",
    "print(pLDDT_score)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['ahl', 'cnp', 'sal'], dtype=object)"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pLDDT_score = pLDDT_score[pLDDT_score['paralogue'] != 'unknown']\n",
    "\n",
    "pLDDT_score.paralogue.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAjIAAAGwCAYAAACzXI8XAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjkuMCwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy80BEi2AAAACXBIWXMAAA9hAAAPYQGoP6dpAAB70ElEQVR4nO3dd3hUZdrH8e+Znp4QEiCU0AklFGnSlGoBVJRd2wKCHSyIKEWKKAoLKgqu2JVV14KuulhfARtSbIAU6TWhBRLSM/28fwwZiVJSZuZMuT/X5eVMZjLnJpnM/OY5z3M/iqqqKkIIIYQQIUindQFCCCGEENUlQUYIIYQQIUuCjBBCCCFClgQZIYQQQoQsCTJCCCGECFkSZIQQQggRsiTICCGEECJkGbQuwN/cbjdOpxOdToeiKFqXI4QQQohKUFUVt9uNwWBApzv7uEvYBxmn08nmzZu1LkMIIYQQ1ZCZmYnJZDrr7WEfZMpTXGZmJnq9XuNqhBBCCFEZLpeLzZs3n3M0BiIgyJSfTtLr9RJkhBBCiBBzvmkhMtlXCCGEECFLgowQQgghQpYEGSGEEEKELAkyQgghhAhZEmSEEEIIEbIkyAghhBAiZEmQEUIIIUTIkiAjhBBCiJAlQUYIIYQQIUuCjBBCCCFClgQZIYQQQoQsCTJCCCGECFkSZIQQQggRsiTICCFC1oEDB/jwww/Jy8vTuhQhhEYMWhcghBDV9dhjj7Fr1y5ycnK48847tS5HCKEBGZERQoSsw4cPA1BQUKBxJUIIrUiQEUKEvJKSEq1LEEJoRIKMECJkud0qAMXFxRpXIoTQigQZIURIcjqdlJWVAlBYWKhxNUIIrUiQEUKEpKKiIu9lmSMjROSSICOECEknT56scNntdmtYjRBCKxJkhBAhKScnx3vZ6XSSn5+vXTFCCM1IkBFChKTyIOM2Rle4LoSILBJkhBAh6ejRowC44upUuC6EiCwSZIQQIengwYMAOBMbVbguhJZsNhsbNmwgOztb61IihmxRIIQISQcPHkQ1mHHF1vFeF0Jrr7/+Ou+++y6KovDZZ58RHR2tdUlhT0ZkhBAhx2azcejQIVyWRFRTDOiN7N27V+uyhODQoUMAqKpKbm6uxtVEBgkyQoiQs2vXLlwuF+6Y2qAoOKOT2bd/P2VlZVqXJiLc6avnZCVdYEiQEUKEnG3btgHgiknx/l91u9m1a5eWZQkhQUYDEmREpTidTl5++WWee+45aQcvNLd582YAXLGeIOM+9f/ffvtNs5qEUFWV48ePe6+ffln4jwQZUSmbN2/mP//5D++//z6rVq3SuhwRwZxOJ7/88gtuSwKqOc7ztbh6oCj8/PPPGlcnIllhYSFWqxW3KQaAY8eOaVxRZJAgIyrl9ElrMoFNaOn333+ntLQUZ3z9P75oMOOMSWXLli2yE7bQTHlwccXWrXBd+JcEGVEpEmREsPjuu+8AcCY2qPB1V0ID3G43a9as0aIsIbwrllyxtUFnkF4yASJBRlTK6V1TpYOq0IrT6WTlypWoxihc8WkVbnMkNwXg//7v/7QoTQj2798PgDsqCVdUIgcPHsTlcmlbVASQICMq5ciRIwCoOj2HDx/WuBoRqX7++Wfy8/Nx1GoKSsWXL9UchzOuLuvXr5d9l4Qm9u3bB3iCjDsqCbvd7n3tFP4jQUZUyv79+1GN0biia3Po0CHsdrvWJYkI9NFHHwHgqN38jLc7kpujqirLli0LZFlCALBnzx5UgwXVYMEVlQjA7t27tS0qAkiQEedVWlrK0aNHcZ36lOF2u6UdvAi43bt389NPP+GMq4c7OvmM93EmN0U1RvHRxx9TWloa4ApFJMvLy/N0m45JAUXBHZMKwNatWzWuLPxJkBHntWPHDgBc0cm4YzxvINu3b9eyJBGB3n33XQDs9dqf/U46A/bUNpQUF/Ppp58GqDIh/ggsrjhPgHHFJIOi8/Y8Ev4jQUac16ZNmwBwxdXxbtAnf5wikHbt2sXKlStxRSf/ZZLvn9lTM0Bv4s233qKoqChAFYpI532dPPUaic6AK6Y2u3btktFBP5MgI85r/fr1gIIrNtXThMxgYf36DaiqqnVpIgKoqsqzzz6LqqrYGnYDRTn3NxjMWNM6UlRYyL///e/AFCkimqqqrF6zBvRG77YZAM74+rhcLn755RcNqwt/EmTEORUWFrJ582acsalgMHs26EtowPHjOTKJTQTEN998w6ZNm3AkNcYVX69S3+NIbY3bksBHH33kXRIrhL8cPHiQw4cO4YhvADq99+vOxEYArF69WqvSIoIEGXFOq1evxu12e/8g4Y8/zvLGZEL4S25uLs88sxB0emwNu1b+G3V6rA274XK5mDv3nzidTv8VKSLeDz/8AIAzqVGFr7uja+E2xbBmzRp5DvqRBBlxTp9//jkAzlpNvF9zJjYAvYkv/+//pNmT8BtVVZk3bx6FhQVYG3bz7qtUWa7Ehthrt2DHju0sWbLEP0WKiKeqqqcJo06PM6Fit2kUBWdSY4qKili3bp02BUYACTLirA4cOOA5rZRQH9Uc+8cNOgP25GacOH6cH3/8UbsCRVj76KOPPMutExrgSMmo1mPYGl2I2xzPf/7ztuyMLfxi8+bNHDx4EEdSY8/p9z9xpLQEkFV0fqRpkCkuLubxxx+nX79+tG/fnuuvv9478xtgypQptGrVqsJ/t9xyi4YVR5b//Oc/ANhTWv/ltvI3lv/85z8y6Vf43Pr163nuuedQjVFYm/Q5/wTfs9EbKWt6ESowc+ZM6bIqfK48oDhSWp3xdndUEs7YVH788SfZRNJPNA0y06dPZ82aNcyfP59PPvmEXr16MWbMmAq/7D59+vDDDz94/1uwYIGGFUeO7Oxsli9fgSu6Fq7Ehn+53R2dhCMxna1bt55a1SSEb2RlZTFjxkxcbpXSZv1RjVFnva++8AhRO5ejLzx7QHHHpmJN70FBQQFTp06lpKTEH2WLCHTs2DFWfv017qjEP5Zdn4EjJQNVdfP+++8HsLrIYdDqwFarla+++orFixfTtatnEt8999zDN998w9tvv82ECRMAMJlMpKSknOuhKkXmclTN4sWLUVU39rSOZ/00bK/fEWP+ARYvXszzzz+PXq8/4/2EqKz8/HymTJlCSUkxZU0vxh139jcHANPhDRiKjoLbQdk5VjQ5UlqhK8tn//6tzJo1i9mzZ2M0Gn1dvogwb731Fi6nE1ujDuccNXTWaor78Ab+t2wZ1157LcnJZ+5MLSqq7Pu2ZkHG6XTicrkwmyueUzSbzRU+4f/000/06NGD+Ph4LrzwQu677z6SkpKqfDxp4FZ5W7du9cyyj6+HMzH9rPdzRydjT2nJnj07ee6557jooosCWKUIN8XFxbzwwgscOXIEW1pHnMnNzvs9istR4f/nYmvYFZ2tkJ9//pkHH3yQkSNHSvgW1Xby5Ek+++xz3JaECoshzkinw1avA7r9P/Cvf/2Lq666KjBFRgjNgkxsbCydOnVi8eLFNG3alNq1a/Ppp5+yceNGGjXyLGHr06cPgwYNokGDBmRlZbFgwQJuu+023nvvvSq/AGVmZsqLViUUFBQwd+5cUHRYG/U879wEW4MuGE8e5P/+7yuuueYaGjRocM77C3EmBQUFPPDAAxw5cgR7nTbY0zr5/iCKjrJm/YjatYLNmzfz6aefMm3aNAwGzV4GRQj75z//idvtwpbW8S87sZ+JM7k57iO/sXbtWm699Vbq16/v/yJDnMvlqtQghKZ/wfPnz+ehhx7ioosuQq/X06ZNG4YMGeLds2LIkCHe+5ZP9h04cKB3lKYq9Hq9BJnzUFWVJ554ghMnTmBr0Bk1KuH832SwYE3vgbLnG2bPns1zzz33l1E2Ic4lPz+fSZMmsXfvXuyprbE17F79yb3nozNQ1nwgUbuW8/3336PT6Zg2bZqcZhJVsmHDBpYvX44rpvb5R2PK6XTYGnRFt+drFi1axBNPPIHir+d5hNF0sm+jRo1466232LBhA99++y0ffPABTqeThg3/OrkUoGHDhiQlJXHgwIEAVxoZ3n77bdauXYszoT72uufYmO9PnLWaYE/JYPfu3SxatEhWMYlKy87OZuy4cezevRt7aga2Rhf6L8SU0xsoazEIZ1w9vv32WyZNmiR7MolKs9vtpxadKFgb96rUaEw5Z1I6zoSG/PLLL3z77bd+qzHSBEUfmejoaFJTUykoKOCHH35gwIABZ7zf0aNHyc/P98nkX1HRihUrePnll1FNsVibXFTlNxNbo264Ymrz2Wef8dZbb/mpShFOtm7dyrhxd3Hk8GFsaZ2wNerh/xBTTm+grOUgHEmN2bBhA/fccw85OTmBObYIaW+++SZZWVnY67TBHV3FSbuKgjX9QtAZWLToWfLz8/1SY6TRNMisWrWK77//nqysLFavXs2oUaNo2rQp11xzDSUlJcybN4+NGzeSnZ3N2rVrGTduHOnp6fTp00fLssPOzz//zNy5/wSDmdKWl5xzuetZ6Tyfct3mOF599VU+++wz3xcqwsa3337LfRMmUFhURFnj3tjrdwpciCmnM2Bt1g97nbbs37+fsWPHsn379sDWIELK+vXreeutt3Cb47DVv6Baj6Ga47DWv4CTJ/OYO3cubrfbx1VGHk3nyBQVFbFgwQKOHj1KYmIil1xyCRMmTMBoNOJyudi5cycff/wxRUVFpKam0qtXL8aPH4/JZNKy7LCydu1aZs6ciUtVKW0xAHdUYrUfSzVGUdryEmK2fcYTTzyBy+Xiyiuv9F2xIuQ5nU5efvll3nvvPdAbKW0xCFeChpMeFQVbo+64zbHkHvyJu+++m/vuu4+hQ4dqV5MISnl5ecye/RgqCmXN+oG++vOqHHXaYig8wo8//sjSpUu5/vrrfVhp5FHUMJ/Q4HK52LhxIx07dpTJvn/y7bffMnv2bFwolDYfiCs+zSePqys9SfTOL1EcZYwbN45rr73WJ48rQlteXh6PPvooGzduxG1JpKx5/xoFZ4Dorf9DX5qLKzqZ0rY1W9KqLzhE1N7vUJxWBg8ezPjx42XiugA8AXzy5Mn8+uuvWBt2x1G3bY0fU3FYifn9f+idZTzzzDO0b1/5eYmRorLv30ExR0YElqqqvP322zzyyCO40FPS4tJKhZjKdFEFT9ffkozBqKYYFi9ezMKFC2Xn1wi3ceNGbrvtNjZu3IijVhNK2lxR4xDja66E+pS0uRJXTG0+//xz7r77brKzs7UuS2hMVVUWLVrEr7/+iiOxEY46bXzzuEaLp+mjCtOmT5fnWg1IkIkwNpuNOXPm8NJLL+E2xlCScfl5u6eWMx3egKEgC9PhDee9r2pJoCRjCK7oWnz00UdMnjxZVoZEIIfDwUsvvcSECRPIzcvD2rA71qZ9azQs70+qOZbSjCHYU1qxa9cubrn1Vj777DNZiRfB3nvvPZYtW4YrOhlr04vPO5ersh/4AFxxdSlL70lRYSGTJk+Wyb/VJEEmgmRnZ3PPPfewfPlynLGpnk/FVZh1X5UuqvDHm4IjKZ1ff/2V22+/nW3btlWrdhF6Dhw4wLhx43j77bdxmeMoyRjqGZIP9t4ZOj22xr0oa9YPm9PTW2nmzJkUFBRoXZkIsG+++YYXXngB1RRLWYtBlQrgVfnAB+BMaYktrSOHDx1i2rRp2Gy2mpYdcSTIRIivvvqKW2+7jZ07d2JPaUlZq8urtzqpqvRGrM36Y0vrxJEjR7n77rt55513ZKZ+GHO73Xz44Yfcdtvt7Nq1C3tKS0raXIU7NrTaJjhrNaG47TCc8fVYtWoVY8aMYd26dVqXJQJk9erVPPbY46A3UdpyEKopulLfV9UPfAD2tE44kpuxdetWpk+fLmGmiiTIhLmCggIee+wx5syZg9XuoqxZX2yNe4MugBOfFQV7/U6UtroMp97Ciy++yAMPPMDRo0cDV4MIiOzsbO677z4WLVqEza1Q1nyA5/kWpKeSzkc1xVDW8jKsDbuRd9KzoeWcOXMoLCzUujThR2vWrGHmzIc9CyFaDMIdVfX9/apEUbA27o0jsRE///wzM2bMkDBTBRJkwpSqqnz77bfcdNNNrFixAldsKsVtr8JZq6lmNbni61Hc9iociY1Yv349N40ezYcffiijM2HA5XLx7rvvMubmm9m0aROOpCaUtLsaZ9LZNx0NGYqCo247StpehSsmha+++opRN93Ed999p3Vlwg+8LSnA0x6gknMIa0ynx9qsH87Ehvz00088/PDD2O32wBw7xEmQCUPHjx9n5syZzJo1i/zCYqwNu1GaMRjVHKd1aZ69mZoPoKzpxVidsGjRIsaPHy/bToSw3bt3c9ddd/HCCy9gV/WUNe+PtXm/wJy6DCB3VBKlrYdgbdiN/IIiHn74YWbOnMmJEye0Lk34yHfffceMGTNwuqG0xSW44uoGtgCdnrJm/XEmNGTdunU89NBDlJaWBraGECRBJow4HA7efvttRowcyapVq3DG1aO47TAcddtVaT8Qv1MUnMnNKGl3NY5aTdi8eTM333wzzz//PCUlJVpXJyqprKyMF154gdtvv53t27fjSG5OcbtrcCY11ro0/1F0OOq288ydiavL999/z6hRo/joo49wuVxaVydqYNmyZcyaNQunqpwaiQlwiCmn01PWvB+OxEb88ssv3H///bKa6TyC6N1N1MRPP/3EmDFjeOmll7C6FMqa9KGs1WWolnitSzsr1RiFtVk/SlsMwmGI4b333mPEyJF89dVXstw1yK1du5abRo/m3XffxWmKpbTlpVibXgSGyGggp1riKWt1OWWNe1Nqd7Fw4ULuuusudu3apXVpoopUVeWNN95gwYIFuA0WSloNxhVfT9uidAaszfvjqN2C7du3c8+998peYOeg6RYFoub27t3Liy++yI8//uiZVFunLba0TmAInW0cXIkNKYmvh+noVk4e+Y05c+bw0UcfMXbsWOl2GWRycnJ47rnnPPNDFB22tI7Y67UHXQS+lCgKzpSWFCc2xJz1E9u3b+eOO+7gb3/7G6NHjyY6unKrXIR2XC4X//rXv/joo49QzXGUtLw0eD78KTqsjXvjNljIOriZcePuYv78eTRtqt08x2AVga8+4eHEiRO89tprfPHFl6iqG2d8mmfPGH/PrvcXnQF7Wgccyc0wZ//Ctm3buPfee+nduze33347jRo10rrCiOZ0Ovnwww957fXXsZaV4Yyriy29Z9B159WCaozC2vRiHLVbYDmwhqVLl7Ly66+59557uOiii1CCvW9OhCotLWX27NmsXbsWV3QtylpcUukl1gGjKNgbdkU1RnEiy7MX2COPPELXrl21riyoSJAJMUVFRbz33nu8//772Gw2XFFJ2Bp2xZXQQOvSfEI1x2Jt1hd7nbaYs3/mhx9+YM2aNQwdOpSRI0eSkhJavUjCwaZNm3j66WfYt28vqtGCtclFOJObBX9juwBzxadR0vZqTEc3k3vkNx5++GG6devGvffeS4MG4fH3GS5OnDjBlClT2L17N86EBjXeBNLfHHXboZpiYN/3TJ48mfvvv182Nj2NBJkQUVZWxocffsjbb79DSUkxqikGa+OuOGs3D66JvD7ijk2hrNXl6POzsGT/zLJly/jiyy+55uqrueGGG0hMTNS6xLCXn5/Piy++yBdffAGAPTUDW/3OETMPplp0euxpHXHUaorl4Drv3LV//OMf3HDDDbIJZRDYvXs3U6ZM5cSJ49hTMrClXxgSr6HOWk0oMcUQvXsFTz75JIcOHeK2225Dpwv+2v1NgkyQs9lsfPrpp7z51lvknzyJarBga9gNR2pG+M9LUBRcSY0oSWyAIXcP6qEN3n1Prr32Wv7+978TGxurdZVhx+1288UXX/DCCy9QVFTk2WMmvWfIdebVkmqJp6zFIAz5B1AP/siSJUtYvnw5EyZMoEuXLlqXF7HWrFnDI48+is1qxdqwG446IbBlxmncsamUZAwletdy3nnnHQ4dOsTUqVOJigqvVgdVFebvhKHLbrfzxRdf8Mabb5J74gToTdjqX4C9TtugHgL1C0WHs3YLims1xXh8B+qR3/j3v//Nf//7X66//nquueYamVjpI3v37mXBggVs2bIF9CasjS70hOYQ+MQadBQFZ1JjiuPrYz68gUOHt/LAAw8wYMAAxo0bR3Jy5fc5EzWjqirvv/8+zz//PKqip6z5gJBt1qha4ilpPZSoPd/w/fffc+TIEebMmRPRp90lyAQZp9PJl19+yRtvvOFZbqc3YKvXHnvddmCwaF2etnR6HHXa4KjdEmPONtSjm3nllVd4b+lSbrzhBoYNGxbxn0yqy2azsWTJEpYuXYrL5cJRqwm2ht2Db/JjKNIbPaOoyc2xHFjDypUrWbt2LXfccQdXXHGFnBrwM6fTycKFC/nkk09QTTGUNh+IOybEQ6TBTFmLSzAfXMuuXTu48847mTt3Li1bttS6Mk1IkAkSTqeT5cuX8+9/v8HRo0c8q3jqtsNeNzPsOqTWmN6Ao14mjtQMTMd+h2NbePHFF3nvvfe48cYbufLKK7FYIjz0VcHGjRt54oknOHToEG5zPNZmPXAl1Ne6rLDjjq5FacYQjCd2QvYvPP3006xcuZIHH3yQhg0bal1eWCouLmbWrFn88ssvuKJrU9ZiYPiEc53Os3LQkkhu1k/cc889PPzww/Ts2VPrygJOgozGXC4XK1euZMm//83hQ4c8kwXrtMFerz2qMUz+4PxFb8Se1gF7amtMx7aSf2wrixcv5p1332XEP/7B0KFDZXLlORQXF/Piiy/yySefgKJgq5uJvX6n8J97pSVFwZHSCmdiI8wH1rJp0yZuvvlmRo8ezXXXXYfBID97X8nJyWHy5Cns27cXR2I61qYXgz7Mfr6KgqNuW9yWeNj7DdOmTWf8+HsZNmyY1pUFVJj9VkOH2+3m+++/59XXXiPr4EFQdNhTW2Ov1yF8PjEEisGEvX4n7HXaYDq2lZPHtvLss8/y9jvvcNOoUQwePFjeIP7k559/5p/z5pF74gSu6FqexlsxtbUuK2KoxiiszfvjPHkA9eBaXn75Zb755humT59O48aNtS4v5O3atYvJkyeTl5fnaRLasGtYz/NyJTakpNVgonev4JlnnuHw4cPceeedEXPaUl7dA0xVVdatW8err77K7t27PQEmJcMzAmOWFTg1YjBjPzUh2nR0M7k5v7NgwQLeeecdxowZw4ABA9Dr9VpXqSmbzcbLL7/MBx984OnMW78z9rqZECEveMHGmZROcVxdzNk/s3v3Tm67/XbGjR3LsGHDpJFeNa1fv56pDz2EzWrzTFav00brkgLCHVPbu6Jp6dKlHD9+nIceegijMfwXh0iQCaDNmzfzwgsvsHXrVkDBkdwcW/1OwbErdTgxmLE36IKjTltMRzZx5Nh25syZw3/+8x9uv/12evbsGZFvEnv37mX27MfYt28v7qhEyppejDs6xCc9hgODGVvj3jgTGxG1/wcWLlzIjz/+yKRJk6hVq5bW1YWUNWvW8PDDD+NwuSltPgBXUmR1BFfNsZRkDCFq90q++eYbysrKeOSRR8L+FLt8DAuArKwsZsyYwT333MPWrVtxJDWmpN3VWJteJCHGj1RjFLZG3SnO/Bv2lFYcOJjFtGnTGD9+PNu3b9e6vID69NNPueOOO9i3by/21DaUtLlSQkyQcSU2oqTt1TgTGrBu3TrGjBnDhg0btC4rZKxcuZIZM2bgcENpi0siLsR4GUyUtRyEM6Eh69atY/LkyZSWlmpdlV9JkPGjgoICFi1axOjRo1m1ahXO2DqUtL4Ca/P+skdNAKmmGGyNe1Hc7mociels2rSJO++8k9mzZ3Ps2DGty/Mrp9PJokWLePLJJ7Gjp7TFIE8nU5nQG5RUYxRlLQZhbXQhBYXFPPDAA/zvf//Tuqyg9+WXX/LYY4/hUvSUtLxU+92rtaYzUNZ8AI5aTdi4cSMTJ04M6zAjQcYP3G43n3/+OSNGjuTDDz/EYYylrPkAyjIGS3dUDamWBKwtBlCaMRhXTG1WrlzJqFGjePvtt3E4HFqX53NFRUVMnjyZDz/8EFd0LUpaX4krUZb5Bj1FwVGnDSWtLsOpM/H000/zzDPP4HQ6ta4sKK1Zs4Z58+bj1pspaTUYd2yq1iUFB53Ou5nptm3bPKNVYfg6BxJkfG7Pnj3ce++9zJ8/n6KSMqwNu3uGi5PSQ6oVdjhzxdWltPUVlDW9GKtbx0svvcStt97Kxo0btS7NZ3Jychg7diy//vorjsR0SjOGyGTyEOOO84zguqJr8fHHHzNlyhRsNpvWZQWVrVu3MmvWI6g6HaUtL8EdLXOKKlB0WBv3wpGUzq+//sq8efNwu91aV+VzEmR8xOVy8eabb3LbbbexZcsWHLWaUNxuOI66bWVFSDBSFJzJzShuNxx7ahsOHDjIfffdxzPPPIPVatW6uhrJy8vj/vvvJzs7G1u99lib94+8bS3ChGqOpTRjCI7ERvzyyy/MnDkTu92udVlB4cCBA0yZMgW7w0Fps/7SPuBsFM/IjDOuLitWrOCFF17QuiKfk3dYHzh69Cj33Xcfr776Ki5DNKUtL8XarJ/0gwkFBhO29AspaXMFrijPJ9/bb7+DXbt2aV1ZtRQUFDBx4gOnQkwH7A26yEhgqNMbsTbrhzOxIT/++COzZ8+O+NNMdrudWbMeoaioiLLGvXElNNC6pOB2as6MOyqJpUuXsmrVKq0r8ikJMjW0Zs0abr75ZjZv3oyjVlOK214l7d1DkDumNqVtrsBepy0HDx5g7NixLFu2TOuyqsThcDB58mTPyqQ6bbHXv0DrkoSv6PSUNeuHMz6NVatWsWDBAq0r0tRrr712agVea5y1m2tdTmgwmClt3h90Bp544klyc3O1rshnJMjUwLJly5g2bTqlNjtlTS7ytMA2hPd6/bCm02Nr1J3SlpfiUIwsWLCAl19+GVVVta6sUt588022b9/u6U/UsJuMxIQbnYGy5gNxxdTm888/D7tP1ZX122+/8d577+G2JGBr0FXrckKKaknA2rArhYUFPPnkkyHz2nY+EmSqQVVVXnvtNRYsWIDb4Jkp76zdXN44woQroT4lrYfitiTwn//8h7lz5wb9UP7OnTt56623cJvjsKb3CP/notOK6dB6dGX5ACj2UnCG9tymStEbKGt6Mej0PLVgAQUFBVpXFFAul8vzBgyUNbko/PZOCgBHSgbO+PqsXbuW77//XutyfEKCTDUsW7aMN954A7clnpKMITLJLAyp5jhKWg/BFZvKV199xUsvvaR1SWelqir//Oc/cbvdWBv3Dv+JvS4H0ds+x3x4I4rqAkDnLCN62+fgCs/lpadTLQlY63cm/+RJFi9erHU5AbVmzRqysrKw124prSyqS1E8H3aAd955JyxGZSTIVNHGjRtZtGgRqjGK0paXoVritS5J+IvBQmnLS3FHJbJ06VK++uorrSs6o82bN7N3714cyc0iohGY+fBG9Nb8v3xdb83HfHhjwOvRgqNOG1xRSaxcuZKioiKtywmY9957DwB7nXYaVxLaVEs8jqTGbN++nc2bN2tdTo1JkKkCm83Go48+isutUtasn/TliAR6I6XNB4LBxBNPPsnx48e1rugvli9fDoCjdkuNKwkMfeGRat0WVhQdjtrNcTqdfPvtt1pXExDbt2/3tLZIbIQalaB1OSHPXtcTBt9//32NK6k5CTJV8MUXX5CXl4etbjtccXW1LidwInU+wimqJZ6yBt1w2O1B90fvdDr5+ptvUE0xEfOcVOwl1bot3DhrNQMUb5ANd7/++isAztotNK4kPLhjU3FZEli/fn3IN8mTIFNJqqryzrvvgs6AI5KGNSN8PkI5Z3JzVFMs//vf/yguLta6HK+CggJKiotxxtYJ/wm+ogLVFI3LEk92drbWpQTEtm3bAHDJ3BifccemUlJSEvLPIQkylXTixAmOHT2KI7EhqtGidTkBI/MRTtHpsCc3xWazsWfPHq2r8Sop8YxAqHqTxpUITehN3udAuNu2bRtuUwyqURqN+orr1EKV7du3a1xJzUiQqaTyxOq2RNa5WZmP8Ae3JRGArKwsbQs5TfmOtmq4r1QSZ6TqjdhstqBvD1BTDoeD3Nxc79+g8I3yn+eRI6H9Wi5BppLKTydE2idfmY/wB9Xg+d0H06mlhARPsNbZg6cmETg6WzFxcXHo9XqtS/ErRU6b+lWo/3wlyFRS/fqebQd01kKNKxFa0ZV5mo81aBA8+7rUrVuXWrWS0RfnaF2KCDDFUYbOVki7du1C/o3ofHTejXdDv+dJcPH8PHUhvrFxaFcfQA0aNECv16MvyYEwaCAkqk5f4gkLjRs31raQ0yiKQmZmO3T2EhRb5PQTEaAvOgpAu3bhv/hAURQUnQ7cLq1LObtQXN156udpMIR2h2QJMpVkMpno1asX+tI8dCXB10tE+JdiL8GYn0Xz5i28o3PBolevXgCYjm3TuBIRSMYcz++7R48eGlfif4qiUD+tPnprkG7JEKKrO3Wnfp7B9ppWVRJkquCaa64BwHQ09DshiqoxHfsdVDfDh18TdMP4/fr1IyUlBdOJHeC0aV2OCABdcQ6GoqNceOGFNG3aVOtyAqJJk8YoTiuKo0zrUv4iVFd36stOAsE1ylwdEmSqoEOHDmRmZmI8eQB9fvCsXBH+pSs9ienYVurWrUf//v21LucvjEYj1113HbgcmI5u0boc4W+q6n1zvPHGG7WtJYCaNGkCgO7Um28wCdXVnbrSkxiNRtLS0rQupUYkyFSBoijcf//96PUGog6uDeohQ+EjqhvL/tWgurn//gmYzWatKzqjIUOGkJKSgvno5qB8oRe+Yzi5D0NBNl26dKF9+/ZalxMwrVu3Bv6YGxRMQnJ1p8uBviyXjIyMkF/1JkGmipo0acKIEf9AsRVj2feDTPwNc6ZD69GX5DBw4EC6deumdTlnFRUVxcSJEz3Ba98PoIZ2y3FxFk4rloPrMJvN3H///VpXE1CZmZkoii4og0wo0hcdBVWlY8eOWpdSYxJkqmHkyJG0b98e48l93gl3Ivzo8w9iPrKJBg0aMGHCBK3LOa8LL7yQQYMGoS85LqeYwpGqYjmwFsVh5dZbbw350wFVFRsbS6tWLTEU54A7vBsABkJ5IOzUqZPGldScBJlqMBgMPPzwwyQl1cKS9VNQnwMV1aMryyd67/eYTGYeffRRYmJitC6pUu6++26Sa9fGfOhX+eQaZozHt2PM20f79u29Cw8iTadOnUB1oy+WlaM1ZSg6gsFgoG3btlqXUmMSZKopOTmZ2bMfxaDXEb3na5RgXRYoqs5pJWrXcnDZmTp1SkitCklISGDWww+j1+mI2vNtUK7wEFWnKz6O5eCPJCYmMnPmzJCf01BdHTp0AEBfJB8ea8TlQF+aS5s2bYJ23l9VSJCpgXbt2jFp0iRw2ojetUKWvoYDt4uo3V+jsxUxevRo+vXrp3VFVZaZmcmdd96J4ijFsucbcMt8mVCmOMqI3vMNCvDwww9Tu3ZtrUvSjMyT8Q198TFQVW8wDHUSZGrokksu4cYbb0RnLSBK3jRCm6piPrAWQ9FR+vXrx0033aR1RdX2t7/9jb59+2IoOoo560etyxHV5XZh2f01ir2YW2+9JSzmM9RETEwMzZo1xVByQhZa1ED5qbnMzEyNK/ENCTI+cOutt9K7d28MhYflTSOEGY/9junETlq1ymDKlClB1/iuKhRFYfLkyTRv3hxTzjaMOdu1LklUlapiPrgWQ/ExBgwYEFE9Y84lPT0d3E4UR6nWpYSs8j0D09PTNa7ENyTI+IBOp+Ohhx6iWbNmmHK2YTixW+uSRBXpC49gyf6J5ORkHn/8sbA4bxwVFcXjjz9OQkICloPrZDg+xBhztmE6vpOWLVsyadKkkA7WvlS+aats4Ft9OlsBJpOJlJQUrUvxCQkyPhIdHc3s2bOJiYkl6sBqdKW5WpckKkmxlxK191v0Oj2PPvpoWM1BqFOnDo899tipyb9fo9iKtS5JVIK+8AiWrB9JSqrFY4+FR7D2lfJl5zrZJLXadLZi6tWrF/K7XpcLj39FkEhLS2PGjOmeCaN7v5NeB6FAVbHsW4XiKOPuu+8Ki6WIf5aZmcmECfehOKxE7V4JLnleBjPFVkT03m8w6PU89thsUlNTtS4pqJSPTMkMmZoJpxE+CTI+duGFFzJ8+HB0ZfmYD63XuhxxHsbjOzAUHqJHjx4MGzZM63L8ZujQoQwbNgx9ae6pLRfkbSAouZ2esOmwcv/994dlsK4pu93uuaCLzCXovqDq9H/8HMOABBk/uP3222nYsCGmo1vQlZzQuhxxFoq9FEv2T8TFx/PAAw+E1SeUM7n77rs9m57m7cF4fIfW5YgzMB/8EX1pHldddRWDBw/WupygZLVaPRckyFSfTv/HzzEMSJDxg9P3QTFn/SyffoOU6fB6cDm58447SE5O1rocvzMYDMycOZP4+AQsWT+iK5F5XMHEcGI3puM7aNmyJXfddZfW5QStnTt3AuA2J2hcSehymxPIy8vj+PHw6JAsQcZPOnXqRK9evTAUHUFfeEjrcsSf6MryMR3fRZOmTbnsssu0LidgUlJSmDFjOorqJmrvN7KDe5BQrIVEHVxDdEwMjzzyCCaTSeuSgpKqqvzyy6+oxmjcUYlalxOynAmeCdO//vqrxpX4hkHrAsLZLbfcwurVqzEd+52yhAZalyNO49nsU+XmMWMirt17165dueGGG3j77bcxZ/+KLf1CrUuqssWLF5/x63feOzHAlfjAqQnnuJxMenA69erV07qioLVv3z7y8nJxJjeHMD8V7E+u+PoA/PLLL2HxQU5GZPyoadOmtG/fHkNBNor0PAgeLgem3D2kpKTSs2dPravRxOjRo0lPT8eU87tseqox47HfMRQfo1+/fvTt21frcoLaBx98AIAzsZHGlYQ2tyUBtzme7777jpycHK3LqTEZkfGzIUOGsGnTJgz5B3HUbad1OQIwFB4Gl53Bgy+PuNGYciaTialTpzJu3Dgs+1dT0u7qkJo8OW7cuDPfYIgKbCE1pNiKsRz6lYSEBMaPH691OUHtwIEDfPHFl7iiknAmhUdHWs0oCrb6ndDt/Y4lS5Z49gwMYZqOyBQXF/P444/Tr18/2rdvz/XXX8+mTZu8t6uqysKFC+nduzft27dn9OjR7N+/X7uCq6FLly4AGORTb9AoH4Ho2rWrxpVoKyMjw9MqwFZ46lSbCDRz9q/gdnL33XeTmJiodTlB7eWXX0ZV3dgadJHTSj7grNUUV3QtvvjiS/bt26d1OTWiaZCZPn06a9asYf78+XzyySf06tWLMWPGcOzYMcDzxH3zzTeZNWsWS5cuJSoqiltuuQWbLXR2mU5OTqZRo0boS0J/+C5c6ItzMJnMZGRkaF2K5kaNGkVcXByWIxtl9/YA0xUfx5i3h1atMhgwYIDW5QS1lStX8sMPP+CMq4tL5hv6hqJgq98ZVXXzz3/OC6n31T/TLMhYrVa++uorHnzwQbp27Up6ejr33HMP6enpvP3226iqyhtvvMHYsWMZOHAgGRkZzJ8/n5ycHFasWKFV2dWSlpaG4rTJCpEgoXMUU7duHQwGObMaFxfn2eXbacd8ZNP5v0H4jPnQLwDcdde4sGkV7w979uxh3rz5oDdhbdxLRmN8yJXQAHvtFuzYsZ1nn31W63KqTbNXcqfTicvl+sseImazmfXr15Odnc3x48crTMaMi4ujQ4cObNiwgSFDhlTpeC6Xyyd1V0d5jxLFUYqql94HmnK7URxWkpOTNX1OBJOhQ4fyzjvvkHt8B7a0jqA3al1S2NOV5mIoPELXrl1p27atPBfPori4mOnTZ2C32yhtMRDVIq+fPqUo2NJ7oC89yaeffkrLli2r/N7qT5X9u9AsyMTGxtKpUycWL15M06ZNqV27Np9++ikbN26kUaNG3kY9f25UlpyczIkTVe+Wu3nzZp/UXR15eXmeCyHaFy+slrqeUlJSwsaNG7UuI2h0796dzz//HOOJnTjqSFt8fzMd3Qp4+k3J8/DM7HY7r7zyCkeOHMaW1hGXrFTyD52Bsub9ifl9GQsXLqSoqCjkTrtrOrY+f/58HnroIS666CL0ej1t2rRhyJAhbN261efHyszM1GyFypdffum5EEKrQsKWTgeKDpPJRMeOHbWuJmg0bdqUFStW4D62DUdqGxm+9yPFYcWYt5f0xo259tprw35rjOqwWq1Mnz6dPXv24EhqjD2tk9YlhTXVHEtZ8/5E7/yKJUuWMHv27KBYDOFyuSo1CKFpkGnUqBFvvfUWpaWlFBcXk5qayn333UfDhg1JSUkBIDc3t8Lur7m5udVKi3q9XrMgk5vraQWvGiyaHL+mwmWpaznVaOHEiRMRu/T6TJKSkrj44otZvnw5utJc3DG1tS4pbBnyD4DqZuiQITJP6wysViszZsxgw4YNOJIaY23aV4J1ALji6lLaYhDRu5YzY8YM5syZExRhpjKCYoZZdHQ0qampFBQU8MMPPzBgwAAaNGhASkoKa9eu9d6vuLiY3377jU6dQiud79mzF7c5HvTyohUMXJYkjh49SmlpqdalBJXyZmyGvNBeihnsyn++F198scaVBJ+SkhKmTp3K+vXr/wgxMhE6YFzx9ShtMRCHS+Whhx5izZo1WpdUKZo+Q1atWsX3339PVlYWq1evZtSoUTRt2pRrrrkGRVEYNWoUzz//PCtXrmTHjh1MmjSJ1NRUBg4cqGXZVXLkyBEKCvJxRdfSuhRxSvnvYts26Z1yui5duhAdHY3x5AGtSwlfThuGoiO0bdu2wkizgBMnTnDvvfdWHImREBNwrvg0b5iZNm06n376qdYlnZemQwRFRUUsWLCAo0ePkpiYyCWXXMKECRMwGj2rJm677TbKysqYOXMmhYWFdO7cmVdeeeUvK52CWfmIkiuhvsaViHKuhAZwdDNr166lc+fOWpcTNMxmM506dWL16tUo9lJUU7TWJYUdffExUFW6deumdSlBZf/+/UyaNImcnBzsqa2xNeoOioQYrbji0yhpdTnRu5fz5JNPkpOTw5gxY4J2PpemQWbw4MEMHjz4rLcrisL48eNDunX3999/D4AzoaHGlYhyrtg6qHoTq1atYtw46eFxuszMTFavXo2++CjOWk21Lifs6Is8zT7bt2+vcSXBY9OmTUyd+hAlJcXYGnTBXjdT5sQEAXdsCiUZQ4ne9RVvvPEGOTk5PPDAA0E5r0tewf0oOzubjRs34oyrJ59ug4lOh6NWY44dOxY229j7SvkbrL5IOlH7g744B71eT+vWrbUuJSgsX76c+++/n5LSUsqaXoy9XnsJMUFEtcRTmjEEV0wKX375JZMmTaKoqEjrsv5CgowfLVu2DABHamityY8EjhTP7+Tjjz/WtpAg06xZMxRFQWc9qXUp4UdV0ZedpGHDhlgsobmC0VdUVWXJkiU8/vjjONFT2vJSnMnNtC5LnIFqjKK01eU4khqzfv167rrrbo4cCa69AyXI+ElBQQHLPvkE1RQjW84HIXdMbVwxqaxZsybkN0zzJbPZTJ26ddGXFWhdSthRHGUoLjvp6ZG9c7PdbmfOnDksWbIEtzmO4oyhuOLraV2WOBe9AWuzftjqZnLw4AHGjh3rl35v1SVBxk8++OADrGVl2OpmSiO8IGVL64Cqqrz11ltalxJUGqenozhKwWXXupSworN6wmEkB5nS0lImT57M8uXLccWmUtp6KGqUbDsQEhQFe8OuWBv3Ir+ggAkT7q/QHkVLEmT84OTJk7z/wQeoxigcKS21LkechSuhAa7oZL7++mv27NmjdTlBo27dugDo7CUaVxJeFHsx8MfPN9KcPHmS++67z7u8urTVZajG0GyqGckcKa0obTEIu8vFtGnT+L//+z+tS5Ig4w9vvPGGZzQmrRPogm+GtzhFUbA16IKqqrz00ktaVxM0yrtqKxJkfEpn9zRgLP/5RpIjR45w9913s3PnTuwprbA26yuvjSHMldCAkpaX4daZmDt3Lu+9956m9UiQ8bHs7GyWLVuG25KAo7aMxgQ7V0J9nPFp/Pjjj6xfv17rcoJC7dqe7QkUu3Q+9iXF4fl5lv98I0V5iDl06BC2tI7Y0ntKj5gw4I5NpSRjMKophueff55///vfmtUizyYfe/HFF3G5XNgadJGulCHC1tCzn8jixYsrvW18OEtMTARA57RqW0iYURyen2dSUpLGlQROXl4e90+cSG5uLtaG3bHXv0CWV4cRd1QiJRlDcFvief311zVbBSrvtD60adMmVq1ahTOurqxUCiHu6GQctVuwe/duvvrqK63L0VxCgmfypeK0aVxJeFFOBcPY2FiNKwmM4uJiHpw0iSOHD2NL64SjblutSxJ+oJpjKW15KaoxmoULF7Jy5cqA1yBBxkdUVeXFF18EwNagq3zqCDG2+heATs/rr7+O3R7Zq3XKgwwyIuNTitNGTGxsUHZG9TWn08n06dPZs3s39tTW2NM6al2S8CPVHEdpy0tQ9UbmzJkb8NP0EmR8ZO3atWzduhVHUmPcsZE3mS/UqaYYbKltyMnJ8TYyjFTlIwaKM7IDna8pLhvxcfFalxEQ7733Hhs3bsSR1ARbowvlg10EcEfXoqT5IFxuN3PmzKWkJHCLBSTI+MDpvUjs9S/QuBpRXfZ67UFv4p1338XhcGhdjmZiYmJQFAXFJaeWfEnnshMXF/6nlQ4cOMCSJUtQTdFYG/eSEBNB3HF1sKV15MSJ47zwwgsBO274j3EGwJYtW/j9999xJKbjjkrUuhxRXQYz9pSW5B7dwjfffMMll1yidUWa0Ol0RMfEUCgjMr6jusHlCPv5MS6Xi/nz5+NwOChrcTEYTFqXFDCLFy8+49fvvHdigCvRlr1uewwn9/PJJ5/Qv39/OnXq5PdjyoiMD/zvf/8DwFG3ncaViJqy12kLihLxezDFx8XLiIwPlZ+mi48P71NLv/zyi+cUe60muGTBQ2TS6bA27g0oLFmyJCCHlBGZGrLb7axevQa3OQ5XbKrW5YgaUk0xOOPq8fvvv5OTk0NqamT+TuPj4zhyTHbA9plTK8Di4uI0LsS/Vq1aBZz6QBBhxo0bd+YbDJHXvdgdUxtnXB02bdpMfn6+t6WDv8iITA1t2LCBsrJSnEmN5VxwmHAmNQZg9erV2haiofj4eHA7wS19dXyhfOm1d0VYGHK5XPzwww+oxmjcMbLgIdI5E9NRVXdAXkclyNTQ3r17AXDG1dG4EuErrlO/y/LfbSQqb9qmOMo0riQ8lDcX9PcnUy3t2rWL/Px8HImN5EOdwJnk2Rx13bp1fj+WBJkaysrKAsBtCd9PWpHGbY4HFA4ePKh1KZqpVasWIEHGV8p/juU/13BUvtJPNVo0rkQEA9XgeR4EYgWoBJkaKi09tR+N3qxtIcJ3dHrQGykri9w3cQkyvlW+z1I4b08QFeWZC6K4Ird1gfiD4vY8D6Kjo/1+LAkyNeRdTumSpaphQ3WDyx72S2XPpXySs85erHEl4UFn8/wc69atq3El/lP+hqXIa6EA73uiBJkQUP4JS2cr0rgS4SvKqTedcP70fD716tUD5HntK4q9GJ1OR0pK+E6CrV27NrGxsRgKsmWSuMCQ7zk137RpU78fS4JMDXXp0gUAQ36WxpUIXyn/XZb/biNR+ciBIkHGJ3S2IlJSUsJ6nyWTycSQIUNQHGUYTh7QuhyhJVXFlLMds8XCoEGD/H44CTI11LZtW+Li4jCe3A8up9bliJpSVYy5u1EUhQsvvFDrajSTkJBAQkIiemu+1qWEPpcDnb2E9PR0rSvxu6uuugpFUTDm/A6qqnU5QiP6gix0tiIuGTQoIL2TJMjUkMFgYNiwYSiOUkzHtmhdjs+ppphq3RaqDHn70Jfm0r9//4g+tQTQpEljdNYiTz8ZUW26spMANG7cWNtCAiAtLY2ePXtiKM7BeHyH1uUILThtRB1ch6IoXH311QE5pAQZH7jhhhtITErCfHRz2A3Fu+LrVeu2kOS0YTn0C0ajkdtuu03rajTXpEkTQEVXlq91KRWEWrjWnwoynp9n+LvvvvtISEjAcnAdupITWpcjAklVidr7PYqtmDFjxgRkfgxIkPGJ6Oho7ho3DlwOonathDBafmhL64jLkviXr7uiErGldQx4PX6juona+x2KrZgRI0aE9eqSysrIyABAX3Jc40oqCrVwrSv2bPVQ/vMMdykpKcycORMFleg9X3u3ZxDhz3R0E4aCLLp3786IESMCdlwJMj4yaNAgrrnmGvRleVj2rfIs4Q0HeiOlrYdgS+uIqugBcBuiKM0YAnqjxsX5iKpizv4FQ0E2PXv2ZOTIkVpXFBTatvXsl6MvDq49l0ItXOuLjxMdHR0Rc2TKde7cmTFjxqDYione+X8SZiKAMWc75uxfSUlJ4aGHHkKnC1y8kCDjQ+PGjaNTp04YT+7Hsvc7cIdJmDGYsde/AHdUIgCqKRoMYdIAUFUxZf+C6egWGjVqxLRp0wL6BxjM6tevT3xCAvriY8E1cTOEwrXisKK35tOmTZuIe16NGDGCIUOGoC85Qcz2z71NAUX4MR7djOXAGhKTkpg3b17A9xSLrL8sPzMYDDz++ON06NABY94+LHu+ln4KwUxVMR9ch/noZho1SmfBggXExATfHAutKIpC5wsuQGcrRrEWal1ORSESrvWFh4DIXMqv0+l44IEH+Pvf/46u7CTR2z/39mgSYUJVMR1ajyXrZ1JSUnh20aKAzYs5nQQZH4uOjmbevHl07doVY/5Bzx+vXT6JBB2XHcvulZhyttG8eXMWLVpI7dq1ta4q6JQvQTcUZGtcSWgq/7l169ZN40q0oSgK48aN46abbkJnLSRm+2foioNrzpWoJrcL84E1mA9vJK1+fZ599lkaNmyoSSkSZPzAYrHw+OOPc9lll6EvOU7M78u8E/6E9hRrATG/f4ox/yBdunTh6aefDutdiWuia9euKIqCIV8anFWZ242hIJuUlNSIWbF0JoqiMGbMGO666y50jjJitn8mS7NDnGIrJnr7Z5iO76Bly5Y8u2iRpgskJMj4iclkYvLkydx7773oXVZitn+B8ejW4JprEIEMuXuJ/f0TdNZ8brjhBubNmxeQhk2hqlatWnTo0AFD0VEUe4nW5YQUfWE2itNGv359URRF63I09/e//50nn3yC+Lg4LPtXY973Q9j0KAq1lgA1oS88TMy2ZehLTjB48GCeffZZkpOTNa1JgowfKYrCNddcw1NPPUViYjyWrB+J2vmVTHrTgsuBZe/3RO39FotJz4wZM7jjjjvQ6/VaVxb0yluMG3L3alxJaDHm7gEISIv2UNG5c2defvklWrXKwHRiJ9HbPkOxFmhdVo2FWkuAalHdmA7/RvSO/8OoOpk4cSIPPvggZrP2c9MkyARAp06dWPL66/To0QND4SFitn4se5EEkK7oGDFb/4cxdzetW7fm1VdeYcCAAVqXFTIuuugijEYjphM7ZUSxkhSHFWP+QdLT02nevLnW5QSVOnXqsGjRQoYOHYq+NJfY3/+H8XhoP7dCrSVAVSm2YqJ2fIn50K/Url2bRYsWccUVVwTNSKMEmQBJTExkzpw5jB8/HpPiJmr3Six7vpP+Cv7kdmI++CMx2z9Db/c0unv22WepX7++1pWFlLi4OAYOHIjOWuBdhSPOzXh8B7hdnu1LguTFPpiYzWYeeOABZs2aRWyUBcv+H7Ds/hqcVq1Lq54QaglQVYbcPcRu/RhD0VH69evHa6+9Sps2bbQuq4IqBZmpU6dSXCzL56qrfO+JV195hbZt22LM20Pslg9ldMYPykdhTMe20rBhQ/71r2e59dZbw3r3YX8aPnw4AKZjWzWuJAS4XZhythEdE8Oll16qdTVBrW/fvrz22mue/lv5B4jd8jH6ghANyyHSEqDSnDYse74jau93RJn0TJ06lZkzZxIfH691ZX9RpSDz8ccfY7PJCEJNNWrUiEWLFjF27FhMisszOrP7GxRHmdalhT6XA/OBtZ5RGFsR119/Pa+cCo6i+po3b06nTp0wFBxCV5KrdTlBzZi7B8VRytAhQ4iOjta6nKCXmprKU089xdixYzGqdqJ3/h/m/avDaquXUKMvyCZ268cY8/bQtm1bXnvtNS699NKgHV2sUpBRQ/gcZrDR6/Vcd911vPbqq7Rv3x7jyX2e0ZkTu0L6XLGW9AXZxG75CFPONho3bsxzz/2LO++8Mygmo4WDUaNGAWA+9KvGlQQxtwvz4Y2YTGauu+46rasJGTqdjuuuu44XX3yR5s2bYzq+g9itH6EvPKJ1aZHF5cC8/weid36F0W3jtttuY+HChdSrF9wTlqs8R6akpITi4uJz/icqr2HDhjzzzDNMmDCBKKOOqH2rPCubpANm5TmtWPZ+R/TOrzC4rIwePZqXX3456M7jhrpOnTpxwQUXYCjIRl90TOtygpLx+A4UezHDh1+j+ZLUUNSsWTOef/55brrpJvSOUqJ3fIH5wDpwhccy7WCmLzxC7NaPMB3fSfPmzXnppZf4xz/+ERKn46tc4bnO+aqqiqIobNu2rUZFRRqdTsdVV11Fjx49WLBgAevWrSN264dY63fBkdoagnQ4T3OqiuHkPiwH16E4rLRu3ZpJkyZFdPMxf7v11lsZN24c5qyfKG09VJ6bp3PaMB/eSHR0NNdff73W1YQso9HImDFj6NmzJ3PnzmX//t8xFGZT1rgP7rg6WpcXflwOzNm/Ysr5Hb1ez8jRoxkxYkRIBJhyVa500aJFAd8QKlKkpqYyd+5cvv76axYuXAQH12HI24u1SR9Ui/zMT6c4SjHvX4Mx/yBms5lb77rLs/u49IXxqzZt2jBw4EBWrFiB8cQuHCkttS4paJizf0VxWhlzx13yGukDrVq14qWXXuL111/n3XffI2b759jqtsNe/wLQyd+5L+iKjhG1fxU6ayGNmzThoalTadky9P6mqxxkLrjgAhky9SNFURgwYACdO3fm2WefZeXKlcRu/Rhr/c446rQBJcJXzKsqhry9RB1cB04bF1xwAQ8++GDQn8MNJ2PHjmXNmjWo2b/gSGoEBovWJWlOV3IC0/HtNG7ShKuvvlrrcsKGyWTijjvuoHfv3syZM4dDhzZjLMiirMlFuGNkb7RqczsxHdqA+egWFEXhhhtvZPTo0ZhMJq0rq5YIf1cMXomJicyYMYPZs2eTmBCPJesnorZ/Hny7EAeQ4ijDsnslUXu/w2LUMXHiRJ566ikJMQGWnJzMzTffjOK0Yjn4k9blaM/twrL/BwAm3HdfSA3Jh4q2bdvy6quvMnz4cHRl+cRs+xTT4Y2gurUuLeToSnOJ+X0Z5qObadCgPv/617PcfvvtIRtioIpBJi0tDZ1Osk8g9enTh38vWcKAAQMwFOcQ+/vHGEK8C2Z16POziNn6Mcb8g1xwwQUsef31oOosGWmuvvpq2rRpgzF3d8T3QTId+Q19aR5XXXUVHTp00LqcsGWxWLjnnnt4+umnSU2pjfnQes+HO1uR1qWFBlXFeHQzMb9/gq4sn+HDh4dNa4oqpZKvv/6apKQkAPLy8ti8eTNbtmzh5MmTfilOeCQkJDBjxgxmzZpFjMVC1P4fsOz5JjK6ArudmA+sI3rXckw4ueeee3jyySc13WlVeNoHTJ06FZPJjOXAmojtgaQrPo75yG/Uq1ePO+64Q+tyIkKnTp147bXXGDhwoOfD3daPpW3FeSj2EqJ2fokl62dqJSXyxBNPcM8992CxhMdp4SoPr+zatYt//OMf9OrVi2uvvZa///3v9OzZk1GjRrF3r2wq5099+/bl9ddfo2PHjhhP7id26//QFR/Xuiy/UayFRG/7FFPO7zRp0pQXX3zRM7Qso4JBoWHDhtxxx+2eU377foi8NxKXg6h936EAU6ZMkeZ3ARQbG8v06dOZMWMG0RYTUftWYdn7LTjtWpcWdAwn93vCXuER+vTpw2uvvUbXrl21LsunqnQy9/jx44wYMYJatWoxZcoUmjZtiqqq7Nmzh6VLl/KPf/yDTz/9VCYD+1F5F8x33nmHV199jZjtn2FtdCGOlFZhtRRWn3+Q6L3fg8vOsGHDGDt2rDS2C0JXX301P/30Ez/++CPGY1tx1G2ndUmBoapY9q9GZy3kHyNGyCkljQwYMIC2bdsyZ84cNm3ahL40l7Jm/XBHy3sQbjfm7J8xHduK2WLh3gkPMnjw4LA8HV+lj7ZLliyhfv36fPzxx9x000306dOHiy66iDFjxvDRRx9Rr149lixZ4qdSRTm9Xs+IESN44on5xMfFYTmwxtPS2x0GE99UFdOhDUTvWoHZoDBt2jTuu+8+CTFBSqfTMXXqVJKTk7Fk/xLWI4SnM5zYhTFvL5mZmYwePVrrciJa3bp1efrppxk1ahQ6ayEx2z71zCOMYIq9hOgdn2M6tpXGjRvz0osvMmTIkLAMMVDFILNmzRpuvfXWM76pWCwWbrnlFn744QefFSfOrUuXLrzyysu0apWB6cROonb9X2jPm3G7sOz7HvPhDdRLS+P5559n0KBBWlclzqN8hZ0CRO/5BsURojsYV5KuJJeog2uJi4tjxowZskopCOj1em6++WbmzZtHXEy0Zx7hvlUR2RFYX3CImK3/Q1+cwyWXXMLzzz9Penq61mX5VZWCTFZW1jlnOLdr146srKwaFyUqLzU1lYULn+Giiy7CUHiEmO2fheb2Bk47UTv/D2PuHtq1a8fzixfTtGlTrasSldSxY0duvfUWFHuxZ65CuC6LddqI3vM1iupm+vTppKamal2ROE337t155ZVXaN26NcYTu4je8TmKo1TrsgLGeOx3ond+hUlx8cADDzB16lSioqK0LsvvqhRkSkpKiI2NPevtMTExlJZGzpMmWFgsFmbNmsV1113n6bGw44vQWpLotBG980sMRUfp168fTz31FImJiVpXJaroxhtvpHfv3hgKD2PKXq91Ob6nuona+y2KrYgxY8bQvXt3rSsSZ1CnTh0WLVrE4MGD0ZecIOb3T9GV5mldln+pbswH1mI5uI5atZJ49tlnGTp0aNieSvozn28aKTtka0On0zF27FhuueUWFFsRMdu/CI3meU4r0Tu+RF9ygiFDhjBjxgyZDxOiFEVh6tSpNGjQAPPRTRjy9mldkk+ZDq3HUHCIHj16MGLECK3LEedgNBp58MEHufPOO9E5SojZ/hn6/DA9W+CyE7VrBaacbTRv3pwXXniBjIwMrasKqCqd3FVVtVKbRgrtjBw5EqPRyAsvvEDMzv+jpPVQVGOQDi26nETvXI6+NJdhw4Zx7733ytLqEBcTE8Pjjz/OnXfeCftWUWKJD4sVJIbcvZiPbKJhw4ZMmzZNnqchQFEUrr/+eurXr8/s2Y/BrhWUNemNs3YLrUvzGcVRRtTO/0NfmkfPnj2ZPn16RLYBqFKQeeONN/xVh/Ch66+/HrvdzmuvvUbUzq8ozRgMeqPWZVWkuona+w36kuNcfvnljB8/XkJwmEhPT2fGjBlMmzaN6N0rKWl9JaoxdBtv6Upzidr/A9HR0Tz++OPnPL0ugk+fPn1YtGghkyZNhn2rsKpuT7uKEKc4Sone8SW6snyGDRvGPffcE7Gb5lYpyHTr1s1fdQgfGzlyJMePH+eTTz7Bsvc7rM0HBFWfGXPWzxjys+jWrRsTJ06UEBNmevbsyc0338yrr76KZc83lLW6NCQ3PFUcZUTvXomiupgxYwaNGjXSuiRRDRkZGTzzzNPcP3Ei+ftXg9vl2YQ3RCn2UqJ3fIHOWsB1113HnXfeGdGvodV6ZSkqKuLLL7/k1Vdf5bXXXuOrr76iuDgEV8qEMUVRGD9+PF26dMGYfxDj0S1al+RlyNvv6W/QpAmzZs2S5athasSIEVx88cUYio5gzgrBzSXdbix7vkGxFXPLLbfQo0cPrSsSNdC0aVMWPvMMtWrVwnJwHcZjv2tdUrV4RmI+R2ctYMSIEREfYqCKIzIA//vf/5g9e/ZfgktcXByPPPIIgwcP9llxomYMBgPTp0/nlltvJffQL7ji6uCO1Xa5qGIr8vR4sETx6COPROT53EihKAqTJ0/m4MEs9u37HVd0ckjNTzBn/YSh6Ch9+/blH//4h9blCB9IT09n0aJFjB8/ntyD61BNMTiTQqjHistB1M4V6KyF3HTTTYwePTriQwxUcURm69atPPTQQwwcOJCPP/6YTZs28dtvv/Hf//6Xfv36MWnSJLZv3+6vWkU1JCYmMuvhh1FQiNq3CtwaNog61dYdl52JE++XYfoIEB0dzZw5jxMXH0/UgTXoSk5oXVKlGE7swpTzO82aNWPy5MnyZhFGGjRowLx587BYooja+13odKNW3Vj2foe+9ARDhw6VEHOaKgWZt956iwEDBvDPf/6TjIwMTCYTZrOZtm3bMn/+fPr378+///1vf9UqqikzM5O///1v6KwFmA7/plkdhhO7MBQeplevXgwcOFCzOkRg1atXj4dnzkRRVc98kyDfKVtXcoKoA2uIjYvjsccei4iGYpGmefPmPPLILHS4id69IiSaiJqzfsaYf5AuXbpw3333SYg5TZWCzPr167n++uvPevv111/P+vVh2AgrDIwZM4Y6detiPrpZm2Z5LjuW7F+IioqSP8II1KVLF26//TYUewmWPd8Gb+dfp9XbuffhmTOpV6+e1hUJP+nevbtntaSjjKhgfk7yx7zCJk2a8sgjj8i8wj+pUpDJycmhcePGZ729cePGHDt2rKY1CT+Iiorizjvu8HSA1KDrqunoFhSnlREjRpCSkhLw4wvtXX/99d7Jv6ZDG7Qu569Ulai933sn93bt2lXrioSfXXXVVQwaNAh9SQ6mI5u0LueMFHspUQdWYzabefTRR4iJidG6pKBTpSBTVlZ2zq6rJpMJm63ymxa6XC6eeeYZ+vfvT/v27Rk4cCDPPfdche7AU6ZMoVWrVhX+u+WWW6pStjilb9++tGqVgTFvD7rSk4E7sNOK+dgWkmvXZvjw4YE7rggqiqIwadIk6tevj/nIb+gLsrUuqQLT0c0YCrK58MILufHGG7UuRwTI+PHjSU1NxXx4Q/DNl1FVz+aXTht33XUXDRs21LqioFTl8alVq1YRFxd3xtuKiqp2yuLll1/mnXfeYd68eTRv3pwtW7YwdepU4uLiGDVqlPd+ffr0Ye7cud7rJpOpqmULPG8kY8aMZsqUKZiObcHapE9AjmvK2Q4uJ/+48UYsltBtjCZqLiYmhlmzZjFu3Dii9n5PSdurUE3af8LUFx3FfOhXUlJSmDp1qnTujSCxsbFMmzaN++67j6j9qyhpOyxoeh4ZcndjKPRsi3HFFVdoXU7QqnKQmTJlis8OvmHDBgYMGEDfvn0Bz2zyzz77jE2bKg7xmUwmOR3hI927dyc9PZ0DB/dgq98Z1eTn5c9uF6acbcTGxnL55Zf791giJLRo0YJ77rmHBQsWYNn3PWUtL9O2WaPTRtTe79EpCrNmzSIhIUG7WoQmOnTowNChQ/nkk08wntiNI6Wl1iWB24nl0HpMJjP333+/zCs8hyoFGV8vre7UqRNLly5l3759NGnShO3bt/Prr7/+JSz99NNP9OjRg/j4eC688ELuu+8+kpKSqnQsl8vly9JD2vDhw1mwYAHG3F3Y63Xw67EM+QdQHGUMHX49JpNJfg8CgMGDB/PTTz/xww8/YDy2FUfddprVYjmwDsVezOibbyYjI0OeoxFq5MiRfPXVV6iH1+Oo1RT02k6oNR37HcVewt//8Q9q1aoVkc/Lyv6bffqbys3N5f333/dsGFcJt99+O8XFxVx++eXo9XpcLhcTJkzgyiuv9N6nT58+DBo0iAYNGpCVlcWCBQu47bbbeO+996q0r8TmzZur/O8JV8nJyRiNRtzHd2Gv296vn4aNx3cBnongGzdu9NtxROgZNGgQG3/7DbJ/wRWfhju6VsBrMOTuxZi3h8aNG5ORkSHP0Qh30UUXsXz5ckw5v2Ov1167Qpx2TEc2ER0TQ5s2beR5eR4+DTLHjx9n4cKFlQ4yX3zxBZ988glPPfUUzZs3Z9u2bcydO5fU1FSuvvpqAIYMGeK9f/lk34EDB3pHaSorMzMzYjfUOpO+ffuyfPlydCUncMf657Sd4ijDUHiYdu3aMWjQIL8cQ4Q2k8nE1KlTsez7gdI2QwM6N0FxlGE5uBaLJYrZs2eTlpYWsGOL4NSiRQtWrfoB9/Ht2OtmanbK05i3B8Vl58YbRnHhhRdqUkMwcLlclRqE0HTsbP78+dx+++3esNKqVSsOHz7Miy++6A0yf9awYUOSkpI4cOBAlYKMXq+XIHOaAQMGsHz5cgwn92P3U5AxnDwAqPTv319+9uKMevToweWXX84XX3yB8djvAT3FZD74I4rTxh13jZfVIAKA+Ph4Bg0ayCeffIK+8DCuhPqBL0JVMR7fgV6vZ8iQIfLaWQmaTs22Wq1/mcCk1+srLL/+s6NHj5Kfny+Tf2voggsuIDo6GuPJA347huHUY/fpE5jVUSI0jR07lsSkJCyH1gesWaM+Pwtj3l7atm3LVVddFZBjitBQ/sHaeHyHJsfXleaiL82jV69eVZ4LGqk0DTL9+vXjhRde4NtvvyU7O5vly5fz+uuve9vXl5SUMG/ePDZu3Eh2djZr165l3LhxpKeny5tjDZlMJrp164bOVohiLfT9AVxODMVHadmypYROcU7x8fGMv/decDsxH/zR/wd0u7Ac/BG93sCDDz4oS61FBa1ataJJkyYYC7I02ZvOcHI/AJdddlnAjx2qqnRq6fReLmeSl5dXpYNPnz6dhQsX8sgjj5Cbm0tqairXXXcdd911F+AZndm5cycff/wxRUVFpKam0qtXL8aPHy+9ZHygW7dufPvttxgKsnFY2vj0sfVFR8Dtolu3bj59XBGe+vbty7Jly9iwYQOOgkN+HdI3HduKzlbI8GuvPWenchGZFEWha9eu7Nu3D33xcVzxgd2mwlB4GL3BQKdOnQJ63FBWpSDz+++/n/c+Xbp0qfTjlTcimjZt2hlvt1gsvPrqq5V+PFE15b8rfeERHHV8G2QMhUcqHEOIc1EUhbvvvptbb70NS9aPlMQP88vEX8VRhvnIbyQkJFRouinE6Tp37szSpUs982QCGWScNvQlubTr0F42K62CKgWZN9988y9fK5/PIs16Qk9qaippaWkcyjkKqurTGfr6oiMYTSZat27ts8cU4a1Zs2ZcccVQli1bhiF3D87aLXx+DNPh38Dl4NZbbyU2Ntbnjy/CQ2ZmJjqdDn1xYPcO1JccB1QZjamian/kef/99xk6dCiZmZlkZmYydOhQ3n//fV/WJgKgY8eOKE4burJ83z2oy46+NI+2bdqcc28uIf5s1KhRGI1GzIc3gtu3uxEr9hJMJ3aQlpYmXabFOUVHR5OWlobeWhDQ4+rKPMdr2rRpQI8b6qoVZBYuXMicOXPo168fCxcuZOHChfTr1485c+awcOFCX9co/CgzMxPAp5889MWeTxXljy1EZdWuXZsrr7wSna0IY+5unz626chv4HYxevRoDAZtu7aK4NewYUMURxk47QE7pu5UcJJ2AFVTrSDzzjvvMHv2bCZOnMiAAQMYMGAAEydOZPbs2bz99tu+rlH4kX+CjOex2rXTru28CF033ngjRpMJ09HNnlOePqA4yjCd2EX9+g3o37+/Tx5ThLfyMKEL4KiMzlaAoijSnLGKqhVknE7nGd+k2rZtG5H7QYSy+vXrE5+QgL44x2eP6RmRgTZtfDuBWESG5ORkLhk0CJ21AH1Btk8e03h8B7hdXHvt32U0RlRKrVqeLTMUpzVgx1QcVuLi4uSUfBVVK8hcddVVvPPOO3/5+tKlS2Wr8RCjKApt27RBZyvyDKPWlKqiLzlOo0aNiIuLq/njiYg0fPhwAEzHttT8wU7bgf2SSy6p+eOJiFA+GVxxBe7UkuKyyyT0aqj2R5MPPviA1atX06GDZ/fkTZs2cfjwYYYNG1ah38zUqVNrXqXwqzZt2rB27Vp0JcdxJTaq0WPprPkoLjtt27b1UXUiEjVt2pQLLriA9evXo1gLUS3x1X4sQ/5BFEcZV/ztBlnSKiqt/IOY4rQF7Jg6p00+AFZDtYLMzp07vacNDh48CEBiYiKJiYns3LnTez9Zkh0aykOHvjin5kHm1GklCTKipoYMGcL69esx5u7GXv+Caj+O8YRnB3ZZqSSqwmKxeC64Azhdwu3847ii0qoVZM7UT0aEroyMDBRF553bUhPlc21kfoyoqd69exMdHY16Yjf2tE7V6nOk2EsxFByiTZs2NGpUs5AuIkv5Zo0Kvm0DcFanJrbLJpFVJ5uMCKKjo2natAmGkuOg1uyPVl+cQ3R0tLR+FzVmNpvp168fir0YXUn1QrZn3xpV5saIKvPuweWjlXPnp1Y8rqg0+YkJ4NRSabcTXWnV9suqwGlDb82nTZs28scofOKiiy4C/thJvaoM+QdQFEU2mRVVFvCpEapGxw0D8m4jgD96vtSkn0z5aaX27dv7pCYhOnXqhCUqCuPJA1X/ZOy0YSg6Sps2bUhOTvZPgSJsuX3cWfq8FI2OGwYkyAjgtMZ4RTUIMkVHAWmEJ3zHZDLRvVs3dLZCFFtRlb7XUHgYVJWePXv6qToRzryBImAjJJ7jqAE7lRU+JMgIAOrWrUtqaiqG4mPVPidsKDqK3mCQib7Cpzp37gyAoehIlb5Pf2oH9vLvF6Iq/hgZCVCQURRAkaay1SBBRnh17NgRxVFWvZbcLgf60lxaZ2TI8kHhUxdc4Fl6rS88XKXvMxQdISYmhubNm/ujLBEh1EAFGQhYZgo3EmSEV/nW8foqfvKFU3NrVDcdO3b0cVUi0tWvX59atZKrtI1GeSDPzMyULQmECHMSZISXN8gUVj3IGE59T/mnZyF8RVEU2rRpjc5eguIordT36EpOANLPSIhIIEFGeNWtW5d69ep55iJUsZ+MvvAwRqNROvoKv2jdujXwR0A5H/2p+2VkZPitJhHevA3xathbq9JUt2evOmmIV2USZEQFXbp0QXHaqtRPRnFY0Zfm0r59e9m1VfhFy5YtAdBX8nmpK82t8H1CVJX3tcztDMwBT22FIK+hVSdBRlTgXSFScKjS36MvOlzhe4XwtfJO0bqyk5W6v74sn6SkJBITE/1XlAhr5YFCCdBeS4oEmWqTICMquOCCC1AUpUorRPQFnvt26dLFX2WJCFe7dm1iYmLQleWf/85uJzpbIU2aNPF7XSJ8RUdHey647IE54KnjeI8rKk2CjKggPj6eVq1aefrJuBzn/wZVxVB4iPiEBFnmKvxGURTS09PRWwvO2+dIZy0EkE0iRY3UqlULAJ2zLCDHUxxlFY4rKk+CjPiLLl26gOr2duo9F8VaiM5eQpfOnWV/JeFXaWlpoLpR7CXnvJ/uVAfg+vXrB6IsEaaio6OxWCzegOFvOgky1SbvPOIvunbtCpxq8X4ehsJDFb5HCH8pDya682xVoNg8IzJpaWl+r0mEt+TkZHT2yi35rynFUeI9pqgaCTLiL9q0aYPFYqnUPJny+8hEX+FvdevWBUCxF5/zfjqb5/Z69er5vSYR3ho2bOjpXeT0/zwZXVmB95iiaiTIiL8wGo107NgRfdnJczcgU90Yi47QsGFDUlNTA1egiEjlzzHdeU4tlZ96SklJ8XtNIrylp6cDoLPm+/1YOms+ik5HgwYN/H6scCNBRpxR+QjLubr86kpOgMshozEiIMqDyXnnyNhLsVgsxMbGBqIsEcbKg4y+ksv+q01V0Zflk5aWhslk8u+xwpAEGXFGldmuwCC7C4sAql27NsB55yzoHKXUrl0bRZEd+ETNtGrVCgBd8XG/HkexFaE4rbSWTtTVIkFGnFHTpk2Jj4/3bFdwFvqiIyiKQocOHQJYmYhUlVpFoqoozjKZMCl8onHjxsTExHjaUfiR/tTjZ2Zm+vU44UqCjDgjnU5Hhw4d0NmKzjyU73ZjKM7xBh4hAiE5Ofmc87YUpxVUVZawCp/Q6/W0a9cOnbXAr8uw9UUSZGpCgow4q/bt2wOcsZ+MrvQEuJ0yGiMCKikpCd2psHImisPqvZ8QvtCxY0eAKnU7r5LTmoqWz8kRVSNBRpxV+acD/RmGVWUoVGghMTHRs0vwWdrGK6e6sMoeS8JXevToAYAhP8svj68rO4nOXkKPCy+Una+rSYKMOKvmzZtjMpvRn5ropuqN3v+Xf61t27aa1SciT3lAUZzWM95ePiIjQUb4Snp6OnXr1sNQmA1ut88f35B/EPgjMImqkyAjzspgMNA6IwN9aR64nNjTOuFMaIg9rRP6kuPUrl1b+seIgEpISABAcdrOeHt5wCm/nxA1pSgKvXr1RHHa0Z9j8UN1GU4eQG8wyKa7NSBBRpyTZ/mhir40F1d8PcpaDsIdlYDOXkLr1q21Lk9EmD+CzFlGZE4FHAkywpf69+8PgDFvr08fVykrQF+ay4Xdu0vfoxqQICPOKeNUXwNdyQnv18ovl/dYECJQylfIlZ9C+rPygCMr6YQvtWnThrp162I8eQDcTp89rjFvD/BHUBLVI0FGnFOLFi0A0Jfleb+mL82rcJsQgeINMmc9tSQjMsL3FEVhwIAB4LL7btKvqmLM3YvFYqFnz56+ecwIJUFGnFNaWhpmsxld6R9Bpvxy8+bNtSpLRKjzz5HxfF1GZISvXXLJJQAYT+zyyePpi4+hsxXSt29foqKifPKYkUqCjDgnvV5PkyZN0FvzPcte8SwXjIuLk6ZjIuC8IzKus59askRFYTQaA1mWiADp6em0bdsWQ8Gh8+73VRnlgWjw4ME1fqxIJ0FGnFd6ejq4XSi2YnC70NsKadKkiexlIwLuj1NLnj4yp7cE8HzdRoKMxgg/8YQOFeOJ3TV7IJcdY94+0urXl15cPiBBRpzX6VvZ62xFoKrSgVJoIjY2FkVRvKeQTm8JAKBz2WV+jPCbfv36YbFYMJ7Yedbu0pVhzN0LbidDBg+WD4Q+IEFGnFeDBg0A0FkLUayFANSvX1/LkkSE0uv1xMTEelcnlbcEcMXX8zQrc9mJi4vTuEoRrqKjoxk4cCA6W1GNtiwwHt+BXq/n8ssv92F1kUuCjDgvb5CxFaKzSZAR2kpIiEc5wxYFiksm+gr/Gzp0KOAJI9WhKzmBvjSXXr16yTxDH5EgI86rbt26AOhsJehsxYBnNZMQWoiPj0d3hlVLsvRaBEKrVq1o1qwZxvyD1doR23h8J/BHIBI1J0FGnFd0dDRxcXEo9iIUuyfI1KlTR+OqRKSKj4/3NCX7U2Oy8iAjp5aEPymK4gkhqhtDbhUn/bocmPL2UqdOHdmSwIckyIhKSU1NRWcvRWcvISoqWtppC838eeWSl5xaEgEycOBAjCYTpuN/nfT755V0pzOc3A8uO4MHD0ank7dfX5GfpKiUlJQUFJcdnbWQlJTaWpcjItgfQaZiLxlphicCJS4ujn59+6KzFqArzqlw259X0p3OeHwniqLIJF8fkyAjKiU5ORkAxe2QCWpCU+Wnjv7c3VdOLYlAuuyyywAw/un0UoWVdKdRrIUYio/RuXNnUlNTA1ZnJJAgIyolKSnpjJeFCLQ/uvtKkBHa6dixI6mpqZjy9lVqI0ljrmeDyEsvvdTfpUUcCTKiUk5fCSJBRmjpjxGZinNk5NSSCCSdTufZf6kyG0mqKsbc3ViioujTp09gCowgEmREpZweZOSNQmjJ+/z786kll4zIiMAaOHAgAIa8fee8n640F52tiIv69MFisQSitIgiQUZUyunhRd4ohJa8IzJ/ObVkr3C7EP7WuHFjGjdujLEgG1yOs96vPOj07ds3QJVFFgkyolJiYmK8l2XptdDSuSb7WiwW2flaBFS/fv3A7Tz76SVVxXhyPzExMdI7xk8kyIhKOT3IREdHa1iJiHRnnSPjsslojAi48jkvZwsyujLPZrs9evTAZDIFsrSIIUFGVEpUVJT3sgQZoaXyEcE/77ekyIaRQgNNmjQhNTUVQ2E2qO6/3G4oOAhAjx49Al1axJAgIyrl9CAjk9WElgwGA2azueIcGVVFcdnltKcIOEVR6NGjB4rThr74+F9u1+dnoyg6unbtqkF1kUGCjKiU08OLBBmhtdjY2IqnltxOUNUKp0CFCJTu3bsDoC88VPEGlwNDSQ5t27aR1Z5+JEFGVMrpEyjNZrOGlQhxKsi4/1glUn6aSUZkhBbat2+PoujQFx2t8HV98TFQVTp27KhNYRFCgoyoFL1e770sc2SE1mJiYirMkVFOLX2VERmhhdjYWFq0aI6h5Di4Xd6vlwcbCTL+ZdC6ABE6nnrqKYqKimSvJaG5mJgYzxuG2wU6PZwKNRKyhVY6duzIzp070ZecwBVXBwB90TH0ej1t27bVuLrwJkFGVFrnzp21LkEI4LTA4naATi8jMkJzGRkZAOjKg4zqxlCaS5OmTSoslhC+p+mpJZfLxTPPPEP//v1p3749AwcO5LnnnkNVVe99VFVl4cKF9O7dm/bt2zN69Gj279+vXdFCCM2VB5byAFP+fxmREVpp2bIlAPrSEwDorAXgdnq/LvxH0yDz8ssv88477zBz5kw+//xzHnjgAV555RXefPPNCvd58803mTVrFkuXLiUqKopbbrkFm812jkcWQoSz8k+45QEGtwQZoa20tDRiYmLQleYCoCvx/F+CjP9pGmQ2bNjAgAED6Nu3Lw0aNOCyyy6jd+/ebNq0CfCMxrzxxhuMHTuWgQMHkpGRwfz588nJyWHFihVali6E0JB3qN7tBP4INDKEL7Si0+lo0qQJemshqG501nwAmjZtqm1hEUDTOTKdOnVi6dKl7Nu3jyZNmrB9+3Z+/fVXpkyZAkB2djbHjx+nZ8+e3u+Ji4ujQ4cObNiwgSFDhlT6WC6X6/x3EkKEhPIWAN5TS6cCjdlslr91oZkGDRqwZcsWFFuR59QSUL9+fXlOVlNlf26aBpnbb7+d4uJiLr/8cvR6PS6XiwkTJnDllVcCcPy4p0ticnJyhe9LTk7mxIkTVTrW5s2bfVO0EEJzubmeYXtvL5lTgSYrKwuDQdYwCG2UP/d01gJ0ZQVERUezb98+FEXRuLLwpulf/BdffMEnn3zCU089RfPmzdm2bRtz584lNTWVq6++2qfHyszMrNALRQgRuo4cOeK5cKpnR/mITPv27WnWrJlWZYkIV1xczKefforOWojeXkR6yxZ06tRJ67JClsvlqtQghKZBZv78+dx+++3eU0StWrXi8OHDvPjii1x99dWkpKQAnk9fqamp3u/Lzc31LnWrLL1eL0FGiDDhnex7KsCUz5WJjo6Wv3OhmXr16gGgL80Ft4u6devK8zEANJ3sa7Va/zLkptfrvcuvGzRoQEpKCmvXrvXeXlxczG+//SYpV4gI5t3vy3Vqsu9pc2SE0Er5B2598bEK14V/aToi069fP1544QXS0tK8p5Zef/11hg8fDnh2FR01ahTPP/886enpNGjQgIULF5KamsrAgQO1LF0IoSHvZF/viIznFJNsaCq0lJCQgMlkwm4rBvCeVRD+pWmQmT59OgsXLuSRRx7xnj667rrruOuuu7z3ue222ygrK2PmzJkUFhbSuXNnXnnlFfnkJUQE8wYWt4zIiOChKAqJiYnk5OQAyHYuAaJpkImNjWXatGlMmzbtrPdRFIXx48czfvz4AFYmhAhmfx2RcaIoSoVd2oXQwulBJiEhQeNqIoPsfi2ECDkmk8lz4bRVSyaTSZa5Cs0lJiZ6L0uQCQwJMkKIkFN+aklRTzXMcrswy/wYEQRiY2O9l+Pi4jSsJHJIkBFChBzvXBjvqiWXzI8RQeH0HdhlN/bAkCAjhAg53jky6h+TfS0SZEQQOD28yCamgSFBRggRcryjL+WTfVWnLL0WQeH0jUulGV5gSJARQoQcnU6H0WhEcZVP9pVTSyI4eCeii4CRICOECElms8UzIqO6PZN9JciIICDPw8CTICOECElRURZPHxnp6iuCiIzIBJ4EGSFESLJYPEFGcTu814XQmsyLCTwJMkKIkFQeZMqXYEuQEcFAgkzgSZARQoSkqKgoz9YEbgkyInhIkAk8CTJCiJBksVhAdaM4bUDFZa9CaEW2yQg8CTJCiJBUHlwUZ1mF60JoSYJM4EmQEUKEJG+QcUiQEcFDgkzgSZARQoQkb5Cxe4KMzJERwUBVVa1LiDgSZIQQIak8yOgcpYDsayOCg9vt1rqEiCNBRggRksqDi5xaEsFERmQCT4KMECIk/TFHRkZkRPBwndr/SwSOBBkhREj6Y0RGgowIHhJkAk+CjBAiJHnnyEgfGRFEnE6n1iVEHAkyQoiQ9OcRGBmREcFAgkzgSZARQoSkPwcXGZERwcDhcGhdQsSRICOECEmnBxedTofJZNKwGiE8ZEQm8CTICCFC0ukjMlFRUdJRVQQFu92udQkRR4KMECIknd7JV+bHiGBxepCRnjKBIUFGCBGSTj+1JNsTiGBxepCR0ZnAkCAjhAhJZrPZezpJJvqKYCFBJvAkyAghQpKiKJjNZkCCjAgeEmQCT4KMECJkGQwGAG+gEUJrVqv1jJeF/0iQEUKEPAkyIljYbLYzXhb+I0FGCBHyJMiIYHH66SQJMoEhQUYIEfKkGZ4IFqefTpIgExgSZIQQIU+CjAgWcmop8CTICCFCngQZESwkyASeBBkhRMjT6/ValyAEIEFGCxJkhBAhT1rBi2Ahc2QCT4KMECJkSYARwcYmDfECToKMEEII4QOqquJwOLzXJcgEhgQZIUTIateuHQC1atXSuBIhwOVyobrdqIrnrVWCTGAYtC5ACCGq64EHHuCnn36iX79+WpcihDe4qHoTitMqQSZAJMgIIUJWSkoKQ4YM0boMIQBwOp2eC3ojOK0VTjMJ/5FTS0IIIYQPlAcXVecZI/AGG+FXEmSEEEIIH3C5XJ4LOn3F68KvJMgIIYQQPlAeXGREJrAkyAghhBA+4Ha7PRcUBZA+R4EiQUYIIYTwgT+Ci+5P14U/SZARQgghfODPIzIyRyYwJMgIIYQQfqCcCjTCvyTICCGEED7wR3BR/3Rd+JMEGSGEEMIHvMFFlSATSBJkhBBCCB/Q6z39Y1DdFa8Lv5IgI4QQQvhAeXBR3M4K14V/SZARQgghfMAbXNyuiteFX0mQEUIIIXzgjxEZCTKBJEFGCCGE8AGTyeS54HJUvC78SoKMEEII4QNGoxEAxW2vcF34l0HrAoQQQohwYDAYPEuuT51akiATGDIiI4QQQviAoigYjX+cTjKbzRpWEzkkyAghhBA+cvq8GJkjExgSZIQQQggfMZtlRCbQJMgIIYQQPmKxWLyXJcgEhgQZIYQQwkdODy8SZAJD01VL/fv359ChQ3/5+o033sjDDz/MyJEj+emnnyrcdt111/Hoo48GqkQhhBCi0iTIBJ6mQeaDDz7A5XJ5r+/atYsxY8Zw2WWXeb927bXXcu+993qvR0VFBbRGIYQQorIkyASepkGmVq1aFa6/9NJLNGrUiG7dunm/ZrFYSElJqfGxTg9MQgghhD+cHl5MJpO899RAZX92QdMQz263s2zZMsaMGeNpKHTKJ598wrJly0hJSaFfv36MGzeuWqMymzdv9mW5QgghxF+UlZV5L+/Zs4eCggINq4kMQRNkVqxYQVFREVdffbX3a0OHDiUtLY3U1FR27NjBk08+yb59+/jXv/5V5cfPzMyUDbyEEEL41Zdffum93LFjR2rXrq1hNaHN5XJVahAiaILMf//7Xy666CLq1Knj/dp1113nvdyqVStSUlIYPXo0Bw8epFGjRlV6fL1eL0FGCCGEX52+/Do6OlredwIgKJZfHzp0iDVr1vC3v/3tnPfr0KEDAAcOHAhEWUIIIUSVSGffwAuKIPPhhx+SnJxM3759z3m/bdu2Afhk8q8QQgjha3+e7Cv8T/NTS263mw8//JBhw4ZhMPxRzsGDB/nkk0+4+OKLSUxMZMeOHcydO5euXbuSkZGhYcVCCCHEmZ0eZE5fuCL8R/Mgs2bNGg4fPszw4cMrfN1oNLJ27VreeOMNSktLqVevHpdccgnjxo3TqFIhhBDi3IxGo9YlRBzNg0zv3r3ZsWPHX75er1493nrrLQ0qEkIIIapHTicFXlDMkRFCCCHCgYzIBJ4EGSGEEMJHZEQm8CTICCGEED5y+qIVERgSZIQQQggfkVNLgSdBRgghhPAR6eQbeBJkhBBCCB+RU0uBJ0FGCCGE8JGkpCQA2rdvr3ElkUOioxBCCOEjrVq1YvHixTRo0EDrUiKGBBkhhBDCRxRFoU2bNlqXEVHk1JIQQgghQpYEGSGEEEKELAkyQgghhAhZEmSEEEIIEbIkyAghhBAiZEmQEUIIIUTIkiAjhBBCiJAlQUYIIYQQIUuCjBBCCCFClgQZIYQQQoQsCTJCCCGECFkSZIQQQggRsiTICCGEECJkhf3u16qqAuByuTSuRAghhBCVVf6+Xf4+fjZhH2TcbjcAmzdv1rgSIYQQQlRV+fv42Sjq+aJOiHO73TidTnQ6HYqiaF2OEEIIISpBVVXcbjcGgwGd7uwzYcI+yAghhBAifMlkXyGEEEKELAkyQgghhAhZEmSEEEIIEbIkyAghhBAiZEmQEUIIIUTIkiAjhBBCiJAlQUYIIYQQIUuCjKiU7OxsWrVqxbZt2856nw8//JAuXboEsCohhAgtrVq1YsWKFVqXEVYkyAghhBAiZEmQEUIIIUTIkiAjvL7//ntuuOEGunTpQvfu3bnjjjs4ePBghftkZWUxcuRIOnTowJVXXsmGDRs0qlaEM7fbzcsvv8ygQYNo164dffv25fnnn/ee4vzqq6/O+jwsP8W5YsUKLrnkEjIzM7nllls4cuSIhv8iEU6+/PJLrrjiCtq3b0/37t0ZPXo0paWlbNq0iTFjxtC9e3c6d+7MiBEj2Lp1q9blhj0JMsKrrKyMMWPG8N///pclS5agKAp33XVXhZ1Hn376aW655RY+/vhjGjduzMSJE3E6nRpWLcLRU089xcsvv8y4ceP4/PPPefLJJ6ldu7b39vM9D61WK88//zzz5s3jnXfeobCwkAkTJmjxTxFhJicnh4kTJzJ8+HA+//xz3njjDQYNGoSqqpSUlDBs2DDefvttli5dSnp6OrfffjvFxcValx3eVCHOIjc3V23ZsqW6Y8cONSsrS23ZsqW6dOlS7+27du1SW7Zsqe7evVtVVVX973//q3bu3FmrckWYKCoqUtu1a1fhuVauss/Dli1bqhs3bvTeZ/fu3WrLli3V3377zf//ABHWtmzZorZs2VLNzs4+731dLpfaqVMn9euvv/Z+rWXLlury5cv9WWLEMWgdpETw2L9/P4sWLeK3337j5MmTqKc2Rj9y5AjNmjUDPDPuy6WkpACQl5fnvV2Imtq7dy92u50LL7zwrPc53/PQYDCQmZnpvU+zZs2Ij49nz549tG/f3k+Vi0iQkZFBjx49uOKKK+jduze9e/fm0ksvJSEhgRMnTvDMM8/w008/kZubi9vtpqysjMOHD2tddliTICO87rzzTurXr89jjz1GamoqbreboUOH4nA4vPcxGo3ey4qiAFQ49SRETZnN5vPeR56HQit6vZ7XX3+d9evXs3r1at58802efvppli5dyqxZs8jPz2fatGmkpaVhMpm47rrrKryGCt+TOTICgJMnT7Jv3z7Gjh1Ljx49aNasGQUFBVqXJSJQ48aNsVgsrFu3rtqP4XQ62bJli/f63r17KSwslJFD4ROKotC5c2fuvfdePv74Y4xGIytWrGD9+vWMHDmSiy++mBYtWmAymTh58qTW5YY9GZERACQkJJCYmMh7771HSkoKhw8f5qmnntK6LBGBzGYzt912G0888QRGo5ELLriAvLw8du3aRY8ePSr1GEajkdmzZzN9+nT0ej2zZ8+mY8eOclpJ1Nhvv/3G2rVr6dWrF8nJyfz222/k5eXRtGlTGjduzLJly8jMzKS4uJj58+djsVi0LjnsSZARAOh0Op5++mkee+wxhg4dSpMmTZg+fTojR47UujQRgcaNG4der2fRokXk5OSQkpLC9ddfX+nvt1gs3HbbbUycOJFjx47RpUsXHn/8cT9WLCJFbGwsP//8M//+978pLi4mLS2NKVOmcPHFF5OSksKMGTO4+uqrqVevHhMmTGD+/Plalxz2FLV8RqcQQoSBDz/8kDlz5vDLL79oXYoQIgBkjowQQgghQpYEGSGEEEKELDm1JIQQQoiQJSMyQgghhAhZEmSEEEIIEbIkyAghhBAiZEmQEUIIIUTIkiAjhBBCiJAlQUYIEXL69+/PkiVLtC5DCBEEJMgIIYQQImRJkBFCBAVVVXE6nVqXIYQIMRJkhBDVMnLkSB599FEeffRROnfuTPfu3XnmmWco77H58ccfc80119CpUyd69erFxIkTyc3N9X7/jz/+SKtWrfjuu++45ppryMzM5Ndff+XgwYOMHTuWnj170qlTJ4YPH86aNWvOWcvhw4cZO3YsnTp14oILLmD8+PGcOHGiwn0WL15Mjx496NSpE9OmTePJJ5/kqquuqvDv+fPGkuPGjWPKlCne63a7nXnz5tGnTx86duzI3//+d3788cdq/wyFEDUnQUYIUW0fffQRer2e999/n2nTprFkyRLef/99AJxOJ+PHj2fZsmU899xzHDp0qEIoKPfUU08xceJEPv/8c1q1akVpaSkXX3wxS5Ys4aOPPqJPnz7ceeedHD58+Iw1uN1uxo0bR0FBAW+++Savv/46WVlZTJgwwXufZcuW8cILL/DAAw/w4YcfUq9ePd55550q/3sfffRRNmzYwNNPP82yZcu47LLLuPXWW9m/f3+VH0sI4RsGrQsQQoSuevXq8dBDD6EoCk2bNmXnzp0sWbKEa6+9lr/97W/e+zVs2JBp06bxt7/9jZKSEmJiYry33XvvvfTq1ct7PTExkYyMDO/1++67jxUrVvD1118zYsSIv9Swdu1adu7cycqVK6lXrx4A8+fPZ8iQIWzatIn27dvz1ltv8be//Y3hw4cDcPfdd7N69WpKS0sr/W89fPgwH374Id988w116tQB4JZbbmHVqlV8+OGH3H///ZV+LCGE70iQEUJUW4cOHVAUxXu9Y8eOvP7667hcLrZt28a//vUvtm/fTkFBgfeU05EjR2jevLn3ezIzMys8ZklJCf/617/49ttvOX78OC6XC6vVetYRmT179lC3bl1viAFo3rw58fHx7N27l/bt27Nv3z5uvPHGCt/Xvn171q1bV+l/686dO3G5XFx22WUVvm6320lMTKz04wghfEuCjBDC52w2G7fccgu9e/fmySefJCkpiSNHjnDLLbfgcDgq3DcqKqrC9Xnz5rFmzRomT55Mo0aNsFgs3HvvvX/5Pl9TFIU/76F7+uTj0tJS9Ho9//3vf9Hr9RXuFx0d7dfahBBnJ0FGCFFtmzZtqnD9t99+Iz09nb1795Kfn88DDzzgHSnZsmVLpR5zw4YNXH311QwaNAjwjNAcOnTorPdv1qwZR48e5ciRI95j7d69m8LCQpo1awZAkyZN2Lx5M8OGDfN+3+bNmys8Tq1atTh+/Lj3usvlYteuXXTv3h2A1q1b43K5yMvLo0uXLpX6twgh/E8m+wohqu3w4cPMnTuXvXv38umnn/LWW28xatQo0tLSMBqNvPnmm2RlZbFy5UoWL15cqcdMT09n+fLlbNu2je3btzNx4kTcbvdZ79+zZ09atmzJAw88wNatW9m0aROTJk2iW7du3tNWI0aM4IMPPuCjjz5i//79LF68mB07dlQ4LXbhhRfy3Xff8e2337Jnzx5mzZpFYWGh9/YmTZpwxRVXMGnSJL766iuysrLYtGkTL774It9++231foBCiBqTERkhRLUNGzYMq9XK3//+d/R6PaNGjeK6665DURT++c9/smDBAt58803atm3L5MmTGTt27Hkfc8qUKTz00ENcf/31JCUlcdttt1FSUnLW+yuKwuLFi5k9ezYjRoxAURT69OnDjBkzvPe58sorycrKYt68edhsNi6//HKuvvrqCqMyw4cPZ/v27UyePBm9Xs/o0aO9ozHl5s6dy/PPP88///lPcnJySExMpGPHjvTt27fqPzwhhE8o6p9PCgshRCWMHDmSjIwMpk2bpnUp1TJmzBhq167NE088oXUpQogakBEZIUTYKysr491336V3797odDo+++wz1qxZw+uvv651aUKIGpIgI4QIe4qi8N133/HCCy9gs9lo0qQJzz77LD179tS6NCFEDcmpJSGEEEKELFm1JIQQQoiQJUFGCCGEECFLgowQQgghQpYEGSGEEEKELAkyQgghhAhZEmSEEEIIEbIkyAghhBAiZEmQEUIIIUTI+n88I9TwNupV8AAAAABJRU5ErkJggg==",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Set the style for better-looking plots\n",
    "sns.set_style(\"whitegrid\")\n",
    "\n",
    "# Violin plot\n",
    "sns.violinplot(x='paralogue', y='pLDDT', data=pLDDT_score)\n",
    "plt.xlabel('paralogue')\n",
    "plt.ylabel('pLDDT')\n",
    "\n",
    "# Adjust layout and display the plot\n",
    "plt.show()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Comparison: ahl vs cnp, p-value: 0.9004, Cohen's d: -0.05\n",
      "Comparison: ahl vs sal, p-value: 0.0059, Cohen's d: 0.34\n",
      "Comparison: cnp vs sal, p-value: 0.0025, Cohen's d: 0.36\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAArcAAAIvCAYAAABuqPNZAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjkuMCwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy80BEi2AAAACXBIWXMAAA9hAAAPYQGoP6dpAAC6oklEQVR4nOzdd3hT5fvH8ffJ7G7pYJZVZO8lG0WGIuAAUUA2ooIKKEMERIZMUZaKLGUJKEMEFBVRGbJk770KhdJBR7qyzu+PSn72y5Bi4STt/bquXtAzkk+aNrnznGcoqqqqCCGEEEIIkQvotA4ghBBCCCFETpHiVgghhBBC5BpS3AohhBBCiFxDilshhBBCCJFrSHErhBBCCCFyDSluhRBCCCFEriHFrRBCCCGEyDWkuBVCCCGEELmGFLdCCCGEECLXkOJWCCGEEELkGlLcCpGHXL16lffee4+GDRtSqVIlmjRpwocffsiNGze0jvbQjRw5kvLly7Nhw4Zb9s2cOZOyZcsycuTILNuPHz9O2bJluXz5cpbt3333He3ataNq1apUr16dzp078/vvv7v2jx07lpYtW942R1RUFOXLl2fTpk2ubbNnz6Z8+fLMmzfvvh7b0KFDKVu27C1fFy9evK/bA/jjjz9o3749VapUoXbt2vTt2/eux8fGxjJ06FAaNmxI1apV6dWrFxcuXHDtT0hIYOzYsTz55JNUqVKFxx9/nA8//JDk5OT7zphTMjIyGD16NHXq1KF69eq89dZbxMbG3vUcVVWZPn06DRs2pEqVKnTv3j3L44XMxzxw4EBq1KhBrVq1GDZsGCkpKVnud+jQobRp04YKFSrc9md8/fp1Bg4cyJNPPkm5cuUYN25cjjxmIXITKW6FyCMiIyNp164dFy9e5JNPPuGXX35h1KhR7Ny5kw4dOpCQkKB1xIcmLS2NH374gVdeeYVVq1bd9hiz2cyqVatuKVD+16RJkxg5ciRPP/00a9euZeXKldSsWZO+ffuyZMkSAF544QXOnTvHvn37bjl/9erVhISE8Nhjj7m2rVq16q7Z7kWjRo3Ytm1blq/w8PD7uq2ff/6ZIUOG0LZtW77//nuWLVtG69at73i8qqq88cYbREZG8vnnn/Pdd99RpEgRevToQWpqKpBZpF2/fp13332X9evXM2HCBLZu3crw4cPvK2NOGj9+PL///jvTpk1j8eLFXL9+nTfffPOu58ydO5fFixczatQovv32W7y9venVqxcZGRmuYwYNGsSZM2f46quv+OKLL9izZ0+WD1AOhwOz2UyXLl2oV6/ebe/HarWSL18++vTpQ7ly5XLmAQuR26hCiDyhV69eauPGjdW0tLQs269fv65WrVpVHTlypGtbkyZN1FmzZqlDhw5Vq1Wrpj722GPq8uXLs5wXFRWl9uvXT61Zs6Zau3Zt9fXXX1cjIyPveP+//fabWrNmTdVut6uqqqrHjh1Ty5Qpo3700UeuY4YNG6YOHDhQVVVVjY+PV99++221YcOGapUqVdTWrVur69atcx373XffqY8++qiakZGR5X769OmjDho06K4/i9WrV6svvviimpSUpFatWlWNiorKsn/GjBnqM888o/bo0UPt16+fa/vNzDcf5/79+9UyZcqoixYtuuU+JkyYoFasWNF1288//7w6bNiwLMc4nU71iSeeyPIz2LVrl9qoUSPVarWqDRs2VPfu3XvXx3I77777rtqnT59sn3c7NptNbdSokfrtt9/e8znnzp1Ty5Qpo546dcq1zeFwqHXr1r3r7fz4449qxYoVVZvNds/3dfO5WrZsmdq4cWO1SpUqar9+/dSkpKR7vo1/SkpKUitWrKhu2LDBte3MmTNqmTJl1P3799/2HKfTqTZo0ECdN29eltupVKmSun79+iy3cejQIdcxmzdvVsuWLateu3btltu8l+ewc+fO6ocffpidhydEniAtt0LkAQkJCWzbto1OnTrh5eWVZV9YWBht2rRhw4YNqKrq2v7VV19RqVIl1qxZQ6dOnRg1ahTnzp0DwGaz0atXL3x9ffn6669ZtmwZPj4+vPLKK1it1ttmqFWrFikpKRw7dgyA3bt3ky9fPnbv3u065q+//qJOnTpAZgtVxYoVmTNnDuvXr+fFF19kyJAhHDp0CICnnnoKh8OR5XJ+XFwcmzdvpl27dnf9eaxcuZJnnnkGf39/GjduzOrVq2973MCBA/nll184fPjwbfevX78eHx8fXnrppVv29ejRA5vNxs8//wxAu3bt2LBhg6vlEmDXrl1cvnw5S96VK1fSqlUrjEYjrVu3ZuXKlVlud/Xq1ZQtW/auj+/ffPHFF1SvXv2uX1FRUQAcO3aM6OhodDodzz33HA0bNuSVV17h1KlTd7z9m78DZrPZtU2n02Eymdi7d+8dz7NYLPj5+WEwGLL1eC5dusSGDRv44osvmDdvHsePH2fUqFGu/WvXrv3Xx7tnzx4Ajhw5gs1mo379+q7zS5UqReHChTlw4MBt7//y5cvExMRkOcff35+qVauyf/9+APbv309AQACVK1d2HVO/fn10Op3rd1oIkTOy9woihPBIFy9eRFVVSpUqddv9pUqVIjExkfj4eEJCQgBo3LgxL7/8MgC9e/dmwYIF7Nq1i4iICH788UecTifjxo1DURQAJkyYQO3atdm9ezcNGza85T78/f0pX748u3fvpnLlyuzevZvu3bvz6aefkpKSgsVi4eLFi9SuXRuAAgUK0KtXL9f5Xbp0Ydu2bWzYsIEqVarg5eVF69atWb16tas/69q1aylUqJCrQL6dCxcucPDgQWbOnAnAM888w4QJE+jbt6/rsdxUsWJFWrZsyZQpU1i4cOFtb6tYsWKYTKZb9hUoUAA/Pz9Xt4Y2bdowadIkfvrpJ9q2bQtkFqo1a9akZMmSQGZx9/PPP/PNN9+4snXq1Inhw4fj6+vr+jnePP5u/vjjD6pXr+76vlGjRsyYMQOADh063LEP8E358+cHMruzAHz66acMHTqUIkWK8NVXX9GlSxd+/vlngoKCbjk3IiKCwoUL8/HHHzNmzBi8vb1ZsGAB165dIyYm5rb3Fx8fz+eff37bDwr/JiMjg8mTJ1OgQAEARowYwWuvvcbQoUMJCwvjiSeeoGrVqne9jZvnxsbGYjQaCQgIyLI/JCTkjtlvbr/5t/PPc2721Y2NjSU4ODjLfoPBQGBg4B1vVwhxf6S4FSIP+WfL7L/5Z+ugoiiEhoYSFxcHwIkTJ7h06RI1atTIck5GRgaXLl1iz5499O7d27V99OjRPPPMM67it2fPnuzZs4d33nmHDRs2sHfvXhITE8mfPz8lSpQAMvsffvHFF/z0009ER0djs9mwWq1ZWp5ffPFFXnjhBaKjoylQoACrV6/m+eefv6VI/adVq1bRsGFDV6HRuHFjhg8fzs6dO2/bz3HAgAE8/fTTbNu27ZbiBe79ZxoQEEDz5s1ZtWoVbdu2xWKx8Msvv2Tpc7l+/XqKFSvm6ktZvnx5ihQpwo8//kj79u0BaN68Oc2bN//X+6tTp06W1ktvb2/X/4OCgm5blN6O0+kE4PXXX+fJJ58EMj/ING7cmJ9++okOHTrcco7RaGTmzJkMHz6cRx99FL1eT7169WjcuPFtf14Wi4XXXnuNUqVK/Wvf1tspVKiQqzgFqF69Ok6nk/PnzxMWFoafnx9+fn7Zvl0hhGeS4laIPKBYsWIoisLZs2dvWxidPXuWwMDALC1L/3tpWFEUV2GSmppKxYoVmTJlyi23FRwcjNFoZM2aNa5tN4vCRx99lFWrVnHixAmMRiOlSpXi0UcfZffu3SQlJfHoo4+6zpk/fz6LFi1i2LBhlC1bFm9vb8aPH4/NZnMdU6FCBcqVK8eaNWto0KABZ86ccbWK3o7D4eC7774jNjaWChUqZNm+cuXK2xa3xYoVo3379nz88ce3jEwvUaIEe/fuxWq13tJ6Gx0djcVicRXrkDmwrHv37ly8eJFdu3ah0+l46qmnXPtXrlzJ6dOns2RzOp2sWrXKVdzeK29vb4oXL37bfV988QWzZ8++6/k//PADhQsXJiwsDCBLq7/JZKJo0aJcvXr1judXqlSJ77//nuTkZGw2G8HBwbRv355KlSplOc5isfDKK6/g6+vLZ599htFovNeHeM/Wrl3LBx98cNdj5s6dS61atQgNDcVms5GUlJSl9TYuLs71s/hfN7fHxcW5Wrxvfn/zg0poaCjx8fFZzrPb7SQmJt7xdoUQ90eKWyHygHz58tGgQQOWLl1K9+7ds7R+xsTEsG7dOp599tm7tnj+U8WKFdmwYQMhISF3bBG7XWF1s9/tggULXN0P6tSpw5w5c0hMTKRnz56uY/ft20fTpk159tlngcwi78KFC7d0rXjhhRdYuHAh0dHR1K9fn0KFCt0x9+bNm0lJSWHNmjXodP8/5OD06dO89957txQ0N73xxhs0b96cH374Icv2Vq1asXjxYr755hu6dOmSZd+XX36J0Wh0tXYC1K1bl/DwcFavXs2uXbto1aoVPj4+AJw8eZIjR46wePFiAgMDXeckJibSpUsXzp49e8duJdmVnW4JlSpVwmQycf78eWrVqgVk9rm+cuUKhQsX/tf78vf3BzK7cBw5coT+/fu79lksFnr16oXJZGLWrFlZ+uhmx9WrV12t9wAHDhxAp9O5um9kp1tCpUqVMBqN7Nixw/XcnTt3jqioKKpVq3bbc8PDwwkLC2PHjh2UL1/e9dgOHjxIx44dgczW5KSkJI4cOeIq8Hfu3InT6aRKlSr39biFELcnA8qEyCPef/99rFYrvXr14q+//uLq1ats2bKFnj17UqBAAd5+++17vq02bdq4piPas2cPkZGR7Nq1iw8//JBr167d8bzAwEDKli3LunXrXK20tWrV4tixY1y4cMFV8EJmcbx9+3b27dvH2bNnGTly5G3nGm3Tpg3R0dF8++239zSQ7PHHH6dcuXKUKVPG9dWyZUsCAgJYt27dbc8LDQ2le/fuLF68OMv26tWr07VrVyZPnsyXX37JpUuXOHv2LFOnTmXRokW8++67WYptRVFo164dy5YtY//+/bzwwgtZst2cQ/af2WrXrk3lypVdA8s2btyYpbX3fgQFBVG8ePG7ft1suffz86NDhw7MnDmTbdu2ce7cOVd3h3/meOqpp9i4caPr+w0bNrBr1y4iIyP59ddf6dmzJ82aNXP1x7ZYLPTs2ZPU1FTGjRuHxWIhJiaGmJgYHA5Hth6P2Wxm6NChnDhxgj179vDhhx/SsmVLV4uon5/fvz7emx/4/P39adeuHRMnTmTnzp0cOXKEYcOGUb169SzF7T8fr6IodO3alVmzZrFp0yZOnjzJkCFDyJ8/P82aNQMyW74bNWrE+++/z6FDh9i7dy9jx46lVatWWbpUnDlzhuPHj5OQkEBycjLHjx/n+PHjWR7vzW0pKSnEx8dz/Phxzpw5k62fmRC5mbTcCpFHlChRglWrVjFz5kwGDBhAYmIioaGhNGvWjDfeeOOe+2BC5iXvJUuWMGXKFN58801SUlIoUKAA9erV+9e+jbVr1+b48eOu4jYoKIhSpUoRFxdHRESE67g+ffoQGRlJr1698Pb25sUXX6RZs2a3TPLv7+9PixYt2Lx5s6uQuJ3Y2Fg2b958264UOp2OZs2asXLlStcguv/Vq1cvli1blmXeUoDhw4dTtmxZli5dyrRp09Dr9VSoUIHPPvuMJ5544pbbadu2LTNnzqR06dKu1kSr1cratWuz9FP+pxYtWvDVV1/xzjvvkJyczPnz5+/4OB+EIUOGYDAYGDJkCOnp6VStWpWFCxdmaWE+f/58lucmJiaGiRMnui7nP/vss1kWJTh69CgHDx4EuKWrzKZNm1xz8j7xxBM8//zzvPXWW3fMV6xYMZo3b07v3r1JTEzk8ccf/9duCHczbNgwdDod/fr1w2q10rBhw1tu738fb+/evUlLS2PkyJEkJSVRs2ZN5s2bl6U1esqUKYwdO5Zu3bqh0+lo0aIFI0aMyHK7r776KleuXHF9/9xzzwGZLfv/uw0yf47r16+nSJEi/Pbbb/f9mIXITRQ1OyNMhBDCDXXr1o3SpUvfUigIz5aWlkadOnWYO3fuHWfAmDlzJr/++ivff//9Q04nhHBX0i1BCOGxEhMT2bhxI7t376ZTp05axxE5bNeuXdStW/euU7sJIcT/km4JQgiP9fzzz5OYmMigQYOydGkQucPjjz/O448/rnUMIYSHkW4JQgghhBAi15BuCUIIIYQQIteQ4lYIIYQQQuQaUtwKIYQQQohcQ4pbIYQQQgiRa0hxK4QQQgghcg0pboUQQgghRK4hxa0QQgghhMg1pLgVQgghhBC5hhS3QgghHohdu3bx/PPPU6lSJZo3b87q1av/9ZwTJ07QqVMnKleuzGOPPcbcuXNvOWbDhg089dRTVK5cmTZt2rB582bXPpvNxkcffUSbNm2oVq0aDRs2ZMiQIURHR2e5jSeeeIKyZctm+ZozZ85/f9BCCM1JcSuEECLHRUZG8tprr1GnTh2+//57unXrxogRI9i6desdz7FYLPTq1YvChQuzevVqhgwZwqeffso333zjOmbfvn0MHDiQF154gTVr1tC0aVPeeOMNTp06BUB6ejrHjh2jT58+rF69mk8//ZTz58/Tp0+fW+6vX79+bNu2zfXVuXPnnP9BCCEeOll+VwghcrkuXbpQunRpAL7//nsMBgMdO3akf//+KIryQO7zo48+YvPmzaxfv9617e233yYpKYn58+ff9pylS5cybdo0tm3bhslkAmDKlCn8+uuv/PTTTwAMGDCAtLQ0Zs+e7TrvxRdfpFy5cowZM+a2t3vo0CHat2/P77//TuHChYHMltuuXbvSvXv3nHi4Qgg3Ii23QgiRB3z33Xfo9XpWrFjB8OHDWbBgAStWrLjj8Xv27KF69ep3/Vq7du0dzz9w4AD16tXLsq1hw4YcOHDgrufUqlXLVdjePOf8+fMkJibe9+1aLBYURSEgICDL9rlz51KnTh2ee+455s2bh91uv+NtCCE8h0HrAEIIIR68QoUKMWzYMBRFISIiglOnTrFgwQJefPHF2x5fqVIl1qxZc9fbDAkJueO+2NhYQkNDs2wLDQ3FYrGQnp6Ol5fXbc8JDw+/5Zyb+wIDA297uyEhIcTGxt42R0ZGBlOmTKFVq1b4+fm5tnfp0oUKFSoQGBjI/v37+eSTT4iJieG9996762MWQrg/KW6FECIPqFq1apYuCNWqVeOrr77C4XCg1+tvOd7Ly4vixYs/zIg5zmaz0b9/f1RVZfTo0Vn29ejRw/X/cuXKYTQa+eCDDxg4cGCWlmMhhOeR4lYIIcQt9uzZQ+/eve96zOjRo3nmmWduuy80NPSW1tTY2Fj8/Pxu22p7t3Nu7rvTMXFxcbe05tpsNgYMGEBUVBQLFy7M0mp7O1WrVsVut3P58mUiIiLueqwQwr1JcSuEEHnAoUOHsnx/8OBBihcvfttWW/jv3RKqVavGli1bsmzbvn071apVu+s506ZNw2azYTQaXeeULFmSwMBA1zE7d+7MMhDsf2/3ZmF78eJFFi1aRL58+e76OACOHz+OTqe762MSQngGGVAmhBB5QFRUFBMmTODcuXOsX7+eJUuW0LVr1zsef7Nbwt2+7tYa2qFDByIjI5k8eTJnz57l66+/ZsOGDVmK0iVLltCtWzfX923atMFoNDJ8+HBOnz7Njz/+yKJFi7J0IejatStbt27lyy+/5OzZs8ycOZMjR464pvGy2Wz069ePI0eOMGXKFBwOBzExMcTExGC1WgHYv38/CxYs4MSJE0RGRrJ27VomTJjAM8884yqihRCeS6YCE0KIXK5Lly488sgjOJ1O1q9fj16vp2PHjgwYMOCBTQUGmYs4TJgwgTNnzlCwYEH69u1L27ZtXftnzpzJd999x2+//ebaduLECcaMGcPhw4fJly8fnTt35tVXX81yuxs2bGDatGlcuXKFEiVKMHjwYB577DEALl++TNOmTW+bZ9GiRdSpU4ejR48yevRozp07h9VqJTw8nGeffZYePXpIf1shcgEpbgGn04ndbken0z3QF3ohhNBCt27dKFeunMwEIITwaKqq4nQ6MRgM6HR37nwgfW4Bu93O4cOHtY4hhBAPhMViISYm5q5zwQohhKeoXLnyXa+ySHELruq/cuXKdxxcIYQQnsrPz4+wsLC7DuYSQgh353A4OHz48F1bbUGKWwBXVwS9Xi/FrRAi11myZInWEYQQIsf8WxdSmS1BCCGEEELkGlLcCiGEuK0uXbowbtw4rWMIIUS2SHErhBBCM7t27eL555+nUqVKNG/enNWrV9/1+HPnztGlSxfq169P5cqVadq0KVOnTsVms2U5LikpidGjR9OwYUMqVarEk08+yebNmx/kQxFCuAnpcyuEEEITkZGRvPbaa3To0IEpU6awY8cORowYQVhYGI0aNbrtOUajkeeee46KFSvi7+/PiRMneP/991FVlXfeeQcAq9VKjx49CAkJYfr06RQoUICoqCgCAgIe5sMTQmhEilshhBCkpqYyatQoNm7ciK+vLz179nzg97l8+XLCw8MZOnQoAKVKlWLv3r0sWLDgjsVt0aJFKVq0qOv7IkWKsHv3bvbs2ePatmrVKhITE1m+fLlrGd/w8PAH+EiEEO5EilshhBBMnjyZv/76i88//5zg4GCmTp3K0aNHKVeu3B3P2bNnD717977r7Y4ePZpnnnnmtvsOHDhAvXr1smxr2LAh48ePv+fcFy9eZOvWrTRv3ty17bfffqNatWqMGTOGTZs2ERwcTOvWrendu7fMiCNEHiDFrRBC5HEpKSmsXLmSjz76yFVsTpw40bWk7Z1UqlSJNWvW3PWYkJCQO+6LjY0lNDQ0y7bQ0FAsFgvp6el4eXnd8dwOHTpw9OhRrFYrL730Ev3793fti4yMZOfOnbRp04Y5c+Zw6dIlRo8ejd1u580337xrXiGE55MBZUIIcQfZHewEcOLECTp16kTlypV57LHHmDt37i3HbNiwgaeeeorKlSvTpk2bWwY6DR06lLJly2b56tWrV5Zjjh49So8ePahVqxZ16tTh/fffJyUl5b4eZ2RkJDabjapVq7q2BQUFUbJkybue5+XlRfHixe/65efnd1+Z/s3UqVP57rvv+Pjjj/njjz+YP3++a5+qqoSEhDB27FgqVarE008/zeuvv87y5csfSBYhhHuRllshhLiN+xnsZLFY6NWrF/Xq1WP06NGcOnWKYcOGERAQwEsvvQTAvn37GDhwIO+88w5NmjRh3bp1vPHGG6xevZoyZcq4bqtRo0ZMmDDB9f0/l5qMjo6mR48etGzZkvfffx+LxcL48eN57733mDFjxgP6idzqv3ZLCA0NJTY2Nsu22NhY/Pz87tpqC1CoUCEAHnnkERwOByNHjqRnz57o9XrCwsIwGAxZuiBEREQQExOD1Wq967KdQgjPJ8WtEMLtdenShdKlSwPw/fffYzAY6NixI/379//XlWru1/0Mdlq7di02m43x48djMpkoXbo0x48f56uvvnIVt4sWLaJRo0a88sorAAwYMIDt27ezZMkSxowZ47otk8lEWFjYbe/njz/+wGAw8MEHH7iWobxZRF68eJHixYtn67EWLVoUo9HIwYMHKVy4MACJiYlcuHCB2rVr3/G8/9otoVq1amzZsiXLtu3bt2d7mWBVVbHb7TidTvR6PTVq1GD9+vU4nU7Xz+fChQuEhYVJYStEHiDFrRDCI3z33Xe88MILrFixgiNHjjBy5EgKFy7Miy++eNvjtRjsdODAAWrVqpWlgGrYsCFz584lMTGRwMBADhw4QPfu3W+53V9//TXLtt27d1OvXj0CAgKoW7cuAwYMIF++fEDmVFdGozHL+uo3Wzr37t2b7eLW19eXdu3a8dFHHxEUFERISAhTp0791w8ON7sl3K8OHTrw9ddfM3nyZNq1a8fOnTvZsGEDs2fPdh2zZMkSNm7cyMKFC4HMDxAGg4GyZctiMpk4fPgwH3/8MS1btnTNjNCxY0eWLFnCuHHj6Ny5MxcvXmT27Nl06dLlvrMKITyHFLdCCI9QqFAhhg0bhqIoREREcOrUKRYsWHDH4laLwU6xsbG3TDl18zZiY2MJDAy87e2GhIRkuTzfqFEjmjdvTnh4OJGRkXzyySf07t2bb775Br1eT926dZk4cSLz5s2ja9eupKWl8fHHHwMQExNz18d8J0OGDCE1NZU+ffrg6+tLjx49sFgs93Vb96po0aLMnj2bCRMmsGjRIgoWLMiHH36YpWX8xo0bREZGur43GAzMmzeP8+fPA1C4cGE6d+6c5QNDoUKFmD9/PhMmTOCZZ56hQIECdO3a9V8/7AghcgcpboUQHqFq1apZWhKrVavGV199hcPhuO30Tv+1VVFLrVq1cv3/5oCyZs2auVpzS5cuzcSJE5k4cSKffPIJOp2OLl26EBoaet/dNHx9ffnoo4+ybLvZdeJBqlOnzl0/hLz11lu89dZbru+ffvppnn766X+93erVq/Ptt9/mREQhhIfRtLj966+/mD9/PkeOHCEmJobPPvuMZs2aufarqsqMGTNYsWIFSUlJ1KhRg1GjRlGiRAnXMQkJCYwdO5bff/8dnU5HixYtGD58OL6+vho8IiGEu9BisNOdzrm5707HxMXF3dKa+09FixYlX758XLx40dVVok2bNrRp04bY2Fi8vb1RFIUFCxZkWeBACCHyIk2L29TUVMqWLUu7du1uO/fg3LlzWbx4MRMnTiQ8PJzp06fTq1cvfvzxR8xmMwCDBg0iJiaGr776CpvNxrBhwxg5cqTrEp0QInc4dOhQlu8PHjxI8eLF7zgpvxaDnapVq8a0adOw2Wyu/p/bt2+nZMmSBAYGuo7ZuXNnlsvo/3a7165dIyEh4bYDzG4WxStXrsRsNtOgQYM73o4QQuQJqpsoU6aMunHjRtf3TqdTbdCggTpv3jzXtqSkJLVSpUrq+vXrVVVV1TNnzqhlypRRDx065Dpm8+bNatmyZdVr167d833b7XZ1z549qt1uz4FHIoTIaZ07d1arVaumjh8/Xj179qy6bt06tVq1auqyZcse2H1eunRJrVq1qjpp0iT1zJkz6pIlS9Ty5curW7ZscR2zePFitWvXrq7vk5KS1Pr166uDBw9WT506pf7www9q1apV1eXLl7uO2bt3r1qhQgV1/vz56pkzZ9QZM2aoFStWVE+ePKmqqqpaLBZ14sSJ6v79+9XIyEh1+/bt6vPPP6+2aNFCzcjIyHLfR44cUc+dO6cuWbJErVKlirpw4cIH9vMQQgit3Wu95rZ9bi9fvkxMTAz169d3bfP396dq1ars37+fVq1asX//fgICAqhcubLrmPr166PT6Th06FCW5RiFEJ7tueeeIz09nfbt26PX6+natatreq0H4X4GO/n7+zN//nzGjBlD27ZtyZcvH3379s2Ss0aNGkyZMoVp06bxySefUKJECT777DPXHLd6vZ5Tp06xZs0akpOTyZ8/Pw0aNKB///5ZZmE4dOgQM2fOJCUlhYiICEaPHs1zzz3n2q+qKqmpqQ/s5yNEbuDj4/PAphMU2nHb4vbmiN//vWz4z1HFsbGxBAcHZ9lvMBgIDAy8rxHDDofjPtMKIR4kVVXR6/WMGDGCkSNHurY7nc4Her+1atVi1apVWbb983Wib9++9O3bN8u20qVLs3jx4jueA9CiRQtatGhx22OMRiNz5sy5bZ5/3s4/F3j43/2qqtK4cWN27Nhxx8cmhMhsENu8ebMUuB7iXus0ty1utXD48GGtIwghbsNisRATE8OBAwe0juIR0tLSpLAV4h5s376dnTt34u3trXUUkYPctri9OXAiLi6O/Pnzu7bHxcVRrlw5IHMgRXx8fJbz7HY7iYmJd1zZ524qV658x8EpQgjt+Pn5ERYWlu2Vq/KqlJQU1/+joqJk9hgh/kdKSoprNb4qVarI34iHcDgc99QQ6bbFbXh4OGFhYezYsYPy5csDma03Bw8epGPHjkDmPIZJSUkcOXKESpUqAbBz506cTidVqlTJ9n3q9XopboVwQ0uWLNE6gkf55+tYQECAvHEL8T/++Tci7/25j6bFbUpKCpcuXXJ9f/nyZY4fP05gYCCFCxema9euzJo1i+LFi7umAsufP79rLtxSpUrRqFEj3n//fUaPHo3NZmPs2LG0atWKAgUKaPWwhBBCCCGERjQtbo8cOULXrl1d398cIPH8888zceJEevfuTVpaGiNHjiQpKYmaNWsyb9481xy3AFOmTGHs2LF069bNtYjDiBEjHvpjEUK4ly5dulCuXDmGDx+udRQhhBAPkabFbZ06dTh58uQd9yuKQv/+/enfv/8djwkKCpIFG4QQmti1axcTJ07k9OnTFCpUiD59+tC2bds7Hn/u3Dk++OADzp4965rmq3Xr1rz55puuRR/+6YcffuCdd96hadOmfP755w/yoQghRK7htn1uhRDCnUVGRvLaa6/RoUMHpkyZwo4dOxgxYgRhYWFZ5sL9J6PRyHPPPUfFihXx9/fnxIkTvP/++6iqyjvvvJPl2MuXLzNp0iRq1ar1MB6OEELkGlLcCiE8XmpqKqNGjWLjxo34+vrSs2fPB36fy5cvJzw8nKFDhwKZYwD27t3LggUL7ljcFi1alKJFi7q+L1KkCLt372bPnj1ZjnM4HAwaNIi33nqLvXv3kpSU9OAeiBBC5DJS3AohPN7kyZP566+/+PzzzwkODmbq1KkcPXrUNW3g7ezZs4fevXvf9XZHjx7NM888c9t9Bw4coF69elm2NWzYkPHjx99z7osXL7J169ZbVlP87LPPCAkJoX379uzdu/eeb08IIYQUt0IID5eSksLKlSv56KOPXMXmxIkTeeyxx+56XqVKlVizZs1dj/nfFRL/KTY2ltDQ0CzbQkNDsVgspKen4+XldcdzO3TowNGjR7Farbz00ktZxhXs2bOHlStX/ms2IYQQtyfFrRDCo0VGRmKz2ahataprW1BQECVLlrzreV5eXhQvXvxBx7utqVOnkpKSwokTJ5g8eTLz58+nd+/eWCwWhgwZwtixY29ZWlwIIcS9keJWCJEn/dduCaGhocTGxmbZFhsbi5+f311bbQEKFSoEwCOPPILD4WDkyJH07NmTyMhIrly5Qp8+fVzHOp1OACpUqMBPP/1EsWLF/vWxCSFEXibFrRDCoxUtWhSj0cjBgwddy2kmJiZy4cIFateufcfz/mu3hGrVqrFly5Ys27Zv357tJYJVVcVut+N0OomIiGDdunVZ9k+bNo2UlBSGDx9OwYIFs3XbQgiRF0lxK4TwaL6+vrRr146PPvqIoKAgQkJCmDp1Koqi3PW8/9otoUOHDnz99ddMnjyZdu3asXPnTjZs2MDs2bNdxyxZsoSNGzeycOFCANauXYvBYKBs2bKYTCYOHz7Mxx9/TMuWLV3z3JYpUybL/QQEBNx2uxBCiNuT4lYI4fGGDBlCamoqffr0wdfXlx49emCxWB7ofRYtWpTZs2czYcIEFi1aRMGCBfnwww+zTAN248YNIiMjXd8bDAbmzZvH+fPnAShcuDCdO3eme/fuDzSrEELkJYqqqqrWIbTmcDg4cOAA1apVQ6/Xax1HCCH+E1VVXUX21q1b/7UVW4i8Rv5GPNO91mvSciuEELmMoihs3brV9X8hRFbyN5K7SXErhBC5kLxhC3F38jeSe+m0DiCEEEIIIUROkZZb4XFUVSU1NVXrGEIIITycj4+PtODmQlLcCo+iqioNGzZk+/btWkcRQgjh4Ro0aCADynIh6ZYgPEpqaqoUtkIIIXLEn3/+KVcCcyFpuRUeKzo6Gl9fX61jCCGE8DApKSkUKFBA6xjiAZHiVngsX19fKW6FEEIIkYV0SxBCCCGEELmGFLdCCCGEECLXkOJWCCGEEELkGlLcCiGEEEKIXEOKWyGEEEIIkWtIcSuEEEIIIXINKW6FEEIIIUSuIcWtEEIIIYTINaS4FUIIIYQQuYYUt0IIIYQQIteQ4lYIIYQQQuQaUtwKIYQQQohcw6B1ACGyw8fHhwYNGrj+L4QQQmSXvJfkblLcCo+iKApbt251/V8IIYTILnkvyd2kuBUeR16IhBBC/FfyXpJ7SZ9bIYQQQgiRa0hxK4QQQgghcg0pboUQQgghRK4hxa0QQuRCly5d4vDhwzidTq2jCCHEQyXFrRBC5DIpKSk899xzdOvWjT/++EPrOEII8VBJcSuEELlMfHy86/9RUVEaJhFCiIdPilshhMhlkpOTXf+3WCwaJhFCiIdPilshhMhl/lnc/vP/QgiRF0hxK4QQuUxCQsJt/y+EEHmBFLdCCJHL/LPPbVxcnIZJhBDi4ZPiVgghcpl/FrT/LHSFECIvkOJWCCFymZiYGAB0Tj3Xr1/XOI0QQjxcUtwKIUQuc/XqVRQU8lnzk5SUREpKitaRhBDioZHiVgghcpkrV65gdvjgYw8AZK5bIUTeIsWtEELkIjabjejoaLztfng7/AC4fPmyxqmEEOLhkeJWCCFykfPnz+N0OvG1BeJrCwTg7NmzGqcSQoiHx6B1ACGEEDnnzJkzAPjbg/C3BWXZJoT4f1988QWHDx+mYcOGdOzYUes4IgdJcSuEELnIzULWz5YPs9MHo9PE6dOnNU4lhHuxWq3MmTMHgKNHj0pxm8tItwQhhMhFDh06hIKCvy1f5r/WYC5cuCDL8ArxD/+cCzopKQmbzaZhGpHTpLgVQohcwmq1cvjwEfytwRhUIwD5rAVQVZUDBw5oG04IN/K/i5vcuHFDoyTiQZDiVgghcomjR49is1nJZ83v2pYvI/P/+/bt0yqWEG7nf5ellmWqcxcpboXHiY6OZt68eSxfvhy73a51HCHcxs6dOwHIl1HAtS3QFopO1bNjx06tYgnhdq5duwaAoXBglu9F7iADyoTHWbx4MUuXLgWgZMmS1KlTR+NEQriH3377DZ2qJzijoGubXjUQnF6QU6dOcuXKFYoUKaJhQiHcw5UrVwAwFg/BHpXo+l7kDtJyKzzOPz9hy6dtITJdunSJs2fPEppe2NXf9qYC6cUA+P3337WIJoTbcRW3xYKzfC9yByluhce5fv266/8xMTEaJhHCfWzcuBGA/GlFb9kXlh6OguI6Roi8LjIyEsVkwFAw0PW9yD2kuBUe55+fsOXTthDgcDhYvWo1BtVI/r9baf/J5PQiNK0Ihw8f5uTJkxokFMJ92Gw2zp8/jz7MD53ZgC7QW+aCzmWkuBUexWKxcOPGDfQB+UFRuHjxotaRhNDcn3/+ydVrVymUUvKWLgk3FU0pA8CKFSseZjQh3M758+ex2+0YwvwBMIT5ExMTI9OB5SJS3AqPcrOY1Xn7o5h8uHDhAqqqapxKCG198803AIT/XcDeTkhGYbztfvz4w48kJSU9rGhCuJ1Tp04BoP+7uNWH+WXZLjyfFLfCoxw/fhwAnU8gep8gEhISpN+tyNMOHTrEjh07yJdRAH97vjsep6BQzFKO9Ix0Fi1a9BATCuFejhw5AoCh4N8tt3/3uz18+LBmmUTOkuJWeJT/L27zofMNzLJNiLzos88+A6B0UrV/PTY8pTReTh+WLl0mk9aLPGvP3j2Zg8kKBABgDA8CBfbu3attMJFjpLgVHuXAgQMoegM6bz90vplTuOzfv1/jVEJoY9euXfz111+EphUh6B+rkt2JHgMRiZVJT0/jyy+/fAgJhXAv8fHxnDt7DkN4EIouswTSeZvQh/lz4MABbDabxglFTpDiVniM69evc/78eXR+oSiKDr1fMIpOz65du7SOJsRDZ7VamTxpMnBvrbY3FU59BB97AN9++630MRR5zp49ewAwFg3Ost1YNJiMjAzpmpBLSHErPMbNpUX1AWEAKDo9Or9gTp06JZdYRZ4zd+5czl84TzFLOfztwf9+wt906Cif8CgOh4NRo0bJEtYiT7m5kImpZGiW7aaI0Cz7hWeT4lZ4jJsT0BuCCrm26YMKoaoqmzZt0iqWEA/dyZMn+eqrr/B2+PFINlptbwrJKESRlEc4ceIEixcvzvmAQrih9PR0Nm/Zgj7YxzVDwk3GYsHovI38+uuvOJ1OjRKKnCLFrfAICQkJ7Ny5E51vPnRevq7thuAigMLPP/+sXTghHqLU1FRGjBiB0+mkwo26d5zX9t+USayJ2enDF7O+4NixYzmcUgj38+eff5KeloapbEEURcmyT9HrMD6Sn+joaNdsCsJzSXErPMK6detwOBwYgsOzbNcZvdAHhLF//37OnTunUTohHg5VVRk9ejRnz56lmKUcIRmF/v2kOzCqJirF18dutzPwnYEygb3I9X788UcAzGUL3na/uWwBAH744YeHlkk8GFLcCrdnt9tZtnw5it6AMbT4LfuN+SMAWL58+cOOJsRDtWjRIjZu3Ei+jAKUSaz5n28vJKMQjyRWJ/p6NEOGDJH+tyLXioqKYvPmzRgKBWLI73/bY4wlQtEFeLNu/XqSk5MfckKRk9y+uLVYLIwbN44mTZpQpUoVOnTowKFDh1z7hw4dStmyZbN89erVS8PEIqdt2rSJa1evYggphmK49RKsPqggOrMv69atk4FlItfatm0bM2fMxMvhQ5X4Ruhy6OW7hKUCBVKLs3fvXqZMmSIr/olc6ZtvvsHpdOJVs9gdj1F0Cl41ipKelsb333//ENOJnOb2xe2IESPYvn07kydPZt26dTRo0IAePXoQHR3tOqZRo0Zs27bN9fXJJ59omFjkJKvVyqeffgqKDmPBR257jKIoGAuVJiMjg9mzZz/khEI8eHv27GHQoEEoTh1V4x7D7PTOsdtWUKiYUA9/Wz6+/fZbZs2alWO3LYQ7SEtL47s136HzM9+xS8JNXlXCUYx6li9fjsPheEgJRU5z6+I2PT2dX375hcGDB1O7dm2KFy/OW2+9RfHixVm6dKnrOJPJRFhYmOsrMDBQw9QiJ61YsYIrV65gzB+Bzux7x+MMocXRefmzevVq6XsrcpXDhw/Tv39/7FY71eIeJ9AW+u8nAfGma+wL/o1407V/PdagGqkR2xRfewDz5s1jwYIF/zG1EO7j66+/xpJswataURT93csenZcRc6XCREVFsX79+oeUUOQ0ty5u7XY7DocDs9mcZbvZbGbfvn2u73fv3k29evV48skn+eCDD2RgRC4RFRXFrFmzUAwmTIXL3vVYRdFhKloJp9PJuHHjZCoXkSucOnWKN998k/S0dKrENc7WALKzAYeI9b7C2YBD/34wYHZ6UzO2Gd4OX2bMmCF92EWucOPGDRYsWIDOx4RXzVvHbNyOd90IFIOeWbNmkZ6e/oATigfBoHWAu/Hz86N69ep8/vnnREREEBoayvr16zlw4ADFimX2m2nUqBHNmzcnPDycyMhIPvnkE3r37s0333yDXq/P1v3JJQj34XQ6GTVqFKmpqZhL1kQxmP71HENQQQzBRdi/fz9Llizh5ZdffghJhXgw9u7dy8CBg0hJsVA5viH504tm63y7Ysvy773wcvhSM6YZf+X/hcmTJ5OYmEivXr1umTZJCE8xZ84cUlNT8W1eHp353koevb8XXrWLc33HOZYuXUq3bt0ecEpxr+61TnPr4hZg8uTJDBs2jMaNG6PX66lQoQKtWrXi6NGjALRq1cp17M0BZc2aNXO15maHLLvnPn755Rf27NmDPqgQhpB7f1M3F6+KIzmOTz/9lICAAIoWzV5BIIQ72LNnD/PmzUO1q1SJb0zB9HtrccoJPo4Aal9vwd7QTcyePZsTJ07QqVMndDq3vtAnxC2uXr3KypUr0efzwatK+L+f8A/ej5Yk48Bl5s2fT4kSJaS7o4dx++K2WLFiLFmyhNTUVCwWC/nz52fAgAF3LFqKFi1Kvnz5uHjxYraL28qVK2e7tVfkvJ07d7Jy1Sp0Ji/MJapnq9VIMZgxl6hO+ukdzJ49m0WLFhEcfO9LkwqhtW+++YY5c+agdxqoHvs4wda7D4B5EHwcATwa8xT7Q35j8+bNKIrC2LFj8fLyeuhZhLgfDoeD6dOn43A48G9S9l/72v4vndmAT+NHsPx8jHXr1vHRRx/JFQw34HA47qkh0u2L25t8fHzw8fEhMTGRbdu2MXjw4Nsed+3aNRISEggLC8v2fej1eiluNXbhwgWGDn0PUDCXqovOaP7Xc/6XIaggpvCKRF8+ypAhQ5g9ezYm0793axBCS1arlY8//pgVK1ZgdvpQI+YJ/O35NMtjdnpTK7YFB4M388cff/Daa68xefJkChcurFkmIe7VkiVLOHLkCOaKhTE/kv++bsNcJZyME9Fs3ryZX375haeffjqHU4oHxe2vM23dupUtW7YQGRnJn3/+SdeuXYmIiKBt27akpKQwadIkDhw4wOXLl9mxYwd9+/alePHiNGrUSOvoIpuioqLo07cvKSkWTCWqo/e7/zd2Y8HSGEKKcvDgQYYOHYrNdu/9DoV42KKjo3nllVdYsWIF/rZ8PHr9SU0L25sMqpHqcU0okvIIx44d4+VOL7Nz506tYwlxV+fOnePzWZ+j8zPj27Tcfd+Ooij4tayIYjIwadIkYmJicjCleJDcvrhNTk5mzJgxtGzZknfffZeaNWsyf/58jEYjer2eU6dO0adPH5566imGDx9OxYoV+frrr6WlzsNERUXxyiu9ib52DVN4JYzZ6Gd7O4qiYC5RHX1Afv744w8pcIXb2r17Nx07duTIkSMUTong0etP4e3w0zqWiw49FRPqUeFGXZKTLLzxxhvMnz9fZiQRbik1NTXz9d5qw+/Jiui8bl34Jzv0Ad74NilLcnIyw4YNk/cRD6GoshwNDoeDAwcOUK1aNemWoIHIyEj69OlDVFQUpvCKmAqVybHbVp0O0k/vxJF0nSZNmjBu3DjpNyjcgt1u56uvvmL2F7NBhbI3ahOeWhqFnOnXtyPsB5JN8fhbg6kX0+rfT7gHScY4DoZsJk2fQsOGDfnggw8ICQnJkdsW4r9SVZV3332XX3/9Fa+axfBrWj7Hbjd57SGsJ6/RsWPHO3aLFA/evdZrbt9yK3K3AwcO0LVrt8zCtkiFHC1sARSdHq9H6qAPCOP333/n9ddfl3mQheYuXbpEr169mDVrFia7N7WvP0nR1DI5Vtg+KAG2EOpcb0VIemG2bdtG+/bt+eOPP7SOJQQAX331Fb/++ivGYsH4Pn73udGzQ1EU/FtWxBDmz7Jly2RpXg8gxa3QzM8//8xrr71GYlIi5hI1/nWhhvul6A14la6PIaQohw4domvXrly4cOGB3JcQd6OqKitWrKDDSx04fPgwhVIjqBfd+p5XHXMHJqeZGnFPUC6hNsk3LLzzzjuMGjUKi8WidTSRh23ZsoXPPvsMfYA3/s9UzfbsCP9GMRnwf74aOm8T48aP5+DBgzl6+yJnSXErHjqbzcbUqVN57733sKvgVaYBxrB7n8fTnhRD2qkd2JPuvXO/otNhLlkTU5HyXLlyhc5duvDrr7/eT3wh7sv169fp168fEyZMwJmuUjWuMZVvNMCoet74AAWFYinlqHv9aQKsIaxdu5aXXnqJPXv2aB1N5EF79+7l3XffBYMuswD1eTB/U/ogH/yeqYLdYadfv36cPn36gdyP+O+kuBUP1ZUrV+jZsyeLFy9G5+2Pd7nGGAKyN22bNeoEjsRrWKNOZOs8RVEwFS6HOaI2aekZDBkyhAkTJpCRkZGt2xEiOxwOB8uXL6dt27b8+eefhKYXod61NhR4iAszPCi+9kAejXmKUklVuHb1Gq+++iqjRo0iISFB62gijzh27Bj9+/fHarfh/3w1DAUCHuj9mYqH4Pd0JZItyfTp04dLly490PsT90eKW/FQqKrKzz//TIcOHTl69CiG0GJ4l38cnfd9vBA57Fn/zSZjSDjeFR5H5xPIihUr6NqtG2fPnr2v2xLibk6ePEn37t2ZPHky9hQHFW/Up3pcE8xOb62j5RgdOkolV6XO9ZYEWINZu3Ytbdu25YcffkDGK4sH6cyZM/Tt25fU9DT82lTBVCJ73XusF+NIXLUP68W4bJ3nVaEwfs0rEB8fz2uvv87Vq1ezdb548KS4FQ9cTEwMgwYN4r333iM1PR1zyZp4layJotduDRGdlz/e5R/DmD+C06dO0bFjR+bMmSPTvIgckZaWxtSpU3n55Zc5evQohVIjqH/tGYqklnL7QWP3K8AWwqMxLSmbUJPkGxbef/99+vbtKy1b4oE4c+YMr/fpQ1JSEn4tK2EuUyDbt5G6/Sy2szGkbs9+44ZXtaL4PF6G6GvXpMB1Q1LcigdGVVXWrl1Lu3bt+P3339H7h+Fd4QmMocW0jgZkzqRgLl4Vr9L1cOqMfPHFF3Tu3Jljx45pHU14qJtXKNo+35bFixfjZfWlZmwzKt9ogMmZ+6eg06GjeEoF6kc/Q1haOLt27aJ9+/Z89tlnpKWlaR1P5BKHDx+mV69exMfF4duiAl4V72/VPNXqyPJvdvk8WhKfBqW4HBlJjx49OH/+/H3djsh5UtyKB+L06dO89tprjBo1ipS0dMwlquFVtgE6L1+to93CEFQQ70pNMYSV5PTp03Tt2pXJkyeTlJSkdTThQU6ePEnv3r157733iLkeS0RSZepFtyYko5DW0R46b4cv1eIfp2pcY/TpJubPn89zzz3HTz/9JF0VxH+ya9cuXnv9NZJTLPi1qox3tf+24M9/5dPgEXyblOX69ev07NVLGkfchBS3IkclJiYyadIkOnbsyJ49e9AHFcK7UjOMYSVRFPe9HKvojXiVqIZX2YZg8mH58uU8++yzrFq1Cofj/j7Vi7zhxo0bjB8/npc7vcy+ffvIn1aM+tfa8EhyNfRo1/VGawoKBdKL0yD6GSKSqhB3PZ5hw4bRs2dPjh8/rnU84YE2bdrEW/36kWGz4f9ctftusc1p3rVL4NeyEomJibz66qvs3btX60h5nhS3IkfYbDZWrFjBc889xzfffAMmX7zK1Me7dF10Js8ZPGMICMO7YlNMRSuRZEll3LhxdO7cWV6sxC1sNhtff/01zz77HCtXrsTHGkDNmGZUi38MH4e/1vHchl418EhyVRpEP0OB1OIcPHiQzp07M2bMGGJi7n06P5F3qarKkiVLGPLuuzgVlYAXamB+JL/WsbLwqlwE/2erkpaRTp++ffnxxx+1jpSn5d1mBZEjnE4nv/76K5999jmRkZdQ9EZMRStjzB+BovPMz06KToepYGkMIUWxXj7mutzcoEED3nrrLcqUydlV1IRnUVWVX3/9lRkzZnDlyhWMqplyibUJTymDTtoL7sjb4UfVG42JT4nmZNBfrFmzhp82/ETXbl3p2rUrPj4+WkcUbshmszFp0iRWr16Nzs9MQLsaD3y6r/tlLlMApX1NLGsOMGLECC5cuMDrr7+OzkPfCz2ZFLfivu3atYsZM2ZkXmJUdBjzl8JYuCw6o1nraDlCZ/TCq2QNHPkjsF4+yp9//sn27dt56qmn6NOnD+Hh4VpHFA/ZgQMHmDp1KocPH84cPJVcgYjkShjV3PE7/zAEWwtQ9/rTRPmc40zgQebMmcOqlavo07cPzzzzDAaDvC2JTElJSQwZMoTdu3djKBCAf9vq6P3de2CmqVgwgZ3rkLRqH/PmzePixYuMGjUKb2/PuYKZG8iriMi2vXv3Mnv2bNdqRIaQopiKlEdndr/BYjlB7xuEd9kG2JNisF4+yoYNG9i4cSPPPvssPXr0oHBh9+j3JR6cS5cuMXPmTDZt2gRAwdQSPJJUTbof3CcFHUVSH6FAWnEu+R3ngnqMDz/8kKVLlzJgwAAaNGjg1n30xYN38eJFBgwYwMWLFzGVKYB/q8ooRr3Wse6JPtiXwM51SP7+IBs3buRK1BWmfjKVsLDsLVgk7p8Ut+KeqKrKnj17mDNnjqv/qT6wIKbwCuh9AjVO93AYAsLQl38Mx40orFeOs2rVKr7//nvatGlDz549KVKkiNYRRQ6Li4tjzpw5rF69GofDQVBGfsok1iDIJm9SOcGgGolIrkKRlNKc9T/E+XOn6devHzVr1mTAgAFUrFhR64hCA5s3b2bEiBGkpKTgXTcCn0aPeNyHHZ23iYD2NbFsPMaxQ8fo2LEjH330EdWrV9c6Wp4gxa24K1VV2b17N3PmzGH//v0A6IMKYipcDr1vPo3TPXyKomAILoI+X2Hs8ZexRZ3ku+++Y+3atbRu3ZoePXpQrJh7zOMr7l9KSgqLFy9m0aLFpKen4WsPpHRiNcLSi+baRRi0ZHZ6UyGxDsVSynE6YB979+6lS5cuNG/enDfeeEP+pvIIp9PJnDlzmDNnDopRj3+bKpjLe+5Ueopeh9+TFTHkDyD+txO8+uqrDBo0iBdffNHjinVPI8WtuC2n08nvv//OV18t4NixowDogwr9XdQGaRvODSiKgjGkKIbgcOw3rmCLOsn333/PunXraN68Od27d6ds2bJaxxTZZLPZWL16NXPmzOHGjRt/F111KZxaSgaLPQR+9kCqxzfhhuk6pwP2sXHjRn777Tfatm3Lq6++SkhIiNYRxQOSnJzMiBEj2Lp1K/ogH/yfr4YhzPO7/SiKgneNYhjy+5P8/UEmTZrEsWPHeO+99/Dycu/+w55MiluRhc1m44cffmDhwoVcvHgRyGypNBYqg94nSOt4bkdRFIzB4RjyFcnsrnD1FD///DM///wzDRo0oHv37tSoUUM+pbs5VVXZtGkTM2d+SmTkJQyqkUeSqlEspRwG1ah1vDwnnzU/tWOfJMYrktOB+1mxYgXr16+nW7dudO7cWWZWyGVOnz7NoMGDiLwUiTEiFP/WVdB55a6/O2N4PgK71iX5+4OsW7eO06dP89FHH0l3tgdEilsBgMVi4bvvvmPJkiWZc08qOgxhJTAVLI3Oy0/reG7vn90VHEkx2K6e4s8//+TPP/+kSpUqdOvWjccee0ymhHFD+/btY/r06a4ZEIpZyhGRXDlPLJfrzhQU8qcXIzQ9nCifM5x1HuKLL75gxbcreL3P6zz77LMys0Iu8MMPP/Dhhx+SkZGBd70IfBp6Xv/ae6X39yKwY21SNp3gxIETdOrUifHjx9OgQQOto+U68sqQx8XExLBs2TJWrFxJisWCojdgLPgIxgKPeNTiC+5CURQMgfkxBObHYYnHevUUhw4dYuDAgRQrVowuXbrQunVrzGaZOkpr58+fZ+bMmfzxxx+AzIDgrnToCE8tQ8G0klz0O85F9Rjjxo3j66+/pn///jRu3DjXFkO5mc1mY8qUKaxYsQKdl5GAdjUwlcr9AzUVvQ6/FhUwFA7E8stx+vXrxyuvvMKrr76KXu8Zs0F4Ailu86hz586xaNEifvzxR+x2O4rRC1N4RYxhJVAMJq3j5Qp6v2C8S9fFmZaMNfoMkZcvMW7cOD7//HM6duxI+/btCQzMGzNNuJMbN24wa9YsVq9ejdPpJF9GAcok1iDQFqp1NHEXBtVIqeQqhKeU5pz/IS5eOMPbb79N9erVGThwIBUqVNA6orhH165dY8iQIRw5cgRDfn/8n6uGPihvdTXxqlQksx/umoPMnTuXo0eP8uGHHxIUFKR1tFxBUVVV1TqE1hwOBwcOHKBatWq5+pOTqqrs3buXRYsWsW3bNgB03v4YC5TGEBKOovOMx5569HecqQnofILwqdhE6zj3zGlLxxZ9DnvMeVS7FbOXF88/9xwvv/yy9Lt6CGw2G9988w1zZs/BkmLB1xZImcQahGYUyZUzIOwI+4FkUzz+1mDqxbTSOk6OSzEkcTpgP9e9L6EoCm3atOGNN96QuUTd3N69exk8ZDAJNxIwVyqMX/MKms1fe2PhDhzRSegLBJCvWz1NMjjTbVh+OIz1bAyFChXik08+kcHId3Gv9Zq03OYBdrud3377jUWLFnHs2DEAdP6hmAo+gj6woFzSe0h0Ri/M4RUwFSqDLfYitugzLF++nG+//ZZmzZrRtWtXaX16AFRVZcuWLXzyyVQiIy9hcpopn/QoRVJKywwIHszXHkC1+MeIN0VzMnAPa9euZePGjfTq1YuXX35Zuv64GVVVWb58OR9//DGqAr4tKuBVNTzPv//ovIz4t61O2s5zXN12hu7duzNy5EhatmypdTSPJsVtLpaWlsb333/PkiVLiIqKAhQM+YpgLFgavV/em6PWXSh6A6YCpTDmL4k9PgrbtdP88ssv/PLLL9SsWZPu3btTv379PP+inxPOnTvHRx99xK5du1D+HixWKqmKLJebiwRbC1A3piVXfM5yxnmATz/9lNWrV/P222/zxBNPyN+RG0hPT2f8+PGsX78enZ+ZgGerYSwSpHUst6EoCj71SmHIH4Bl/WGGDx/O8eOZ/XFl0OT9kZ9aLpSYmMi3337L0mXLSExIQNHpMeaPyBwk5pU7l8j1RIqiwxgSjiG4CI7kWGzXTrN371727t1L6TJl6NG9O82aNZMXt/uQkZHBl19+yVdffYXdbic0vQhlE2via5c+zrmRgo7w1NIUTCvOOf8jXIo6zuDBg2nUqBFDhw6lUCHPXQjA00VHR/POO+9w/PhxDEWCCHi2Gjo/+XB5O6ZSYQR2rUPSdwdYsmQJJ0+eZNKkSdIP9z7Iu2YuEh0dzddff83KlatIT09DMZgwFS6HMX8EilFeTNyVoigYAsIwBIThSE3Edu00p0+fZtiwYXz66ad07dqVZ555Rib8vkd//fUX48aN49KlS3g7fKl8oxFhGeFaxxIPgUE1USapBuEpj3A8aDdbt27lr7/20LdvHzp06CAfFB+ykydP8la/t4iNicWraji+zcqj6KUr0N3o8/kS+HIdLBuO8Ndff9G9e3c+/fRTwsPlNSw75LcsF4iKiuLDDz+kTZs2LFmyBKsTTEUr41PlSUxFykth60H0PoF4RdTCp3JzjPkjuHotmokTJ9KqVSsWL15MWlqa1hHdVkJCAh988AGvvfYakZciKZ5cgXrRbaSwzYN8HAHUiGtK5fgGOFNVPvnkE7p27crx48e1jpZn/Pnnn/Ts2ZPY2Fh8m5bD78mKUtjeI53ZgP+zVfGuF8GlS5fo2q0rhw8f1jqWR5HfNA8WFRXF2LFjefbZZzOnNTJ4YS5ZA+/KLTAVfARFL60Unkpn9sVcvCo+VZ7EWLgsCUkWpk6dSqtWrVm4cKEUuf9jx44dvNDuBdatW0eANZg615+mbFJNWV0sD1NQKJQWQf3oZyiSUooTJ07QpXMX5s6di8Ph0DperrZ69Wr69+9Pus2K/3PV8K5ZXOtIHkdRFHwblcbvyYokJCby6quv8ttvv2kdy2NIceuBrl696ipqv/vuO1SjD+aIWnhXaoYxtDiKrIKVayhGM+YiFfCp0gJT4fIkWixMnz6dVq1asXDhQtLT07WOqCmbzcaMGTN44403SLiRQJnEmtSJaUmALVjraMJNmJxmKibUp1ZMc8x2b2bNmsXrr7/O9evXtY6WK82ZM4cPP/wQvA0EdKyFuXQBrSN5NK+q4QS0q4FVdTB48GBWrVqldSSPIFWQB0lOTmb69Ok899xzmUWtyQdzRG28KzXFGFJURgXnYorBhKlIOXwqZ3Y1SbSkun4X1q5dmydboq5cuUKvXr1YsGABvvYAHr3ekhKWCijysiZuI9hakLrRrSmQVoy9e/fy0osvsWXLFq1j5Srz5s3jiy++QB/kQ+DLdTAWCtI6Uq5gKhlKYKfaKD4mxo0bx5o1a7SO5PbkXcADWK1Wvv76a9q0acPChQtx6IyZLbUVm2IMkXkC8xLFYMRUuBw+VVpgLFSWmNg4Ro0aRadOndi+fbvW8R6arVu30uGlDhw5coTCKaWoc/1paa0V/8qomqgS35gKN+pgSbIwYMAApk+fjtPp1Dqax1u4cCGff/45+kBvAjrUynMrjj1ohvwBBLxUC52PibFjx7J+/XqtI7k1KW7d3I4dO2jXrh0ff/wxyanpmIpWyux+IC21eZqiN2IOr4B35eYYQotz+vQZ3nzzTd58802uXLmidbwHav369bz99tukp2RQOb4BlRLqS99acc8UFMJTy/BodEt87YEsXLiQUaNGYbPZtI7msb7++mumT5+OPsCbgA610Qd4ax0pVzKE+hHwYi0ULwMfjBrFTz/9pHUktyXFrZu6ceMG77//Pm+88QZXoqIwFngEn8rNMRUs7THL5IoHT2fyxqtkDbwrPoE+ID/bt2/nhfbtWbJkCXa7Xet4OW7p0qWMHDkSvcNIrZjmFEqL0DqS8FD+9nw8ev1JgjLCWL9+PYMHD87zfdjvx6ZNm/j444/R+XtlttgGSmH7IBny+2cWuCY9I0aMYP/+/VpHcktS3LoZVVXZsGEDbdu244cffkDnmw/vCk0wF6uMYjBpHU+4Kb1PAF5l6mOOqI3VkTn1Uffu3Tl16pTW0XKEqqrMmjWLKVOm4OX0ofb1FgTaQrWOJTycUTVTM64ZoemF2bJlC2+++SYWi0XrWB7j8uXLjBo9GsWoJ+DFmtIV4SExFAjAv211nKgMHTqUGzduaB3J7Uhx60YyMjIYO3Ysw4cPJ8liwVSsCt7lH0PvI6sqiX+nKArGkHB8KjXDEFqMY8eO0aVLF9auXat1tP/s+++/Z+7cufjYA6h9/Un87EFaRxK5hF41UC3ucQqmlmDfvn2MHDkSVVW1juX2rFYr7777LikWC75PVsAQ4qd1pDzFGJ4Pn8aliYmJYcSIEdJv/H9IcesmoqOjeeWVV1izZk1ma23FppgKlJJ+tSLbFIMJr5I18SpTH7uqMGrUKCZNmuSxfQpPnTrFxIkTM1vZYpvh7ZA3UZGzdOipfKMBIemF+OOPP/j666+1juT2pk2bxvHjx/GqGo5XhcJax8mTvGuXwFQqjB07dvDVV19pHcetSHHrBo4fP06nTp04evQohtBieJdrhM4sl3fEf2MILIB3hcfR+QTyzTff8Prrr5OSkqJ1rGyxWCwMHjwEq9VKpbgGeDt8tY7k9qy6DM74HyDFkAhAhj4Vqy5D41TuT0FHpRsNMDt9mD59OgcPHtQ6kts6ffo0y5cvRx/mh+8T5bSOk2cpioLf05XQB3gxZ+4cYmJitI7kNqS41dilS5d44403uHEjAXPxqphL1JABYyLH6My+eJdrjCG4CPv372fQoEEe1YI7c+ZMIiMvUTK5EmEZRbSO4/bsio2/Qn/mXMBhnLrMuY+t+nT+Cv0Zu+I5z7tWzE5vqsQ1xOlwMmzYMI/6W3mY5s2bB4Dv42VRjPJ+pSWdtwnvBo9gs9pYuHCh1nHchhS3GoqNjc1cWSkhAXOJqhjzR0g3BJHjFL0Bc0Rt9PkKs2vXLj744AOP6J+VnJzM2rVr8bH7UyqpqtZxPMI5/0OkGBNv2Z5iTOSc/yENEnmefNYCFLWU5erVq7LIw22cO3eOX3/9FUOhQIwlQrSOIwBzhULoA71ZuXIlsbGxWsdxC1LcakRVVYYPH86VK1cwFSmPMayk1pFELqYoCl4RtdD5h/DTTz+xdOlSrSP9qx9++IGMjAzCU8qgk5eqexJvjr6vfSKroillAGSp09v48ssvUVUVn/oyJsRdKHod3nUjXAs+CSluNbNjxw7++usv9IEFMRYqq3UckQcoOj3ej9RFMZiYN28eycnJWke6I1VVWbFiJTpVT+FUmcv2XqXr79yn+m77RFa+9kCC0wuyc+dOLl26pHUct+F0Otm6bSu6IB+METIVnzsxVyqMYjawdetWraO4BSluNeB0Opk5cyYApqIV5dPvPVLtGWRcOYYzLbMoc1rTUO0yUCY7FIMJY8EyJCUlsWjRIq3j3FFKSgrnz58jOKMAJqeX1nFEHlQgrTgAhw8f1jiJ+7hw4QLJSckYi+aT9y03o+h1GAoHcu7cORITb+2alNdIcauBCxcucPLkSQz5iqD3DtA6jkdQHTbSjm/FFnUS1MyBMtgzSDu+FdUhgz6yw1ggAsVgcuulG2+2KpscUtgKbZicZgBZ1OEfbs4gYSwcpG0QcVs3n5dDh6R/vRS3Gjh//jwAOr9gjZN4DmvUSZzpt15Gd6YnY406qUEiz6Xo9Oh8AomKiiItLU3rOLd1s7g1qLIqn9DGzd89d+6+87AdOXIEAENhWVjIHRmKBAH//zzlZVLcasBV3Hr7a5zEcziS7jx/3932idvTefmjqqrb9ieUKZiE1hQ187K71WrVOIn7yMjI7AameBk1TiJuRzFnPi/yOyvFrSa8vb0z/+OwaxvEg6jWO7cw3m2fuD3179891++imylVqhRGg5EEk3xwEdpIMF8HoEKFChoncR86nZQM7i1z2Wh5nqS41UTp0qUBcKRJp2+hDWdaEl5eXoSHh2sd5ba8vLyoUrUKyaZ4bIq0QoiHL858DZ2io2bNmlpHcRuuQWSqqm0QcXt/Py0y2E+KW02UKVMGRVFwJF5HlRcJ8ZA5rWk405IoW7asW3/Cr127NioqcV5RWkcReYxNsZJoiqVsubL4+0v3sZu8vDIHeKoZnn3V0ZlmJWXraRxxmYMFnZZ0nGme/yFatWY+L2azWeMk2nPfd7ZcLCgoiKZNm+JMuYEjWVYTEQ+X7doZUJ20bdtW6yh31aJFC3SKjnP+R1CRD4Hi4bngfxSn4qBly5ZaR3ErJUtmLjbkiPHcGSScGXYSl+4mbcc5sGeu1KimWElcuhunhxftN5+XiAiZG1yKW4306NEDAFvUCWm9FQ+N05aOPfYCBQsW5KmnntI6zl2VKFGC1m1aYzHe4Jr3Ba3jiDwiQ5fGJb8T5M+fn/bt22sdx62UKZO5cps9xnNnkEjbcRZH3K0LmjjiUkjbcVaDRDnHfj3zebn5POVlUtxqpHz58jz++OM4kmOxx5zXOo7IA1RVJeP8PlSHnVdeeQWj0f1HPL/22msYDUbOBhzEiUPrOCIPOOd/CIdi57XXXpPLu//j5ngR+/UkjZPcP9ul+Pva5wns15Pw9vGmSJEiWkfRnBS3Gho2bBiBgUFYI4+4Vt0S4kGxx1zAkRhNgwYNeP7557WOc08KFSrESx1eItWQzKnAfVrHEblcnDmKSL9TlChegtatW2sdx+34+/tTvHhx7FGJqA6n1nHuiyM5/b72uTtnmhVHbAoVK1R067EUD4v8BDQUGhrKyJHvozodpJ/djWqXuT3Fg+GwxGONPExgYCAjR470qNG0ffv2pVSpUlzyO8F1r0it44hcKkOXxpHg7RgNRsZPGO8RVza0UK9ePdQMO/arMtuPO7FdjAdVpV69elpHcQtS3GqsSZMmdOnSBWdaUmaBq3rmp2HhvpwZKWSc2YmCyvjx4wkLC9M6UrZ4eXkxceJEzCYzx4J3kK67tb+cEP+FisqRfNvJ0KUx4O0BlCtXTutIbqtu3boAWM/LYGh3cvP5uPn85HVS3LqBfv36Zfa/TbpOxsWDMsBM5BjVbiX99E6ctgyGDh3qsZ/qS5UqxeAhg7EqGRwI2YxD8exRzcK9nAnYT5xXFI0bN6ZDhw5ax3FrtWrVwmAwYLsQp3UU8TdVVbFdjCMoXz7Kli2rdRy3IMWtG9Dr9YwbN47y5ctjj7mA9cpxrSOJXEB12Ek7vQNnWhJdunThhRde0DrSf/L888/z/PPPk2SK43C+bajIVQ7x3132Oc15/6OUKF6CMWPGeFSXHS34+PhQtWpV7NeScKZLVzp34IhPxZmUTt06daS/7d/kp+AmvL29mTlzJsWKFcN29STWa6e1jiQ8mOp0kn52N05LPK1bt6Z///5aR/rPFEVh6NCh1K1bl+vekTLATPxnseYojufbRVBQEDNmziAgIEDrSB6hTp06oKoeP7tAbmG7mNmKXqdOHY2TuA8pbt1IcHAws2bNIn/+/Fgjj2CLuah1JOGBVFUl49weHInRPPbYY4wcOTLXfJo3Go1MmjSJRx55hIt+x7noK1c5xP1JMsZzKGQLRpOR6dOnu+1S1O7oZr9O6ZrgHm4+D48++qjGSdxH7njHy0UKFSrErFmzCAwMIuPCfuzxV7SOJDyIqqpkXDyA/cYVatasycSJEzEYDFrHylH+/v7MmDGDsLAwTgbtIcr7nNaRhIdJ0SexL3QTTp2D8ePHU7lyZa0jeZTy5cvj6+uL7fINraPkeaqqYr9yg/DwcAoVKqR1HLchxa0bKlmyJJ999ik+Pt5knN+DPfG61pGEh7BePoY95gIVKlRg2rRpuXYS+oIFCzJr1iwC/AM4GrxdpggT9yxdl8q+sF+x6tIZPmI4TzzxhNaRPI5eryciIgLnjVRUp/R915KaZsOZZnMtsCEySXHrpipUqMD06dMxGgxknN2NI1XmFBR3Z7t+Dtu1U5QoUYKZM2fi6+urdaQHKiIigk8/+xSzlxeHQ7YSb4rWOpJwczYlg32hv5KmT6Ffv34es5iJOypZsiSqw4kzIU3rKHmaI84CZD4f4v9JcevGatasybhx48BpJ+P0DpxWeRERt2dPuErGxUMEh4Tw6aefki9fPq0jPRSVKlVi6tRPUPQKB0P/INkgA1zE7dkVG/tCf8NiTKRLly50795d60gerUSJEgDY42TeaS05/v75S3GblRS3bq5p06a8/fbbOK1ppJ/egep0aB1JuBlHaiIZZ/fg5WVm5owZFC5cWOtID1WdOnUYN34cdp2NfWG/kaqXpaxFVk4cHAzeQqIplmeeeYYBAwZoHcnjBQcHA6BmyHRgWro5HdvN50NkkuLWA7z88su0a9cOZ2oiGZcOaR1HuBHVYSPj7G5Up50JEyZQvnx5rSNponnz5rz33ntk6NLYF7aJDJ1c5RCZbq4+FucVxWOPPcaIESNkLtsckJGRAYBi0GucJG+7+fO/+XyITFLcegBFURg8eLBrkQdb7CWtIwk3oKoqGRf240y30KNHDx577DGtI2nqhRdeoE+fPqTqk9kf8jt2RVqUBJwO2M81nwtUr16dCRMm5LrZQ7TiKm6NUtxqSTFmlnFS3GYlxa2HMJlMTJ48GT8/P6yXDkn/W4E9/gr2+CvUqFGDPn36aB3HLbzyyiu0bduWJFMcR/L9iYosZZ2XXfY5zYW/Vx+bNm0aXl5eWkfKNSyWzIFMGKSM0NTfLbfJydId65/kt9KDFClShIEDB6I6bFgvHdY6jtCQardhizyM2Wxm9OjR0hr1N0VRePfdd6lTpw7XvSM5HbBf60hCI3GmqxzPt5ugwMzVx/z9/bWOlKvs27cPFDCEyc9VS4YCmavq7d8vr3X/JMWth2nTpg3Vq1fHfuOKzH+bh1mjjuO0pfPqq69SpEgRreO4FaPRyOTJkylZsiQX/I9yxees1pHEQ5aqT+ZQ6BYMRj1Tp02V1cdyWHJyMvv278NQKAidj0nrOHmaPsQXXaA32/7cht1u1zqO25DmHg+j0+l499136dChA9arJzAE5tc6knjInLYM7DHnCQ8Pp3PnzlrHcUs3VzHr1KkTJ9TdBFpD8LMHaR3roVi4cOFtt/fu/NpDTqINJw4OBW/BplgZO3IsVatW1TpSrrNjxw6cDidepcK0jpLnKYqCqVQYyfsucfjwYapXr651JLcgLbceqEyZMjRs2BBnchwOi8zrmdfYrp9FdTrp2rUrRqNR6zhuq0iRIowZMwaHYudgyBYZYJZHnAzcS5IpnrZt29KqVSut4+Q6qqqyYsUKAEyPSHHrDkx/f8i4+bwIabn1WN26dWPbtm3Yrp9H7yfz2+UVqqpij7lAYFCQvHHfg8cee4wuXbqwePFijgftpvKNBlpHeuC6det22+0mcv9gqmteF4n0O0mZ0mUYNGiQ1nFypU2bNrF3715MpfNLf1s3YSwRgqFwID/99BMvvvgi1apV0zqS5ty+5dZisTBu3DiaNGlClSpV6NChA4cO/f9cr6qqMn36dBo2bEiVKlXo3r07Fy5c0C7wQ1KjRg3CwvLjSIpGVWVEeF7hTLmBasug6RNP4O3trXUcj/Dmm29SpUoVrvqcI8Z8Res44gGx6jI4kW83Xl7eTP5ossyM8ACkp6fzydSpKHodvk3Kah1H/E1RFHyblgNg8uTJOByy2JPbF7cjRoxg+/btTJ48mXXr1tGgQQN69OhBdHTmOvJz585l8eLFjBo1im+//RZvb2969eqV6+d8UxSFhg0boNoycKYmaB1HPCSOxMzf+wYNcn8LZE4xGo2MHDkSg8HAiXy7pHtCLnUqYC9WXTpvvvkGxYoV0zpOrvTll19y7epVvGqXQB/ko3Uc8Q/GQkGYKxXmxIkTrFq1Sus4mnPr4jY9PZ1ffvmFwYMHU7t2bYoXL85bb71F8eLFWbp0KaqqsmjRIvr06UOzZs0oV64ckydP5vr16/z6669ax3/gatSoAYAzJUHbIOKhcfz9QaZWrVraBvEwERER9OjRgzR9Cmf9ZZW/3CbedI0o37OUL1+el156Ses4udKmTZuYN28e+kBvfOqW1DqOuA3fxmXQeRuZMmUKe/fu1TqOpty6uLXb7TgcDsxmc5btZrOZffv2cfnyZWJiYqhfv75rn7+/P1WrVs0Tc77dnALKaU3VOIl4WNSMNPz8/WXOzvvQs2dPihUrxiX/46QYkrSOI3KIisqJoL/QKTpGjBiBXi8rZuW0I0eOMHz4cBSTAf+21VFMMlzHHen8zPg/Vw2H6mTgwIFcupR3VzN1699QPz8/qlevzueff05ERAShoaGsX7+eAwcOUKxYMWJiYgAICQnJcl5ISAixsbHZvj9P66cSFpY5QlKV1cryDNWWRoHCRT3ud9UdGAwG3nrrLQYPHsw5/8N5YnBZXhDtdQmLMYFn2jxDmTJl5G8jh0VFRdG/f3+sdhsB7WrIIDI3ZywajO+TFUj68QhvvvkmX331FUFBQVrHyjH3+vft1sUtZHaOHjZsGI0bN0av11OhQgVatWrF0aNHc/y+Dh/2rFW/4uNvTgOmaJrjYbnT/J3de736kJNoKyMjgwMHDmgdwyMFBgZStGhRIiPPE5FUGV9HgNaRxH+gonIu4BA6RUfdunXl7yKHxcXF8cnUqdy4cQPfFhUwlQzVOpK4B16ViuBISOPy9rP07t2b/v374+fnp3Wsh8rti9tixYqxZMkSUlNTsVgs5M+fnwEDBlC0aFFXy2VcXBz58///YgZxcXGUK1cu2/dVuXJlj7qkdf78eQAUD8os/iOdHlVVZaqX/6Bfv34MHjyY8/5HqJRQ/99PEG4rxisSizGBNq3b0Lx5c63j5CoXLlxg+PDhXL9+He96EXhXK6p1JJENPg1K4bSkc+HQBWbMmMGnn36apU7yVA6H454aIt2+uL3Jx8cHHx8fEhMT2bZtG4MHDyY8PJywsDB27NhB+fLlgcypww4ePEjHjh2zfR96vd6jituEhITM/+jzxvKHd5q/UzGYb7s9N1IMJuLj41EUBZ3OrbvMu60nnniC4sWLc/nCRcol1sKg5o2/n9zosu8ZALp37+5Rr93u7tixY/R94w2SEhPxebwsPo+W0DqSyCZFUfB7siI6LyPnd5+n1yu9mPX5LIoXL651tIfC7d8dt27dypYtW4iMjOTPP/+ka9euRERE0LZtWxRFoWvXrsyaNYtNmzZx8uRJhgwZQv78+WnWrJnW0R+4EydOAKD3CdQ4iXhYdN6BpKWlERkZqXUUj6UoCm3atMGh2In2zrsDLjxdhi6VOK8oqlSpQsmSMno/p+zevZvevXuTlJyEX8tKUth6MEVR8H28LD6PleHa1Wv06NmD48ePax3roXD74jY5OZkxY8bQsmVL3n33XWrWrMn8+fNdy4727t2bzp07M3LkSF544QVSU1OZN2/eLTMs5EbHjh0DQOcbpG0Q8dDcfK4fRJ/zvKRVq1YoisIVn7NaRxH3KcrnPCoqbdq00TpKrrFy5UreeOMN0m0Z+D9bDa/KRbSOJHKAT52S+D1ZkYSERHr26pUnpkp1+24JTz/9NE8//fQd9yuKQv/+/enfv/9DTKU9m83Gtm1/oph8UEwymXZeoffPHNCxZcuWu/5diLsrUKAAtWvXZvfu3WTo0jA7ZbU3T3Pd+xIGg4EWLVpoHcXj2Ww2Jk+ezKpVq9D5mAh4rhrG8HxaxxI5yKtqODo/M5Z1hxgyZAivvPIKr7/+eq7t3pY7H1UesHv3biyWZAz5CqMoeWO2BAE67wB0Xn5s2bKV9PR0reN4tJvzY8ebr2mcRGSXTbGSZIqjSpUqMufzfxQfH8/rr7/OqlWrMBQIILBrXSlscylTqTACutRBn8+HefPmMXDgQCwWi9axHggpbj3U2rVrATAEy2WjvERRFPTBRUhPT8sTl5YepDp16gBS3HqiG+ZoVFTXcyjuz4kTJ3i5c2f279+PqVxBAjs9ij5ArmLkZoYQPwK71MVYIoTNmzfTrXu3XLnYgxS3HigqKopNmzah8wlC5yufsPMaY2gJUBQWL16Mqqpax/FYpUuXJigwSIpbD3TzOXv00Uc1TuK51q1bR/fu3YmOvoZP49L4t6mCYpQZJ/ICnZeRgBdq4F27BOfPneflzp3ZsmWL1rFylBS3Hujrr7/G6XRiLFhauiTkQTqzD4Z84Zw+fZqdO3dqHcdj6XQ6ylcoT5rBgl2xah1HZEOy8QY6nY6yZctqHcXj2Gw2Jk6cyAcffIBdpxLwQk186kbIe0keo+h0+DYpi3/rKqSmpzFgwAC++OILnE6n1tFyhBS3HiYqKooVK1eiM/tiCC6sdRyhEWOh0gB89tlnuebFSAtlypQBINmYoG0Qcc9UVCzGGxQvXhwvLy+t43iUmJgYXn31Vb799lsM+f0J7FpXVh3L48wVChHYuQ76IB/mzJnDgAEDSE5O1jrWfybFrYf5/PPPsdtsGItUQFHk6cur9D6BGEKKcuzYMTZu3Kh1HI9VunTmh4Rk4w2Nk4h7laFLxaazup47cW+OHj1Kp5df5uDBg5kFzcuZBY0QNz/oGCNC2bZtGy937syFCxe0jvWfSHXkQY4cOcKPP/6IzidIBpIJTEXKg6JjxowZpKWlaR3HI5UoUQKANL3nt1TkFamGzOcqr6y0lBM2btxIr169iIuLw7dpOfxaVZb+tSILnZeRgHY18K4XweXISLp268quXbu0jnXfpLj1EA6HgwkTJgBgLlZF+kcJdGZfjAUf4erVqyxYsEDrOB6pUKFCAKQZUjROIu7Vzefq5nMn7kxVVebNm8e7776LXXFmDiKqWVzeP8RtKYqCb6PS+LWuTEpaKm+88QYrV67UOtZ9keLWQ6xevZrjx49jCC2G3j9E6zjCTZgKlUVn8uGrBQty5XQuD1pQUBBms5l0vRS3nuLmcyXF7d1lZGQwYsQIPv/8c/RBPgS8XEf614p74lWhMAEv1QZvA+PHj+ejjz7C4XBoHStbpLj1ALGxscyYORPFYMQUXknrOMKNKHoDpmKVsdtsTJgwQaYGyyZFUShYsCAZhlSto4h7lKHPfK4KFiyocRL3lZGRwaBBg9iwYQOG8HwEdq6DIdRP61jCgxiLBBHYuS76MD+WLVvG6NGjParAleLWA3zyySekWCyYilREZzRrHUe4GX1QIfSBBdm1axc//fST1nE8TnBwMFZdBirywcATWHWZK/MFBwdrnMQ93Sxs//zzT0ylwgh8sRY6H5PWsYQH0gd6E9ipDoYiQaxfv96jClwpbt3czYJF5xuMIayE1nGEG1IUBXPxqig6PVOmTMkV07g8TMHBwag4Za5bD2HVpWM0GPHzk5bI//W/ha3/c9VQDPI2L+6fzmwg4IWargJ3zJgxHlHgym+9G7PZbEyePBlQMJeoJoMAxB3pzD4YC5fjxo0bzJkzR+s4HuVmC2CGPl3jJOJeWPXp5MuXT14P/4fD4WDw4MH8+eefGEuF4f9sNRS9vMWL/+6fBe66dev48MMP3b4LnPzmu7FvvvmG8+fPY8xfEr1PoNZxhJszFiiFziuzf9TZs2e1juMxAgMz/7ZsugyNk4h7YdNnEJQvSOsYbmfBggVs27YNY0QoAc9Ki63IWa4Ct1Ag33//PevWrdM60l3Jb7+bSk5OZu7cuSgGU+Z8pkL8C0Wnx1S0Mk6nk88++0zrOB5DilvP4cSJTbG6njOR6ejRo3zxxRfoA7zwb11FClvxQOjMBvyfrYrOy8ikSZOIjIzUOtIdGbQOIG5v+fLlJCcnYwqviGKQwQDi3ugDC6DzC+GPP/7g+PHjlC8vH4z+jRS3nsOuy+wXLcXt/0tNTWXY8GE4nA4Cn66BzsuodSS3t3Dhwttu796390NO4nn0Ad74tqhA8tqDDB8+nPnz52M0ut/vnHy8c0OpqaksXrwYxWjGmD9C6zjCgyiK4mrpnzdvnsZpPEO+fPkAsEpx6/ZufgAJCgrSNogbmT17NpGXIvF+tCTGYjKDhHjwzOUKYq5UmCNHjrB8+XKt49yWtNy6od9++w2LxYKpSHkUvTxFInsMAWHofPOxefNm4uLiCAmRRT/u5mZxa9PJgDJ3d3MasJvPWV5ns9lYu3YtOj8zPg0f0TqOx+jWrdtttyu+cpX0Xvk2LYf1ZDRr1qyhc+fObjfAU1pu3dAPP/wAgCGkqMZJhKcyhBTD6XTy888/ax3F7d1sBZSWW/d38zmS4jbT7t27SUxMxFS2oMyMIB4qndmIsVQY58+f5/Tp01rHuYX8NbiZlJQUdu/ejc4vGJ3ZV+s4wkMZg4sACn/88YfWUdyeq1uCPk3jJOLfSMttVjc/vJoryGpt4uEzl8/8vXPHxYOkuHUz586dQ1VV9L7Sd0rcP8VoRvHy5cyZM1pHcXve3t74+fqRoZPi1t3dXHo3LCxM4yTuYceOHej8vTAUlAF24uEzlQxDMejZuXOn1lFuka3i9r333sNisTyoLAJc85PqvAM0TiI8nc47gISEBOLj47WO4vbyF8hPhkGKW3d3s7jNnz+/xkncg96gB4PO7fo7irxBMehAAYPB/cYGZau4XbNmDRkZ0i/tQUpJSQGQ6b/Ef3bzd0g+kP67/PnzY9Wl48D9l5XMy9L/Lm5DQ0M1TuIe/Hz9wCq/s0IbqlNFtTnw9XW/LpTZKm7dfbm13ODm/I2qQ9a5F/+Nas/8HQoIkKsA/6ZIkSIApBnkg4A7SzNYCAsNw8vLS+sobsHX1xc1w651DJFHqdbM3z13LG6z3ZackpKC2Wy+6zF+fn73HSivuzlyW7XKJVLx36jWNHQ6Hf7+/lpHcXtFi2bOTJJqSMLPLv0X3ZETJ2mGFMoVK611FLdRuHBhDh8+jC0qAWPhIK3jiDzGdimzy1uhQoU0TnKrbBe3Tz755B33qaqKoigcP378P4XKy6pUqYLBYMB+4yqmwuW0jiM8lNOahjPlBrVq1UKv12sdx+39f3GbrHEScSfp+hRUnK7nSkD79u35+eefSdt9HuNz1bWOI/IQVVVJ230eRVFo37691nFuke3idsaMGbL04QMUEBBA3bp12bZtG850CzovaQUX2We/EQVAixYtNE7iGUqUKAGAxZCgaQ5xZxbjDeD/nysB1atXp1KlShw5egRHfAr6YPe7PCxyJ/uVBOxRiTRp0oRixYppHecW2S5ua9SoISsePWCtW7dm27ZtWC8fw+uRR7WOIzyMardhv3YKk8lE06ZNtY7jEYoVK4aX2Ytk6w2to4g7SPq7uC1XTq5o3aQoCt26dWPw4MGkbDuDf5sqMnOCeOBUp0rqtsxpJu+02pvWZJ5bN9SsWTOqVauG/cYV7AnXtI7jFhST933ty4usV47htKbz6quvymT390iv11OmbBksxgScMmOCW0o2ZvbvK1OmjMZJ3Mvjjz9O5cqVsZ64RtpfF7WOI/KAlD9OYrsUz+OPP06VKlW0jnNb2SpuCxcujE4n9fCDptPpGDFiBAaDAevFAzhlcBn6gDtP2n63fXmNPeEatuvniIiIoEuXLlrH8SjlypVDVZwkGz239dbLcefL0nfb5+5UVBLNsRQsWFA+sP0PvV7Pxx9/TP78+UndfBLr2RitI4lcLP3QZdL3XCQiIoIxY8ZoHeeOslWp/vbbb64Xlvj4eA4fPsyRI0e4ccNz3wzcVUREBG+99RZOaxrpp7a7pnXKq0yFy6LzunXUv87bXwbe/c1hiSfj7F+YvbwYM2YMRqNR60gepWbNmgDEm6M1TnL/gjMK3Nc+d5dqSMKqS6dGjRpaR3FLoaGhTJs2DbPZC8u6Q9hjZGCkyHm2yBukbDxOYFAQ06dPd+uZsbLdDHv69GlefvllGjRowIsvvkj79u2pX78+Xbt25dy5cw8iY57VuXNnOnbsiDMtibQzu1AdeXc+Q0VvxLt8Y4yFy4Hy9+h/gxnvco1R9O63OsrD5kxLIv30DnSKypSPPqJChQpaR/I4N4vbGx5c3EYkV8HXduuAX19bIBHJ7nn58F7EmzKfk1q1ammcxH2VK1eOD8eOxWm1k/TNHmxXE7WOJHIR6/lYklftQ6/o+HjKFNfc4O4qW8VtTEwMnTt3Jj4+nqFDhzJnzhxmz57NkCFDiImJ4eWXXyYuLu5BZc1zFEVh4MCBPPnkkziTY0k7uQ2nLV3rWJpRDCbMRcqj885swdWZvGUlN8CRFEvaia2odisffPABDRo00DqSRwoODiYiIoIE83WP7XdrUI08GvMkEUlV0DkzPwSaHF48GvMkBtVzW/Ljva4C//8BRNxe06ZNGTFiBKTbSVr+Fxmnr2sdSeQC6YevkLRqHwZ0TJw40SOuoGSruF2wYAFFihRhzZo1dOvWjUaNGtG4cWN69OjBd999R6FChViwYMEDipo36XQ6xowZw/PPP48z5QbpxzfjSEvSOpZwE7bYS6Sd+hM9Kh9++CGtW7fWOpJHa9CgAXbFRpzZcwdyGlUzjyRXxffvxSjMDh+M6t0X3nFnDuzEekVRvHhxwsPDtY7j9tq2bcv06dMxG4wkrzlA2r5LWkcSHkpVVVK2ncGy4QiBAYHMnj2bJ554QutY9yRbxe327dt55ZVXbrtCmZeXF7169WLbtm05Fk5kMhqNjBgxIrMPbkYq6ce3YI+/onUsoSHV6SQj8ggZ5/cS4O/HrFmf8/TTT2sdy+PdnDrturcUBO4izusqDsVO06ZNZZqre9SgQQPmz5tPSEgIKb8ex/LbCVSHU+tYwoOoNgeWH4+Qtv0sRYsVZeGCBVStWlXrWPcsW8VtZGQkFStWvOP+SpUqERkZ+Z9DiVspikKPHj2YNGkSZqOe9LO7Sb+wP0/3w82rnOkW0o5vxnbtNMWLF2fhwoVyuTaHVKpUibDQMGK8L+NEigF3EP33Bw1PaTFyF+XLl2fRwoVERESQvuciiUt347iRqnUs4QHs15NIWLSDjKNRVKlahQVfLXDLhRruJlvFbUpKyl1Hx/n6+pKaKn88D1Lz5s1ZtmwZ5cqVwx5zgbRjf+BITdA6lngIVFXFFnuRtGO/40xN4Pnnn2fp0qUUL15c62i5hk6no3mL5lh16cR4XdY6Tp5nU6xc975EeHg45cuX1zqOxylUqBCLFi3imWeewX41kcSFO0g/GqV1LE3p/b3ua19eoKoqaXsvkrh4F464FLp06cKc2XM8cvq9bM+WkJKSgsViueOXqqoPIqf4h5utdV26dMGZnkzasT/IuHwU1emZg2DEv3NmpJJ+egcZ5/fh4+3FpEmTeP/99/H2lgUsclrbtm0BuOx7WuMk4qrPeRyKnbZt20qXhPvk4+PDqFGjmDBhAt5GM5YfDpO8/hDOjLx51c9YLPi+9uV2zlQrSav2k7LpBPkCg/jss894++23MZk8c9B2tuZQUlWVJ5988q775QXo4TAajbz99tvUr1+fsWPHEhV1CseNKMwlqqP3D9U6nsghqqpiu34O25VjqA479erVY8SIERQqVEjraLlWREQENWrUYN++faTqk/Fx3Dq/snjwVFQu+57CYDDw7LPPah3H4z355JNUrlyZYcOGcejQIexXEvB9qiKm4iFaR3uovOuVwno2BkdcSpbt+lA/fOqX0iiVtjJORZOy8TjOlAwaNmzIqFGjCA727EI/W8XtokWLHlQOcZ/q1KnDihUrmDVrFl8vXUraia0YwkpgDq+AYvDcEdICHKkJZFw4iDMlnsDAQAYNGsTTTz8tHyAfghdeeIF9+/Zxye8E5RJrax0nT4o3X8NiTODJpk965GVRd1S4cGHmzZvH3LlzmT9/Pknf7MFcJRzfJmXQmT13qrjs0JkNBHZ6lLS9l0jbfR7sThRfE4Eda6OY8tac6c6UDCy/Hsd6MhqTyUS/QYPo2LFjrniPydYz+eijjz6oHOI/8Pb25p133uHJJ59kzJgxnD59GseNKEzhFTCElsgVv6h5iWq3Yr1yHNv184BKy5YtGTRokLzBP0TNmjXjs88+4/Ll05RMrojZ6aN1pDxFReWs/yEAevTooXGa3MVgMNCnTx+aNGnCqFGjOHXoFLZzMfi2qID5kfxax3sodN4mfBs+ktmCG52Ezs8LnbdnXn6/H6qqknE0itTfTuJMt1GjRg1GjhzpcYPG7ua+PqYkJyfz559/cuXKFRRFITw8nPr167v1Umx5QcWKFfn666/59ttvmTVrFikXDmCLuYi5WFX0flIYuTtVVbHHXcJ6+SiqLYOSJUvy7rvvyodKDRgMBnr16sWYMWO44HeMskmyMtbDdMMUTYL5Ok2aNKFMmTJax8mVypUrx+LFi1m8eDGz58wmefV+MsoVxK9pOXS+ctUvt3IkpWH5+Ri287H4+PgwYNhg2rZti06X7SFYbi3bxe3333/P2LFjsVgsWbb7+/szevRomWtTYwaDgU6dOtGiRQtmzJjB+vXrSTv+B4bQ4pjCK6Az5u3RoO7KYYkn49IhnCk38PLy5vU3BtCxY0eMxrxxqdAdtWrVirlz5nJZPU3xlPJ4OXy1jpQnqKicCTgAwCuvvKJtmFzOaDTSs2dPmjRpwpgxYzh48CAJF+PweawM5spF5KpfLqI6naTvvUTqtjOoNgcNGzZk2LBhFCxYUOtoD0S2SvWjR48ybNgwmjVrxpo1azh06BAHDx5k1apVNGnShCFDhnDixIkHlVVkQ2hoKGPGjGH+/PmUKVMGe+xF0g7/ivXaaVSnzN/pLpzWdNLP7yXt+GacKTd46qmnWLPmO7p27SqFrcaMRiN93+iLQ7FzKmCf1nHyjGveF0gwx9C8eXOZ/ushKVmyJPPnz+fdd9/FW2fC8tNREpf9hT3W8u8nC7dnu5pA4qKdpPx+kkA/fz788EOmT5+eawtbyGZxu2TJEpo2bcrEiRMpV64cJpMJs9lMxYoVmTx5Mk888QQLFy58UFnFfahevTpff/01I0aMIMDfF2vkEdKObsKe4LnLi+YGqtOB9eop0o5sxB57iXLlyvHll18yfvx48ufPG/3ePMHTTz9NlSpVuOZzgXhTtNZxcj27YuNU0D7MJjMDBgzQOk6eotPpeOmll1i9ejXNmzfHfvkGiQt3kLLlNKpNppn0RM4MG5aNx0hcsgv79WSef/55vlv9XZ4YmJyt4nbfvn106NDhjvs7dOjAvn3SwuFu9Ho9bdu2Ze3339OpUycUWxrpp3eQdmo7zvRkrePlKaqqYk+4StqRTVgvHyXQ34+RI0eyePFiqlWrpnU88T8URWHIkCEoisKJoN2yatkDds7/MBm6VHr26inT3WkkLCyMSZMmMWPGDArmL0DaznMkfLUd6/lYraOJe6SqKhknrpEw70/S90cSUTKCL7/8kvfff5/AwECt4z0U2Spur1+/TokSJe64v0SJEkRHS+uGu/L392fQoEF888031K1bF0diNKlHfiPj0mFUu03reLmeMy2Z9FPbST+9E8WeTufOnfn+++957rnn0Ov1WscTd1ChQgXatWuHxZjAOf/DWsfJtRKMMVz0P0aRwkXo0qWL1nHyvIYNG7Jy5Uq6d+8OyRkkrdhL8rqDOC0ZWkcTd+FISCVp1T6S1x7EaIe33nqLZcuW5bnGk2wNKEtLS8NsvvMoSpPJREaG/OK7u4iICD777DO2bNnCxx9/zOXLZ7DHX8YUXhFDSNFcf7niYVMddqxRJ7BFnwFVpX79+gwcOJCSJUtqHU3co/79+/Pnn39ynsPkTw8nwJa3Jr5/0BzYORq8HRQYPWY0Xl4y8NUdeHt7069fP1q2bMm4ceM4dOgQtvNx+DQujblquLxXuBHV4SRtzwXS/jyHandQv359hg4dSnh4uNbRNJHt2RK2bt2Kv//tV+xJTpZL3J5CURQee+wx6tWrx5IlS5g7bx4Z5/dii7mAuXhV9D5549LFg6SqKvYbV7BFHsFpTaNIkSIMHjyYRo0ayZuCh/H19WX06NG8+uqrHMn3J3Wut0KPtLbnlDOBB0gxJNGlcxdq1KihdRzxP0qXLs2XX37J6tWrmT5jBpZfjpF+NAq/FhUwhMkKflqzXUnA8stRHDEWQkJCGDx4MM2bN8/T7zPZLm6HDh36IHIIjZhMJnr27MnTTz/NJ598wq+//krasd8x5o/AVKQ8il5G7N8PZ3oyGRcP4kiKwWQy0atPH7p27XrXKx/CvdWqVYuOHTuybNkyTgbuoUJiHa0j5QoxXpe56HeckiVK0qdPH63jiDvQ6XS88MILPP7443zyySf89NNPJCzcgXedkvjUi0AxyIe9h82ZYSd1yynSD0SioNC+fXvefPPNOzZA5iXZKm5lmq/cq2DBgkyePJmdO3cyceJELl06i+PGVUzFq2IIyr3TheQ01enEdu00tqsnUJ1OHnvsMQYNGkSRIkW0jiZyQL9+/di/fz8nTpwgyBpG4bQIrSN5tDS9hSPB2zGbzEycNFG6I3iA0NBQxo8fT+vWrRk/fjxRO85hPRmNX8tKGIsEaR0vz7CeiyHll2M4ktIpVaoUI0aMoGrVqlrHchs5uiRFXFwcX3zxRU7epHjI6taty7fffsvrr7+OTrWRfnoH6Wf/wmmTvtT/xmGJJ+3Y71ivHCM0JISpU6cydepUKWxzEbPZzOTJk/Hz9eN4vl1YDAlaR/JYThwcDN6CTclg2PBhlC5dWutIIhvq16/PihUr6NSpE84bqSQu3YVl03FUq13raLmaM81K8g+HSVq5DyXVzuuvv87SpUulsP0fOVrcxsTEMH369Jy8SaEBk8nEq6++yvJly6hatSr2+MukHfkVW/xlraO5JdXpICPyMGnHN6OmJ9O+fXtWrVrFY489pnU08QCEh4czZuwYHIqdAyGbsSnywS+7VFSOB+0myRTH888/T5s2bbSOJO6Dt7c3gwYN4ssvv6RkiZKk772UOW3YxTito+VKGSevkfDldjKORlGxYkWWLl3Kq6++Kgv+3EbuWkxY5KiIiAjmz5/Pe++9h5dRT8bZv0g/uwfVbtU6mttwpCaSduwPbNfOULx4cdfPy8/PT+to4gF6/PHH6dmzJ6mGJA4Gb5H5b7Ppku8JrvieoUKFigwePFjrOOI/qlq1KsuWLctcLtliJembPVh+PS6LP+QQZ7qN5PWHSP4+c3qvd955hwULFvDII49oHc1tSXEr7kqn09G+fXu++eabv1txI0k7+hv2pBito2lKVVWsV0+RfvwPnGlJdOjQgaVLl+a5uQTzsr59+9KkSRPiva5xMnCP1nE8Roz5CqeC9hIWFsbUqZ9IP9tcwmQy0bdvX5YsWUKpUqVI33eJhEU7sEcnaR3No9kuxZO4YAcZx65StWpVvvnmGzp37ixzo/8LKW7FPQkPD2fevHm8+eabKA4r6Se3YY06gaqqWkd76FS7lfTTO7BePkpIcAifffYZQ4YMwdvbW+to4iHS6XSMHTuWMmXKEOl3kku+MuD23yQbbnA4ZCsmk4lp06YRFhamdSSRw8qWLcuSJUt4+eWXccSlkLhkF6m7zqM68957xX+hOpykbD5F4jd/QYqVvn37MnfuXIoVK6Z1NI+QrdkSJkyYcNf98fHx/ymMcG96vZ6ePXtSt25dBg8ezNUrx3FY4vGKqIViMGkd76FwpNwg4+xunBmpNGzYkDFjxhAUFKR1LKERHx8fpk2bRpfOXTjBX3g5fMifLm8+t5OuS2F/2G/YFRsTxn1E+fLltY4kHhCz2czAgQNp2LAh7498n9jNp7Cei8G/dRX0/tJS/2/scSlY1h3Efj2ZYsWKMW7cOCpWrKh1LI+SrZbbY8eO3fXr2rVr1KpV60FlFW6iQoUKLF26lIYNG+JIjCbt2O84UnP/pSdb7CXST2xFtabRt29fpk2bJoWtoGDBgsz8dCbe3t4cDtlGgilvd9m5HZtiZV/ob6TrUnnnnXdo2rSp1pHEQ1CnTh1WfLuCZs2aYY+8QeKiHVgvSSPY3WSciiZp8U7s15N54YUXWLZsmRS29yFbLbeLFy++ZdvNy9J5eSWMvCgwMJBp06Yxf/58Zs2aRfrJLZhL1cEQkPsuM6qqiu3qSaxXjhMQEMDEiROpW7eu1rGEGylXrhxTpkyhX79+HAj9ndrXn8LXHqB1LLfgxMGBkD+wGBN4+eWX6dy5s9aRxEMUGBjIpEmTWLVqFZMnTybp2z34NC6Nd+0SUjf8g+p0krr1DGm7zuPl7cXoSeNo3ry51rE81n33uV2xYgWtW7emcuXKVK5cmdatW7NixYqczCbcnE6no3fv3kycOBG9AumntmOLvaR1rBylOp1kXNiP9cpxihQpwsKFC6WwFbdVr149PvjgA6xKBvtCN5GuS9U6kuZUnBzO9yc3zNE0b96ct99+W+tIQgOKovDCCy8wf/58wkJDSf3jFMnfH8SZIXPiAjhTMkj6di9pu85TvHhxlixeIoXtf3Rfxe306dMZP348TZo0Yfr06UyfPp0mTZowfvx4mec2D2rRogVfzJqFv58vGef3Yrt+XutIOUJ1Okk/uxt77EUqVqzIwoULKV68uNaxhBtr3bo1b731Fml6C/tCN2FT8u60eSoqJwL3EO1zkZo1azJmzBh0OhnDnJdVrlyZZUuXUatWLaynoklcshNHYprWsTRlj0kmcdFObJfiadq0KYsXLyYiQlY+/K/u65Vm2bJljB07loEDB9K0aVOaNm3KwIEDGTt2LEuXLs3pjMID1KhRg6+++org4GAyLh7AFuPZBa7qdJJ+7i8cCVepU6cOc+bMITg4WOtYwgN0796dTp06YTEmsD/kdxzkzdapc/6HifQ7SdmyZZk6dSpms1nrSMINBAcH8/nnn9OlSxfXbAr267l/zMbt2C7Fk7T0L5yWDPr165e5+qHMkZ4j7qu4tdvtVKpU6ZbtFStWxOGQSZvzqoiICObMmUO+fPnIuHAAW+xFrSPdF1VVyTi/F8eNKGrXrs0nn3wi03yJe6YoCu+88w4tW7YkwXydQ8Fb89wiD5E+pzgbcJDwIuF8+umn8oYtsjAYDLz99tsMHToUNdVK0rK/8tyqZhknr5G0Yi86Z+ZMVN27d5c+yDnovorbZ599lmXLlt2y/dtvv5VlFPO4iIgIZs+eTWBgEBkX9mNPuq51pGyzXj6KPf4yNWrUYNq0aVLYimzT6XSMGjWKevXqEeN9meNBO1HJG/N8XvO6yPF8uwgODuazzz8jJCRE60jCTb344otMnjwZvaqQvHIfGSeuaR3poUjbd5HktQfx9vLms08/pUWLFlpHynWyNVvCP61cuZL/a+/eo6Mq772Bf/fcMpM7uUlCIIGEBAMJkBCSIBBAEFCuRQQ8QvHwAgVXpRRPxYMiVlsPIAVprRfswiOtnrKUesRX21OO2i5FXi9AuBoQ0BAChISEZJK57r3fP5IZMhAgwMzsPXu+n7VYzJ7Zs/cvhMl888xvP8/nn3+OgQMHAgAOHDiAmpoaTJs2zWc+3CeeeOL2q6SQkp2djU2bNmLRokVwnPgSun7l0FlilC6rS1wXvofr3HFk9u6NjRs3MtjSLTMajXjhhRewePFiHDp0CCbJjL5NhUqXFVD1prM4lPgZoiKj8NJLL6Fnz55Kl0Qqd/fdd+P38b/Hz5YvR/POCsiSBHNemtJlBYztq+/R8kklEpOS8NLvfoecnBylS9KkWxq5PXbsGPLy8tCtWzdUVVWhqqoK8fHxyMvLw7Fjx7zz3h49etTf9VKIGDhwINasWQPZ7YL9uz2QRfX3HYrWi3D8UIHYuDhsfvFFxMSERiAn9bJYLNi8eTN6Z/bGqZjD+CHqiNIlBUyT8SIqkv4BvVGPTS9uQm5urtIlUYgoKirCH15/HTHRMbB+eAiO70LvE7+usB+oRssnlUhJScEbW7cy2AbQLY3cdjbfLdGVJk6ciOPHj+ONN96Ao+oAzL3VO2oliy44Tn4NQQBeWL8e6enpSpdEGhEfH4+Xfv8S5s+fj0p8A5NkQaqtt9Jl+VWrvhn7kj6GpBOx7vl1KCoqUrokCjE5OTn43e9+h8U/WQzr+xUQ7i+CqZd2LuJ1fHsO1r8dQXy3eLzyyivo0aOH0iVpGudloYBasmQJ8vL6w133A9wXzyhdzjU5qg5AcrTgXx9+mKvskd9179697cKqqGgcTtiNiybt9BY6dXbsTfoYDp0NK59YiTFjxihdEoWo/Px8vLjpRRh0ejTv2AfX2UtKl+QXzlN1sP7fg4iKisLLv38ZmZmZSpekeQy3FFBGoxG//vWvYDab4aiqgOxW37yf7ku1cNdVoX///li0aJHS5ZBGZWdnY+OmjdAb9Nif9CmaDQ1Kl3TbRLixL+FTtBqasGDBAtx///1Kl0Qhrri4GOvWroPgltD83n5Irep7z7gZYkMrrO9XwGQw4rebN7NdJ0gYbingevXqhSVLlkB2OeA8o64+bFkS4ayqgE6nw+rVq2E0GpUuiTSsqKgIz/3qOYg6N/YlfQJHCK9iJkPGoYTPcSniAiZPnoylS5cqXRJpRHl5OR599FFIzXY0f3AAshSaM43IbtG7Etvq1asxaNAgpUsKG6oOt6IoYtOmTRgzZgwKCgowduxYvPTSS5Dly//RV65cidzcXJ8/CxYsULBq6szs2bPRu3cfuGpPQWxVz0dNrvMnINmtmDNnDvr27at0ORQGxo0bh2XLlsGub8G+xE8hCuq/2LIz38Xux3lLFYqLi/Hkk09yjk7yq7lz52LUqFFwfV8P2xcnlC7nlrT877dw1zbh/vvvx8SJE5UuJ6yoOtxu2bIFb7/9NlavXo0PP/wQjz32GF5//fWrLmgbMWIEPvvsM++f3/zmNwpVTNdiNBrxi1/8GwBZNaO3suiC69xxxMXHY/HixUqXQ2Fk7ty5mD59OppM9TjY7fOQmwP3TOQJnIo5hIyMDKxfv56feJDfCYKAZ555BmlpaWjdfSLkFnlwHD0Le0U1+vXrhxUrVihdTthRdbjdt28f7r77bowaNQrp6emYMGEChg8fjgMHDvjsZzKZkJyc7P0TFxenUMV0PUOHDkVhYSHExrMQW5TvN3SdPwHZ7cT8H/+YKyhRUAmCgJUrV6K4uBi1liqciKlQuqQuazDV4mi3PYiLjcPmzZsRGxurdEmkUTExMVi3bh10gg4tu45CFkNjpT/J4UbLx5WIMJuxbt06Lj2tAFWH28GDB2PPnj04deoUAODbb7/FN998g5EjR/rs9+WXX6KsrAzjx4/H008/jYYG5YMTXU0QBCxZsgQA4Dx7TNFaZNEN1/kTiI/vhgceeEDRWig8GY1GrF+/Huk90nEy9iBqzaeVLumGHLpWHEj8JwS9gBc2vMBFGijg8vLyMHPmTIj1LbDvrVK6nC6x7T4BqcWBhf/n/3BaSYXc8gplwbBo0SJYrVZMnDgRer0eoihi+fLlmDJlinefESNGYNy4cUhPT8fp06fxm9/8BgsXLsSf//xn6PX6mzqfKIr+/hLoCoMGDUJeXh6OHDkKydEKXUSkInW4L56G7HbigQfmw2Qy8XtPioiKisL6F9bj4fkP41DC5yipnYgotzo/eZIgoiLhn3DobFjxsxUYNGgQXzcUFIsWLcJf//ZXNH1+AhF3pkIXrd6RUHe9FbZvfkB6ejrmzJnD14ifdfXfU9Xh9qOPPsLOnTuxYcMGZGdn4+jRo3j++eeRkpKC6dOnAwDuu+8+7/6eC8rGjh3rHc29GQcPHvRr/dS50tJSHDlyBK7aU4jo2T/o55dlGa7zJ6HX65GTk4P9+/cHvQaijub9eB5ee+017E/4B0ouTIRBVl8Pa2XcN2iMuIDS0lLk5ubydUNBNWXyFGzbtg2tu08g+p48pcu5ptZ/HAckGdOnT8eRI9pdkVDtVB1u161bh0WLFnkDbG5uLmpqavDqq696w+2VevbsiW7duuGHH3646XCbn59/06O9dPPy8vLwzrvvorm+Cqb0vKBfZS21NkKyNeGee+5BeXl5UM9N1JlBgwbBarXirbfeQmXc1+jfeHM/uwKt1lyF09GVyM7Oxvr162E2m5UuicJMfn4+Pv7kY9QcqkHkyL7QmdX3C6DY0ArniVoUFBRg3rx5SpejSaIodmkgUtXh1m63XxV89Hq9z1RgVzp37hwaGxuRnJx80+fT6/UMt0FgsVgwYfx4bN++HWJTLQxxdwT1/O66tr6tyZMn8/tNqrFs2TLs378fR44cQaIjFd1tmUqXBACw61pwOGEPzBFmrF27FlFRUUqXRGFIr9dj9qzZ2LBhA+wHqhE5VH1LWNv2VQEy8OCDD/K9RWGqvqBs9OjReOWVV/Dpp5+iuroaf//737F161aMHTsWANDS0oK1a9di//79qK6uxhdffIGlS5ciIyMDI0aMULh6uh7PaLy7PrgX0ciyDHfDGSQkJKCkpCSo5ya6nrbV/H4Ni8WCI93+H2x6q9IlQYaEgwmfwyU48G+/+Df07q2+QEHhY8qUKTBbLLDvPa26hR1kpxuOg2eQnJKM0aNHK11O2FN1uH3yyScxfvx4PPPMM7j33nuxdu1azJo1C8uWLQPQ9pvcsWPHsGTJEkyYMAGrVq1C//798ac//Qkmk0nh6ul6BgwYgNTUVIiXzkOWgje9i2Sth+xyYOzYsTAYVP3BBYWhXr164YknnoBbcOJw/BeKz39bFVWJhojzGDt2LKZNm6ZoLUQxMTGYdN99kJpscJ2+qHQ5Phzf1UJ2uHH/jPs577MKqPrdPTo6GqtWrcKqVas6fdxsNuMPf/hDkKsifxAEAaNGjcLbb78N0VoHQ2xKUM7rbqgBAP5mTap133334ZNPPsEnn3yCM5HfIb1VmZXzWvXN+C5uP+Lju+Hf//3fuQIZqcLo0aPxzjvvwHWqDqaMRKXL8XKdrAPA9xa1UPXILWmb54eA2HguaOcUG88hOjoahYWFQTsn0c3wLPAQHRWNY/F7Yde1Br0GGTKOxO+BKLixcuXjiI+PD3oNRJ0pLCyE2WyGsz1MqoEsy3B9X4+UlBRkZWUpXQ6B4ZYUNHDgQERGRkK8dD4o55PsVkiOFpSWlvJjI1K15ORk/HzFz+EWnPg2/qugn/+s5SQums9h1KhRGDduXNDPT3QtERERKC4uhlhnhdhsV7ocAID7XBOkVifuuusufsKhEgy3pBij0YiSkhJv6Aw0sakWADBs2LCAn4vodk2dOhWFhYWotVThoik4vwACgFtw4Xj8PpgjzFi5ciXfrEl1hgwZAgBwn7ukcCVtPHUUFRUpXAl5MNySojwzFohNFwJ+Lnf7OYYOHRrwcxHdLkEQsGLFCgiCgMr4ryAjOBdenoo+BIfOhn9d8K9ISQlOLzzRzejTpw8AQKwL/KBIV4h1bTObZGdnK1wJeTDckqKKi4sBAGJzYPunZFmG1FyH9PR0pKWlBfRcRP5y5513YsqUKWg2NuBM5ImAn8+mt+KH2KPofkd3PPTQQwE/H9Gt8IbbeuWnywMAsb4FOp0OvXr1UroUasdwS4rKzMxEUlISxKYL112c43ZJtkuQ3U5vmCYKFY888gjMEWacjD0ICYFdp/5kTNs5fvroT7kKGalW9+7dYbZY4K5XycjtxRakpaXxNaMiDLekKEEQUFRUBNllhxzAvluxqW1kmD1RFGqSkpIw84GZsOtbUBPA0Vub3oqaqJPo3bs37rnnnoCdh+h2CYKA5KQkyK1OpUtp+1SwxckWHpVhuCXFeQJnIFsTPMdmuKVQNG/ePEREROBU7KGAjd6eijkEGRIWLlzIpUNJ9eLi4iDbXUqXAdnpBmQZcXFxSpdCHTDckuI8c84GKtzKsgzJWo8ePXrgjjvuCMg5iAIpMTERM2fOhE3fgnOWH/x+fIfOhpqoE8jIyODUXxQSYmNjIbtEyO7AturciGxzeesh9WC4JcX17t0b3bp1g9hcH5DjS7ZmyG4nR20ppM2ZMwc6QYeq6G/9fuwzkd9BgoR/+Zd/4agthYSYmBgAgOxwK1qH5/zR0dGK1kG+GG5Jcd6+W2drQOa7ZUsCaUFqairKR5WjyVSPS0b/fcohQUJ19DFER0Xj3nvv9dtxiQLJYDAAAGQpcBcid4Xn/J56SB0YbkkVAtl3KzZf8DkHUaiaNWsWAOB0VKXfjnnBXA27vhVTpk5BZGSk345LFEg6XXt8CeAsO13Sfn5vPaQK/G6QKnhWnPHMauAvbfPb1iM1NY3z21LIKy4uRnp6OmojT0MU/PNx7NnIkwCAH/3oR345HlEweNtnFB659Zyf7TzqwnBLqtCnTx/Ed+sGsbnOr/PdSrYmyG4HhgzhqC2FPkEQMHHiRLgFFy6Yq2/7eC7BgTrLGeTk5HonxicKBWpbFlpt9YQ7hltSBUEQMLS4GLKz1a/z3Ypccpc0ZsKECQCAs5ZTt32s85YqSJAwceKE2z4WUTCJYvssCUqHyvbze+shVWC4JdXwBFBPIPUHhlvSmt69eyMnJwf15rO33ZpQazkNAFy0gUKOJEltN3QKh1tv66/C7RHkg+GWVKOkpAQA4G6q9cvxZEmC1FyHPn2ykJyc7JdjEqnBiBEjIAkiLprO3fIxRLjREHEeffv2RWpqqh+rIwo8z0ip0u0AAkduVYnhllSjR48e6NmrF6SmC5Bl6baPJ1kvQpbcKCsr9UN1ROoxfPhwAMAF85lbPsbFiPMQBbf3WEQhSelW1/Zwy5FbdWG4JVUZVlYGWXRBsl687WO5L50HAJSWMtyStgwYMABxcXGoN9fc8jE8z73rrrv8VRYRkSow3JKqeEaRPMH0doiXziEiIoLz25Lm6PV6FBYWwmawwq5rvaVjNJhqYTKZkJ+f7+fqiIiUxXBLqlJUVASTyQTxNsOt5LRBsjWhuLgYZrPZT9URqcfgwYMBAI0RN9+j7hZcsJoakJ+fD6PR6O/SiIJH6W4Apc9PnWK4JVUxm80YOnQopNZLkJy2Wz6O2Nh2oQ0/ciWtGjRoEIC2Edibdcl0ATJk7zGIQo1n0EJ2K3shl+f8HERRF4ZbUp0RI0YAuBxQb4X70jmfYxFpTW5uLgwGA5qNN9+f3tT+nAEDBvi7LKKguBxub//i49vhCbcWi0XROsgXwy2pjrfvtvHsLT1flkRITReQnZ3NJXdJs4xGI7KystBsaoCMm3uD94Tbfv36BaI0ooDzhluXwiO3rrbXHkdu1YXhllQnNTUVffv2hdRcB1m8+UnqxaZayJKIkSNHBqA6IvXIzc2FKLjRami+qec1my4iPi4eKSkpAaqMKLCio6MBALLdpWgdnvNHRUUpWgf5YrglVSovL4csibe0Wpm7vZ2hvLzc32URqUrfvn0BAFZDY5efI0JEq6EZfXP6Kj4BPtGt8izMI1kditbhOT9/UVQXhltSJc+o6822JsiyDLHxHLp164b+/fsHojQi1cjMzAQAtBiauvyc1vZ9Pc8lCkXqCbd2AOAqmCrDcEuqlJeXh8TERIiXzt3Uyi9SayNklx3l5eXQ6fjfm7TNG26NXQ+3LQy3pAHecNtsV7QOT7hOSkpStA7yxXd/UiWdToeRI0dCdjkgtTR0+XmelgT221I46N69O4xGk3c0tis8+2ZkZASqLKKAS0tLg06ng3ixRdE6xPoWxMXFISYmRtE6yBfDLamWtzXhJhZ0EBvPwWg0oqSkJFBlEamGXq9HWloqbAZrl5/j2Tc9PT1QZREFnNlsRnrPdIgXrDf16Z4/yS4RYkMr+vZl/7raMNySag0dOhRGoxFiF/tuJacNUmsjiouLOecghY20tDQ4dXaI6NrMIjZ9W7jt3r17IMsiCri+2X0h2ZyQW5yKnF+sbwFkGdnZ2Yqcn66N4ZZUy2KxoLi4uH21shv3VXmW7PXMk0sUDlJTUwEANkPXPp61G1qQlJQEk8kUyLKIAs4zW4j7fNfbcvzJc15PHaQeDLekap7lc8UutCa4GW4pDHlGYB36G4dbGTLs+laO2pImDB48GADgOn3zq/T5g+e8njpIPRhuSdW6Gm5lWYLUdAE9e/ViLyGFFc/8mna97Yb7unROSILIaYtIEwoKCmA0GeGqCn64lWUZrqqLSExM5MWZKsRwS6rWs2dPpKamQWy+cN2LBiRrA2TRhWFlZUGsjkh5nnDr0LXecF/PPpxwnrQgIiICAwsGwn2+GVKQVyqTGlohWR0oLi7mxWQqxHBLqiYIAkpLSyC7nZBaG6+5n9hUCwAoLS0NUmVE6uANt/ouhNv20V2O3JJWlJSUALIM16m6oJ7XeaJt9cyhQ4cG9bzUNQy3pHqeab2utxSvu/kCdDodioqKglUWkSokJiYCABz6G1906Qm3nHCetGL06NEAAMexrk8Z6Q+OY+eh0+m4zLtKMdyS6g0ZMgQAIDa3/2auN/j8LUsiJGsD7rzzTkRHRytRIpFiYmNjYTAY4NTduOfWs48nEBOFuj59+iAzMxOuk3WQXWJQzilZHXCfaURRURG6desWlHPSzWG4JdVLSEhAnz59IFnrIcsSTGn9oI/rDlNaPwCAaL0IyJI3BBOFE0EQkJCQ4B2VvR7PPgy3pCV33303ZJcIZ5BaExzHznnPS+rEcEshobCwELLohtTaBENsMiw5ZTDEtq8tbq0HwOlYKHwlJibCpXfccD+nzu7dn0grxo8fDwBwHDoTlPM5DtVAp9cx3KoYwy2FhIKCAgCA2B5kOxKtF332IQo3CQkJcAuuG65S5gm38fHxQaiKKDiys7ORl5cH58k6SC03/iXvdrgvNMN9rgkjho/gL4kqxnBLIWHgwIEAAMnqO5+hLMuQWi4iIyODb9gUthISEgAAzhtcVObUOxATEwOj0RiMsoiCZurUqYAkw364a8u13yr7wTOXz0eqxXBLISE9PR1xcXEQr5gOTHa0QHa7kJ+fr0xhRCrg+cXOqbv+qJVLZ/cGYSItmTBhAkwmExwHqq87J/rtkN0SnEfOolu3bt4FhkidGG4pJAiCgLy8PMh2K2S303u/2NIAAMjLy1OqNCLFeUduddceuZUhw8lwSxoVExOD8ePHQ7zYErAVyxyV5yC1OjFt2jR++qFyDLcUMvr1a5sdQWpt8t7nWdjhzjvvVKIkIlXwTEd0vbYEt+CELMicuog0a+bMmQAA+/7TATm+ff9pCIKAGTNmBOT45D8MtxQycnJyAACi7ZL3Pk/Qzc7OVqQmIjXwBFZXh5Fbg2z0+dsTfBluSav69++PO++8E87jtRCbb7yoyc1wn2+C+0wjRowYgbS0NL8em/yP4ZZChifcSq0dwq2tCT169EBUVJRSZREpzjty26HnNqupAEm2HshqKvB5jOGWtEoQBDzwwANtF5b5efTWtrcKANqOT6rHcEsho2fPnjAYDJBszQAA2e2E7LIjKytL4cqIlHU53F4erUpwdkfhxTFIcHb3eYzhlrRswoQJiIuPh6OiGrLbPyuWSa1OOI+cRUZGBkpLS/1yTAoshlsKGQaDAb169YLssLZNAdYecnv37q1wZUTK6izcXsnFcEthICIiAvfPmAGp1QnH0XN+Oaa9ohqyKGH27NnQ6RibQgG/SxRSMjMz22ZLcDsh2a0AgIyMDIWrIlKWxWKByWS67iplbEugcDFz5kzo9XrYvvnhtqcFk0UJ9v2nERUVhUmTJvmpQgo0hlsKKT179gQASI4WSI4Wn/uIwpUgCIiPj7/uPLeeC8q42AlpXUpKCu6++26Itc1wn2m8rWM5j9dCarZj6tSpvLYjhDDcUkhJT08H0BZuZYZbIq+EhAS4rjMVmIsjtxRGZs+eDeDyhWC3yra36vKFahQyGG4ppHimYJEdrZCcrTAYDEhKSlK4KiLlxcfHwy24IKHzi2g8o7ocuaVwMHDgQOTm5sJ57Pw1pwUTTHqfv6/kPt8Ed3UD7rrrLvTq1StgtZL/MdxSSOneve3Kb8lpg+y04Y477mCDPxE6nw6sI6fOjsjISJhMpmCWRaQIQRAwa9astmnBKqo73SdyWBaMWcmIHNb5jDu2fW2jvrNmzQpYnRQYTAUUUu644w4AgOxsheyye8MuUbjzjMi6rjFjgktvZ0sChZXx48cjOiYajgPVkCXpqsdNGYmIm1EIU0biVY9JDjecR8+hR48eKCsrC0a55EcMtxRSIiMjYbFY2hZykGW2JBC1u7wEb+cjty6dg+GWworFYsHkSZMhWR1wnqi7qec6jpyF7BLxox/9iJ8OhiB+xyjkJCcnQ3a1jU4x3BK1uTxye3W4FQU3REFkvy2FnRkzZgAA7BVdX7FMlmXYK05Dr9djypQpgSqNAojhlkJOQkJCp7eJwtn1FnLgHLcUrvr06YOBAwfCdar+mheWXUmsbYZY24xRo0YhMfHqlgVSP4ZbCjkd36A5EkXUxvNa6OyCMk8fblxcXDBLIlKFyZMnA7IMx9GzXdrffrjm8vMoJDHcUsjp+AbNN2uiNtdrS+A0YBTOxo0bB6PJCEd7aL0eWZLgPHoO8d268UKyEMZwSyEnJibGezs2NlbBSojUw/OLXmfh1sVwS2EsJiYG5SPLIV6wwl3bdN19Xd/XQ2pxYOKECTAajUGqkPyN4ZZCTsdw2/E2UTjrSrjlJx0UriZMmAAAcFSev+5+nsfHjx8f8JoocBhuKeRER0d7b3Otb6I2RqMRkZGRDLdEnSgrK0NERAScx2uvuY8syXCduICk5GQMGDAgiNWRvzHcUsixWCze2wy3RJfFxcXBpXNedb/nPoZbClcWiwXDhg2DWGeFeLGl033cZxogtToxetQozm0b4vjdo5DTMdB2DLpE4S4uLg4ufWfhliO3RKNHjwYAOL670OnjjvZR3TFjxgStJgoMhlsKOREREZ3eJgp3sbGxcAtOSPBdatQzcssedQpnntkPXD90vlqZ64d6mM1mDB48OJhlUQAw3FLIMZvN3tuCIChYCZG6eMKrW+fyud8tOGE0GH1eO0ThJjExEX379oX7dCNkt+8vgJLVAfGCFYWFhTCZTApVSP7CcEshJyMjAxZLJAYOGqR0KUSq4pkazy34tia4dE7ExMbwl0EKe6WlpZDdIlw1jT73O6vqvY9T6GO4pZCTkpKCTz/9BH94/XWlSyFSFc9MIldeVObWOdmSQASguLgYAOA+fdHnfvfpBp/HKbQZlC6A6FZwcm2iq3nbEoQrw62L4ZYIQH5+PgDAdfaSz/2us5dgtliQnZ2tRFnkZ6oeuRVFEZs2bcKYMWNQUFCAsWPH4qWXXoIsy959ZFnGiy++iOHDh6OgoADz58/H999/r1zRREQK8Yzcduy5lSBBFNw+80MThau4uDj06tUL4tlL3iwhO90QL1jRPy8Per1e4QrJH1Qdbrds2YK3334bq1evxocffojHHnsMr7/+OrZt2+azz7Zt27BmzRps374dFosFCxYsgMNx9UTmRERa1lm4FQWXz2NE4S4/Px+SzQWpoRUA4D7XBMgyF27QEFWH23379uHuu+/GqFGjkJ6ejgkTJmD48OE4cOAAgLZR2zfffBNLlizB2LFj0a9fP6xbtw61tbXYtWuXwtUTEQVXZGQkgMuBFrgcdD2PEYW7O++8EwDgvtDs83e/fv0Uq4n8S9XhdvDgwdizZw9OnToFAPj222/xzTffYOTIkQCA6upqXLhwAcOGDfM+JyYmBgMHDsS+ffsUqZmISCmeBU7cHcNt+22u5kfUJisrCwAg1ll9/ma/rXao+oKyRYsWwWq1YuLEidDr9RBFEcuXL8eUKVMAABcutK0ykpiY6PO8xMRE1NV1Pknz9YiiePtFExEpxDOPrdixLaH9tsVi4c84IgCZmZkAAHd7qHXXWaHX69GjRw++RlSuq98fVYfbjz76CDt37sSGDRuQnZ2No0eP4vnnn0dKSgqmT5/u9/MdPHjQ78ckIgqWmpoaAIBbcHvv89xubGzE/v37lSiLSFVkWUZkZCQc9S0AALG+Bd1TUnD48GGFKyN/UXW4XbduHRYtWoT77rsPAJCbm4uamhq8+uqrmD59OpKTkwEA9fX1SElJ8T6vvr7+lnpn8vPzeaUkEYWs7t27AwDEDuHWczsrKwuDuPAJEYC2xYC+/e4YJIcLst2F7Oxsvj5CgCiKXRqIVHW4tdvtV62oo9frvdN3pKenIzk5GV988YW3QdxqtaKiogJz5sy56fPp9XqGWyIKWZ6+WrGT2RKioqL4842oXY8ePXD06FG4zzZ5t/n60A5Vh9vRo0fjlVdeQVpamrctYevWrZgxYwYAQBAEzJs3Dy+//DIyMjKQnp6OF198ESkpKRg7dqzC1RMRBZdnRoSObQmiru22xWJRpCYiNUpNTQUAuNpXKktLS1OyHPIzVYfbJ598Ei+++CKeeeYZb+vBrFmz8Mgjj3j3WbhwIWw2G1avXo2mpiYUFRXh9ddfR0REhIKVExEFn9FohE6ng9RJWwKnAiO6zBNu3dUNPtukDaoOt9HR0Vi1ahVWrVp1zX0EQcCyZcuwbNmyIFZGRKQ+giDAYrZAtF8dbj0zKRDR5VmWPHPcXjnrEoU2Vc9zS0REN8dsMXd6QRnbEogu84RZuf0XQYZbbWG4JSLSEIvF4u2zBQBRaJsXkiO3RJd1Nj8+aQfDLRGRhlgsFm+gBThyS9SZ+Ph4722j0ciedI1huCUi0pC2cMueW6Lr6bgcdXRMtIKVUCAw3BIRaYjFYoEkiJAhAbg8FRjDLdFlBoMBlsi2TzNiY2IVrob8jeGWiEhDPCHW05ogCm4IgsBwS3SF6OiY9r85cqs1DLdERBri6a31tCOIghtms/mq1R6Jwl1k+2uF/ejaw3BLRKQhnYVbvnkTXc3zCx8/1dAehlsiIg253JbAcEvUFVzRVHsYbomINMQzpZHnQjJR5+Y0R0Sd8IzcmkwmhSshf2O4JSLSEE+QdQsuAG0jtwy3RNem0zEKaQ2/o0REGtKx51aCBEkQ2ZZARGGF4ZaISEO8bQmCC2L76C1HbomuJsuyz9+kHQy3REQa4ll5ya1zw93ed9txNSYi8sVwqz0Mt0REGtLZyC3bEoiu5rmgjHNAaw/DLRGRhngvKNO5vBeVceSWiMIJwy0RkYZ4lhJ1Cy64dS6f+4joamxL0B6GWyIiDbncc8sLyoi6gm0J2sNwS0SkIZ5wK3LklojCFMMtEZGGdFzEwdNzy3BLROGE4ZaISEP0ej0iIyPh1jk5cktEYYnhlohIY6KjojlbAtEN8EIy7WK4JSLSmOiY9nCrc7Ztc+SW6JoYcrWH4ZaISGOioqJ8em45ckt0bZIkKV0C+RnDLRGRxkRFRUESRLh0Du82EXXO6XQqXQL5GcMtEZHGeNoQ7PpWGI0mmEwmhSsiUi+Xy6V0CeRnDLdERBrjCbcOfSuiOWpLdF0cudUehlsiIo3xzHUr6tyIjOLqZESd8YzYMtxqD8MtEZHGdOyx5UwJRJ3zhFqGW+1huCUi0hjPyO2Vt4noMrvDDgBwOBwKV0L+xnBLRKQxHUduOVMCUecc9rZQa7fbFa6E/I3hlohIYzqO1losFgUrIVInSZK87QgMt9rDcEtEpDFsSyC6vo6tCHa2JWgOwy0RkcZ0HK3lyC3R1TqO1jo4cqs5DLdERBrDcEt0fR3Drdvt5kIOGsNwS0SkMWaz2Xub4ZboajabzWebfbfawnBLRKQxHcNtx9tE1ObKMMtwqy0Mt0REGhMREdHpbSJqc+XI7ZXbFNoYbomINIbhluj6OHKrbQy3REQaw3BLdH2eMCuYDT7bpA0Mt0REGmMymTq9TURtLodbk882aQPDLRGRxuj1eu9thluiq3nCrM5i9NkmbWC4JSLSMKPRqHQJRKrjHblluNUkhlsiIg1juCW6mtPpBADozEafbdIGhlsiIg2KiYkBAHTr1k3hSojUx+FwAACECIPPNmmDQekCiIjI/95++23U19cjMzNT6VKIVMczUitw5FaTGG6JiDQoLS0NaWlpSpdBpErecGtqu/iSI7fawrYEIiIiCisulwsAIBjbwq3b7VayHPIzhlsiIiIKK54wKxgNPtukDQy3REREFFYuh9u2kVvPSC5pA8MtERERhRVJktpu6NtikCzLClZD/sZwS0RERGHFE24FneCzTdrAcEtERERhxRtmGW41ieGWiIiIwoq3DYHhVpMYbomIiCg8tWdcQRCUrYP8iuGWiIiIwoo3zLaP4DLcagvDLREREYUVhlttY7glIiKisKLTtU8BJjLcahHDLREREYUVvb5t8Qa4RQCAwWBQsBryN4ZbIiIiCiueMCu7JZ9t0gaGWyIiIgor3nDr4sitFjHcEhERUVgxGo0AANnl9tkmbWC4JSIiorASEREBAJAdbp9t0gaGWyIiIgorJpMJwOVw69kmbWC4JSIiorDiDbd2hlstYrglIiKisOJpQ5DsLp9t0gaGWyIiIgorZrMZACDbnD7bpA0Mt0RERBRWOHKrbQy3REREFFa8I7etDLdaxHBLREREYcUbbp1tF5RZLBYlyyE/U/2SHGPGjMGZM2euuv/BBx/E008/jblz5+LLL7/0eWzWrFn45S9/GawSiYiIKIRcGWYZbrVF9eH2nXfegSiK3u3jx4/j4YcfxoQJE7z3PfDAA3j00Ue92/xPSkRERNdy5QVkvKBMW1QfbhMSEny2X3vtNfTq1QtDhw713mc2m5GcnBzs0oiIiCgEceRW20Kq59bpdOL999/HjBkzIAiC9/6dO3eipKQEkyZNwoYNG2Cz2RSskoiIiNSMI7fapvqR24527dqF5uZmTJ8+3XvfpEmTkJaWhpSUFFRWVuKFF17AqVOn8Lvf/e6mj9+x/YGIiIi0qeOKZAajEYIgMAOEgK5+j0Iq3L777rsYOXIk7rjjDu99s2bN8t7Ozc1FcnIy5s+fj6qqKvTq1eumjn/w4EG/1UpERETq5HK5vLeNBgP279+vXDHkdyETbs+cOYPdu3fjt7/97XX3GzhwIADghx9+uOlwm5+fD71ef8s1EhERkfrJsgxBECDLMqKiozBo0CClS6IuEEWxSwORIRNud+zYgcTERIwaNeq6+x09ehQAbukCM71ez3BLREQUBiLMZthtNljMFr73a0xIhFtJkrBjxw5MmzYNBsPlkquqqrBz506Ul5cjPj4elZWVeP7551FcXIx+/fopWDERERGpmTkiAnabjReTaVBIhNvdu3ejpqYGM2bM8LnfaDTiiy++wJtvvonW1lakpqbinnvuwdKlSxWqlIiIiEKBZ8ldLr2rPSERbocPH47Kysqr7k9NTcUf//hHBSoiIiKiUOYZseXIrfaE1Dy3RERERP6g07VFII7cag/DLREREYWtjnPekjYw3BIREVHYkWUZAMOtFjHcEhERUdjiNGDaw3BLREREYcszgkvawXBLREREYUcQBKVLoABhuCUiIqKwwxFb7WK4JSIiIiLNYLglIiKisMX2BO1huCUiIqKwM2XKFABAaWmpwpWQv4XE8rtERERE/jR//nzMnDkTUVFRSpdCfsaRWyIiIgpLDLbaxHBLRERERJrBcEtEREREmsFwS0RERESawXBLRERERJrBcEtEREREmsFwS0RERESawXBLRERERJrBcEtEREREmsFwS0RERESawXBLRERERJrBcEtEREREmsFwS0RERESawXBLRERERJrBcEtEREREmsFwS0RERESawXBLRERERJrBcEtEREREmmFQugA1kGUZACCKosKVEBEREVFnPDnNk9uuheEWgCRJAICDBw8qXAkRERERXY8nt12LIN8o/oYBSZLgdruh0+kgCILS5RARERHRFWRZhiRJMBgM0Omu3VnLcEtEREREmsELyoiIiIhIMxhuiYiIiEgzGG6JiIiISDMYbomIiIhIMxhuiYiIiEgzGG6JiIiISDMYbomIiIhIMxhuKeRUV1cjNzcXR48eveY+O3bswJAhQ4JYFRERaUlubi527dqldBl0CxhuiYiIiEgzGG6JiIiISDMYbkmV/vnPf2LOnDkYMmQISkpKsHjxYlRVVfnsc/r0acydOxcDBw7ElClTsG/fPoWqJQoeSZKwZcsWjBs3DgMGDMCoUaPw8ssve9t1/ud//uearwtPu86uXbtwzz33ID8/HwsWLMDZs2cV/IqIAuevf/0rJk+ejIKCApSUlGD+/PlobW3FgQMH8PDDD6OkpARFRUV46KGHcPjwYaXLJT9huCVVstlsePjhh/Huu+/ijTfegCAIeOSRRyBJknefjRs3YsGCBXjvvfeQmZmJFStWwO12K1g1UeBt2LABW7ZswdKlS/Hhhx/ihRdeQFJSkvfxG70u7HY7Xn75ZaxduxZvv/02mpqasHz5ciW+FKKAqq2txYoVKzBjxgx8+OGHePPNNzFu3DjIsoyWlhZMmzYNb731FrZv346MjAwsWrQIVqtV6bLJH2SiEFBfXy/n5OTIlZWV8unTp+WcnBx5+/bt3sePHz8u5+TkyN99950sy7L87rvvykVFRUqVSxQQzc3N8oABA3z+73t09XWRk5Mj79+/37vPd999J+fk5MgVFRWB/wKIgujQoUNyTk6OXF1dfcN9RVGUBw8eLH/88cfe+3JycuS///3vgSyRAsSgdLgm6sz333+PzZs3o6KiAg0NDZBlGQBw9uxZZGVlAWi7ktUjOTkZAHDx4kXv40Rac/LkSTidTpSWll5znxu9LgwGA/Lz8737ZGVlITY2FidOnEBBQUGAKicKvn79+qGsrAyTJ0/G8OHDMXz4cIwfPx5xcXGoq6vDpk2b8OWXX6K+vh6SJMFms6GmpkbpsskPGG5JlX7yk5+gR48eeO6555CSkgJJkjBp0iS4XC7vPkaj0XtbEAQA8GlbINKaiIiIG+7D1wVRG71ej61bt2Lv3r34/PPPsW3bNmzcuBHbt2/HmjVr0NjYiFWrViEtLQ0mkwmzZs3yeY+h0MWeW1KdhoYGnDp1CkuWLEFZWRmysrJw6dIlpcsiUlxmZibMZjP27Nlzy8dwu904dOiQd/vkyZNoamriJx6kSYIgoKioCI8++ijee+89GI1G7Nq1C3v37sXcuXNRXl6Ovn37wmQyoaGhQelyyU84ckuqExcXh/j4ePz5z39GcnIyampqsGHDBqXLIlJcREQEFi5ciPXr18NoNKKwsBAXL17E8ePHUVZW1qVjGI1GPPvss3jyySeh1+vx7LPPYtCgQWxJIM2pqKjAF198gbvuuguJiYmoqKjAxYsX0adPH2RmZuL9999Hfn4+rFYr1q1bB7PZrHTJ5CcMt6Q6Op0OGzduxHPPPYdJkyahd+/eePLJJzF37lylSyNS3NKlS6HX67F582bU1tYiOTkZs2fP7vLzzWYzFi5ciBUrVuD8+fMYMmQIfvWrXwWwYiJlREdH46uvvsJ//ud/wmq1Ii0tDStXrkR5eTmSk5Px1FNPYfr06UhNTcXy5cuxbt06pUsmPxFkz5U6RESkaTt27MCvf/1rfP3110qXQkQUMOy5JSIiIiLNYLglIiIiIs1gWwIRERERaQZHbomIiIhIMxhuiYiIiEgzGG6JiIiISDMYbomIiIhIMxhuiYg0Ijc3F7t27VK6DCIiRXGFMiIiP1u5ciX+8pe/AGhb7jY1NRVTp07FT37yExgM/LFLRBRI/ClLRBQAI0aMwPPPPw+n04l//OMf+OUvfwmj0YjFixff1HFEUYQgCNDp+EEbEVFX8KclEVEAmEwmJCcno0ePHnjwwQcxbNgwfPzxx9i6dSsmT56MQYMGoby8HGvWrEFLS4v3eTt27MCQIUPwv//7v7j33nuRn5+PmpoaHDhwAA8//DBKSkpQVFSEhx56CIcPH75uDZWVlZg3bx4KCgpQUlKCp556yudcbrcbzz33HIYMGYKSkhKsX78ejz/+OJYuXerdZ8yYMXjjjTd8jjt16lT89re/9W43NTVh1apVKC0tRWFhIebNm4dvv/32Nv8FiYhuDcMtEVEQREREwOVyQRAErFq1Ch988AH+4z/+A3v27MH69et99rXb7diyZQuee+45fPDBB0hMTERLSwumTZuGt956C9u3b0dGRgYWLVoEq9Xa6flaW1uxYMECxMXF4Z133sGmTZuwe/duPPvss959tmzZgp07d+L555/HW2+9BavVeks9u8uWLUN9fT22bNmCHTt2oH///vjxj3+MxsbGmz4WEdHtYrglIgogWZaxe/dufPbZZygpKcH8+fNRWlqK9PR0lJWV4Wc/+xk++ugjn+e4XC6sWbMGhYWF6NOnDywWC8rKyjB16lRkZWUhKysLzz77LGw2G7766qtOz/vBBx/A6XRi7dq1yMnJQVlZGVavXo3//u//Rl1dHQDgj3/8IxYtWoRx48YhKysLq1evRmxs7E19fV9//TUOHDiAzZs3Iz8/H5mZmXj88ccRGxuLv/3tb7f2j0ZEdBvYc0tEFACffvopBg8eDJfLBVmWMWnSJPz0pz/F7t278eqrr+LkyZOwWq0QRREOhwM2mw0WiwVA20Voubm5Pserq6vDpk2b8OWXX6K+vh6SJMFms6GmpqbT8584cQK5ubmIjIz03ldYWAhJknDq1ClERESgrq4OBQUF3sf1ej369+8PSZK6/HVWVlaitbUVJSUlPvfb7XZUVVV1+ThERP7CcEtEFAAlJSVYs2YNjEYjUlJSYDAYUF1djcWLF2POnDlYvnw54uLi8M0332DVqlVwuVzecGs2myEIgs/xHn/8cTQ2NmLVqlVIS0uDyWTCrFmz4HK5Avp1XFkH0Nar69HS0oLk5GRs27btqv1iYmICWhsRUWcYbomIAsBisSAjI8PnvsOHD0OWZaxcudI7+8GVLQnXsnfvXjz99NMoLy8HAJw9exYNDQ3X3D8rKwt/+ctf0Nra6h293bt3L3Q6HXr37o2YmBgkJSXh4MGDKC4uBtA2M8ORI0fQr18/73ESEhJQW1vr3bZaraiurvZu9+/fH3V1ddDr9UhPT+/S10JEFEjsuSUiCpKMjAy4XC5s27YNp0+fxnvvvYf/+q//6tJzMzMz8f777+PEiROoqKjAY489BrPZfM39J0+eDJPJhJUrV+LYsWPYs2cPnn32WUydOhVJSUkAgIceegivvvoqdu3ahZMnT+JXv/oVLl265DNaW1paivfffx9ff/01Kisr8fjjj/tMSzZs2DAMGjQIjzzyCD777DNUV1dj79692LhxIw4ePHiL/1JERLeO4ZaIKEj69euHJ554Alu2bMGkSZOwc+dO/PznP+/Scz3Bc/r06fjFL36BuXPnIjEx8Zr7WywW/OEPf0BjYyPuv/9+LFu2DGVlZXjqqae8+yxcuBCTJk3C448/jtmzZyMyMhLDhw9HRESEd5/FixejuLgYixcvxuLFizF27Fj06tXL+7ggCHjttddQXFyMJ554AhMmTMDPf/5znDlzxhuiiYiCSZBlWVa6CCIiUp4kSZg4cSImTpyIn/3sZ0qXQ0R0S9hzS0QUps6cOYPPP/8cxcXFcDqd+NOf/oQzZ85g8uTJSpdGRHTLGG6JiMKUTqfDjh07sHbtWsiyjJycHGzduhVZWVlKl0ZEdMvYlkBEREREmsELyoiIiIhIMxhuiYiIiEgzGG6JiIiISDMYbomIiIhIMxhuiYiIiEgzGG6JiIiISDMYbomIiIhIMxhuiYiIiEgzGG6JiIiISDP+P36eVpHH9LkEAAAAAElFTkSuQmCC",
      "text/plain": [
       "<Figure size 800x600 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Multiple Comparison of Means - Tukey HSD, FWER=0.05 \n",
      "====================================================\n",
      "group1 group2 meandiff p-adj   lower   upper  reject\n",
      "----------------------------------------------------\n",
      "   ahl    cnp   0.1378 0.9004 -0.6048  0.8804  False\n",
      "   ahl    sal  -0.9607 0.0059 -1.6903 -0.2311   True\n",
      "   cnp    sal  -1.0985 0.0025 -1.8685 -0.3285   True\n",
      "----------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "from scipy import stats\n",
    "from statsmodels.stats.multicomp import pairwise_tukeyhsd\n",
    "\n",
    "df = pLDDT_score\n",
    "\n",
    "\n",
    "def cohens_d(group1, group2):\n",
    "    \"\"\"Calculate Cohen's d for two groups\"\"\"\n",
    "    n1, n2 = len(group1), len(group2)\n",
    "    var1, var2 = np.var(group1, ddof=1), np.var(group2, ddof=1)\n",
    "    pooled_se = np.sqrt(((n1 - 1) * var1 + (n2 - 1) * var2) / (n1 + n2 - 2))\n",
    "    return (np.mean(group1) - np.mean(group2)) / pooled_se\n",
    "\n",
    "# Perform one-way ANOVA\n",
    "categories = df['paralogue'].unique()\n",
    "f_statistic, p_value = stats.f_oneway(*[df[df['paralogue'] == category]['pLDDT'] for category in categories])\n",
    "\n",
    "# Perform Tukey's HSD test for pairwise comparisons\n",
    "tukey_results = pairwise_tukeyhsd(df['pLDDT'], df['paralogue'])\n",
    "\n",
    "# Create the violin plot\n",
    "plt.figure(figsize=(8, 6))\n",
    "ax = sns.violinplot(x='paralogue', y='pLDDT', data=df, hue='paralogue', palette=['#31688EFF', '#8302a2', '#35B779FF'], legend=False)\n",
    "\n",
    "# Add ANOVA result\n",
    "plt.text(0.5, 1.05, f'One-way ANOVA: F={f_statistic:.2f}, p={p_value:.4f}', \n",
    "         horizontalalignment='center', transform=ax.transAxes)\n",
    "\n",
    "# Add significance annotations and Cohen's d\n",
    "y_max = df['pLDDT'].max()\n",
    "y_min = df['pLDDT'].min()\n",
    "y_range = y_max - y_min\n",
    "\n",
    "for idx, row in enumerate(tukey_results._results_table.data[1:], 1):\n",
    "    group1, group2 = row[0], row[1]\n",
    "    p_value = row[3]\n",
    "    \n",
    "    # Calculate Cohen's d\n",
    "    group1_data = df[df['paralogue'] == group1]['pLDDT']\n",
    "    group2_data = df[df['paralogue'] == group2]['pLDDT']\n",
    "    d = cohens_d(group1_data, group2_data)\n",
    "    \n",
    "    print(f\"Comparison: {group1} vs {group2}, p-value: {p_value:.4f}, Cohen's d: {d:.2f}\")\n",
    "    \n",
    "    if p_value < 0.05:\n",
    "        try:\n",
    "            x1, x2 = categories.tolist().index(group1), categories.tolist().index(group2)\n",
    "            y = y_max + (idx * 0.07 * y_range)\n",
    "            plt.plot([x1, x1, x2, x2], [y, y + 0.05 * y_range, y + 0.05 * y_range, y], lw=1.5, c='k')\n",
    "            plt.text((x1 + x2) * 0.4, y + 0.06 * y_range, f'p = {p_value:.4f}\\nd = {d:.2f}', ha='center', va='bottom')\n",
    "        except ValueError as e:\n",
    "            print(f\"Error plotting comparison {group1} vs {group2}: {e}\")\n",
    "\n",
    "plt.xlabel('Paralogue')\n",
    "plt.ylabel('pLDDT')\n",
    "\n",
    "# plt.tight_layout()\n",
    "plt.show()\n",
    "\n",
    "# Print Tukey's HSD test results\n",
    "print(tukey_results)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Heat Map"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import seaborn as sns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "At_v_Sc = pd.read_csv(\"/home/mattm/projects/np_paper_structural/20240328_colabfold/AF2_structures_all/At_vs_Sc/At_v_Sc_all-v-all_EMD_dx.csv\") \n",
    "\n",
    "subset_df = At_v_Sc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "subset_df = subset_df.loc[(subset_df['id1'].isin(['ScHAL2', 'AtSAL1'])) & (subset_df['id2'].isin(['ScHAL2', 'AtSAL1']))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>basename2</th>\n",
       "      <th>AtAHL-pH_7_0</th>\n",
       "      <th>AtAHL-pH_7_3</th>\n",
       "      <th>AtAHL-pH_7_6</th>\n",
       "      <th>AtAHL-pH_7_9</th>\n",
       "      <th>AtAHL-pH_8_2</th>\n",
       "      <th>AtCNP-pH_7_0</th>\n",
       "      <th>AtCNP-pH_7_3</th>\n",
       "      <th>AtCNP-pH_7_6</th>\n",
       "      <th>AtCNP-pH_7_9</th>\n",
       "      <th>AtCNP-pH_8_2</th>\n",
       "      <th>AtSAL1-pH_7_0</th>\n",
       "      <th>AtSAL1-pH_7_3</th>\n",
       "      <th>AtSAL1-pH_7_6</th>\n",
       "      <th>AtSAL1-pH_7_9</th>\n",
       "      <th>AtSAL1-pH_8_2</th>\n",
       "      <th>ScHAL2-pH_7_0</th>\n",
       "      <th>ScHAL2-pH_7_3</th>\n",
       "      <th>ScHAL2-pH_7_6</th>\n",
       "      <th>ScHAL2-pH_7_9</th>\n",
       "      <th>ScHAL2-pH_8_2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>basename1</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AtAHL-pH_7_0</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002695</td>\n",
       "      <td>0.004700</td>\n",
       "      <td>0.004700</td>\n",
       "      <td>0.004700</td>\n",
       "      <td>0.004700</td>\n",
       "      <td>0.001272</td>\n",
       "      <td>0.006951</td>\n",
       "      <td>0.002798</td>\n",
       "      <td>0.002798</td>\n",
       "      <td>0.002798</td>\n",
       "      <td>0.002798</td>\n",
       "      <td>0.003905</td>\n",
       "      <td>0.003905</td>\n",
       "      <td>0.002881</td>\n",
       "      <td>0.002881</td>\n",
       "      <td>0.002881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AtAHL-pH_7_3</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002695</td>\n",
       "      <td>0.004700</td>\n",
       "      <td>0.004700</td>\n",
       "      <td>0.004700</td>\n",
       "      <td>0.004700</td>\n",
       "      <td>0.001272</td>\n",
       "      <td>0.006951</td>\n",
       "      <td>0.002798</td>\n",
       "      <td>0.002798</td>\n",
       "      <td>0.002798</td>\n",
       "      <td>0.002798</td>\n",
       "      <td>0.003905</td>\n",
       "      <td>0.003905</td>\n",
       "      <td>0.002881</td>\n",
       "      <td>0.002881</td>\n",
       "      <td>0.002881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AtAHL-pH_7_6</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002695</td>\n",
       "      <td>0.004700</td>\n",
       "      <td>0.004700</td>\n",
       "      <td>0.004700</td>\n",
       "      <td>0.004700</td>\n",
       "      <td>0.001272</td>\n",
       "      <td>0.006951</td>\n",
       "      <td>0.002798</td>\n",
       "      <td>0.002798</td>\n",
       "      <td>0.002798</td>\n",
       "      <td>0.002798</td>\n",
       "      <td>0.003905</td>\n",
       "      <td>0.003905</td>\n",
       "      <td>0.002881</td>\n",
       "      <td>0.002881</td>\n",
       "      <td>0.002881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AtAHL-pH_7_9</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002695</td>\n",
       "      <td>0.004700</td>\n",
       "      <td>0.004700</td>\n",
       "      <td>0.004700</td>\n",
       "      <td>0.004700</td>\n",
       "      <td>0.001272</td>\n",
       "      <td>0.006951</td>\n",
       "      <td>0.002798</td>\n",
       "      <td>0.002798</td>\n",
       "      <td>0.002798</td>\n",
       "      <td>0.002798</td>\n",
       "      <td>0.003905</td>\n",
       "      <td>0.003905</td>\n",
       "      <td>0.002881</td>\n",
       "      <td>0.002881</td>\n",
       "      <td>0.002881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AtAHL-pH_8_2</th>\n",
       "      <td>0.002697</td>\n",
       "      <td>0.002697</td>\n",
       "      <td>0.002697</td>\n",
       "      <td>0.002697</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.007393</td>\n",
       "      <td>0.007393</td>\n",
       "      <td>0.007393</td>\n",
       "      <td>0.007393</td>\n",
       "      <td>0.003224</td>\n",
       "      <td>0.009560</td>\n",
       "      <td>0.003704</td>\n",
       "      <td>0.003704</td>\n",
       "      <td>0.003704</td>\n",
       "      <td>0.003704</td>\n",
       "      <td>0.006315</td>\n",
       "      <td>0.006315</td>\n",
       "      <td>0.002163</td>\n",
       "      <td>0.002163</td>\n",
       "      <td>0.002163</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AtCNP-pH_7_0</th>\n",
       "      <td>0.004345</td>\n",
       "      <td>0.004345</td>\n",
       "      <td>0.004345</td>\n",
       "      <td>0.004345</td>\n",
       "      <td>0.006830</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.003922</td>\n",
       "      <td>0.003074</td>\n",
       "      <td>0.004628</td>\n",
       "      <td>0.004628</td>\n",
       "      <td>0.004628</td>\n",
       "      <td>0.004628</td>\n",
       "      <td>0.001885</td>\n",
       "      <td>0.001885</td>\n",
       "      <td>0.006150</td>\n",
       "      <td>0.006150</td>\n",
       "      <td>0.006150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AtCNP-pH_7_3</th>\n",
       "      <td>0.004345</td>\n",
       "      <td>0.004345</td>\n",
       "      <td>0.004345</td>\n",
       "      <td>0.004345</td>\n",
       "      <td>0.006830</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.003922</td>\n",
       "      <td>0.003074</td>\n",
       "      <td>0.004628</td>\n",
       "      <td>0.004628</td>\n",
       "      <td>0.004628</td>\n",
       "      <td>0.004628</td>\n",
       "      <td>0.001885</td>\n",
       "      <td>0.001885</td>\n",
       "      <td>0.006150</td>\n",
       "      <td>0.006150</td>\n",
       "      <td>0.006150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AtCNP-pH_7_6</th>\n",
       "      <td>0.004345</td>\n",
       "      <td>0.004345</td>\n",
       "      <td>0.004345</td>\n",
       "      <td>0.004345</td>\n",
       "      <td>0.006830</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.003922</td>\n",
       "      <td>0.003074</td>\n",
       "      <td>0.004628</td>\n",
       "      <td>0.004628</td>\n",
       "      <td>0.004628</td>\n",
       "      <td>0.004628</td>\n",
       "      <td>0.001885</td>\n",
       "      <td>0.001885</td>\n",
       "      <td>0.006150</td>\n",
       "      <td>0.006150</td>\n",
       "      <td>0.006150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AtCNP-pH_7_9</th>\n",
       "      <td>0.004345</td>\n",
       "      <td>0.004345</td>\n",
       "      <td>0.004345</td>\n",
       "      <td>0.004345</td>\n",
       "      <td>0.006830</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.003922</td>\n",
       "      <td>0.003074</td>\n",
       "      <td>0.004628</td>\n",
       "      <td>0.004628</td>\n",
       "      <td>0.004628</td>\n",
       "      <td>0.004628</td>\n",
       "      <td>0.001885</td>\n",
       "      <td>0.001885</td>\n",
       "      <td>0.006150</td>\n",
       "      <td>0.006150</td>\n",
       "      <td>0.006150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AtCNP-pH_8_2</th>\n",
       "      <td>0.001172</td>\n",
       "      <td>0.001172</td>\n",
       "      <td>0.001172</td>\n",
       "      <td>0.001172</td>\n",
       "      <td>0.002967</td>\n",
       "      <td>0.003908</td>\n",
       "      <td>0.003908</td>\n",
       "      <td>0.003908</td>\n",
       "      <td>0.003908</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.005927</td>\n",
       "      <td>0.001976</td>\n",
       "      <td>0.001976</td>\n",
       "      <td>0.001976</td>\n",
       "      <td>0.001976</td>\n",
       "      <td>0.003411</td>\n",
       "      <td>0.003411</td>\n",
       "      <td>0.002527</td>\n",
       "      <td>0.002527</td>\n",
       "      <td>0.002527</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AtSAL1-pH_7_0</th>\n",
       "      <td>0.006862</td>\n",
       "      <td>0.006862</td>\n",
       "      <td>0.006862</td>\n",
       "      <td>0.006862</td>\n",
       "      <td>0.009431</td>\n",
       "      <td>0.003283</td>\n",
       "      <td>0.003283</td>\n",
       "      <td>0.003283</td>\n",
       "      <td>0.003283</td>\n",
       "      <td>0.006351</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.006902</td>\n",
       "      <td>0.006902</td>\n",
       "      <td>0.006902</td>\n",
       "      <td>0.006902</td>\n",
       "      <td>0.004986</td>\n",
       "      <td>0.004986</td>\n",
       "      <td>0.008726</td>\n",
       "      <td>0.008726</td>\n",
       "      <td>0.008726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AtSAL1-pH_7_3</th>\n",
       "      <td>0.002736</td>\n",
       "      <td>0.002736</td>\n",
       "      <td>0.002736</td>\n",
       "      <td>0.002736</td>\n",
       "      <td>0.003619</td>\n",
       "      <td>0.004895</td>\n",
       "      <td>0.004895</td>\n",
       "      <td>0.004895</td>\n",
       "      <td>0.004895</td>\n",
       "      <td>0.002097</td>\n",
       "      <td>0.006836</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.005084</td>\n",
       "      <td>0.005084</td>\n",
       "      <td>0.003665</td>\n",
       "      <td>0.003665</td>\n",
       "      <td>0.003665</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AtSAL1-pH_7_6</th>\n",
       "      <td>0.002736</td>\n",
       "      <td>0.002736</td>\n",
       "      <td>0.002736</td>\n",
       "      <td>0.002736</td>\n",
       "      <td>0.003619</td>\n",
       "      <td>0.004895</td>\n",
       "      <td>0.004895</td>\n",
       "      <td>0.004895</td>\n",
       "      <td>0.004895</td>\n",
       "      <td>0.002097</td>\n",
       "      <td>0.006836</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.005084</td>\n",
       "      <td>0.005084</td>\n",
       "      <td>0.003665</td>\n",
       "      <td>0.003665</td>\n",
       "      <td>0.003665</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AtSAL1-pH_7_9</th>\n",
       "      <td>0.002736</td>\n",
       "      <td>0.002736</td>\n",
       "      <td>0.002736</td>\n",
       "      <td>0.002736</td>\n",
       "      <td>0.003619</td>\n",
       "      <td>0.004895</td>\n",
       "      <td>0.004895</td>\n",
       "      <td>0.004895</td>\n",
       "      <td>0.004895</td>\n",
       "      <td>0.002097</td>\n",
       "      <td>0.006836</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.005084</td>\n",
       "      <td>0.005084</td>\n",
       "      <td>0.003665</td>\n",
       "      <td>0.003665</td>\n",
       "      <td>0.003665</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AtSAL1-pH_8_2</th>\n",
       "      <td>0.002736</td>\n",
       "      <td>0.002736</td>\n",
       "      <td>0.002736</td>\n",
       "      <td>0.002736</td>\n",
       "      <td>0.003619</td>\n",
       "      <td>0.004895</td>\n",
       "      <td>0.004895</td>\n",
       "      <td>0.004895</td>\n",
       "      <td>0.004895</td>\n",
       "      <td>0.002097</td>\n",
       "      <td>0.006836</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.005084</td>\n",
       "      <td>0.005084</td>\n",
       "      <td>0.003665</td>\n",
       "      <td>0.003665</td>\n",
       "      <td>0.003665</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ScHAL2-pH_7_0</th>\n",
       "      <td>0.004223</td>\n",
       "      <td>0.004223</td>\n",
       "      <td>0.004223</td>\n",
       "      <td>0.004223</td>\n",
       "      <td>0.006824</td>\n",
       "      <td>0.002205</td>\n",
       "      <td>0.002205</td>\n",
       "      <td>0.002205</td>\n",
       "      <td>0.002205</td>\n",
       "      <td>0.004005</td>\n",
       "      <td>0.005462</td>\n",
       "      <td>0.005623</td>\n",
       "      <td>0.005623</td>\n",
       "      <td>0.005623</td>\n",
       "      <td>0.005623</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.005975</td>\n",
       "      <td>0.005975</td>\n",
       "      <td>0.005975</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ScHAL2-pH_7_3</th>\n",
       "      <td>0.004223</td>\n",
       "      <td>0.004223</td>\n",
       "      <td>0.004223</td>\n",
       "      <td>0.004223</td>\n",
       "      <td>0.006824</td>\n",
       "      <td>0.002205</td>\n",
       "      <td>0.002205</td>\n",
       "      <td>0.002205</td>\n",
       "      <td>0.002205</td>\n",
       "      <td>0.004005</td>\n",
       "      <td>0.005462</td>\n",
       "      <td>0.005623</td>\n",
       "      <td>0.005623</td>\n",
       "      <td>0.005623</td>\n",
       "      <td>0.005623</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.005975</td>\n",
       "      <td>0.005975</td>\n",
       "      <td>0.005975</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ScHAL2-pH_7_6</th>\n",
       "      <td>0.003012</td>\n",
       "      <td>0.003012</td>\n",
       "      <td>0.003012</td>\n",
       "      <td>0.003012</td>\n",
       "      <td>0.002259</td>\n",
       "      <td>0.006956</td>\n",
       "      <td>0.006956</td>\n",
       "      <td>0.006956</td>\n",
       "      <td>0.006956</td>\n",
       "      <td>0.002867</td>\n",
       "      <td>0.009241</td>\n",
       "      <td>0.003919</td>\n",
       "      <td>0.003919</td>\n",
       "      <td>0.003919</td>\n",
       "      <td>0.003919</td>\n",
       "      <td>0.005777</td>\n",
       "      <td>0.005777</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ScHAL2-pH_7_9</th>\n",
       "      <td>0.003012</td>\n",
       "      <td>0.003012</td>\n",
       "      <td>0.003012</td>\n",
       "      <td>0.003012</td>\n",
       "      <td>0.002259</td>\n",
       "      <td>0.006956</td>\n",
       "      <td>0.006956</td>\n",
       "      <td>0.006956</td>\n",
       "      <td>0.006956</td>\n",
       "      <td>0.002867</td>\n",
       "      <td>0.009241</td>\n",
       "      <td>0.003919</td>\n",
       "      <td>0.003919</td>\n",
       "      <td>0.003919</td>\n",
       "      <td>0.003919</td>\n",
       "      <td>0.005777</td>\n",
       "      <td>0.005777</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ScHAL2-pH_8_2</th>\n",
       "      <td>0.003012</td>\n",
       "      <td>0.003012</td>\n",
       "      <td>0.003012</td>\n",
       "      <td>0.003012</td>\n",
       "      <td>0.002259</td>\n",
       "      <td>0.006956</td>\n",
       "      <td>0.006956</td>\n",
       "      <td>0.006956</td>\n",
       "      <td>0.006956</td>\n",
       "      <td>0.002867</td>\n",
       "      <td>0.009241</td>\n",
       "      <td>0.003919</td>\n",
       "      <td>0.003919</td>\n",
       "      <td>0.003919</td>\n",
       "      <td>0.003919</td>\n",
       "      <td>0.005777</td>\n",
       "      <td>0.005777</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "basename2      AtAHL-pH_7_0  AtAHL-pH_7_3  AtAHL-pH_7_6  AtAHL-pH_7_9  \\\n",
       "basename1                                                               \n",
       "AtAHL-pH_7_0       0.000000      0.000000      0.000000      0.000000   \n",
       "AtAHL-pH_7_3       0.000000      0.000000      0.000000      0.000000   \n",
       "AtAHL-pH_7_6       0.000000      0.000000      0.000000      0.000000   \n",
       "AtAHL-pH_7_9       0.000000      0.000000      0.000000      0.000000   \n",
       "AtAHL-pH_8_2       0.002697      0.002697      0.002697      0.002697   \n",
       "AtCNP-pH_7_0       0.004345      0.004345      0.004345      0.004345   \n",
       "AtCNP-pH_7_3       0.004345      0.004345      0.004345      0.004345   \n",
       "AtCNP-pH_7_6       0.004345      0.004345      0.004345      0.004345   \n",
       "AtCNP-pH_7_9       0.004345      0.004345      0.004345      0.004345   \n",
       "AtCNP-pH_8_2       0.001172      0.001172      0.001172      0.001172   \n",
       "AtSAL1-pH_7_0      0.006862      0.006862      0.006862      0.006862   \n",
       "AtSAL1-pH_7_3      0.002736      0.002736      0.002736      0.002736   \n",
       "AtSAL1-pH_7_6      0.002736      0.002736      0.002736      0.002736   \n",
       "AtSAL1-pH_7_9      0.002736      0.002736      0.002736      0.002736   \n",
       "AtSAL1-pH_8_2      0.002736      0.002736      0.002736      0.002736   \n",
       "ScHAL2-pH_7_0      0.004223      0.004223      0.004223      0.004223   \n",
       "ScHAL2-pH_7_3      0.004223      0.004223      0.004223      0.004223   \n",
       "ScHAL2-pH_7_6      0.003012      0.003012      0.003012      0.003012   \n",
       "ScHAL2-pH_7_9      0.003012      0.003012      0.003012      0.003012   \n",
       "ScHAL2-pH_8_2      0.003012      0.003012      0.003012      0.003012   \n",
       "\n",
       "basename2      AtAHL-pH_8_2  AtCNP-pH_7_0  AtCNP-pH_7_3  AtCNP-pH_7_6  \\\n",
       "basename1                                                               \n",
       "AtAHL-pH_7_0       0.002695      0.004700      0.004700      0.004700   \n",
       "AtAHL-pH_7_3       0.002695      0.004700      0.004700      0.004700   \n",
       "AtAHL-pH_7_6       0.002695      0.004700      0.004700      0.004700   \n",
       "AtAHL-pH_7_9       0.002695      0.004700      0.004700      0.004700   \n",
       "AtAHL-pH_8_2       0.000000      0.007393      0.007393      0.007393   \n",
       "AtCNP-pH_7_0       0.006830      0.000000      0.000000      0.000000   \n",
       "AtCNP-pH_7_3       0.006830      0.000000      0.000000      0.000000   \n",
       "AtCNP-pH_7_6       0.006830      0.000000      0.000000      0.000000   \n",
       "AtCNP-pH_7_9       0.006830      0.000000      0.000000      0.000000   \n",
       "AtCNP-pH_8_2       0.002967      0.003908      0.003908      0.003908   \n",
       "AtSAL1-pH_7_0      0.009431      0.003283      0.003283      0.003283   \n",
       "AtSAL1-pH_7_3      0.003619      0.004895      0.004895      0.004895   \n",
       "AtSAL1-pH_7_6      0.003619      0.004895      0.004895      0.004895   \n",
       "AtSAL1-pH_7_9      0.003619      0.004895      0.004895      0.004895   \n",
       "AtSAL1-pH_8_2      0.003619      0.004895      0.004895      0.004895   \n",
       "ScHAL2-pH_7_0      0.006824      0.002205      0.002205      0.002205   \n",
       "ScHAL2-pH_7_3      0.006824      0.002205      0.002205      0.002205   \n",
       "ScHAL2-pH_7_6      0.002259      0.006956      0.006956      0.006956   \n",
       "ScHAL2-pH_7_9      0.002259      0.006956      0.006956      0.006956   \n",
       "ScHAL2-pH_8_2      0.002259      0.006956      0.006956      0.006956   \n",
       "\n",
       "basename2      AtCNP-pH_7_9  AtCNP-pH_8_2  AtSAL1-pH_7_0  AtSAL1-pH_7_3  \\\n",
       "basename1                                                                 \n",
       "AtAHL-pH_7_0       0.004700      0.001272       0.006951       0.002798   \n",
       "AtAHL-pH_7_3       0.004700      0.001272       0.006951       0.002798   \n",
       "AtAHL-pH_7_6       0.004700      0.001272       0.006951       0.002798   \n",
       "AtAHL-pH_7_9       0.004700      0.001272       0.006951       0.002798   \n",
       "AtAHL-pH_8_2       0.007393      0.003224       0.009560       0.003704   \n",
       "AtCNP-pH_7_0       0.000000      0.003922       0.003074       0.004628   \n",
       "AtCNP-pH_7_3       0.000000      0.003922       0.003074       0.004628   \n",
       "AtCNP-pH_7_6       0.000000      0.003922       0.003074       0.004628   \n",
       "AtCNP-pH_7_9       0.000000      0.003922       0.003074       0.004628   \n",
       "AtCNP-pH_8_2       0.003908      0.000000       0.005927       0.001976   \n",
       "AtSAL1-pH_7_0      0.003283      0.006351       0.000000       0.006902   \n",
       "AtSAL1-pH_7_3      0.004895      0.002097       0.006836       0.000000   \n",
       "AtSAL1-pH_7_6      0.004895      0.002097       0.006836       0.000000   \n",
       "AtSAL1-pH_7_9      0.004895      0.002097       0.006836       0.000000   \n",
       "AtSAL1-pH_8_2      0.004895      0.002097       0.006836       0.000000   \n",
       "ScHAL2-pH_7_0      0.002205      0.004005       0.005462       0.005623   \n",
       "ScHAL2-pH_7_3      0.002205      0.004005       0.005462       0.005623   \n",
       "ScHAL2-pH_7_6      0.006956      0.002867       0.009241       0.003919   \n",
       "ScHAL2-pH_7_9      0.006956      0.002867       0.009241       0.003919   \n",
       "ScHAL2-pH_8_2      0.006956      0.002867       0.009241       0.003919   \n",
       "\n",
       "basename2      AtSAL1-pH_7_6  AtSAL1-pH_7_9  AtSAL1-pH_8_2  ScHAL2-pH_7_0  \\\n",
       "basename1                                                                   \n",
       "AtAHL-pH_7_0        0.002798       0.002798       0.002798       0.003905   \n",
       "AtAHL-pH_7_3        0.002798       0.002798       0.002798       0.003905   \n",
       "AtAHL-pH_7_6        0.002798       0.002798       0.002798       0.003905   \n",
       "AtAHL-pH_7_9        0.002798       0.002798       0.002798       0.003905   \n",
       "AtAHL-pH_8_2        0.003704       0.003704       0.003704       0.006315   \n",
       "AtCNP-pH_7_0        0.004628       0.004628       0.004628       0.001885   \n",
       "AtCNP-pH_7_3        0.004628       0.004628       0.004628       0.001885   \n",
       "AtCNP-pH_7_6        0.004628       0.004628       0.004628       0.001885   \n",
       "AtCNP-pH_7_9        0.004628       0.004628       0.004628       0.001885   \n",
       "AtCNP-pH_8_2        0.001976       0.001976       0.001976       0.003411   \n",
       "AtSAL1-pH_7_0       0.006902       0.006902       0.006902       0.004986   \n",
       "AtSAL1-pH_7_3       0.000000       0.000000       0.000000       0.005084   \n",
       "AtSAL1-pH_7_6       0.000000       0.000000       0.000000       0.005084   \n",
       "AtSAL1-pH_7_9       0.000000       0.000000       0.000000       0.005084   \n",
       "AtSAL1-pH_8_2       0.000000       0.000000       0.000000       0.005084   \n",
       "ScHAL2-pH_7_0       0.005623       0.005623       0.005623       0.000000   \n",
       "ScHAL2-pH_7_3       0.005623       0.005623       0.005623       0.000000   \n",
       "ScHAL2-pH_7_6       0.003919       0.003919       0.003919       0.005777   \n",
       "ScHAL2-pH_7_9       0.003919       0.003919       0.003919       0.005777   \n",
       "ScHAL2-pH_8_2       0.003919       0.003919       0.003919       0.005777   \n",
       "\n",
       "basename2      ScHAL2-pH_7_3  ScHAL2-pH_7_6  ScHAL2-pH_7_9  ScHAL2-pH_8_2  \n",
       "basename1                                                                  \n",
       "AtAHL-pH_7_0        0.003905       0.002881       0.002881       0.002881  \n",
       "AtAHL-pH_7_3        0.003905       0.002881       0.002881       0.002881  \n",
       "AtAHL-pH_7_6        0.003905       0.002881       0.002881       0.002881  \n",
       "AtAHL-pH_7_9        0.003905       0.002881       0.002881       0.002881  \n",
       "AtAHL-pH_8_2        0.006315       0.002163       0.002163       0.002163  \n",
       "AtCNP-pH_7_0        0.001885       0.006150       0.006150       0.006150  \n",
       "AtCNP-pH_7_3        0.001885       0.006150       0.006150       0.006150  \n",
       "AtCNP-pH_7_6        0.001885       0.006150       0.006150       0.006150  \n",
       "AtCNP-pH_7_9        0.001885       0.006150       0.006150       0.006150  \n",
       "AtCNP-pH_8_2        0.003411       0.002527       0.002527       0.002527  \n",
       "AtSAL1-pH_7_0       0.004986       0.008726       0.008726       0.008726  \n",
       "AtSAL1-pH_7_3       0.005084       0.003665       0.003665       0.003665  \n",
       "AtSAL1-pH_7_6       0.005084       0.003665       0.003665       0.003665  \n",
       "AtSAL1-pH_7_9       0.005084       0.003665       0.003665       0.003665  \n",
       "AtSAL1-pH_8_2       0.005084       0.003665       0.003665       0.003665  \n",
       "ScHAL2-pH_7_0       0.000000       0.005975       0.005975       0.005975  \n",
       "ScHAL2-pH_7_3       0.000000       0.005975       0.005975       0.005975  \n",
       "ScHAL2-pH_7_6       0.005777       0.000000       0.000000       0.000000  \n",
       "ScHAL2-pH_7_9       0.005777       0.000000       0.000000       0.000000  \n",
       "ScHAL2-pH_8_2       0.005777       0.000000       0.000000       0.000000  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subset_df2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Axes: xlabel='basename2', ylabel='basename1'>"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAocAAAIPCAYAAAAB93RXAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjkuMCwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy80BEi2AAAACXBIWXMAAA9hAAAPYQGoP6dpAACPg0lEQVR4nOzdeVxU9f4/8NcMw6YgiohAKiCoICqYCKLmUi6QadzIkFxzy1K/brlxU9NKuC5ZqUldB9DrAi5pZkYXNRQVJRTMBTRNo1TQUiBQkeXz+8Mf5zYNyDAMzhFfz/s4j5xzPuczr8PNfPv5nPM5CiGEABERERERAKWxAxARERGRfLA4JCIiIiIJi0MiIiIikrA4JCIiIiIJi0MiIiIikrA4JCIiIiIJi0MiIiIikrA4JCIiIiIJi0MiIiIikqiMHYDql/wHrY0doUofNYgwdgR6Sr3Q7aKxIzzRpqY2NXYEqiOnH7xdp/03sow0WF8F9+YZrC+5Y3FIRERE9ZKJUBg7whOJ08pEREREJOHIIREREdVLJuDIoT5YHBIREVG9xOJQPywOiYiIqF4yEbx7Th/8qRERERGRhCOHREREVC8pOa2sFxaHREREVC+xONQPp5WJiIiISMKRQyIiIqqX+LSyflgcEhERUb3EaWX9yG5aOSUlBSYmJhg0aJDG/vfeew8+Pj5a7U+fPo0hQ4bA3t4eFhYWcHFxQWhoKG7evKnVNiIiAiYmJli+fLnWsdjYWDRu3LjKXIcPH8bgwYPh5OQEhUKB3bt31/TSqlRVf2PGjEFwcLBOfbi4uEChUGhtkydP1un8+/fvY/LkyWjatCmsrKwQEhKC3NzcGlwFERER1QeyKw7VajWmTp2Kw4cP4/r1649se+vWLbzwwguwtbXFd999h8zMTMTExMDJyQlFRUVa7aOjozFnzhxER0fXOFdRURG8vb2xdu3aGp/7OPzwww+4ceOGtCUmJgIAhg4dqtP5M2bMwNdff43t27fj0KFDuH79Ol555ZW6jExERFSnlFAYbHuayGpaubCwEPHx8UhLS0NOTg5iY2MRHh6O2NhYLF68GMDDUTYAiImJQePGjZGfn4/169dDpXp4Ka6urujbt69W34cOHcK9e/ewZMkSbNy4EceOHUP37t11zhYUFISgoKAaX5NCocBnn32GPXv2ICkpCY6Ojli2bBleffXVGvf1KM2aNdP4HBkZCTc3N/Tu3bvac/Pz86FWq7FlyxY8//zzAB7+fD09PXH8+HF069bNoFmJiIgeB6UwdoInk6xGDrdt2wYPDw+0a9cOI0aMQHR0NIQQCA0NxaxZs+Dl5SWNjIWGhsLBwQGlpaXYtWsXhHj0vwFqtRphYWEwNTVFWFgY1Gr1Y7oqYMGCBQgJCcHp06cxfPhwDBs2DJmZmXX2fQ8ePMCmTZswduxYqZh+lJMnT6KkpAT9+vWT9nl4eKBVq1ZISUmps5xEREQkP7IqDtVqNUaMGAEACAwMRH5+Pg4dOgRLS0tYWVlBpVLBwcEBDg4OsLS0RLdu3RAeHo7XX38ddnZ2CAoKwvLly7XulSsoKMCOHTukvkeMGIFt27ahsLDwsVzX0KFDMX78eLRt2xbvv/8+fH19sXr1ao02YWFhsLKy0tg2b96s1/ft3r0beXl5GDNmjE7tc3JyYGZmpnXPZfPmzZGTk6NXBiIiImPjtLJ+ZFMcXrhwAampqQgLCwMAqFQqhIaGVjvC9+GHHyInJwdRUVHw8vJCVFQUPDw8cObMGanN1q1b4ebmBm9vbwCAj48PnJ2dER8fb7D8S5cu1SjssrOzpWMBAQEabQMCArRGDletWoWMjAyNbciQIXplUavVCAoKgpOTk17n66q4uBgFBQUaW3Exx/CJiEgeWBzqRzbFoVqtRmlpKZycnKBSqaBSqbBu3Trs3LkT+fn5jzy3adOmGDp0KFasWIHMzEw4OTlhxYoVGn2fO3dO6lelUuH8+fN6PZhSlUmTJmkUdjUtzBwcHODu7q6xWVtb1zjHL7/8gv3792P8+PE1+u4HDx4gLy9PY39ubi4cHByqPC8iIgI2NjYa20fL8qpsT0RE9DixONSPLB5IKS0txcaNG7Fy5UoMGDBA41hwcDC2bt0KMzMzlJWVVduXmZkZ3NzcpKeVz5w5g7S0NCQlJcHW1lZqd/v2bfTp0wdZWVnw8PCo9TXY2tpq9P9Xx48fx6hRozQ+d+7cudbfWZmYmBjY29trLQX0KF26dIGpqSkOHDiAkJAQAA9HcrOzs7VGPf9q/vz5mDlzpsa++wpv/YITERGRLMiiONy7dy/u3LmDcePGwcbGRuNYSEgI1Go1ZsyYgStXriAjIwMtWrSAtbU1EhMTERcXh2HDhqFt27YQQuDrr7/Gvn37EBMTA+DhqKGfnx969eql9b1du3aFWq2W1j0sKytDRkaGRhtzc3N4enqisLAQly5dkvZXZLG1tUWrVq0eeX3bt2+Hr68vevbsic2bNyM1NbVOHogpLy9HTEwMRo8eLT29rQsbGxuMGzcOM2fOhK2tLRo1aoSpU6ciICDgkU8qm5ubw9zcXGOfePB0/e2KiIjk62kb8TMUWRSHarUa/fr10yoMgYfF4bJly+Dl5YXAwED07dsXeXl5iImJQa9evdCgQQPMmjULv/76K8zNzdGmTRusX78eI0eOlJ7anTt3bqXfGxISgpUrV2Lp0qUAHi6l8/cRPTc3N1y6dAlpaWkaS+RUjJiNHj0asbGxj7y+xYsXIy4uDm+//TYcHR2xdetWtG/fviY/Ip3s378f2dnZGDt2bI3PXbVqFZRKJUJCQlBcXIyBAwfis88+M3hGIiKix0XB4lAvClHdGjBUKwqFArt27dL5TSdPuvwHrY0doUofNYgwdgR6Sr3Q7aKxIzzRpqY2NXYEqiOnH7xdp/37mK0zWF8ZD94yWF9yJ4uRQyIiIiJD47SyfmTztDJVLTs7W2sNxKqWzanM5s2bqzzXy8vrMV0FERHR46U04PY04chhHTPErL2Tk5PWgzJ/P/4oQ4YMgb+/f6XHTE1NaxONiIiI6hkWh08AlUoFd3d3vc+3trbWa81EIiKiJ5lScFpZHywOiYiIqF7iPYf6YXFIRERE9RKXstHP03aPJRERERE9AkcOiYiIqF7iCJh+WBwSERFRvcR7DvXDopqIiIiIJBw5JCIionqJI4f6YXFIRERE9RJLQ/1wWpmIiIiIJBw5JCIionqJ08r6YXFIBvVJw6XGjlClmXfnGzsCPaXS+4wzdoQn2t4xacaOQE8ovj5PP5xWJiIiIiIJRw6JiIioXuK4oX5YHBIREVG9xHsO9cPikIiIiOol3junH/7ciIiIiEjCkUMiIiKqlxScVtYLi0MiIiKql3jPoX44rUxERERkYGvXroWLiwssLCzg7++P1NTUR7bfvn07PDw8YGFhgY4dO2Lfvn0ax4UQWLhwIRwdHWFpaYl+/frhp59+0mhz8eJFvPzyy7Czs0OjRo3Qs2dPfP/99zXOzuKQiIiI6iWFMNxWE/Hx8Zg5cyYWLVqEU6dOwdvbGwMHDsTNmzcrbX/s2DGEhYVh3LhxSE9PR3BwMIKDg3H27FmpzbJly/Dpp58iKioKJ06cQMOGDTFw4EDcv39favPSSy+htLQUBw8exMmTJ+Ht7Y2XXnoJOTk5Nfy5CVHDSyaq2hLTOGNHqNK0onBjR6CnFN+QUjtuHS4bOwLVkZZfRNdp/y+b/MdgfX1VNlLntv7+/ujatSvWrFkDACgvL0fLli0xdepUzJs3T6t9aGgoioqKsHfvXmlft27d4OPjg6ioKAgh4OTkhFmzZuGdd94BAOTn56N58+aIjY3FsGHD8Pvvv6NZs2Y4fPgwnnvuOQDAn3/+iUaNGiExMRH9+vXTOT9HDomIiIiqUVxcjIKCAo2tuLhYq92DBw9w8uRJjWJMqVSiX79+SElJqbTvlJQUreJt4MCBUvsrV64gJydHo42NjQ38/f2lNk2bNkW7du2wceNGFBUVobS0FJ9//jns7e3RpUuXGl2r7IrDlJQUmJiYYNCgQRr733vvPfj4+Gi1P336NIYMGQJ7e3tYWFjAxcUFoaGhlQ7dRkREwMTEBMuXL9c6Fhsbi8aNG1eZ6/Dhwxg8eDCcnJygUCiwe/fuml5alarqb8yYMQgODtapDxcXFygUCq1t8uTJOp3/5ptvws3NDZaWlmjWrBlefvllZGVl1eAqiIiI5EVpwC0iIgI2NjYaW0REhNZ3/v777ygrK0Pz5s019jdv3rzK6d2cnJxHtq/456PaKBQK7N+/H+np6bC2toaFhQU++ugjJCQkoEmTJjr8tP5HdsWhWq3G1KlTcfjwYVy/fv2RbW/duoUXXngBtra2+O6775CZmYmYmBg4OTmhqKhIq310dDTmzJmD6OiaD2MXFRXB29sba9eurfG5j8MPP/yAGzduSFtiYiIAYOjQoTqd36VLF8TExCAzMxPfffcdhBAYMGAAysrK6jI2ERFRnVEY8H/z589Hfn6+xjZ//nxjX6JECIHJkyfD3t4eycnJSE1NRXBwMAYPHowbN27UqC9ZLWVTWFiI+Ph4pKWlIScnB7GxsQgPD0dsbCwWL14M4GFlDAAxMTFo3Lgx8vPzsX79eqhUDy/F1dUVffv21er70KFDuHfvHpYsWYKNGzfi2LFj6N69u87ZgoKCEBQUVONrUigU+Oyzz7Bnzx4kJSXB0dERy5Ytw6uvvlrjvh6lWbNmGp8jIyPh5uaG3r1763T+xIkTpV+7uLjggw8+gLe3N65evQo3NzeDZiUiInrSmJubw9zcvNp2dnZ2MDExQW5ursb+3NxcODg4VHqOg4PDI9tX/DM3NxeOjo4abSpmVQ8ePIi9e/fizp07aNSoEQDgs88+Q2JiIjZs2FDpvY5VkdXI4bZt2+Dh4YF27dphxIgRiI6OhhACoaGhmDVrFry8vKSRsdDQUDg4OKC0tBS7du1Cdc/VqNVqhIWFwdTUFGFhYVCr1Y/pqoAFCxYgJCQEp0+fxvDhwzFs2DBkZmbW2fc9ePAAmzZtwtixY6ViuiaKiooQExMDV1dXtGzZsg4SEhER1T1DTivryszMDF26dMGBAwekfeXl5Thw4AACAgIqPScgIECjPQAkJiZK7V1dXeHg4KDRpqCgACdOnJDa3L179+E1KzXTKpVKlJeX1+AKZFYcqtVqjBgxAgAQGBiI/Px8HDp0CJaWlrCysoJKpYKDgwMcHBxgaWmJbt26ITw8HK+//jrs7OwQFBSE5cuXa1XfBQUF2LFjh9T3iBEjsG3bNhQWFj6W6xo6dCjGjx+Ptm3b4v3334evry9Wr16t0SYsLAxWVlYa2+bNm/X6vt27dyMvLw9jxoyp0XmfffaZ9N3ffvstEhMTYWZmplcGIiIiY1NCYbCtJmbOnIl///vf2LBhAzIzM/HWW2+hqKgIb7zxBgBg1KhRGlPS06ZNQ0JCAlauXImsrCy89957SEtLw5QpUwA8nIWcPn06PvjgA+zZswdnzpzBqFGj4OTkJD2bEBAQgCZNmmD06NE4ffo0Ll68iNmzZ+PKlStaz3FU/3OTiQsXLiA1NRVhYWEAAJVKhdDQ0GpH+D788EPk5OQgKioKXl5eiIqKgoeHB86cOSO12bp1K9zc3ODt7Q0A8PHxgbOzM+Lj4w2Wf+nSpRqFXXZ2tnTs739TCAgI0Bo5XLVqFTIyMjS2IUOG6JVFrVYjKCgITk5ONTpv+PDhSE9Px6FDh9C2bVu89tprGusn/V1lT26VihK9MhMRERmasdY5DA0NxYoVK7Bw4UL4+PggIyMDCQkJ0gMl2dnZGvcBdu/eHVu2bMEXX3wBb29v7NixA7t370aHDh2kNnPmzMHUqVMxceJEdO3aFYWFhUhISICFhQWAh9PZCQkJKCwsxPPPPw9fX18cOXIEX331lVT/6Eo29xyq1WqUlpZqFDRCCJibm0vrBFWladOmGDp0KIYOHYqlS5eic+fOWLFiBTZs2CD1fe7cOem+RODhEG90dDTGjTPM+mOTJk3Ca6+9Jn2uaWHm4OAAd3d3jX3W1tbIy8urUT+//PIL9u/fjy+//LJG5wGQnr5q06YNunXrhiZNmmDXrl1Swf53ERER0r2gFfooQtDXxLD3UxIRET1ppkyZIo38/V1SUpLWvoo6pioKhQJLlizBkiVLqmzj6+uL7777rsZZ/04WxWFpaSk2btyIlStXYsCAARrHgoODsXXrVpiZmen05KyZmRnc3Nykp5XPnDmDtLQ0JCUlwdbWVmp3+/Zt9OnTB1lZWfDw8Kj1Ndja2mr0/1fHjx/HqFGjND537ty51t9ZmZiYGNjb29d4CPnvhBAQQlS6hlOF+fPnY+bMmRr7Vth+VavvJSIiMhTZTI8+YWRRHFY8XTNu3DjY2NhoHAsJCYFarcaMGTNw5coVZGRkoEWLFrC2tkZiYiLi4uIwbNgwtG3bFkIIfP3119i3bx9iYmIAPBw19PPzQ69evbS+t2vXrlCr1dK6h2VlZcjIyNBoY25uDk9PTxQWFuLSpUvS/oostra2aNWq1SOvb/v27fD19UXPnj2xefNmpKam1skDMeXl5YiJicHo0aM1Rkmr8/PPPyM+Ph4DBgxAs2bN8NtvvyEyMhKWlpZ48cUXqzyvsie3VApTvfMTEREZkqKG9wrSQ7IoDtVqNfr166dVGAIPi8Nly5bBy8sLgYGB6Nu3L/Ly8hATE4NevXqhQYMGmDVrFn799VeYm5ujTZs2WL9+PUaOHCk9tTt37txKvzckJAQrV67E0qVLATxcSufvI3pubm64dOkS0tLSNJbIqRgxGz16NGJjYx95fYsXL0ZcXBzefvttODo6YuvWrWjfvn1NfkQ62b9/P7KzszF27NganWdhYYHk5GR8/PHHuHPnDpo3b45evXrh2LFjsLe3N3hOIiIiki++W7mOKRQK7Nq1S+c3nTzp+G5lIm18t3Lt8N3K9Vddv1t5hHKLwfraVP66wfqSO1mMHBIREREZGqeV9cN7NZ8A2dnZWmsgVrVsTmU2b95c5bleXl6P6SqIiIjoScCRwzpmiFl7JycnrQdl/n78UYYMGQJ/f/9Kj5ma8gESIiKqnzgCph8Wh08AlUqltQZiTVhbW8Pa2tqAiYiIiOSPxaF++HMjIiIiIglHDomIiKheUgg+kKIPFodERERUL3F6VD8sDomIiKhe4rihflhUExEREZGEI4dERERUL3EETD8sDomIiKhe4htS9MOimoiIiIgkHDkkIiKieokjYPphcUhERET1EieV9cOimoiIiIgkHDkkIiKieokjYPphcUhERET1EotD/fDnRkREREQSjhwSERFRvcQHUvTD4pCIiIjqJU6P6ofFIREREdVLCg4d6oVFNRERERFJOHJIRERE9RJHwPTD4pCIiIjqJc4q64dFNRERERFJOHJIRERE9RJHwPQju59bSkoKTExMMGjQII397733Hnx8fLTanz59GkOGDIG9vT0sLCzg4uKC0NBQ3Lx5U6ttREQETExMsHz5cq1jsbGxaNy4cZW5Dh8+jMGDB8PJyQkKhQK7d++u6aVVqar+xowZg+DgYJ36cHFxgUKh0NomT56sc46UlBQ8//zzaNiwIRo1aoRevXrh3r17Op9PREQkJwoDbk8T2RWHarUaU6dOxeHDh3H9+vVHtr116xZeeOEF2Nra4rvvvkNmZiZiYmLg5OSEoqIirfbR0dGYM2cOoqOja5yrqKgI3t7eWLt2bY3PfRx++OEH3LhxQ9oSExMBAEOHDtXp/JSUFAQGBmLAgAFITU3FDz/8gClTpkCplN2/IkRERFSHZDWtXFhYiPj4eKSlpSEnJwexsbEIDw9HbGwsFi9eDODhKBsAxMTEoHHjxsjPz8f69euhUj28FFdXV/Tt21er70OHDuHevXtYsmQJNm7ciGPHjqF79+46ZwsKCkJQUFCNr0mhUOCzzz7Dnj17kJSUBEdHRyxbtgyvvvpqjft6lGbNmml8joyMhJubG3r37q3T+TNmzMD//d//Yd68edK+du3aGTQjERHR46R82ob8DERWw0Lbtm2Dh4cH2rVrhxEjRiA6OhpCCISGhmLWrFnw8vKSRsZCQ0Ph4OCA0tJS7Nq1C0KIR/atVqsRFhYGU1NThIWFQa1WP6arAhYsWICQkBCcPn0aw4cPx7Bhw5CZmVln3/fgwQNs2rQJY8eOlYrpR7l58yZOnDgBe3t7dO/eHc2bN0fv3r1x5MiROstIRERU1zitrB9ZFYdqtRojRowAAAQGBiI/Px+HDh2CpaUlrKysoFKp4ODgAAcHB1haWqJbt24IDw/H66+/Djs7OwQFBWH58uXIzc3V6LegoAA7duyQ+h4xYgS2bduGwsLCx3JdQ4cOxfjx49G2bVu8//778PX1xerVqzXahIWFwcrKSmPbvHmzXt+3e/du5OXlYcyYMTq1//nnnwE8vK9zwoQJSEhIwLPPPosXXngBP/30k14ZiIiI6Mkkm+LwwoULSE1NRVhYGABApVIhNDS02hG+Dz/8EDk5OYiKioKXlxeioqLg4eGBM2fOSG22bt0KNzc3eHt7AwB8fHzg7OyM+Ph4g+VfunSpRmGXnZ0tHQsICNBoGxAQoDVyuGrVKmRkZGhsQ4YM0SuLWq1GUFAQnJycdGpfXl4OAHjzzTfxxhtvoHPnzli1ahXatWv3yPszi4uLUVBQoLGVihK9MhMRERmaUmG47Wkim+JQrVajtLQUTk5OUKlUUKlUWLduHXbu3In8/PxHntu0aVMMHToUK1asQGZmJpycnLBixQqNvs+dOyf1q1KpcP78eb0eTKnKpEmTNAo7XQuzCg4ODnB3d9fYrK2ta5zjl19+wf79+zF+/Hidz3F0dAQAtG/fXmO/p6enRpH7dxEREbCxsdHYksu/qnFmIiKiusBpZf3I4oGU0tJSbNy4EStXrsSAAQM0jgUHB2Pr1q0wMzNDWVlZtX2ZmZnBzc1Nelr5zJkzSEtLQ1JSEmxtbaV2t2/fRp8+fZCVlQUPD49aX4Otra1G/391/PhxjBo1SuNz586da/2dlYmJiYG9vb3WUkCP4uLiAicnJ1y4cEFj/8WLFx/5EM78+fMxc+ZMjX0rbFkcEhGRPDxtI36GIovicO/evbhz5w7GjRsHGxsbjWMhISFQq9WYMWMGrly5goyMDLRo0QLW1tZITExEXFwchg0bhrZt20IIga+//hr79u1DTEwMgIejhn5+fujVq5fW93bt2hVqtVpa97CsrAwZGRkabczNzeHp6YnCwkJcunRJ2l+RxdbWFq1atXrk9W3fvh2+vr7o2bMnNm/ejNTU1Dp5IKa8vBwxMTEYPXq09PS2LhQKBWbPno1FixbB29sbPj4+2LBhA7KysrBjx44qzzM3N4e5ubnGPpXCVO/8REREZHyyKA7VajX69eunVRgCD4vDZcuWwcvLC4GBgejbty/y8vIQExODXr16oUGDBpg1axZ+/fVXmJubo02bNli/fj1GjhwpPbU7d+7cSr83JCQEK1euxNKlSwE8XErn7yN6bm5uuHTpEtLS0jSWyKkYMRs9ejRiY2MfeX2LFy9GXFwc3n77bTg6OmLr1q1aU7iGsH//fmRnZ2Ps2LE1Pnf69Om4f/8+ZsyYgdu3b8Pb2xuJiYlwc3MzeE4iIqLHQYFHr2RClVOI6taAoVpRKBTYtWuXzm86edItMY0zdoQqTSsKN3YEekql9xln7AhPNLcOl40dgepIyy8Md+9/ZT4wM9yfSe8+GGawvuRONg+kEBEREZHxsTh8AmRnZ2utgVjVsjmV2bx5c5Xnenl5PaarICIierz4tLJ+ZHHPYX1miFl7JycnrQdl/n78UYYMGQJ/f/9Kj5ma8gESIiKqn5QK3jmnDxaHTwCVSgV3d3e9z7e2ttZrzUQiIiJ6+rA4JCIionpJ8bTNBxsIi0MiIiKql1gc6ocPpBARERGRhCOHREREVC/xgRT9sDgkIiKieonTyvphcUhERET1kpLFoV54zyERERERSThySERERPWSgvcc6oXFIREREdVLvOdQP5xWJiIiIiIJRw6JiIioXuJSNvphcUhERET1EqeV9cNpZSIiIiKScOSQiIiI6iVOK+uHxSERERHVS1zKRj+cViYiIiIiCUcOiYiIqF5ScAhMLywOiYiIqF7iPYf6YXFIRERE9RLvOdQPB1yJiIiISMKRQyIiIqqXuAi2flgcEhERUb3Eew71I7tp5ZSUFJiYmGDQoEEa+9977z34+PhotT99+jSGDBkCe3t7WFhYwMXFBaGhobh586ZW24iICJiYmGD58uVax2JjY9G4ceMqcx0+fBiDBw+Gk5MTFAoFdu/eXdNLq1JV/Y0ZMwbBwcE69eHi4gKFQqG1TZ48WafzL1++jH/84x9o1qwZGjVqhNdeew25ubk1uAoiIiKqD2RXHKrVakydOhWHDx/G9evXH9n21q1beOGFF2Bra4vvvvsOmZmZiImJgZOTE4qKirTaR0dHY86cOYiOjq5xrqKiInh7e2Pt2rU1Pvdx+OGHH3Djxg1pS0xMBAAMHTq02nOLioowYMAAKBQKHDx4EEePHsWDBw8wePBglJeX13V0IiKiOqFQCINtTxNZFYeFhYWIj4/HW2+9hUGDBiE2NhbAw1G9xYsX4/Tp09KIWGxsLI4ePYr8/HysX78enTt3hqurK/r27YtVq1bB1dVVo+9Dhw7h3r17WLJkCQoKCnDs2LEaZQsKCsIHH3yAf/zjHzU6T6FQYN26dQgKCoKlpSVat26NHTt21KgPXTRr1gwODg7StnfvXri5uaF3797Vnnv06FFcvXoVsbGx6NixIzp27IgNGzYgLS0NBw8eNHhWIiKix0GhMNxWU2vXroWLiwssLCzg7++P1NTUR7bfvn07PDw8YGFhgY4dO2Lfvn0ax4UQWLhwIRwdHWFpaYl+/frhp59+0urnm2++gb+/PywtLdGkSROdZyD/SlbF4bZt2+Dh4YF27dphxIgRiI6OhhACoaGhmDVrFry8vKSRsdDQUDg4OKC0tBS7du2CEI+u6tVqNcLCwmBqaoqwsDCo1erHdFXAggULEBISgtOnT2P48OEYNmwYMjMz6+z7Hjx4gE2bNmHs2LFQ6PBvdHFxMRQKBczNzaV9FhYWUCqVOHLkSJ3lJCIiqo/i4+Mxc+ZMLFq0CKdOnYK3tzcGDhxY6S1vAHDs2DGEhYVh3LhxSE9PR3BwMIKDg3H27FmpzbJly/Dpp58iKioKJ06cQMOGDTFw4EDcv39farNz506MHDkSb7zxBk6fPo2jR4/i9ddfr3F+WRWHarUaI0aMAAAEBgYiPz8fhw4dgqWlJaysrKBSqaSRMUtLS3Tr1g3h4eF4/fXXYWdnh6CgICxfvlzrXrmCggLs2LFD6nvEiBHYtm0bCgsLH8t1DR06FOPHj0fbtm3x/vvvw9fXF6tXr9ZoExYWBisrK41t8+bNen3f7t27kZeXhzFjxujUvlu3bmjYsCHmzp2Lu3fvoqioCO+88w7Kyspw48YNvTIQEREZm1IhDLbVxEcffYQJEybgjTfeQPv27REVFYUGDRpUeVvbJ598gsDAQMyePRuenp54//338eyzz2LNmjUAHo4afvzxx3j33Xfx8ssvo1OnTti4cSOuX78uPbNQWlqKadOmYfny5Zg0aRLatm2L9u3b47XXXqv5z63GZ9SRCxcuIDU1FWFhYQAAlUqF0NDQakf4PvzwQ+Tk5CAqKgpeXl6IioqCh4cHzpw5I7XZunUr3Nzc4O3tDQDw8fGBs7Mz4uPjDZZ/6dKlGoVddna2dCwgIECjbUBAgNbI4apVq5CRkaGxDRkyRK8sarUaQUFBcHJy0ql9s2bNsH37dnz99dewsrKCjY0N8vLy8Oyzz0KprPpfkeLiYhQUFGhspaJEr8xERESGZsh7Div7M6+4uFjrOx88eICTJ0+iX79+0j6lUol+/fohJSWl0pwpKSka7QFg4MCBUvsrV64gJydHo42NjQ38/f2lNqdOncK1a9egVCrRuXNnODo6IigoSGP0UVeyKQ7VajVKS0vh5OQElUoFlUqFdevWYefOncjPz3/kuU2bNsXQoUOxYsUKZGZmwsnJCStWrNDo+9y5c1K/KpUK58+f1+vBlKpMmjRJo7DTtTCr4ODgAHd3d43N2tq6xjl++eUX7N+/H+PHj6/ReQMGDMDly5dx8+ZN/P777/jPf/6Da9euoXXr1lWeExERARsbG40tufyrGmcmIiKSu8r+zIuIiNBq9/vvv6OsrAzNmzfX2N+8eXPk5ORU2ndOTs4j21f881Ftfv75ZwAPV3d59913sXfvXjRp0gR9+vTB7du3a3StsljnsLS0FBs3bsTKlSsxYMAAjWPBwcHYunUrzMzMUFZWVm1fZmZmcHNzk55WPnPmDNLS0pCUlARbW1up3e3bt9GnTx9kZWXBw8Oj1tdga2ur0f9fHT9+HKNGjdL43Llz51p/Z2ViYmJgb2+vtRSQruzs7AAABw8exM2bNx85ejl//nzMnDlTY98KWxaHREQkDwql4Z4yruzPvL/eq29sFauL/POf/0RISAiAhzVBixYtsH37drz55ps69yWL4nDv3r24c+cOxo0bBxsbG41jISEhUKvVmDFjBq5cuYKMjAy0aNEC1tbWSExMRFxcHIYNG4a2bdtCCIGvv/4a+/btQ0xMDICHo4Z+fn7o1auX1vd27doVarVaWvewrKwMGRkZGm3Mzc3h6emJwsJCXLp0SdpfkcXW1hatWrV65PVt374dvr6+6NmzJzZv3ozU1NQ6eSCmvLwcMTExGD16NFSqmv1fGxMTA09PTzRr1gwpKSmYNm0aZsyYgXbt2lV5jrm5udZvDJXCVK/sREREhmbIN6RU9mdeZezs7GBiYqL1/ENubi4cHBwqPcfBweGR7Sv+mZubC0dHR402FWtAV+xv3769RubWrVtr3OqmC1lMK6vVavTr10+rMAQeFodpaWnw8vJCYGAg+vbti2bNmmHr1q1o3749GjRogFmzZsHHxwfdunXDtm3bsH79eowcOVJ6areigq6s740bN6Kk5OF9coWFhejcubPGNnjwYABAWlqatA8AZs6cic6dO2PhwoXVXt/ixYsRFxcn3UBakd3Q9u/fj+zsbIwdO7bG5164cAHBwcHw9PTEkiVL8M9//lNjap6IiOhJY4wHUszMzNClSxccOHBA2ldeXo4DBw5oPYNQISAgQKM9ACQmJkrtXV1d4eDgoNGmoKAAJ06ckNp06dIF5ubmuHDhgtSmpKQEV69ehbOzs875AUAhqlsDhmpFoVBg165deq0z9CRaYhpn7AhVmlYUbuwI9JRK7zPO2BGeaG4dLhs7AtWRll8Y7t7/ymxx+cJgfb1+daLObePj4zF69Gh8/vnn8PPzw8cff4xt27YhKysLzZs3x6hRo/DMM89I9yweO3YMvXv3RmRkJAYNGoS4uDgsXboUp06dQocOHQAA//rXvxAZGYkNGzbA1dUVCxYswI8//ojz58/DwsICADB9+nTs2LED0dHRcHZ2xvLly/H1118jKysLTZo00Tm/LKaViYiIiAzNWG82CQ0Nxa1bt7Bw4ULk5OTAx8cHCQkJ0gMl2dnZGquBdO/eHVu2bMG7776L8PBwtGnTBrt375YKQwCYM2cOioqKMHHiROTl5aFnz55ISEiQCkMAWL58OVQqFUaOHIl79+7B398fBw8erFFhCHDksM4ZYuQwOzv7kdPQ58+ff+R9j5s3b67yRlRnZ2ecO3dO72x/x5FDIm0cOawdjhzWX3U9chjXOspgfQ37eZLB+pI7jhzWMUPU3k5OTloPyvz9+KMMGTIE/v7+lR4zNeUDJERERPQ/LA6fACqVCu7u7nqfb21trdeaiURERE8yY00rP+lYHBIREVG9VNPX3tFDsljKhoiIiIjkgSOHREREVC9xWlk/LA6JiIioXlJwflQv/LERERERkYQjh0RERFQvcVpZPywOiYiIqF5icagfFodERERUL3EpG/3wnkMiIiIikhi0OCwtLUV2drYhuyQiIiLSi0IhDLY9TQxaHJ47dw6urq6G7JKIiIhILywO9cNpZSIiIiKS1OiBlGefffaRx+/du1erMERERESG8rSN+BlKjYrD8+fPY9iwYVVOHd+4cQMXL140SDAiIiKi2lAoWRzqo0bFYYcOHeDv74+33nqr0uMZGRn497//bZBgRERERPT41ag47NGjBy5cuFDlcWtra/Tq1avWoYiIiIhqi9PK+qlRcfjJJ5888ribmxu+//77WgUiIiIiMgQWh/rh08pEREREJNG7OExOTsaIESMQEBCAa9euAQD+85//4MiRIwYLR0RERKQvrnOoH72Kw507d2LgwIGwtLREeno6iouLAQD5+flYunSpQQMSERER6YPFoX70Kg4/+OADREVF4d///jdMTU2l/T169MCpU6cMFo6IiIhIX0qFMNj2NNGrOLxw4UKlTyXb2NggLy+vtpmIiIiIyEj0Kg4dHBxw6dIlrf1HjhxB69ataxUoJSUFJiYmGDRokMb+9957Dz4+PlrtT58+jSFDhsDe3h4WFhZwcXFBaGgobt68qdU2IiICJiYmWL58udax2NhYNG7cuMpchw8fxuDBg+Hk5ASFQoHdu3fX9NKqVFV/Y8aMQXBwsE59lJWVYcGCBXB1dYWlpSXc3Nzw/vvvQ4jq/7ZTUlKCuXPnomPHjmjYsCGcnJwwatQoXL9+vYZXQkREJB8KheG2p4lexeGECRMwbdo0nDhxAgqFAtevX8fmzZvxzjvvVLlAtq7UajWmTp2Kw4cPV1uc3Lp1Cy+88AJsbW3x3XffITMzEzExMXByckJRUZFW++joaMyZMwfR0dE1zlVUVARvb2+sXbu2xuc+Dv/617+wbt06rFmzBpmZmfjXv/6FZcuWYfXq1dWee/fuXZw6dQoLFizAqVOn8OWXX+LChQsYMmTIY0hORERUN3jPoX5qtM5hhXnz5qG8vBwvvPAC7t69i169esHc3BzvvPMOpk6dqneYwsJCxMfHIy0tDTk5OYiNjUV4eDhiY2OxePFiAA9H2QAgJiYGjRs3Rn5+PtavXw+V6uGluLq6om/fvlp9Hzp0CPfu3cOSJUuwceNGHDt2DN27d9c5W1BQEIKCgmp8TQqFAp999hn27NmDpKQkODo6YtmyZXj11Vdr3NejHDt2DC+//LI04uri4oKtW7ciNTW12nNtbGyQmJiosW/NmjXw8/NDdnY2WrVqZdCsREREJF96jRwqFAr885//xO3bt3H27FkcP34ct27dwvvvv1+rMNu2bYOHhwfatWuHESNGIDo6GkIIhIaGYtasWfDy8sKNGzdw48YNhIaGwsHBAaWlpdi1a1e106dqtRphYWEwNTVFWFgY1Gp1rbLWxIIFCxASEoLTp09j+PDhGDZsGDIzMw36Hd27d8eBAwekd1ufPn0aR44c0augBR4+ea5QKB451U5ERCRnCqUw2PY0qdUi2GZmZmjfvj38/PxgZWVV6zBqtRojRowAAAQGBiI/Px+HDh2CpaUlrKysoFKp4ODgAAcHB1haWqJbt24IDw/H66+/Djs7OwQFBWH58uXIzc3V6LegoAA7duyQ+h4xYgS2bduGwsLCWmfWxdChQzF+/Hi0bdsW77//Pnx9fbWme8PCwmBlZaWxbd68WefvmDdvHoYNGwYPDw+Ympqic+fOmD59OoYPH17jvPfv38fcuXMRFhaGRo0a1fh8IiIiOeC0sn70Kg7v37+P5cuX48UXX4Svry+effZZjU0fFy5cQGpqKsLCwgAAKpUKoaGh1Y7wffjhh8jJyUFUVBS8vLwQFRUFDw8PnDlzRmqzdetWuLm5wdvbGwDg4+MDZ2dnxMfH65W1MkuXLtUo7LKzs6VjAQEBGm0DAgK0Rg5XrVqFjIwMja0m9/xt27YNmzdvxpYtW3Dq1Cls2LABK1aswIYNG2p0HSUlJXjttdcghMC6dese2ba4uBgFBQUaW6koqdH3ERERkbzodc/huHHj8N///hevvvoq/Pz8pPsAa0OtVqO0tBROTk7SPiEEzM3NsWbNmkee27RpUwwdOhRDhw7F0qVL0blzZ43CSK1W49y5c9J9iQBQXl6O6OhojBs3rtbZAWDSpEl47bXXpM9/vQ5dODg4wN3dXWOftbW1zksDzZ49Wxo9BICOHTvil19+QUREBEaPHq1THxWF4S+//IKDBw9WO2oYEREh3QtaoY8iBH1NDHs/JRERkT6ethE/Q9GrONy7dy/27duHHj16GCREaWkpNm7ciJUrV2LAgAEax4KDg7F161aYmZmhrKys2r7MzMzg5uYmPa185swZpKWlISkpCba2tlK727dvo0+fPsjKyoKHh0etr8HW1laj/786fvw4Ro0apfG5c+fOtf7Ov7p79y6USs2BYBMTE5SXl+t0fkVh+NNPP+H7779H06ZNqz1n/vz5mDlzpsa+FbZf6R6aiIioDrE41I9exeEzzzwDa2trg4XYu3cv7ty5g3HjxsHGxkbjWEhICNRqNWbMmIErV64gIyMDLVq0gLW1NRITExEXF4dhw4ahbdu2EELg66+/xr59+xATEwPg4aihn59fpYt2d+3aFWq1Wlr3sKysDBkZGRptzM3N4enpicLCQo21HSuy2NraVvs07/bt2+Hr64uePXti8+bNSE1NNfgDMYMHD8aHH36IVq1awcvLC+np6fjoo48wduzYas8tKSnBq6++ilOnTmHv3r0oKytDTk4OgIdFr5mZWaXnmZubw9zcXGOfSmFaaVsiIiJ6MuhVHK5cuRJz585FVFQUnJ2dax1CrVajX79+WoUh8LA4XLZsGby8vBAYGIi+ffsiLy8PMTEx6NWrFxo0aIBZs2bh119/hbm5Odq0aYP169dj5MiRePDgATZt2oS5c+dW+r0hISFYuXKl9D7owsJCrRE9Nzc3XLp0CWlpaRpL5FSMmI0ePRqxsbGPvL7FixcjLi4Ob7/9NhwdHbF161a0b9++Jj+iaq1evRoLFizA22+/jZs3b8LJyQlvvvkmFi5cWO25165dw549ewBAa6Hx77//Hn369DFoViIioseBI4f6UQhdXqHxN7du3cJrr72Gw4cPo0GDBhrvVwYeTtnSQwqFArt27dL5TSdPuiWmccaOUKVpReHGjkBPqfQ+hrm3+Wnl1uGysSNQHWn5Rc1fSlETyc8tMVhfzyVXP9hSX+g1chgWFoZr165h6dKlaN68uUEeSCEiIiIyJI4c6kev4vDYsWNISUmRloahupWdnf3Iaejz588/8r7H5OTkRy6G/bjWeyQiIiL506s49PDwwL179wydpV7SY9Zei5OTk9aDMn8//ii+vr6PPJ+IiKg+4sihfvQqDiMjIzFr1ix8+OGH6Nixo9Y9h3yrhmGpVCqtNRBrwtLSslbnExERPYmettfeGYpexWFgYCAA4IUXXtDYL4SAQqHQaT1CIiIiIpIfvYrD77//3tA5iIiIiAyK08r60as47N27t6FzEBERERkUi0P96FUcVrh79y6ys7Px4MEDjf2dOnWqVSgiIiIiMg69isNbt27hjTfewLffflvpcd5zSERERMbGkUP9KPU5afr06cjLy8OJEydgaWmJhIQEbNiwAW3atJFew0ZERERkTAqFMNj2NNFr5PDgwYP46quv4OvrC6VSCWdnZ/Tv3x+NGjVCREQEBg0aZOicRERERPQY6DVyWFRUBHt7ewBAkyZNcOvWLQBAx44dcerUKcOlIyIiItKTQmG47WmiV3HYrl07XLhwAQDg7e2Nzz//HNeuXUNUVBQcHR0NGpCIiIhIH5xW1o9e08rTpk3DjRs3AACLFi1CYGAgNm/eDDMzM8TGxhoyHz1hnvW4YewIVTroPcPYEegp9fzpVcaO8ETb4z7f2BGojoys4/6ftqLOUPQqDkeMGCH9ukuXLvjll1+QlZWFVq1awc7OzmDhiIiIiOjxqtU6hxUaNGiAZ5991hBdERERERkG362sF72Kw7KyMsTGxuLAgQO4efMmysvLNY4fPHjQIOGIiIiI9MVpZf3ofc9hbGwsBg0ahA4dOkDxtD3GQ0RERFRP6VUcxsXFYdu2bXjxxRcNnYeIiIjIIDhyqB+9ikMzMzO4u7sbOgsRERGRwbA41I9e6xzOmjULn3zyCYTgD52IiIioPtFr5PDIkSP4/vvv8e2338LLywumpqYax7/88kuDhCMiIiLSF0cO9aNXcdi4cWP84x//MHQWIiIiIoNhcagfvYrDmJgYQ+cgIiIiIhnQ655DACgtLcX+/fvx+eef488//wQAXL9+HYWFhQYLR0RERKQvvltZP3qNHP7yyy8IDAxEdnY2iouL0b9/f1hbW+Nf//oXiouLERUVZeicRERERDWi0HsI7Omm149t2rRp8PX1xZ07d2BpaSnt/8c//oEDBw7UqK9bt27hrbfeQqtWrWBubg4HBwcMHDgQR48erfbcpKQkKBQK5OXlaR1zcXHBxx9/rLU/IiICJiYmWL58udax2NhYNG7cuMrvi4iIQNeuXWFtbQ17e3sEBwfjwoUL1easztWrV6FQKJCRkaF1rE+fPpg+fbpO/SgUikq3yq61Mrdv38bw4cPRqFEjNG7cGOPGjeNIMBERPbE4cqgfvYrD5ORkvPvuuzAzM9PY7+LigmvXrtWor5CQEKSnp2PDhg24ePEi9uzZgz59+uCPP/7QJ1q1oqOjMWfOHERHR9f43EOHDmHy5Mk4fvw4EhMTUVJSggEDBqCoqKgOktbcjRs3NLbo6GgoFAqEhITodP7w4cNx7tw5JCYmYu/evTh8+DAmTpxYx6mJiIhITvQqDsvLy1FWVqa1/7fffoO1tbXO/eTl5SE5ORn/+te/0LdvXzg7O8PPzw/z58/HkCFDpDZvvvkmmjdvDgsLC3To0AF79+7VJzYOHTqEe/fuYcmSJSgoKMCxY8dqdH5CQgLGjBkDLy8veHt7IzY2FtnZ2Th58mSV51SMCsbFxaF79+7SNRw6dEiva3gUBwcHje2rr75C37590bp162rPzczMREJCAtavXw9/f3/07NkTq1evRlxcHK5fv27wrERERHWNI4f60as4HDBggMaUrUKhQGFhIRYtWlSjV+pZWVnBysoKu3fvRnFxsdbx8vJyBAUF4ejRo9i0aRPOnz+PyMhImJiY6BMbarUaYWFhMDU1RVhYGNRqtV79VMjPzwcA2NraVtt29uzZmDVrFtLT0xEQEIDBgwfX2egoAOTm5uKbb77BuHHjdGqfkpKCxo0bw9fXV9rXr18/KJVKnDhxoq5iEhER1R2FMNz2FNHrgZSVK1di4MCBaN++Pe7fv4/XX38dP/30E+zs7LB161bdv1ylQmxsLCZMmICoqCg8++yz6N27N4YNG4ZOnTph//79SE1NRWZmJtq2bQsAlY6CtWjRQmvf3bt3NT4XFBRgx44dSElJAQCMGDECzz33HD755BNYWVnV5PIBPCxcp0+fjh49eqBDhw7Vtp8yZYo0vbtu3TokJCRArVZjzpw5Upvu3btDqdSs1+/duwcfH58a59uwYQOsra3xyiuv6NQ+JycH9vb2GvtUKhVsbW2Rk5NT4+8nIiKiJ5NexWGLFi1w+vRpxMfH4/Tp0ygsLMS4ceMwfPhwjQdUdBESEoJBgwYhOTkZx48fx7fffotly5Zh/fr1uHnzJlq0aCEVhlVJTk7Wms7u06ePxuetW7fCzc0N3t7eAAAfHx84OzsjPj5e59G1v5o8eTLOnj2LI0eOSPsmTZqETZs2SZ//+jBHQECA9GuVSgVfX19kZmZq9BkfHw9PT0+NfcOHD69xNuDhvZXDhw+HhYWFXufrori4WGvEt6S8FKZKvf61IiIiMqinbTrYUPT+U1ylUmH48OF6Fy9/ZWFhgf79+6N///5YsGABxo8fj0WLFuGdd97R6XxXV1etp4xVKs1LU6vVOHfunMb+8vJyREdH17g4nDJlivTAxl9HLZcsWaJz5sq0bNkS7u7uGvtqWmwDD4vlCxcuID4+XudzHBwccPPmTY19paWluH37NhwcHCo9JyIiAosXL9bYF9ZsAF5vHljjzERERIbG4lA/et1zuGHDBnzzzTfS5zlz5qBx48bo3r07fvnll1qHat++PYqKitCpUyf89ttvuHjxYq36O3PmDNLS0pCUlISMjAxpS0pKQkpKCrKysnTqRwiBKVOmYNeuXTh48CBcXV01jtvb28Pd3V3a/ur48ePSr0tLS3Hy5EmtUUJDUavV6NKlizRKqouAgADk5eVpPFxz8OBBlJeXw9/fv9Jz5s+fj/z8fI1taLN+tc5PRERExqNXcbh06VJpRCslJQVr1qzBsmXLYGdnhxkzZujczx9//IHnn38emzZtwo8//ogrV65g+/btWLZsGV5++WX07t0bvXr1QkhICBITE3HlyhV8++23SEhIqFFetVoNPz8/9OrVCx06dJC2Xr16oWvXrhoPppSVlWkUkBkZGdL07+TJk7Fp0yZs2bIF1tbWyMnJQU5ODu7du1dthrVr12LXrl3IysrC5MmTcefOHYwdO7ZG16GLgoICbN++HePHj6/ReZ6enggMDMSECROQmpqKo0ePYsqUKRg2bBicnJwqPcfc3ByNGjXS2DilTEREcmHMp5XXrl0LFxcXWFhYwN/fH6mpqY9sv337dnh4eMDCwgIdO3bEvn37NI4LIbBw4UI4OjrC0tIS/fr1w08//VRpX8XFxfDx8alyDeXq6FUc/vrrr9LI2O7du/Hqq69i4sSJiIiIQHJyss79WFlZwd/fH6tWrZIKtwULFmDChAlYs2YNAGDnzp3o2rUrwsLC0L59e8yZM6fSZXSq8uDBA2zatKnKtf5CQkKwceNGlJSUAHh4n2Dnzp01tsGDBwN4+CBJfn4++vTpA0dHR2nTZfo2MjISkZGR8Pb2xpEjR7Bnzx7Y2dnpfB26iouLgxACYWFhNT538+bN8PDwwAsvvIAXX3wRPXv2xBdffGHwjERERI+DsYrD+Ph4zJw5E4sWLcKpU6fg7e2NgQMHat2+VeHYsWMICwvDuHHjkJ6ejuDgYAQHB+Ps2bNSm2XLluHTTz9FVFQUTpw4gYYNG2LgwIG4f/++Vn9z5sypcmBHFwohRI3LYXt7e3z33XdS8TRz5kyMHDkSly9fhre3N9+q8RdXr16Fq6sr0tPT9Xrq+Emzt+MqY0eoUkkpRzXJOJ4/Ld/fF0+CPe7zjR2B6sjI7Al12v/FsMkG66vt1rU6t/X390fXrl2lga7y8nK0bNkSU6dOxbx587Tah4aGoqioSGMd527dusHHxwdRUVEQQsDJyQmzZs2Snm3Iz89H8+bNERsbi2HDhknnffvtt5g5cyZ27twJLy8vveoPvUYO+/fvj/Hjx2P8+PG4ePGitLbhuXPn4OLiok+XRERERAalUBpuKy4uRkFBgcZW2RrNDx48wMmTJ9Gv3//uwVcqlejXr5+0nN7fpaSkaLQHgIEDB0rtr1y5gpycHI02NjY28Pf31+gzNzcXEyZMwH/+8x80aNBA75+bXsXh2rVrERAQgFu3bmHnzp1o2rQpAODkyZN6TWdS9ZKTk6VFwyvbqrN06dIqzw0KCnoMV0BERPSYGXAR7IiICNjY2GhsERERWl/5+++/o6ysDM2bN9fY37x58yrXDc7JyXlk+4p/PqqNEAJjxozBpEmTNF5ooQ+95tkaN24sDZX+1d+XNaGH75vWY+Zei6+vr143lVaYNGkSXnvttUqP6bNcDhERkdwZcimb+fPnY+bMmRr7zM3NDdZ/ba1evRp//vkn5s+v/W0YtboJ6+7du8jOzsaDBw809nfq1KlWoUibpaWl1vI4NWFra6vTa/6IiIhIm7m5uU7FoJ2dHUxMTJCbm6uxPzc3t8p1gx0cHB7ZvuKfubm5cHR01GhTcT/hwYMHkZKSopXR19cXw4cPx4YNG6rNXkGvaeVbt25h0KBBsLa2hpeXl9bTvURERETGZoynlc3MzNClSxccOHBA2ldeXo4DBw5ovC3trwICAjTaA0BiYqLU3tXVFQ4ODhptCgoKcOLECanNp59+itOnT0vL8FUshRMfH48PP/xQ5/yAniOH06dPR35+Pk6cOIE+ffpg165dyM3NxQcffICVK1fq0yURERGRQRnrDSkzZ87E6NGj4evrCz8/P3z88ccoKirCG2+8AQAYNWoUnnnmGemexWnTpqF3795YuXIlBg0ahLi4OKSlpUnLySkUCkyfPh0ffPAB2rRpA1dXVyxYsABOTk4IDg4GALRq1UojQ8XzCG5ubhpvc9OFXsXhwYMH8dVXX8HX1xdKpRLOzs7o378/GjVqhIiICAwaNEifbomIiIieeKGhobh16xYWLlyInJwc+Pj4ICEhQXqgJDs7G0rl/yZvu3fvji1btuDdd99FeHg42rRpg927d6NDhw5Smzlz5qCoqAgTJ05EXl4eevbsiYSEBFhYWBg8v17rHDZq1Ag//vgjXFxc4OzsjC1btqBHjx64cuUKvLy8cPfuXYMHpScD1zkk0sZ1DmuH6xzWX3W9zuGVMRMN1pdr7NPzUgi97jls164dLly4AADw9vbG559/jmvXriEqKkrjRkkiIiIio1EKw21PEb2GUqZNm4YbN24AABYtWoTAwEBs2rQJZmZmNXoahoiIiIjkRa/icMSIEdKvn332Wfzyyy/IyspCq1at6uR9wUREREQ1ZawHUp50ek0rA4BarUaHDh1gYWGBJk2aYNSoUdi9e7cBoxERERHpT6Ew3PY00WvkcOHChfjoo48wdepUaX2dlJQUzJgxA9nZ2ViyZIlBQxIRERHR46FXcbhu3Tr8+9//1niP8pAhQ9CpUydMnTqVxSEREREZHaeV9aNXcVhSUlLpS527dOmC0tLSWociIiIiqi0Wh/rR657DkSNHYt26dVr7v/jiCwwfPrzWoYiIiIhqTSEMtz1FdB45nDlzpvRrhUKB9evX47///S+6desGADhx4gSys7MxatQow6ckIiIiosdC5+IwPT1d43OXLl0AAJcvXwYA2NnZwc7ODufOnTNgPCIiIiL9KJ6yxasNRefi8Pvvv6/LHEREREQGxXsO9aP3OodEREREVP/o9bQyERERkdxx5FA/LA6JiIiofnrK3mxiKJxWJiIiIiIJRw6JiIioXuK0sn5YHBIREVG9xKVs9MNpZSIiIiKSGL04vHXrFt566y20atUK5ubmcHBwwMCBA3H06NFqz01KSoJCoUBeXp7WMRcXF3z88cda+yMiImBiYoLly5drHYuNjUXjxo2r/L6IiAh07doV1tbWsLe3R3BwMC5cuFBtzupcvXoVCoUCGRkZWsf69OmD6dOn69SPQqGodKvsWiszZMgQtGrVChYWFnB0dMTIkSNx/fr1GlwJERGRjPD1eXoxenEYEhKC9PR0bNiwARcvXsSePXvQp08f/PHHH3XyfdHR0ZgzZw6io6NrfO6hQ4cwefJkHD9+HImJiSgpKcGAAQNQVFRUB0lr7saNGxpbdHQ0FAoFQkJCdDq/b9++2LZtGy5cuICdO3fi8uXLePXVV+s4NRERUd1QKITBtqeJUe85zMvLQ3JyMpKSktC7d28AgLOzM/z8/DTazJ07F7t370Z+fj7c3d0RGRmJl156qcbfd+jQIdy7dw9LlizBxo0bcezYMXTv3l3n8xMSEjQ+x8bGwt7eHidPnkSvXr0qPefq1atwdXXF1q1b8emnn+LUqVNwd3fH2rVrpWs2FAcHB43PX331Ffr27YvWrVvrdP6MGTOkXzs7O2PevHkIDg5GSUkJTE1NDZqViIiorj1tRZ2hGHXk0MrKClZWVti9ezeKi4u1jpeXlyMoKAhHjx7Fpk2bcP78eURGRsLExESv71Or1QgLC4OpqSnCwsKgVqtrlT8/Px8AYGtrW23b2bNnY9asWUhPT0dAQAAGDx5cZ6OjAJCbm4tvvvkG48aN0+v827dvY/PmzejevTsLQyIioqeIUYtDlUqF2NhYbNiwAY0bN0aPHj0QHh6OH3/8EQCwf/9+pKam4ssvv0T//v3RunVrvPTSSwgKCtLop0WLFlKhWbFlZ2drtCkoKMCOHTswYsQIAMCIESOwbds2FBYW6pW9vLwc06dPR48ePdChQ4dq20+ZMgUhISHw9PTEunXrYGNjo1Wcdu/eXes6kpOT9cq3YcMGWFtb45VXXqnReXPnzkXDhg3RtGlTZGdn46uvvtLr+4mIiIyO9xzqRRb3HF6/fh179uxBYGAgkpKS8OyzzyI2NhYZGRlo0aIF2rZt+8g+kpOTkZGRobE5OTlptNm6dSvc3Nzg7e0NAPDx8YGzszPi4+P1yj158mScPXsWcXFx0r5JkyZpFHZ/FRAQIP1apVLB19cXmZmZGm3i4+O1rsPX11evfNHR0Rg+fDgsLCxqdN7s2bORnp6O//73vzAxMcGoUaMgROW/KYqLi1FQUKCxlZSX6pWXiIjI0BRKw21PE1msc2hhYYH+/fujf//+WLBgAcaPH49FixbhnXfe0el8V1dXraeMVSrNS1Or1Th37pzG/vLyckRHR9d46nXKlCnYu3cvDh8+jBYtWkj7lyxZonPmyrRs2RLu7u4a+ywtLWvcT3JyMi5cuKBX4WtnZwc7Ozu0bdsWnp6eaNmyJY4fP65R3FaIiIjA4sWLNfaFNRuA15sH1vh7iYiISB5kWQu3b98eRUVF6NSpE3777TdcvHixVv2dOXMGaWlpSEpK0hiVS0pKQkpKCrKysnTqRwiBKVOmYNeuXTh48CBcXV01jtvb28Pd3V3a/ur48ePSr0tLS3Hy5El4enrW6rqqolar0aVLF2mUVF/l5eUAUOn9oAAwf/585Ofna2xDm/Wr1XcSEREZCp9W1o9RRw7/+OMPDB06FGPHjkWnTp1gbW2NtLQ0LFu2DC+//DJ69+6NXr16ISQkBB999BHc3d2RlZUFhUKBwEDdR6fUajX8/PwqfaK4a9euUKvV0lqAZWVlWusNmpubw9PTE5MnT8aWLVvw1VdfwdraGjk5OQAAGxubakf41q5dizZt2sDT0xOrVq3CnTt3MHbsWJ2vQVcFBQXYvn07Vq5cWaPzTpw4gR9++AE9e/ZEkyZNcPnyZSxYsABubm6VjhoCD38u5ubmGvtMlbIYjCYiInrq7hU0FKP+SW5lZQV/f3+sWrUKly9fRklJCVq2bIkJEyYgPDwcALBz50688847CAsLQ1FRkbSUja4ePHiATZs2Ye7cuZUeDwkJwcqVK7F06VIAQGFhITp37qzRxs3NDZcuXcK6desAPFyY+q9iYmIwZsyYR+aIjIxEZGQkMjIy4O7ujj179sDOzk7n69BVXFwchBAICwur0XkNGjTAl19+iUWLFqGoqAiOjo4IDAzEu+++q1UAEhERUf2lEFU9bUAGUbHOYXp6Onx8fIwdp87t7bjK2BGqVFLKUU0yjudPy/f3xZNgj/t8Y0egOjIye0Kd9v/HvKEG66tp5HaD9SV3/NOSiIiI6iWFkuNf+pDlAymkLTk5WWsNxKqWzanM0qVLqzz37+tGEhER0dOLI4d1zMXFpcp1AmvC19dX60GZmpg0aRJee+21So/ps1wOERGR7PGBFL2wOHxCWFpaai2PUxO2trY6veaPiIiovnjalqAxFBaHREREVC8pFMZO8GTiPYdEREREJOHIIREREdVPfFpZLywOiYiIqF7iPYf64bQyEREREUk4ckhERET1E0cO9cLikIiIiOolTivrh9PKRERERCThyCERERHVT3xaWS8sDsmgXr9UbOwIVUofesHYEegpleA529gRnmhDLkUYOwLVmQl12junlfXD4pCIiIjqJb4hRT+855CIiIiIJBw5JCIiovqJ9xzqhcUhERER1Uu851A/nFYmIiIiIglHDomIiKh+4sihXlgcEhERUb2k4D2HeuG0MhERERFJOHJIRERE9ROnlfXC4pCIiIjqJT6trB9OKxMRERGRhCOHREREVD/xgRS9GHXk8NatW3jrrbfQqlUrmJubw8HBAQMHDsTRo0erPTcpKQkKhQJ5eXlax1xcXPDxxx9r7Y+IiICJiQmWL1+udSw2NhaNGzeu8vsiIiLQtWtXWFtbw97eHsHBwbhw4UK1Oatz9epVKBQKZGRkaB3r06cPpk+frlM/CoWi0q2ya63KN998A39/f1haWqJJkyYIDg7W+VwiIiK5USgMtz1NjFochoSEID09HRs2bMDFixexZ88e9OnTB3/88UedfF90dDTmzJmD6OjoGp976NAhTJ48GcePH0diYiJKSkowYMAAFBUV1UHSmrtx44bGFh0dDYVCgZCQEJ3O37lzJ0aOHIk33ngDp0+fxtGjR/H666/XcWoiIqI6pBCG254iRptWzsvLQ3JyMpKSktC7d28AgLOzM/z8/DTazJ07F7t370Z+fj7c3d0RGRmJl156qcbfd+jQIdy7dw9LlizBxo0bcezYMXTv3l3n8xMSEjQ+x8bGwt7eHidPnkSvXr0qPefq1atwdXXF1q1b8emnn+LUqVNwd3fH2rVrpWs2FAcHB43PX331Ffr27YvWrVtXe25paSmmTZuG5cuXY9y4cdL+9u3bGzQjERERyZ/RRg6trKxgZWWF3bt3o7i4WOt4eXk5goKCcPToUWzatAnnz59HZGQkTExM9Po+tVqNsLAwmJqaIiwsDGq1ulb58/PzAQC2trbVtp09ezZmzZqF9PR0BAQEYPDgwXU2OgoAubm5+OabbzQKvUc5deoUrl27BqVSic6dO8PR0RFBQUE4e/ZsnWUkIiKqc0phuO0pYrTiUKVSITY2Fhs2bEDjxo3Ro0cPhIeH48cffwQA7N+/H6mpqfjyyy/Rv39/tG7dGi+99BKCgoI0+mnRooVUaFZs2dnZGm0KCgqwY8cOjBgxAgAwYsQIbNu2DYWFhXplLy8vx/Tp09GjRw906NCh2vZTpkxBSEgIPD09sW7dOtjY2GgVp927d9e6juTkZL3ybdiwAdbW1njllVd0av/zzz8DAN577z28++672Lt3L5o0aYI+ffrg9u3bemUgIiIyNoVCGGx7mhj1aeWQkBAMGjQIycnJOH78OL799lssW7YM69evx82bN9GiRQu0bdv2kX0kJyfD2tpaY1+fPn00Pm/duhVubm7w9vYGAPj4+MDZ2Rnx8fE6j6791eTJk3H27FkcOXJE2jdp0iRs2rRJ+vzXwjMgIED6tUqlgq+vLzIzMzX6jI+Ph6enp8a+4cOH1zgb8PDeyuHDh8PCwkKn9uXl5QCAf/7zn9I9ijExMWjRogW2b9+ON998s9LziouLtUZ9hSiFQsGH4ImIiJ5URv9T3MLCAv3790f//v2xYMECjB8/HosWLcI777yj0/murq5aTxmrVJqXpVarce7cOY395eXliI6OrnFxOGXKFOzduxeHDx9GixYtpP1LlizROXNlWrZsCXd3d419lpaWNe4nOTkZFy5cQHx8vM7nODo6AtC8x9Dc3BytW7fWGoX9q4iICCxevFhjn5nJCzA37VfD1ERERHXgKRvxMxTZLYLdvn17FBUVoVOnTvjtt99w8eLFWvV35swZpKWlISkpCRkZGdKWlJSElJQUZGVl6dSPEAJTpkzBrl27cPDgQbi6umoct7e3h7u7u7T91fHjx6Vfl5aW4uTJk1qjhIaiVqvRpUsXaZRUF126dIG5ubnG0jwlJSW4evUqnJ2dqzxv/vz5yM/P19jMVH1qE5+IiMhgFEphsO1pYrSRwz/++ANDhw7F2LFj0alTJ1hbWyMtLQ3Lli3Dyy+/jN69e6NXr14ICQnBRx99BHd3d2RlZUGhUCAwMFDn71Gr1fDz86v0ieKuXbtCrVZLawGWlZVprTdobm4OT09PTJ48GVu2bMFXX30Fa2tr5OTkAABsbGyqHeFbu3Yt2rRpA09PT6xatQp37tzB2LFjdb4GXRUUFGD79u1YuXJljc5r1KgRJk2ahEWLFqFly5ZwdnaWfiZDhw6t8jxzc3OYm5tr7OOUMhER0ZPNaH+SW1lZwd/fH6tWrcLly5dRUlKCli1bYsKECQgPDwfwcO29d955B2FhYSgqKpKWstHVgwcPsGnTJsydO7fS4yEhIVi5ciWWLl0K4OF9gp07d9Zo4+bmhkuXLmHdunUAtO9njImJwZgxYx6ZIzIyEpGRkcjIyIC7uzv27NkDOzs7na9DV3FxcRBCICwsrMbnLl++HCqVCiNHjsS9e/fg7++PgwcPokmTJgbPSURE9FhwWlkvCiEEf3J1pGKdw/T0dPj4+Bg7zmPRyFL34v1xSx961dgR6CmVdlT32zxIW2Cm7m96oieLjdnPddr/g38bbk1hswmHDNaX3MnunkMiIiIiMh4Wh0+A5ORkrTUQ/7pVZ+nSpVWe+/d1I4mIiOoLrnOoHz49UIdcXFxgiFl7X19frQdlamLSpEl47bXXKj2mz3I5RERET4SnrKgzFI4cPgEsLS01lsn5+1YdW1vbKs995plnHsMVEBERGYERX5+3du1auLi4wMLCAv7+/khNTX1k++3bt8PDwwMWFhbo2LEj9u3bp3FcCIGFCxfC0dERlpaW6NevH3766Sfp+NWrVzFu3Di4urrC0tISbm5uWLRoER48eFDj7CwOiYiIiAwoPj4eM2fOxKJFi3Dq1Cl4e3tj4MCBuHnzZqXtjx07hrCwMIwbNw7p6ekIDg5GcHAwzp49K7VZtmwZPv30U0RFReHEiRNo2LAhBg4ciPv37wMAsrKyUF5ejs8//xznzp3DqlWrEBUVJa0AUxN8WpkMik8rE2nj08q1w6eV66+6flq5NLaHwfpSjTmqc1t/f3907doVa9asAfDwrWwtW7bE1KlTMW/ePK32oaGhKCoqwt69e6V93bp1g4+PD6KioiCEgJOTE2bNmiW9jS0/Px/NmzdHbGwshg0bVmmO5cuXY926dfj555r9nDlySERERPWTAaeVi4uLUVBQoLEVFxdrfeWDBw9w8uRJ9Ov3v1fJKpVK9OvXDykpKZXGTElJ0WgPAAMHDpTaX7lyBTk5ORptbGxs4O/vX2WfwMMC0tbWtkY/MoDFIREREVG1IiIiYGNjo7FFRERotfv9999RVlaG5s2ba+xv3ry59Ha1v8vJyXlk+4p/1qTPS5cuYfXq1XjzzTd1u8C/4NPKREREVD8Z8Gnl+fPnY+bMmRr7/v4KWbm4du0aAgMDMXToUEyYMKHG57M4JCIionpJoTBcX+bm5joVg3Z2djAxMUFubq7G/tzcXDg4OFR6joODwyPbV/wzNzcXjo6OGm3+/ga269evo2/fvujevTu++OKLavNWhtPKRERERAZiZmaGLl264MCBA9K+8vJyHDhwAAEBAZWeExAQoNEeABITE6X2rq6ucHBw0GhTUFCAEydOaPR57do19OnTB126dEFMTAyUSv3KPI4cEhERUf2kx/qEhjBz5kyMHj0avr6+8PPzw8cff4yioiK88cYbAIBRo0bhmWeeke5ZnDZtGnr37o2VK1di0KBBiIuLQ1pamjTyp1AoMH36dHzwwQdo06YNXF1dsWDBAjg5OSE4OBjA/wpDZ2dnrFixArdu3ZLyVDViWRUWh0RERFQ/GekNKaGhobh16xYWLlyInJwc+Pj4ICEhQXqgJDs7W2NUr3v37tiyZQveffddhIeHo02bNti9ezc6dOggtZkzZw6KioowceJE5OXloWfPnkhISICFhQWAhyONly5dwqVLl9CiRQuNPDVdtZDrHJJBcZ1DIm1c57B2uM5h/VXX6xyWbfU3WF8mYScM1pfcceSQiIiI6icjTSs/6VgcEhERUf1kpGnlJx2LQyIiIqqXFBw51AuXsiEiIiIiCUcOiYiIqH4y4CLYTxMWh0RERFQ/cVpZL5xWJiIiIiIJRw6JiIiofuLTynphcUhERET1E6eV9cJpZSIiIiKScOSQiIiI6idOK+uFxSERERHVT5wf1YtRf2y3bt3CW2+9hVatWsHc3BwODg4YOHAgjh49Wu25SUlJUCgUyMvL0zrm4uKCjz/+WGt/REQETExMsHy59kvcY2Nj0bhx4yq/LyIiAl27doW1tTXs7e0RHByMCxcuVJuzOlevXoVCoUBGRobWsT59+mD69Ok69aNQKCrdKrvWypw6dQr9+/dH48aN0bRpU0ycOBGFhYU1uBIiIiKqD4xaHIaEhCA9PR0bNmzAxYsXsWfPHvTp0wd//PFHnXxfdHQ05syZg+jo6Bqfe+jQIUyePBnHjx9HYmIiSkpKMGDAABQVFdVB0pq7ceOGxhYdHQ2FQoGQkJBqz71+/Tr69esHd3d3nDhxAgkJCTh37hzGjBlT98GJiIjqikIYbnuKGG1aOS8vD8nJyUhKSkLv3r0BAM7OzvDz89NoM3fuXOzevRv5+flwd3dHZGQkXnrppRp/36FDh3Dv3j0sWbIEGzduxLFjx9C9e3edz09ISND4HBsbC3t7e5w8eRK9evWq9JyrV6/C1dUVW7duxaeffopTp07B3d0da9eula7ZUBwcHDQ+f/XVV+jbty9at25d7bl79+6Fqakp1q5dC6Xy4d8XoqKi0KlTJ1y6dAnu7u4GzUpERPQ48N3K+jHayKGVlRWsrKywe/duFBcXax0vLy9HUFAQjh49ik2bNuH8+fOIjIyEiYmJXt+nVqsRFhYGU1NThIWFQa1W1yp/fn4+AMDW1rbatrNnz8asWbOQnp6OgIAADB48uM5GRwEgNzcX33zzDcaNG6dT++LiYpiZmUmFIQBYWloCAI4cOVInGYmIiOocRw71YrTiUKVSITY2Fhs2bEDjxo3Ro0cPhIeH48cffwQA7N+/H6mpqfjyyy/Rv39/tG7dGi+99BKCgoI0+mnRooVUaFZs2dnZGm0KCgqwY8cOjBgxAgAwYsQIbNu2Te976srLyzF9+nT06NEDHTp0qLb9lClTEBISAk9PT6xbtw42NjZaxWn37t21riM5OVmvfBs2bIC1tTVeeeUVndo///zzyMnJwfLly/HgwQPcuXMH8+bNA/BwupqIiIieHka/5/D69evYs2cPAgMDkZSUhGeffRaxsbHIyMhAixYt0LZt20f2kZycjIyMDI3NyclJo83WrVvh5uYGb29vAICPjw+cnZ0RHx+vV+7Jkyfj7NmziIuLk/ZNmjRJo7D7q4CAAOnXKpUKvr6+yMzM1GgTHx+vdR2+vr565YuOjsbw4cNhYWGhU3svLy9s2LABK1euRIMGDeDg4ABXV1c0b95cYzTx74qLi1FQUKCxCVGqV2YiIiKDUxpwe4oYfSkbCwsL9O/fH/3798eCBQswfvx4LFq0CO+8845O57u6umo9ZaxSaV6WWq3GuXPnNPaXl5cjOjpa56nXClOmTMHevXtx+PBhtGjRQtq/ZMkSnTNXpmXLllr39lVM7dZEcnIyLly4UOPC9/XXX8frr7+O3NxcNGzYEAqFAh999NEj71mMiIjA4sWLNfaZmbwAc9N+Nc5NRERkcE/ZdLChyK4Wbt++PYqKitCpUyf89ttvuHjxYq36O3PmDNLS0pCUlKQxKpeUlISUlBRkZWXp1I8QAlOmTMGuXbtw8OBBuLq6ahy3t7eHu7u7tP3V8ePHpV+Xlpbi5MmT8PT0rNV1VUWtVqNLly7SKGlNNW/eHFZWVoiPj5cK96rMnz8f+fn5GpuZqo+eyYmIiEgOjDZy+Mcff2Do0KEYO3YsOnXqBGtra6SlpWHZsmV4+eWX0bt3b/Tq1QshISH46KOP4O7ujqysLCgUCgQGBur8PWq1Gn5+fpU+Udy1a1eo1WppLcCysjKt9QbNzc3h6emJyZMnY8uWLfjqq69gbW2NnJwcAICNjU21I3xr165FmzZt4OnpiVWrVuHOnTsYO3asztegq4KCAmzfvh0rV66s8blr1qyR7ntMTEzE7NmzERkZ+ci1H83NzWFubq6xT6Ew+mA0ERHRQ3xaWS9G+5PcysoK/v7+WLVqFS5fvoySkhK0bNkSEyZMQHh4OABg586deOeddxAWFoaioiJpKRtdPXjwAJs2bcLcuXMrPR4SEoKVK1di6dKlAIDCwkJ07txZo42bmxsuXbqEdevWAXi4MPVfxcTEVLseYGRkJCIjI5GRkQF3d3fs2bMHdnZ2Ol+HruLi4iCEQFhYWI3PTU1NxaJFi1BYWAgPDw98/vnnGDlypMEzEhERPTYKYwd4MimEECyr60jFOofp6enw8fExdpzHopGl7sX745Y+9KqxI9BTKu2ofrd50EOBmbq96YmePDZmP9dp/+J7L4P1peh7zmB9yR3nAImIiKh+4rSyXmT3QAppS05O1loDsaplcyqzdOnSKs/9+7qRRERE9YVQGm57mnDksA65uLjAELP2vr6+Wg/K1MSkSZPw2muvVXpMn+VyiIiIqP5icfgEsLS0rNX7jW1tbXV6zR8REVG9wnUO9cLikIiIiOon3nOoFxaHREREVD9xKRu9PGW3WBIRERHRo3DkkIiIiOonDoHphcUhERER1U+851AvrKmJiIiISMKRQyIiIqqf+ECKXlgcEhERUb0kOK2sF04rExEREZGEI4dERERUP3FaWS8sDomIiKh+4vyoXvhjIyIiIiIJRw6JiIiofuIQmF5YHBIREVG9JBR8WlkfLA6JiIiofuLIoV74YyMiIiIiCUcOiYiIqH7iEJheWBwSERFRvSS4zqFeWFMTERERkYQjh0RERFQ/cQhMLywOiYiIqF4SLA71wh8bEREREUmMWhzeunULb731Flq1agVzc3M4ODhg4MCBOHr0aLXnJiUlQaFQIC8vT+uYi4sLPv74Y639ERERMDExwfLly7WOxcbGonHjxlV+X0REBLp27Qpra2vY29sjODgYFy5cqDZnda5evQqFQoGMjAytY3369MH06dN16qewsBBTpkxBixYtYGlpifbt2yMqKkqnc2/fvo2pU6eiXbt2sLS0RKtWrfB///d/yM/Pr8GVEBERyYxCYbjtKWLU4jAkJATp6enYsGEDLl68iD179qBPnz74448/6uT7oqOjMWfOHERHR9f43EOHDmHy5Mk4fvw4EhMTUVJSggEDBqCoqKgOktbczJkzkZCQgE2bNiEzMxPTp0/HlClTsGfPnmrPvX79Oq5fv44VK1bg7NmziI2NRUJCAsaNG/cYkhMREdUNoTTc9jQx2uXm5eUhOTkZ//rXv9C3b184OzvDz88P8+fPx5AhQ6Q2b775Jpo3bw4LCwt06NABe/fu1ev7Dh06hHv37mHJkiUoKCjAsWPHanR+QkICxowZAy8vL3h7eyM2NhbZ2dk4efJkledUjArGxcWhe/fu0jUcOnRIr2t4lGPHjmH06NHo06cPXFxcMHHiRHh7eyM1NbXaczt06ICdO3di8ODBcHNzw/PPP48PP/wQX3/9NUpLSw2elYiIiOTLaMWhlZUVrKyssHv3bhQXF2sdLy8vR1BQEI4ePYpNmzbh/PnziIyMhImJiV7fp1arERYWBlNTU4SFhUGtVtcqf8WUq62tbbVtZ8+ejVmzZiE9PR0BAQEYPHiwwUdHu3fvjj179uDatWsQQuD777/HxYsXMWDAAL36y8/PR6NGjaBS8ZklIiJ6QikNuD1FjPYnv0qlQmxsLCZMmICoqCg8++yz6N27N4YNG4ZOnTph//79SE1NRWZmJtq2bQsAaN26tVY/LVq00Np39+5djc8FBQXYsWMHUlJSAAAjRozAc889h08++QRWVlY1zl5eXo7p06ejR48e6NChQ7Xtp0yZgpCQEADAunXrkJCQALVajTlz5khtunfvDqVS89++e/fuwcfHR6dMq1evxsSJE9GiRQuoVCoolUr8+9//Rq9evXS/sP/v999/x/vvv4+JEyfW+FwiIiK54CLY+jHqsFBISAgGDRqE5ORkHD9+HN9++y2WLVuG9evX4+bNm2jRooVUGFYlOTkZ1tbWGvv69Omj8Xnr1q1wc3ODt7c3AMDHxwfOzs6Ij4/X6766yZMn4+zZszhy5Ii0b9KkSdi0aZP0ubCwUPp1QECA9GuVSgVfX19kZmZq9BkfHw9PT0+NfcOHD9c50+rVq3H8+HHs2bMHzs7OOHz4MCZPngwnJyf069dP534KCgowaNAgtG/fHu+9994j2xYXF2uN+gpRCoWCo41ERCQDT9mIn6EY/U9xCwsL9O/fH/3798eCBQswfvx4LFq0CO+8845O57u6umo9Zfz3qVC1Wo1z585p7C8vL0d0dHSNi8MpU6Zg7969OHz4sMao5ZIlS3TOXJmWLVvC3d1dY5+lpaVO5967dw/h4eHYtWsXBg0aBADo1KkTMjIysGLFCp2Lwz///BOBgYGwtrbGrl27YGpq+sj2ERERWLx4scY+M5MXYG6qezFKRERE8iK7mrp9+/YoKipCp06d8Ntvv+HixYu16u/MmTNIS0tDUlISMjIypC0pKQkpKSnIysrSqR8hBKZMmYJdu3bh4MGDcHV11Thub28Pd3d3afur48ePS78uLS3FyZMntUYJa6OkpAQlJSVa09ImJiYoLy/XqY+CggIMGDAAZmZm2LNnDywsLKo9Z/78+cjPz9fYzFR99LkEIiIigxNKhcG2p4nRRg7/+OMPDB06FGPHjkWnTp1gbW2NtLQ0LFu2DC+//DJ69+6NXr16ISQkBB999BHc3d2RlZUFhUKBwMBAnb9HrVbDz8+v0nvvunbtCrVaLa17WFZWprXeoLm5OTw9PTF58mRs2bIFX331FaytrZGTkwMAsLGxqXaEb+3atWjTpg08PT2xatUq3LlzB2PHjtX5GqrTqFEj9O7dG7Nnz4alpSWcnZ1x6NAhbNy4ER999FG151cUhnfv3sWmTZtQUFCAgoICAECzZs2qfAjI3Nwc5ubmGvs4pUxERLIhuyGwJ4PR/iS3srKCv78/Vq1ahcuXL6OkpAQtW7bEhAkTEB4eDgDYuXMn3nnnHYSFhaGoqAju7u6IjIzU+TsePHiATZs2Ye7cuZUeDwkJwcqVK7F06VIAD+8T7Ny5s0YbNzc3XLp0CevWrQOgfT9jTEwMxowZ88gckZGRiIyMREZGBtzd3bFnzx7Y2dnpfB26iIuLw/z58zF8+HDcvn0bzs7O+PDDDzFp0qRqzz116hROnDgBAFqjnleuXIGLi4tBsxIREZF8KYQQwtgh6qurV6/C1dUV6enpOj91/KRrZKl78f64pQ+9auwI9JRKO+pt7AhPtMBM7bdaUf1gY/ZznfZ/92Y7g/XVwL72b0V7UnAOkIiIiOqnp+xeQUPhbPwTIDk5WVo0vLKtOps3b67yXC8vr8dwBURERPSk4MhhHXJxcYEhZu19fX21HpSpiSFDhsDf37/SY9UtV0NERPSketreiWwoLA6fAJaWlloPitSEtbW11kLhRERE9R3fkKIf1tREREREJOHIIREREdVPHALTC4tDIiIiqpd4z6F+WBwSERFRvSSUXMpZH6ypiYiIiEjC4pCIiIjqJ4Uw3FZDa9euhYuLCywsLODv74/U1NRHtt++fTs8PDxgYWGBjh07Yt++fRrHhRBYuHAhHB0dYWlpiX79+uGnn37SaHP79m0MHz4cjRo1QuPGjTFu3DgUFhbWODuLQyIiIqqXhFIYbKuJ+Ph4zJw5E4sWLcKpU6fg7e2NgQMH4ubNm5W2P3bsGMLCwjBu3Dikp6cjODgYwcHBOHv2rNRm2bJl+PTTTxEVFYUTJ06gYcOGGDhwIO7fvy+1GT58OM6dO4fExETs3bsXhw8fxsSJE2v8c+O7lcmg+G5lIm18t3Lt8N3K9Vddv1s5/34bg/VlY/FT9Y3+P39/f3Tt2hVr1qwBAJSXl6Nly5aYOnUq5s2bp9U+NDQURUVF2Lt3r7SvW7du8PHxQVRUFIQQcHJywqxZs/DOO+8AAPLz89G8eXPExsZi2LBhyMzMRPv27fHDDz/A19cXAJCQkIAXX3wRv/32G5ycnHTOz5FDIiIiqp+UZQbbiouLUVBQoLEVFxdrfeWDBw9w8uRJ9OvX738xlEr069cPKSkplcZMSUnRaA8AAwcOlNpfuXIFOTk5Gm1sbGzg7+8vtUlJSUHjxo2lwhAA+vXrB6VSiRMnTtTs5yaIZOr+/fti0aJF4v79+8aOokXO2YRgvtqQczYhmK825JxNCOaTu0WLFgkAGtuiRYu02l27dk0AEMeOHdPYP3v2bOHn51dp36ampmLLli0a+9auXSvs7e2FEEIcPXpUABDXr1/XaDN06FDx2muvCSGE+PDDD0Xbtm21+m7WrJn47LPPdL5OIQRXACL5Ki4uxuLFiyv9m5mxyTkbwHy1IedsAPPVhpyzAcwnd/Pnz0d+fr7GNn/+fGPHqhNc55CIiIioGubm5jA3N6+2nZ2dHUxMTJCbm6uxPzc3Fw4ODpWe4+Dg8Mj2Ff/Mzc2Fo6OjRhsfHx+pzd8feCktLcXt27er/N6qcOSQiIiIyEDMzMzQpUsXHDhwQNpXXl6OAwcOICAgoNJzAgICNNoDQGJiotTe1dUVDg4OGm0KCgpw4sQJqU1AQADy8vJw8uRJqc3BgwdRXl4Of3//Gl0DRw6JiIiIDGjmzJkYPXo0fH194efnh48//hhFRUV44403AACjRo3CM888g4iICADAtGnT0Lt3b6xcuRKDBg1CXFwc0tLS8MUXXwAAFAoFpk+fjg8++ABt2rSBq6srFixYACcnJwQHBwMAPD09ERgYiAkTJiAqKgolJSWYMmUKhg0bVqMnlQEWhyRj5ubmWLRokU7D+I+bnLMBzFcbcs4GMF9tyDkbwHz1SWhoKG7duoWFCxciJycHPj4+SEhIQPPmzQEA2dnZUCr/N3nbvXt3bNmyBe+++y7Cw8PRpk0b7N69Gx06dJDazJkzB0VFRZg4cSLy8vLQs2dPJCQkwMLCQmqzefNmTJkyBS+88AKUSiVCQkLw6aef1jg/1zkkIiIiIgnvOSQiIiIiCYtDIiIiIpKwOCQiIiIiCYtDIiIiIpKwOCQiIiIiCYtDIiIiIpJwnUOSjdTUVKSkpCAnJwfAw1cBBQQEwM/Pz6i5hBC4evUqWrZsCZVKhQcPHmDXrl0oLi7Giy++CDs7O6NlKy4uhlKphKmpKQDg8uXLiI6ORnZ2NpydnTFu3Di4uroaLV+FgwcP4siRI7hx4waUSiVat26NIUOGoE2bNsaOht9++w2NGzeGlZWVxv6SkhKkpKSgV69eRkr2Pzk5OThx4oTG7w1/f/8avxKrrpSVlcHExET6fOLECRQXFyMgIED6d9PY8vLysH37dun3xtChQ2FjY2PsWPj555+1fm/0798fjRo1MnY0AA/f7PHX9fj+uv+3335Dq1atjJCK6pwgMrLc3FzRs2dPoVAohLOzs/Dz8xN+fn7C2dlZKBQK0bNnT5Gbm2uUbFlZWcLZ2VkolUrh7u4ufv75Z9GlSxfRsGFD0aBBA2FnZycuXrxolGxCCNG7d2+xfft2IYQQR44cEebm5qJTp04iNDRUdO7cWTRo0EAcO3bMaPlyc3OFn5+fUCqVQqVSCaVSKbp06SIcHByEiYmJmD17ttGyXb9+XXTt2lUolUphYmIiRo4cKf7880/peE5OjlAqlUbLJ4QQhYWFYvjw4cLExESoVCphb28v7O3thUqlEiYmJmLEiBGiqKjIaPmuX78uevToIUxMTESvXr3E7du3xaBBg4RCoRAKhUK0bdtWXL9+3SjZ/vGPf0i/N86ePSvs7OxEs2bNhL+/v2jevLlwcHAQ58+fN0o2IR7+f/vqq69KPyulUin9vrCyshJr1qwxWjYhhMjPzxdDhw4VFhYWwt7eXixYsECUlpZKx+Xw+4PqDqeVyejefvttlJWVITMzE1evXsWJEydw4sQJXL16FZmZmSgvL8fkyZONkm3u3Lnw9vZGRkYGXnrpJQwaNAgtWrTAnTt3cPv2bQQEBGDJkiVGyQYA6enp8Pb2BgD885//xNtvv43Tp08jLi4Op06dwsyZMzF79myj5fu///s/ODk54c6dOygsLMTbb78NLy8v3LhxA//9738RHR2NTz75xCjZ5s2bB6VSiRMnTiAhIQHnz59H3759cefOHamNMPI7AqZNm4bU1FR88803uH//PnJzc5Gbm4v79+9j3759SE1NxbRp04yWb+7cuRBCYNeuXXB0dMRLL72EgoIC/Prrr7h69SqaNWuGDz/80CjZkpKSpLdLzJ49GwMGDMBvv/2G48eP49dff8WgQYMwffp0o2QDHr5e7caNG/jxxx9x8eJFvPLKKxg1ahQKCgrwySefYM6cOdiyZYvR8i1YsACnT5/Gf/7zH3z44YfYuHEjXn75ZTx48EBqY+zfH1SHjFycEgkrKytx6tSpKo+npaUJKyurx5jof5o1aybS09OFEA//pq9QKERycrJ0/OjRo6JVq1ZGySaEEA0bNhSZmZlCCCGaN28uMjIyNI5funTJaD87IYRo1KiROHv2rPS5sLBQmJqaivz8fCGEEP/5z39Eu3btjJLNyclJnDhxQvp8//59MXjwYOHj4yP++OMPWYyMNG7cWBw9erTK40eOHBGNGzd+jIk0OTo6ipSUFCGEEH/88YdQKBRi//790vEDBw6I1q1bGyWbpaWluHTpkpTz7/+NuXDhgrCxsTFCsofs7OxEWlqa9Pn27dvCwsJCGgles2aN8PHxMVY80apVK/H9999Ln2/duiX8/PzEgAEDxP3792Xx+4PqDkcOyejMzc1RUFBQ5fE///zTaO/yLCwshK2tLQCgYcOGaNiwIRwdHaXjLVu2RG5urlGyAYC/vz++/vprAICbmxtOnz6tcTwjI0PKbwzm5uZQKBTSZ6VSibKyMpSWlgJ4+D7Rq1evGiVbfn4+mjRpIn02NzfHl19+CRcXF/Tt2xc3b940Sq6/Ki8vh5mZWZXHzczMUF5e/hgTabpz5w6eeeYZAICtrS0aNGgAZ2dn6bi7uztu3LhhlGydOnXCwYMHATy8R/OXX37ROP7LL7/A0tLSGNEAAKWlpRr3FVpZWaG0tBRFRUUAgAEDBiArK8tY8XDr1i2N/y/t7Oywf/9+/Pnnn3jxxRdx9+5do2WjusfikIwuNDQUo0ePxq5duzSKxIKCAuzatQtvvPEGwsLCjJLNyckJ2dnZ0udly5bB3t5e+nzr1i2NAuNx++CDD/Dhhx/ivffeQ1hYGGbNmoUFCxZgy5YtWLRoEcaPH2+0KXkA6NmzJxYuXIiioiKUlJQgPDwcrVu3lgpWY/78WrdujR9//FFjn0qlwvbt29G6dWu89NJLRsn1Vy+99BImTpyI9PR0rWPp6el46623MHjwYCMke8je3l6j+JsyZYrGX0bu3LmDhg0bGiMaFixYgHnz5iE2Nhb/93//hxkzZkCtVuPYsWOIiYnBuHHjMHLkSKNkA4CuXbtq3FLxySefoFmzZmjWrBmAh38x/ftDUo9Tq1atkJmZqbHP2toa//3vf3Hv3j384x//MFIyeiyMPXRJdP/+fTFp0iRhZmYmlEqlsLCwEBYWFkKpVAozMzPx1ltvifv37xsl25tvvin+/e9/V3k8IiJCvPjii48xkbZjx46Jbt26STe2V2zPPPOM+Pjjj42a7fLly8LNzU2oVCphamoqGjduLBITE6XjMTExYt68eUbJNmfOHDFgwIBKj5WUlIghQ4YYfdrs9u3bIjAwUCgUCmFrays8PDyEh4eHsLW1FUqlUgQFBYk7d+4YLd+QIUMe+e/YmjVrxPPPP/8YE2nasWOHaNGihVAqlRq/NywsLMT06dM1HrB43E6ePClsbW2Fg4ODaNWqlTAzMxNbt26Vjq9Zs0aMGjXKaPmmTp0qXn311UqPFRQUCH9/f6P//qC6oxCCd5SSPBQUFODkyZMay3V06dJFNks6VObKlSuwsLDQmGo2llu3buHnn39GeXk5HB0d4eLiYuxIAIC7d+/i6NGjKC4uRrdu3Yy69M9flZaW4u7du1X++1VaWopr165pTK0ZS1ZWVqXLPHl4eBg52aOlpqaiQYMG0oMhxlBWVoZTp05p/N7o0qULrK2tjZapwo0bN7B3714UFxfj+eefR/v27Y0dSXLnzh1cv34dXl5elR7/888/cerUKfTu3fsxJ6PHgcUhPXE6duyIffv2oWXLlsaOokXO2QDmq41GjRohIyMDrVu3NnaUKg0aNAjr16+XxV9WKiPnfHLOBsg/n5x/71LN8Z5DeuJcvXoVJSUlxo5RKTlnA5ivNp6Ev0cfPnwY9+7dM3aMKsk5n5yzAfLPJ+ffu1RzLA6JiIiISMLikIiIiIgkLA6JiIiISMLikIhIB39dzJuIqD5jcUhEpIMn4YEUIiJDYHFIT5zPP/8czZs3N3aMSsk5G8B8tfHtt99Kr4qTq/DwcKO+LrE6cs4n52yA/PPJ+fcu6cGYK3ATCSHElClTxOHDh40do1JyziYE89XW6tWrxciRI6U3U2zcuFF4enqKdu3aifnz54uSkhKj5luxYoW4evWqUTM8ipzzyTmbEPLPJ4QQX3/9tViwYIE4cuSIEEKIAwcOiKCgIDFw4EDx+eefGzkd1SUWh2R0CoVCKJVK0aZNGxEZGSlu3Lhh7EgSOWcTgvlq4/333xfW1tYiJCREODg4iMjISNG0aVPxwQcfiKVLl4pmzZqJhQsXGjWjQqEQJiYmol+/fiIuLk4UFxcbNc/fyTmfnLMJIf98UVFRQqVSiS5duohGjRqJ//znP8La2lqMHz9evPnmm8LS0tLor+ekusPikIxOoVCI/fv3i2nTpgk7OzthamoqhgwZIr7++mtRVlbGbMxXJ9zc3MTOnTuFEEJkZGQIExMTsWnTJun4l19+Kdzd3Y0VTwjx8OcXExMjXn75ZWFqaiqaNm0qpk2bJs6cOWPUXBXknE/O2YSQf7727duLL774QgghxMGDB4WFhYVYu3atdDwmJkZ4enoaKx7VMRaHZHQKhULk5uYKIYR48OCBiI+PFwMHDhQmJibCyclJhIeHi59++onZmM+gLC0txS+//CJ9NjU1FWfPnpU+X716VTRo0MAY0SR//fnl5uaKf/3rX8LDw0MolUrRtWtX8cUXX4iCggLme8KyPQn5Kvv98dfC9cqVK0b//UF1h8UhGd1f/yP5V7/88otYtGiRcHZ2Fkql0gjJ5J1NCOarDVdXV/Htt98KIYS4ePGiUCqVYtu2bdLxb775Rri4uBglW4Wqfn6HDx8Wo0ePFg0bNhQNGzY0QrKH5JxPztmEkH++Fi1aSPcLX7t2TSgUCvHNN99Ix5OSkkSLFi2MFY/qGItDMrqq/iNZoby8XPz3v/99jIn+R87ZhGC+2nj33XdFs2bNxPjx44Wrq6uYN2+eaNWqlVi3bp2IiooSLVu2FDNmzDBKtgpKpfKRP7/8/Hxp6s8Y5JxPztmEkH++yZMnizZt2ogPPvhA+Pn5idGjRwsPDw/x7bffioSEBNGxY0cxduxYo+WjuqUQgot3kXG5uroiLS0NTZs2NXYULXLOBjBfbZSXlyMyMhIpKSno3r075s2bh/j4eMyZMwd3797F4MGDsWbNGjRs2NBoGZVKJXJycmBvb2+0DI8i53xyzgbIP19RURFmzJgh/f5YvXo1Pv30U/zzn/9ESUkJevfujfj4eNnmp9phcUhPnN9++w1OTk5QKuW3TKecswHMVxtyzkb0uNy/fx8lJSWwtrY2dhSqQ/yvHD1x2rdvj6tXrxo7RqXknA1gvtqQc7YKjRo1ws8//2zsGFWScz45ZwPkk8/CwqLSwlAu+cgwWBzSE0fOg91yzgYwX23IOVsFuWeUcz45ZwOYjx4vFodEREREJGFxSEREREQSFodEREREJGFxSE8chUJh7AhVknM2gPlqQ87ZKsg9o5zzyTkbwHz0eLE4pCeOnG98lnM2gPlqQ87ZKsg9o5zzyTkbwHz0eKmMHYCops6fPw8nJydjx6iUnLMBzFcbcs5W4dtvv8Uzzzxj7BhVknM+OWcDmI8eLy6CTUb3yiuv6NTuyy+/rOMk2uScDWC+2pBztgozZ87Uqd1HH31Ux0kqJ+d8cs4GMB/JG0cOyehsbGw0Pm/ZsgWDBw+WxQr8cs4GMF9tyDlbhfT0dI3PR44cQZcuXWBpaSntM+a9XnLOJ+dsAPORvHHkkGTH2toap0+fRuvWrY0dRYucswHMVxtyzlZB7hnlnE/O2QDmI3nhAylEREREJGFxSEREREQSFodEREREJOEDKWR0e/bs0fhcXl6OAwcO4OzZsxr7hwwZ8jhjAZB3NoD5akPO2Sr8+OOPGp+FEMjKykJhYaHG/k6dOj3OWBI555NzNoD5SN74QAoZnVJZ/QC2QqFAWVnZY0ijSc7ZAOarDTlnq6BUKqFQKCpdYLhiv7H//5VrPjlnA5iP5I0jh2R05eXlxo5QJTlnA5ivNuScrcKVK1eMHeGR5JxPztkA5iN548ghEREREUk4ckiycuHCBaxevRqZmZkAAE9PT0yZMgUeHh5GTibvbADz1Yacs1W4c+cO1Gq1RsaxY8fC1tbWyMkeknM+OWcDmI/kh08rk2zs3LkTHTp0wMmTJ+Ht7Q1vb2+cOnUKHTt2xM6dO5mN+Z66bBUOHz4MFxcXfPrpp7hz5w7u3LmD1atXw9XVFYcPHzZ2PFnnk3M25iPZEkQy0bp1a7FgwQKt/QsXLhStW7c2QqL/kXM2IZivNuScrUKHDh3EhAkTRGlpqbSvtLRUTJw4UXTo0MGIyR6Scz45ZxOC+UieWBySbFhaWoqffvpJa//FixeFpaWlERL9j5yzCcF8tSHnbBUsLCxEVlaW1v6srCxhYWFhhESa5JxPztmEYD6SJ04rk2z06dMHycnJWvuPHDmC5557zgiJ/kfO2QDmqw05Z6vw7LPPSvd7/VVmZia8vb2NkEiTnPPJORvAfCRPfCCFZGPIkCGYO3cuTp48iW7dugEAjh8/ju3bt2Px4sUaixY/7oWJ5ZyN+epvtgr/93//h2nTpuHSpUsaGdeuXYvIyEiNBYuNsSixnPPJORvzkVxxKRuSDV0WJQaMszCxnLMBzFcbcs5WobqMxl6UWM755JwNYD6SJ44ckmzIeVFiOWcDmK825JytgtwXJJZzPjlnA5iP5IkjhyRLv/32G5ycnHQe1Xmc5JwNYL7akHM2IqLHhf8FJFlq3749rl69auwYlZJzNoD5akPO2So0atQIP//8s7FjVEnO+eScDWA+kg8WhyRLch7QlnM2gPlqQ87ZKsg9o5zzyTkbwHwkHywOiYiIiEjC4pBkKTw8XLbv7ZRzNoD5akPO2SqMGDECjRo1MnaMKsk5n5yzAcxH8sEHUoiIiIhIwpFDkr1ff/0VY8eONXaMSsk5G8B8tSGXbPfu3cORI0dw/vx5rWP379/Hxo0bjZDqf+ScT87ZAOYjGXusL+sj0kNGRoZQKpXGjlEpOWcTgvlqQw7ZLly4IJydnYVCoRBKpVL06tVLXLt2TTqek5Nj1IxyzifnbMxHcsdFsMno/vp6ssoYc+kEOWcDmK825Jytwty5c9GhQwekpaUhLy8P06dPR8+ePZGUlIRWrVoZO56s88k5G/OR3PGeQzI6pVIpvYKpKsZ8dZRcswHMVxtyzlahefPm2L9/Pzp27Ajg4VIib7/9Nvbt24fvv/8eDRs2hJOTk9EyyjmfnLMxH8kd7zkko3N0dMSXX36J8vLySrdTp04xG/M9Vdkq3Lt3DyrV/yZ4FAoF1q1bh8GDB6N37964ePGiEdPJO5+cswHMR/LG4pCMrkuXLjh58mSVx6sb3alLcs4GMF9tyDlbBQ8PD6SlpWntX7NmDV5++WUMGTLECKn+R8755JwNYD6SNxaHZHSzZ89G9+7dqzzu7u6O77///jEm+h85ZwOYrzbknK3CP/7xD2zdurXSY2vWrEFYWJhRC1g555NzNoD5SN54zyERERERSThySLKXlZWFtm3bGjtGpeScDWC+2pBztgpyzyjnfHLOBjAfGReLQ5K94uJiXL582dgxKiXnbADz1Yacs1WQe0Y555NzNoD5yLhYHBIRERGRhMUhEREREUlYHBIRERGRhK/PI6Nr0qQJFApFlcdLS0sfYxpNcs4GMF9tyDlbBblnlHM+OWcDmI/kjcUhGd3HH39s7AhVknM2gPlqQ87ZKsg9o5zzyTkbwHwkb1znkJ4IZWVlMDExMXaMSsk5G8B8tSHnbBXknlHO+eScDWA+Mh7ec0iydvHiRcydOxctWrQwdhQtcs4GMF9tyDlbBblnlHM+OWcDmI+Mj8Uhyc7du3cRExOD5557Du3bt8ehQ4cwc+ZMY8cCIO9sAPPVhpyzVZB7Rjnnk3M2gPlIZgSRTKSkpIhx48aJRo0aiQ4dOggTExNx+PBhY8cSQsg7mxDMVxtyzlZB7hnlnE/O2YRgPpInjhyS0a1cuRJeXl549dVX0aRJExw+fBhnzpyBQqFA06ZNmY35nrpsFeSeUc755JyN+Uj2jF2dEpmYmIjw8HBRWlqqsV+lUolz584ZKdVDcs4mBPPVhpyzVZB7Rjnnk3M2IZiP5I0jh2R077//PrZv3w5XV1fMnTsXZ8+eNXYkiZyzAcxXG3LOVkHuGeWcT87ZAOYjmTN2dUpUISkpSYwaNUo0aNBAdOrUSZiYmIgjR44YO5YQQt7ZhGC+2pBztgpyzyjnfHLOJgTzkTyxOCTZKSgoEFFRUcLPz0+YmJiIgIAAsXLlSmPHEkLIO5sQzFcbcs5WQe4Z5ZxPztmEYD6SFy6CTbJ25swZqNVqbNmyBTdv3jR2HA1yzgYwX23IOVsFuWeUcz45ZwOYj4yPxSE9EUpKSmBqamrsGJWSczaA+WpDztkqyD2jnPPJORvAfGQ8fLcyGd3vv/+OoqIiODs7S/vOnTuHFStWoKioCMHBwXj99deZjfmemmwVPv3002rbKBQKTJ069TGk0SbnfHLOBjAfyRtHDsnowsLC4OTkhJUrVwIAbt68CQ8PDzg5OcHNzQ3ffvst1Go1Ro4cyWzM91Rkq+Dq6qrx+ddff4WjoyNUqv/9vV6hUODnn39+3NEAyDufnLMBzEcyZ8wbHomEEMLFxUUkJSVJn5cvXy7c3NxESUmJ9Nnf35/ZKsF89TNbVaysrMTly5eNHaNKcs4n52xCMB/JC9c5JKPLycmBi4uL9PngwYN45ZVXpL+hDhkyBD/99BOzVYL56mc2IiJjYnFIRteoUSPk5eVJn1NTU+Hv7y99VigUKC4uNkIyeWcDmK825JyNiMiYWByS0XXr1g2ffvopysvLsWPHDvz55594/vnnpeMXL15Ey5YtmY35nppsRETGxKeVyejef/99vPDCC9i0aRNKS0sRHh6OJk2aSMfj4uLQu3dvZmO+pyZbhYKCAo3PCoUChYWFWvsbNWr0OGNJ5JxPztkA5iN549PKJAu///47jh49CgcHB42pPQD45ptv4OXlpXF/GLP9D/PVz2wAoFQqoVAopM9CiEo/l5WVGSOerPPJORvAfCRvHDkkWbCzs0N+fj4CAwO1jvXv3x9xcXFG+0NaztkA5qsNOWcDgO+//95o360LOeeTczaA+UjeOHJIsmFiYoIbN27A3t5eY/8ff/wBe3t7o/4NVc7ZAOarDTlnIyIyBj6QQrLx92mLCr/99htsbGyMkOh/5JwNYL7akHO2Cvv27cN3332ntf+7777Dt99+a4REmuScT87ZAOYjeeK0Mhld586doVAooFAo8MILL2iswF9WVoYrV65UOu33tGdjvvqb7e/mzZuHyMhIrf1CCMybNw9BQUFGSPU/cs4n52wA85E8sTgkowsODgYAZGRkYODAgbCyspKOmZmZwcXFBR06dGC2SjBf/cz2dz/99BPat2+vtd/DwwOXLl0yQiJNcs4n52wA85FMPb6XsRA9WmxsrLh37570uaCgQHz++eeia9euQqlUGjGZvLMJwXy1IedsFZo3by4OHDigtT8xMVE0a9bMCIk0yTmfnLMJwXwkTywOSXYOHTokRo0aJRo2bCjatGkj5s6dK1JTU40dSwgh72xCMF9tyDnbxIkTRceOHcWlS5ekfT/99JPo1KmTGDdunBGTPSTnfHLOJgTzkTyxOCRZuHHjhoiIiBDu7u7C3t5eTJkyRahUKnHu3DljR5N1NiGYr75m+6u8vDzRrVs3oVKphIuLi3BxcREmJiaib9++4s6dO8aOJ+t8cs7GfCRXXMqGjG7w4ME4fPgwBg0ahOHDhyMwMBAmJiYwNTXF6dOnK73fhdmYrz5nq4wQAomJiTh9+jQsLS3h7e2N5557ztixJHLOJ+dsAPORDBmxMCUSQghhYmIiZsyYIS5evKixXw4jOHLOJgTz1Yacs1U4duyY+PrrrzX2xcbGCmdnZ9GsWTMxYcIEcf/+fSOlk3c+OWcTgvlI3rjOIRndkSNH8Oeff6JLly7w9/fHmjVr8Pvvvxs7FgB5ZwOYrzbknK3CkiVLcO7cOenzmTNnMGHCBPTv3x/z5s3D119/jYiICOZ7wrIxH8mesatTogqFhYVCrVaLHj16CFNTU6FUKsXHH38sCgoKjB1N1tmEYL76ms3BwUH88MMP0ufw8HDRo0cP6fO2bduEp6enMaIJIeSdT87ZhGA+kjcWhyRLWVlZYvbs2cLBwUFYWFiIwYMHGzuSRM7ZhGC+2pBbNnNzc5GdnS197tGjh/jggw+kz1euXBFWVlbGiCaEkHc+OWcTgvlI3jitTLLUrl07LFu2DL/99hu2bt1q7Dga5JwNYL7akFu25s2b48qVKwCABw8e4NSpU+jWrZt0/M8//4Spqamx4sk6n5yzAcxHMmfs6pSIiCo3adIkERAQIA4fPixmzpwpmjZtKoqLi6XjmzZtEr6+vsz3hGVjPpI7FodERDJ169Yt8dxzzwmFQiGsra3Fl19+qXH8+eefF+Hh4UZKJ+98cs72/9q725Cm2gYO4P+hKRubxXKWhAsFiWNg71EfylJy9kEWQYUvNE1YhJIpvRBkBRLzw4jGgvoQtPlSRhIVmVjBLHH0ZjBoVrSVrT6IUiGZrTSv+8P9ePHsvr1vvbt70Gf9fzDYOee6rvPnfPpzDjsTgvloZuN7DomIZrjBwUFotVrExMRE7P/w4QO0Wi3i4uKmKdnvZnK+mZwNYD6amVgOiYiIiEjiD1KIiIiISGI5JCIiIiKJ5ZCIiIiIJJZDIooKGzZswL59+6Y7BhHR/z2WQyKiKOfz+VBQUICUlBSo1WooigKHwzHdsYhohoqd7gBERPS/1d3djaSkJDQ2NiIlJQVerxdWqxUxMTGoqKiY7nhENMPwziERRY3R0VFUVFRg9uzZSExMRE1NDcbf1tXQ0ICVK1dCp9Nh/vz5KCwsRH9/v5z78eNHFBUVwWAwQK1WIz09HefPn5fH3759i+3bt2POnDnQ6/Uwm83o7e2Vx0tKSrBlyxbY7XYkJydj7ty5KC8vx8jIiBwzWYaOjg6oVCq0t7dj2bJlUKvVyM7ORn9/P9ra2qAoChISElBYWIjh4WE5b2xsDDabDampqVCr1ViyZAlaWlrk8V27dsHhcCArKwtpaWkoLi5GaWkprly58lOvPxFFB5ZDIooabrcbsbGxePjwIRwOB06ePIlz584BAEZGRlBbWwufz4erV6+it7cXJSUlcm5NTQ16enrQ1taGZ8+e4cyZM0hMTJRzTSYTdDodOjs70dXVBa1Wi7y8PHz79k2u4fF4EAwG4fF44Ha74XK54HK55PHJMow7fvw4Tp8+Da/XK0vpqVOncOHCBbS2tuLWrVtwOp1yvM1mQ319Pc6ePQu/34+qqioUFxfj7t27f3mtBgcHodfrf/BKE1FUm9b/ZyEi+kmysrKEoihibGxM7jt06JBQFGXC8Y8ePRIAxKdPn4QQQuTn54vS0tIJxzY0NIhFixZFrP3161ehVqtFe3u7EEIIi8UiFi5cKEZHR+WYbdu2iR07dvxl5j9m8Hg8AoC4c+eOHGOz2QQAEQwG5b7du3cLk8kkhBAiHA4LjUYjvF5vxNplZWWioKBgwvN2dXWJ2NhYmZ2I6L/xziERRY01a9ZApVLJ7bVr1+Lly5f4/v07uru7kZ+fD6PRCJ1Oh6ysLABAKBQCAOzZswfNzc1YunQpDh48CK/XK9fx+XwIBALQ6XTQarXQarXQ6/UIh8MIBoNy3OLFiyP+Ziw5OTnisfFkGcZlZmbK7/PmzYNGo0FaWlrEvvF1A4EAhoeHsWnTJplNq9Wivr4+Itu4p0+fwmw249ixY8jNzf0HV5eIfhX8QQoRRb1wOAyTyQSTyYSmpiYYDAaEQiGYTCb5WHjz5s148+YNbt68idu3byMnJwfl5eWw2+0YGhrCihUr0NTU9Ke1DQaD/D5r1qyIYyqVCmNjYwCAz58/T5phonVUKtXfrjs0NAQAaG1txYIFCyLGxcfHR2z39PQgJycHVqsVR44cmfzCEdEvieWQiKLGgwcPIrbv37+P9PR0PH/+HO/fv0ddXR1SUlIAAI8fP/7TfIPBAIvFAovFgnXr1uHAgQOw2+1Yvnw5Ll26hKSkJCQkJPxQtqlm+KcyMjIQHx+PUCgk70ROxO/3Izs7GxaLBSdOnPjX5yWi6MXHykQUNUKhEKqrq/HixQtcvHgRTqcTlZWVMBqNiIuLg9PpxKtXr3D9+nXU1tZGzD169CiuXbuGQCAAv9+PGzduQFEUAEBRURESExNhNpvR2dmJ169fo6OjA3v37sW7d++mlG0qGX6ETqfD/v37UVVVBbfbjWAwiCdPnsDpdMLtdgP4/VHyxo0bkZubi+rqavT19aGvrw8DAwP/+vxEFH1YDokoauzcuRNfvnzB6tWrUV5ejsrKSlitVhgMBrhcLly+fBkZGRmoq6uD3W6PmBsXF4fDhw8jMzMT69evR0xMDJqbmwEAGo0G9+7dg9FoxNatW6EoCsrKyhAOh6d8J3EqGX5UbW0tampqYLPZoCgK8vLy0NraitTUVABAS0sLBgYG0NjYiOTkZPlZtWrVTzk/EUUXlRD/eQkYEREREf3yeOeQiIiIiCSWQyIiIiKSWA6JiIiISGI5JCIiIiKJ5ZCIiIiIJJZDIiIiIpJYDomIiIhIYjkkIiIiIonlkIiIiIgklkMiIiIiklgOiYiIiEhiOSQiIiIi6TdIAbRQzktAlAAAAABJRU5ErkJggg==",
      "text/plain": [
       "<Figure size 640x480 with 2 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "subset_df2 = subset_df.pivot(index=\"basename1\", columns=\"basename2\", values=\"emd_score\")\n",
    "sns.heatmap(subset_df2, cmap = 'plasma_r')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Axes: xlabel='basename2', ylabel='basename1'>"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAo8AAAIPCAYAAAASIDSjAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjkuMCwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy80BEi2AAAACXBIWXMAAA9hAAAPYQGoP6dpAADoYUlEQVR4nOzdeVxTV/o/8E8gLGExiAiRumCRRaOCFUHUWmwpYK3KlGKM4orYWqQKjAtWLK5YFNRxo9YLyFdExG0stTpoC2pBKQrWDdSOSqeAS5VQUEEgvz/4Gb0mQBITAX3e8zqvTu4959znpvTFw7n3nMORSqVSEEIIIYQQogSd1g6AEEIIIYS0H5Q8EkIIIYQQpVHySAghhBBClEbJIyGEEEIIURolj4QQQgghRGmUPBJCCCGEEKVR8kgIIYQQQpRGySMhhBBCCFEaJY+EEEIIIURp3NYOgLxeJLVvt3YI7VaHE0Ya7a+myEqj/b1JdEaWaLS/e+vcNdpfl7EFGu3vTVJfbtLaITRL17KqtUN4pTjeF7Tafwfeao31Vfloocb6au8oeSSEEELIa0lXymntEF5L9NiaEEIIIYQorV0lj7m5udDV1cWoUaNYx6OiouDs7Nxku+joaOjq6mLNmjVy55KSkmBmZqawHYfDwcGDB5v8rAkeHh6YO3euSnG9aOrUqeBwOHJFKBQqHcfmzZthY2MDQ0NDuLm5IS8vT+m2hBBCSFukC47GCnmmXSWPDMMgJCQEJ06cQGlpqdLtEhISMH/+fCQkJGgxutazYcMGlJWVycoff/wBc3Nz+Pv7K9U+LS0NYWFh+Prrr3Hu3Dk4OTnB29sbd+7c0XLkhBBCiPZQ8qgd7SZ5rKqqQlpaGmbNmoVRo0YhKSkJQOMI3dKlS3H+/HnZiNvTcwCQnZ2NR48eYdmyZaisrEROTs4ri9nDwwOzZ8/G7NmzwefzYWFhgcjISEilUo1eh8/nQyAQyEp+fj4ePHiAadOmKdU+Li4OQUFBmDZtGvr06YP4+HgYGRm9tsk2IYSQN4OuVEdjhTzTbr6NPXv2wNHREQ4ODggICEBCQgKkUilEIhHCw8MhFAplI28ikUjWjmEYiMVi6OnpQSwWg2GYVxr3jh07wOVykZeXhw0bNiAuLg7bt2/X6jUZhoGnpyd69OjRYt3a2lqcPXsWnp6esmM6Ojrw9PREbm6uNsMkhBBCSDvUbmZbMwyDgIAAAICPjw8kEgmys7Ph4eEBExMTcLlcCAQCVpvKykrs3btXlgQFBATg3XffxYYNG2Bi8my5BolEwvqsSd26dcO6devA4XDg4OCACxcuYN26dQgKCpLV2bJli1xCWVdXB0NDQ5WvV1paih9//BG7du1Sqv69e/dQX18PKyv2si5WVlYoKipS+fqEEEJIW6FDj5u1ol2MPBYXFyMvLw9isRgAwOVyIRKJWhxFTE1Nha2tLZycnAAAzs7O6NGjB9LS0lj1TE1NUVhYKFeUdfLkSZiYmMhKSkqK7NzgwYPB4Tz74XV3d8e1a9dQX18vOzZx4kS5ay9btkzp6z9vx44dMDMzg6+vr1rtVVFTU4PKykpWqanR7CN5QgghRF064GiskGfaxcgjwzCoq6uDtbW17JhUKoWBgQE2bdrUbLtLly6By312mw0NDUhISEBgYKDsmI6ODnr16qV2fC4uLqxk88VRvJbw+Xy561taWqoch1QqRUJCAiZNmgR9fX2l2lhYWEBXVxe3b99mHb99+7bcSO6LoqOjsXTpUtaxBYvNEBHZUbXACSGEENJutPnksa6uDsnJyYiNjYWXlxfrnK+vL1JTU6Gvr88ayQOACxcuID8/H1lZWTA3N5cdv3//Pjw8PFBUVARHR0eNxMjj8ZpMPs+cOcP6fPr0adjZ2UFXV1cj135ednY2rl+/zkqMW6Kvr4+BAwfi+PHjstHKhoYGHD9+HLNnz262bUREBMLCwljHHnOcVI6bEEII0QaaJa0dbT55zMjIwIMHDxAYGAg+n8865+fnB4ZhEBoaihs3bqCwsBBdu3aFqakpGIaBq6srhg8fLtfnoEGDwDCMwnUfW/L0Os+zs7ODsbGxwvolJSUICwvDZ599hnPnzmHjxo2IjY1V+brKYBgGbm5u6Nu3r0rtwsLCMGXKFLi4uMDV1RXr169HdXV1i7O1DQwMYGBgwDomraX/UAkhhLQN9LhZO9p88vh05vCLiSPQmDzGxMRAKBTCx8cHI0aMQEVFBb799lvs3LkTCxYsUNinn58fYmNjsWrVKpXjeXGkDWh853HYsGEK60+ePBmPHj2Cq6srdHV1MWfOHMycOVPl67ZEIpFg37592LBhg8ptRSIR7t69iyVLlqC8vBzOzs44cuSIyo/fCSGEEPL640g1veggkfHw8ICzszPWr1/f2qG8MpLat1s7hHarwwkjjfZXU0TJv7p0RpZotL9769w12l+XsQUa7e9NUl+unZU1NEXXsqq1Q3ilON4XtNq/ncG/NNbXtZovNdZXe9fmRx4JIYQQQtShQ8NjWtEulup50wmFQtZSQE0tC6RISUlJk21NTExQUqLZERZCCCGEAJs3b4aNjQ0MDQ3h5uaGvLy8Zuunp6fD0dERhoaG6NevHw4fPsw6v3//fnh5eaFTp07gcDhy8y9u3rwp22nvxZKeni6rp+j87t27Vbo3GnnUoqysLI30c/jwYTx58kThuZbeS7S2tm52zcrnlz8ihBBCXietNWEmLS0NYWFhiI+Ph5ubG9avXw9vb28UFxcrXIovJycHYrEY0dHR+Pjjj7Fr1y74+vri3Llzskmw1dXVGDZsGMaNG8faaOSpbt26oaysjHVs27ZtWLNmDUaOHMk6npiYCB8fH9lnMzMzle6P3nkkGkXvPKqP3nlsO+idx9cXvfPYtmj7nUehftNrQavqUm3zy9c9z83NDYMGDZKtRd3Q0IBu3bohJCQECxculKsvEolQXV2NjIwM2bHBgwfD2dkZ8fHxrLo3b95Ez549UVBQAGdn52bjGDBgAN555x3WpiocDgcHDhx4qc1E6LE1IYQQQl5LrbHDTG1tLc6ePQtPT89ncejowNPTU7Zd8otyc3NZ9QHA29u7yfrKOHv2LAoLCxWu/RwcHAwLCwu4uroiISEBqo4j0mNrQgghhJAW1NTUoKamhnVM0XrH9+7dQ319vdxrZVZWVigqKlLYd3l5ucL65eXlasfLMAx69+6NIUOGsI4vW7YM77//PoyMjPCf//wHX3zxBaqqqvDll8rPJqeRR0IIIYS8ljQ58hgdHQ0+n88q0dHRrX2LCj169Ai7du1SOOoYGRmJoUOHYsCAAViwYAHmz5+v8qYplDwSQggh5LXE0eD/IiIiIJFIWCUiIkLumhYWFtDV1cXt27dZx2/fvg2BQKAwToFAoFL9luzduxcPHz7E5MmTW6zr5uaG//3vf3Kjqs1pV8ljbm4udHV1MWrUKNbxqKioZl8ajY6Ohq6ursLMOikpqclZRhwOBwcPHmzysyZ4eHhg7ty5KsX1oqlTpyqcei8UCpVqv3XrVvTv3x8dOnRAhw4d4O7ujh9//FGFuyCEEEJebwYGBrLfk0/Li4+sAUBfXx8DBw7E8ePHZccaGhpw/PhxuLsrnjzn7u7Oqg8AmZmZTdZvCcMwGDNmDDp37txi3cLCQnTs2FHhvTSlXb3zyDAMQkJCwDAMSktLlV5mJiEhAfPnz0dCQgLmzZun5ShfvQ0bNmD16tWyz3V1dXBycoK/v79S7bt27YrVq1fDzs4OUqkUO3bswNixY1FQUKB0AkoIIYS0Na21VE9YWBimTJkCFxcXuLq6Yv369aiursa0adMANG5d/NZbb8kee8+ZMwfvvfceYmNjMWrUKOzevRv5+fnYtm2brM/79++jpKQEpaWlAIDi4mIAjaOWz49QXr9+HSdOnJBbJxIAvv/+e9y+fRuDBw+GoaEhMjMzsWrVKvzzn/9U6f7azchjVVUV0tLSMGvWLIwaNQpJSUkAGkfoli5divPnz8tG3J6eA4Ds7Gw8evQIy5YtQ2VlJXJycl5ZzB4eHpg9ezZmz54NPp8PCwsLREZGqjyrqSV8Pl/2wyMQCJCfn48HDx7IfkhbMnr0aHz00Uews7ODvb09Vq5cCRMTE5w+fVqjcRJCCCGvko4GiypEIhHWrl2LJUuWwNnZGYWFhThy5IhsUkxJSQlrTcYhQ4Zg165d2LZtG5ycnLB3714cPHhQtsYjABw6dAgDBgyQPX0dP348BgwYILeUT0JCArp27QovLy+5uPT09LB582a4u7vD2dkZ3377LeLi4vD111+rdH/tZuRxz549cHR0hIODAwICAjB37lxERERAJBLh4sWLOHLkCI4dOwagMZl6imEYiMVi6OnpQSwWg2EYuZlH2rRjxw4EBgYiLy8P+fn5mDlzJrp3765wgU9NYRgGnp6e6NGjh8pt6+vrkZ6ejurqarWHywkhhJA33dPBI0UUbSLi7+/f7BPDqVOnYurUqS1ed9WqVVi1apXCcz4+PqzFwdXVbpJHhmEQEBAAoPHmJRIJsrOz4eHhARMTE3C5XLkXSysrK7F3717ZOkkBAQF49913sWHDBpiYPFsoViKRsD5rUrdu3bBu3TpwOBw4ODjgwoULWLduHSt53LJlC7Zv385qV1dXB0NDQ5WvV1paih9//BG7du1Sqd2FCxfg7u6Ox48fw8TEBAcOHECfPn2abaNo2YIajhQGBq3zmIAQQgh5no6Ufh9pQ7t4bF1cXIy8vDyIxWIAAJfLhUgkYq2YrkhqaipsbW3h5OQEAHB2dkaPHj2QlpbGqmdqaorCwkK5oqyTJ082ud/04MGDweE8++F1d3fHtWvXUF9fLzs2ceJEuWsvW7ZM6es/b8eOHTAzM1N55XgHBwcUFhbizJkzmDVrFqZMmYLLly8320bRsgVxMRVqxU0IIYRoWmssEv4maBcjjwzDoK6ujjVBRiqVwsDAQLb1T1PtLl26BC732W02NDQgISGBtfaRjo4OevXqpXZ8Li4urGSzpf2mX8Tn8+Wur2jvy5ZIpVIkJCRg0qRJ0NfXV6mtvr6+LIaBAwfi119/xYYNG/Dtt9822SYiIgJhYWGsY485TirHTQghhGgDh5I+rWjzyWNdXR2Sk5MRGxsr9/Knr68vUlNToa+vzxrJAxofw+bn5yMrKwvm5uay4/fv34eHhweKiorg6OiokRh5PF6TyeeZM2dYn0+fPg07Ozvo6upq5NrPy87OxvXr1xUuCqqqhoaGFtd8UrSyvrSW/kMlhBBCXmdtPnnMyMjAgwcPEBgYyJoIAwB+fn5gGAahoaG4ceMGCgsL0bVrV5iamoJhGLi6umL48OFyfQ4aNAgMw6i8ojoA2XWeZ2dnB2NjY4X1S0pKEBYWhs8++wznzp3Dxo0bERsbq/J1lcEwDNzc3Fizs5QRERGBkSNHonv37vj777+xa9cuZGVl4ejRo1qJkxBCCHkV2sW7ee1Qm08en84cfjFxBBqTx5iYGAiFQvj4+GDEiBGoqKjAt99+i507d2LBggUK+/Tz80NsbGyTs5Ga8+JjWqDxncdhw4YprD958mQ8evQIrq6u0NXVxZw5czBz5kyVr9sSiUSCffv2YcOGDSq3vXPnDiZPnoyysjLw+Xz0798fR48exYcffqjxOAkhhJBXhd5V1A6OVNOLDhIZDw8PODs7Y/369a0dyisjqX27tUNotzqcMNJofzVFqr17S57RGVmi0f7urdPsslddxhZotL83SX25dlbW0BRdy6rWDuGV4nhf0Gr/7+ptb7mSkk4+maGxvtq7Nj/ySAghhBCiDhp51A56HaAdEAqFrKWAmloWSJGSkpIm25qYmKCkRLMjLIQQQkhbwdFgIc/QyKMWKVpBXh2HDx/GkydPFJ5raVkga2vrZtesVHZ/cEIIIYQQgJLHdkGdbQaf4nK5L7WGJSGEENJe0WNr7aDkkRBCCCGvJdqeUDvonUdCCCGEEKI0GnkkhBBCyGuJxh21o12NPObm5kJXVxejRo1iHY+KioKzs3OT7aKjo6Grq6twR5mkpCSYmZkpbMfhcHDw4MEmP2uCh4cH5s6dq1JcL5o6dSo4HI5cEQqFSsfx559/IiAgAJ06dQKPx0O/fv2Qn5+vdHtCCCGkrdEBR2OFPNOukkeGYRASEoITJ06gtLRU6XYJCQmYP38+EhIStBhd69mwYQPKyspk5Y8//oC5uTn8/f2Vav/gwQMMHToUenp6+PHHH3H58mXExsaiY8eOWo6cEEII0R4dDRbyTLv5PqqqqpCWloZZs2Zh1KhRSEpKAtA4Qrd06VKcP39eNuL29BwAZGdn49GjR1i2bBkqKyuRk5PzymL28PDA7NmzMXv2bPD5fFhYWCAyMhKa3tSHz+dDIBDISn5+Ph48eIBp06Yp1f6bb75Bt27dkJiYCFdXV/Ts2RNeXl6wtbXVaJyEEEIIaf/aTfK4Z88eODo6wsHBAQEBAUhISIBUKoVIJEJ4eDiEQqFs5E0kEsnaMQwDsVgMPT09iMViMAzzSuPesWMHuFwu8vLysGHDBsTFxWH7ds1tl6TI0/3AlV3i59ChQ3BxcYG/vz8sLS0xYMAAfPfdd1qNkRBCCNE2jgb/R55pNxNmGIZBQEAAAMDHxwcSiQTZ2dnw8PCAiYkJuFwuBAIBq01lZSX27t2L3NxcAEBAQADeffddbNiwASYmz/Y3lUgkrM+a1K1bN6xbtw4cDgcODg64cOEC1q1bh6CgIFmdLVu2yCWUdXV1MDQ0VPl6paWl+PHHH7Fr1y6l2/z3v//F1q1bERYWhkWLFuHXX3/Fl19+CX19fUyZMkXlGAghhJC2gN5V1I52MfJYXFyMvLw8iMViAI0LX4tEohZHEVNTU2FrawsnJycAgLOzM3r06IG0tDRWPVNTUxQWFsoVZZ08ebLJLQMHDx4MDufZD6+7uzuuXbuG+vp62bGJEyfKXXvZsmVKX/95O3bsgJmZGXx9fZVu09DQgHfeeQerVq3CgAEDMHPmTAQFBSE+Pr7ZdjU1NaisrGSVmhrNPpInhBBCSNvSLkYeGYZBXV0days9qVQKAwMDbNq0qdl2ly5dApf77DYbGhqQkJCAwMBA2TEdHZ2X2oXFxcWFlWy2tGXgi/h8vtz1LS0tVY5DKpUiISEBkyZNgr6+vtLtunTpgj59+rCO9e7dG/v27Wu2XXR0NJYuXco6tmCxGSIiaaINIYSQ1seh8QytaPPJY11dHZKTkxEbGwsvLy/WOV9fX6SmpkJfX581kgcAFy5cQH5+PrKysmBubi47fv/+fXh4eKCoqAiOjo4aiZHH4zWZfJ45c4b1+fTp07Czs4Ourq5Grv287OxsXL9+nZUYK2Po0KEoLi5mHbt69WqL70xGREQgLCyMdewxx0mlaxNCCCHaQo+ttaPNJ48ZGRl48OABAgMDwefzWef8/PzAMAxCQ0Nx48YNFBYWomvXrjA1NQXDMHB1dcXw4cPl+hw0aBAYhlG47mNLnl7neXZ2djA2NlZYv6SkBGFhYfjss89w7tw5bNy4EbGxsSpfVxkMw8DNzQ19+/ZVqV1oaCiGDBmCVatWYdy4ccjLy8O2bduwbdu2ZtsZGBjAwMCAdUxaS/+hEkIIIa+zNv/O49OZwy8mjkBj8pifnw+hUAgfHx+MGDECnTt3xo4dO7Bz5074+fkp7NPPzw/Jycl48uSJyvGEhYVhwIABrFJQUNBk/cmTJ+PRo0dwdXVFcHAw5syZg5kzZ6p83ZZIJBLs27dP5VFHoDGZPnDgAFJTU9G3b18sX74c69evx8SJEzUeJyGEEPKq0DqP2sGRanrRQSLj4eEBZ2dnrF+/vrVDeWUktW+3dgjtVocTRhrtr6ZItXdvyTM6I0s02t+9de4a7a/L2Kb/YCXNqy/XzsoamqJrWdXaIbxSHO8LWu3/E92dGutrf32Axvpq7yiZJoQQQgghSqPksR0QCoWspYCaWhZIkZKSkibbmpiYoKREsyMshBBCSFtBj621o81PmGnPsrKyNNLP4cOHm3w/s6VlgaytrZtds/L55Y8IIYSQ1wnNttYOSh7bAWW3GVSEy+W+1BqWhBBCSHtF6zxqB43EEkIIIYQQpdHIIyGEEEJeSzRCph2UPBJCCCHktcShdx61gpJyQgghhBCitHaVPObm5kJXVxejRo1iHY+KioKzs3OT7aKjo6Grq6twO8KkpCSYmZkpbMfhcHDw4MEmP2uCh4cH5s6dq1JcL5o6dSo4HI5cEQqFSrX/+++/MXfuXPTo0QM8Hg9DhgzBr7/+qsJdEEIIIW0PLdWjHe3q+2AYBiEhIThx4gRKS0uVbpeQkID58+cjISFBi9G1ng0bNqCsrExW/vjjD5ibm8Pf31+p9jNmzEBmZib+7//+DxcuXICXlxc8PT3x559/ajlyQgghRHs4GvwfeabdJI9VVVVIS0vDrFmzMGrUKCQlJQFoHKFbunQpzp8/Lxtxe3oOALKzs/Ho0SMsW7YMlZWVyMnJeWUxe3h4YPbs2Zg9ezb4fD4sLCwQGRkJTe8IyefzIRAIZCU/Px8PHjzAtGnTWmz76NEj7Nu3DzExMRg+fDh69eqFqKgo9OrVC1u3btVonIQQQghp/9pN8rhnzx44OjrCwcEBAQEBSEhIgFQqhUgkQnh4OIRCoWzkTSQSydoxDAOxWAw9PT2IxWIwDPNK496xYwe4XC7y8vKwYcMGxMXFYfv27Vq9JsMw8PT0VGp9yLq6OtTX18PQ0JB1nMfj4dSpU9oKkRBCCNE6emytHe1mtjXDMAgIaNyU3MfHBxKJBNnZ2fDw8ICJiQm4XC4EAgGrTWVlJfbu3Yvc3FwAQEBAAN59911s2LABJiYmsnoSiYT1WZO6deuGdevWgcPhwMHBARcuXMC6desQFBQkq7Nlyxa5hLKurk4uoVNGaWkpfvzxR+zatUup+qampnB3d8fy5cvRu3dvWFlZITU1Fbm5uS0uLl5TU4Oamhr2MY4UBgY0vE8IIaT1UdKnHe3iey0uLkZeXh7EYjGAxl1TRCJRi6OIqampsLW1hZOTEwDA2dkZPXr0QFpaGqueqakpCgsL5YqyTp482eR+04MHDwaH8yyZcnd3x7Vr11BfXy87NnHiRLlrL1u2TOnrP2/Hjh0wMzODr6+v0m3+7//+D1KpFG+99RYMDAzwr3/9C2KxGDo6zf94REdHg8/ns0pcTIVacRNCCCGkfWgXySPDMKirq4O1tTW4XC64XC62bt2Kffv2QSKRNNvu0qVLsjZcLheXL1+Wmzijo6ODXr16yRVlubi4sBK/MWPGqHR/fD5f7tqWlpYq9QEAUqkUCQkJmDRpEvT19ZVuZ2tri+zsbFRVVeGPP/5AXl4enjx5grfffrvZdhEREZBIJKwSNt9M5bgJIYQQbeBIORorqtq8eTNsbGxgaGgINzc35OXlNVs/PT0djo6OMDQ0RL9+/XD48GHW+f3798PLywudOnUCh8NROMjl4eEht/LK559/zqpTUlKCUaNGwcjICJaWlpg3bx7q6upUurc2/9i6rq4OycnJiI2NhZeXF+ucr68vUlNToa+vzxrJA4ALFy4gPz8fWVlZMDc3lx2/f/8+PDw8UFRUBEdHR43EyOPxmkw2z5w5w/p8+vRp2NnZQVdXVyPXfl52djauX7+OwMBAtdobGxvD2NgYDx48wNGjRxETE9NsfQMDAxgYGLCOSWvpkTUhhJC2obVGyNLS0hAWFob4+Hi4ublh/fr18Pb2RnFxscLBoZycHIjFYkRHR+Pjjz/Grl274Ovri3PnzqFv374AgOrqagwbNgzjxo1jvfr2oqCgINbTSyMjI9n/r6+vx6hRoyAQCJCTk4OysjJMnjwZenp6WLVqldL31+aTx4yMDDx48ACBgYHg8/msc35+fmAYBqGhobhx4wYKCwvRtWtXmJqagmEYuLq6Yvjw4XJ9Dho0CAzDKFz3sSVPr/M8Ozs7GBsbK6xfUlKCsLAwfPbZZzh37hw2btyI2NhYla+rDIZh4ObmJvtBU9bRo0chlUrh4OCA69evY968eXB0dFRqtjYhhBDSVrXWcEZcXByCgoJkv0fj4+Pxww8/ICEhAQsXLpSrv2HDBvj4+GDevHkAgOXLlyMzMxObNm1CfHw8AGDSpEkAgJs3bzZ7bSMjI7k5IE/95z//weXLl3Hs2DFYWVnB2dkZy5cvx4IFCxAVFaX0U8s2/9j66czhFxNHoDF5zM/Ph1AohI+PD0aMGIHOnTtjx44d2LlzJ/z8/BT26efnh+TkZDx58kTleMLCwjBgwABWKSgoaLL+5MmT8ejRI7i6uiI4OBhz5szBzJkzVb5uSyQSCfbt26fWqKNEIkFwcDAcHR0xefJkDBs2DEePHoWenp7G4ySEEEJeZ7W1tTh79iw8PT1lx3R0dODp6SmbwPui3NxcVn0A8Pb2brJ+c1JSUmBhYYG+ffsiIiICDx8+ZF2nX79+sLKyYl2nsrISly5dUvoabX7k8fvvv2/ynKurq2zNxL1797LONZegzZ8/H/PnzwfQuDvL1KlTFdZ7cT1GddZn1NPTw/r165tcMzErK0vh8ebiUoTP57N+QFQxbtw4jBs3Tq22hBBCSFulyREyRSuMKHp96969e6ivr2claABgZWWFoqIihX2Xl5crrF9eXq5SjBMmTECPHj1gbW2N3377DQsWLEBxcTH279/f7HWenlNWm08eCSGEEELUocmdYaKjo7F06VLWsa+//hpRUVEau8bLen7grF+/fujSpQs++OAD/P7777C1tdXYddr8Y2sCCIVC1lJATS0LpEhJSUmTbU1MTFBSUvKK7oIQQghpvxStMBIRESFXz8LCArq6urh9+zbr+O3bt5t8F1EgEKhUX1lubm4AgOvXrzd7nafnlEUjj1rU1CNpVR0+fLjJ9zNfHH5+kbW1dbNrVlpbW79MaIQQQkibpckRMkWPqBXR19fHwIEDcfz4cdmayw0NDTh+/Dhmz56tsI27uzuOHz+OuXPnyo5lZmbC3d39pWJ++vu/S5cusuusXLkSd+7ckc36zszMRIcOHdCnTx+l+6XksR1QZpvBpnC5XJXWrCSEEEJeF6012zosLAxTpkyBi4sLXF1dsX79elRXV8tmX0+ePBlvvfUWoqOjAQBz5szBe++9h9jYWIwaNQq7d+9Gfn4+tm3bJuvz/v37KCkpQWlpKYDGDVSAxhFDgUCA33//Hbt27cJHH32ETp064bfffkNoaCiGDx+O/v37AwC8vLzQp08fTJo0CTExMSgvL8fixYsRHBysVGL8FCWPRKM6nDBquRJRqHK4ehOemiJJs9Fof2+Srt3va7S/kivq/wGoiNXQYo329yapuNittUNolvknv7V2CK/U67oysEgkwt27d7FkyRKUl5fD2dkZR44ckT0tLCkpYe3iNmTIEOzatQuLFy/GokWLYGdnh4MHD7KW3jt06BBrCb3x48cDePbepb6+Po4dOyZLVLt16wY/Pz8sXrxY1kZXVxcZGRmYNWsW3N3dYWxsjClTpqi8qx1Hqs4UYkKaID2m2hqT5BmNJ4/BIzTa35uk6+hzGu0vb8PHGu1v0Iz/aLS/N8mDQs0m8pr2piWPOm6KZx9rSggnreVKStooFWmsr/aORh4JIYQQ8lqiWcHa0a6+19zcXOjq6mLUqFGs41FRUXB2dm6yXXR0NHR1dRXuKJOUlAQzMzOF7TgcDg4ePNjkZ03w8PBgvSCrTFyKpKSkwMnJCUZGRujSpQumT5+Ov/76S6m23333Hd5991107NgRHTt2hKenZ4t7cBJCCCHkzdSukkeGYRASEoITJ07IXhhVRkJCAubPn4+EhAQtRtd6fvnlF0yePBmBgYG4dOkS0tPTkZeX1+zel8/LysqCWCzGzz//jNzcXHTr1g1eXl74888/tRw5IYQQoj0cDRbyTLtJHquqqpCWloZZs2Zh1KhRSEpKAtA4Qrd06VKcP38eHA4HHA5Hdg4AsrOz8ejRIyxbtgyVlZXIycl5ZTF7eHhg9uzZmD17Nvh8PiwsLBAZGanWTjXNyc3NhY2NDb788kv07NkTw4YNw2effab06GFKSgq++OILODs7w9HREdu3b5ctK0AIIYS0VzoaLOSZdvN97NmzB46OjnBwcEBAQAASEhIglUohEokQHh4OoVCIsrIylJWVQSR69lIrwzAQi8XQ09ODWCwGwzCvNO4dO3aAy+UiLy8PGzZsQFxcHLZv367Ra7i7u+OPP/7A4cOHIZVKcfv2bezduxcfffSRWv09fPgQT548gbm5uUbjJIQQQl4lDkdzhTzTbibMMAyDgIAAAICPjw8kEgmys7Ph4eEBExMTcLlcudXRKysrsXfvXtnG4gEBAXj33XexYcMGmJiYyOpJJBLWZ03q1q0b1q1bBw6HAwcHB1y4cAHr1q1jPVLesmWLXEJZV1cHQ0NDpa4xdOhQpKSkQCQS4fHjx6irq8Po0aOxefNmtWJesGABrK2t5TZpJ4QQQghpFyOPxcXFyMvLg1gsBtC48LVIJGpxFDE1NRW2trZwcnICADg7O6NHjx5IS2NP3Tc1NUVhYaFcUdbJkyeb3DJw8ODB4Dz3J4u7uzuuXbuG+vp62bGJEyfKXVuVNZcuX76MOXPmYMmSJTh79iyOHDmCmzdv4vPPP1e6j6dWr16N3bt348CBAy0mrzU1NaisrGSVmtoGla9JCCGEaAM9ttaOdjHyyDAM6urqWFvpSaVSGBgYYNOmTc22u3TpErjcZ7fZ0NCAhIQEBAYGyo7p6Oi81C4sLi4urGSzpS0DX8Tn8+Wu/3TbIGVER0dj6NChmDdvHgCgf//+MDY2xrvvvosVK1bItiVqydq1a7F69WocO3ZMthp9S9d9cZP4JZM6I2qy8rETQggh2kJPm7WjzSePdXV1SE5ORmxsLLy8vFjnfH19kZqaCn19fdZIHgBcuHAB+fn5yMrKYr27d//+fXh4eKCoqAiOjo4aiZHH4zWZfJ45c4b1+fTp07Czs4Ourq5Grg00vqP4fIIMQNa/spNzYmJisHLlShw9ehQuLi5KtYmIiEBYWBjrmP6pwUq1JYQQQkj71OaTx4yMDDx48ACBgYHg8/msc35+fmAYBqGhobhx4wYKCwvRtWtXmJqagmEYuLq6Yvjw4XJ9Dho0CAzDKFz3sSVPr/M8Ozs7GBsbK6xfUlKCsLAwfPbZZzh37hw2btyI2NhYla/bnNGjRyMoKAhbt26Ft7c3ysrKMHfuXLi6urJGa5vyzTffYMmSJdi1axdsbGxQXl4OALLH8E1RtEm8VJ8G9wkhhLQN9BtJO9r898owDDw9PeUSR6AxeczPz4dQKISPjw9GjBiBzp07Y8eOHdi5cyf8/PwU9unn54fk5GQ8efJE5XjCwsIwYMAAVikoKGiy/uTJk/Ho0SO4uroiODgYc+bMwcyZM1W+bnOmTp2KuLg4bNq0CX379oW/vz8cHBywf/9+pdpv3boVtbW1+PTTT9GlSxdZWbt2rUbjJIQQQl4lWudRO2hvay3y8PCAs7Mz1q9f39qhvDK0t7X6aG/rtoP2tn590d7WbYu297aO0NHc3tbRDbS39VNt/rE1IYQQQog6dGjIUCva/GNrAgiFQtZSQE0tC9SUptqamJjg5MmTr+AOCCGEkFePHltrB408alFWVpZG+jl8+HCT72cqsyxQc2tWvvXWW+qGRQghhJA3ECWP7UCPHi/3js7LrGFJCCGEtFf02Fo7KHkkGvXkRqfWDqHdevS9s0b742/+WaP9vUka0lRb6L8lrmHfa7S/Jzdp33l1dRQVtnYIzarY2/IGDa8Tczft9k+5o3ZQ8kgIIYSQ1xKNPGoHTZghhBBCCCFKo5FHQgghhLyWOKClrLWhXY085ubmQldXF6NGjWIdj4qKgrOzs8I2BQUF8Pf3h5WVFQwNDWFnZ4egoCBcvXoVAHDz5k1wOBxYWlri77//ZrV1dnZGVFSU7LOHhwc4HA44HA4MDQ3Rp08fbNmy5aXvy8bGRuFC4s3d14uej+358uJ31RSpVIolS5agS5cu4PF48PT0xLVr11S4C0IIIaRt0eForpBn2lXyyDAMQkJCcOLECZSWlrZYPyMjA4MHD0ZNTQ1SUlJw5coV7Ny5E3w+H5GRkay6f//9t1Lb8QUFBaGsrAyXL1/GuHHjEBwcjNTUVLXvSVP279+PsrIyWbl48SJ0dXXh7++vVPuYmBj861//Qnx8PM6cOQNjY2N4e3vj8ePHWo6cEEIIIe1Ju0keq6qqkJaWhlmzZmHUqFFISkoCACQlJWHp0qU4f/68bLQtKSkJDx8+xLRp0/DRRx/h0KFD8PT0RM+ePeHm5oa1a9fi22+/ZfUfEhKCuLg43Llzp9k4jIyMIBAI8PbbbyMqKgp2dnY4dOhQk/VtbGywfPlyiMViGBsb46233sLmzZtf+vt4kbm5OQQCgaxkZmbCyMhIqeRRKpVi/fr1WLx4McaOHYv+/fsjOTkZpaWlOHjwoMZjJYQQQl4FWiRcO9pN8rhnzx44OjrCwcEBAQEBSEhIgFQqhUgkQnh4OIRCoWzUTSQS4ejRo7h37x7mz5+vsD8zMzPWZ7FYjF69emHZsmUqxcXj8VBbW9tsnTVr1sDJyQkFBQVYuHAh5syZg8zMTJWuoyqGYTB+/HgYGxu3WPfGjRsoLy+Hp6en7Bifz4ebmxtyc3O1GSYhhBCiNTocqcYKeabdJI8MwyAgIAAA4OPjA4lEguzsbPB4PJiYmIDL5cpG3Xg8nux9PUdHR6X653A4WL16NbZt24bff/+9xfr19fXYuXMnfvvtN7z//vvN1h06dCgWLlwIe3t7hISE4NNPP8W6detYdRYsWCC3deCqVauUiv1FeXl5uHjxImbMmKFU/fLycgDyu9VYWVnJzilSU1ODyspKVql50qBWzIQQQghpH9pF8lhcXIy8vDyIxWIAAJfLhUgkAsMwTbaRSlX/K8Hb2xvDhg2Tex/yeVu2bIGJiQl4PB6CgoIQGhqKWbNmISUlpck9o93d3Vl9uLu748qVK6xj8+bNQ2FhIat8/vnnKt8D0Jho9+vXD66urmq1V1Z0dDT4fD6rxPxYotVrEkIIIcricDRXyDPtYqkehmFQV1cHa2tr2TGpVAoDAwNs2rRJYRt7e3sAQFFRkVzy1pzVq1fD3d0d8+bNU3h+4sSJ+Oqrr8Dj8dClSxfo6DTm32PGjIGb27Ol8lXdM9rCwkJuG0Fzc9V3kaiursbu3btVevwuEAgAALdv30aXLl1kx2/fvt3sbO+IiAiEhYWxjnF2Kje7mxBCCNE2Svq0o80nj3V1dUhOTkZsbCy8vLxY53x9fZGamgp9fX3U19ezznl5ecHCwgIxMTE4cOCAXL8VFRVy7z0CgKurKz755BMsXLhQYTx8Pl/hXtGmpqYwNTVV2Ob06dNyn3v37q2w7stKT09HTU2N7BG/Mnr27AmBQIDjx4/LksXKykqcOXMGs2bNarKdgYEBDAwMWMdq9drFYDYhhBBC1NTmk8eMjAw8ePAAgYGB4PP5rHN+fn5gGAahoaG4ceMGCgsL0bVrV5iamsLY2Bjbt2+Hv78/xowZgy+//BK9evXCvXv3sGfPHpSUlGD37t0Kr7ly5UoIhUJwuZr5en755RfExMTA19cXmZmZSE9Pxw8//KCRvl/EMAx8fX3RqZPye0xzOBzMnTsXK1asgJ2dHXr27InIyEhYW1vD19dXK3ESQggh2kYTXbSjzQ8TMQwDT09PucQRaEwe8/PzIRQK4ePjgxEjRqBz586ydRfHjh2LnJwc6OnpYcKECXB0dIRYLIZEIsGKFSuavKa9vT2mT5+usTUOw8PDkZ+fjwEDBmDFihWIi4uDt7e3Rvp+XnFxMU6dOoXAwECV286fPx8hISGYOXMmBg0ahKqqKhw5cgSGhoYaj5MQQgh5FeidR+3gSNWZWUKUZmNjg7lz52Lu3LmtHcorUfvde60dQrt1/2I3jfbHW5Oj0f7eJMZpVi1XUoGO+SON9vfkpurvQ5NGXPf/tXYIzZLs7d/aIbxS5qv2arX/OJNdGusrrGqCxvpq79r8yCMhhBBCCGk72vw7j6SRiYlJk+d+/PFHvPvuu02eP3nyJEaOHNnk+aqqqpeKjRBCCGmLOPTOo1ZQ8qhlN2/e1Eg/hYWFTZ5raVkgFxeXZtsTQgghryN6V1E7KHlsJxQtD6QsHo/3Uu0JIYQQQp6i5JEQQgghryVaqkc7KHkkhBBCyGuJHltrB822JoQQQgghSmtXyWNubi50dXUxahR7/+SoqKgm92AuKCiAv78/rKysYGhoCDs7OwQFBeHq1asAGie0cDgcWFpa4u+//2a1dXZ2RlRUlOyzh4cHOBwOOBwODA0N0adPH2zZsuWl78vGxgbr16+XO97cfb3o+dieLy9+V02JioqCo6MjjI2N0bFjR3h6euLMmTMq3AUhhBDStuhwpBor5Jl2lTwyDIOQkBCcOHECpaWlLdbPyMjA4MGDUVNTg5SUFFy5cgU7d+4En89HZGQkq+7ff/+NtWvXtthnUFAQysrKcPnyZYwbNw7BwcGyHW1a0/79+1FWViYrFy9ehK6uLvz9/ZVqb29vj02bNuHChQs4deoUbGxs4OXlhbt372o5ckIIIUQ7OBypxoqqNm/eDBsbGxgaGsLNzQ15eXnN1k9PT4ejoyMMDQ3Rr18/HD58mHV+//798PLyQqdOncDhcORWUbl//z5CQkLg4OAAHo+H7t2748svv4REInnhO5EfaGpqu+amtJvksaqqCmlpaZg1axZGjRqFpKQkAEBSUhKWLl2K8+fPy76EpKQkPHz4ENOmTcNHH32EQ4cOwdPTEz179oSbmxvWrl2Lb7/9ltV/SEgI4uLicOfOnWbjMDIygkAgwNtvv42oqCjY2dnh0KFDTda3sbHB8uXLIRaLYWxsjLfeegubN29+6e/jRebm5hAIBLKSmZkJIyMjpZPHCRMmwNPTE2+//TaEQiHi4uJQWVmJ3377TeOxEkIIIa+ztLQ0hIWF4euvv8a5c+fg5OQEb2/vJnOMnJwciMViBAYGoqCgAL6+vvD19cXFixdldaqrqzFs2DB88803CvsoLS1FaWkp1q5di4sXLyIpKQlHjhxRuGVxYmIia8DJ19dXpftrN8njnj174OjoCAcHBwQEBCAhIQFSqRQikQjh4eEQCoWyL0EkEuHo0aO4d+8e5s+fr7A/MzMz1mexWIxevXph2bJlKsXF4/FQW1vbbJ01a9bAyckJBQUFWLhwIebMmYPMzEyVrqMqhmEwfvx4GBsbq9y2trYW27ZtA5/Ph5OTkxaiI4QQQrSPo6O5ooq4uDgEBQVh2rRp6NOnD+Lj42FkZISEhASF9Tds2AAfHx/MmzcPvXv3xvLly/HOO+9g06ZNsjqTJk3CkiVL4OnpqbCPvn37Yt++fRg9ejRsbW3x/vvvY+XKlfj+++9RV1fHqmtmZsYacDI0NFTp/tpN8sgwDAICAgAAPj4+kEgkyM7OBo/Hg4mJCbhcruxL4PF4uHbtGgDA0dFRqf45HA5Wr16Nbdu24ffff2+xfn19PXbu3InffvsN77//frN1hw4dioULF8Le3h4hISH49NNPsW7dOladBQsWwMTEhFVWrVqlVOwvysvLw8WLFzFjxgyV2mVkZMDExASGhoZYt24dMjMzYWFhoVYMhBBCSGvT5DuPNTU1qKysZJWamhq5a9bW1uLs2bOsJE9HRweenp7Izc1VGGdubq5cUujt7d1kfWVJJBJ06NABXC57cZ3g4GBYWFjA1dVVNhininaRPBYXFyMvLw9isRgAwOVyIRKJwDBMk21U/SKAxn9Rw4YNk3sf8nlbtmyBiYkJeDwegoKCEBoailmzZiElJYWV+J08eVLWxt3dndWHu7s7rly5wjo2b948FBYWssrnn3+u8j0AjYl2v3794OrqqlK7ESNGoLCwEDk5OfDx8cG4ceOafYyv8D+kJw1qxUwIIYRomibfeYyOjgafz2eV6OhouWveu3cP9fX1sLKyYh23srJCeXm5wjjLy8tVqq+Me/fuYfny5Zg5cybr+LJly7Bnzx5kZmbCz88PX3zxBTZu3KhS3+1inUeGYVBXVwdra2vZMalUCgMDA9aQ7vPs7e0BAEVFRXLJW3NWr14Nd3d3zJs3T+H5iRMn4quvvgKPx0OXLl2go9OYf48ZMwZubm6yei1tGfgiCwsLuV1gzM3NVeoDaHwnYvfu3So/fgcAY2Nj9OrVC7169cLgwYNhZ2cHhmEQERGhsH50dDSWLl3KOrb44+6IHGOj8rUJIYSQtiwiIgJhYWGsYwYGBq0UTfMqKysxatQo9OnTh7VqDADWANmAAQNQXV2NNWvW4Msvv1S6/zafPNbV1SE5ORmxsbHw8vJinfP19UVqair09fVRX1/POufl5QULCwvExMTgwIEDcv1WVFTIvfcIAK6urvjkk0+wcOFChfHw+XyFW/2ZmprC1NRUYZvTp0/Lfe7du7fCui8rPT0dNTU1skf8L6OhoUHhkPxTiv5D4uxUbmkgQgghRNs0uUi4gYGBUsmihYUFdHV1cfv2bdbx27dvQyAQKGwjEAhUqt+cv//+Gz4+PjA1NcWBAwegp6fXbH03NzcsX74cNTU1SifDbf6xdUZGBh48eIDAwED07duXVfz8/MAwDGxsbHDjxg0UFhbi3r17qKmpgbGxMbZv344ffvgBY8aMwbFjx3Dz5k3k5+dj/vz5zT4SXrlyJX766ScUFxdr5B5++eUXxMTE4OrVq9i8eTPS09MxZ84cjfT9IoZh4Ovri06dOindprq6GosWLcLp06dx69YtnD17FtOnT8eff/7Z7GxtAwMDdOjQgVUM9Nr8jxQhhJA3RGus86ivr4+BAwfi+PHjsmMNDQ04fvx4k09C3d3dWfUBIDMzU6Unp0DjiKOXlxf09fVx6NAhpSbCFBYWomPHjiqNorb53/QMw8DT0xN8Pl/unJ+fH/Lz8yEUCuHj44MRI0agc+fOsnUXx44di5ycHOjp6WHChAlwdHSEWCyGRCLBihUrmrymvb09pk+fjsePH2vkHsLDw5Gfn48BAwZgxYoViIuLg7e3t0b6fl5xcTFOnTqlcFp+c3R1dVFUVAQ/Pz/Y29tj9OjR+Ouvv3Dy5EkIhUKNx0kIIYS8zsLCwvDdd99hx44duHLlCmbNmoXq6mpMmzYNADB58mTWK2Fz5szBkSNHEBsbi6KiIkRFRSE/Px+zZ8+W1bl//z4KCwtx+fJlAI2/8wsLC2XvRT5NHKurq8EwDCorK1FeXo7y8nLZ09nvv/8e27dvx8WLF3H9+nVs3boVq1atQkhIiEr3x5GqM7OEKM3GxgZz587F3LlzWzuUV6L2u/daO4R26/7Fbhrtj7cmR6P9vUmM06xarqQCHfNHGu3vyU3V34cmjbju/2vtEJol2du/tUN4pcxX7dVq/wldFC+No47pZdNVqr9p0yasWbMG5eXlcHZ2xr/+9S/Z3AgPDw/Y2NjI1qwGGl87W7x4MW7evAk7OzvExMTgo48+kp1PSkqSJZ/P+/rrrxEVFYWsrCyMGDFCYSw3btyAjY0Njhw5goiICFy/fh1SqRS9evXCrFmzEBQUJJvDoQxKHrWMkkeiLEoe2w5KHl9flDy2LdpOHhOtNZc8TitVLXl8nbX5x9ak0YtrQDa1LJAiJ0+ebLY9IYQQQoiy2vxs6/bu5s2bGunnxT0sn9fSskAuLi7NtieEEEJeR6pMdCHKo+SxnVC0PJCyeDzeS7UnhBBC2iMOJY9aQY+tCSGEEEKI0mjkkRBCCCGvJY4OjTxqAyWPhBBCCHktaXKHGfIMJY+EEEIIeS3RhBntaDfvPObm5kJXVxejRrH3To6KioKzs7PCNgUFBfD394eVlRUMDQ1hZ2eHoKAgXL16FUDjTGgOhwNLS0v8/fffrLbOzs6szcQ9PDzA4XDA4XBgaGiIPn36YMuWLS99XzY2Nli/fr3c8ebu60XPx/Z8efG7as6VK1cwZswY8Pl8GBsbY9CgQSgpKVG6PSGEEELeDO0meWQYBiEhIThx4gRKS0tbrJ+RkYHBgwejpqYGKSkpuHLlCnbu3Ak+n4/IyEhW3b///htr165tsc+goCCUlZXh8uXLGDduHIKDg2VbIbam/fv3o6ysTFYuXrwIXV3dZvelft7vv/+OYcOGwdHREVlZWfjtt98QGRmp1J6YhBBCSFvF4Ug1Vsgz7eKxdVVVFdLS0pCfn4/y8nIkJSVh0aJFSEpKwtKlSwEAnP//YkNiYiLGjRuHadOm4aOPPsKBAwdk/fTs2RNubm6oqKhg9R8SEoK4uDgEBwfD0tKyyTiMjIwgEAgANI4M7tq1C4cOHYJYLFZY38bGBoGBgbh8+TIOHToEMzMzLFq0CMHBwS/zdcgxN2fvNrF7924YGRkpnTx+9dVX+OijjxATEyM7Zmtrq9EYCSGEkFeNkj7taBcjj3v27IGjoyMcHBwQEBCAhIQESKVSiEQihIeHQygUykbdRCIRjh49inv37mH+/PkK+zMzM2N9FovF6NWrF5YtW6ZSXDweD7W1tc3WWbNmDZycnFBQUICFCxdizpw5yMzMVOk6qmIYBuPHj4exsXGLdRsaGvDDDz/A3t4e3t7esLS0hJubGw4ePKjVGAkhhBDSPrWL5JFhGAQEBAAAfHx8IJFIkJ2dDR6PBxMTE3C5XAgEAggEAvB4PFy7dg0A4OjoqFT/HA4Hq1evxrZt2/D777+3WL++vh47d+7Eb7/9hvfff7/ZukOHDsXChQthb2+PkJAQfPrpp1i3bh2rzoIFC+S2DFy1apVSsb8oLy8PFy9exIwZM5Sqf+fOHVRVVWH16tXw8fHBf/7zH/zjH//AJ598guzs7Gbb1tTUoLKyklVqnjSoFTchhBCiafTYWjvafPJYXFyMvLw82aNhLpcLkUgEhmGabCOVqv4v2dvbG8OGDZN7H/J5W7ZsgYmJCXg8HoKCghAaGopZs2YhJSWlyb2m3d3dWX24u7vjypUrrGPz5s1DYWEhq3z++ecq3wPQmGj369cPrq6uStVvaGhM9saOHYvQ0FA4Oztj4cKF+PjjjxEfH99s2+joaPD5fFaJ+ZEm2RBCCGkbdDhSjRXyTJt/55FhGNTV1cHa2lp2TCqVwsDAAJs2bVLYxt7eHgBQVFQkl7w1Z/Xq1XB3d8e8efMUnp84cSK++uor8Hg8dOnSBTo6jbn3mDFj4ObmJqvX0l7TL7KwsJDbPvDF9xiVUV1djd27d6v0+N3CwgJcLhd9+vRhHe/duzdOnTrVbNuIiAiEhYWxjnF2Kj/DmxBCCCHtT5tOHuvq6pCcnIzY2Fh4eXmxzvn6+iI1NRX6+vqor69nnfPy8oKFhQViYmJYE2aeqqiokHvvEQBcXV3xySefYOHChQrj4fP5CveINjU1hampqcI2p0+flvvcu3dvhXVfVnp6OmpqamSP+JWhr6+PQYMGobi4mHX86tWr6NGjR7NtDQwMYGBgwDpWq9fmB7MJIYS8Iehxs3a06eQxIyMDDx48QGBgIPh8Puucn58fGIZBaGgobty4gcLCQnTt2hWmpqYwNjbG9u3b4e/vjzFjxuDLL79Er169cO/ePezZswclJSXYvXu3wmuuXLkSQqEQXK5mvppffvkFMTEx8PX1RWZmJtLT0/HDDz9opO8XMQwDX19fdOrUSaV28+bNg0gkwvDhwzFixAgcOXIE33//PbKysrQSJyGEEPIqcGg8Qyva9NfKMAw8PT3lEkegMXnMz8+HUCiEj48PRowYgc6dO8vWXRw7dixycnKgp6eHCRMmwNHREWKxGBKJBCtWrGjymvb29pg+fToeP36skXsIDw9Hfn4+BgwYgBUrViAuLg7e3t4a6ft5xcXFOHXqFAIDA1Vu+49//APx8fGIiYlBv379sH37duzbtw/Dhg3TeJyEEEIIad84UnVmlxCl2NjYYO7cuZg7d25rh/LK1H73XmuH0G7dv9hNo/3x1uRotL83iXGalUb70zF/pNH+ntxU/Z1o0ojr/r/WDqFZkr39WzuEV8p81V6t9r/fUfHcCHV8UjRbY321d236sTUhhBBCiLronUftoOSxHTAxMWny3I8//oh33323yfMnT57EyJEjmzxfVVX1UrERQgghbRUtsaMdlDxq0c2bNzXST2FhYZPnWloWyMXFpdn2hBBCCCGq0GjyWFdXh9LSUnTv3l2T3b7xFC0PpCwej/dS7QkhhJD2ih5ba4dGZ1tfunQJPXv21GSXhBBCCCFqoe0JtaNNL9VDCCGEEELaFpUeW7/zzjvNnn/0SLPLURBCCCGEqItGDLVDpZHHy5cvo3///hg7dqzC8t572lvjLzc3F7q6uhg1ir13clRUFJydnRW2KSgogL+/P6ysrGBoaAg7OzsEBQXh6tWrABontHA4HFhaWuLvv/9mtXV2dkZUVJTss4eHBzgcDjgcDgwNDdGnTx9s2bLlpe/LxsYG69evlzve3H296PnYni8vfldNuX37NqZOnQpra2sYGRnBx8cH165dU+EuCCGEkLaHoyPVWCHPqDTy2LdvX7i5uWHWrFkKzxcWFuK7777TSGAvYhgGISEhYBgGpaWlsLa2brZ+RkYG/Pz84O3tjZSUFNja2uLOnTtIT09HZGQk0tLSZHX//vtvrF27FkuXLm22z6CgICxbtgwPHz5EcnIygoOD0bFjR4jFYo3co7r279+P2tpa2ee//voLTk5O8Pf3b7GtVCqFr68v9PT08O9//xsdOnRAXFwcPD09cfnyZRgbG2szdEIIIYS0MyqNPA4dOhTFxcVNnjc1NcXw4cNfOqgXVVVVIS0tDbNmzcKoUaOQlJQEAEhKSsLSpUtx/vx52WhbUlISHj58iGnTpuGjjz7CoUOH4OnpiZ49e8LNzQ1r167Ft99+y+o/JCQEcXFxuHPnTrNxGBkZQSAQ4O2330ZUVBTs7Oxw6NChJuvb2Nhg+fLlEIvFMDY2xltvvYXNmze/9PfxInNzcwgEAlnJzMyEkZGRUsnjtWvXcPr0aWzduhWDBg2Cg4MDtm7dikePHsm2eiSEEELaI5owox0qJY8bNmxQ+Ij1KVtbW/z8888vG5OcPXv2wNHREQ4ODggICEBCQgKkUilEIhHCw8MhFApRVlaGsrIyiEQiHD16FPfu3cP8+fMV9mdmZsb6LBaL0atXLyxbtkyluHg8HmvET5E1a9bAyckJBQUFWLhwIebMmYPMzEyVrqMqhmEwfvx4pUYNa2pqAACGhoayYzo6OjAwMMCpU6e0FiMhhBCibZQ8ake7mG3NMAwCAgIAAD4+PpBIJMjOzgaPx4OJiQm4XK5s1I3H48ne13N0dFSqfw6Hg9WrV2Pbtm34/fffW6xfX1+PnTt34rfffsP777/fbN2hQ4di4cKFsLe3R0hICD799FOsW7eOVWfBggUwMTFhlVWrVikV+4vy8vJw8eJFzJgxQ6n6jo6O6N69OyIiIvDgwQPU1tbim2++wf/+9z+UlZWpFQMhhBBCXl9qJ48nT55EQEAA3N3d8eeffwIA/u///k/jo1XFxcXIy8uTvVfI5XIhEonAMEyTbaRS1f9C8Pb2xrBhwxAZGdlknS1btsDExAQ8Hg9BQUEIDQ3FrFmzkJKSwkr8Tp48KWvj7u7O6sPd3R1XrlxhHZs3bx4KCwtZ5fPPP1f5HoDGRLtfv35wdXVVqr6enh7279+Pq1evwtzcHEZGRvj5558xcuRI6Og0/+NRU1ODyspKVql50qBW3IQQQoim0cijdqiVPO7btw/e3t7g8XgoKCiQPfqUSCRqj5g1hWEY1NXVwdraGlwuF1wuF1u3bsW+ffsgkUgUtrG3twcAFBUVqXSt1atXIy0tDQUFBQrPT5w4EYWFhbhx4waqq6sRFxcHHR0djBkzhpX4ubi4qHRdCwsL9OrVi1XMzc1V6gMAqqursXv3bgQGBqrUbuDAgSgsLERFRQXKyspw5MgR/PXXX3j77bebbRcdHQ0+n88qMT+WqBw3IYQQog2UPGqHWsnjihUrEB8fj++++w56enqy40OHDsW5c+c0FlxdXR2Sk5MRGxvLSs7Onz8Pa2trpKamQl9fH/X19ax2Xl5esLCwQExMjMJ+KyoqFB53dXXFJ598goULFyo8z+fz0atXL7z11lusUTlTU1NW4sfj8WTnTp8+zerj9OnT6N27tzK3r7L09HTU1NTIHvGris/no3Pnzrh27Rry8/MxduzYZutHRERAIpGwyvyRtDUlIYSQtkGHI9VYIc+otbd1cXGxwlnVfD6/ycRMHRkZGXjw4AECAwPB5/NZ5/z8/MAwDEJDQ3Hjxg0UFhaia9euMDU1hbGxMbZv3w5/f3+MGTMGX375JXr16oV79+5hz549KCkpwe7duxVec+XKlRAKheByNbPt9y+//IKYmBj4+voiMzMT6enp+OGHHzTS94sYhoGvry86deqkUrv09HR07twZ3bt3x4ULFzBnzhz4+vrCy8ur2XYGBgYwMDBgHavVaxev0RJCCCFETWr9phcIBLh+/brc8VOnTrX4qFMVDMPA09NTLnEEGpPH/Px8CIVC+Pj4YMSIEejcubNseZmxY8ciJycHenp6mDBhAhwdHSEWiyGRSLBixYomr2lvb4/p06fj8ePHGrmH8PBw5OfnY8CAAVixYgXi4uLg7e2tkb6fV1xcjFOnTqn8yBoAysrKMGnSJDg6OuLLL7/EpEmTaJkeQggh7R6Ho7lCnuFI1ZhdEh0djZ07dyIhIQEffvghDh8+jFu3biE0NBSRkZEICQnRRqztjo2NDebOnYu5c+e2diivTO132ttl6HV3/2I3jfbHW5Oj0f7eJMZpVhrtT8dcs1u3Prmp+jvRpBHX/X+tHUKzJHv7t3YIr5T5qr1a7f+462qN9fVBnuJX2t5Eaj2bXbhwIRoaGvDBBx/g4cOHGD58OAwMDPDPf/6TEkdCCCGEkNeYWskjh8PBV199hXnz5uH69euoqqpCnz59YGJioun4CNDs9/rjjz/i3XffbfL8yZMnMXLkyCbPV1VVvVRshBBCSFtFe1Jrx0vNCtHX10efPn00Fctr5+bNmxrpp7CwsMlzb731VrNtXVxcmm1PCCGEvK5oiR3tUGvCzOPHj7FmzRp89NFHcHFxwTvvvMMqRLNeXAOyqWWBFOHxeM22J4QQQojmbd68GTY2NjA0NISbmxvy8vKarZ+eng5HR0cYGhqiX79+OHz4MOv8/v374eXlhU6dOoHD4SgcGHr8+DGCg4PRqVMnmJiYwM/PD7dv32bVKSkpwahRo2BkZARLS0vMmzcPdXV1Kt2bWiOPgYGB+M9//oNPP/0Urq6u4NA0JPLUB7Slobos36rQaH8NGp708SapFt1uuZIKpMudNdpfh7GXNNrfG+VGh9aOoFkdvS+3dgivldYaeUxLS0NYWBji4+Ph5uaG9evXw9vbG8XFxbC0tJSrn5OTA7FYjOjoaHz88cfYtWsXfH19ce7cOfTt2xdA40Ygw4YNw7hx4xAUFKTwuqGhofjhhx+Qnp4OPp+P2bNn45NPPsEvv/wCoHF75VGjRkEgECAnJwdlZWWYPHky9PT0VNrkRa3Z1nw+H4cPH8bQoUNVbUpec7X/tW/tENotblHzo8iqavjLSKP9vUkoeXyN3TJt7QiaxbF42NohvFKc97SbLGcPbXppPlW998tipeu6ublh0KBB2LRpEwCgoaEB3bp1Q0hIiMKNSEQiEaqrq5GRkSE7NnjwYDg7OyM+Pp5V9+bNm+jZsycKCgrg7OwsOy6RSNC5c2fs2rULn376KYDGnfZ69+6N3NxcDB48GD/++CM+/vhjlJaWwsqqcYAhPj4eCxYswN27d6Gvr6/U/an12Pqtt96CqWnb/g+QEEIIIURTampqUFlZySpPt2d+Xm1tLc6ePQtPT0/ZMR0dHXh6eiI3N1dh37m5uaz6AODt7d1kfUXOnj2LJ0+esPpxdHRE9+7dZf3k5uaiX79+ssTx6XUqKytx6ZLyf5SqlTzGxsZiwYIFuHXrljrNCSGEEEK0TpN7W0dHR4PP57NKdHS03DXv3buH+vp6VoIGAFZWVigvL1cYZ3l5uUr1m+pDX18fZmZmTfbT1HWenlOWWsmji4sLHj9+jLfffhumpqYwNzdnFW3Izc2Frq4uRo0axToeFRXFGrZ9XkFBAfz9/WFlZQVDQ0PY2dkhKCgIV69eBdA49MvhcGBpaYm///6b1dbZ2RlRUVGyzx4eHuBwOOBwODA0NESfPn2wZcuWl74vGxsbrF+/Xu54c/elyPr16+Hg4AAej4du3bohNDRU6V1yoqOjMWjQIJiamsLS0hK+vr4oLi5W+tqEEEJIW6TJ5DEiIgISiYRVIiIiWvsWW4VaE2bEYjH+/PNPrFq1ClZWVq9kwgzDMAgJCQHDMCgtLYW1tXWz9TMyMuDn5wdvb2+kpKTA1tYWd+7cQXp6OiIjI5GWliar+/fff2Pt2rVYunRps30GBQVh2bJlePjwIZKTkxEcHIyOHTtCLBZr5B7VtWvXLixcuBAJCQkYMmQIrl69iqlTp4LD4SAuLq7F9tnZ2QgODsagQYNQV1eHRYsWwcvLC5cvX4axsfEruANCCCFE8zQ5YcbAwAAGBgYt1rOwsICurq7cLOfbt29DIBAobCMQCFSq31QftbW1qKioYI0+Pt+PQCCQm/X99LqqXEut5DEnJwe5ublwcnJSp7nKqqqqkJaWhvz8fJSXlyMpKQmLFi1CUlKSLOF7msAmJiZi3LhxmDZtGj766CMcOHBA1k/Pnj3h5uaGiooKVv8hISGIi4tDcHCwwllQTxkZGcm+3KioKOzatQuHDh1qMnm0sbFBYGAgLl++jEOHDsHMzAyLFi1CcHDwy3wdcnJycjB06FBMmDBBdl2xWIwzZ84o1f7IkSOsz0lJSbC0tMTZs2cxfPhwjcZKCCGEvM709fUxcOBAHD9+HL6+vgAaJ8wcP34cs2fPVtjG3d0dx48fZ21nnJmZCXd3d6WvO3DgQOjp6eH48ePw8/MDABQXF6OkpETWj7u7O1auXIk7d+7I8p3MzEx06NBBpXW71Xps7ejoiEePNLtXa3P27NkDR0dHODg4ICAgAAkJCZBKpRCJRAgPD4dQKERZWRnKysogEolw9OhR3Lt3D/Pnz1fY34vvA4jFYvTq1QvLli1TKS4ej4fa2tpm66xZswZOTk4oKCjAwoULMWfOHGRmZqp0nZYMGTIEZ8+elf018d///heHDx/GRx99pFZ/EokEALT2CgIhhBDyKmjysbUqwsLC8N1332HHjh24cuUKZs2aherqakybNg0AMHnyZNYj7zlz5uDIkSOIjY1FUVERoqKikJ+fz0o279+/j8LCQly+3DhDvbi4GIWFhbJ3Ffl8PgIDAxEWFoaff/4ZZ8+exbRp0+Du7o7BgwcDALy8vNCnTx9MmjQJ58+fx9GjR7F48WIEBwcrNar6lFojj6tXr0Z4eDhWrlyJfv36QU9Pj3W+QwfNrqPFMAwCAgIAAD4+PpBIJMjOzoaHhwdMTEzA5XJZw63Xrl0D0JjkKoPD4WD16tUYPXo0QkNDYWtr22z9+vp6pKam4rfffsPMmTObrTt06FDZtHx7e3v88ssvWLduHT788ENZnQULFmDxYvYSALW1tUr/FTBhwgTcu3cPw4YNg1QqRV1dHT7//HMsWrRIqfbPa2howNy5czF06FDZ2lJNqampkZtpxqlpgIGBWn+TEEIIIRrVWtsTikQi3L17F0uWLEF5eTmcnZ1x5MgR2eSUkpIS6Og8+105ZMgQ7Nq1C4sXL8aiRYtgZ2eHgwcPsn4PHzp0SJZ8AsD48eMBAF9//bVsjsa6deugo6MDPz8/1NTUwNvbmzU/Q1dXFxkZGZg1axbc3d1hbGyMKVOmqDx4plby6OPjAwD44IMPWMelUik4HA7q6+vV6Vah4uJi5OXlyR4/c7lciEQiMAwDDw8PhW3UWLoS3t7eGDZsGCIjI7Fr1y6FdbZs2YLt27ejtrYWurq6CA0NxaxZs5CSkoLPPvtMVu/5/aZfHHJ2d3eXmyAzb948TJ06lXXsX//6F06cOKFU7FlZWVi1ahW2bNkCNzc3XL9+HXPmzMHy5csRGRmpVB9PBQcH4+LFizh16lSLdaOjo+XeE138pTki53RS6ZqEEELI62b27NlNPqbOysqSO+bv7w9/f/8m+5s6dapcrvAiQ0NDbN68GZs3b26yTo8ePeR2r1GVWsnjzz///FIXVQXDMKirq2NNkJFKpTAwMJAtvvkie/vGhaqLiopUel9g9erVcHd3x7x58xSenzhxIr766ivweDx06dJF9lfDmDFj4ObmJqvX0n7TL7KwsJDbKlCVR8aRkZGYNGkSZsyYAQDo168fqqurMXPmTHz11Vesv26aM3v2bGRkZODEiRPo2rVri/UjIiIQFhbGOsb5k7anJIQQ0jbQ3tbaoVby+N5772k6DoXq6uqQnJyM2NhYeHl5sc75+voiNTUV+vr6ciOdXl5esLCwQExMDGvCzFMvzkR6ytXVFZ988onC1d+BxvcJFO0HbWpq2uSi6adPn5b73Lt3b4V11fXw4UO5BFFXVxeAcqOwUqkUISEhOHDgALKystCzZ0+lrqto5lntPXpkTQghpG2g5FE71Eoen3r48CFKSkrkJo3079//pYJ6KiMjAw8ePEBgYCD4fD7rnJ+fHxiGQWhoKG7cuIHCwkJ07doVpqamMDY2xvbt2+Hv748xY8bgyy+/RK9evXDv3j3s2bMHJSUl2L17t8Jrrly5EkKhEFzuS301Mr/88gtiYmLg6+uLzMxMpKen44cfftBI30+NHj0acXFxGDBggOyxdWRkJEaPHi1LIpsTHByMXbt24d///jdMTU1ZL9/yeJrdMo8QQggh7ZtaGdLdu3cxbdo0/PjjjwrPa+qdR4Zh4OnpKZc4Ao3JY0xMDIRCIXx8fDBixAhUVFQgMTERU6dOxdixY5GTk4Po6GhMmDABlZWV6NatG95//32sWNH0Xpf29vaYPn06tm3bppF7CA8PR35+PpYuXYoOHTogLi4O3t7eGun7qcWLF4PD4WDx4sX4888/0blzZ4wePRorV65Uqv3WrVsBQO4d0qffJSGEENIe0cijdnCkaswumThxIm7duoX169fDw8MDBw4cwO3bt7FixQrExsbK7QLzprKxscHcuXNZ6za97mr/a9/aIbRb3CLNjvI2/GWk0f7eJNWi2y1XUoF0ubNG++swVvk9aMkLbil+xait4Fg8bO0QXinOe5e12n/eh19prC/XTOUGZN4Eao08/vTTT/j3v/8NFxcX6OjooEePHvjwww/RoUMHREdHU/JICCGEEPKaUmt2Q3V1tWxl8o4dO+Lu3bsAGmf5njt3TnPREQCAiYlJk+XkyZPNti0pKWm2fUlJySu6C0IIIeTV4nA0V8gzao08Ojg4oLi4GDY2NnBycsK3334LGxsbxMfHo0uXLpqOsd26efOmRvopLCxs8lxLywJZW1s3276lPcIJIYSQ9oreedQOtZLHOXPmoKysDEDjyuY+Pj5ISUmBvr4+kpKSNBkfARQuD6QsLpf7Uu1VdX+D6yu71uum5HIPjfbnEqx4Qhtp2ZNIF432p7c8X6P93Q4fqtH+3iRmPTX7PqumXf+/4a0dwivVT8sr/1HyqB1qJY9PtwoEGjfivnXrFoqKitC9e3dYWFhoLDhCCCGEENK2aGQxQyMjI7zzDu0sQgghhJA2pJX2tn7dqZU81tfXIykpCcePH8edO3fQ0NDAOv/TTz9pJDhCCCGEEHXRY2vtUGu29Zw5czBnzhzU19ejb9++cHJyYpWXkZubC11dXbnlfqKiouDs7CxX//z58xgzZgwsLS1haGgIGxsbiEQi3LlzR65udHQ0dHV1sWbNGrlzSUlJCrcsfOrEiRMYPXo0rK2tweFwcPDgQVVvrUlN9Td16lT4+voq1YeNjQ04HI5cCQ4OVqr948ePERwcjE6dOsHExAR+fn64fbttvxtECCGEkFdPrZHH3bt3Y8+ePfjoo480HQ8YhkFISAgYhkFpaWmzs4Hv3r2LDz74AB9//DGOHj0KMzMz3Lx5E4cOHUJ1dbVc/YSEBMyfPx8JCQmYN2+eSnFVV1fDyckJ06dPxyeffKLyfWnbr7/+ytrZ5+LFi/jwww/h7++vVPvQ0FD88MMPSE9PB5/Px+zZs/HJJ5/gl19+0VbIhBBCiFbRyKN2qJU86uvra2UGb1VVFdLS0pCfn4/y8nIkJSVh0aJFSEpKwtKlSwE0jtIBjVvnmZmZQSKRYPv27bK9qHv27IkRI0bI9Z2dnY1Hjx5h2bJlSE5ORk5ODoYMGaJ0bCNHjsTIkSNVvicOh4MtW7bg0KFDyMrKQpcuXRATE4NPP/1U5b6a07lzZ9bn1atXw9bWFu+91/JUNolEAoZhsGvXLrz//vsAGr/f3r174/Tp0xg8eLBGYyWEEEJeBUoetUOtx9bh4eHYsGED1NjZsFl79uyBo6MjHBwcEBAQgISEBEilUohEIoSHh0MoFKKsrAxlZWUQiUQQCASoq6vDgQMHWoyFYRiIxWLo6elBLBaDYRiNxt6cyMhI+Pn54fz585g4cSLGjx+PK1euaO16tbW12LlzJ6ZPny5Ltptz9uxZPHnyBJ6enrJjjo6O6N69O3Jzc7UWJyGEEELaH7VGHk+dOoWff/4ZP/74I4RCIfT09Fjn9+/fr1YwDMPIlgHy8fGBRCJBdnY2PDw8YGJiAi6XC4FAIKs/ePBgLFq0CBMmTMDnn38OV1dXvP/++5g8eTKsrKxk9SorK7F3715ZIhQQEIB3330XGzZsgImJiVqxqsLf3x8zZswAACxfvhyZmZnYuHEjtmzZIqsjFouhq6vLaldTU6PWVo8HDx5ERUUFpk6dqlT98vJy6Ovry73zaWVlhfLycpWvTwghhLQFNPKoHWqNPJqZmeEf//gH3nvvPVhYWIDP57OKOoqLi5GXlwexWAygcXFrkUjU4gjhypUrUV5ejvj4eAiFQsTHx8PR0REXLlyQ1UlNTYWtra1sMo+zszN69OiBtLQ0tWJVZNWqVU1u++fu7s6q6+7uLjfyuG7dOhQWFrLKmDFj1IqFYRiMHDlS67vH1NTUoLKyklVq6upbbkgIIYS8AhyOVGOFPKPWyGNiYqKm4wDDMKirq2MlPFKpFAYGBti0aVOzbTt16gR/f3/4+/tj1apVGDBgANauXYsdO3bI+r506ZLsvUgAaGhoQEJCAgIDAzUS/+eff45x48bJPquauAkEArn3SE1NTVFRUaFSP7du3cKxY8dUGv0VCASora1FRUUFa/Tx9u3brJHeF0VHR8veRX0q3LUf/jm4v0oxE0IIIaT9UHuR8Lq6OmRlZeH333/HhAkTYGpqitLSUnTo0EHlR8F1dXVITk5GbGwsvLy8WOd8fX2RmpoKfX191mzipujr68PW1lY22/rChQvIz89HVlYWzM3NZfXu378PDw8PFBUVwdHRUaV4FTE3N2f1/7zTp09j8uTJrM8DBgx46WsqkpiYCEtLS5Uedw8cOBB6eno4fvw4/Pz8ADSOBJeUlMiNmj4vIiICYWFhrGMPIj5TL3BCCCFEw2jEUDvUSh5v3boFHx8flJSUoKamBh9++CFMTU3xzTffoKamBvHx8Sr1l5GRgQcPHiAwMFDusbefnx8YhkFoaChu3LiBwsJCdO3aFaampsjMzMTu3bsxfvx42NvbQyqV4vvvv8fhw4dlo6MMw8DV1RXDh8vvFzpo0CAwDCNb97G+vh6FhYWsOgYGBujduzeqqqpw/fp12fGnsZibm6N79+7N3l96ejpcXFwwbNgwpKSkIC8vTysTdhoaGpCYmIgpU6awRllbwufzERgYiLCwMJibm6NDhw4ICQmBu7t7szOtDQwMYGBgwDr2kKvbRG1CCCHk1eKo9XIeaYlayeOcOXPg4uKC8+fPo1OnTrLj//jHPxAUFKRyfwzDwNPTU+H7kn5+foiJiYFQKISPjw9GjBiBiooKJCYmYvjw4TAyMkJ4eDj++OMPGBgYwM7ODtu3b8ekSZNks44XLFig8Lp+fn6IjY3FqlWrADQuFfTiiKCtrS2uX7+O/Px81hJAT0fcpkyZgqSkpGbvb+nSpdi9eze++OILdOnSBampqejTp48qX5FSjh07hpKSEkyfPl3ltuvWrYOOjg78/PxQU1MDb29v1oQeQgghpL2hkUft4EjVWG+nU6dOyMnJgYODA0xNTXH+/Hm8/fbbuHnzJvr06YOHDx9qI9Z2icPh4MCBA0rvFNPelc8JaO0Q2q2Syz002p9L8I8a7e9Ncj9Xs+vY6i3P12h/D8OHarS/N4lZz7a9c9b1E31bO4RXqt/BOK32f8kvVGN9Cfet01hf7Z1aI48NDQ0K3z/83//+B1NT05cOihBCCCHkZdHIo3ao9TaAl5cX1q9fL/vM4XBQVVWFr7/+WitbFr7pSkpKWMsAvVieXxZIkZSUlCbbCoXCV3QXhBBCyCvGkWquEBm1Rh5jY2Ph7e2NPn364PHjx5gwYQKuXbsGCwsLpKamajrGdk0Tu/BYW1vLTeR58XxzxowZAzc3N4XnXlzgnRBCCCGkOWolj127dsX58+eRlpaG8+fPo6qqCoGBgZg4cSJ4PJ6mY3zjcbncl9pL3NTUlF4nIIQQ8sahx9baodaEGUKaIj32Zr3srUkNfxlptL8n5ert9kQA/cF/aLS/O7tcNNofb02ORvt7k+httGvtEJpl0PVBa4fwSumK8rTaf5EoRGN9OaZt1Fhf7Z1a7zzu2LEDP/zwg+zz/PnzYWZmhiFDhuDWrVsaC44QQgghhLQtaiWPq1atkj2ezs3NxaZNmxATEwMLCwuEhmpuWjwhhBBCiLpob2vtUOudxz/++EP2Dt7Bgwfx6aefYubMmRg6dCg8PDw0GR8hhBBCiFoo6dMOtUYeTUxM8NdffwEA/vOf/+DDDz8EABgaGuLRo0cvFVBubi50dXXl9maOioqCs7OzXP3z589jzJgxsLS0hKGhIWxsbCASiXDnzh25utHR0dDV1ZVtR/i8pKQkmJmZNRnXiRMnMHr0aFhbW4PD4eDgwYOq3lqTmupv6tSpSi8ubmNjAw6HI1eCg4OVav/ZZ5/B1tYWPB4PnTt3xtixY1FUVKTCXRBCCCHkTaBW8vjhhx9ixowZmDFjBq5evSpb2/HSpUuwsbF5qYAYhkFISAhOnDiB0tLSZuvevXsXH3zwAczNzXH06FFcuXIFiYmJsLa2RnV1tVz9hIQEzJ8/HwkJCSrHVV1dDScnJ2zevFnltq/Cr7/+irKyMlnJzMwEAPj7+yvVfuDAgUhMTMSVK1dw9OhRSKVSeHl5KVwMnhBCCGkPODqaK+QZtR5bb968GYsXL8Yff/yBffv2yfa3Pnv2LMRisdrBVFVVIS0tDfn5+SgvL0dSUhIWLVqEpKQkLF26FEDjKB0AJCYmwszMDBKJBNu3bweX23grPXv2ZO1B/VR2djYePXqEZcuWITk5GTk5ORgyZIjSsY0cORIjR45U+Z44HA62bNmCQ4cOISsrC126dEFMTAw+/fRTlftqTufOnVmfV69eDVtbW7z33ntKtZ85c6bs/9vY2GDFihVwcnLCzZs3YWtrq9FYCSGEkFeCHltrhVq5tJmZGTZt2oR///vf8PHxkR1funQpvvrqK7WD2bNnDxwdHeHg4ICAgAAkJCRAKpVCJBIhPDwcQqFQNrImEokgEAhQV1eHAwcOtLgYN8MwEIvF0NPTg1gsBsMwasepqsjISPj5+eH8+fOYOHEixo8fjytXrmjterW1tdi5cyemT58uS7ZVUV1djcTERPTs2RPdunXTQoSEEEKI9tGEGe14qYHYhw8foqioCL/99hurqIthGAQEBAAAfHx8IJFIkJ2dDR6PBxMTE3C5XAgEAggEAvB4PAwePBiLFi3ChAkTYGFhgZEjR2LNmjW4fZu98X1lZSX27t0r6zsgIAB79uxBVVWV+jevAn9/f8yYMQP29vZYvnw5XFxcsHEje70osVgst3VgSkqKWtc7ePAgKioqMHXqVJXabdmyRXbtH3/8EZmZmdDX12+yfk1NDSorK1mlprZBrZgJIYQQ0j6olTzevXsXo0aNgqmpKYRCIQYMGMAq6iguLkZeXp7ssTeXy4VIJGpxhHDlypUoLy9HfHw8hEIh4uPj4ejoiAsXLsjqpKamwtbWFk5OTgAAZ2dn9OjRA2lpaWrFqsiqVaua3G/a3d2dVdfd3V1u5HHdunUoLCxklTFjxqgVC8MwGDlyZIvbFr5o4sSJKCgoQHZ2Nuzt7TFu3Dg8fvy4yfrR0dHg8/msEr37nloxE0IIIZpGI4/aoVbyOHfuXEgkEpw5cwY8Hg9HjhzBjh07YGdnh0OHDqkVCMMwqKurg7W1NbhcLrhcLrZu3Yp9+/ZBIpE027ZTp07w9/fH2rVrceXKFVhbW2Pt2rWsvi9duiTrl8vl4vLly2pNnGnK559/zkr8VE3cBAIBevXqxSrqbCl469YtHDt2DDNmzFC5LZ/Ph52dHYYPH469e/eiqKgIBw4caLJ+REQEJBIJq0SMt1D5uoQQQog2UPKoHWpNmPnpp5/w73//Gy4uLtDR0UGPHj3w4YcfokOHDoiOjpZbZqcldXV1SE5ORmxsLLy8vFjnfH19kZqaCn19faVm/urr68PW1lY22/rChQvIz89HVlYWzM3NZfXu378PDw8PFBUVwdHRUaV4FTE3N2f1/7zTp09j8uTJrM/qjtC2JDExEZaWlir/O3iRVCqFVCpFTU1Nk3UMDAxgYGDAbqdPU9IIIYSQ15lav+mrq6thaWkJAOjYsSPu3r0LAOjXrx/OnTuncn8ZGRl48OABAgMD0bdvX1bx8/MDwzCwsbHBjRs3UFhYiHv37qGmpgYZGRkICAhARkYGrl69iuLiYqxduxaHDx/G2LFjATSOOrq6umL48OGsfocPH45BgwaxHovX19fLPTp++ni5qqpKdgyALJbnH083JT09HQkJCbh69Sq+/vpr5OXlYfbs2Sp/Ty1paGhAYmIipkyZIpt9roz//ve/iI6OxtmzZ1FSUoKcnBz4+/uDx+PJlmEihBBC2pvWHHncvHkzbGxsYGhoCDc3N+TlNb+Pd3p6OhwdHWFoaIh+/frh8OHDrPNSqRRLlixBly5dwOPx4OnpiWvXrsnOZ2VlKVzvmcPh4NdffwUA3Lx5U+H506dPq3RvaiWPDg4OKC4uBgA4OTnh22+/xZ9//on4+Hh06dJF5f4YhoGnpyf4fL7cOT8/P+Tn50MoFMLHxwcjRoxA586dkZqaij59+sDIyAjh4eFwdnbG4MGDsWfPHmzfvh2TJk2SzTr28/NTeF0/Pz8kJyfjyZMnABoTxBff3xw9ejQAID8/n/VOZ1hYGAYMGIAlS5a0eH9Lly7F7t270b9/fyQnJ8ti17Rjx46hpKQE06dPV6mdoaEhTp48iY8++gi9evWCSCSCqakpcnJyZH8kEEIIIe2OjlRzRQVpaWkICwvD119/jXPnzsHJyQne3t4KNzABgJycHIjFYgQGBqKgoAC+vr7w9fXFxYsXZXViYmLwr3/9C/Hx8Thz5gyMjY3h7e0tm5swZMgQ1nrPZWVlmDFjBnr27AkXFxfW9Y4dO8aqN3DgQJXujyNtaY0bBXbu3Im6ujpMnToVZ8+ehY+PD/766y/o6+tjx44dEIlEqnb52uJwODhw4IDSO8W0d9JjfVs7hHar4S8jjfb3pFz+jzGiHP3Bf2i0vzu7XFqupALemhyN9vcm0dto19ohNMug64PWDuGV0hU1Pxr3sm5MD9JYXz0TvlO6rpubGwYNGoRNmzYBaHwy2K1bN4SEhGDhwoVy9UUiEaqrq5GRkSE7NnjwYDg7OyM+Ph5SqRTW1tYIDw/HP//5TwCARCKBlZUVkpKSMH78eLk+nzx5grfeegshISGIjIwE0Djy2LNnTxQUFCjctU9Zao08BgQEyJaBeeedd3Dr1i3k5+fjf//7HyWOhBBCCGkTNPnYWuHydArmBdTW1uLs2bPw9PSUHdPR0YGnpydyc3MVxpmbm8uqDwDe3t6y+jdu3EB5eTmrDp/Ph5ubW5N9Hjp0CH/99RemTZsmd+7pts7Dhg1Ta6Kz2rMbGIZB3759YWhoiI4dO2Ly5Mka3e+ZPFNSUiK3BmRTywIpkpKS0mRboVD4iu6CEEIIebU4HM0VhcvTRUfLXfPevXuor6+HlZUV67iVlRXKy8sVxlleXt5s/af/VKVPhmHg7e2Nrl27yo6ZmJggNjYW6enp+OGHHzBs2DD4+vqqnECqNdt6yZIliIuLQ0hIiGwNw9zcXISGhqKkpATLli1Tp9vXkhpvBcixtraWTdRp6nxzxowZAzc3N4Xn9PT0XiY0Qggh5I0QERGBsLAw1rEXVxxpK/73v//h6NGj2LNnD+u4hYUF6x4GDRqE0tJSrFmzRqW1pdVKHrdu3YrvvvuOtY/1mDFj0L9/f4SEhFDyqGFcLhe9evVSu72pqalaa0YSQggh7Zkm12dUtDydIhYWFtDV1ZXb7e727dsQCAQK2wgEgmbrP/3n7du3WROTb9++rfDdxcTERHTq1EmphNDNzQ2ZmZkt1nueWo+tnzx5IjdzBwAGDhyIuro6dbokhBBCCNGo1liqR19fHwMHDsTx48dlxxoaGnD8+HG5Heeecnd3Z9UHgMzMTFn9nj17QiAQsOpUVlbizJkzcn1KpVIkJiZi8uTJSj1dLCwsVHmlHLVGHidNmoStW7ciLi6OdXzbtm2YOHGiOl0SQgghhGhWK+0MExYWhilTpsDFxQWurq5Yv349qqurZZNXJk+ejLfeekv2zuScOXPw3nvvITY2FqNGjcLu3buRn5+Pbdu2Nd4Gh4O5c+dixYoVsLOzQ8+ePREZGQlra2u51Vx++ukn3LhxQ+FOczt27IC+vr5s2cH9+/cjISEB27dvV+n+lE4en39GzuFwsH37dvznP//B4MGDAQBnzpxBSUkJaycVQgghhJA3jUgkwt27d7FkyRKUl5fD2dkZR44ckU14KSkpgY7Os4e/Q4YMwa5du7B48WIsWrQIdnZ2OHjwIPr2fbb83fz581FdXY2ZM2eioqICw4YNw5EjR2BoaMi6NsMwGDJkSJO75y1fvhy3bt0Cl8uFo6Mj0tLS8Omnn6p0f0o/ti4oKJCVCxcuYODAgejcuTN+//13/P7777CwsMA777yDS5cuqRTAi3Jzc6Grqyu3vV5UVJTC5/rnz5+XTTk3NDSEjY0NRCKRwoU4o6OjoaurizVr1sidS0pKgpmZWZNxnThxAqNHj4a1tTU4HI5GZ5Y31d/UqVOVXh/SxsZG4arxwcHBSseRm5uL999/H8bGxujQoQOGDx+OR48eKd2eEEIIaUs4OlKNFVXNnj0bt27dQk1NDc6cOcOauJqVlYWkpCRWfX9/fxQXF6OmpgYXL16U2+GNw+Fg2bJlKC8vx+PHj3Hs2DHY29vLXXfXrl345ZdfFMY0ZcoUXL58GdXV1ZBIJDhz5ozKiSOgwsjjzz//rHLn6mAYBiEhIWAYBqWlpc3OJL579y4++OADfPzxxzh69CjMzMxw8+ZNHDp0SLa39fMSEhIwf/58JCQkYN68eSrFVV1dDScnJ0yfPh2ffPKJyvelbb/++itr7++LFy/iww8/hL+/v1Ltc3Nz4ePjg4iICGzcuBFcLhfnz59n/WVECCGEtCeanDBDnlHrnUdtqaqqQlpaGvLz81FeXo6kpCQsWrQISUlJWLp0KYDGzBtonElkZmYGiUSC7du3y/Zy7tmzJ0aMGCHXd3Z2Nh49eoRly5YhOTkZOTk5GDJkiNKxjRw5EiNHjlT5njgcDrZs2YJDhw4hKysLXbp0QUxMjFqZfnM6d+7M+rx69WrY2trivffeU6p9aGgovvzyS9bK9w4ODhqNkRBCCCHtX5saVtqzZw8cHR3h4OCAgIAAJCQkQCqVQiQSITw8HEKhULYPo0gkgkAgQF1dHQ4cONDieooMw0AsFkNPTw9isRgMw7yiuwIiIyPh5+eH8+fPY+LEiRg/fjyuXLmites93dN7+vTpsmS7OXfu3MGZM2dgaWmJIUOGwMrKCu+99x5OnTqltRgJIYQQbWuN2dZvgjaVPDIMg4CAAACAj48PJBIJsrOzwePxYGJiAi6XC4FAAIFAAB6Ph8GDB2PRokWYMGECLCwsMHLkSKxZs0ZuraTKykrs3btX1ndAQAD27NmDqqqqV3Jf/v7+mDFjBuzt7bF8+XK4uLhg48aNrDpisVhu95eUlBS1rnfw4EFUVFTItpBsyX//+18Aje+VBgUF4ciRI3jnnXfwwQcf4Nq1a2rFQAghhLQ6jgYLkWkzyWNxcTHy8vJkC49zuVyIRKIWRwhXrlyJ8vJyxMfHQygUIj4+Ho6Ojrhw4YKsTmpqKmxtbeHk5AQAcHZ2Ro8ePZCWlqax+FetWtXkloEvrsHk7u4uN/K4bt06FBYWsooqq70/j2EYjBw5ssWdZ55qaGgAAHz22WeYNm0aBgwYgHXr1sHBwQEJCQlNtlO4z2dtg1oxE0IIIaR9aDPJI8MwqKurg7W1NbhcLrhcLrZu3Yp9+/ZBIpE027ZTp07w9/fH2rVrceXKFVhbW2Pt2rWsvi9duiTrl8vl4vLly80mRqr6/PPPWYmfsonbUwKBAL169WIVdXaFuXXrFo4dO6ZwfaemPF0ctE+fPqzjvXv3bnbfbIX7fO6+p3LMhBBCiDbQY2vtaBMTZurq6pCcnIzY2Fh4eXmxzvn6+iI1NRX6+vqs2cRN0dfXh62trWy29YULF5Cfn4+srCyYm5vL6t2/fx8eHh4oKipqci0kVZibm7P6f97p06dZ61+ePn1atkCnpiUmJsLS0lJuqaPm2NjYwNraGsXFxazjV69ebXaSkKJ9PvVPDVYtYEIIIURL1Flih7SsTSSPGRkZePDgAQIDA8Hn81nn/Pz8wDAMQkNDcePGDRQWFqJr164wNTVFZmYmdu/ejfHjx8Pe3h5SqRTff/89Dh8+jMTERACNo46urq4YPny43HUHDRoEhmFk6z7W19ejsLCQVcfAwAC9e/dGVVUVrl+/Ljv+NBZzc3N079692ftLT0+Hi4sLhg0bhpSUFOTl5Wllwk5DQwMSExMxZcoU2exzZXA4HMybNw9ff/01nJyc4OzsjB07dqCoqAh79+5tsp2ifT6l+m1mMJsQQgghWtAmkkeGYeDp6SmXOAKNyWNMTAyEQiF8fHwwYsQIVFRUIDExEcOHD4eRkRHCw8Pxxx9/wMDAAHZ2dti+fTsmTZokm3W8YMEChdf18/NDbGwsVq1aBaBxqaAXRwRtbW1x/fp15Ofns5YAejriNmXKFLmFPl+0dOlS7N69G1988QW6dOmC1NRUuUfEmnDs2DGUlJRg+vTpKredO3cuHj9+jNDQUNy/fx9OTk7IzMyEra2txuMkhBBCXgl63KwVHGlLa9yQl8LhcHDgwAGld4pp76TH+rZciSjU8JeRRvt7Ui7/xxhRjv7gPzTa351dLhrtj7cmR6P9vUn0Ntq1dgjNMuj6oLVDeKV0RXla7f926ASN9WW1bpfG+mrv2sTIIyGEEEKIptFEF+2gF9TagZKSErk1IJtaFkiRlJSUJtsKhcJXdBeEEEIIeR3QyKOWaeKtAGtra7mJPC+eb86YMWNYG7I/T09P72VCI4QQQtouGnnUCkoe2wEul4tevXqp3d7U1FStNSMJIYSQ9oxDz1e1gr5WQgghhBCiNBp5JIQQQshriSbMaAclj4QQQgh5PVHyqBVt7rF1bm4udHV15bbXi4qKgrOzs1z98+fPY8yYMbC0tIShoSFsbGwgEolw584dubrR0dHQ1dWV7SjzvKSkJJiZmTUZ14kTJzB69GhYW1uDw+Hg4MGDqt5ak5rqb+rUqUqvD2ljYwMOhyNXgoODlWr/+++/4x//+Ac6d+6MDh06YNy4cbh9+7YKd0EIIYSQN0GbSx4ZhkFISAhOnDiB0tLSZuvevXsXH3zwAczNzXH06FFcuXIFiYmJsLa2lu1t/byEhATMnz8fCQkJKsdVXV0NJycnbN68WeW2r8Kvv/6KsrIyWcnMzAQA+Pv7t9i2uroaXl5e4HA4+Omnn/DLL7+gtrYWo0ePRkNDg7ZDJ4QQQrSCw5FqrJBn2tRj66qqKqSlpSE/Px/l5eVISkrCokWLkJSUhKVLlwJoHKUDgMTERJiZmUEikWD79u2yvZx79uzJ2kbwqezsbDx69AjLli1DcnIycnJyMGTIEKVjGzlyJEaOHKnyPXE4HGzZsgWHDh1CVlYWunTpgpiYGHz66acq99Wczp07sz6vXr0atra2eO+991ps+8svv+DmzZsoKChAhw4dAAA7duxAx44d8dNPP8HT01OjsRJCCCGvAkeHkj5taFMjj3v27IGjoyMcHBwQEBCAhIQESKVSiEQihIeHQygUykbWRCIRBAIB6urqcODAgRbXU2QYBmKxGHp6ehCLxWAY5hXdFRAZGQk/Pz+cP38eEydOxPjx43HlyhWtXe/pnt7Tp0+XJdvNqampAYfDgYGBgeyYoaEhdHR0cOrUKa3FSQghhJD2p00ljwzDICAgAADg4+MDiUSC7Oxs8Hg8mJiYgMvlQiAQQCAQgMfjYfDgwVi0aBEmTJgACwsLjBw5EmvWrJF7V6+yshJ79+6V9R0QEIA9e/agqqrqldyXv78/ZsyYAXt7eyxfvhwuLi7YuHEjq45YLJbb/SUlJUWt6x08eBAVFRWYOnWqUvUHDx4MY2NjLFiwAA8fPkR1dTX++c9/or6+HmVlZWrFQAghhLQ6jlRzhci0meSxuLgYeXl5EIvFABoXxhaJRC2OEK5cuRLl5eWIj4+HUChEfHw8HB0dceHCBVmd1NRU2NrawsnJCQDg7OyMHj16IC0tTWPxr1q1qsktA93d3Vl13d3d5UYe161bh8LCQlYZM2aMWrEwDIORI0e2uPPMU507d0Z6ejq+//57mJiYgM/no6KiAu+88w50dJr+EampqUFlZSWr1NTSO5KEEELaBnrnUTvazDuPDMOgrq6OlfBIpVIYGBhg06ZNzbbt1KkT/P394e/vj1WrVmHAgAFYu3YtduzYIev70qVLsvciAaChoQEJCQkIDAzUSPyff/45xo0bJ/usbOL2lEAgkNtFxtTUFBUVFSr1c+vWLRw7dgz79+9XqZ2Xlxd+//133Lt3D1wuF2ZmZhAIBHj77bebbBMdHS17F/WpJZM6I2qypUrXJoQQQrRBiTe3iBraRPJYV1eH5ORkxMbGwsvLi3XO19cXqamp0NfXR319fYt96evrw9bWVjbb+sKFC8jPz0dWVhbMzc1l9e7fvw8PDw8UFRXB0dHxpe/B3Nyc1f/zTp8+jcmTJ7M+Dxgw4KWvqUhiYiIsLS3lljpSloWFBQDgp59+wp07d5od/YyIiEBYWBjrmP6pwWpdlxBCCCHtQ5tIHjMyMvDgwQMEBgaCz+ezzvn5+YFhGISGhuLGjRsoLCxE165dYWpqiszMTOzevRvjx4+Hvb09pFIpvv/+exw+fBiJiYkAGkcdXV1dMXz4cLnrDho0CAzDyNZ9rK+vR2FhIauOgYEBevfujaqqKly/fl12/Gks5ubm6N69e7P3l56eDhcXFwwbNgwpKSnIy8vTyoSdhoYGJCYmYsqUKaxRVmUkJiaid+/e6Ny5M3JzczFnzhyEhobCwcGhyTYGBgasSTYAINVvM29CEEIIedPRbGutaBPJI8Mw8PT0lEscgcbkMSYmBkKhED4+PhgxYgQqKiqQmJiI4cOHw8jICOHh4fjjjz9gYGAAOzs7bN++HZMmTZLNOl6wYIHC6/r5+SE2NharVq0C0LhU0Isjgra2trh+/Try8/NZSwA9HXGbMmUKkpKSmr2/pUuXYvfu3fjiiy/QpUsXpKamok+fPqp8RUo5duwYSkpKMH36dJXbFhcXIyIiAvfv34eNjQ2++uorhIaGajxGQggh5FWhdxW1gyNtaY0b8lI4HA4OHDig9E4x7Z30WN/WDqHdavjLSKP9PSmX/2OMKEd/8B8a7e/OLheN9sdbk6PR/t4kehvtWjuEZhl0fdDaIbxSuqI8rfYvifLVWF/8qIMa66u9axMjj4QQQgghGkcjj1pBL6i1AyUlJXJrQDa1LJAiKSkpTbYVCoWv6C4IIYSQV4uW6tEOGnnUMk28FWBtbS03kefF880ZM2YM3NzcFJ7T09N7mdAIIYQQ8oah5LEd4HK5cmtAqsLU1BSmpqYajIgQQghpB2i2tVZQ8kgIIYSQ1xI9btYOSh4JIYQQ8lqiHWa0gybMEEIIIYQQpdHIIyGEEEJeT/TOo1a0uZHH3Nxc6Orqyu3NHBUVBWdnZ7n658+fx5gxY2BpaQlDQ0PY2NhAJBLhzp07cnWjo6Ohq6sr247weUlJSTAzM2syrhMnTmD06NGwtrYGh8PBwYMHVb21JjXV39SpU5VeXLy+vh6RkZHo2bMneDwebG1tsXz5cqVmez958gQLFixAv379YGxsDGtra0yePBmlpaUq3gkhhBDSdtBSPdrR5pJHhmEQEhKCEydOtJi83L17Fx988AHMzc1x9OhRXLlyBYmJibC2tkZ1dbVc/YSEBMyfPx8JCQkqx1VdXQ0nJyds3rxZ5bavwjfffIOtW7di06ZNuHLlCr755hvExMRg48aNLbZ9+PAhzp07h8jISJw7dw779+9HcXExxowZ8woiJ4QQQkh70qYeW1dVVSEtLQ35+fkoLy9HUlISFi1ahKSkJCxduhRA4ygdACQmJsLMzAwSiQTbt28Hl9t4Kz179mTtQf1UdnY2Hj16hGXLliE5ORk5OTkYMmSI0rGNHDkSI0eOVPmeOBwOtmzZgkOHDiErKwtdunRBTEwMPv30U5X7ak5OTg7Gjh0rG7G1sbFBamoq8vJa3vqJz+cjMzOTdWzTpk1wdXVFSUkJunfvrtFYCSGEkFeCRgy1ok2NPO7ZsweOjo5wcHBAQEAAEhISIJVKIRKJEB4eDqFQiLKyMpSVlUEkEkEgEKCurg4HDhxo8fEswzAQi8XQ09ODWCwGwzCv6K6AyMhI+Pn54fz585g4cSLGjx+PK1euaPQaQ4YMwfHjx3H16lUAjY/zT506pVbCCwASiQQcDqfZR/mEEEJIW8bRkWqsqGrz5s2wsbGBoaEh3NzcWhzMSU9Ph6OjIwwNDdGvXz8cPnyYdV4qlWLJkiXo0qULeDwePD09ce3aNVYdGxsbcDgcVlm9ejWrzm+//YZ3330XhoaG6NatG2JiYlS+tzaVPDIMg4CAAACAj48PJBIJsrOzwePxYGJiAi6XC4FAAIFAAB6Ph8GDB2PRokWYMGECLCwsMHLkSKxZswa3b99m9VtZWYm9e/fK+g4ICMCePXtQVVX1Su7L398fM2bMgL29PZYvXw4XFxe5x8lisVhu68CUlBSlr7Fw4UKMHz8ejo6O0NPTw4ABAzB37lxMnDhR5XgfP36MBQsWQCwWo0OHDiq3J4QQQt5kaWlpCAsLw9dff41z587ByckJ3t7eCudjAI1PD8ViMQIDA1FQUABfX1/4+vri4sWLsjoxMTH417/+hfj4eJw5cwbGxsbw9vbG48ePWX0tW7ZMNtBWVlaGkJAQ2bnKykp4eXmhR48eOHv2LNasWYOoqChs27ZNpftrM8ljcXEx8vLyIBaLATTuqiISiVocIVy5ciXKy8sRHx8PoVCI+Ph4ODo64sKFC7I6qampsLW1hZOTEwDA2dkZPXr0QFpamsbiX7VqVZP7Tbu7u7Pquru7y408rlu3DoWFhayiyjuHe/bsQUpKCnbt2oVz585hx44dWLt2LXbs2KHSfTx58gTjxo2DVCrF1q1bm61bU1ODyspKVqmpbVDpeoQQQojWcKSaKyqIi4tDUFAQpk2bhj59+iA+Ph5GRkZNzrnYsGEDfHx8MG/ePPTu3RvLly/HO++8g02bNgFoHHVcv349Fi9ejLFjx6J///5ITk5GaWmp3IRbU1NT2UCbQCCAsbGx7FxKSgpqa2uRkJAAoVCI8ePH48svv0RcXJxK99dmkkeGYVBXVwdra2twuVxwuVxs3boV+/btg0QiabZtp06d4O/vj7Vr1+LKlSuwtrbG2rVrWX1funRJ1i+Xy8Xly5fVmjjTlM8//5yV+LW03/SLBAIBevXqxSqqbCk4b9482ehjv379MGnSJISGhiI6OlrpPp4mjrdu3UJmZmaLo47R0dHg8/msEr37ntLXI4QQQrRJk7OtFQ6Y1NTIXbO2thZnz56Fp6en7JiOjg48PT2Rm5urMM7c3FxWfQDw9vaW1b9x4wbKy8tZdfh8Ptzc3OT6XL16NTp16oQBAwZgzZo1qKurY11n+PDh0NfXZ12nuLgYDx48UPp7bRMTZurq6pCcnIzY2Fh4eXmxzvn6+iI1NRX6+vqor69vsS99fX3Y2trKZltfuHAB+fn5yMrKgrm5uaze/fv34eHhgaKiIjg6Or70PZibm7P6f97p06cxefJk1ucBAwa89DWf9/DhQ+josP8W0NXVRUODciOBTxPHa9eu4eeff0anTp1abBMREYGwsDDWMf1Tg5UPmhBCCGknoqOjZZN3n/r6668RFRXFOnbv3j3U19fDysqKddzKygpFRUUK+y4vL1dYv7y8XHb+6bGm6gDAl19+iXfeeQfm5ubIyclBREQEysrKZCOL5eXl6Nmzp1wfT8917Nixyft/XptIHjMyMvDgwQMEBgaCz+ezzvn5+YFhGISGhuLGjRsoLCxE165dYWpqiszMTOzevRvjx4+Hvb09pFIpvv/+exw+fBiJiYkAGkcdXV1dMXz4cLnrDho0CAzDyNZ9rK+vR2FhIauOgYEBevfujaqqKly/fl12/Gks5ubmLc5GTk9Ph4uLC4YNG4aUlBTk5eVpfMLO6NGjsXLlSnTv3h1CoRAFBQWIi4vD9OnTW2z75MkTfPrppzh37hwyMjJQX18v+2E0Nzdn/YXyPAMDAxgYGLCOSfXbzGA2IYSQN50GFwlXNGDy4u/A1vZ8fP3794e+vj4+++wzREdHazTWNpE8MgwDT09PucQRaEweY2JiIBQK4ePjgxEjRqCiogKJiYkYPnw4jIyMEB4ejj/++AMGBgaws7PD9u3bMWnSJNTW1mLnzp1YsGCBwuv6+fkhNjYWq1atAtC4VNCLI4K2tra4fv068vPzWUsAPf0XNGXKFCQlJTV7f0uXLsXu3bvxxRdfoEuXLkhNTUWfPn1U+YpatHHjRkRGRuKLL77AnTt3YG1tjc8++wxLlixpse2ff/6JQ4cOAYDcQuw///wzPDw8NBorIYQQ8ipocm9rRQMmilhYWEBXV1du8u7t27chEAgUthEIBM3Wf/rP27dvo0uXLqw6ijZQecrNzQ11dXW4efMmHBwcmrzO89dQBkeqzBYkRG0cDgcHDhxQeqeY9k56rG9rh9BuNfxlpNH+npTL/zFGlKM/+A+N9ndnl4tG++OtydFof28SvY12rR1Cswy6Kv/e2etAV9TyWsQv4/G/PFuupCTDL48pXdfNzQ2urq6ylVUaGhrQvXt3zJ49GwsXLpSrLxKJ8PDhQ3z//feyY0OGDEH//v0RHx8PqVQKa2tr/POf/0R4eDiAxpnTlpaWSEpKwvjx4xXGkZKSgsmTJ+PevXvo2LEjtm7diq+++gq3b9+Gnp4eAGDRokXYv39/k4/UFaFnjIQQQgghGhQWFobvvvsOO3bswJUrVzBr1ixUV1dj2rRpAIDJkycjIiJCVn/OnDk4cuQIYmNjUVRUhKioKOTn52P27NkAGgei5s6dixUrVuDQoUO4cOECJk+eDGtra9ngVG5uLtavX4/z58/jv//9L1JSUhAaGoqAgADZu4wTJkyAvr4+AgMDcenSJaSlpWHDhg1yj+Nb0iYeW5PmlZSUNPuY+/Lly82+d3ny5MlmFwt/VetdEkIIIa+UBt95VIVIJMLdu3exZMkSlJeXw9nZGUeOHJFNTikpKWFNch0yZAh27dqFxYsXY9GiRbCzs8PBgwfRt++zp3nz589HdXU1Zs6ciYqKCgwbNgxHjhyBoaEhgMbH6rt370ZUVBRqamrQs2dPhIaGshJDPp+P//znPwgODsbAgQNhYWGBJUuWYObMmSrdHz22bgeevq/QFBsbG9n2jIo8evQIf/75Z5Pne/Xq9TLhsdBja/XRY+u2gx5bv77osXXbou3H1jWb39dYXwbBP2msr/aORh7bAS6X+1IJHo/H02iCSAghhJA3FyWPRKPqbim3RhSR99dvPTTan+X4fI329yZpKFa8Zqu6TKwqNNpfQ4xQo/29SZ7Mv9TaITSrdrlza4fwSpmJtHwBFXeGIcqh5JEQQgghryVOK73z+Lqj2daEEEIIIURprZ483r17F7NmzUL37t1hYGAAgUAAb29v/PLLLy22zcrKAofDQUVFhdw5GxsbrF+/Xu54dHQ0dHV1ZbvKPC8pKQlmZmZNXi86OhqDBg2CqakpLC0t4evri+Li4hbjbMnNmzfB4XDkdrcBAA8PD8ydO1epfjgcjsKi6F4VuX//PiZOnIgOHTrAzMwMgYGBNBObEEJI+8WRaq4QmVZPHv38/FBQUIAdO3bg6tWrOHToEDw8PPDXX39p5XoJCQmYP38+EhISVG6bnZ2N4OBgnD59GpmZmXjy5Am8vLxk+2i3trKyMlZJSEgAh8OBn5+fUu0nTpyIS5cuITMzExkZGThx4oTK0/cJIYSQNkNHg4XItOrXUVFRgZMnT+Kbb77BiBEj0KNHD7i6uiIiIgJjxoyR1fnss89gZWUFQ0ND9O3bFxkZGWpdLzs7G48ePcKyZctQWVmJnBzVlrs4cuQIpk6dCqFQCCcnJyQlJaGkpARnz55tss3TUcXdu3djyJAhsnvIzs5W6x6aIxAIWOXf//43RowYgbfffrvFtleuXMGRI0ewfft2uLm5YdiwYdi4cSN2796N0tJSjcdKCCGEkPapVZNHExMTmJiY4ODBg6ipqZE739DQgJEjR+KXX37Bzp07cfnyZaxevRq6urpqXY9hGIjFYujp6UEsFoNhmJeKXyKRAADMzVuemTlv3jyEh4ejoKAA7u7uGD16tNZGV4HGvSp/+OEHBAYGKlU/NzcXZmZmcHF5th6dp6cndHR0cObMGW2FSQghhGgNhyPVWCHPtOpsay6Xi6SkJAQFBSE+Ph7vvPMO3nvvPYwfPx79+/fHsWPHkJf3/9q787ioyvZ/4J+ZYRmUAURERBAQNCEEDYXABS0XTEWeSAs1NXF9oDRz/6aWlpC5lfpgCwhmKu6SpuWGIogGijvgRj4uuAuCsjnX7w9/nscJkDM4wwxwvV+v84o59zmfuQ4DdHuW+z6G8+fPo3Xr1gBQ4Vk0Ozu7cuseP36s8jo/Px+bNm3CkSNHAABDhw5Fly5d8N1338HU1FTt2pVKJSZOnIhOnTqpjABfmfDwcOHycVRUFHbv3o3o6GhMnTpV2MbPz09lxHng2QDfL5v0vDJxcXFQKBR49913RW2fm5sLa2trlXUGBgawtLREbm6u2u/PGGOM6Rx3+rRC50P1BAcHo2/fvkhKSkJqaip27dqFBQsW4Oeff8bt27dhZ2cndBwrk5SUBIVCobKuW7duKq/XrVsHZ2dneHp6AgDatWsHBwcHxMfHiz4796KwsDCcOXMGhw8fFtaNGzcOa9asEV6/+LCJr6+v8LWBgQE6dOiA8+fPq2TGx8fD1dVVZd2QIUPUrg14dm/nkCFDhGmLtKG4uLjcGWNpqRLGhnxzCGOMMT3AQ/VohV78X14ul6Nnz56YNWsWUlJSMGLECMyZMwcmJiai9ndycoKLi4vK8s/p+qKjo3H27FkYGBgIy7lz56r14Ex4eDh27NiBAwcOqJz1nDt3LjIyMoRFXfb29uWOQ+z34EVJSUnIysrCqFGjRO9jY2OD27dvq6wrKyvD/fv3YWNjU+E+ERERMDc3V1m++V2z07oxxhhjTL/oRefxn9zc3FBYWAgPDw9cu3YN2dnZr5R3+vRppKWlITExUaVzl5iYiCNHjiAzM1NUDhEhPDwcW7duxf79++Hk5KTSbm1trdLxe1FqaqrwdVlZGdLT08udZdSU6OhoeHl5CWdZxfD19cXDhw9VHv7Zv38/lEolfHx8KtxnxowZyMvLU1mmvWP/yvUzxhhjmsD3PGqHTi9b37t3DwMHDsTIkSPh4eEBhUKBtLQ0LFiwAAMGDIC/vz+6du2K4OBgLF68GC4uLsjMzIREIkFAQIDo94mOjoa3tze6du1arq1jx46Ijo4WxkJ8+vRpubOGxsbGcHV1RVhYGNauXYvt27dDoVAI9wKam5tXeYZwxYoVaNWqFVxdXbFkyRI8ePAAI0eOFH0MYuXn52Pjxo1YtGiRWvu5uroiICBAuP+0tLQU4eHh+OCDD2Bra1vhPsbGxjA2NlZZV8qXrBljjOkLvmytFTrtPJqamsLHxwdLlizBpUuXUFpaCnt7e4wePRozZ84EAGzevBmTJ09GSEgICgsL4eLigsjISNHvUVJSgjVr1mDatGkVtgcHB2PRokWYP38+gGf3KbZv315lG2dnZ1y8eBFRUVEAyt9PuWrVKowYMeKldURGRiIyMhIZGRlwcXFBQkICrKysRB+HWOvXrwcRISQkRO19f/31V4SHh+Ptt9+GVCpFcHAwvv/+e43XyBhjjLHaS0JE3C3XopycHDg5OeHEiRPVemq6timN7qLrEmqte6ccNJpn/UGaRvPqE+WFqoffUkfRNc3mKZUSjebVJzT1rK5LeCma107XJdQoi3lbtJpf9otv1RuJZPDhEY1l1XY6f9qaMcYYY0wbJPzvLK3gG9RqiaSkJGFQ9YqWqsyfP7/Sffv06VMDR8AYY4yxuoDPPGqZo6MjNHFnQIcOHao1/M9z48aNw6BBgypsq85wQIwxxpje4wdmtII7j7WEiYlJueF/1GFpaSlqGkXGGGOszuAhdrSCL1szxhhjjDHR+MwjY4wxxuomvmytFdx5ZIwxxljdxJettYI7j4wxxhirkyR85lErdH7P4507dzB+/Hi0aNECxsbGsLGxQe/evZGcnFzlvomJiZBIJHj48GG5NkdHRyxdurTc+oiICMhkMmE6whfFxsbCwsKi0veLiIhAx44doVAoYG1tjaCgIGRlZVVZZ1VycnIgkUgqfJq6W7dumDhxoqgciURS4VLRsVYkMDAQLVq0gFwuR7NmzfDhhx/ixo0bahwJY4wxxuo6nXceg4ODceLECcTFxSE7OxsJCQno1q0b7t27p5X3i4mJwdSpUxETE6P2vgcPHkRYWBhSU1OxZ88elJaWolevXigsLNRCpeq7efOmyhITEwOJRILg4GBR+3fv3h0bNmxAVlYWNm/ejEuXLuG9997TctWMMcaYlkg0uDCBTi9bP3z4EElJSUhMTIS/vz8AwMHBAd7e3irbTJs2Ddu2bUNeXp4wt3W/fv3Ufr+DBw/iyZMnmDt3LlavXo2UlBT4+fmJ3n/37t0qr2NjY2FtbY309HR07dq1wn2eT0+4bt06fP/99zh+/DhcXFywYsUK4Zg1xcbGRuX19u3b0b17d7Rs2VLU/p9++qnwtYODA6ZPn46goCCUlpbC0NBQo7UyxhhjWseXrbVCp2cen89wsm3bNhQXF5drVyqV6NOnD5KTk7FmzRqcO3cOkZGRkMlk1Xq/6OhohISEwNDQECEhIYiOjn6l+vPy8gBA1PiJU6ZMwWeffYYTJ07A19cX/fv319rZVQC4desWdu7cidDQ0Grtf//+ffz666/w8/PjjiNjjDHGBDrtPBoYGCA2NhZxcXGwsLBAp06dMHPmTJw6dQoAsHfvXhw7dgxbtmxBz5490bJlS/Tr16/cdHp2dnblpty7evWqyjb5+fnYtGkThg4dCgAYOnQoNmzYgIKCgmrVrlQqMXHiRHTq1Anu7u5Vbh8eHo7g4GC4uroiKioK5ubm5Tqvfn5+5Y4jKSmpWvXFxcVBoVDg3XffVWu/adOmoWHDhmjcuDGuXr2K7du3V+v9GWOMMZ2TkOYWJtCLex5v3LiBhIQEBAQEIDExEW+88QZiY2ORkZEBOzs7tG7d+qUZSUlJyMjIUFlsbW1Vtlm3bh2cnZ3h6ekJAGjXrh0cHBwQHx9frbrDwsJw5swZrF+/Xlg3bty4Sueb9vX1Fb42MDBAhw4dcP78eZVt4uPjyx1Hhw4dqlVfTEwMhgwZArlcrtZ+U6ZMwYkTJ/Dnn39CJpNh2LBhlU6vWFxcjPz8fJWluFRZrXoZY4wxjZOS5hYm0IuheuRyOXr27ImePXti1qxZGDVqFObMmYPJkyeL2t/JyancU9IGBqqHFh0djbNnz6qsVyqViImJUfvSbnh4OHbs2IFDhw7Bzs5OWD937lzRNVfE3t6+3BSE1Zl3OikpCVlZWdXqGFtZWcHKygqtW7eGq6sr7O3tkZqaqtL5fS4iIgJffvmlyrrP+9tj9gAHtd+XMcYYY7WDzs88VsTNzQ2FhYXw8PDAtWvXkJ2d/Up5p0+fRlpaGhITE1XO6iUmJuLIkSPIzMwUlUNECA8Px9atW7F//344OTmptFtbW8PFxUVYXpSamip8XVZWhvT0dLi6ur7ScVUmOjoaXl5ewlnW6lIqn51FrOh+VACYMWMG8vLyVJZp79i/0nsyxhhjGsOXrbVCp2ce7927h4EDB2LkyJHw8PCAQqFAWloaFixYgAEDBsDf3x9du3ZFcHAwFi9eDBcXF2RmZkIikSAgIED0+0RHR8Pb27vCJ6I7duyI6OhoYSzEp0+flhtv0djYGK6urggLC8PatWuxfft2KBQK5ObmAgDMzc2rPEO4YsUKtGrVCq6urliyZAkePHiAkSNHij4GsfLz87Fx40YsWrRIrf2OHj2Kv/76C507d0ajRo1w6dIlzJo1C87OzhWedQSefV+MjY1V1pUa6uW/RxhjjNVH/L8krdBp59HU1BQ+Pj5YsmQJLl26hNLSUtjb22P06NGYOXMmAGDz5s2YPHkyQkJCUFhYKAzVI1ZJSQnWrFmDadOmVdgeHByMRYsWYf78+QCAgoICtG/fXmUbZ2dnXLx4EVFRUQCeDdz9olWrVmHEiBEvrSMyMhKRkZHIyMiAi4sLEhISYGVlJfo4xFq/fj2ICCEhIWrt16BBA2zZsgVz5sxBYWEhmjVrhoCAAHz++eflOoiMMcYYq78kVNnTEEwjno/zeOLECbRr107X5WhdaXQXXZdQa907pdl7Ra0/SNNoXn2ivFD18FvqKLqm2Tylkkcsri6aelbXJbwUzWun6xJqlMW8LVrNV/7+ardvvUj6zkmNZdV2evHADGOMMcaYpvHc1trBdwPUEklJSeXGgKxsWKCKzJ8/v9J9/zluJmOMMVYn8AMzWsFnHrXM0dGx0nES1dGhQ4dyD/KoY9y4cRg0aFCFbdUZDogxxhhj9RN3HmsJExOTcsP/qMPS0lLUNIqMMcZYncHXV7WCO49Mo2TN8nVdQq1l7arZB1xyf/XWaF590qz/CY3mZa3112ie55BEjebVJ4URbXVdwktJZmXouoS6hS83awX3yRljjDHGNGzFihVwdHSEXC6Hj48Pjh079tLtN27ciDZt2kAul6Nt27b4/fffVdqJCLNnz0azZs1gYmKCHj164MKFC0J7Tk4OQkND4eTkBBMTEzg7O2POnDkoKSlR2UYikZRbXpzIRAzuPDLGGGOsbtLR3Nbx8fGYNGkS5syZg+PHj8PT0xO9e/fG7du3K9w+JSUFISEhCA0NxYkTJxAUFISgoCCcOXNG2GbBggX4/vvvsXLlShw9ehQNGzZE7969UVRUBADIzMyEUqnEDz/8gLNnz2LJkiVYuXKlMG72i/bu3YubN28Ki5eXl1rHp9PO4507dzB+/Hi0aNECxsbGsLGxQe/evZGcnFzlvomJiZBIJHj48GG5NkdHRyxdurTc+oiICMhkMmE2mRfFxsaWmx/7n/t27NgRCoUC1tbWCAoKQlZWVpV1VuX5vwIqehimW7dumDhxoqiciv4lIZFIKjzWyuzcuRM+Pj4wMTFBo0aNEBQUJHpfxhhjTO9INLioYfHixRg9ejQ++ugjuLm5YeXKlWjQoAFiYmIq3P67775DQEAApkyZAldXV8ybNw9vvPEGli9fDuDZWcelS5fi888/x4ABA+Dh4YHVq1fjxo0b2LZtGwAgICAAq1atQq9evdCyZUsEBgZi8uTJ2LKl/FiajRs3ho2NjbAYGhqqdXw67TwGBwfjxIkTiIuLQ3Z2NhISEtCtWzfcu3dPK+8XExODqVOnVvrhvczBgwcRFhaG1NRU7NmzB6WlpejVqxcKCwu1UKn6XvwXxM2bNxETEwOJRILg4GBR+2/evBkffvghPvroI5w8eRLJyckYPHiwlqtmjDHGaofi4mLk5+erLMXFxeW2KykpQXp6Onr06CGsk0ql6NGjB44cOVJh9pEjR1S2B4DevXsL21+5cgW5ubkq25ibm8PHx6fSTADIy8ur8GHZwMBAWFtbo3PnzkhISHj5gVdAZ53Hhw8fIikpCd988w26d+8OBwcHeHt7Y8aMGQgMDBS2GTt2LJo2bQq5XA53d3fs2LGjWu938OBBPHnyBHPnzkV+fj5SUlLU2n/37t0YMWIEXn/9dXh6eiI2NhZXr15Fenp6pfs8P6u4fv16+Pn5Ccdw8ODBah3Dy7z4LwgbGxts374d3bt3R8uWLavct6ysDBMmTMC3336LcePGoXXr1nBzc6t0aB/GGGOsVtDgZeuIiAiYm5urLBEREeXe8u7du3j69CmaNm2qsr5p06bIzc2tsMzc3NyXbv/8v+pkXrx4EcuWLcPYsWOFdaampli0aBE2btyInTt3onPnzggKClK7A6mzp62fD1C9bds2vPnmm+XmT1YqlejTpw8ePXqENWvWwNnZGefOnYNMJqvW+0VHRyMkJASGhoYICQlBdHQ0/Pz8ql1/Xl4eAIga/mbKlClYunQp3NzcsHjxYvTv3x9XrlxB48aNq/3+L3Pr1i3s3LkTcXFxorY/fvw4rl+/DqlUivbt2yM3Nxft2rXDt99+C3d3d63UyBhjjGkbafAU2YwZMzBp0iSVdf/su+iL69evIyAgAAMHDsTo0aOF9VZWVirH0LFjR9y4cQPffvutcOJODJ2deTQwMEBsbCzi4uJgYWGBTp06YebMmTh16hSAZzdzHjt2DFu2bEHPnj3RsmVL9OvXr9xsKHZ2duVmTLl69arKNvn5+di0aROGDh0KABg6dCg2bNiAgoKCatWuVCoxceJEdOrUSVTnKjw8HMHBwXB1dUVUVBTMzc0RHR2tso2fn1+540hKSqpWfXFxcVAoFHj33XdFbX/58mUAwBdffIHPP/8cO3bsQKNGjdCtWzfcv3+/WjUwxhhjdYmxsTHMzMxUloo6j1ZWVpDJZLh165bK+lu3bsHGxqbCbBsbm5du//y/YjJv3LiB7t27w8/PDz/++GOVx+Xj44OLFy9Wud2LdH7P440bN5CQkICAgAAkJibijTfeQGxsLDIyMmBnZ4fWrVu/NCMpKQkZGRkqi62trco269atg7OzMzw9n02Q3q5dOzg4OCA+Pr5adYeFheHMmTNYv369sG7cuHGVThfo6+srfG1gYIAOHTrg/PnzKtvEx8eXO44OHTpUq76YmBgMGTIEcrlc1PZKpRIA8H//938IDg6Gl5cXVq1aBYlEgo0bN1a6X4X3f5Qqq1UzY4wxpnE6mJ7QyMgIXl5e2Ldvn7BOqVRi3759Kv2BF/n6+qpsDwB79uwRtndycoKNjY3KNvn5+Th69KhK5vXr19GtWzfh/+NSadXdvIyMDDRr1kz08QF6MEi4XC5Hz5490bNnT8yaNQujRo3CnDlzMHnyZFH7Ozk5lXtK2sBA9bCio6Nx9uxZlfVKpRIxMTEIDQ1Vq97w8HDs2LEDhw4dgp2dnbB+7ty5omuuiL29fbkZZKozbWBSUhKysrLU6hg//6Fxc3MT1hkbG6Nly5blzuK+KCIiAl9++aXKutmDrTFnSMX/smKMMcZqlJpD7GjKpEmTMHz4cHTo0AHe3t5YunQpCgsL8dFHHwEAhg0bhubNmwv3TE6YMAH+/v5YtGgR+vbti/Xr1yMtLU04cyiRSDBx4kR89dVXaNWqFZycnDBr1izY2toKI6M87zg6ODhg4cKFuHPnjlDP87OTcXFxMDIyQvv27QEAW7ZsQUxMDH7++We1jk/nncd/cnNzw7Zt2+Dh4YFr164hOzu7yrOPL3P69GmkpaUhMTFR5f7E+/fvo1u3bsjMzESbNm2qzCEifPzxx9i6dSsSExPh5OSk0m5tbQ1ra+sK901NTUXXrl0BPHs4JT09HeHh4dU+ppeJjo6Gl5eXcJZVDC8vLxgbGyMrKwudO3cGAJSWliInJwcODg6V7lfR/R+GBzpVr3DGGGNM09QcYkdT3n//fdy5cwezZ88WniPYvXu38MDL1atXVc4K+vn5Ye3atfj8888xc+ZMtGrVCtu2bVO5NW7q1KkoLCzEmDFj8PDhQ3Tu3Bm7d+8WrjLu2bMHFy9exMWLF1VObgHP+jDPzZs3D3///TcMDAzQpk0bxMfH47333lPr+HTWebx37x4GDhyIkSNHwsPDAwqFAmlpaViwYAEGDBgAf39/dO3aFcHBwVi8eDFcXFyQmZkJiUSCgIAA0e8THR0Nb29vofP2oo4dOyI6OloYC/Hp06flxls0NjaGq6srwsLCsHbtWmzfvh0KhUJ4usnc3LzKM4QrVqxAq1at4OrqiiVLluDBgwcYOXKk6GMQKz8/Hxs3bsSiRYvU2s/MzAzjxo3DnDlzYG9vDwcHB+F7MnDgwEr3MzY2Lv+gkyGPO88YY4yFh4dXeqIoMTGx3LqBAwe+9P+5EokEc+fOxdy5cytsHzFiBEaMGPHSmoYPH47hw4e/dBsxdPq0tY+PD5YsWYJLly6htLQU9vb2GD16tDAa+ubNmzF58mSEhISgsLAQLi4uiIyMFP0eJSUlWLNmDaZNm1Zhe3BwMBYtWoT58+cDAAoKCoRTuc85Ozvj4sWLiIqKAvBs4O4XrVq1qsoPKzIyEpGRkcjIyICLiwsSEhJgZWUl+jjEWr9+PYgIISEhau/77bffwsDAAB9++CGePHkCHx8f7N+/H40aNdJ4nYwxxliN4PMZWiGhF89lMo3KycmBk5MTTpw4gXbt2um6nBqh/F385XL2DxblB5t9FblrvTWaV580639Co3nHl/TTaJ7nkESN5tUnhZcrvr1Ib8w4resKapS50WWt5iv/ek1jWdKOrz6rXF3BfXLGGGOMMSYadx5rgaSkpHJjQFY2LFBF5s+fX+m+/xw3kzHGGKszdDS3dV2nd09b1yWOjo7QxF0BHTp0KPcgjzrGjRtX6VSD1RkOiDHGGKsNSEdD9dR13HmsBUxMTMqNAakOS0tLUdMoMsYYY4xVhTuPjDHGGKub+HKzVnDnkTHGGGN1Ez/ZoRX8bWWMMcYYY6LxmUfGGGOM1U18ikwrdPptvXPnDsaPH48WLVrA2NgYNjY26N27N5KTk6vcNzExERKJBA8fPizX5ujoiKVLl5ZbHxERAZlMJky996LY2FhYWFhU+n4RERHo2LEjFAoFrK2tERQUhKysVx8wNCcnBxKJpMKnqbt164aJEyeKypFIJBUuFR1rRY4fP46ePXvCwsICjRs3xpgxY1BQUKDGkTDGGGP6hSSksYX9j047j8HBwThx4gTi4uKQnZ2NhIQEdOvWDffu3dPK+8XExGDq1KmIiYlRe9+DBw8iLCwMqamp2LNnD0pLS9GrVy8UFhZqoVL13bx5U2WJiYmBRCJBcHBwlfveuHEDPXr0gIuLC44ePYrdu3fj7NmzVU67yBhjjOk1qQYXJtDZt+Phw4dISkrCN998g+7du8PBwQHe3t6YMWMGAgMDhW3Gjh2Lpk2bQi6Xw93dHTt27KjW+x08eBBPnjzB3LlzkZ+fj5SUFLX23717N0aMGIHXX38dnp6eiI2NxdWrV5Genl7pPs/PKq5fvx5+fn7CMRw8eLBax/AyNjY2Ksv27dvRvXt3tGzZssp9d+zYAUNDQ6xYsQKvvfYaOnbsiJUrV2Lz5s24ePGixmtljDHGWO2ls87j8xlOtm3bhuLi8nP6KpVK9OnTB8nJyVizZg3OnTuHyMhIyGSyar1fdHQ0QkJCYGhoiJCQEERHR79S/Xl5eQAgavzEKVOm4LPPPsOJEyfg6+uL/v37a+3sKgDcunULO3fuRGhoqKjti4uLYWRkBKn0fz8OzwcPP3z4sFZqZIwxxrSOzzxqhc6+HQYGBoiNjUVcXBwsLCzQqVMnzJw5E6dOnQIA7N27F8eOHcOWLVvQs2dPtGzZEv369Ss3nZ6dnV25KfeuXr2qsk1+fj42bdqEoUOHAgCGDh2KDRs2VPuePqVSiYkTJ6JTp05wd3evcvvw8HAEBwfD1dUVUVFRMDc3L9d59fPzK3ccSUlJ1aovLi4OCoUC7777rqjt33rrLeTm5uLbb79FSUkJHjx4gOnTpwN4djmcMcYYq41IormF/Y/O73m8ceMGEhISEBAQgMTERLzxxhuIjY1FRkYG7Ozs0Lp165dmJCUlISMjQ2WxtbVV2WbdunVwdnaGp6cnAKBdu3ZwcHBAfHx8teoOCwvDmTNnsH79emHduHHjKp1v2tfXV/jawMAAHTp0wPnz51W2iY+PL3ccHTp0qFZ9MTExGDJkCORyuajtX3/9dcTFxWHRokVo0KABbGxs4OTkhKZNm6qcjfyn4uJi5OfnqyzFpcpq1cwYY4yx2kHnJ2Llcjl69uyJWbNmISUlBSNGjMCcOXNEz7ns5OQEFxcXlcXAQHUEoujoaJw9exYGBgbCcu7cuWo9OBMeHo4dO3bgwIEDsLOzE9bPnTtXpeOnLnt7+3LHUZ15p5OSkpCVlYVRo0aptd/gwYORm5uL69ev4969e/jiiy9w586dl94zGRERAXNzc5UlcsNttWtmjDHGtIIvW2uF3n073NzcUFhYCA8PD1y7dg3Z2dmvlHf69GmkpaUhMTFRpXOXmJiII0eOIDMzU1QOESE8PBxbt27F/v374eTkpNJubW2t0vF7UWpqqvB1WVkZ0tPT4erq+krHVZno6Gh4eXkJZ1nV1bRpU5iamiI+Pl7o2FdmxowZyMvLU1mmD7KubumMMcaYRpFUcwv7H50NEn7v3j0MHDgQI0eOhIeHBxQKBdLS0rBgwQIMGDAA/v7+6Nq1K4KDg7F48WK4uLggMzMTEokEAQEBot8nOjoa3t7e6Nq1a7m2jh07Ijo6WhgL8enTp+XOGhobG8PV1RVhYWFYu3Yttm/fDoVCgdzcXACAubl5lWcIV6xYgVatWsHV1RVLlizBgwcPMHLkSNHHIFZ+fj42btyIRYsWqb3v8uXLhfsu9+zZgylTpiAyMvKlY18aGxvD2NhYZZ3SkH/DGGOMsbpMZ51HU1NT+Pj4YMmSJbh06RJKS0thb2+P0aNHY+bMmQCAzZs3Y/LkyQgJCUFhYSFcXFwQGRkp+j1KSkqwZs0aTJs2rcL24OBgLFq0CPPnzwcAFBQUoH379irbODs74+LFi4iKigLwbODuF61atarK8RAjIyMRGRmJjIwMuLi4ICEhAVZWVqKPQ6z169eDiBASEqL2vseOHcOcOXNQUFCANm3a4IcffsCHH36o8RoZY4yxGiPhJ120QUJEPGy6luTk5MDJyQknTpxAu3btdF1OjVD+Xr3L5QyARfkhq15F7lpvjebVJ836n9Bo3vEl/TSa5zkkUaN59UnhZT2/tWbGaV1XUKPMjS5rNb845+UP3arD2PHVbqOrS/gaI2OMMcYYE407j7VAUlJSuTEgKxsWqCLz58+vdN9/jpvJGGOM1Rn8tLVW6Oyex/rA0dERmrgroEOHDtUa/ue5cePGYdCgQRW2VWc4IMYYY6w24MG9tYM7j7WAiYlJueF/1GFpaSlqGkXGGGOsTuEzhlrB31bGGGOMMSYan3lkjDHGWJ1EUr5urQ3ceWSMMcZY3cTXV7VCp9/WO3fuYPz48WjRogWMjY1hY2OD3r17Izk5ucp9ExMTIZFI8PDhw3Jtjo6OWLp0abn1ERERkMlkwowyL4qNjX3pbCoRERHo2LEjFAoFrK2tERQUhKysrCrrrEpOTg4kEkmFD8R069YNEydOFJVTUFCA8PBw2NnZwcTEBG5ubli5cqWofe/fv4+PP/4Yr732GkxMTNCiRQt88sknyMvLU+NIGGOMMVYf6LTzGBwcjBMnTiAuLg7Z2dlISEhAt27dcO/ePa28X0xMDKZOnYqYmBi19z148CDCwsKQmpqKPXv2oLS0FL169UJhYaEWKlXfpEmTsHv3bqxZswbnz5/HxIkTER4ejoSEhCr3vXHjBm7cuIGFCxfizJkziI2Nxe7duxEaGloDlTPGGGPaQRLNLex/dNZ5fPjwIZKSkvDNN9+ge/fucHBwgLe3N2bMmIHAwEBhm7Fjx6Jp06aQy+Vwd3fHjh07qvV+Bw8exJMnTzB37lzk5+cjJSVFrf13796NESNG4PXXX4enpydiY2Nx9epVpKenV7rP87OK69evh5+fn3AMBw8erNYxvExKSgqGDx+Obt26wdHREWPGjIGnpyeOHTtW5b7u7u7YvHkz+vfvD2dnZ7z11lv4+uuv8dtvv6GsrEzjtTLGGGM1QirR3MIEOus8Ph+ketu2bSguLj8tm1KpRJ8+fZCcnIw1a9bg3LlziIyMhEwmq9b7RUdHIyQkBIaGhggJCUF0dPQr1f/8kq6YIXCmTJmCzz77DCdOnICvry/69++v8bOrfn5+SEhIwPXr10FEOHDgALKzs9GrV69q5eXl5cHMzAwGBnxbLGOMMcb+R2c9AwMDA8TGxmL06NFYuXIl3njjDfj7++ODDz6Ah4cH9u7di2PHjuH8+fNo3frZ3JQtW7Ysl2NnZ1du3ePHj1Ve5+fnY9OmTThy5AgAYOjQoejSpQu+++47UTO0/JNSqcTEiRPRqVMnuLu7V7l9eHg4goODAQBRUVHYvXs3oqOjMXXqVGEbPz8/SKWqffknT56InhN72bJlGDNmDOzs7GBgYACpVIqffvoJXbt2FX9g/9/du3cxb948jBkzRu19GWOMMX1B/MCMVuj0tFJwcDD69u2LpKQkpKamYteuXViwYAF+/vln3L59G3Z2dkLHsTJJSUlQKBQq67p166byet26dXB2doanpycAoF27dnBwcEB8fHy17usLCwvDmTNncPjwYWHduHHjsGbNGuF1QUGB8LWvr6/wtYGBATp06IDz58+rZMbHx8PV1VVl3ZAhQ0TXtGzZMqSmpiIhIQEODg44dOgQwsLCYGtrix49eojOyc/PR9++feHm5oYvvvjipdsWFxeXO2tsWKqEsSH/tjLGGNM9vldRO3R+TVIul6Nnz57o2bMnZs2ahVGjRmHOnDmYPHmyqP2dnJzKPSX9z0ut0dHROHv2rMp6pVKJmJgYtTuP4eHh2LFjBw4dOqRy1nPu3Lmia66Ivb19uVlkxE4d+OTJE8ycORNbt25F3759AQAeHh7IyMjAwoULRXceHz16hICAACgUCmzduhWGhoYv3T4iIgJffvmlyrrZg60xZ4iNqPdjjDHGWO2jd6eI3NzcUFhYCA8PD1y7dg3Z2dmvlHf69GmkpaUhMTERGRkZwpKYmIgjR44gMzNTVA4RITw8HFu3bsX+/fvh5OSk0m5tbQ0XFxdheVFqaqrwdVlZGdLT08udZXwVpaWlKC0tLXfZWyaTQalUisrIz89Hr169YGRkhISEBMjl8ir3mTFjBvLy8lSW6YOsq3UMjDHGmMZJNbgwgc7OPN67dw8DBw7EyJEj4eHhAYVCgbS0NCxYsAADBgyAv78/unbtiuDgYCxevBguLi7IzMyERCJBQECA6PeJjo6Gt7d3hff+dezYEdHR0cK4j0+fPi033qKxsTFcXV0RFhaGtWvXYvv27VAoFMjNzQUAmJubV3mGcMWKFWjVqhVcXV2xZMkSPHjwACNHjhR9DFUxMzODv78/pkyZAhMTEzg4OODgwYNYvXo1Fi9eXOX+zzuOjx8/xpo1a5Cfn4/8/HwAQJMmTSp9SMnY2BjGxsYq65R8yZoxxpie4HsetUNnnUdTU1P4+PhgyZIluHTpEkpLS2Fvb4/Ro0dj5syZAIDNmzdj8uTJCAkJQWFhIVxcXBAZGSn6PUpKSrBmzRpMmzatwvbg4GAsWrQI8+fPB/DsPsX27durbOPs7IyLFy8iKioKQPn7KVetWoURI0a8tI7IyEhERkYiIyMDLi4uSEhIgJWVlejjEGP9+vWYMWMGhgwZgvv378PBwQFff/01xo0bV+W+x48fx9GjRwGg3FnTK1euwNHRUaO1MsYYYzWBpKTrEuokCRHxd1ZLcnJy4OTkhBMnToh+arq2U/7uqesSai+L8kNWvYrctd4azatPmvU/odG840v6aTTPc0iiRvPqk8LLen5rzYzTuq6gRpkbXdZqfn7haxrLMmv46rPK1RV8QpcxxhhjdZOENLeoacWKFXB0dIRcLoePj0+Vk3Zs3LgRbdq0gVwuR9u2bfH777+rtBMRZs+ejWbNmsHExAQ9evTAhQsXVLa5f/8+hgwZAjMzM1hYWCA0NFRl9BcAOHXqFLp06QK5XA57e3ssWLBA7WPjzmMtkJSUJAyqXtFSlV9//bXSfV9//fUaOALGGGOs5pGUNLaoIz4+HpMmTcKcOXNw/PhxeHp6onfv3rh9+3aF26ekpCAkJAShoaE4ceIEgoKCEBQUhDNnzgjbLFiwAN9//z1WrlyJo0ePomHDhujduzeKioqEbYYMGYKzZ89iz549wsgwL47Z/PwZBwcHB6Snp+Pbb7/FF198gR9//FGt4+PL1rXAkydPcP369Urb/3mf4j89evQIt27dqrDN0NAQDg4Or1Tfi/iy9Svgy9Z6gy9b11182Vq/aPuydV5RK41lmcsvVL3R/+fj44OOHTti+fLlAJ4ND2hvb4+PP/4Y06dPL7f9+++/j8LCQpUpmN988020a9cOK1euBBHB1tYWn332mTAsYF5eHpo2bYrY2Fh88MEHOH/+PNzc3PDXX3+hQ4cOAJ5NrfzOO+/g2rVrsLW1RVRUFP7v//4Pubm5MDIyAgBMnz4d27ZtEz36DKAH4zyyqpmYmFTZQXwZhUJRbiB1xhhjrM6TPtVYVEUTY1Q06khJSQnS09MxY8aM/5UhlaJHjx7CTHf/dOTIEUyaNEllXe/evbFt2zYAzx5ezc3NVRm32dzcHD4+Pjhy5Ag++OADHDlyBBYWFkLHEQB69OgBqVSKo0eP4l//+heOHDmCrl27Ch3H5+/zzTff4MGDB2jUqJG4bwYxVsOKiopozpw5VFRUpHd5+lwb5+lPFufpV54+18Z5dcecOXMIgMoyZ86ccttdv36dAFBKSorK+ilTppC3t3eF2YaGhrR27VqVdStWrCBra2siIkpOTiYAdOPGDZVtBg4cSIMGDSIioq+//ppat25dLrtJkyb0n//8h4iIevbsSWPGjFFpP3v2LAGgc+fOveToVXHnkdW4vLw8AkB5eXl6l6fPtXGe/mRxnn7l6XNtnFd3FBUVUV5enspSUQe5PnQe+bI1Y4wxxlgVKrpEXRErKyvIZLJyzxrcunULNjYVT99rY2Pz0u2f//fWrVto1qyZyjbPhwK0sbEp90BOWVkZ7t+/r5JT0fu8+B5i8NPWjDHGGGMaYmRkBC8vL+zbt09Yp1QqsW/fPvj6+la4j6+vr8r2ALBnzx5heycnJ9jY2Khsk5+fj6NHjwrb+Pr64uHDh0hPTxe22b9/P5RKJXx8fIRtDh06hNLSUpX3ee2118Tf7wjuPDLGGGOMadSkSZPw008/IS4uDufPn8f48eNRWFiIjz76CAAwbNgwlQdqJkyYgN27d2PRokXIzMzEF198gbS0NISHhwMAJBIJJk6ciK+++goJCQk4ffo0hg0bBltbWwQFBQEAXF1dERAQgNGjR+PYsWNITk5GeHg4PvjgA9ja2gIABg8eDCMjI4SGhuLs2bOIj4/Hd999V+5hnSqJvsDNmIbo883Z+lwb5+lPFufpV54+18Z59deyZcuoRYsWZGRkRN7e3pSamiq0+fv70/Dhw1W237BhA7Vu3ZqMjIzo9ddfp507d6q0K5VKmjVrFjVt2pSMjY3p7bffpqysLJVt7t27RyEhIWRqakpmZmb00Ucf0aNHj1S2OXnyJHXu3JmMjY2pefPmFBkZqfax8TiPjDHGGGNMNL5szRhjjDHGROPOI2OMMcYYE407j4wxxhhjTDTuPDLGGGOMMdG488gYY4wxxkTjziNjjDHGGBONpydkWnX37l3ExMTgyJEjyM3NBfBsCiQ/Pz+MGDECTZo0USvv/PnzSE1Nha+vL9q0aYPMzEx89913KC4uxtChQ/HWW2+Jyjl+/DgaNWoEJycnAMAvv/yClStX4urVq3BwcBAGVlXHzZs3ERUVhcOHD+PmzZuQSqVo2bIlgoKCMGLECMhkMrXz9u3bB0tLS/To0QNGRkZCW2FhIRYtWoTZs2eLzispKcG2bdsq/CwGDBigki/GvXv3cOrUKXh6esLS0hJ3795FdHQ0iouLMXDgQLi6uorOunbtGuRyOaysrAAASUlJKp9HWFhYpTMzVObJkydYt25dhZ/H22+/rXZWeno6LC0t4ebmptJWVFSEDRs2YNiwYWplHjt2rNxn4evrC29vb7VyAICIkJOTA3t7exgYGKCkpARbt25FcXEx3nnnHeH7KkZxcTGkUikMDQ0BAJcuXUJMTIzwWYSGhgq/N+rYv39/uc8iMDAQrVq1UjsLePYzY2FhAVNTU5X1paWlOHLkCLp27apWXm5uLo4eParyefj4+Kg1ZduLnj59qvI7f/ToURQXF8PX11f43lbHw4cPsXHjRuHzGDhwIMzNzdXOuXz5crnPo2fPnjAzM6tWXUqlElJp+fNRSqUS165dQ4sWLaqVy/SU2iNDMibSsWPHqFGjRtS8eXMaPnw4TZ06laZOnUrDhw8nOzs7srS0pL/++kt03q5du8jIyIgsLS1JLpfTrl27qEmTJtSjRw966623SCaT0b59+0RleXh40J49e4iI6KeffiITExP65JNPKCoqiiZOnEimpqYUHR0tura//vqLzM3NycvLizp37kwymYw+/PBDev/998nCwoL8/PwoPz9fdN6xY8fIwsKCzMzMyMTEhFxcXOjMmTNCe25uLkmlUtF5Fy5coJYtW5JcLid/f38aNGgQDRo0iPz9/Ukul5OLiwtduHBBdN7Ro0fJ3NycJBIJNWrUiNLS0sjJyYlatWpFzs7OZGJiQunp6aLzvL296bfffiMiom3btpFUKqXAwECaNm0a/etf/yJDQ0OhXezxOjg4kLW1Ndnb25NEIqG+ffuSj48PyWQyGjhwIJWWlorKysrKIgcHB5JIJCSVSqlr165048YNoV3dz+LWrVvUuXNnkkgk5ODgQN7e3uTt7S28R+fOnenWrVui8zIzM8nBwYGkUim5uLjQ5cuXycvLixo2bEgNGjQgKysrys7OFp3n7+9PGzduJCKiw4cPk7GxMXl4eND7779P7du3pwYNGlBKSopax+vt7U1SqZQMDAxIKpWSl5cX2djYkEwmoylTpojOIiK6ceMGdezYkaRSqfB79uIgyOp+HgUFBTRkyBCSyWRkYGBA1tbWZG1tTQYGBiSTyWjo0KFUWFioVn2dOnUimUxGXbt2pfv371Pfvn1JIpGQRCKh1q1bq/z8VOVf//qX8HmcOXOGrKysqEmTJuTj40NNmzYlGxsbOnfunFrH+9577wn1SKVS4bMwNTWl5cuXi84iIsrLy6OBAweSXC4na2trmjVrFpWVlQnt6n4erHbgziPTGh8fHxozZgwplcpybUqlksaMGUNvvvmm6DxfX1/6v//7PyIiWrduHTVq1IhmzpwptE+fPp169uwpKsvExIRycnKIiKh9+/b0448/qrT/+uuv5ObmJrq2Tp060RdffCG8/uWXX8jHx4eIiO7fv0/t2rWjTz75RHRejx496KOPPqKnT59Sfn4+jR8/nho3bkzHjx8nIvX/IPfo0YMGDBhAeXl55dry8vJowIAB1KtXL7XyRo0aRfn5+fTtt9+SnZ0djRo1Smj/6KOPKCgoSHRew4YN6fLly0T07OfmnzMeLFu2jNq3by86r0+fPjR27FjhZy8yMpL69OlDRETZ2dnk6OhIc+bMEZUVFBREffv2pTt37tCFCxeob9++5OTkRH///TcRqf9ZBAcHk6+vL2VmZpZry8zMJD8/P3rvvfdE5w0YMIACAwPp1KlTNHHiRHJ1daUBAwZQSUkJFRUVUf/+/Wno0KGi88zMzITOpr+/P3366acq7Z9//jl16tRJdN77779PQUFBlJeXR0VFRRQeHk7Dhg0jIqJ9+/ZR48aNaenSpaLzhg0bRj4+PvTXX3/Rnj17yMvLizp06ED3798nomefh0QiEZ0XGhpKrVq1ot27d6t0esrKyuiPP/6g1q1bq/xsV+XDDz8kPz8/SkhIoPfff5/8/PyoS5cudO3aNfr777+pU6dOFBYWJjqvUaNGdP78eSJ69nM9ePBgKi4uJiKikpISCg0NVet3d8yYMdSpUyc6ffo0Xbhwgd577z2aOnUqFRYWUnR0NDVo0IB+/fVX0XmffPIJtW7dmjZu3Eg//fQTOTg4UN++fYUa1f08WO3AnUemNXK5XPijV5Hz58+TXC4XnWdmZiacHXv69CkZGBgInSkiotOnT1PTpk1FZTVu3JjS0tKIiMja2poyMjJU2i9evEgmJiaiazMxMaFLly4Jr58+fUqGhoaUm5tLRER//vkn2drais5r1KhRuWmnIiIiqFGjRnTs2DG1OywmJiZ0+vTpSttPnTql1vE2atRIONtRUlJCUqmUjh49KrSnp6dT8+bNReeZm5vTyZMniejZ5/H86+cuXrxIDRo0EJ3XoEEDlbNtxcXFZGhoSHfv3iWiZ2c3HR0dRWVZW1vTqVOnhNdKpZLGjRtHLVq0oEuXLqn9WZiamqr83P5TWloamZqais5r0qQJnThxgoienVWSSCSUlJQktCcnJ1OLFi1E5zVs2FD4vW3atGmFvxvq1GdmZqZy1rygoIAMDQ2Ff8j88ssv9Nprr4nOs7W1VflZe95BbteuHd27d0/tz8PCwoKSk5MrbT98+DBZWFiIzmvWrBkdOXKEiJ5NFSeRSGjv3r1C+759+6hly5ai80xMTOjixYtC9j9/drKyssjc3Fx0npWVlfC3j+jZP27lcrlwdnX58uXUrl070XktWrSgAwcOCK/v3LlD3t7e1KtXLyoqKuIzj3UUPzDDtMbGxgbHjh2rtP3YsWNo2rSpWpkSiQQAIJVKIZfLVe71USgUyMvLE5XTp08fREVFAQD8/f2xadMmlfYNGzbAxcVFdF3W1ta4efOm8PrWrVsoKysT7h9q1aoV7t+/LzoPeHYv3YumT5+OmTNnolevXkhJSVEry8LCAjk5OZW25+TkwMLCQnReSUkJTExMAACGhoZo0KCByn11VlZWuHfvnug8f39/rFu3DgDQvn17JCYmqrQfOHAAzZs3F51nYWGBR48eCa8fP36MsrIy4b5ODw8Plc/rZZ48eQIDg//dHi6RSBAVFYX+/fvD398f2dnZousCAGNjY+Tn51fa/ujRIxgbG4vOKygogKWlJQCgYcOGaNiwIZo1aya029vb49atW6LzfHx88NtvvwEAnJ2dcfLkSZX2jIwM4f3EMDY2Fn5vgWe/u0+fPkVZWRkAwM/P76U/m/+Ul5eHRo0aqeRv2bIFjo6O6N69O27fvi06C3h2T97L7vc1MjKCUqkUnffgwQPhZ9XS0hINGjSAg4OD0O7i4iL6Zw949rO6f/9+AM/+pv79998q7X///bfwuyjGi3+XAMDU1BRlZWUoLCwEAPTq1QuZmZmi8+7cuaNyfFZWVti7dy8ePXqEd955B48fPxadxWoRXfdeWd21fPlyMjY2pk8++YS2b99OqamplJqaStu3b6dPPvmETExMaMWKFaLzPDw8aNeuXcLr06dPq9y3dujQIXJychKVdf36dXJ0dKSuXbvSpEmTyMTEhDp37kyjR4+mrl27kpGRUblJ6V9mwoQJ5O7uTrt27aL9+/dT9+7dqVu3bkL77t27ydnZWXRely5dKCoqqsK2b775hoyNjdX61/ysWbOoUaNGtHjxYjp58iTl5uZSbm4unTx5khYvXkyWlpaiL+MSEbVp00bl/tIdO3bQ48ePhdepqalkZ2cnOu/cuXPUuHFjGjZsGM2bN49MTU1p6NCh9PXXX9OwYcPI2NiYVq1aJTpv+PDh5O/vT+fPn6fLly8L9+s9l5iYSPb29qKyOnbsSKtXr66wLSwsjCwsLNT6LP7973+Tg4MDbdmyReU2gry8PNqyZQs5OjpSeHi46DxnZ2eVM43/+c9/VO6vTU9PJxsbG9F5KSkpZG5uTnPmzKFly5aRlZUVff755/Trr7/S7NmzycLCgr755hvRef/6178oODiYCgoKqKSkhCZOnEguLi5Ce2pqqlr1tW3bljZt2lRufWlpKQUFBVGLFi3U+jwGDx5M7du3r/Bs8PHjx8nLy4uGDBkiOq9FixYqZ0anTZtG9+7dE15nZGSQlZWV6LwdO3aQpaUlrVq1ilatWkWOjo70888/U3JyMsXExJC9vb1a94327NlT5bL5t99+S82aNRNeHz9+XK36XnvttQr/Vj569Ih8fX3J09OTzzzWQdx5ZFq1fv168vHxIQMDA+EGbQMDA/Lx8aH4+Hi1sqKiomjHjh2Vts+YMYNCQ0NF5z148ICmTZtGbm5uJJfLycjIiBwcHGjw4MFqPchD9OwP5aBBg4Tj9PPzE+7hIyL6448/aMOGDaLzfvrpp5fepxYZGSn6suuL+zRr1ky4SV4qlZJEIqFmzZqp1RkgIvriiy9o3bp1lbbPnDmT3n33XbUyL168SB988AEpFArhZ8XQ0JD8/Pxo69atamXdunWL3nzzTeFYHRwcVDoHGzdupO+//15U1vz584X7JSsyfvx4te7pKioqonHjxpGRkRFJpVKSy+Ukl8tJKpWSkZERjR8/noqKikTnjR07ln766adK2yMiIuidd94RnUf0rAP5/Pv34tK8eXO17k8kIrp06RI5OzuTgYEBGRoakoWFhfCwGhHRqlWraPr06aLzpk6dWuk9fqWlpRQYGKhWZ+X+/fsUEBBAEomELC0tqU2bNtSmTRuytLQkqVRKffr0oQcPHojOCwwMfOn3aPny5fTWW2+JziMi2rRpE9nZ2Qm/s88XuVxOEydOVLlXsyrp6elkaWlJNjY21KJFCzIyMlL5XV6+fLlwT6oYH3/8caX36Obn55OPjw93HusgCRGRrs9+srqvtLQUd+/eBfDsskZFQ1Vcu3YNtra2FQ73UB2azBObVVRUhLKysnLDh2izNgBITk5Ghw4dRF3uvHLlispwJNUZdqUqjx8/hkwmU+vy63NEhNu3b0OpVFb6syLWhQsXUFxcjDZt2qhcetYmsZ9tfn4+0tPTVT4LLy+vag+VUpkrV65ALperXMoW686dO7h8+TKUSiWaNWsGR0fHatXw+PFjJCcno7i4GG+++aZaQwf9U1lZGR4/flzp96msrAzXr19XuZQqRmZmZoVDJ7Vp06batVbk2LFjaNCgAdzd3dXa7+nTpzh+/LjK5+Hl5QWFQqF2DTdv3sSOHTtQXFyMt956q9zwU+p48OABbty4gddff73C9kePHuH48ePw9/ev9nswPaTjzitjAoVCofLQiT7l6XNtnKdfeZquzd3dna5evcp51aTpz+Odd95Ra6gdzlOl6c+X6QY/MMP0Bmn4JLgm8/S5Ns7TrzxN15aTk4PS0lLOqyZNfx6HDh3CkydPOK+aNP35Mt3gziNjjDHGGBONO4+MMcYYY0w07jwyxhhjjDHRuPPI9MaLAwnrW54+18Z5+pWn6drYq+HPgzHN484j0xv16SEIzqu7eZqujb0a/jwY07yaGfiMMRHOnTsHW1tbvczTl9pu3boFIoKNjY3K+hen4tOEXbt2qTUdYH3NI6JyZ7Y0/bPyww8/qD2NJ+f9j6Z/VmbOnKnW9Iycp0rTny/TkZobFYjVN+7u7jR37lyNjemlyTx9ro2I6N69exQcHEz29vY0btw4Kisro9DQUGHGFF9fX7XGXuvXrx+tXr1aZQrBV1Gf8oqKiuizzz6jLl26UGRkJBERzZs3jxo2bEgNGzakkJAQlWkGqxIeHk6HDh165brqax4R0bJly+jDDz8UZkZZvXo1ubq60muvvUYzZsxQmba0KgsXLqScnByN1Vbf8oiIfvvtN5o1axYdPnyYiIj27dtHffr0od69e9MPP/yg0fdi+oE7j0xrJBIJNW7cmGQyGfXu3Zs2bdqk1h91bebpc21ERCNHjiR3d3datmwZ+fv704ABA8jDw4MOHz5MKSkp1LFjR7WmEHs+LaS5uTmNGzeO0tLSql1bfcv79NNPydbWlj777DNydXWlf//739SiRQtas2YNrV27llxcXOjjjz9WqzapVEqtWrWiyMhIunnzZrVrq4958+bNI4VCQcHBwWRjY0ORkZHUuHFj+uqrr2j+/PnUpEkTmj17tlr1yWQy6tGjB61fv56Ki4tfqb76lrdy5UoyMDAgLy8vMjMzo19++YUUCgWNGjWKxo4dSyYmJmpPacn0H3cemdZIJBK6fv06bd26lfr3708GBgbUpEkT+uyzz+jcuXM6zdPn2oiImjVrRsnJyURElJubSxKJhP7880+h/fDhw9S8eXO16jt79iwtWbKE2rZtS1KplDw9PWnZsmV0//59teurT3n29vbCXMyXLl0iqVRK27ZtE9r//PNPcnBwUKu2vXv30oQJE8jKyooMDQ0pMDCQfvvtN3r69KlatdXHPGdnZ9q8eTMREWVkZJBMJqM1a9YI7Vu2bCEXFxe16lu1ahUNGDCADA0NqXHjxjRhwgQ6ffq02rXVxzw3Nzf68ccfiYho//79JJfLacWKFUL7qlWryNXVtVrZTH9x55FpjUQioVu3bgmvb9y4QfPnz6dWrVoJl16jo6N1kqfPtRERNWjQQOXSkqGhocof98uXL1PDhg2rXd/Ro0dpzJgxZG5uTiYmJhQSEkL79u3jvAqYmJjQ33//Lbw2NDSkM2fOCK+vXLlCDRo0qFZtJSUlFB8fT7179yaZTEa2trY0c+ZMunDhAudVoqrPIycnp9qfx61bt+ibb76hNm3akFQqpY4dO9KPP/5I+fn5nFeJij6PF/9Wqfv7wWoH7jwyrZFKpSr/A3/RgQMHaOjQoWp1gDSZp8+1ERF5enrS8uXLiYjo999/J4VCQYsWLRLao6KiyN3dXXTePztTzxUWFtKqVauoc+fOJJVKOa8Cr732Gq1fv56IiI4dO0ZGRkYUExMjtK9fv55atWr1yrX9/fffNGfOHHJwcNDIsdbVPCcnJ9q1axcREWVnZ5NUKqUNGzYI7Tt37iRHR8dXru/QoUM0fPhw4d5WzquYnZ2dcE/r9evXSSKR0M6dO4X2xMREsrOzE53HagfuPDKtqeyP1IvUedBAk3n6XBsR0Zo1a0gmk5GLiwsZGxvTxo0bydbWlgYNGkQffPABGRkZCZ1LTdWXlZXFeRVYsmQJyeVy6tGjBzVq1Ii+//57srGxoalTp9L06dPJ3Nyc5s6dq7HalEqlyi0KnKfq888/pyZNmtCoUaPIycmJpk+fTi1atKCoqChauXIl2dvb06effio672X/8CN69nv7/LIs55UXFhZGrVq1oq+++oq8vb1p+PDh1KZNG9q1axft3r2b2rZtSyNHjhSdx2oH7jwyrRkxYoRalz9qMk+fa3vu8OHDtHDhQuHex7Nnz9KHH35IwcHBFBsbq1ZWt27d6MGDBxqrrb7l/frrrxQeHk5r164lomdnk7t06UJeXl70xRdfqHXvnqOjI929e1djtdW3vKdPn9LXX39N/fr1o/nz55NSqaR169aRvb09NW7cmEaMGEEFBQWi88T8Q0Md9S2voKCARo8eTe7u7jRmzBgqLi6mb7/9loyMjEgikVC3bt00+n5MP0iIeARVph/WrVuHwMBANGzYUO/y9Lk2beQlJyejQ4cOMDY25jwdZgHAtWvXYGtrC6lUM3M6cB6rCUVFRSgtLYVCodB1KUwL+LeJ6Y2xY8fi1q1bepmnz7VpI69Pnz64fv065+k4CwDc3NyQk5PDeXqSZ2ZmhsuXL3NeFeRyeYUdR03Xx3SDO49Mb2j6JLgm8/S5Ns7Trzx9ro3zOK+u5THd4M4jY4wxxhgTjTuPjDHGGGNMNO48MsYYY4wx0bjzyBgrRyKRcJ4eZHEe53Ee00fceWR6w8HBAYaGhnqZp8+1aSNP32+S1+c8fa6N8zivruUx3eBxHhljTI/997//ha2tLWQyGefpQd7hw4fRsWNHjY3jyXmsNuLOI9OaRo0aibpEcf/+/RrP0+fatJHXvn17UXnHjx/nPC3X9u6774rabsuWLZxXA3mTJk0Std3ixYs5rwbyWO1goOsCWN21dOlS4Wsiwvjx4zF37lxYW1vrPE+fa9NGXlBQkEpeREQExo0bB0tLS86r4drMzc1VXq9duxb9+/ev9kwcnPdqeSdOnFB5ffjwYXh5ecHExERYp859epz3anmsltDwdIeMVcrU1JQuXbqkl3n6XBvn6VeePtfGeZxX1/KYfuIHZhhjjDHGmGjceWSMMcYYY6Jx55ExxhhjjInGD8wwrfnnU3glJSX4+uuvy93wXt2n+l4lT59r00be999/r/K6rKwMsbGxsLKyUln/ySefcJ6Wa0tISFB5rVQqsW/fPpw5c0ZlfWBgIOfVQN6pU6dUXhMRMjMzUVBQoLLew8OD82ogj9UOPFQP05ru3btXuY1EIsH+/ftrPE+fa9NGnpOTk6i8y5cvc56Wa5NKq77gI5FI8PTpU86roTyJRFLh4NXP13NezeWx2oE7j4wxxuqtv//+W9R2Dg4OnFcDeax24M4jq1HPf9w0Ne6XJvP0uTZt5DHGGGPVwQ/MsBoRHR0Nd3d3yOVyyOVyuLu74+eff9aLPH2uTRt5+/btQ79+/eDs7AxnZ2f069cPe/fu5Twd1JaVlYXw8HC8/fbbePvttxEeHo7MzEzO01HegwcPsHDhQoSGhiI0NBQLFy4UPYsT52k+j+kv7jwyrZs9ezYmTJiA/v37Y+PGjdi4cSP69++PTz/9FLNnz9Zpnj7Xpo28//znPwgICIBCocCECRMwYcIEmJmZ4Z133sGKFSs4rwZr27x5M9zd3ZGeng5PT094enri+PHjaNu2LTZv3sx5NZx36NAhODo64vvvv8eDBw/w4MEDLFu2DE5OTjh06BDn1XAe03M1NRo5q7+srKxo7dq15davXbuWGjdurNM8fa5NG3nNmzenZcuWlVu/fPlysrW15bwarK1ly5Y0a9ascutnz55NLVu25LwaznN3d6fRo0dTWVmZsK6srIzGjBlD7u7unFfDeUy/ceeRaZ25uTllZ2eXW5+VlUXm5uY6zdPn2rSR17BhQ7pw4UK59dnZ2dSwYUPOq8HaTExMKs0zMTHhvBrOk8vllJmZWW59ZmYmyeVyzqvhPKbf+LI107oPP/wQUVFR5db/+OOPGDJkiE7z9Lk2beQFBgZi69at5dZv374d/fr147warK1bt25ISkoqt/7w4cPo0qUL59Vw3htvvIHz58+XW3/+/Hl4enpyXg3nMf3GT1szrfv444+xevVq2Nvb48033wQAHD16FFevXsWwYcNgaGgobCtm0GtN5ulzbdrI++qrr7Bw4UJ06tQJvr6+AIDU1FQkJyfjs88+g5mZmbCtmEGv61OepmtbuXIlZs+ejUGDBgmfbWpqKjZu3Igvv/wStra2wrZiBrzmvFfLi4+Px9SpU/Hxxx+r5K1YsQKRkZFwdXUVthUz4DXnvVoe02/ceWRaJ2bAa0CzA3yLzdPn2rSRJ2bA6+d5mhpAu67kabo2MYNdP88TM8Ay52k3T90Brznv1fKYfuPOI2OMsXpP7GDXgLgBrznv1fKYfuPOI6tR69atQ2BgIBo2bKh3efpcmzbykpOT0aFDBxgbG3OeDrMA4Nq1a7C1tRV9No3ztJvHGHs5/k1jNWrs2LG4deuWXubpc23ayOvTpw+uX7/OeTrOAgA3Nzfk5ORwnp7kmZmZiZ6rnPO0n8f0D3ceWY3S9IluTebpc22cp195+lwb53FeXctj+oc7j4wxxhhjTDTuPLIatWvXLjRv3hwA8PjxY6SkpOhNnj7Xpo28H374AU2bNn2ljPqap+naZs6cCUtLS87Tk7yhQ4eqDL3EebrNY/qHH5hhOnPy5Em88cYbGhu2QZN5+lxbdfPi4+ORkJCAkpISvP322xg3btwr1VCf8jRdG2OM1WZ85pGxeiAqKgohISFIS0vDhQsXEBYWhilTpnCeDmqryn//+1+MHDmS82ow78mTJzh8+DDOnTtXrq2oqAirV6/mvBrMY7WAlqc/ZKxSGRkZJJVK9TJPn2urTp6bmxt98cUXwutffvmFGjRoUO33r095mq6tKrr+WalveVlZWeTg4EASiYSkUil17dqVrl+/LrTn5uZyXg3msdrBQNedV8aY9l2+fBnDhw8XXg8ePBihoaG4efMmmjVrxnk1WFtCQkKV78d5NZc3bdo0uLu7Iy0tDQ8fPsTEiRPRuXNnJCYmokWLFmplcd6r57HagTuPTGuq+iN/5coVneXpc23ayCsuLlYZXFwqlcLIyAhPnjxRK6c+5mm6tqCgIGGqtspIJBLOq6G8lJQU7N27F1ZWVrCyssJvv/2Gf//73+jSpQsOHDig9qD8nPdqeax24M4j05qgoKAqt1H3fxqaytPn2rSRBwCzZs1CgwYNhNclJSX4+uuvYW5uLqxbvHgx52k5q1mzZvjPf/6DAQMGVNiekZEBLy8vUVmc9+p5T548gYHB//5XKJFIEBUVhfDwcPj7+2Pt2rWiszjv1fNY7cCdR6Y1SqVSb/P0uTZt5HXt2hVZWVkq6/z8/FQu8anTGa1PeZquzcvLC+np6ZV2fqo6q8Z5ms1r06YN0tLS4OrqqrJ++fLlAIDAwEDRWZz36nmsltDVzZaMMVYfHTp0iHbt2lVpe0FBASUmJnJeDeXNnz+f+vTpU2n7+PHjSSKRcF4N5bHagcd5ZFpT1X17z4n9l6km8/S5Nm3kMcYYY5rCnUemNVKp6jCiFV1Okkgkoge61mSePtemjby5c+eK2m727Nmcp8UsMTIzMxEYGIjs7GzO4zzOY3qJO4+sxigUCpw8eRItW7bUuzx9rk0Tee3bt6+0TSKRICsrC0VFRaI7o/UpT9O1VUUfZiPiPM7T1zymH/iBGcbqgRMnTlS4PiMjA9OnT8eZM2cwevRozquB2hhjrLbj6QkZq4euXLmCoUOHomPHjjA3N8fZs2excuVKztNBbYwxVttw55GxeuTu3bv4+OOP0aZNG9y8eRMpKSmIj49Hq1atOK+Ga2OMsdqKL1uzGiORSNQe2Lqm8vS5Nk3kFRYWYuHChVi8eDFcXFzw22+/oVevXpyng9oaNWr00s+yrKyM8ziv3uax2oE7j0xr/vlHpaCgAO3bty/3JPH9+/drPE+fa9NGnrOzMx49eoSPP/4YISEhkEgkOHXqVLntPDw8OE/LtS1dulTUdmJxHufVpTxWO/DT1kxr4uLiRG03fPjwGs/T59q0kfdip/Ofw/48f13doYTqep6maxPj6dOnkMlknMd5nMf0EncemU7p8x8pfa5N3by///5b1HYODg6cp8WsqmRnZyM6OhqrV6/GzZs3OY/zOI/ppxqby4axF2RlZdHUqVPJxsZG7/L0uTZt5DHdKiwspJiYGOrcuTPJZDLy8fGhBQsWcB7ncR7TW9x5ZDVGn/9I6XNtmsjLzs6mDz74gPLy8sq1PXz4kEJCQujSpUucVwO1PXfkyBEKDQ0lMzMzcnd3J5lMRocOHVI7h/M4ry7mMf3GnUemdfr8R0qfa9Nk3ujRo2nKlCmVtk+dOpXGjRvHeTVQ28KFC8nNzY2aN29OkydPpoyMDCIiMjAwoLNnz4rO4TzOq4t5rHbgziPTGn3+I6XPtWkjr3Xr1nTs2LFK29PS0qh169acVwO1yWQymjlzJpWVlamsr+5ny3mcV5fyWO3AnUemNfr8R0qfa9NGnlwup5ycnErbc3JyyMTEhPNqoLb58+dTq1atyN7enqZOnUqnT58moup/tpzHeXUpj9UO3HlkWqPPf6T0uTZt5DVt2pT27dtXafvevXupadOmnFcDtT2XmJhIw4YNowYNGpCHhwfJZDI6fPiw2jmcx3l1MY/pN+48Mq3T5z9S+lybJvMGDhxIQUFBlbYHBgbSe++9x3k1UNs/5efn08qVK8nb25tkMhn5+vrSokWLOI/zOI/pLe48shqjz3+k9Lk2TeQdP36cjI2NKTg4mI4ePUoPHz6khw8fUmpqKr377rtkbGxM6enpnFcDtb3MqVOnaMKECdSkSRPO4zzOY3qLO49MJ/T5j5Q+1/Yqeb/99hs1adKEpFKpytKkSRPavn272nXUpzxN11aVkpISzuM8zmN6i2eYYVq3evVqvP/++zA2NlZZX1JSgtWrV2PUqFE6y9Pn2rSR9+TJE/zxxx+4cOECiAitW7dGr1690KBBA7Vy6mOeprLu3r2LwsJClRlpzp49i4ULF6KwsBBBQUEYPHgw59VQ3vfff1/lNhKJBB9//DHn1UAeqyV02HFl9YRUKqVbt26VW3/37l2SSqU6zdPn2rSRFxcXR0VFReXWFxcXU1xcHOfVQNYHH3xAkyZNEl7funWLGjVqRK+//joFBgaSoaEhrV69mvNqKM/R0VFlkclkZGdnp7LOycmJ82ooj9UO3HlkWieRSOj27dvl1mdkZFCjRo10mqfPtWkjT987t/qcp6ksR0dHSkxMFF5/++235OzsTKWlpcJrHx8fzquhvH8yNTWt1oxBnKedPKafDHR95pPVXe3bt4dEIoFEIsHbb78NA4P//bg9ffoUV65cQUBAgE7y9Lk2beQ9R0SQSCTl1l+7dg3m5uacVwNZubm5cHR0FF7v378f7777rvAZBwYGIiIigvNqKI8xpj7uPDKtCQoKAgBkZGSgd+/eMDU1FdqMjIzg6OgId3d3neTpc23ayNP3zq0+52m6NjMzMzx8+FC4Z+/YsWMIDQ0V2iUSCYqLizmvhvIYY+rjziPTmjlz5gAAHB0d8f7770MulwMAHj16hHXr1mHJkiVIT0/H06dPazxPn2vTRp6+d271OU/Ttb355pv4/vvv8dNPP2HLli149OgR3nrrLaE9Ozsb9vb2nFdDeYyxatDpRXNWrxw8eJCGDRtGDRs2pFatWtG0adNeOmdwTebpc22azIuNjaUnT54Ir/Pz8+mHH36gjh07VuuewvqUp6mskydPkpWVFRkZGZFUKqXPP/9cpX3o0KE0duxYzquhvLy8PJVFoVDQyZMny63nvJrJY7UDdx6ZVt28eZMiIiLIxcWFrK2tKTw8/JXmPNVknj7Xpo28F+lr57Y25Gki686dO7Rt2zZKTU0t17Zjxw66cuUK59VQnkQiURm7s7LXnFczeax24MvWTGv69++PQ4cOoW/fvli6dCkCAgIgk8mwcuVKnefpc23ayAOePWgQGxuL6Oho5OfnY9CgQSguLsa2bdvg5ubGeTVYm5WVFfLy8iq8V7Jnz55Yv369ykMhnKe9vAMHDoh+X87Tfh6rJXTde2V1l0wmo08//ZSys7NV1lf37Jkm8/S5Nm3k9evXj8zMzCgkJIR27NhBZWVlnKej2p7T52GJ6mMeY0w8qa47r6zuOnz4MB49egQvLy/4+Phg+fLluHv3rl7k6XNt2sjbtWsXQkND8eWXX6Jv376QyWTVzqpveZqu7TnS42GJ6mPe77//jj/++KPc+j/++AO7du3ivBrOY3pOp11XVi8UFBRQdHQ0derUiQwNDUkqldLSpUspPz9f53n6XJsm844cOUKjRo0ihUJB3t7etGzZMrpz5061z57VpzxN19auXTtq3749SaVSatu2LbVv315YPDw8SKFQ0MCBAzmvhvKea9u2Le3cubPc+l27dpGHhwfn1XAe0288tzWrUVlZWYiOjsYvv/yChw8fomfPnkhISNCLPH2uTVN5hYWFiI+PR0xMDI4dO4anT59i8eLFGDlyJBQKhdo11ac8TWV9+eWXwn8/++yzSof+adu2LefVQN5zJiYmOH/+fLl7JXNycvD666+jsLCQ82owj+k5XfdeWf1UVlZGW7dupf79++tdnj7Xpsm8zMxMmjJlCtnY2JBcLue8Gs7S52GJ6mNe06ZNad++feXW79mzh5o0acJ5NZzH9Bt3Hhmr5/S1c1sb8jSRpc/DEtWnvDFjxlDbtm3p4sWLwroLFy6Qh4cHhYaGcl4N5zH9xp1HxhirYfo+Jmh9yyMievjwIb355ptkYGBAjo6O5OjoSDKZjLp3704PHjzgvBrOY/qN73lkjLEa9OIYnkOGDBHG8DQ0NMTJkyfVHjeS814t70VEhD179uDkyZMwMTGBp6cnunTpwnk6ymN6TIcdV8YYq3f0fUzQ+paXkpJCv/32m8q62NhYcnBwoCZNmtDo0aOpqKiI82ooj9UOPM4jY4zVIH0fE7S+5c2dOxdnz54VXp8+fRqjR49Gz549MX36dPz222+IiIjgvBrKY7WErnuvjDFWH+nrmKD1Lc/Gxob++usv4fXMmTOpU6dOwusNGzaQq6sr59VQHqsduPPIGGM6ps/DEtX1PGNjY7p69arwulOnTvTVV18Jr69cuUKmpqacV0N5rHbgy9aMMaZjr732GhYsWIBr165h3bp1nFeDeU2bNsWVK1cAACUlJTh+/DjefPNNof3Ro0cwNDTkvBrKY7WErnuvjDHGmK6MGzeOfH196dChQzRp0iRq3LgxFRcXC+1r1qyhDh06cF4N5bHagTuPjDHG6q07d+5Qly5dSCKRkEKhoC1btqi0v/XWWzRz5kzOq6E8VjvwOI+MMcbqvby8PJiamkImk6msv3//PkxNTWFkZMR5NZjH9Bt3HhljjDHGmGj8wAxjjDHGGBONO4+MMcYYY0w07jwyxhhjjDHRuPPIGKsTunXrhokTJ+q6DMYYq/O488gYY3XcyZMnERISAnt7e5iYmMDV1RXfffedrstijNVSBrougDHGmHalp6fD2toaa9asgb29PVJSUjBmzBjIZDKEh4frujzGWC3DZx4ZY3VGWVkZwsPDYW5uDisrK8yaNQvPRyP75Zdf0KFDBygUCtjY2GDw4MG4ffu2sO+DBw8wZMgQNGnSBCYmJmjVqhVWrVoltP/3v//FoEGDYGFhAUtLSwwYMAA5OTlC+4gRIxAUFISFCxeiWbNmaNy4McLCwlBaWipsU1UNiYmJkEgk+OOPP9C+fXuYmJjgrbfewu3bt7Fr1y64urrCzMwMgwcPxuPHj4X9lEolIiIi4OTkBBMTE3h6emLTpk1C+8iRI/Hdd9/B398fLVu2xNChQ/HRRx9hy5YtGv3+M8bqB+48MsbqjLi4OBgYGODYsWP47rvvsHjxYvz8888AgNLSUsybNw8nT57Etm3bkJOTgxEjRgj7zpo1C+fOncOuXbtw/vx5REVFwcrKSti3d+/eUCgUSEpKQnJyMkxNTREQEICSkhIh48CBA7h06RIOHDiAuLg4xMbGIjY2VmivqobnvvjiCyxfvhwpKSlCp3Xp0qVYu3Ytdu7ciT///BPLli0Tto+IiMDq1auxcuVKnD17Fp9++imGDh2KgwcPVvq9ysvLg6WlZTW/04yxek2n89swxpiG+Pv7k6urKymVSmHdtGnTyNXVtcLt//rrLwJAjx49IiKi/v3700cffVThtr/88gu99tprKtnFxcVkYmJCf/zxBxERDR8+nBwcHKisrEzYZuDAgfT+++9XWvM/azhw4AABoL179wrbREREEAC6dOmSsG7s2LHUu3dvIiIqKiqiBg0aUEpKikp2aGgohYSEVPi+ycnJZGBgINTOGGPq4DOPjLE6480334REIhFe+/r64sKFC3j69CnS09PRv39/tGjRAgqFAv7+/gCAq1evAgDGjx+P9evXo127dpg6dSpSUlKEnJMnT+LixYtQKBQwNTWFqakpLC0tUVRUhEuXLgnbvf766yrTszVr1kzlsnRVNTzn4eEhfN20aVM0aNAALVu2VFn3PPfixYt4/PgxevbsKdRmamqK1atXq9T23JkzZzBgwADMmTMHvXr1UuO7yxhjz/ADM4yxOq+oqAi9e/dG79698euvv6JJkya4evUqevfuLVx27tOnD/7++2/8/vvv2LNnD95++22EhYVh4cKFKCgogJeXF3799ddy2U2aNBG+NjQ0VGmTSCRQKpUAgMLCwiprqChHIpG8NLegoAAAsHPnTjRv3lxlO2NjY5XX586dw9tvv40xY8bg888/r/obxxhjFeDOI2Oszjh69KjK69TUVLRq1QqZmZm4d+8eIiMjYW9vDwBIS0srt3+TJk0wfPhwDB8+HF26dMGUKVOwcOFCvPHGG4iPj4e1tTXMzMyqVZvYGtTl5uYGY2NjXL16VTiTWZGzZ8/irbfewvDhw/H111+/8vsyxuovvmzNGKszrl69ikmTJiErKwvr1q3DsmXLMGHCBLRo0QJGRkZYtmwZLl++jISEBMybN09l39mzZ2P79u24ePEizp49ix07dsDV1RUAMGTIEFhZWWHAgAFISkrClStXkJiYiE8++QTXrl0TVZuYGqpDoVBg8uTJ+PTTTxEXF4dLly7h+PHjWLZsGeLi4gA8u1TdvXt39OrVC5MmTUJubi5yc3Nx586dV35/xlj9w51HxlidMWzYMDx58gTe3t4ICwvDhAkTMGbMGDRp0gSxsbHYuHEj3NzcEBkZiYULF6rsa2RkhBkzZsDDwwNdu3aFTCbD+vXrAQANGjTAoUOH0KJFC7z77rtwdXVFaGgoioqKRJ+JFFNDdc2bNw+zZs1CREQEXF1dERAQgJ07d8LJyQkAsGnTJty5cwdr1qxBs2bNhKVjx44aeX/GWP0iIfr/g6AxxhhjjDFWBT7zyBhjjDHGROPOI2OMMcYYE407j4wxxhhjTDTuPDLGGGOMMdG488gYY4wxxkTjziNjjDHGGBONO4+MMcYYY0w07jwyxhhjjDHRuPPIGGOMMcZE484jY4wxxhgTjTuPjDHGGGNMNO48MsYYY4wx0f4fr1+ZCixXE64AAAAASUVORK5CYII=",
      "text/plain": [
       "<Figure size 640x480 with 2 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "At_v_Sc = At_v_Sc.pivot(index=\"basename1\", columns=\"basename2\", values=\"emd_score\")\n",
    "sns.heatmap(At_v_Sc, cmap = 'plasma_r', vmin=0, vmax=0.0185)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>basename2</th>\n",
       "      <th>AtAHL-pH_7_0</th>\n",
       "      <th>AtAHL-pH_7_3</th>\n",
       "      <th>AtAHL-pH_7_6</th>\n",
       "      <th>AtAHL-pH_7_9</th>\n",
       "      <th>AtAHL-pH_8_2</th>\n",
       "      <th>AtCNP-pH_7_0</th>\n",
       "      <th>AtCNP-pH_7_3</th>\n",
       "      <th>AtCNP-pH_7_6</th>\n",
       "      <th>AtCNP-pH_7_9</th>\n",
       "      <th>AtCNP-pH_8_2</th>\n",
       "      <th>AtSAL1-pH_7_0</th>\n",
       "      <th>AtSAL1-pH_7_3</th>\n",
       "      <th>AtSAL1-pH_7_6</th>\n",
       "      <th>AtSAL1-pH_7_9</th>\n",
       "      <th>AtSAL1-pH_8_2</th>\n",
       "      <th>ScHAL2-pH_7_0</th>\n",
       "      <th>ScHAL2-pH_7_3</th>\n",
       "      <th>ScHAL2-pH_7_6</th>\n",
       "      <th>ScHAL2-pH_7_9</th>\n",
       "      <th>ScHAL2-pH_8_2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>basename1</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AtAHL-pH_7_0</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002695</td>\n",
       "      <td>0.004700</td>\n",
       "      <td>0.004700</td>\n",
       "      <td>0.004700</td>\n",
       "      <td>0.004700</td>\n",
       "      <td>0.001272</td>\n",
       "      <td>0.006951</td>\n",
       "      <td>0.002798</td>\n",
       "      <td>0.002798</td>\n",
       "      <td>0.002798</td>\n",
       "      <td>0.002798</td>\n",
       "      <td>0.003905</td>\n",
       "      <td>0.003905</td>\n",
       "      <td>0.002881</td>\n",
       "      <td>0.002881</td>\n",
       "      <td>0.002881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AtAHL-pH_7_3</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002695</td>\n",
       "      <td>0.004700</td>\n",
       "      <td>0.004700</td>\n",
       "      <td>0.004700</td>\n",
       "      <td>0.004700</td>\n",
       "      <td>0.001272</td>\n",
       "      <td>0.006951</td>\n",
       "      <td>0.002798</td>\n",
       "      <td>0.002798</td>\n",
       "      <td>0.002798</td>\n",
       "      <td>0.002798</td>\n",
       "      <td>0.003905</td>\n",
       "      <td>0.003905</td>\n",
       "      <td>0.002881</td>\n",
       "      <td>0.002881</td>\n",
       "      <td>0.002881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AtAHL-pH_7_6</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002695</td>\n",
       "      <td>0.004700</td>\n",
       "      <td>0.004700</td>\n",
       "      <td>0.004700</td>\n",
       "      <td>0.004700</td>\n",
       "      <td>0.001272</td>\n",
       "      <td>0.006951</td>\n",
       "      <td>0.002798</td>\n",
       "      <td>0.002798</td>\n",
       "      <td>0.002798</td>\n",
       "      <td>0.002798</td>\n",
       "      <td>0.003905</td>\n",
       "      <td>0.003905</td>\n",
       "      <td>0.002881</td>\n",
       "      <td>0.002881</td>\n",
       "      <td>0.002881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AtAHL-pH_7_9</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002695</td>\n",
       "      <td>0.004700</td>\n",
       "      <td>0.004700</td>\n",
       "      <td>0.004700</td>\n",
       "      <td>0.004700</td>\n",
       "      <td>0.001272</td>\n",
       "      <td>0.006951</td>\n",
       "      <td>0.002798</td>\n",
       "      <td>0.002798</td>\n",
       "      <td>0.002798</td>\n",
       "      <td>0.002798</td>\n",
       "      <td>0.003905</td>\n",
       "      <td>0.003905</td>\n",
       "      <td>0.002881</td>\n",
       "      <td>0.002881</td>\n",
       "      <td>0.002881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AtAHL-pH_8_2</th>\n",
       "      <td>0.002697</td>\n",
       "      <td>0.002697</td>\n",
       "      <td>0.002697</td>\n",
       "      <td>0.002697</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.007393</td>\n",
       "      <td>0.007393</td>\n",
       "      <td>0.007393</td>\n",
       "      <td>0.007393</td>\n",
       "      <td>0.003224</td>\n",
       "      <td>0.009560</td>\n",
       "      <td>0.003704</td>\n",
       "      <td>0.003704</td>\n",
       "      <td>0.003704</td>\n",
       "      <td>0.003704</td>\n",
       "      <td>0.006315</td>\n",
       "      <td>0.006315</td>\n",
       "      <td>0.002163</td>\n",
       "      <td>0.002163</td>\n",
       "      <td>0.002163</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AtCNP-pH_7_0</th>\n",
       "      <td>0.004345</td>\n",
       "      <td>0.004345</td>\n",
       "      <td>0.004345</td>\n",
       "      <td>0.004345</td>\n",
       "      <td>0.006830</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.003922</td>\n",
       "      <td>0.003074</td>\n",
       "      <td>0.004628</td>\n",
       "      <td>0.004628</td>\n",
       "      <td>0.004628</td>\n",
       "      <td>0.004628</td>\n",
       "      <td>0.001885</td>\n",
       "      <td>0.001885</td>\n",
       "      <td>0.006150</td>\n",
       "      <td>0.006150</td>\n",
       "      <td>0.006150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AtCNP-pH_7_3</th>\n",
       "      <td>0.004345</td>\n",
       "      <td>0.004345</td>\n",
       "      <td>0.004345</td>\n",
       "      <td>0.004345</td>\n",
       "      <td>0.006830</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.003922</td>\n",
       "      <td>0.003074</td>\n",
       "      <td>0.004628</td>\n",
       "      <td>0.004628</td>\n",
       "      <td>0.004628</td>\n",
       "      <td>0.004628</td>\n",
       "      <td>0.001885</td>\n",
       "      <td>0.001885</td>\n",
       "      <td>0.006150</td>\n",
       "      <td>0.006150</td>\n",
       "      <td>0.006150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AtCNP-pH_7_6</th>\n",
       "      <td>0.004345</td>\n",
       "      <td>0.004345</td>\n",
       "      <td>0.004345</td>\n",
       "      <td>0.004345</td>\n",
       "      <td>0.006830</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.003922</td>\n",
       "      <td>0.003074</td>\n",
       "      <td>0.004628</td>\n",
       "      <td>0.004628</td>\n",
       "      <td>0.004628</td>\n",
       "      <td>0.004628</td>\n",
       "      <td>0.001885</td>\n",
       "      <td>0.001885</td>\n",
       "      <td>0.006150</td>\n",
       "      <td>0.006150</td>\n",
       "      <td>0.006150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AtCNP-pH_7_9</th>\n",
       "      <td>0.004345</td>\n",
       "      <td>0.004345</td>\n",
       "      <td>0.004345</td>\n",
       "      <td>0.004345</td>\n",
       "      <td>0.006830</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.003922</td>\n",
       "      <td>0.003074</td>\n",
       "      <td>0.004628</td>\n",
       "      <td>0.004628</td>\n",
       "      <td>0.004628</td>\n",
       "      <td>0.004628</td>\n",
       "      <td>0.001885</td>\n",
       "      <td>0.001885</td>\n",
       "      <td>0.006150</td>\n",
       "      <td>0.006150</td>\n",
       "      <td>0.006150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AtCNP-pH_8_2</th>\n",
       "      <td>0.001172</td>\n",
       "      <td>0.001172</td>\n",
       "      <td>0.001172</td>\n",
       "      <td>0.001172</td>\n",
       "      <td>0.002967</td>\n",
       "      <td>0.003908</td>\n",
       "      <td>0.003908</td>\n",
       "      <td>0.003908</td>\n",
       "      <td>0.003908</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.005927</td>\n",
       "      <td>0.001976</td>\n",
       "      <td>0.001976</td>\n",
       "      <td>0.001976</td>\n",
       "      <td>0.001976</td>\n",
       "      <td>0.003411</td>\n",
       "      <td>0.003411</td>\n",
       "      <td>0.002527</td>\n",
       "      <td>0.002527</td>\n",
       "      <td>0.002527</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AtSAL1-pH_7_0</th>\n",
       "      <td>0.006862</td>\n",
       "      <td>0.006862</td>\n",
       "      <td>0.006862</td>\n",
       "      <td>0.006862</td>\n",
       "      <td>0.009431</td>\n",
       "      <td>0.003283</td>\n",
       "      <td>0.003283</td>\n",
       "      <td>0.003283</td>\n",
       "      <td>0.003283</td>\n",
       "      <td>0.006351</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.006902</td>\n",
       "      <td>0.006902</td>\n",
       "      <td>0.006902</td>\n",
       "      <td>0.006902</td>\n",
       "      <td>0.004986</td>\n",
       "      <td>0.004986</td>\n",
       "      <td>0.008726</td>\n",
       "      <td>0.008726</td>\n",
       "      <td>0.008726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AtSAL1-pH_7_3</th>\n",
       "      <td>0.002736</td>\n",
       "      <td>0.002736</td>\n",
       "      <td>0.002736</td>\n",
       "      <td>0.002736</td>\n",
       "      <td>0.003619</td>\n",
       "      <td>0.004895</td>\n",
       "      <td>0.004895</td>\n",
       "      <td>0.004895</td>\n",
       "      <td>0.004895</td>\n",
       "      <td>0.002097</td>\n",
       "      <td>0.006836</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.005084</td>\n",
       "      <td>0.005084</td>\n",
       "      <td>0.003665</td>\n",
       "      <td>0.003665</td>\n",
       "      <td>0.003665</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AtSAL1-pH_7_6</th>\n",
       "      <td>0.002736</td>\n",
       "      <td>0.002736</td>\n",
       "      <td>0.002736</td>\n",
       "      <td>0.002736</td>\n",
       "      <td>0.003619</td>\n",
       "      <td>0.004895</td>\n",
       "      <td>0.004895</td>\n",
       "      <td>0.004895</td>\n",
       "      <td>0.004895</td>\n",
       "      <td>0.002097</td>\n",
       "      <td>0.006836</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.005084</td>\n",
       "      <td>0.005084</td>\n",
       "      <td>0.003665</td>\n",
       "      <td>0.003665</td>\n",
       "      <td>0.003665</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AtSAL1-pH_7_9</th>\n",
       "      <td>0.002736</td>\n",
       "      <td>0.002736</td>\n",
       "      <td>0.002736</td>\n",
       "      <td>0.002736</td>\n",
       "      <td>0.003619</td>\n",
       "      <td>0.004895</td>\n",
       "      <td>0.004895</td>\n",
       "      <td>0.004895</td>\n",
       "      <td>0.004895</td>\n",
       "      <td>0.002097</td>\n",
       "      <td>0.006836</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.005084</td>\n",
       "      <td>0.005084</td>\n",
       "      <td>0.003665</td>\n",
       "      <td>0.003665</td>\n",
       "      <td>0.003665</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AtSAL1-pH_8_2</th>\n",
       "      <td>0.002736</td>\n",
       "      <td>0.002736</td>\n",
       "      <td>0.002736</td>\n",
       "      <td>0.002736</td>\n",
       "      <td>0.003619</td>\n",
       "      <td>0.004895</td>\n",
       "      <td>0.004895</td>\n",
       "      <td>0.004895</td>\n",
       "      <td>0.004895</td>\n",
       "      <td>0.002097</td>\n",
       "      <td>0.006836</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.005084</td>\n",
       "      <td>0.005084</td>\n",
       "      <td>0.003665</td>\n",
       "      <td>0.003665</td>\n",
       "      <td>0.003665</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ScHAL2-pH_7_0</th>\n",
       "      <td>0.004223</td>\n",
       "      <td>0.004223</td>\n",
       "      <td>0.004223</td>\n",
       "      <td>0.004223</td>\n",
       "      <td>0.006824</td>\n",
       "      <td>0.002205</td>\n",
       "      <td>0.002205</td>\n",
       "      <td>0.002205</td>\n",
       "      <td>0.002205</td>\n",
       "      <td>0.004005</td>\n",
       "      <td>0.005462</td>\n",
       "      <td>0.005623</td>\n",
       "      <td>0.005623</td>\n",
       "      <td>0.005623</td>\n",
       "      <td>0.005623</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.005975</td>\n",
       "      <td>0.005975</td>\n",
       "      <td>0.005975</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ScHAL2-pH_7_3</th>\n",
       "      <td>0.004223</td>\n",
       "      <td>0.004223</td>\n",
       "      <td>0.004223</td>\n",
       "      <td>0.004223</td>\n",
       "      <td>0.006824</td>\n",
       "      <td>0.002205</td>\n",
       "      <td>0.002205</td>\n",
       "      <td>0.002205</td>\n",
       "      <td>0.002205</td>\n",
       "      <td>0.004005</td>\n",
       "      <td>0.005462</td>\n",
       "      <td>0.005623</td>\n",
       "      <td>0.005623</td>\n",
       "      <td>0.005623</td>\n",
       "      <td>0.005623</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.005975</td>\n",
       "      <td>0.005975</td>\n",
       "      <td>0.005975</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ScHAL2-pH_7_6</th>\n",
       "      <td>0.003012</td>\n",
       "      <td>0.003012</td>\n",
       "      <td>0.003012</td>\n",
       "      <td>0.003012</td>\n",
       "      <td>0.002259</td>\n",
       "      <td>0.006956</td>\n",
       "      <td>0.006956</td>\n",
       "      <td>0.006956</td>\n",
       "      <td>0.006956</td>\n",
       "      <td>0.002867</td>\n",
       "      <td>0.009241</td>\n",
       "      <td>0.003919</td>\n",
       "      <td>0.003919</td>\n",
       "      <td>0.003919</td>\n",
       "      <td>0.003919</td>\n",
       "      <td>0.005777</td>\n",
       "      <td>0.005777</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ScHAL2-pH_7_9</th>\n",
       "      <td>0.003012</td>\n",
       "      <td>0.003012</td>\n",
       "      <td>0.003012</td>\n",
       "      <td>0.003012</td>\n",
       "      <td>0.002259</td>\n",
       "      <td>0.006956</td>\n",
       "      <td>0.006956</td>\n",
       "      <td>0.006956</td>\n",
       "      <td>0.006956</td>\n",
       "      <td>0.002867</td>\n",
       "      <td>0.009241</td>\n",
       "      <td>0.003919</td>\n",
       "      <td>0.003919</td>\n",
       "      <td>0.003919</td>\n",
       "      <td>0.003919</td>\n",
       "      <td>0.005777</td>\n",
       "      <td>0.005777</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ScHAL2-pH_8_2</th>\n",
       "      <td>0.003012</td>\n",
       "      <td>0.003012</td>\n",
       "      <td>0.003012</td>\n",
       "      <td>0.003012</td>\n",
       "      <td>0.002259</td>\n",
       "      <td>0.006956</td>\n",
       "      <td>0.006956</td>\n",
       "      <td>0.006956</td>\n",
       "      <td>0.006956</td>\n",
       "      <td>0.002867</td>\n",
       "      <td>0.009241</td>\n",
       "      <td>0.003919</td>\n",
       "      <td>0.003919</td>\n",
       "      <td>0.003919</td>\n",
       "      <td>0.003919</td>\n",
       "      <td>0.005777</td>\n",
       "      <td>0.005777</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "basename2      AtAHL-pH_7_0  AtAHL-pH_7_3  AtAHL-pH_7_6  AtAHL-pH_7_9  \\\n",
       "basename1                                                               \n",
       "AtAHL-pH_7_0       0.000000      0.000000      0.000000      0.000000   \n",
       "AtAHL-pH_7_3       0.000000      0.000000      0.000000      0.000000   \n",
       "AtAHL-pH_7_6       0.000000      0.000000      0.000000      0.000000   \n",
       "AtAHL-pH_7_9       0.000000      0.000000      0.000000      0.000000   \n",
       "AtAHL-pH_8_2       0.002697      0.002697      0.002697      0.002697   \n",
       "AtCNP-pH_7_0       0.004345      0.004345      0.004345      0.004345   \n",
       "AtCNP-pH_7_3       0.004345      0.004345      0.004345      0.004345   \n",
       "AtCNP-pH_7_6       0.004345      0.004345      0.004345      0.004345   \n",
       "AtCNP-pH_7_9       0.004345      0.004345      0.004345      0.004345   \n",
       "AtCNP-pH_8_2       0.001172      0.001172      0.001172      0.001172   \n",
       "AtSAL1-pH_7_0      0.006862      0.006862      0.006862      0.006862   \n",
       "AtSAL1-pH_7_3      0.002736      0.002736      0.002736      0.002736   \n",
       "AtSAL1-pH_7_6      0.002736      0.002736      0.002736      0.002736   \n",
       "AtSAL1-pH_7_9      0.002736      0.002736      0.002736      0.002736   \n",
       "AtSAL1-pH_8_2      0.002736      0.002736      0.002736      0.002736   \n",
       "ScHAL2-pH_7_0      0.004223      0.004223      0.004223      0.004223   \n",
       "ScHAL2-pH_7_3      0.004223      0.004223      0.004223      0.004223   \n",
       "ScHAL2-pH_7_6      0.003012      0.003012      0.003012      0.003012   \n",
       "ScHAL2-pH_7_9      0.003012      0.003012      0.003012      0.003012   \n",
       "ScHAL2-pH_8_2      0.003012      0.003012      0.003012      0.003012   \n",
       "\n",
       "basename2      AtAHL-pH_8_2  AtCNP-pH_7_0  AtCNP-pH_7_3  AtCNP-pH_7_6  \\\n",
       "basename1                                                               \n",
       "AtAHL-pH_7_0       0.002695      0.004700      0.004700      0.004700   \n",
       "AtAHL-pH_7_3       0.002695      0.004700      0.004700      0.004700   \n",
       "AtAHL-pH_7_6       0.002695      0.004700      0.004700      0.004700   \n",
       "AtAHL-pH_7_9       0.002695      0.004700      0.004700      0.004700   \n",
       "AtAHL-pH_8_2       0.000000      0.007393      0.007393      0.007393   \n",
       "AtCNP-pH_7_0       0.006830      0.000000      0.000000      0.000000   \n",
       "AtCNP-pH_7_3       0.006830      0.000000      0.000000      0.000000   \n",
       "AtCNP-pH_7_6       0.006830      0.000000      0.000000      0.000000   \n",
       "AtCNP-pH_7_9       0.006830      0.000000      0.000000      0.000000   \n",
       "AtCNP-pH_8_2       0.002967      0.003908      0.003908      0.003908   \n",
       "AtSAL1-pH_7_0      0.009431      0.003283      0.003283      0.003283   \n",
       "AtSAL1-pH_7_3      0.003619      0.004895      0.004895      0.004895   \n",
       "AtSAL1-pH_7_6      0.003619      0.004895      0.004895      0.004895   \n",
       "AtSAL1-pH_7_9      0.003619      0.004895      0.004895      0.004895   \n",
       "AtSAL1-pH_8_2      0.003619      0.004895      0.004895      0.004895   \n",
       "ScHAL2-pH_7_0      0.006824      0.002205      0.002205      0.002205   \n",
       "ScHAL2-pH_7_3      0.006824      0.002205      0.002205      0.002205   \n",
       "ScHAL2-pH_7_6      0.002259      0.006956      0.006956      0.006956   \n",
       "ScHAL2-pH_7_9      0.002259      0.006956      0.006956      0.006956   \n",
       "ScHAL2-pH_8_2      0.002259      0.006956      0.006956      0.006956   \n",
       "\n",
       "basename2      AtCNP-pH_7_9  AtCNP-pH_8_2  AtSAL1-pH_7_0  AtSAL1-pH_7_3  \\\n",
       "basename1                                                                 \n",
       "AtAHL-pH_7_0       0.004700      0.001272       0.006951       0.002798   \n",
       "AtAHL-pH_7_3       0.004700      0.001272       0.006951       0.002798   \n",
       "AtAHL-pH_7_6       0.004700      0.001272       0.006951       0.002798   \n",
       "AtAHL-pH_7_9       0.004700      0.001272       0.006951       0.002798   \n",
       "AtAHL-pH_8_2       0.007393      0.003224       0.009560       0.003704   \n",
       "AtCNP-pH_7_0       0.000000      0.003922       0.003074       0.004628   \n",
       "AtCNP-pH_7_3       0.000000      0.003922       0.003074       0.004628   \n",
       "AtCNP-pH_7_6       0.000000      0.003922       0.003074       0.004628   \n",
       "AtCNP-pH_7_9       0.000000      0.003922       0.003074       0.004628   \n",
       "AtCNP-pH_8_2       0.003908      0.000000       0.005927       0.001976   \n",
       "AtSAL1-pH_7_0      0.003283      0.006351       0.000000       0.006902   \n",
       "AtSAL1-pH_7_3      0.004895      0.002097       0.006836       0.000000   \n",
       "AtSAL1-pH_7_6      0.004895      0.002097       0.006836       0.000000   \n",
       "AtSAL1-pH_7_9      0.004895      0.002097       0.006836       0.000000   \n",
       "AtSAL1-pH_8_2      0.004895      0.002097       0.006836       0.000000   \n",
       "ScHAL2-pH_7_0      0.002205      0.004005       0.005462       0.005623   \n",
       "ScHAL2-pH_7_3      0.002205      0.004005       0.005462       0.005623   \n",
       "ScHAL2-pH_7_6      0.006956      0.002867       0.009241       0.003919   \n",
       "ScHAL2-pH_7_9      0.006956      0.002867       0.009241       0.003919   \n",
       "ScHAL2-pH_8_2      0.006956      0.002867       0.009241       0.003919   \n",
       "\n",
       "basename2      AtSAL1-pH_7_6  AtSAL1-pH_7_9  AtSAL1-pH_8_2  ScHAL2-pH_7_0  \\\n",
       "basename1                                                                   \n",
       "AtAHL-pH_7_0        0.002798       0.002798       0.002798       0.003905   \n",
       "AtAHL-pH_7_3        0.002798       0.002798       0.002798       0.003905   \n",
       "AtAHL-pH_7_6        0.002798       0.002798       0.002798       0.003905   \n",
       "AtAHL-pH_7_9        0.002798       0.002798       0.002798       0.003905   \n",
       "AtAHL-pH_8_2        0.003704       0.003704       0.003704       0.006315   \n",
       "AtCNP-pH_7_0        0.004628       0.004628       0.004628       0.001885   \n",
       "AtCNP-pH_7_3        0.004628       0.004628       0.004628       0.001885   \n",
       "AtCNP-pH_7_6        0.004628       0.004628       0.004628       0.001885   \n",
       "AtCNP-pH_7_9        0.004628       0.004628       0.004628       0.001885   \n",
       "AtCNP-pH_8_2        0.001976       0.001976       0.001976       0.003411   \n",
       "AtSAL1-pH_7_0       0.006902       0.006902       0.006902       0.004986   \n",
       "AtSAL1-pH_7_3       0.000000       0.000000       0.000000       0.005084   \n",
       "AtSAL1-pH_7_6       0.000000       0.000000       0.000000       0.005084   \n",
       "AtSAL1-pH_7_9       0.000000       0.000000       0.000000       0.005084   \n",
       "AtSAL1-pH_8_2       0.000000       0.000000       0.000000       0.005084   \n",
       "ScHAL2-pH_7_0       0.005623       0.005623       0.005623       0.000000   \n",
       "ScHAL2-pH_7_3       0.005623       0.005623       0.005623       0.000000   \n",
       "ScHAL2-pH_7_6       0.003919       0.003919       0.003919       0.005777   \n",
       "ScHAL2-pH_7_9       0.003919       0.003919       0.003919       0.005777   \n",
       "ScHAL2-pH_8_2       0.003919       0.003919       0.003919       0.005777   \n",
       "\n",
       "basename2      ScHAL2-pH_7_3  ScHAL2-pH_7_6  ScHAL2-pH_7_9  ScHAL2-pH_8_2  \n",
       "basename1                                                                  \n",
       "AtAHL-pH_7_0        0.003905       0.002881       0.002881       0.002881  \n",
       "AtAHL-pH_7_3        0.003905       0.002881       0.002881       0.002881  \n",
       "AtAHL-pH_7_6        0.003905       0.002881       0.002881       0.002881  \n",
       "AtAHL-pH_7_9        0.003905       0.002881       0.002881       0.002881  \n",
       "AtAHL-pH_8_2        0.006315       0.002163       0.002163       0.002163  \n",
       "AtCNP-pH_7_0        0.001885       0.006150       0.006150       0.006150  \n",
       "AtCNP-pH_7_3        0.001885       0.006150       0.006150       0.006150  \n",
       "AtCNP-pH_7_6        0.001885       0.006150       0.006150       0.006150  \n",
       "AtCNP-pH_7_9        0.001885       0.006150       0.006150       0.006150  \n",
       "AtCNP-pH_8_2        0.003411       0.002527       0.002527       0.002527  \n",
       "AtSAL1-pH_7_0       0.004986       0.008726       0.008726       0.008726  \n",
       "AtSAL1-pH_7_3       0.005084       0.003665       0.003665       0.003665  \n",
       "AtSAL1-pH_7_6       0.005084       0.003665       0.003665       0.003665  \n",
       "AtSAL1-pH_7_9       0.005084       0.003665       0.003665       0.003665  \n",
       "AtSAL1-pH_8_2       0.005084       0.003665       0.003665       0.003665  \n",
       "ScHAL2-pH_7_0       0.000000       0.005975       0.005975       0.005975  \n",
       "ScHAL2-pH_7_3       0.000000       0.005975       0.005975       0.005975  \n",
       "ScHAL2-pH_7_6       0.005777       0.000000       0.000000       0.000000  \n",
       "ScHAL2-pH_7_9       0.005777       0.000000       0.000000       0.000000  \n",
       "ScHAL2-pH_8_2       0.005777       0.000000       0.000000       0.000000  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "At_v_Sc"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Make Network for SSN"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "\"\"\"\n",
    "Created by Matt Mortimer based off the sequence\n",
    "cleaner script from BioPython.\n",
    "26 Oct 21\n",
    "matthew.mortimer@anu.edu.au\n",
    "ORCID id: https://orcid.org/0000-0002-8135-9319\n",
    "Python 3\n",
    "\n",
    "Version 1.3.3 (211118)\n",
    "\"\"\"\n",
    "\n",
    "from Bio import SeqIO\n",
    "from datetime import datetime\n",
    "\n",
    "# Formats the current date, assigns to variable date\n",
    "DATE = datetime.now().strftime(\"%y%m%d\")\n",
    "\n",
    "\n",
    "def cleaner(file, data_source, por_n=0):\n",
    "\n",
    "    \"\"\"\n",
    "    Modification of the sequence cleaner script from BioPython.\n",
    "    Takes four arguments (one default):\n",
    "        The fasta file name as a string with the file extension,\n",
    "        The name of the data source as a sting\n",
    "        por_n value as the acceptable freq of 'X' residues (default 0)\n",
    "    Outputs a file with the duplicate sequeces and sequences with 'X' residues\n",
    "        (exceeding the set freq)\n",
    "    Summary is outputed to log.\n",
    "    \"\"\"\n",
    "\n",
    "    # Creates 2 dicts and set 2 counters to 0\n",
    "    sequences = {}  # Dict as a hash table\n",
    "    duplicate = {}\n",
    "    dup_count = 0\n",
    "    d_count = 0\n",
    "\n",
    "    # Uses the Biopython fasta parse\n",
    "    for seq_record in SeqIO.parse(file, \"fasta\"):\n",
    "        # Take the current sequence\n",
    "        sequence = str(seq_record.seq).upper()\n",
    "\n",
    "        # If the sequence passed in the test \"is it clean?\" and it isn't in\n",
    "        # the hash table, the sequence and its id are going to be in the hash\n",
    "        if (float(sequence.count(\"X\")) / float(len(sequence))) * 100 <= por_n:\n",
    "            if sequence not in sequences:\n",
    "                sequences[sequence] = seq_record.description\n",
    "            else:  # Places duplicate sequences into the 'duplicate' dict\n",
    "                duplicate[sequence] = seq_record.id\n",
    "                dup_count += 1\n",
    "        else:  # Places sequences with ambiguous residues into the same dict\n",
    "            duplicate[sequence] = seq_record.id\n",
    "            dup_count += 1\n",
    "            # A count is required to check if some sequences were duplicated\n",
    "            # multiple times\n",
    "\n",
    "    # Write the clean sequences\n",
    "    # Create a file with the cleaned sequences\n",
    "    with open(f\"output/{DATE}_{data_source}_cleaned.fasta\", \"w+\") as output_file:\n",
    "        # Just read the hash table and write on the file as a fasta format\n",
    "        # as key and variable\n",
    "        for sequence in sequences:\n",
    "            output_file.write(\">\" + sequences[sequence] + \"\\n\" + sequence + \"\\n\")\n",
    "\n",
    "    # Creates a file for dropped sequences using the same methods as above but\n",
    "    # taking the sequences from the 'duplicate' dict.\n",
    "    with open(\n",
    "        f\"output/{DATE}_{data_source}_cleaner_dropped_seq.fasta\", \"w+\"\n",
    "    ) as dropped_file:\n",
    "        for sequence in duplicate:\n",
    "            dropped_file.write(\">\" + duplicate[sequence] + \"\\n\" + sequence + \"\\n\")\n",
    "            d_count += 1\n",
    "\n",
    "    # Printing and logging the summary\n",
    "    print(\n",
    "        f\"{len(sequences)} were kept, output was written to {DATE}\\\n",
    "_{data_source}_cleaned.fasta. {d_count} sequences were dropped and \\\n",
    "written to output/{DATE}_{data_source}_cleaner_dropped_seq.fasta\"\n",
    "    )\n",
    "    print(f\"There were {dup_count} instances of duplicate or ambiguous sequences.\")\n",
    "\n",
    "    seq_file = f\"output/{DATE}_{data_source}_cleaned.fasta\"\n",
    "\n",
    "    print(seq_file)\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1533 were kept, output was written to 240626_all_cleaned.fasta. 4 sequences were dropped and written to output/240626_all_cleaner_dropped_seq.fasta\n",
      "There were 4 instances of duplicate or ambiguous sequences.\n",
      "output/240626_all_cleaned.fasta\n"
     ]
    }
   ],
   "source": [
    "cleaner(\"/home/mattm/projects/np_paper_structural/240626_SSN/240626_SSN_seqs.fasta\", \"all\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "\"\"\"\n",
    "Created by Matt Mortimer using code from Matt Spence\n",
    "26 Oct 21\n",
    "matthew.mortimer@anu.edu.au\n",
    "ORCID id: https://orcid.org/0000-0002-8135-9319\n",
    "Python 3\n",
    "\n",
    "Version 1.1.2 (211129)\n",
    "\"\"\"\n",
    "\n",
    "import pandas as pd\n",
    "import os as os\n",
    "from datetime import datetime, timedelta\n",
    "from time import time\n",
    "\n",
    "\n",
    "def blast(infile, E_value_threshold=\"10e-10\", cpus=\"2\", path=''):\n",
    "    \"\"\"\n",
    "    IMPORTANT: This function assumes blast has been added to the path environment\n",
    "    Runs BLAST locally using a fasta file to generate the database. A all v.\n",
    "    all search is then preformed.\n",
    "    Takes the following arguments:\n",
    "        blast_path (full dir path for blast)\n",
    "        infile as a fasta file as a string\n",
    "        E_value_threshold (default = \"10e-10\")\n",
    "        cpus; the number of CPU to use (default 2)\n",
    "    Outputs a network file which can be viewed in Cytoscape.\n",
    "    *Modified from code contributed by Matt Spence*\n",
    "    \"\"\"\n",
    "    # Start point for run time calc.\n",
    "    start = time()\n",
    "\n",
    "    # Format the date\n",
    "    date = datetime.now().strftime(\"%y%m%d\")\n",
    "\n",
    "    # Sets the name for the database files\n",
    "    db_file = f\"output/BLAST/{date}_dataset_db\"\n",
    "\n",
    "    # Make a BLAST database using the non-redundant/annotated sequence dataset\n",
    "    os.system(f\"{path}makeblastdb -in {infile} -dbtype prot -out {db_file}\")\n",
    "\n",
    "    # Perform an all.vs.all BLAST within the database\n",
    "    outblast = f\"{db_file}_blast_{E_value_threshold}\"\n",
    "\n",
    "    # Runs blastp, calling blast+\n",
    "    os.system(\n",
    "        f\"{path}blastp -db {db_file} -query {infile} \\\n",
    "-outfmt 6 -out {outblast} -evalue {E_value_threshold} -num_threads {cpus}\"\n",
    "    )\n",
    "\n",
    "    cols = [\n",
    "        \"Query\",\n",
    "        \"Target\",\n",
    "        \"% Identity\",\n",
    "        \"Length\",\n",
    "        \"Mistmatch\",\n",
    "        \"Gapopen\",\n",
    "        \"Qstart\",\n",
    "        \"Qend\",\n",
    "        \"Sstart\",\n",
    "        \"Send\",\n",
    "        \"E-value\",\n",
    "        \"Bit-score\",\n",
    "    ]\n",
    "\n",
    "    # Labels columns in a csv and writes to file\n",
    "    blast_out = pd.read_csv(f\"{outblast}\", delimiter=\"\\t\", names=cols)\n",
    "\n",
    "    blast_out.to_csv(\n",
    "        f\"output/BLAST/{date}_dataset_network_{E_value_threshold}\\\n",
    ".csv\"\n",
    "    )\n",
    "\n",
    "    # Run time calc\n",
    "    end = time()\n",
    "    elapsed_time = end - start\n",
    "    f_elapsed_time = str(timedelta(seconds=elapsed_time))\n",
    "    run_time = f_elapsed_time.split(\".\")[0]\n",
    "\n",
    "    # Printing and logging the summary\n",
    "    print(f\"Run time: {run_time} hh:mm:ss\")\n",
    "\n",
    "    print(\n",
    "        f\"Network file output/BLAST/{date}_dataset_network_\\\n",
    "{E_value_threshold}.csv generated\"\n",
    "    )\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "Building a new DB, current time: 06/26/2024 13:28:50\n",
      "New DB name:   /home/mattm/projects/np_paper_structural/output/BLAST/240626_dataset_db\n",
      "New DB title:  /home/mattm/projects/np_paper_structural/240626_SSN/output/240626_all_cleaned.fasta\n",
      "Sequence type: Protein\n",
      "Deleted existing Protein BLAST database named /home/mattm/projects/np_paper_structural/output/BLAST/240626_dataset_db\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 1536 sequences in 0.020977 seconds.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_686809/3522783759.py:67: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  blast_out = pd.read_csv(f\"{outblast}\", delimiter=\"\\t\", names=cols)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Run time: 0:00:53 hh:mm:ss\n",
      "Network file output/BLAST/240626_dataset_network_10e-10.csv generated\n"
     ]
    }
   ],
   "source": [
    "in_fasta = '/home/mattm/projects/np_paper_structural/240626_SSN/output/240626_all_cleaned.fasta' \n",
    "\n",
    "blast(in_fasta, '10e-10', cpus='40')\n",
    "\n",
    "DATE = '260626'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_686809/3168359867.py:7: DtypeWarning: Columns (1) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  netwrk = pd.read_csv(input_network)\n",
      "/tmp/ipykernel_686809/3168359867.py:15: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  nw_mod['Unid'] = nw_mod['Unid'].astype(\"string\")\n",
      "/tmp/ipykernel_686809/3168359867.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  index['Unid'] = index['Unid'].astype(\"string\")\n"
     ]
    }
   ],
   "source": [
    "# Set input and output files as variables \n",
    "input_network = f'/home/mattm/projects/np_paper_structural/240626_SSN/output/BLAST/240626_dataset_network_10e-10.csv'\n",
    "input_index = f'/home/mattm/projects/np_paper_structural/240626_SSN/output/BLAST/240626_wrk_index_updated.csv'\n",
    "output_network = f'/home/mattm/projects/np_paper_structural/240626_SSN/output/BLAST/{DATE}_SSN_10e-10.csv'\n",
    "\n",
    "# Read the input files\n",
    "netwrk = pd.read_csv(input_network)\n",
    "master_index = pd.read_csv(input_index)\n",
    "\n",
    "# Make a new dataframe for the input_files with a subset of columns, then rename\n",
    "# the first column of the input_network dataframe so that it can be easily merged \n",
    "# with the index file\n",
    "nw_mod = netwrk[['Query', 'Target', '% Identity', 'Length', 'E-value', 'Bit-score']]\n",
    "nw_mod.columns = ['Unid', 'Target', '% Identity', 'Length', 'E-value', 'Bit-score']\n",
    "nw_mod['Unid'] = nw_mod['Unid'].astype(\"string\")\n",
    "\n",
    "index = master_index[['Unid', 'Species', 'Brief_Classification']]\n",
    "index['Unid'] = index['Unid'].astype(\"string\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Unid          string[python]\n",
       "Target                object\n",
       "% Identity           float64\n",
       "Length                 int64\n",
       "E-value              float64\n",
       "Bit-score            float64\n",
       "dtype: object"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nw_mod.dtypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Unid                    string[python]\n",
       "Species                         object\n",
       "Brief_Classification            object\n",
       "dtype: object"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "index.dtypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "ssn_df = nw_mod.merge(index, on='Unid', how='left')\n",
    "\n",
    "ssn_df.to_csv(output_network, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unid</th>\n",
       "      <th>Target</th>\n",
       "      <th>% Identity</th>\n",
       "      <th>Length</th>\n",
       "      <th>E-value</th>\n",
       "      <th>Bit-score</th>\n",
       "      <th>Species</th>\n",
       "      <th>Brief_Classification</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>27611</td>\n",
       "      <td>27611</td>\n",
       "      <td>100.000</td>\n",
       "      <td>368</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>753.0</td>\n",
       "      <td>Micromonas_sp.</td>\n",
       "      <td>Mamiellophyceae</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>27611</td>\n",
       "      <td>18271</td>\n",
       "      <td>86.685</td>\n",
       "      <td>368</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>635.0</td>\n",
       "      <td>Micromonas_sp.</td>\n",
       "      <td>Mamiellophyceae</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>27611</td>\n",
       "      <td>95405</td>\n",
       "      <td>79.235</td>\n",
       "      <td>366</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>603.0</td>\n",
       "      <td>Micromonas_sp.</td>\n",
       "      <td>Mamiellophyceae</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>27611</td>\n",
       "      <td>28251</td>\n",
       "      <td>60.743</td>\n",
       "      <td>377</td>\n",
       "      <td>5.630000e-146</td>\n",
       "      <td>414.0</td>\n",
       "      <td>Micromonas_sp.</td>\n",
       "      <td>Mamiellophyceae</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>27611</td>\n",
       "      <td>84068</td>\n",
       "      <td>56.284</td>\n",
       "      <td>366</td>\n",
       "      <td>8.200000e-138</td>\n",
       "      <td>393.0</td>\n",
       "      <td>Micromonas_sp.</td>\n",
       "      <td>Mamiellophyceae</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>754095</th>\n",
       "      <td>AT4G39120.1</td>\n",
       "      <td>67548</td>\n",
       "      <td>28.261</td>\n",
       "      <td>138</td>\n",
       "      <td>1.260000e-10</td>\n",
       "      <td>56.2</td>\n",
       "      <td>Arabidopsis_thaliana</td>\n",
       "      <td>CoreEudicots/Rosids</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>754096</th>\n",
       "      <td>AT4G39120.1</td>\n",
       "      <td>85977</td>\n",
       "      <td>27.778</td>\n",
       "      <td>126</td>\n",
       "      <td>1.910000e-10</td>\n",
       "      <td>55.8</td>\n",
       "      <td>Arabidopsis_thaliana</td>\n",
       "      <td>CoreEudicots/Rosids</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>754097</th>\n",
       "      <td>AT4G39120.1</td>\n",
       "      <td>32778</td>\n",
       "      <td>29.496</td>\n",
       "      <td>139</td>\n",
       "      <td>3.780000e-10</td>\n",
       "      <td>54.7</td>\n",
       "      <td>Arabidopsis_thaliana</td>\n",
       "      <td>CoreEudicots/Rosids</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>754098</th>\n",
       "      <td>AT4G39120.1</td>\n",
       "      <td>95737</td>\n",
       "      <td>26.613</td>\n",
       "      <td>124</td>\n",
       "      <td>5.280000e-10</td>\n",
       "      <td>54.3</td>\n",
       "      <td>Arabidopsis_thaliana</td>\n",
       "      <td>CoreEudicots/Rosids</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>754099</th>\n",
       "      <td>AT4G39120.1</td>\n",
       "      <td>57187</td>\n",
       "      <td>23.565</td>\n",
       "      <td>331</td>\n",
       "      <td>9.080000e-10</td>\n",
       "      <td>53.9</td>\n",
       "      <td>Arabidopsis_thaliana</td>\n",
       "      <td>CoreEudicots/Rosids</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>754100 rows  8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               Unid Target  % Identity  Length        E-value  Bit-score  \\\n",
       "0             27611  27611     100.000     368   0.000000e+00      753.0   \n",
       "1             27611  18271      86.685     368   0.000000e+00      635.0   \n",
       "2             27611  95405      79.235     366   0.000000e+00      603.0   \n",
       "3             27611  28251      60.743     377  5.630000e-146      414.0   \n",
       "4             27611  84068      56.284     366  8.200000e-138      393.0   \n",
       "...             ...    ...         ...     ...            ...        ...   \n",
       "754095  AT4G39120.1  67548      28.261     138   1.260000e-10       56.2   \n",
       "754096  AT4G39120.1  85977      27.778     126   1.910000e-10       55.8   \n",
       "754097  AT4G39120.1  32778      29.496     139   3.780000e-10       54.7   \n",
       "754098  AT4G39120.1  95737      26.613     124   5.280000e-10       54.3   \n",
       "754099  AT4G39120.1  57187      23.565     331   9.080000e-10       53.9   \n",
       "\n",
       "                     Species Brief_Classification  \n",
       "0             Micromonas_sp.      Mamiellophyceae  \n",
       "1             Micromonas_sp.      Mamiellophyceae  \n",
       "2             Micromonas_sp.      Mamiellophyceae  \n",
       "3             Micromonas_sp.      Mamiellophyceae  \n",
       "4             Micromonas_sp.      Mamiellophyceae  \n",
       "...                      ...                  ...  \n",
       "754095  Arabidopsis_thaliana  CoreEudicots/Rosids  \n",
       "754096  Arabidopsis_thaliana  CoreEudicots/Rosids  \n",
       "754097  Arabidopsis_thaliana  CoreEudicots/Rosids  \n",
       "754098  Arabidopsis_thaliana  CoreEudicots/Rosids  \n",
       "754099  Arabidopsis_thaliana  CoreEudicots/Rosids  \n",
       "\n",
       "[754100 rows x 8 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ssn_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "old = pd.read_csv('/home/mattm/projects/np_paper_structural/240626_SSN/output/BLAST/master_index_w_outg_controls.csv')\n",
    "new = pd.read_csv('/home/mattm/projects/np_paper_structural/240626_SSN/output/BLAST/240626_wrk_index_updated.csv')\n",
    "\n",
    "old['Unid'] = old['Unid'].astype(\"string\")\n",
    "new['Unid'] = new['Unid'].astype(\"string\")\n",
    "\n",
    "old_mod = (old.merge(new, on='Unid', how='left', indicator=True)\n",
    "     .query('_merge == \"left_only\"')\n",
    "     .drop('_merge', axis=1))\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "old_mod.to_csv('/home/mattm/projects/np_paper_structural/240626_SSN/output/BLAST/check.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['PACid_x', 'Unid', 'SAL_variant', 'Datasource_x', 'Header_x',\n",
       "       'Sequence_x', 'Species_x', 'Onekp_index_id_x', 'Scaffold_x', 'Tax_1_x',\n",
       "       'Tax_2_x', 'Subject Seq-id_x', 'PACid_y', 'Species_y',\n",
       "       'Brief_Classification', 'Classification', 'Very_Brief_Classification',\n",
       "       'Full_Classification', 'Datasource_y', 'Header_y', 'Sequence_y',\n",
       "       'Onekp_index_id_y', 'Scaffold_y', 'Tax_1_y', 'Tax_2_y',\n",
       "       'Subject Seq-id_y'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "old_mod.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "old_mod = old_mod[['Unid', 'Header_x', 'Sequence_x', 'Species_x', 'Tax_1_x',\n",
    "       'Tax_2_x', 'Subject Seq-id_x']]\n",
    "\n",
    "old_mod.columns = ['Unid', 'Header', 'Sequence', 'Species', 'Tax_1',\n",
    "       'Tax_2', 'Seq-id']\n",
    "\n",
    "\n",
    "new = new[['Unid', 'Header', 'Sequence', 'Species', 'Brief_Classification', 'Tax_1', 'Tax_2',\n",
    "       'Subject Seq-id']]\n",
    "\n",
    "new.columns = ['Unid', 'Header', 'Sequence', 'Species', 'Brief_Classification', 'Tax_1', 'Tax_2',\n",
    "       'Seq-id']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "concat with the updated wrking index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unid</th>\n",
       "      <th>Header</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Species</th>\n",
       "      <th>Brief_Classification</th>\n",
       "      <th>Tax_1</th>\n",
       "      <th>Tax_2</th>\n",
       "      <th>Seq-id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>79283</td>\n",
       "      <td>&gt;79283 gnl|onekp|MGBF_scaffold_2010030 Acacia_...</td>\n",
       "      <td>WLTMETEKYAKELEVAVRVVHVACALCGRVQERLFSVGDDHVMSKD...</td>\n",
       "      <td>Acacia_pycnantha</td>\n",
       "      <td>CoreEudicots/Rosids</td>\n",
       "      <td>Core_Eudicots/Rosids</td>\n",
       "      <td>Fabaceae</td>\n",
       "      <td>gnl|onekp|MGBF_scaffold_2010030</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>37580</td>\n",
       "      <td>&gt;37580 gnl|onekp|MGBF_scaffold_2063174 Acacia_...</td>\n",
       "      <td>LNIRSSLPLPVKKAKYNKELEAAVDAVEKACRLCVDVKASLFSSDG...</td>\n",
       "      <td>Acacia_pycnantha</td>\n",
       "      <td>CoreEudicots/Rosids</td>\n",
       "      <td>Core_Eudicots/Rosids</td>\n",
       "      <td>Fabaceae</td>\n",
       "      <td>gnl|onekp|MGBF_scaffold_2063174</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>60720</td>\n",
       "      <td>&gt;60720 gnl|onekp|BMJR_scaffold_2061134 Adiantu...</td>\n",
       "      <td>SYEHELHLAVRAARLAARLCQTVQKKLLTKETQSKADKSPVTVADY...</td>\n",
       "      <td>Adiantum_raddianum</td>\n",
       "      <td>LeptosporangiateMonilophytes</td>\n",
       "      <td>Leptosporangiate_Monilophytes</td>\n",
       "      <td>Pteridaceae</td>\n",
       "      <td>gnl|onekp|BMJR_scaffold_2061134</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>99841</td>\n",
       "      <td>&gt;99841 gnl|onekp|BMJR_scaffold_2009359 Adiantu...</td>\n",
       "      <td>RLFQGNYDQELEVAVRAVQLACILTQRTQERILRNEEKAGSKTDRS...</td>\n",
       "      <td>Adiantum_raddianum</td>\n",
       "      <td>LeptosporangiateMonilophytes</td>\n",
       "      <td>Leptosporangiate_Monilophytes</td>\n",
       "      <td>Pteridaceae</td>\n",
       "      <td>gnl|onekp|BMJR_scaffold_2009359</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>15943</td>\n",
       "      <td>&gt;15943 gnl|onekp|GBCQ_scaffold_2001355 Aerva_p...</td>\n",
       "      <td>EKYSQELDVAVKAVQLACSLCQRVQQNLISTSSSLVQSKDDNSLVT...</td>\n",
       "      <td>Aerva_persica</td>\n",
       "      <td>CoreEudicots_Caryophyllales</td>\n",
       "      <td>Core_Eudicots</td>\n",
       "      <td>Amaranthaceae</td>\n",
       "      <td>gnl|onekp|GBCQ_scaffold_2001355</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30795</th>\n",
       "      <td>87894</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MGQRPGPVLSAVAVLGQVAKRKVAWLPRWKAVTRTETAGNSSGVYG...</td>\n",
       "      <td>Pan_troglodytes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mammalian</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30796</th>\n",
       "      <td>27803</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MKMADPWQECMDYAVTLARRAGEVVCEALKNEMNVMLKSSPVDLVT...</td>\n",
       "      <td>Papio_anubis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mammalian</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30797</th>\n",
       "      <td>23604</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MADPWQECMDYAVTLARQAGEVVCEALKNEMNVMLKSSPVDLVTAT...</td>\n",
       "      <td>Macaca_fascicularis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30798</th>\n",
       "      <td>49932</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MADPWQECMDYAVTLARQAGEVVREAIKNEMNVMVKSSPADLVTAT...</td>\n",
       "      <td>Felis_catus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mammalian</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30799</th>\n",
       "      <td>53531</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MADPWQECMDYAVTLAGQAGEVVREALKNEMNIMVKSSPADLVTAT...</td>\n",
       "      <td>Bos_taurus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mammalian</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>30819 rows  8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Unid                                             Header  \\\n",
       "0      79283  >79283 gnl|onekp|MGBF_scaffold_2010030 Acacia_...   \n",
       "1      37580  >37580 gnl|onekp|MGBF_scaffold_2063174 Acacia_...   \n",
       "2      60720  >60720 gnl|onekp|BMJR_scaffold_2061134 Adiantu...   \n",
       "3      99841  >99841 gnl|onekp|BMJR_scaffold_2009359 Adiantu...   \n",
       "4      15943  >15943 gnl|onekp|GBCQ_scaffold_2001355 Aerva_p...   \n",
       "...      ...                                                ...   \n",
       "30795  87894                                                NaN   \n",
       "30796  27803                                                NaN   \n",
       "30797  23604                                                NaN   \n",
       "30798  49932                                                NaN   \n",
       "30799  53531                                                NaN   \n",
       "\n",
       "                                                Sequence              Species  \\\n",
       "0      WLTMETEKYAKELEVAVRVVHVACALCGRVQERLFSVGDDHVMSKD...     Acacia_pycnantha   \n",
       "1      LNIRSSLPLPVKKAKYNKELEAAVDAVEKACRLCVDVKASLFSSDG...     Acacia_pycnantha   \n",
       "2      SYEHELHLAVRAARLAARLCQTVQKKLLTKETQSKADKSPVTVADY...   Adiantum_raddianum   \n",
       "3      RLFQGNYDQELEVAVRAVQLACILTQRTQERILRNEEKAGSKTDRS...   Adiantum_raddianum   \n",
       "4      EKYSQELDVAVKAVQLACSLCQRVQQNLISTSSSLVQSKDDNSLVT...        Aerva_persica   \n",
       "...                                                  ...                  ...   \n",
       "30795  MGQRPGPVLSAVAVLGQVAKRKVAWLPRWKAVTRTETAGNSSGVYG...      Pan_troglodytes   \n",
       "30796  MKMADPWQECMDYAVTLARRAGEVVCEALKNEMNVMLKSSPVDLVT...         Papio_anubis   \n",
       "30797  MADPWQECMDYAVTLARQAGEVVCEALKNEMNVMLKSSPVDLVTAT...  Macaca_fascicularis   \n",
       "30798  MADPWQECMDYAVTLARQAGEVVREAIKNEMNVMVKSSPADLVTAT...          Felis_catus   \n",
       "30799  MADPWQECMDYAVTLAGQAGEVVREALKNEMNIMVKSSPADLVTAT...           Bos_taurus   \n",
       "\n",
       "               Brief_Classification                          Tax_1  \\\n",
       "0               CoreEudicots/Rosids           Core_Eudicots/Rosids   \n",
       "1               CoreEudicots/Rosids           Core_Eudicots/Rosids   \n",
       "2      LeptosporangiateMonilophytes  Leptosporangiate_Monilophytes   \n",
       "3      LeptosporangiateMonilophytes  Leptosporangiate_Monilophytes   \n",
       "4       CoreEudicots_Caryophyllales                  Core_Eudicots   \n",
       "...                             ...                            ...   \n",
       "30795                           NaN                      Mammalian   \n",
       "30796                           NaN                      Mammalian   \n",
       "30797                           NaN                            NaN   \n",
       "30798                           NaN                      Mammalian   \n",
       "30799                           NaN                      Mammalian   \n",
       "\n",
       "               Tax_2                           Seq-id  \n",
       "0           Fabaceae  gnl|onekp|MGBF_scaffold_2010030  \n",
       "1           Fabaceae  gnl|onekp|MGBF_scaffold_2063174  \n",
       "2        Pteridaceae  gnl|onekp|BMJR_scaffold_2061134  \n",
       "3        Pteridaceae  gnl|onekp|BMJR_scaffold_2009359  \n",
       "4      Amaranthaceae  gnl|onekp|GBCQ_scaffold_2001355  \n",
       "...              ...                              ...  \n",
       "30795            NaN                              NaN  \n",
       "30796            NaN                              NaN  \n",
       "30797            NaN                              NaN  \n",
       "30798            NaN                              NaN  \n",
       "30799            NaN                              NaN  \n",
       "\n",
       "[30819 rows x 8 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "full = pd.concat([new, old_mod])\n",
    "full"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "full.to_csv('/home/mattm/projects/np_paper_structural/240626_SSN/output/BLAST/240628_master_index_updated.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is just taking the SAL cluster from the SSN and checking the taxonomy to make sure none are in red algae or glycophyta. There are some in there, but they fall in the outgroup or AHL/CNP in the phylogeny. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('/home/mattm/projects/np_paper_structural/240626_SSN/output/BLAST/SAL_seqs.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df[['name', 'Species']]\n",
    "df.columns = ['Unid', 'Species']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "index_df = pd.read_csv('/home/mattm/projects/np_paper_structural/240626_SSN/output/BLAST/240628_master_index_updated.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "index_df = index_df[['Unid', 'Species', 'Brief_Classification', 'Tax_1', 'Tax_2']]\n",
    "index_df.columns = ['']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_df = pd.merge(df, index_df, on='Unid', how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unid</th>\n",
       "      <th>Species_x</th>\n",
       "      <th>Species_y</th>\n",
       "      <th>Brief_Classification</th>\n",
       "      <th>Tax_1</th>\n",
       "      <th>Tax_2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>64292</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Ahnfeltiopsis_flabelliformis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Red_Algae</td>\n",
       "      <td>Phyllophoraceae</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>85050</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Triglochin_maritima</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Monocots</td>\n",
       "      <td>Juncaginaceae</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>47415</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Haemaria_discolor</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Monocots</td>\n",
       "      <td>Orchidaceae</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>40479</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Licania_michauxii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Core_Eudicots/Rosids</td>\n",
       "      <td>Chrysobalanaceae</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>98289</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Morus_nigra</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Core_Eudicots/Rosids</td>\n",
       "      <td>Moraceae</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1530</th>\n",
       "      <td>84068</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Monomastix_opisthostigma</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Green_Algae</td>\n",
       "      <td>Monomasticaceae</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1531</th>\n",
       "      <td>28251</td>\n",
       "      <td>Dolichomastix_tenuilepis</td>\n",
       "      <td>Dolichomastix_tenuilepis</td>\n",
       "      <td>Mamiellophyceae</td>\n",
       "      <td>Green_Algae</td>\n",
       "      <td>Dolichomastigaceae</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1532</th>\n",
       "      <td>95405</td>\n",
       "      <td>Ostreococcus_lucimarinus</td>\n",
       "      <td>Ostreococcus_lucimarinus</td>\n",
       "      <td>Mamiellophyceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1533</th>\n",
       "      <td>18271</td>\n",
       "      <td>Micromonas_pusilla</td>\n",
       "      <td>Micromonas_pusilla</td>\n",
       "      <td>Mamiellophyceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1534</th>\n",
       "      <td>27611</td>\n",
       "      <td>Micromonas_sp.</td>\n",
       "      <td>Micromonas_sp.</td>\n",
       "      <td>Mamiellophyceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1535 rows  6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Unid                 Species_x                     Species_y  \\\n",
       "0     64292                       NaN  Ahnfeltiopsis_flabelliformis   \n",
       "1     85050                       NaN           Triglochin_maritima   \n",
       "2     47415                       NaN             Haemaria_discolor   \n",
       "3     40479                       NaN             Licania_michauxii   \n",
       "4     98289                       NaN                   Morus_nigra   \n",
       "...     ...                       ...                           ...   \n",
       "1530  84068                       NaN      Monomastix_opisthostigma   \n",
       "1531  28251  Dolichomastix_tenuilepis      Dolichomastix_tenuilepis   \n",
       "1532  95405  Ostreococcus_lucimarinus      Ostreococcus_lucimarinus   \n",
       "1533  18271        Micromonas_pusilla            Micromonas_pusilla   \n",
       "1534  27611            Micromonas_sp.                Micromonas_sp.   \n",
       "\n",
       "     Brief_Classification                 Tax_1               Tax_2  \n",
       "0                     NaN             Red_Algae     Phyllophoraceae  \n",
       "1                     NaN              Monocots       Juncaginaceae  \n",
       "2                     NaN              Monocots         Orchidaceae  \n",
       "3                     NaN  Core_Eudicots/Rosids    Chrysobalanaceae  \n",
       "4                     NaN  Core_Eudicots/Rosids            Moraceae  \n",
       "...                   ...                   ...                 ...  \n",
       "1530                  NaN           Green_Algae     Monomasticaceae  \n",
       "1531      Mamiellophyceae           Green_Algae  Dolichomastigaceae  \n",
       "1532      Mamiellophyceae                   NaN                 NaN  \n",
       "1533      Mamiellophyceae                   NaN                 NaN  \n",
       "1534      Mamiellophyceae                   NaN                 NaN  \n",
       "\n",
       "[1535 rows x 6 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_df.to_csv('/home/mattm/projects/np_paper_structural/240626_SSN/output/BLAST/SAL_seqs_w_tax.csv', index='False')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.14"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
